# Analysis Notebook - Fisher exact test

This notebook looks at specific Genes and Gene Subcategories

```{r}
defaultW <- getOption("warn")  # suppress warnings for this cell
options(warn = -1) 
library(dplyr)
library(multtest)
library(R.utils)

options(warn = defaultW)
```

### 1  Read in all and significant alternative splicing and differential gene expression results

The summary data captured in the now saved **all_gene_as_gene_names.tsv**, **all_genes_dge_data** and significant results captured in **gene_as.tsv** and **gene_dge.tsv**

```{r}
results_dir  <- "../data/"
all_genes_as_data  <- read.table("../assets/all_gene_as_gene_names.tsv", header=TRUE, sep="\t",
                               skipNul=FALSE, stringsAsFactors = FALSE)
names(all_genes_as_data) <- c("GeneSymbol", "ensg")
all_genes_dge_data <- read.table("../assets/all_gene_dge_gene_names.tsv", header=TRUE, sep="\t",
                               skipNul=FALSE, stringsAsFactors = FALSE)
sig_gene_as  <- read.table(file="../data/gene_as.tsv", header=TRUE, sep="\t",
                               skipNul=FALSE, stringsAsFactors = FALSE)
sig_gene_dge  <- read.table(file="../data/gene_dge.tsv", header=TRUE, sep="\t",
                               skipNul=FALSE, stringsAsFactors = FALSE)
head(sig_gene_as,2)
head(sig_gene_dge,2)
head(all_genes_as_data,2)
head(all_genes_dge_data,2)
```

### 2  We then do a hypergeometric/Fisher test to look for overrepresentation



### 2.1 DGE vs DAS

Comparing differentially expressed genes with differentially alternatively spliced:

|  	|  DGE+| DGE-|
|-	|-	|-	|
| DAS+|  a|  b|
| DAS-|  c| d|

```{r}
sigASGenes  <- unique(sort(sig_gene_as$GeneSymbol))
sigDGEGenes <- unique(sort(sig_gene_dge$GeneSymbol))
allASGenes  <- unique(sort(all_genes_as_data$GeneSymbol))
allDGEGenes <- unique(sort(all_genes_dge_data$GeneSymbol))
```

```{r}
message("Number of sigASGenes ", length(sigASGenes))
notSigAS <- setdiff(allASGenes,sigASGenes)
message("Number of genes that are NOT sigAS ", length(notSigAS))
message("Number of sigDGEgenes ", length(sigDGEGenes))
notDGE <- setdiff(allDGEGenes,sigDGEGenes)
message("Number of genes that are NOT DGE ", length(notDGE))
a <- intersect(sigASGenes, sigDGEGenes)
b <- intersect(sigASGenes, notDGE)
c <- intersect(notSigAS, allDGEGenes)
d <- intersect(notSigAS, notDGE)
message("a: ", length(a), "; b: ",  length(b), "; c: ",  length(c), "; d: ",  length(d))
```

```{r}
m <- matrix(c(length(a),length(b),length(c),length(d)), nrow=2,byrow = TRUE)
fisher.test(m)
```

<!-- #region -->
### 2.2 IGH vs DGE

It was noted that there are a number of immune genes showing up disproportionally in DGE



|  	| DGE+| DGE-|
|-	|-	|-	|
| IGH+ |  a|  b|
| IGH- |  c| d|
<!-- #endregion -->

```{r}
IGH         <- all_genes_dge_data[grepl("IGH",all_genes_dge_data$GeneSymbol ),]
sigDGEGenes <- unique(sort(sig_gene_dge$GeneSymbol))
IGHGenes    <- unique(sort(IGH$GeneSymbol))
AllGenes    <- unique(sort(all_genes_dge_data$GeneSymbol))
length(IGHGenes)
length(AllGenes)
length(sigDGEGenes)
```

```{r}
message("Number of IGH genes ", length(IGHGenes))
notIGH <- setdiff(AllGenes,IGHGenes)
message("Number of genes that are NOT IGH ", length(notIGH))
message("Number of sigDGEgenes ", length(sigDGEGenes))
notDGE <- setdiff(AllGenes,sigDGEGenes)
message("Number of genes that are NOT DGE ", length(notDGE))
a <- intersect(IGHGenes, sigDGEGenes)
b <- intersect(IGHGenes, notDGE)
c <- intersect(notIGH,   sigDGEGenes)
d <- intersect(notIGH,   notDGE)
message("a: ", length(a), "; b: ",  length(b), "; c: ",  length(c), "; d: ",  length(d))
```

```{r}
m <- matrix(c(length(a),length(b),length(c),length(d)), nrow=2,byrow = TRUE)
fisher.test(m)
```

### Appendix - Metadata

For replicability and reproducibility purposes, we also print the following metadata:

1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**
2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)


### Appendix 1. Checksums with the sha256 algorithm

```{r}
notebookid   = "FisherExactTests"
notebookid

```

###Â Appendix 2. Libraries metadata

```{r}
dev_session_info   <- devtools::session_info()
utils_session_info <- utils::sessionInfo()

message("Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..")
saveRDS(dev_session_info, file = paste0("../metadata/", notebookid, "_devtools_session_info.rds"))
message("Done!\n")

message("Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..")
saveRDS(utils_session_info, file = paste0("../metadata/", notebookid ,"_utils_info.rds"))
message("Done!\n")

dev_session_info$platform
dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]
```

```{r}

```
