# Changes in Access to GTEx data

This repository documents the analysis performed for [The impact of sex on alternative splicing](https://www.biorxiv.org/content/10.1101/490904v2.full);

1. Get access to the [Genotype-Tissue Expression (GTEx)](https://www.gtexportal.org/home/) RNAseq data (an application to dbGAP for access to the dataset phs000424.v8.v2 is required)
2. Search through the gen3.anvil.io/explorer.
3. Search for the Transcriptome sequencing data
4. Obtain using the GA4GH FASP scripts to obtain the aligned bam files
5. Convert those files to fastq's to prep for re-alignment

and then continue with the previous step 2, now step 6

6. Align each RNAseq sample using hisat2 and create a matrix of counts for each of a variety of splicing types was generated by the rMATS. Specifically, rMATS was run as a [nextflow script](https://github.com/lifebit-ai/rmats-nf/). The script may be modified to run on any platform, the results from this study was performed on the [cloudOS/lifebit platform](https://lifebit.ai/). 
7. Run the Jupyter notebooks from this repository to perform the individual analyses.

