
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4179559.svg)](https://doi.org/10.5281/zenodo.4179559)

# The impact of sex on alternative splicing

This repository documents the analysis performed for [The impact of sex on alternative splicing](https://www.biorxiv.org/content/10.1101/490904v1.full);
note that a manuscript with a modified version of the analysis has been submitted. To reproduce the analysis, users will need to go through several steps.

1. Get access to the [Genotype-Tissue Expression (GTEx)](https://www.gtexportal.org/home/) RNAseq data (an application to dbGAP for access to the dataset phs000424.v8.v2 is required)
2. Align each RNAseq sample using hisat2 and create a matrix of counts for each of a variety of splicing types was generated by the rMATS. Specifically, rMATS was run as a [nextflow script](https://github.com/lifebit-ai/rmats-nf/). The script may be modified to run on any platform, the results from this study was performed on the [cloudOS/lifebit platform](https://lifebit.ai/). 
3. Run the Jupyter notebooks from this repository to perform the individual analyses.


This repository documents the interactive analysis for the results of running the rmats-nf pipeline.


## 1. Get access

The RNA-seq samples analyzed in this project are restricted access (dbGAP phs000424.v8.v2). See the
[database of Genotypes and Phenotypes (dbGaP)](https://www.ncbi.nlm.nih.gov/gap/) for details.

## 2. Processing the RNA-seq samples

See the manuscript for methods details. In brief, we ran the nextflow script at https://github.com/lifebit-ai/rmats-nf  to align the RNA-seq samples with hisat2 and to characterize splicing events with rMATS. Results from individual samples are summarized in 'matrix' files. To run the Jupyter scripts in the
next section, you will need to place these files in a results bucket (if you are using the cloudos system) or in some other defined location.

## 3. Running the notebooks

Each of the results described in the manuscript was generated by one or more Jupyter notebooks in this repository.
There are a number of R packages that need to be installed prior to running the notebooks. This process is described from the
cloudos environment in [this document](https://github.com/TheJacksonLaboratory/sbas/blob/master/SettingUpRenvironment.MD). If running the notebooks in another environment, simply run the 
setup scripts. 

### 3.1 Summarizing events

Most of the notebooks require that the raw rMATS files are first processed to generate summary files. This is done by the notebook
[countGenesAndEvents.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/countGenesAndEvents.ipynb). Additionally, two notebooks are used to
perform DGE and DAS analysis. These three notebooks should be run first.



1. [differentialGeneExpressionAnalysis.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/differentialGeneExpressionAnalysis.ipynb). Perform differential gene analysis with voom.
2. [differentialSplicingJunctionAnalysis.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/differentialSplicingJunctionAnalysis.ipynb). Regression analysis to characterize sex-biased alternative splicing events.
3. [countGenesAndEvents.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/countGenesAndEvents.ipynb). Set up the overall analysis. Write various files to the ``data`` subdirectory that will be used by other scripts.

The remaining notebooks can be run in any order. Most of the notebooks generate a Figure or a Table or a result that is described in the manuscript.


* [expressionHeatplot.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/expressionHeatplot.ipynb). Generate a heatplot representing expression across tissues.
* [totalDGEByTissue.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/totalDGEByTissue.ipynb). Generate a plot representing counts of expression events across tissues.
* [alternativeSplicingHeatplot.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/alternativeSplicingHeatplot.ipynb). Generate a heatplot representing alternative splicing across tissues.
* [totalAlternativeSplicingByTissue.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/totalAlternativeSplicingByTissue.ipynb). Generate a plot representing counts of alternative splicing across tissues.
* [XchromosomalEscape.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/XchromosomalEscape.ipynb). Investigate the overlap of alternative splicing and genes on the X chromosome that escape inactivation.
* [splicingIndex.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/splicingIndex.ipynb). Calculate the splicing index for each chromosome.
* [spliceTypeByChromosome.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/spliceTypeByChromosome.ipynb). Calculate the distribution of the 5 types of alternative splicing event analyzed in this manuscript for each chromosome.
* [altSplicing_events_per_gene.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/altSplicing_events_per_gene.ipynb). Create a plot showing genes that display alternative splicing in many tissues.
* [tissue_piechart.ipynb](https://github.com/TheJacksonLaboratory/sbas/blob/master/jupyter/tissue_piechart.ipynb). Create a piechart showing distribution of genes according to number of tissues showing differential alternative splicing.



## 4. Reproducibility note

To facilitate reproducing the results from the secondary analysis that generates all the plots and tables of the publication, we have created a helper bash script that can be run to perform the following:

1. Prepare the environment by installing dependencies
2. Retrieve the data that we have made available via Zenodo [/DOI/10.5281/zenodo.4179559](https://doi.org/10.5281/zenodo.4179559)
3. Programmatically executing all Jupyter Notebooks leveraging the [papermill](https://papermill.readthedocs.io/en/latest/) library.

You can find the file at [`./reproduce.sh`](reproduce.sh). The only prerequisite is a machine with `conda` installed.

> IMPORTANT NOTE:
Before executing the bash script, make sure your terminal is initialises for using `conda`.
You can do so by running the following command, depending on you default shell:

i) for `zsh`
```
## Initialise the terminal for use of conda
conda init zsh && exec -l zsh
```
ii) for `bash`

```
## Initialise the terminal for use of conda
conda init bash && exec -l bash
```

<details>
<summary> The contents of file are displayed below:

</summary>

```bash
# Clone sbas repo
git clone https://github.com/TheJacksonLaboratory/sbas

# cd into repo
cd sbas

# Install dependencies in your linux machine with conda available
## Install mamba, a faster alternative/implementation compared conda 
conda install mamba -y

## Create a new isolated environment for the analysis
mamba env create --name sbas -f environment.yml 

## Activate the new environment
conda activate sbas

# Retrieve prerequisite input files for Jupyter Notebooks from ZENODO
wget https://zenodo.org/record/4179559/files/as.tar.gz
wget https://zenodo.org/record/4179559/files/dge.tar.gz
wget https://zenodo.org/record/4179559/files/fromGTF.tar.gz
wget https://zenodo.org/record/4179559/files/gtex.tar.gz 
wget https://zenodo.org/record/4179559/files/rmats_final.tar.gz
wget https://zenodo.org/record/4179559/files/srr.tar.gz

# Decompress archives into the empty data folder and delete the archives after 
tar xzvf as.tar.gz -C data && rm as.tar.gz
tar xzvf dge.tar.gz -C data && rm dge.tar.gz
tar xzvf fromGTF.tar.gz -C data && rm fromGTF.tar.gz
tar xzvf gtex.tar.gz  -C data && rm  gtex.tar.gz
tar xzvf rmats_final.tar.gz -C data && rm rmats_final.tar.gz
tar xzvf srr.tar.gz -C data && rm srr.tar.gz

# cd into jupyter
cd jupyter

# Execute programmatically the notebooks with Papermill
papermill countGenesAndEvents.ipynb countGenesAndEvents.ipynb
papermill expressionHeatplot.ipynb expressionHeatplot.ipynb 
papermill totalDGEByTissue.ipynb totalDGEByTissue.ipynb 
papermill alternativeSplicingHeatplot.ipynb alternativeSplicingHeatplot.ipynb 
papermill totalAlternativeSplicingByTissue.ipynb totalAlternativeSplicingByTissue.ipynb
papermill XchromosomalEscape.ipynb XchromosomalEscape.ipynb
papermill splicingIndex.ipynb splicingIndex.ipynb
papermill spliceTypeByChromosome.ipynb spliceTypeByChromosome.ipynb
papermill altSplicing_events_per_gene.ipynb altSplicing_events_per_gene.ipynb
papermill tissue_piechart.ipynb tissue_piechart.ipynb    
```
    
    
</details>
