{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# AllTissueAllSplicingJunctionAnalysis as a Notebook \n",
    "\n",
    "rMATS 3.2.5 was run on controlled access RNASeq files retrieved experiments stored in the Sequence Read Archive with controlled access managed by dbGaP.   The data were generated under the Gene Tissue Expression.\n",
    "\n",
    "## rMATS RNASeq-MATS.py produces 10 different output types which get assembled into as type junction ID by sample ID matrices\n",
    "\n",
    "### Alternative Splice Site Types are: (se, a3ss, a5ss, mxe, ri)\n",
    "\n",
    " This is input as ARGV1 into variable 'astype'\n",
    "\n",
    "  * Skipped Exon events (se),\n",
    "  * Alternative 3' splice site (a3ss),\n",
    "  * Alternative 5' splice site (a5ss),\n",
    "  * Mutually exclusive exon (mxe),\n",
    "  * and retention intron (ri)\n",
    "\n",
    "### There are two different kinds of junction counts\n",
    "\n",
    "  * jc = junction counts - reads that cross the junction\n",
    "  * jcec = junction counts plus reads on the target (such as included exon\n",
    "\n",
    "### And the count type -- there are 5 types\n",
    "\n",
    "  * inclusion levels (percent spliced in)\n",
    "  * included junction counts (ijc)\n",
    "  * skipped junction counts (sjc)\n",
    "  * inclusion length (inclen)\n",
    "  * skipped length (skiplen)\n",
    "\n",
    "### function: fit_iso_tissue \n",
    "\n",
    "fit_iso_tissue expects the following input:\n",
    "\n",
    "  * the tissue of interest (SMSTD) \n",
    "  * an ordered_merged_rmats -- which will be ordered to fit the count matrix\n",
    "  * count matrix (inc or ijc & sjc merged)\n",
    "  * splice type (a3ss, a5ss, mxe, ri or se)\n",
    "  * junction_count type (jc or jcec)\n",
    "  * count type (inc or the merged ijc,sjc)\n",
    "  \n",
    "### reordering to match annotations between count matrix and annotation matrix\n",
    "\n",
    "Common problem is to match specifically the rows of an annotation matrix with the columns of a count matrix\n",
    "`match` is the function that gives the re-ordering index required to accomplish this\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **NOTE**:\n",
    "\n",
    "We assume that you have cloned the analysis repository and have `cd` into the parent directory. Before starting with the analysis make sure you have first completed the dependencies set up by following the instructions described in the **`dependencies/README.md`** document. All paths defined in this Notebook are relative to the parent directory (repository). Please close this Notebook and start again by following the above guidelines if you have not completed the aforementioned steps.\n",
    "\n",
    "## rMATS-final-merged\n",
    "the rmats-nf NextFlow was executed and the results released here:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: R.oo\n",
      "Loading required package: R.methodsS3\n",
      "R.methodsS3 v1.8.0 (2020-02-14 07:10:20 UTC) successfully loaded. See ?R.methodsS3 for help.\n",
      "R.oo v1.23.0 successfully loaded. See ?R.oo for help.\n",
      "\n",
      "Attaching package: ‘R.oo’\n",
      "\n",
      "The following object is masked from ‘package:R.methodsS3’:\n",
      "\n",
      "    throw\n",
      "\n",
      "The following objects are masked from ‘package:methods’:\n",
      "\n",
      "    getClasses, getMethods\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    attach, detach, load, save\n",
      "\n",
      "R.utils v2.9.2 successfully loaded. See ?R.utils for help.\n",
      "\n",
      "Attaching package: ‘R.utils’\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    timestamp\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    cat, commandArgs, getOption, inherits, isOpen, nullfile, parse,\n",
      "    warnings\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(limma)\n",
    "library(piggyback)\n",
    "library(multtest)\n",
    "library(Biobase)\n",
    "library(edgeR)\n",
    "library(tibble)\n",
    "#install.packages('R.utils')\n",
    "library(R.utils)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis \n",
    "\n",
    "This analysis uses edgeR.  Normalization takes the form of correction factors computed internally by edgeR functions, but it is also possible for a user to supply them. The correction factors may take the form of scaling factors for the library sizes, such as computed by calcNormFactors, which are then used to compute the effective library sizes. \n",
    "\n",
    "In this analysis, we are using the raw counts as provided by rMATS 3.2.5.  The raw counts we are using in the model are `ijc` and `sjc`, the sample specific raw read counts as they align to the junctions of the `included exon (ijc)` and the junctions of the `excluded or skipped exon (sjc)` respectively.\n",
    "\n",
    "\n",
    "Be sure to set your GITHUB_TOKEN, prior to downloading files\n",
    "\n",
    "One suggestion is change it to your token and then run it then immediately change it back to this:\n",
    "\n",
    "Sys.setenv(GITHUB_TOKEN = \"your-very-own-github-token\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Did you remember?\n",
    "Did you remember to delete your private github token?  Now is a good time to do so, before you save your work and checkit in inadvertantly...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n",
      "Warning message in get_token():\n",
      "“Using default public GITHUB_TOKEN.\n",
      "                     Please set your own token”All files up-to-date already\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!(\"rmats_final.se.jc.ijc.txt.gz\" %in% list.files(\"../significant_events/\"))) {    \n",
    "    # SE\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    # RI\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    # MXE\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    # A3SS\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "     # A5SS\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "} "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in the count data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading a3ss.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a3ss.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a5ss.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a5ss.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading mxe.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading mxe.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading ri.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading ri.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading se.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading se.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"loading a3ss.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "a3ss.jc.ijc <- data.table::fread(\"../rmats_counts/rmats_final.a3ss.jc.ijc.txt.gz\")\n",
    "message(\"done!\\n\")\n",
    "message(\"loading a3ss.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "a3ss.jc.sjc <- data.table::fread(\"../rmats_counts/rmats_final.a3ss.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading a5ss.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "a5ss.jc.ijc <- data.table::fread(\"../rmats_counts/rmats_final.a5ss.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading a5ss.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "a5ss.jc.sjc <- data.table::fread(\"../rmats_counts/rmats_final.a5ss.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading mxe.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "mxe.jc.ijc  <- data.table::fread(\"../rmats_counts/rmats_final.mxe.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading mxe.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "mxe.jc.sjc  <- data.table::fread(\"../rmats_counts/rmats_final.mxe.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading ri.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "ri.jc.ijc   <- data.table::fread(\"../rmats_counts/rmats_final.ri.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading ri.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "ri.jc.sjc   <- data.table::fread(\"../rmats_counts/rmats_final.ri.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading se.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "se.jc.ijc   <- data.table::fread(\"../rmats_counts/rmats_final.se.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading se.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "se.jc.sjc   <- data.table::fread(\"../rmats_counts/rmats_final.se.jc.sjc.txt.gz\")\n",
    "message(\"done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in metadata \n",
    "\n",
    "- `Sequence Read Archive (SRA)` Accession Data, `SRR` numbers\n",
    "- `Genome Tissue Expression (GTEx)` Clinical Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\n",
      "\n",
      "done!\n",
      "\n",
      "Loading obj GTEx v8 rds object with readRDS from ../data/gtex.rds ..\n",
      "\n",
      "Done!\n",
      "\n",
      "Generating sha256sum for gtex.rds ..\n",
      "\n",
      "c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0  ../data/gtex.rds\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!(\"SraRunTable.txt.gz\" %in% list.files(\"../data/\"))) {\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"TheJacksonLaboratory/sbas\", \n",
    "        file = \"SraRunTable.txt.gz\",\n",
    "        tag  = \"GTExV8.v1.0\", \n",
    "        dest = \"../data/\")\n",
    "    \n",
    "    message(\"Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\\n\")   \n",
    "    metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    message(\"done!\")\n",
    "} else {\n",
    "    message(\"Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\\n\")   \n",
    "    metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    message(\"done!\\n\")\n",
    "}\n",
    "\n",
    "if (!(\"gtex.rds\" %in% list.files(\"../data/\"))) {\n",
    "    message(\"Downloading and loading obj with GTEx v8 with 'yarn::downloadGTExV8()'\\n\")\n",
    "    obj <- yarn::downloadGTExV8(type='genes',file='../data/gtex.rds')\n",
    "    message(\"Done!\\n\")\n",
    "\n",
    "} else {\n",
    "# Load with readRDS() if gtex.rds available in data/\n",
    "    message(\"Loading obj GTEx v8 rds object with readRDS from ../data/gtex.rds ..\\n\")   \n",
    "    obj <- readRDS(file = \"../data/gtex.rds\")\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Generating sha256sum for gtex.rds ..\\n\")    \n",
    "    message(system(\"sha256sum ../data/gtex.rds\", intern = TRUE))\n",
    "    message(\"Done!\\n\")\n",
    "} \n",
    "if (! (file.exists(\"../data/fromGTF.tar.gz\"))) {\n",
    "    system(\"mkdir -p ../data\", intern = TRUE)\n",
    "    message(\"Fetching fromGTF.tar.gz from GitHub ..\")\n",
    "    # Download archive from GitHub release with tag \"dge\"\n",
    "    piggyback::pb_download(file = \"fromGTF.tar.gz\",\n",
    "                           dest = \"../data\",\n",
    "                           repo = \"adeslatt/sbas_gtf\",\n",
    "                           tag  = \"rMATS.3.2.5.gencode.v30\",\n",
    "                           show_progress = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Decompressing fromGTF.tar.gz into ../data\")\n",
    "    system(\"mkdir -p ../data && tar xvfz ../data/fromGTF.tar.gz -C ../data\", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Decompressing fromGTF.*.txt.gz into ../data\")\n",
    "    system(\"gunzip  ../data/fromGTF*.txt.gz \", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Reading fromGTF.SE.txt into fromGTF.SE\")    \n",
    "    fromGTF.SE <- read.table(\"../data/fromGTF.SE.txt\", header=TRUE)\n",
    "    message(\"Done!\\n\") \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quality control, preprocessing of data \n",
    "\n",
    "We observed above that our phenotype data have 2 more observations than our expression data. Let's inspect what these samples are:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-YF7O-2326-101833-SM-5CVN9</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>Skin   </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>GTEX-YEC3-1426-101806-SM-5PNXX</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>Stomach</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-YF7O-2326-101833-SM-5CVN9 & NA & NA & NA & NA & NA & Skin    & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tGTEX-YEC3-1426-101806-SM-5PNXX & NA & NA & NA & NA & NA & Stomach & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-YF7O-2326-101833-SM-5CVN9 | NA | NA | NA | NA | NA | Skin    | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| GTEX-YEC3-1426-101806-SM-5PNXX | NA | NA | NA | NA | NA | Stomach | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "                               SAMPID SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS   \n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA     NA       NA       NA       NA    Skin   \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA     NA       NA       NA       NA    Stomach\n",
       "                               SMTSD SMUBRID SMTSISCH SMTSPAX ⋯ SME1PCTS\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA    NA      NA       NA      ⋯ NA      \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA    NA      NA       NA      ⋯ NA      \n",
       "                               SMRRNART SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA       NA       NA       NA       NA      \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA       NA       NA       NA       NA      \n",
       "                               SUBJID SEX AGE DTHHRDY\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA     NA  NA  NA     \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA     NA  NA  NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17384</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17384\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17384\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17384    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55878</li><li>17382</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55878\n",
       "\\item 17382\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55878\n",
       "2. 17382\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55878 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17382"
      ],
      "text/latex": [
       "17382"
      ],
      "text/markdown": [
       "17382"
      ],
      "text/plain": [
       "[1] 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The non-overlapping IDs between pheno and count data are:\n",
      "\n",
      "GTEX-YF7O-2326-101833-SM-5CVN9\n",
      "GTEX-YEC3-1426-101806-SM-5PNXX\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "logical_match_names\n",
       "FALSE  TRUE \n",
       "    2 17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-ZZPU-2626-SM-5E45Y</th><td>GTEX-ZZPU-2626-SM-5E45Y</td><td> 1</td><td>B1</td><td>2 pieces, includes internal fat (up to 20%) and tendon</td><td> 8.50000</td><td>Muscle        </td><td>Muscle - Skeletal     </td><td>0011907</td><td>  248</td><td> 428</td><td>⋯</td><td>50.6647</td><td>0.010466900</td><td>0.9956840</td><td>NA</td><td> 0</td><td>49.6941</td><td>GTEX-ZZPU</td><td>2</td><td>50-59</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>GTEX-ZZPU-2726-SM-5NQ8O</th><td>GTEX-ZZPU-2726-SM-5NQ8O</td><td> 1</td><td>B1</td><td>2 pieces                                              </td><td> 6.80000</td><td>Adipose Tissue</td><td>Adipose - Subcutaneous</td><td>0002190</td><td>  249</td><td> 427</td><td>⋯</td><td>50.3168</td><td>0.005856410</td><td>0.9958590</td><td>NA</td><td> 0</td><td>50.0760</td><td>GTEX-ZZPU</td><td>2</td><td>50-59</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-ZZPU-2626-SM-5E45Y & GTEX-ZZPU-2626-SM-5E45Y &  1 & B1 & 2 pieces, includes internal fat (up to 20\\%) and tendon &  8.50000 & Muscle         & Muscle - Skeletal      & 0011907 &   248 &  428 & ⋯ & 50.6647 & 0.010466900 & 0.9956840 & NA &  0 & 49.6941 & GTEX-ZZPU & 2 & 50-59 &  0\\\\\n",
       "\tGTEX-ZZPU-2726-SM-5NQ8O & GTEX-ZZPU-2726-SM-5NQ8O &  1 & B1 & 2 pieces                                               &  6.80000 & Adipose Tissue & Adipose - Subcutaneous & 0002190 &   249 &  427 & ⋯ & 50.3168 & 0.005856410 & 0.9958590 & NA &  0 & 50.0760 & GTEX-ZZPU & 2 & 50-59 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-ZZPU-2626-SM-5E45Y | GTEX-ZZPU-2626-SM-5E45Y |  1 | B1 | 2 pieces, includes internal fat (up to 20%) and tendon |  8.50000 | Muscle         | Muscle - Skeletal      | 0011907 |   248 |  428 | ⋯ | 50.6647 | 0.010466900 | 0.9956840 | NA |  0 | 49.6941 | GTEX-ZZPU | 2 | 50-59 |  0 |\n",
       "| GTEX-ZZPU-2726-SM-5NQ8O | GTEX-ZZPU-2726-SM-5NQ8O |  1 | B1 | 2 pieces                                               |  6.80000 | Adipose Tissue | Adipose - Subcutaneous | 0002190 |   249 |  427 | ⋯ | 50.3168 | 0.005856410 | 0.9958590 | NA |  0 | 50.0760 | GTEX-ZZPU | 2 | 50-59 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "                        SAMPID                  SMATSSCR SMCENTER\n",
       "GTEX-ZZPU-2626-SM-5E45Y GTEX-ZZPU-2626-SM-5E45Y  1       B1      \n",
       "GTEX-ZZPU-2726-SM-5NQ8O GTEX-ZZPU-2726-SM-5NQ8O  1       B1      \n",
       "                        SMPTHNTS                                              \n",
       "GTEX-ZZPU-2626-SM-5E45Y 2 pieces, includes internal fat (up to 20%) and tendon\n",
       "GTEX-ZZPU-2726-SM-5NQ8O 2 pieces                                              \n",
       "                        SMRIN    SMTS           SMTSD                  SMUBRID\n",
       "GTEX-ZZPU-2626-SM-5E45Y  8.50000 Muscle         Muscle - Skeletal      0011907\n",
       "GTEX-ZZPU-2726-SM-5NQ8O  6.80000 Adipose Tissue Adipose - Subcutaneous 0002190\n",
       "                        SMTSISCH SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT \n",
       "GTEX-ZZPU-2626-SM-5E45Y   248     428    ⋯ 50.6647  0.010466900 0.9956840\n",
       "GTEX-ZZPU-2726-SM-5NQ8O   249     427    ⋯ 50.3168  0.005856410 0.9958590\n",
       "                        SMNUM5CD SMDPMPRT SME2PCTS SUBJID    SEX AGE   DTHHRDY\n",
       "GTEX-ZZPU-2626-SM-5E45Y NA        0       49.6941  GTEX-ZZPU 2   50-59  0     \n",
       "GTEX-ZZPU-2726-SM-5NQ8O NA        0       50.0760  GTEX-ZZPU 2   50-59  0     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55878</li><li>17382</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55878\n",
       "\\item 17382\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55878\n",
       "2. 17382\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55878 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail (pData(obj),2)\n",
    "dim(pData(obj))\n",
    "dim(exprs(obj))\n",
    "dim(obj)\n",
    "sample_names=as.vector(as.character(colnames(exprs(obj))))\n",
    "pheno_sample_names=as.vector(as.character(rownames(pData(obj))))\n",
    "length(pheno_sample_names)\n",
    "length(sample_names)\n",
    "\n",
    "if (length(pheno_sample_names) > length(sample_names)) {\n",
    "    superset <- pheno_sample_names\n",
    "    subset   <- sample_names    \n",
    "} \n",
    "\n",
    "if (length(pheno_sample_names) < length(sample_names)) {\n",
    "    superset <- sample_names\n",
    "    subset   <- pheno_sample_names   \n",
    "} \n",
    "\n",
    "non_overlaps <- setdiff( superset, subset)\n",
    "\n",
    "message(\"The non-overlapping IDs between pheno and count data are:\\n\\n\", \n",
    "        paste(non_overlaps, collapse = \"\\n\") )\n",
    "logical_match_names=superset %in% subset\n",
    "length(logical_match_names)\n",
    "table(logical_match_names)\n",
    "pData(obj) <- (pData(obj)[logical_match_names==TRUE,])\n",
    "\n",
    "tail (pData(obj),2)\n",
    "dim(pData(obj))\n",
    "dim(exprs(obj))\n",
    "dim(obj)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in junction Annotation Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading fromGTF.A3SS.txt into fromGTF.A3SS\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.A5SS.txt into fromGTF.A5SS\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.MXE.txt into fromGTF.MXE\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.RI.txt into fromGTF.RI\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.SE.txt into fromGTF.SE\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"Reading fromGTF.A3SS.txt into fromGTF.A3SS\")    \n",
    "fromGTF.A3SS <- read.table(\"../data/fromGTF.A3SS.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.A5SS.txt into fromGTF.A5SS\")    \n",
    "fromGTF.A5SS <- read.table(\"../data/fromGTF.A5SS.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.MXE.txt into fromGTF.MXE\")    \n",
    "fromGTF.MXE <- read.table(\"../data/fromGTF.MXE.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.RI.txt into fromGTF.RI\")    \n",
    "fromGTF.RI <- read.table(\"../data/fromGTF.RI.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.SE.txt into fromGTF.SE\")    \n",
    "fromGTF.SE <- read.table(\"../data/fromGTF.SE.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preparing our Data\n",
    "### Aligning Annotations with Run Data and with Count Data\n",
    "\n",
    "A description of the GTEx V8 release may be found here https://www.gtexportal.org/home/datasets.   To facilitate the analysis, the Quackenbush lab's Joe Paulson's Yarn Package, https://github.com/jnpaulson/yarn, forked and upgraded to GTEx V8 here https://github.com/TheJacksonLaboratory/yarn/tree/annes-changes.  \n",
    "\n",
    "The sequences we used for our analysis with rMATS are specified by their SRR number and were obtained from dbGaP, with our application to the controlled access human data.  Beginning with these raw unfiltered fastq files, we ran them through the NextFlow workflow rmats-nf https://github.com/lifebit-ai/rmats-nf. The accession data was downloaded with a repository key from the google cloud bucket (g3).\n",
    "\n",
    "- `obj`      : GTEx V8 expressionSet \n",
    "- `metadata` : SraRunTable.txt holds the SRR accession numbers used to extract samples from dbGaP SRA\n",
    "- `ijc`      : inclusion junction counts as reported by `rMATS 3.2.5`\n",
    "- `sjc`      : skipped junction counts as reported by `rMATS 3.2.5`  \n",
    "\n",
    "For each of the splicing events:\n",
    "\n",
    "- `a3ss`     : alternative 3` splice site\n",
    "- `a5ss`     : alternative 5` splice site\n",
    "- `mxe`      : mutually exclusive exon\n",
    "- `ri`       : retention intron\n",
    "- `se`       : skipped exon\n",
    "\n",
    "In `rMATS 3.2.5`, each junction is encoded in a unique `ID` that is immutable regardless of comparison.  \n",
    "This `ID` is different for each of the differing splicing events."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Move Junction ID to Row definition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rownames(a3ss.jc.ijc) <- a3ss.jc.ijc$ID\n",
    "rownames(a3ss.jc.sjc) <- a3ss.jc.sjc$ID\n",
    "a3ss.jc.ijc <- a3ss.jc.ijc[,-1]\n",
    "a3ss.jc.sjc <- a3ss.jc.sjc[,-1]\n",
    "\n",
    "rownames(a5ss.jc.ijc) <- a3ss.jc.ijc$ID\n",
    "rownames(a5ss.jc.sjc) <- a3ss.jc.sjc$ID\n",
    "a5ss.jc.ijc <- a5ss.jc.ijc[,-1]\n",
    "a5ss.jc.sjc <- a5ss.jc.sjc[,-1]\n",
    "\n",
    "rownames(mxe.jc.ijc)  <- mxe.jc.ijc$ID\n",
    "rownames(mxe.jc.sjc)  <- mxe.jc.sjc$ID\n",
    "mxe.jc.ijc <- mxe.jc.ijc[,-1]\n",
    "mxe.jc.sjc <- mxe.jc.sjc[,-1]\n",
    "\n",
    "rownames(ri.jc.ijc)   <- ri.jc.ijc$ID\n",
    "rownames(ri.jc.sjc)   <- ri.jc.sjc$ID\n",
    "ri.jc.ijc <- ri.jc.ijc[,-1]\n",
    "ri.jc.sjc <- ri.jc.sjc[,-1]\n",
    "\n",
    "rownames(se.jc.ijc)   <- se.jc.ijc$ID\n",
    "rownames(se.jc.sjc)   <- se.jc.sjc$ID\n",
    "se.jc.ijc <- se.jc.ijc[,-1]\n",
    "se.jc.sjc <- se.jc.sjc[,-1]\n",
    "\n",
    "dim(a3ss.jc.ijc)\n",
    "dim(a3ss.jc.sjc)\n",
    "dim(a5ss.jc.ijc)\n",
    "dim(a5ss.jc.sjc)\n",
    "dim(mxe.jc.ijc)\n",
    "dim(mxe.jc.sjc)\n",
    "dim(ri.jc.ijc)\n",
    "dim(ri.jc.sjc)\n",
    "dim(se.jc.ijc)\n",
    "dim(se.jc.sjc)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Synchronize Clinical Annotations and Accession Run, reducing to only tissues of interest\n",
    "\n",
    "Join the yarn metadata with the metadata we have (there are redundant samples that have been sequenced multiple times). We want to be sure that we can obtain all required Clinical Annotation information from the YARN GTEx Annotation information, as the SRA metadata is not as reliable.    This will be a one-to-many mapping, as there are multiple sequence runs per 69 samples -- expanding our data set.  There are only a handful of annotations we require: SEX, AGE, DTHHRDY (which is cause of death), SMCENTER.\n",
    "\n",
    "Note that the numbers in specific age groups expand because of the one to many relationship from sample to sequencing runs. \n",
    "\n",
    "#### Using results from analysis of number of samples stored in `tissues.tsv` we keep only those that are members of this reduced tissue list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       " 0  1 \n",
       "11 39 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>39</li><li>5</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 39\n",
       "\\item 5\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 39\n",
       "2. 5\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 39  5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "16923   459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>459</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 459\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   459\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>459</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 459\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 459\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 459  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 68</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th><th scope=col>tissue</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-1117F-2826-SM-5GZXL</th><td>GTEX-1117F-2826-SM-5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>Breast - Mammary Tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td><td>breast_mammary_tissue</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 68\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY & tissue\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-1117F-2826-SM-5GZXL & GTEX-1117F-2826-SM-5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & Breast - Mammary Tissue & 0008367 &  1340 & 1008 & ⋯ & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4 & breast\\_mammary\\_tissue\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 68\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; | tissue &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-1117F-2826-SM-5GZXL | GTEX-1117F-2826-SM-5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | Breast - Mammary Tissue | 0008367 |  1340 | 1008 | ⋯ | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 | breast_mammary_tissue |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX-1117F-2826-SM-5GZXL GTEX-1117F-2826-SM-5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX-1117F-2826-SM-5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                   SMUBRID\n",
       "GTEX-1117F-2826-SM-5GZXL  5.80000 Breast Breast - Mammary Tissue 0008367\n",
       "                         SMTSISCH SMTSPAX ⋯ SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX-1117F-2826-SM-5GZXL  1340    1008    ⋯ 0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX-1117F-2826-SM-5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     \n",
       "                         tissue               \n",
       "GTEX-1117F-2826-SM-5GZXL breast_mammary_tissue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "54"
      ],
      "text/latex": [
       "54"
      ],
      "text/markdown": [
       "54"
      ],
      "text/plain": [
       "[1] 54"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 1851 15531 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17382"
      ],
      "text/latex": [
       "17382"
      ],
      "text/markdown": [
       "17382"
      ],
      "text/plain": [
       "[1] 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>15531</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 15531\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   15531\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    15531 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>15531</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 15531\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 15531\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 15531    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "15072   459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>459</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 459\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   459\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>459</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 459\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 459\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 459  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 68</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th><th scope=col>tissue</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-1117F-2826-SM-5GZXL</th><td>GTEX-1117F-2826-SM-5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>Breast - Mammary Tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td><td>breast_mammary_tissue</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 68\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY & tissue\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-1117F-2826-SM-5GZXL & GTEX-1117F-2826-SM-5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & Breast - Mammary Tissue & 0008367 &  1340 & 1008 & ⋯ & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4 & breast\\_mammary\\_tissue\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 68\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; | tissue &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-1117F-2826-SM-5GZXL | GTEX-1117F-2826-SM-5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | Breast - Mammary Tissue | 0008367 |  1340 | 1008 | ⋯ | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 | breast_mammary_tissue |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX-1117F-2826-SM-5GZXL GTEX-1117F-2826-SM-5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX-1117F-2826-SM-5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                   SMUBRID\n",
       "GTEX-1117F-2826-SM-5GZXL  5.80000 Breast Breast - Mammary Tissue 0008367\n",
       "                         SMTSISCH SMTSPAX ⋯ SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX-1117F-2826-SM-5GZXL  1340    1008    ⋯ 0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX-1117F-2826-SM-5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     \n",
       "                         tissue               \n",
       "GTEX-1117F-2826-SM-5GZXL breast_mammary_tissue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read in all requirements so that the stage is properly set -- \n",
    "# if it is clear here -- it will remain clear for the rest of the time\n",
    "# tissues.tsv contains the subset of files desired for analysis.\n",
    "tissue_reduction <- read.table(file=\"../assets/tissues.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "\n",
    "# only include those tissues we wish to continue with\n",
    "table(tissue_reduction$include)\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$include==1,]\n",
    "\n",
    "# create a matching tissue name to go with the expressionSet phenotype object\n",
    "pData(obj)$tissue      <- factor(snakecase::to_snake_case(as.character(pData(obj)$SMTSD)))\n",
    "tissue_reduction$SMTSD <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "\n",
    "dim(tissue_reduction)\n",
    "dim(obj)\n",
    "dim(pData(obj))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(obj)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = obj[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "pData(tobj)[1,]\n",
    "# end test\n",
    "\n",
    "length(levels(pData(obj)$tissue))\n",
    "length(levels(tissue_reduction$SMTSD))\n",
    "\n",
    "keep <- pData(obj)$tissue %in% tissue_reduction$SMTSD\n",
    "table(keep)\n",
    "length(keep)\n",
    "\n",
    "dim(obj)\n",
    "dim(pData(obj))\n",
    "reduced_obj <- obj[,keep==TRUE]\n",
    "dim(reduced_obj)\n",
    "dim(pData(reduced_obj))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "pData(tobj)[1,]\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Synchronize accession metadata and phenotype data\n",
    "A kind of transitive closure.   Metadata links the count data to the phenotype data.\n",
    "Begin with synchronizing accession metadata and phenotype data - which has been reduced - `reduced_obj` inputs here "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24667</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24667\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24667\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24667    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>15531</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 15531\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   15531\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    15531 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>15531</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 15531\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 15531\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 15531    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 5159 10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>10372</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 10372\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   10372\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10372</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10372\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10372\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10372    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 68</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th><th scope=col>tissue</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX.1117F.2826.SM.5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>Breast - Mammary Tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td><td>breast_mammary_tissue</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 68\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY & tissue\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX.1117F.2826.SM.5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & Breast - Mammary Tissue & 0008367 &  1340 & 1008 & ⋯ & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4 & breast\\_mammary\\_tissue\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 68\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; | tissue &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX.1117F.2826.SM.5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | Breast - Mammary Tissue | 0008367 |  1340 | 1008 | ⋯ | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 | breast_mammary_tissue |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX.1117F.2826.SM.5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX.1117F.2826.SM.5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                   SMUBRID\n",
       "GTEX.1117F.2826.SM.5GZXL  5.80000 Breast Breast - Mammary Tissue 0008367\n",
       "                         SMTSISCH SMTSPAX ⋯ SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX.1117F.2826.SM.5GZXL  1340    1008    ⋯ 0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX.1117F.2826.SM.5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     \n",
       "                         tissue               \n",
       "GTEX.1117F.2826.SM.5GZXL breast_mammary_tissue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "24667"
      ],
      "text/latex": [
       "24667"
      ],
      "text/markdown": [
       "24667"
      ],
      "text/plain": [
       "[1] 24667"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "10372"
      ],
      "text/latex": [
       "10372"
      ],
      "text/markdown": [
       "10372"
      ],
      "text/plain": [
       "[1] 10372"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>10372</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 10372\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   10372\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10372</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10372\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10372\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10372    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24667</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24667\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24667\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24667    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 6453 18214 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>18214</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 18214\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 18214\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 18214    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "17722   492 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "492"
      ],
      "text/latex": [
       "492"
      ],
      "text/markdown": [
       "492"
      ],
      "text/plain": [
       "[1] 492"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "286"
      ],
      "text/latex": [
       "286"
      ],
      "text/markdown": [
       "286"
      ],
      "text/plain": [
       "[1] 286"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR821498</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>7879119272</td><td>PRJNA75899</td><td>SAMN01994192</td><td>GTEx</td><td>GTEX-WI4N-1426-SM-3LK7H</td><td>Breast - Mammary Tissue</td><td>⋯</td><td></td><td></td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.WI4N.1426.SM.3LK7H</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR821498 & RNA:Total RNA & RNA-Seq & 152 & 7879119272 & PRJNA75899 & SAMN01994192 & GTEx & GTEX-WI4N-1426-SM-3LK7H & Breast - Mammary Tissue & ⋯ &  &  & NA & NA &  &  &  &  &  & GTEX.WI4N.1426.SM.3LK7H\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR821498 | RNA:Total RNA | RNA-Seq | 152 | 7879119272 | PRJNA75899 | SAMN01994192 | GTEx | GTEX-WI4N-1426-SM-3LK7H | Breast - Mammary Tissue | ⋯ | <!----> | <!----> | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.WI4N.1426.SM.3LK7H |\n",
       "\n"
      ],
      "text/plain": [
       "  Run       analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR821498 RNA:Total RNA RNA-Seq    152        7879119272 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN01994192 GTEx                   GTEX-WI4N-1426-SM-3LK7H         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯                                                    \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.WI4N.1426.SM.3LK7H"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# let's limit the phenotype object and then align the metadata file\n",
    "metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "metadata$SAMPID   <- gsub('-','\\\\.',metadata$'Sample Name')\n",
    "pData(reduced_obj)$SAMPID <- gsub('-','\\\\.',pData(reduced_obj)$SAMPID)\n",
    "\n",
    "dim(metadata)\n",
    "dim(reduced_obj)\n",
    "dim(pData(reduced_obj))\n",
    "rownames(pData(reduced_obj))<- pData(reduced_obj)$SAMPID\n",
    "\n",
    "# keep only those runs (as epitomized by the metadata_samples) in the phenotype set\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj)$SAMPID)\n",
    "\n",
    "# any undefined (N/A) sample names? These results will be zero\n",
    "sum(is.na(metadata_samples))\n",
    "sum(is.na(phenotype_samples))\n",
    "\n",
    "rm(keep)\n",
    "keep <- phenotype_samples %in% metadata_samples\n",
    "table(keep)\n",
    "reduced_obj2 <- reduced_obj[,keep==TRUE]\n",
    "dim(reduced_obj2)\n",
    "dim(pData(reduced_obj2))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "pData(tobj)[1,]\n",
    "# end test\n",
    "\n",
    "# now go the other way - make sure the metadata samples are in sync with the phenotype samples\n",
    "# note that we are now with `reduced_obj2`\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj2)$SAMPID)\n",
    "length(metadata_samples)\n",
    "length(phenotype_samples)\n",
    "dim(reduced_obj2)\n",
    "dim(pData(reduced_obj2))\n",
    "dim(metadata)\n",
    "rm(keep)\n",
    "keep <- metadata_samples %in% phenotype_samples\n",
    "table(keep)\n",
    "\n",
    "reduced_metadata <- metadata[keep==TRUE,]\n",
    "dim(reduced_metadata)\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "breast_metadata_samples   <- as.character(reduced_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "rm(keep)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "breast_samples <- reduced_metadata[keep,]\n",
    "length(breast_samples$SAMPID)\n",
    "length(pData(tobj)$SAMPID)\n",
    "breast_samples[1,]\n",
    "# end test\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  Limit obj and metadata to size of the ijc and sjc columns \n",
    "\n",
    "Things are complicated and can easily lead to data disaster.\n",
    "We need to go forward and backwards with these data - and see how to have them arranged.  \n",
    "Things we know to be true:\n",
    "1. Run numbers are unique\n",
    "2. Samples (which are Donor and Tissue combined) have been sequenced more than once.\n",
    "\n",
    "We need to keep this clear as the metadata are used to sort out tissues, sex, and other factors - and these are based upon a Donor.  Relationships are as follows:\n",
    "\n",
    "* One Donor can have many tissues (one donor has as many as 54 tissues)\n",
    "* One Tissue may have many sequencing runs (One Tissue has been as of this writing sequenced as many as 3 times)\n",
    "\n",
    "The analyst must keep this information in mind as manipulations occur.\n",
    "\n",
    "Here we use `se.jc.ijc` as exemplar as well as model for all subsequent matrices.  Arrangements of `se.jc.ijc` will ripple to all other matrices because sample order is the same between all samples.\n",
    "\n",
    "Entering into this step we have for our phenotype object:\n",
    "1. reduced_obj2\n",
    "2. reduced_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "metadata_for_counts\n",
       "FALSE  TRUE \n",
       " 1653  7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "counts_for_metadata\n",
       "FALSE  TRUE \n",
       "11194  7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>18214</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 18214\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 18214\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 18214    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SRR821653'</li><li>'SRR821690'</li><li>'SRR821715'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SRR821653'\n",
       "\\item 'SRR821690'\n",
       "\\item 'SRR821715'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SRR821653'\n",
       "2. 'SRR821690'\n",
       "3. 'SRR821715'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SRR821653\" \"SRR821690\" \"SRR821715\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SRR821653'</li><li>'SRR821690'</li><li>'SRR821715'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SRR821653'\n",
       "\\item 'SRR821690'\n",
       "\\item 'SRR821715'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SRR821653'\n",
       "2. 'SRR821690'\n",
       "3. 'SRR821715'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SRR821653\" \"SRR821690\" \"SRR821715\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "6891"
      ],
      "text/latex": [
       "6891"
      ],
      "text/markdown": [
       "6891"
      ],
      "text/plain": [
       "[1] 6891"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "15"
      ],
      "text/latex": [
       "15"
      ],
      "text/markdown": [
       "15"
      ],
      "text/plain": [
       "[1] 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "33"
      ],
      "text/latex": [
       "33"
      ],
      "text/markdown": [
       "33"
      ],
      "text/plain": [
       "[1] 33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata_for_counts = colnames(se.jc.ijc)%in% reduced_metadata$Run\n",
    "counts_for_metadata = reduced_metadata$Run %in% colnames(se.jc.ijc)\n",
    "table(metadata_for_counts)\n",
    "table(counts_for_metadata)\n",
    "\n",
    "counts          <- data.matrix(se.jc.ijc)\n",
    "counts_metadata <- reduced_metadata\n",
    "dim(counts)\n",
    "dim(counts_metadata)\n",
    "\n",
    "counts          <- counts         [                         ,metadata_for_counts==TRUE]\n",
    "counts_metadata <- counts_metadata[counts_for_metadata==TRUE,                         ]\n",
    "\n",
    "# now we arrange metadata and counts to be in the same order\n",
    "# Run numbers are unique, but we will replace the columns with SAMPID as metadata is by SAMPID\n",
    "i = order(colnames(counts))\n",
    "j = order(counts_metadata$Run)\n",
    "\n",
    "ose.jc.ijc       = counts[,i]\n",
    "ocounts_metadata = counts_metadata[j,]\n",
    "\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(ocounts_metadata$Run,3)\n",
    "\n",
    "colnames(ose.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(ocounts_metadata$SAMPID,3)\n",
    "sum(table(colnames(ose.jc.ijc))==1)\n",
    "sum(table(colnames(ose.jc.ijc))==2)\n",
    "sum(table(colnames(ose.jc.ijc))==3)\n",
    "sum(table(colnames(ose.jc.ijc))==4)\n",
    "dim(ose.jc.ijc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now extend this organization to all count matrices\n",
    "now that the data are organized and accounted for, extend to the rest of the splicing factors and their count matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "oa3ss.jc.ijc  <- data.matrix(a3ss.jc.ijc)\n",
    "oa3ss.jc.sjc  <- data.matrix(a3ss.jc.sjc)\n",
    "oa3ss.jc.ijc  <- oa3ss.jc.ijc[,metadata_for_counts==TRUE]\n",
    "oa3ss.jc.sjc  <- oa3ss.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "oa5ss.jc.ijc  <- data.matrix(a5ss.jc.ijc)\n",
    "oa5ss.jc.sjc  <- data.matrix(a5ss.jc.sjc)\n",
    "oa5ss.jc.ijc  <- oa5ss.jc.ijc[,metadata_for_counts==TRUE]\n",
    "oa5ss.jc.sjc  <- oa5ss.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "omxe.jc.ijc   <- data.matrix(mxe.jc.ijc)\n",
    "omxe.jc.sjc   <- data.matrix(mxe.jc.sjc)\n",
    "omxe.jc.ijc   <- omxe.jc.ijc[,metadata_for_counts==TRUE]\n",
    "omxe.jc.sjc   <- omxe.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "ori.jc.ijc    <- data.matrix(ri.jc.ijc)\n",
    "ori.jc.sjc    <- data.matrix(ri.jc.sjc)\n",
    "ori.jc.ijc    <- ori.jc.ijc[,metadata_for_counts==TRUE]\n",
    "ori.jc.sjc    <- ori.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "ose.jc.ijc    <- data.matrix(se.jc.ijc)\n",
    "ose.jc.sjc    <- data.matrix(se.jc.sjc)\n",
    "ose.jc.ijc    <- ose.jc.ijc[,metadata_for_counts==TRUE]\n",
    "ose.jc.sjc    <- ose.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "oa3ss.jc.ijc  <- oa3ss.jc.ijc[,i]\n",
    "oa3ss.jc.sjc  <- oa3ss.jc.sjc[,i]\n",
    "\n",
    "oa5ss.jc.ijc  <- oa5ss.jc.ijc[,i]\n",
    "oa5ss.jc.sjc  <- oa5ss.jc.sjc[,i]\n",
    "\n",
    "omxe.jc.ijc   <- omxe.jc.ijc[,i]\n",
    "omxe.jc.sjc   <- omxe.jc.sjc[,i]\n",
    "\n",
    "ori.jc.ijc    <- ori.jc.ijc[,i]\n",
    "ori.jc.sjc    <- ori.jc.sjc[,i]\n",
    "\n",
    "ose.jc.ijc    <- ose.jc.ijc  [,i]\n",
    "ose.jc.sjc    <- ose.jc.sjc  [,i]\n",
    "\n",
    "\n",
    "colnames(oa3ss.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(oa3ss.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(oa5ss.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(oa5ss.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(omxe.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(omxe.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(ori.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(ori.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(ose.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(ose.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "tail(colnames(oa3ss.jc.ijc),3)\n",
    "tail(colnames(oa3ss.jc.sjc),3)\n",
    "\n",
    "tail(colnames(oa5ss.jc.ijc),3)\n",
    "tail(colnames(oa5ss.jc.sjc),3)\n",
    "\n",
    "tail(colnames(omxe.jc.ijc),3)\n",
    "tail(colnames(omxe.jc.sjc),3)\n",
    "\n",
    "tail(colnames(ori.jc.ijc),3)\n",
    "tail(colnames(ori.jc.sjc),3)\n",
    "\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(colnames(ose.jc.sjc),3)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sanity Check\n",
    "\n",
    "Spot check one tissue and make sure there isn't nonsense\n",
    "Our current matrices are:\n",
    "1. ocounts_metadata for the connection from counts to phenotype via the Run and SAMPID\n",
    "2. pData(reduced_obj2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 6842   178 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "178"
      ],
      "text/latex": [
       "178"
      ],
      "text/markdown": [
       "178"
      ],
      "text/plain": [
       "[1] 178"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "286"
      ],
      "text/latex": [
       "286"
      ],
      "text/markdown": [
       "286"
      ],
      "text/plain": [
       "[1] 286"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR1068977</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>6706681864</td><td>PRJNA75899</td><td>SAMN02465522</td><td>GTEx</td><td>GTEX-XBED-1626-SM-47JYN</td><td>Breast - Mammary Tissue</td><td>⋯</td><td>P-RNA-0002</td><td>P-RNA-0002</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.XBED.1626.SM.47JYN</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR1068977 & RNA:Total RNA & RNA-Seq & 152 & 6706681864 & PRJNA75899 & SAMN02465522 & GTEx & GTEX-XBED-1626-SM-47JYN & Breast - Mammary Tissue & ⋯ & P-RNA-0002 & P-RNA-0002 & NA & NA &  &  &  &  &  & GTEX.XBED.1626.SM.47JYN\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR1068977 | RNA:Total RNA | RNA-Seq | 152 | 6706681864 | PRJNA75899 | SAMN02465522 | GTEx | GTEX-XBED-1626-SM-47JYN | Breast - Mammary Tissue | ⋯ | P-RNA-0002 | P-RNA-0002 | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.XBED.1626.SM.47JYN |\n",
       "\n"
      ],
      "text/plain": [
       "  Run        analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR1068977 RNA:Total RNA RNA-Seq    152        6706681864 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN02465522 GTEx                   GTEX-XBED-1626-SM-47JYN         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯ P-RNA-0002                P-RNA-0002               \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.XBED.1626.SM.47JYN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "breast_metadata_samples   <- as.character(ocounts_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "rm(keep)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "breast_samples <- ocounts_metadata[keep,]\n",
    "length(breast_samples$SAMPID)\n",
    "length(pData(tobj)$SAMPID)\n",
    "breast_samples[1,]\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### which metadata shall we use?\n",
    "There is a lack of correspondeence between the metadata in the Sequence Read Archive table as obtained from dbGaP SRR Run Explorer and the GTEx V8 Release.   Not sure which one to use!  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "20-29 30-39 40-49 50-59 60-69 70-79 \n",
       "  760   771  1729  3484  3312   316 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "   0    1    2    3    4 \n",
       "5523  467 2738  466 1082 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "        B1     B1, A1         C1     C1, A1 C1, B1, A1         D1     D1, A1 \n",
       "      6154        377       3110        603          3         70         55 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "   1    2 \n",
       "6796 3576 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "female   male \n",
       "  2501   4519 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(pData(reduced_obj2)$AGE)\n",
    "table(pData(reduced_obj2)$DTHHRDY)\n",
    "table(pData(reduced_obj2)$SMCENTER)\n",
    "table(pData(reduced_obj2)$SEX)\n",
    "table(ocounts_metadata$sex)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Exploring the ijc and sjc Count data \n",
    "\n",
    "For each sample, we have ijc and sjc count data, for each alternative splicing type, we have the following number of alternative splicing events:\n",
    "\n",
    "* `A3SS`: ` 8,920`\n",
    "* `A5SS`: ` 5,584`\n",
    "* `MXE `: ` 2,979`\n",
    "* `RI`  : ` 6,312`\n",
    "* `SE  `: `42,611`\n",
    "\n",
    "For exon skipping events (SE), we have 42,611 non-zero junction IDs the (first dimension of the ijc and sjc cout table) for the skipped exon event for breast-Mammary Tissue, 191 individuals.  These are healthy individuals, and we are studying the impact of sex on the occurrence or non-occurance of specific alternative splicing events.   We explore the information we ahve about these junctions and create a construct, as_event, which accounts for the junction under exploration.\n",
    "\n",
    "The `IJC` and `SJC` counts are in many ways two sides of the same coin.  Both our the observational output and we wish to see how robust each are in their ability to separate out the samples to provide for us differentially expressed isoform events as measured by their counts.   Each junction is in a manner a specific marker to specific isoform events that may or may not be shared between the genders.   If there is significant results, then this is indicative of the separation achieved by isoform specific differentiation.   In our model we will use these in combination, it is important to see if they will yield the results we are looking for.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reduce count matrices based upon the reduced tissue information - count matrix by column adjustment\n",
    "\n",
    "We wish to keep only those samples (columns) in our count data that are a member of the reduced tissue set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "keep\n",
       "TRUE \n",
       "7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata_names <- as.character(ocounts_metadata$SAMPID)\n",
    "count_names    <- as.character(colnames(ose.jc.ijc))\n",
    "\n",
    "keep  <- count_names %in% metadata_names\n",
    "\n",
    "table(keep)\n",
    "length(keep)\n",
    "length(metadata_names)\n",
    "length(count_names)\n",
    "\n",
    "a3ss.jc.ijc2 = oa3ss.jc.ijc[,keep==TRUE] \n",
    "a3ss.jc.sjc2 = oa3ss.jc.sjc[,keep==TRUE] \n",
    "\n",
    "a5ss.jc.ijc2 = oa5ss.jc.ijc[,keep==TRUE] \n",
    "a5ss.jc.sjc2 = oa5ss.jc.sjc[,keep==TRUE] \n",
    "\n",
    "mxe.jc.ijc2 = omxe.jc.ijc[,keep==TRUE] \n",
    "mxe.jc.sjc2 = omxe.jc.sjc[,keep==TRUE] \n",
    "\n",
    "ri.jc.ijc2 = ori.jc.ijc[,keep==TRUE] \n",
    "ri.jc.sjc2 = ori.jc.sjc[,keep==TRUE] \n",
    "\n",
    "se.jc.ijc2 = ose.jc.ijc[,keep==TRUE] \n",
    "se.jc.sjc2 = ose.jc.sjc[,keep==TRUE] \n",
    "\n",
    "dim(a3ss.jc.ijc2)\n",
    "dim(a3ss.jc.sjc2)\n",
    "dim(a5ss.jc.ijc2)\n",
    "dim(a5ss.jc.sjc2)\n",
    "dim(mxe.jc.ijc2)\n",
    "dim(mxe.jc.sjc2)\n",
    "dim(ri.jc.ijc2)\n",
    "dim(ri.jc.sjc2)\n",
    "dim(se.jc.ijc2)\n",
    "dim(se.jc.sjc2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Keeping only chromosomes shared male female - count matrix by junction row adjustment \n",
    "\n",
    "The Y chromosome spans more than 59 million base pairs of DNA and represents almost 2 percent of the total DNA in cells. Each person normally has one pair of sex chromosomes in each cell. The Y chromosome is present in males, who have one X and one Y chromosome, while females have two X chromosomes. Since our analysis is on the comparative differences, we must eliminate chrY from our analyses.\n",
    "\n",
    "To do so, we grab the annotation from the GTF file and remove those junctions that correspond to the genes on this chromosome.\n",
    "\n",
    "So this is a count matrix row adjustment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611    11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5177"
      ],
      "text/latex": [
       "5177"
      ],
      "text/markdown": [
       "5177"
      ],
      "text/plain": [
       "[1] 5177"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3824"
      ],
      "text/latex": [
       "3824"
      ],
      "text/markdown": [
       "3824"
      ],
      "text/plain": [
       "[1] 3824"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1619"
      ],
      "text/latex": [
       "1619"
      ],
      "text/markdown": [
       "1619"
      ],
      "text/plain": [
       "[1] 1619"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3617"
      ],
      "text/latex": [
       "3617"
      ],
      "text/markdown": [
       "3617"
      ],
      "text/plain": [
       "[1] 3617"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "13390"
      ],
      "text/latex": [
       "13390"
      ],
      "text/markdown": [
       "13390"
      ],
      "text/plain": [
       "[1] 13390"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  862   230   643   552   108   304   326   537   626   118   783   578   241 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   85   246   514   260   337   388   377   255   264   269    17 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  496   155   386   354    59   212   188   348   410    74   400   387   135 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   60   129   368   184   210   216   284   166   162   191    10 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  286   120   152   166    46   109    97   137   154    70   116   386    67 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   40    49   205   144   127   155   106   104    74    61     8 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  604   117   475   423    46   241   191   441   541    40   637   327   153 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   64   194   371   133   181   284   250   185   192   207    15 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       " 3812  1458  2541  2540   663  1551  1544  2039  2538   742  2552  3547   939 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "  493   895  2938  1803  1863  1890  2053  1636  1313  1134   127 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A3SS.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   17  8903 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "8920"
      ],
      "text/latex": [
       "8920"
      ],
      "text/markdown": [
       "8920"
      ],
      "text/plain": [
       "[1] 8920"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A5SS.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   10  5574 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5584"
      ],
      "text/latex": [
       "5584"
      ],
      "text/markdown": [
       "5584"
      ],
      "text/plain": [
       "[1] 5584"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "MXE.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "    8  2971 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2979"
      ],
      "text/latex": [
       "2979"
      ],
      "text/markdown": [
       "2979"
      ],
      "text/plain": [
       "[1] 2979"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RI.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   15  6297 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "6312"
      ],
      "text/latex": [
       "6312"
      ],
      "text/markdown": [
       "6312"
      ],
      "text/plain": [
       "[1] 6312"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "SE.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "  127 42484 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "42611"
      ],
      "text/latex": [
       "42611"
      ],
      "text/markdown": [
       "42611"
      ],
      "text/plain": [
       "[1] 42611"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484    11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(fromGTF.A3SS)\n",
    "dim(fromGTF.A5SS)\n",
    "dim(fromGTF.MXE)\n",
    "dim(fromGTF.RI)\n",
    "dim(fromGTF.SE)\n",
    "\n",
    "A3SS.genes <- factor(fromGTF.A3SS$geneSymbol)\n",
    "A5SS.genes <- factor(fromGTF.A5SS$geneSymbol)\n",
    "MXE.genes  <- factor(fromGTF.MXE$geneSymbol)\n",
    "RI.genes   <- factor(fromGTF.RI$geneSymbol)\n",
    "SE.genes   <- factor(fromGTF.SE$geneSymbol)\n",
    "\n",
    "length(levels(A3SS.genes))    \n",
    "length(levels(A5SS.genes))    \n",
    "length(levels(MXE.genes))    \n",
    "length(levels(RI.genes))    \n",
    "length(levels(SE.genes))    \n",
    "\n",
    "table(fromGTF.A3SS$chr)\n",
    "table(fromGTF.A5SS$chr)\n",
    "table(fromGTF.MXE$chr)\n",
    "table(fromGTF.RI$chr)\n",
    "table(fromGTF.SE$chr)\n",
    "\n",
    "A3SS.keepAllJunctionsButChrY <- (fromGTF.A3SS$chr != \"chrY\")\n",
    "A5SS.keepAllJunctionsButChrY <- (fromGTF.A5SS$chr != \"chrY\")\n",
    "MXE.keepAllJunctionsButChrY  <- (fromGTF.MXE$chr  != \"chrY\")\n",
    "RI.keepAllJunctionsButChrY   <- (fromGTF.RI$chr   != \"chrY\")\n",
    "SE.keepAllJunctionsButChrY   <- (fromGTF.SE$chr   != \"chrY\")\n",
    "\n",
    "table(A3SS.keepAllJunctionsButChrY)\n",
    "sum(table(A3SS.keepAllJunctionsButChrY))\n",
    "\n",
    "table(A5SS.keepAllJunctionsButChrY)\n",
    "sum(table(A5SS.keepAllJunctionsButChrY))\n",
    "\n",
    "table(MXE.keepAllJunctionsButChrY)\n",
    "sum(table(MXE.keepAllJunctionsButChrY))\n",
    "\n",
    "table(RI.keepAllJunctionsButChrY)\n",
    "sum(table(RI.keepAllJunctionsButChrY))\n",
    "\n",
    "table(SE.keepAllJunctionsButChrY)\n",
    "sum(table(SE.keepAllJunctionsButChrY))\n",
    "\n",
    "a3ss_fromGTF_no_chrY          <- fromGTF.A3SS[A3SS.keepAllJunctionsButChrY,]\n",
    "a3ss_jc_ijc_no_chrY           <- a3ss.jc.ijc2[A3SS.keepAllJunctionsButChrY,]\n",
    "a3ss_jc_sjc_no_chrY           <- a3ss.jc.sjc2[A3SS.keepAllJunctionsButChrY,]\n",
    "rownames(a3ss_jc_ijc_no_chrY) <- rownames(a3ss_fromGTF_no_chrY)\n",
    "rownames(a3ss_jc_sjc_no_chrY) <- rownames(a3ss_fromGTF_no_chrY)\n",
    "\n",
    "a5ss_fromGTF_no_chrY          <- fromGTF.A5SS[A5SS.keepAllJunctionsButChrY,]\n",
    "a5ss_jc_ijc_no_chrY           <- a5ss.jc.ijc2[A5SS.keepAllJunctionsButChrY,]\n",
    "a5ss_jc_sjc_no_chrY           <- a5ss.jc.sjc2[A5SS.keepAllJunctionsButChrY,]\n",
    "rownames(a5ss_jc_ijc_no_chrY) <- rownames(a5ss_fromGTF_no_chrY)\n",
    "rownames(a5ss_jc_sjc_no_chrY) <- rownames(a5ss_fromGTF_no_chrY)\n",
    "\n",
    "mxe_fromGTF_no_chrY          <- fromGTF.MXE[MXE.keepAllJunctionsButChrY,]\n",
    "mxe_jc_ijc_no_chrY           <- mxe.jc.ijc2[MXE.keepAllJunctionsButChrY,]\n",
    "mxe_jc_sjc_no_chrY           <- mxe.jc.sjc2[MXE.keepAllJunctionsButChrY,]\n",
    "rownames(mxe_jc_ijc_no_chrY) <- rownames(mxe_fromGTF_no_chrY)\n",
    "rownames(mxe_jc_sjc_no_chrY) <- rownames(mxe_fromGTF_no_chrY)\n",
    "\n",
    "ri_fromGTF_no_chrY          <- fromGTF.RI[RI.keepAllJunctionsButChrY,]\n",
    "ri_jc_ijc_no_chrY           <- ri.jc.ijc2[RI.keepAllJunctionsButChrY,]\n",
    "ri_jc_sjc_no_chrY           <- ri.jc.sjc2[RI.keepAllJunctionsButChrY,]\n",
    "rownames(ri_jc_ijc_no_chrY) <- rownames(ri_fromGTF_no_chrY)\n",
    "rownames(ri_jc_sjc_no_chrY) <- rownames(ri_fromGTF_no_chrY)\n",
    "\n",
    "se_fromGTF_no_chrY          <- fromGTF.SE[SE.keepAllJunctionsButChrY,]\n",
    "se_jc_ijc_no_chrY           <- se.jc.ijc2[SE.keepAllJunctionsButChrY,]\n",
    "se_jc_sjc_no_chrY           <- se.jc.sjc2[SE.keepAllJunctionsButChrY,]\n",
    "rownames(se_jc_ijc_no_chrY) <- rownames(se_fromGTF_no_chrY)\n",
    "rownames(se_jc_sjc_no_chrY) <- rownames(se_fromGTF_no_chrY)\n",
    "\n",
    "dim(a3ss_fromGTF_no_chrY)\n",
    "dim(a5ss_fromGTF_no_chrY)\n",
    "dim(mxe_fromGTF_no_chrY)\n",
    "dim(ri_fromGTF_no_chrY)\n",
    "dim(se_fromGTF_no_chrY)\n",
    "\n",
    "dim(a3ss_jc_ijc_no_chrY)\n",
    "dim(a3ss_jc_sjc_no_chrY)\n",
    "\n",
    "dim(a5ss_jc_ijc_no_chrY)\n",
    "dim(a5ss_jc_sjc_no_chrY)\n",
    "\n",
    "dim(mxe_jc_ijc_no_chrY)\n",
    "dim(mxe_jc_sjc_no_chrY)\n",
    "\n",
    "dim(ri_jc_ijc_no_chrY)\n",
    "dim(ri_jc_sjc_no_chrY)\n",
    "\n",
    "dim(se_jc_ijc_no_chrY)\n",
    "dim(se_jc_sjc_no_chrY)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exploratory and Differential analysis as_event:ijc, sjc \n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) to transform junction counts (reads that were aligned to junctions when an exon is included - ijc, and reads that were aligned to junctions when the exon is excluded - sjc) with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma.    In each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression: \n",
    "\n",
    "           y = B0 + B1 sex + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the included exon junction count expression; sex denotes the reported sex of the subject\n",
    "\n",
    "## Differential analysis as_event (combined ijc and sjc)\n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) to transform junction counts (reads that were aligned to junctions when an exon is included - ijc, and reads that were aligned to junctions when the exon is excluded - sjc) with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma.    In each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression: \n",
    "\n",
    "           y = B0 + B1 sex + B2 as_event + B3 sex*as_event + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the alternative splicing event expression; sex denotes the reported sex of the subject, as_event represents the specific alternative splicing event - either included exon junction counts or skipped exon junction counts and their interaction terms.   Donor is added to our model as a blocking variable used in both the calculation of duplicate correlation as well as in the linear fit.\n",
    "\n",
    "### Voom, limma's lmFit and eBayes\n",
    "\n",
    "Using sample as a blocking variable, we are able to model the effects of the donor on the results, which improves the power.  This topic is discussed in biostars https://www.biostars.org/p/54565/.  And Gordon Smyth answers the question here https://mailman.stat.ethz.ch/pipermail/bioconductor/2014-February/057887.html.  The method of modeling is a random effects approach in which the intra-donor correlation is incorporated into the covariance matrix instead of the linear predictor.   And though as Gordon Smyth states both are good method and the twoway anova approach makes fewer assumptions, the random effects approach is statistically more powerful.  \n",
    "\n",
    "We have a balanced design in which all donors receive all stimuli (which is really in healthy human donors, life and all of its factors!) Our measurement has so many points -- we are measuring in the skipped exon approach, 42,611 junctions!   It is not possible to encorporate those measurements into the linear predictor.  A two-way ANOVA approach is virtually as powerful as the random effects approach \n",
    "and hence is preferable as it makes fewer assumptions.\n",
    "\n",
    "For an unbalanced design in which each donor receives only a subset of the stimula, the random effects approach is more powerful.\n",
    "\n",
    "Random effects approach is equivalent to The first method is twoway anova, a generalization of a paired analysis.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "print_exploratory_plots <- function (plot, tissue_of_interest, splice_type, fromGTF, tissue_list, ijc, sjc, obj, metadata ) {\n",
    "\n",
    "    fromGTF           <- fromGTF\n",
    "    tissue_true       <- tissue_list == tissue_of_interest\n",
    "\n",
    "    # test to make sure we don't have nonsense\n",
    "    # like proteins - this process moves in one direction\n",
    "    # tissue will get us the phenotype selection \n",
    "    # from the phenotype SAMPID we get the metadata SAMPID\n",
    "    # which leads us to counts - it goes in this direction.\n",
    "    # first phenotype obj subsetting for the tissue_of_interest\n",
    "\n",
    "    message(\"Limiting phenotype data to tissue of interest\")    \n",
    "    keep = pData(reduced_obj2)$tissue== tissue_of_interest\n",
    "    table(keep)\n",
    "    tissue_obj<- reduced_obj2[,keep]\n",
    "    message(\"tissue_obj now reduced to the tissue of interest\")    \n",
    "    dim(pData(tissue_obj))\n",
    "    pData(tissue_obj)[1,]\n",
    "\n",
    "\n",
    "    # now we get the runs via the metadata ocounts_metadata matrix via SAMPID\n",
    "    # limit the phenotype info to just the samples we care about\n",
    "    metadata_samples          <- as.character(ocounts_metadata$SAMPID)\n",
    "    tissue_phenotype_samples  <- as.character(pData(tissue_obj)$SAMPID)\n",
    "\n",
    "    message(\"Limit the phenotypes to those we have samples for\")    \n",
    "    rm(keep)\n",
    "    keep = tissue_phenotype_samples %in% metadata_samples\n",
    "    table(keep)\n",
    "    message(\"tissue_count_obj now has these count specific phenotypes\")    \n",
    "    tissue_count_obj        <- tissue_obj[keep,]\n",
    "    pData(tissue_count_obj) <- pData(tissue_obj)[keep,]\n",
    "    dim(pData(tissue_count_obj))\n",
    "    dim(tissue_count_obj)\n",
    "\n",
    "    # and vice-versa, limit our samples to the ones we have phenotype for\n",
    "    rm(keep)\n",
    "    message(\"Limit the counts now has these count specific phenotypes\")    \n",
    "    metadata_samples       <- as.character(ocounts_metadata$SAMPID)\n",
    "    phenotype_samples      <- as.character(pData(tissue_count_obj)$SAMPID)\n",
    "    keep = metadata_samples %in% phenotype_samples\n",
    "    message(\"metadata_tissue now has these phenotype specific counts\")    \n",
    "    table(keep)\n",
    "\n",
    "    ijc_tissue        <- ijc     [          ,keep==TRUE]\n",
    "    sjc_tissue        <- sjc     [          ,keep==TRUE]\n",
    "    metadata_tissue   <- metadata[keep==TRUE,          ]\n",
    "\n",
    "    message(\"dimensions of the ijc_tissue,sjc_tissue, metadata_tissue and pData(tissue_count_obj)\")\n",
    "    dim(ijc_tissue)\n",
    "    dim(sjc_tissue)\n",
    "    dim(metadata_tissue)\n",
    "    metadata_samples       <- as.character(metadata_tissue$SAMPID)\n",
    "    phenotype_samples      <- as.character(pData(tissue_count_obj)$SAMPID)\n",
    "\n",
    "    length(metadata_samples)\n",
    "    length(phenotype_samples)\n",
    "    non_overlaps <- setdiff( metadata_samples, phenotype_samples)\n",
    "    message(\"The non-overlapping IDs between pheno and count data are:\\n\\n\", \n",
    "        paste(length(non_overlaps), collapse = \"\\n\") )\n",
    "\n",
    "    for (i in (1:length(metadata_samples))) {\n",
    "        sample = metadata_tissue[i,]$SAMPID\n",
    "        sample_sex = pData(tissue_count_obj)$SEX[pData(tissue_count_obj)$SAMPID == sample]\n",
    "        if (i==1) {\n",
    "            sex = sample_sex\n",
    "        } else {\n",
    "            sex = c(sex,sample_sex)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    message(\"sex samples:\\n\\n\",\n",
    "        paste0(table(sex), collapse=\"\\n\"))\n",
    "\n",
    "\n",
    "    ijc.dm            <- data.matrix(ijc_tissue)\n",
    "    sjc.dm            <- data.matrix(sjc_tissue)    \n",
    "\n",
    "\n",
    "    sex      <- ifelse(sex == 1,\"male\",\"female\")\n",
    "    sex      <- factor(sex,levels=c(\"male\",\"female\"))\n",
    "    message(\"sex samples:\\n\\n\",\n",
    "        paste0(table(sex), collapse=\"\\n\"))\n",
    "\n",
    "    design    <- model.matrix ( ~ sex)\n",
    "\n",
    "    colnames(design) <- c(\"intercept\",\"sex\")\n",
    "\n",
    "    y_ijc <- DGEList(counts=ijc.dm, group = sex)\n",
    "    y_ijc <- calcNormFactors(y_ijc, method=\"RLE\")\n",
    "    y_ijc_voom <- voom (y_ijc, design=design, plot=plot)\n",
    "\n",
    "    fit_ijc <- lmFit(y_ijc_voom, design)\n",
    "    fit_ijc <- eBayes(fit_ijc)\n",
    "\n",
    "    ijc_sex_results                    <- topTable(fit_ijc, coef='sex', number=nrow(y_ijc_voom))\n",
    "    ijc_sex_results_refined            <- ijc_sex_results$adj.P.Val <= 0.05 & abs(ijc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    ijc_sex_rnResults                  <- rownames(ijc_sex_results)\n",
    "    ijc_sex_resultsAnnotations         <- fromGTF[ijc_sex_rnResults,]\n",
    "\n",
    "    ijc_sex_results_refinedAnnotations <- ijc_sex_resultsAnnotations[ijc_sex_results_refined      ==TRUE,]\n",
    "    dim (ijc_sex_results_refinedAnnotations)\n",
    "    head(ijc_sex_results_refinedAnnotations)\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    ijc_sex_geneSymbols               <- ijc_sex_resultsAnnotations$geneSymbol\n",
    "    ijc_sex_refined_geneSymbols       <- ijc_sex_results_refinedAnnotations$geneSymbol\n",
    "    length(ijc_sex_refined_geneSymbols)\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    ijc_sex_results_rn         <- paste(ijc_sex_geneSymbols,       ijc_sex_rnResults, sep=\"-\")\n",
    "    rownames(ijc_sex_results)  <- ijc_sex_results_rn\n",
    "\n",
    "    length(ijc_sex_refined_geneSymbols)\n",
    "    y_sjc <- DGEList(counts=sjc.dm, group = sex)\n",
    "    y_sjc <- calcNormFactors(y_sjc, method=\"RLE\")\n",
    "    y_sjc_voom <- voom (y_sjc, design=design, plot=plot)\n",
    "\n",
    "    fit_sjc <- lmFit(y_sjc_voom, design)\n",
    "    fit_sjc <- eBayes(fit_sjc)\n",
    "\n",
    "    sjc_sex_results                    <- topTable(fit_sjc, coef='sex', number=nrow(y_sjc_voom))\n",
    "    sjc_sex_results_refined            <- sjc_sex_results$adj.P.Val <= 0.05 & abs(sjc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    sjc_sex_rnResults                  <- rownames(sjc_sex_results)\n",
    "    sjc_sex_resultsAnnotations         <- fromGTF[sjc_sex_rnResults,]\n",
    "\n",
    "    sjc_sex_results_refinedAnnotations <- sjc_sex_resultsAnnotations[sjc_sex_results_refined      ==TRUE,]\n",
    "    dim (sjc_sex_results_refinedAnnotations)\n",
    "    head(sjc_sex_results_refinedAnnotations)\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sjc_sex_geneSymbols               <- sjc_sex_resultsAnnotations$geneSymbol\n",
    "    sjc_sex_refined_geneSymbols       <- sjc_sex_results_refinedAnnotations$geneSymbol\n",
    "    length(sjc_sex_refined_geneSymbols)\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sjc_sex_results_rn         <- paste(sjc_sex_geneSymbols,       sjc_sex_rnResults, sep=\"-\")\n",
    "    head(sjc_sex_results_rn)\n",
    "    rownames(sjc_sex_results)       <- sjc_sex_results_rn\n",
    "\n",
    "    length(sjc_sex_results_refined) \n",
    "   # print these exploratory plots for ijc and sjc alone\n",
    "    Gender <- substring(sex,1,1)\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_ijc, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-ijc-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "        plotMDS(y_ijc_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    " \n",
    "    fit_ijc <- lmFit(y_ijc_voom, design)\n",
    "    fit_ijc <- eBayes(fit_ijc)\n",
    "    ijc_sex_results          <- topTable(fit_ijc, coef='sex', number=nrow(y_ijc_voom))\n",
    "    ijc_sex_results_refined  <- ijc_sex_results$adj.P.Val < 0.05 & abs(ijc_sex_results$logFC) > abs(log2(1.5))\n",
    "    ijc_sex_rnResults        <- rownames(ijc_sex_results)\n",
    "\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "           plotMDS(y_sjc, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-sjc-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "          plotMDS(y_sjc_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "        \n",
    "    fit_sjc <- lmFit(y_sjc_voom, design)\n",
    "    fit_sjc <- eBayes(fit_sjc)\n",
    "        \n",
    "    sjc_sex_results                    <- topTable(fit_sjc, coef='sex', number=nrow(y_sjc_voom))\n",
    "    sjc_sex_results_refined            <- sjc_sex_results$adj.P.Val <= 0.05 & abs(sjc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    sjc_sex_rnResults                  <- rownames(sjc_sex_results)\n",
    "    sjc_sex_resultsAnnotations         <- fromGTF[sjc_sex_rnResults,]\n",
    "    sjc_sex_results_refinedAnnotations <- sjc_sex_resultsAnnotations[sjc_sex_results_refined ==TRUE,]\n",
    "\n",
    "    head(sjc_sex_resultsAnnotations)\n",
    "    head(sjc_sex_results_refinedAnnotations)\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sjc_sex_geneSymbols               <- sjc_sex_resultsAnnotations$geneSymbol\n",
    "    sjc_sex_refined_geneSymbols       <- sjc_sex_results_refinedAnnotations$geneSymbol\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sjc_sex_results_rn             <- paste(sjc_sex_geneSymbols,       sjc_sex_rnResults, sep=\"-\")\n",
    "    head(sjc_sex_results_rn)    \n",
    "    rownames(sjc_sex_results)      <- sjc_sex_results_rn\n",
    "    sjc_sex_filename               = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sjc_sex.csv',sep='')\n",
    "    sjc_sex_refined_filename       = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sjc_sex_refined.csv',sep='')\n",
    "    sjc_sex_genesFilename          = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sjc_sex_universe.txt',sep='')\n",
    "    sjc_sex_refined_genesFilename  = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sjc_sex_gene_set.txt',sep='')\n",
    "\n",
    "    write.table(sjc_sex_results, \n",
    "                file = sjc_sex_filename        , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_results [sjc_sex_results_refined      ,], \n",
    "                file = sjc_sex_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_geneSymbols,        \n",
    "                file = sjc_sex_genesFilename        , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_refined_geneSymbols,\n",
    "                file = sjc_sex_refined_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    \n",
    "    sample_names <- as.character(colnames(ijc.dm))\n",
    "    # we will add donor as a blocking parameter\n",
    "    # rather than sample name -- we should use donor for real\n",
    "    sample     <- factor(sample_names)\n",
    "    \n",
    "    donor    <- rep(sample, 2)\n",
    "\n",
    "    \n",
    "    ijc_names <- as.character(colnames(ijc.dm))\n",
    "    sjc_names <- as.character(colnames(sjc.dm))\n",
    "    sjc_names <- paste0(sjc_names,\"-sjc\")\n",
    "    ijc_names <- paste0(ijc_names,\"-ijc\")\n",
    "\n",
    "    colnames(ijc.dm) <- ijc_names\n",
    "    colnames(sjc.dm) <- sjc_names\n",
    "\n",
    "    as_matrix <- cbind(ijc.dm,sjc.dm)\n",
    "    sex3      <- c(rep(sex,2))\n",
    "    head(sex)\n",
    "    as_event  <- c(rep(\"ijc\",dim(ijc.dm)[2]), rep(\"sjc\", dim(sjc.dm)[2]))\n",
    "    as_event  <- factor(as_event, levels=c(\"ijc\", \"sjc\"))\n",
    "\n",
    "    length(as_event)\n",
    "\n",
    "    design    <- model.matrix( ~ sex3 + as_event + sex3*as_event)\n",
    "\n",
    "    colnames(design) <- c(\"intercept\",\"sex\", \"as_event\",\"sex*as_event\")\n",
    "\n",
    "    dim(as_matrix)\n",
    "    table(sex3)\n",
    "    table(as_event)\n",
    "    head(design)\n",
    "\n",
    "    y <- DGEList(counts=as_matrix, group = sex3)\n",
    "    y <- calcNormFactors(y, method=\"RLE\")\n",
    "    y_voom <- voom (y, design=design, plot = plot)\n",
    "\n",
    "#    commenting this out for speed\n",
    "#    dup_cor <- duplicateCorrelation(y_voom$E, design=design, ndups=2, block=donor, weights=y$samples$norm.factors)\n",
    "#    dup_cor$consensus.correlation \n",
    "#    y_dup_voom <- voom (y, design=design, plot = plot, block = donor, correlation = dup_cor$consensus.correlation)\n",
    "    \n",
    "    Gender <- substring(sex3,1,1)\n",
    "     # print the combined exploratory plot\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-voom-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_voom[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(\"../pdf/\", snakecase::to_snake_case(tissue_of_interest)),\"-y-dup-voom-ijc-MDSplot-100.pdf\")\n",
    "#    pdf (filename)\n",
    "#        plotMDS(y_dup_voom[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "#            gene.selection=\"common\")\n",
    "#    dev.off()\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-voom-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_voom[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", snakecase::to_snake_case(splice_type)),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-dup-voom-sjc-MDSplot-100.pdf\")\n",
    "#    pdf (filename)\n",
    "#        plotMDS(y_dup_voom[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "#            gene.selection=\"common\")\n",
    "#    dev.off()\n",
    "                       \n",
    "# for comment this out for speed\n",
    "#   fit <- lmFit(y_dup_voom, design=design, block=donor, correlation = dup_cor$consensus.correlation)\n",
    "    fit <- lmFit(y_voom, design=design)\n",
    "    fit <- eBayes(fit, robust=TRUE)    \n",
    "    \n",
    "    sex_as_events_results         <- topTable(fit, coef=\"sex*as_event\", number=nrow(y_voom))\n",
    "    sex_as_events_results_refined <- sex_as_events_results$adj.P.Val <= 0.05 & abs(sex_as_events_results$logFC) >= abs(log2(1.5))\n",
    "\n",
    "    sex_results                   <- topTable(fit, coef=\"sex\", number=nrow(y_voom))\n",
    "    sex_results_refined           <- sex_results$adj.P.Val <= 0.05 & abs(sex_results$logFC) >= abs(log2(1.5))\n",
    "\n",
    "    sex_as_events_rnResults <- rownames(sex_as_events_results)\n",
    "    sex_rnResults           <- rownames(sex_results)\n",
    "    head(sex_as_events_rnResults)\n",
    "    head(ijc_sex_rnResults)\n",
    "    head(sex_rnResults)\n",
    "    head(fromGTF[sex_as_events_rnResults,])\n",
    "\n",
    "    # use the junctionIDs to get the annotations\n",
    "    sex_as_events_resultsAnnotations      <- fromGTF[sex_as_events_rnResults,]\n",
    "    sex_resultsAnnotations                <- fromGTF[sex_rnResults,]\n",
    "    ijc_sex_resultsAnnotations            <- fromGTF[ijc_sex_rnResults,]\n",
    "    head(sex_as_events_resultsAnnotations)\n",
    "    head(sex_resultsAnnotations)\n",
    "    head(ijc_sex_resultsAnnotations)\n",
    "    \n",
    "    sex_as_events_results_refinedAnnotations<- sex_as_events_resultsAnnotations[sex_as_events_results_refined==TRUE,]\n",
    "    sex_results_refinedAnnotations          <- sex_resultsAnnotations          [sex_results_refined          ==TRUE,]\n",
    "    ijc_sex_results_refinedAnnotations      <- ijc_sex_resultsAnnotations      [ijc_sex_results_refined      ==TRUE,]\n",
    "    head(sex_as_events_results_refinedAnnotations)\n",
    "    head(sex_results_refinedAnnotations)\n",
    "    head(ijc_sex_results_refinedAnnotations)\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sex_as_events_geneSymbols         <- sex_as_events_resultsAnnotations$geneSymbol\n",
    "    sex_geneSymbols                   <- sex_resultsAnnotations$geneSymbol\n",
    "    ijc_sex_geneSymbols               <- ijc_sex_resultsAnnotations$geneSymbol\n",
    "\n",
    "    sex_as_events_refined_geneSymbols <- sex_as_events_results_refinedAnnotations$geneSymbol\n",
    "    sex_refined_geneSymbols           <- sex_results_refinedAnnotations$geneSymbol\n",
    "    ijc_sex_refined_geneSymbols       <- ijc_sex_results_refinedAnnotations$geneSymbol\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sex_as_events_results_rn   <- paste(sex_as_events_geneSymbols, sex_as_events_rnResults, sep=\"-\")\n",
    "    sex_results_rn             <- paste(sex_geneSymbols,           sex_rnResults, sep=\"-\")\n",
    "    ijc_sex_results_rn         <- paste(ijc_sex_geneSymbols,       ijc_sex_rnResults, sep=\"-\")\n",
    "    head(sex_as_events_results_rn)\n",
    "    head(sex_results_rn)\n",
    "    head(ijc_sex_results_rn)\n",
    "    rownames(sex_as_events_results) <- sex_as_events_results_rn\n",
    "    rownames(sex_results)           <- sex_results_rn\n",
    "    rownames(ijc_sex_results)       <- ijc_sex_results_rn\n",
    "    \n",
    "    sex_as_events_filename         = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_as_events.csv')\n",
    "    sex_filename                   = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex.csv',sep='')\n",
    "    ijc_sex_filename               = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_ijc_sex.csv',sep='')\n",
    "    sex_as_events_refined_filename = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_as_events_refined.csv',sep='')\n",
    "    sex_refined_filename           = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_refined.csv',sep='')\n",
    "    ijc_sex_refined_filename       = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_ijc_sex_refined.csv',sep='')\n",
    "    sex_as_events_genesFilename    = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_as_events_universe.txt',sep='')\n",
    "    sex_genesFilename              = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_universe.txt',sep='')\n",
    "    ijc_sex_genesFilename          = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_ijc_sex_universe.txt',sep='')\n",
    "    sex_as_events_refined_genesFilename = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                        snakecase::to_snake_case(tissue_of_interest)),'_sex_as_events_gene_set.txt',sep='')\n",
    "    sex_refined_genesFilename           = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                        snakecase::to_snake_case(tissue_of_interest)),'_sex_gene_set.txt',sep='')\n",
    "    ijc_sex_refined_genesFilename       = paste0(paste0(paste0('../significant_events/', splice_type),\n",
    "                                                        snakecase::to_snake_case(tissue_of_interest)),'_ijc_sex_gene_set.txt',sep='')\n",
    "\n",
    "    write.table(sex_as_events_results, file = sex_as_events_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_results,           file = sex_filename          , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_results,       file = ijc_sex_filename      , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "\n",
    "    write.table(sex_as_events_results[sex_as_events_results_refined,], file = sex_as_events_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_results          [sex_results_refined          ,], file = sex_refined_filename          , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_results      [ijc_sex_results_refined      ,], file = ijc_sex_refined_filename      , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "\n",
    "    write.table(sex_as_events_geneSymbols, file = sex_as_events_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_geneSymbols,           file = sex_genesFilename          , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_geneSymbols,       file = ijc_sex_genesFilename      , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "\n",
    "    write.table(sex_as_events_refined_geneSymbols,file = sex_as_events_refined_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_refined_geneSymbols,          file = sex_refined_genesFilename          , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_refined_geneSymbols,      file = ijc_sex_refined_genesFilename      , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "\n",
    "    return(0)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# tissue_index -- enables this to run as a NextFlow notebook\n",
    "# devtools::install_github(\"ropensci/piggyback@87f71e8\", upgrade=\"never\")\n",
    "#parameters for running the notebook as NextFlow\n",
    "\n",
    "splice_list       = c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\",\"se_\")\n",
    "\n",
    "tissue_index <- 17\n",
    "splice_index <- 5\n",
    "\n",
    "tissue_list_m_f     = levels(reduced_metadata_pData$tissue)\n",
    "tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "splice_type         = splice_list    [splice_index] \n",
    "fromGTF             = a3ss_fromGTF_no_chrY\n",
    "metadata            = ocounts_metadata\n",
    "ijc                 = a3ss_jc_ijc_no_chrY\n",
    "sjc                 = a3ss_jc_sjc_no_chrY\n",
    "plot                = FALSE\n",
    "obj                 = reduced_obj2\n",
    "\n",
    "for (tissue_index in 1:length(tissue_list_m_f)) {\n",
    "\n",
    "    # a3ss\n",
    "    splice_index        = 1\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = a3ss_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = a3ss_jc_ijc_no_chrY\n",
    "    sjc                 = a3ss_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # a5ss\n",
    "    splice_index        = 2\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = a5ss_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = a5ss_jc_ijc_no_chrY\n",
    "    sjc                 = a5ss_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # mxe\n",
    "    splice_index        = 3\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = mxe_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = mxe_jc_ijc_no_chrY\n",
    "    sjc                 = mxe_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # ri\n",
    "    splice_index        = 4\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = ri_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = ri_jc_ijc_no_chrY\n",
    "    sjc                 = ri_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # se\n",
    "#    splice_index        = 5\n",
    "#    splice_type         = splice_list    [splice_index] \n",
    "#    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "#    fromGTF             = se_fromGTF_no_chrY\n",
    "#    metadata            = reduced_metadata_pData\n",
    "#    ijc                 = se_jc_ijc_no_chrY\n",
    "#    sjc                 = se_jc_sjc_no_chrY\n",
    "#    print_exploratory_plots (plot, \n",
    "#                             tissue_of_interest, \n",
    "#                             splice_type, \n",
    "#                             fromGTF, \n",
    "#                             tissue_list, \n",
    "#                             ijc, \n",
    "#                             sjc, \n",
    "#                             obj, \n",
    "#                             metadata )\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Limiting phenotype data to tissue of interest\n",
      "Warning message in rm(keep):\n",
      "“object 'keep' not found”tissue_obj now reduced to the tissue of interest\n",
      "Limit the phenotypes to those we have samples for\n",
      "tissue_count_obj now has these count specific phenotypes\n",
      "Limit the counts now has these count specific phenotypes\n",
      "metadata_tissue now has these phenotype specific counts\n",
      "dimensions of the ijc_tissue,sjc_tissue, metadata_tissue and pData(tissue_count_obj)\n",
      "The non-overlapping IDs between pheno and count data are:\n",
      "\n",
      "0\n",
      "sex samples:\n",
      "\n",
      "103\n",
      "75\n",
      "sex samples:\n",
      "\n",
      "103\n",
      "75\n",
      "Repeated column names found in count matrix\n",
      "Repeated column names found in count matrix\n",
      "Repeated column names found in count matrix\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in xy.coords(x, y, xlabel, ylabel, log): 'x' and 'y' lengths differ\n",
     "output_type": "error",
     "traceback": [
      "Error in xy.coords(x, y, xlabel, ylabel, log): 'x' and 'y' lengths differ\nTraceback:\n",
      "1. print_exploratory_plots(plot, tissue_of_interest, splice_type, \n .     fromGTF, tissue_list, ijc, sjc, obj, metadata)",
      "2. plotMDS(y[, c(1:dim(ijc.dm)[2])], labels = Gender, top = 100, \n .     col = ifelse(Gender == \"m\", \"blue\", \"red\"), gene.selection = \"common\")   # at line 263-264 of file <text>",
      "3. plotMDS.DGEList(y[, c(1:dim(ijc.dm)[2])], labels = Gender, top = 100, \n .     col = ifelse(Gender == \"m\", \"blue\", \"red\"), gene.selection = \"common\")",
      "4. plotMDS(y, top = top, labels = labels, pch = pch, cex = cex, \n .     dim.plot = dim.plot, ndim = ndim, gene.selection = gene.selection, \n .     xlab = xlab, ylab = ylab, plot = plot, ...)",
      "5. plotMDS.default(y, top = top, labels = labels, pch = pch, cex = cex, \n .     dim.plot = dim.plot, ndim = ndim, gene.selection = gene.selection, \n .     xlab = xlab, ylab = ylab, plot = plot, ...)",
      "6. plotMDS(mds, labels = labels, pch = pch, cex = cex, xlab = xlab, \n .     ylab = ylab, ...)",
      "7. plotMDS.MDS(mds, labels = labels, pch = pch, cex = cex, xlab = xlab, \n .     ylab = ylab, ...)",
      "8. plot(c(left.x, right.x), c(x$y, x$y), type = \"n\", xlab = xlab, \n .     ylab = ylab, ...)",
      "9. plot.default(c(left.x, right.x), c(x$y, x$y), type = \"n\", xlab = xlab, \n .     ylab = ylab, ...)",
      "10. xy.coords(x, y, xlabel, ylabel, log)",
      "11. stop(\"'x' and 'y' lengths differ\")"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzde7wVdb3/8c/asG9y1QQ2FxVwc7iaKEg7kCAvoGCIRBFa6k8jL1Rawgm8\nUahdTsciQbKDWkoqISQihZYiCWoBCoaBJgKaxF3lvtlsWL8/5jhnmMt3vjNrZq1Zs1/PP3ys\nPXvmO5+Ztdjz9vv9zqxMNpsVAAAAFL+SQhcAAACAaBDsAAAAUoJgBwAAkBIEOwAAgJQg2AEA\nAKQEwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg2AEAAKQE\nwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg2AEAAKQEwQ4A\nACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg2AEAAKQEwQ4AACAl\nCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg2AEAAKQEwQ4AACAlCHYA\nAAApQbADgMKbN29eJpPJZDIVFRWFriXlFixYYJzqxo0bF7oWIHoEOwBRevDBBzMWDz30kHOd\nRx55xLrOgw8+mP86ASCV+P8VADFasGDBtddea1u4cOHCghSTZL179/75z38uIunoRtq+fXv7\n9u2PHj26fv36bt26FbocoAFJw18QAIn1/PPPHzhwoEmTJuaSw4cP/+lPfypgSclUXV198803\nF7qKyMydO/fo0aOFrgJoiBiKBRCLDh06iEhtbe1zzz1nXb5kyZL9+/eLSPv27QtTGeL3u9/9\nrtAlAA0UwQ6I1wUXXGDMJBsyZIjtV7NnzzZ+VVpaunPnTmPh3r17f/jDH9bU1Jx00kllZWVt\n2rQZOnTor3/9a9f+D82VH374YWNHn/vc50Rk3rx5Z5111gknnNCxY8fbb7/9yJEjIrJ+/foR\nI0aceOKJTZs2HTJkyD/+8Q9rC6+++uq8efPmzZv36quvah54v379GjVqJCJPP/20dbkxDtuo\nUaM+ffq4bvj2229ff/31Xbp0qaioaN68+TnnnHPffffV19db18lms3PmzBk6dGjr1q1LS0ub\nN2/+mc98Zvr06dYDtx31smXLLrzwQuMABw4c+Pzzz/seQqD3LlBJgwYNqq+vv+mmm1q1atWm\nTRvxvnkipiPdsmXLd7/73R49ejRp0qSysrJnz56TJk0yP4SB3gubSy65JJPJvPzyy8aP3bt3\nz2QyRmek4vA1dxfiSH/zm9/07du3SZMmJ5100vDhw1etWpXJZBT1A0UvCyBO5t0DZWVle/fu\ntf5q1KhRxq+GDRtmLFmzZo1XP9ZnP/vZXbt2WTfXX3nOnDnG8l69es2dO9d2YRs/fvzGjRtP\nPvlk68JWrVp9/PHHZguXXnqpsfzSSy9VH++sWbOMNb/61a/W1NSIyEknnVRfX2+uYPTkffaz\nnx09erSx5qxZs8zfzp8/3/W20PPOO+/QoUPmaldccYXrgV9yySXHjh1zHvVzzz1XVlZmXbNR\no0bPP/98hO+dTkmPP/64sfDTn/70vffea1aSzWaffPJJ48fy8nLrjuI40iVLlrRo0cLZZps2\nbf7+978HfS9shg8f7tzkpptuUh++5u6CHuktt9xia628vPz222+37RpIE4IdEK+PPvrIvPzM\nnTvXXH7o0CFz5tnjjz+ezWY//PBDM6h16tTpl7/85YIFC773ve+Zs+mHDx9ubh5oZTM0tG3b\ntkOHDiNGjPjmN79pXtrLy8svvvjidu3a3XTTTZ/5zGfMS+DPf/5zs4UQwe7LX/7ylClTjNdL\nliwxfvvaa68ZS6ZMmTJy5EjjtRnsNm7cWFlZaSycOHHi22+/vXLlykGDBhlLJk+ebKz2zDPP\nGEtKSkpmzpy5du3ahx56yDxw8yRbj7pjx469e/eePHmyte+tX79+Ub13QUvq2LHjKaecUlpa\n2rt3765du2Y9gl0cR7p169aTTjrJWD5o0KB58+b99re/7d27t7Gka9euR44c0X8vnP7xj39Y\n+2gfe+yxZcuWbdy4UX34mrsLdKR/+9vfzOXnnXfe73//+4ULF1500UXmCSTYIZUIdkDszFT0\nta99zVxoXvyaNWt24MCBbDb7gx/8wFjSvHnzLVu2mGvOnj3bvD6tWrXKWBhoZfNyKCJf+tKX\njIXz5883F1ZUVLz77rvZbPbw4cNdu3Y1Fn7hC19wHoJ+sPviF7+4YsUK47XRYZPNZs2ot2LF\nCrNNM9h985vfNJYMHjzYbHDnzp1NmzY1TpTRczNjxozhw4cPHz7cbDabzY4YMcLY9sorr3Qe\n9cCBA81eH7MPrKSkpK6uTn04mu9diJK6dOnyr3/9y1zZNdjFcaSTJ082Fp588slG8dlsdseO\nHWaumjdvnv574Wrr1q1mPevXr3ceo/PwNXcX6EjHjRtnLGzVqtX+/fuNhUeOHDHv0iXYIZUI\ndkDszPGjT33qU+ag5NVXX20svOqqq4wlZ555prHk6quvtm5eX19/4oknGr+aOnVqiJWtl8MV\nK1YYC+vq6srLy42FY8eONVuYMGGCsbB3794hDtYa7I4dO2Z0K3bs2NH47VlnnSUi7du3P3bs\nmDPYnX766caS22677ZCFMaFKRBSDp9/61reMdYYMGeI8arPLMJvNWmdibd68WX04mu9diJKM\nrj6T11Bs5Efaq1cvY8nXv/51a7Mvv/zy4sWLFy9e/M4772Rzey90gp3t8DV3F+hIu3fvbiy5\n5pprrPu66667jOUEO6QSN08AsRsxYoTR8bB79+5XXnlFRI4ePbpo0SLjt1/96ldFJJvNmvcr\nnHHGGdbNGzVqZPYxrF+/PujKNuZ1vbS0tFWrVsbrT3/60+YKbdu2NV7s27cvxMFaZTIZI71t\n3rz573//+wcffLB69WoRGTlypHMCezab3bRpk/H6nnvuqbR46aWXjOXmUT///PMjR448/fTT\nKyoqjNn006dPN37lepdJ3759zdennXaa+do4xr1793Z0OHz4sOi9d+FKOu+88/zPYNRHms1m\nzU+FmaUM/fv3v+iiiy666KLq6upA70U41sMPtzv1kYrI5s2bjRddunSxbtizZ89cKgcSjufY\nAbGrrKy89NJLH3vsMRF55plnBg4cuGzZsl27dolIu3btjCvcwYMHzbv/jCRhZc7oMi5agVa2\nKi8vN0fcjB+NFy1btjQX2iak52jkyJEzZ84UkT//+c8nnHCCudC55sGDB48dO6Zu7cMPPxSR\nX/7ylzfeeKOxpEmTJp07dy4rK9uyZYtxSp3Ky8ubNWtm/uicoX/s2LH33nvPtjCbzYreexei\npEaNGpmpWiHyIz148KAZB82eXSf99yIc2+GH2J3vkWaz2UOHDhmvzQ+e649AytBjB+TD2LFj\njRfGwz4WLFhg/Hj55ZeXlJSIyAknnGDO6d67d69tczOiGXc8BFq5sAYPHmykxiVLlrzwwgsi\n0rJlS3NSvNUJJ5xgPB5FRKZNm+Y6xPD9739///795mDx5ZdfvnPnznXr1q1Zs+bLX/5yTIfg\n+96FKKlx48bGtgpxHGllZaW53wMHDnitpvlehC7Ddvhx7M764BjbkTr/yQBpQrAD8mHIkCGf\n+tSnROTtt9/+5z//aYYDcywvk8mYg6Rr1qyxbnvkyJF169YZr40x00ArF1ZpaemwYcNEZNmy\nZS+++KKIDB8+vLS01LlmJpMxBwfNgTmn1atXHzx40Hg9ceJEswPy7bffDl1ky5YtnUnCjAW+\n710cJcXUbElJiTkuaRupf/zxx+++++6777772Wef1XwvohLT7k499VTjxVtvvWVdbswHANKK\nYAfkQ2lpqfnYtqlTpxoDf2eccYZ5D4SImI9GW7BgwQcffGAu/81vfmN0wmUyGXMQM9DKufvb\n3/62YMGCBQsWWB8hockoY9++fcZomqKqoUOHGi+efPJJM9McPXr0iiuuuOaaayZPnrxlyxZj\n6pvBeLSyiKxbt85IjdaFUfF972IqKaZmjZwtIgsWLNizZ4/x+qOPPvrmN795xx133HHHHcbH\nSee98NqFdQKl16ixTS678/LZz37WeLFw4cKPPvrIeL1//37rneNACgW+3QJAKEuXLrX96/vJ\nT35iXeGjjz4yHt4rItXV1ffff//8+fMnTpxozoSz3sYYaGWvOy7NbpJf/vKX5kJzbv7pp59u\nLgz3uBNjyb59+8yqKioq9u3bZ2vTvCv23XffNfulBgwY8Ic//OG5554zI2zPnj3r6+u3bNli\njuJ94QtfWLt27cKFC9u3b28+paV58+avvvrq9u3bvY76X//6l/kWrF271u99y2b93rvcS3J9\nj2I60vfee8+cnfaZz3zmd7/73WOPPWZ+C8ipp55qPBlE573wOl319fVmp+zAgQPnzp373HPP\nuR6jSXN3gY7U+q716dNnzpw5jzzyyDnnnGNOSy0pKdF594HiQrAD8sR89od5UbE+x8ug+DKJ\nUaNG2Z4cpr9yYYNdNps1e4msj012BjujVDMFWrVv3958cIb5vA9Tu3btNm3a1K5dO3PJlClT\nIgx2vu9djiVlPd6jmI500aJFrjcQVFVVrVmzJtB74eXiiy+2bmK87+pHuujsLuiR/r//9/9s\nrTVp0mTGjBnmj0ePHlUfCFB0GIoF8iSTyYwZM8b8cdCgQWaXm+nMM89ct27d3Xfffc4557Ro\n0aK0tLRt27aXXXbZwoULnV+4FGjlwrrsssuMF76jw6NHj169evU111zTqVOn8vLyE044oVev\nXrfddtvatWvNx7jce++9P/nJT3r06FFZWdm+ffuvf/3rK1eu7Nix4yOPPNK1a9fGjRt36NDB\nfIZZJHzfu5hKiqnZ4cOHv/nmmzfccEOXLl2Mp4r07Nlz8uTJa9eutc4N0HkvvMyaNWvkyJEt\nW7asqKjo1KlT//79favKZXdeHnzwwXvvvbdr165lZWWtW7cePXr03/72t89//vPmCoo7SIAi\nlclms4WuAQAAABGgxw4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAA\ngJQg2AEAAKQEwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg\n2AEAAKQEwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg2AEA\nAKQEwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBIEOwAAgJQg2AEAAKQE\nwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAAkBKNC11AcXjjjTfq6+sLXQUA\nAEiExo0bn3nmmYWuwgXBzt+qVavOOeecQlcBAAASZOXKlX379i10FXYEO391dXUicvjw4bKy\nskLXAgAACqyurq68vNyIB0nDHDsAAICUINgBAACkBMEOAAAgJQh2AAAAKUGwAwAASAmCHQAA\nQEoQ7AAAAFKCYAcAAJASBDsAAICUINgBAACkBMEOAAAgJQh2AAAAKUGwAwAASAmCHQAAQEoQ\n7AAAAFKCYAcAAJASBDsAAICUINgBAACkBMEOAAAgJQh2AAAAKUGwAxBMJpPJZDKFrgIA4IJg\nBwAAkBKNC10AgCKTzWYLXQIAwB09dgAAAClBsAMAAEgJgh0AAEBKEOwAAABSgmAHAACQEgQ7\nAACAlCDYAQAApATBDgAAICUIdgAAAClBsAMAAEgJgh0A5Ekmkyl0CQBSjmAHAPlAqgOQBwQ7\nAMiHbDZb6BIApB/BDgDyhGwHIG4EOwAAgJQg2AEAAKQEwQ6IC5PlAQB5RrADYkGqAwDkH8EO\niAXT5AEA+UewA+JCtgMA5BnBDgAAICUIdgAAAClBsAMAAEgJgh0AAEBKEOwAAABSgmAHAACQ\nEgQ7AACAlCDYAQAApATBDgAAICUIdgAAAClBsAMAAEgJgh0AAEBKEOwAAABSgmAHAACQEgQ7\nAACAlCDYAQAApATBDgAAICUIdgAAAClBsAMAAEgJgh0AAEBKEOwAAABSgmAHAACQEgQ7AACA\nlCDYAYheJpMpdAkA0BAR7ADEgmwHAPlHsAMQi2w2W+gSAKDBIdgBiB6pDgAKgmAHAACQEgQ7\nAACAlCDYAQAApATBDgAAICUIdgAAAClBsAMAAEgJgh1QHDKZDI/8BQCoEewAAABSonGhCwCg\nhUf+AgB80WMHAACQEgQ7AACAlCDYAQAApATBDgAAICUIdgAAAClBsAMAAEgJgh2AYHhUMgAk\nFsEOAAAgJQh2QMOSe2dbNpvlackAkEwEOyAR8jO+aeyCgVQASCuCHZKL/BE5o6eN/rYGiH9N\nQANBsEOiNZyrUd7GN0l1aqn8yNFTCzQcjQtdAKBCCkH+ZTKZlH3wUnY4ABTosUNycTVCQfDB\nA1C8CHYA8H9IdQCKGsEOAKLEA5wBFBDBDgAAICW4eQIAosRgLoACoscOAAAgJQh2QLFqgHO5\nGuAhA0AgBDsAAICUYI4dUKwa4FyuBnjIABAIPXYAAAApQbADAABICYIdAABAShDsAAAAUoJg\nBwAAkBIEOwAAgJQg2AEAAKQEwQ4AACAlCHYAAAApQbADAABICYIdAABAShDsAAAAUoJgBwAA\nkBIEOyBBMplMJpMpdBUAgGJFsAMQGYIpABRW40IXAOD/ZLPZQpcAAChiBDsAkSGYAkBhMRQL\nAACQEgQ7AACAlCDYAQAApATBDoged4YCAAqCYAdEzEh1ZDsAQP4R7ICIGXeGRnh/KA+HQ8PE\nJx8IgWAHRC/yp37wGBEAgA6eYwckHakODROffCAEeuyA8BgqAgAkCsEOAAAgJRiKBcJjqAgA\nkCj02AFA2jBJAGiwCHYAAAApwVAsAKQNkwSABoseOyC5GFADAARCsAMAAEgJhmKB5GJADQAQ\nCD12AHwwIgwAxYJgBwAAkBIMxQLwwYgwABQLeuwAAABSgmAHAACQEgQ7IN8Sey9CYgsDAGgi\n2AEAAKQEN08A+ZbYexESWxgAQBM9dkDxYcAUAOCKYAcknW3qG6kOAOCFYAcUmWw2W6gxUzIl\nACQcwQ5IugImOSsj1ZHtACDJCHZoQAI9zoNnf9gY4TIJERMA4IVgh4aFXJILzh408T9FQKEQ\n7JBazi63QLkk3AAo1zOAfwVAARHskGYF6WEKcVVLzbBvag4EuUjIrFCgYSLYIbUKdWnhkgYA\nKBS+eQKIUrhUl5osmJoDAYAiRY8dkqtYxvWKpU4AQOoR7IAAyHAAgCRjKBbJFdO4XiaTUbds\nRDf9vTP+CABICIIdGiKvbOfbG0eGAwAkGcEODZE6n5HeAABFijl2KA4RzmxT5DaevwUAKGoE\nOySC+qYEna+f57YGAAAIdigCfP08AAA6mGOHRPANbbmvUBBBb7BNK987kQEAkaDHDkC8dEbS\nAQCRoMcOOE60fWx0UwknAQDyiB47AACAlKDHDjgO3UsAgOJFjx0QPZ69kmeccAAwFHePXV1d\n3RtvvLF///6OHTt26tSp0OUARY/beAGgqBVNj93dd9/94osvWpf86le/qqqq6tev33nnnde5\nc+e+ffuuWbOmUOUBVnyDRZ5xwgHAUDTB7o477njuuefMH//whz9cf/31Bw8evOyyy6677roB\nAwa89tprgwcPfvfddwtYJFDsSEhRYWgYQEEU61Dsd77znRYtWrz66qvdu3c3lvz+978fPXr0\nPffc8/DDDxe2NsSHgUIUBfPRfZF8VvnYA9BXND12Vjt37nznnXfGjx9vpjoRGTVq1KWXXvqn\nP/2pgIUhJkyNR3HhS/AAFEpRBrva2loRsaY6Q69evXbs2FGIipAnCbxSEjrhKsLPKuPjAPQV\nZbBr165dixYtPvjgA9vyf//7382aNStISYiJEZvMC1sCv5yKKy4AIDmKKdi9//77q1at2rBh\nw0cffXTjjTc+9NBDBw8eNH/71ltv/e53vxswYEABK0Tcsp+IbxcheuDIdgCAhCimmyeeeOKJ\nJ554wrpk8eLFX/ziF0Xk8ccf/8Y3vnHo0KE77rijQNUhFgXMTMxYLyBOPgCEUzTB7te//vXH\nFnv27Pn4449PPPFE47cff/xxy5Yt58yZc8455xS2ThS7NIUJ4hEANDTR3I1fcPv37z/hhBNK\nSmIZWX7llVcGDBhw+PDhsrKyONpHcYnqGRZ5QLALROd0FdG7DyA+dXV15eXlL7/8cv/+/Qtd\ni13R9NipNW3atNAloAEplqt7URRZdIrl3QfQMKUk2AH5ZL2uh+sVM+/PKHhEIKaYNM8DpwtA\nkqUn2L377rvXXXediDz//PP6W73//vsXXnhhfX29Yp0DBw6IyNGjR3OsEOkQyXU9m80mJ1El\np5Lk40QBSLj0BLt9+/a98MILQbdq167dj3/8Y3Ww+/Of/zxr1iyCHVyFvtInJyIkpxIAQI7S\nE+y6deu2du3aoFs1btz4sssuU6/z4Ycfzpo1K2xdKG6pv/8gaYdG9yEA5CI9wa6ioqJXr16F\nrgKJRmgIJ2/nzfxmEd4mAAinmL55wtfu3bs3bNhQ6CqQaCG+VSKOkFFE3zCbzzrz8M0iAJBu\nqQp2P/3pT7t06VLoKpBc+qGhiIJX3IhZAFBE0jMUCxSR4kpLxVUtADRkBDvABVEGAFCMiibY\n9e3b13edLVu25KESIP+4nwAAoKNogt3q1atFpLS0VLGO+nF0QJHiXlEAgKaiuXli4sSJTZo0\nefPNN2u9TZgwodBlIkYN9m4G7hUFAGgqmmB31113VVdXjx079siRI4WuBQXTYLOdQvLPCbcY\nA0DeFE2wKy0tfeyxx/7xj3/ceuutha4FBUOvlY05SlvoQpASpHCg2BXNHDsR6d69+7Zt2xQT\n6S6++OKWLVvmsyTkUxJSXdImumWz2aSV5ORank7Zqf8+NwCIXDEFOxFp3ry54reDBg0aNGhQ\n3oqBVUO4BifzJoZEFaOJPqHEKsaPEwCrIgt2gIJOuMwlgHLNi4rR0aizWh6KAYA00Q12mzdv\n3rBhw65du/bs2dOiRYuTTz65urq6Y8eOcdaGYlKk12Df7rdAQTBpnXlJxokCgDj4BLuNGzdO\nmzZt8eLFGzZscP62urp62LBhN998c6dOneIpDwhAJytY17ENrZp9SDrtOAOfThdUQxiwBgAU\nkGew27lz56RJkx599NH6+vrWrVuPGTOmS5curVu3btmy5ccff7xjx4533nlnyZIl991338yZ\nM6+88sof//jHrVq1ymfpgCavjjTbnQeu44P6fXWRrBM3kiUApJt7sFu6dOmYMWN27979la98\n5bvf/e5ZZ53lek3KZrOrV6/+2c9+9sgjjyxatGju3Lncu4BkUmQ7xY8KOq3pbwgAQCTcn2M3\nZMiQbt26vf3227/97W/PPvtsr56GTCZz9tln//a3v33rrbe6det24YUXxlkqUiiSh2b5NpKH\nr23Q2UUSvj0iCTUUo8wnCl0IAPhwD3aTJk1asmTJ6aefrtlKdXX1kiVLJk2aFF1hQJS4KiMX\nRhomEwNIPveh2KlTpwZtqFGjRiG2QgMXyZUy7sst89IgfAAAFAmeY4dYWMNQoYKRdb9clQEA\nDQHBDgkS6IEjeZOoYgAAUCDYIRbWMBToVtMIn/Gr3w6DrQCAdHC/eQIolESlK/17IbkzAwCQ\nBAQ7wPMhIJrz88xvsIijNgAA9BHs0HBF9T301vxHvAMAFJD/HLtsNjtv3rxHH330gw8+OHLk\niHOFN998M4bCgJCss/QUk+dsXxSbI1IdACAJ/IPdvffeO3HiRBE54YQTSktL4y8JCE8/rumE\nv6ASNUEQANAA+Qe7X/ziF0OHDp05c2bnzp3zUBDgyxbFrA9Jsd1XG216K+zNs+ZxJfke3iTX\nBgANgX+w2759+7x580h1SBTb41Rcw1yIphLL1hOpqJloBQANmX+wa9OmDRcJJIrzA6k/8Brt\nfr1Enq6S/G8wxDd8RPi0wiQjZAPIP/+7YseOHTt79uw8lALAl+/9GV6PbkkabjQBgDj499jd\neeedo0ePvuKKK6688spTTz3Vef9EdXV1PLUB0ctPJ0p87RvjzjE1Hk64gy2K9JmjhnCMAJLG\nP9g1a9bMePH444+7rsAfLyRBtKN7SR5ES2ZVgVgPIcmnGgCKjn+wGzt2bFlZWePGfKsskivE\nU06QHLwpABAV/7jm1VEH5JM6tHn9KnQ3XgGjRgO5scDEHb4AEKEA/XC7du165513Dhw40KxZ\ns65du7Zs2TK+spAaEcYUW1OuV33rwqTNRdMXebYr3oRUjDUDQAFpfVfs8uXLa2pqWrVq1b9/\n/wsvvLCmpuakk0664IIL+DIx6IgqYEX4gLo8Z76guyPNGIr3PGQ+UehCADQ4/j12K1asuOCC\nC+rr688999yuXbtWVlYeOHBg3bp1S5YsGTBgwIoVK7p27ZqHQlG8YnqGnPP+UOd1NA/fEusr\nCamueBNSJIq3wxIAgvIPdnfffXerVq3+/Oc/d+vWzbp89erVF1100Q9+8AMm4UEh9NU0vjsh\nbN9UkSPf0JDAB5QgbsU+GQBA8fIfin3llVduvPFGW6oTkbPOOuvGG29csmRJPIUBMfZ15RI3\nQ1ytw+2OsbyoFOqhzfQRAsg//x67PXv2dOjQwfVXHTt2/PDDD6MuCQg/dhbrpdQas8wKi/3i\nzTBlUWho90oDCM0/2LVu3Xr9+vWuv1q3bl3r1q2jLgkogAQ+Ay/Q7vKQz8xcS8LIszzPCgVQ\n1PyHYocMGTJ9+vSnn37a+jclm80+9dRT999//8UXXxxneWigrD1hgUYkQ6wcenQ1HVdZ/QMx\nVkvHURcXzjwAff49dlOmTPnjH/84cuTIqqqqHj16NGnSxLgrdtu2bW3btp0yZUoeqgQUnJ1V\nmn1LmndRFEVPVX5qS/IZSDfOPABN/j12HTt2XLVq1VVXXXXo0KElS5Y888wzSy3RRGoAACAA\nSURBVJYsqaur+/rXv/7aa695Tb9DAxH3BP+gI4zZT2huZd69WPDbFIIWUPCCAQAJpPXNE6ec\ncspvfvObbDa7bdu2AwcONG3atKqqKu7KACfXnOca4DSzoGuPnXOhrTsw8u6ToJOoSHUAAFfu\nwW7btm3l5eUnnnii8dpcnslkmjZtaltIyGvI4h4hiqN9Mz8pxlitGcs5yBt5tovqSzWYXw8A\nDZx7sGvbtu3QoUOfffZZ47W6CS4kiJW1l07nC+PVo7fWvq4Qjw6O5NMe0x2sdOMBANyD3Zgx\nY3r37m2+zmM9SKJ89gPF/dgOnWHcIsVXXAAA3IPdnDlzXF8jfTSDVKBsF204C1pb7vuN+wFy\neYutAICGxv+u2OXLl3t9vcSKFSvmz58fdUlIorwlBvVj1Zy3r1onvRFrAAANnH+wGzhw4Esv\nveT6q2XLlo0bNy7qkpBXOnkoxNT+HDNWQaa+xcTrWSpxPGMlCc9tAQAUkOfjTjZs2LBhwwbj\n9erVqysqKmwrHDp0aO7cuYcPH46xOqSa710O1ntXvW6eiGNimdcTj5McHwFf3DQNNASewW7e\nvHmTJ082Xk+dOtVrtdGjR0dfFBDwocSR71o/LIbu77R+Z5pmU5rFEEPhRFcu0EB4BrtJkyZd\nddVVK1euvPTSS7/2ta/16NHDtkKjRo06d+48YsSImCtEaimSh+1mCHOEMZ9T/Ww/GjVEVYDt\nOXkFvO+YXpwGgpumgQZC9c0Tbdu2HTFixPDhw2+88caamhrnCgcOHNi9ezcPKEZ+uOaPJPdO\nWUeT1U85Luzjjgue7ZL8JqYJZxhoCPxvnli0aJFrqhORp59++uyzz466JMAuCXe8Bq3BTG/m\nC2tTUrirrPNACn5uAQBR0fqu2F27ds2ZM2fz5s319fXmwtra2kWLFu3fvz+22hCjPPSR5Kcf\nKM+hRP+g1NPddBpJ5QlMbA3FqOBdrQASyD/Ybd68uV+/fjt37nTZuHHjO+64I4aqUPS8ZvM4\nL0Vetw64RqJYv4xL50nI0e43IftCMYrpa4sBFDv/YHf77bfX1tbOmDGje/fu559//oMPPtih\nQ4elS5fOnj37oYceGjp0aB6qROTivhi4ztR2vRTp3zpgbTBcwssxF8bx3bJeX3HL1VpTg52f\n1wAPGYAO/2C3bNmy8ePHjx8/vra2VkR69uxZU1MzdOjQMWPGnH/++QsXLhwwYED8daL46DyF\nzlzTeXnW7NjLhbnT+B6tEih2KB7pF2LXiAQnH0Bx8b95YuvWrZ07dxaRkpISEamrqzOW9+7d\ne/z48VOmTIm1PsCQ/USINGZrxHgdyVin8/vNnM2q6zRLIjqEk4QbawAgOfyDXbNmzbZv3y4i\nZWVlTZs23bhxo/mrHj16rFq1KsbqUPxsWcorTgW6POeeySLp/9MpI5LMEeKGXEVtfO1YIARH\nAMVF67tiH3jggaVLl4rIGWeccf/995t3wi5ZsqS8vDzW+lAo8V3+c282xwttVMfleq+rojbn\nKY3pJBNE8oaUDCBp/IPdrbfeunv37gkTJojIuHHjVq1a1aNHj1GjRp111lmzZs268MIL4y8S\naRBfnArRgnX0U/PaHGKYtYCylq/rcP4qsWUDAHLkf/NEv379li9fvmLFChG5+uqr33nnnWnT\npj311FOZTGbEiBHTpk2Lv0gUgP7tnOGaDfSEkRwHTF03DzdFL3QZrpsTsCJX2K+eA4CC03pA\ncZ8+ffr06SMimUzmhz/84Z133rlt27Y2bdpUVlbGXB7SI45LoH46tK4T6NpfdANt5sGSOQCg\nAXIPdtu2bSsvLz/xxBON184VKioq9uzZs2fPHhHhu2IbpnC5Ieg9rZEMvIZYoUgfcmH9+rKi\nKz4SDfOoAcDkHuzatm07dOjQZ5991nitboK/pClTRLHAtR/Ot/5AhxbiwXUFPHXmrEHFI/Fy\nL6+IPiEA0NC4B7sxY8b07t3bfJ3HepBOvh11ZuBQhwbrwGiswSKX/siCf8uT+uwVvDwAQHzc\ng92cOXNcX6MhyNtV3xbm1IHDd66buW1hO8wKG5vUsdhZXo53wOQNfYQAoMn/cSeDBg2aNWvW\nxx9/nIdqkEpGmLDdD2v+1xo1vK7f1ucJm7z2VdjbHSIMH74HEuJgyUYAkG7+wW7ZsmXf+MY3\nqqqqvvjFLz711FPmV4oBOpyP5DVeOPuNvH4US57LsZKgMci2SbjUmPlE0K3EO9tF9Yy6Ynmm\nXbHUmUoF/58lAIH4B7v333//5z//+dlnn/3UU0+NGjWqqqrquuuuW7ZsGX9ni1SEf6bVkcW1\n+y3otzXo7MU28U7/ton8XK6cDx8x96s4NJ1HlhB3AAA2/sGuQ4cON9988yuvvGIkvG7dus2a\nNetzn/tcp06dbrvttvXr1+ehSsQqdNRThA9bh5Mte3ndS+EMXq49fF7UETBEqrNVGDpIOVOd\nc+/OytUJlUiH/ODDBhQX/2BnsiW800477b/+67969OgRX3GIg/rPtHXqm7lEkYGs0+Ns6U2U\n0cTWvvXWB2faCxRxdO6ojVagwVbrsHIkQ8wAAJgCBDtTkyZNPvWpT3Xo0KF58+aRF4T8M7OF\nel6XbyNeS2zZxbV9111bCwvRwebbvg5z13HPNIo14TFNCgAaCK2vFDPs2LFjwYIF8+fPX7Jk\nSX19fYsWLUaNGjV27Nj4ikOeZd2e7qufNjTXdHawuc7GC8119NZ1HDaqPdLlBgBICP9gt2XL\nlqeeemr+/PnLli07evRoZWXlyJEjL7/88mHDhpWXl+ehRERLnWkizCjmjgLNk4upqkzYx8tZ\n+x1j2kU4ivfRWQnRM2mi/V+L5O8XQN74B7tTTjklm802btz4wgsvHDt27GWXXdasWbM8VIZ0\ncKY619hhm2mXC2uUtNZgm8lnWyGSXec523nVEEclBILIcTIBxME/2A0YMGDs2LFf+tKXWrVq\nlYeCEDf1LLQILzbWeXu2vbju2nW5tRHnRD11v6O5uSJTSqRJKJ+XavW+CA0JV6g3iA8GkHr+\nwW7ZsmXGi3379r3//vvt27dv2bJlzFUhWSLJfJlPvn/C2Tln9rFlHN9CoVmb2Y7rTtUNxj3T\nLgndeLlLwSHEh+5MAMmhdVfsX/7yl759+zZv3rxXr15//etfjYUjRox44YUX4qwNRUDxpI9A\nd2I6H49i66vzegZKUXRchb4jNcTdrDw/BQAaMv9gt2LFiiFDhvzzn/8cOnSouXDnzp0rV64c\nNmzYa6+9Fmd5yB9FIFD/SpT5KWshjr4NW3BxzTG5PKfDtbVwDebyxJAQScu5Ox5ZkliEaQDJ\n4R/spk6dWlVVtW7dut/85jfmwlatWr3xxhtVVVV33XVXjNWhGGjGwXCdT+a2Xu0o5uep64mP\nrcJcdkpiQD7xPw9ACvgHu7/+9a833HBDhw4dbMtbt259/fXXv/TSS/EUhnSyJRXrj9aOPYM6\nzykine1WXOt4sWZUcrav2FCzmEAUo88KXJUBoIHzv3liz549p5xyiuuv2rZtu3///qhLQqJp\n9kU57xhQ342rbi3reBieebOFODr2jJWdN2fkreurUH1syU913GSQcLw1QAr4B7uqqqr169e7\n/uqll15q165d1CWhAEJfcRUb5pilnOHMujvrj+LINLb9ZiyPTdG8dzWSzKr+la+g5zbreIBf\nJJKcxpJcGwAUhP9Q7LBhw2bOnPn6669bF3700Ue33Xbbr3/96+HDh8dWGyIT4dQZ/VlfOuOG\nmU+egeJ6O0WON4RaR2DVbervyNaaPmslOb4XihpyjzjWkxY5pgwCQNz8e+x+8IMfLF68+DOf\n+cynP/1pEZk8efLkyZPXr19/+PDhU0899c4774y/SMQu9OVWfbesjU7PmbmmayP63VfWoVuv\n0V7XddTyOaob6NzGt+skRzHN/lcAaDj8e+yqqqpWrVo1bty49957T0TWrFmzZs2aZs2a3XDD\nDStXrmzTpk38RSJXkfeUePWxuXLtBMoc/7xia8Zy3kURiPPmCXO5Yiv93YUrzDwi61EHbSSX\nbj/NTZxnPvlT92xC96qmUqxdsAASyL/HTkRat249c+bM+++/f8eOHfv27WvWrBl5DkHZesUi\nCZo6k9s085xXr56NbTX9niGvbsUEdixlgnz/RwJnueX5XpmE42wADY1WsDNkMpk2bdoQ6Rog\n28Xb9qP+oKqZ7VxH+qy/yiUuqO97sK6jOdRrW8G4TFon7YXIdjkeWojN9YsUt6/ZLZRwoaTg\nZScKZwNoUNyDXU1Njeb2dXV1tvsq0EDE1A3gO6or3je9unKdP6cTIjXn+UV1K0lQMXWVBW0w\nD6EhHR1O6TgKAMnnHuxWrVpl/bGkpOTIkSPGa+ufpxYtWjRv3jzW+pAEzihj69exMTOHon/O\ndRPX1dQJRn/ykHVMKverbHIu0vENhiZkmLXgBUSFbAcgD9xvnqi32LlzZ01Nzfjx49esWXPo\n0KFjx47t3bt3+fLlX/nKV/r06bN27do8V4w42Gbl+6alHO9vCNSsV1+a+HW2ubbvbNPrQOI4\nQE36Nzrkc3eRbBVUmpJQmo4FQGL53xU7YcKEtm3bzpgx48wzz6yoqBCRZs2aDRgw4Iknnqis\nrLzlllviLxJJEfQOO+tYp3pDa4pyrmkLZDqz+mw3asQxEmqLwrkEnRC3uIol18aUGNRvSgEV\n452epDoA+eEf7J555pmhQ4e6/mrw4MELFy6MuiQUgFc40Lx26l9lnWtal/g2Yt64oMg04a6g\noVOCNWtGtWvfrJbjnQ22dyFcNCxgj2ZCbuwAgATyvyt27969O3fudP3V7t279+7dG3VJKABF\nTso47uUMdH+D9YUz0il+VLev5nvJdx3DDTQFyjlx0Lcf0WsF/QmFXlvlQdJSVO71MOMNQCr5\n99j16NFj+vTpK1eutC1fsWLFww8/3K1bt3gKQyEpusSsuUoxIuZc6Nqg87Xm9D6dNW3H4ktx\nf4bzqHUatG4SaP24FbCzLVGS9r4AQO78e+ymTp06cuTIfv36VVdXd+rUqaKiora2dtOmTRs2\nbMhkMjNmzMhDlYib1zis60CnkWycXXEZy5dJ6Pc5WWdxqetR1KnYxHUdazS0FuB1F0WOCUC/\nBcVRmC34Fhw5/X7E4pK+I5L0vlkANPkHu0suuWTp0qX33HPP0qVLN2zYYCwsKysbPHjwpEmT\nvKbfoSg4s4L5o850N7GEM99YZosjzlBia0SdWvQ77QJ176kbcb724ry4KroDc5mZV9TjiYUt\nvnjPGwAoaH3zxLnnnrt48eJjx45t3br14MGDlZWVVVVVjRsH+NYKFJ2glz39uGP90ZntXNdU\nLPTdtTW8es3/U5QUq0Bx0+tc5UEc+2IYNCbOPnUADUqAcFZSUtK+ffv4SkFMFEMzkcxAd23H\n1oHnTFdyfF+a+dq1C9CL87YDZzHO3kH1NS+fV8TcB3mLF7EDAOJAr1vKhc4Nzvln5o+hBzez\nyq+ssP3KGftcB3BtWVA9Yy97/He8iqP3LtqYpZMRY803DWG6VUM4xqA4G0BD5n9XLIpa9hPG\nj4r7WDVZO8ZsjZvti/dwp+sNE74dV2bms+7Ctty5L2fNZtleO1L/VvySn/PwbQUDABAreuwa\nFt9bVm0dZtYNneuI32Q457Q2caQ69V7MjjTXHfneXSHe4U8xjOs6OmxtSme/tm7CqDqWArVT\npD03DeEYASAm9Ng1OOpUp9hQM52YaSxQ15drV5miHs0+MOdeXFez9WjaNrfFPmtQ82rH1hTh\nI6j4zhgdqDrUZ4lzCCQZwa7BUfxR1pln5uzJcx3ttbUT1WXAd6jUub7ra8Wa1sNxJlTXfj7R\nuNQFrTzydpwVJvbyTA4GgNAIdvg/zgu/9Udb7HMNcOoGncuNmGJtyvYr666tldi2Eu+YYg1n\nipq9OiNdQ4YiXVn7NSMJKLHGrzgiVDLDoimq90Ut4SfBV45TUQEUkPscu5qaGs3t6+rqXn/9\n9ejqQex8O66MJGGLceo5dhm3W1m9durVk+cbEJ2T3mwVWitXHLLrAGsI+t2BuXM9LsXK4j1L\nMtrc6VVAQ77wF3uqA1DU3IPdqlWrrD+WlJQcOXLEeG39k92iRYvmzZvHWh/yxvdqpFghq3yO\niRwfNbIej7UTjXgUaGVFnTq7tsUjRRC0HZ25MJdwY41f1l5Srxp8mwq6VaDGvUJkQ2OejYaQ\n7Rp4fAcSy30ott5i586dNTU148ePX7NmzaFDh44dO7Z3797ly5d/5Stf6dOnz9q1a/NcMXyF\nGLyzRgfrDDOTZiecrRPLNQMF6qWzNWXbqXoQ1pVt8lzWcSeETmuae/Qa8PVt1raa11vgJdZx\nNN9jbyCxRi2BiSeOMX3eaCCB/OfYTZgwoW3btjNmzDjzzDMrKipEpFmzZgMGDHjiiScqKytv\nueWW+ItEvDTThniPqIpjSpzztTpq2PZlTWzqeOTszfK92GhedG0BSJGHNJf7nmf15tYwqtOg\ndcP4Bl6dLTfki32sA9wJ1KAOFigW/sHumWeeGTp0qOuvBg8evHDhwqhLQq7UV5eMhbm++F3+\nvX7lvKjbOtJce6Gc+1LnFdes4NqI+V9nKPQKHOb66iO17cKZcZ1J13Vb61a2MGoNwb5n23Wn\neeBam1f6L7g4uqmKVOShM1FvNACTf7Dbu3fvzp07XX+1e/fuvXv3Rl0SCkBxqXYNT+avbJu4\nvrZuZUtdzlDlVaFtHWt2tK3jeiH3/ZXthU4itGZHr7hjWxK0z0/Btx+0gJJZVcKRQQFEwj/Y\n9ejRY/r06StXrrQtX7FixcMPP9ytW7d4CkNcshbGEsUVRWf00HeM0tnP5NqULSzaXmQd9zo4\ns6P1uBSdeZqcfX7qlQMtD7eaTYjDCbEXzUqSGeYSW5iJPAcgWv5fKTZ16tSRI0f269evurq6\nU6dOFRUVtbW1mzZt2rBhQyaTmTFjRh6qRI5cO9hCs3a/+bas3qntkpZx+7Iy20LXxGZr0BoK\nfdOnoiRr/6Kt09G6jnOhTh9khIEj2vcXBcHblxrOPwhAPvkHu0suuWTp0qX33HPP0qVLN2zY\nYCwsKysbPHjwpEmTvKbfoYj4Tufy2sprvDLrePqJ18isrR1rv5pX4xnLM/OCdnXY9uiMaIof\nzc1ds50cn65c056C75q5z2DLej/nz2t3XJzygJOsULyfQ7IdCsg/2InIueeeu3jx4mPHjm3d\nuvXgwYOVlZVVVVWNG2ttiySI+09M7iOJOrHDmhetu3btwNPpqHP2//kWbN1dHtKY166tu/A9\nxkgKAKCJf24oLP9wtnDhwtNPP71nz54lJSXt27fPQ02Ile1Krwgfof+n0zXnOTvYrLv2+pV1\nc9c1XSt03g9hm2Nny4iKsV05/lyFPiHWDZ0jzvrDxEHfJmeHqO8m+vUAseJzCITgf/PEmDFj\nFi1alIdSkE+KPjbrbG6vvOWc8Z39hPmjWPrhrOsomvWt2TbgG/TvvnP+nHoXvkUq1redENsp\njepuCZ2T4DX/z0vQk1BwRXT/Qe6lFtHBAigI/2B37rnn/uUvfzl27FgeqkEeOBOYNW8pruhm\nInGmK9skM2eaEbcw5xpKdC5axoaZ4zlbdtZm1u/sJ/M6duvKXhHK68CNbV07R23n32un1h9d\nz6oXdeL03dyrTcXek5Y2klBPQwthDe14gWTyH4r97W9/+53vfGf48OFXXnnlf/zHf7Ro0cK2\nQnV1dTy1IUrOKBN0W5us280T5l40w4dzbFSOD0C2HxXNOotxRjrfkgLVrC7Jq3cwktGlTKhB\n4YzlvhPFal6/zXrfEaJ/bsNVrhbufw8KIvdjZ3QSgJp/sKuqqjJePPvss64r8IcmsXTCnHUd\nr3FSRQ+c/nXaKxbYllj3pa5fpwfLGjRdQ5hXsrT1QTp7BJ070k9vvu+L85zY4mnQ4VevOp0Z\nOpfMpy5JQuUt58dP5/OmWU/cGtrfxoZ2vEAy+Qe7MWPGlJWVlZaWJuEPJULz7WixXTUVv1I0\nYs1Jro0oUqPXEq+jsA6A2prNsa8uUHyxpUbn4biWHZRr5lZzrcf30hu6Ry3HvKU4ezpJOlw9\nAJA+/sFuzpw5Xr86cODAvn37Iq0HUdK81povrHO5XEcznfFCpzdLfwhV0Yi53BkNddp0lm3b\nkesx2nKqF6/cpt8nF0jobX27S3NpPHQN+ps4pzCS3gDAxv/mCYWnn3767LPPjqoU5J9XCHMO\n/HmNeyo6YEJ0mCmu07YOv+wn1K25spaX+YS5ic4R6RTv7HzyOpm2YnyPyKs22+aufZmurLX5\nrhwhZ8G2FbyON+s3U1D/I+d7zpEEvEeAPq2HDO/atWvOnDmbN2+ur683F9bW1i5atGj//v2x\n1YZ8M6+Xrr1ovtvaevu8mrLuy9Znph6nszboGghsXXohenR8u4K8To7XqVMMoToPJFrWtyP5\nPVtBK4wk1aEo8IYCgfgHu82bN/fr12/nzp0uGzdufMcdd8RQFXLiFXrE+3LozGRego6r5r6m\nbw1er+X4o3aeAWs7XsN8zjoVw8Rev3XdMOu4myTCMVazKusLRVr1ajnWnrz4mo2kkxgJEegN\nBeAf7G6//fba2toZM2Z07979/PPPf/DBBzt06LB06dLZs2c/9NBDfFds0mjO8XJuYg0Zzn6v\n0H9bXTuxXKNY1nHvp/O1eCczTb69ib69jNZ2so57JlyX+yYq1/yUe6hS9y96yb2Hz3lO8pyf\niGumggyyR67Y6wfyyT/YLVu2bPz48ePHj6+trRWRnj171tTUDB06dMyYMeeff/7ChQsHDBgQ\nf53Q5ewHsnL9lW1805kGbDEr0Fit75qK0VXfUOJVjCvX/jlxu2Yo8pbXufW6fMZ3WdVs2TUs\nqj8his1DSH5fS1EMVQOAJv+bJ7Zu3dq5c2cRKSkpEZG6ujpjee/evcePHz9lypRY60MIvlcp\nzeFF/c1D78irQ8t5E4BOAYqJ9q6tWV9bOwtNzl+ZK1iLt8VEr+TnNdPOK1h4NaXDmpWtu9YZ\nL8495UTSSD4lP3168fpcmYruvQCQI/9g16xZs+3bt4tIWVlZ06ZNN27caP6qR48eq1atirE6\nRCf7CeNH83pgCzHWlSPcqdmga1+ga/7z7QJ0NqhITnJ8ylF0SdrK0FzuPGOusU8h6AlXt2x7\nu3PZUY6KIlgkv0KkRvH+XwSKhX+wGzhw4AMPPLB06VIROeOMM+6//37zTtglS5aUl5fHWh8i\n5Ps/99bVnAOjoffomqWCNqJfiWKoVL2Jcy+2bjxra86I7LpHa5vOZiVI7glxDs3uQNuuc3k7\n0sTaXVqkJ6QocjNMRfoxQ3HxD3a33nrr7t27J0yYICLjxo1btWpVjx49Ro0addZZZ82aNevC\nCy+Mv0hEzNqdYxuadF1Z549R0AuMa8oxuS60jXja/utc7tzKdUfOFKvYu1fvozUCelUi0SWq\ncFnZq380byIPlNE2yEUXcSOFIw/8b57o16/f8uXLV6xYISJXX331O++8M23atKeeeiqTyYwY\nMWLatGnxFwktGb/Z9LYspY5xWbd7GnT2HvTqqNgke/y3dSnGT535SbG7EJXYtrXGOMWJMvOf\naxAMMfZqTY22KOn65jpHh0Ps10uETRWK6+g5ECs+Zoib1gOK+/Tp06dPHxHJZDI//OEP77zz\nzm3btrVp06aysjLm8hCxjOOuUvEeJbTSTGy+61hjUPb4b7mwrSbKHibX3jLXHKPY0DhwZ2eb\nazG+yxV/rzOOR8nkMrrtGto0RZjqCrj3OBqMpCnf/7kCgLhpBTubioqKjh07Rl0JcqU/VcuW\nqHQGQ6MapXINZF5FWpeE6At05kLXRtRdhl6N20Z7rbO1rHu3rWzLtfrHoqDOlzFFjRBvh6/4\nUlEKOhfzjDPWQPBGp5J7sKupqdHcvq6u7vXXX4+uHuSDVyeZ70BkLsOs4TaX4DPD1HnRVkyg\ngWbbEuuwtW1Nr0l+Gb0H77lGHMWQawhRpShnVQW/TnidvTwUVvBjj1AS3krEikmlaeUe7GwP\nMSkpKTly5Ijx2vqvvUWLFs2bN4+1PkRIf5w03ObqTUL0t/lOv/Ot0Dm5TTRSnW0F38ub6yUw\n0OC11y5yv7hGeG3OHH8vsDhOYI7XCdeOT51iNBvPobS8Ssh4bsELQNzi6HdHErjfFVtvsXPn\nzpqamvHjx69Zs+bQoUPHjh3bu3fv8uXLv/KVr/Tp02ft2rV5rhhBec1jc13TuTDCCebRhgxF\nm9ZZg67r+I6xKgZVbS2b2dEMJdaBV9eVFYfjXNN6pJqtmRsqRtt9N9ep0KRoU/Hx0/9k6pRk\nq822xLZHrmdqpLoGgjc6lfwfdzJhwoS2bdvOmDHjzDPPrKioEJFmzZoNGDDgiSeeqKysvOWW\nW+IvEu50rovqUOK1lW8ECffnwJZ71DRXc0141uu3+dqZQkLs1JreNHOJbTVFhnNdxysqWQ9N\nRx6iTO5BLVDcDH1NCp3tcjxATUEzd1BBPzkAiot/sHvmmWeGDh3q+qvBgwcvXLgw6pIQgCJ+\n+Q6TZbwn+7v+3fdaOSidfkGvGrw4u9lc92vtwcocf6eqomWd7OW1mpqZMLzaty23lu27L/0c\nr1lnOIrTYubjQLuwvXchivE9gUWde3yL1/z8AChS/sFu7969O3fudP3V7t279+7dG3VJCMP2\n19w5Oub8O551TLDwDUaR1JlL477dbOrLtjMnOTcUtwio6EJTV+7MZ66bq2O0V8u2zKSI5jle\nxV1PqTNA5NKFlnvOCPF/AkHF3ZeWN+k4CgCu/INdjx49pk+fvnLlStvyFStWPPzww926dYun\nMPjwvYb59tY4M4drm8npulB3T5oJKcRAW4jBKcV503lrFJ2FivoVwSL08KImzSgQOjAF2kWI\nnlHXHusIdxFOHO+XV/FF3Q0JQJ//c+ymTp06cuTIfv36VVdXd+rUqaKiMbhPCQAAIABJREFU\nora2dtOmTRs2bMhkMjNmzMhDlRBHDsta7vf0vd5bV5bj++ry/Lfe2U2oz7dfR2cEyrVvzHo2\nFOOGrh1UXgOpOmHFlg7NTTKOZ6lY3zvXZrO5PfdYn/5YsHND162iqjlQO0mIOIqB+ITzeisB\nJIF/sLvkkkuWLl16zz33LF26dMOGDcbCsrKywYMHT5o0yWv6HfImxJ/XAl7Vcty1/oitTrdZ\n6F4ua8wKlFRcr+LOTBZotNe2ji0CKlbW/OS49uwW10VdZx5C/uX5HBbXWwYgNK1vnjj33HMX\nL1587NixrVu3Hjx4sLKysqqqqnHjMN9agdD0J43xF1y8+70UHXKur0WZqzT78Gyz35y7yB7/\nJGfXMjRrCLStb7NesTVEtnOek8J+UIvxn0nBT5pBc2oHgIIIEM5KSkrat28fXykIzfdPqm0e\nvTOOFLz3Iih1zc7uK9do5ToI6xywVgzCKvZry5TqmjUHkb3CfUbvDl+zkUBXX2fZuY8hFu8o\nZAHp/yMlYwENmX+wy2az8+bNe/TRRz/44APz+yes3nzzzRgKgy5naFNnFHULyafucvM9Ftch\nS0X+c+7UGtpc967ornPNdjqDp9bfOufe6VzCXY9Rf2qmYrA4RIzQmRGY7nQS7qTpNFuoM5bW\ndwooOv7B7t577504caKInHDCCaWlpfGXBBXn9cBrUM81o7jOxy8uITq9nCnEdUPXrKbTvniE\nM9cYF2L+nHXNcJdPa9dj0Hff638GcryQ+24eeUdyA+kgbCCHCcCLf7D7xS9+MXTo0JkzZ3bu\n3DkPBSEE/SG/YqF5CF7JSR3yvAZYxbuPSr2mtWXXrTRDjOvkPFudzvda3bK1fa8Nvf5vwbbc\n+qPify0iFLRl327p5ISemMpIyNEBKCD/YLd9+/Z58+aR6hIi0KCbeI/92WaYJU0uhUV1UK7R\nzboL5zm05TNr76l6BNmVs+fVljudmyuSjeYlXzMTazaVnJyRnEqKTrrHxIH08Q92bdq04Z90\n8qknSwX9Vd7kGC6Dbus1hU6zTcUwrhfbyLjvHEevXjGvPeYYntTXbOdy/X0V5NPFXyoA8P/m\nibFjx86ePTsPpSB3RneRfNKZ5Jx+pxhbLIj4yrAdrOaONDOW7dzaBi51bixwlmRuqJiEp7jv\nwdpgJGc12il0IUoKfRT6bzd0OP+SIHd8RBEf/x67O++8c/To0VdcccWVV1556qmnOu+fqK6u\njqc2aLEO9rnOx/L6Md1C9K55sZ5VZ8LzatnrVgzbJDzzqmmNZer87Zwn5zuZT92xlzn+aSmB\nTpR68DeXcCZ+ZbtuRQRB8jWoP8XIP/9g16xZM+PF448/7roCf0kLyzVqQIdX8giaSLKOO5EV\n2Us8HpiiWaH+pDrbOrbQY26oGJKO6rMUIuFlg39Dmm+6Fb3TBcQtyfObkQL+wW7s2LFlZWV8\nz0QxUgQXKeb/azSv+vqH4LzLQbGyc0zTurlm4HDeaeFszWs817qmV5LTLMM3yrhOpPNq3Llc\nXUMun7Sg8SuSiyXJD/nBZwzx8Y9rXh11InLgwIF9+/ZFWg8io7jwF2+kM+Q+YSvH67ci3/je\nVarIiDqdc+oyFAXL8YHSFrkUh5PjvMx8fthc5zJalxf8akpwBBA3/5snFJ5++umzzz47qlIQ\nIdvEO2NhsV9Ocp+PH3q/XjP2zHn6tttWbJsr3gKvuyhcO9I0W7OWFCj5eR2j1+68DsRag1m2\nThkRcp5Y5xIn1zOPlNH5JADFS2uAddeuXXPmzNm8eXN9fb25sLa2dtGiRfv374+tNoTk1StQ\n7H/Loq1ff+TOdzWv2Bduipgmr/FcdZgLMZCqv7643cObcdwsom4hWolKac5+00JX1EAxxQ3p\n5h/sNm/e3K9fv507d7ps3LjxHXfcEUNViBJ/wqLlDFKaG7rOC1TMQnONQb6TC539eVnvezvM\nNr0WOper9+sbWfKW7QKlUv35l0gH3mWkmH+wu/3222tra2fMmNG9e/fzzz//wQcf7NChw9Kl\nS2fPnv3QQw8NHTo0D1XC5LxwWpfYfkukUwh044VtEzOd2O6HcOYDWyeN1yw6W1az7kgcUdL1\ntVc4s46E6k809O2D9NrW93qZ2AtqHnpxUjMpAkCS+Qe7ZcuWjR8/fvz48bW1tSLSs2fPmpqa\noUOHjhkz5vzzz1+4cOGAAQPirxPH8erLUfwIU7hLuC2Z+Z5tZ4bT2anXfD4znwXqIJTjPyqK\neXuuE/K88kfoj5aZL30jqRdbUFbXGbSwguQthmUBRMv/5omtW7caXxRbUlIiInV1dcby3r17\njx8/fsqUKbHWBxvnFcg53ZupwQZ155Z6fSvXrjjntupwoJj46GzfNYqp31bnx8DrJgxrPQqu\nIc+sIcebDPh8Ikf8lQO8+Ae7Zs2abd++XUTKysqaNm26ceNG81c9evRYtWpVjNXBDf9zr8k2\nTpqffTl35PV+ueYwr0Zck5/mtU1nNVtQ07/9QrFT9e40C1O0oE6uxUL/EFxPV4SfgRCIVkAC\n+Qe7gQMHPvDAA0uXLhWRM8444/777zfvhF2yZEl5eXms9UHN2d9jKOzVrolIJ5HOIu1FThRp\nWqAygs411OwP87qrwGtzaxeXbUJk5pPn0bgOTWY/ucXVd9jdulzzYHOJRL4buk4o1N9cvWv9\nzYkdsSrs6U1Bpgdi4j/H7tZbbx00aNCECRNWrVo1bty4a665pkePHn379t20adOaNWuuuOKK\nPFTZwLnOqHNdJ/9/ak8WGSxyjkg7kfYiVSIdRJq5rXlU5N8ib4u8JbJO5C2R9SLb4qwt97OR\nPf5hws6Frmu6LvF67bpT52rOWWXOT4VrqaIMUragad1RjvFL50Mb64VZcZK9onnCec0BCL1t\njgryBweAL/9g169fv+XLl69YsUJErr766nfeeWfatGlPPfVUJpMZMWLEtGnT4i8SntnO629r\nrH9wW4oMEvm8yHkivUT2ivxV5N8iy0S2i3zwyX+NvtwWIiUizUUaibQX6S7SVWSoSCeRxiK7\nRV4SeUHkBZG34qv4eL4XJDOXqLtCXfOBdatwA44Z5UN9Fd14ttrCXcvV3Zb6jdviabTBQqdN\n19RrntjQ5wdWuc+z5F0AIhfmr1ttbe22bdvatGlTWVkZR01J86tf/er666/ft29f06aFGVT0\nvQjlZzJZmchlIjeInCtySGSZyBKRpSKrRY6Gaq1a5EyRwSIXiHQW+UBkicgLIn8U2RV18bGy\n9niZS3T6UF3fVtcLnrM70GtNLzor+67jzKyiEfg0U5T+4YTIBKFjRGHzR4rTT5IPLcm1ISHq\n6urKy8tffvnl/v37F7oWO/85dsuXL//www+tSyoqKjp27FhZWblixYr58+fHVhv+l2aqk9g6\n6tqL/EBks8iDIutFBomcJDJM5L9FVmmnOlsnU53IOpEnRK4TOV3kdJEfiJSJ/FRkq8gfRL5a\nuMl5pkDzeJydefqz2q29eoreOHXkCtE1qLNr68rq1rxWi/wC6VWnokhzzmIk/0Y024lqd6nE\nJDkgJlo3T7z00kuuv1q2bNm4ceOiLgla8nPN+JzIkyKbRL4s8mORDiI3iLwsciR4U+qCN4o8\nKDJWpK3IEJF/i9wnsl1kjsgIkbJcjsFDoI4uUT49xBmPrJs49+uV0qyT5Gyc6/vGLHND44Wz\nW9FarTPkBVWM12mdgy3scRXjWU0BTjuKmuccuw0bNmzYsMF4vXr16oqKCtsKhw4dmjt37uHD\nh2OsDoXTXuQXIiNFnhEZJvKCSH7+zh0TeVHkRZHxIheLXC7yhMhBkVkivxT5V3Q70uxxsb1W\n/Ll3zq5z/so6uytjmSpn/a26JGtq9Iprrp24imNx7td5sLY5cxnlPD/fc+W6I9sugrags1Nn\nr2o4mld9wgGA/PMMdvPmzZs8ebLxeurUqV6rjR49OvqiICLKx2rYAkG0GomMF7lL5C2RviJr\nIt+BXoKpE3la5GmRZiKXi3xLZKLIApH7RJbFvOtwrWU0bnAJumvfmOJsMHP83D7N43UNRq5R\nSdFgxmNaYaCgJsdn1vhE+0kAgITwDHaTJk266qqrVq5ceemll37ta1/r0aOHbYVGjRp17tx5\nxIgRMVcIl8QQ3wXpbJFfiXQRuVXklyLH4tmLun7bFXefyAPZbEkmc57It0SWivxdZLrIYyJe\n3cW5p17NjjRFV5lObjDbt03/8trWtlDR5ebchVg625wHaNvK2qeo6L1T7866frimbH14XrnZ\ntykvobNjuC7JoLugww+5C/0JRPFSPe6kbdu2I0aMGD58+I033lhTU5O3mtSy2eymTZs2bty4\nb98+EWnRokWXLl1OOeWUQtcVPd9JURJpr0NTkbtEviXye5FLRf6dc4Oha/Pq8jGeitJJ5EaR\ne0V+IPJTkVkih9xa8PpD5lqSrVTnmbf2IamPyxZivHahXu7Mi66HYxvHtIUnZxnqqpwpKsJP\nV6GuK3Fc1fLQz0dXIoDQ/J9jt2jRIhE5evRoo0aNjCWHDx9es2ZNWVlZ79698/kH6KOPPrrn\nnntmz569Y8cO269OPfXUr3/96xMmTEjfE1icPStxDMW2E/mDSEuRESJ/jKRFjdrC5YZNIhNF\npop8U+R2kVtF7hX5pcj+gHtXFOPVD+cMWIquNUWK8urWEkeadNbgnIhmzWHOrdSZJtp/v75v\nqH7Acj05rmwnM+6OrhDDyiH2koccTEdOA8Fb3AD5B7ujR49++9vf3rFjx5NPPikimzdvPv/8\n841vjD333HMXL16cn6e7bd26dcCAAZs2berSpcuwYcNOO+20Jk2aiMjevXvffffdv/zlL3fe\neef8+fNffPHFE088MQ/1FIo1GUSlp8gfRd4XOV/kQ//VI2AbeQxhn8iPRO4T+YbIBJHvifxC\n5D6RPR778m1QfyKa5gR8rwlnthUUTdlW8J3AJ8qrtVdktO3INYzqh8XCXkW8urcjlJ/5fwAQ\nXtbPj370IxH57ne/a/w4bNiwTCZzww033HjjjSUlJT/60Y98W4jEtddeW1paOnfuXNff1tfX\n33///ZlM5qabbop81w888ICI7Nu3L/KWw4n2AzBY5CORJ0Xstz3HJvJDqBC5UeQ9kd0iE0WC\n9tlGWI/zDTJ/zLpFNNc31HVNcfun6lyuaNZZp7ma+qPlu1P9j6v+VqF3mstekizcaY+wQQBW\nxiNBXn755UIX4sL/H3avXr1GjRplvP7ggw8ymcy1115r/HjNNdf07t07xuosqqqqrrnmGvU6\nY8aMOeWUUyLfdaGCnddl2/dKrG+sSK3Iz3SeZxiD3Ou3KhP5lsg2kX+JjNPpi460DHUjWcsg\nqdcbba7p+kZrfkLUjVvb913Zt1r1fkOvr2jE+ToOtvZj3Ze+oGX4nqWEHBdQpJIc7Pwv6Js3\nbx4yZIjx+rnnnstms2PHjjV+7NOnz+bNm31biMTu3btPP/109Trdu3ffvn17furJg6xHX45z\neTjfE5kt8j2R7wa/+zVruXK4LteR8XgWWjh1ItNFqkX+R+S/Rd4U+ZKIzlivbYwyXD3qkVxz\nAD37yUQ6rxsprC+yjgFr1w0zx3O2abIV49zKtrLzVASdYSbHJyTXppIvCaUG/Vj6fpKj+ncH\nIGn8g531j9rzzz/fpEmTgQMHGj9ms9kjR0J8B0EY7dq1e+ONN9TrrF69ul27dvmpJz9sf3yt\nl2RzhXB/oO8T+b7IGJFfBC9DLJ8KxcwtdQvOzQMV4GW/yF0inUX+IPKoyAqRz2luGbCkEFyP\nwnmjhvn+Wk+yIrRp7t0WLvU532JnRgzUoGvLtuWKNiP5nwHFLpztk4EAFBH/YHfaaacZXym2\nffv2Z555ZsiQIWVl//sNT2+88UaHDh3iLfATI0eOfPLJJ//7v//b9bsuDhw4MGXKlKeffnrM\nmDH5qScJvK73vm4XuUbkQhGdL/oN1E3oez12zYi+l+qgx7hb5BaR/xBZJ/KiyDyRznobBg2a\nIUKSLROrm7KeGesJ9OpmU2dra1ecsxvPliNtZVvbsS609f95HaxXSYrfOovXXFkhxD+WvKW6\nJPQLAkgB/5lIl19++a233rpp06b33ntv//79N910k7H80UcffeSRR8wf4/b9739/2bJlEydO\nnDp1ar9+/U455ZSmTZtms9n9+/e/9957K1asOHjw4MCBA2+//fb81FNA2dxuKb1cZIrIKJHl\neutbA4Sik8NrczOIuG6ufyyKzb02+ZfIVSLTRX4m8m55+U8PH77H7bbZEKwZSF2DF/PArVlN\nLKfLuiOd1Bu06865ftD315rwssc/yUWH86jVu9OR8XgMsvlb0fsfiTio36Zw/5yDvvUAGoSs\nn0OHDl199dWVlZUtWrS47777zOVt27Y944wzPvzwQ98WonL48OGf/exnvXv3Np+oZygtLa2p\nqfmf//mf+vr6OPabnLtizbfM+t4Fers/J1Ir8s0oPjly/BXFWaRXbUFrVuw9qx0IRotkO3fe\nIXK9SCOvlTTaiYnzzdX5rfOEqz88rm+K6wdMv1nXj6jXts7Gw+1F0aZi79aVvSrPpRLXMnI8\nnzr7imo1APqSfPNE+Acy/fWvf+3bt2/jxvp3H0amtrb2X//6l/HNE82bNz/11FPN0eE4/OpX\nv7r++uv37duXnyf2+QrdV9dV5BWRR0S+m8Pes47vKnDtajI/Vxm3L4YK2vemU4z/yrW1/1lR\ncZvIJpEbRV4NujPHro0XioMK0aatEes/T9t4qOuJ9WrZuYJ60FynzaCcXb/OetRLFM16reb8\nQIZrR5/rZzvHNnNExx4Qubq6uvLy8pdffrl///6FrsUufCwr4JeMVVRUdOnSpVB7L5Qc/zq3\nEvmDyEsiE3Kowbl3c/DLmjOsqykKtmVBr4XmEmuaMV/oJ6pMRYWIzBbZ+rWvvfLYY3L11fKT\nn2RatdLZ1lah1yw012TmeiyKdaxLvM6DeL8X4vioOM+PesjVdfPcY4H17XNtLdwufCOdZsua\ne/c9G9nkPcE4RFwGULwK8vwyhBeuW6hS5GmR3SJXBH+yiW8xik4mL0Z3sXNb9SbWefrmi6An\nZJtIZvbsQceOvfnwwx+2anWd5d+AZtrwvSFA3XNm60My/2vrx3K2YztY1zIUJ8RZlXVH6rL1\n9+JVmHh8WtTl5dIP6nUa42OelkAZEQkX4o8MUFgFGEiNybvvvnvdddeJyPPPP6+/1Y4dO669\n9tra2lrFOlu2bJGE/T9uoAteicijIm1FakQOarcv3olNPZbqNfhlGzq09eqZW7l2FwX6w6ro\nqTK9JHK2yLdFfipyrcgNIqu8v/VVf9eZ4++l8EqKzoOyvVZ/2LzOlTUaevVXOTdx1i9659BZ\np7Uw16DmtYnv8UbbmZcLr6CvH+as/7g03+sIDyRRf8cAxCE9wW7fvn0vvPBC0K0qKyt79+6t\nfhpfo0aN1q9fX8D/abP+cc8Evw3zFpELRT4rov/s5nBBynfsTz++uIYezU4+zVN0RORekTki\nPxP5m8gvMplmIvvU22hwVu48LbYgq0jMtijsFT1tb4FmaHCt3JnOnTnS9/Q6h3RzDCghsl3+\nx0MDZbsQhxN0R+EwVmvDqUDRSdZckFzU1tZu2LBBRHr16hVtywW/eSKXTNlLZJXIN0QeDb6t\n2cUSrqPFydafZI2q1hUU9fiuE85FIjNFSkW+JbIg4La2LjpFX6bXEmdPm1cYcq6m7pkTj/fI\n9e1Tt6lu0FUkwS7EfsXtQ5LnP3Rexxs6bhLsvBRdwUiHJN88kZ45dhUVFb169Yo81SVB6HlC\nZSKzRf4YMNWZM67Cdfz4tuxc7nrxzlo41/RqP1xhz4r0EnlMZK5I9tJLTwnSrNkBo5nqxHEU\ntt677PE3Z1hfO0+gdYn5fqnPkmvL1qas/3V9azTbtx5L6M+wc+/Oyr3WL8jFPqaPqOs/hwiZ\npzS+XQDID/dg99prr4VoK9xWEdq9e7fRaQfDFJG2ItcF3CpQH4nvJdaV4uLhulyzry6XzryD\nIpNE+oi8+vTT60RutjzuTrMfMZe9ex21s8Ms3Ci5yTdl+m5oG6vNWFhXjqpjNXTOMLOdbwu5\nV2u2YP1/odTEI+f7K5G+xTlK06mGmutHEU7uwa5///7Tp08P1ND06dML3iH505/+NK2PQQnx\nUa4R+U+R60R2xlCPVwdV6Kacy229VrnvSMdakXNFJopMEXlZpIf3mtbIpSjPqxdEcTXy6lGz\npQdb76Zry84WFAei4Mz6+pfSOHJVJNfyOK4QsXaqxcr3/7WcH+B8lAVYFLAnvri4B7tvfOMb\n3/72t/v377948WLfJhYvXty/f/9vf/vb48aNi7o8/K+gH+UmIo+KzBZ5OmDLrr1Eim3N9TUv\nk3FEtGj/nR8TeUCkl8hOkddFbhcpdVvNNoipbtMaxVxPlzXMKVrTzFWutQUt2NrtpMgrWQvb\nEltrrn176uJ9i9RcUy33KKaZ8gNJVM+E11uvs22iDgTFjlSnw/2uWKP77Vvf+tawYcO6d+8+\nZMiQz3/+89XV1SeffHKLFi327Nmza9euDRs2vPjii3/605/Wr1//qU996vHHHx87dmyeq08x\nZ69MIP8lUipys/aObPtyDrs4i1H3G6n35dWmsylzNa+womjB63Ki09e4ReQLIpeLPHbyyV/c\ntesakdXeKwe6bvmunHG7AdYam8wrZfb4my3MbV0P0Lmy8x3X+bw53xqvoxDHx8M3v5pdkprJ\n1XdNBZ1dmCncd/2CX29y/IsBIDVUfxb3798/Y8aMadOmbd/u+aCMqqqqm2++efz48XHfMdq3\nb1/fdbZs2bJt27bI/7QV5K5Y559p/fQwROSPIheILFWulnW7B1O9XKdC3+ul/sVSsZrtiq7T\nueiaEX21FpkucpnIf4ncJXJYe0PNHdnOhleRXh1vrvnDua0zFvvuyOT7UbT9ypYyndnL9b3T\nCevOAlzbj0TQYJdL+1E1GKg11/VJh4CmJN8Vq3qOXdOmTSdNmvSf//mfr7322gsvvPDOO+/s\n2rVr7969zZs3P/nkk7t06XLBBRecffbZJSX5uLV29erVIlJa6jos9r/q6+vzUEl+hP7beqLI\nQyL3eac667XcuRfrcKG6knAVag4mqtn6dQINAQcdFdoh8uVsdlQmM1NkpMjVIquCFKm/muKE\nKPq9vFZz7sLrvXaurPnWOEOV7X3JetxYbWtf/aPXIVjfekXNoZNKoK5o2x4LEoyCFunVCMOm\nQLHzf0BxSUnJOeecc8455+ShGoWJEyfOnDnz9ddfr66u9lpn0qRJP/nJT/JZVT5p/s39iche\nkVu9V/AaD839apTn65lXh6LrUGaIxq0tPCXyF5HpIq+K/FjkLpE67W01V7P1dblukstBeaUr\n32o1E6G4RT1Fj12ILjFrZFS0X0D6QbOwNSuGxaMqLFHvC9Cg+He2LV++/MMPP3T91YoVK+bP\nnx91Se7uuuuu6urqsWPHqr8lIq3MaVX/n73zjpOiyN//MyxLkCWp3JlQiYKiCHgeioggnoqZ\ncOd5BgTOgOFEMOCpiCL6U8xfPDOeeGa98xTTgYCeCRFETEhYAygmlAy77Pbvj5G+2kpd3RO6\nZ+Z5/7Gv2e6qT32quqfrmae6Z+zsCJwKXALYfiItqy3Ggq8GJFUnbQk1r1gKp2OuAv4EDAH+\nDLwLdDOHclkX1sopi3unmnNiB6WDJc3TkqmmtqsKL3t5sZilaXFLSkDtjh1L4Wi7whI4CD7S\ncck1mbxJLWvrmZPYSwchpUCwsOvdu/drr72m3fX666/n7UnY8vLyf/zjHx999NFll1ncqFLn\nXKASmOZcPnbnIBBThi56JVRkqbrJQ/oX0AX4BHgHGKd7YDZwPMMOuLqgKUoldXz87WooyxCp\nisRRnloKWA6K1JwlVCilHk04ZohjW6bjlViypRoJIXnGuBS7ZMkS/8t+58+f36hRI6nAxo0b\nn3jiic2b3e8mz5TOnTuvXLnSciPdkUce2aJFi7zlk0+kZUctTYCzgEuBwGuqy2W34C7N6qqc\n5CRpjSX7wKZ0q7o/ACcCQ4DJwLHAqcBHQVU8841xgWiNPd/zE3WPtlFfoUo+nOn4agdQlblS\nN9WAajKm7rigtTDFQdDmZlLnmbfuvjenZKXpHOVfcBcQQooGo7B76qmnxo4dm3599dVXm4oN\nHjw4+0mZadasmWVvnz59+vTpk7dk8onLdHg6UA1MNewt0OusODdL87Q6bWuLiYXFf7WLoaYE\nVJ7wvF+nUncDc4G/ArcCteYqodSMb8uZljhdDmVWLCutGtMqVBeZaC/pkoaag6W86nTmAvfI\nEXLQZp4ttRpILprIW/KElDJGYXfppZeedtpp77777nHHHXfKKafsuaf8JfxlZWVt27Y99thj\nc5whcZqky4ALgDvNd9fldHpLJqou8eres28q5hj5O+AE4HTgVuBY4DTgi7plXFYz1XYD7xtL\nBT0GITZtklaSYhb9TtWHMw2RZcE37C4TmbjLqombHByHwpJ/5iKJMouQosT2VOyOO+547LHH\nHnXUUSNHjuzZs2feciIijmrjeGAn4G+5zibvWJwq98XEUAt5LojVpwAzgQeBD4ALgClbc8jE\nDgwsafdyxBeBvRNvShMraqtLCi/PskkVmmIyMB/6xMo7O9qcE94R+1AnPHlCioPghyfOOOOM\npk2b5iEVkgmjgb8H/SxsVpbnEk5qK6a92teRG0q//hzoB4wH7gT+CfzKHDwtB+2RXcrAel+g\ntoAqiy22n+MEbFIeXt37/0yNwnyHvnocRempjemiX/OJ5TxENp5bSkIEiVK4whCSfIKF3R/+\n8Ifnn38+D6mQNOKU5nihPBD4LXBrpIbCJ1gAiK6Vo7Fnn4MDW6zxvJs8bz9gN+AD4BhDMcfF\nWe32wGlYFGdiLfGM8h0s9QTzC/sFxC2BqCIssmjQtmhXb9oVZ09NN0c+AAAgAElEQVRZknY8\n51V/tIjfLFkk60qREBKBYGF30EEHzZ49u7a2NrAkyQrSapcLo4HngUVKHH8udJEFhYKjIaeW\nVKVehvpD5SOgJ/AA8E/gHiDwF+hMy6lqAYu2ELdLR1ktLxY26Sf1DLSfimE/h0jp2bWstCVC\nfK0BaR9Py7/uRHjHubeVI6FJ/UpIERD8yxMPP/zwqFGjjjrqqFNPPbVjx47NmzeXClh+CoK4\n409y4iTtcpFtDxwH9NMFtE8txaHzIiDO947+TWBJf28VcBkwDXgIWNuuHaZOTdX9JUFVa5pC\nmQrYE4DOupPw7StTNNHfUhdPLf+GTdXSOoKUXyDqsrJW4anyUey4JJS1o6GOZ4Sc7SdDjt6t\nWf9sQwiJnWBht8MOO6RfvPTSS9oCvChkF3U2snMBMA/QfoW0GKdoruB2qWpyvKDMvv60rVbR\nGl2hknwD2Be4ZenSoQce6F1+eYMJE6qdlbojnu4ZWDVbUZVqFYxUxWSSiRG0654mY0xb3rQx\n61j6Hlg3Qo8yOb7a6qaBzWSZ2+XDXqhLUNGTa3lNSHYJFnZ/+MMfGjRoUF5eTos+p0gSxHG0\ntwWGAsMdYpYIolklrg9q53XTIAcaWi6sBUYAzwH3TJjwJnCym1MVyqKzGHKinous6bV3BZgU\nksUYC7TNVNxzVks61tUWE7WySb/C/HxutLVXu8Oa9Xex2DVtcMsSdmkS9sM2IfESLOwee+yx\nPORBos2+ZwHfAy6/11tqVyXtXV+iQWWyr+xqT4oTyLPA28D9wDxgNHCXQ86W1tXu2BdbA5Wi\nKmu0twTATeQlavLzO6I9cBZBYylg2V4ouORf6H3MBRwTUkAEPzxB8kmom5fLgfOA2wD1R9ak\n+awE8eWIPxTqtO0JoO61Wy2QErC0qPItcAwwBrgJeA74VdS+aDHdBqemak9easiyoq1NAAYl\nDeFUtLcu7c36CayOhqmJrLdr+TdUN0NdHAghJUuwY7fvvvs2bNjQ8YLy9ttvZ5xSiRJh+e9g\nYLut34grEXn1rZgwyRSXKllp/X/+EPA3YCbwMLAQGAZMCxktF2aYSb15hvvS7HulZUqLOSrl\nYO+XVrbaA0rdcY9ssfEiLPVCELviu9uUf+aHmO/6DOEAkuIgWNitXLlyzZo1GzduTP8rXn0a\nN25cVVWVw+xKj1Dy7hhgNrBaqe4eoRSQLtbakTENl7tKUIOoMT8FDgCuBp4F7gHGABu27rKs\nmQauEUORDmLdzDuiTUl6Hdh3S9MW8yyw42pYkwLz81RfWHLLEK32jSBztUlShRBCtAQvxX7y\nySc9evQ455xz5s2bt3Hjxtra2tWrV8+ePXvgwIG9e/detWrVFoE8ZEx8jgGeq7tFvEeKV/w0\n6uQqratmIoLDDnI1MBY4FBgAvAf02LrdMt9rFzcdS1r0nHbdU+tOBY6PFMpeWMUlN2llPGwT\nalvSC8v5oLaozcHxdLInn+F7lu/6DOEAkuIgWNiNHj26ffv2//d//9etW7dGjRoBaNas2cEH\nH/z000/Xq1dv9OjRuU+y5HC5uHQB2irCjmhxGU9xphdlhL1WNBk9G+gKzAPeBC4FyoRQjklq\n/7Uk6btikvhwydyxgHbcAuOrLpTv0qkCS9rooqIsItjFDtRWNG2UrMGwkcNKijyrkAw/AmU9\nDiHERLCwe/7553v37q3d1b9//3//+9/ZTqlECXuxOxZYCFQq20vwE2cE4001gSTFgLpHxB48\nwly1GvgTcDpwCfAqsJtDwtrE7BlK3pu0S+vMqVXU4dWqLm0yoUbG4i+61AosZikZSiSpccTT\nyT0ld0rwTU0IiUywsFuzZs3KlSu1u7777rvVq1drd5FQRJgJjgGeVTaW5gQgzqkRKkq467nM\neQToCnjAAuAkczFtnvZei/6ce6007qLNXt00vHZhalp1VXVn5OMOQYFJrWTLwsyWnaYV3LGQ\nrR7l2WgkpAQJFnZ77rnn5MmT33nnHWn7G2+88cADD3Tq1Ck3iZUWprnTxK+B/YHnc5lSYZH5\nVOEyu+diTvoS6AdcB0wBHgFaKMnAaqeJWbksekoLnaIbpzqXqCuzxC0uWBRJoOenltGaYe7J\nmI6d2kRgHJfmuOCYLTiShIQl+KnYq666auDAgT179mzTpk27du0aN268cePGZcuWLVu2LJVK\n3XWX/StXiSuhFMPRwHfAu8r2VJK+IbbgkIbO/9cXKGFXe7Vrdinll2drgf8HvAL8A1gAnOZg\nhpn+DZukPT4MZ5SllcAI2lVvizDNyikdtheh4uSUhLydTR3nBYeQBBIs7I499tgZM2Zcd911\ns2fPrqz85Z6uBg0a9OvXb+zYsf37989xhqWFOM9ZZp1jgeeBWmU7L7LZJZNlWa3MstwkNx/o\nAdwITAduAq4A3L9JyNSQpWTgSi6UadvuwLnrJK05p263SMBQ2G+ts1SJIGX8vcX0TvRP2rAL\nC1lB/XyVh0YJKWiChR2APn369OnTp7a29ptvvtmwYUPjxo133HHHsrKyXCdXstivmI2B/sCJ\nynZe8jJHK6w95fe1sjWlSdplI3AucM4LL5w6YMBhwMnAx+aKYhq+rnJJzNQ7dbHVrnvgJnzV\n+PbyaqqOhcNiEgoRhKBYMexicfL1il39E0KSRvA9drW1vxhD9erV23nnnVu0aPHOO+88/PDD\nX375ZY5zKwmku51crpVpj3R63Y28yGaCeKuZj2+9OM70ptvy7E1rChx55A7fffcVMBc4F0jp\nioXScLDqUe1NZtoBUcvD8KCAJ6BWV2+eM90AZ+pOYMwIRBBkLkORayyHKT+OWt7I/9gSUqDY\nhN1rr73WrVu3mTNn+lv+85//tG/ffvDgwUOHDm3Xrt0dd9yR+wxLC5dr8THADGCjsIXXu0yw\nz+jqEqf2GKn6T9RS9gOkEW2/+tVxwAXA9cCLwE5BJ4ZJX4ofFcTPD6YI9kU3tRVpi6hsxIa0\nU7JFBol1/Q88Wm2qVaURUIWmNhOX6rDqWpXsesClDMeQkDRGYTdv3rwjjzzy/fff/+abb9Jb\n1q1bd9JJJ1VVVY0ZM+aaa6751a9+dcEFF7z33nv5SrU48a/+jtNSPeBogF8emHUCZ2KTBDGV\nd9EEdqvpHqAb0BxYsd123lNP2bLfmqHjwqIquUTpJklD8YWoeBzPWKmP0jhbuq/u0v4rSUBt\no9qAYhr2vvgV1QhqGnkgUDRbthcrYce/+BxNQnyMwm7SpElVVVXTpk07+eST01seeeSRH374\n4cYbb7zxxhsvv/zymTNnlpWV/e1vf8tXqkWO41VmP+DXyhed8AqVCZYpUJz4U+afNIVOKkVI\nQ924GOgNXPXjjzjxxAeBZg5xxEzs8T3zF91pA8JNz4kem/3M1AovdSQl+aVVV2rXTLtMKWk7\nrlV+qth1P+haPeGPlcsbOYKCCVXerxXXVSVauyWlYgmxYxR2b7755oABAwYMGOBveeGFFxo0\naHDKKaek/+3YsePvfve7119/Pec5EoFjgXcB9Qujqe3ygMnWkspI5pYWca9dA20BxgM9t2w5\nrWPHBYD2R2BUg8puParLf5JmUtWen6GaZ2QRII6ki96ylFHVlWUE3PWTqObVVhx9vmho65pU\nuz2I1m4MJBaplIn9GSrhQC3OKyopXIzCbuXKld26dfP/9Tzv9ddf79mzZ/Pmzf2Ne+yxx/Ll\ny3ObYMng+Ln/mLrrsKW24JJ/1IlcK3rUA+FuVrnMuO8AFZ999hIwE7geaBD0rKJ2IVL72nIK\n+du1GZrSloSjNohd/trzyfCEd3zLBL4fHYM4tq6OFXSnUNgRiDxicV1VsnKIM4eqjhQ0tocn\nmjZt6r/++OOPV61a1atXL6lAVZX7l22RYOwXlN2BfXQ32FHeZUIoC8pfntMuI/r/Rpv17awH\nzgaOA4YC7wB7m59mVVsRnR6X/qo2VWDOFh3piDipe4alSa/uV8+I+ai2pbYvYbMKS1hTTYtF\nVoYNKx67grhKmD6Z5DmHghgrQrQYhd0OO+zw7bff+v/OmDEDQO/edRaCvv32W9HAI7nmGGAZ\n8CEAXnqyisnNkjaKf929HPFfSYhESHUa8Ovvv68E3gVGb30DW0KZXB8t6ZPKJI8cc05X11ov\nfhD17PWEpV77pC5O/BmudWZdPUReSeTbmRCSLYzCrkuXLk8//XRNTQ2A6urq++67r0mTJocc\ncohfoKamZtq0aZ07d85DlqWD/eJ+LDBt62suFmQRk/YSN0qg7mSsWkfiv9ISm72tQFKtWg0E\nzgauBKYDu7p1ypSDujyq/deStt1jg7OctQyvNmFpo8XlsgdRe23xa+3vuxjFWZFdEKh0CYmM\nUdiddtppy5Yt69+//+TJkwcNGrRw4cLhw4c3btw4vbe2tvbSSy/96quvjjvuuHylWuQE+gfN\ngYP5RSc5QJ0/AicV8WA5uj5Zn6UeBLoCZcAHwCmGJEUxZzKTArvv1xWjqSrKcSnWroAtWWlD\niT5iNJ8s6+XzrEjEMSwybUcIiYbxJ8UGDRo0ePDgp556atasWQB69OgxYcIEf++xxx47bdq0\n9u3bn3XWWXnIkgA4HNgIzI47DYKt7pRneJTVtF0l2kzsx/8c6AuMAe4FjgfOAr5X4tuTkVw9\nSWlpNR/qLraa0pPalYJbLDFRq/mh7Jlrt0hi1ITWfbREyJtoc8xfpDgsrggdJ4SIGIVdvXr1\nnnjiiVmzZn388cetW7ceMGBA/fr/K7zLLrscccQR9957b0VFRV7yLH4C1cAA4GWgWiifh6yI\nFt8iMikPUZ0gq1aK1FYtcAPwIjAVWAj8GXiubmG/afW1lKSKWF7tiF3bQRmEwO5IjqA4emHP\ndrVRSS7k+u2jqpNc65XYLwgUZIQkBKOwA5BKpfr27du3b1911+TJk8vKynKWVSlin/9SwO+A\nv279l1fPeBGFkagVVCUXKG7c7T0psshCYH9gHPBP4O/AKGCNNXk/VHrdUOqLVhWZ8tQKR1Oq\nfhlLr0XRDOVUF9WD/1q0AyX9ah6GrBFBdzrGsYQV+55501kh7JlsiZN5EEJKGdvXnVigqssz\n+wI7AC8D4IUvA7K4QqrevmYJZTGKoq0YqlQBfwV6AwcDC4A+QnC/IXEhVUpAuwIr6gapgC+n\n1EzU9VxTr+3YR9V0c57YaGAadlyqiFLe3pxlxKAchZwu5ecOXpoISQIRhR3JLoEX6MOBhcDX\n+cmmSMnnZCnN6y73lpnqhsrqLWBf4EVgBnAL0Nj6PKykM0xNmJKx342nFVvi30y6qcYUA6oj\nbxHfdl3ukoC7mrEbt2KozIV+tDSkYlrvNmlqkhAiQWGXCAIv00cAL+YnleLFcS4MNWWqK4PS\nrmhEq5tOYz0wEhgADAbmAXjnHRg6pS53an0jVcBpnSdtzuIiqVZEprYi1VKHNF3Mq0tKQTsm\nUhDH5NUyloouCk8rZN31tFpAPHCRu+COvS41HyEJgcIufkRxoKUZcODWdViSCVnXdloJIqGq\nFksceyhtRe32V4C9gXeALT17TkylUFVlMsks64Ooe+OXquq0MUUXzSVVv6HA11AUoSWg2KKv\nArVlXI6g+/F1TClsK9nC3WLUHsHMfVbqP0JyDYVdIrBf7A4FNgFvAOBdLIkkmgPkgnZ50dSQ\n1OjPwFBgMDAc+KBhw311ZpV2i+pySQmYVjy12YrumtQpqadSXyQZESiIAzEpTu1G1Wg0tZXJ\nW9IUM6z6cRyH3F09UrpfEyGExAKFXfwEXg2PAF4F0j/Kyw+7SUCc9dX1wcDqprnc3pA2gqq3\nJJ4FugCLgDnA5dbH4LVyx5KeOALagOK6p/t56z6MUhV7Gf9IiYm5ZGUX1v72QFcvcLgCVWwo\nIhuQ0Qglr6n/CMkpFHaJwH6lOxx4KW+pEDfs872pPOqaW9LGDDEF+QH4PXAacAHwNrCXkJJ2\nlhWXOy3x7X2390jSVRY/Uuspasu4Kwa7fhIdStWtzBvujVo0d5ZzMpDT8cmFDCWkuKGwSwSW\nK1dnYDfhBjt+2E0CgT5ZYC1kNmNFqPgYsBewHPiwYcNLgTJFNrnHFEtKrp5p+dKiYkWZZfcI\nM9HB0uCb2lIlkeMQ2WWlLxOz6FdZTMdAl5GXEUKKGAq7pHME8ClQGXcaJAIWNZMVE0I7PVum\n82+B44FTNm++GHgD6BRUxR7TVxUmoaBqPkmI+BVTdZGCqGanv8txGFXFKe3V7rLYmaZs84nd\ndwwr3ZLQIy2UoYSEhcIuTlyupEfQris0UrqvAhaNK4vL4r82+VhqW+5ZpV88DHQBfgDmA2O2\nWnemTKSN6qKnKO+0+dszNJ3/0XxEUXVJoSwmVqDZph6LCO9ESyuR5VQSRE8CtSAhJQ6FXaLZ\nBjiYN9gVFCb7xz75ebrvChYNLb9YYEyXWfZr4Gig0QMP/BV4XbDupJS0jbq3IhaWtJGkSKS2\nArWge9MikhgVP1aJI+yoltxLhiIunaQ9LqVGMkVqAlMiCYfCLk4Cr6SHAB4we2vhfOVFouC7\nWf4W6eCa9BkMQlA1dC2X+LCnh+d5OP30LsBPwHzgYqCsrq5SPTPVhHN05kzSULX6TFJVKmnR\nf35JtReRR09rBLpgqSJuVxV8AVHiWjDXmD4oEmLB8u0HJE9Y3rRHAK8BG/OZDckY02Ir6npX\n2SXCpT9d5ShgKHALMBDY0/odGeK/kp6za6ZAB04rDkSJplqVWi9TaijQN9W6hqYkLXvD4g+a\naWnYsjfraWTeSh5SzQ8J7IX2/CfEDh27mLFPGP4XnfCNnXy0N2Ohri+rrm8GHlmLI+JYV2o6\njXjiPQi0WLHiB2AecEndT3smJ0ly0VTXzbTKCcXH0m50bE7yvMUmpP5ahjGUFefiwLk0Ct3h\nk6y7LKrJ3Fk+NJNyDS/+JCwUdsmlLdBx60/EJvPmDxIKk9SQ1ImEaqu4L36Ji7km5fHLq512\nOsrzzgQuAd4C9hYiaPWHlJ7doRQTlnSeZQkyvVdaic5QgUlj7hgq0NgLpcNcfDJ1uBzRLj2b\n0gs1CCZKR3aIbyVejUmSobBLLkcAnwOL4k6DOBK48KfOB+rKoGmuDVxkzJxUKvUQsCfwFfAe\n4I0b1yDq45+OzQVWlFZaIyRgiqne4iaKYHva4uyu+ogmd9MeWS0QYeRNTdg/OUQg81QLnRLs\nMikgKOxixnKBOEJYh83udZlknVArdOIu7azveLh9VZHFc2MlMBA4Gfh2/Pi5AN5919QuDMLU\npEHtykb0qCy6wSTCTK8jEHZRVXXgtPkH2n6hGlVzSFkf3AlL4Hlbgoh2b9y5EGKDwi5OLJ/j\nGwJ9twq7kr2SFhCma73WMVKnYWmXVDes2WPJR62rbeIJoAuwAMABB+CSS7wNGxyjiX6Y1sfy\nBAKztasTSeGp46adhi2GqEVcqhHU3Cz+YuChtPfUxUpU08gQS6qm8nm+UvHCSIgWCrvYsF+V\nDgIaAK/mLRuSMZbp3P9XnL8lraNdwtM2FDifaaOFwvO8H4BTAPzrXytuuGHxNtv00ZWxB7Hk\nb5dN2siiV+fYol/YYqdZEFWvtpY2eO7snAiD5lLXEjPsrvxbWdR2hKhQ2MWGfYH1COANYC2/\nJqpwcJ9jTE4P3A63y5JQhudMyv8KkqOP3gt4FXgVuBtobkjb3rRJIZl0mFrdtF07FKpfqMbR\nam6YxbfUnKmnFnybNvKhcamojS85mnZMdrJ73QgVI2M6voSUMhR2CeVw4GXezFE42G0k05Qj\nqQTJkTJJf3Xx0ZSMyfoypapOk6lUajVwFtAXOAT4GDjOXDdQtajyy/9rMixFUQhFuEgaTrIq\nPd3TDNoI4pCqPpxW56k9chFt0ZRHhp/u/EOTU+kjKek8yCxeHgnRQmEXG5b5fkegi/ATsST5\nuM8xknNjmbMtZ4hLK6YM7RpUu93zvNeAjhs3PgQ8BTwB7CDskiwuVVqJ8VXTS1veolktfpKo\naFXtqw61VjtCJyJdZCvc1EyGckdqIuv6KdeeYu7IUP4SUjRQ2CWR/sD3wAdxp0FCYZpUJCfD\ngigsVJ/JFFzd5W+UFEBgAqYmfqnbqNFYYH+gHfAxMBzwamvF+FrXTdqo6hIxZ9Uw0yqqlPL1\neH5hMWxgPp6Cmpi7crIX07YSGDArTUs5OLYeFrszSoqJPNixJBMo7GLDctU7DJgO1PLW4GJB\n8qhUmWI3ohwnSK0uiZynGir9Yj6wPzARuB2YXa9eJ6tFB7OgcVS6okSThkjca3cKI6AqY0t6\ngSW1tUIdI0nyioLYPYg9coZB8tkcIcQChV3iSAH9gelxp0Gyi8m7sthXcJAXWm1hKm93jHxn\nTrIY1Rc1wCSgC7AJeB+4AkBVld+0Rblq8xFz1qbn6zntmKhGkemFH18ddmnkI2hQqXCgdnHR\ndn622hzUsyUy2dLBPjlSb4UrCgs0bRO0YxNO/eAiJGd4uq+l6ALsCPxna4H8Z0XyjCRuxO2i\nfFEVg/b8saBKBJMKMS1KqhX/BNwMoHv3A4E3db3QZmiyBu2qSGtq2kNJaWg3WrCPsOntqfUX\nHetaQolS1aUJx+CWCIEFxDIRfOUSISsHixB36NgljsOAT4DlJXkFLGK0PpB2OtTqPFHxBBpv\nLsn4TahySjLYJEmBrcIiXeAfwJ7AlI8+eg24C2jp0DtP+GJhtY/ufXEcBMnU1HqKIqqHqnVV\noRwgdZTCpm3piCr6M3FN7IpWOjQR3KYcOToFahRleLAICQuFXeI4bKtdV2TuPQnEv/qrh15d\nOrTYWoENqaaaiytjKvkjMAw4bOv3ofxRUH4m6WPHUXWpuySHSfImJTmr1ZSqBNTuhU7zQRGU\nqCsr1X9dyFwWaFsMFLiBEcT0wuYT4eLG6yEhjlDYxYl6qWoIHLxV2JESQdQZ4gtpyrSrvUBE\ndaLOxHbBpP7rKcwEugJ3AVMAHH54u7pN+4LVNKmLBew5iCOjLR9BhfhV1AKSqlOVdzSN4qJQ\nM8cSyqLGLIOQi0wcqyMB2i6Lh4aQ3EFhlyx6AfWBWQC4FFsyaA0hz/B1sqKWcgzrvzat64VF\nVaJpsbXJ88YDXYGZr7yyELgMKBfKB6YamJXLO0LKKlRFsZbjCJuKabdnUa75L6LFtMtov0wE\ns9BFR2ZoQBYQFIIkFijskkV/4B1gXdxpkFhwURKiorIXVp02rVfnnpVdqfh6aBFwKND4wQev\n3X7794GDDfmouYmt2H01u8Fm0Q2SdNa2JQ2y/yLQU9RuF5Pxm7Mrm/Re03AFHn2pj6ZiuUDM\nTR2QTJKxnBX5pHQkKSloKOySxWF8HrYkEUWGSamEOiWk8qokcqwYWFebmAekhg7d/ocf3gRm\nAt5pp7Vyzlm0AC32j0X3SK6bJMhSdVdyoSghu6Kyb1EzUTO3VJEia505sVOmg+VoFLmfVO4B\nEdTHDF3GwoJCkMQChV1sqFe37YHuwCuFeQkjWcQyQbqfG2GnT7sjZUrAl1Bqiz8Cf047dvPn\nfwr8eevlRpUmoqSz5K8KMnf/Uszc5A6KXVOVk1cXbSit+BP7FUohRUDNLYK2iKy9UHcwo/Ui\nctOEkDQUdgmiH7AGmMtLG9mKaWpM1f21WVOBUPEdzzpJyqCuDlD3/tfz8N5729544z1NmvwX\n2CfoqV41Q22SgZmr3ptWNYYSyuoqZ44+g6WUL64LbMhkVbrUCiwWGM39kmUP5S7TE0UmUpiQ\nrMMvKE4QhwGvAjVxp0GSjzjlq2ojFbSOGbYtbSito6YqPACp8nIArYHbgQX169+6Zcs4YI1Q\nzNQLU4u+qDLpM9HJU304dZdJHJt2qbVMq65SEEexpWYemFV6NFz0X2AmkZ22zGWuYy8IIRbo\n2CUI3mBHVLRmgLrRlxGBnkdKwFRGWtGzuyzquqS6C8BXwAnA0Vu2XNC27SfAiXVTUpMUX6i6\nTVve0i/JAFMLmCpqPTC1Ia1/GQ13v02tGK0tSwGpOy7DmyGFePXLYvcJyRwKu3hQL44dgd2A\n/xTmdY2Ewn3utygeP056Ukkp36YbLb4UObCkfVcacdqbBuDDD+8HHgSmA3voSpqkg70VbaOm\nWpJKUwvYRyYwpn1jBLK12BdZNfrVM88hd0QbpVJYRS2FPhIfCrukcBiwDFgadxrFSrbmxbgQ\nfTjT3JyyrsBGbhS6WcFdHGicrcaNrwT2BmqAD4CJgLd+Pep+z0hwEF0ZT8ClX+oWMQgEH1TN\nAQaXztK6tndZPDPVUCkBRJrd1e4kXNuFJfLIFByl0EeShvfYJQXpl8SK7OpJRFwOrigUUtbf\nihD3mrSdp9zEZlpGdOxCWMdRPKvTrz/zvFQqNRi4BfiySZOBDk+YeoZ7+KTuBMZRw/oVxXG2\nHCZRg0rF1J6qBQJPALWM/2/kK4P02cBSpqQuPqb3V/FRCn0kaejYJYL6QF9getxpFDGFfhOM\n6NXZ12eh66x96hJNnQhYNIdW9IivnwI6A48CjwIvAR0Nmsnx8Kl+lSVb6J7ntQyCKYeU7h7E\ndGQXr8uvK7lrajT7ARI7ItmHEZZfs+gjxoV7F+K9MuRnqAv66kfCQmGXCH4LNAFe3fov34TE\n7qmk6t5RJ03bWjdO2mK3o9wTU5csxe2BGhTAOuBSoCtQD/gAuC6VarI1gt9ZSXWZPDnJb/Pq\nPjYrLbC69E4VbWoQSZap1aX4FvNVG9BdnQSqt8Ayufv8kzf5wosnIVyKjQfJd+kPzAVWbd0V\nV1YkUWidMPW1qjBg0AppAlcqHZ0h8V9RRXnmW+W0+Gkc5nlDUqknd9llbCqFm292iWAXUoER\nJOVncstgeFea3qp+j/wCYhApoKTUTTm7XxbEyIG1TL1L4FVIK98tJLALWgolT1JA0LFLBIdx\nHZZYsUglaZek/CTTKw+EbUjM7SmgYvly/OlPVUOGTAf23FrG23ojlGrJaHWtlIbqw4ntWhI2\nuWtizqK09QWuNqa/XfTktOrc/6tmrmpZS/55O+iBZO6lBX/Atp0AACAASURBVLqthJA0FHbx\nIF5wmwO/5TfYEQOBKkHdrn0dlrCnorvVZJrj0xHWA7juugaLFlUBH5WXTwK81au18lSKY0lA\nXNNUxZxWirmsmUKnESOMv90OzEQSuXQhFhzTsHxoiZGEjGEuyOcnQJI7KOzipw+wCXgr7jRI\nknF3qsSNUi3J8cqi2WNaJvMcHvXQB9xjjwHA8dXVAwF06nQK4NdJ1b3vLdTEL+YpjobWNrPk\nLFbUZgLrIbOkJ6lPl5imsCa3L4JUCjXfu4s2x4C5ILKCCXR5A/cSkmso7OLnUOB1oCruNEgC\nCauuUsL3FSOD2SvaslegPBJTdWniWWAvYNw339wNvA50F6rDQctqX6irpf64pZR1UpP9JlVX\n3TWL2JKktqnvuRMHWrkpNed+5qQU1GjaHFzOrkzcyhwhnUKmgSpQbZfAAScRoLCLn37C87CE\n+JhsMHGvP69ofTitHeVkmLlNz2otVTD5mah+odQ7cZdfciNwNbAn0GvgwHeBu4Ht62aodddU\ngaWdsRw7aJFiFokZ2Iqaj6RKVQ0aFsmcM03blsOtnmBqztBJ5wgZ5pNcKxjKIxIjfCo2ZnYA\n9gJmAOC1gNTFs351qr9Xcr9Sum/Z1boyuUtbbctdTWqp9DwAv0ulbgM+A8YBt1dXp8rLTbW0\nG7VCSq2SUr7iRI0m6hitq2fST4HvcbuKClVFlGuSntaadtotlpNQ+/khVIb+3kK89FlyLsTu\nkGKCjl0MiDNBX2AVsEDZTgiCZghPdwebdoLXuj7uy2EuZbSWoUtAu2wSY84AugEtb731auDD\n8vK+SjR702GXF03pSeNpH0ZTiy7J+HrI4ttZgoRVh5JC9XH06qKRa+csc3hZTmM5YzMxlUku\noLCLmUOBmUBt3GmQAiXQinNcjJMCmiJYirlvtC9cwqoqqoHUBRd0BN4GXq1X70mgTdA3wCHo\nmQNfoLh4MKr6UYOYxiGCD6cmoE3MNMgWk8w9EwumYye9Tr56M0G9QgoRCruYOXTrOmwaXkdI\nINrPx/YPzdLMaploXSxAtZiow9SFYAvRPut/D/wZ2K+2dkfgYwBXXumtXy81F2ikicabqY+S\nLEaQdDP1SGtS2r1G7fKoo8hOme+Z0ypUmIfLXZBFvnYl2e8pXEmadSxDwVFKGhR2cdIW2H3r\nkxN8Y5BoBKo0k4CwL7RJQRzRrugF5uxoIko223tAb2AEgAce+KpJkxOBeualosCUApHUUqBw\nlP61LGOZgmjloAsmG89lr7ZpKQ1pPMXXonvK+Z5kTpJ1f2KhsIsB/2LXD/gK+EzYzusgCUR7\nnriYcKb5WHKhwsaPgNR64IVbbf0X6QA8AlSsWPEQMAV4DfDee0/biiVyYHmtErJoVqlAoBOm\nPUBqW6ZDYPcITUczME7gEffLi2LXUt5E5qdWoc/9hZ4/SRoUdnFyqPBFJ3xjk8zRrh5KeMqD\nnOrKnfZszPopmp7R7WFVi1Etsx64HNgTOGjQIPzmN/cCv94aPzCH1Fa0uWmdRbv9ZlF7Ul+k\nttxbkVoMLONCKDtQGlhJ+MbyAZUfiYsV+h0RoLCLjRTQl99gR3KAKFZEleALKfVCKRXWxky/\nyOQiK9lU7kaFS7FKAE89hf/8Z3/gM+DiVKqh0NOw7p2Up6WMxefTzkn+UPsHxd46hLlNHTFJ\nVEXzfqIdXG0HIyeg1hI/cthjFvrET+1CsguFXQykL1JdgF8Lwo5vbBIZR5Eklokwl7gYbPa6\nESr6OJpq6Ndvny1bLgEuBj4EjnPOVqveJLNQlFlakWoSc1JYSV5LkcXeIYys0dqBps5qe2Ty\nFAOxJyntVf/V1sr1JTHamUxI8qGwi41+wKfA8rjTIEWAb9tohYVWiGjjBDo3kedCVXpGm7ZN\nsqnO9rKyu4AOwDTg2fLyV4AuzkmqiVnytMs4SRRCEV7awdQaclKLUlg7mXh4lvPEosbsR1Yr\n0GEYZ/HEdsg6BPYOElLQUNjFxqFchyXZw6IAtBslq0YqbNccgROtyzQcbU41LdipqvFnYBSA\nhQsPGzBgYf363tlne99/b8pWK3zd5/7A/ooFLEPttyj1xSJ6xIoWZa/WtXwGEAs46kL7KeGS\nUp6xi3JCChoKu3ioD/ThOizJGXZPRSoWVmOJy46mAqECqmQSvM56YqdOqRdeOHLLFsyahY4d\nRwHlQhNSeV/KYOs3d6gCKPICpbu6sjekxtSKXbEVS5ywS6KiqRnqQIjRpJGPyzaLsWlCcgqF\nXTzsB1QAM+NOgxQZ7v6KWtHfqBUcdhViCmsvZi+vLkpq05bCarv/EoAFCzB+/OXAQuBoXXxJ\n3mlln1TYXRmoQUQzzGQNumyBMjJqv9TXjpGhU3VZQR0NlyqW/AkhPhR28dAPmA+sijsNUqy4\nLDOpqkgraKQq4uQqTc8m2WFqV2vnWPpikSzaLXUGobw8df75HYCXgWeAl7feeKeqIotcCxxV\nVR1K5VWNCN3ga4fUxV1Tg0MnHLUt2nWzFDld3jJWKesX79m7YCfD6j4u7xF7GtSXoeCI5Q0K\nu3jgDXYkFzjqOf/yqoqPwIuv3e1TC4S6lGs1X2AEi4CQAq4C/gJ0BWqA+YB39tkw3Hhn0Sta\nY8/Sa19FaSOovbYPoOX4as1F0+DYFXyE1i21tE2HFVVSRzLRZHaoP0ihQ2EXA42BAynsSKxo\n9ZkkR0TVIq0bqnXV13b7LSz2uTadXmDw9N5PgAHpBdnXXkPHjpg0ydu82ZSzSQCZpn9P+EYY\n0f9T7UZxrxf1e2QciRZc6ziqHwzU+IGH2O4UWtBq31jI0PArQThieYPCLt+kUqkDgXrAf7du\n4blO8ol/ebUvCKp2lxonQuvR7BBVLNqX+cQykh/pb3kZwPvv49prf7jooiUNG3rPPGOyyqTh\nclctjnu1qk7N3N6oWkVLYEORiebthcWxj2HPMdNHGntJSZ0TkhAo7PKN53n9gDnAuq1beFEg\nsZCqC6x+lRbtFGhycUwRAmO6TOTaFrWTrtjTVHk5Ro7sAPwbqBo4EH37Yv58bXz7v9LomaxN\nSVKrulNFKmAqHKgt7ErFXZ1Ycjbp7Mx1T37Ek0v8LH7CISR3UNjlm1QqxRvsSBKQjChp7gyU\nFFrhosaxoy1pElKe8iu32jKOTfsVfwZGAw0++wwtWtR2734/sJMhGdXpVPM3DYu/Vzs+gcJL\nPByBZp7fhNhWhkaj/bCa6mq94bAEBrGfuo5NWISplIlai9qOJAoKu3zTHNgPmC5s4UWBxIVo\nHaW3qGrJxVWybAy1gulZbzWL7Nk4mV4dO+Kf/6w3c2Y3YEVFxeVAY6G8JI+k6dzT/fCraG5p\ngwSm6oL7cJlKqiZcdu0xF8M1sLnkXCSTkwkhJijs8k0fYDMwJ+40CJHwZ3etE+YiIEx2mqOD\n4r4aCOFuOSmy+4qkPqVDDulWU4PbbhsJLALwj3/UM3uK0gqsH9au20wej0ldaY+LvTlLDpF1\niSVmtoRg5DgRLLrYIxOSOyjs8k0f4HVgc9xpEOIiFLSiQYwQOBOnBDJItk5A8YXFiFKrQFGc\nmmzLyjBsWEeg9eWXbzz55HeAg5S6kiEHxZMTX4hrqZHHIbCz9iNl2iIl6b5um0UsTq3jcGXX\nYiSk0KkfdwIlx4NxJ0CIT6ruV9BJSArPfe4MVTiTupJK064ji76afXVSVGPp8o3POGPRrrvO\nBp4G2oXPyiQoxfS0FVWhlu6Iu96SmhYDSk07rthaiLB8nHVDkRDiQ2GXbxbGnQAhPqbJ3l5Y\nEhmiHLEbS2FzCxXH4mZpHR17cHHv7cDNwMfA7UCLVGq1QSZ6yp12/nbUHeGwyYhx7DmbErMU\ntktGlxNDCuiy3TRKEo46L3M56N5NQpIPl2JjhpcSEhcRjBZJw0mCKStmT4aiUPXATMLCPea7\nwEGe1/DJJwcBS4BzgHK3tT/VeFMXgsUM/eVddWwDMUlqsRVxi1o9k3VPtXUxoFrF8dzLyhqr\nKQdCihUKO0JI8E3ilknR5QZzuyGkLZ9SvuPD3oQfMPCGM2nFU7pbTir8v12DB7fdvHn7G2+c\nACwEjtZVlwwzk5gIzFCMpu2jZUtKeKxEK3P9BAK1juQyWsprmxO7YBreXOMo5lzOYUsTlIwk\nUVDYxQyvCCRpqD6cZPb4JpNpOgx0yAInUVEb+S8kW0ttVHWnLHh1fyct0HpMpVKphg1TF13U\n4vvvO5177jPAjFQKCxYgSPSI6flpSxrLNAhqWClh985axtwkBMUWtaHUDNV+ZaKZMqnrR8hW\nKDu8kpPkQGEXM/n88EpIJlhMF8kDCxvNRKruKqokZaR8fLmpjWNvxZSh3oLafnvcccfewHoA\n3btj+PCddNXVVkS0Gy0C0S7gLLIssEf28o5NuOTsqEHdcRe1WWzUFJ9XcpIcKOwIIXWwr6nB\n/DgCDHOt1hAyiTB3RDWpbkRd00hb3SJVLY2m+dTzjgP61dZi/vwVTZqMA5rocouGSSir26UC\nEXypUHrLpD6zaIzZ88m6OswKVHUkUVDYEUJspKfqUI6RuBjnmb9nxK7t3CVCoMq0Z+4YWZvP\nTKBs/vzT16//M/AZgClTvJqawC5IMsguo9WSktTLgw1mb8LxMKk9zTBz7amlJYFyMGn5kGKC\nwo4QkhPEBcfAMpYCWl9N6wu6ZKW1D6VdUhOSCyhtrAUeBDoCdwHrhg1Djx6H6jIUxZwvfKG7\nhdFkQIbqJhx0rTgUJukjpmTyCy2yyZKtaZXWLugDW9QaipYcYtR81HYkR1DYxQkNfFIoSIrH\n4kVJykksHNb4EY09NZq2sCmaqqUsk7opE20r6cLrPe8aoOLrr9Gjx8vA80Bn3cKli2UVar6X\nRkPtkV0ABQZUc3a/aqnK1RQn8pXQogIlOR6tiZwqP17/SY6gsIsTfmIjiUL0eEQpo7422S2m\naPZ2tfOuJIa0+YgFXAh0egIjWPIEkNppJ9x3XzegHPgA+Bvwa10rWgNM1U921WLJ1lexkhy3\n1HKRPqYC0WRTKtI9joGIbqK2UXUE7AahVMByHobNM3LdHJE7CUvyDIUdISQ0Yacl04Sq1QSB\nLpQaPEJKjnmGyiRdZiFwOHA0cBCwsmnTy4DGuoS1mJqzyzi7trZ4q/a+aGuZRL+lil0Nm7Zk\nQhZlk6rqpI8chCQNCrs44XWBJA3fovBPTpe1uUDPQ0uqLtL2UAlnF3Hmtkgry7B4nvcysC+A\nm246F1gEYOpU8bkKCOMmjoBJn0m7Ag+KVhRqFXOg+actkLlfZRFMYatHbjdsHPGjiHrsCp3I\no0qSBoVdbPAtRBKIfYrSzsdSdcnPkFSFyaWzNG1Xjb5dJ1VXjbew7zjLQqGoySyOVA2AP/+5\nI/AgsP7UU+eVlXmvvurYaKCOFIvZRZ5lozuqTWU5lGItF9EjSqXConAzJ0UMhR0h5BciqB+L\nwyRFNhVwEQeBu1xWSNU0AhF1m9brQl3fSxt/HXAlsAewAED//v8GOis5p1sRg6h/tT0VhaA4\nGhYZrYotbRAxMbWAo55zP7L51EbZ9dio6kjSoLAjhERENK5M1pHk20GRJhZJFDaZXNQSNVwo\n4SsVXu55p3se5s075rDDPq5f/2/Ar+q2or5WLUPtkqXJyZOkJxTx59gRUzIRKkoZRkiDEBII\nhR0hJAqWdVh1LnfRT6HKu+SjTU+1r0wLl3YzUith7foVALp2xSuvDNiy5SDg26ZNce2121gz\nEe1ACAae1hOVLDpVUmu76TuFjqamy6GxKGC1ofxrO5f0CClcKOzige49KWgku0XVE2p5yfey\nu1+mydXeiqWKSUbY/SRLTEsVU1v+CLy49bkKTJ68CBgK1At6UENt2i5B/GKS0yYOoFrGLnfs\nYxJBEvEySEguoLAjhIRGlAjijK6aSVofSOsPaX2+QOEVuDxqESXRhIXn/FsLWmstJTxXgc8+\n22XcuClNmrwH9A/6BhBto+Iyq7oxrHR2EXCOWMZcEr6JkneBpxMhyYfCLt/wwkGKGFVkqAUk\nX0q1lLSFxS2SOIgg2iwenlpM0q/aZVBLzrZkKipw1VVYvHgu8DLwAtDFIXm1QOCwayOkBKS9\n0nYXk1W719KEhWyJyyKj+Ial+HqUECjsCCEZoc7o2n8DxYq7Aog8GUhiTnW2tA5ihJR8Lejv\nVUfpf53dcccRnrcvkAIWlpV5I0bsvFW/BjYnZq59rf4L3bEI1G1hEdd8A8uQeIlXXVHbZR0K\nO0JIDrErDNGBMy3Laud+VSqZZI17npaVUzuq9WiKY0nsQ+BIAC+9hLlzPwPGA1i3zs9NG0fd\nHsoCsctZmJ8gUTEV8/N0PCJ+77Tmn7olD5rA4mjmummJHOngCHZvtsj6xwmShsKOEJI1fHkk\nmlXqa+0FXXWhAhFnI1Hq+XrRUl7bkKh1xCAuKdk9NscE0L8/3nuvyYMPDgNWNm16BuBVV0O3\nSA1FBmVrbvaPTqgpP/PWtRGk8yfDJhwblShu5eF+hpNCgcKOEJIREdwLSZekN/ovTLLMx+Ku\nqTrAPmOpSlSsKyWGupJUa0ZqbTA78opwvXo47bSOwO3AjcDH5eV4/nmpFVOedv/Db8jxkIWa\n8h21r6VdSZcHNpG52RNZ1RWZz1RMfSGgsCOEZAVxMhanPem1WD4woKlksO8lbLdIBElASP9q\nsxUlaUpZRDbZS6Lss2gCscpG4DqgPTATwMCB6NsXc+dKAU1ja0cUzaZkZK0ZhL1YBOmvZpgL\n8UFBQ4oSCjtCSEZEnnpNqkLdbpIvFukGs2NnryVVt5t5Fg9MEnza6mqnJDn4needC+xRXY1t\nt8Vvf4s//Qmff27J1tKK35B0sAJVV6DPF7g9UEdq83TRghEko7YtNaYlbCaNZp4wIYFQ2BFC\nskBYa8ck6RD0bSkQbuq3iySxirRFUlom8WffYsFug6WUb2zRJuyX/wzA00/jtdfefuSRzW3a\n4KKL8NNPUkCximlw3HthXxLVZmuJ495uqOCZFI4LeoQkD1DYEUJyiMn/CHSJpC1ar05d2fQr\nWuzAwJy1BqSjgvELp9EKxEC3TD9cvXodCJwM4F//Qvv2uPnmhobWw2pQe/l0AVP3LWatdFzC\nCi8Xe8909DMhbKNZj09IhlDYEUJyiDrxm5wqUxXP/EsPomoUV4Qtc6epsKQRxRYt0SyrxvY+\nRsMDngIaLFly/qpVuO66Tbvv7j3yCLbmL657+mosUECrBiGUUfpfAlG1cn7IViYZSjdC4oXC\njhCSW0xqwKSBJKMrhKeV8dQuLWKaloa1RpGjsrTEcSkPoBq4A8CSJfjjHzF8OPbfHzNnwtx3\nd7fMLmGRgU7Ng1SiFCMkDYUdISTfqHOwuoQqagivLpbIqbpo20JdEam9Kc2uY8Qq0gsYhFTK\n/uMTdXHpKZo3x8SJ+OwzdOmC/v1x9NH46KOw4kZsRXJVPeX7CENF9mup40MIySkUdoSQfCNN\n8xYFFs3Tktoy5WCKELjg6Cm4JybpTm18dX3ZyC67YMoUzJ+Pmhp07YoRI7Bihb1HpgQsuKxK\nh8LeuotKzq5MzErAhIjXJORA4oXCjhASG1qFl95octpUbw+6pzj9VVG/rqMo0cZH3fnSJMss\nYlENYsc3zLRWn5RwqmtXvPjiYTU18++/Hx07XptKNXfThZLCUzvlCY9NhFJ1kjB1F+gR5Ga2\nFFXsyizzBPx3UJYy0sSncEw+FHaEkHxjmeZNIkMtIyktNaBX91lOi0PmRzAJRFMaduvLLmUk\nt08MEjg42gLTgf2AUzZsOBlYAuD//g/V1ZZ21RGAbthNhU1GmrrL3yLuDRwcsXXTXm1bln8t\nzWVuRoYKkiOF5G4hkyKGwo4QEhuqyLDIHam8XbdZbD+4mU92IWgqLDYqWUqRNYc9N3G4aoGH\ngU7ADQCuvBJ77oknn4RukVfVkeJfqSFT5pbxD8zZVMBRlIjFREVuSs8xZj4lkXaoE67JsjJE\ntP1yDYUdISQpBIoYbflQ3o8kIu0NaWcgqZb6r6q6LA6cqj4lyWJJT0rSH41NwCRgu59+mrRk\nCU49FQccgNde01Y0+Xbia5MQVPsO3SCj7iD4ilwrFgM1mWSmqg3Z00sO2qGLJRNSfFDYEUKS\niGWek2SBSXOYZKLLoqcqF7SeHLJhtLgISpN9qN2e/rsKuAjYfdMmdOxY26fPv1MpfPyxqWnt\nkLpLTL+81pvUBrEgHTtTTO12u+w2YdeReXOYSkTb5dkZLUEo7Agh8WCZLH0ryz4HmFYkHVuR\n6rrYY6a9fraSnySVV2WHmrNJNonlpYDajqc3fgHgoYd6AI2Amr328kaM8FasgFWoWSy6wEYd\nUTse2BHR8LNUiSDCAi1hkjlcfs0nFHaEkNiIdq33xY1lJVE1WnxXya/uF4iQhnYR0O69idLE\nZLmp3USQkDJtF8dnvucdDhwJ4L330KED/vpXrF6tzVm7jqkmaVKoWonsD3JkP8+x7y7epzaI\nvWnH40VIQqCwI4TEhmXC9hx+EUvEN/nCtiVW1zZnWaeTqvsNqb6aqAz8VE1Onj15d50hdeo/\nAObOxT334JFH0L49brutgb1V812P7sfFxfTKrlqyD07Yk0r9l9ouAnb1TLILhR0hJB4yXMuD\n9eEDdf3O5BiZzCe1RdHN8v0/taTahLZp02t3JJ2htqJafamyMvzpT/j0U4wdi2uu2dy2rffo\no15trUUMRZuSRaPOHieLOknbROTTTJVxqh9J3AklqUkmUNgRQhKKZRKVVJGoYNT5WKu0LNOz\nNJdbCljiWJSfI9qJ0KTe1IQlVVfHR2zYEBdeiCVLMGQIhg17r149zJgBZy9Qa0BqFaE2PVPC\nLt03lQzM077RUj3XYrQU4HDlGQo7QkjhobXl/F0uC7Ip4RY0i0tnDyitkGpbUatI5p82DVME\ntbNSMcskKmvNFi1w/fWtN258H6jp3/8loFvGnoqoxnzBJwpuLS7Dq21C3Bgx4/ConxyIhUDL\n1iUCRzsUFHaEkILBvs4oEnYKUYWRZ73JLz1LSTpGLRDYnCVhrbZTrUcpyTrOnLIkqrbyleeN\nAMo+/PDwY46ZX6/eVMCrrJTaUlsRt1icLSltNXlLlVBHMBNn1AV1hHPdYo4oRJEUONSF2Kmc\nQmFHCCk2LPaVuoKpzgqqw2SfOUyumym3wBy0fp57DqYCqKvz6qSx117497/71Na2B9Cpkzdq\n1HaWPgStUIv5mPS3i8jTdlntprvGiqAAAqtkRVVkS5okTeJkK5kCldFxQWFHCCkYXJwSi7wQ\nt1siS4tHWjNPG0cVT4E9chc0UhMucVC3X4Hq6jXgQACPPIIXXlgKXAZgwwYxSLaMq1DWXeZk\nLneKRljkyGs0jbBJ1meXwjVQcwSFHSGkqJDUmLoXVhlk2igpP1G6aWc1VQWqqPpPaxOKi56m\nVMV/3UWMVCvdw9SgQfjww+Z3330u8HWTJrjnHmzZ4hjNxVyEMt+bhijrE3aEgIFVspJktnqa\nT4njeJpRcuUfCjtCSBGiijC74JNcOrWu+5KcapJpDULHjqTqPvOLoMlbykHK2d9u0akAUuXl\nOOOMDsCdAC6+GHvvPSiVSimWpBZtboHCNDIublwCHZ2krZlGQ/uuIbFDYUcIKU7sJpZ2i9Zm\nCzUBSxN2aiuOebq8tuOSudYjlJZWU6nUOs+7FsCSJTjiiEeANwHMng1B3mkja01HO44jnBJw\nKZ9hc7mIVhx6zieUmCuyvicZCjtCSKLJ7t1RgdaCVjqkdD9fZg8SmJIkpKBTRWr+YoaegLhL\nXeIUI4svtELWT++Xvdtvj1tuaVhZ2fOUU9CvH446Ch98YO++3Rw1DYg7UramIdLWzYW2iBBT\ntFQjVE+CurXoexIvFHaEkOQiLRGGrWiqK62WWoSFaZd2dVW70GmpaJJWJrSK090OFPOUCqTq\noqm7++546CHMnw/PQ7du3qmnep9/rm1OVdKmTMReWLqgBs9ETORi0TCUn5qEJctsqbFQb0/T\nGj2lYdahsCOEJBdfJ4WqZVolVLdIG1UPTC0cKC9MHqFUS5VoUg4unXKUg9DdJugikuT4++yD\nF17AjBlYtAh77IFRo/DDD5a6LqouF/iWYRYVjOPhDks0qZehQMzW4TCdq1oif0gjYaGwI4Qk\nmmjTnvpa612Jr02CTK0SQTF4ytemSJrVfyEpPKlpSSZqUddzA3MT0wieqg85pN477wzavPnT\nW29d06oVJkzA+vWW4v6MbtFG7gPrizY/uEsfIxyyUFUKTq/kR2qrBrDLuyzDhgiFHSGkCDGJ\nNq1rJSEtSprmnsA5SWpXbcVvy6t7R5qkhMRMLKFM3QnMxzIpmvRTrec9A+wNjAZWXHEF2rfH\nnXeiulqKmbk2smx0lICOrWceisIikDwbtyULhR0hpHRRxZNLedM0L0aTtquFXWSfNriL9aiK\nKq1w1PYxMBO/oWrPuw/YecMGXHABLr8cnTvjscdgcCJddLC2d2q2oVYAHVvPvErpqBZ3vZ5F\nbZ2QhgoCCjtCSAETdglGOwFIESz+nFZUqYuDUIRIYFbaDC0VLTovMJS00RPQBrGMsOd5aNwY\nl1yCpUsxaBCGDUOPHnj5ZT942KNjT0A93FmZ0cOaiyqlpiqy2F+uomYdCjtCSELJfLrVBjTt\n1UofUxxTTFEkibsimEympt3NEjFVaTD9ZDzrQwaSi2ZrumXL1A037LJxI/bbb8sRR6BfP7zz\njmOnfNQjbrLxTNXDthihlaygHfMC0jcFlGoJQmFHCEkoLtNthmrJUt3fJc7B6nwmul+mUOos\nblIwakU1uNZxVDMU/TyximlKdle9FlYAqXvv7QI8NXNmbc+e3sCB3iefZEUwieain7BazLdL\nLXst8fNGoG2cQHJkreVzFbVE3EEKO0JIcsnuFd+y2ihttxsqUuFAaeiCqMPEhlQ70H9h0ZHa\nUKr95pd08eTcu7MIGAIcAGD1anTpghEj8NVXqCs6LTDZ6wAAIABJREFUtQ25+5q+gFMj2Ksk\nB/UsKpT13AJKtTShsCOEEBvSNBZqSpOUk/iv1kVTFY+j6jJVcdSgapKmfAI9Dz/OHCA1YwZe\nfBELFqBjx5tSqe2yp65UJ1JMTxo0Vb9mQuQuZN0uyrNULQI9VwRdcIHCjhBCfkG79Jl5NOlf\nVWqIOkzcImksU0PpWn5Y+9qxVMteRsrfvowrCcpf/j3sMMyZg4ceGr3HHksB75prsG4dQupj\nU/7aLTmduZNj+7kc4py2HrbdDHVtckY++VDYEUIIkIOZQ7vWZpIjpi1awyzzVLWyQFzDNaUX\nVvKmUqlUvXoYMgQfftj87rtx111o3967/XZUVQVXDN9N7WBmUeplEso9E5e+hxLlycely0XT\n2VxDYUcIIYDDTGmfmLUzk2N56a44k8CyCD6E8avEG9TsGWobUrGMzP+216+PM87A4sUYPRpX\nXYVOnTB1KmprA4Pbyfpk76ioXOLkWojEuLAY7b1gvxMgkxaJSGELu6qqqnfffXfmzJmVlZVx\n50IIKXiy5YJIos00x0sTlVbqBQsmZaNUS9SLopgTl02lkv6/6upqUNeDUm3cGBddhGXLcNJJ\nOPtsdO2Kf/9brSKJTtMAuktPF8S17EyCZJ5JqKFOgtxRD5A9K+2bIgkdKQ4KRthNmDBh5syZ\n4pa77757hx122H///fv169e2bdv99tvv/fffjys9QkjREFm+iMLIbk5otZd762EXQ6ETc9Lk\nqspKONtOod2p5s0xYQKWLMEhh2DIEBx4IGbNsoeyCFkY5FTYrFSlGxaxuVJ2mFzsN5diJBoF\nI+yuuOKKl19+2f932rRpZ5111oYNG0444YQzzzyzV69e77333iGHHLJ06dIYkySEFDpZn2xU\nGecjKg+xmN2o86xfKSzFl177f03V8zrX7rAD7rgDn36K9u1x6KE44ogehodXXJbwsrL0maEa\nK1kxF6HjpTlQ+aFghJ3EqFGjmjdvPn/+/Geeeeauu+7673//+/TTT69Zs+baa6+NOzVCSGGT\noUTI2+xuX6AU8zHt1bqMEZZfM+pymzZ46CEsWICGDecCTwDep59qC1rMPJOfZ9LT0TLN9T1z\nhGSFghR233///eLFi88555zOnTv7GwcOHHjccce98sorMSZGCCllLIowwu3kFsRaFm1nkmii\nO+goVqRiWbHH6tClC559NvXmm0MOOUT8TmN37IMpJawmH9gjqjpSKBSksNu0aRMAUdWl6dKl\ny3fffRdHRoSQ4iHrTk9WkJSHXRGalJ/UtZRh6VNtOtR2F/RC6oADMHMmnn8e77+Pjh1x4YX4\n/ntT8mKcsJlE0NOOKjz7kjcZZOUuRpIfClLY7bTTTs2bN1++fLm0/euvv27atGksKRFCihvx\nHjUL9nvjTGEdkSLY796L3Iq2XW3T2sIR2pKrHH443n0XU6fipZfQrh3GjcOaNZbc4CDUJMPS\nXoBooYYrFApJ2H355Zdz585dsmTJTz/9NHLkyPvvv3/Dhg3+3k8//fTxxx/v1atXjBkSQooV\nRwERavLLRNWFCh5WbkZOKbBH0vMixoqpFAYPxsKFuP12/P3vaNsWkyZh40bHNGKkiAWi2q8i\n7mxBU0jC7tFHH/3Nb37ToUOHVq1aXXfddUuWLHnxxRfTux555JH99ttv48aNV1xxRbxJEkKK\nlVC2UIYBtSuMjitf0v1zmUy99oVO7V2DLump9qe+YlkZhg7FokW48kpMmoQOHXDXXaiuds/f\nlKdaIEd2VNEsVlLAFRD1407AlSlTpvwssHr16p9//rlly5bpvT///HOLFi0ee+yx3/zmN/Hm\nSQgh7mQ+X2oNsGwFT4swVSZaxIpladhXnH6ZVN1fmNXTsCHOPx/Dhl3etOmEsWMxaRKuugon\nnYR6TsaEo66icCFFQ6o4zuZ169Zts8029dze52G5++67zzrrrLVr11ZUVOQiPiGEuOBrrMDt\n9m88MQVHXd2mjelSwJSPKckQ6a1ahRtuWH/99ZVAl3/9C8ceC7fv87PfiVgc8yDJJ1VVVQ0b\nNnzjjTcOPPDAuHORKaSlWAsVFRU5UnWEEJIE7E9vqEpOtccCg9uRnj8I9fCEuDosPYobaAH6\nhQGgZUtcd117YBaA3/8ePXti+nSXzB3zFPcWxxIqKUEohgghJB5CqQfLM7D2WhEKR7sp3v7Y\naSCO2i6VSn3jeed6HhYtwp574ogj0K8f3norQrbRhpSQhFMw99gFsnTp0jPPPBPAdIcPcD7r\n1q274YYbqqqqLGX4E7SEkGyhLkq6P+JgKql9OjVbj3GEQrp5LlRDlmdsJVfvl+C7744pU3DJ\nJRg3Dr164aijMGECunbNvBf2PAlJOMUj7NauXTtjxoywtdavXz9v3rzNmzdbyvzwww8A6tcv\nnrEihCSBwFXIUI+1SoXDarssErheHPY5D6kvdQp36oTHH8ell+Lyy9G9O4YMwfjx2GOPaJkT\nUgQUycMTADZt2rRkyRIAXbp0yW7kN998s1evXps3b27QoEF2IxNCkkbmXxGSRSILO62qsz8r\nkF0hGCEBU5wQ5d94A3/9K954Y8qWLad//jl22801XUJCwocn8kGjRo26dOmSdVVHCCHZxf3W\nulB3gEkPN4RKw/F3NdwxrcOG6lHofHr1wqxZv9uypQtQtfvuOO88fPNNuAilBx8TKT4Kb3nR\n87zKysply5atXbsWQPPmzTt06NC6deu48yKEFAOZ3OuWLbJuniFII2q/oyRyQ9mqHi2rVzwP\nnodnn8UVV+CBB3Duubj4Ymy3XbSsCCk4CknY/fTTT9dee+3UqVO/++47adeuu+46YsSIMWPG\nNG7cOJbcCCHEkcD7yWJJIyGrzxJhs/pFCKZSOP54HHssHnsMV12Fu+7ChRdi1Cg0a5ajPAuX\nZB53kgkFc4/dN99806tXr8rKyg4dOvTq1Wu33XZr0qQJgDVr1ixdunT27Nlff/11165dZ86c\n6f8cRbbgPXaEkHwS43MPmRP7Qxty61u24O9/x9VXY/16XHwxzj0X22wTS3pI2B2cJBOSfI9d\nwTh2V1xxxfLly5944okhQ4aoe2tqau6+++5zzz13/Pjxt956a/7TI4SQLBKvtousP/x79bKS\nfNg0vK2/QluH+vUxfDhOPhn33IOJE3HLLbjsMpxxBho2zDxDQhJIwTw8MW3atFNOOUWr6gCU\nlZWNHDny97///TPPPJPnxAghJCyprcSdSJax/NZFfjprVIENG+K887B0KUaNwvjx6NgR992H\n6uo8pCTC70MmeaBghN2PP/7Yrl07e5nOnTt/++23+cmHEEIiY5/dY5/+M0nAUjGstsv+OGyz\nDS6+GMuWYdgwjBmDPffE1Kmorc1mE4TETcEIu5122mnBggX2MvPnz99pp53ykw8hhGRC3tRb\n7qzBsJGTYlY1a4Zx47BsGYYMwdlnY++98dRTyOUzzkXpzpLEUjDC7vjjj3/yyScnTZqk/ZWI\n9evXjxs37tlnn/3DH/6Q/9wIIYTYCaXq8qGEtt0WEydi6VL87nc45RT06IHnn89FbpblaUJy\nQcE8e/Xzzz8feuih8+bNa9q06f7779+6deuKigrP89atW/fFF1/MmTNnw4YNvXv3fuGFFyoq\nKrLbNJ+KJYSQfBLhNyoyYsUKXHst7r8f3brhmmtw2GGBubn/HEihTLIkFHwqNgu0aNHirbfe\nmjx58kMPPTRr1qyamhp/V3l5eY8ePYYNGzZs2LCysrIYkySEEJI5+udbc8fOO+POO3Hxxbj6\nagwYgAMPxIQJ6N3blJtjVC6/klgoyA8TmzZt+uqrr9K/PNGsWbNdd901p14aHTtCSNIo0G9E\ni/zrt/njs88wfjweewz9+2P8ePTsmdfWSYGQZMeuYO6xE2nUqFGHDh26d+/evXv39u3bU28R\nQkiuyc8TAH4r0drKQpIdO+If/8CCBaiowIEH4phjMH9+RgEJyS8FKewIIaTEif0rUVRcRFXY\ntGPrY5cuePppzJ0LAD16YPBgfPRRPJkQEhIKO0IIIcHkR0qKrURoLstJdu+O557Dm29izRrs\nsw/+9CcsWpS14ITkBgo7QghJNIXyRWjZFVUJ8iN79sQrr2DWLKxYgS5dMHQoli2LOydCjFDY\nEUJIouEXoSWC3r0xaxZefBGffYZOnXDGGfjyy7hzIkQDhR0hhCQdqrqk0L8/3nwTzz6L+fPR\noQPOOQcrVsSdEyF1oLAjhBASAwWxvqznyCMxZw6eeAJvvIH27XHBBVi5Mu6cCPkFCjtCCCk8\nIkuiQpRTScw5lcJxx2H+fEydiunT0a4dxozB99/HnRYhFHaEEFJoZPJNb8lZ1U3gN7aEJpXC\n4MH44APcdx+efx5t2+Kyy7BqlVvV5KlVUhRQ2BFCSIGR4eMUBSenki4B69XDH/+Ijz7CnXfi\nySfRpg2uuAI//RR3WqREobAjhJDCI9FCpzQpK8Mpp+CTT3DLLXj4YbRti6uvxpo1puJJV6uk\nYKGwI4QQQrJE/foYNgyLFuH663H//WjTBhMnYu3auNMiJQSFHSGEkBxSijeTNWiAM8/E4sW4\n+mrceSfatsUNN2D9+rjTIiUBhR0hhJASJbeis0EDnHMOlizBFVfg1lvRti1uugkbNuSqOUIA\nUNgRQgjJKaV+M1mjRjj/fCxdiksvxY03ol073HorNm6MOy1StFDYEUIIKVHyJzobN8aoUVi2\nDKNHY+JEtGuHO+7Apk35aJqUGBR2hBBCSF7YZhuMGYPKSvzlL7j6arRvj8mTsXlz3GmRooLC\njhBCCMkjTZrgkktQWYmRI3HllejQAXfdhaqquNMiRQKFHSGEEJJ3Kipw2WWorMSf/4yxY9Gx\nI+69F9XVcadFCh4KO0IIISQmmjXDFVfg889x+um4+GJ07Ij77qO8I5lAYUcIIYTESvPmGDcO\nlZU49VSMGYM99sADD1DekWhQ2BFCCCEJoEULjB+PykqcfDJGjUKnTpgyBVu2xJ0WKTAo7Agh\nhJDE0LIlrr4alZU46ST85S/o1Al//zvlHXGHwo4QQghJGNtui2uuQWUl/vAHnHceOnfGQw9R\n3hEXKOwIIYSQRLLddrj2WlRWYsgQnHMO9twTDz2Empq40yKJhsKOEEIISTDbbYeJE1FZiUGD\ncM456NwZU6dS3hETFHaEEEJI4tl+e1x3HSorMXAgRo7EnntS3hEtFHaEEEJIgbD99rj+elRW\n4oQTfpF3Dz9MeUdEKOwIIYSQgkKUd2efjb32orwjPhR2hBBCSAGSlnfLluH443H22VycJWko\n7AghhJCCpVUreXGW8q60obAjhBBCChx/cZaPVpQ8FHaEEEJIUZB+cvbzz/HHP+K889ChA+65\nh19rXGpQ2BFCCCFFxHbb4aqrsGQJTjwRY8Zgzz35o2QlBYUdIYQQUnRsv/0vX2s8ZAjOOw+d\nOmHKFMq7UoDCjhBCCClS0j9Kll6cveAC7LEHHngA1dVxp0VySP24EygAGjRoAKBhw4ZxJ0II\nIYREZFtg5Jo1rYcPrxw+/G/A6rjzKQLS8iBppDzPizuHAmDBggVbitHBXrNmTb9+/SZOnNi6\ndeu4c0kiP/300/nnn3/jjTfusMMOceeSRL777rvRo0ffdttt2267bdy5JJHly5ePHTv2zjvv\nbNq0ady5JJHKysorr7zy3nvvbdSoUdy5JJFFixZNmDDhoYceSqVSceeSRBYuXHjzzTe//fbb\ncSVQv379rl27xtW6BQq7kmbVqlXbbbfdggUL9tlnn7hzSSJff/31zjvvvGjRoo4dO8adSxJZ\nunRp+/btv/zyS34w0PLhhx/uvffe33///fbbbx93Lklk7ty5v/nNb9auXVtRURF3Lknk9ddf\nP/jgg7ds2VJWVhZ3LknklVdeOeaYYzZv3hx3IomD99gRQgghhBQJFHaEEEIIIUUChR0hhBBC\nSJFAYUcIIYQQUiRQ2BFCCCGEFAkUdoQQQgghRQKFHSGEEEJIkUBhRwghhBBSJFDYEUIIIYQU\nCRR2JU15eXkqlUrmr90lgQYNGnB8LKRHhuNjIn3+lJeXx51IQmnQoEG9evXq1+dPlutp0KBB\n+hIddyIJpUGDBrz4aOFPipU6y5Yta9u2bdxZJBeOjx2Ojx2Ojx2OjwXP8yorKzk+JjzP+/zz\nz9u0aRN3IomDwo4QQgghpEjgUiwhhBBCSJFAYUcIIYQQUiRQ2BFCCCGEFAkUdoQQQgghRQKF\nHSGEEEJIkUBhRwghhBBSJFDYEUIIIYQUCRR2hBBCCCFFAoUdIYQQQkiRQGFHCCGEEFIkUNgR\nQgghhBQJFHaEEEIIIUUChR0hhBBCSJFAYUcIIYQQUiRQ2BFCCCGEFAkUdiXKzz//fMEFF+y+\n++4NGjTYaaedRowY8c0338SdVIL46aefxowZs9tuuzVs2LBNmzbHH3/822+/HXdSCeXCCy9M\npVIjRoyIO5Fk8eKLL/bp06dp06YtWrTo16/frFmz4s4oQXz66aennHLKjjvuWF5e3qpVqxNO\nOGHOnDlxJxUn1dXVY8eOLSsr22+//dS9vFzbx4eXa4mU53lx50DyTVVV1QEHHDBv3rxBgwZ1\n79596dKlU6dO3WWXXd57772WLVvGnV38rFq1qkePHp9//vlRRx3VvXv3ZcuWPf744/Xr158z\nZ87ee+8dd3bJYu7cuT179qypqRk+fPh9990XdzpJYcqUKcOGDWvXrt0f//jHTZs2/f3vf1+9\nevXMmTMPPPDAuFOLn48++uiAAw4oLy8/99xz27dv/8UXX0yePPmHH354+eWX+/XrF3d2MfDJ\nJ5+cfPLJixcvXr9+fbdu3ebOnSvu5eXaPj68XGvwSOlx8803A/h//+//+Vsef/xxAKNHj44x\nq+RwzjnnALjjjjv8LU8//TSAAQMGxJhVAqmurt533327du0KYPjw4XGnkxS+/fbbioqKbt26\nrVu3Lr1l8eLFFRUVI0eOjDexhHDSSScBePXVV/0tCxYsAHDIIYfEmFVcrF69unHjxvvtt9/i\nxYsbNmzYo0cPqUCJX64Dx4eXaxUKu1Jk3333bdq06aZNm8SN7du3/9WvflVbWxtXVsnhggsu\nOPTQQ6uqqvwttbW1jRs33m233eJLKolcf/31qVTqxRdfpLATufHGGwG89NJL4ka+s3x++9vf\nAhDfX57nNWvWbPfdd48rpRj58ccfR48enR4NrXAp8ct14Pjwcq3Ce+xKjk2bNi1cuHD//fdv\n2LChuP2ggw767rvvKisr40osOdxyyy3Tp08vLy/3t1RVVW3ZsmWXXXaJMauksXTp0vHjx591\n1lk9e/aMO5dkMX369MaNG6dXFTdv3rxmzRoAqVQq7rySQqdOnQAsWrTI3/LDDz+sW7euc+fO\n8SUVG9tuu+2kSZPEq40IL9f28QEv1zoo7EqOr776qqampnXr1tL23XbbDcCyZcviSCrp3H33\n3dXV1SeeeGLciSSIM888s0WLFtddd13ciSSOTz/9tE2bNh9++OFBBx3UuHHj5s2bt2/f/sEH\nH4w7r6RwySWXtGzZ8uSTT/7vf/+7cuXK+fPnn3jiiY0aNRo3blzcqSUOXq4jwMs1hV3JsXbt\nWgBNmjSRtldUVPh7icjs2bMvuuiigw466Kyzzoo7l6Tw4IMPzpgx44477mjevHncuSSOVatW\nrV+//qijjurZs+eTTz552223VVdXn3766Y888kjcqSWCzp07v/XWW9XV1b17995xxx27d+++\nePHi6dOnp5doiQgv12Hh5RoUdiWLujDkeZ52e4nz6KOPHn744V26dHn22Wfr168fdzqJ4Lvv\nvhs9evTRRx89aNCguHNJIlVVVV988cX1118/adKkQYMGnX/++W+//XZFRcXo0aNramrizi5+\nPvnkkyOPPHLt2rU33XTTc889d//99zdt2vTII4+cPn163KklFF6uHeHlOg2FXcnRrFkz6D7q\npe8Eatq0aQw5JRLP88aNG3fSSSf17dt31qxZ2267bdwZJYW//OUvVVVVkydPjjuRhFJRUVFW\nVjZ48GB/y4477njkkUeuXLny448/jjGxhDBs2LBvv/32rbfeuvDCC48++uhhw4bNmTOnoqJi\n6NCh1dXVcWeXLHi5doSXaxEKu5Jj1113rV+//hdffCFtX7p0KYAOHTrEkVTi8DxvxIgRV199\n9Xnnnff888/zAurz4osvPvbYY6NGjapXr97y5cuXL1/+9ddfA9iwYcPy5cvT802Js/vuuwOQ\nbvdu1aoVuHYGrFu37p133vntb3+78847+xu32WabQw89dMWKFZ999lmMuSUQXq5d4OVagsKu\n5GjQoEGPHj3mzJmzYcMGf2Ntbe3s2bNbt2696667xphbchg1atQDDzwwceLE22+/vaysLO50\nEsSMGTMAXHPNNa23stdeewF49NFHW7duPXHixLgTjJ8DDjigpqZm3rx54sYlS5YAUO+CLzU2\nbtzoed6mTZuk7ekt6vYSh5drF3i5lqCwK0WGDx++YcOG9Ldtpbnnnnu+/vpr/ipUmmeeeea2\n2277y1/+Mnbs2LhzSRzDhw9/ri6PPfYYgN/97nfPPffc0KFD404wfoYOHZpKpS677LLNmzen\nt8ydO3f69On77LMPhV2rVq3atGkzd+5c0Zz7+eefp0+f3qxZsy5dusSYWzLh5doOL9cq/Emx\nUqSmpqZv376vv/76cccd1717908++eTxxx/v0qXL22+/vc0228SdXfy0b99+6dKl5513njoa\n6W9qiCWrxPLzzz+3bNmSPykmMmrUqFtvvXXfffc94YQTli9f/vDDD9fU1Lz88suHHHJI3KnF\nzz//+c/Bgwe3bNnyrLPOateu3TfffHPfffdVVlZOnjx55MiRcWeXb2bPnp3+im8AkyZNatWq\n1WmnnZb+96KLLtpuu+1K/HIdOD68XGuI77uRSZysXbs2/avJ5eXlO++88znnnPPjjz/GnVRS\nsLxfKisr484ucfz000/gL0/Upba29q677uratWujRo2aN28+YMCAOXPmxJ1UgnjzzTePP/74\nVq1a1a9fv2XLlv379582bVrcScWD5ZsgFy9enC5TypfrwPHh5VqFjh0hhBBCSJHAe+wIIYQQ\nQooECjtCCCGEkCKBwo4QQgghpEigsCOEEEIIKRIo7AghhBBCigQKO0IIIYSQIoHCjhBCCCGk\nSKCwI4QQQggpEijsCCGEEEKKBAo7QgghhJAigcKOEEIIIaRIoLAjhBBCCCkSKOwIIYQQQooE\nCjtCCCGEkCKBwo4QQgghpEigsCOEEEIIKRIo7AghhBBCigQKO0IIIYSQIoHCjhBCCCGkSKCw\nI4QQQggpEijsCCGEEEL+f3v3HhRV2ccB/FnYXWSRm3JzF1wkFfCKC6k4mAqWgAbkhZAII0ld\nFfGColGCmoZD5R0bSyWgQWQimEEcyrFI8oKhEgqoKKyUS+oKCnFrYd8/Hj3tuyzLqijN8fv5\ny33Oc57zO2dn8DvnOedZlkCwAwAAAGAJBDsAAAAAlkCwAwAAAGAJBDsAAAAAlkCwAwAAAGAJ\nBDsAAAAAlkCwAwAAAGAJBDsAAAAAlkCwAwAAAGAJBDsAAAAAlkCwAwAAAGAJBDuAl11ISAiH\nw/njjz9e8HGXL19uZGRUUlLygo/7wtALW1dX19eFvFAbN27k8/mFhYV9XQjASwrBDgB6WX19\nfUxMjFgsNjIyGjJkSFBQ0NmzZzX6ZGRk7Nu377PPPnN3d++TIvWRmJhYVVX11Lu7ubnNmDHD\nyMioF0t6dg0NDStXrnR0dOTz+UKhMDIyUi6X6+ifkpLC0eaTTz7R2j8+Pt7T0zM4OPju3bvP\n5wwAQBeOSqXq6xoAoC+FhIRkZmbW1tba29s/+2j37993d3evqamZOXOmRCK5efNmZmYml8st\nLi4ePXo07dPU1OTo6Dhs2LAzZ848+xGfE7lcLhQKjx8/7uvr29e19Jr29nZPT88LFy7MmTNH\nIpHcuHEjLS3N3t6+pKTE0tJS6y47d+5ctWrV/PnzBw8erN4+Y8aMadOmad2lqqrKxcVlwYIF\nBw8e7P1zAADdVADwcnv77bcJIbW1tb0y2rJlywghe/bsYVq+++47Qoi/vz/TkpiYSAg5duxY\nrxzxOcnNzSWEHD9+vK8L6ZlYLF6zZo0+Pb/44gtCyPbt25mWzMxMQoiO3ePj4wkh58+ff6KS\nQkNDuVzuzZs3n2gvAHh2mIoFgP8jk8kiIiJEIhGfz7eysgoICCguLlbvcOzYsfHjxwsEAjs7\nu+jo6JaWFgcHB4lEQrfyeDwfH5/Fixcz/d966y1jY+MrV67Qj52dnTt37nRxcfH399ddSV1d\nXWRkpEgkMjExGTt27K5du5RKpZ51zpo1i8PhNDQ0MC1KpZLD4UyfPp1+DA0N5XA4TU1NsbGx\njo6ORkZGDg4OO3bsUKlUdPfAwEBCiJ+fH4fDKSoq0lphW1tbUlLS2LFjzc3NTU1Nx4wZk5SU\n1NnZSbcyz9jV1NRonc20srJihvrrr7+WLVsmFov5fL61tXVQUND58+d1X5+nkJqaampqGh0d\nzbQEBwcPHTo0LS1N1c3sDb2GFhYWT3Sg1atXK5XKnTt3Pku1APAUuH1dAAD8h9TW1o4fP765\nuVkqlY4cOfLPP/9MTk5+7bXXTpw44eXlRQj55ZdfAgMDra2t169fb2VllZWVFRIS0tjYKBKJ\n6Ag7duzQGLO9vV2pVDLzvBcuXKirqwsODtZdyd27dz08PJqamsLDw8Vi8c8//7xy5cqysrKv\nv/5anzp7xOfzCSFz584dMmTIkSNHOjs7N23atHr1agsLi4iIiI8++mjAgAFpaWkbN24cN27c\niBEjtA4ilUoPHz4cGhoqlUo5HE5BQcG6detkMtnevXvVu1lZWX311VfqLaWlpXv37nVxcWFO\ndsKECQ0NDUuWLBk1alRtbW1ycvLkyZMLCgrbjKZPAAAKqElEQVSmTJmiz+noo7W1taysbOrU\nqRqP/Xl5eaWkpFRXVzs5OXXdiwl2HR0dcrm8X79+6nm0OxKJxNraOj8/f9euXb1VPwDopa9v\nGQJAH1Ofil2wYAEhJDs7m9laXl5uaGg4ceJE+vH1118nahNzSqWSPmg1YcKE7san/7Uzk7Of\nfvopISQnJ0d3VVKplBBSUFDAtMycOZMQcvnyZX3qpJ3r6+uZDv/88w8hxMfHh35cuHAhIWT+\n/PlMhxs3bhBCZs2apV6n7qlYgUDg6emp3rJq1ao5c+YolUrV4wsrl8s19rp//76Tk5OVlZVM\nJmNOlsvlqk933rp1y9TU1MPDo7tDX1cjEokWLlzIfOx6ROratWuEkPfee0+jnU62/vjjj1r3\nCgoKIoTExcUxD+ENHz7822+/7a4wBj396urqHnsCQC/CHTsAeESlUuXk5Nja2tL/yylXV1dP\nT8+ioiKFQjFw4MBTp065uLh4eHjQrYaGhrGxsT/99FN3YxYWFq5du9bLy2vJkiW05fr164SQ\noUOH6q7k6NGjDg4ONEdSu3fvXrNmja2trT516nnKNCBSTk5OAoHgiZZ94fF4Mpnszp07NjY2\ntIU+xKaDSqUKCwuTyWQFBQX0dQSVSpWVlTVmzBh7e3tmbRQejzdp0qSCgoKmpqb+/ftrDKJU\nKocNG6becvDgQeZNhcDAwJycnK6HbmxsJISYmJhotNPx6dau6B27jIyMdevWiUSiioqKffv2\nvfPOO42NjeoT7l3RCquqqhwdHXV0A4DehWAHAI/U1dU9ePDA3d2dw+Gotzs7OxcVFV27ds3V\n1bW1tVUjk02aNKm7ATMyMiIiIkaNGpWbm8vlPvprc+/ePUKI7uk8uVyuUCgkEol6JU5OTnSu\nUC6X667T09NTz1PWeNOTx+PRG3t62rx5c3R09LBhwwIDA6dNm/bGG28wU9Ld2bRpU35+fmJi\noo+PD225c+fOvXv37t27N2jQoK79b9261XUi2NDQMCsri/m4dOnSiRMnhoeH04+6a9C4aIQQ\nlUqltZ36+OOPly9f7uvryyTCsLAwiUTy4YcfRkRE0BltrWjYpV83ALwwCHYA8Mjff/9NtN3R\nMTY2plsVCgUhRCAQqG81NTU1NDTU2EWlUiUkJGzevNnX1/fo0aOmpqbMpocPHxJCzM3NdVTS\n0tJCCOluBbge69QxsgYej6d/565WrFgxatSoPXv2ZGdnp6WlcTgcPz+/5ORksVistX9+fv7m\nzZtnz54dGxvLNNJbZW5ubnTyV4NQKOzayOFw5s6dy3yMiYkZPny4eotWZmZmRNudOfqNqH9H\n6ry9vTVaRowY4e/v//3335eWlr766qvdHY6+b/HgwQPdVQFA70KwA4BH6JRc12BEW0xNTWkM\nam1tVd/a3Nzc0dGh3qJSqSIjIw8dOhQVFbVjxw6N2EfjxYMHD/r169ddJXZ2duTxJOBT1Kl1\nr/b29u4O9yy8vb29vb3b2tpOnTqVnp6empo6ffr0K1eudL2VdfPmzbCwMGdn55SUFPV2puDn\nvWDe4MGDuVyuTCbTaKcPF2rM7epG78Y1NTXp6EO/Pt0JHgB6HZY7AYBH7OzsBgwYUFFRofr/\nlS/Ky8s5HI6zs7OdnZ2BgYFGMjh37pzGOKtWrTp06NC2bdt2797d9WYenYSlN/+6Y2JiYm1t\nXVFRoT4xevXq1b179165cqXHOsnjW3Hqu1dXV/d0AZ6ekZHR9OnTU1JSlixZUlVVdenSJY0O\nLS0ts2fPViqV2dnZGtHT1tbWysqqsrJSI8j2+i838Pl8d3f34uLi5uZmprGzs7OwsNDBwUFj\nVppqamrav39/RkaGRjtdvKa7G5MUrV+fV2gBoBch2AHAv2bPni2Xy+navNSlS5eKi4u9vb0t\nLCz4fL6Hh8fvv/9eWVlJt3Z0dGzfvl19hOzs7F27dkVHR2/YsEHrIZhn6nVXEhgYqFAovvnm\nG6YlISEhKiqqra2txzoJIfR5tYqKCqZDamqqXpfgMRpJ6aSwVmfPnhWJRBrDGhgYEG0zvIsX\nLy4tLT18+LCrq2vXoebNm9fa2pqUlMS03L17d8yYMW+++eYT1dyjhQsXNjc3qx/owIEDt2/f\njoyMpB9bW1svXbpE7+ERQgQCwdatWxctWsR844SQ3NzcoqKicePGaV0ehaHPWzIA0OswFQsA\n/9q0aVNeXt677767YsUKZ2fnmpqaffv29e/fn3nZc+3atfPmzfP391+6dKmZmVl6erqTk5P6\nw3Dr1q0jhHR2dq5fv15j8NjYWEtLS/rSwMmTJwMCAnRUEh8fn5eXJ5VKS0tLxWJxYWFhXl5e\neHg4XQm5xzrDw8P379+/evXqpKQkgUCQm5t75syZ7mZptaKpJTExsbq6evLkyV0fJvPw8Bgw\nYMAHH3xQVFTk5ubG4XB+++23lJQULy8vNzc39Z7p6elpaWlubm719fV0HT6Gr6+vvb19QkLC\nsWPHtm3bJpfLp0yZcvv27S+//FKhUKxYsUKfUmtqavQ8qffffz8tLS0hIeHixYsSiaSioiIz\nM3P06NExMTG0Q1VV1bhx43x8fE6cOEEIMTAwSE5ODgoK8vDwCAkJEQqFly9fzsnJMTMz0zgR\nDSqV6uTJk0OHDsUrsQAvWh8tswIA/xUaPyl269atiIiIQYMGcblcGxubkJCQ8vJy9f4HDx50\ndnbm8/lisTguLq69vZ3P50+aNIlu1fHXhi5p1tHRYWtr6+rq2mNhNTU1YWFhNjY2PB7Pycnp\n888/p+vD6VlnSkrKiBEjjI2NbW1tFy1a1NDQIBQKvby86Fa6jt3169fVdzE3Nx85ciT9d3t7\n+5w5c4yNjS0tLbOysrRWqFAoVq5c+corrwgEAnNz87Fjx27btq2xsVH9wsrl8ri4uO6uCbNO\nnlwul0qlDg4OXC7XwsIiICDg3LlzPV6ip9DY2BgTEyMWi3k8nkgkWrZsmUKhYLaWlZURtdX+\nqNOnT/v5+VlYWHC5XKFQGB4ernHduiopKSGEREVFPY9TAAAdOCqdf4gBAHR7+PChubl5QECA\n+sSobomJiRs2bMjPz/fz83uutUFfCQsLy8zMvHr1qu7pWgDodXjGDgCewOHDh6dOnUrvx1D0\nHU89f8iLWr58+cCBA7ds2dLr5cF/wY0bN44cORIeHo5UB/Di4Y4dADyBc+fOTZkyxdLSUiqV\nCoXCixcvHjhwQCgUlpaWPtHvxGdkZISGhu7evTsqKur5VQsvXkdHh7e3d2Vl5eXLl62trfu6\nHICXDoIdADyZX3/9devWrSUlJfX19TY2NjNmzNiyZYvWdXR1i4qKOnDgwOnTp93d3Z9HndAn\nNm7cmJiY+MMPP0ydOrWvawF4GSHYAQAAALAEnrEDAAAAYAkEOwAAAACWQLADAAAAYAkEOwAA\nAACWQLADAAAAYAkEOwAAAACWQLADAAAAYAkEOwAAAACWQLADAAAAYAkEOwAAAACWQLADAAAA\nYAkEOwAAAACWQLADAAAAYAkEOwAAAACWQLADAAAAYAkEOwAAAACWQLADAAAAYAkEOwAAAACW\nQLADAAAAYAkEOwAAAACWQLADAAAAYAkEOwAAAACWQLADAAAAYAkEOwAAAACWQLADAAAAYAkE\nOwAAAACW+B/ERn99iol+CwAAAABJRU5ErkJggg==",
      "text/plain": [
       "Plot with title “voom: Mean-variance trend”"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#     se\n",
    "    splice_list       = c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\",\"se_\")\n",
    "    tissue_list_m_f     = levels(pData(reduced_obj2)$tissue)\n",
    "    splice_index        = 5\n",
    "    tissue_index        = 21\n",
    "    plot                = TRUE\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = se_fromGTF_no_chrY\n",
    "    metadata            = ocounts_metadata\n",
    "    ijc                 = se_jc_ijc_no_chrY\n",
    "    sjc                 = se_jc_sjc_no_chrY\n",
    "    tissue_list         = tissue_list_m_f\n",
    "    obj                 = reduced_obj2\n",
    "    print_exploratory_plots (plot, \n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"AllTissueJunctionAnalysis\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the artefacts in the `..data/` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -type f -exec sha256sum {} \\\\;  >  ../metadata/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\")\n",
    "\n",
    "data.table::fread(paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\"), header = FALSE, col.names = c(\"sha256sum\", \"file\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../metadata/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../metadata/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
