{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# AllTissueAllSplicingJunctionAnalysis as a Notebook \n",
    "\n",
    "rMATS 3.2.5 was run on controlled access RNASeq files retrieved experiments stored in the Sequence Read Archive with controlled access managed by dbGaP.   The data were generated under the Gene Tissue Expression.\n",
    "\n",
    "## rMATS RNASeq-MATS.py produces 10 different output types which get assembled into as type junction ID by sample ID matrices\n",
    "\n",
    "### Alternative Splice Site Types are: (se, a3ss, a5ss, mxe, ri)\n",
    "\n",
    " This is input as ARGV1 into variable 'astype'\n",
    "\n",
    "  * Skipped Exon events (se),\n",
    "  * Alternative 3' splice site (a3ss),\n",
    "  * Alternative 5' splice site (a5ss),\n",
    "  * Mutually exclusive exon (mxe),\n",
    "  * and retention intron (ri)\n",
    "\n",
    "### There are two different kinds of junction counts\n",
    "\n",
    "  * jc = junction counts - reads that cross the junction\n",
    "  * jcec = junction counts plus reads on the target (such as included exon\n",
    "\n",
    "### And the count type -- there are 5 types\n",
    "\n",
    "  * inclusion levels (percent spliced in)\n",
    "  * included junction counts (ijc)\n",
    "  * skipped junction counts (sjc)\n",
    "  * inclusion length (inclen)\n",
    "  * skipped length (skiplen)\n",
    "\n",
    "### function: fit_iso_tissue \n",
    "\n",
    "fit_iso_tissue expects the following input:\n",
    "\n",
    "  * the tissue of interest (SMSTD) \n",
    "  * an ordered_merged_rmats -- which will be ordered to fit the count matrix\n",
    "  * count matrix (inc or ijc & sjc merged)\n",
    "  * splice type (a3ss, a5ss, mxe, ri or se)\n",
    "  * junction_count type (jc or jcec)\n",
    "  * count type (inc or the merged ijc,sjc)\n",
    "  \n",
    "### reordering to match annotations between count matrix and annotation matrix\n",
    "\n",
    "Common problem is to match specifically the rows of an annotation matrix with the columns of a count matrix\n",
    "`match` is the function that gives the re-ordering index required to accomplish this\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **NOTE**:\n",
    "\n",
    "We assume that you have cloned the analysis repository and have `cd` into the parent directory. Before starting with the analysis make sure you have first completed the dependencies set up by following the instructions described in the **`dependencies/README.md`** document. All paths defined in this Notebook are relative to the parent directory (repository). Please close this Notebook and start again by following the above guidelines if you have not completed the aforementioned steps.\n",
    "\n",
    "## rMATS-final-merged\n",
    "the rmats-nf NextFlow was executed and the results released here:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: R.oo\n",
      "Loading required package: R.methodsS3\n",
      "R.methodsS3 v1.8.0 (2020-02-14 07:10:20 UTC) successfully loaded. See ?R.methodsS3 for help.\n",
      "R.oo v1.23.0 successfully loaded. See ?R.oo for help.\n",
      "\n",
      "Attaching package: ‘R.oo’\n",
      "\n",
      "The following object is masked from ‘package:R.methodsS3’:\n",
      "\n",
      "    throw\n",
      "\n",
      "The following objects are masked from ‘package:methods’:\n",
      "\n",
      "    getClasses, getMethods\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    attach, detach, load, save\n",
      "\n",
      "R.utils v2.9.2 successfully loaded. See ?R.utils for help.\n",
      "\n",
      "Attaching package: ‘R.utils’\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    timestamp\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    cat, commandArgs, getOption, inherits, isOpen, nullfile, parse,\n",
      "    warnings\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(limma)\n",
    "library(piggyback)\n",
    "library(multtest)\n",
    "library(Biobase)\n",
    "library(edgeR)\n",
    "library(tibble)\n",
    "#install.packages('R.utils')\n",
    "library(R.utils)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis \n",
    "\n",
    "This analysis uses edgeR.  Normalization takes the form of correction factors computed internally by edgeR functions, but it is also possible for a user to supply them. The correction factors may take the form of scaling factors for the library sizes, such as computed by calcNormFactors, which are then used to compute the effective library sizes. \n",
    "\n",
    "In this analysis, we are using the raw counts as provided by rMATS 3.2.5.  The raw counts we are using in the model are `ijc` and `sjc`, the sample specific raw read counts as they align to the junctions of the `included exon (ijc)` and the junctions of the `excluded or skipped exon (sjc)` respectively.\n",
    "\n",
    "\n",
    "Be sure to set your GITHUB_TOKEN, prior to downloading files\n",
    "\n",
    "One suggestion is change it to your token and then run it then immediately change it back to this:\n",
    "\n",
    "Sys.setenv(GITHUB_TOKEN = \"your-very-own-github-token\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Did you remember?\n",
    "Did you remember to delete your private github token?  Now is a good time to do so, before you save your work and checkit in inadvertantly...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!(\"rmats_final.se.jc.ijc.txt.gz\" %in% list.files(\"../rmats_counts/\"))) {    \n",
    "    # SE\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    # RI\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    # MXE\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    # A3SS\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "     # A5SS\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../rmats_counts/\")\n",
    "} "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in the count data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading a3ss.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a3ss.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a5ss.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a5ss.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading mxe.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading mxe.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading ri.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading ri.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading se.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading se.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>0</td><td>0</td><td> 2</td><td>0</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td>0</td><td>10</td><td>0</td></tr>\n",
       "\t<tr><td>3</td><td>0</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td>4</td><td>2</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td>5</td><td>1</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 0 & 0 &  2 & 0\\\\\n",
       "\t 2 & 0 & 0 & 10 & 0\\\\\n",
       "\t 3 & 0 & 0 &  0 & 0\\\\\n",
       "\t 4 & 2 & 0 &  0 & 0\\\\\n",
       "\t 5 & 1 & 0 &  0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 0 | 0 |  2 | 0 |\n",
       "| 2 | 0 | 0 | 10 | 0 |\n",
       "| 3 | 0 | 0 |  0 | 0 |\n",
       "| 4 | 2 | 0 |  0 | 0 |\n",
       "| 5 | 1 | 0 |  0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  0          0           2         0         \n",
       "2 2  0          0          10         0         \n",
       "3 3  0          0           0         0         \n",
       "4 4  2          0           0         0         \n",
       "5 5  1          0           0         0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>9</td><td>151</td><td>51</td><td>263</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>3</td><td>2</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td>0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>5</td><td>0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 9 & 151 & 51 & 263\\\\\n",
       "\t 2 & 0 &   0 &  0 &   0\\\\\n",
       "\t 3 & 2 &   0 &  0 &   0\\\\\n",
       "\t 4 & 0 &   0 &  0 &   0\\\\\n",
       "\t 5 & 0 &   0 &  0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 9 | 151 | 51 | 263 |\n",
       "| 2 | 0 |   0 |  0 |   0 |\n",
       "| 3 | 2 |   0 |  0 |   0 |\n",
       "| 4 | 0 |   0 |  0 |   0 |\n",
       "| 5 | 0 |   0 |  0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  9          151        51         263       \n",
       "2 2  0            0         0           0       \n",
       "3 3  2            0         0           0       \n",
       "4 4  0            0         0           0       \n",
       "5 5  0            0         0           0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 0</td><td> 0</td><td> 0</td><td> 2</td></tr>\n",
       "\t<tr><td>2</td><td> 0</td><td> 0</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td>3</td><td> 1</td><td> 0</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td>4</td><td>14</td><td>14</td><td> 3</td><td> 9</td></tr>\n",
       "\t<tr><td>5</td><td>22</td><td>67</td><td>10</td><td>20</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  0 &  0 &  0 &  2\\\\\n",
       "\t 2 &  0 &  0 &  0 &  1\\\\\n",
       "\t 3 &  1 &  0 &  0 &  1\\\\\n",
       "\t 4 & 14 & 14 &  3 &  9\\\\\n",
       "\t 5 & 22 & 67 & 10 & 20\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  0 |  0 |  0 |  2 |\n",
       "| 2 |  0 |  0 |  0 |  1 |\n",
       "| 3 |  1 |  0 |  0 |  1 |\n",
       "| 4 | 14 | 14 |  3 |  9 |\n",
       "| 5 | 22 | 67 | 10 | 20 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   0          0          0          2        \n",
       "2 2   0          0          0          1        \n",
       "3 3   1          0          0          1        \n",
       "4 4  14         14          3          9        \n",
       "5 5  22         67         10         20        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>56</td><td>79</td><td>30</td><td>778</td></tr>\n",
       "\t<tr><td>2</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>3</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td>25</td><td>13</td><td>12</td><td> 74</td></tr>\n",
       "\t<tr><td>5</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 56 & 79 & 30 & 778\\\\\n",
       "\t 2 &  0 &  0 &  0 &   0\\\\\n",
       "\t 3 &  0 &  0 &  0 &   0\\\\\n",
       "\t 4 & 25 & 13 & 12 &  74\\\\\n",
       "\t 5 &  0 &  0 &  0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 56 | 79 | 30 | 778 |\n",
       "| 2 |  0 |  0 |  0 |   0 |\n",
       "| 3 |  0 |  0 |  0 |   0 |\n",
       "| 4 | 25 | 13 | 12 |  74 |\n",
       "| 5 |  0 |  0 |  0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  56         79         30         778       \n",
       "2 2   0          0          0           0       \n",
       "3 3   0          0          0           0       \n",
       "4 4  25         13         12          74       \n",
       "5 5   0          0          0           0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 6</td><td>5</td><td>3</td><td> 9</td></tr>\n",
       "\t<tr><td>2</td><td>13</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>3</td><td>10</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>4</td><td> 1</td><td>1</td><td>0</td><td> 1</td></tr>\n",
       "\t<tr><td>5</td><td> 8</td><td>0</td><td>0</td><td>17</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  6 & 5 & 3 &  9\\\\\n",
       "\t 2 & 13 & 0 & 0 &  0\\\\\n",
       "\t 3 & 10 & 0 & 0 &  0\\\\\n",
       "\t 4 &  1 & 1 & 0 &  1\\\\\n",
       "\t 5 &  8 & 0 & 0 & 17\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  6 | 5 | 3 |  9 |\n",
       "| 2 | 13 | 0 | 0 |  0 |\n",
       "| 3 | 10 | 0 | 0 |  0 |\n",
       "| 4 |  1 | 1 | 0 |  1 |\n",
       "| 5 |  8 | 0 | 0 | 17 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   6         5          3           9        \n",
       "2 2  13         0          0           0        \n",
       "3 3  10         0          0           0        \n",
       "4 4   1         1          0           1        \n",
       "5 5   8         0          0          17        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>22</td><td>36</td><td>7</td><td>40</td></tr>\n",
       "\t<tr><td>2</td><td>10</td><td> 0</td><td>4</td><td> 0</td></tr>\n",
       "\t<tr><td>3</td><td> 1</td><td> 0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>4</td><td> 4</td><td> 0</td><td>2</td><td> 0</td></tr>\n",
       "\t<tr><td>5</td><td>51</td><td> 4</td><td>8</td><td>24</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 22 & 36 & 7 & 40\\\\\n",
       "\t 2 & 10 &  0 & 4 &  0\\\\\n",
       "\t 3 &  1 &  0 & 0 &  0\\\\\n",
       "\t 4 &  4 &  0 & 2 &  0\\\\\n",
       "\t 5 & 51 &  4 & 8 & 24\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 22 | 36 | 7 | 40 |\n",
       "| 2 | 10 |  0 | 4 |  0 |\n",
       "| 3 |  1 |  0 | 0 |  0 |\n",
       "| 4 |  4 |  0 | 2 |  0 |\n",
       "| 5 | 51 |  4 | 8 | 24 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  22         36         7          40        \n",
       "2 2  10          0         4           0        \n",
       "3 3   1          0         0           0        \n",
       "4 4   4          0         2           0        \n",
       "5 5  51          4         8          24        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>  0</td><td> 0</td><td>  5</td><td> 0</td></tr>\n",
       "\t<tr><td>2</td><td>  3</td><td> 0</td><td>  1</td><td> 6</td></tr>\n",
       "\t<tr><td>3</td><td>  2</td><td> 0</td><td>  0</td><td> 6</td></tr>\n",
       "\t<tr><td>4</td><td>  9</td><td> 0</td><td>  4</td><td>15</td></tr>\n",
       "\t<tr><td>5</td><td>112</td><td>64</td><td>113</td><td>59</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &   0 &  0 &   5 &  0\\\\\n",
       "\t 2 &   3 &  0 &   1 &  6\\\\\n",
       "\t 3 &   2 &  0 &   0 &  6\\\\\n",
       "\t 4 &   9 &  0 &   4 & 15\\\\\n",
       "\t 5 & 112 & 64 & 113 & 59\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |   0 |  0 |   5 |  0 |\n",
       "| 2 |   3 |  0 |   1 |  6 |\n",
       "| 3 |   2 |  0 |   0 |  6 |\n",
       "| 4 |   9 |  0 |   4 | 15 |\n",
       "| 5 | 112 | 64 | 113 | 59 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1    0         0           5         0        \n",
       "2 2    3         0           1         6        \n",
       "3 3    2         0           0         6        \n",
       "4 4    9         0           4        15        \n",
       "5 5  112        64         113        59        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>2</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>3</td><td> 2</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td>88</td><td>98</td><td>25</td><td>206</td></tr>\n",
       "\t<tr><td>5</td><td>70</td><td>11</td><td>34</td><td> 37</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  0 &  0 &  0 &   0\\\\\n",
       "\t 2 &  0 &  0 &  0 &   0\\\\\n",
       "\t 3 &  2 &  0 &  0 &   0\\\\\n",
       "\t 4 & 88 & 98 & 25 & 206\\\\\n",
       "\t 5 & 70 & 11 & 34 &  37\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  0 |  0 |  0 |   0 |\n",
       "| 2 |  0 |  0 |  0 |   0 |\n",
       "| 3 |  2 |  0 |  0 |   0 |\n",
       "| 4 | 88 | 98 | 25 | 206 |\n",
       "| 5 | 70 | 11 | 34 |  37 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   0          0          0           0       \n",
       "2 2   0          0          0           0       \n",
       "3 3   2          0          0           0       \n",
       "4 4  88         98         25         206       \n",
       "5 5  70         11         34          37       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>2</td><td>26</td><td>247</td><td>103</td><td>620</td></tr>\n",
       "\t<tr><td>3</td><td> 1</td><td>  0</td><td>  1</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td> 0</td><td>  1</td><td>  1</td><td>  2</td></tr>\n",
       "\t<tr><td>5</td><td> 3</td><td>  0</td><td>  2</td><td>  3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  0 &   0 &   0 &   0\\\\\n",
       "\t 2 & 26 & 247 & 103 & 620\\\\\n",
       "\t 3 &  1 &   0 &   1 &   0\\\\\n",
       "\t 4 &  0 &   1 &   1 &   2\\\\\n",
       "\t 5 &  3 &   0 &   2 &   3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  0 |   0 |   0 |   0 |\n",
       "| 2 | 26 | 247 | 103 | 620 |\n",
       "| 3 |  1 |   0 |   1 |   0 |\n",
       "| 4 |  0 |   1 |   1 |   2 |\n",
       "| 5 |  3 |   0 |   2 |   3 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   0           0          0          0       \n",
       "2 2  26         247        103        620       \n",
       "3 3   1           0          1          0       \n",
       "4 4   0           1          1          2       \n",
       "5 5   3           0          2          3       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>2</td><td>0</td><td>1</td><td>3</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>3</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>4</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>5</td><td>0</td><td>5</td><td>3</td><td>8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 2 & 0 & 1 & 3\\\\\n",
       "\t 2 & 0 & 0 & 0 & 1\\\\\n",
       "\t 3 & 0 & 0 & 0 & 1\\\\\n",
       "\t 4 & 0 & 0 & 0 & 1\\\\\n",
       "\t 5 & 0 & 5 & 3 & 8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 2 | 0 | 1 | 3 |\n",
       "| 2 | 0 | 0 | 0 | 1 |\n",
       "| 3 | 0 | 0 | 0 | 1 |\n",
       "| 4 | 0 | 0 | 0 | 1 |\n",
       "| 5 | 0 | 5 | 3 | 8 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  2          0          1          3         \n",
       "2 2  0          0          0          1         \n",
       "3 3  0          0          0          1         \n",
       "4 4  0          0          0          1         \n",
       "5 5  0          5          3          8         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "message(\"loading a3ss.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "a3ss.jc.ijc <- data.table::fread(\"../rmats_counts/rmats_final.a3ss.jc.ijc.txt.gz\")\n",
    "message(\"done!\\n\")\n",
    "message(\"loading a3ss.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "a3ss.jc.sjc <- data.table::fread(\"../rmats_counts/rmats_final.a3ss.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading a5ss.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "a5ss.jc.ijc <- data.table::fread(\"../rmats_counts/rmats_final.a5ss.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading a5ss.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "a5ss.jc.sjc <- data.table::fread(\"../rmats_counts/rmats_final.a5ss.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading mxe.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "mxe.jc.ijc  <- data.table::fread(\"../rmats_counts/rmats_final.mxe.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading mxe.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "mxe.jc.sjc  <- data.table::fread(\"../rmats_counts/rmats_final.mxe.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading ri.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "ri.jc.ijc   <- data.table::fread(\"../rmats_counts/rmats_final.ri.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading ri.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "ri.jc.sjc   <- data.table::fread(\"../rmats_counts/rmats_final.ri.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading se.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "se.jc.ijc   <- data.table::fread(\"../rmats_counts/rmats_final.se.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading se.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "se.jc.sjc   <- data.table::fread(\"../rmats_counts/rmats_final.se.jc.sjc.txt.gz\")\n",
    "message(\"done!\\n\")\n",
    "\n",
    "a3ss.jc.ijc[1:5,1:5]\n",
    "a3ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "a5ss.jc.ijc[1:5,1:5]\n",
    "a5ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "mxe.jc.ijc[1:5,1:5]\n",
    "mxe.jc.sjc[1:5,1:5]\n",
    "\n",
    "ri.jc.ijc[1:5,1:5]\n",
    "ri.jc.sjc[1:5,1:5]\n",
    "\n",
    "se.jc.ijc[1:5,1:5]\n",
    "se.jc.sjc[1:5,1:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in metadata \n",
    "\n",
    "- `Sequence Read Archive (SRA)` Accession Data, `SRR` numbers\n",
    "- `Genome Tissue Expression (GTEx)` Clinical Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\n",
      "\n",
      "done!\n",
      "\n",
      "Loading obj GTEx v8 rds object with readRDS from ../data/gtex.rds ..\n",
      "\n",
      "Done!\n",
      "\n",
      "Generating sha256sum for gtex.rds ..\n",
      "\n",
      "c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0  ../data/gtex.rds\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!(\"SraRunTable.txt.gz\" %in% list.files(\"../data/\"))) {\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"TheJacksonLaboratory/sbas\", \n",
    "        file = \"SraRunTable.txt.gz\",\n",
    "        tag  = \"GTExV8.v1.0\", \n",
    "        dest = \"../data/\")\n",
    "    \n",
    "    message(\"Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\\n\")   \n",
    "    metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    message(\"done!\")\n",
    "} else {\n",
    "    message(\"Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\\n\")   \n",
    "    metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    message(\"done!\\n\")\n",
    "}\n",
    "\n",
    "if (!(\"gtex.rds\" %in% list.files(\"../data/\"))) {\n",
    "    message(\"Downloading and loading obj with GTEx v8 with 'yarn::downloadGTExV8()'\\n\")\n",
    "    obj <- yarn::downloadGTExV8(type='genes',file='../data/gtex.rds')\n",
    "    message(\"Done!\\n\")\n",
    "\n",
    "} else {\n",
    "# Load with readRDS() if gtex.rds available in data/\n",
    "    message(\"Loading obj GTEx v8 rds object with readRDS from ../data/gtex.rds ..\\n\")   \n",
    "    obj <- readRDS(file = \"../data/gtex.rds\")\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Generating sha256sum for gtex.rds ..\\n\")    \n",
    "    message(system(\"sha256sum ../data/gtex.rds\", intern = TRUE))\n",
    "    message(\"Done!\\n\")\n",
    "} \n",
    "if (! (file.exists(\"../data/fromGTF.tar.gz\"))) {\n",
    "    system(\"mkdir -p ../data\", intern = TRUE)\n",
    "    message(\"Fetching fromGTF.tar.gz from GitHub ..\")\n",
    "    # Download archive from GitHub release with tag \"dge\"\n",
    "    piggyback::pb_download(file = \"fromGTF.tar.gz\",\n",
    "                           dest = \"../data\",\n",
    "                           repo = \"adeslatt/sbas_gtf\",\n",
    "                           tag  = \"rMATS.3.2.5.gencode.v30\",\n",
    "                           show_progress = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Decompressing fromGTF.tar.gz into ../data\")\n",
    "    system(\"mkdir -p ../data && tar xvfz ../data/fromGTF.tar.gz -C ../data\", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Decompressing fromGTF.*.txt.gz into ../data\")\n",
    "    system(\"gunzip  ../data/fromGTF*.txt.gz \", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Reading fromGTF.SE.txt into fromGTF.SE\")    \n",
    "    fromGTF.SE <- read.table(\"../data/fromGTF.SE.txt\", header=TRUE)\n",
    "    message(\"Done!\\n\") \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quality control, preprocessing of data \n",
    "\n",
    "We observed above that our phenotype data have 2 more observations than our expression data. Let's inspect what these samples are:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-YF7O-2326-101833-SM-5CVN9</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>Skin   </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>GTEX-YEC3-1426-101806-SM-5PNXX</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>Stomach</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-YF7O-2326-101833-SM-5CVN9 & NA & NA & NA & NA & NA & Skin    & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tGTEX-YEC3-1426-101806-SM-5PNXX & NA & NA & NA & NA & NA & Stomach & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-YF7O-2326-101833-SM-5CVN9 | NA | NA | NA | NA | NA | Skin    | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| GTEX-YEC3-1426-101806-SM-5PNXX | NA | NA | NA | NA | NA | Stomach | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "                               SAMPID SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS   \n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA     NA       NA       NA       NA    Skin   \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA     NA       NA       NA       NA    Stomach\n",
       "                               SMTSD SMUBRID SMTSISCH SMTSPAX ⋯ SME1PCTS\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA    NA      NA       NA      ⋯ NA      \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA    NA      NA       NA      ⋯ NA      \n",
       "                               SMRRNART SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA       NA       NA       NA       NA      \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA       NA       NA       NA       NA      \n",
       "                               SUBJID SEX AGE DTHHRDY\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA     NA  NA  NA     \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA     NA  NA  NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17384</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17384\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17384\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17384    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55878</li><li>17382</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55878\n",
       "\\item 17382\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55878\n",
       "2. 17382\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55878 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17382"
      ],
      "text/latex": [
       "17382"
      ],
      "text/markdown": [
       "17382"
      ],
      "text/plain": [
       "[1] 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The non-overlapping IDs between pheno and count data are:\n",
      "GTEX-YF7O-2326-101833-SM-5CVN9\n",
      "GTEX-YEC3-1426-101806-SM-5PNXX\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "logical_match_names\n",
       "FALSE  TRUE \n",
       "    2 17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-ZZPU-2626-SM-5E45Y</th><td>GTEX-ZZPU-2626-SM-5E45Y</td><td> 1</td><td>B1</td><td>2 pieces, includes internal fat (up to 20%) and tendon</td><td> 8.50000</td><td>Muscle        </td><td>Muscle - Skeletal     </td><td>0011907</td><td>  248</td><td> 428</td><td>⋯</td><td>50.6647</td><td>0.010466900</td><td>0.9956840</td><td>NA</td><td> 0</td><td>49.6941</td><td>GTEX-ZZPU</td><td>2</td><td>50-59</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>GTEX-ZZPU-2726-SM-5NQ8O</th><td>GTEX-ZZPU-2726-SM-5NQ8O</td><td> 1</td><td>B1</td><td>2 pieces                                              </td><td> 6.80000</td><td>Adipose Tissue</td><td>Adipose - Subcutaneous</td><td>0002190</td><td>  249</td><td> 427</td><td>⋯</td><td>50.3168</td><td>0.005856410</td><td>0.9958590</td><td>NA</td><td> 0</td><td>50.0760</td><td>GTEX-ZZPU</td><td>2</td><td>50-59</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-ZZPU-2626-SM-5E45Y & GTEX-ZZPU-2626-SM-5E45Y &  1 & B1 & 2 pieces, includes internal fat (up to 20\\%) and tendon &  8.50000 & Muscle         & Muscle - Skeletal      & 0011907 &   248 &  428 & ⋯ & 50.6647 & 0.010466900 & 0.9956840 & NA &  0 & 49.6941 & GTEX-ZZPU & 2 & 50-59 &  0\\\\\n",
       "\tGTEX-ZZPU-2726-SM-5NQ8O & GTEX-ZZPU-2726-SM-5NQ8O &  1 & B1 & 2 pieces                                               &  6.80000 & Adipose Tissue & Adipose - Subcutaneous & 0002190 &   249 &  427 & ⋯ & 50.3168 & 0.005856410 & 0.9958590 & NA &  0 & 50.0760 & GTEX-ZZPU & 2 & 50-59 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-ZZPU-2626-SM-5E45Y | GTEX-ZZPU-2626-SM-5E45Y |  1 | B1 | 2 pieces, includes internal fat (up to 20%) and tendon |  8.50000 | Muscle         | Muscle - Skeletal      | 0011907 |   248 |  428 | ⋯ | 50.6647 | 0.010466900 | 0.9956840 | NA |  0 | 49.6941 | GTEX-ZZPU | 2 | 50-59 |  0 |\n",
       "| GTEX-ZZPU-2726-SM-5NQ8O | GTEX-ZZPU-2726-SM-5NQ8O |  1 | B1 | 2 pieces                                               |  6.80000 | Adipose Tissue | Adipose - Subcutaneous | 0002190 |   249 |  427 | ⋯ | 50.3168 | 0.005856410 | 0.9958590 | NA |  0 | 50.0760 | GTEX-ZZPU | 2 | 50-59 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "                        SAMPID                  SMATSSCR SMCENTER\n",
       "GTEX-ZZPU-2626-SM-5E45Y GTEX-ZZPU-2626-SM-5E45Y  1       B1      \n",
       "GTEX-ZZPU-2726-SM-5NQ8O GTEX-ZZPU-2726-SM-5NQ8O  1       B1      \n",
       "                        SMPTHNTS                                              \n",
       "GTEX-ZZPU-2626-SM-5E45Y 2 pieces, includes internal fat (up to 20%) and tendon\n",
       "GTEX-ZZPU-2726-SM-5NQ8O 2 pieces                                              \n",
       "                        SMRIN    SMTS           SMTSD                  SMUBRID\n",
       "GTEX-ZZPU-2626-SM-5E45Y  8.50000 Muscle         Muscle - Skeletal      0011907\n",
       "GTEX-ZZPU-2726-SM-5NQ8O  6.80000 Adipose Tissue Adipose - Subcutaneous 0002190\n",
       "                        SMTSISCH SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT \n",
       "GTEX-ZZPU-2626-SM-5E45Y   248     428    ⋯ 50.6647  0.010466900 0.9956840\n",
       "GTEX-ZZPU-2726-SM-5NQ8O   249     427    ⋯ 50.3168  0.005856410 0.9958590\n",
       "                        SMNUM5CD SMDPMPRT SME2PCTS SUBJID    SEX AGE   DTHHRDY\n",
       "GTEX-ZZPU-2626-SM-5E45Y NA        0       49.6941  GTEX-ZZPU 2   50-59  0     \n",
       "GTEX-ZZPU-2726-SM-5NQ8O NA        0       50.0760  GTEX-ZZPU 2   50-59  0     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55878</li><li>17382</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55878\n",
       "\\item 17382\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55878\n",
       "2. 17382\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55878 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail (pData(obj),2)\n",
    "dim(pData(obj))\n",
    "dim(exprs(obj))\n",
    "dim(obj)\n",
    "sample_names=as.vector(as.character(colnames(exprs(obj))))\n",
    "pheno_sample_names=as.vector(as.character(rownames(pData(obj))))\n",
    "length(pheno_sample_names)\n",
    "length(sample_names)\n",
    "\n",
    "if (length(pheno_sample_names) > length(sample_names)) {\n",
    "    superset <- pheno_sample_names\n",
    "    subset   <- sample_names    \n",
    "} \n",
    "\n",
    "if (length(pheno_sample_names) < length(sample_names)) {\n",
    "    superset <- sample_names\n",
    "    subset   <- pheno_sample_names   \n",
    "} \n",
    "\n",
    "non_overlaps <- setdiff( superset, subset)\n",
    "\n",
    "message(\"The non-overlapping IDs between pheno and count data are:\\n\", \n",
    "        paste(non_overlaps, collapse = \"\\n\") )\n",
    "logical_match_names=superset %in% subset\n",
    "length(logical_match_names)\n",
    "table(logical_match_names)\n",
    "pData(obj) <- (pData(obj)[logical_match_names==TRUE,])\n",
    "\n",
    "tail (pData(obj),2)\n",
    "dim(pData(obj))\n",
    "dim(exprs(obj))\n",
    "dim(obj)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in junction Annotation Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading fromGTF.A3SS.txt into fromGTF.A3SS\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.A5SS.txt into fromGTF.A5SS\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.MXE.txt into fromGTF.MXE\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.RI.txt into fromGTF.RI\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.SE.txt into fromGTF.SE\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"Reading fromGTF.A3SS.txt into fromGTF.A3SS\")    \n",
    "fromGTF.A3SS <- read.table(\"../data/fromGTF.A3SS.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.A5SS.txt into fromGTF.A5SS\")    \n",
    "fromGTF.A5SS <- read.table(\"../data/fromGTF.A5SS.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.MXE.txt into fromGTF.MXE\")    \n",
    "fromGTF.MXE <- read.table(\"../data/fromGTF.MXE.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.RI.txt into fromGTF.RI\")    \n",
    "fromGTF.RI <- read.table(\"../data/fromGTF.RI.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.SE.txt into fromGTF.SE\")    \n",
    "fromGTF.SE <- read.table(\"../data/fromGTF.SE.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preparing our Data\n",
    "### Aligning Annotations with Run Data and with Count Data\n",
    "\n",
    "A description of the GTEx V8 release may be found here https://www.gtexportal.org/home/datasets.   To facilitate the analysis, the Quackenbush lab's Joe Paulson's Yarn Package, https://github.com/jnpaulson/yarn, forked and upgraded to GTEx V8 here https://github.com/TheJacksonLaboratory/yarn/tree/annes-changes.  \n",
    "\n",
    "The sequences we used for our analysis with rMATS are specified by their SRR number and were obtained from dbGaP, with our application to the controlled access human data.  Beginning with these raw unfiltered fastq files, we ran them through the NextFlow workflow rmats-nf https://github.com/lifebit-ai/rmats-nf. The accession data was downloaded with a repository key from the google cloud bucket (g3).\n",
    "\n",
    "- `obj`      : GTEx V8 expressionSet \n",
    "- `metadata` : SraRunTable.txt holds the SRR accession numbers used to extract samples from dbGaP SRA\n",
    "- `ijc`      : inclusion junction counts as reported by `rMATS 3.2.5`\n",
    "- `sjc`      : skipped junction counts as reported by `rMATS 3.2.5`  \n",
    "\n",
    "For each of the splicing events:\n",
    "\n",
    "- `a3ss`     : alternative 3` splice site\n",
    "- `a5ss`     : alternative 5` splice site\n",
    "- `mxe`      : mutually exclusive exon\n",
    "- `ri`       : retention intron\n",
    "- `se`       : skipped exon\n",
    "\n",
    "In `rMATS 3.2.5`, each junction is encoded in a unique `ID` that is immutable regardless of comparison.  \n",
    "This `ID` is different for each of the differing splicing events."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Move Junction ID to Row definition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td>0</td><td> 2</td><td>0</td><td>2</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>10</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td> 0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td> 0</td><td>0</td><td>2</td></tr>\n",
       "\t<tr><td>1</td><td>0</td><td> 0</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 0 & 0 &  2 & 0 & 2\\\\\n",
       "\t 0 & 0 & 10 & 0 & 0\\\\\n",
       "\t 0 & 0 &  0 & 0 & 1\\\\\n",
       "\t 2 & 0 &  0 & 0 & 2\\\\\n",
       "\t 1 & 0 &  0 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 0 | 0 |  2 | 0 | 2 |\n",
       "| 0 | 0 | 10 | 0 | 0 |\n",
       "| 0 | 0 |  0 | 0 | 1 |\n",
       "| 2 | 0 |  0 | 0 | 2 |\n",
       "| 1 | 0 |  0 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 0          0           2         0          2         \n",
       "2 0          0          10         0          0         \n",
       "3 0          0           0         0          1         \n",
       "4 2          0           0         0          2         \n",
       "5 1          0           0         0          1         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>9</td><td>151</td><td>51</td><td>263</td><td>292</td></tr>\n",
       "\t<tr><td>0</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>2</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>0</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>0</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 9 & 151 & 51 & 263 & 292\\\\\n",
       "\t 0 &   0 &  0 &   0 &   0\\\\\n",
       "\t 2 &   0 &  0 &   0 &   0\\\\\n",
       "\t 0 &   0 &  0 &   0 &   0\\\\\n",
       "\t 0 &   0 &  0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 9 | 151 | 51 | 263 | 292 |\n",
       "| 0 |   0 |  0 |   0 |   0 |\n",
       "| 2 |   0 |  0 |   0 |   0 |\n",
       "| 0 |   0 |  0 |   0 |   0 |\n",
       "| 0 |   0 |  0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 9          151        51         263        292       \n",
       "2 0            0         0           0          0       \n",
       "3 2            0         0           0          0       \n",
       "4 0            0         0           0          0       \n",
       "5 0            0         0           0          0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td> 2</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td> 1</td><td>  0</td></tr>\n",
       "\t<tr><td> 1</td><td> 0</td><td> 0</td><td> 1</td><td>  0</td></tr>\n",
       "\t<tr><td>14</td><td>14</td><td> 3</td><td> 9</td><td>  4</td></tr>\n",
       "\t<tr><td>22</td><td>67</td><td>10</td><td>20</td><td>167</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &  0 &  2 &   1\\\\\n",
       "\t  0 &  0 &  0 &  1 &   0\\\\\n",
       "\t  1 &  0 &  0 &  1 &   0\\\\\n",
       "\t 14 & 14 &  3 &  9 &   4\\\\\n",
       "\t 22 & 67 & 10 & 20 & 167\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |  0 |  2 |   1 |\n",
       "|  0 |  0 |  0 |  1 |   0 |\n",
       "|  1 |  0 |  0 |  1 |   0 |\n",
       "| 14 | 14 |  3 |  9 |   4 |\n",
       "| 22 | 67 | 10 | 20 | 167 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  0          0          0          2           1       \n",
       "2  0          0          0          1           0       \n",
       "3  1          0          0          1           0       \n",
       "4 14         14          3          9           4       \n",
       "5 22         67         10         20         167       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>56</td><td>79</td><td>30</td><td>778</td><td>293</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>25</td><td>13</td><td>12</td><td> 74</td><td> 10</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 56 & 79 & 30 & 778 & 293\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t 25 & 13 & 12 &  74 &  10\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 56 | 79 | 30 | 778 | 293 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "| 25 | 13 | 12 |  74 |  10 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 56         79         30         778        293       \n",
       "2  0          0          0           0          0       \n",
       "3  0          0          0           0          0       \n",
       "4 25         13         12          74         10       \n",
       "5  0          0          0           0          0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 6</td><td>5</td><td>3</td><td> 9</td><td>4</td></tr>\n",
       "\t<tr><td>13</td><td>0</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td>10</td><td>0</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td> 1</td><td>1</td><td>0</td><td> 1</td><td>0</td></tr>\n",
       "\t<tr><td> 8</td><td>0</td><td>0</td><td>17</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  6 & 5 & 3 &  9 & 4\\\\\n",
       "\t 13 & 0 & 0 &  0 & 0\\\\\n",
       "\t 10 & 0 & 0 &  0 & 0\\\\\n",
       "\t  1 & 1 & 0 &  1 & 0\\\\\n",
       "\t  8 & 0 & 0 & 17 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  6 | 5 | 3 |  9 | 4 |\n",
       "| 13 | 0 | 0 |  0 | 0 |\n",
       "| 10 | 0 | 0 |  0 | 0 |\n",
       "|  1 | 1 | 0 |  1 | 0 |\n",
       "|  8 | 0 | 0 | 17 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  6         5          3           9         4         \n",
       "2 13         0          0           0         0         \n",
       "3 10         0          0           0         0         \n",
       "4  1         1          0           1         0         \n",
       "5  8         0          0          17         0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>22</td><td>36</td><td>7</td><td>40</td><td>10</td></tr>\n",
       "\t<tr><td>10</td><td> 0</td><td>4</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td> 1</td><td> 0</td><td>0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 4</td><td> 0</td><td>2</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td>51</td><td> 4</td><td>8</td><td>24</td><td> 2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 22 & 36 & 7 & 40 & 10\\\\\n",
       "\t 10 &  0 & 4 &  0 &  1\\\\\n",
       "\t  1 &  0 & 0 &  0 &  0\\\\\n",
       "\t  4 &  0 & 2 &  0 &  1\\\\\n",
       "\t 51 &  4 & 8 & 24 &  2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 22 | 36 | 7 | 40 | 10 |\n",
       "| 10 |  0 | 4 |  0 |  1 |\n",
       "|  1 |  0 | 0 |  0 |  0 |\n",
       "|  4 |  0 | 2 |  0 |  1 |\n",
       "| 51 |  4 | 8 | 24 |  2 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 22         36         7          40         10        \n",
       "2 10          0         4           0          1        \n",
       "3  1          0         0           0          0        \n",
       "4  4          0         2           0          1        \n",
       "5 51          4         8          24          2        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  5</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  3</td><td> 0</td><td>  1</td><td> 6</td><td> 1</td></tr>\n",
       "\t<tr><td>  2</td><td> 0</td><td>  0</td><td> 6</td><td> 2</td></tr>\n",
       "\t<tr><td>  9</td><td> 0</td><td>  4</td><td>15</td><td> 7</td></tr>\n",
       "\t<tr><td>112</td><td>64</td><td>113</td><td>59</td><td>43</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t   0 &  0 &   5 &  0 &  0\\\\\n",
       "\t   3 &  0 &   1 &  6 &  1\\\\\n",
       "\t   2 &  0 &   0 &  6 &  2\\\\\n",
       "\t   9 &  0 &   4 & 15 &  7\\\\\n",
       "\t 112 & 64 & 113 & 59 & 43\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|   0 |  0 |   5 |  0 |  0 |\n",
       "|   3 |  0 |   1 |  6 |  1 |\n",
       "|   2 |  0 |   0 |  6 |  2 |\n",
       "|   9 |  0 |   4 | 15 |  7 |\n",
       "| 112 | 64 | 113 | 59 | 43 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1   0         0           5         0          0        \n",
       "2   3         0           1         6          1        \n",
       "3   2         0           0         6          2        \n",
       "4   9         0           4        15          7        \n",
       "5 112        64         113        59         43        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 2</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>88</td><td>98</td><td>25</td><td>206</td><td>150</td></tr>\n",
       "\t<tr><td>70</td><td>11</td><td>34</td><td> 37</td><td> 56</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t  2 &  0 &  0 &   0 &   0\\\\\n",
       "\t 88 & 98 & 25 & 206 & 150\\\\\n",
       "\t 70 & 11 & 34 &  37 &  56\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "|  2 |  0 |  0 |   0 |   0 |\n",
       "| 88 | 98 | 25 | 206 | 150 |\n",
       "| 70 | 11 | 34 |  37 |  56 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  0          0          0           0          0       \n",
       "2  0          0          0           0          0       \n",
       "3  2          0          0           0          0       \n",
       "4 88         98         25         206        150       \n",
       "5 70         11         34          37         56       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>  0</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>26</td><td>247</td><td>103</td><td>620</td><td>494</td></tr>\n",
       "\t<tr><td> 1</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td>  1</td><td>  1</td><td>  2</td><td>  0</td></tr>\n",
       "\t<tr><td> 3</td><td>  0</td><td>  2</td><td>  3</td><td>  6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 &   0 &   0 &   0 &   1\\\\\n",
       "\t 26 & 247 & 103 & 620 & 494\\\\\n",
       "\t  1 &   0 &   1 &   0 &   0\\\\\n",
       "\t  0 &   1 &   1 &   2 &   0\\\\\n",
       "\t  3 &   0 &   2 &   3 &   6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  0 |   0 |   0 |   0 |   1 |\n",
       "| 26 | 247 | 103 | 620 | 494 |\n",
       "|  1 |   0 |   1 |   0 |   0 |\n",
       "|  0 |   1 |   1 |   2 |   0 |\n",
       "|  3 |   0 |   2 |   3 |   6 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  0           0          0          0          1       \n",
       "2 26         247        103        620        494       \n",
       "3  1           0          1          0          0       \n",
       "4  0           1          1          2          0       \n",
       "5  3           0          2          3          6       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>2</td><td>0</td><td>1</td><td>3</td><td>6</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>5</td><td>3</td><td>8</td><td>4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 2 & 0 & 1 & 3 & 6\\\\\n",
       "\t 0 & 0 & 0 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 1 & 0\\\\\n",
       "\t 0 & 5 & 3 & 8 & 4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 2 | 0 | 1 | 3 | 6 |\n",
       "| 0 | 0 | 0 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 1 | 0 |\n",
       "| 0 | 5 | 3 | 8 | 4 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 2          0          1          3          6         \n",
       "2 0          0          0          1          0         \n",
       "3 0          0          0          1          0         \n",
       "4 0          0          0          1          0         \n",
       "5 0          5          3          8          4         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rownames(a3ss.jc.ijc) <- a3ss.jc.ijc$ID\n",
    "rownames(a3ss.jc.sjc) <- a3ss.jc.sjc$ID\n",
    "a3ss.jc.ijc <- a3ss.jc.ijc[,-1]\n",
    "a3ss.jc.sjc <- a3ss.jc.sjc[,-1]\n",
    "\n",
    "rownames(a5ss.jc.ijc) <- a3ss.jc.ijc$ID\n",
    "rownames(a5ss.jc.sjc) <- a3ss.jc.sjc$ID\n",
    "a5ss.jc.ijc <- a5ss.jc.ijc[,-1]\n",
    "a5ss.jc.sjc <- a5ss.jc.sjc[,-1]\n",
    "\n",
    "rownames(mxe.jc.ijc)  <- mxe.jc.ijc$ID\n",
    "rownames(mxe.jc.sjc)  <- mxe.jc.sjc$ID\n",
    "mxe.jc.ijc <- mxe.jc.ijc[,-1]\n",
    "mxe.jc.sjc <- mxe.jc.sjc[,-1]\n",
    "\n",
    "rownames(ri.jc.ijc)   <- ri.jc.ijc$ID\n",
    "rownames(ri.jc.sjc)   <- ri.jc.sjc$ID\n",
    "ri.jc.ijc <- ri.jc.ijc[,-1]\n",
    "ri.jc.sjc <- ri.jc.sjc[,-1]\n",
    "\n",
    "rownames(se.jc.ijc)   <- se.jc.ijc$ID\n",
    "rownames(se.jc.sjc)   <- se.jc.sjc$ID\n",
    "se.jc.ijc <- se.jc.ijc[,-1]\n",
    "se.jc.sjc <- se.jc.sjc[,-1]\n",
    "\n",
    "dim(a3ss.jc.ijc)\n",
    "dim(a3ss.jc.sjc)\n",
    "dim(a5ss.jc.ijc)\n",
    "dim(a5ss.jc.sjc)\n",
    "dim(mxe.jc.ijc)\n",
    "dim(mxe.jc.sjc)\n",
    "dim(ri.jc.ijc)\n",
    "dim(ri.jc.sjc)\n",
    "dim(se.jc.ijc)\n",
    "dim(se.jc.sjc)\n",
    "\n",
    "a3ss.jc.ijc[1:5,1:5]\n",
    "a3ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "a5ss.jc.ijc[1:5,1:5]\n",
    "a5ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "mxe.jc.ijc[1:5,1:5]\n",
    "mxe.jc.sjc[1:5,1:5]\n",
    "\n",
    "ri.jc.ijc[1:5,1:5]\n",
    "ri.jc.sjc[1:5,1:5]\n",
    "\n",
    "se.jc.ijc[1:5,1:5]\n",
    "se.jc.sjc[1:5,1:5]\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Synchronize Clinical Annotations and Accession Run, reducing to only tissues of interest\n",
    "\n",
    "Join the yarn metadata with the metadata we have (there are redundant samples that have been sequenced multiple times). We want to be sure that we can obtain all required Clinical Annotation information from the YARN GTEx Annotation information, as the SRA metadata is not as reliable.    This will be a one-to-many mapping, as there are multiple sequence runs per 69 samples -- expanding our data set.  There are only a handful of annotations we require: SEX, AGE, DTHHRDY (which is cause of death), SMCENTER.\n",
    "\n",
    "Note that the numbers in specific age groups expand because of the one to many relationship from sample to sequencing runs. \n",
    "\n",
    "#### Using results from analysis of number of samples stored in `tissues.tsv` we keep only those that are members of this reduced tissue list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       " 0  1 \n",
       "11 39 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>39</li><li>5</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 39\n",
       "\\item 5\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 39\n",
       "2. 5\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 39  5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "16923   459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>459</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 459\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   459\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>459</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 459\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 459\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 459  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 68</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th><th scope=col>tissue</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-1117F-2826-SM-5GZXL</th><td>GTEX-1117F-2826-SM-5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>Breast - Mammary Tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td><td>breast_mammary_tissue</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 68\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY & tissue\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-1117F-2826-SM-5GZXL & GTEX-1117F-2826-SM-5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & Breast - Mammary Tissue & 0008367 &  1340 & 1008 & ⋯ & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4 & breast\\_mammary\\_tissue\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 68\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; | tissue &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-1117F-2826-SM-5GZXL | GTEX-1117F-2826-SM-5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | Breast - Mammary Tissue | 0008367 |  1340 | 1008 | ⋯ | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 | breast_mammary_tissue |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX-1117F-2826-SM-5GZXL GTEX-1117F-2826-SM-5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX-1117F-2826-SM-5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                   SMUBRID\n",
       "GTEX-1117F-2826-SM-5GZXL  5.80000 Breast Breast - Mammary Tissue 0008367\n",
       "                         SMTSISCH SMTSPAX ⋯ SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX-1117F-2826-SM-5GZXL  1340    1008    ⋯ 0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX-1117F-2826-SM-5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     \n",
       "                         tissue               \n",
       "GTEX-1117F-2826-SM-5GZXL breast_mammary_tissue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "54"
      ],
      "text/latex": [
       "54"
      ],
      "text/markdown": [
       "54"
      ],
      "text/plain": [
       "[1] 54"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 1851 15531 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17382"
      ],
      "text/latex": [
       "17382"
      ],
      "text/markdown": [
       "17382"
      ],
      "text/plain": [
       "[1] 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>15531</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 15531\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   15531\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    15531 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>15531</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 15531\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 15531\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 15531    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "15072   459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>459</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 459\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   459\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      459 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>459</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 459\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 459\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 459  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 68</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th><th scope=col>tissue</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-1117F-2826-SM-5GZXL</th><td>GTEX-1117F-2826-SM-5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>Breast - Mammary Tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td><td>breast_mammary_tissue</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 68\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY & tissue\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-1117F-2826-SM-5GZXL & GTEX-1117F-2826-SM-5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & Breast - Mammary Tissue & 0008367 &  1340 & 1008 & ⋯ & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4 & breast\\_mammary\\_tissue\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 68\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; | tissue &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-1117F-2826-SM-5GZXL | GTEX-1117F-2826-SM-5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | Breast - Mammary Tissue | 0008367 |  1340 | 1008 | ⋯ | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 | breast_mammary_tissue |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX-1117F-2826-SM-5GZXL GTEX-1117F-2826-SM-5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX-1117F-2826-SM-5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                   SMUBRID\n",
       "GTEX-1117F-2826-SM-5GZXL  5.80000 Breast Breast - Mammary Tissue 0008367\n",
       "                         SMTSISCH SMTSPAX ⋯ SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX-1117F-2826-SM-5GZXL  1340    1008    ⋯ 0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX-1117F-2826-SM-5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     \n",
       "                         tissue               \n",
       "GTEX-1117F-2826-SM-5GZXL breast_mammary_tissue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read in all requirements so that the stage is properly set -- \n",
    "# if it is clear here -- it will remain clear for the rest of the time\n",
    "# tissues.tsv contains the subset of files desired for analysis.\n",
    "tissue_reduction <- read.table(file=\"../assets/tissues.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "\n",
    "# only include those tissues we wish to continue with\n",
    "table(tissue_reduction$include)\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$include==1,]\n",
    "\n",
    "# create a matching tissue name to go with the expressionSet phenotype object\n",
    "pData(obj)$tissue      <- factor(snakecase::to_snake_case(as.character(pData(obj)$SMTSD)))\n",
    "tissue_reduction$SMTSD <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "\n",
    "dim(tissue_reduction)\n",
    "dim(obj)\n",
    "dim(pData(obj))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(obj)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = obj[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "pData(tobj)[1,]\n",
    "# end test\n",
    "\n",
    "length(levels(pData(obj)$tissue))\n",
    "length(levels(tissue_reduction$SMTSD))\n",
    "\n",
    "keep <- pData(obj)$tissue %in% tissue_reduction$SMTSD\n",
    "table(keep)\n",
    "length(keep)\n",
    "\n",
    "dim(obj)\n",
    "dim(pData(obj))\n",
    "reduced_obj <- obj[,keep==TRUE]\n",
    "dim(reduced_obj)\n",
    "dim(pData(reduced_obj))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "pData(tobj)[1,]\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Synchronize accession metadata and phenotype data\n",
    "A kind of transitive closure.   Metadata links the count data to the phenotype data.\n",
    "Begin with synchronizing accession metadata and phenotype data - which has been reduced - `reduced_obj` inputs here "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24667</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24667\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24667\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24667    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>15531</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 15531\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   15531\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    15531 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>15531</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 15531\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 15531\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 15531    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 5159 10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>10372</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 10372\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   10372\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10372</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10372\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10372\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10372    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 68</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th><th scope=col>tissue</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX.1117F.2826.SM.5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>Breast - Mammary Tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td><td>breast_mammary_tissue</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 68\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY & tissue\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX.1117F.2826.SM.5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & Breast - Mammary Tissue & 0008367 &  1340 & 1008 & ⋯ & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4 & breast\\_mammary\\_tissue\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 68\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; | tissue &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX.1117F.2826.SM.5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | Breast - Mammary Tissue | 0008367 |  1340 | 1008 | ⋯ | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 | breast_mammary_tissue |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX.1117F.2826.SM.5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX.1117F.2826.SM.5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                   SMUBRID\n",
       "GTEX.1117F.2826.SM.5GZXL  5.80000 Breast Breast - Mammary Tissue 0008367\n",
       "                         SMTSISCH SMTSPAX ⋯ SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX.1117F.2826.SM.5GZXL  1340    1008    ⋯ 0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX.1117F.2826.SM.5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     \n",
       "                         tissue               \n",
       "GTEX.1117F.2826.SM.5GZXL breast_mammary_tissue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "24667"
      ],
      "text/latex": [
       "24667"
      ],
      "text/markdown": [
       "24667"
      ],
      "text/plain": [
       "[1] 24667"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "10372"
      ],
      "text/latex": [
       "10372"
      ],
      "text/markdown": [
       "10372"
      ],
      "text/plain": [
       "[1] 10372"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>10372</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 10372\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   10372\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10372</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10372\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10372\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10372    68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24667</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24667\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24667\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24667    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 6453 18214 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>18214</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 18214\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 18214\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 18214    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "17722   492 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "492"
      ],
      "text/latex": [
       "492"
      ],
      "text/markdown": [
       "492"
      ],
      "text/plain": [
       "[1] 492"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "286"
      ],
      "text/latex": [
       "286"
      ],
      "text/markdown": [
       "286"
      ],
      "text/plain": [
       "[1] 286"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR821498</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>7879119272</td><td>PRJNA75899</td><td>SAMN01994192</td><td>GTEx</td><td>GTEX-WI4N-1426-SM-3LK7H</td><td>Breast - Mammary Tissue</td><td>⋯</td><td></td><td></td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.WI4N.1426.SM.3LK7H</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR821498 & RNA:Total RNA & RNA-Seq & 152 & 7879119272 & PRJNA75899 & SAMN01994192 & GTEx & GTEX-WI4N-1426-SM-3LK7H & Breast - Mammary Tissue & ⋯ &  &  & NA & NA &  &  &  &  &  & GTEX.WI4N.1426.SM.3LK7H\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR821498 | RNA:Total RNA | RNA-Seq | 152 | 7879119272 | PRJNA75899 | SAMN01994192 | GTEx | GTEX-WI4N-1426-SM-3LK7H | Breast - Mammary Tissue | ⋯ | <!----> | <!----> | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.WI4N.1426.SM.3LK7H |\n",
       "\n"
      ],
      "text/plain": [
       "  Run       analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR821498 RNA:Total RNA RNA-Seq    152        7879119272 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN01994192 GTEx                   GTEX-WI4N-1426-SM-3LK7H         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯                                                    \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.WI4N.1426.SM.3LK7H"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# let's limit the phenotype object and then align the metadata file\n",
    "metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "metadata$SAMPID   <- gsub('-','\\\\.',metadata$'Sample Name')\n",
    "pData(reduced_obj)$SAMPID <- gsub('-','\\\\.',pData(reduced_obj)$SAMPID)\n",
    "\n",
    "dim(metadata)\n",
    "dim(reduced_obj)\n",
    "dim(pData(reduced_obj))\n",
    "rownames(pData(reduced_obj))<- pData(reduced_obj)$SAMPID\n",
    "\n",
    "# keep only those runs (as epitomized by the metadata_samples) in the phenotype set\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj)$SAMPID)\n",
    "\n",
    "# any undefined (N/A) sample names? These results will be zero\n",
    "sum(is.na(metadata_samples))\n",
    "sum(is.na(phenotype_samples))\n",
    "\n",
    "rm(keep)\n",
    "keep <- phenotype_samples %in% metadata_samples\n",
    "table(keep)\n",
    "reduced_obj2 <- reduced_obj[,keep==TRUE]\n",
    "dim(reduced_obj2)\n",
    "dim(pData(reduced_obj2))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "pData(tobj)[1,]\n",
    "# end test\n",
    "\n",
    "# now go the other way - make sure the metadata samples are in sync with the phenotype samples\n",
    "# note that we are now with `reduced_obj2`\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj2)$SAMPID)\n",
    "length(metadata_samples)\n",
    "length(phenotype_samples)\n",
    "dim(reduced_obj2)\n",
    "dim(pData(reduced_obj2))\n",
    "dim(metadata)\n",
    "rm(keep)\n",
    "keep <- metadata_samples %in% phenotype_samples\n",
    "table(keep)\n",
    "\n",
    "reduced_metadata <- metadata[keep==TRUE,]\n",
    "dim(reduced_metadata)\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "breast_metadata_samples   <- as.character(reduced_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "rm(keep)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "breast_samples <- reduced_metadata[keep,]\n",
    "length(breast_samples$SAMPID)\n",
    "length(pData(tobj)$SAMPID)\n",
    "breast_samples[1,]\n",
    "# end test\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  Limit obj and metadata to size of the ijc and sjc columns \n",
    "\n",
    "Things are complicated and can easily lead to data disaster.\n",
    "We need to go forward and backwards with these data - and see how to have them arranged.  \n",
    "Things we know to be true:\n",
    "1. Run numbers are unique\n",
    "2. Samples (which are Donor and Tissue combined) have been sequenced more than once.\n",
    "\n",
    "We need to keep this clear as the metadata are used to sort out tissues, sex, and other factors - and these are based upon a Donor.  Relationships are as follows:\n",
    "\n",
    "* One Donor can have many tissues (one donor has as many as 54 tissues)\n",
    "* One Tissue may have many sequencing runs (One Tissue has been as of this writing sequenced as many as 3 times)\n",
    "\n",
    "The analyst must keep this information in mind as manipulations occur.\n",
    "\n",
    "Here we use `se.jc.ijc` as exemplar as well as model for all subsequent matrices.  Arrangements of `se.jc.ijc` will ripple to all other matrices because sample order is the same between all samples.\n",
    "\n",
    "Entering into this step we have for our phenotype object:\n",
    "1. reduced_obj2\n",
    "2. reduced_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "metadata_for_counts\n",
       "FALSE  TRUE \n",
       " 1653  7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "counts_for_metadata\n",
       "FALSE  TRUE \n",
       "11194  7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>18214</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 18214\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 18214\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 18214    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SRR821653'</li><li>'SRR821690'</li><li>'SRR821715'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SRR821653'\n",
       "\\item 'SRR821690'\n",
       "\\item 'SRR821715'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SRR821653'\n",
       "2. 'SRR821690'\n",
       "3. 'SRR821715'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SRR821653\" \"SRR821690\" \"SRR821715\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SRR821653'</li><li>'SRR821690'</li><li>'SRR821715'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SRR821653'\n",
       "\\item 'SRR821690'\n",
       "\\item 'SRR821715'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SRR821653'\n",
       "2. 'SRR821690'\n",
       "3. 'SRR821715'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SRR821653\" \"SRR821690\" \"SRR821715\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "6891"
      ],
      "text/latex": [
       "6891"
      ],
      "text/markdown": [
       "6891"
      ],
      "text/plain": [
       "[1] 6891"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "15"
      ],
      "text/latex": [
       "15"
      ],
      "text/markdown": [
       "15"
      ],
      "text/plain": [
       "[1] 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "33"
      ],
      "text/latex": [
       "33"
      ],
      "text/markdown": [
       "33"
      ],
      "text/plain": [
       "[1] 33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata_for_counts = colnames(se.jc.ijc)%in% reduced_metadata$Run\n",
    "counts_for_metadata = reduced_metadata$Run %in% colnames(se.jc.ijc)\n",
    "table(metadata_for_counts)\n",
    "table(counts_for_metadata)\n",
    "\n",
    "counts          <- data.matrix(se.jc.ijc)\n",
    "counts_metadata <- reduced_metadata\n",
    "dim(counts)\n",
    "dim(counts_metadata)\n",
    "\n",
    "counts          <- counts         [                         ,metadata_for_counts==TRUE]\n",
    "counts_metadata <- counts_metadata[counts_for_metadata==TRUE,                         ]\n",
    "\n",
    "# now we arrange metadata and counts to be in the same order\n",
    "# Run numbers are unique, but we will replace the columns with SAMPID as metadata is by SAMPID\n",
    "i = order(colnames(counts))\n",
    "j = order(counts_metadata$Run)\n",
    "\n",
    "ose.jc.ijc       = counts[,i]\n",
    "ocounts_metadata = counts_metadata[j,]\n",
    "\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(ocounts_metadata$Run,3)\n",
    "\n",
    "colnames(ose.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(ocounts_metadata$SAMPID,3)\n",
    "sum(table(colnames(ose.jc.ijc))==1)\n",
    "sum(table(colnames(ose.jc.ijc))==2)\n",
    "sum(table(colnames(ose.jc.ijc))==3)\n",
    "sum(table(colnames(ose.jc.ijc))==4)\n",
    "dim(ose.jc.ijc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now extend this organization to all count matrices\n",
    "now that the data are organized and accounted for, extend to the rest of the splicing factors and their count matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td> 2</td><td>2</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>10</td><td>0</td><td>4</td><td>4</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 0 &  2 & 2 & 0 & 0\\\\\n",
       "\t 0 & 10 & 0 & 4 & 4\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\t 0 &  0 & 2 & 0 & 1\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 0 |  2 | 2 | 0 | 0 |\n",
       "| 0 | 10 | 0 | 4 | 4 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "| 0 |  0 | 2 | 0 | 1 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 0           2         2          0          0         \n",
       "[2,] 0          10         0          4          4         \n",
       "[3,] 0           0         1          0          1         \n",
       "[4,] 0           0         2          0          1         \n",
       "[5,] 0           0         1          0          1         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>151</td><td>51</td><td>292</td><td>59</td><td>91</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 151 & 51 & 292 & 59 & 91\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 151 | 51 | 292 | 59 | 91 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 151        51         292        59         91        \n",
       "[2,]   0         0           0         0          0        \n",
       "[3,]   0         0           0         0          0        \n",
       "[4,]   0         0           0         0          0        \n",
       "[5,]   0         0           0         0          0        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  1</td><td> 2</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td> 0</td></tr>\n",
       "\t<tr><td>14</td><td> 3</td><td>  4</td><td> 3</td><td>11</td></tr>\n",
       "\t<tr><td>67</td><td>10</td><td>167</td><td>26</td><td>31</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   1 &  2 &  0\\\\\n",
       "\t  0 &  0 &   0 &  0 &  0\\\\\n",
       "\t  0 &  0 &   0 &  1 &  0\\\\\n",
       "\t 14 &  3 &   4 &  3 & 11\\\\\n",
       "\t 67 & 10 & 167 & 26 & 31\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   1 |  2 |  0 |\n",
       "|  0 |  0 |   0 |  0 |  0 |\n",
       "|  0 |  0 |   0 |  1 |  0 |\n",
       "| 14 |  3 |   4 |  3 | 11 |\n",
       "| 67 | 10 | 167 | 26 | 31 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]  0          0           1         2          0        \n",
       "[2,]  0          0           0         0          0        \n",
       "[3,]  0          0           0         1          0        \n",
       "[4,] 14          3           4         3         11        \n",
       "[5,] 67         10         167        26         31        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>79</td><td>30</td><td>293</td><td>120</td><td>107</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>13</td><td>12</td><td> 10</td><td>  8</td><td> 18</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 79 & 30 & 293 & 120 & 107\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t 13 & 12 &  10 &   8 &  18\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 79 | 30 | 293 | 120 | 107 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "| 13 | 12 |  10 |   8 |  18 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 79         30         293        120        107       \n",
       "[2,]  0          0           0          0          0       \n",
       "[3,]  0          0           0          0          0       \n",
       "[4,] 13         12          10          8         18       \n",
       "[5,]  0          0           0          0          0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>1</td><td> 4</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td> 0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>1</td><td>0</td><td>0</td><td>3</td><td> 2</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>7</td><td>32</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 5 & 3 & 4 & 1 &  4\\\\\n",
       "\t 0 & 0 & 0 & 1 &  0\\\\\n",
       "\t 0 & 0 & 0 & 0 &  0\\\\\n",
       "\t 1 & 0 & 0 & 3 &  2\\\\\n",
       "\t 0 & 0 & 0 & 7 & 32\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 5 | 3 | 4 | 1 |  4 |\n",
       "| 0 | 0 | 0 | 1 |  0 |\n",
       "| 0 | 0 | 0 | 0 |  0 |\n",
       "| 1 | 0 | 0 | 3 |  2 |\n",
       "| 0 | 0 | 0 | 7 | 32 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 5          3          4          1           4        \n",
       "[2,] 0          0          0          1           0        \n",
       "[3,] 0          0          0          0           0        \n",
       "[4,] 1          0          0          3           2        \n",
       "[5,] 0          0          0          7          32        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>36</td><td>7</td><td>10</td><td>15</td><td> 1</td></tr>\n",
       "\t<tr><td> 0</td><td>4</td><td> 1</td><td> 2</td><td> 2</td></tr>\n",
       "\t<tr><td> 0</td><td>0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td>2</td><td> 1</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 4</td><td>8</td><td> 2</td><td>25</td><td>74</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 36 & 7 & 10 & 15 &  1\\\\\n",
       "\t  0 & 4 &  1 &  2 &  2\\\\\n",
       "\t  0 & 0 &  0 &  0 &  0\\\\\n",
       "\t  0 & 2 &  1 &  0 &  0\\\\\n",
       "\t  4 & 8 &  2 & 25 & 74\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 36 | 7 | 10 | 15 |  1 |\n",
       "|  0 | 4 |  1 |  2 |  2 |\n",
       "|  0 | 0 |  0 |  0 |  0 |\n",
       "|  0 | 2 |  1 |  0 |  0 |\n",
       "|  4 | 8 |  2 | 25 | 74 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 36         7          10         15          1        \n",
       "[2,]  0         4           1          2          2        \n",
       "[3,]  0         0           0          0          0        \n",
       "[4,]  0         2           1          0          0        \n",
       "[5,]  4         8           2         25         74        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>  5</td><td> 0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  1</td><td> 1</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td>  0</td><td> 2</td><td> 0</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  4</td><td> 7</td><td> 4</td><td>  2</td></tr>\n",
       "\t<tr><td>64</td><td>113</td><td>43</td><td>40</td><td>247</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t  0 &   5 &  0 &  1 &   1\\\\\n",
       "\t  0 &   1 &  1 &  0 &   0\\\\\n",
       "\t  0 &   0 &  2 &  0 &   1\\\\\n",
       "\t  0 &   4 &  7 &  4 &   2\\\\\n",
       "\t 64 & 113 & 43 & 40 & 247\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|  0 |   5 |  0 |  1 |   1 |\n",
       "|  0 |   1 |  1 |  0 |   0 |\n",
       "|  0 |   0 |  2 |  0 |   1 |\n",
       "|  0 |   4 |  7 |  4 |   2 |\n",
       "| 64 | 113 | 43 | 40 | 247 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]  0           5         0          1           1       \n",
       "[2,]  0           1         1          0           0       \n",
       "[3,]  0           0         2          0           1       \n",
       "[4,]  0           4         7          4           2       \n",
       "[5,] 64         113        43         40         247       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>98</td><td>25</td><td>150</td><td>64</td><td> 37</td></tr>\n",
       "\t<tr><td>11</td><td>34</td><td> 56</td><td>38</td><td>125</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   0 &  1 &   1\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t 98 & 25 & 150 & 64 &  37\\\\\n",
       "\t 11 & 34 &  56 & 38 & 125\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   0 |  1 |   1 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "| 98 | 25 | 150 | 64 |  37 |\n",
       "| 11 | 34 |  56 | 38 | 125 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]  0          0           0         1           1       \n",
       "[2,]  0          0           0         0           0       \n",
       "[3,]  0          0           0         0           0       \n",
       "[4,] 98         25         150        64          37       \n",
       "[5,] 11         34          56        38         125       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>247</td><td>103</td><td>494</td><td>145</td><td>145</td></tr>\n",
       "\t<tr><td>  0</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  1</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  0</td><td>  2</td><td>  6</td><td>  1</td><td>  1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t   0 &   0 &   1 &   0 &   0\\\\\n",
       "\t 247 & 103 & 494 & 145 & 145\\\\\n",
       "\t   0 &   1 &   0 &   0 &   1\\\\\n",
       "\t   1 &   1 &   0 &   0 &   1\\\\\n",
       "\t   0 &   2 &   6 &   1 &   1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|   0 |   0 |   1 |   0 |   0 |\n",
       "| 247 | 103 | 494 | 145 | 145 |\n",
       "|   0 |   1 |   0 |   0 |   1 |\n",
       "|   1 |   1 |   0 |   0 |   1 |\n",
       "|   0 |   2 |   6 |   1 |   1 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]   0          0          1          0          0       \n",
       "[2,] 247        103        494        145        145       \n",
       "[3,]   0          1          0          0          1       \n",
       "[4,]   1          1          0          0          1       \n",
       "[5,]   0          2          6          1          1       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td>1</td><td>6</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>0</td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 0 & 1 & 6 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 1\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 5 & 3 & 4 & 0 & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 0 | 1 | 6 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 1 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 5 | 3 | 4 | 0 | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 0          1          6          1          0         \n",
       "[2,] 0          0          0          0          1         \n",
       "[3,] 0          0          0          0          0         \n",
       "[4,] 0          0          0          0          0         \n",
       "[5,] 5          3          4          0          3         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "oa3ss.jc.ijc  <- data.matrix(a3ss.jc.ijc)\n",
    "oa3ss.jc.sjc  <- data.matrix(a3ss.jc.sjc)\n",
    "oa3ss.jc.ijc  <- oa3ss.jc.ijc[,metadata_for_counts==TRUE]\n",
    "oa3ss.jc.sjc  <- oa3ss.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "oa5ss.jc.ijc  <- data.matrix(a5ss.jc.ijc)\n",
    "oa5ss.jc.sjc  <- data.matrix(a5ss.jc.sjc)\n",
    "oa5ss.jc.ijc  <- oa5ss.jc.ijc[,metadata_for_counts==TRUE]\n",
    "oa5ss.jc.sjc  <- oa5ss.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "omxe.jc.ijc   <- data.matrix(mxe.jc.ijc)\n",
    "omxe.jc.sjc   <- data.matrix(mxe.jc.sjc)\n",
    "omxe.jc.ijc   <- omxe.jc.ijc[,metadata_for_counts==TRUE]\n",
    "omxe.jc.sjc   <- omxe.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "ori.jc.ijc    <- data.matrix(ri.jc.ijc)\n",
    "ori.jc.sjc    <- data.matrix(ri.jc.sjc)\n",
    "ori.jc.ijc    <- ori.jc.ijc[,metadata_for_counts==TRUE]\n",
    "ori.jc.sjc    <- ori.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "ose.jc.ijc    <- data.matrix(se.jc.ijc)\n",
    "ose.jc.sjc    <- data.matrix(se.jc.sjc)\n",
    "ose.jc.ijc    <- ose.jc.ijc[,metadata_for_counts==TRUE]\n",
    "ose.jc.sjc    <- ose.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "oa3ss.jc.ijc  <- oa3ss.jc.ijc[,i]\n",
    "oa3ss.jc.sjc  <- oa3ss.jc.sjc[,i]\n",
    "\n",
    "oa5ss.jc.ijc  <- oa5ss.jc.ijc[,i]\n",
    "oa5ss.jc.sjc  <- oa5ss.jc.sjc[,i]\n",
    "\n",
    "omxe.jc.ijc   <- omxe.jc.ijc[,i]\n",
    "omxe.jc.sjc   <- omxe.jc.sjc[,i]\n",
    "\n",
    "ori.jc.ijc    <- ori.jc.ijc[,i]\n",
    "ori.jc.sjc    <- ori.jc.sjc[,i]\n",
    "\n",
    "ose.jc.ijc    <- ose.jc.ijc  [,i]\n",
    "ose.jc.sjc    <- ose.jc.sjc  [,i]\n",
    "\n",
    "oa3ss.jc.ijc[1:5,1:5]\n",
    "oa3ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "oa5ss.jc.ijc[1:5,1:5]\n",
    "oa5ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "omxe.jc.ijc[1:5,1:5]\n",
    "omxe.jc.sjc[1:5,1:5]\n",
    "\n",
    "ori.jc.ijc[1:5,1:5]\n",
    "ori.jc.sjc[1:5,1:5]\n",
    "\n",
    "ose.jc.ijc[1:5,1:5]\n",
    "ose.jc.sjc[1:5,1:5]\n",
    "\n",
    "colnames(oa3ss.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(oa3ss.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(oa5ss.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(oa5ss.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(omxe.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(omxe.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(ori.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(ori.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(ose.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(ose.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "tail(colnames(oa3ss.jc.ijc),3)\n",
    "tail(colnames(oa3ss.jc.sjc),3)\n",
    "\n",
    "tail(colnames(oa5ss.jc.ijc),3)\n",
    "tail(colnames(oa5ss.jc.sjc),3)\n",
    "\n",
    "tail(colnames(omxe.jc.ijc),3)\n",
    "tail(colnames(omxe.jc.sjc),3)\n",
    "\n",
    "tail(colnames(ori.jc.ijc),3)\n",
    "tail(colnames(ori.jc.sjc),3)\n",
    "\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(colnames(ose.jc.sjc),3)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sanity Check\n",
    "\n",
    "Spot check one tissue and make sure there isn't nonsense\n",
    "Our current matrices are:\n",
    "1. ocounts_metadata for the connection from counts to phenotype via the Run and SAMPID\n",
    "2. pData(reduced_obj2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>68</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 68\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 68\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 6842   178 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "178"
      ],
      "text/latex": [
       "178"
      ],
      "text/markdown": [
       "178"
      ],
      "text/plain": [
       "[1] 178"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "286"
      ],
      "text/latex": [
       "286"
      ],
      "text/markdown": [
       "286"
      ],
      "text/plain": [
       "[1] 286"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR1068977</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>6706681864</td><td>PRJNA75899</td><td>SAMN02465522</td><td>GTEx</td><td>GTEX-XBED-1626-SM-47JYN</td><td>Breast - Mammary Tissue</td><td>⋯</td><td>P-RNA-0002</td><td>P-RNA-0002</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.XBED.1626.SM.47JYN</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR1068977 & RNA:Total RNA & RNA-Seq & 152 & 6706681864 & PRJNA75899 & SAMN02465522 & GTEx & GTEX-XBED-1626-SM-47JYN & Breast - Mammary Tissue & ⋯ & P-RNA-0002 & P-RNA-0002 & NA & NA &  &  &  &  &  & GTEX.XBED.1626.SM.47JYN\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR1068977 | RNA:Total RNA | RNA-Seq | 152 | 6706681864 | PRJNA75899 | SAMN02465522 | GTEx | GTEX-XBED-1626-SM-47JYN | Breast - Mammary Tissue | ⋯ | P-RNA-0002 | P-RNA-0002 | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.XBED.1626.SM.47JYN |\n",
       "\n"
      ],
      "text/plain": [
       "  Run        analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR1068977 RNA:Total RNA RNA-Seq    152        6706681864 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN02465522 GTEx                   GTEX-XBED-1626-SM-47JYN         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯ P-RNA-0002                P-RNA-0002               \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.XBED.1626.SM.47JYN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$tissue== \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "breast_metadata_samples   <- as.character(ocounts_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "rm(keep)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "breast_samples <- ocounts_metadata[keep,]\n",
    "length(breast_samples$SAMPID)\n",
    "length(pData(tobj)$SAMPID)\n",
    "breast_samples[1,]\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### which metadata shall we use?\n",
    "There is a lack of correspondeence between the metadata in the Sequence Read Archive table as obtained from dbGaP SRR Run Explorer and the GTEx V8 Release.   Not sure which one to use!  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "20-29 30-39 40-49 50-59 60-69 70-79 \n",
       "  760   771  1729  3484  3312   316 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "   0    1    2    3    4 \n",
       "5523  467 2738  466 1082 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "        B1     B1, A1         C1     C1, A1 C1, B1, A1         D1     D1, A1 \n",
       "      6154        377       3110        603          3         70         55 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "   1    2 \n",
       "6796 3576 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "female   male \n",
       "  2501   4519 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(pData(reduced_obj2)$AGE)\n",
    "table(pData(reduced_obj2)$DTHHRDY)\n",
    "table(pData(reduced_obj2)$SMCENTER)\n",
    "table(pData(reduced_obj2)$SEX)\n",
    "table(ocounts_metadata$sex)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Exploring the ijc and sjc Count data \n",
    "\n",
    "For each sample, we have ijc and sjc count data, for each alternative splicing type, we have the following number of alternative splicing events:\n",
    "\n",
    "* `A3SS`: ` 8,920`\n",
    "* `A5SS`: ` 5,584`\n",
    "* `MXE `: ` 2,979`\n",
    "* `RI`  : ` 6,312`\n",
    "* `SE  `: `42,611`\n",
    "\n",
    "For exon skipping events (SE), we have 42,611 non-zero junction IDs the (first dimension of the ijc and sjc cout table) for the skipped exon event for breast-Mammary Tissue, 191 individuals.  These are healthy individuals, and we are studying the impact of sex on the occurrence or non-occurance of specific alternative splicing events.   We explore the information we ahve about these junctions and create a construct, as_event, which accounts for the junction under exploration.\n",
    "\n",
    "The `IJC` and `SJC` counts are in many ways two sides of the same coin.  Both our the observational output and we wish to see how robust each are in their ability to separate out the samples to provide for us differentially expressed isoform events as measured by their counts.   Each junction is in a manner a specific marker to specific isoform events that may or may not be shared between the genders.   If there is significant results, then this is indicative of the separation achieved by isoform specific differentiation.   In our model we will use these in combination, it is important to see if they will yield the results we are looking for.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reduce count matrices based upon the reduced tissue information - count matrix by column adjustment\n",
    "\n",
    "We wish to keep only those samples (columns) in our count data that are a member of the reduced tissue set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "keep\n",
       "TRUE \n",
       "7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td> 2</td><td>2</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>10</td><td>0</td><td>4</td><td>4</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 0 &  2 & 2 & 0 & 0\\\\\n",
       "\t 0 & 10 & 0 & 4 & 4\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\t 0 &  0 & 2 & 0 & 1\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 0 |  2 | 2 | 0 | 0 |\n",
       "| 0 | 10 | 0 | 4 | 4 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "| 0 |  0 | 2 | 0 | 1 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 0                        2                      2                      \n",
       "[2,] 0                       10                      0                      \n",
       "[3,] 0                        0                      1                      \n",
       "[4,] 0                        0                      2                      \n",
       "[5,] 0                        0                      1                      \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 0                       0                      \n",
       "[2,] 4                       4                      \n",
       "[3,] 0                       1                      \n",
       "[4,] 0                       1                      \n",
       "[5,] 0                       1                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>151</td><td>51</td><td>292</td><td>59</td><td>91</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 151 & 51 & 292 & 59 & 91\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 151 | 51 | 292 | 59 | 91 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 151                     51                      292                    \n",
       "[2,]   0                      0                        0                    \n",
       "[3,]   0                      0                        0                    \n",
       "[4,]   0                      0                        0                    \n",
       "[5,]   0                      0                        0                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 59                      91                     \n",
       "[2,]  0                       0                     \n",
       "[3,]  0                       0                     \n",
       "[4,]  0                       0                     \n",
       "[5,]  0                       0                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  1</td><td> 2</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td> 0</td></tr>\n",
       "\t<tr><td>14</td><td> 3</td><td>  4</td><td> 3</td><td>11</td></tr>\n",
       "\t<tr><td>67</td><td>10</td><td>167</td><td>26</td><td>31</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   1 &  2 &  0\\\\\n",
       "\t  0 &  0 &   0 &  0 &  0\\\\\n",
       "\t  0 &  0 &   0 &  1 &  0\\\\\n",
       "\t 14 &  3 &   4 &  3 & 11\\\\\n",
       "\t 67 & 10 & 167 & 26 & 31\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   1 |  2 |  0 |\n",
       "|  0 |  0 |   0 |  0 |  0 |\n",
       "|  0 |  0 |   0 |  1 |  0 |\n",
       "| 14 |  3 |   4 |  3 | 11 |\n",
       "| 67 | 10 | 167 | 26 | 31 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]  0                       0                        1                    \n",
       "[2,]  0                       0                        0                    \n",
       "[3,]  0                       0                        0                    \n",
       "[4,] 14                       3                        4                    \n",
       "[5,] 67                      10                      167                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]  2                       0                     \n",
       "[2,]  0                       0                     \n",
       "[3,]  1                       0                     \n",
       "[4,]  3                      11                     \n",
       "[5,] 26                      31                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>79</td><td>30</td><td>293</td><td>120</td><td>107</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>13</td><td>12</td><td> 10</td><td>  8</td><td> 18</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 79 & 30 & 293 & 120 & 107\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t 13 & 12 &  10 &   8 &  18\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 79 | 30 | 293 | 120 | 107 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "| 13 | 12 |  10 |   8 |  18 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 79                      30                      293                    \n",
       "[2,]  0                       0                        0                    \n",
       "[3,]  0                       0                        0                    \n",
       "[4,] 13                      12                       10                    \n",
       "[5,]  0                       0                        0                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 120                     107                    \n",
       "[2,]   0                       0                    \n",
       "[3,]   0                       0                    \n",
       "[4,]   8                      18                    \n",
       "[5,]   0                       0                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>1</td><td> 4</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td> 0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>1</td><td>0</td><td>0</td><td>3</td><td> 2</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>7</td><td>32</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 5 & 3 & 4 & 1 &  4\\\\\n",
       "\t 0 & 0 & 0 & 1 &  0\\\\\n",
       "\t 0 & 0 & 0 & 0 &  0\\\\\n",
       "\t 1 & 0 & 0 & 3 &  2\\\\\n",
       "\t 0 & 0 & 0 & 7 & 32\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 5 | 3 | 4 | 1 |  4 |\n",
       "| 0 | 0 | 0 | 1 |  0 |\n",
       "| 0 | 0 | 0 | 0 |  0 |\n",
       "| 1 | 0 | 0 | 3 |  2 |\n",
       "| 0 | 0 | 0 | 7 | 32 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 5                       3                       4                      \n",
       "[2,] 0                       0                       0                      \n",
       "[3,] 0                       0                       0                      \n",
       "[4,] 1                       0                       0                      \n",
       "[5,] 0                       0                       0                      \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 1                        4                     \n",
       "[2,] 1                        0                     \n",
       "[3,] 0                        0                     \n",
       "[4,] 3                        2                     \n",
       "[5,] 7                       32                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>36</td><td>7</td><td>10</td><td>15</td><td> 1</td></tr>\n",
       "\t<tr><td> 0</td><td>4</td><td> 1</td><td> 2</td><td> 2</td></tr>\n",
       "\t<tr><td> 0</td><td>0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td>2</td><td> 1</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 4</td><td>8</td><td> 2</td><td>25</td><td>74</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 36 & 7 & 10 & 15 &  1\\\\\n",
       "\t  0 & 4 &  1 &  2 &  2\\\\\n",
       "\t  0 & 0 &  0 &  0 &  0\\\\\n",
       "\t  0 & 2 &  1 &  0 &  0\\\\\n",
       "\t  4 & 8 &  2 & 25 & 74\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 36 | 7 | 10 | 15 |  1 |\n",
       "|  0 | 4 |  1 |  2 |  2 |\n",
       "|  0 | 0 |  0 |  0 |  0 |\n",
       "|  0 | 2 |  1 |  0 |  0 |\n",
       "|  4 | 8 |  2 | 25 | 74 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 36                      7                       10                     \n",
       "[2,]  0                      4                        1                     \n",
       "[3,]  0                      0                        0                     \n",
       "[4,]  0                      2                        1                     \n",
       "[5,]  4                      8                        2                     \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 15                       1                     \n",
       "[2,]  2                       2                     \n",
       "[3,]  0                       0                     \n",
       "[4,]  0                       0                     \n",
       "[5,] 25                      74                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>  5</td><td> 0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  1</td><td> 1</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td>  0</td><td> 2</td><td> 0</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  4</td><td> 7</td><td> 4</td><td>  2</td></tr>\n",
       "\t<tr><td>64</td><td>113</td><td>43</td><td>40</td><td>247</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t  0 &   5 &  0 &  1 &   1\\\\\n",
       "\t  0 &   1 &  1 &  0 &   0\\\\\n",
       "\t  0 &   0 &  2 &  0 &   1\\\\\n",
       "\t  0 &   4 &  7 &  4 &   2\\\\\n",
       "\t 64 & 113 & 43 & 40 & 247\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|  0 |   5 |  0 |  1 |   1 |\n",
       "|  0 |   1 |  1 |  0 |   0 |\n",
       "|  0 |   0 |  2 |  0 |   1 |\n",
       "|  0 |   4 |  7 |  4 |   2 |\n",
       "| 64 | 113 | 43 | 40 | 247 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]  0                        5                      0                     \n",
       "[2,]  0                        1                      1                     \n",
       "[3,]  0                        0                      2                     \n",
       "[4,]  0                        4                      7                     \n",
       "[5,] 64                      113                     43                     \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]  1                        1                    \n",
       "[2,]  0                        0                    \n",
       "[3,]  0                        1                    \n",
       "[4,]  4                        2                    \n",
       "[5,] 40                      247                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>98</td><td>25</td><td>150</td><td>64</td><td> 37</td></tr>\n",
       "\t<tr><td>11</td><td>34</td><td> 56</td><td>38</td><td>125</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   0 &  1 &   1\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t 98 & 25 & 150 & 64 &  37\\\\\n",
       "\t 11 & 34 &  56 & 38 & 125\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   0 |  1 |   1 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "| 98 | 25 | 150 | 64 |  37 |\n",
       "| 11 | 34 |  56 | 38 | 125 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]  0                       0                        0                    \n",
       "[2,]  0                       0                        0                    \n",
       "[3,]  0                       0                        0                    \n",
       "[4,] 98                      25                      150                    \n",
       "[5,] 11                      34                       56                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]  1                        1                    \n",
       "[2,]  0                        0                    \n",
       "[3,]  0                        0                    \n",
       "[4,] 64                       37                    \n",
       "[5,] 38                      125                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>247</td><td>103</td><td>494</td><td>145</td><td>145</td></tr>\n",
       "\t<tr><td>  0</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  1</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  0</td><td>  2</td><td>  6</td><td>  1</td><td>  1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t   0 &   0 &   1 &   0 &   0\\\\\n",
       "\t 247 & 103 & 494 & 145 & 145\\\\\n",
       "\t   0 &   1 &   0 &   0 &   1\\\\\n",
       "\t   1 &   1 &   0 &   0 &   1\\\\\n",
       "\t   0 &   2 &   6 &   1 &   1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|   0 |   0 |   1 |   0 |   0 |\n",
       "| 247 | 103 | 494 | 145 | 145 |\n",
       "|   0 |   1 |   0 |   0 |   1 |\n",
       "|   1 |   1 |   0 |   0 |   1 |\n",
       "|   0 |   2 |   6 |   1 |   1 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]   0                       0                       1                    \n",
       "[2,] 247                     103                     494                    \n",
       "[3,]   0                       1                       0                    \n",
       "[4,]   1                       1                       0                    \n",
       "[5,]   0                       2                       6                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]   0                       0                    \n",
       "[2,] 145                     145                    \n",
       "[3,]   0                       1                    \n",
       "[4,]   0                       1                    \n",
       "[5,]   1                       1                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td>1</td><td>6</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>0</td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 0 & 1 & 6 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 1\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 5 & 3 & 4 & 0 & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 0 | 1 | 6 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 1 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 5 | 3 | 4 | 0 | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 0                       1                       6                      \n",
       "[2,] 0                       0                       0                      \n",
       "[3,] 0                       0                       0                      \n",
       "[4,] 0                       0                       0                      \n",
       "[5,] 5                       3                       4                      \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 1                       0                      \n",
       "[2,] 0                       1                      \n",
       "[3,] 0                       0                      \n",
       "[4,] 0                       0                      \n",
       "[5,] 0                       3                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata_names <- as.character(ocounts_metadata$SAMPID)\n",
    "count_names    <- as.character(colnames(ose.jc.ijc))\n",
    "\n",
    "keep  <- count_names %in% metadata_names\n",
    "\n",
    "table(keep)\n",
    "length(keep)\n",
    "length(metadata_names)\n",
    "length(count_names)\n",
    "\n",
    "a3ss.jc.ijc2 = oa3ss.jc.ijc[,keep==TRUE] \n",
    "a3ss.jc.sjc2 = oa3ss.jc.sjc[,keep==TRUE] \n",
    "\n",
    "a5ss.jc.ijc2 = oa5ss.jc.ijc[,keep==TRUE] \n",
    "a5ss.jc.sjc2 = oa5ss.jc.sjc[,keep==TRUE] \n",
    "\n",
    "mxe.jc.ijc2 = omxe.jc.ijc[,keep==TRUE] \n",
    "mxe.jc.sjc2 = omxe.jc.sjc[,keep==TRUE] \n",
    "\n",
    "ri.jc.ijc2 = ori.jc.ijc[,keep==TRUE] \n",
    "ri.jc.sjc2 = ori.jc.sjc[,keep==TRUE] \n",
    "\n",
    "se.jc.ijc2 = ose.jc.ijc[,keep==TRUE] \n",
    "se.jc.sjc2 = ose.jc.sjc[,keep==TRUE] \n",
    "\n",
    "dim(a3ss.jc.ijc2)\n",
    "dim(a3ss.jc.sjc2)\n",
    "dim(a5ss.jc.ijc2)\n",
    "dim(a5ss.jc.sjc2)\n",
    "dim(mxe.jc.ijc2)\n",
    "dim(mxe.jc.sjc2)\n",
    "dim(ri.jc.ijc2)\n",
    "dim(ri.jc.sjc2)\n",
    "dim(se.jc.ijc2)\n",
    "dim(se.jc.sjc2)\n",
    "\n",
    "a3ss.jc.ijc2[1:5,1:5]\n",
    "a3ss.jc.sjc2[1:5,1:5]\n",
    "\n",
    "a5ss.jc.ijc2[1:5,1:5]\n",
    "a5ss.jc.sjc2[1:5,1:5]\n",
    "\n",
    "mxe.jc.ijc2[1:5,1:5]\n",
    "mxe.jc.sjc2[1:5,1:5]\n",
    "\n",
    "ri.jc.ijc2[1:5,1:5]\n",
    "ri.jc.sjc2[1:5,1:5]\n",
    "\n",
    "se.jc.ijc2[1:5,1:5]\n",
    "se.jc.sjc2[1:5,1:5]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Keeping only chromosomes shared male female - count matrix by junction row adjustment \n",
    "\n",
    "The Y chromosome spans more than 59 million base pairs of DNA and represents almost 2 percent of the total DNA in cells. Each person normally has one pair of sex chromosomes in each cell. The Y chromosome is present in males, who have one X and one Y chromosome, while females have two X chromosomes. Since our analysis is on the comparative differences, we must eliminate chrY from our analyses.\n",
    "\n",
    "To do so, we grab the annotation from the GTF file and remove those junctions that correspond to the genes on this chromosome.\n",
    "\n",
    "So this is a count matrix row adjustment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611    11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5177"
      ],
      "text/latex": [
       "5177"
      ],
      "text/markdown": [
       "5177"
      ],
      "text/plain": [
       "[1] 5177"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3824"
      ],
      "text/latex": [
       "3824"
      ],
      "text/markdown": [
       "3824"
      ],
      "text/plain": [
       "[1] 3824"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1619"
      ],
      "text/latex": [
       "1619"
      ],
      "text/markdown": [
       "1619"
      ],
      "text/plain": [
       "[1] 1619"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3617"
      ],
      "text/latex": [
       "3617"
      ],
      "text/markdown": [
       "3617"
      ],
      "text/plain": [
       "[1] 3617"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "13390"
      ],
      "text/latex": [
       "13390"
      ],
      "text/markdown": [
       "13390"
      ],
      "text/plain": [
       "[1] 13390"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  862   230   643   552   108   304   326   537   626   118   783   578   241 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   85   246   514   260   337   388   377   255   264   269    17 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  496   155   386   354    59   212   188   348   410    74   400   387   135 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   60   129   368   184   210   216   284   166   162   191    10 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  286   120   152   166    46   109    97   137   154    70   116   386    67 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   40    49   205   144   127   155   106   104    74    61     8 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  604   117   475   423    46   241   191   441   541    40   637   327   153 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   64   194   371   133   181   284   250   185   192   207    15 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       " 3812  1458  2541  2540   663  1551  1544  2039  2538   742  2552  3547   939 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "  493   895  2938  1803  1863  1890  2053  1636  1313  1134   127 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A3SS.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   17  8903 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "8920"
      ],
      "text/latex": [
       "8920"
      ],
      "text/markdown": [
       "8920"
      ],
      "text/plain": [
       "[1] 8920"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A5SS.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   10  5574 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5584"
      ],
      "text/latex": [
       "5584"
      ],
      "text/markdown": [
       "5584"
      ],
      "text/plain": [
       "[1] 5584"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "MXE.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "    8  2971 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2979"
      ],
      "text/latex": [
       "2979"
      ],
      "text/markdown": [
       "2979"
      ],
      "text/plain": [
       "[1] 2979"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RI.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   15  6297 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "6312"
      ],
      "text/latex": [
       "6312"
      ],
      "text/markdown": [
       "6312"
      ],
      "text/plain": [
       "[1] 6312"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "SE.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "  127 42484 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "42611"
      ],
      "text/latex": [
       "42611"
      ],
      "text/markdown": [
       "42611"
      ],
      "text/plain": [
       "[1] 42611"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484    11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(fromGTF.A3SS)\n",
    "dim(fromGTF.A5SS)\n",
    "dim(fromGTF.MXE)\n",
    "dim(fromGTF.RI)\n",
    "dim(fromGTF.SE)\n",
    "\n",
    "A3SS.genes <- factor(fromGTF.A3SS$geneSymbol)\n",
    "A5SS.genes <- factor(fromGTF.A5SS$geneSymbol)\n",
    "MXE.genes  <- factor(fromGTF.MXE$geneSymbol)\n",
    "RI.genes   <- factor(fromGTF.RI$geneSymbol)\n",
    "SE.genes   <- factor(fromGTF.SE$geneSymbol)\n",
    "\n",
    "length(levels(A3SS.genes))    \n",
    "length(levels(A5SS.genes))    \n",
    "length(levels(MXE.genes))    \n",
    "length(levels(RI.genes))    \n",
    "length(levels(SE.genes))    \n",
    "\n",
    "table(fromGTF.A3SS$chr)\n",
    "table(fromGTF.A5SS$chr)\n",
    "table(fromGTF.MXE$chr)\n",
    "table(fromGTF.RI$chr)\n",
    "table(fromGTF.SE$chr)\n",
    "\n",
    "A3SS.keepAllJunctionsButChrY <- (fromGTF.A3SS$chr != \"chrY\")\n",
    "A5SS.keepAllJunctionsButChrY <- (fromGTF.A5SS$chr != \"chrY\")\n",
    "MXE.keepAllJunctionsButChrY  <- (fromGTF.MXE$chr  != \"chrY\")\n",
    "RI.keepAllJunctionsButChrY   <- (fromGTF.RI$chr   != \"chrY\")\n",
    "SE.keepAllJunctionsButChrY   <- (fromGTF.SE$chr   != \"chrY\")\n",
    "\n",
    "table(A3SS.keepAllJunctionsButChrY)\n",
    "sum(table(A3SS.keepAllJunctionsButChrY))\n",
    "\n",
    "table(A5SS.keepAllJunctionsButChrY)\n",
    "sum(table(A5SS.keepAllJunctionsButChrY))\n",
    "\n",
    "table(MXE.keepAllJunctionsButChrY)\n",
    "sum(table(MXE.keepAllJunctionsButChrY))\n",
    "\n",
    "table(RI.keepAllJunctionsButChrY)\n",
    "sum(table(RI.keepAllJunctionsButChrY))\n",
    "\n",
    "table(SE.keepAllJunctionsButChrY)\n",
    "sum(table(SE.keepAllJunctionsButChrY))\n",
    "\n",
    "a3ss_fromGTF_no_chrY          <- fromGTF.A3SS[A3SS.keepAllJunctionsButChrY,]\n",
    "a3ss_jc_ijc_no_chrY           <- a3ss.jc.ijc2[A3SS.keepAllJunctionsButChrY,]\n",
    "a3ss_jc_sjc_no_chrY           <- a3ss.jc.sjc2[A3SS.keepAllJunctionsButChrY,]\n",
    "rownames(a3ss_jc_ijc_no_chrY) <- rownames(a3ss_fromGTF_no_chrY)\n",
    "rownames(a3ss_jc_sjc_no_chrY) <- rownames(a3ss_fromGTF_no_chrY)\n",
    "\n",
    "a5ss_fromGTF_no_chrY          <- fromGTF.A5SS[A5SS.keepAllJunctionsButChrY,]\n",
    "a5ss_jc_ijc_no_chrY           <- a5ss.jc.ijc2[A5SS.keepAllJunctionsButChrY,]\n",
    "a5ss_jc_sjc_no_chrY           <- a5ss.jc.sjc2[A5SS.keepAllJunctionsButChrY,]\n",
    "rownames(a5ss_jc_ijc_no_chrY) <- rownames(a5ss_fromGTF_no_chrY)\n",
    "rownames(a5ss_jc_sjc_no_chrY) <- rownames(a5ss_fromGTF_no_chrY)\n",
    "\n",
    "mxe_fromGTF_no_chrY          <- fromGTF.MXE[MXE.keepAllJunctionsButChrY,]\n",
    "mxe_jc_ijc_no_chrY           <- mxe.jc.ijc2[MXE.keepAllJunctionsButChrY,]\n",
    "mxe_jc_sjc_no_chrY           <- mxe.jc.sjc2[MXE.keepAllJunctionsButChrY,]\n",
    "rownames(mxe_jc_ijc_no_chrY) <- rownames(mxe_fromGTF_no_chrY)\n",
    "rownames(mxe_jc_sjc_no_chrY) <- rownames(mxe_fromGTF_no_chrY)\n",
    "\n",
    "ri_fromGTF_no_chrY          <- fromGTF.RI[RI.keepAllJunctionsButChrY,]\n",
    "ri_jc_ijc_no_chrY           <- ri.jc.ijc2[RI.keepAllJunctionsButChrY,]\n",
    "ri_jc_sjc_no_chrY           <- ri.jc.sjc2[RI.keepAllJunctionsButChrY,]\n",
    "rownames(ri_jc_ijc_no_chrY) <- rownames(ri_fromGTF_no_chrY)\n",
    "rownames(ri_jc_sjc_no_chrY) <- rownames(ri_fromGTF_no_chrY)\n",
    "\n",
    "se_fromGTF_no_chrY          <- fromGTF.SE[SE.keepAllJunctionsButChrY,]\n",
    "se_jc_ijc_no_chrY           <- se.jc.ijc2[SE.keepAllJunctionsButChrY,]\n",
    "se_jc_sjc_no_chrY           <- se.jc.sjc2[SE.keepAllJunctionsButChrY,]\n",
    "rownames(se_jc_ijc_no_chrY) <- rownames(se_fromGTF_no_chrY)\n",
    "rownames(se_jc_sjc_no_chrY) <- rownames(se_fromGTF_no_chrY)\n",
    "\n",
    "dim(a3ss_fromGTF_no_chrY)\n",
    "dim(a5ss_fromGTF_no_chrY)\n",
    "dim(mxe_fromGTF_no_chrY)\n",
    "dim(ri_fromGTF_no_chrY)\n",
    "dim(se_fromGTF_no_chrY)\n",
    "\n",
    "dim(a3ss_jc_ijc_no_chrY)\n",
    "dim(a3ss_jc_sjc_no_chrY)\n",
    "\n",
    "dim(a5ss_jc_ijc_no_chrY)\n",
    "dim(a5ss_jc_sjc_no_chrY)\n",
    "\n",
    "dim(mxe_jc_ijc_no_chrY)\n",
    "dim(mxe_jc_sjc_no_chrY)\n",
    "\n",
    "dim(ri_jc_ijc_no_chrY)\n",
    "dim(ri_jc_sjc_no_chrY)\n",
    "\n",
    "dim(se_jc_ijc_no_chrY)\n",
    "dim(se_jc_sjc_no_chrY)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exploratory and Differential analysis as_event:ijc, sjc \n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) to transform junction counts (reads that were aligned to junctions when an exon is included - ijc, and reads that were aligned to junctions when the exon is excluded - sjc) with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma.    In each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression: \n",
    "\n",
    "           y = B0 + B1 sex + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the included exon junction count expression; sex denotes the reported sex of the subject\n",
    "\n",
    "## Differential analysis as_event (combined ijc and sjc)\n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) to transform junction counts (reads that were aligned to junctions when an exon is included - ijc, and reads that were aligned to junctions when the exon is excluded - sjc) with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma.    In each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression: \n",
    "\n",
    "           y = B0 + B1 sex + B2 as_event + B3 sex*as_event + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the alternative splicing event expression; sex denotes the reported sex of the subject, as_event represents the specific alternative splicing event - either included exon junction counts or skipped exon junction counts and their interaction terms.   Donor is added to our model as a blocking variable used in both the calculation of duplicate correlation as well as in the linear fit.\n",
    "\n",
    "### Voom, limma's lmFit and eBayes\n",
    "\n",
    "Using sample as a blocking variable, we are able to model the effects of the donor on the results, which improves the power.  This topic is discussed in biostars https://www.biostars.org/p/54565/.  And Gordon Smyth answers the question here https://mailman.stat.ethz.ch/pipermail/bioconductor/2014-February/057887.html.  The method of modeling is a random effects approach in which the intra-donor correlation is incorporated into the covariance matrix instead of the linear predictor.   And though as Gordon Smyth states both are good method and the twoway anova approach makes fewer assumptions, the random effects approach is statistically more powerful.  \n",
    "\n",
    "We have a balanced design in which all donors receive all stimuli (which is really in healthy human donors, life and all of its factors!) Our measurement has so many points -- we are measuring in the skipped exon approach, 42,611 junctions!   It is not possible to encorporate those measurements into the linear predictor.  A two-way ANOVA approach is virtually as powerful as the random effects approach \n",
    "and hence is preferable as it makes fewer assumptions.\n",
    "\n",
    "For an unbalanced design in which each donor receives only a subset of the stimula, the random effects approach is more powerful.\n",
    "\n",
    "Random effects approach is equivalent to The first method is twoway anova, a generalization of a paired analysis.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "print_exploratory_plots <- function (plot, dup, tissue_of_interest, splice_type, fromGTF, tissue_list, ijc, sjc, obj, metadata ) {\n",
    "\n",
    "    fromGTF           <- fromGTF\n",
    "    tissue_true       <- tissue_list == tissue_of_interest\n",
    "\n",
    "    # test to make sure we don't have nonsense\n",
    "    # like proteins - this process moves in one direction\n",
    "    # tissue will get us the phenotype selection \n",
    "    # from the phenotype SAMPID we get the metadata SAMPID\n",
    "    # which leads us to counts - it goes in this direction.\n",
    "    # first phenotype obj subsetting for the tissue_of_interest\n",
    "\n",
    "    message(\"\\nLimiting phenotype data to tissue of interest\\n\",\n",
    "           paste(tissue_of_interest),collapse=\" \")    \n",
    "    keep = pData(obj)$tissue== tissue_of_interest\n",
    "    message(\"\\nkeep\\n\",\n",
    "           paste(table(keep)),collapse=\" \")    \n",
    "    tissue_obj<- obj[,keep]\n",
    "    message(\"\\ntissue_obj now reduced to the tissue of interest\\n\",\n",
    "            paste(dim(pData(tissue_obj)), collapse=\" \"))\n",
    "    message(\"\\nsample value\\n\",\n",
    "            paste(   pData(tissue_obj)[1,], collapse=\" \"))\n",
    "\n",
    "    # now we get the runs via the metadata matrix via SAMPID\n",
    "    # limit the phenotype info to just the samples we care about\n",
    "    metadata_samples          <- as.character(metadata$SAMPID)\n",
    "    tissue_phenotype_samples  <- as.character(pData(tissue_obj)$SAMPID)\n",
    "\n",
    "    rm(keep)\n",
    "    keep = tissue_phenotype_samples %in% metadata_samples\n",
    "    message(\"\\nLimit the phenotypes to those we have samples for\\n\", \n",
    "        paste(table(keep), collapse = \" \") )\n",
    "    message(\"\\ntissue_count_obj now has these count specific phenotypes\")    \n",
    "    tissue_count_obj        <- tissue_obj[keep,]\n",
    "    pData(tissue_count_obj) <- pData(tissue_obj)[keep,]\n",
    "    message(\"\\nDimensions of tissue_count_obj\\n\", \n",
    "        paste(dim(pData(tissue_count_obj)), collapse = \" \") )\n",
    "    message(\"\\nDimensions of tissue_count_obj \\n\", \n",
    "        paste( dim(tissue_count_obj), collapse = \" \") )\n",
    "\n",
    "    # and vice-versa, limit our samples to the ones we have phenotype for\n",
    "    rm(keep)\n",
    "    metadata_samples       <- as.character(metadata$SAMPID)\n",
    "    phenotype_samples      <- as.character(pData(tissue_count_obj)$SAMPID)\n",
    "    keep = metadata_samples %in% phenotype_samples\n",
    "    message(\"\\nLimit the counts now has these count specific phenotypes \\n\", \n",
    "        paste( table(keep), collapse = \" \") )\n",
    "    table(keep)\n",
    "\n",
    "    ijc_tissue        <- ijc     [          ,keep==TRUE]\n",
    "    sjc_tissue        <- sjc     [          ,keep==TRUE]\n",
    "    metadata_tissue   <- metadata[keep==TRUE,          ]\n",
    "\n",
    "    message(\"\\ndimensions of the ijc_tissue \\n\", \n",
    "        paste(  dim(ijc_tissue), collapse = \" \") )\n",
    "    message(\"\\ndimensions of the ijc_tissue \\n\", \n",
    "        paste(  dim(sjc_tissue), collapse = \" \") )\n",
    "    message(\"\\ndimensions of the metadata_tissue \\n\", \n",
    "        paste(  dim(metadata_tissue), collapse = \" \") )\n",
    "   \n",
    "    metadata_samples       <- as.character(metadata_tissue$SAMPID)\n",
    "    phenotype_samples      <- as.character(pData(tissue_count_obj)$SAMPID)\n",
    "\n",
    "    length(metadata_samples)\n",
    "    length(phenotype_samples)\n",
    "    non_overlaps <- setdiff( metadata_samples, phenotype_samples)\n",
    "    message(\"\\nThe non-overlapping IDs between pheno and count data are:\\n\", \n",
    "        paste(length(non_overlaps), collapse = \" \") )\n",
    "\n",
    "    for (i in (1:length(metadata_samples))) {\n",
    "        sample = metadata_tissue[i,]$SAMPID\n",
    "        sample_sex = pData(tissue_count_obj)$SEX[pData(tissue_count_obj)$SAMPID == sample]\n",
    "        if (i==1) {\n",
    "            sex = sample_sex\n",
    "        } else {\n",
    "            sex = c(sex,sample_sex)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    message(\"sex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\"\\n\"))\n",
    "\n",
    "    ijc.dm            <- data.matrix(ijc_tissue)\n",
    "    sjc.dm            <- data.matrix(sjc_tissue)    \n",
    "\n",
    "    sex      <- ifelse(sex == 1,\"male\",\"female\")\n",
    "    sex      <- factor(sex,levels=c(\"male\",\"female\"))\n",
    "    message(\"\\nsex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\" \"))\n",
    "\n",
    "    design    <- model.matrix ( ~ sex)\n",
    "    message(\"\\ndesign matrix ijc, alone:\\n\",\n",
    "        paste0(head(design), collapse=\"\\n\"))\n",
    "\n",
    "    colnames(design) <- c(\"intercept\",\"sex\")\n",
    "\n",
    "    y_ijc <- DGEList(counts=ijc.dm, group = sex)\n",
    "    y_ijc <- calcNormFactors(y_ijc, method=\"RLE\")\n",
    "    y_ijc_voom <- voom (y_ijc, design=design, plot=plot)\n",
    "\n",
    "    Gender <- substring(sex,1,1)\n",
    "    if (dup == TRUE) {\n",
    "        pdf_sub_directory = '../pdf_dup_corr/'\n",
    "        csv_sub_directory = '../significant_events_dup_corr/'\n",
    "    } else {\n",
    "        pdf_sub_directory = '../pdf_no_dup_corr/'\n",
    "        csv_directory = '../significant_events_no_dup_corr/'\n",
    "    }\n",
    "    message(\"mkdir -p pdf_sub_directory\")\n",
    "    system (\"mkdir -p pdf_sub_directory\")\n",
    "    message(\"mkdir -p csv_sub_directory\")\n",
    "    system (\"mkdir -p csv_sub_directory\")\n",
    "    \n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_ijc, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-ijc-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "        plotMDS(y_ijc_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    " \n",
    "    fit_ijc <- lmFit(y_ijc_voom, design)\n",
    "    fit_ijc <- eBayes(fit_ijc)\n",
    "\n",
    "    ijc_sex_results                    <- topTable(fit_ijc, coef='sex', number=nrow(y_ijc_voom))\n",
    "    ijc_sex_results_refined            <- ijc_sex_results$adj.P.Val <= 0.05 & abs(ijc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    ijc_sex_rnResults                  <- rownames(ijc_sex_results)\n",
    "    ijc_sex_resultsAnnotations         <- fromGTF[ijc_sex_rnResults,]\n",
    "\n",
    "    ijc_sex_results_refinedAnnotations <- ijc_sex_resultsAnnotations[ijc_sex_results_refined      ==TRUE,]\n",
    "    message(\"\\ndimensions of the ijc_sex_results_refined_annotations \\n\", \n",
    "        paste(dim (ijc_sex_results_refinedAnnotations), collapse = \" \") )\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    ijc_sex_geneSymbols               <- ijc_sex_resultsAnnotations$geneSymbol\n",
    "    ijc_sex_refined_geneSymbols       <- ijc_sex_results_refinedAnnotations$geneSymbol\n",
    "    message(\"\\nlength ijc_sex_results_refined_geneSymbols\\n\", \n",
    "        paste(length(ijc_sex_refined_geneSymbols), collapse = \" \") )\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    ijc_sex_results_rn         <- paste(ijc_sex_geneSymbols,       ijc_sex_rnResults, sep=\"-\")\n",
    "    rownames(ijc_sex_results)  <- ijc_sex_results_rn    \n",
    "\n",
    "    ijc_sex_filename               = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_ijc_sex.csv',sep='')\n",
    "    ijc_sex_refined_filename       = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_ijc_sex_refined.csv',sep='')\n",
    "    ijc_sex_genesFilename          = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_ijc_sex_universe.txt',sep='')\n",
    "    ijc_sex_refined_genesFilename  = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_ijc_sex_gene_set.txt',sep='')\n",
    "    write.table(ijc_sex_results, \n",
    "                file = ijc_sex_filename        , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_results [ijc_sex_results_refined      ,], \n",
    "                file = ijc_sex_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_geneSymbols,        \n",
    "                file = ijc_sex_genesFilename        , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_refined_geneSymbols,\n",
    "                file = ijc_sex_refined_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "   \n",
    "    message(\"\\nstarting sjc\\n\")\n",
    "\n",
    "    y_sjc <- DGEList(counts=sjc.dm, group = sex)\n",
    "    y_sjc <- calcNormFactors(y_sjc, method=\"RLE\")\n",
    "    y_sjc_voom <- voom (y_sjc, design=design, plot=plot)\n",
    "\n",
    "    Gender <- substring(sex,1,1)\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_sjc, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-sjc-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "        plotMDS(y_sjc_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    " \n",
    "    fit_sjc <- lmFit(y_sjc_voom, design)\n",
    "    fit_sjc <- eBayes(fit_sjc)\n",
    "\n",
    "    sjc_sex_results                    <- topTable(fit_sjc, coef='sex', number=nrow(y_sjc_voom))\n",
    "    sjc_sex_results_refined            <- sjc_sex_results$adj.P.Val <= 0.05 & abs(sjc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    sjc_sex_rnResults                  <- rownames(sjc_sex_results)\n",
    "    sjc_sex_resultsAnnotations         <- fromGTF[sjc_sex_rnResults,]\n",
    "\n",
    "    sjc_sex_results_refinedAnnotations <- sjc_sex_resultsAnnotations[sjc_sex_results_refined      ==TRUE,]\n",
    "    message(\"\\ndimensions of the sjc_sex_results_refined_annotations \\n\", \n",
    "        paste(dim (sjc_sex_results_refinedAnnotations), collapse = \" \") )\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sjc_sex_geneSymbols               <- sjc_sex_resultsAnnotations$geneSymbol\n",
    "    sjc_sex_refined_geneSymbols       <- sjc_sex_results_refinedAnnotations$geneSymbol\n",
    "    message(\"\\nlength sjc_sex_results_refined_geneSymbols\\n\", \n",
    "        paste(length(sjc_sex_refined_geneSymbols), collapse = \" \") )\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sjc_sex_results_rn         <- paste(sjc_sex_geneSymbols, sjc_sex_rnResults, sep=\"-\")\n",
    "    rownames(sjc_sex_results)  <- sjc_sex_results_rn    \n",
    "\n",
    "    sjc_sex_filename               = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sjc_sex.csv',sep='')\n",
    "    sjc_sex_refined_filename       = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sjc_sex_refined.csv',sep='')\n",
    "    sjc_sex_genesFilename          = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sjc_sex_universe.txt',sep='')\n",
    "    sjc_sex_refined_genesFilename  = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sjc_sex_gene_set.txt',sep='')\n",
    "    write.table(sjc_sex_results, \n",
    "                file = sjc_sex_filename        , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_results [sjc_sex_results_refined      ,], \n",
    "                file = sjc_sex_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_geneSymbols,        \n",
    "                file = sjc_sex_genesFilename        , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_refined_geneSymbols,\n",
    "                file = sjc_sex_refined_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    \n",
    "   \n",
    "    message(\"starting the y prediction\\n\")\n",
    "    sample_names <- as.character(colnames(ijc.dm))\n",
    "    # we will add donor as a blocking parameter\n",
    "    # rather than sample name -- we should use donor for real\n",
    "    sample     <- factor(sample_names)\n",
    "    \n",
    "    donor    <- rep(sample, 2)\n",
    "    message(\"\\ndonor size\", \n",
    "        paste(length(donor), collapse = \" \") )\n",
    "    \n",
    "    ijc_names <- as.character(colnames(ijc.dm))\n",
    "    sjc_names <- as.character(colnames(sjc.dm))\n",
    "    sjc_names <- paste0(sjc_names,\"-sjc\")\n",
    "    ijc_names <- paste0(ijc_names,\"-ijc\")\n",
    "\n",
    "    colnames(ijc.dm) <- ijc_names\n",
    "    colnames(sjc.dm) <- sjc_names\n",
    "\n",
    "    as_matrix <- cbind(ijc.dm,sjc.dm)\n",
    "    message(\"\\ndim as_matrix\", \n",
    "        paste(dim(as_matrix), collapse = \" \")) \n",
    "            \n",
    "    for (i in (1:length(metadata_samples))) {\n",
    "        sample = metadata_tissue[i,]$SAMPID\n",
    "        sample_sex = pData(tissue_count_obj)$SEX[pData(tissue_count_obj)$SAMPID == sample]\n",
    "        if (i==1) {\n",
    "            sex = sample_sex\n",
    "        } else {\n",
    "            sex = c(sex,sample_sex)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    message(\"sex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\"\\n\"))\n",
    "    sex2      <- c(rep(sex,2))\n",
    "    table(sex2)\n",
    "    message(\"\\nlength sex2\\n\", \n",
    "        paste(length(sex2), collapse = \" \") )\n",
    "    message(\"table sex2\\n\", \n",
    "        paste(table(sex2), collapse = \"\\n\") )\n",
    "    as_event  <- c(rep(\"ijc\",dim(ijc.dm)[2]), rep(\"sjc\", dim(sjc.dm)[2]))\n",
    "    as_event  <- factor(as_event, levels=c(\"ijc\", \"sjc\"))\n",
    "    message(\"\\nlength as_event\\n\", \n",
    "        paste(length(as_event), collapse = \" \") )\n",
    "\n",
    "    design    <- model.matrix( ~ sex2 + as_event + sex2*as_event)\n",
    "    dim(design)\n",
    "    colnames(design) <- c(\"intercept\",\"sex\", \"as_event\",\"sex*as_event\")\n",
    "    message(\"\\ndim design <- model.matrix( ~sex + as_event + sex*as_event)\\n\", \n",
    "        paste(head(design), collapse = \"\\n\") )\n",
    "\n",
    "    y <- DGEList(counts=as_matrix, group = sex2)\n",
    "    y <- calcNormFactors(y, method=\"RLE\")\n",
    "    y_voom <- voom (y, design=design, plot = plot)\n",
    "\n",
    "    if (dup==TRUE) {\n",
    "        dup_cor <- duplicateCorrelation(y_voom$E, design=design, ndups=2, block=donor, weights=y$samples$norm.factors)\n",
    "        dup_cor$consensus.correlation \n",
    "        y_dup_voom <- voom (y, design=design, plot = plot, block = donor, correlation = dup_cor$consensus.correlation) \n",
    "    }\n",
    "    \n",
    "    Gender <- substring(sex[1:dim(ijc.dm)[2]],1,1)\n",
    "    message(\"\\nGenders new size\\n\", \n",
    "        paste(length(Gender), collapse = \" \") )\n",
    "    message(\"\\nplotting y for ijc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")\n",
    "     # print the combined exploratory plot\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    message(\"\\nplotting y_voom for ijc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")\n",
    "    filename <- paste0(paste0(paste0(\"../pdf/\", splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-voom-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_voom[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    if (dup == TRUE) {\n",
    "        filename <- paste0(paste0(pdf_sub_directory, snakecase::to_snake_case(tissue_of_interest)),\"-y-dup-voom-ijc-MDSplot-100.pdf\")\n",
    "        pdf (filename)\n",
    "            plotMDS(y_dup_voom[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "        dev.off()\n",
    "    }\n",
    "    message(\"\\nplotting y for sjc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    \n",
    "    if (dup == TRUE) {\n",
    "        message(\"\\nplotting y_voom for sjc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")    \n",
    "        filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-voom-sjc-MDSplot-100.pdf\")\n",
    "        pdf (filename)\n",
    "            plotMDS(y_voom[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "        dev.off()\n",
    "        filename <- paste0(paste0(paste0(pdf_sub_directory, snakecase::to_snake_case(splice_type)),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-dup-voom-sjc-MDSplot-100.pdf\")\n",
    "        pdf (filename)\n",
    "            plotMDS(y_dup_voom[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "        dev.off()\n",
    "        \n",
    "        fit <- lmFit(y_dup_voom, design=design, block=donor, correlation = dup_cor$consensus.correlation)\n",
    "    } else {\n",
    "        fit <- lmFit(y_voom, design=design)\n",
    "        fit <- eBayes(fit, robust=TRUE)    \n",
    "    }\n",
    "    \n",
    "    \n",
    "    sex_as_events_results         <- topTable(fit, coef=\"sex*as_event\", number=nrow(y_voom))\n",
    "    sex_as_events_results_refined <- sex_as_events_results$adj.P.Val <= 0.05 & abs(sex_as_events_results$logFC) >= abs(log2(1.5))\n",
    "\n",
    "    sex_results                   <- topTable(fit, coef=\"sex\", number=nrow(y_voom))\n",
    "    sex_results_refined           <- sex_results$adj.P.Val <= 0.05 & abs(sex_results$logFC) >= abs(log2(1.5))\n",
    "\n",
    "    sex_as_events_rnResults <- rownames(sex_as_events_results)\n",
    "    sex_rnResults           <- rownames(sex_results)\n",
    "    head(sex_as_events_rnResults)\n",
    "    head(ijc_sex_rnResults)\n",
    "    head(sex_rnResults)\n",
    "    head(fromGTF[sex_as_events_rnResults,])\n",
    "\n",
    "    # use the junctionIDs to get the annotations\n",
    "    sex_as_events_resultsAnnotations      <- fromGTF[sex_as_events_rnResults,]\n",
    "    sex_resultsAnnotations                <- fromGTF[sex_rnResults,]\n",
    "    ijc_sex_resultsAnnotations            <- fromGTF[ijc_sex_rnResults,]\n",
    "    head(sex_as_events_resultsAnnotations)\n",
    "    head(sex_resultsAnnotations)\n",
    "    head(ijc_sex_resultsAnnotations)\n",
    "    \n",
    "    sex_as_events_results_refinedAnnotations<- sex_as_events_resultsAnnotations[sex_as_events_results_refined==TRUE,]\n",
    "    sex_results_refinedAnnotations          <- sex_resultsAnnotations          [sex_results_refined          ==TRUE,]\n",
    "    sjc_sex_results_refinedAnnotations       <- sjc_sex_resultsAnnotations      [sjc_sex_results_refined      ==TRUE,]\n",
    "    head(sex_as_events_results_refinedAnnotations)\n",
    "    head(sex_results_refinedAnnotations)\n",
    "    head(sjc_sex_results_refinedAnnotations)\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sex_as_events_geneSymbols         <- sex_as_events_resultsAnnotations$geneSymbol\n",
    "    sex_as_events_refined_geneSymbols <- sex_as_events_results_refinedAnnotations$geneSymbol\n",
    "    sex_geneSymbols                   <- sex_resultsAnnotations$geneSymbol\n",
    "    sex_refined_geneSymbols           <- sex_results_refinedAnnotations$geneSymbol\n",
    "    ijc_sex_geneSymbols               <- ijc_sex_resultsAnnotations$geneSymbol\n",
    "    ijc_sex_refined_geneSymbols       <- ijc_sex_results_refinedAnnotations$geneSymbol\n",
    "    sjc_sex_geneSymbols               <- sjc_sex_resultsAnnotations$geneSymbol\n",
    "    sjc_sex_refined_geneSymbols       <- sjc_sex_results_refinedAnnotations$geneSymbol\n",
    "\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sex_as_events_results_rn   <- paste(sex_as_events_geneSymbols, sex_as_events_rnResults, sep=\"-\")\n",
    "    sex_results_rn             <- paste(sex_geneSymbols,           sex_rnResults, sep=\"-\")\n",
    "    ijc_sex_results_rn         <- paste(ijc_sex_geneSymbols,       ijc_sex_rnResults, sep=\"-\")\n",
    "    sjc_sex_results_rn         <- paste(sjc_sex_geneSymbols,       sjc_sex_rnResults, sep=\"-\")\n",
    "    message(\"\\n sex_as_events\\n\", \n",
    "        paste(head(sex_as_events_results_rn), collapse = \" \") )\n",
    "    message(\"\\n ijc_sex_results\\n\", \n",
    "        paste(head(ijc_sex_results_rn), collapse = \" \") )\n",
    "    message(\"\\n sjc_sex_results\\n\", \n",
    "        paste(head(sjc_sex_results_rn), collapse = \" \") )\n",
    "    rownames(sex_as_events_results) <- sex_as_events_results_rn\n",
    "    rownames(sex_results)           <- sex_results_rn\n",
    "    rownames(ijc_sex_results)       <- ijc_sex_results_rn\n",
    "    rownames(sjc_sex_results)       <- sjc_sex_results_rn\n",
    "\n",
    "    sex_as_events_filename         = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_as_events.csv')\n",
    "    sex_as_events_refined_filename = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_as_events_refined.csv',sep='')\n",
    "    sex_filename                   = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex.csv',sep='')\n",
    "    sex_refined_filename           = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_refined.csv',sep='')\n",
    "    sex_as_events_genesFilename    = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_as_events_universe.txt',sep='')\n",
    "    sex_as_events_refined_genesFilename = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_as_events_gene_set.txt',sep='')\n",
    "    sex_genesFilename              = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_universe.txt',sep='')\n",
    "    sex_refined_genesFilename      = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_gene_set.txt',sep='')\n",
    "\n",
    "    write.table(sex_as_events_results, file = sex_as_events_filename, \n",
    "                row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_as_events_results[sex_as_events_results_refined,], \n",
    "                file = sex_as_events_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_results,           file = sex_filename          , \n",
    "                row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_results [sex_results_refined          ,], file = sex_refined_filename, \n",
    "                row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_as_events_geneSymbols, file = sex_as_events_genesFilename, \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_as_events_refined_geneSymbols,file = sex_as_events_refined_genesFilename, \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_geneSymbols,           file = sex_genesFilename          , \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_refined_geneSymbols,          file = sex_refined_genesFilename          , \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "\n",
    "    return(0)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# tissue_index -- enables this to run as a NextFlow notebook\n",
    "# devtools::install_github(\"ropensci/piggyback@87f71e8\", upgrade=\"never\")\n",
    "#parameters for running the notebook as NextFlow\n",
    "\n",
    "splice_list       = c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\",\"se_\")\n",
    "\n",
    "tissue_index <- 17\n",
    "splice_index <- 5\n",
    "\n",
    "tissue_list_m_f     = levels(reduced_metadata_pData$tissue)\n",
    "tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "splice_type         = splice_list    [splice_index] \n",
    "fromGTF             = a3ss_fromGTF_no_chrY\n",
    "metadata            = ocounts_metadata\n",
    "ijc                 = a3ss_jc_ijc_no_chrY\n",
    "sjc                 = a3ss_jc_sjc_no_chrY\n",
    "plot                = FALSE\n",
    "dup                 = FALSE\n",
    "obj                 = reduced_obj2\n",
    "\n",
    "for (tissue_index in 1:length(tissue_list_m_f)) {\n",
    "\n",
    "    # a3ss\n",
    "    splice_index        = 1\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = a3ss_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = a3ss_jc_ijc_no_chrY\n",
    "    sjc                 = a3ss_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # a5ss\n",
    "    splice_index        = 2\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = a5ss_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = a5ss_jc_ijc_no_chrY\n",
    "    sjc                 = a5ss_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # mxe\n",
    "    splice_index        = 3\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = mxe_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = mxe_jc_ijc_no_chrY\n",
    "    sjc                 = mxe_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # ri\n",
    "    splice_index        = 4\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = ri_fromGTF_no_chrY\n",
    "    metadata            = reduced_metadata_pData\n",
    "    ijc                 = ri_jc_ijc_no_chrY\n",
    "    sjc                 = ri_jc_sjc_no_chrY\n",
    "    print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    # se\n",
    "#    splice_index        = 5\n",
    "#    splice_type         = splice_list    [splice_index] \n",
    "#    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "#    fromGTF             = se_fromGTF_no_chrY\n",
    "#    metadata            = reduced_metadata_pData\n",
    "#    ijc                 = se_jc_ijc_no_chrY\n",
    "#    sjc                 = se_jc_sjc_no_chrY\n",
    "#    print_exploratory_plots (plot, \n",
    "#                             dup,\n",
    "#                             tissue_of_interest, \n",
    "#                             splice_type, \n",
    "#                             fromGTF, \n",
    "#                             tissue_list, \n",
    "#                             ijc, \n",
    "#                             sjc, \n",
    "#                             obj, \n",
    "#                             metadata )\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Testing Snippet\n",
    "\n",
    "To use this snippet - set *test = `TRUE`*.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "ijc\n",
      "42484\n",
      "7020\n",
      "\n",
      "head(ijc)\n",
      "0 247 0 1 0 0 103 1 1 2 1 494 0 0 6 0 145 0 0 1 0 145 1 1 1\n",
      "\n",
      "sjc\n",
      "42484\n",
      "7020\n",
      "\n",
      "sjc\n",
      "0 0 0 0 5 1 0 0 0 3 6 0 0 0 4 1 0 0 0 0 0 1 0 0 3\n",
      "\n",
      "obj\n",
      "55878\n",
      "10372\n",
      "\n",
      "Limiting phenotype data to tissue of interest\n",
      "breast_mammary_tissue \n",
      "\n",
      "keep\n",
      "10086286 \n",
      "\n",
      "tissue_obj now reduced to the tissue of interest\n",
      "286 68\n",
      "\n",
      "sample value\n",
      "GTEX.1117F.2826.SM.5GZXL c(`GTEX-1117F-2826-SM-5GZXL` = 2) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = 2996) c(`GTEX-1117F-2826-SM-5GZXL` = 13) c(`GTEX-1117F-2826-SM-5GZXL` = 8) c(`GTEX-1117F-2826-SM-5GZXL` = 21) c(`GTEX-1117F-2826-SM-5GZXL` = 43) c(`GTEX-1117F-2826-SM-5GZXL` = 1519) c(`GTEX-1117F-2826-SM-5GZXL` = 695) c(`GTEX-1117F-2826-SM-5GZXL` = 1037) c(`GTEX-1117F-2826-SM-5GZXL` = 6) c(`GTEX-1117F-2826-SM-5GZXL` = 580) c(`GTEX-1117F-2826-SM-5GZXL` = 261) c(`GTEX-1117F-2826-SM-5GZXL` = 79) c(`GTEX-1117F-2826-SM-5GZXL` = 18) c(`GTEX-1117F-2826-SM-5GZXL` = 3) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 4869) c(`GTEX-1117F-2826-SM-5GZXL` = 8616) c(`GTEX-1117F-2826-SM-5GZXL` = 9439) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 5382) c(`GTEX-1117F-2826-SM-5GZXL` = 5615) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 4197) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 9139) c(`GTEX-1117F-2826-SM-5GZXL` = 78) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = 4393) c(`GTEX-1117F-2826-SM-5GZXL` = 6486) c(`GTEX-1117F-2826-SM-5GZXL` = 12930) c(`GTEX-1117F-2826-SM-5GZXL` = 4084) c(`GTEX-1117F-2826-SM-5GZXL` = 10262) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 4210) c(`GTEX-1117F-2826-SM-5GZXL` = 4295) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = 12882) c(`GTEX-1117F-2826-SM-5GZXL` = 5382) c(`GTEX-1117F-2826-SM-5GZXL` = 6362) c(`GTEX-1117F-2826-SM-5GZXL` = 4393) c(`GTEX-1117F-2826-SM-5GZXL` = 11516) c(`GTEX-1117F-2826-SM-5GZXL` = 4250) c(`GTEX-1117F-2826-SM-5GZXL` = 4663) c(`GTEX-1117F-2826-SM-5GZXL` = 4087) c(`GTEX-1117F-2826-SM-5GZXL` = 66) c(`GTEX-1117F-2826-SM-5GZXL` = 4097) c(`GTEX-1117F-2826-SM-5GZXL` = 6386) c(`GTEX-1117F-2826-SM-5GZXL` = 10322) c(`GTEX-1117F-2826-SM-5GZXL` = 4306) c(`GTEX-1117F-2826-SM-5GZXL` = 6395) c(`GTEX-1117F-2826-SM-5GZXL` = 13962) c(`GTEX-1117F-2826-SM-5GZXL` = 6128) c(`GTEX-1117F-2826-SM-5GZXL` = NA) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = 1236) c(`GTEX-1117F-2826-SM-5GZXL` = 1) c(`GTEX-1117F-2826-SM-5GZXL` = 2) c(`GTEX-1117F-2826-SM-5GZXL` = 5) c(`GTEX-1117F-2826-SM-5GZXL` = 5) 21\n",
      "\n",
      "Limit the phenotypes to those we have samples for\n",
      "110 176\n",
      "\n",
      "tissue_count_obj now has these count specific phenotypes\n",
      "\n",
      "Dimensions of tissue_count_obj\n",
      "176 68\n",
      "\n",
      "Dimensions of tissue_count_obj \n",
      "34401 286\n",
      "\n",
      "Limit the counts now has these count specific phenotypes \n",
      "6842 178\n",
      "\n",
      "dimensions of the ijc_tissue \n",
      "42484 178\n",
      "\n",
      "dimensions of the ijc_tissue \n",
      "42484 178\n",
      "\n",
      "dimensions of the metadata_tissue \n",
      "178 80\n",
      "\n",
      "The non-overlapping IDs between pheno and count data are:\n",
      "0\n",
      "sex samples:\n",
      "103\n",
      "75\n",
      "\n",
      "sex samples:\n",
      "103 75\n",
      "\n",
      "design matrix ijc, alone:\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "0\n",
      "0\n",
      "1\n",
      "0\n",
      "0\n",
      "1\n",
      "Repeated column names found in count matrix\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in pdf(filename): cannot open file '../pdf_no_dup_corr/se_breast_mammary_tissue-ijc-MDSplot-100.pdf'\n",
     "output_type": "error",
     "traceback": [
      "Error in pdf(filename): cannot open file '../pdf_no_dup_corr/se_breast_mammary_tissue-ijc-MDSplot-100.pdf'\nTraceback:\n",
      "1. print_exploratory_plots(plot, dup, tissue_of_interest, splice_type, \n .     fromGTF, tissue_list, ijc, sjc, obj, metadata)   # at line 34-43 of file <text>",
      "2. pdf(filename)   # at line 112 of file <text>"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzdebgU1Z3/8dMsd5E9CZsgAsKwaURRJCIDI5EbxSASI5LEZRwdFZIxJvgL\nuMBIxBknY0Iikvi4SzQEQRBJ0ESRCGoEFBQjMSJgIgG8EvCyw4X+/VGPNcU5p06dqq7qrq77\nfv3hc6mu5VvV7a3P/Z6q6lw+nxcAAAAof41KXQAAAADiQbADAADICIIdAABARhDsAAAAMoJg\nBwAAkBEEOwAAgIwg2AEAAGQEwQ4AACAjCHYAAAAZQbADAADICIIdAABARhDsAAAAMoJgBwAA\nkBEEOwAAgIwg2AEAAGQEwQ4AACAjCHYAAAAZQbADAADICIIdAABARhDsAAAAMoJgBwAAkBEE\nOwAAgIwg2AEAAGQEwQ4AACAjCHYAAAAZQbADAADICIIdAABARhDsAAAAMoJgBwAAkBEEOwAA\ngIwg2AEAAGQEwQ4AACAjCHYAAAAZQbADAADICIIdAABARhDsAAAAMoJgBwAAkBEEOwAAgIwg\n2AEAAGQEwQ4ASm/evHm5XC6Xy1VVVZW6loxbuHChc6ibNGlS6lqA+BHsAMTpwQcfzHk89NBD\n6jyPPfaYd54HH3yw+HUCQCbx9wqABC1cuPDf/u3fpImLFi0qSTFp1r9//5/85CdCiGy0kbZv\n396pU6cjR46sX7++d+/epS4HaECy8BsEQGq98MILe/fubdasmTvl4MGDv/vd70pYUjr16NHj\nu9/9bqmriM3cuXOPHDlS6iqAhoihWACJ6Ny5sxDiwIEDzz//vHf60qVL9+zZI4To1KlTaSpD\n8n7961+XugSggSLYAcn68pe/7FxJNmLECOml2bNnOy81bdq0trbWmVhXV3fXXXcNGjToc5/7\nXEVFRfv27Wtqah555BFt/8Ny5ocfftjZ0D//8z8LIebNm3faaacdd9xxXbt2ve222w4fPiyE\nWL9+/ahRo9q0adO8efMRI0b86U9/8q7htddemzdv3rx581577TXLHR84cGDjxo2FEM8884x3\nujMO27hx4wEDBmgXfO+9966//vqePXtWVVW1bNnyzDPP/NnPflZfX++dJ5/Pz5kzp6ampl27\ndk2bNm3ZsuVZZ5117733endc2uvly5efd955zg4OGTLkhRdeCNyFUO9dqJKGDh1aX19/4403\ntm3btn379sL/5omE9nTLli3f+973+vbt26xZs+rq6n79+k2aNMn9EIZ6LyQXXnhhLpd75ZVX\nnH/26dMnl8s5zUjD7ltuLsKePvroo2eccUazZs0+97nPjRw5cvXq1blczlA/UPbyAJLk3j1Q\nUVFRV1fnfWnMmDHOSxdccIEzZe3atX59rC996UuffPKJd3H7mefMmeNMP/nkk+fOnSud2CZM\nmLBx48YvfOEL3olt27bdtWuXu4aLLrrImX7RRReZ9/eBBx5w5vzWt741aNAgIcTnPve5+vp6\ndwank/elL33pkksuceZ84IEH3Ffnz5+vvS303HPP3b9/vzvbN7/5Te2OX3jhhUePHlX3+vnn\nn6+oqPDO2bhx4xdeeCHG986mpCeffNKZ+MUvfvGee+5xK8nn80899ZTzz8rKSu+GktjTpUuX\ntmrVSl1n+/bt33777bDvhWTkyJHqIjfeeKN59y03F3ZPv//970trq6ysvO2226RNA1lCsAOS\ntXPnTvf0M3fuXHf6/v373SvPnnzyyXw+/49//MMNat26dfv5z3++cOHCH/zgB+7V9CNHjnQX\nDzWzGxo6duzYuXPnUaNGffvb33ZP7ZWVleeff/7xxx9/4403nnXWWe4p8Cc/+Ym7hgjB7tJL\nL506darz89KlS51X33jjDWfK1KlTR48e7fzsBruNGzdWV1c7E2+++eb33ntv1apVQ4cOdaZM\nnjzZme3ZZ591pjRq1GjWrFnr1q176KGH3B13D7J3r7t27dq/f//Jkyd7e28DBw6M670LW1LX\nrl1POOGEpk2b9u/fv1evXnmfYJfEnm7duvVzn/ucM33o0KHz5s375S9/2b9/f2dKr169Dh8+\nbP9eqP70pz95e7RPPPHE8uXLN27caN59y82F2tPXX3/dnX7uuec+/fTTixYt+spXvuIeQIId\nMolgByTOTUWXX365O9E9+bVo0WLv3r35fP6OO+5wprRs2XLLli3unLNnz3bPT6tXr3YmhprZ\nPR0KIb7+9a87E+fPn+9OrKqq+uCDD/L5/MGDB3v16uVM/OpXv6rugn2w+9rXvrZy5UrnZ6dh\nk8/n3ai3cuVKd51usPv2t7/tTBk2bJi7wtra2ubNmzsHyunczJw5c+TIkSNHjnRXm8/nR40a\n5Sx7xRVXqHs9ZMgQt+vj9sAaNWp06NAh8+5YvncRSurZs+ff/vY3d2ZtsEtiTydPnuxM/MIX\nvuAUn8/nP/74YzdXzZs3z/690Nq6datbz/r169V9VHffcnOh9vTaa691JrZt23bPnj3OxMOH\nD7t36RLskEkEOyBx7vjR5z//eXdQ8qqrrnImXnnllc6UU0891Zly1VVXeRevr69v06aN89K0\nadMizOw9Ha5cudKZeOjQocrKSmfiuHHj3DVMnDjRmdi/f/8IO+sNdkePHnXail27dnVePe20\n04QQnTp1Onr0qBrsTjrpJGfKrbfeut/DuaBKCGEYPP3Od77jzDNixAh1r92WYT6f916JtXnz\nZvPuWL53EUpyWn0uv6HY2Pf05JNPdqZcc8013tW+8sorS5YsWbJkyfvvv58v7L2wCXbS7ltu\nLtSe9unTx5ly9dVXe7f1wx/+0JlOsEMmcfMEkLhRo0Y5jYcdO3a8+uqrQogjR44sXrzYefVb\n3/qWECKfz7v3K5xyyinexRs3buz2GNavXx92Zol7Xm/atGnbtm2dn7/4xS+6M3Ts2NH5Yffu\n3RF21iuXyznpbfPmzW+//fZHH320Zs0aIcTo0aPVC9jz+fymTZucn6dPn17t8fLLLzvT3b1+\n4YUXRo8efdJJJ1VVVTlX0997773OS9q7TM444wz35xNPPNH92dnHurq6roqDBw8Ku/cuWknn\nnntu8BGMe0/z+bz7qXCzlOPss8/+yle+8pWvfKVHjx6h3otovLsfbXPmPRVCbN682fmhZ8+e\n3gX79etXSOVAyvEcOyBx1dXVF1100RNPPCGEePbZZ4cMGbJ8+fJPPvlECHH88cc7Z7h9+/a5\nd/85ScLLvaLLOWmFmtmrsrLSHXFz/un80Lp1a3eidEF6gUaPHj1r1iwhxO9///vjjjvOnajO\nuW/fvqNHj5rX9o9//EMI8fOf/3z8+PHOlGbNmnXv3r2iomLLli3OIVVVVla2aNHC/ad6hf7R\no0c//PBDaWI+nxd2712Ekho3buymaoPY93Tfvn1uHHQ7uyr79yIaafcjbC5wT/P5/P79+52f\n3Q+e9p9AxtCxA4ph3Lhxzg/Owz4WLlzo/PMb3/hGo0aNhBDHHXece013XV2dtLgb0Zw7HkLN\nXFrDhg1zUuPSpUtffPFFIUTr1q3di+K9jjvuOOfxKEKIGTNmaIcY/vM//3PPnj3uYPE3vvGN\n2trad999d+3atZdeemlCuxD43kUoqUmTJs6yBknsaXV1tbvdvXv3+s1m+V5ELkPa/SQ2531w\njLSn6v8yQJYQ7IBiGDFixOc//3khxHvvvfeXv/zFDQfuWF4ul3MHSdeuXetd9vDhw++++67z\nszNmGmrm0mratOkFF1wghFi+fPlLL70khBg5cmTTpk3VOXO5nDs46A7MqdasWbNv3z7n55tv\nvtltQL733nuRi2zdurWaJNxYEPjeJVFSQqtt1KiROy4pjdQ/+eSTd95555133vncc89Zvhdx\nSWhzXbp0cX7485//7J3uXA8AZBXBDiiGpk2buo9tmzZtmjPwd8opp7j3QAgh3EejLVy48KOP\nPnKnP/roo04TLpfLuYOYoWYu3Ouvv75w4cKFCxd6HyFhySlj9+7dzmiaoaqamhrnh6eeesrN\nNEeOHPnmN7959dVXT548ecuWLc6lbw7n0cpCiHfffddJjd6JcQl87xIqKaHVOjlbCLFw4cJP\nP/3U+Xnnzp3f/va3b7/99ttvv935ONm8F36b8F5A6TdqLClkc36+9KUvOT8sWrRo586dzs97\n9uzx3jkOZFDo2y0ARLJs2TLp/767777bO8POnTudh/cKIXr06HHffffNnz//5ptvdq+E897G\nGGpmvzsu3TbJz3/+c3eie23+SSed5E6M9rgTZ8ru3bvdqqqqqnbv3i2t070r9oMPPnD7UoMH\nD/7Nb37z/PPPuxG2X79+9fX1W7ZscUfxvvrVr65bt27RokWdOnVyn9LSsmXL1157bfv27X57\n/be//c19C9atWxf0vuXzQe9d4SVp36OE9vTDDz90r04766yzfv3rXz/xxBPut4B06dLFeTKI\nzXvhd7jq6+vdpuyQIUPmzp37/PPPa/fRZbm5UHvqfdcGDBgwZ86cxx577Mwzz3QvS23UqJHN\nuw+UF4IdUCTusz/ck4r3OV4Ow5dJjBkzRnpymP3MpQ12+Xze7RJ5H5usBjunVDcFenXq1Ml9\ncIb7vA/X8ccfv2nTpuOPP96dMnXq1BiDXeB7V2BJeZ/3KKE9Xbx4sfYGgg4dOqxduzbUe+Hn\n/PPP9y7ivO/mR7rYbC7snv7rv/6rtLZmzZrNnDnT/eeRI0fMOwKUHYZigSLJ5XJjx451/zl0\n6FC35eY69dRT33333TvvvPPMM89s1apV06ZNO3bsePHFFy9atEj9wqVQM5fWxRdf7PwQODp8\nySWXrFmz5uqrr+7WrVtlZeVxxx138skn33rrrevWrXMf43LPPffcfffdffv2ra6u7tSp0zXX\nXLNq1aquXbs+9thjvXr1atKkSefOnd1nmMUi8L1LqKSEVjty5Mh33nnnhhtu6Nmzp/NUkX79\n+k2ePHndunXeawNs3gs/DzzwwOjRo1u3bl1VVdWtW7ezzz47sKpCNufnwQcfvOeee3r16lVR\nUdGuXbtLLrnk9ddf/5d/+Rd3BsMdJECZyuXz+VLXAAAAgBjQsQMAAMgIgh0AAEBGEOwAAAAy\ngmAHAACQEQQ7AACAjCDYAQAAZATBDgAAICMIdgAAABlBsAMAAMgIgh0AAEBGEOwAAAAygmAH\nAACQEQQ7AACAjCDYAQAAZATBDgAAICMIdgAAABlBsAMAAMgIgh0AAEBGEOwAAAAygmAHAACQ\nEQQ7AACAjCDYAQAAZATBDgAAICMIdgAAABlBsAMAAMgIgh0AAEBGEOwAAAAygmAHAACQEQQ7\nAACAjCDYAQAAZATBDgAAICMIdgAAABlBsAMAAMgIgh0AAEBGEOwAAAAygmAHAACQEQQ7AACA\njCDYAQAAZESTUhdQHt566636+vpSVwEAAFKhSZMmp556aqmr0CDYBVu9evWZZ55Z6ioAAECK\nrFq16owzzih1FTKCXbBDhw4JIQ4ePFhRUVHqWgAAQIkdOnSosrLSiQdpwzV2AAAAGUGwAwAA\nyAiCHQAAQEYQ7AAAADKCYAcAAJARBDsAAICMINgBAABkBMEOAAAgIwh2AAAAGUGwAwAAyAiC\nHQAAQEYQ7AAAADKCYAcAAJARBDsAAICMINgBAABkBMEOAAAgI5qUuoDQ8vn8pk2bNm7cuHv3\nbiFEq1atevbsecIJJ5S6LgAAgBIrp2C3c+fO6dOnz549++OPP5Ze6tKlyzXXXDNx4sTq6uqS\n1AYAAFByZRPstm7dOnjw4E2bNvXs2fOCCy448cQTmzVrJoSoq6v74IMP/vCHP0yZMmX+/Pkv\nvfRSmzZtSl0sAABACZRNsLv99ts/+uijuXPnfv3rX1dfPXLkyP333//tb3/7jjvumDFjRvHL\nAwAAKLmyuXniN7/5zeWXX65NdUKIxo0bjx8//tJLL3366aeLXBhQLnK5XC6XK3UVAIAElU2w\n27Fjx0knnWSep0+fPtu3by9OPUDDRDoEgDQrm6HY448//q233jLPs2bNmuOPP7449QBlJ5/P\nl7oEAECyyqZjN3r06Keeeup///d/Dx48qL66d+/eqVOnPvPMM2PHji1+bUDDkc/nCYgAkFq5\ncvkdvWvXruHDh7/55pstWrQYOHDgCSec0Lx583w+v2fPng8//HDlypX79u0bMmTIb3/72+bN\nm8e76VdffXXw4MEHDx6sqKiId80AAKDsHDp0qLKy8pVXXjn77LNLXYusbIZiW7du/dprr913\n332PP/74smXLjhw54r7UtGnTAQMGXH311VdffXXjxo1LWCQAAEAJlU2wE0JUVFTcdNNNN910\n04EDB/72t7853zzRsmXLLl260EsD4uLeG1Eu7XwAgKucgp2rqqqqZ8+epa4CyKZ8Pp/Llc1F\nGgAAr7K5eQJA0ZDqAKBMlWXHTuuDDz647rrrhBAvvPCC/VL19fWLFy8+fPiwYZ7169cXWhwA\nAEDyshPsdu/e/eKLL4ZdasuWLRMmTNi/f79hHucBK4cOHeJKPgAAkGbZCXa9e/det25d2KVO\nPPHELVu2mOe5//77r7/++qh1AaXk3AnB0CoANBDZCXZVVVUnn3xyqasAAAAomUzdPLFz587N\nmzeXugogRfiiCABoUMop2L399tsjR47s2rXrkCFDZs2a5X1GsePuu+/u1q1bSWoDAAAoubIZ\nin3llVeGDx9+8ODB44477u9///uKFSvmzp27YMGCNm3alLo0oIxxER4AZEnZdOz+67/+6+jR\nowsWLNizZ8/u3bt//OMfv/rqqzU1NXv37i11aQAAAKlQNsHu7bffHjt27OjRo3O5XGVl5U03\n3fTcc8+99dZbl156qTomC8ASF+EBQJaUTbDbtm1b9+7dvVPOPffcBx988Le//e33vve9UlUF\nAACQHmVzjV379u3Xrl0rTbz88svXr1//X//1X507d7755ptLUhgAAEBKlE2wGzNmzL333jtz\n5szrrruuadOm7vTp06f//e9//3//7//9/e9/Z0wWAAA0ZGUT7KZMmbJw4cLvfOc7zzzzzO9/\n/3t3ei6Xe+SRR1q1ajVjxowSlgcAAFByZXON3ec///k33nhj/Pjx6tdL5HK5n/70p/Pnzz/p\npJNKUhsAAEAalE3HTgjxhS984b777vN7dcyYMWPGjClmPQAAAKlSNh07AAAAmBHsAAAAMoJg\nBwAAkBEEOwAAgIwg2AFoiHK5XC6XK3UVABAzgh2AEAhDAJBmBDsAtpxUl41sl8/n8/l8qasA\ngJiV03PsAJQWSQgAUo6OHQAAQEYQ7AAAADKCYAcAAJARBDsAAICMINgBAABkBMEOAAAgIwh2\nABLEFzwAQDER7AAAADKCBxQDSBDPNAaAYqJjByAcRlcBILUIdgDCoQkHAKlFsAMQGtkOANKJ\nYAcAAJARBDsAAICMINgBAABkBMEOAAAgIwh2AAAAGUGwAwAAyAiCHQAAQEYQ7AAAADKCYAek\nGl/eBQCwR7AD0otUBwAIpUmpCwDgi2/uAgCEQscOAEosl8vRnQUQC4IdAABARjAUCzQsTmeI\nQd5U4e0AEBc6dgAAABlBxw5oWGgOAUCG0bEDAADICIIdAABARhDsAAAAMoJgBwAAkBEEOwAA\ngIwg2AEAAGQEwQ4AACAjCHYAAAAZQbADAADICIIdAABARhDsAAAAMoJgBwAAkBEEOwAAgIwg\n2AEAAGQEwQ4AACAjCHYAAAAZQbADAADICIIdAABARhDsAAAAMoJgB8BKLpcrdQkAgAAEOwDB\nnFRHtgOAlCPYAQiWz+fd/wIAUotgB8AKqQ4A0o9gBwAAkBEEOyBxXJoGACgOgh2QLG47AAAU\nTZNSFwBkHJemAQCKho4dAABARhDsAAAAMoJgBwAAkBEEOwAAgIwg2AEAAGQEwQ4oe7lcjsep\nAAAEwQ4AACAzeI4dUPbsH5XnNPZ4tB4AZBUdOyDjGKUFgIaDYAdkmZTq8vl8Pp8n6gFAVhHs\ngCxTR1357loAyDCCHZBxUrZz/un07Yh3AJAxBDugweHmCQDIKoIdAABARhDsgOxj1BUAGgie\nYwc0CN7hV55mBwBZRccOiE1qu2JkOEu0NgGUO4IdEI8yeoyI8zS7aMsSfQAgzQh2QDzcx4iU\nupD4NZwwV0jkBYA04Bo7IDYNIRM0hH0EgPJFsAMQgDAHAOWCYAdkUCH3vbqjruQ5ACg7XGMH\nlF7uM6UuRIhMXywIAJlHsAMKEksgiz1FSTcBaCs0VE6qA4AyRbADUiG5+zHL6DksAIACcY0d\nYMXvqrX0NLfCVpieygEAcaFjB/yf9FzopkptYQCA9KBjB1hJQ38rl8sZykhDhQCA0qJjB/yf\nlH/xgFtbmjuLsMQ7CCAJBDugPKQ5cSIash2A2DEUC5QfQl428D4CiB0dO6BBM4/qMuabHFId\ngCQQ7ICMixbOShLpCJEAUCCCHVBKyeUny9Wa7xcp/t0kZDsAKATX2AFZID2d2P22icjJrFQD\nhQxQAkAh6NgBGeTEo7ILSWVXMACkDR07oJTiijJp/q4z4f91ZwCAeNGxAwAAyAiCHbKGq++L\nxv7Oj5R/pQcAZAbBDpni3jRQ6kIAACgBgh0ypUxvGihT9OEAIG0IdsgaogYAoMEi2AEAAGQE\nwQ5pwYVxqcU3xgJAuSDYIRXIDV42QYqwBQBQEeyQClwYl2bcJIES4m8YIBS+eQJpUdbRwXvi\ncXakkO9asFkqn89ztgMASOjYoYyl50/5wlNphB0p6ygMWKJhDIRCsAPiIT1CzzkbhYqefnOm\nJ78CAFKOoViUsVT9HV9g9krVvgAAyhTBDhlRyDVtSYixnvTsFBqUXC7HZw8oOwzFAqnGOCxK\niM8eUHbo2CEjSt5akAoopB6/bl/aupLIPD5sQNkh2KGBKpeQlP4KkVV89oByxFAsGqg0n7Tc\n5ztI47BxPfehvMbXGIwGAHsEOxRP2s7Qac52yXHeglS9ETbKrmAAKAmGYoFkFXJrYRLRs+zi\nrPebPAAAZnTsUDwN9gnyhJLCNcxPDgCERbAD4pG2gWYAQANEsAOSFUufkshYIGI3gAaCYAfE\nI7mB5jK93QEAUHzcPAGkHZeXaYV6EiHHEEADQccOKI3cZwJnS2jTsa8WAFByBDsgigKzUahl\nkwhhGehgNdibrAHAgKFYIH6Bo4R+L2kXjD2+kIcAIKsIdkAUkbNR2MxHCAMA2GMoFoiZG90i\nDKES4wAAhSDYASaRr6Xze0YJV4YBAJLDUCwQMze3FfItsQAAREDHDjAp5BvoSXX2eAILAMSC\nYAeUGTcDEYYAABKGYoEAxW+8hfpOhWxoUDsLAMkh2KFIShhWMpaT3B0Ju0clPw5cdAgASWMo\nFlkWYbAyDYObCd05a15n0gO7frcJAwZcbwCERbBDkZTwMR/Spg3niWjhI9S5p7QnqhI2zJxN\n07Fr4AhqQNIYikWWGb65S/tSqmJHkQcui7CtVB1elETYB3fzmQHCItihIYp2tpCuUXPPT6HW\nZjmz2zssoxNbya/hQ1ngEwIkimCH+EVLPIVv0XJzhVTlXdbpPSS0j5z8AAARcI0d4hf5aqpC\nvryrCNQ9sm+/+V1alKXrjUr+bWlcvwUABDskomipLi6xZwLvY4QN88S4RQAAGIpFnAq5yirs\nVdXeBe1nNlRY/EFVdZeTG9ttCBfAZXvvAMAGwQ4p4v1i1iKfpGPfXIQL/iI38Cwvaoy8jw0h\nFAJANjAUiziV/CqrQGmuMHJhNhc1Gp78Em2jAIAUItghdZLIXrHHF/WaPMMdEvb1FJjtwrJ8\nIHOa0zAAwItgh+wrMNWFuq/CJgDZxKni3ODJt0EAQMZwjR2yrzjfu2q/lUK6a5EXj7cYAEA6\nEexQrop2RX9Joo/fRslhCFT8J4QDSI/yDnaHDh1666239uzZ07Vr127dupW6HAAovUS/EwVA\nypXNNXZ33nnnSy+95J1y//33d+jQYeDAgeeee2737t3POOOMtWvXlqo8FF8ZXdHPNyKgyMrl\nfw0AsbPt2G3evHnDhg2ffPLJp59+2qpVqy984Qs9evTo2rVrkrUd4/bbb//BD37wL//yL84/\nf/Ob31x//fWVlZUXX3xxu3bt3nnnnVdeeWXYsGFvvPHGSSedVLSqEAsek2aD4wMACBQQ7DZu\n3DhjxowlS5Zs2LBBfbVHjx4XXHDBd7/73eIPg950002tWrV67bXX+vTp40x5+umnL7nkkunT\npz/88MNFLgaZJ0VP7z9tUmnhmYyRNdI/ANjwDXa1tbWTJk16/PHH6+vr27VrN3bs2J49e7Zr\n165169a7du36+OOP33///aVLl/7sZz+bNWvWFVdc8d///d9t27YtTtG1tbXvv//+Lbfc4qY6\nIcSYMWMuuuii3/3ud8WpATFKydk6VHgqflaL5Yp44hEAZJs+2C1btmzs2LE7duy47LLLvve9\n75122mnaK4Ty+fyaNWt+/OMfP/bYY4sXL547d+7QoUMTLlgIIQ4cOCCE8KY6x8knn/yb3/ym\nCAUge8zXwElJyPvPAp9dYs/7xbINMJ81qJ0FgMj0N0+MGDGid+/e77333i9/+cvTTz/d7ySU\ny+VOP/30X/7yl3/+85979+593nnnJVnq/zn++ONbtWr10UcfSdP//ve/t2jRojg1IM0i3Kxg\nzg3uCuO6DSJsTHGfJFxgvimjO06i4T4VAA2cPthNmjRp6dKl9nch9OjRY+nSpZMmTYqvMI2/\n/vWvq1ev3rBhw86dO8ePH//QQw/t27fPffXPf/7zr3/968GDBydaA5KQ6MnYfuVFTjyBmzOU\nnfl8hjQgJQPlqGyuyNb+fpk3b97XvvY1IcSTTz757//+7/v37//jH/945plnxrvp+++///rr\nr9+9e3fz5s3jXTMcfgOLsQw4JjRqWYS7GRrgeKtBuRyNcqnTEnftAFqHDh2qrKx85ZVXzj77\n7FLXIiubBxQ/8sgjuzw+/fTTXbt2tWnTxnl1165drVu3njNnTuypDkWQ6JkjlgotjIIAACAA\nSURBVLscpPW4w7IRVm6/lHa2cskN5VKnveztkY2Gtr9ABpRNsLvqqqsMr15xxRXXX399o0Zl\n87xl2EjtScV7H0MENtkulk5JNr5aqrxGA8v6UAPIgLIJdmYMkiI5aozz3p0a7TYIM/PVdeqc\n2gfsCd1XS7kzJN1/0m40rhWWRMkLAAAbtLjQ0NlcIa4+7kTKUvGKMUNkII5wp4if8uplAiiO\njHTshBAffPDBddddJ4R44YUX7Jeqq6ubMmXK/v37DfOsX7++0OIasMjdGnMzrFTXdHu3a26e\nFSjadXiBS/kVHwu/N4VYloRCLvQEkGHZCXa7d+9+8cUXwy518ODB2traw4cPG+ZxHqrCb89i\nMrci4m1U2L+ziTZI1EHVwEfr+c1QkpM9OaPIOM4AtIKDXT6fnzdv3uOPP/7RRx9pA9A777yT\nQGGh9e7de926dWGXatu27RNPPGGe5/7773/jjTcY9Ygm2ukn8lev+gUL7XTLHpu7rPmeCfP6\nQ/XzAkOSNIN3zgI/qJH7jgk9rQYAEEpwsLvnnntuvvlmIcRxxx3XtGnT5EuKqKqq6uSTTy51\nFSglw1ekiJCZz2+dxbnbQL3pwTCztHcF3rEblzTUkD00RAEECg52P/3pT2tqambNmtW9e/ci\nFFSIHTt27Ny5s0ePHqUuBKVh/jZV9Yzozp9Qb6mQr5T19vnMi2v3K9S24lo2ifVAQrYDYBZ8\nV+z27dvvuOOO9Kc6IcSPfvSjnj17lroKlJJ2QNBwW6XN3QahzqPxfguT24SLa4VJy32m1IWU\nXhLHwfk0cpABGAR37Nq3b19G5xWUnYQ6EFLHy2Z+v2+YKO3nP96tF/jI4ngPSBoOb9oqAYAC\nBXfsxo0bN3v27CKUggZIG7widyPUpez7bbH0P8zdwUJWm8L2jLaq/GdiWVuM85sl9KdFQkkx\n8kEG0BAEd+ymTJlyySWXfPOb37ziiiu6dOmi3j9RnGvazjjjjMB5tmzZUoRKEKNoV/pr+yuF\nP24j2sVq3ltTCyygCAqsLd5dS9WBSlUxABBZcLBr0aKF88OTTz6pnaE4vxDXrFkjhDDflltf\nX1+EShAvyzgVOFhmuJPUb9kCxyXFsX2+wFtZM6a0Ia/hHGcACCU42I0bN66ioqJJkxI/yvjm\nm2+eNWvWm2++aWgQTpo06e677y5mVYiRXx/OvpEW1zPz7POZ1HEszsNQAADwExzX/Bp1RfbD\nH/7wd7/73bhx41599dU0P04v84p/mXnkC9fcUkMFwbAjqkU4FOWY6rgdAQBKIvjmCdcnn3zy\n2muvvfDCC6+//vquXbuSq0mradOmTzzxxJ/+9KdbbrmlyJtGcXgTmPfSeMMQqg3DVfbxXv5v\nudHkJLTRIuxICm8NKQmeYwKgcFYDrCtWrJg4ceLrr7/uTsnlcueee+6MGTOK+WUPffr02bZt\nm+FCuvPPP79169ZFq6dhKnIPxi/V2XTUtE88sR/whSjgjhD7+cviphN7hbcq03wcMvM2ARkW\nHOxWrlz55S9/ub6+/pxzzunVq1d1dfXevXvffffdpUuXDh48eOXKlb169SpCoY6WLVsaXh06\ndOjQoUOLVgySY75JIvKqpJDnPutVxHHG8p7RSxIcvf3OuDZXhDtCUnjTSanqSdVB8JO2NwuA\nJDjY3XnnnW3btv3973/fu3dv7/Q1a9Z85StfueOOO1JyER7KQiFnhRgv25JCWKLnqpJ0pOLN\ndrGsp7SbCCvyAUzhvsQr/TvI9Z1o4IKvsXv11VfHjx8vpTohxGmnnTZ+/PilS5cmUxgywtsk\ncyNO0TZt+chiv7sopOLNlftdmVfWvZ+GdsmXu7/EAi0OC5B+wR27Tz/9tHPnztqXunbt+o9/\n/CPukpAd2qHPGB8gHHkN0vhs4WtOVRpISRnliwNY1nj70MAFB7t27dqtX79e+9K7777brl27\nuEtCdkT7LocYt+53d63hejjv4oZ/hq0k8rKxC5VBU1V5ETS0/QWQPcFDsSNGjLj33nufeeYZ\n76+8fD6/YMGC++677/zzz0+yPJRGeQ3AFVJqqJs3DYdFGoctrwMIAMiM4I7d1KlTf/vb344e\nPbpDhw59+/Zt1qyZc1fstm3bOnbsOHXq1CJUCXipPSe/4dTAbpy0oKGb5X7JRLz3zxYfTSkA\nyLDgjl3Xrl1Xr1595ZVX7t+/f+nSpc8+++zSpUsPHTp0zTXXvPHGG36X36GsaZ8AlwRvZyvU\nY4QlhXTL1AsBbWYLrCed+YlWIgBkm9UDik844YRHH300n89v27Zt7969zZs379ChQ9KVoeRi\njCbmHpU329lsVL36TXshnY2wF9KpD6iznBhqK4G8m/D7OXY8wAwA0k8f7LZt21ZZWdmmTRvn\nZ3d6Lpdr3ry5NJGQl0nFOYVLrUHvCKn7T8uoFznbhSrVniF7FefYakOee0NJtN0h2wFAyumD\nXceOHWtqap577jnnZ/Mq+EWPQH4PihOlG7U0NLfs+16GC/u0I55xDYNKdzKZ64l9iwZJx2sA\ngJk+2I0dO7Z///7uz0WsBxlR+JhggYOqMYq2L9q85e0simNzXuz1Wz7GBUihVD0YEigv+mA3\nZ84c7c9osOx/z0rBJfKlZuZl4+oMuVuxH/wNPBTmGUIVXHant3hLZeQXAMIKvit2xYoVfl8v\nsXLlyvnz58ddEsqD+f7K/GeirTyW8Vnzk+e8sxWhGMOaI6zfvF9pu/U1Wj3uVX0JVIS0S+19\n5UD6BQe7IUOGvPzyy9qXli9ffu2118ZdEkpPPRPb/54tzm9kdys2J35tsHDX4P43f+w3Vfit\nIXDTMR4BaVXmqCq9VJxIlFyILPBvg1RJW9QGkGG+jzvZsGHDhg0bnJ/XrFlTVVUlzbB///65\nc+cePHgwweqQYkU+49o8MEW6qdYhXdYmiWskt3D219vlg75y17zLxZeNcAYAZcE32M2bN2/y\n5MnOz9OmTfOb7ZJLLom/KJRaGs7ENpeXBdapXjMX7QK4lNyI4L0Jw1xM0QpLw0cl/ThKZmV3\nLSmQZr7BbtKkSVdeeeWqVasuuuiiyy+/vG/fvtIMjRs37t69+6hRoxKuEMWW0BXrBf7uDpvw\n7MOcPWk9bmMslsMV4zGP9thk+1UB5Y4PNjLM9M0THTt2HDVq1MiRI8ePHz9o0CB1hr179+7Y\nsYMHFGdJep5DazPqan4WXWBbC0AalOS6DiCrgm+eWLx4sTbVCSGeeeaZ008/Pe6SUErJXbGe\n3GqdHwzXp6ubVu8zsLm8XV2P/Q0clrRlhLoHVnqAi7fUCMc/9h0ESsv9a5C/8ZBVVt8V+8kn\nn8yZM2fz5s319fXuxAMHDixevHjPnj2J1QYcQ+3PpeTSt3jZFG9zK0ksGk6ks+wEJ7fRBqLk\nu5y2W4uA2AUHu82bNw8cOLC2tlazcJMmt99+ewJVIftiH+21XJv2d7p0O22owoqwF9HuIImx\nsMBV+R23ko/pRy6gQYWthoY3F9kWHOxuu+22AwcOzJw5s0+fPsOHD3/wwQc7d+68bNmy2bNn\nP/TQQzU1NUWoEhkT9kq+aKfnwAeIRD7rp6HrYHgpsCFhv+OF7GPJr9cM1ZjhnuLiaIC7DBRZ\ncLBbvnz5hAkTJkyYcODAASFEv379Bg0aVFNTM3bs2OHDhy9atGjw4MHJ14lMCXwSm8pmfm3e\n8gYd70OG3djhTlGXjZ1aYRK5x7zCJMahinyfimWwJkMAaICCb57YunVr9+7dhRCNGjUSQhw6\ndMiZ3r9//wkTJkydOjXR+pBJYbOFzZXOgd8t4U5RezM2d2AY6inwewWKfMUPcQcAMiw42LVo\n0WL79u1CiIqKiubNm2/cuNF9qW/fvqtXr06wOpQ/begJ26vzu1FUu2bLnKTGviLwJkKpWSiK\nmPCijWun55Jzbmk0SNU7BaD4rL4r9he/+MWyZcuEEKeccsp9993n3gm7dOnSysrKROtDVkU+\nMTvnLTWTOSd7aYA1QlX5z74xNtQJMu//PbNaftefFX5KLseTenJZJOyzYwLnITYBSLngYHfL\nLbfs2LFj4sSJQohrr7129erVffv2HTNmzGmnnfbAAw+cd955yReJMmY5iqo9WboZLuwlXAWm\nxrCLCP845bdCd6e0ySNwSqhiYski5dgka5gJrBzfKQAxCr55YuDAgStWrFi5cqUQ4qqrrnr/\n/fdnzJixYMGCXC43atSoGTNmJF8k0q7wu0TdlBPXFfHqQGeoYkLdN1rItvxWJbGsp0zP6AmV\nrT2eBR7GMj3CpVL4bwYAYVk9oHjAgAEDBgwQQuRyubvuumvKlCnbtm1r3759dXV1wuUhs3LK\nFyQUeCmetPJoa/COqHrPSVK16o23fmWYC1Bf9ZtfnW7YR+mlhnxajX3fi59UyEYAQtEHu23b\ntlVWVrZp08b5WZ2hqqrq008//fTTT4UQfFcstGedwBOSlHv8rjaT7j81r9ObwwwjcYb1qAv6\nXQ9nFm0pLUN085vfcrvucSA92DPkePMMDRNHAygyfbDr2LFjTU3Nc8895/xsXgX/3yKawG6W\nN52Yz5pSLim8UyUlTu3tGoFriDHbaVceYcRQG+Aa5rVo0RT/1125/4JN4vMPwEAf7MaOHdu/\nf3/35yLWg+wwd9q02cJviNO+FxLLtWj2I6SBW4mWmWz2N5aTZYwXCCYhQh+xhK1H6eMabw3l\n21Itx5qBsqYPdnPmzNH+jPIS7W/luE4hoTKWdvTT24Gzb1CZ55TWnHQ7oZBEmERh2szaMHsq\n5RuVygiHFyi+4MedDB069IEHHti1a1cRqkHskns8mPmpHNKr7iMYAh//4f7TuyF3TDPwyR3S\negL3QjsxDUOT9jf/xvIok+JsyJ6zuVDvpsOwSNF2IULZJVkngEwKDnbLly//93//9w4dOnzt\na19bsGCB+5ViKAsRTgbRTiHeO0mjbde7KumatshnNcPTQ9zVliTGqSEjDWlSpCPX2hRQeJ0J\nRSWbwkp+hAFkWHCw++tf//qTn/zk9NNPX7BgwZgxYzp06HDdddctX76cPx/TL7n3SD0p5j/j\nnaiewOzPpu4J0ua6Or/em1qDVKc6j1ShZYDQ9i/Dnr8jnO8jHM+wJXlDcNjFC2HZRwy7wpSE\nquTKSMkOAiiV4GDXuXPn7373u6+++qqT8Hr37v3AAw/88z//c7du3W699db169cXoUqUi8h9\nFJul1PFZ8+a0WVO7zlgiS9iQkVe+hcyyfZhoQysNQ342NUQuMloOtj+elsUndJBjT3Vp6OAC\nCCU42LmkhHfiiSf+z//8T9++fZMrDumn/b0ftsejba0Fclde4GkyMPlZrlzbwgxVhnQcDG3I\nCKIdpTTkPD+WtXk/oqndl7ik+f0CUBwhgp2rWbNmn//85zt37tyyZcvYC0IZ0Y60SjNEaMV5\nE5u0IemSO/cHqbliLjiJDoT5bOp3HKTTsNo+VBf07rK5Wyms97QhtGTCdp7Ut6ZhBqYGu+NA\n+bL6SjHHxx9/vHDhwvnz5y9durS+vr5Vq1ZjxowZN25ccsUhabnCnnOR1z0pI9oKvaOQ2pSW\n83lEsDSDG938yvBmuwLPWGFXoq3KLVvdO3VkNpaypfWY406Bn5A0SNWldQCQtOBgt2XLlgUL\nFsyfP3/58uVHjhyprq4ePXr0N77xjQsuuKCysrIIJSIhhZzqLAdbpZZb2Jml/JHz/8JW93o1\nbdZUtxKYZgKrDeRdiboL3nncn9UtBu6IYXH7EWS/wfQsKfd4CgCWgoPdCSeckM/nmzRpct55\n540bN+7iiy9u0aJFESpD0op5qtOmB1XO8+iywHSV033hmPqSYakI1LhmT9trVMed7dcgrSpy\nJZY7Ele/EACQnOBgN3jw4HHjxn39619v27ZtEQpC+kU+wUuXyknrMY+iaid6h1+lTYRajygs\nsWnXo+3M+a3ZnPkMpZpXG1lKhi+Jkn44MgD8BAe75cuXOz/s3r37r3/9a6dOnVq3bp1wVWiI\ntJfZGdpyQslh0hiuV+7YrzEIdUb0u59De8GcYdfUqGef50p1Ik80PgIAYmd1V+wf/vCHM844\no2XLlieffPIf//hHZ+KoUaNefPHFJGtDSkW4Uc57c4BhpNV9VcpAgbffqourKSqw/2RIVNqb\nGGxudJVWHti6sxHh+EfeUBG2Yi6g5DWkE0cGgJ/gYLdy5coRI0b85S9/qampcSfW1tauWrXq\nggsueOONN5IsD9kR9lEj6qnL2zbTrseNgNpHhJhPhOZBUm0AVRexOdcasp1fMNVOsRH2GR+l\nUhZFIlXK5bMNFF9wsJs2bVqHDh3efffdRx991J3Ytm3bt956q0OHDj/84Q8TrA4Z4k1I6u2u\nzjzqL2tvhAq8fs5L22OLQKqtwHMJjRZV2MSfGQ1wlwEUQXCw++Mf/3jDDTd07txZmt6uXbvr\nr7/+5ZdfTqYw2IolcCRHesaHX5029fs18Lxb0V5F5xcdbA6dtMLyimVlkSNt+qkll9D/Yqn9\n31aVtl8y6f/MAKUSfPPEp59+esIJJ2hf6tix4549e+IuCeH4jUsWmbcG8y9c6Q6JvM/jedVF\nQv0qV2+YjXwa0IbFwnFjI8r63ecDDKRTcLDr0KHD+vXrtS+9/PLLxx9/fNwlIbRi/m4N+9vc\nO2f+2JtD1UaadFtrtK2oLxluVrVcvw2pZstd4OxYFpJ4g8rrTS+vaoGGLHgo9oILLpg1a9ab\nb77pnbhz585bb731kUceGTlyZGK1Ie28ozN5D8MiUs4LXL935X6b9k6RenveexH8FjHXEC+p\ngxjtZGlZc4G7lrahN6QNg6FAOgUHuzvuuKN58+ZnnXWWk+EmT5582mmndezY8a677urSpcuU\nKVOSLxKpFvb0r71Dwv1BHZM1J4wIl97bzBwh1kjnOW0YVbduk1nNK9HO784W+5GJjKSIJPC5\nAiTBwa5Dhw6rV6++9tprP/zwQyHE2rVr165d26JFixtuuGHVqlXt27dPvkiklOXf6+pgq/qz\nduX5Yx9ZbIgvNv3CvOfW2ggX7RXI230sZD2W93BE3oqhSxphVZxxAaDIgq+xE0K0a9du1qxZ\n991338cff7x79+4WLVqQ57LNcOGXNFHblAq8n8NNJznjl7p651cbXdqL8wJ3x2+j0sTYA592\nu9qyw4Y2v/kj7EISMTfwzpiyULRLIbnmMiyOFSCxCnaOXC7Xvn17Ih1seHOMttMWtvOkzikN\n11oW5pfq1H/Gft+DNtUVuM40Mwd3iGy93QBSQh/sBg0aZLn8oUOHpPsqUF4MTSPvS+psgd0m\nQyYzrMFyyFJKje69EX5dRvOwoLnFmOjZN9sn9UJGhO2XLUJ8LNrblO3PA4Ai0Ae71atXe//Z\nqFGjw4cPOz97f4e2atWqZcuWidaHMmJohhlGOXOeB855r6uzHKX1m2Jfs7tdvwq9dXqn+KVJ\ncyBW55HWGbjdRKWnh2S544GfljQrx5oBpJz+5ol6j9ra2kGDBk2YMGHt2rX79+8/evRoXV3d\nihUrLrvssgEDBqxbt67IFSNehmvk3Zdyx36pgzYMWd6tKSx6ctKGDDOIY/uCNgv6jQ4bxHL2\ntb9rxG+pMmV5C4U6m+Vhz8AFfAXiJhUAXsF3xU6cOLFjx44zZ8489dRTq6qqhBAtWrQYPHjw\nr371q+rq6u9///vJF4mYhToTqInNO1F6NcL9FuZFQmU7Q8Eqt3ibTGCIv+aZDXHTcruWG5VE\nPtmH2tPkhKohDQUDQEoEB7tnn322pqZG+9KwYcMWLVoUd0lIF/uOiDfB5DyPIJHmyXkIi+hm\nE1AMY5reYryvJpQGApuF2q0bcli8dXqPfOBs7s8RNuH+0/JQFzNQZqy/lZIsDiAlgu+Kraur\nq62t1b60Y8eOurq6uEtC4sKeBvx6bOoPXtp+mHpO9QuO0thl2JptFjGU7feqYQ1qEjKvIdQ1\nZAW+Zd7pYe9LCLVdadkUBo6yviYPAAIFd+z69u177733rlq1Spq+cuXKhx9+uHfv3skUhhJT\nx17NDTapD2fOc34DlIYukX3CMMc1bc3ef0orMWxXbU0ZCjDXU+SmS2DTzq3Hsiq1RWc5IF58\nBV6TZ9nyTEKptgugvAR37KZNmzZ69OiBAwf26NGjW7duVVVVBw4c2LRp04YNG3K53MyZM4tQ\nJUpC7WoYuh1uKyinPKNOWsrv9gXtyi17gTZLqaT2VV556Im2JGkRaYXqdO/8buXqtgxF2sxm\nL2zTLnBO+zfORkLdWa9CDmnYowcARRYc7C688MJly5ZNnz592bJlGzZscCZWVFQMGzZs0qRJ\nfpffIQO052lDHPHrhUj9G79gpA2FLsPZ1Ltabb/Qfkg0r3vYirrvOeWiPcPK1eLtFy9chAPi\nXcpmIDjGHYkwThqtiRVhE6E6skkgTQKwYfXNE+ecc86SJUuOHj26devWffv2VVdXd+jQoUmT\nEN9agbJj6IeF4oYYm46OOboJn5hiFuHqNCleqBsN1e6ShiljHE0LdTSiXatXZBHKK2ZrEADS\nL0Q4a9SoUadOnZIrBakixRpDL01d0PnB70Ir6To2tZsllFhp3oq51WcWKl9q+3lhN1HkMBFt\nc1J3KrkMVMJ0pb3MINTnAQBSKPjmCTQc0tXZ0Xo83oBl7nK5m5Macu5LDmkl6vnYXYn91eWh\nrkP3bjRafEyI9mjYz2wpuQv2tbm/OAp5B7X/myQq1McVQANHsEMU6plGGnA0LKsmDG9r0Ly4\nlN7sr3IrhLSGUNdmRQiaBRYcewJI7nLACCk5uYjjt5vmPxhKmE0BQIvr5PB/1AabZdNOHaB0\nF5QGZNVVGVZu85L5sjzLi+T8xn+9l8SZV6hOVE/25oOpHfgOq8BGVIFriCCuSzm9KxGx7oX5\n8lD1Vb+qQv0PpdYQan4ADRkdO0TpgriJzXui8k5x1+y3Re06zZvzq1kawLVfrbYwbX9Oe/I2\nHDQ1BZoldx2e/ZubRCcyrMDVmj8JCQnsQJsXp58HoJjo2MGXOthq7jr4DSaaz3w53YNF3IwY\nuKy6RXWgNh90L6plH9G8QrVbGbjOAttL5jciwmpj75+FrafwLRazuWW5X95PC703AEkj2DVc\n3tFSw2zSEKH6kjq/3yik4dW88sg07cCoWpJ01lQTnjdvGcZtpaajYdzWb6L2NB/viTxCYitw\ncNZy32OR4dATy66VZKwcQNnRB7tBgwZZLn/o0KE333wzvnpQPIY+lvaitJzPA1DUxp70T3Pv\nShsZ/TocheQMv3aUtDt+19X57UvkerRzBkZPm5njOveHHcX2WyTpLFKqrEPGApBC+mC3evVq\n7z8bNWp0+PBh52fvCaZVq1YtW7ZMtD4kKlrvR40+gUv5pSibTbjTI3cstKlRSnKBLbrAEBy2\nMPvdKeZAXuTdKZVyqdMVy0A5APjR3zxR71FbWzto0KAJEyasXbt2//79R48eraurW7FixWWX\nXTZgwIB169YVuWJEZrjMPOwV6M6J328R7XCqNCX3Gb96tD8beNfml8zcYVapfjc4eldS+AVw\n9kfVPKdlDaHewdjXEC0LSjse9nMIAJAE3xU7ceLEjh07zpw589RTT62qqhJCtGjRYvDgwb/6\n1a+qq6u///3vJ18kooh8jvQmG/Ma4urrSCtRr5lTT//S1qX2nlS2dv3SELDNsfLrMpoPgrpm\nb740rDkU7YEyz6/O6UZedc2Ii3SEObwA4hUc7J599tmamhrtS8OGDVu0aFHcJSEphgjivhR5\n1C/vYTm/+7OUcgwF5H2+pkxaShrS1a7fbyuGYd+w/c5oI93a9Ru2rl3cr18YNu5HiL826/R+\nzPLHPjcn6VgZqpOanJSUASB7gu+Kraurq62t1b60Y8eOurq6uEtCPCI3gfJ2jxoJtRJtwtAu\n6E1U2sab2pDzW7k5Kqk5MvAsGzaomdNnLKuKvNrAmd1jmPO5aaZo4t1iXrmksoRSUgaALAnu\n2PXt2/fee+9dtWqVNH3lypUPP/xw7969kykMsdGOY6ozSGFIPeVIK9EuZcM7s7fJF9jACOwe\nabtW6s/a1YqgdpFhH83tQENV5sK8M4Qd9fbrnoZdj1A+EoVnEXUs0qaXHKM0xKkk9gsAhE3H\nbtq0aaNHjx44cGCPHj26detWVVV14MCBTZs2bdiwIZfLzZw5swhVIi5qwpNGLd0pUlfD3K3x\ndnfctalb9JviFeps563Hu8W8clOtt+2kzazSq4F9QTWa+O1OMftDMW7LcuS6QKXqAsYiwl81\nKd8QgAwIDnYXXnjhsmXLpk+fvmzZsg0bNjgTKyoqhg0bNmnSJL/L75AeUocssNGlnSgFlMAL\noQyjkNqXvK01Q0gy/NOb5wxB0y9x+iW/wDFQm8grjWZK74h5/WFP58UJSYatqPtoyCWlCitE\nJQBZZfXNE+ecc86SJUuOHj26devWffv2VVdXd+jQoUkTvrWi/EjjmH5RyZtC3ClSJDIPTarR\nStqKN/GYr0UzRCttgDMMufplC+1LflvXRl6/Cr1Z0xwjvDOEih3emYuZVMq0zRajou1+Az/O\nAEIJDmeLFi066aST+vXr16hRo06dOhWhJhRI2zIp/Nzg18oyjLpKtFFJvY5Nux5DnJLinU3n\nzLxy7YYCA6i6Oe+orjl4BR46v5WXkGVjNV5S/I28rZIfvbKTkk8dgEDBN0+MHTt28eLFRSgF\n9gKvuNcuIjyNN+cHm/VIDTbvSqR5vKv1vmQeQpUW1F7KZk5OgdQM6m5R2yRzd1Z70Nz57d8C\nm/FWtTZpFwyxspDTbYTPkkjmYjvLUW/zFBhwuICGIDjYnXPOOX/4wx+OHj1ahGoQC+2ZXg0K\n2mUjnFyliWpDzttaUzNfYJfOnUFKWlKCzCk3V5rHcP2qCiTNr128wLBlFrjyQvYuCZZHI7Cd\naflHgrtUyXc8hSIfk0Q/0oXgjQYkwUOxv/zlL2+66aaRI0deccUV//RPgew9vgAAIABJREFU\n/9SqVStphh49eiRTG3yF/Q2rTXXqSJa2W6Ztk2jzU9j+mV+dalxTw6LU3yr+b3abnTWMFaoN\nQrXTadhcIaOQkpScrQOjauDfKrDBQUsC49RIleBg16FDB+eH5557TjsDn+ayk/e/XSDwz193\nHkOnzXI288CuW4k3OGqjp9+a1UWk379SkDLvuF/qMtcQYwITShz0W3NedzuweW2xi3Hl0VbC\n7yVVJo9JJncKKERwsBs7dmxFRUXTpk1pd5cjv/OrX6/OkG8Ck5xfARGaMX5BUIqkanzxG9K1\nbEZ6V6Ut25t9bbpxZtLwceCB0tapbjS1/6vS2MieeP90KVN+v8E4MiiJ4GA3Z84cv5f27t27\ne/fuWOtBaeSPvXNTe9FYYEMr73MDadgr2NQ1aC+nM2zdnSj1/PxK0v5eVoebbcZ8LXc2sJHm\nN49bg6EbapM+zS8VroGf0sJen1BGtH9ZAUiP4JsnDJ555pnTTz89rlKQBJtLntUEYH820l5s\nF+qvVTWHCZ+wIo4NWOqFd96f7TuLUsPMEAclfg0/6ZhL+xU42O23oDC+R9JG3YxuKMOmhsJJ\nAb2QoFNGMUI9/qLML/P3/k+R4dgalwI/6kAhrB4y/Mknn8yZM2fz5s319fXuxAMHDixevHjP\nnj2J1YakqCkhsEGl7Q+pfa/AV/2aXqFSlHRekaKDtKA3OPpdzBd2BNOm3+ZXjLffpk3AoQ6U\noWbL80qiY0ZSc7HwVZWRLJ3XDUkuS7sJZENwsNu8efPAgQNra2s1CzdpcvvttydQFUrGPIQa\neYBVKH/rC5/GmzbQ+PXG1EVyx161Jq0/d+xwsxpbxbFh0Tu/VLO2Vek3eKruoM0VclKT0rDv\nNpf4qD/7vYPa3O+3d2Y2g9f2q4plPZJYQqellGSgCO9mjO8jgKQFB7vbbrvtwIEDM2fO7NOn\nz/Dhwx988MHOnTsvW7Zs9uzZDz30EN8VW46kdKKGFe2J3xs4onXdvAX4vaR2vPzaV+Zcom5R\nGtb0ezUUN0d6Y2vgKVONiepEQ6nCE0cKzyU2YdSGX7pNSZrRirGhmHkcIqBcBAe75cuXT5gw\nYcKECQcOHBBC9OvXb9CgQTU1NWPHjh0+fPiiRYsGDx6cfJ2IkyEShf317e2NaV81B6bAMVC1\nz2dYlSF6+iUk9dXAbpyUg6UeoQ2/QWHtP/26cdp2o3cpNTKq89gUY1iDVhnlJL9+p1kZ7aBW\nWRcPIFDwzRNbt27t3r27EKJRo0ZCiEOHDjnT+/fvP2HChKlTpyZaHyypY5qWSzk/2DSZpLTh\nxoLIm5aW1SYYy/ZP4BnaO6KqVl5g18qbsaSf/RqN3r0OLFud6LeU2u/UHlWpIxj57VOrCrse\nv7UVTbSaGZcEkFrBwa5Fixbbt28XQlRUVDRv3nzjxo3uS3379l29enWC1SFu6qif+5K392Y4\n0XpTkWFD2tO82t4zBBdtwebh18CS3F1zN2HuZkmbkNKYTZTUTpdisbp+875IedRwcPLKpYTa\nNJb+Fk7k5Bd7AjM3QQGg5IKD3ZAhQ37xi18sW7ZMCHHKKafcd9997p2wS5curaysTLQ+WLI5\nPbvpRwoQ3iRnSE7mHCN0g5vS2tRUIYIilN+C2mX9gmNeuUtD2h3L/plaoTcjek/55jil3Sn1\npcBQoj2eBt5Ab45KNsHdpkIzNWdHqMRmK4Us7i0j9pgIALELDna33HLLjh07Jk6cKIS49tpr\nV69e3bdv3zFjxpx22mkPPPDAeeedl3yRiIcUhvLGL5kwj/T5tfrCliSFLUOTT1pKij7mDpw5\nrarblVpiKm3LU51BrUrNB9pQKCwiprZC7QGUsmai3aYI/T+bz0/ktqLa/S1EEg1O7Z8cMa4f\nQEMTfPPEwIEDV6xYsXLlSiHEVVdd9f7778+YMWPBggW5XG7UqFEzZsxIvkjExg1zNicnbfLz\nxg41XXn/qfaTzH24wCmGJKot3m+ieSWWm8h7bqFwj4ma2KRK8sfe4WGzIb9NmytXo6q6EuGT\nDi1rKzzfqC3eJLbiXb/2jwcb6Rl7LUI6L5UM7xpQTFYPKB4wYMCAAQOEELlc7q677poyZcq2\nbdvat29fXV2dcHkoBm3a84YVvyQhhQO/fCOdUyM3JMxDh1KQMpzC1d1Uw6jf7hhWW2B0MJSn\nTg+Vcb3LajuU5m2ZmU/GBf4VEaPUxoXAnisAhGIV7CRVVVVdu3aNuxIURE1XfnNqR1G9KxHH\n5gChJCrt2JZfNtLO411PqA6Z+rNWtIzlrcfQKXRXq/bk1O2qb4d3bYV077RttsDcqU4pPJMF\nsnwjSphpyqhXpM39ZVF5oGzsBVBy+mA3aNAgy+UPHTr05ptvxlcPChKq5+G24tTrkNTumtTy\n0a5QDUDel9TpgdV6+4U2AdF+tFe7d4Z61CwrhRXLYWXveqTFte1Sv6QeeQjVMvHb05Zd4Dq1\nDE3luDZRdsxRHkDDpA920kNMGjVqdPjwYedn7++RVq1atWzZMtH6EIH6i147qiiUGKcupV5I\nF7YSQyDzm00cm7TMm9Y2wGwERkB3uppUzIPL6uCaOe+ah5jVFVruprSs2kSMcJld5BhhE8LU\nzB04qh6X8s1G5Vs5gITo74qt96itrR00aNCECRPWrl27f//+o0eP1tXVrVix4rLLLhswYMC6\ndeuKXDEiCPztn/uMd4owRjEp7piHR21GTg1TDPUHttnyHmrl7krU0dUI3S/vUubF1aPntx77\nrTsMLVXtzIaV+P0Z4DePTdnajQZ+cqS+cqjNRRDqGGZMQ953IBuCH3cyceLEjh07zpw589RT\nT62qqhJCtGjRYvDgwb/61a+qq6u///3vJ18kggWe5GI/BXrHMf1OBoHjp95/euOXtP5CKszp\nCJ84JTX/3MAqpRlpQ9L4oLfyvGcsVZssA/uR2t1Xj5Lf4uZ1hspGhXyEAvdaim7ew1XgprUh\nNbXKqFQA6RQc7J599tmamhrtS8OGDVu0aFHcJcFX4G987flPPVWoTaxolRTSMjF3g6SXYj/V\naVOmtiStuLpHNvulplJpQ2p6tol9UgCVXvX7YHgPVIRd1i5oPgiGD4alCB/yRHuBSaw2Rsnt\nO4DiCL4rtq6urra2VvvSjh076urq4i4JJjmfa/aliX6/mrWZIGd8oIm2QZXE+anwU7gfv6Cj\n9ue0s3kPmreBl1fubPA7htIUqR1oqNwv6Uol+S0lFWbZ5xPHfjDMc9qHgAjNRb/p3t2Rdq3A\nIhNlc1RTUiqA8hXcsevbt++99967atUqafrKlSsffvjh3r17J1MYNArskEmtHUPDxjubd+vS\nRL9BLpsi/bpBNsJ2X4RuhNRvnYauoTkXal9SV+K3Xb/iJdL61cahzWr9SKvN6wbZDR9Cm96q\n9ggk1x5LVYcsVO8wVZUDKCPBHbtp06aNHj164MCBPXr06NatW1VV1YEDBzZt2rRhw4ZcLjdz\n5swiVAk/fi0Zg9xnFzOpbacI3aNovLlBKsxydNJ+K9qxS7UAQ5fOsDlDwZZ9U8Mbp01vgYfI\nsPLcsfeISA2kUB1Tbdnm2vL+z+aw//S6q9L+nH6hUp1NxxQAJMHB7sILL1y2bNn06dOXLVu2\nYcMGZ2JFRcWwYcMmTZrkd/kdSkU72OqXbNQ5pVX5nafNp3DziKTTRzFHmUA2EVA70iqdMv0C\nXGBrzXBI7Yeqwx5D7aa984ca0BRBAcI73T5kWCZaqQZ1iraHGuovmVSlolDhtfDKw2blAreV\nqkMNNHBW3zxxzjnnLFmy5OjRo1u3bt23b191dXWHDh2aNInyrRUoDnXQMH/sBWGGYUSHdrAy\nWu/KryTDzIEsu4x+W9Seh7Qhz92EtqFoHhL1SySBqdrbUjXvjiGx+W1F7f9J/Uu1eKlI8+Bp\nqHN82PxBhrDh/b876cNFcxH2+JwUR4hw1qhRo06dOiVXCgoROChmP9Dpzu83RbueklwSJO2U\nzQ66141544v6s/YHc0tPLcAw1Kuu2W/g2EaE0CwdNJvRYfO4s4h0tZx2nX7N0cBonpBQ6bOY\nrbJARSjDMMIOqPi0FEFwsMvn8/PmzXv88cc/+ugj9/snvN55550ECkP8bC6i0nZr8srFcMkU\nGIU5sBp6bNI4o/0M0vrFsQc2sHWn3QVpfvVnv00Hrly7iLYGN5YZMmgSv5FDXVoX9i+TeAtO\nzzkp2sh7QlJyTFAW+LQUQXCwu+eee26++WYhxHHHHde0adPkS0IMQmWvwF5RaZNc2EFe6SXv\nCKP0g+V1bDaj1RHisiV10DnUtW6G9pvfP83rVz8nxexRleSsoMZ3QyV+/zeV7/ksVT1IFF/Y\n3zwGfIqKIzjY/fSnP62pqZk1a1b37t2LUBCi0WYXw8iguqw49n/glLTlog0fa8dApVac5bie\n4ZhEHhmUzvp+UUCdTTu8HsvvSpteVIwfiSKkw9hPIdr/p7SK8P9OgXsX46m6hEicxcFoe9kJ\nDnbbt2+fN28eqS61zP0q54ewHaNoZ6YC42CMl+6ZQ636G8pwpZ2hkrC/7/zmVwd/zYOz3r3T\nrjCw/VbIxWrmzdkHxKRPFeazfrStW45HFy1qRxbqo8sZHXwGykvwA4rbt2/Pm5p+budGe1LP\n5XKGS9Rj7PokvXjkOm1W7j1K6lWG0oGVTo2Wg8VSBlK3ov2negmgX0jVvtG5z/iVavgAeBe0\n/JxYfgzUnS38Q6jdfe1s0vwl70+bW+lJyMBv9bh+cQEZExzsxo0bN3v27CKUgsi0gcB+ZM3y\ndFhezI3MCDlPSlduIJDCgV/TUR1FdX/2G5TULmh/JnMLC/V5KPyTUMxzreGtdMO34U8adyVJ\n1ffZ+kPFTQAoRPBQ7JQpUy655JJvfvObV1xxRZcuXdT7J3r06JFMbbBiP56SvTNH0julXb/f\nuLYUxdRhU63csV8E4p0ulJDn19IzTFF3xKYRKG1OGraTqvLOafg0uiuRCijJdXWWYddS4Tti\n/1YCgFlwsGvRooXzw5NPPqmdgV9ApWI+nRhaROVLyklq30u7m5b5T7sG9Vo9w7LSzH7dO+1K\nzJFIW1LOeJmUGsUCSS0uNbcZ5HwGpg3vSxGk4bdTGmoA0HAEB7tx48ZVVFTwPRNlJD2n1WIy\nd8UKX4l5NrUdpc6s9sPUfp7UvfOLpJbZTprfm9u0HTjv/Nrpfq+a07BhR+x3QVpEW1jY2eJq\nGRLdAKRHcFzza9QJIfbu3bt79+5Y60EI6kiZ9hyZpVRn33szDJX68YtrUgZSx0O9GSswVWvb\nWn7rNxSszXbqEKo7s7anqF2t9me/Cr1zqpvzW7O0CzYM2y0XuWPvevYb6c6Y2AfcAZgF3zxh\n8Mwzz5x++ulxlQJ72vEy6TIptfeTgd+t9iOqEZb1XiimzcfSEXanh9pi4Duipkm/mc1vqH1m\nCvvZcE/VYceUhXInQeCmtUPblgV7Z/NrfBbtfwr1w+OdUkZ/fUnvYODM0g8lFKpyoHxZDbB+\n8sknc+bM2bx5c319vTvxwIEDixcv3rNnT2K1IYB5tCtU78c8Q3mx73/4dYz8ro3zW9bbq5MO\nptSksSnGmy/VQVs18BlW6Fe58IQzc1USv6US6jn5NTsLXI99ter/RJFr8L59eWW0XfvmqsWI\ngltfRe6fqZ/qspalfUGGBQe7zZs3Dxw4sLa2VrNwkya33357AlUhgF8iUbOdzdoyE+lcEfYo\ncHgxcFlpENOv1aSWatnw87Z2/BbJf3YVnTcdGioPlVS8KVMYP2yGIkOdFKUAJIxH0nI9boV+\n7VJp5YZoHjbn+YXywsdkLaOhzXrUOiVhN5GeGJSeSoBEBQe722677cCBAzNnzuzTp8/w4cMf\nfPDBzp07L1u2bPbs2Q899FBNTU0RqoTKfGovcjGJ8ksS2nlCMYfgwB6n2rQzb8Wvq+edzTDe\nGmopQw3qS/YNTvN0vyNmLtJcgLTywPji96p9p9OvbG2QjZylIqRe8wfMrx51L+yLpEEl4VCg\nLAQHu+XLl0+YMGHChAkHDhwQQvTr12/QoEE1NTVjx44dPnz4okWLBg8enHyd0FBTQiZ/71he\npmNosEmnZO/gl/qq38oD5/Gu02+pwPV430d3j7yXgkVusGk3EY3aF5RqizDCa9k/M6/cL52H\nHb/2q01qrcU7AF34IG/h9fi9C5n8xQJkVfDNE1u3bnW+KLZRo0ZCiEOHDjnT+/fvP2HChKlT\npyZaHwL5tZ0a2u9iJ2r4vaT9pzYLes/i0uiqNEInRS51VdpKLNfvXaFh19xd8O6UXzdIajFK\nPxsCtFqAuhWbIrWvCrvPqt9+hdpQKIEdsrDdr7CFqYv4rcT7l4BaaiGRMdqCcER404HCBQe7\nFi1abN++XQhRUVHRvHnzjRs3ui/17dt39erVCVYHH95+id/vdJfNGbGYkijGppkXWIMUs4Tn\nvOgd7ZK2687j1+xx3yB1JYbV+u2LNmP57ZE7g9RzkhZR1+BXWCiWiyf60S1wDdrFvW9o0qft\nkv+f63cAyStAagUHuyFDhvziF79YtmyZEOKUU06577773Dthly5dWllZmWh9UJnbcqEGLksi\n3lOCeUekFo53KM1vcbVJ5hfIDAVox/7Uhp92094Z8h72u6lO9Pbk1Fyi3TW/t8nQljNEQ7/g\nq21JGgJT4IenOH/GSJnYcpHAmaUd12bKUHuX0NEg1VlK2x/VaCCCg90tt9yyY8eOiRMnCiGu\nvfba1atX9+3bd8yYMaeddtoDDzxw3nnnJV8krBh+g4TqCaWZTb7xUkeyvD+reSJsAW548q4q\nf+xNEoErVztn5iaWd4s26VNq7ho27Z1u6NPYHyi/LBstumm3HuMnOWwHzv6cXYTenqVYKiGs\nAGkWfPPEwIEDV6xYsXLlSiHEVVdd9f7778+YMWPBggW5XG7UqFEzZsxIvkgcI+9zAb46sTij\nRcUU2OUqZOV+BzbvuSw92iakROKNgO4U71itdOK037p9dNNOccvIGe/YsDyv+0VDdbWW68/7\n3B+gnVg4bzpXk7ph5sjUvwSkHrPfny7FT1pkOyC1rB5QPGDAgAEDBgghcrncXXfdNWXKlG3b\ntrVv3766ujrh8iDTnmwij1tlhuEgFL6I2ijK+191584gLa6dQcpz2jWYi/SLgIGkD48aOv0W\nKVwhWcQcT707FUvy8EbPQna/wGJi/B+ZQAZkXvBQ7IoVK/7xj394p1RVVXXt2rW6unrlypXz\n589PrDaE1sB/a9vsvuWAtXlVOeV+WO9S6iCvMA5uSqOWfps2h5XAgv1GadUi/Ur1vmQeS9UO\nuZpX7q3HZs0G2lFpy+Fjb23R+pTJMbwpxS8GQGpZ3Tzx8ssva19avnz5tddeG3dJMFF/j/t1\nejIpML4ELuJmMnPfSDuuLWUvmyNv3oq7Tve/ajT0bisw2+U9g+/SP/3m9+6XuV1nHgcPRX1T\nvDsb+TNsTqXaLSbEZv2B83g/b9n+/xpAjHyHYjds2LBhwwbn5zVr1lRVVUkz7N+/f+7cuQcP\nHkywOhzLZgwr28OvEQadvR0jbS9HBI17utHKmwi18xiqijb+qJZX+PurrcRvv/wY5s8rV+mF\n2vfIXUnDnNGaoBHkdFcQ2i8rjg1zMVYV7woBpJlvsJs3b97kyZOdn6dNm+Y32yWXXBJ/USh/\nOSHSdhoJDH/qOdXv+jNt3sor91io6Ueabsg9+WMfJiyV5/eSODaGSpuQgmZg+jF30RIKCtqo\nHfu2Ys9zEdYfOE/subMkvH9ZlbYSoIHwDXaTJk268sorV61addFFF11++eV9+/aVZmjcuHH3\n7t1HjRqVcIX4P35jTH4xImmthThdiDOEOFmI1kK0EaL1Zz80E0IIsVOI3ULsFqJOiN1CbBHi\nHSHeEuJtIWqLWaidsEcv57n1QRwbngrpuxjKkEKbmji1Y8Qi7s+GGji8h0IaN7RJZoWc8mNp\nhYZaym/BsHttWFyE/zTarxlAtpnuiu3YseOoUaNGjhw5fvz4QYMGFa0m2PP7lZ1cyPsnIS4U\n4gwhBgjRU4h6Id4R4m0h3hdilxA7Pf+tEqKFEC2EaClECyFaCXGCEFcK0UeIpkJsFWKdEGuE\neFGIl4VI1Yi+/aFTx3nd6X7duGhnWZvL4MwVSmtzu4zamd0und91dWrHMa7Pm+UoauwsA6i2\ni5ZEkd53vKyTWVkXD5Sj4MedLF68WAhx5MiRxo0bO1MOHjy4du3aioqK/v37Z/uKrpQz/Imf\n0PvyVSGeFOJvQvxRiBlCvCnE2vCZrEKIPkJ8UYhThBgoxHeFqBdiqRBLhFgixOYk6o7KPh/b\nN00DZ/AbzPVOUS/4ky4EDLU7fulN2q5Qgmwu6E4Uc1qKMbIU3iGLd6nkLt0DALPgYHfkyJH/\n+I//+Pjjj5966ikhxObNm4cPH+58Y+w555yzZMmS5s2bJ15mA+bXVpGaQNKwYBK+J8TdQvxQ\niB8Wdv3cISHeEuKtz/7ZXIhzhThfiB8IMUuI9UI8LcTjQvwlhpILYnmZlP2le9L8fu0fdQ0i\nKHtJnTNDey9wjNjvs+R3W4DNFVSG3TS0NqVNWCYbmwykHij79SchcAfTkOoSutgRQOyCH3fy\nox/9aNasWV26dHH+OWHChE2bNt1www3jx49/9dVXZ86cmXCFDZphkMvplwTOFosmQtwvxHQh\nLhdiWtx3RewRYpEQNwjRVYh+QjwixEgh3hPiFSGuE6J1rNsKxXuEVdqr2bQjs37r8cYLd23S\nnNKNC+4/vVPUXppUjOXeeXtvfvFI2rRh/eo82pQZ14VlUvHm9UhHVaW+ra5o5RVCOoaRa2B0\nBWgggjt2TzzxxJgxY+655x4hxJYtW5YsWXL11VfPmjVLCHHgwIFf//rXkyZNSrzMhsrb1JE6\nNEX7Nd1aiKeE+KIQ5wrxWsLbeleId4X4kRBfFOIqIe4QYoYQzwjxqBC/E+JowlsPJdrxV99E\nm3ik9u0M/TbDVuzHi9XVahex79+Y86K5qsDr3qSV5H2uIFSXkubxG4CWdiTG3qG6C+YV2sxm\nXkPklhu9OqBcBHfsNm/ePGLECOfn559/Pp/Pjxs3zvnngAEDNm/enFxxEEGPmfBe55TE1k8S\n4lUhOghxVvKpzuttIb4nRGchLhWisRDPCPFnIW4UomV8m0j0RGVOYGqqcDsxagfLcqw28NI6\nw7Cs+rPa01ILC2wkq4UFRjS/oVhDppFeNfcIpQUD1yCVZ1ibduXazp/NFLXywExvI/KypepW\nAoggONh5/39+4YUXmjVrNmTIEOef+Xz+8OHDSZXWsFmekxLVWYjXhPhQiMEluqehXohnhfi6\nECcK8YQQPxDiIyFmCtErjpUHjtbFcgGWdmjVHFO0SxnikTcKeIO+dnhXHcyV6lSzpj2p4yVt\n182XgZ/tAmcQ1sPEhgwnHXC/42YuwD5lWlbrF3ztS0JkpFuUi+Bgd+KJJzpfKbZ9+/Znn312\nxIgRFRUVzktvvfVW586dky2wAdNmAnW6SOw3zi+E2CDEV4Woi33VIW0T4g4hugpxvRADhFgv\nxHNCfDnJLYY6pN6zvk2DTUpdhp6reTAxsJtrWb/w7+qp9Uu7IG3Ovj8XqrawxyFwKXXOCP8H\nmY+Y9pBKJalF2nyKpBpE8n/phSoJQGkFB7tvfOMbTz755Nlnn3366afv2bPnxhtvdKY//vjj\njz32GA8oTk5pf5N+S4gvC/FvQtSXsIhjHRLiSSG+JMRZQnwixBIh/ijEKCHiOqcZDni0QT3D\n/G4MklaidnpsavBGE79c5dfD05Yhjj2XS502bSXq5rQb8rbuDPvit6c2M0Tgl1YN82vrt9yv\nGP8SCxsEERnHGeUiONjddNNNV1111dq1a/fu3fuzn/1s6NChzvRJkyb16tXL/doxxEtqIUid\nkqR/xbQTYoYQPxRifXLbKMAqIb4lRC8h3hJirhBrhbhMiMYFr9bmXGsZvLwzmPtqEbhNGvcj\nIaUuKYf5fVqkeKH9XEn7a5N9C0wtfstqV5tQr9pSqAwae6ml3fciyPwOAgkJDnZVVVWPPPLI\nvn37du3a9Z3vfMed/vTTT7/55ptt2rRJsryGzvB7LdFfefcK8Tch/ie5DcRhoxDXCXGSEEuF\neEiId4W4Mo54Z2Y58iWFcucHNTlpk4r2n37TDT8LTydPTXveRcwvaaOeDW+P0O8lm72z/KjH\nkgPUlWhX6xfpLP/iKuQPM+KOAUcGEDbBzs+gQYOaNAl+Wgoik86LRRsIGC3EGCH+TYiyuC9m\nixA3CdFViHlCzBTibSEujm9wVhTQXVMXNEcotbWjBiO/dppf9lK3JbUSpYacd35DB07KZFK1\nbpr0ridUFI6QHQsUYT1hA1bs//9mfmQw7A5aftKAzCu/ZJbP5zdt2rRx48bdu3cLIVq1atWz\nZ88TTjih1HUViXsSTWj9bYSYJcSPhHgzoQ0ko1aIW4X4qRC3CvErId4S4hYhXoxjzYHnibzu\nuns34vgtnvdcEqfGJm2jzjvdMDaqZjW/ZYUSGbU74hcrVdqd9RsF9tuLQhqElnParyT2/90K\nX2G2w1wh8tYPhgSyrZyC3c6dO6dPnz579uyPP/5YeqlLly7XXHPNxIkTq6urS1JboqSTtDj2\nfBzvX6j3CPGpENNiXGMRfSzEjUL8WIj/FOJ5IV4S4hYhVlkvbpPD1CnaRUI1qLzrFLp3WSjR\nJ/fZY3i1fUFhTH5+gVItIDCeCt2HUyWVFO0TGypWxnV2N38kYtkEYsSbAogyCnZbt24dPHjw\npk2bevbsecEFF5x44onNmjUTQtTV1X3wwQd/+MMfpkyZMn/+/Jdeeil7l/0Z/gyNN9WdJ8QV\nQgwV4kCMKy26D4X4VyF+JMSdQvxRiF8JMVmIvwkhgjKKzUvmPGdgH8G1c/5/9s48Xoriav/P\ncFkjW1wQUFwQFRBfRQTZVzUKEXfRJC4xaoiiwRc1LsHduJEYX1zUY1lqAAAgAElEQVRAo+Z1\n3/3pK0oUBGQRUEAS16gYFRQEkU1Bltu/PwbaunWqTlfvM3PP94/7mdtddepUdU/X00/1zLgk\nYGuikOZXWNMcYNGmhTx+mzVVh6zg8EO37tEEQRASoWyE3ejRoxcvXvzkk0+eeOKJdO+WLVvG\njx8/YsSIa6655q9//Wv26aVNoeYviqbRRGPgHuBOYGa8OJFNxGTdx/eA44CewG3AB8AY4JYM\nn7+hfbE1rWkgR3FQUH44Sw1l8+fUXfQsUps2Lrxq8sWWKpO8lm2y+iYNr442YVyltQ0j/TcD\nCqZfSEsvAVn3FITSJPqHJzJmwoQJp556qlHVAaiqqjr33HNPOumkZ599NuPE0iYzu+U6AMAV\nseO4P2jlUjEms4DuwHDgN8CHwGmxP1fBL8y5rDbSZ8hCHV9fHjlO25HPnIICnwltyO+R37to\nIsPlAwpqZKPWjDC2TEN+GbWDxqZtG1Mls1uXaL516eNyyglCiWMWdvPmzYsQK1otR7755pu9\n9tqLL9OhQ4dly5all0P2UNcnpYvOAcAIYASwLmoEfgZ1V3uJz4Ue8BCwL/A34G5gDtA92QYA\nOM9zVARoFRNxWUKdJzQZW0BNroXKRK0SqnfuheMPmv/CuMzNOJS2YeHHM433slFep4c4doJQ\ngpiFXc+ePceOHRsq0NixY3v27JlESmZat269cOFCvsyCBQtat26dXg7Z4y+B+TNrGq3UAe4C\n/g+YECNIhInKWD4l5fodcDXQHvgImAncB+yUdBO2oxM439MuU0/INkkX2F+GdcxZC641x/fL\nuCLJyFzP7bfL+KZt0MjRBoQvwJcJtP3SM4SS8iYd20pKSZcUGahhQUgbs7A755xzLrjggp49\ne7788suBIV5++eWePXtecMEFZ599dtLp/cgxxxzz1FNPjRkz5ocffqB7v/vuu6uuuur5558f\nNmxYejlkifGePqVL4a+BA4CRNTc6Xt00C8rmakS+XCZ7kf0C+CUwEOgGfAD8LrlnETz7A4Lq\ncXRfW0TNZ/Ntyt7fYlOHRgtQMwj9JowKD/bDZ1R1Wm5xjqBx6Ao1CWsf0r772cY8V7W2mL02\npa72K2brgRHylS+M9BcEISZWE+ixxx47//zzv/nmmw4dOhx++OEDBgxo167djjvu2KxZs9Wr\nV69YseLjjz+eMmXKK6+88v777++www5jx4495ZRT0kt01apVgwYNmj9/fpMmTbp169amTZvG\njRt7nrdu3brPPvts7ty533//fZ8+fV566aXGjRsn2/T48eOHDx++du3axCPz+Fc9f3ZHCt9v\nsgPwIXDLtt+ZsMVXc1C30ITdI/jbs7++1wUuAK4G/g2cC8wNWZ3P2dZTJo6xCm1Fk3HqIXA8\nW4yZ+1WY+HyjgV2LTGHbx0Rs+YeyqdQqfHqByWvRjLtc0rOdADEFccwIGWC8URGEcmHjxo0N\nGjSYOXNmqmuVEfHsrF279sYbb9x5552Z6i1btrzpppvWrl3LxEmKH3744S9/+cuBBx5YVVXj\nh6Pq1avXvXv3e+65Z/PmzWm0O27cOADZ9FFDPUYpnQD3Au8C9ZQtXs0VQFsCtmxhn1GMJVPt\nHU9r4FFgCzAeCPUdOYEJ2zrl2FPjsNBTwjj4LqHUaOoLLUnmbOTTMybgDtMQn6SxlnsrthaN\nTcTpoHsagiCUJsXFw5kzZ+adiIHge6bq6up58+ZNnjz5o48+WrFixZo1a5o2bbrjjjvuvffe\nhx566EEHHVSnTtYfrd2wYcMXX3xR/OWJpk2b7rbbbvXr10+vubwcO6Tm0vn0AGYAA4FpNbdr\nZ4Xqmvi7XAwPDVtH0uugCwOAu4CfAiOBx00FPNYxspFsZ72ajo72rzFPRD1/tLpMSVuZAvne\nDXr+RAgLi+MLi9eVlCdHW2Fyc+mge61UvTfqmZWF1ScIpUApO3bB32NXp06drl27du3aNYNs\nHGnYsOHee++ddxZZ4IV50jwsdYG7gUeIqjMSuG6iaoitNw1kEY0PzszNyaK1NQU4ELgU+Dtw\nOnAu8GnNwlqqLjGZkmE1lvGZsGjLWEY5CJI8LaAps0CJTwtrveCTd+kao5sdQ/HijBHlEaRb\n2DQyQJ5vE4RKJdhsmzFjxsqVK4275s6d+8wzzySdkpARI4DdgIuVLT8auURNUqVVLKnF9Gdx\noziwPUKkvU57qqO5/QBcAxwINALeAS7ZdsdjE9bGDCMrrWgRCjW/c9izLJrTmP5R9v81ii2/\njFGZwdnd0dpy7yATjXZW61T85tQmbOeqMTfjLqM6d0wj/qCFOofFrhOEcifYsevTp89zzz13\nzDHH0F3Tp0+/4YYbjj/++BQSC80nn3zy29/+FsCkSZPca3355ZcnnHDCxo0bmTLLly9HtjfW\nmgBK4966NXANcBmwjNgqRg1HE9MwOluFmr+5BMXzUK0Rm11km3tSMhs+AAZs+zmyXwBnhVlQ\ndk/JqJZsQQLD2g6NqiQYa0rbokYL9Jm0Q2yELsLy76Ow65LqCeb4DjU2wZt/ERYotSrauyAX\n6BDFORaCIJQsVmH38ccff/zxx8XXCxYsaNiwoVZg/fr1Tz75pPHLR3Jh7dq1kydPDltr++23\nHzZs2IYN3I+jzpkz5/PPP6+wlYs/A/8GxgMwaQLbPMTMuMY5Q9NzUASNR9ZeHWVNqgfCA+4H\nXgT+CrwBjAGuIb+cm9LqMGpOpY4Lu/H3IupDeH6e2gnj77Ud0KS0gmaDMdqOUYRQzkb3xOgt\nEC+C/ReB45yGnKJ3U2m3WFJUfAcFQcN6Pbrpppsuu+yywPonnHDCU089lXRWUdiwYUNRiXbq\n1CnZyKXw4Qkkqif6AK/XqYM33igccoi6Pez0RpNktgdacUz5XBgK3A2sAc4E3ki5LXXK50eA\nTs/utqJxF9VkGrybZSzJ69Q4U6zjyRa4VzWMI2Rl7Fr8fvmkpEKM7+6K1z0V30EhF8rywxOX\nXnrp6aef/uabbx599NGnnnpqx44dtQJVVVVt27YdOnRoyhm60rBhw8QlXV6keiWqA9wO/G91\n9endumlzfARVB5MBYFx0C4zsL8vaREl6i7BGXgCmA7cB04HbgdHA96m1ZVuqDqxSfCiKVrFJ\nN7UVNZq2Jq7KR9siuxrcFlbDKPhCSQ3aRMHyHcX8+q+2SBoBLX9m8TrwbRVKGjpeHIzFjCI+\nY8WTvczKXdKJshQyhnvGrlWrVkOHDh0yZMi5557bvXsaP7CZMN988823337brl27vBNJngQF\nza+BvYGfA6crG5PSTIGLYrYqdAaKoHWM1ePwLXAG8AQwHhgKnAlMjx+0JrYlS5v2ssWhFpSt\nIaMw4pPkx98ooWyWmE19wnQm2LZoOUe7IdGajmZUB9ZVx5xvwsXGrm3U8u4LQjSCPxV7zjnn\nNGnSJINU4nPrrbdWxtegaA5EgqquKXADcCPwJWnLf6E6N2ExGiS2gMyUrwUxrh4yJGvsvQx0\nAqYAU4A/A42CytvSsw2OWiBwJgvsmtG7Yo4p9epcFmS9bdCG3LO1tWXbDssYJjLxM7KVGTrf\nMbWFVd9ftmhqf/mDBcvg0Cr8GOYolXJsPUu/XyXfARdqIcHCbtiwYS+++GIGqQgaaVyGVl9y\nyffAX9hphslHs1uMdR0vYZr1QlejGJ0XamQSuaSu9rxzgCHAMGAewH+pI+9sOVYMHBB1o3Hm\ncOm4pp5Vkce0y0cuRjAqP5gUYaBy4lu3ictAyUULMyVdEuPju0dWyyeuCTITN3HuD30S7L5q\nnQpCZRMs7Hr37j1t2rTq6uoMshFUjOZEHNoBP9xyyyXABnZFLPG5xHGCt9VFTa1gLMCQyKW8\nGOQfQCdgHjALuK7m77CFDQX2wS/b0bF1Vp1E6QsEHVPbXk32MY0yKRkjFxTUbG2Fba3w2KSw\n0VZkblT4LsD5ZNaixbw1cknSSKjxZI5yeYkk8cyE2kPw99g9/PDDF1544ZAhQ0477bR99tmn\nWbNmWoFsnmk7+OCDA8ssWbIkg0yyJNByCMUYYDbwNInveMnjnZLIMG4NLZnZ1dlTVkXVDFcB\npwLPAuO2Paf4z6hN2CZ1VfkZ3TKjLjHu8pTn6hwPn2/aGaPBYWFd20g1h5Ywc5LzJiWNxvTL\niMv7iw6v35BtbNX4fPIZE631aGq7ZIVUlpcRQciFYGHXsmXL4ouJEycaC2TzJlmwYAGAevU4\nl2Tz5s0ZZFJ2FA/QYYXCz4GqBQvQuTPvxGRM4OzuGCHZrHyMkZ8DZgDjgLlAgz//uc6oUYkM\nYsH0mQCbtvNlFtVbnuVzEsa2NP1qdA3VklomMB1BLTemrraLVtFGwCZq6RbNGwOr/IyTvZYA\nI7KjaYWwHlu0t6qWWyKqLnIyjqQkf0Pdx9rIV5oLggvBwm7YsGH169evV69evsb7xRdffNdd\nd82fP58xCC+99NKbb745y6zSJpExLxQK3qZNr+6/P3r0KHTuDOd5KO1LmMuUo+ZgdJJCDZFN\noNjaZVgOHA+cBtwxatQrwOnK51Fc0gj0qLRUjUadr10C5Y4xFO+H8TGp/8dIN4pLdT4O1Z2O\nTRtL8ofbplOL48/X1d5E7rJSixBYjM8hgp1ZkcgIlD4ineMTLOwef/zxDPII5LrrrnvllVdO\nOeWUWbNm8b6dYODee/H555g0yRs/3mgLZZ+ROl1p2/mJPI7YtdWN7Pk9CMwAHgb+CZwDPOO2\nkOrYFtWvLgJOa0vbEqq8WotaesZiUPQ3agogY8I2xUMPtypNbOMQ6NKpspVxE/lUbZnHeSvx\n67kxb/ACb5nSJmxb+U7q/HEUwZE2qa7A1BKCPzxRItSrV++RRx559913L7/88rxzyYikTu6f\nAivOPRejR6NFC9S8MAUqg/SuYoyHoZkxRj9G3R7HsAlML5BFQF/gDuAJAGef7f84SbTDF3Pl\niz+4mjq0FaZiyBhK1V4qttZpL7QqTHM0jk2sMFaolptRDhorGrdoQWwmKBPcMQHEO5/dXepU\nKRc9lPtACSifs6VkCXbsDjzwwAYNGjie7rNnz46dkpUOHTosXbqUeZDuyCOPbN68eXoJZEPi\nV5YrgFXAjuefT3fl+/6xKTNjVoxjVLD/DmZkjF6X7dBsBq7yPMyatahXr/nAL4C32PwdF/60\nLTQULaxZZTQmdQFtTQdaaGpzfF+MuKg3mLpmK+AS2dgpKnBVi5HRyjYxZwturBtBskejYHny\nIQPrrozmabGLhAogWNgtXbp0zZo169evL/6rXh0aNWq0cePGFLMjNG3alNnbr1+/fv36ZZZM\nGiR+TdkDGAH8Cni6QYOyuLxGXgTxlI9zIkiKoeYVnMog22Kfja2z45o1DzVtOhP4IzAG8NPl\nV/2MqpSm57k9+h24hhgW41Rnc0l5O5Bx2lSVb+umcbtf3kUuh8XllDMePhfZSrtTIEvYEXJm\nWgxbxV13VhK1sMtChRG8FPv+++936dLlvPPOmz9//vr166urq1evXj1t2rTjjjuuT58+K1eu\n3KyQQcaVh+qjJH4pvx6YDzwTL3IJ3sL6Go5qBd8Lsc2OgStfkftbaNr0NOA3wB+BiUBLt4Ca\nB+biA9HyRVR5xK+l0iraFhpWG7fCNph+8Sar7SjwOpjuUqWV0ZRyhz836PlmHBl1O9+W+i9/\n3BmYU4U5BGoacYxVQRBKjWBhN2rUqHbt2t1xxx2dO3du2LAhgKZNm/bt2/eZZ56pU6fOqFGj\n0k+yNpKMwps375d16vSYMSN+rNK5vtvEBOMu0BmUVyR89UAeBg4CfgosBI5QGqVhjRIBimGj\n6bPApjWN6NIRm21pC8tvodJHG2pNudI+2g6NTcFHQ70xiFDdpWLYyNrJEKq6bcQC0ygo2LIK\n9WZJg9wTEITyIljYvfjii3369DHuOvTQQ1944YWkU6p1GO/141/IPM/DxRfj6KPRq1d8mVgK\nyxOatekPVJzrvku/IgT/BOgFPAC8CPwZqG8JWyCPcDGGnK+B1C00rFbGFtzWcWrwqMpPi2BU\nzFoco6dlEy6aFnHXNzbPjPZCw3jyRD6jYkoQ23ufD8uMUsy3bUrv+pjvWUEQeIKfsVuzZs3S\npUuNu77++uvVq1cnnVKtQzMwEos7YQJmzMA778SPVCKqzn/tkaeajFLGq/nUHY8xgju0+ibg\nUmAy8CDQDxgGfGKq6N6oVlLTSaqi4tfd1PONaSLwBkN1vBjZZDQmGYwBqdB0F6ZQ3mIuR9mP\nkPxb0p5noeazoWk35+P41nAJ7hgNzk+LuicgCIJKsGPXsWPHO++8c86cOdr2mTNn3n///e3b\nt08nsVqENvMlciPrbd787s9/ftemTdhnn/jRSgfNjtLUDCNueB2AoGH3FPiUNF4FDgCWA/OB\nYVzPrJ8MsOXD5Mn8y9RiLDRjQF+BqeYcDaIZfrZ/1S2MZcVUN6Ipe6P6928AaMCwkkJVumEl\nOyO13Z3LbNB6F0GSllR3BKHCCHbsrr766uOOO6579+577rnnXnvt1ahRo/Xr1y9atGjRokWF\nQmHcuHEZZFkbSPhK98ADuwHXAOcmGTRPjKLHOBdSKaPaFZofY6xuFBZMFX4K/xoYDPwBeAgY\nCIwE1ltKanEC3TJakokQqBr9PjJuiqbDjEaai9hyt3Zou7aUeHnhsUvY2l5+EBLB3cUMlQk1\nAhFJF0Yw4ULFFwQhPYKF3dChQydPnnzjjTdOmzbt008/LW6sX7/+wIEDL7vsskMPPTTlDGsL\n2hXZZYI04nle40IBV17Z5Prrl11xRZIplgyaaUS3GylsWxa0uSmq8jNWj5guUO15hULhdeAx\nYDZwMvB+5FhRsalVYxl1l7pwZvNpQg2OrSKVI3QX2LeGKi6NOs+m2mnrxjI0oI0IQsfWhEso\nm/iLcxlhBGVMGUePryAICRIs7LDt++Gqq6u/+uqr77//vlGjRq1ataqqqko7uVqIzXoJFcG7\n5hrccw8uvDC5vPIkwjTg6Dn5sxezFuaYnoavWlQdOQvoDNwPvAls9/e/F844g6nItMhoU60M\nLFN7hC7TNcrAVUvPtGiOmvILFm1H21WFl/Zai+P/Sw1FRsvS5oynCiIReA4z/rFjcLW82hzd\n4oh6CrnXEgQhd4Kfsauurt5atE6dXXbZpXnz5nPmzHn44Yc///zzlHOrXcR/jKZYfWcAY8bg\n2mvxk58kk1n5YBQrgat4Rt2TUjIrgWOB7W67beMZZzwA0CMUqNgCM4xgodEIvnZ09C8D13kR\n5gw3GqvGVUVVu3gKxiRtvVCr2Hwv2y4+uNEdZO4EtN7FPxniow6m40kVWCz+tU4QBAZO2L3+\n+uudO3eeMmWKv+XVV19t167dCSeccMYZZ+y1115jx45NP8NaRJzJuIjneUuHD8cee+D00xNJ\nqRRgpgF1smFWsjSDx98SOIuHdTjgoKs8ACNH9gb6AnOBjs6RVQLVKl+Smk+qsqFTuE2l+dAq\nxghGcaZGg0lAGIWaVoyac8a0tSDUvdP2hpUgjG4LjOOuRwObptI8mpaKUCX+RSwC7qJTEGoD\nVmE3f/78I4888u233/7qq6+KW9atW/eLX/xi48aNF1100XXXXdeiRYuRI0fOmzcvq1QrmeKF\nKdpdbI1aH3+M++7DjTei9i2UUyPHuFed6hh94xfmG9IKe26PNBUKhTeBLsCHwLvbbXeaQ3lN\nuCRieBjVlU0I0vLaXgab+cf3wijBqY1Hlb0WXFU2zImhtWvrBZOwo7tmlOlaL9QBdzftmBsV\nPu3ImUdoThCEtLEKuzFjxmzcuHHChAm/+tWvilseffTRFStW3Hrrrbfeeusf//jHKVOmVFVV\n3X333VmlKlj58Zp75ZXo3h1DhuSdUUZok42LAaPtoj6TTYWoZZgW3ZNfBZwAjPzuu3uB+7Yt\ny0YWTwxGAWeMSd0dm1ago0er2GSxX5cqVFXfaAKOiawdTepXGWUTTwQlaiwTykxK5Chrvl20\nTOLAjFJKCUTzIwWhUrF+eGLWrFmDBw8ePHiwv+Wll16qX7/+qaeeWvx3n332Ofzww6dPn556\njrUA91tzjrffxhNPYNq0ZHKqfTj6bSoxD5kH3A68ATwJzAFOTGfmC7QSjR4SU0CTR4y61YQF\njWxrzi9stFSNkstWBSbRaUzAOA5qi9q//mvfMFPVqrv9GSFhHtuhMaaRmSRiDpAgCAlideyW\nLl3auXNn/1/P86ZPn969e/dmzZr5G/fdd9/Fixenm2BtwrZaxFf58Z/LL8fgwejdO/nMyhyj\nD+cpqBuzT28ucBCwCHgTOMUtB17/MQIFpoVR/1+j66aWVwv4soZRgVTkaRFoXWZVke8vtWMd\n31DU/9P+ZXSSTbrZ1C2fhkvCLsabzWfN8vS2nRglouoy8y8FIXu4rztp0qSJ//q9995buXJl\nr169tAIbN25MK7XaSqjLjW/G9AWmAHUWLkwvsXIh0BJIxh9NlJXAMcAo4EHg7kKhAfCDpaQq\nhiJYjC7YYhr9M2Nhr+YXnWgDTpV0wfTVJLzjpRZgTDKbFUd7Sp05oz9HC2sJ2Hqq1aW9Nm7h\nMfbOUT/xHmdKxOyvEAH+LSBUJFbHrmXLlsuWLfP/nTx5MoA+ffqoZZYtW6YaeEIEkrpxvBF4\nFMB//Vf8UGUN9eTg7NwEYouQyEXTA8YAA4ChwEygraUYY4/ZyvgbmTxjLvnx57A2tRjPeV9k\n2MSWMVXmyFIN4fJGowLRsY8geo5v0e+v2rSjueh+sKg7a0wg/iXI6M/F957TI2P/Ml9qT0+F\nIlZh16lTp2eeeWbLli0ANm3a9Le//W277bbr37+/X2DLli0TJkzo0KFDBlnWEooXxwhvQu/5\n53vWr/+rT4y/Ml+74K/X6l5t1c8j39nhuO4GMj1HzR0AZgCdgS6HHz4POMZeLKYO00JpLqCt\niZSUsaM/B3b81fyp8rNFMJ4t6s0An7mnoCkkapsZzy5mSI2CTBPHftMR5FGy9zyRSanppG6Y\nKwBRdbUQq7A7/fTTFy1adOihh955553HH3/8v/71r9/85jeNGjUq7q2urr700ku/+OKLo48+\nOqtUKxN6cx/6elRdjSuuwNlno63N5RGsUIcmsDBP/OlkOYCXX74NeBoYA9Q16Y/oZ4udUHOh\nr8YCx0QtQ2vRjvDGnhaTKaNGtklGYy/om5EXYZqsVJM0lncpkNTx5XVbglN+NHUomkMQ0sD6\njN3xxx9/wgknPP3001OnTgXQpUuX66+/3t87dOjQCRMmtGvXbvjw4RlkKdjwPA8PPYRPP8Wr\nr+adS3njWb7WruD2m2OO0tCxcKGqCsAs4FHgEGDXQmGJsVhNB8hRmwY0bX82jpZ0sdmMW/gl\nRVu7VCDSXfRfP0mm72GXOLW2GKPdpoADR8A4Yv5rT/kErjpc/utofYkzDiVFBXRBECJjFXZ1\n6tR58sknp06d+t5777Vp02bw4MF16/5YeNdddz3iiCPuvffexo0bZ5JnJcNPnwEVN27EVVdh\n5Ei0bJlCapWGbd5i5jPHGSLU4XMsPAnoDDwBzAd+AUx2b8CCsS9MMsYlZvV0VZ0qZgyNhhZM\nw26Uqp79MX+PfMSBJqkFL1i+nUSLWaj5OWJGU9LtgTrSBdvZGCjaAg9oequfIqcEoRTgPhVb\nKBQGDBgwYMAAuuvOO++sqn2/bVBSbL2Gjh+P1atx0UV5p1OihFJyRosocG6OpcuDKi4BBgA3\nApPq1sVVV1WNHl3NBuQ9RbUYLayaPS7iwObVMUYaP/erYlFd/bTpQhCtpm2nuHiQWnkX4WvU\nW0zato0uNxj0Bb89FDGVWeloO3cJyx8mQShHuN+KZRBVVxJ89x1uuAGXXormzfNOpTywPQnk\n8oQQv9YWAduzXxqbgIuA4zZvxpgxLwI7OEejOsPHfQkVNcUQXagNrEgz4SMz2svvo3+8tL8+\nfigNmkBYR9avQtdGjR13lwsuJ4PjOWMjVD5Qzpk0gqdK2ExKJ3NBiE9EYSckTpTL4tixqKrC\niBHpZFQJ2FSCNtrG2cvRBOJ3MVmF2vUc0G716lbAfMCbPduxCdtJxSfs6Lvw9hL1k3i3TEsb\nDlaKcSGVZsi3FVjMFo0foghvZ7XvfJnM0FKKICu1KnFUaVgcx6qk9KggJIIIu1Ih9EVz9Wrc\neisuvxzbPqosJIuvG6gDFD9yhF2fAD2BScDG7t0dtbzRxjN6SyAWlPbaphQZa01tnVm31SLT\nnDWXLrCPCKO6+GOh7rUZfvyYBGpo2gQdPa1RJmAgLtcZvtcxyVLbCULtRIRd2XLbbWjSBGed\nlXcepQ4jxdSlPbUwlTuOUiBZaNrrgd8Aw4FbgMcA5oNLgZLOL2ZsiAnIdL/YhFGWGV8YW+Fz\nUMvYVKmajGPy2nGn2tfR+DT2Mew9W3oGkksazEC539Xw4+wSoTSJ4FkKQvaIsCsJwl4sdgBw\n220YPRoNGqSTUeXgeC2mVpY22RttjFQnKlvaDwDdgS7Am8B+IQMyzlmo3ALNM6OVZROXjEFF\na/EyPVBsGe00rWKBfAzW/2uTrbCcIVp6MI2SrcvMQYmgMIyC2Nh0qLC2hkpBANnOh1LITRDS\nQ4RdWXIRgBYtcPrpeSdS3vgrX/4W4+wLMq/bBIExfrLZFvkncDDwHjAH+KW9vCpAGcOMMaVs\nYR2nRuNwMU4bI6Ztii1QW2s+HD3otpxtatI/6P5Q+P/SirS6sbmw8OqWqZKBYaY2FPPOJ2MF\nFqj5yt1xFGoJIuzKDM/zdgbOB3DVVajLfVuNUCT+tTiCoeVXDFWe+lvqCy3aGuAEYDTwAHA3\n0NDUekH51anAVF2y9V2xQEOOCWhUljZbLuYY0r2aOjf2wm86/plDt9iUa1j8syKUyHYsxjiO\n7rnFSSNUSSYNo8iO/HYWq08oC0TY5U/Yi8XSkSO369gRp5ySUj61Cv8qr+oVEDOGGkv0hcts\nwZexKR6rBwPcBgwAhu+yywzA2/ZjwVoyLqrOMUOGwIr8ZOwooFQAACAASURBVE+NUlsX1ALq\nMXLJQUvGdviM5pbWrrqRPw00B4vu0g4Qr4lthiuTAG2CxqRjGCqHUDhWz94bE0NOqAxE2OVM\n6EvkkiUYNw5XXw35KsEUsE1ymqXkF1C1oKPjFTaZwGIzASxY0OXww9Gly9E1Tak0Zim1s4Fq\ngIozTTqrhTX1RrugRQtlINn0FlPFWNdTfgYjMAHtxLC1GNZ7M+apYr0TsDzPBzLO9H6GOckd\nz3ybRcpQUjJLZJ9QLshaXrlx/fVo3x4nnJB3HpWGR36PQdMZtms6sysRtMnVaNgUCoU6wB+B\nZ4BbC4W6wGbn+Komo34SbY6Oj81T5LxGNymsFaMZUq8LyqHkg/PYZKVN8dDqjgnQHjEVXWL6\nGRrdQcewgceONqqOjwggQcgREXY54272ANgD2DhuXP0XXoA86pEaeQm4wCUw4y5/Aq4GrgXe\nAB4BegAnA0tM/lmB/OwEo18DpRvfF9uioercaG6WTZQYszW2pbpNWl9ceqHpEpuA47WjmgB/\nzqjNaTEjK6REdC3CHHSXtoyK3GjohstSEASCLMXmj8vVszhDXAksAHDUUeknVeswKgYfYxl1\nb+Bym0sC0RbjVF4FDgIAzAcOMzWBoI9tMj6Nu+VjC+KuWdWh0I4Ck566qkhFg4vS8svzJeP7\naklVNJ4zxsNhW0VlDgr1/AIzdBxhQRDSQ4RdzoSYyP/971/XrXvIK6+kmU7txVE3aPMWVWMx\nlVlYaHOLgQHAg8DLwDWA7UnMQClGu8nUpbrWpnQZzcTLvrA6DyY9V4zjmH9B+fCvv91RtbgX\nM0pq/iyiasxFMWegtxiZqL1T6BHUjlHiCWQWIS/KN3MhcUTYlQeFQgHXXYeePXEYNWKEZOAd\nGl8laMtngXVdmrAplWiT8WbgYuBYYATwKtCSFHA0aWytuxtvTGFNOTlG87Z9NCRQQKg2lXrI\naJmwOJpSoY6dMROv5ueyjdpa1UN8NGNztG58HM+uyEEEQeCRZ+zKA++DD7Dffnj11bwTqRW4\nmBwuU6Nn/30tYzRq3sSZ3v4POAh4AlgA/BJ4zZK/lm0iTRvxTE+/+R2nYwWTbrYJRxcDD5aD\n5XfcF+vGo6bW9cvQdjX5VQh6FE9t1KUvNBNbcL47NDEmVUeY6sbjC3svAt96Cd4IJRshL8o3\ncyFxxLErE667Dr16YcCAvPOoFWhuHCwrR4FxqIAILO8opxwv4p8BfYHHgVeAK4Pe7S5y1pZG\n2EnFqCHoCLtba2rytgOndTCs0RVB6WquoSqaeRXlp+0nqXXEJRnqVmoVqapW/2qh3LtPe+p3\nyiWgoyEqCIINcezKgH0BPP642HXpQT2AQAeIFwRGdycpvJofJuWDbwQuBF4HHgD6AKcCS92a\nKL5w1D2OOkPNlq9ic5sCm4Dp2KmNBkamR01VRaG0rO/YMSImcAsTHErXtLM37CmnpWpTYyDn\nti1gsrcoMauEJb55KQg5Io5dGfDBL38pdl3uGFfKXGZQRylDy7sUcIz5HNAZaAosAA51SMDF\nJwulV/hs1bE1emz0X1Vgaf6Wv5HPwajwIhwsrSRtmookY3rGigXTZ01C+ceaDtOwBdEOQSgX\nTS1sMxo1VRpIKMswkbrJ3owJQpaIY1fqyNN1GRBq0qK+BcWfj92nh1A+nEsQbeOnQG/gRmAi\ncCNwNbAlieDGApoCcxdJ/qgGHhHfXqJCjTatyRRVrFDVpek8qvwCbSpeMzlGoJ01lqQFAr00\nHlWT0XbVvcbmIgjBtIn2biqd/AUhLOLYlTqPtG8/bcsWsetyh3FiYFn+c1ltRJBGDAUvGjYB\nFwHHAL8DpgC72lOyBadtqVDvyhhWc4yMpg4dPTo+oUbMd7+Y7hgPGVV+anpa8rSuf3CpL+hy\nhhi1sgtxLC41AWPm0dLQ3kF+71xiMkrRJUJYlSaqTihrRNjlDH8FaQ+cDFwj6wKlhHGO0XwL\nbQLjo9EtjofbGJm3oAC8CHQGCsACYIilrq05Km0Dp9VAAykRCaLF0cSEup3KC+0Fk61Rv9IW\n1S1GXRioG7SOGMtrqlrbTkvSJiKcZsbTlW89bXjBJ1dOoRYiwi5n+OvOH4Gqfv2myB1krmg2\nEuNA+H9tJgd1jFysqcT5AhgA3AM8D9wGNLCXNNpUtIzqYKnl+cLaduYktykb4y7e4jKmamvF\nqGW1ULbjq0pGbXuxlktwpkeOqkU7ewPLa4TSo8bqgTc2LjI38JbDtlGunDZE9VYwIuxKl6Jd\nh6uvlmtT6WOzxzSSvZLGibYZuAL4GTAMmAXsQwpQheqShk2vUAtKE1j8NOOXNCrjwMRslhWz\nFxZlpjpk/EFnzDa1DNWFNn0Z5zqgjr86mIGze9jp31jePUiy9zly5RRqJyLsSpc/AjMA9O+f\ndyK1HXUiN65zgUw/2S9IUVxmxMnAAcBSYB7g/f3vxuphvTSjCFPLeyEfZmfi0DI2/8z4r+pm\n0S5o/xr1irH7vNak2ldLg4al1iAszybSvfFFYbKofTH2SytcfEEFvRCTkjorhGQRYZcnzBWq\naNf1mzIly3yECNCDqLo7CTYUuKTlWFJjOfBz4ErghzPOeBhoUjNCoKljM6tsUkPdTq0vilfz\nCTnN5PMjMEdBTUZVOWpYpmv0r637YQ+3rV1NkKmjpI6euwdGw6ptBZ5XofplPCJqEPWI08hG\nvU4leMVru4rvoJAqIuxKFLHryg5NczCXZsZEYSZR92t92FnBA24DegLdgAXAIXbRY9OsLnO/\ncZVN02o0lIvqYppTCxQU+OQZj5A/slBOAGMv6OlBhY5Lklr3bacTLcZkHgc61AhSrlp1W2Tj\nmZaN1RThliapdlOKLNQSRNiVIvts+zCsUPrwlkYooeZF/fq6wLbUMrZi84GDgNeBGcVP7ZjK\naCJM2x45K8aIsilI1ZCjhlAoVBWi2mNUS4V1y2grKr5SNJZXO6hVoWVohmp5xzEJ1KzuUPll\nC+5+vOIcYncCB0GV3SkpsGxkq1DBiLArRS4DZgOyClt2OE5dtsnDOCu7N+0yzfDF1gFnAqcA\n/w1MBfbYlgafJ59qfM1H/y32wua0qWqGJsZLberq8VrKU2DGlknDFtDY/fiEVW+R1V4oxRZY\nOBetwySWtnGoercpNSFUMCLsSo49gV8C18t9W3miLUipPpDxgAZaemFPg/inzdPAAcAm4G3g\nl/bcVGmS9gxkHD3VPzM6KMahcFfPgV4aM+sbvTTjKKlSlWkrsAnbRnfiVA+sGzM3DeNgJnIS\nundEtJdQsoiwKzn+ALwNTMw7DSEU1NcxTu1GdcLPJWGnDb684+T6BXAo0OyWW+4HHi0UmrEN\nhTLwwmbiN8HoHpvS8qd/3l9hRsymw5is1EZ5v8fWoqYCvagL9EbJqNmBgZGTFWTxsb2PAquk\nlE+qg1Nqgy+UCyLsSotdgTOAG/JOQ0gEm62CmvOBo7BI5BLvMpEXX1QDhUsu6QEcBKxq02ZQ\noq3A8kCYLRlbAUZXuQT3TL8Va6xFhTstbMQXZyo0f9/+cVefvF6JcD9Q7s5TXjJItJdQaoiw\nKy0uBj4AXpCLRWnjz4LUDoGDUDMaSI7TUrRa7mjJFz9R8T9ffPEP4HagkVsQm8kUzcZj9LGW\ns23kGddKS9imwNSKWrRAw1JNL1A/qVYflZW0ehzLzdidzIjWqCrHHc988b2E2oYIu9yg17Wd\ngbOBGwC5CJUy7hOSsaRRhbhP+XReDzVBRpjh1gO/B34GHAssALqxwXkZEdhBukXVyryc8lth\nklGljDp0jmNo1PFacF6Das0ZXcBQqoVmVfzXRc2oFbV7ksATMqYcpAc0FCLUBIFBhF0JMQr4\nD/BM3mkIPOr8zUsHf6OmBmyTInVoNHJcLJsM7A/MAWYC3ujR3saNWgGjYFI74uItGbWdS6+N\n4smvq+bAROPlgnYQvZrfnBzYtPbaqCw1o86WISMuI6Aeo8wEk/8mcrmlsUXQtsTUmoJQMYiw\nKxV2AH4H/AmozjsTISZ0dlFVIO+C+C945Zdgbu6sBk4HhgHLr7tufv36+9dMyRjZZa5VI2iK\nVtMcKrQhNYhxr5YJjeNny6Tt16Jijm7UtlNBZrwxMLbLZMIMmhHbkVJ3BQZxcQRtrRvHIUKo\n0iGyPBWENBBhVypcCCwFHgdQ/pc5AZbnonxjRp2SbaYOhZ85sjltngX2Bz4D3gK8a6+t7zyZ\nGdML3KiJXZcZ1KjtjJEZiROqRa2iUfzRvZ7i3Wb2llcFXKBXR4VvGikl1ffIWjORpiHXbaFk\nEGFXEjQHRgA3AZvl6lD+GLWa8bBSIyfO3OlSN5GzaxlwHHA6sOLKK+cAB7jVsi0g0u5rOtgI\n71HZKjKaj4lPgwQONS/paBAm4UATMVRizNlYTI86kTbTMdq5qt7tlLLFFTa9Er9ul/hoC8ki\nwq4kOB9YDTwEINfnqIT0UD0Sx4tsGlNFgmfX48B+wCLgTeAqoF4Yy4RKN1XxJJUh33Q0g4ev\nblylZbbzDQUakzbHkanIS1sXJa1qsjjaLkfkGitUNiLs8qcx8HvgZkB/HF2oCGyziNEgcalI\n40TMLDZfAycA9Z544jxgLtCFlRQx82Q8OZdGtVVvX17b1iJt/qKtFUeZqOlCYy1VmfmKUD1b\nbH4nH017oWprRqVpg+Ybfo6Lj7Qh9Q4nx1PXxdrMXYAmSIV1R+ARYZcP6mVlOLARuD/HbIQU\noMaJ0ezxahKtoZipxqQwbFgn4N/AHOAW03fdRfCrDK2E1Mdao5qkcI/v72IOkNEkcxc9tpI2\n0enYSqBQ01QXSB9tEQKbZqqXCBWvcjT5LtQqRNjlTAPgQuAvwIa8MxFSwt3LoeuSKI0ZKFB3\nfg0MA44HTgH+BQysudfFCopsT1LLildgxi2O2pqqH6NO0uLQ13TG1cpoTWi71Ez4VAvk646Z\npsPqXUdCqWG+xbBKxVY+2fdUaYon9aALtQ0Rdvngv99OBxoB4/PNRkgBOrs7XmS1K3La04a7\n6GQ8s+KLF4D9gH8ArwL3AT+1zy5ezQcNNUUb1oWicRz7YvSujLinFChWqHFrW6/U/mU8YNSU\nlXzaWgFH19AR7f7E5RaFnl1hBR9tOjP8Q5Nxuy6Iqqu1iLDLh+KFoAq4GLgDWKvskndjbaBA\n0JRcBk6D2lxSrAHOA/oBPYD3gGH2D3vSjb7IiJOSWtdlrZCuPDK1bKYXX0uTrXx5X9ryMoVX\nwMa92sgUMUpDo0YMxJawTeppVqKjHg2VkvvdVEzkoi2UFCLs8uREoBXwP8oWuUDUWgrsxxUd\nzY8SYQbQGbgXeBCYCLRzE2qBes6l1zZNYFtwdBSRtJimQf3Dp4o/Y7vaoQzUiKoAYs4Bda9R\nrhm3+AqP7zszStquaFrK8dzmI2cm43JvVBB4RNjlRgG4FPgbsELdWJKWvpA4Xk2KG1WVADJV\na1ohssqJk7Nxu/Gk/QG4EjgQqA+8A4wGGoQM61hYU06q56SOpz+kVIioQRgRw4+/p9iNWruq\n1LP1SC2jCS9bJny2NmwBYVeQDNQfNerUHKVPhCGK0ESq8QUhLCLscuNIoAPw55ob5eavdmJ0\nLBznJOacCVzLCwWVm4F8AAwCzgbOAzbss8+h8ZLUBoTx4YxBbGUCu2P02NTRsEk9X+tQgc4n\nEOoc4LUU1VtMd4wwR9y4K7LQyUaEJdtE4OgJQvaIsMuNy4BHgC/yTkMoETT3Tt2uFqCmTvaT\nSqgWPeBhoD2AQYP+ATwKeIsXp2olFmqugVLfTq1oVFpqWBrc9loLEqjeaNOMTLSVNxYw9sim\nz2wnXgQi3y24R9DO/Jh3PvGxHSxByBERdvnQB+gJ3Jx3GkK+aDOT8e7fpiQooWYXP5TRLHSp\nGCqHVQDuuqvOnDn7AGjf/opCoaFDdUZvIWiOLzj8epgtvqqubP0NNVbF9Fz8RVscJg0trPHe\nIDPlofqUgYXDqjqXMvSeh3ErEyFUSoKQASLs8uFS4Dngw7zTEEoNZgIIVCoRpkk/bFITHp8/\nunXrBpy1bt0FwLvAsWGqUweOqcUvkPGdpb5Xwb7q6hczijajTlV1j8uwa5YtPViqboisHSPA\n6JX4OobXsv4LmwepvhA7TahtiLDLg4ULjwRurLkty7tqoUQIPOhUGTBVmO3GWlQchJqPo52u\n1cB9wL7As8ATwCRgvxjRoqFJsbAVwVp9MA24+i+1lGh8bSPYDz0YM3HvC986D38qJmtTOaan\npVQKF1W5tgsZI8IuD2688VVgXs1t4tULRj+GL6lutIkVm2iLMItrAiVUdbXKauBiYH9gI/A2\ncAewk/NHBDTBRNNwnES1FTr1daA4MLbrh/XFn1qGN7eMMsuxF/wJYzv68aUGc7KFisOrZEEQ\nwiLCLnM+/hhPP31T3lkIJYhtTc0mp3it5j6/umuICNM2w4fAYGAoMBD4CLjY/pUoWhrG18aN\nvGJTqzDLdupBKaI14Tl8CMMG3es4yL6TV1BgIhiFpm2smEYd7xMc708c23UMKAgCRNjlwL33\nomvXKXlnIZQ+/DRG1UYmSZkJ27qa8MvAfwFXABcD7wEnAlTUaPosWlaBgonRZ9p2NbItpvG4\n0I5E08qhlJ979chnEZWJ6i4m28TP22TvPQShHBFhlznnn99m9my6WW5GBZs+o15LBGcoVQIF\nE199M3AXsDfwNPAQMB04xB6/4Pxpj+JAGRdSjQvZjmqJCqaCw3oo7Yiajy0rtQDNkHqH2guQ\nGwCtOh0EW5JqFRfB6kj5irDyzVyoeETYZc6uu34hGk6I96y6NnNnMMdEUI3q8rGLiloN/AHo\nACwBZtep8xSwryUy02WbWFH3GhWSS0e0mNrhiOZXqeLMD6vpJGPCxqw0TzHQKivUhK8SByas\ntitmArZDnCyi6oRSRoRdDtCLgth1AoM/ixvtGR+jO5UUMZcL3RfjPgWGAYdUV+8AvAOMB3Zx\niK+KSH6d0WUV0sUR9O1Ao9qmGfJB/MKhFCdTjBdSaZhqtkFTD42tontb2dzGBCJXbKGUEWFX\nEpTI1UrIksDFu8CJUF0Xc599I6TqEjZQCdkwZj4XGAgcBXQD/g3cCDTf1lDkDPl2qQzl10/9\nXard5Zk+DUCVnzZWBft3F1O70ZiPlrzq2xnNXT9neuBoFVvf3WEyt42VmnCEy6PtbHQM5VJM\ntJ1QsoiwE4TygM7Q/PRjm9gcS4Yiwuzr0uhEoAtwNnAS8O32218KNHb+0hOE0Shhh9G4S5VN\nqKkLNd1Gh8vXWI4jabT3bOlR6a8J0PjqLZFbU20A02vXWKsC7q6jiWCh8hBhVxJEdjuECoYx\nb4wFUFNDFCwaKDBINjhOP9XAo0DbH37AtdeeD3wCjAS89evDtuIrBvf+2kwmf6/NGAtMRmvC\nVsAPa5ODajHe+oIi5oy2orEX2ugZe6SmxBez7aIb1ZGnhqIxTqhGA4lwQS4dRSVTiVA37wQE\nIOWno4QyRTsrikrCZkFpOkNzg4yvja0YI+ROoUEDAI2Ac4HLgcWNGt0A1Ac2FvcWCoEJ+wVc\n+sVYgL7aqJGeslFLxn1s6eEz9ssYPJTF5S7uXfrLFHNspWD5EkHbIYum80AOE31/uYdKlqSu\n/zKJCEXEsROEEsV9AqbreqEaMqo996zSwNjKemCM5+0FjAP+BHwAnAnUA2D/RGpgTCOMvcQH\nCbWSGKjwaHkQma4ees2TC1yYczzuvCCjvQjrXXn2D7EGHjL+RGWq82+lUMdRbS7C7VBJ3UEJ\nlYEIO0EoRQLND8cFuESaU8lgHmImyEKhsBa4AWgLPALcBnwInAPUD4qZ1EpZoIfk0rS6Hhrn\nIPLSny7japqP5mncRQWiMUNeRdnUm9GBps1pxdSjEOE2htp1Wj68RAsrl3nEZhMSR4RdSSDv\nbaFIoMtixLiSFbbdyHUZmDW+mMmsAkYDewAPAjcBHwPnAQ2jZRkedfFRHXzeJSqi6hLNbPML\nqK/VmMYXtEWbdmHkcsHyHcsuPXIpY2vUjxC2OsKf9kYZHS0sc2JH6Itc/4VkEWEnCCVENImm\nCgWbOQRlYnNfooo55TBKIk5Yn2+Bq4E9gXuAa4CPgd8DP3GoGG0C9iuq3hJTEooy8/81VuSt\nLDWaVkw94sZkqH7irTvHkWGK2XYxVULlYFS6oXDvY6jtglAiiLDLH7lMCCr++eCufqgIoDO3\n5/Z1/FRw5Hh+qk0zaawGrgf2AMYClwP/Aa7Y9r13YD0ttUAc00gNayxmG3mmUc1CUx04Rnyr\nPiIjDcMuJkYeHzV5W8yU3GJ3NHGcRj6BmjsXch95ISVE2AlC6RLo66i6DWQCdjRINErnTkNb\nlPRfG3XGOuBmYA/gOuAc4DPgRsBbutRRzqrWmqNVo0lnaozRzLX8w86smjWrtWtrjqbEBw+V\nEg0SNoLx1I0crQSJMLAZ9LoCBlawIcIua4xzcy6ZCKWPcapzLEzX4IpzZwTTzj3hbGDm+/XA\n/3heO+D3wKXt269v2fIOYE/T02lqNDUsVZCah6oqaU298fqJ14W2njIBtWg2P49x+NQOGqsn\nBR82kUYjG2/qEMX0Jm3xYeqjra1EFLZjVkJFIsIua4xv71wyEUocfiZQTSZaN3BWMNbK8lRM\nta1NwN+Bqg8+OBU4BPgIeKJQOJhtOlAPaRilg4tKYxpVt2hqkslKVZOORqCv56iW1RrVCgQK\nJibDCESTWfElUShdWKiJMYLn9iCEXzgNfWlsKO0mhFwQYZc1Jeh/CJWKZhVoEw8tH+fkDDtJ\nZPBGqAaeAboChwNNgLnAVAAvvgiLFFa7oM7TaqpUTAfOwUyBQEUVaNmqh1jLWUubarVQh8B2\nwthaCRU5zsng101JDDmml6rfGUjMMRQqDBF2glCW8NOYtr4W+bpfjvf0tKevAUOA/YFFwMaj\njkKnTmeZvhtFlT5Gw8zfQtUMb1NpQYyHI+Z6qHbEteaMWdlOIdXM00xixiG2mVVGklIhVHnD\nMgKhWgylEY0jTyNkY8Jlg+jIEkeEnSDUIrS5J7C8++U7beMkPu8Cv/a8+kuW4KijbgE+B64D\nWlkKa2Io5oKgTQxpW7yaPyZmLAPi0hkj8NXVHrks/NnKGOULE6eIZhuHHV6qKpjE4kuQkj2f\nVcoiSSEz5LdiBaECUWd3H8/+G7K2MtGaLuU5prDLLgC2A04Dfg9cAjwJ/BWYx1RxGBC+jE1+\n+f/SMloBozWlFTZ6hLxGNNYN7EWgCuTjaHlGO1vCOnARmnDMoZTP9pSohV0uL8SxE4RKwOZM\nuDgW2nJb8smVDH7vvgPuBjoAxwAtgDeB6cBJ2355NqVGmS0+/PKov0gK08JoHHfKtjRMBaiW\nHnPihU1AhSmpLn06eo2hMolPsiuVsu4phEWEXc7IrY+QNppoo1OdbdqIfHLm6KaEiuYBLwM/\nAzoB7wL3A/8BRgM7J5hQYA41PwZrnMVVucMszqr/amFpKCaapq7U17bh1TRW/NVPY0p8meyR\nBVChNBFhlydyURCSgj5U5JEHtqh0MHowZe3eRcv5Xc8bDuwK/Bk4DfgceBjoHjKIcTBpAduW\nCAu+qkqzKTmXFhmbkEnPb9fWr1BWcShd6Ghiufh/aZCs4BP5KIRFhJ0gVA7U1WDmS17VGf9N\ng/jzblJJrgL+AuwL1J8wYXtgJvAWcBawXZhMGHNLW9b0lO82s62E0n9tnVXVvOa3qS3CQSj4\npw2t67haqiVGk2GaNpaMJm4CpbYgVCTy4QlBqByKk7HLTObiqZTIjBjW1nIJRa0v//WWbYJm\nb+C3wE3AGOB/gfHAe2xwY0yPfBjFI99Va1sPVY+RUajR6pp2tJ0Mfm78RrUJ3m50xxhKU5Ba\nZ+M0zZR0fKdUALWnp0IRcewEoaKwXcGNa14UbSU38YelIhDoEoV1j5jyqqr4CLgI2BW4AOgG\nvAtMBU4GvA0bQvmaRv2k7jVG85RvkguMaYxm04vMoipqKkhVU/pnhebemfscxiDkexEH2hdt\nV/artNlTe3oq+IiwE4TaiCYatEVbf6NWy90L9FtR48fKOCqhmqbz3wbgQaAHcCDwATAewK67\n/rlQ2NdSi1+B5Rt1sVHVYja9EtiQuiWwiruYC8R2LFQxqg1mqGNn7KAtE1gOVtj7hGRLJo7x\nrS1UNiLsBKG2YLy+axu1KZZ3m4y4G0XZELZp2xS4ECh+wOKcFSv6AR8A0wDvoYe0X7AIpbSY\nAoy+oT2iB1EVLrQwanbT0V2jKXnKI32+MxRoi/KR6b/RMA4gL8EFoTIQYScItRHbFJ7GFFtJ\nrAXuBboCnYF3gdWnnroEuB04oGYxd9FgNDgpgaYgrWtTdeoCJX1BZZlNqdtSDYtm64a1l/xs\nGRUb2B21fKhj51JYTU8QMkCEnSAIESmLuSqCUHAp9jZwLrALcBFwEPA28CYwHGi2LYit0bAL\n3Ixoo3VtoSI4ptQmNL6mCURb+0vjXFJVnSof3SMkol9d7qDKkRxteIFHhF1ulMWkKNQ2bMu1\n2osijHzJDF5DeEl8BIThO+ABoA/QAZgCXA18BTwEDADquH3UgxFtDPxSLGr6WIHCwiMf1FC3\nqEu32r804QhDzSQftrr2mqbneLpGWE0OTNLYSlLxBUFFhF1uyLtaKAXoehzIQ1fGMrYtSRFq\nDmb2xozvWPcD4BKgDfALoDnwCvAJ4F199R72CFQhaTnb9IrRPLMtmKrrgLwmo42G2uUF/aAt\nT0r6Ww0bwbNMPJ8s46dNBF9WyAYRdoJQ27HNfPletdO+83GMb/NajNs3Af8POArYDbgLeO/q\nqxcBrwGnAj8hEXwZZMsklIDzpZVNfPsJq8kXlF8l8cgXJof1O21SNSxqMtTgdM8n8inEyO6k\nWhRJJKSHCLvckDe2UCJoTgydUG2rXagFp7FtHPgp/CtgDLAf0BP4EPgfYClwP9AP8Ktpw2gb\nSU1h2GwzzZrytqHt9f+6CJdo66phq/gVYy7p2mKqJ5xQKgAAIABJREFUxI8ZSOksxZROJkLG\niLAThFoKfayKL+m/1na5tJXGnBpHQ0SL424X+THf8LzfAa2A3wItgcnAIuBaYG9i+xXI1wrS\nmEbpo6oW/4XNujOKcqOOd/ER1Y2acIypKrTuaLnFiVwkEdFjDFJStzolpe1SWm0XKCLsBEGo\ngS8U6IU4whKVrzMSz9PWFkIaUYEL0BGmak0HbwAeAwYDuwJjgaHAv4GZwLmFwvZKMZdRctRb\nxmH35Roj/oxNaN3RPF261xiTX1SltYxnIGqKyGjnlftoBwbRYGzUXCgplSlkhgi7fJD3m5A7\ndKnONg8ZV7Icfa/0LL1A/eHeqG2NlfeuwuJ53lLgL8CBwIHAHGA08BXw/wDvqae89etpR1BT\nS2mJ+QVs2pT6c6qlR9ti0Fq3bee1NT3HNIEYX2y5RIi2MqulWvrX8FLLMLPVcEGEnSAIZtIw\n22Lqv2RbD5QUoaQGP29puxYC/w20AY4G1gDrTjwRrVp5Z53Vb9tF2W/XqKKYxJiEbdqdGrFx\nnLD4Jam3p/5Ll6QzdsWMh4Yplhm5G4RCiSDCThCEGo+Za9vVfws10XYlkkb8IAyBScZJoDgm\nTATa+hZgInAa0BI4ddUqLF48tarqM+BW4MCgBVObJWas4lm+JEVrwihZNLVne/pNLc8IL9s5\nplmJfluhTNNonlDY4DbD0uibikclZI8Iu6wRO1oofVTpxsgL20ofc5KHsmqyx0X5peHTfAc8\nDBQmTsSSJbcAvYEFADp2/CPQVmlaa8gov4wJu2QVwQJEzQEJvLJ5ymNzEVZCjU37HYzWR02Z\nuYs8LYKmjGmxDJDJRSgiwk4QBFfUKZxZvCvlJaGYM59vy8XXJeZiLVuOBXoA+OQT/OIXvyh+\n0fEhh3i33YYlS5iYxviatxpoyqoFAgsbXVtbN40jZtzCKFS+CUeMEejZG8HGcz8rSvkNIlQA\nIuwEoRbhOKNQc0h74U6puQiJTKiBw8h02cVmA4C2bQujR3fwPCxYgP79cfvt1bvuOhX4HeAt\nX24TSYx5ZmxLXT62ibZAF9aI0fGlbRlj2gQlM+a8SaytBauh/HzcTwzRZEKJUzfvBARByA6v\n5u8+uUzVWskCeYzMU75Ew+iFlJSwi4xt7g+lCVxVnaoeDjyw0LlzATgEGAbc1bo1WrX6x+bN\nvwGeA1ZaMuFNKTUB7djRIF7NL0nh83c51i4d15xgKj0R6dSiglJVeMWA/FsjEVWXxjsi2gJ3\nJSUg+IhjJwi1DkfTTl1jQk13p6AA1ncJpXsSJHGnkLeLHAur7pE6nkyjW0cYmA1cCOCLL/DK\nK58CfwKWAi8BZwDNg0xEqslsnhljbtHkmWVNrQnN8+PrwuEUVe9PAvvOSFJmKGzRAsuUAoGm\nslDBiLAThFpN4ARAC9jmY0cHJZupMcFZzVF7Rasemjp1CgMH/g5oDRwJfAE8sNNOy4AXgdOB\nn9oVreNiJVOdFva3RFjQNGJUmY63DUzTjllRSaruUofIJVrG8HcyGei8xG+lhMiIsBOE2kiE\nSzC13+gWl8kvMyOhLBwLmiQzdOquLcAkz/stUG/58voTJw4566wxwFIARx6J++7bgXxoFMR7\nM46PZsup7WpGl1ZLK2Oz9Hyfkheggecn0wVa0r18lidngm0Zz6JSkFliHOaCCDtBqF34V3z/\nLz8BaAW0Wkanx99FL+sRLL0SmaICYfywaOVtvpG2fTOAn/2s8Le/tQKGAOMmTlx21llLgVcL\nBYwfj2XLNFXH5M94dcYZWjuXws7ivLJk7DotN14LMno0VTKzco0OOn3f5dL3snjnVh4i7ARB\nCIfqyhh38VJG038U6vHESDYB3BfyArvmQ71Ppl1qj2kVi1s2A5OA3wG7AIcDHwFfDh9e3bLl\n68BIYPeardMmjKoRlk+VarvUMmrXqJ7Qbgk0E5EOgg1NU7oUjiMyYlbXfFDjOEcOW+LiqcTT\nq0hE2AmC8CN0BnKpFWjYFOdF9+UzmlXYKrmgDh3sU5pHnkVjXDpmXrRJri3AFOA8oA3QG5gD\nnA98CngHH3wZ0IFdY7U5rKoW1KQbH0Hb6J8DNrGlKjZG+jPnEn+a0ZFnBj8OWZpkqZrcEd6z\n5eKyVyrydSeCIESE13OqFIgwMZSmmAuVmKO24HeF0taaY1oNzNq2pTNw7FtvnQL8CfgQeA64\ndPZseF6hTh2YHpdk0jCKP03beeTXIDRvSdO+jGXoiLGu+0a1L4ENOZY0hk1Q8ZSseIowREKC\niGMnCMKPqLfaEW676TodM4nyu0K1mxkREmPG0OZX8cMeeFCMBRYAVwL/BewN3Af0BdCz55I6\nde4CDgewcSOVa1Tq0fSoeRaoDvkOuoywp6C2rukJdaNW1zEZmlvJnpkxYbom9lvZIcJOEIRY\nMDZMICmtgqkkHjlswLD+XKB6YNYfQWZoTVt7nveR590K9AKwePG1wJ7ACwBatPBOPvlkoJml\nIZfZ3W/L2/Zlv0xhW0BtIx0NXoLQf42t0CFiUqWt8F2Lpv8qQDWqb2fRgjkiS7GCIMRFWwFU\nNzJVjOUTn9vclzKZkp7b96VFiJxsEE2KqTlrq4GFQqHQujWAe4AmwJGrVz9RVXU38BNgalHq\nff65Wj5Chi7ng5ZwYCvGdkGEGoJUnftBCYxWptBTIn7AOBW9oN/8ENwRx04QhGQwLiOqLo5a\nDJav88gmVYqjquMJ1C5qyVATWCjzzFZA9YTUAmuBJwE8/PBOwM+Bw0eMuGP33bH77vMB78or\nvTffBHlc0nigtbBeTeB2fDXjijbEhAqlfekJaSOs1IjsVyWirtwHIdm3m3uvjUmWu1tZUoiw\nEwQheYwraLRYXvNfSrhPTlS+hGqFF4v8xGlLslAobAZeBTB2LP7znwOB5wBMnOh17bqkTp3x\nhQJeeqmhEoFfd2NUu7pqrJYPHD1efPN1I5R0L5/IKmqWZ7V2Z1XYhkvdxBWYdteXbPDaiSzF\nCoKQIsYJ3rYxwoKj1lYa9/3aUlEahJU4xjIF08dO1SowWTW25hYCC4Hr5s5tBQwFxg0ZsmHI\nkG+AScD/AfjqKxo2ju/lJ+ZZPh4baDK5i4Ow6sF4xpYm6QlclZgDYqxbLiNcFoiwEwQhSWxz\nfOBinLpia3s8i1c8qa7mZLBUZOs7VZYRFo5Vg834RJrtxZfbsvoJcBgwBLgG8Fq3fhM4+Jpr\nMG9enYMPpk3Q+LSntL/MWrbn/AyWYzFHVMVpK1OmosTvVBouppAjshQrCEJc3Ndx6HIhNWaM\nXpTRxakwoi2hUgK9MaaA2px2pL4HngfOAXYFDgFeBvD88+jWbTFwD3A0sB0JQoO7tG57Wivy\ns2vRULN1UZMZSP9oTTDnVYSAaR+CbEayshHHThCEJAn7uBizgkbNJ+pd0TLua6a8iZjI2mt6\nC7ihumkcxkA0B+7Hf4G52w7WLoXCEGAI8AhQBUwFJgATarZui6z5drb1ZVsQG4koj0AfMfvg\nLt6hO5V9j1TLEcdOEIS4RLBSVOvO9vQ0neapFGMsrkCM3oDjsq97E/GD8AQ+W+aSA6NuwZoo\nXwL3Akd73o7AscDHwNg99lgEYL/9cPHFmDoVmzer0YxPzkERefENm0JNogXxthGqvGNuqaZk\nPKXLSMaVV7aliQg7QRDyRBVS/JwX6pnrQLljo0SWgSLM39HWan1FpY1YcQuzqKoWLhQKG4CJ\nwPlA4T//6Qj84b33Xh8zZvOAAdhpJ5x00pmFQkvyJSYuXYg8zUd7eiwbPIVEAvrvHf9vvgua\nspyaLyLsBEFIF+0qr/7LTGxGd4eRAnH8qgThe+ReN9oyq21G12QEtTyLtbRV0bBD58d/H7jZ\n8/oBLQCMH/+/Tz11E/AlgK5dcdVVmDsX1dV+cPpCGwdGXBp3+T11v2egkd0LhyJtOypZOSv6\nrEwRYScIQkmgzXmO+s9GXhMS0y6ziy4x8yWZmKqsCSxsS0Adeeot+YV9OUgbLUZY6Xk46aQz\ngJZAnTlzMGQIXn4ZPXsur6p6CDgF8Fas0JqOLEpcVIj7WeESKpFzLJE4NsM1+0xoPoGRRT4m\njnx4QhCEdKFGFD/laL6RttFWy7aLWU/kc4g82RiNMZ7Akn4+0WKqrz3y0ROXJVewY6K6fcbt\n1f6vx159Nb7+etTOOw8BHt1+e+y8M7p1w+DB3ltv1Tn4YK/mcXc5T2ij/CAUnD984F4yWVz6\nnkYoenyz7H6CvRbEsRMEoRQxXuKpdeTiTNgKpDSL2NZD3aGJhV1JZLpm1Dp+uy7jaVTb6sRM\n3aMaFVu0eNDzTgbqrVyJqVMxcODbo0d7Bx9cvfPOfweGAVi5Uu2I44pqYOZqMoGh4CArXc6f\nwDOhGCc91yqCb5dGDkwaEcxFgUeEnSAI2cHM074+0FwiZk3HUUJFsF7yXRtyaZ32yJcIjhGM\n7arjSRdYtbaM/wYuBfpNbAbQu3fhhhs6A7sAuPHGhsDjzZujRQuvR48/AgcDqK4OTJtpy1MA\nUZzMyZOU1ODjGBNIUOiEjWMsb0wycSUq8i5BRNgJgpAnqhChC3lG6RZqAnBcp8tmUkmwFc3N\nCmyRt0zollCuocvEz8f80vPw618P8zysWIEpU9C//7HAXAAtW+LUU/HII1i+nO9FWDKTEYzA\nVcsYByf7GwzjyZBxDkJMRNgJgpAPmuUDh6XAZG/rNS8wfhB+o9oK4zA5tkXX74r/Gs0zsELQ\nJsv8+NoyawT7kwp0o5EGAFVV6NMHf/rTQZ5XWLoUt96KTZvw+9+jZcu5wHWFAmbNwpYtILon\nggyifSkFIZU20QaKbilHwZeLVs4e+fCEIAi5QbWdca8vWfx/UVOvaFtoQzbPz2N/xMIFl+Zo\nfFtDfAIu6WkD5ZHvQ4FDTz3LxxeYubxg+vyES8I2ee15HnbeuXDGGQCqgG7AEcCV3bqhT59v\nq6t/euKJvwEmBkavmRtqnjy0mFY4VWjH3UfbEfruyIXsrfFajgg7QRDywTbfRJ5ZNT/Mpips\nsibmrXygWqIyK05kx4VRd5mrjZ7LIWCyUvWEdkBdliY1tgBvALOKFVesOG+nnX721FN/a9kS\nS5di//3xs5/hiCO89evRsCGTqvuytS23VEWJY3D3TFS3VWtIu1MqQVIa8FLucoLIUqwgCCUH\n/3iW9i+dopgFF15aaYuYPHGeRmJWskLNPWkvTNt0oXFdVStjXmk1oUZTl2jVf38MsuOOjwGn\ne15h6dLOwB/eeQfz52PIEOywAwYPxu2348MPbZkYUw0kKR+XwXGIAsMaRzL3ZdNSyKFWIY6d\nIAj5YPTqbLsCg2gWEY1jMzDUUJpYCbsw6jLvaq9pKxHCalAHTl1kNC5b08TUXUbN51txNvtH\nM10CPRitpLaRliw2vaBYYN06TJmCV1/9ZOTIvQDsvvu9n332D2CyqQuRYeIY7y5s+UeAGp9x\nWslFY4Va6U6qp7UTcewEQSg5HBWV0cNgVJ1NJWgqh8Ho0rlXd4zpDtNZfzt9wgxRnyI3ruGC\nyESwgsy43X0Q/MxrJNO4MY46qjB2bDugHYBLLmkB3A98W7fuTOAqAG+8UfzIRQQCrcfAe4ZE\ncNRDJSt9SjaxykMcO0EQSgUXheR+30/FnMsibOAqpGf6OIIWyubG0Vqh1EBgQM/yWZCw1iPv\nVmojbzwQhaDPUqgDTgPa/EImZyjnT+G88zzPw6ZNmD17at++gwH07o2mTTFoEA4/HIcdhj33\ntPXOBu81FtyeI4TptIxDemopMM8IHYmTrehCd8SxEwShtOBdB/V5Ha1k4DIfLf+jFLCshxqX\nhGwZMuubfHruZQItGU2h8i6aLZr2nFaE1pktxobcUXMIGK569dCnz+We183zsGwZxo1Ds2a4\n9lq0bftRoYARI/DCC1izxiWHQIMzcdkRzVItd2pnrxNHhJ0gCJWDTabYDCR1IqGTik01ak+V\nuU/qji4jUzJw5tP0HG9Y+h0JNGaYvT5aEE0dqoocit9Da7mMLZOzWV7vuCOGDcN992Hx4k7A\nnQAWLcIpp2xu1mw6cGWhgNmzI6/VOuIy2iql4FHxxm0pZChQRNgJglCuGFWOpgkY3aYWY4SU\nzRekzpyL2RDWZVT/Nco1F0tMHRNNYNkyV0fP0UexGZ+agOYHnI+shWIS0PJXi70L3A7gpZca\nfv993ddemwFc26ULevXCjjviuONw9934+GMtcvYiJnfNFHiY+AMRudFoLQoq5f2M3caNGxcu\nXLhu3bo99thjz/DPTAiCkDvU6VF3FV8EaheqgZjynumJqILpq1yNsiAORs8ssKRn+pEJWwRq\nTKKm9KSDo7UV1IkaUJnLBHdpxeXYuWdFj7XKD0Bh4MCte4s/Zfbqq5+fe+5uAPbcE4cdhsMO\nw4AB2GEHvyNx9FbuWs0dvrOOnnEZ9bfCKBvH7vrrr58yZYq6Zfz48S1btuzWrdvAgQPbtm17\n8MEHv/3223mlJwhCZGzzRKBn4FKAQhvVBJyvohzNLRfcbSr36dBY0t1do9F4fRzWmNQiqyNv\n9PZsbpx/LBg565iJmg+0Lu+4I048EffcszuwD4D//m8sXYrf/AYtWqBrV1x22SDA+vXHFYfj\niWosFsGRFZKlbBy70aNH/+EPfxgwYEDx3wkTJgwfPrxBgwbHHntsixYt3nnnnZkzZ/bv33/e\nvHl77bVXvqkKghCKyP6N++zOOBCOMiWsnvOdM6OFxsCXpLYiU9e9v3xJL+gDtozn55FvB6TR\nGH2gNeEbq5pAT1BM/BhkxAhs3ow5czBpEiZNmlSvHurWxc9+hkGDcNhhOOAA1MnHGYnvHboQ\n+V3pUkBIlbJx7DQuvPDCZs2aLViw4Nlnnx03btyMGTOeeeaZNWvW3HDDDXmnJghCCBKcA+h6\nZaAKsaXkq7GwK4laK3QV1R3N9FLDqvFtxgnIo2nGJnylpZYpvlbVkupyGZtQy8BNfDguAdsM\nV6PnxzQXhbp10asXrroK06dj5Uo88QQ6dMCDD6JLF+y8M046Cffei0WLEm6UJZRvWmEwp7qg\nUpbCbvny5R999NF5553XoUMHf+Nxxx139NFHv/LKKzkmJghClrhc6FUBRPUKLLrB2Jb2IjCN\nyKIQxCSLplCNjTIdMUJXSF10pJakOry2tDWV6ZIb2MEPDOLY0NZijRvjqKPw17/inXeweDFu\nuw2NGuHaa7HXXmjbFmefjSeewNdfO6YdGf5cDdEdoXIpS2G3YcMGAKqqK9KpU6ev039fCYJQ\nmmizHXW56C4NVZeonpPmZtEWVfmitUItLj4HOKz8qu06LszRAtRsc9G1xu2qn+fSQYRUw8a6\nWnV3la8lT8sE59C6NX71K/zv/+KLL/D++xg1CitWYPhwtGyJAw/EqFF46SWsWxfYkWShqjrV\ntmhDqUpG0aPulM0zdiqtW7du1qzZ4sWLte1ffvllkyZNcklJEIQSxFOex1KVGS9T4DA72koW\nLI9/UYwio1Dzly08049YGCUao2k0hapF08LSrByHQlMVjhrRNlaObqIP3wutjJaMizLm9rZv\nj/btCyNGeJs34623MHkyXnsNd92FLVvQrRsGDcKgQejeHfXrh+lQFDz2I8BqsbQzSYnyzTxj\nysmx+/zzz996662PP/7422+/Pffcc++7777vv//e3/vBBx888cQTvXr1yjFDQRBKDffJgFoC\nqs5w8Z8cHQU+rKZRjJEZ98Klvy5ajaphTboxQTTpZvMskZANY3NkqaXnEk1L2HFxEwCqqnDI\nIbj8ckyahJUr8fLL6NcPEydi4EBsvz2OOAI334y33or8NcguY5WZ9DGeui6maZwWU4pceZTN\nN80YT+inn376+OOPB/Doo4+ec84569evnz17dteuXZNtevz48cOHD1+7dm3jxo2TjSwIQja4\nuDJMGWr7MT5cWIXn6MmptYxShiZpbMiWhrEvTItMnowNpm2nDpMagdqi/LDT+LSAMQ0eFxuM\nK7NmDaZO3erkvfsumjdHv34YOBADB2K//VwScMk5VI/ypYxSZdi4cWODBg1mzpzZs2fPvHPR\nKZul2AceeGCVwurVq1etWvXTn/60uHfVqlXNmzd//PHHE1d1giCUI4FLe1Qr2JSB71S52HL8\nEqSj4NM0DZU+HvkOEXWXbZnYV0vU/zMmr6orXneqcUKtadJuaiPvOGh8MfUIBoayBQ+7Vvtj\nlaZNMXQohg4FgK+/3qrwbr8dF1yAnXfGgAEYOBADBqBdu8A0IiTPUxkaS9AoG8eOZ926dT/5\nyU/qpPOtQuLYCULZ4fKwl3G2Nk51gT6TsaLRJaJZBT4fxqgWz/4Qnq05NVVVOQVKtwjCzqaH\n6HgGRnbxzJgIcXA5l6JU+ewzTJmC117Da69hyRLsttuPIq9Nm3gpuyLCLjLi2KWOSC5BEFR8\nmcJoAuN2l0nORSw6umIguk2Tjy5CU42jblQjU2moPTNni++/DhwZm4oNJFTkOGWiESGyU5Xd\nd8cZZ+CMMwDgww+3iryLL8bXX6NdO/Tvj4ED0b8/WrUy1k5Ek4mkq0gqRNgJgiBopDdp2XSV\nVkDbG0Fiap8z8Etq2pGuOGtLqHRN1pakLf9AAsfEZXuERiM/OZcxAbntuy/23RfDh8Pz8O67\nmDLluQsuOPbZZ7FyJTp0QP/+GDAA/fqhRYtMkxbKkwpZigXwySef/Pa3vwUwadKkULXat2+/\nefPmwJLr1q3bbrvtoucnCEKlwH8yAPZPD0RYgrSVt5l8WuuBth/TOp9S4qLK2Fk+YIQq7tGS\nJWzwQqFQB9gyfz6mTMGUKZg+HWvWYL/9tiq8fv2w445p5Ck4UspLsZUj7N5+++3OnTsj/Nty\n4cKFvLD75z//eeaZZ/7www/10/8iIkEQShaXjyUWCdRSFPqsm++rBXqB2i7jw3NG9cOYiPwu\nLe1EtJRNpTHDHphkNM0apgeJQQ90jUy2bEFR5E2bhhkzsHYtOnXCgAHo3x99+2KHHXLIuHZT\nysKucpZi27dv/69//StCxQMOOIAv8MMPP0TKSBCEsiGpSV1dFS2ugWoPvcFklamrpXT51fbc\nHi0PRQlpyRiDR5ZNtC/xMT4yyMdPVoTlbnOoA64nU1WFrl3RtSsuuQSbN2PePEydiqlTcf/9\n+P57dOqE/v1F5AlFKkfYNWzYsFOnTnlnIQhCxcJP/I6ygEorrbr2LJ0qv/ySNrlj3EVbYT5R\nQXvEyES1jG1hOuyg0VA56q0sPbwQrdSti0MOwSGH4A9/QPHnLqZN2yryvvtuq8jr1w99+2Kn\nndJMWShRyk/YeZ736aefLlq0aO3atQCaNWu29957t8nqw+GCIFQkqX7Sgppb/ENyai2qoihU\nz2kBNXPOaguxMRn1FliMga7GGvvOVM/dacuTunXRvTu6d98q8ubPx9SpmDYNf/871q1Dx44/\nirydd847VyEjyknYffvttzfccMNDDz309ddfa7t22223s84666KLLmrUqFEuuQmCUPE4+jdG\nrVN8oT6R5rHfRcJ/LkHVbZoG0p5XC/zwgaMqsrl3NKAf1vh8YSKup7G5mKESySdn6tZFt27o\n1m3rcu2CBZg2DdOm4eyzsXo1OnRA377o2xf9+6N167xzFVKkbO51vvrqq169en366ad77713\nr169dt999+JnVNesWfPJJ59Mmzbtyy+/POCAA6ZMmeL/HEVSzJo1q1evXvLhCUGo5bh8eII+\n4qY6drbPb9o+aurvjaAmYVJj1B5TSzLBmacDjQKOxsz3owmlmUlGbNmChQvx+uuYOhUzZuCb\nb9CuHfr23erk7bFH3vmVJfLhiQQYPXr04sWLn3zyyRNPPJHu3bJly/jx40eMGHHNNdf89a9/\nzT49QRAqHncpYHTXtI1GXWXUdjEliE0vun/6gVm6NW5Jz0IrNQItzJKgqgoHHYSDDsLIkaiu\nxrvvYupUvP46LrkEy5ahTRv0748+fdCnD9q3zztXIQHKxrFr1arV4MGD77vvPqbMySefPGvW\nrM8//zzZpsWxEwQhDu4WES0ZzV5iBIe7URcngezJJc/yEHYMH3yA11/fauYtWYIWLbYu1/bt\ni/33Rzq/0lkZlLJjVzaH7Ztvvtlrr734Mh06dFi2bFk2+QiCICRO4FNxYaMx29Xl1AS/tYQn\nsK3MMuFzcEzD20baKaVF+/Y45xw8/DAWL8aiRbjlFjRtirFjceCB2GEH/PznuPlmvPEGNm3K\nO1EhBGWzFNu6deuFCxfyZRYsWNBangkVBKHEYCb+QJ/J8em6wCf/tIDRlpXjEPlJQUeimZpl\nrMkSZ889seeeOP10APjqq63fhPzww7jsMjRqhO7dty7Xdu8O+RGm0qZsHLtjjjnmqaeeGjNm\njPHrgr/77rurrrrq+eefHzZsWPa5CYIg5Av9uhNVwBW2QWtpn2BNyb3TPi3LfA+fsWIGaJ9Q\ndhd86oiVgt2YDK1a4eSTcccd+Ne/sGIFHnsMnTtj4kQccQSaN0f37rj4YrzwAr75Ju9EBQNl\n84zdqlWrBg0aNH/+/CZNmnTr1q1NmzaNGzf2PG/dunWfffbZ3Llzv//++z59+rz00kuNGzdO\ntml5xk4QhDRIyjRyMe2KBH5S1Rgqke+KK5i+aTmwimPJmNg+OOxYV/uKmXRyLA2++w6zZ2P6\ndLz+OubMwfr16NABffqgd2/06YPdd887v+wo5WfsymYptnnz5m+88cadd9754IMPTp06dcuW\nLf6uevXqdenS5cwzzzzzzDOrqqpyTFIQhNpJGpO6e8zAL4eLE4r5rrhQ8OvRxr22D9giac1n\n/BRz2LoVLumKbLcdBg3CoEEAsGkT5s3DzJl4/XWMHIlvvkGbNujbF716oU8fdOwon73Ii7Jx\n7FQ2bNjwxRdfFH95omnTprvttluqXpo4doIg8GQs7FKy+mxh46i6wFSjSbRaYY+VEZ6H99/H\n9OmYMQPTp+Ozz9C8OXr1Qu/e6N0bXbuiQYOKjINvAAAY1klEQVS8U0wYcewSpmHDhnvvvXfe\nWQiCIGwlDYXBf6JCXdOMb6fx6ipQ+cUhWjSRdKVFoYCOHdGxI377WwBYvBgzZmDmTDz2GK64\nAvXqoWtX9O6NXr3Qsye23z7vdCucshR2giAIpUwiD6X5ocB+M3CCnyTNWJ4KFcuuu+Lkk3Hy\nyQCwejVmzcLMmZg+Hbffjg0b0LHjVpHXqxfats071wpEhJ0gCEKSJPVQmiPMh0xDBUmwmCD8\nSLNmOPJIHHkkAGzahLfewqxZmD4do0Zh+XK0arV1xbZnT3TujLqiSRJABlEQBCFJElFaarRQ\nZeThM6F0qVcPPXqgRw+MGgUAH364VeSNG4eRI7HddujWbavI69EDzZrlnW65IsJOEAQhYSpM\nV9m+A4VuzCCNjBsVUmTffbHvvvj1rwFg+fKtK7aTJ+OWW7BpE/bbb+szebJiGxIRdoIgCJVD\nBNGT6qpxgsGZbz8Ryp6ddsLRR+PoowFgwwbMm7dV5xVXbFu2RM+e6N0bPXrgoIMg31DBIsJO\nEASh9uLyRCD/UVmXJhLUdonEEUqahg23frTi4osB4MMP8cYbmDkT996LUaPQoAEOPhg9eqBX\nL/TogRYt8k635BBhJwiCUHvJQCqJGhNiUVyxPeMMAFi7FnPmYMYMzJuHe+/FqlVo1WrrZ2y7\ndMEhh6BevZyzLQFE2AmCIAhxCfxaFkFIgCZNcOihOPRQANiyBR98sPXXL+65B++/j+22wwEH\n/PiFeTvskHe6+SDCThAEQRCEcqOqCvvth/32w2mnAcDSpXjzza06r/iFeW3bbv0ulV690LEj\nkvuseokjwk4QBEFwhfld18h1BSEBWrbEUUfhqKOAbR+/eOMNzJqFq67C0qXYaSf06IGePTFs\nGPbYI+dUU0aEnSAIguBE/O/nE20nZIH/8YsiixZh1iy88QYefxyFAi65JNfkUkeEnSAIguBE\nTE0mkk7Ih7Zt0bYtfvWrvPPIiDp5JyAIgiAIgiAkgwg7QRAEQRCECkGEnSD8//buNSqK847j\n+LOyLIoiF7kYECFUgheisFATLFYFW5GkQiUxaCiWSlW8ovESY70Hg0dTLxHtMVWJ2BLk1ECP\nl0ObY0qlUUkgUlGMctXUpSKKskVEYPtiku12uYgJumH2+3nlPvPMzH8Gz/DjeWZmAQCQCYId\nAKD7KRSK7/6wBYDHRbADAACQCZ6KBQB0P56BBUyCETsAAACZINgBAADIBMEOAABAJgh2AAAA\nMkGwAwAAkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGw\nAwAAkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMqE0dQE9\ngEqlEkJYWVmZuhAAAPB9IcWD7xuFTqczdQ09QFFRUXNzs6mr6B4fffTR+++//+6775q6EEBc\nuXJl06ZNH3zwQa9ezB7AxOrr6+fNm3fkyBEvLy9T14IeQKlUjho1ytRVtINgZ3YOHDiQlJRU\nVlZm6kIAkZeXN3bs2IcPHyqVzB7AxGpqapydnYuLi0eMGGHqWoBvj7+SAQAAZIJgBwAAIBME\nOwAAAJkg2AEAAMgEwQ4AAEAmCHYAAAAyQbADAACQCYIdAACATBDsAAAAZIJgZ3ZUKtX38+vt\nYIZUKpWlpaVCoTB1IYCQ/ityeURPx1eKmZ2HDx9WV1e7u7ubuhBA6HS6iooKvpoT3xPl5eX8\nb0RPR7ADAACQCaZiAQAAZIJgBwAAIBMEOwAAAJkg2AEAAMgEwQ4AAEAmCHYAAAAyQbADAACQ\nCYIdAACATBDsAAAAZIJgBwAAIBMEOwAAAJkg2AEAAMgEwQ4AAEAmCHYAAAAyQbADAACQCYKd\nGamrq0tMTPT09FSpVK6urvHx8RqNxtRFwRylpqYq2vP222+bujSYi4cPH65atcrCwiIwMLDt\nUq6W6LmUpi4AT0lTU1NoaGhhYWFUVJRarS4rKzt06NCpU6cKCgrs7e1NXR3MS11dnRBi+vTp\ngwcPNmz/0Y9+ZKKKYF5KSkpiYmKuXr3a7lKulujRCHbmIiUlpbCwcMuWLStWrJBaJk2a9Npr\nryUlJW3bts20tcHcSMFu6dKl7Q6WAE/UvXv3AgICRowYUVhY6Ovr27YDV0v0aAqdTmfqGvA0\n+Pv7l5WV1dTUWFlZ6Ru9vb3v3btXXV2tUChMWBvMTWJi4s6dO69evTpkyBBT1wKzc/v27c2b\nN7/zzjuWlpa9e/f29fX9/PPPDTtwtUSPxj12ZqGxsfHChQujR482vE4JIYKDg2/evFlRUWGq\nwmCepBE7Ozu7lpaWr7766tatW6auCGbEwcFh27ZtlpaW7S7laomejmBnFq5fv97S0uLu7m7U\n7uHhIYQoLy83RVEwX3fv3hVC7Nixw8nJyd3d3cnJycfH549//KOp6wK4WqLH4x47s1BfXy+E\n6Nu3r1F7v3799EuBp0YasUtPT1+xYoWbm1tJSUlKSsrrr79eX18/Z84cU1cHs8bVEj0dwc6M\ntL01RLrDkltG8JStWbNmwYIFYWFh+l+fMTExarX6rbfeiouLU6lUpi0P4GqJnoupWLPQv39/\n0d7fmvfu3RNC2NjYmKAmmLGQkJCoqCjDQZHhw4eHh4ffvn27qKjIhIUBXC3R0xHszMLgwYOV\nSmVVVZVRe1lZmRDC29vbFEUB/8fZ2VkIodVqTV0IzBpXS/R0BDuzoFKpAgIC8vPzGxoa9I2t\nra25ubnu7u5GL4kFniitVrt379709HSj9osXL4pvblEHTIWrJXo6gp25mDVrVkNDw9atW/Ut\n+/btu3HjRnx8vAmrghmytrZOSkqaPXv25cuX9Y3Z2dl5eXn+/v5eXl4mrA0QXC3Rw/GCYnPR\n0tIyYcKE06dPR0REqNXqkpKSjIwMX1/fs2fPWltbm7o6mJc///nPkZGR1tbW0dHRrq6uxcXF\nWVlZNjY2n3zyiVqtNnV1kLnc3NyTJ09K/962bZuTk9PMmTOlj8uXLx8wYABXS/RoBDszotVq\nN2zYkJmZeePGDWdn58jIyI0bNzo4OJi6LpijM2fObNq06cyZM1qt1tnZeeLEiWvWrOGLKPAU\nJCcnr1q1qt1F+m9D4WqJnotgBwAAIBPcYwcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAA\nMkGwAwAAkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGw\nAwAAkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGwAwAA\nkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGwAwAAkAmC\nHQAAgEwQ7AAAAGSCYAeYu+joaIVC8dVXXz3l/S5YsMDKyqqgoOAp7/epkU5sdXW1qQt5qtau\nXatSqXJzc01dCGCmCHYAutmdO3eWLVvm4eFhZWX17LPPRkZGnj171qhPenp6SkrKtm3bAgIC\nTFJkVyQnJ5eWln7r1f38/CZNmmRlZdWNJX13dXV1iYmJnp6eKpXK1dU1Pj5eo9F00j81NVXR\nnrfffrvd/uvWrQsKCpo2bVpNTc2TOQIAnVHodDpT1wDAlKKjozMyMq5fvz5o0KDvvrXbt28H\nBARUVla+9NJLarW6vLw8IyNDqVTm5+c///zzUh+tVuvp6ent7X3mzJnvvscnRKPRuLq6njx5\nMiwszNS1dJumpqagoKDCwsKoqCi1Wl1WVpaWljZo0KCCggJ7e/t2V9mxY8eSJUumT58+ePBg\nw/ZJkyZNmDCh3VVKS0uHDh06c+bM/fv3d/8xAOicDoB5e+2114QQ169f75atzZ8/Xwjx3nvv\n6Vv+9Kc/CSHCw8P1LcnJyUKI48ePd8sen5Ds7GwhxMmTJ01dyKN5eHi88cYbXen529/+Vgix\nZcsWfUtGRoYQopPV161bJ4T47LPPHqukGTNmKJXK8vLyx1oLwHfHVCyA/1NVVRUXF+fm5qZS\nqRwdHadMmZKfn2/Y4fjx46NHj7a2th44cODixYvv37/v7u6uVqulpZaWlqGhoXPmzNH3//nP\nf96nT5+LFy9KH1tbW3fs2DF06NDw8PDOK6muro6Pj3dzc+vbt++oUaN27tzZ3NzcxTpffvll\nhUJRV1enb2lublYoFBMnTpQ+zpgxQ6FQaLXalStXenp6WllZubu7b9++XafTSatHREQIISZP\nnqxQKPLy8tqt8MGDB1u3bh01apStra2Njc3IkSO3bt3a2toqLdXfY1dZWdnubKajo6N+U//+\n97/nz5/v4eGhUqmcnJwiIyM/++yzzs/Pt3Do0CEbG5vFixfrW6ZNmzZkyJC0tDRdB7M30jm0\ns7N7rB0tXbq0ubl5x44d36VaAN+C0tQFAPgeuX79+ujRoxsaGhISEkaMGPGvf/1rz549P/7x\njz/++OPg4GAhxN///veIiAgnJ6c333zT0dExMzMzOjq6vr7ezc1N2sL27duNttnU1NTc3Kyf\n5y0sLKyurp42bVrnldTU1AQGBmq12tjYWA8Pj7/97W+JiYkXLlz4/e9/35U6H0mlUgkhXnnl\nlWefffbDDz9sbW3dsGHD0qVL7ezs4uLifvOb3zg4OKSlpa1du9bf33/48OHtbiQhIeHgwYMz\nZsxISEhQKBQ5OTkrVqyoqqravXu3YTdHR8f333/fsKWoqGj37t1Dhw7VH+wLL7xQV1c3d+5c\nX1/f69ev79mzZ+zYsTk5OePGjevK4XRFY2PjhQsXxo8fb3TbX3BwcGpqakVFhZeXV9u19MGu\npaVFo9H07t3bMI92RK1WOzk5nThxYufOnd1VP4AuMfWQIQATM5yKnTlzphDi6NGj+qWXLl2y\nsLB48cUXpY8/+clPhMHEXHNzs3Sj1QsvvNDR9qVf7frJ2XfeeUcIkZWV1XlVCQkJQoicnBx9\ny0svvSSEKC4u7kqdUuc7d+7oOzx8+FAIERoaKn2cNWuWEGL69On6DmVlZUKIl19+2bDOzqdi\nra2tg4KCDFuWLFkSFRXV3Nys++bEajQao7Vu377t5eXl6OhYVVWlP1ilUmk43Xnt2jUbG5vA\nwMCOdn3VgJub26xZs/Qf2+5RcuXKFSHEL3/5S6N2abL1r3/9a7trRUZGCiFWr16tvwnvueee\n+8Mf/tBRYXrS4VdUVDyyJ4BuxIgdgK/pdLqsrCwXFxfpd7lk2LBhQUFBeXl5tbW1AwYMOH36\n9NChQwMDA6WlFhYWK1eu/OSTTzraZm5u7vLly4ODg+fOnSu1XL16VQgxZMiQzis5cuSIu7u7\nlCMlu3bteuONN1xcXLpSZxcPWQqIEi8vL2tr68d67YulpWVVVdXNmzednZ2lFukmtk7odLqY\nmJiqqqqcnBzpcQSdTpeZmTly5MhBgwbp341iaWk5ZsyYnJwcrVbbr18/o400Nzd7e3sbtuzf\nv1//pEJERERWVlbbXdfX1wsh+vbta9QubV9a2pY0Ypeenr5ixQo3N7eSkpKUlJTXX3+9vr7e\ncMK9LanC0tJST0/PTroB6F4EOwBfq66uvnv3bkBAgEKhMGz38fHJy8u7cuXKsGHDGhsbjTLZ\nmDFjOtpgenp6XFycr69vdna2Uvn11ebWrVtCiM6n8zQaTW1trVqtNqzEy8tLmivUaDSd1xkU\nFNTFQzZ60tPS0lIa2OuijRs3Ll682NvbOyIiYsKECT/96U/1U9Id2bBhw4kTJ5KTk0NDQ6WW\nmzdv3rp169atW88880zb/teuXWs7EWxhYZGZman/OG/evBdffDE2Nlb62HkNRidNCKHT6dpt\nl6xZs2bBggVhYWH6RBgTE6NWq9966624uDhpRrtdUtiVftwAnhqCHYCv/ec//xHtjej06dNH\nWlpbWyuEsLa2NlxqY2NjYWFhtIpOp1u/fv3GjRvDwsKOHDliY2OjX3Tv3j0hhK2tbSeV3L9/\nXwjR0RvgHllnJ1s2Ymlp2fXObS1atMjX1/e99947evRoWlqaQqGYPHnynj17PDw82u1/4sSJ\njRs3Tp06deXKlfpGaajMz89Pmvw14urq2rZRoVC88sor+o/Lli177rnnDFva1b9/f9HeyJz0\nEzH8GRkKCQkxahk+fHh4ePhHH31UVFT0wx/+sKPdSc9b3L17t/OqAHQvgh2Ar0lTcm2DkdRi\nY2MjxaDGxkbDpQ0NDS0tLYYtOp0uPj7+wIEDCxcu3L59u1Hsk+LF3bt3e/fu3VElAwcOFN9M\nAn6LOttdq6mpqaPdfRchISEhISEPHjw4ffr04cOHDx06NHHixIsXL7YdyiovL4+JifHx8UlN\nTTVs1xf8pF+YN3jwYKVSWVVVZdQu3VxoNLfbOWk0TqvVdtJH+vF1nuABdDtedwLgawMHDnRw\ncCgpKdH9/5svLl26pFAofHx8Bg4c2KtXL6NkcO7cOaPtLFmy5MCBA5s3b961a1fbwTxpElYa\n/OtI3759nZycSkpKDCdGv/zyy927d1+8ePGRdYpvhuIMV6+oqHjUCfj2rKysJk6cmJqaOnfu\n3NLS0vPnzxt1uH///tSpU5ubm48ePWoUPV1cXBwdHS9fvmwUZLv9mxtUKlVAQEB+fn5DQ4O+\nsbW1NTc3193d3WhWWqLVavfu3Zuenm7ULr28pqOBSYlUf1ceoQXQjQh2AP5n6tSpGo1Gejev\n5Pz58/n5+SEhIXZ2diqVKjAw8J///Ofly5elpS0tLVu2bDHcwtGjR3fu3Ll48eJVq1a1uwv9\nPfWdVxIREVFbW/vBBx/oW9avX79w4cIHDx48sk4hhHS/WklJib7DoUOHunQKviFFUmlSuF1n\nz551c3Mz2myvXr1EezO8c+bMKSoqOnjw4LBhw9pu6tVXX21sbNy6dau+paamZuTIkT/72c8e\nq+ZHmjVrVkNDg+GO9u3bd+PGjfj4eOljY2Pj+fPnpTE8IYS1tXVSUtLs2bP1P3EhRHZ2dl5e\nnr+/f7uvR9HrylMyALodU7EA/mfDhg3Hjh37xS9+sWjRIh8fn8rKypSUlH79+ukf9ly+fPmr\nr74aHh4+b968/v37Hz582MvLy/BmuBUrVgghWltb33zzTaONr1y50t7eXnpo4NSpU1OmTOmk\nknXr1h07diwhIaGoqMjDwyM3N/fYsWOxsbHSm5AfWWdsbOzevXuXLl26detWa2vr7OzsM2fO\ndDRL2y4ptSQnJ1dUVIwdO7btzWSBgYEODg6//vWv8/Ly/Pz8FArF559/npqaGhwc7OfnZ9jz\n8OHDaWlpfn5+d+7ckd7DpxcWFjZo0KD169cfP3588+bNGo1m3LhxN27c+N3vfldbW7to0aKu\nlFpZWdnFg/rVr36Vlpa2fv36L774Qq1Wl5SUZGRkPP/888uWLZM6lJaW+vv7h4aGfvzxx0KI\nXr167dmzJzIyMjAwMDo62tXVtbi4OCsrq3///kYHYkSn0506dWrIkCE8Egs8bSZ6zQqA7wuj\nrxS7du1aXFzcM888o1QqnZ2do6OjL126ZNh///79Pj4+KpXKw8Nj9erVTU1NKpVqzJgx0tJO\nrjbSK81aWlpcXFyGDRv2yMIqKytjYmKcnZ0tLS29vLzeffdd6f1wXawzNTV1+PDhffr0cXFx\nmT17dl1dnaura3BwsLRUeo/d1atXDVextbUdMWKE9O+mpqaoqKg+ffrY29tnZma2W2FtbW1i\nYuIPfvADa2trW1vbUaNGbd68ub6+3vDEajSa1atXd3RO9O/J02g0CQkJ7u7uSqXSzs5uypQp\n586de+Qp+hbq6+uXLVvm4eFhaWnp5uY2f/782tpa/dILFy4Ig7f9ST799NPJkyfb2dkplUpX\nV9fY2Fij89ZWQUGBEGLhwoVP4hAAdEKh6/RCDACdu3fvnq2t7ZQpUwwnRjuXnJy8atWqEydO\nTJ48+YnWBlOJiYnJyMj48ssvO5+uBdDtuMcOwGM4ePDg+PHjpfEYifSMZxe/yEuyYMGCAQMG\nbNq0qdvLw/dBWVnZhx9+GBsbS6oDnj5G7AA8hnPnzo0bN87e3j4hIcHV1fWLL77Yt2+fq6tr\nUVHRY31PfHp6+owZM3bt2rVw4cInVy2evpaWlpCQkMuXLxcXFzs5OZm6HMDsEOwAPJ5//OMf\nSUlJBQUFd+7ccXZ2njRp0qZNm9p9j27nFi5cuG/fvk8//TQgIOBJ1AmTWLt2bXJy8l/+8pfx\n48ebuhbAHBHsAAAAZIJ77AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGw\nAwAAkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGwAwAA\nkAmCHQAAgEwQ7AAAAGSCYAcAACATBDsAAACZINgBAADIBMEOAABAJgh2AAAAMkGwAwAAkAmC\nHQAAgEwQ7AAAAGSCYAcAACAT/wV6F2lQ7x/gmAAAAABJRU5ErkJggg==",
      "text/plain": [
       "Plot with title “voom: Mean-variance trend”"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#     Testing\n",
    "# set test = TRUE to test\n",
    "# set test = FALSE to ignore\n",
    "\n",
    "test = TRUE\n",
    "dup  = FALSE\n",
    "if (test == TRUE) {\n",
    "    splice_list       = c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\",\"se_\")\n",
    "    tissue_list_m_f     = levels(pData(reduced_obj2)$tissue)\n",
    "    splice_index        = 5\n",
    "    tissue_index        = 21\n",
    "    plot                = TRUE\n",
    "    splice_type         = splice_list    [splice_index] \n",
    "    tissue_of_interest  = tissue_list_m_f[tissue_index]\n",
    "    fromGTF             = se_fromGTF_no_chrY\n",
    "    metadata            = ocounts_metadata\n",
    "    ijc                 = se_jc_ijc_no_chrY\n",
    "\n",
    "    message(\"\\nijc\\n\", \n",
    "        paste(dim(ijc), collapse = \"\\n\") )\n",
    "    message(\"\\nhead(ijc)\\n\", \n",
    "        paste(ijc[1:5,1:5], collapse = \" \") )\n",
    "    sjc                 = se_jc_sjc_no_chrY\n",
    "    message(\"\\nsjc\\n\", \n",
    "        paste(dim(sjc), collapse = \"\\n\") )\n",
    "    message(\"\\nsjc\\n\", \n",
    "        paste(sjc[1:5,1:5], collapse = \" \") )\n",
    "    tissue_list         = tissue_list_m_f\n",
    "    obj                 = reduced_obj2\n",
    "    message(\"\\nobj\\n\", \n",
    "        paste(dim(obj), collapse = \"\\n\") )\n",
    "\n",
    "    \n",
    "    print_exploratory_plots (plot,\n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    message(\"\\ndone!\\n\") \n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"AllTissueJunctionAnalysis\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the artefacts in the `..data/` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -type f -exec sha256sum {} \\\\;  >  ../metadata/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\")\n",
    "\n",
    "data.table::fread(paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\"), header = FALSE, col.names = c(\"sha256sum\", \"file\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../metadata/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../metadata/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
