{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# AllTissueAllSplicingJunctionAnalysis as a Notebook \n",
    "\n",
    "rMATS 3.2.5 was run on controlled access RNASeq files retrieved experiments stored in the Sequence Read Archive with controlled access managed by dbGaP.   The data were generated under the Gene Tissue Expression.\n",
    "\n",
    "## rMATS RNASeq-MATS.py produces 10 different output types which get assembled into as type junction ID by sample ID matrices\n",
    "\n",
    "### Alternative Splice Site Types are: (se, a3ss, a5ss, mxe, ri)\n",
    "\n",
    " This is input as ARGV1 into variable 'astype'\n",
    "\n",
    "  * Skipped Exon events (se),\n",
    "  * Alternative 3' splice site (a3ss),\n",
    "  * Alternative 5' splice site (a5ss),\n",
    "  * Mutually exclusive exon (mxe),\n",
    "  * and retention intron (ri)\n",
    "\n",
    "### There are two different kinds of junction counts\n",
    "\n",
    "  * jc = junction counts - reads that cross the junction\n",
    "  * jcec = junction counts plus reads on the target (such as included exon\n",
    "\n",
    "### And the count type -- there are 5 types\n",
    "\n",
    "  * inclusion levels (percent spliced in)\n",
    "  * included junction counts (ijc)\n",
    "  * skipped junction counts (sjc)\n",
    "  * inclusion length (inclen)\n",
    "  * skipped length (skiplen)\n",
    "\n",
    "### function: fit_iso_tissue \n",
    "\n",
    "fit_iso_tissue expects the following input:\n",
    "\n",
    "  * the tissue of interest (SMSTD) \n",
    "  * an ordered_merged_rmats -- which will be ordered to fit the count matrix\n",
    "  * count matrix (inc or ijc & sjc merged)\n",
    "  * splice type (a3ss, a5ss, mxe, ri or se)\n",
    "  * junction_count type (jc or jcec)\n",
    "  * count type (inc or the merged ijc,sjc)\n",
    "  \n",
    "### reordering to match annotations between count matrix and annotation matrix\n",
    "\n",
    "Common problem is to match specifically the rows of an annotation matrix with the columns of a count matrix\n",
    "`match` is the function that gives the re-ordering index required to accomplish this\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **NOTE**:\n",
    "\n",
    "We assume that you have cloned the analysis repository and have `cd` into the parent directory. Before starting with the analysis make sure you have first completed the dependencies set up by following the instructions described in the **`dependencies/README.md`** document. All paths defined in this Notebook are relative to the parent directory (repository). Please close this Notebook and start again by following the above guidelines if you have not completed the aforementioned steps.\n",
    "\n",
    "## rMATS-final-merged\n",
    "the rmats-nf NextFlow was executed and the results released here:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: R.oo\n",
      "Loading required package: R.methodsS3\n",
      "R.methodsS3 v1.8.0 (2020-02-14 07:10:20 UTC) successfully loaded. See ?R.methodsS3 for help.\n",
      "R.oo v1.23.0 successfully loaded. See ?R.oo for help.\n",
      "\n",
      "Attaching package: ‘R.oo’\n",
      "\n",
      "The following object is masked from ‘package:R.methodsS3’:\n",
      "\n",
      "    throw\n",
      "\n",
      "The following objects are masked from ‘package:methods’:\n",
      "\n",
      "    getClasses, getMethods\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    attach, detach, load, save\n",
      "\n",
      "R.utils v2.9.2 successfully loaded. See ?R.utils for help.\n",
      "\n",
      "Attaching package: ‘R.utils’\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    timestamp\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    cat, commandArgs, getOption, inherits, isOpen, nullfile, parse,\n",
      "    warnings\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(limma)\n",
    "library(piggyback)\n",
    "library(multtest)\n",
    "library(Biobase)\n",
    "library(edgeR)\n",
    "library(tibble)\n",
    "#install.packages('R.utils')\n",
    "library(R.utils)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis \n",
    "\n",
    "This analysis uses edgeR.  Normalization takes the form of correction factors computed internally by edgeR functions, but it is also possible for a user to supply them. The correction factors may take the form of scaling factors for the library sizes, such as computed by calcNormFactors, which are then used to compute the effective library sizes. \n",
    "\n",
    "In this analysis, we are using the raw counts as provided by rMATS 3.2.5.  The raw counts we are using in the model are `ijc` and `sjc`, the sample specific raw read counts as they align to the junctions of the `included exon (ijc)` and the junctions of the `excluded or skipped exon (sjc)` respectively.\n",
    "\n",
    "\n",
    "Be sure to set your GITHUB_TOKEN, prior to downloading files\n",
    "\n",
    "One suggestion is change it to your token and then run it then immediately change it back to this:\n",
    "\n",
    "Sys.setenv(GITHUB_TOKEN = \"your-very-own-github-token\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Did you remember?\n",
    "Did you remember to delete your private github token?  Now is a good time to do so, before you save your work and checkit in inadvertantly...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!(\"rmats_final.se.jc.ijc.txt.gz\" %in% list.files(\"../data/\"))) {    \n",
    "    # SE\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.se.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    # RI\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.ri.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    # MXE\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.mxe.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    # A3SS\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a3ss.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "     # A5SS\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.ijc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.sjc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.inc.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.inclen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"adeslatt/sbas_test\", \n",
    "        file = \"rmats_final.a5ss.jc.skiplen.txt.gz\",\n",
    "        tag  = \"rMATS.3.2.5.GTEx.V8.final_matrices\", \n",
    "        dest = \"../data/\")\n",
    "} "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in the count data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading a3ss.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a3ss.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a5ss.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading a5ss.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading mxe.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading mxe.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading ri.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading ri.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading se.jc.ijc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n",
      "loading se.jc.sjc rMATS 3.2.5 counts\n",
      "\n",
      "done!\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>0</td><td>0</td><td> 2</td><td>0</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td>0</td><td>10</td><td>0</td></tr>\n",
       "\t<tr><td>3</td><td>0</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td>4</td><td>2</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td>5</td><td>1</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 0 & 0 &  2 & 0\\\\\n",
       "\t 2 & 0 & 0 & 10 & 0\\\\\n",
       "\t 3 & 0 & 0 &  0 & 0\\\\\n",
       "\t 4 & 2 & 0 &  0 & 0\\\\\n",
       "\t 5 & 1 & 0 &  0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 0 | 0 |  2 | 0 |\n",
       "| 2 | 0 | 0 | 10 | 0 |\n",
       "| 3 | 0 | 0 |  0 | 0 |\n",
       "| 4 | 2 | 0 |  0 | 0 |\n",
       "| 5 | 1 | 0 |  0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  0          0           2         0         \n",
       "2 2  0          0          10         0         \n",
       "3 3  0          0           0         0         \n",
       "4 4  2          0           0         0         \n",
       "5 5  1          0           0         0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>9</td><td>151</td><td>51</td><td>263</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>3</td><td>2</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td>0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>5</td><td>0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 9 & 151 & 51 & 263\\\\\n",
       "\t 2 & 0 &   0 &  0 &   0\\\\\n",
       "\t 3 & 2 &   0 &  0 &   0\\\\\n",
       "\t 4 & 0 &   0 &  0 &   0\\\\\n",
       "\t 5 & 0 &   0 &  0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 9 | 151 | 51 | 263 |\n",
       "| 2 | 0 |   0 |  0 |   0 |\n",
       "| 3 | 2 |   0 |  0 |   0 |\n",
       "| 4 | 0 |   0 |  0 |   0 |\n",
       "| 5 | 0 |   0 |  0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  9          151        51         263       \n",
       "2 2  0            0         0           0       \n",
       "3 3  2            0         0           0       \n",
       "4 4  0            0         0           0       \n",
       "5 5  0            0         0           0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 0</td><td> 0</td><td> 0</td><td> 2</td></tr>\n",
       "\t<tr><td>2</td><td> 0</td><td> 0</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td>3</td><td> 1</td><td> 0</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td>4</td><td>14</td><td>14</td><td> 3</td><td> 9</td></tr>\n",
       "\t<tr><td>5</td><td>22</td><td>67</td><td>10</td><td>20</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  0 &  0 &  0 &  2\\\\\n",
       "\t 2 &  0 &  0 &  0 &  1\\\\\n",
       "\t 3 &  1 &  0 &  0 &  1\\\\\n",
       "\t 4 & 14 & 14 &  3 &  9\\\\\n",
       "\t 5 & 22 & 67 & 10 & 20\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  0 |  0 |  0 |  2 |\n",
       "| 2 |  0 |  0 |  0 |  1 |\n",
       "| 3 |  1 |  0 |  0 |  1 |\n",
       "| 4 | 14 | 14 |  3 |  9 |\n",
       "| 5 | 22 | 67 | 10 | 20 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   0          0          0          2        \n",
       "2 2   0          0          0          1        \n",
       "3 3   1          0          0          1        \n",
       "4 4  14         14          3          9        \n",
       "5 5  22         67         10         20        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>56</td><td>79</td><td>30</td><td>778</td></tr>\n",
       "\t<tr><td>2</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>3</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td>25</td><td>13</td><td>12</td><td> 74</td></tr>\n",
       "\t<tr><td>5</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 56 & 79 & 30 & 778\\\\\n",
       "\t 2 &  0 &  0 &  0 &   0\\\\\n",
       "\t 3 &  0 &  0 &  0 &   0\\\\\n",
       "\t 4 & 25 & 13 & 12 &  74\\\\\n",
       "\t 5 &  0 &  0 &  0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 56 | 79 | 30 | 778 |\n",
       "| 2 |  0 |  0 |  0 |   0 |\n",
       "| 3 |  0 |  0 |  0 |   0 |\n",
       "| 4 | 25 | 13 | 12 |  74 |\n",
       "| 5 |  0 |  0 |  0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  56         79         30         778       \n",
       "2 2   0          0          0           0       \n",
       "3 3   0          0          0           0       \n",
       "4 4  25         13         12          74       \n",
       "5 5   0          0          0           0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 6</td><td>5</td><td>3</td><td> 9</td></tr>\n",
       "\t<tr><td>2</td><td>13</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>3</td><td>10</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>4</td><td> 1</td><td>1</td><td>0</td><td> 1</td></tr>\n",
       "\t<tr><td>5</td><td> 8</td><td>0</td><td>0</td><td>17</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  6 & 5 & 3 &  9\\\\\n",
       "\t 2 & 13 & 0 & 0 &  0\\\\\n",
       "\t 3 & 10 & 0 & 0 &  0\\\\\n",
       "\t 4 &  1 & 1 & 0 &  1\\\\\n",
       "\t 5 &  8 & 0 & 0 & 17\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  6 | 5 | 3 |  9 |\n",
       "| 2 | 13 | 0 | 0 |  0 |\n",
       "| 3 | 10 | 0 | 0 |  0 |\n",
       "| 4 |  1 | 1 | 0 |  1 |\n",
       "| 5 |  8 | 0 | 0 | 17 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   6         5          3           9        \n",
       "2 2  13         0          0           0        \n",
       "3 3  10         0          0           0        \n",
       "4 4   1         1          0           1        \n",
       "5 5   8         0          0          17        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>22</td><td>36</td><td>7</td><td>40</td></tr>\n",
       "\t<tr><td>2</td><td>10</td><td> 0</td><td>4</td><td> 0</td></tr>\n",
       "\t<tr><td>3</td><td> 1</td><td> 0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>4</td><td> 4</td><td> 0</td><td>2</td><td> 0</td></tr>\n",
       "\t<tr><td>5</td><td>51</td><td> 4</td><td>8</td><td>24</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 22 & 36 & 7 & 40\\\\\n",
       "\t 2 & 10 &  0 & 4 &  0\\\\\n",
       "\t 3 &  1 &  0 & 0 &  0\\\\\n",
       "\t 4 &  4 &  0 & 2 &  0\\\\\n",
       "\t 5 & 51 &  4 & 8 & 24\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 22 | 36 | 7 | 40 |\n",
       "| 2 | 10 |  0 | 4 |  0 |\n",
       "| 3 |  1 |  0 | 0 |  0 |\n",
       "| 4 |  4 |  0 | 2 |  0 |\n",
       "| 5 | 51 |  4 | 8 | 24 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  22         36         7          40        \n",
       "2 2  10          0         4           0        \n",
       "3 3   1          0         0           0        \n",
       "4 4   4          0         2           0        \n",
       "5 5  51          4         8          24        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>  0</td><td> 0</td><td>  5</td><td> 0</td></tr>\n",
       "\t<tr><td>2</td><td>  3</td><td> 0</td><td>  1</td><td> 6</td></tr>\n",
       "\t<tr><td>3</td><td>  2</td><td> 0</td><td>  0</td><td> 6</td></tr>\n",
       "\t<tr><td>4</td><td>  9</td><td> 0</td><td>  4</td><td>15</td></tr>\n",
       "\t<tr><td>5</td><td>112</td><td>64</td><td>113</td><td>59</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &   0 &  0 &   5 &  0\\\\\n",
       "\t 2 &   3 &  0 &   1 &  6\\\\\n",
       "\t 3 &   2 &  0 &   0 &  6\\\\\n",
       "\t 4 &   9 &  0 &   4 & 15\\\\\n",
       "\t 5 & 112 & 64 & 113 & 59\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |   0 |  0 |   5 |  0 |\n",
       "| 2 |   3 |  0 |   1 |  6 |\n",
       "| 3 |   2 |  0 |   0 |  6 |\n",
       "| 4 |   9 |  0 |   4 | 15 |\n",
       "| 5 | 112 | 64 | 113 | 59 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1    0         0           5         0        \n",
       "2 2    3         0           1         6        \n",
       "3 3    2         0           0         6        \n",
       "4 4    9         0           4        15        \n",
       "5 5  112        64         113        59        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>2</td><td> 0</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>3</td><td> 2</td><td> 0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td>88</td><td>98</td><td>25</td><td>206</td></tr>\n",
       "\t<tr><td>5</td><td>70</td><td>11</td><td>34</td><td> 37</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  0 &  0 &  0 &   0\\\\\n",
       "\t 2 &  0 &  0 &  0 &   0\\\\\n",
       "\t 3 &  2 &  0 &  0 &   0\\\\\n",
       "\t 4 & 88 & 98 & 25 & 206\\\\\n",
       "\t 5 & 70 & 11 & 34 &  37\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  0 |  0 |  0 |   0 |\n",
       "| 2 |  0 |  0 |  0 |   0 |\n",
       "| 3 |  2 |  0 |  0 |   0 |\n",
       "| 4 | 88 | 98 | 25 | 206 |\n",
       "| 5 | 70 | 11 | 34 |  37 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   0          0          0           0       \n",
       "2 2   0          0          0           0       \n",
       "3 3   2          0          0           0       \n",
       "4 4  88         98         25         206       \n",
       "5 5  70         11         34          37       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>2</td><td>26</td><td>247</td><td>103</td><td>620</td></tr>\n",
       "\t<tr><td>3</td><td> 1</td><td>  0</td><td>  1</td><td>  0</td></tr>\n",
       "\t<tr><td>4</td><td> 0</td><td>  1</td><td>  1</td><td>  2</td></tr>\n",
       "\t<tr><td>5</td><td> 3</td><td>  0</td><td>  2</td><td>  3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 &  0 &   0 &   0 &   0\\\\\n",
       "\t 2 & 26 & 247 & 103 & 620\\\\\n",
       "\t 3 &  1 &   0 &   1 &   0\\\\\n",
       "\t 4 &  0 &   1 &   1 &   2\\\\\n",
       "\t 5 &  3 &   0 &   2 &   3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 |  0 |   0 |   0 |   0 |\n",
       "| 2 | 26 | 247 | 103 | 620 |\n",
       "| 3 |  1 |   0 |   1 |   0 |\n",
       "| 4 |  0 |   1 |   1 |   2 |\n",
       "| 5 |  3 |   0 |   2 |   3 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1   0           0          0          0       \n",
       "2 2  26         247        103        620       \n",
       "3 3   1           0          1          0       \n",
       "4 4   0           1          1          2       \n",
       "5 5   3           0          2          3       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>2</td><td>0</td><td>1</td><td>3</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>3</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>4</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>5</td><td>0</td><td>5</td><td>3</td><td>8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " ID & SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & 2 & 0 & 1 & 3\\\\\n",
       "\t 2 & 0 & 0 & 0 & 1\\\\\n",
       "\t 3 & 0 & 0 & 0 & 1\\\\\n",
       "\t 4 & 0 & 0 & 0 & 1\\\\\n",
       "\t 5 & 0 & 5 & 3 & 8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| ID &lt;int&gt; | SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | 2 | 0 | 1 | 3 |\n",
       "| 2 | 0 | 0 | 0 | 1 |\n",
       "| 3 | 0 | 0 | 0 | 1 |\n",
       "| 4 | 0 | 0 | 0 | 1 |\n",
       "| 5 | 0 | 5 | 3 | 8 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID SRR1068788 SRR1068808 SRR1068832 SRR1068855\n",
       "1 1  2          0          1          3         \n",
       "2 2  0          0          0          1         \n",
       "3 3  0          0          0          1         \n",
       "4 4  0          0          0          1         \n",
       "5 5  0          5          3          8         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "message(\"loading a3ss.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "a3ss.jc.ijc <- data.table::fread(\"../data/rmats_final.a3ss.jc.ijc.txt.gz\")\n",
    "message(\"done!\\n\")\n",
    "message(\"loading a3ss.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "a3ss.jc.sjc <- data.table::fread(\"../data/rmats_final.a3ss.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading a5ss.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "a5ss.jc.ijc <- data.table::fread(\"../data/rmats_final.a5ss.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading a5ss.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "a5ss.jc.sjc <- data.table::fread(\"../data/rmats_final.a5ss.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading mxe.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "mxe.jc.ijc  <- data.table::fread(\"../data/rmats_final.mxe.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading mxe.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "mxe.jc.sjc  <- data.table::fread(\"../data/rmats_final.mxe.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading ri.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "ri.jc.ijc   <- data.table::fread(\"../data/rmats_final.ri.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading ri.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "ri.jc.sjc   <- data.table::fread(\"../data/rmats_final.ri.jc.sjc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "\n",
    "message(\"loading se.jc.ijc rMATS 3.2.5 counts\\n\")\n",
    "se.jc.ijc   <- data.table::fread(\"../data/rmats_final.se.jc.ijc.txt.gz\")    \n",
    "message(\"done!\\n\")\n",
    "message(\"loading se.jc.sjc rMATS 3.2.5 counts\\n\")\n",
    "se.jc.sjc   <- data.table::fread(\"../data/rmats_final.se.jc.sjc.txt.gz\")\n",
    "message(\"done!\\n\")\n",
    "\n",
    "a3ss.jc.ijc[1:5,1:5]\n",
    "a3ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "a5ss.jc.ijc[1:5,1:5]\n",
    "a5ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "mxe.jc.ijc[1:5,1:5]\n",
    "mxe.jc.sjc[1:5,1:5]\n",
    "\n",
    "ri.jc.ijc[1:5,1:5]\n",
    "ri.jc.sjc[1:5,1:5]\n",
    "\n",
    "se.jc.ijc[1:5,1:5]\n",
    "se.jc.sjc[1:5,1:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in metadata \n",
    "\n",
    "- `Sequence Read Archive (SRA)` Accession Data, `SRR` numbers\n",
    "- `Genome Tissue Expression (GTEx)` Clinical Annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\n",
      "\n",
      "done!\n",
      "\n",
      "Loading obj GTEx v8 rds object with readRDS from ../data/gtex.rds ..\n",
      "\n",
      "Done!\n",
      "\n",
      "Generating sha256sum for gtex.rds ..\n",
      "\n",
      "c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0  ../data/gtex.rds\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!(\"SraRunTable.txt.gz\" %in% list.files(\"../data/\"))) {\n",
    "    piggyback::pb_download(\n",
    "        show_progress = TRUE,\n",
    "        repo = \"TheJacksonLaboratory/sbas\", \n",
    "        file = \"SraRunTable.txt.gz\",\n",
    "        tag  = \"GTExV8.v1.0\", \n",
    "        dest = \"../data/\")\n",
    "    \n",
    "    message(\"Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\\n\")   \n",
    "    metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    message(\"done!\")\n",
    "} else {\n",
    "    message(\"Loading metadata from SraRunTable.txt.gz ../data/gtex.rds ..\\n\")   \n",
    "    metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    message(\"done!\\n\")\n",
    "}\n",
    "\n",
    "if (!(\"gtex.rds\" %in% list.files(\"../data/\"))) {\n",
    "    message(\"Downloading and loading obj with GTEx v8 with 'yarn::downloadGTExV8()'\\n\")\n",
    "    obj <- yarn::downloadGTExV8(type='genes',file='../data/gtex.rds')\n",
    "    message(\"Done!\\n\")\n",
    "\n",
    "} else {\n",
    "# Load with readRDS() if gtex.rds available in data/\n",
    "    message(\"Loading obj GTEx v8 rds object with readRDS from ../data/gtex.rds ..\\n\")   \n",
    "    obj <- readRDS(file = \"../data/gtex.rds\")\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Generating sha256sum for gtex.rds ..\\n\")    \n",
    "    message(system(\"sha256sum ../data/gtex.rds\", intern = TRUE))\n",
    "    message(\"Done!\\n\")\n",
    "} \n",
    "if (! (file.exists(\"../data/fromGTF.tar.gz\"))) {\n",
    "    system(\"mkdir -p ../data\", intern = TRUE)\n",
    "    message(\"Fetching fromGTF.tar.gz from GitHub ..\")\n",
    "    # Download archive from GitHub release with tag \"dge\"\n",
    "    piggyback::pb_download(file = \"fromGTF.tar.gz\",\n",
    "                           dest = \"../data\",\n",
    "                           repo = \"adeslatt/sbas_gtf\",\n",
    "                           tag  = \"rMATS.3.2.5.gencode.v30\",\n",
    "                           show_progress = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Decompressing fromGTF.tar.gz into ../data\")\n",
    "    system(\"mkdir -p ../data && tar xvfz ../data/fromGTF.tar.gz -C ../data\", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Decompressing fromGTF.*.txt.gz into ../data\")\n",
    "    system(\"gunzip  ../data/fromGTF*.txt.gz \", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Reading fromGTF.SE.txt into fromGTF.SE\")    \n",
    "    fromGTF.SE <- read.table(\"../data/fromGTF.SE.txt\", header=TRUE)\n",
    "    message(\"Done!\\n\") \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quality control, preprocessing of data \n",
    "\n",
    "We observed above that our phenotype data have 2 more observations than our expression data. Let's inspect what these samples are:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-YF7O-2326-101833-SM-5CVN9</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>Skin   </td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>GTEX-YEC3-1426-101806-SM-5PNXX</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>Stomach</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-YF7O-2326-101833-SM-5CVN9 & NA & NA & NA & NA & NA & Skin    & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tGTEX-YEC3-1426-101806-SM-5PNXX & NA & NA & NA & NA & NA & Stomach & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-YF7O-2326-101833-SM-5CVN9 | NA | NA | NA | NA | NA | Skin    | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| GTEX-YEC3-1426-101806-SM-5PNXX | NA | NA | NA | NA | NA | Stomach | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "                               SAMPID SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS   \n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA     NA       NA       NA       NA    Skin   \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA     NA       NA       NA       NA    Stomach\n",
       "                               SMTSD SMUBRID SMTSISCH SMTSPAX ⋯ SME1PCTS\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA    NA      NA       NA      ⋯ NA      \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA    NA      NA       NA      ⋯ NA      \n",
       "                               SMRRNART SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA       NA       NA       NA       NA      \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA       NA       NA       NA       NA      \n",
       "                               SUBJID SEX AGE DTHHRDY\n",
       "GTEX-YF7O-2326-101833-SM-5CVN9 NA     NA  NA  NA     \n",
       "GTEX-YEC3-1426-101806-SM-5PNXX NA     NA  NA  NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17384</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17384\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17384\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17384    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55878</li><li>17382</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55878\n",
       "\\item 17382\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55878\n",
       "2. 17382\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55878 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17382"
      ],
      "text/latex": [
       "17382"
      ],
      "text/markdown": [
       "17382"
      ],
      "text/plain": [
       "[1] 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The non-overlapping IDs between pheno and count data are:\n",
      "GTEX-YF7O-2326-101833-SM-5CVN9\n",
      "GTEX-YEC3-1426-101806-SM-5PNXX\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "logical_match_names\n",
       "FALSE  TRUE \n",
       "    2 17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-ZZPU-2626-SM-5E45Y</th><td>GTEX-ZZPU-2626-SM-5E45Y</td><td> 1</td><td>B1</td><td>2 pieces, includes internal fat (up to 20%) and tendon</td><td> 8.50000</td><td>Muscle        </td><td>Muscle - Skeletal     </td><td>0011907</td><td>  248</td><td> 428</td><td>⋯</td><td>50.6647</td><td>0.010466900</td><td>0.9956840</td><td>NA</td><td> 0</td><td>49.6941</td><td>GTEX-ZZPU</td><td>2</td><td>50-59</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>GTEX-ZZPU-2726-SM-5NQ8O</th><td>GTEX-ZZPU-2726-SM-5NQ8O</td><td> 1</td><td>B1</td><td>2 pieces                                              </td><td> 6.80000</td><td>Adipose Tissue</td><td>Adipose - Subcutaneous</td><td>0002190</td><td>  249</td><td> 427</td><td>⋯</td><td>50.3168</td><td>0.005856410</td><td>0.9958590</td><td>NA</td><td> 0</td><td>50.0760</td><td>GTEX-ZZPU</td><td>2</td><td>50-59</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-ZZPU-2626-SM-5E45Y & GTEX-ZZPU-2626-SM-5E45Y &  1 & B1 & 2 pieces, includes internal fat (up to 20\\%) and tendon &  8.50000 & Muscle         & Muscle - Skeletal      & 0011907 &   248 &  428 & ⋯ & 50.6647 & 0.010466900 & 0.9956840 & NA &  0 & 49.6941 & GTEX-ZZPU & 2 & 50-59 &  0\\\\\n",
       "\tGTEX-ZZPU-2726-SM-5NQ8O & GTEX-ZZPU-2726-SM-5NQ8O &  1 & B1 & 2 pieces                                               &  6.80000 & Adipose Tissue & Adipose - Subcutaneous & 0002190 &   249 &  427 & ⋯ & 50.3168 & 0.005856410 & 0.9958590 & NA &  0 & 50.0760 & GTEX-ZZPU & 2 & 50-59 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-ZZPU-2626-SM-5E45Y | GTEX-ZZPU-2626-SM-5E45Y |  1 | B1 | 2 pieces, includes internal fat (up to 20%) and tendon |  8.50000 | Muscle         | Muscle - Skeletal      | 0011907 |   248 |  428 | ⋯ | 50.6647 | 0.010466900 | 0.9956840 | NA |  0 | 49.6941 | GTEX-ZZPU | 2 | 50-59 |  0 |\n",
       "| GTEX-ZZPU-2726-SM-5NQ8O | GTEX-ZZPU-2726-SM-5NQ8O |  1 | B1 | 2 pieces                                               |  6.80000 | Adipose Tissue | Adipose - Subcutaneous | 0002190 |   249 |  427 | ⋯ | 50.3168 | 0.005856410 | 0.9958590 | NA |  0 | 50.0760 | GTEX-ZZPU | 2 | 50-59 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "                        SAMPID                  SMATSSCR SMCENTER\n",
       "GTEX-ZZPU-2626-SM-5E45Y GTEX-ZZPU-2626-SM-5E45Y  1       B1      \n",
       "GTEX-ZZPU-2726-SM-5NQ8O GTEX-ZZPU-2726-SM-5NQ8O  1       B1      \n",
       "                        SMPTHNTS                                              \n",
       "GTEX-ZZPU-2626-SM-5E45Y 2 pieces, includes internal fat (up to 20%) and tendon\n",
       "GTEX-ZZPU-2726-SM-5NQ8O 2 pieces                                              \n",
       "                        SMRIN    SMTS           SMTSD                  SMUBRID\n",
       "GTEX-ZZPU-2626-SM-5E45Y  8.50000 Muscle         Muscle - Skeletal      0011907\n",
       "GTEX-ZZPU-2726-SM-5NQ8O  6.80000 Adipose Tissue Adipose - Subcutaneous 0002190\n",
       "                        SMTSISCH SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT \n",
       "GTEX-ZZPU-2626-SM-5E45Y   248     428    ⋯ 50.6647  0.010466900 0.9956840\n",
       "GTEX-ZZPU-2726-SM-5NQ8O   249     427    ⋯ 50.3168  0.005856410 0.9958590\n",
       "                        SMNUM5CD SMDPMPRT SME2PCTS SUBJID    SEX AGE   DTHHRDY\n",
       "GTEX-ZZPU-2626-SM-5E45Y NA        0       49.6941  GTEX-ZZPU 2   50-59  0     \n",
       "GTEX-ZZPU-2726-SM-5NQ8O NA        0       50.0760  GTEX-ZZPU 2   50-59  0     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55872</li><li>17382</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55872\n",
       "\\item 17382\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55872\n",
       "2. 17382\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55872 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55872</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55872\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55872Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55872    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tail (pData(obj),2)\n",
    "dim(pData(obj))\n",
    "dim(exprs(obj))\n",
    "dim(obj)\n",
    "sample_names=as.vector(as.character(colnames(exprs(obj))))\n",
    "pheno_sample_names=as.vector(as.character(rownames(pData(obj))))\n",
    "length(pheno_sample_names)\n",
    "length(sample_names)\n",
    "\n",
    "if (length(pheno_sample_names) > length(sample_names)) {\n",
    "    superset <- pheno_sample_names\n",
    "    subset   <- sample_names    \n",
    "} \n",
    "\n",
    "if (length(pheno_sample_names) < length(sample_names)) {\n",
    "    superset <- sample_names\n",
    "    subset   <- pheno_sample_names   \n",
    "} \n",
    "\n",
    "non_overlaps <- setdiff( superset, subset)\n",
    "\n",
    "message(\"The non-overlapping IDs between pheno and count data are:\\n\", \n",
    "        paste(non_overlaps, collapse = \"\\n\") )\n",
    "logical_match_names=superset %in% subset\n",
    "length(logical_match_names)\n",
    "table(logical_match_names)\n",
    "obj        <- obj       [logical_match_names==TRUE,]\n",
    "pData(obj) <- pData(obj)[logical_match_names==TRUE,]\n",
    "\n",
    "tail (pData(obj),2)\n",
    "dim(pData(obj))\n",
    "dim(exprs(obj))\n",
    "dim(obj)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in junction Annotation Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading fromGTF.A3SS.txt into fromGTF.A3SS\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.A5SS.txt into fromGTF.A5SS\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.MXE.txt into fromGTF.MXE\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.RI.txt into fromGTF.RI\n",
      "Done!\n",
      "\n",
      "Reading fromGTF.SE.txt into fromGTF.SE\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"Reading fromGTF.A3SS.txt into fromGTF.A3SS\")    \n",
    "fromGTF.A3SS <- read.table(\"../data/fromGTF.A3SS.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.A5SS.txt into fromGTF.A5SS\")    \n",
    "fromGTF.A5SS <- read.table(\"../data/fromGTF.A5SS.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.MXE.txt into fromGTF.MXE\")    \n",
    "fromGTF.MXE <- read.table(\"../data/fromGTF.MXE.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.RI.txt into fromGTF.RI\")    \n",
    "fromGTF.RI <- read.table(\"../data/fromGTF.RI.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Reading fromGTF.SE.txt into fromGTF.SE\")    \n",
    "fromGTF.SE <- read.table(\"../data/fromGTF.SE.txt\", header=TRUE)\n",
    "message(\"Done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preparing our Data\n",
    "### Aligning Annotations with Run Data and with Count Data\n",
    "\n",
    "A description of the GTEx V8 release may be found here https://www.gtexportal.org/home/datasets.   To facilitate the analysis, the Quackenbush lab's Joe Paulson's Yarn Package, https://github.com/jnpaulson/yarn, forked and upgraded to GTEx V8 here https://github.com/TheJacksonLaboratory/yarn/tree/annes-changes.  \n",
    "\n",
    "The sequences we used for our analysis with rMATS are specified by their SRR number and were obtained from dbGaP, with our application to the controlled access human data.  Beginning with these raw unfiltered fastq files, we ran them through the NextFlow workflow rmats-nf https://github.com/lifebit-ai/rmats-nf. The accession data was downloaded with a repository key from the google cloud bucket (g3).\n",
    "\n",
    "- `obj`      : GTEx V8 expressionSet \n",
    "- `metadata` : SraRunTable.txt holds the SRR accession numbers used to extract samples from dbGaP SRA\n",
    "- `ijc`      : inclusion junction counts as reported by `rMATS 3.2.5`\n",
    "- `sjc`      : skipped junction counts as reported by `rMATS 3.2.5`  \n",
    "\n",
    "For each of the splicing events:\n",
    "\n",
    "- `a3ss`     : alternative 3` splice site\n",
    "- `a5ss`     : alternative 5` splice site\n",
    "- `mxe`      : mutually exclusive exon\n",
    "- `ri`       : retention intron\n",
    "- `se`       : skipped exon\n",
    "\n",
    "In `rMATS 3.2.5`, each junction is encoded in a unique `ID` that is immutable regardless of comparison.  \n",
    "This `ID` is different for each of the differing splicing events."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Move Junction ID to Row definition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td>0</td><td> 2</td><td>0</td><td>2</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>10</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td> 0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>2</td><td>0</td><td> 0</td><td>0</td><td>2</td></tr>\n",
       "\t<tr><td>1</td><td>0</td><td> 0</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 0 & 0 &  2 & 0 & 2\\\\\n",
       "\t 0 & 0 & 10 & 0 & 0\\\\\n",
       "\t 0 & 0 &  0 & 0 & 1\\\\\n",
       "\t 2 & 0 &  0 & 0 & 2\\\\\n",
       "\t 1 & 0 &  0 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 0 | 0 |  2 | 0 | 2 |\n",
       "| 0 | 0 | 10 | 0 | 0 |\n",
       "| 0 | 0 |  0 | 0 | 1 |\n",
       "| 2 | 0 |  0 | 0 | 2 |\n",
       "| 1 | 0 |  0 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 0          0           2         0          2         \n",
       "2 0          0          10         0          0         \n",
       "3 0          0           0         0          1         \n",
       "4 2          0           0         0          2         \n",
       "5 1          0           0         0          1         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>9</td><td>151</td><td>51</td><td>263</td><td>292</td></tr>\n",
       "\t<tr><td>0</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>2</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>0</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>0</td><td>  0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 9 & 151 & 51 & 263 & 292\\\\\n",
       "\t 0 &   0 &  0 &   0 &   0\\\\\n",
       "\t 2 &   0 &  0 &   0 &   0\\\\\n",
       "\t 0 &   0 &  0 &   0 &   0\\\\\n",
       "\t 0 &   0 &  0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 9 | 151 | 51 | 263 | 292 |\n",
       "| 0 |   0 |  0 |   0 |   0 |\n",
       "| 2 |   0 |  0 |   0 |   0 |\n",
       "| 0 |   0 |  0 |   0 |   0 |\n",
       "| 0 |   0 |  0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 9          151        51         263        292       \n",
       "2 0            0         0           0          0       \n",
       "3 2            0         0           0          0       \n",
       "4 0            0         0           0          0       \n",
       "5 0            0         0           0          0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td> 2</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td> 1</td><td>  0</td></tr>\n",
       "\t<tr><td> 1</td><td> 0</td><td> 0</td><td> 1</td><td>  0</td></tr>\n",
       "\t<tr><td>14</td><td>14</td><td> 3</td><td> 9</td><td>  4</td></tr>\n",
       "\t<tr><td>22</td><td>67</td><td>10</td><td>20</td><td>167</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &  0 &  2 &   1\\\\\n",
       "\t  0 &  0 &  0 &  1 &   0\\\\\n",
       "\t  1 &  0 &  0 &  1 &   0\\\\\n",
       "\t 14 & 14 &  3 &  9 &   4\\\\\n",
       "\t 22 & 67 & 10 & 20 & 167\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |  0 |  2 |   1 |\n",
       "|  0 |  0 |  0 |  1 |   0 |\n",
       "|  1 |  0 |  0 |  1 |   0 |\n",
       "| 14 | 14 |  3 |  9 |   4 |\n",
       "| 22 | 67 | 10 | 20 | 167 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  0          0          0          2           1       \n",
       "2  0          0          0          1           0       \n",
       "3  1          0          0          1           0       \n",
       "4 14         14          3          9           4       \n",
       "5 22         67         10         20         167       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>56</td><td>79</td><td>30</td><td>778</td><td>293</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>25</td><td>13</td><td>12</td><td> 74</td><td> 10</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 56 & 79 & 30 & 778 & 293\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t 25 & 13 & 12 &  74 &  10\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 56 | 79 | 30 | 778 | 293 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "| 25 | 13 | 12 |  74 |  10 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 56         79         30         778        293       \n",
       "2  0          0          0           0          0       \n",
       "3  0          0          0           0          0       \n",
       "4 25         13         12          74         10       \n",
       "5  0          0          0           0          0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 6</td><td>5</td><td>3</td><td> 9</td><td>4</td></tr>\n",
       "\t<tr><td>13</td><td>0</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td>10</td><td>0</td><td>0</td><td> 0</td><td>0</td></tr>\n",
       "\t<tr><td> 1</td><td>1</td><td>0</td><td> 1</td><td>0</td></tr>\n",
       "\t<tr><td> 8</td><td>0</td><td>0</td><td>17</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  6 & 5 & 3 &  9 & 4\\\\\n",
       "\t 13 & 0 & 0 &  0 & 0\\\\\n",
       "\t 10 & 0 & 0 &  0 & 0\\\\\n",
       "\t  1 & 1 & 0 &  1 & 0\\\\\n",
       "\t  8 & 0 & 0 & 17 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  6 | 5 | 3 |  9 | 4 |\n",
       "| 13 | 0 | 0 |  0 | 0 |\n",
       "| 10 | 0 | 0 |  0 | 0 |\n",
       "|  1 | 1 | 0 |  1 | 0 |\n",
       "|  8 | 0 | 0 | 17 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  6         5          3           9         4         \n",
       "2 13         0          0           0         0         \n",
       "3 10         0          0           0         0         \n",
       "4  1         1          0           1         0         \n",
       "5  8         0          0          17         0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>22</td><td>36</td><td>7</td><td>40</td><td>10</td></tr>\n",
       "\t<tr><td>10</td><td> 0</td><td>4</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td> 1</td><td> 0</td><td>0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 4</td><td> 0</td><td>2</td><td> 0</td><td> 1</td></tr>\n",
       "\t<tr><td>51</td><td> 4</td><td>8</td><td>24</td><td> 2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 22 & 36 & 7 & 40 & 10\\\\\n",
       "\t 10 &  0 & 4 &  0 &  1\\\\\n",
       "\t  1 &  0 & 0 &  0 &  0\\\\\n",
       "\t  4 &  0 & 2 &  0 &  1\\\\\n",
       "\t 51 &  4 & 8 & 24 &  2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 22 | 36 | 7 | 40 | 10 |\n",
       "| 10 |  0 | 4 |  0 |  1 |\n",
       "|  1 |  0 | 0 |  0 |  0 |\n",
       "|  4 |  0 | 2 |  0 |  1 |\n",
       "| 51 |  4 | 8 | 24 |  2 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 22         36         7          40         10        \n",
       "2 10          0         4           0          1        \n",
       "3  1          0         0           0          0        \n",
       "4  4          0         2           0          1        \n",
       "5 51          4         8          24          2        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  5</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  3</td><td> 0</td><td>  1</td><td> 6</td><td> 1</td></tr>\n",
       "\t<tr><td>  2</td><td> 0</td><td>  0</td><td> 6</td><td> 2</td></tr>\n",
       "\t<tr><td>  9</td><td> 0</td><td>  4</td><td>15</td><td> 7</td></tr>\n",
       "\t<tr><td>112</td><td>64</td><td>113</td><td>59</td><td>43</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t   0 &  0 &   5 &  0 &  0\\\\\n",
       "\t   3 &  0 &   1 &  6 &  1\\\\\n",
       "\t   2 &  0 &   0 &  6 &  2\\\\\n",
       "\t   9 &  0 &   4 & 15 &  7\\\\\n",
       "\t 112 & 64 & 113 & 59 & 43\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|   0 |  0 |   5 |  0 |  0 |\n",
       "|   3 |  0 |   1 |  6 |  1 |\n",
       "|   2 |  0 |   0 |  6 |  2 |\n",
       "|   9 |  0 |   4 | 15 |  7 |\n",
       "| 112 | 64 | 113 | 59 | 43 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1   0         0           5         0          0        \n",
       "2   3         0           1         6          1        \n",
       "3   2         0           0         6          2        \n",
       "4   9         0           4        15          7        \n",
       "5 112        64         113        59         43        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 2</td><td> 0</td><td> 0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>88</td><td>98</td><td>25</td><td>206</td><td>150</td></tr>\n",
       "\t<tr><td>70</td><td>11</td><td>34</td><td> 37</td><td> 56</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t  0 &  0 &  0 &   0 &   0\\\\\n",
       "\t  2 &  0 &  0 &   0 &   0\\\\\n",
       "\t 88 & 98 & 25 & 206 & 150\\\\\n",
       "\t 70 & 11 & 34 &  37 &  56\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "|  0 |  0 |  0 |   0 |   0 |\n",
       "|  2 |  0 |  0 |   0 |   0 |\n",
       "| 88 | 98 | 25 | 206 | 150 |\n",
       "| 70 | 11 | 34 |  37 |  56 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  0          0          0           0          0       \n",
       "2  0          0          0           0          0       \n",
       "3  2          0          0           0          0       \n",
       "4 88         98         25         206        150       \n",
       "5 70         11         34          37         56       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>  0</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>26</td><td>247</td><td>103</td><td>620</td><td>494</td></tr>\n",
       "\t<tr><td> 1</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td>  1</td><td>  1</td><td>  2</td><td>  0</td></tr>\n",
       "\t<tr><td> 3</td><td>  0</td><td>  2</td><td>  3</td><td>  6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t  0 &   0 &   0 &   0 &   1\\\\\n",
       "\t 26 & 247 & 103 & 620 & 494\\\\\n",
       "\t  1 &   0 &   1 &   0 &   0\\\\\n",
       "\t  0 &   1 &   1 &   2 &   0\\\\\n",
       "\t  3 &   0 &   2 &   3 &   6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "|  0 |   0 |   0 |   0 |   1 |\n",
       "| 26 | 247 | 103 | 620 | 494 |\n",
       "|  1 |   0 |   1 |   0 |   0 |\n",
       "|  0 |   1 |   1 |   2 |   0 |\n",
       "|  3 |   0 |   2 |   3 |   6 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1  0           0          0          0          1       \n",
       "2 26         247        103        620        494       \n",
       "3  1           0          1          0          0       \n",
       "4  0           1          1          2          0       \n",
       "5  3           0          2          3          6       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 5 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068788</th><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068855</th><th scope=col>SRR1068880</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>2</td><td>0</td><td>1</td><td>3</td><td>6</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>5</td><td>3</td><td>8</td><td>4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068788 & SRR1068808 & SRR1068832 & SRR1068855 & SRR1068880\\\\\n",
       " <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 2 & 0 & 1 & 3 & 6\\\\\n",
       "\t 0 & 0 & 0 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 1 & 0\\\\\n",
       "\t 0 & 5 & 3 & 8 & 4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 5\n",
       "\n",
       "| SRR1068788 &lt;int&gt; | SRR1068808 &lt;int&gt; | SRR1068832 &lt;int&gt; | SRR1068855 &lt;int&gt; | SRR1068880 &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 2 | 0 | 1 | 3 | 6 |\n",
       "| 0 | 0 | 0 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 1 | 0 |\n",
       "| 0 | 5 | 3 | 8 | 4 |\n",
       "\n"
      ],
      "text/plain": [
       "  SRR1068788 SRR1068808 SRR1068832 SRR1068855 SRR1068880\n",
       "1 2          0          1          3          6         \n",
       "2 0          0          0          1          0         \n",
       "3 0          0          0          1          0         \n",
       "4 0          0          0          1          0         \n",
       "5 0          5          3          8          4         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rownames(a3ss.jc.ijc) <- a3ss.jc.ijc$ID\n",
    "rownames(a3ss.jc.sjc) <- a3ss.jc.sjc$ID\n",
    "a3ss.jc.ijc <- a3ss.jc.ijc[,-1]\n",
    "a3ss.jc.sjc <- a3ss.jc.sjc[,-1]\n",
    "\n",
    "rownames(a5ss.jc.ijc) <- a3ss.jc.ijc$ID\n",
    "rownames(a5ss.jc.sjc) <- a3ss.jc.sjc$ID\n",
    "a5ss.jc.ijc <- a5ss.jc.ijc[,-1]\n",
    "a5ss.jc.sjc <- a5ss.jc.sjc[,-1]\n",
    "\n",
    "rownames(mxe.jc.ijc)  <- mxe.jc.ijc$ID\n",
    "rownames(mxe.jc.sjc)  <- mxe.jc.sjc$ID\n",
    "mxe.jc.ijc <- mxe.jc.ijc[,-1]\n",
    "mxe.jc.sjc <- mxe.jc.sjc[,-1]\n",
    "\n",
    "rownames(ri.jc.ijc)   <- ri.jc.ijc$ID\n",
    "rownames(ri.jc.sjc)   <- ri.jc.sjc$ID\n",
    "ri.jc.ijc <- ri.jc.ijc[,-1]\n",
    "ri.jc.sjc <- ri.jc.sjc[,-1]\n",
    "\n",
    "rownames(se.jc.ijc)   <- se.jc.ijc$ID\n",
    "rownames(se.jc.sjc)   <- se.jc.sjc$ID\n",
    "se.jc.ijc <- se.jc.ijc[,-1]\n",
    "se.jc.sjc <- se.jc.sjc[,-1]\n",
    "\n",
    "dim(a3ss.jc.ijc)\n",
    "dim(a3ss.jc.sjc)\n",
    "dim(a5ss.jc.ijc)\n",
    "dim(a5ss.jc.sjc)\n",
    "dim(mxe.jc.ijc)\n",
    "dim(mxe.jc.sjc)\n",
    "dim(ri.jc.ijc)\n",
    "dim(ri.jc.sjc)\n",
    "dim(se.jc.ijc)\n",
    "dim(se.jc.sjc)\n",
    "\n",
    "a3ss.jc.ijc[1:5,1:5]\n",
    "a3ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "a5ss.jc.ijc[1:5,1:5]\n",
    "a5ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "mxe.jc.ijc[1:5,1:5]\n",
    "mxe.jc.sjc[1:5,1:5]\n",
    "\n",
    "ri.jc.ijc[1:5,1:5]\n",
    "ri.jc.sjc[1:5,1:5]\n",
    "\n",
    "se.jc.ijc[1:5,1:5]\n",
    "se.jc.sjc[1:5,1:5]\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Synchronize Clinical Annotations and Accession Run, reducing to only tissues of interest\n",
    "\n",
    "Join the yarn metadata with the metadata we have (there are redundant samples that have been sequenced multiple times). We want to be sure that we can obtain all required Clinical Annotation information from the YARN GTEx Annotation information, as the SRA metadata is not as reliable.    This will be a one-to-many mapping, as there are multiple sequence runs per 69 samples -- expanding our data set.  There are only a handful of annotations we require: SEX, AGE, DTHHRDY (which is cause of death), SMCENTER.\n",
    "\n",
    "Note that the numbers in specific age groups expand because of the one to many relationship from sample to sequencing runs. \n",
    "\n",
    "#### Using results from analysis of number of samples stored in `tissues.tsv` we keep only those that are members of this reduced tissue list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "size tissue_reduction\n",
      "50 5\n",
      "\n",
      "size obj\n",
      "5587217382\n",
      "\n",
      "\n",
      "size pData(obj)\n",
      "17382 67\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " 0  1 \n",
       "11 39 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "TEST: how many to keep of obj in breast_mammary_tissue\n",
      "17382\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55872 0\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "0 67\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>NA</th><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tNA & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "   SAMPID SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS SMTSD SMUBRID SMTSISCH SMTSPAX\n",
       "NA NA     NA       NA       NA       NA    NA   NA    NA      NA       NA     \n",
       "   ⋯ SME1PCTS SMRRNART SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS SUBJID SEX AGE\n",
       "NA ⋯ NA       NA       NA       NA       NA       NA       NA     NA  NA \n",
       "   DTHHRDY\n",
       "NA NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "length tissues in phenotype data\n",
      "54\n",
      "\n",
      "length tissues in tissue_reduction data\n",
      "39\n",
      "\n",
      "length tissue in samples phenotype data\n",
      "17382\n",
      "\n",
      "length keep obj \n",
      "17382\n",
      "\n",
      "how many to keep in phenotype data\n",
      "1851 15531\n",
      "\n",
      "size reduced_obj\n",
      "5587215531 \n",
      "\n",
      "size pData(reduced_obj)\n",
      "15531 67\n",
      "\n",
      "length tissues in phenotype data\n",
      "54\n",
      "\n",
      "TEST: how many to keep in to have only breast_mammary_tissue\n",
      "15072 459\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55872 459\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "459 67\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-1117F-2826-SM-5GZXL</th><td>GTEX-1117F-2826-SM-5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>breast_mammary_tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>50.2455</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-1117F-2826-SM-5GZXL & GTEX-1117F-2826-SM-5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & breast\\_mammary\\_tissue & 0008367 &  1340 & 1008 & ⋯ & 50.2455 & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;fct&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-1117F-2826-SM-5GZXL | GTEX-1117F-2826-SM-5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | breast_mammary_tissue | 0008367 |  1340 | 1008 | ⋯ | 50.2455 | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX-1117F-2826-SM-5GZXL GTEX-1117F-2826-SM-5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX-1117F-2826-SM-5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                 SMUBRID SMTSISCH\n",
       "GTEX-1117F-2826-SM-5GZXL  5.80000 Breast breast_mammary_tissue 0008367  1340   \n",
       "                         SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX-1117F-2826-SM-5GZXL 1008    ⋯ 50.2455  0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX-1117F-2826-SM-5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read in all requirements so that the stage is properly set -- \n",
    "# if it is clear here -- it will remain clear for the rest of the time\n",
    "# tissues.tsv contains the subset of files desired for analysis.\n",
    "tissue_reduction <- read.table(file=\"../assets/tissues.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "\n",
    "message(\"\\nsize tissue_reduction\\n\",\n",
    "        paste(dim(tissue_reduction), collapse=\" \"))\n",
    "message(\"\\nsize obj\\n\",\n",
    "        paste(dim(obj)), collapse=\"\\n\")\n",
    "message(\"\\nsize pData(obj)\\n\",\n",
    "        paste(dim(pData(obj)), collapse=\" \"))\n",
    "\n",
    "# only include those tissues we wish to continue with\n",
    "table(tissue_reduction$include)\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$include==1,]\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep <- (pData(obj)$SMTSD== \"breast_mammary_tissue\")\n",
    "message(\"\\nTEST: how many to keep of obj in breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = obj      [,keep]\n",
    "pData(tobj) = pData(obj[,keep])\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "# end test\n",
    "\n",
    "# create a matching tissue name to go with the expressionSet phenotype object\n",
    "pData(obj)$SMTSD       <- factor(snakecase::to_snake_case(as.character(pData(obj)$SMTSD)))\n",
    "tissue_reduction$SMTSD <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "\n",
    "message(\"\\nlength tissues in phenotype data\\n\",\n",
    "        paste(length(levels(pData(obj)$SMTSD)), collapse = \" \"))\n",
    "message(\"\\nlength tissues in tissue_reduction data\\n\",\n",
    "        paste(length(tissue_reduction$SMTSD), collapse = \" \"))\n",
    "\n",
    "keep <- pData(obj)$SMTSD %in% tissue_reduction$SMTSD\n",
    "message(\"\\nlength tissue in samples phenotype data\\n\",\n",
    "        paste(length(pData(obj)$SMTSD), collapse = \" \"))\n",
    "message(\"\\nlength keep obj \\n\",\n",
    "        paste(length(keep), collapse = \" \"))\n",
    "message(\"\\nhow many to keep in phenotype data\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "\n",
    "# both obj and pData(obj) need to be adjusted\n",
    "reduced_obj        <- obj       [          ,keep==TRUE]\n",
    "pData(reduced_obj) <- pData(obj)[keep==TRUE,          ]\n",
    "rm(keep)\n",
    "message(\"\\nsize reduced_obj\\n\",\n",
    "        paste(dim(reduced_obj)), collapse=\" \")\n",
    "message(\"\\nsize pData(reduced_obj)\\n\",\n",
    "        paste(dim(pData(reduced_obj)), collapse=\" \"))\n",
    "message(\"\\nlength tissues in phenotype data\\n\",\n",
    "        paste(length(levels(pData(reduced_obj)$SMTSD)), collapse = \" \"))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj)$SMTSD== \"breast_mammary_tissue\"\n",
    "message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = reduced_obj       [          ,keep==TRUE]\n",
    "pData(tobj) = pData(reduced_obj)[keep==TRUE,          ]\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Synchronize accession metadata and phenotype data\n",
    "A kind of transitive closure.   Metadata links the count data to the phenotype data.\n",
    "Begin with synchronizing accession metadata and phenotype data - which has been reduced - `reduced_obj` inputs here "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "size accession SraRunTable \n",
      "24667 80\n",
      "\n",
      "size reduced_obj\n",
      "55872 15531\n",
      "\n",
      "size pData(reduced_obj)\n",
      "15531 67\n",
      "\n",
      " any undefined (NA) sample names?\n",
      "0\n",
      "\n",
      " any undefined (NA) sample names?\n",
      "0\n",
      "\n",
      "length keep metadata samples in phenotype samples \n",
      "15531\n",
      "\n",
      "how many to keep in phenotype obj\n",
      "5159 10372\n",
      "\n",
      "size reduced_obj2\n",
      "55872 10372\n",
      "\n",
      "size pData(reduced_obj2)\n",
      "10372 67\n",
      "\n",
      "TEST: how many to keep in to have only breast_mammary_tissue\n",
      "10086 286\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55872 286\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "286 67\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX.1117F.2826.SM.5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>breast_mammary_tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>50.2455</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX.1117F.2826.SM.5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & breast\\_mammary\\_tissue & 0008367 &  1340 & 1008 & ⋯ & 50.2455 & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX.1117F.2826.SM.5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | breast_mammary_tissue | 0008367 |  1340 | 1008 | ⋯ | 50.2455 | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX.1117F.2826.SM.5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX.1117F.2826.SM.5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                 SMUBRID SMTSISCH\n",
       "GTEX.1117F.2826.SM.5GZXL  5.80000 Breast breast_mammary_tissue 0008367  1340   \n",
       "                         SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX.1117F.2826.SM.5GZXL 1008    ⋯ 50.2455  0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX.1117F.2826.SM.5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "length of metadata_samples\n",
      "24667\n",
      "\n",
      "length of phenotype_samples\n",
      "10372\n",
      "\n",
      "dimension of metadata\n",
      "24667 80\n",
      "\n",
      "length keep phenotype samples in metadata samples \n",
      "24667\n",
      "\n",
      "how many to keep in metadata samples\n",
      "\n",
      "\n",
      "dimension of reduced_metadata\n",
      "18214 80\n",
      "\n",
      "TEST: how many to keep in to have only breast_mammary_tissue\n",
      "10086 286\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55872 286\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "286 67\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX.1117F.2826.SM.5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>breast_mammary_tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>50.2455</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX.1117F.2826.SM.5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & breast\\_mammary\\_tissue & 0008367 &  1340 & 1008 & ⋯ & 50.2455 & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX.1117F.2826.SM.5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | breast_mammary_tissue | 0008367 |  1340 | 1008 | ⋯ | 50.2455 | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX.1117F.2826.SM.5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX.1117F.2826.SM.5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                 SMUBRID SMTSISCH\n",
       "GTEX.1117F.2826.SM.5GZXL  5.80000 Breast breast_mammary_tissue 0008367  1340   \n",
       "                         SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX.1117F.2826.SM.5GZXL 1008    ⋯ 50.2455  0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX.1117F.2826.SM.5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "17722   492 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "TEST: how many breast_mammary_tissue samples in reduced_metadata\n",
      "17722 492\n",
      "\n",
      "TEST: number reduced_metadata breast_samples\n",
      "492\n",
      "\n",
      "TEST: number phenotype obj breast_samples\n",
      "286\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR821498</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>7879119272</td><td>PRJNA75899</td><td>SAMN01994192</td><td>GTEx</td><td>GTEX-WI4N-1426-SM-3LK7H</td><td>Breast - Mammary Tissue</td><td>⋯</td><td></td><td></td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.WI4N.1426.SM.3LK7H</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR821498 & RNA:Total RNA & RNA-Seq & 152 & 7879119272 & PRJNA75899 & SAMN01994192 & GTEx & GTEX-WI4N-1426-SM-3LK7H & Breast - Mammary Tissue & ⋯ &  &  & NA & NA &  &  &  &  &  & GTEX.WI4N.1426.SM.3LK7H\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR821498 | RNA:Total RNA | RNA-Seq | 152 | 7879119272 | PRJNA75899 | SAMN01994192 | GTEx | GTEX-WI4N-1426-SM-3LK7H | Breast - Mammary Tissue | ⋯ | <!----> | <!----> | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.WI4N.1426.SM.3LK7H |\n",
       "\n"
      ],
      "text/plain": [
       "  Run       analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR821498 RNA:Total RNA RNA-Seq    152        7879119272 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN01994192 GTEx                   GTEX-WI4N-1426-SM-3LK7H         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯                                                    \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.WI4N.1426.SM.3LK7H"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# let's limit the phenotype object and then align the metadata file\n",
    "metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "metadata$SAMPID           <- gsub('-','\\\\.',metadata$'Sample Name')\n",
    "pData(reduced_obj)$SAMPID <- gsub('-','\\\\.',pData(reduced_obj)$SAMPID)\n",
    "\n",
    "message(\"\\nsize accession SraRunTable \\n\",\n",
    "        paste(dim(metadata), collapse=\" \"))\n",
    "message(\"\\nsize reduced_obj\\n\",\n",
    "        paste(dim(reduced_obj), collapse=\" \"))\n",
    "message(\"\\nsize pData(reduced_obj)\\n\",\n",
    "        paste(dim(pData(reduced_obj)), collapse=\" \"))\n",
    "rownames(pData(reduced_obj))<- pData(reduced_obj)$SAMPID\n",
    "\n",
    "# keep only those runs (as epitomized by the metadata_samples) in the phenotype set\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj)$SAMPID)\n",
    "\n",
    "# any undefined (N/A) sample names? These results will be zero\n",
    "message(\"\\n any undefined (NA) sample names?\\n\",\n",
    "        paste(sum(is.na(metadata_samples)), collapse=\" \"))\n",
    "message(\"\\n any undefined (NA) sample names?\\n\",\n",
    "        paste(sum(is.na(phenotype_samples)), collapse=\" \"))\n",
    "\n",
    "keep <- phenotype_samples %in% metadata_samples\n",
    "message(\"\\nlength keep metadata samples in phenotype samples \\n\",\n",
    "        paste(length(keep), collapse = \" \"))\n",
    "message(\"\\nhow many to keep in phenotype obj\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "\n",
    "reduced_obj2        <- reduced_obj       [          ,keep==TRUE]\n",
    "pData(reduced_obj2) <- pData(reduced_obj)[keep==TRUE,          ]\n",
    "message(\"\\nsize reduced_obj2\\n\",\n",
    "        paste(dim(reduced_obj2), collapse=\" \"))\n",
    "message(\"\\nsize pData(reduced_obj2)\\n\",\n",
    "        paste(dim(pData(reduced_obj2)), collapse=\" \"))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj2)$SMTSD == \"breast_mammary_tissue\"\n",
    "message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = reduced_obj2       [          ,keep==TRUE]\n",
    "pData(tobj) = pData(reduced_obj2)[keep==TRUE,          ]\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "# end test\n",
    "\n",
    "# now go the other way - make sure the metadata samples are in sync with the phenotype samples\n",
    "# note that we are now with `reduced_obj2`\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj2)$SAMPID)\n",
    "message(\"\\nlength of metadata_samples\\n\",\n",
    "        paste(length(metadata_samples), collapse=\" \"))\n",
    "message(\"\\nlength of phenotype_samples\\n\",\n",
    "        paste(length(phenotype_samples), collapse=\" \"))\n",
    "message(\"\\ndimension of metadata\\n\",\n",
    "        paste(dim(metadata), collapse=\" \"))\n",
    "keep <- metadata_samples %in% phenotype_samples\n",
    "message(\"\\nlength keep phenotype samples in metadata samples \\n\",\n",
    "        paste(length(keep), collapse = \" \"))\n",
    "message(\"\\nhow many to keep in metadata samples\\n\")\n",
    "\n",
    "reduced_metadata <- metadata[keep==TRUE,]\n",
    "message(\"\\ndimension of reduced_metadata\\n\",\n",
    "        paste(dim(reduced_metadata), collapse=\" \"))\n",
    "rm(keep)\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj2)$SMTSD== \"breast_mammary_tissue\"\n",
    "message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = reduced_obj2       [          ,keep==TRUE]\n",
    "pData(tobj) = pData(reduced_obj2)[keep==TRUE,          ]\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "\n",
    "breast_metadata_samples   <- as.character(reduced_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "message(\"\\nTEST: how many breast_mammary_tissue samples in reduced_metadata\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "breast_samples <- reduced_metadata[keep,]\n",
    "message(\"\\nTEST: number reduced_metadata breast_samples\\n\",\n",
    "        paste(length(breast_samples$SAMPID), collapse=\" \"))\n",
    "message(\"\\nTEST: number phenotype obj breast_samples\\n\",\n",
    "        paste(length(pData(tobj)$SAMPID), collapse=\" \"))\n",
    "breast_samples[1,]\n",
    "# end test\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  Limit obj and metadata to size of the ijc and sjc columns \n",
    "\n",
    "Things are complicated and can easily lead to data disaster.\n",
    "We need to go forward and backwards with these data - and see how to have them arranged.  \n",
    "Things we know to be true:\n",
    "1. Run numbers are unique\n",
    "2. Samples (which are Donor and Tissue combined) have been sequenced more than once.\n",
    "\n",
    "We need to keep this clear as the metadata are used to sort out tissues, sex, and other factors - and these are based upon a Donor.  Relationships are as follows:\n",
    "\n",
    "* One Donor can have many tissues (one donor has as many as 54 tissues)\n",
    "* One Tissue may have many sequencing runs (One Tissue has been as of this writing sequenced as many as 3 times)\n",
    "\n",
    "The analyst must keep this information in mind as manipulations occur.\n",
    "\n",
    "Here we use `se.jc.ijc` as exemplar as well as model for all subsequent matrices.  Arrangements of `se.jc.ijc` will ripple to all other matrices because sample order is the same between all samples.\n",
    "\n",
    "Entering into this step we have for our phenotype object:\n",
    "1. reduced_obj2\n",
    "2. reduced_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "metadata_for_counts\n",
       "FALSE  TRUE \n",
       " 1653  7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "counts_for_metadata\n",
       "FALSE  TRUE \n",
       "11194  7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>8673</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 8673\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 8673\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  8673"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>18214</li><li>80</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 18214\n",
       "\\item 80\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 18214\n",
       "2. 80\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 18214    80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SRR821653'</li><li>'SRR821690'</li><li>'SRR821715'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SRR821653'\n",
       "\\item 'SRR821690'\n",
       "\\item 'SRR821715'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SRR821653'\n",
       "2. 'SRR821690'\n",
       "3. 'SRR821715'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SRR821653\" \"SRR821690\" \"SRR821715\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SRR821653'</li><li>'SRR821690'</li><li>'SRR821715'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SRR821653'\n",
       "\\item 'SRR821690'\n",
       "\\item 'SRR821715'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SRR821653'\n",
       "2. 'SRR821690'\n",
       "3. 'SRR821715'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SRR821653\" \"SRR821690\" \"SRR821715\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "6891"
      ],
      "text/latex": [
       "6891"
      ],
      "text/markdown": [
       "6891"
      ],
      "text/plain": [
       "[1] 6891"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "15"
      ],
      "text/latex": [
       "15"
      ],
      "text/markdown": [
       "15"
      ],
      "text/plain": [
       "[1] 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "33"
      ],
      "text/latex": [
       "33"
      ],
      "text/markdown": [
       "33"
      ],
      "text/plain": [
       "[1] 33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata_for_counts = colnames(se.jc.ijc)%in% reduced_metadata$Run\n",
    "counts_for_metadata = reduced_metadata$Run %in% colnames(se.jc.ijc)\n",
    "table(metadata_for_counts)\n",
    "table(counts_for_metadata)\n",
    "\n",
    "counts          <- data.matrix(se.jc.ijc)\n",
    "counts_metadata <- reduced_metadata\n",
    "dim(counts)\n",
    "dim(counts_metadata)\n",
    "\n",
    "counts          <- counts         [                         ,metadata_for_counts==TRUE]\n",
    "counts_metadata <- counts_metadata[counts_for_metadata==TRUE,                         ]\n",
    "\n",
    "# now we arrange metadata and counts to be in the same order\n",
    "# Run numbers are unique, but we will replace the columns with SAMPID as metadata is by SAMPID\n",
    "i = order(colnames(counts))\n",
    "j = order(counts_metadata$Run)\n",
    "\n",
    "ose.jc.ijc       = counts[,i]\n",
    "ocounts_metadata = counts_metadata[j,]\n",
    "\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(ocounts_metadata$Run,3)\n",
    "\n",
    "colnames(ose.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(ocounts_metadata$SAMPID,3)\n",
    "sum(table(colnames(ose.jc.ijc))==1)\n",
    "sum(table(colnames(ose.jc.ijc))==2)\n",
    "sum(table(colnames(ose.jc.ijc))==3)\n",
    "sum(table(colnames(ose.jc.ijc))==4)\n",
    "dim(ose.jc.ijc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now extend this organization to all count matrices\n",
    "now that the data are organized and accounted for, extend to the rest of the splicing factors and their count matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td> 2</td><td>2</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>10</td><td>0</td><td>4</td><td>4</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 0 &  2 & 2 & 0 & 0\\\\\n",
       "\t 0 & 10 & 0 & 4 & 4\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\t 0 &  0 & 2 & 0 & 1\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 0 |  2 | 2 | 0 | 0 |\n",
       "| 0 | 10 | 0 | 4 | 4 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "| 0 |  0 | 2 | 0 | 1 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 0           2         2          0          0         \n",
       "[2,] 0          10         0          4          4         \n",
       "[3,] 0           0         1          0          1         \n",
       "[4,] 0           0         2          0          1         \n",
       "[5,] 0           0         1          0          1         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>151</td><td>51</td><td>292</td><td>59</td><td>91</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 151 & 51 & 292 & 59 & 91\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 151 | 51 | 292 | 59 | 91 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 151        51         292        59         91        \n",
       "[2,]   0         0           0         0          0        \n",
       "[3,]   0         0           0         0          0        \n",
       "[4,]   0         0           0         0          0        \n",
       "[5,]   0         0           0         0          0        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  1</td><td> 2</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td> 0</td></tr>\n",
       "\t<tr><td>14</td><td> 3</td><td>  4</td><td> 3</td><td>11</td></tr>\n",
       "\t<tr><td>67</td><td>10</td><td>167</td><td>26</td><td>31</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   1 &  2 &  0\\\\\n",
       "\t  0 &  0 &   0 &  0 &  0\\\\\n",
       "\t  0 &  0 &   0 &  1 &  0\\\\\n",
       "\t 14 &  3 &   4 &  3 & 11\\\\\n",
       "\t 67 & 10 & 167 & 26 & 31\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   1 |  2 |  0 |\n",
       "|  0 |  0 |   0 |  0 |  0 |\n",
       "|  0 |  0 |   0 |  1 |  0 |\n",
       "| 14 |  3 |   4 |  3 | 11 |\n",
       "| 67 | 10 | 167 | 26 | 31 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]  0          0           1         2          0        \n",
       "[2,]  0          0           0         0          0        \n",
       "[3,]  0          0           0         1          0        \n",
       "[4,] 14          3           4         3         11        \n",
       "[5,] 67         10         167        26         31        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>79</td><td>30</td><td>293</td><td>120</td><td>107</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>13</td><td>12</td><td> 10</td><td>  8</td><td> 18</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 79 & 30 & 293 & 120 & 107\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t 13 & 12 &  10 &   8 &  18\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 79 | 30 | 293 | 120 | 107 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "| 13 | 12 |  10 |   8 |  18 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 79         30         293        120        107       \n",
       "[2,]  0          0           0          0          0       \n",
       "[3,]  0          0           0          0          0       \n",
       "[4,] 13         12          10          8         18       \n",
       "[5,]  0          0           0          0          0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>1</td><td> 4</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td> 0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>1</td><td>0</td><td>0</td><td>3</td><td> 2</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>7</td><td>32</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 5 & 3 & 4 & 1 &  4\\\\\n",
       "\t 0 & 0 & 0 & 1 &  0\\\\\n",
       "\t 0 & 0 & 0 & 0 &  0\\\\\n",
       "\t 1 & 0 & 0 & 3 &  2\\\\\n",
       "\t 0 & 0 & 0 & 7 & 32\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 5 | 3 | 4 | 1 |  4 |\n",
       "| 0 | 0 | 0 | 1 |  0 |\n",
       "| 0 | 0 | 0 | 0 |  0 |\n",
       "| 1 | 0 | 0 | 3 |  2 |\n",
       "| 0 | 0 | 0 | 7 | 32 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 5          3          4          1           4        \n",
       "[2,] 0          0          0          1           0        \n",
       "[3,] 0          0          0          0           0        \n",
       "[4,] 1          0          0          3           2        \n",
       "[5,] 0          0          0          7          32        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>36</td><td>7</td><td>10</td><td>15</td><td> 1</td></tr>\n",
       "\t<tr><td> 0</td><td>4</td><td> 1</td><td> 2</td><td> 2</td></tr>\n",
       "\t<tr><td> 0</td><td>0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td>2</td><td> 1</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 4</td><td>8</td><td> 2</td><td>25</td><td>74</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 36 & 7 & 10 & 15 &  1\\\\\n",
       "\t  0 & 4 &  1 &  2 &  2\\\\\n",
       "\t  0 & 0 &  0 &  0 &  0\\\\\n",
       "\t  0 & 2 &  1 &  0 &  0\\\\\n",
       "\t  4 & 8 &  2 & 25 & 74\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 36 | 7 | 10 | 15 |  1 |\n",
       "|  0 | 4 |  1 |  2 |  2 |\n",
       "|  0 | 0 |  0 |  0 |  0 |\n",
       "|  0 | 2 |  1 |  0 |  0 |\n",
       "|  4 | 8 |  2 | 25 | 74 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 36         7          10         15          1        \n",
       "[2,]  0         4           1          2          2        \n",
       "[3,]  0         0           0          0          0        \n",
       "[4,]  0         2           1          0          0        \n",
       "[5,]  4         8           2         25         74        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>  5</td><td> 0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  1</td><td> 1</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td>  0</td><td> 2</td><td> 0</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  4</td><td> 7</td><td> 4</td><td>  2</td></tr>\n",
       "\t<tr><td>64</td><td>113</td><td>43</td><td>40</td><td>247</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t  0 &   5 &  0 &  1 &   1\\\\\n",
       "\t  0 &   1 &  1 &  0 &   0\\\\\n",
       "\t  0 &   0 &  2 &  0 &   1\\\\\n",
       "\t  0 &   4 &  7 &  4 &   2\\\\\n",
       "\t 64 & 113 & 43 & 40 & 247\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|  0 |   5 |  0 |  1 |   1 |\n",
       "|  0 |   1 |  1 |  0 |   0 |\n",
       "|  0 |   0 |  2 |  0 |   1 |\n",
       "|  0 |   4 |  7 |  4 |   2 |\n",
       "| 64 | 113 | 43 | 40 | 247 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]  0           5         0          1           1       \n",
       "[2,]  0           1         1          0           0       \n",
       "[3,]  0           0         2          0           1       \n",
       "[4,]  0           4         7          4           2       \n",
       "[5,] 64         113        43         40         247       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>98</td><td>25</td><td>150</td><td>64</td><td> 37</td></tr>\n",
       "\t<tr><td>11</td><td>34</td><td> 56</td><td>38</td><td>125</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   0 &  1 &   1\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t 98 & 25 & 150 & 64 &  37\\\\\n",
       "\t 11 & 34 &  56 & 38 & 125\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   0 |  1 |   1 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "| 98 | 25 | 150 | 64 |  37 |\n",
       "| 11 | 34 |  56 | 38 | 125 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]  0          0           0         1           1       \n",
       "[2,]  0          0           0         0           0       \n",
       "[3,]  0          0           0         0           0       \n",
       "[4,] 98         25         150        64          37       \n",
       "[5,] 11         34          56        38         125       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>247</td><td>103</td><td>494</td><td>145</td><td>145</td></tr>\n",
       "\t<tr><td>  0</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  1</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  0</td><td>  2</td><td>  6</td><td>  1</td><td>  1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t   0 &   0 &   1 &   0 &   0\\\\\n",
       "\t 247 & 103 & 494 & 145 & 145\\\\\n",
       "\t   0 &   1 &   0 &   0 &   1\\\\\n",
       "\t   1 &   1 &   0 &   0 &   1\\\\\n",
       "\t   0 &   2 &   6 &   1 &   1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "|   0 |   0 |   1 |   0 |   0 |\n",
       "| 247 | 103 | 494 | 145 | 145 |\n",
       "|   0 |   1 |   0 |   0 |   1 |\n",
       "|   1 |   1 |   0 |   0 |   1 |\n",
       "|   0 |   2 |   6 |   1 |   1 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,]   0          0          1          0          0       \n",
       "[2,] 247        103        494        145        145       \n",
       "[3,]   0          1          0          0          1       \n",
       "[4,]   1          1          0          0          1       \n",
       "[5,]   0          2          6          1          1       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SRR1068808</th><th scope=col>SRR1068832</th><th scope=col>SRR1068880</th><th scope=col>SRR1068929</th><th scope=col>SRR1068953</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td>1</td><td>6</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>0</td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " SRR1068808 & SRR1068832 & SRR1068880 & SRR1068929 & SRR1068953\\\\\n",
       "\\hline\n",
       "\t 0 & 1 & 6 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 1\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 5 & 3 & 4 & 0 & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| SRR1068808 | SRR1068832 | SRR1068880 | SRR1068929 | SRR1068953 |\n",
       "|---|---|---|---|---|\n",
       "| 0 | 1 | 6 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 1 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 5 | 3 | 4 | 0 | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "     SRR1068808 SRR1068832 SRR1068880 SRR1068929 SRR1068953\n",
       "[1,] 0          1          6          1          0         \n",
       "[2,] 0          0          0          0          1         \n",
       "[3,] 0          0          0          0          0         \n",
       "[4,] 0          0          0          0          0         \n",
       "[5,] 5          3          4          0          3         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GTEX.WOFL.0006.SM.3TW8K'</li><li>'GTEX.NPJ8.0011.R1a.SM.33HCB'</li><li>'GTEX.OHPK.0226.SM.3MJH6'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "\\item 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "\\item 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GTEX.WOFL.0006.SM.3TW8K'\n",
       "2. 'GTEX.NPJ8.0011.R1a.SM.33HCB'\n",
       "3. 'GTEX.OHPK.0226.SM.3MJH6'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GTEX.WOFL.0006.SM.3TW8K\"     \"GTEX.NPJ8.0011.R1a.SM.33HCB\"\n",
       "[3] \"GTEX.OHPK.0226.SM.3MJH6\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "oa3ss.jc.ijc  <- data.matrix(a3ss.jc.ijc)\n",
    "oa3ss.jc.sjc  <- data.matrix(a3ss.jc.sjc)\n",
    "oa3ss.jc.ijc  <- oa3ss.jc.ijc[,metadata_for_counts==TRUE]\n",
    "oa3ss.jc.sjc  <- oa3ss.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "oa5ss.jc.ijc  <- data.matrix(a5ss.jc.ijc)\n",
    "oa5ss.jc.sjc  <- data.matrix(a5ss.jc.sjc)\n",
    "oa5ss.jc.ijc  <- oa5ss.jc.ijc[,metadata_for_counts==TRUE]\n",
    "oa5ss.jc.sjc  <- oa5ss.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "omxe.jc.ijc   <- data.matrix(mxe.jc.ijc)\n",
    "omxe.jc.sjc   <- data.matrix(mxe.jc.sjc)\n",
    "omxe.jc.ijc   <- omxe.jc.ijc[,metadata_for_counts==TRUE]\n",
    "omxe.jc.sjc   <- omxe.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "ori.jc.ijc    <- data.matrix(ri.jc.ijc)\n",
    "ori.jc.sjc    <- data.matrix(ri.jc.sjc)\n",
    "ori.jc.ijc    <- ori.jc.ijc[,metadata_for_counts==TRUE]\n",
    "ori.jc.sjc    <- ori.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "ose.jc.ijc    <- data.matrix(se.jc.ijc)\n",
    "ose.jc.sjc    <- data.matrix(se.jc.sjc)\n",
    "ose.jc.ijc    <- ose.jc.ijc[,metadata_for_counts==TRUE]\n",
    "ose.jc.sjc    <- ose.jc.sjc[,metadata_for_counts==TRUE]\n",
    "\n",
    "oa3ss.jc.ijc  <- oa3ss.jc.ijc[,i]\n",
    "oa3ss.jc.sjc  <- oa3ss.jc.sjc[,i]\n",
    "\n",
    "oa5ss.jc.ijc  <- oa5ss.jc.ijc[,i]\n",
    "oa5ss.jc.sjc  <- oa5ss.jc.sjc[,i]\n",
    "\n",
    "omxe.jc.ijc   <- omxe.jc.ijc[,i]\n",
    "omxe.jc.sjc   <- omxe.jc.sjc[,i]\n",
    "\n",
    "ori.jc.ijc    <- ori.jc.ijc[,i]\n",
    "ori.jc.sjc    <- ori.jc.sjc[,i]\n",
    "\n",
    "ose.jc.ijc    <- ose.jc.ijc  [,i]\n",
    "ose.jc.sjc    <- ose.jc.sjc  [,i]\n",
    "\n",
    "oa3ss.jc.ijc[1:5,1:5]\n",
    "oa3ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "oa5ss.jc.ijc[1:5,1:5]\n",
    "oa5ss.jc.sjc[1:5,1:5]\n",
    "\n",
    "omxe.jc.ijc[1:5,1:5]\n",
    "omxe.jc.sjc[1:5,1:5]\n",
    "\n",
    "ori.jc.ijc[1:5,1:5]\n",
    "ori.jc.sjc[1:5,1:5]\n",
    "\n",
    "ose.jc.ijc[1:5,1:5]\n",
    "ose.jc.sjc[1:5,1:5]\n",
    "\n",
    "colnames(oa3ss.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(oa3ss.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(oa5ss.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(oa5ss.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(omxe.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(omxe.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(ori.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(ori.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "colnames(ose.jc.ijc) = as.character(ocounts_metadata$SAMPID)\n",
    "colnames(ose.jc.sjc) = as.character(ocounts_metadata$SAMPID)\n",
    "\n",
    "tail(colnames(oa3ss.jc.ijc),3)\n",
    "tail(colnames(oa3ss.jc.sjc),3)\n",
    "\n",
    "tail(colnames(oa5ss.jc.ijc),3)\n",
    "tail(colnames(oa5ss.jc.sjc),3)\n",
    "\n",
    "tail(colnames(omxe.jc.ijc),3)\n",
    "tail(colnames(omxe.jc.sjc),3)\n",
    "\n",
    "tail(colnames(ori.jc.ijc),3)\n",
    "tail(colnames(ori.jc.sjc),3)\n",
    "\n",
    "tail(colnames(ose.jc.ijc),3)\n",
    "tail(colnames(ose.jc.sjc),3)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sanity Check\n",
    "\n",
    "Spot check one tissue and make sure there isn't nonsense\n",
    "Our current matrices are:\n",
    "1. ocounts_metadata for the connection from counts to phenotype via the Run and SAMPID\n",
    "2. pData(reduced_obj2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "10086   286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55872</dd><dt>Samples</dt><dd>286</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55872\n",
       "\\item[Samples] 286\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55872Samples\n",
       ":   286\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55872      286 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>286</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 286\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 286\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 286  67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       " 6842   178 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "178"
      ],
      "text/latex": [
       "178"
      ],
      "text/markdown": [
       "178"
      ],
      "text/plain": [
       "[1] 178"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "286"
      ],
      "text/latex": [
       "286"
      ],
      "text/markdown": [
       "286"
      ],
      "text/plain": [
       "[1] 286"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR1068977</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>6706681864</td><td>PRJNA75899</td><td>SAMN02465522</td><td>GTEx</td><td>GTEX-XBED-1626-SM-47JYN</td><td>Breast - Mammary Tissue</td><td>⋯</td><td>P-RNA-0002</td><td>P-RNA-0002</td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.XBED.1626.SM.47JYN</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR1068977 & RNA:Total RNA & RNA-Seq & 152 & 6706681864 & PRJNA75899 & SAMN02465522 & GTEx & GTEX-XBED-1626-SM-47JYN & Breast - Mammary Tissue & ⋯ & P-RNA-0002 & P-RNA-0002 & NA & NA &  &  &  &  &  & GTEX.XBED.1626.SM.47JYN\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR1068977 | RNA:Total RNA | RNA-Seq | 152 | 6706681864 | PRJNA75899 | SAMN02465522 | GTEx | GTEX-XBED-1626-SM-47JYN | Breast - Mammary Tissue | ⋯ | P-RNA-0002 | P-RNA-0002 | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.XBED.1626.SM.47JYN |\n",
       "\n"
      ],
      "text/plain": [
       "  Run        analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR1068977 RNA:Total RNA RNA-Seq    152        6706681864 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN02465522 GTEx                   GTEX-XBED-1626-SM-47JYN         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯ P-RNA-0002                P-RNA-0002               \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.XBED.1626.SM.47JYN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# test to make sure we don't have nonsense\n",
    "rm(keep)\n",
    "keep = pData(reduced_obj2)$SMTSD == \"breast_mammary_tissue\"\n",
    "table(keep)\n",
    "tobj = reduced_obj2[,keep]\n",
    "dim(tobj)\n",
    "dim(pData(tobj))\n",
    "breast_metadata_samples   <- as.character(ocounts_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "rm(keep)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "breast_samples <- ocounts_metadata[keep,]\n",
    "length(breast_samples$SAMPID)\n",
    "length(pData(tobj)$SAMPID)\n",
    "breast_samples[1,]\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### which metadata shall we use?\n",
    "There is a lack of correspondeence between the metadata in the Sequence Read Archive table as obtained from dbGaP SRR Run Explorer and the GTEx V8 Release.   Not sure which one to use!  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "20-29 30-39 40-49 50-59 60-69 70-79 \n",
       "  760   771  1729  3484  3312   316 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "   0    1    2    3    4 \n",
       "5523  467 2738  466 1082 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "        B1     B1, A1         C1     C1, A1 C1, B1, A1         D1     D1, A1 \n",
       "      6154        377       3110        603          3         70         55 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "   1    2 \n",
       "6796 3576 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "female   male \n",
       "  2501   4519 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(pData(reduced_obj2)$AGE)\n",
    "table(pData(reduced_obj2)$DTHHRDY)\n",
    "table(pData(reduced_obj2)$SMCENTER)\n",
    "table(pData(reduced_obj2)$SEX)\n",
    "table(ocounts_metadata$sex)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Exploring the ijc and sjc Count data \n",
    "\n",
    "For each sample, we have ijc and sjc count data, for each alternative splicing type, we have the following number of alternative splicing events:\n",
    "\n",
    "* `A3SS`: ` 8,920`\n",
    "* `A5SS`: ` 5,584`\n",
    "* `MXE `: ` 2,979`\n",
    "* `RI`  : ` 6,312`\n",
    "* `SE  `: `42,611`\n",
    "\n",
    "For exon skipping events (SE), we have 42,611 non-zero junction IDs the (first dimension of the ijc and sjc cout table) for the skipped exon event for breast-Mammary Tissue, 191 individuals.  These are healthy individuals, and we are studying the impact of sex on the occurrence or non-occurance of specific alternative splicing events.   We explore the information we ahve about these junctions and create a construct, as_event, which accounts for the junction under exploration.\n",
    "\n",
    "The `IJC` and `SJC` counts are in many ways two sides of the same coin.  Both our the observational output and we wish to see how robust each are in their ability to separate out the samples to provide for us differentially expressed isoform events as measured by their counts.   Each junction is in a manner a specific marker to specific isoform events that may or may not be shared between the genders.   If there is significant results, then this is indicative of the separation achieved by isoform specific differentiation.   In our model we will use these in combination, it is important to see if they will yield the results we are looking for.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reduce count matrices based upon the reduced tissue information - count matrix by column adjustment\n",
    "\n",
    "We wish to keep only those samples (columns) in our count data that are a member of the reduced tissue set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "keep\n",
       "TRUE \n",
       "7020 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7020"
      ],
      "text/latex": [
       "7020"
      ],
      "text/markdown": [
       "7020"
      ],
      "text/plain": [
       "[1] 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td> 2</td><td>2</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>10</td><td>0</td><td>4</td><td>4</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td> 0</td><td>1</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 0 &  2 & 2 & 0 & 0\\\\\n",
       "\t 0 & 10 & 0 & 4 & 4\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\t 0 &  0 & 2 & 0 & 1\\\\\n",
       "\t 0 &  0 & 1 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 0 |  2 | 2 | 0 | 0 |\n",
       "| 0 | 10 | 0 | 4 | 4 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "| 0 |  0 | 2 | 0 | 1 |\n",
       "| 0 |  0 | 1 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 0                        2                      2                      \n",
       "[2,] 0                       10                      0                      \n",
       "[3,] 0                        0                      1                      \n",
       "[4,] 0                        0                      2                      \n",
       "[5,] 0                        0                      1                      \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 0                       0                      \n",
       "[2,] 4                       4                      \n",
       "[3,] 0                       1                      \n",
       "[4,] 0                       1                      \n",
       "[5,] 0                       1                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>151</td><td>51</td><td>292</td><td>59</td><td>91</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>  0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 151 & 51 & 292 & 59 & 91\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\t   0 &  0 &   0 &  0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 151 | 51 | 292 | 59 | 91 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "|   0 |  0 |   0 |  0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 151                     51                      292                    \n",
       "[2,]   0                      0                        0                    \n",
       "[3,]   0                      0                        0                    \n",
       "[4,]   0                      0                        0                    \n",
       "[5,]   0                      0                        0                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 59                      91                     \n",
       "[2,]  0                       0                     \n",
       "[3,]  0                       0                     \n",
       "[4,]  0                       0                     \n",
       "[5,]  0                       0                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  1</td><td> 2</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td> 0</td></tr>\n",
       "\t<tr><td>14</td><td> 3</td><td>  4</td><td> 3</td><td>11</td></tr>\n",
       "\t<tr><td>67</td><td>10</td><td>167</td><td>26</td><td>31</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   1 &  2 &  0\\\\\n",
       "\t  0 &  0 &   0 &  0 &  0\\\\\n",
       "\t  0 &  0 &   0 &  1 &  0\\\\\n",
       "\t 14 &  3 &   4 &  3 & 11\\\\\n",
       "\t 67 & 10 & 167 & 26 & 31\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   1 |  2 |  0 |\n",
       "|  0 |  0 |   0 |  0 |  0 |\n",
       "|  0 |  0 |   0 |  1 |  0 |\n",
       "| 14 |  3 |   4 |  3 | 11 |\n",
       "| 67 | 10 | 167 | 26 | 31 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]  0                       0                        1                    \n",
       "[2,]  0                       0                        0                    \n",
       "[3,]  0                       0                        0                    \n",
       "[4,] 14                       3                        4                    \n",
       "[5,] 67                      10                      167                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]  2                       0                     \n",
       "[2,]  0                       0                     \n",
       "[3,]  1                       0                     \n",
       "[4,]  3                      11                     \n",
       "[5,] 26                      31                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>79</td><td>30</td><td>293</td><td>120</td><td>107</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>13</td><td>12</td><td> 10</td><td>  8</td><td> 18</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 79 & 30 & 293 & 120 & 107\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\t 13 & 12 &  10 &   8 &  18\\\\\n",
       "\t  0 &  0 &   0 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 79 | 30 | 293 | 120 | 107 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "| 13 | 12 |  10 |   8 |  18 |\n",
       "|  0 |  0 |   0 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 79                      30                      293                    \n",
       "[2,]  0                       0                        0                    \n",
       "[3,]  0                       0                        0                    \n",
       "[4,] 13                      12                       10                    \n",
       "[5,]  0                       0                        0                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 120                     107                    \n",
       "[2,]   0                       0                    \n",
       "[3,]   0                       0                    \n",
       "[4,]   8                      18                    \n",
       "[5,]   0                       0                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>1</td><td> 4</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>1</td><td> 0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td> 0</td></tr>\n",
       "\t<tr><td>1</td><td>0</td><td>0</td><td>3</td><td> 2</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>7</td><td>32</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 5 & 3 & 4 & 1 &  4\\\\\n",
       "\t 0 & 0 & 0 & 1 &  0\\\\\n",
       "\t 0 & 0 & 0 & 0 &  0\\\\\n",
       "\t 1 & 0 & 0 & 3 &  2\\\\\n",
       "\t 0 & 0 & 0 & 7 & 32\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 5 | 3 | 4 | 1 |  4 |\n",
       "| 0 | 0 | 0 | 1 |  0 |\n",
       "| 0 | 0 | 0 | 0 |  0 |\n",
       "| 1 | 0 | 0 | 3 |  2 |\n",
       "| 0 | 0 | 0 | 7 | 32 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 5                       3                       4                      \n",
       "[2,] 0                       0                       0                      \n",
       "[3,] 0                       0                       0                      \n",
       "[4,] 1                       0                       0                      \n",
       "[5,] 0                       0                       0                      \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 1                        4                     \n",
       "[2,] 1                        0                     \n",
       "[3,] 0                        0                     \n",
       "[4,] 3                        2                     \n",
       "[5,] 7                       32                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>36</td><td>7</td><td>10</td><td>15</td><td> 1</td></tr>\n",
       "\t<tr><td> 0</td><td>4</td><td> 1</td><td> 2</td><td> 2</td></tr>\n",
       "\t<tr><td> 0</td><td>0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 0</td><td>2</td><td> 1</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td> 4</td><td>8</td><td> 2</td><td>25</td><td>74</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 36 & 7 & 10 & 15 &  1\\\\\n",
       "\t  0 & 4 &  1 &  2 &  2\\\\\n",
       "\t  0 & 0 &  0 &  0 &  0\\\\\n",
       "\t  0 & 2 &  1 &  0 &  0\\\\\n",
       "\t  4 & 8 &  2 & 25 & 74\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 36 | 7 | 10 | 15 |  1 |\n",
       "|  0 | 4 |  1 |  2 |  2 |\n",
       "|  0 | 0 |  0 |  0 |  0 |\n",
       "|  0 | 2 |  1 |  0 |  0 |\n",
       "|  4 | 8 |  2 | 25 | 74 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 36                      7                       10                     \n",
       "[2,]  0                      4                        1                     \n",
       "[3,]  0                      0                        0                     \n",
       "[4,]  0                      2                        1                     \n",
       "[5,]  4                      8                        2                     \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 15                       1                     \n",
       "[2,]  2                       2                     \n",
       "[3,]  0                       0                     \n",
       "[4,]  0                       0                     \n",
       "[5,] 25                      74                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td>  5</td><td> 0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  1</td><td> 1</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td>  0</td><td> 2</td><td> 0</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td>  4</td><td> 7</td><td> 4</td><td>  2</td></tr>\n",
       "\t<tr><td>64</td><td>113</td><td>43</td><td>40</td><td>247</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t  0 &   5 &  0 &  1 &   1\\\\\n",
       "\t  0 &   1 &  1 &  0 &   0\\\\\n",
       "\t  0 &   0 &  2 &  0 &   1\\\\\n",
       "\t  0 &   4 &  7 &  4 &   2\\\\\n",
       "\t 64 & 113 & 43 & 40 & 247\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|  0 |   5 |  0 |  1 |   1 |\n",
       "|  0 |   1 |  1 |  0 |   0 |\n",
       "|  0 |   0 |  2 |  0 |   1 |\n",
       "|  0 |   4 |  7 |  4 |   2 |\n",
       "| 64 | 113 | 43 | 40 | 247 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]  0                        5                      0                     \n",
       "[2,]  0                        1                      1                     \n",
       "[3,]  0                        0                      2                     \n",
       "[4,]  0                        4                      7                     \n",
       "[5,] 64                      113                     43                     \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]  1                        1                    \n",
       "[2,]  0                        0                    \n",
       "[3,]  0                        1                    \n",
       "[4,]  4                        2                    \n",
       "[5,] 40                      247                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 1</td><td>  1</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td> 0</td><td> 0</td><td>  0</td><td> 0</td><td>  0</td></tr>\n",
       "\t<tr><td>98</td><td>25</td><td>150</td><td>64</td><td> 37</td></tr>\n",
       "\t<tr><td>11</td><td>34</td><td> 56</td><td>38</td><td>125</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t  0 &  0 &   0 &  1 &   1\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t  0 &  0 &   0 &  0 &   0\\\\\n",
       "\t 98 & 25 & 150 & 64 &  37\\\\\n",
       "\t 11 & 34 &  56 & 38 & 125\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|  0 |  0 |   0 |  1 |   1 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "|  0 |  0 |   0 |  0 |   0 |\n",
       "| 98 | 25 | 150 | 64 |  37 |\n",
       "| 11 | 34 |  56 | 38 | 125 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]  0                       0                        0                    \n",
       "[2,]  0                       0                        0                    \n",
       "[3,]  0                       0                        0                    \n",
       "[4,] 98                      25                      150                    \n",
       "[5,] 11                      34                       56                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]  1                        1                    \n",
       "[2,]  0                        0                    \n",
       "[3,]  0                        0                    \n",
       "[4,] 64                       37                    \n",
       "[5,] 38                      125                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "\t<tr><td>247</td><td>103</td><td>494</td><td>145</td><td>145</td></tr>\n",
       "\t<tr><td>  0</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  1</td><td>  1</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "\t<tr><td>  0</td><td>  2</td><td>  6</td><td>  1</td><td>  1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t   0 &   0 &   1 &   0 &   0\\\\\n",
       "\t 247 & 103 & 494 & 145 & 145\\\\\n",
       "\t   0 &   1 &   0 &   0 &   1\\\\\n",
       "\t   1 &   1 &   0 &   0 &   1\\\\\n",
       "\t   0 &   2 &   6 &   1 &   1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "|   0 |   0 |   1 |   0 |   0 |\n",
       "| 247 | 103 | 494 | 145 | 145 |\n",
       "|   0 |   1 |   0 |   0 |   1 |\n",
       "|   1 |   1 |   0 |   0 |   1 |\n",
       "|   0 |   2 |   6 |   1 |   1 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,]   0                       0                       1                    \n",
       "[2,] 247                     103                     494                    \n",
       "[3,]   0                       1                       0                    \n",
       "[4,]   1                       1                       0                    \n",
       "[5,]   0                       2                       6                    \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,]   0                       0                    \n",
       "[2,] 145                     145                    \n",
       "[3,]   0                       1                    \n",
       "[4,]   0                       1                    \n",
       "[5,]   1                       1                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 5 × 5 of type int</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GTEX.QV44.0008.SM.447AX</th><th scope=col>GTEX.UJMC.0426.SM.4IHJF</th><th scope=col>GTEX.O5YT.0008.SM.4E3IQ</th><th scope=col>GTEX.PWN1.0726.SM.48TDR</th><th scope=col>GTEX.PWN1.1526.SM.48TDA</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0</td><td>1</td><td>6</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>5</td><td>3</td><td>4</td><td>0</td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 5 × 5 of type int\n",
       "\\begin{tabular}{lllll}\n",
       " GTEX.QV44.0008.SM.447AX & GTEX.UJMC.0426.SM.4IHJF & GTEX.O5YT.0008.SM.4E3IQ & GTEX.PWN1.0726.SM.48TDR & GTEX.PWN1.1526.SM.48TDA\\\\\n",
       "\\hline\n",
       "\t 0 & 1 & 6 & 1 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 1\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t 5 & 3 & 4 & 0 & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 5 × 5 of type int\n",
       "\n",
       "| GTEX.QV44.0008.SM.447AX | GTEX.UJMC.0426.SM.4IHJF | GTEX.O5YT.0008.SM.4E3IQ | GTEX.PWN1.0726.SM.48TDR | GTEX.PWN1.1526.SM.48TDA |\n",
       "|---|---|---|---|---|\n",
       "| 0 | 1 | 6 | 1 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 1 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 0 | 0 | 0 | 0 | 0 |\n",
       "| 5 | 3 | 4 | 0 | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "     GTEX.QV44.0008.SM.447AX GTEX.UJMC.0426.SM.4IHJF GTEX.O5YT.0008.SM.4E3IQ\n",
       "[1,] 0                       1                       6                      \n",
       "[2,] 0                       0                       0                      \n",
       "[3,] 0                       0                       0                      \n",
       "[4,] 0                       0                       0                      \n",
       "[5,] 5                       3                       4                      \n",
       "     GTEX.PWN1.0726.SM.48TDR GTEX.PWN1.1526.SM.48TDA\n",
       "[1,] 1                       0                      \n",
       "[2,] 0                       1                      \n",
       "[3,] 0                       0                      \n",
       "[4,] 0                       0                      \n",
       "[5,] 0                       3                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata_names <- as.character(ocounts_metadata$SAMPID)\n",
    "count_names    <- as.character(colnames(ose.jc.ijc))\n",
    "\n",
    "keep  <- count_names %in% metadata_names\n",
    "\n",
    "table(keep)\n",
    "length(keep)\n",
    "length(metadata_names)\n",
    "length(count_names)\n",
    "\n",
    "a3ss.jc.ijc2 = oa3ss.jc.ijc[,keep==TRUE] \n",
    "a3ss.jc.sjc2 = oa3ss.jc.sjc[,keep==TRUE] \n",
    "\n",
    "a5ss.jc.ijc2 = oa5ss.jc.ijc[,keep==TRUE] \n",
    "a5ss.jc.sjc2 = oa5ss.jc.sjc[,keep==TRUE] \n",
    "\n",
    "mxe.jc.ijc2 = omxe.jc.ijc[,keep==TRUE] \n",
    "mxe.jc.sjc2 = omxe.jc.sjc[,keep==TRUE] \n",
    "\n",
    "ri.jc.ijc2 = ori.jc.ijc[,keep==TRUE] \n",
    "ri.jc.sjc2 = ori.jc.sjc[,keep==TRUE] \n",
    "\n",
    "se.jc.ijc2 = ose.jc.ijc[,keep==TRUE] \n",
    "se.jc.sjc2 = ose.jc.sjc[,keep==TRUE] \n",
    "\n",
    "dim(a3ss.jc.ijc2)\n",
    "dim(a3ss.jc.sjc2)\n",
    "dim(a5ss.jc.ijc2)\n",
    "dim(a5ss.jc.sjc2)\n",
    "dim(mxe.jc.ijc2)\n",
    "dim(mxe.jc.sjc2)\n",
    "dim(ri.jc.ijc2)\n",
    "dim(ri.jc.sjc2)\n",
    "dim(se.jc.ijc2)\n",
    "dim(se.jc.sjc2)\n",
    "\n",
    "a3ss.jc.ijc2[1:5,1:5]\n",
    "a3ss.jc.sjc2[1:5,1:5]\n",
    "\n",
    "a5ss.jc.ijc2[1:5,1:5]\n",
    "a5ss.jc.sjc2[1:5,1:5]\n",
    "\n",
    "mxe.jc.ijc2[1:5,1:5]\n",
    "mxe.jc.sjc2[1:5,1:5]\n",
    "\n",
    "ri.jc.ijc2[1:5,1:5]\n",
    "ri.jc.sjc2[1:5,1:5]\n",
    "\n",
    "se.jc.ijc2[1:5,1:5]\n",
    "se.jc.sjc2[1:5,1:5]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Keeping only chromosomes shared male female - count matrix by junction row adjustment \n",
    "\n",
    "The Y chromosome spans more than 59 million base pairs of DNA and represents almost 2 percent of the total DNA in cells. Each person normally has one pair of sex chromosomes in each cell. The Y chromosome is present in males, who have one X and one Y chromosome, while females have two X chromosomes. Since our analysis is on the comparative differences, we must eliminate chrY from our analyses.\n",
    "\n",
    "To do so, we grab the annotation from the GTF file and remove those junctions that correspond to the genes on this chromosome.\n",
    "\n",
    "So this is a count matrix row adjustment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8920</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8920\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8920\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8920   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5584</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5584\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5584\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5584   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2979</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2979\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2979\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2979   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6312</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6312\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6312\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6312   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42611</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42611\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42611\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42611    11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5177"
      ],
      "text/latex": [
       "5177"
      ],
      "text/markdown": [
       "5177"
      ],
      "text/plain": [
       "[1] 5177"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3824"
      ],
      "text/latex": [
       "3824"
      ],
      "text/markdown": [
       "3824"
      ],
      "text/plain": [
       "[1] 3824"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1619"
      ],
      "text/latex": [
       "1619"
      ],
      "text/markdown": [
       "1619"
      ],
      "text/plain": [
       "[1] 1619"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "3617"
      ],
      "text/latex": [
       "3617"
      ],
      "text/markdown": [
       "3617"
      ],
      "text/plain": [
       "[1] 3617"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "13390"
      ],
      "text/latex": [
       "13390"
      ],
      "text/markdown": [
       "13390"
      ],
      "text/plain": [
       "[1] 13390"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  862   230   643   552   108   304   326   537   626   118   783   578   241 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   85   246   514   260   337   388   377   255   264   269    17 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  496   155   386   354    59   212   188   348   410    74   400   387   135 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   60   129   368   184   210   216   284   166   162   191    10 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  286   120   152   166    46   109    97   137   154    70   116   386    67 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   40    49   205   144   127   155   106   104    74    61     8 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       "  604   117   475   423    46   241   191   441   541    40   637   327   153 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "   64   194   371   133   181   284   250   185   192   207    15 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 chr20 \n",
       " 3812  1458  2541  2540   663  1551  1544  2039  2538   742  2552  3547   939 \n",
       "chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9  chrX  chrY \n",
       "  493   895  2938  1803  1863  1890  2053  1636  1313  1134   127 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A3SS.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   17  8903 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "8920"
      ],
      "text/latex": [
       "8920"
      ],
      "text/markdown": [
       "8920"
      ],
      "text/plain": [
       "[1] 8920"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A5SS.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   10  5574 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5584"
      ],
      "text/latex": [
       "5584"
      ],
      "text/markdown": [
       "5584"
      ],
      "text/plain": [
       "[1] 5584"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "MXE.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "    8  2971 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2979"
      ],
      "text/latex": [
       "2979"
      ],
      "text/markdown": [
       "2979"
      ],
      "text/plain": [
       "[1] 2979"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RI.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "   15  6297 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "6312"
      ],
      "text/latex": [
       "6312"
      ],
      "text/markdown": [
       "6312"
      ],
      "text/plain": [
       "[1] 6312"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "SE.keepAllJunctionsButChrY\n",
       "FALSE  TRUE \n",
       "  127 42484 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "42611"
      ],
      "text/latex": [
       "42611"
      ],
      "text/markdown": [
       "42611"
      ],
      "text/plain": [
       "[1] 42611"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>13</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 13\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 13\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971   13"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297   11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>11</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 11\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 11\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484    11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>8903</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 8903\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 8903\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 8903 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>5574</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 5574\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 5574\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 5574 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>2971</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2971\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2971\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2971 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>6297</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6297\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6297\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 6297 7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>42484</li><li>7020</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 42484\n",
       "\\item 7020\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 42484\n",
       "2. 7020\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 42484  7020"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(fromGTF.A3SS)\n",
    "dim(fromGTF.A5SS)\n",
    "dim(fromGTF.MXE)\n",
    "dim(fromGTF.RI)\n",
    "dim(fromGTF.SE)\n",
    "\n",
    "A3SS.genes <- factor(fromGTF.A3SS$geneSymbol)\n",
    "A5SS.genes <- factor(fromGTF.A5SS$geneSymbol)\n",
    "MXE.genes  <- factor(fromGTF.MXE$geneSymbol)\n",
    "RI.genes   <- factor(fromGTF.RI$geneSymbol)\n",
    "SE.genes   <- factor(fromGTF.SE$geneSymbol)\n",
    "\n",
    "length(levels(A3SS.genes))    \n",
    "length(levels(A5SS.genes))    \n",
    "length(levels(MXE.genes))    \n",
    "length(levels(RI.genes))    \n",
    "length(levels(SE.genes))    \n",
    "\n",
    "table(fromGTF.A3SS$chr)\n",
    "table(fromGTF.A5SS$chr)\n",
    "table(fromGTF.MXE$chr)\n",
    "table(fromGTF.RI$chr)\n",
    "table(fromGTF.SE$chr)\n",
    "\n",
    "A3SS.keepAllJunctionsButChrY <- (fromGTF.A3SS$chr != \"chrY\")\n",
    "A5SS.keepAllJunctionsButChrY <- (fromGTF.A5SS$chr != \"chrY\")\n",
    "MXE.keepAllJunctionsButChrY  <- (fromGTF.MXE$chr  != \"chrY\")\n",
    "RI.keepAllJunctionsButChrY   <- (fromGTF.RI$chr   != \"chrY\")\n",
    "SE.keepAllJunctionsButChrY   <- (fromGTF.SE$chr   != \"chrY\")\n",
    "\n",
    "table(A3SS.keepAllJunctionsButChrY)\n",
    "sum(table(A3SS.keepAllJunctionsButChrY))\n",
    "\n",
    "table(A5SS.keepAllJunctionsButChrY)\n",
    "sum(table(A5SS.keepAllJunctionsButChrY))\n",
    "\n",
    "table(MXE.keepAllJunctionsButChrY)\n",
    "sum(table(MXE.keepAllJunctionsButChrY))\n",
    "\n",
    "table(RI.keepAllJunctionsButChrY)\n",
    "sum(table(RI.keepAllJunctionsButChrY))\n",
    "\n",
    "table(SE.keepAllJunctionsButChrY)\n",
    "sum(table(SE.keepAllJunctionsButChrY))\n",
    "\n",
    "a3ss_fromGTF_no_chrY          <- fromGTF.A3SS[A3SS.keepAllJunctionsButChrY,]\n",
    "a3ss_jc_ijc_no_chrY           <- a3ss.jc.ijc2[A3SS.keepAllJunctionsButChrY,]\n",
    "a3ss_jc_sjc_no_chrY           <- a3ss.jc.sjc2[A3SS.keepAllJunctionsButChrY,]\n",
    "rownames(a3ss_jc_ijc_no_chrY) <- rownames(a3ss_fromGTF_no_chrY)\n",
    "rownames(a3ss_jc_sjc_no_chrY) <- rownames(a3ss_fromGTF_no_chrY)\n",
    "\n",
    "a5ss_fromGTF_no_chrY          <- fromGTF.A5SS[A5SS.keepAllJunctionsButChrY,]\n",
    "a5ss_jc_ijc_no_chrY           <- a5ss.jc.ijc2[A5SS.keepAllJunctionsButChrY,]\n",
    "a5ss_jc_sjc_no_chrY           <- a5ss.jc.sjc2[A5SS.keepAllJunctionsButChrY,]\n",
    "rownames(a5ss_jc_ijc_no_chrY) <- rownames(a5ss_fromGTF_no_chrY)\n",
    "rownames(a5ss_jc_sjc_no_chrY) <- rownames(a5ss_fromGTF_no_chrY)\n",
    "\n",
    "mxe_fromGTF_no_chrY          <- fromGTF.MXE[MXE.keepAllJunctionsButChrY,]\n",
    "mxe_jc_ijc_no_chrY           <- mxe.jc.ijc2[MXE.keepAllJunctionsButChrY,]\n",
    "mxe_jc_sjc_no_chrY           <- mxe.jc.sjc2[MXE.keepAllJunctionsButChrY,]\n",
    "rownames(mxe_jc_ijc_no_chrY) <- rownames(mxe_fromGTF_no_chrY)\n",
    "rownames(mxe_jc_sjc_no_chrY) <- rownames(mxe_fromGTF_no_chrY)\n",
    "\n",
    "ri_fromGTF_no_chrY          <- fromGTF.RI[RI.keepAllJunctionsButChrY,]\n",
    "ri_jc_ijc_no_chrY           <- ri.jc.ijc2[RI.keepAllJunctionsButChrY,]\n",
    "ri_jc_sjc_no_chrY           <- ri.jc.sjc2[RI.keepAllJunctionsButChrY,]\n",
    "rownames(ri_jc_ijc_no_chrY) <- rownames(ri_fromGTF_no_chrY)\n",
    "rownames(ri_jc_sjc_no_chrY) <- rownames(ri_fromGTF_no_chrY)\n",
    "\n",
    "se_fromGTF_no_chrY          <- fromGTF.SE[SE.keepAllJunctionsButChrY,]\n",
    "se_jc_ijc_no_chrY           <- se.jc.ijc2[SE.keepAllJunctionsButChrY,]\n",
    "se_jc_sjc_no_chrY           <- se.jc.sjc2[SE.keepAllJunctionsButChrY,]\n",
    "rownames(se_jc_ijc_no_chrY) <- rownames(se_fromGTF_no_chrY)\n",
    "rownames(se_jc_sjc_no_chrY) <- rownames(se_fromGTF_no_chrY)\n",
    "\n",
    "dim(a3ss_fromGTF_no_chrY)\n",
    "dim(a5ss_fromGTF_no_chrY)\n",
    "dim(mxe_fromGTF_no_chrY)\n",
    "dim(ri_fromGTF_no_chrY)\n",
    "dim(se_fromGTF_no_chrY)\n",
    "\n",
    "dim(a3ss_jc_ijc_no_chrY)\n",
    "dim(a3ss_jc_sjc_no_chrY)\n",
    "\n",
    "dim(a5ss_jc_ijc_no_chrY)\n",
    "dim(a5ss_jc_sjc_no_chrY)\n",
    "\n",
    "dim(mxe_jc_ijc_no_chrY)\n",
    "dim(mxe_jc_sjc_no_chrY)\n",
    "\n",
    "dim(ri_jc_ijc_no_chrY)\n",
    "dim(ri_jc_sjc_no_chrY)\n",
    "\n",
    "dim(se_jc_ijc_no_chrY)\n",
    "dim(se_jc_sjc_no_chrY)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exploratory and Differential analysis as_event:ijc, sjc \n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) to transform junction counts (reads that were aligned to junctions when an exon is included - ijc, and reads that were aligned to junctions when the exon is excluded - sjc) with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma.    In each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression: \n",
    "\n",
    "           y = B0 + B1 sex + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the included exon junction count expression; sex denotes the reported sex of the subject\n",
    "\n",
    "## Differential analysis as_event (combined ijc and sjc)\n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) to transform junction counts (reads that were aligned to junctions when an exon is included - ijc, and reads that were aligned to junctions when the exon is excluded - sjc) with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma.    In each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression: \n",
    "\n",
    "           y = B0 + B1 sex + B2 as_event + B3 sex*as_event + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the alternative splicing event expression; sex denotes the reported sex of the subject, as_event represents the specific alternative splicing event - either included exon junction counts or skipped exon junction counts and their interaction terms.   Donor is added to our model as a blocking variable used in both the calculation of duplicate correlation as well as in the linear fit.\n",
    "\n",
    "### Voom, limma's lmFit and eBayes\n",
    "\n",
    "Using sample as a blocking variable, we are able to model the effects of the donor on the results, which improves the power.  This topic is discussed in biostars https://www.biostars.org/p/54565/.  And Gordon Smyth answers the question here https://mailman.stat.ethz.ch/pipermail/bioconductor/2014-February/057887.html.  The method of modeling is a random effects approach in which the intra-donor correlation is incorporated into the covariance matrix instead of the linear predictor.   And though as Gordon Smyth states both are good method and the twoway anova approach makes fewer assumptions, the random effects approach is statistically more powerful.  \n",
    "\n",
    "We have a balanced design in which all donors receive all stimuli (which is really in healthy human donors, life and all of its factors!) Our measurement has so many points -- we are measuring in the skipped exon approach, 42,611 junctions!   It is not possible to encorporate those measurements into the linear predictor.  A two-way ANOVA approach is virtually as powerful as the random effects approach \n",
    "and hence is preferable as it makes fewer assumptions.\n",
    "\n",
    "For an unbalanced design in which each donor receives only a subset of the stimula, the random effects approach is more powerful.\n",
    "\n",
    "Random effects approach is equivalent to The first method is twoway anova, a generalization of a paired analysis.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "print_exploratory_plots <- function (plot, dup, tissue_of_interest, splice_type, fromGTF, tissue_list, ijc, sjc, obj, metadata ) {\n",
    "\n",
    "    fromGTF           <- fromGTF\n",
    "    tissue_true       <- tissue_list == tissue_of_interest\n",
    "\n",
    "    # test to make sure we don't have nonsense\n",
    "    # like proteins - this process moves in one direction\n",
    "    # tissue will get us the phenotype selection \n",
    "    # from the phenotype SAMPID we get the metadata SAMPID\n",
    "    # which leads us to counts - it goes in this direction.\n",
    "    # first phenotype obj subsetting for the tissue_of_interest\n",
    "\n",
    "    message(\"\\nLimiting phenotype data to tissue of interest\\n\",\n",
    "           paste(tissue_of_interest),collapse=\" \")    \n",
    "    keep = pData(obj)$SMTSD== tissue_of_interest\n",
    "    message(\"\\nkeep\\n\",\n",
    "           paste(table(keep)),collapse=\" \")    \n",
    "    tissue_obj<- obj[,keep]\n",
    "    message(\"\\ntissue_obj now reduced to the tissue of interest\\n\",\n",
    "            paste(dim(pData(tissue_obj)), collapse=\" \"))\n",
    "    message(\"\\nsample value\\n\",\n",
    "            paste(   pData(tissue_obj)[1,], collapse=\" \"))\n",
    "\n",
    "    # now we get the runs via the metadata matrix via SAMPID\n",
    "    # limit the phenotype info to just the samples we care about\n",
    "    metadata_samples          <- as.character(metadata$SAMPID)\n",
    "    tissue_phenotype_samples  <- as.character(pData(tissue_obj)$SAMPID)\n",
    "\n",
    "    rm(keep)\n",
    "    keep = tissue_phenotype_samples %in% metadata_samples\n",
    "    message(\"\\nLimit the phenotypes to those we have samples for\\n\", \n",
    "        paste(table(keep), collapse = \" \") )\n",
    "    message(\"\\ntissue_count_obj now has these count specific phenotypes\")    \n",
    "    tissue_count_obj        <- tissue_obj[keep,]\n",
    "    pData(tissue_count_obj) <- pData(tissue_obj)[keep,]\n",
    "    message(\"\\nDimensions of tissue_count_obj\\n\", \n",
    "        paste(dim(pData(tissue_count_obj)), collapse = \" \") )\n",
    "    message(\"\\nDimensions of tissue_count_obj \\n\", \n",
    "        paste( dim(tissue_count_obj), collapse = \" \") )\n",
    "\n",
    "    # and vice-versa, limit our samples to the ones we have phenotype for\n",
    "    rm(keep)\n",
    "    metadata_samples       <- as.character(metadata$SAMPID)\n",
    "    phenotype_samples      <- as.character(pData(tissue_count_obj)$SAMPID)\n",
    "    keep = metadata_samples %in% phenotype_samples\n",
    "    message(\"\\nLimit the counts now has these count specific phenotypes \\n\", \n",
    "        paste( table(keep), collapse = \" \") )\n",
    "    table(keep)\n",
    "\n",
    "    ijc_tissue        <- ijc     [          ,keep==TRUE]\n",
    "    sjc_tissue        <- sjc     [          ,keep==TRUE]\n",
    "    metadata_tissue   <- metadata[keep==TRUE,          ]\n",
    "\n",
    "    message(\"\\ndimensions of the ijc_tissue \\n\", \n",
    "        paste(  dim(ijc_tissue), collapse = \" \") )\n",
    "    message(\"\\ndimensions of the ijc_tissue \\n\", \n",
    "        paste(  dim(sjc_tissue), collapse = \" \") )\n",
    "    message(\"\\ndimensions of the metadata_tissue \\n\", \n",
    "        paste(  dim(metadata_tissue), collapse = \" \") )\n",
    "   \n",
    "    metadata_samples       <- as.character(metadata_tissue$SAMPID)\n",
    "    phenotype_samples      <- as.character(pData(tissue_count_obj)$SAMPID)\n",
    "\n",
    "    length(metadata_samples)\n",
    "    length(phenotype_samples)\n",
    "    non_overlaps <- setdiff( metadata_samples, phenotype_samples)\n",
    "    message(\"\\nThe non-overlapping IDs between pheno and count data are:\\n\", \n",
    "        paste(length(non_overlaps), collapse = \" \") )\n",
    "\n",
    "    ijc.dm            <- data.matrix(ijc_tissue)\n",
    "    sjc.dm            <- data.matrix(sjc_tissue)    \n",
    "\n",
    "    ## remove features (junctions) that have zero sums\n",
    "    ijc.rs <- rowSums(ijc.dm) \n",
    "    keep.ijc.rs <- (ijc.rs > 10)\n",
    "    sjc.rs <- rowSums(sjc.dm) \n",
    "    keep.sjc.rs <- (sjc.rs > 10)\n",
    "    keep <- keep.ijc.rs | keep.sjc.rs\n",
    "    message(\"\\nijc.rs > 10 \\n\", \n",
    "        paste(table(keep.ijc.rs), collapse = \" \") )                    \n",
    "    message(\"\\nsjc.rs > 10 \\n\", \n",
    "        paste(table(keep.sjc.rs), collapse = \" \") )\n",
    "    message(\"\\nkeep combined ijc | sjc \\n\", \n",
    "        paste(table(keep), collapse = \" \") )                    \n",
    "    \n",
    "    # ensure the matrices do not have too many zeros\n",
    "    ijc.dm <- ijc.dm[keep,]\n",
    "    sjc.dm <- sjc.dm[keep,]\n",
    "    message(\"\\ndim(ijc.dm)\\n\", \n",
    "        paste(dim(ijc.dm), collapse = \" \") )                    \n",
    "    message(\"\\ndim(sjc.dm)\\n\", \n",
    "        paste(dim(sjc.dm), collapse = \" \") )\n",
    "    message(\"\\n\")\n",
    "    \n",
    "    for (i in (1:length(metadata_samples))) {\n",
    "        sample = metadata_tissue[i,]$SAMPID\n",
    "        sample_sex = pData(tissue_count_obj)$SEX[pData(tissue_count_obj)$SAMPID == sample]\n",
    "        if (i==1) {\n",
    "            sex = sample_sex\n",
    "        } else {\n",
    "            sex = c(sex,sample_sex)\n",
    "        }\n",
    "    }\n",
    "    message(\"\\nsex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\"\\n\"))\n",
    "    sex      <- ifelse(sex == 1,\"male\",\"female\")\n",
    "    sex      <- factor(sex,levels=c(\"male\",\"female\"))\n",
    "    message(\"\\nsex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\" \"))\n",
    "\n",
    "    design    <- model.matrix ( ~ sex)\n",
    "    message(\"\\ndesign matrix ijc, alone:\\n\",\n",
    "        paste0(head(design), collapse=\"\\n\"))\n",
    "\n",
    "    colnames(design) <- c(\"intercept\",\"sex\")\n",
    "\n",
    "    y_ijc <- DGEList(counts=ijc.dm, group = sex)\n",
    "    y_ijc <- calcNormFactors(y_ijc, method=\"RLE\")\n",
    "    y_ijc_voom <- voom (y_ijc, design=design, plot=plot)\n",
    "\n",
    "    Gender <- substring(sex,1,1)\n",
    "    pdf_sub_directory = '../pdf/'\n",
    "    csv_sub_directory = '../data/'\n",
    "    \n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_ijc, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-ijc-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "        plotMDS(y_ijc_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    " \n",
    "    fit_ijc <- lmFit(y_ijc_voom, design)\n",
    "    fit_ijc <- eBayes(fit_ijc)\n",
    "\n",
    "    ijc_sex_results                    <- topTable(fit_ijc, coef='sex', number=nrow(y_ijc_voom))\n",
    "    ijc_sex_results_refined            <- ijc_sex_results$adj.P.Val <= 0.05 & abs(ijc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    ijc_sex_rnResults                  <- rownames(ijc_sex_results)\n",
    "    ijc_sex_resultsAnnotations         <- fromGTF[ijc_sex_rnResults,]\n",
    "\n",
    "    ijc_sex_results_refinedAnnotations <- ijc_sex_resultsAnnotations[ijc_sex_results_refined      ==TRUE,]\n",
    "    message(\"\\ndimensions of the ijc_sex_results_refined_annotations \\n\", \n",
    "        paste(dim (ijc_sex_results_refinedAnnotations), collapse = \" \") )\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    ijc_sex_geneSymbols               <- ijc_sex_resultsAnnotations$geneSymbol\n",
    "    ijc_sex_refined_geneSymbols       <- ijc_sex_results_refinedAnnotations$geneSymbol\n",
    "    message(\"\\nlength ijc_sex_results_refined_geneSymbols\\n\", \n",
    "        paste(length(ijc_sex_refined_geneSymbols), collapse = \" \") )\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    ijc_sex_results_rn         <- paste(ijc_sex_geneSymbols,       ijc_sex_rnResults, sep=\"-\")\n",
    "    rownames(ijc_sex_results)  <- ijc_sex_results_rn    \n",
    "\n",
    "    ijc_sex_filename               = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_ijc_sex.csv',sep='')\n",
    "    ijc_sex_refined_filename       = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_ijc_sex_refined.csv',sep='')\n",
    "    ijc_sex_genesFilename          = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_ijc_sex_universe.txt',sep='')\n",
    "    ijc_sex_refined_genesFilename  = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_ijc_sex_gene_set.txt',sep='')\n",
    "    write.table(ijc_sex_results, \n",
    "                file = ijc_sex_filename        , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_results [ijc_sex_results_refined      ,], \n",
    "                file = ijc_sex_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_geneSymbols,        \n",
    "                file = ijc_sex_genesFilename        , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(ijc_sex_refined_geneSymbols,\n",
    "                file = ijc_sex_refined_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "   \n",
    "    message(\"\\nstarting sjc\\n\")\n",
    "\n",
    "    y_sjc <- DGEList(counts=sjc.dm, group = sex)\n",
    "    y_sjc <- calcNormFactors(y_sjc, method=\"RLE\")\n",
    "    y_sjc_voom <- voom (y_sjc, design=design, plot=plot)\n",
    "\n",
    "    Gender <- substring(sex,1,1)\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_sjc, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                                     snakecase::to_snake_case(tissue_of_interest)),\"-sjc-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "        plotMDS(y_sjc_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    " \n",
    "    fit_sjc <- lmFit(y_sjc_voom, design)\n",
    "    fit_sjc <- eBayes(fit_sjc)\n",
    "\n",
    "    sjc_sex_results                    <- topTable(fit_sjc, coef='sex', number=nrow(y_sjc_voom))\n",
    "    sjc_sex_results_refined            <- sjc_sex_results$adj.P.Val <= 0.05 & abs(sjc_sex_results$logFC) >= abs(log2(1.5))\n",
    "    sjc_sex_rnResults                  <- rownames(sjc_sex_results)\n",
    "    sjc_sex_resultsAnnotations         <- fromGTF[sjc_sex_rnResults,]\n",
    "\n",
    "    sjc_sex_results_refinedAnnotations <- sjc_sex_resultsAnnotations[sjc_sex_results_refined      ==TRUE,]\n",
    "    message(\"\\ndimensions of the sjc_sex_results_refined_annotations \\n\", \n",
    "        paste(dim (sjc_sex_results_refinedAnnotations), collapse = \" \") )\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sjc_sex_geneSymbols               <- sjc_sex_resultsAnnotations$geneSymbol\n",
    "    sjc_sex_refined_geneSymbols       <- sjc_sex_results_refinedAnnotations$geneSymbol\n",
    "    message(\"\\nlength sjc_sex_results_refined_geneSymbols\\n\", \n",
    "        paste(length(sjc_sex_refined_geneSymbols), collapse = \" \") )\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sjc_sex_results_rn         <- paste(sjc_sex_geneSymbols, sjc_sex_rnResults, sep=\"-\")\n",
    "    rownames(sjc_sex_results)  <- sjc_sex_results_rn    \n",
    "\n",
    "    sjc_sex_filename               = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sjc_sex.csv',sep='')\n",
    "    sjc_sex_refined_filename       = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sjc_sex_refined.csv',sep='')\n",
    "    sjc_sex_genesFilename          = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sjc_sex_universe.txt',sep='')\n",
    "    sjc_sex_refined_genesFilename  = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sjc_sex_gene_set.txt',sep='')\n",
    "    write.table(sjc_sex_results, \n",
    "                file = sjc_sex_filename        , row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_results [sjc_sex_results_refined      ,], \n",
    "                file = sjc_sex_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_geneSymbols,        \n",
    "                file = sjc_sex_genesFilename        , row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sjc_sex_refined_geneSymbols,\n",
    "                file = sjc_sex_refined_genesFilename, row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    \n",
    "   \n",
    "    message(\"starting the y prediction\\n\")\n",
    "    sample_names <- as.character(colnames(ijc.dm))\n",
    "    # we will add donor as a blocking parameter\n",
    "    # rather than sample name -- we should use donor for real\n",
    "    sample     <- factor(sample_names)\n",
    "    \n",
    "    donor    <- rep(sample, 2)\n",
    "    message(\"\\ndonor size\", \n",
    "        paste(length(donor), collapse = \" \") )\n",
    "    \n",
    "    ijc_names <- as.character(colnames(ijc.dm))\n",
    "    sjc_names <- as.character(colnames(sjc.dm))\n",
    "    sjc_names <- paste0(sjc_names,\"-sjc\")\n",
    "    ijc_names <- paste0(ijc_names,\"-ijc\")\n",
    "\n",
    "    colnames(ijc.dm) <- ijc_names\n",
    "    colnames(sjc.dm) <- sjc_names\n",
    "\n",
    "    as_matrix <- cbind(ijc.dm,sjc.dm)\n",
    "    message(\"\\ndim as_matrix\", \n",
    "        paste(dim(as_matrix), collapse = \" \")) \n",
    "            \n",
    "    for (i in (1:length(metadata_samples))) {\n",
    "        sample = metadata_tissue[i,]$SAMPID\n",
    "        sample_sex = pData(tissue_count_obj)$SEX[pData(tissue_count_obj)$SAMPID == sample]\n",
    "        if (i==1) {\n",
    "            sex = sample_sex\n",
    "        } else {\n",
    "            sex = c(sex,sample_sex)\n",
    "        }\n",
    "    }\n",
    "    message(\"sex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\"\\n\"))\n",
    "    sex      <- ifelse(sex == 1,\"male\",\"female\")\n",
    "    sex      <- factor(sex,levels=c(\"male\",\"female\"))\n",
    "    message(\"\\nsex samples:\\n\",\n",
    "        paste0(table(sex), collapse=\" \"))\n",
    "    sex2      <- c(rep(sex,2))\n",
    "    table(sex2)\n",
    "    message(\"\\nlength sex2\\n\", \n",
    "        paste(length(sex2), collapse = \" \") )\n",
    "    message(\"table sex2\\n\", \n",
    "        paste(table(sex2), collapse = \"\\n\") )\n",
    "    as_event  <- c(rep(\"ijc\",dim(ijc.dm)[2]), rep(\"sjc\", dim(sjc.dm)[2]))\n",
    "    as_event  <- factor(as_event, levels=c(\"ijc\", \"sjc\"))\n",
    "    message(\"\\nlength as_event\\n\", \n",
    "        paste(length(as_event), collapse = \" \") )\n",
    "\n",
    "    design    <- model.matrix( ~ sex2 + as_event + sex2*as_event)\n",
    "    dim(design)\n",
    "    colnames(design) <- c(\"intercept\",\"sex\", \"as_event\",\"sex*as_event\")\n",
    "    message(\"\\ndim design <- model.matrix( ~sex + as_event + sex*as_event)\\n\", \n",
    "        paste(head(design), collapse = \"\\n\") )\n",
    "\n",
    "    y <- DGEList(counts=as_matrix, group = sex2)\n",
    "    y <- calcNormFactors(y, method=\"RLE\")\n",
    "    y_voom <- voom (y, design=design, plot = plot)\n",
    "\n",
    "    if (dup==TRUE) {\n",
    "        dup_cor <- duplicateCorrelation(y_voom$E, design=design, ndups=2, block=donor, weights=y$samples$norm.factors)\n",
    "        dup_cor$consensus.correlation \n",
    "        y_dup_voom <- voom (y, design=design, plot = plot, block = donor, correlation = dup_cor$consensus.correlation) \n",
    "    }\n",
    "    \n",
    "    Gender <- substring(sex[1:dim(ijc.dm)[2]],1,1)\n",
    "    message(\"\\nGenders new size\\n\", \n",
    "        paste(length(Gender), collapse = \" \") )\n",
    "    message(\"\\nplotting y for ijc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")\n",
    "     # print the combined exploratory plot\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    message(\"\\nplotting y_voom for ijc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-voom-ijc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_voom[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    if (dup == TRUE) {\n",
    "        filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                           snakecase::to_snake_case(tissue_of_interest)),\"-y-dup-voom-ijc-MDSplot-100.pdf\")\n",
    "        pdf (filename)\n",
    "            plotMDS(y_dup_voom[,c(1:dim(ijc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "        dev.off()\n",
    "    }\n",
    "    message(\"\\nplotting y for sjc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")\n",
    "    filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-sjc-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "            gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    \n",
    "    if (dup == TRUE) {\n",
    "        message(\"\\nplotting y_voom for sjc portion of design <- model.matrix( ~sex + as_event + sex*as_event\\n\")    \n",
    "        filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-voom-sjc-MDSplot-100.pdf\")\n",
    "        pdf (filename)\n",
    "            plotMDS(y_voom[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "        dev.off()\n",
    "        filename <- paste0(paste0(paste0(pdf_sub_directory, splice_type),\n",
    "                              snakecase::to_snake_case(tissue_of_interest)),\"-y-dup-voom-sjc-MDSplot-100.pdf\")\n",
    "        pdf (filename)\n",
    "            plotMDS(y_dup_voom[,c((dim(ijc.dm)[2]+1)):(dim(ijc.dm)[2]+dim(sjc.dm)[2])], labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "        dev.off()\n",
    "        \n",
    "        fit <- lmFit(y_dup_voom, design=design, block=donor, correlation = dup_cor$consensus.correlation)\n",
    "    } else {\n",
    "        fit <- lmFit(y_voom, design=design)\n",
    "    }\n",
    "        \n",
    "    fit <- eBayes(fit, robust=TRUE)\n",
    "    \n",
    "    sex_as_events_results         <- topTable(fit, coef=\"sex*as_event\", number=nrow(y_voom))\n",
    "    sex_as_events_results_refined <- sex_as_events_results$adj.P.Val <= 0.05 & abs(sex_as_events_results$logFC) >= abs(log2(1.5))\n",
    "\n",
    "    sex_results                   <- topTable(fit, coef=\"sex\", number=nrow(y_voom))\n",
    "    sex_results_refined           <- sex_results$adj.P.Val <= 0.05 & abs(sex_results$logFC) >= abs(log2(1.5))\n",
    "\n",
    "    sex_as_events_rnResults <- rownames(sex_as_events_results)\n",
    "    sex_rnResults           <- rownames(sex_results)\n",
    "    head(sex_as_events_rnResults)\n",
    "    head(ijc_sex_rnResults)\n",
    "    head(sex_rnResults)\n",
    "    head(fromGTF[sex_as_events_rnResults,])\n",
    "\n",
    "    # use the junctionIDs to get the annotations\n",
    "    sex_as_events_resultsAnnotations      <- fromGTF[sex_as_events_rnResults,]\n",
    "    sex_resultsAnnotations                <- fromGTF[sex_rnResults,]\n",
    "    ijc_sex_resultsAnnotations            <- fromGTF[ijc_sex_rnResults,]\n",
    "    head(sex_as_events_resultsAnnotations)\n",
    "    head(sex_resultsAnnotations)\n",
    "    head(ijc_sex_resultsAnnotations)\n",
    "    \n",
    "    sex_as_events_results_refinedAnnotations<- sex_as_events_resultsAnnotations[sex_as_events_results_refined==TRUE,]\n",
    "    sex_results_refinedAnnotations          <- sex_resultsAnnotations          [sex_results_refined          ==TRUE,]\n",
    "    sjc_sex_results_refinedAnnotations       <- sjc_sex_resultsAnnotations      [sjc_sex_results_refined      ==TRUE,]\n",
    "    head(sex_as_events_results_refinedAnnotations)\n",
    "    head(sex_results_refinedAnnotations)\n",
    "    head(sjc_sex_results_refinedAnnotations)\n",
    "\n",
    "    # geneSymbols are in the annotations \n",
    "    sex_as_events_geneSymbols         <- sex_as_events_resultsAnnotations$geneSymbol\n",
    "    sex_as_events_refined_geneSymbols <- sex_as_events_results_refinedAnnotations$geneSymbol\n",
    "    sex_geneSymbols                   <- sex_resultsAnnotations$geneSymbol\n",
    "    sex_refined_geneSymbols           <- sex_results_refinedAnnotations$geneSymbol\n",
    "    ijc_sex_geneSymbols               <- ijc_sex_resultsAnnotations$geneSymbol\n",
    "    ijc_sex_refined_geneSymbols       <- ijc_sex_results_refinedAnnotations$geneSymbol\n",
    "    sjc_sex_geneSymbols               <- sjc_sex_resultsAnnotations$geneSymbol\n",
    "    sjc_sex_refined_geneSymbols       <- sjc_sex_results_refinedAnnotations$geneSymbol\n",
    "\n",
    "\n",
    "    # adjust the rownames to be the geneSymbols rather than junction IDs\n",
    "    sex_as_events_results_rn   <- paste(sex_as_events_geneSymbols, sex_as_events_rnResults, sep=\"-\")\n",
    "    sex_results_rn             <- paste(sex_geneSymbols,           sex_rnResults, sep=\"-\")\n",
    "    ijc_sex_results_rn         <- paste(ijc_sex_geneSymbols,       ijc_sex_rnResults, sep=\"-\")\n",
    "    sjc_sex_results_rn         <- paste(sjc_sex_geneSymbols,       sjc_sex_rnResults, sep=\"-\")\n",
    "    message(\"\\n sex_as_events\\n\", \n",
    "        paste(head(sex_as_events_results_rn), collapse = \" \") )\n",
    "    message(\"\\n ijc_sex_results\\n\", \n",
    "        paste(head(ijc_sex_results_rn), collapse = \" \") )\n",
    "    message(\"\\n sjc_sex_results\\n\", \n",
    "        paste(head(sjc_sex_results_rn), collapse = \" \") )\n",
    "    rownames(sex_as_events_results) <- sex_as_events_results_rn\n",
    "    rownames(sex_results)           <- sex_results_rn\n",
    "    rownames(ijc_sex_results)       <- ijc_sex_results_rn\n",
    "    rownames(sjc_sex_results)       <- sjc_sex_results_rn\n",
    "\n",
    "    sex_as_events_filename         = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_as_events.csv')\n",
    "    sex_as_events_refined_filename = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_as_events_refined.csv',sep='')\n",
    "    sex_filename                   = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex.csv',sep='')\n",
    "    sex_refined_filename           = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_DGE_sex_refined.csv',sep='')\n",
    "    sex_as_events_genesFilename    = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_as_events_universe.txt',sep='')\n",
    "    sex_as_events_refined_genesFilename = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_as_events_gene_set.txt',sep='')\n",
    "    sex_genesFilename              = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_universe.txt',sep='')\n",
    "    sex_refined_genesFilename      = paste0(paste0(paste0(csv_sub_directory, splice_type),\n",
    "                                                   snakecase::to_snake_case(tissue_of_interest)),'_sex_gene_set.txt',sep='')\n",
    "\n",
    "    write.table(sex_as_events_results, file = sex_as_events_filename, \n",
    "                row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_as_events_results[sex_as_events_results_refined,], \n",
    "                file = sex_as_events_refined_filename, row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_results,           file = sex_filename          , \n",
    "                row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_results [sex_results_refined          ,], file = sex_refined_filename, \n",
    "                row.names = T, col.names = T, quote = F, sep = \",\")\n",
    "    write.table(sex_as_events_geneSymbols, file = sex_as_events_genesFilename, \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_as_events_refined_geneSymbols,file = sex_as_events_refined_genesFilename, \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_geneSymbols,           file = sex_genesFilename          , \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "    write.table(sex_refined_geneSymbols,          file = sex_refined_genesFilename          , \n",
    "                row.names = F, col.names = F, quote = F, sep = \",\")\n",
    "\n",
    "    return(0)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Testing Snippet\n",
    "\n",
    "To use this snippet - set *test = `TRUE`*.\n",
    "To play with using or not using duplicate correlation, set *dup = `TRUE`* accordingly."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Execution of All Tissues and All Splicing Variants\n",
    "\n",
    "Additional values set to enable this notebook to be executed as a nextflow workflow or to run in place with appropriate settings.\n",
    "\n",
    "### parameters Setting\n",
    "\n",
    "1. Setting `dup=TRUE` causes lengthy execution times.\n",
    "\n",
    "2. Setting `plot=TRUE` can overwhelm the saving capacity within a jupyter-lab notebook - \n",
    "   this sets to print all the voom plots.\n",
    "   \n",
    "3. Adjusting `splice_type` will allow you to play with a variety of results\n",
    "\n",
    "   a. all splice types desired to be run:\n",
    "    \n",
    "    `splice_list       = c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\",\"se_\")`\n",
    "    \n",
    "   b. a subset (leaving out say `splice_type = \"se_\"` since it is the largest, for example)\n",
    "    \n",
    "    `splice_list       = c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\")`\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>adipose_subcutaneous</li><li>adipose_visceral_omentum</li><li>adrenal_gland</li><li>artery_aorta</li><li>artery_coronary</li><li>artery_tibial</li><li>brain_caudate_basal_ganglia</li><li>brain_cerebellar_hemisphere</li><li>brain_cerebellum</li><li>brain_cortex</li><li>brain_frontal_cortex_ba_9</li><li>brain_hippocampus</li><li>brain_hypothalamus</li><li>brain_nucleus_accumbens_basal_ganglia</li><li>brain_putamen_basal_ganglia</li><li>brain_spinal_cord_cervical_c_1</li><li>breast_mammary_tissue</li><li>cells_cultured_fibroblasts</li><li>cells_ebv_transformed_lymphocytes</li><li>colon_sigmoid</li><li>colon_transverse</li><li>esophagus_gastroesophageal_junction</li><li>esophagus_mucosa</li><li>esophagus_muscularis</li><li>heart_atrial_appendage</li><li>heart_left_ventricle</li><li>liver</li><li>lung</li><li>muscle_skeletal</li><li>nerve_tibial</li><li>pancreas</li><li>pituitary</li><li>skin_not_sun_exposed_suprapubic</li><li>skin_sun_exposed_lower_leg</li><li>small_intestine_terminal_ileum</li><li>spleen</li><li>stomach</li><li>thyroid</li><li>whole_blood</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'adipose_subcutaneous'</li><li>'adipose_visceral_omentum'</li><li>'adrenal_gland'</li><li>'artery_aorta'</li><li>'artery_coronary'</li><li>'artery_tibial'</li><li>'brain_caudate_basal_ganglia'</li><li>'brain_cerebellar_hemisphere'</li><li>'brain_cerebellum'</li><li>'brain_cortex'</li><li>'brain_frontal_cortex_ba_9'</li><li>'brain_hippocampus'</li><li>'brain_hypothalamus'</li><li>'brain_nucleus_accumbens_basal_ganglia'</li><li>'brain_putamen_basal_ganglia'</li><li>'brain_spinal_cord_cervical_c_1'</li><li>'breast_mammary_tissue'</li><li>'cells_cultured_fibroblasts'</li><li>'cells_ebv_transformed_lymphocytes'</li><li>'colon_sigmoid'</li><li>'colon_transverse'</li><li>'esophagus_gastroesophageal_junction'</li><li>'esophagus_mucosa'</li><li>'esophagus_muscularis'</li><li>'heart_atrial_appendage'</li><li>'heart_left_ventricle'</li><li>'liver'</li><li>'lung'</li><li>'muscle_skeletal'</li><li>'nerve_tibial'</li><li>'pancreas'</li><li>'pituitary'</li><li>'skin_not_sun_exposed_suprapubic'</li><li>'skin_sun_exposed_lower_leg'</li><li>'small_intestine_terminal_ileum'</li><li>'spleen'</li><li>'stomach'</li><li>'thyroid'</li><li>'whole_blood'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item adipose\\_subcutaneous\n",
       "\\item adipose\\_visceral\\_omentum\n",
       "\\item adrenal\\_gland\n",
       "\\item artery\\_aorta\n",
       "\\item artery\\_coronary\n",
       "\\item artery\\_tibial\n",
       "\\item brain\\_caudate\\_basal\\_ganglia\n",
       "\\item brain\\_cerebellar\\_hemisphere\n",
       "\\item brain\\_cerebellum\n",
       "\\item brain\\_cortex\n",
       "\\item brain\\_frontal\\_cortex\\_ba\\_9\n",
       "\\item brain\\_hippocampus\n",
       "\\item brain\\_hypothalamus\n",
       "\\item brain\\_nucleus\\_accumbens\\_basal\\_ganglia\n",
       "\\item brain\\_putamen\\_basal\\_ganglia\n",
       "\\item brain\\_spinal\\_cord\\_cervical\\_c\\_1\n",
       "\\item breast\\_mammary\\_tissue\n",
       "\\item cells\\_cultured\\_fibroblasts\n",
       "\\item cells\\_ebv\\_transformed\\_lymphocytes\n",
       "\\item colon\\_sigmoid\n",
       "\\item colon\\_transverse\n",
       "\\item esophagus\\_gastroesophageal\\_junction\n",
       "\\item esophagus\\_mucosa\n",
       "\\item esophagus\\_muscularis\n",
       "\\item heart\\_atrial\\_appendage\n",
       "\\item heart\\_left\\_ventricle\n",
       "\\item liver\n",
       "\\item lung\n",
       "\\item muscle\\_skeletal\n",
       "\\item nerve\\_tibial\n",
       "\\item pancreas\n",
       "\\item pituitary\n",
       "\\item skin\\_not\\_sun\\_exposed\\_suprapubic\n",
       "\\item skin\\_sun\\_exposed\\_lower\\_leg\n",
       "\\item small\\_intestine\\_terminal\\_ileum\n",
       "\\item spleen\n",
       "\\item stomach\n",
       "\\item thyroid\n",
       "\\item whole\\_blood\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'adipose\\_subcutaneous'\n",
       "\\item 'adipose\\_visceral\\_omentum'\n",
       "\\item 'adrenal\\_gland'\n",
       "\\item 'artery\\_aorta'\n",
       "\\item 'artery\\_coronary'\n",
       "\\item 'artery\\_tibial'\n",
       "\\item 'brain\\_caudate\\_basal\\_ganglia'\n",
       "\\item 'brain\\_cerebellar\\_hemisphere'\n",
       "\\item 'brain\\_cerebellum'\n",
       "\\item 'brain\\_cortex'\n",
       "\\item 'brain\\_frontal\\_cortex\\_ba\\_9'\n",
       "\\item 'brain\\_hippocampus'\n",
       "\\item 'brain\\_hypothalamus'\n",
       "\\item 'brain\\_nucleus\\_accumbens\\_basal\\_ganglia'\n",
       "\\item 'brain\\_putamen\\_basal\\_ganglia'\n",
       "\\item 'brain\\_spinal\\_cord\\_cervical\\_c\\_1'\n",
       "\\item 'breast\\_mammary\\_tissue'\n",
       "\\item 'cells\\_cultured\\_fibroblasts'\n",
       "\\item 'cells\\_ebv\\_transformed\\_lymphocytes'\n",
       "\\item 'colon\\_sigmoid'\n",
       "\\item 'colon\\_transverse'\n",
       "\\item 'esophagus\\_gastroesophageal\\_junction'\n",
       "\\item 'esophagus\\_mucosa'\n",
       "\\item 'esophagus\\_muscularis'\n",
       "\\item 'heart\\_atrial\\_appendage'\n",
       "\\item 'heart\\_left\\_ventricle'\n",
       "\\item 'liver'\n",
       "\\item 'lung'\n",
       "\\item 'muscle\\_skeletal'\n",
       "\\item 'nerve\\_tibial'\n",
       "\\item 'pancreas'\n",
       "\\item 'pituitary'\n",
       "\\item 'skin\\_not\\_sun\\_exposed\\_suprapubic'\n",
       "\\item 'skin\\_sun\\_exposed\\_lower\\_leg'\n",
       "\\item 'small\\_intestine\\_terminal\\_ileum'\n",
       "\\item 'spleen'\n",
       "\\item 'stomach'\n",
       "\\item 'thyroid'\n",
       "\\item 'whole\\_blood'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. adipose_subcutaneous\n",
       "2. adipose_visceral_omentum\n",
       "3. adrenal_gland\n",
       "4. artery_aorta\n",
       "5. artery_coronary\n",
       "6. artery_tibial\n",
       "7. brain_caudate_basal_ganglia\n",
       "8. brain_cerebellar_hemisphere\n",
       "9. brain_cerebellum\n",
       "10. brain_cortex\n",
       "11. brain_frontal_cortex_ba_9\n",
       "12. brain_hippocampus\n",
       "13. brain_hypothalamus\n",
       "14. brain_nucleus_accumbens_basal_ganglia\n",
       "15. brain_putamen_basal_ganglia\n",
       "16. brain_spinal_cord_cervical_c_1\n",
       "17. breast_mammary_tissue\n",
       "18. cells_cultured_fibroblasts\n",
       "19. cells_ebv_transformed_lymphocytes\n",
       "20. colon_sigmoid\n",
       "21. colon_transverse\n",
       "22. esophagus_gastroesophageal_junction\n",
       "23. esophagus_mucosa\n",
       "24. esophagus_muscularis\n",
       "25. heart_atrial_appendage\n",
       "26. heart_left_ventricle\n",
       "27. liver\n",
       "28. lung\n",
       "29. muscle_skeletal\n",
       "30. nerve_tibial\n",
       "31. pancreas\n",
       "32. pituitary\n",
       "33. skin_not_sun_exposed_suprapubic\n",
       "34. skin_sun_exposed_lower_leg\n",
       "35. small_intestine_terminal_ileum\n",
       "36. spleen\n",
       "37. stomach\n",
       "38. thyroid\n",
       "39. whole_blood\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'adipose_subcutaneous'\n",
       "2. 'adipose_visceral_omentum'\n",
       "3. 'adrenal_gland'\n",
       "4. 'artery_aorta'\n",
       "5. 'artery_coronary'\n",
       "6. 'artery_tibial'\n",
       "7. 'brain_caudate_basal_ganglia'\n",
       "8. 'brain_cerebellar_hemisphere'\n",
       "9. 'brain_cerebellum'\n",
       "10. 'brain_cortex'\n",
       "11. 'brain_frontal_cortex_ba_9'\n",
       "12. 'brain_hippocampus'\n",
       "13. 'brain_hypothalamus'\n",
       "14. 'brain_nucleus_accumbens_basal_ganglia'\n",
       "15. 'brain_putamen_basal_ganglia'\n",
       "16. 'brain_spinal_cord_cervical_c_1'\n",
       "17. 'breast_mammary_tissue'\n",
       "18. 'cells_cultured_fibroblasts'\n",
       "19. 'cells_ebv_transformed_lymphocytes'\n",
       "20. 'colon_sigmoid'\n",
       "21. 'colon_transverse'\n",
       "22. 'esophagus_gastroesophageal_junction'\n",
       "23. 'esophagus_mucosa'\n",
       "24. 'esophagus_muscularis'\n",
       "25. 'heart_atrial_appendage'\n",
       "26. 'heart_left_ventricle'\n",
       "27. 'liver'\n",
       "28. 'lung'\n",
       "29. 'muscle_skeletal'\n",
       "30. 'nerve_tibial'\n",
       "31. 'pancreas'\n",
       "32. 'pituitary'\n",
       "33. 'skin_not_sun_exposed_suprapubic'\n",
       "34. 'skin_sun_exposed_lower_leg'\n",
       "35. 'small_intestine_terminal_ileum'\n",
       "36. 'spleen'\n",
       "37. 'stomach'\n",
       "38. 'thyroid'\n",
       "39. 'whole_blood'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] adipose_subcutaneous                  adipose_visceral_omentum             \n",
       " [3] adrenal_gland                         artery_aorta                         \n",
       " [5] artery_coronary                       artery_tibial                        \n",
       " [7] brain_caudate_basal_ganglia           brain_cerebellar_hemisphere          \n",
       " [9] brain_cerebellum                      brain_cortex                         \n",
       "[11] brain_frontal_cortex_ba_9             brain_hippocampus                    \n",
       "[13] brain_hypothalamus                    brain_nucleus_accumbens_basal_ganglia\n",
       "[15] brain_putamen_basal_ganglia           brain_spinal_cord_cervical_c_1       \n",
       "[17] breast_mammary_tissue                 cells_cultured_fibroblasts           \n",
       "[19] cells_ebv_transformed_lymphocytes     colon_sigmoid                        \n",
       "[21] colon_transverse                      esophagus_gastroesophageal_junction  \n",
       "[23] esophagus_mucosa                      esophagus_muscularis                 \n",
       "[25] heart_atrial_appendage                heart_left_ventricle                 \n",
       "[27] liver                                 lung                                 \n",
       "[29] muscle_skeletal                       nerve_tibial                         \n",
       "[31] pancreas                              pituitary                            \n",
       "[33] skin_not_sun_exposed_suprapubic       skin_sun_exposed_lower_leg           \n",
       "[35] small_intestine_terminal_ileum        spleen                               \n",
       "[37] stomach                               thyroid                              \n",
       "[39] whole_blood                          \n",
       "39 Levels: adipose_subcutaneous adipose_visceral_omentum ... whole_blood"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'adipose_subcutaneous'</li><li>'adipose_visceral_omentum'</li><li>'adrenal_gland'</li><li>'artery_aorta'</li><li>'artery_coronary'</li><li>'artery_tibial'</li><li>'brain_caudate_basal_ganglia'</li><li>'brain_cerebellar_hemisphere'</li><li>'brain_cerebellum'</li><li>'brain_cortex'</li><li>'brain_frontal_cortex_ba_9'</li><li>'brain_hippocampus'</li><li>'brain_hypothalamus'</li><li>'brain_nucleus_accumbens_basal_ganglia'</li><li>'brain_putamen_basal_ganglia'</li><li>'brain_spinal_cord_cervical_c_1'</li><li>'breast_mammary_tissue'</li><li>'cells_cultured_fibroblasts'</li><li>'cells_ebv_transformed_lymphocytes'</li><li>'colon_sigmoid'</li><li>'colon_transverse'</li><li>'esophagus_gastroesophageal_junction'</li><li>'esophagus_mucosa'</li><li>'esophagus_muscularis'</li><li>'heart_atrial_appendage'</li><li>'heart_left_ventricle'</li><li>'liver'</li><li>'lung'</li><li>'muscle_skeletal'</li><li>'nerve_tibial'</li><li>'pancreas'</li><li>'pituitary'</li><li>'skin_not_sun_exposed_suprapubic'</li><li>'skin_sun_exposed_lower_leg'</li><li>'small_intestine_terminal_ileum'</li><li>'spleen'</li><li>'stomach'</li><li>'thyroid'</li><li>'whole_blood'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'adipose\\_subcutaneous'\n",
       "\\item 'adipose\\_visceral\\_omentum'\n",
       "\\item 'adrenal\\_gland'\n",
       "\\item 'artery\\_aorta'\n",
       "\\item 'artery\\_coronary'\n",
       "\\item 'artery\\_tibial'\n",
       "\\item 'brain\\_caudate\\_basal\\_ganglia'\n",
       "\\item 'brain\\_cerebellar\\_hemisphere'\n",
       "\\item 'brain\\_cerebellum'\n",
       "\\item 'brain\\_cortex'\n",
       "\\item 'brain\\_frontal\\_cortex\\_ba\\_9'\n",
       "\\item 'brain\\_hippocampus'\n",
       "\\item 'brain\\_hypothalamus'\n",
       "\\item 'brain\\_nucleus\\_accumbens\\_basal\\_ganglia'\n",
       "\\item 'brain\\_putamen\\_basal\\_ganglia'\n",
       "\\item 'brain\\_spinal\\_cord\\_cervical\\_c\\_1'\n",
       "\\item 'breast\\_mammary\\_tissue'\n",
       "\\item 'cells\\_cultured\\_fibroblasts'\n",
       "\\item 'cells\\_ebv\\_transformed\\_lymphocytes'\n",
       "\\item 'colon\\_sigmoid'\n",
       "\\item 'colon\\_transverse'\n",
       "\\item 'esophagus\\_gastroesophageal\\_junction'\n",
       "\\item 'esophagus\\_mucosa'\n",
       "\\item 'esophagus\\_muscularis'\n",
       "\\item 'heart\\_atrial\\_appendage'\n",
       "\\item 'heart\\_left\\_ventricle'\n",
       "\\item 'liver'\n",
       "\\item 'lung'\n",
       "\\item 'muscle\\_skeletal'\n",
       "\\item 'nerve\\_tibial'\n",
       "\\item 'pancreas'\n",
       "\\item 'pituitary'\n",
       "\\item 'skin\\_not\\_sun\\_exposed\\_suprapubic'\n",
       "\\item 'skin\\_sun\\_exposed\\_lower\\_leg'\n",
       "\\item 'small\\_intestine\\_terminal\\_ileum'\n",
       "\\item 'spleen'\n",
       "\\item 'stomach'\n",
       "\\item 'thyroid'\n",
       "\\item 'whole\\_blood'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'adipose_subcutaneous'\n",
       "2. 'adipose_visceral_omentum'\n",
       "3. 'adrenal_gland'\n",
       "4. 'artery_aorta'\n",
       "5. 'artery_coronary'\n",
       "6. 'artery_tibial'\n",
       "7. 'brain_caudate_basal_ganglia'\n",
       "8. 'brain_cerebellar_hemisphere'\n",
       "9. 'brain_cerebellum'\n",
       "10. 'brain_cortex'\n",
       "11. 'brain_frontal_cortex_ba_9'\n",
       "12. 'brain_hippocampus'\n",
       "13. 'brain_hypothalamus'\n",
       "14. 'brain_nucleus_accumbens_basal_ganglia'\n",
       "15. 'brain_putamen_basal_ganglia'\n",
       "16. 'brain_spinal_cord_cervical_c_1'\n",
       "17. 'breast_mammary_tissue'\n",
       "18. 'cells_cultured_fibroblasts'\n",
       "19. 'cells_ebv_transformed_lymphocytes'\n",
       "20. 'colon_sigmoid'\n",
       "21. 'colon_transverse'\n",
       "22. 'esophagus_gastroesophageal_junction'\n",
       "23. 'esophagus_mucosa'\n",
       "24. 'esophagus_muscularis'\n",
       "25. 'heart_atrial_appendage'\n",
       "26. 'heart_left_ventricle'\n",
       "27. 'liver'\n",
       "28. 'lung'\n",
       "29. 'muscle_skeletal'\n",
       "30. 'nerve_tibial'\n",
       "31. 'pancreas'\n",
       "32. 'pituitary'\n",
       "33. 'skin_not_sun_exposed_suprapubic'\n",
       "34. 'skin_sun_exposed_lower_leg'\n",
       "35. 'small_intestine_terminal_ileum'\n",
       "36. 'spleen'\n",
       "37. 'stomach'\n",
       "38. 'thyroid'\n",
       "39. 'whole_blood'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"adipose_subcutaneous\"                 \n",
       " [2] \"adipose_visceral_omentum\"             \n",
       " [3] \"adrenal_gland\"                        \n",
       " [4] \"artery_aorta\"                         \n",
       " [5] \"artery_coronary\"                      \n",
       " [6] \"artery_tibial\"                        \n",
       " [7] \"brain_caudate_basal_ganglia\"          \n",
       " [8] \"brain_cerebellar_hemisphere\"          \n",
       " [9] \"brain_cerebellum\"                     \n",
       "[10] \"brain_cortex\"                         \n",
       "[11] \"brain_frontal_cortex_ba_9\"            \n",
       "[12] \"brain_hippocampus\"                    \n",
       "[13] \"brain_hypothalamus\"                   \n",
       "[14] \"brain_nucleus_accumbens_basal_ganglia\"\n",
       "[15] \"brain_putamen_basal_ganglia\"          \n",
       "[16] \"brain_spinal_cord_cervical_c_1\"       \n",
       "[17] \"breast_mammary_tissue\"                \n",
       "[18] \"cells_cultured_fibroblasts\"           \n",
       "[19] \"cells_ebv_transformed_lymphocytes\"    \n",
       "[20] \"colon_sigmoid\"                        \n",
       "[21] \"colon_transverse\"                     \n",
       "[22] \"esophagus_gastroesophageal_junction\"  \n",
       "[23] \"esophagus_mucosa\"                     \n",
       "[24] \"esophagus_muscularis\"                 \n",
       "[25] \"heart_atrial_appendage\"               \n",
       "[26] \"heart_left_ventricle\"                 \n",
       "[27] \"liver\"                                \n",
       "[28] \"lung\"                                 \n",
       "[29] \"muscle_skeletal\"                      \n",
       "[30] \"nerve_tibial\"                         \n",
       "[31] \"pancreas\"                             \n",
       "[32] \"pituitary\"                            \n",
       "[33] \"skin_not_sun_exposed_suprapubic\"      \n",
       "[34] \"skin_sun_exposed_lower_leg\"           \n",
       "[35] \"small_intestine_terminal_ileum\"       \n",
       "[36] \"spleen\"                               \n",
       "[37] \"stomach\"                              \n",
       "[38] \"thyroid\"                              \n",
       "[39] \"whole_blood\"                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tissue_list  <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "tissue_list\n",
    "length(tissue_list)\n",
    "pdata_tissue_list <- factor(snakecase::to_snake_case(as.character(pData(reduced_obj2)$SMTSD)))\n",
    "levels(pdata_tissue_list)\n",
    "length(levels(pdata_tissue_list))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>adipose_subcutaneous</li><li>adipose_visceral_omentum</li><li>adrenal_gland</li><li>artery_aorta</li><li>artery_coronary</li><li>artery_tibial</li><li>brain_caudate_basal_ganglia</li><li>brain_cerebellar_hemisphere</li><li>brain_cerebellum</li><li>brain_cortex</li><li>brain_frontal_cortex_ba_9</li><li>brain_hippocampus</li><li>brain_hypothalamus</li><li>brain_nucleus_accumbens_basal_ganglia</li><li>brain_putamen_basal_ganglia</li><li>brain_spinal_cord_cervical_c_1</li><li>breast_mammary_tissue</li><li>cells_cultured_fibroblasts</li><li>cells_ebv_transformed_lymphocytes</li><li>colon_sigmoid</li><li>colon_transverse</li><li>esophagus_gastroesophageal_junction</li><li>esophagus_mucosa</li><li>esophagus_muscularis</li><li>heart_atrial_appendage</li><li>heart_left_ventricle</li><li>liver</li><li>lung</li><li>muscle_skeletal</li><li>nerve_tibial</li><li>pancreas</li><li>pituitary</li><li>skin_not_sun_exposed_suprapubic</li><li>skin_sun_exposed_lower_leg</li><li>small_intestine_terminal_ileum</li><li>spleen</li><li>stomach</li><li>thyroid</li><li>whole_blood</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'adipose_subcutaneous'</li><li>'adipose_visceral_omentum'</li><li>'adrenal_gland'</li><li>'artery_aorta'</li><li>'artery_coronary'</li><li>'artery_tibial'</li><li>'brain_caudate_basal_ganglia'</li><li>'brain_cerebellar_hemisphere'</li><li>'brain_cerebellum'</li><li>'brain_cortex'</li><li>'brain_frontal_cortex_ba_9'</li><li>'brain_hippocampus'</li><li>'brain_hypothalamus'</li><li>'brain_nucleus_accumbens_basal_ganglia'</li><li>'brain_putamen_basal_ganglia'</li><li>'brain_spinal_cord_cervical_c_1'</li><li>'breast_mammary_tissue'</li><li>'cells_cultured_fibroblasts'</li><li>'cells_ebv_transformed_lymphocytes'</li><li>'colon_sigmoid'</li><li>'colon_transverse'</li><li>'esophagus_gastroesophageal_junction'</li><li>'esophagus_mucosa'</li><li>'esophagus_muscularis'</li><li>'heart_atrial_appendage'</li><li>'heart_left_ventricle'</li><li>'liver'</li><li>'lung'</li><li>'muscle_skeletal'</li><li>'nerve_tibial'</li><li>'pancreas'</li><li>'pituitary'</li><li>'skin_not_sun_exposed_suprapubic'</li><li>'skin_sun_exposed_lower_leg'</li><li>'small_intestine_terminal_ileum'</li><li>'spleen'</li><li>'stomach'</li><li>'thyroid'</li><li>'whole_blood'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item adipose\\_subcutaneous\n",
       "\\item adipose\\_visceral\\_omentum\n",
       "\\item adrenal\\_gland\n",
       "\\item artery\\_aorta\n",
       "\\item artery\\_coronary\n",
       "\\item artery\\_tibial\n",
       "\\item brain\\_caudate\\_basal\\_ganglia\n",
       "\\item brain\\_cerebellar\\_hemisphere\n",
       "\\item brain\\_cerebellum\n",
       "\\item brain\\_cortex\n",
       "\\item brain\\_frontal\\_cortex\\_ba\\_9\n",
       "\\item brain\\_hippocampus\n",
       "\\item brain\\_hypothalamus\n",
       "\\item brain\\_nucleus\\_accumbens\\_basal\\_ganglia\n",
       "\\item brain\\_putamen\\_basal\\_ganglia\n",
       "\\item brain\\_spinal\\_cord\\_cervical\\_c\\_1\n",
       "\\item breast\\_mammary\\_tissue\n",
       "\\item cells\\_cultured\\_fibroblasts\n",
       "\\item cells\\_ebv\\_transformed\\_lymphocytes\n",
       "\\item colon\\_sigmoid\n",
       "\\item colon\\_transverse\n",
       "\\item esophagus\\_gastroesophageal\\_junction\n",
       "\\item esophagus\\_mucosa\n",
       "\\item esophagus\\_muscularis\n",
       "\\item heart\\_atrial\\_appendage\n",
       "\\item heart\\_left\\_ventricle\n",
       "\\item liver\n",
       "\\item lung\n",
       "\\item muscle\\_skeletal\n",
       "\\item nerve\\_tibial\n",
       "\\item pancreas\n",
       "\\item pituitary\n",
       "\\item skin\\_not\\_sun\\_exposed\\_suprapubic\n",
       "\\item skin\\_sun\\_exposed\\_lower\\_leg\n",
       "\\item small\\_intestine\\_terminal\\_ileum\n",
       "\\item spleen\n",
       "\\item stomach\n",
       "\\item thyroid\n",
       "\\item whole\\_blood\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'adipose\\_subcutaneous'\n",
       "\\item 'adipose\\_visceral\\_omentum'\n",
       "\\item 'adrenal\\_gland'\n",
       "\\item 'artery\\_aorta'\n",
       "\\item 'artery\\_coronary'\n",
       "\\item 'artery\\_tibial'\n",
       "\\item 'brain\\_caudate\\_basal\\_ganglia'\n",
       "\\item 'brain\\_cerebellar\\_hemisphere'\n",
       "\\item 'brain\\_cerebellum'\n",
       "\\item 'brain\\_cortex'\n",
       "\\item 'brain\\_frontal\\_cortex\\_ba\\_9'\n",
       "\\item 'brain\\_hippocampus'\n",
       "\\item 'brain\\_hypothalamus'\n",
       "\\item 'brain\\_nucleus\\_accumbens\\_basal\\_ganglia'\n",
       "\\item 'brain\\_putamen\\_basal\\_ganglia'\n",
       "\\item 'brain\\_spinal\\_cord\\_cervical\\_c\\_1'\n",
       "\\item 'breast\\_mammary\\_tissue'\n",
       "\\item 'cells\\_cultured\\_fibroblasts'\n",
       "\\item 'cells\\_ebv\\_transformed\\_lymphocytes'\n",
       "\\item 'colon\\_sigmoid'\n",
       "\\item 'colon\\_transverse'\n",
       "\\item 'esophagus\\_gastroesophageal\\_junction'\n",
       "\\item 'esophagus\\_mucosa'\n",
       "\\item 'esophagus\\_muscularis'\n",
       "\\item 'heart\\_atrial\\_appendage'\n",
       "\\item 'heart\\_left\\_ventricle'\n",
       "\\item 'liver'\n",
       "\\item 'lung'\n",
       "\\item 'muscle\\_skeletal'\n",
       "\\item 'nerve\\_tibial'\n",
       "\\item 'pancreas'\n",
       "\\item 'pituitary'\n",
       "\\item 'skin\\_not\\_sun\\_exposed\\_suprapubic'\n",
       "\\item 'skin\\_sun\\_exposed\\_lower\\_leg'\n",
       "\\item 'small\\_intestine\\_terminal\\_ileum'\n",
       "\\item 'spleen'\n",
       "\\item 'stomach'\n",
       "\\item 'thyroid'\n",
       "\\item 'whole\\_blood'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. adipose_subcutaneous\n",
       "2. adipose_visceral_omentum\n",
       "3. adrenal_gland\n",
       "4. artery_aorta\n",
       "5. artery_coronary\n",
       "6. artery_tibial\n",
       "7. brain_caudate_basal_ganglia\n",
       "8. brain_cerebellar_hemisphere\n",
       "9. brain_cerebellum\n",
       "10. brain_cortex\n",
       "11. brain_frontal_cortex_ba_9\n",
       "12. brain_hippocampus\n",
       "13. brain_hypothalamus\n",
       "14. brain_nucleus_accumbens_basal_ganglia\n",
       "15. brain_putamen_basal_ganglia\n",
       "16. brain_spinal_cord_cervical_c_1\n",
       "17. breast_mammary_tissue\n",
       "18. cells_cultured_fibroblasts\n",
       "19. cells_ebv_transformed_lymphocytes\n",
       "20. colon_sigmoid\n",
       "21. colon_transverse\n",
       "22. esophagus_gastroesophageal_junction\n",
       "23. esophagus_mucosa\n",
       "24. esophagus_muscularis\n",
       "25. heart_atrial_appendage\n",
       "26. heart_left_ventricle\n",
       "27. liver\n",
       "28. lung\n",
       "29. muscle_skeletal\n",
       "30. nerve_tibial\n",
       "31. pancreas\n",
       "32. pituitary\n",
       "33. skin_not_sun_exposed_suprapubic\n",
       "34. skin_sun_exposed_lower_leg\n",
       "35. small_intestine_terminal_ileum\n",
       "36. spleen\n",
       "37. stomach\n",
       "38. thyroid\n",
       "39. whole_blood\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'adipose_subcutaneous'\n",
       "2. 'adipose_visceral_omentum'\n",
       "3. 'adrenal_gland'\n",
       "4. 'artery_aorta'\n",
       "5. 'artery_coronary'\n",
       "6. 'artery_tibial'\n",
       "7. 'brain_caudate_basal_ganglia'\n",
       "8. 'brain_cerebellar_hemisphere'\n",
       "9. 'brain_cerebellum'\n",
       "10. 'brain_cortex'\n",
       "11. 'brain_frontal_cortex_ba_9'\n",
       "12. 'brain_hippocampus'\n",
       "13. 'brain_hypothalamus'\n",
       "14. 'brain_nucleus_accumbens_basal_ganglia'\n",
       "15. 'brain_putamen_basal_ganglia'\n",
       "16. 'brain_spinal_cord_cervical_c_1'\n",
       "17. 'breast_mammary_tissue'\n",
       "18. 'cells_cultured_fibroblasts'\n",
       "19. 'cells_ebv_transformed_lymphocytes'\n",
       "20. 'colon_sigmoid'\n",
       "21. 'colon_transverse'\n",
       "22. 'esophagus_gastroesophageal_junction'\n",
       "23. 'esophagus_mucosa'\n",
       "24. 'esophagus_muscularis'\n",
       "25. 'heart_atrial_appendage'\n",
       "26. 'heart_left_ventricle'\n",
       "27. 'liver'\n",
       "28. 'lung'\n",
       "29. 'muscle_skeletal'\n",
       "30. 'nerve_tibial'\n",
       "31. 'pancreas'\n",
       "32. 'pituitary'\n",
       "33. 'skin_not_sun_exposed_suprapubic'\n",
       "34. 'skin_sun_exposed_lower_leg'\n",
       "35. 'small_intestine_terminal_ileum'\n",
       "36. 'spleen'\n",
       "37. 'stomach'\n",
       "38. 'thyroid'\n",
       "39. 'whole_blood'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] adipose_subcutaneous                  adipose_visceral_omentum             \n",
       " [3] adrenal_gland                         artery_aorta                         \n",
       " [5] artery_coronary                       artery_tibial                        \n",
       " [7] brain_caudate_basal_ganglia           brain_cerebellar_hemisphere          \n",
       " [9] brain_cerebellum                      brain_cortex                         \n",
       "[11] brain_frontal_cortex_ba_9             brain_hippocampus                    \n",
       "[13] brain_hypothalamus                    brain_nucleus_accumbens_basal_ganglia\n",
       "[15] brain_putamen_basal_ganglia           brain_spinal_cord_cervical_c_1       \n",
       "[17] breast_mammary_tissue                 cells_cultured_fibroblasts           \n",
       "[19] cells_ebv_transformed_lymphocytes     colon_sigmoid                        \n",
       "[21] colon_transverse                      esophagus_gastroesophageal_junction  \n",
       "[23] esophagus_mucosa                      esophagus_muscularis                 \n",
       "[25] heart_atrial_appendage                heart_left_ventricle                 \n",
       "[27] liver                                 lung                                 \n",
       "[29] muscle_skeletal                       nerve_tibial                         \n",
       "[31] pancreas                              pituitary                            \n",
       "[33] skin_not_sun_exposed_suprapubic       skin_sun_exposed_lower_leg           \n",
       "[35] small_intestine_terminal_ileum        spleen                               \n",
       "[37] stomach                               thyroid                              \n",
       "[39] whole_blood                          \n",
       "39 Levels: adipose_subcutaneous adipose_visceral_omentum ... whole_blood"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "processing adipose_subcutaneous \n",
      "\n",
      "splice_list does contain\n",
      "se_ continuing with processing\n",
      "\n",
      "\n",
      "Limiting phenotype data to tissue of interest\n",
      "adipose_subcutaneous \n",
      "\n",
      "keep\n",
      "9945427 \n",
      "\n",
      "tissue_obj now reduced to the tissue of interest\n",
      "427 67\n",
      "\n",
      "sample value\n",
      "GTEX.1117F.0226.SM.5GZZ7 c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = 1657) c(`GTEX-1117F-0226-SM-5GZZ7` = 23) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) 1 c(`GTEX-1117F-0226-SM-5GZZ7` = 30) c(`GTEX-1117F-0226-SM-5GZZ7` = 1393) c(`GTEX-1117F-0226-SM-5GZZ7` = 812) c(`GTEX-1117F-0226-SM-5GZZ7` = 1045) c(`GTEX-1117F-0226-SM-5GZZ7` = 6) c(`GTEX-1117F-0226-SM-5GZZ7` = 595) c(`GTEX-1117F-0226-SM-5GZZ7` = 261) c(`GTEX-1117F-0226-SM-5GZZ7` = 79) c(`GTEX-1117F-0226-SM-5GZZ7` = 18) c(`GTEX-1117F-0226-SM-5GZZ7` = 3) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 4430) c(`GTEX-1117F-0226-SM-5GZZ7` = 10620) c(`GTEX-1117F-0226-SM-5GZZ7` = 5305) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 5255) c(`GTEX-1117F-0226-SM-5GZZ7` = 3092) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 4350) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 7872) c(`GTEX-1117F-0226-SM-5GZZ7` = 68) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = 3866) c(`GTEX-1117F-0226-SM-5GZZ7` = 12372) c(`GTEX-1117F-0226-SM-5GZZ7` = 1693) c(`GTEX-1117F-0226-SM-5GZZ7` = 3547) c(`GTEX-1117F-0226-SM-5GZZ7` = 11684) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 4365) c(`GTEX-1117F-0226-SM-5GZZ7` = 3764) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = 14999) c(`GTEX-1117F-0226-SM-5GZZ7` = 5255) c(`GTEX-1117F-0226-SM-5GZZ7` = 3804) c(`GTEX-1117F-0226-SM-5GZZ7` = 3866) c(`GTEX-1117F-0226-SM-5GZZ7` = 11023) c(`GTEX-1117F-0226-SM-5GZZ7` = 3493) c(`GTEX-1117F-0226-SM-5GZZ7` = 6574) c(`GTEX-1117F-0226-SM-5GZZ7` = 3501) c(`GTEX-1117F-0226-SM-5GZZ7` = 56) c(`GTEX-1117F-0226-SM-5GZZ7` = 3540) c(`GTEX-1117F-0226-SM-5GZZ7` = 7254) c(`GTEX-1117F-0226-SM-5GZZ7` = 9437) c(`GTEX-1117F-0226-SM-5GZZ7` = 3556) c(`GTEX-1117F-0226-SM-5GZZ7` = 4463) c(`GTEX-1117F-0226-SM-5GZZ7` = 2569) c(`GTEX-1117F-0226-SM-5GZZ7` = 6540) c(`GTEX-1117F-0226-SM-5GZZ7` = NA) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = 3039) c(`GTEX-1117F-0226-SM-5GZZ7` = 1) c(`GTEX-1117F-0226-SM-5GZZ7` = 2) c(`GTEX-1117F-0226-SM-5GZZ7` = 5) c(`GTEX-1117F-0226-SM-5GZZ7` = 5)\n",
      "\n",
      "Limit the phenotypes to those we have samples for\n",
      "133 294\n",
      "\n",
      "tissue_count_obj now has these count specific phenotypes\n",
      "\n",
      "Dimensions of tissue_count_obj\n",
      "294 67\n",
      "\n",
      "Dimensions of tissue_count_obj \n",
      "38458 427\n",
      "\n",
      "Limit the counts now has these count specific phenotypes \n",
      "6726 294\n",
      "\n",
      "dimensions of the ijc_tissue \n",
      "42484 294\n",
      "\n",
      "dimensions of the ijc_tissue \n",
      "42484 294\n",
      "\n",
      "dimensions of the metadata_tissue \n",
      "294 80\n",
      "\n",
      "The non-overlapping IDs between pheno and count data are:\n",
      "0\n",
      "\n",
      "ijc.rs > 10 \n",
      "6157 36327\n",
      "\n",
      "sjc.rs > 10 \n",
      "9131 33353\n",
      "\n",
      "keep combined ijc | sjc \n",
      "3855 38629\n",
      "\n",
      "dim(ijc.dm)\n",
      "38629 294\n",
      "\n",
      "dim(sjc.dm)\n",
      "38629 294\n",
      "\n",
      "\n",
      "\n",
      "sex samples:\n",
      "190\n",
      "104\n",
      "\n",
      "sex samples:\n",
      "190 104\n",
      "\n",
      "design matrix ijc, alone:\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "0\n",
      "0\n",
      "1\n",
      "0\n",
      "\n",
      "dimensions of the ijc_sex_results_refined_annotations \n",
      "99 11\n",
      "\n",
      "length ijc_sex_results_refined_geneSymbols\n",
      "99\n",
      "Warning message in file(file, ifelse(append, \"a\", \"w\")):\n",
      "“cannot open file '../csv/se_adipose_subcutaneous_DGE_ijc_sex.csv': No such file or directory”"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in file(file, ifelse(append, \"a\", \"w\")): cannot open the connection\n",
     "output_type": "error",
     "traceback": [
      "Error in file(file, ifelse(append, \"a\", \"w\")): cannot open the connection\nTraceback:\n",
      "1. print_exploratory_plots(plot, dup, tissue_of_interest, splice_type, \n .     fromGTF, tissue_list, ijc, sjc, obj, metadata)",
      "2. write.table(ijc_sex_results, file = ijc_sex_filename, row.names = T, \n .     col.names = T, quote = F, sep = \",\")   # at line 168-169 of file <text>",
      "3. file(file, ifelse(append, \"a\", \"w\"))"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzdebzV9J3/8c8B7gZcFpV9kXVYrSiKVKQyUrkVLCKlIlqXsVoV7FgVZ0AF\npij212ltsQLVuosLIggibi0iCmoFFBQrdWSzFdlElMtygQvn90eGTEi++eZ7zknOkvt6/sHj\n3Jzkm0+Sw837fr9JTiKZTAoAAAAKX61cFwAAAIBwEOwAAABigmAHAAAQEwQ7AACAmCDYAQAA\nxATBDgAAICYIdgAAADFBsAMAAIgJgh0AAEBMEOwAAABigmAHAAAQEwQ7AACAmCDYAQAAxATB\nDgAAICYIdgAAADFBsAMAAIgJgh0AAEBMEOwAAABigmAHAAAQEwQ7AACAmCDYAQAAxATBDgAA\nICYIdgAAADFBsAMAAIgJgh0AAEBMEOwAAABigmAHAAAQEwQ7AACAmCDYAQAAxATBDgAAICYI\ndgAAADFBsAMAAIgJgh0AAEBMEOwAAABigmAHAAAQEwQ7AACAmCDYAQAAxATBDgAAICYIdgAA\nADFBsAOA3JszZ04ikUgkEqWlpbmuJebmz59v7eo6derkuhYgfAQ7AGF66KGHEg4PP/ywd57H\nH3/cOc9DDz2U/ToBIJb4ewVAhObPn//Tn/7UNXHBggU5KSaf9erV6/e//72IxKMbadu2ba1a\ntTp8+PDatWu7du2a63KAGiQOv0EA5K1Fixbt3bu3Xr169pQDBw78+c9/zmFJ+alTp06/+MUv\ncl1FaGbPnn348OFcVwHURAzFAohE69atRaSqquq1115zTl+8ePGePXtEpFWrVrmpDNF79tln\nc10CUEMR7IBoff/737euJBs0aJDrrZkzZ1pvFRUV7dixw5q4e/fuu+++u2/fvscdd1xxcXGz\nZs0qKioeffRRZf+H4cyPPPKItaLvfe97IjJnzpxTTjmlbt267dq1u+OOOw4dOiQia9euHTp0\naOPGjevXrz9o0KC//e1vzhbefffdOXPmzJkz59133zXc8D59+tSuXVtEXnjhBed0axy2du3a\nvXv3Vi746aefXnfddZ07dy4tLW3QoMHpp5/+hz/8obq62jlPMpmcNWtWRUVF06ZNi4qKGjRo\ncMYZZ9x3333ODXdt9dKlS88991xrA/v3779o0aLATUjp2KVU0tlnn11dXX3jjTc2adKkWbNm\n4n/zRERbunnz5ptvvrl79+716tUrKyvr0aPHuHHj7A9hSsfC5fzzz08kEm+//bb1Y7du3RKJ\nhNUZqdl8w9WlsaWPPfbYaaedVq9eveOOO27IkCErV65MJBKa+oGClwQQJfvugeLi4t27dzvf\nGj58uPXW4MGDrSmrV6/268f67ne/+9VXXzkXN5951qxZ1vSePXvOnj3bdWIbM2bMhg0bTjjh\nBOfEJk2afPPNN3YLF1xwgTX9ggsu0G/vgw8+aM35k5/8pG/fviJy3HHHVVdX2zNYPXnf/e53\nR4wYYc354IMP2u/OnTtXeVvoOeecs3//fnu2Sy+9VLnh559//pEjR7xb/dprrxUXFzvnrF27\n9qJFi0I8diYlPf3009bE73znO/fcc49dSTKZfO6556wfS0pKnCuKYksXL17csGFDb5vNmjX7\n6KOPUj0WLkOGDPEucuONN+o333B1qW7pLbfc4mqtpKTkjjvucK0aiBOCHRCtXbt22aef2bNn\n29P3799vX3n29NNPJ5PJr7/+2g5q7du3/+Mf/zh//vz//M//tK+mHzJkiL14SjPboaFFixat\nW7ceOnToDTfcYJ/aS0pKzjvvvJYtW954441nnHGGfQr8/e9/b7eQRrC76KKLJk2aZL1evHix\n9e77779vTZk0adKwYcOs13aw27BhQ1lZmTXx1ltv/fTTT1esWHH22WdbU8aPH2/N9uKLL1pT\natWqNWPGjDVr1jz88MP2hts72bnV7dq169Wr1/jx4519b3369Anr2KVaUrt27dq0aVNUVNSr\nV68uXbokfYJdFFu6ZcuW4447zpp+9tlnz5kz58knn+zVq5c1pUuXLocOHTI/Fl5/+9vfnH20\nTz311NKlSzds2KDffMPVpbSl7733nj39nHPOef755xcsWPCDH/zA3oEEO8QSwQ6InJ2KLrvs\nMnuiffIrLy/fu3dvMpn85S9/aU1p0KDB5s2b7Tlnzpxpn59WrlxpTUxpZvt0KCI//vGPrYlz\n5861J5aWlq5fvz6ZTB44cKBLly7WxB/+8IfeTTAPdj/60Y+WL19uvbY6bJLJpB31li9fbrdp\nB7sbbrjBmjJgwAC7wR07dtSvX9/aUVbPzbRp04YMGTJkyBC72WQyOXToUGvZyy+/3LvV/fv3\nt3t97D6wWrVqHTx4UL85hscujZI6d+78z3/+055ZGeyi2NLx48dbE0844QSr+GQyuX37djtX\nzZkzx/xYKG3ZssWuZ+3atd5t9G6+4epS2tJrrrnGmtikSZM9e/ZYEw8dOmTfpUuwQywR7IDI\n2eNHxx9/vD0oeeWVV1oTr7jiCmvKySefbE258sornYtXV1c3btzYemvy5MlpzOw8HS5fvtya\nePDgwZKSEmviqFGj7BbGjh1rTezVq1caG+sMdkeOHLG6Fdu1a2e9e8opp4hIq1atjhw54g12\nHTt2tKbcfvvt+x2sC6pERDN4+vOf/9yaZ9CgQd6ttrsMk8mk80qsTZs26TfH8NilUZLV1Wfz\nG4oNfUt79uxpTbn66qudzb799tuvvPLKK6+88tlnnyUzOxYmwc61+YarS2lLu3XrZk256qqr\nnOu68847rekEO8QSN08AkRs6dKjV8bBz58533nlHRA4fPrxw4ULr3Z/85Ccikkwm7fsVTjrp\nJOfitWvXtvsY1q5dm+rMLvZ5vaioqEmTJtbr73znO/YMLVq0sF5UVlamsbFOiUTCSm+bNm36\n6KOPvvjii1WrVonIsGHDvBewJ5PJjRs3Wq+nTJlS5vDWW29Z0+2tXrRo0bBhwzp27FhaWmpd\nTX/fffdZbynvMjnttNPs1yeeeKL92trG3bt3t/M4cOCAmB279Eo655xzgvdg2FuaTCbtT4Wd\npSxnnnnmD37wgx/84AedOnVK6Vikx7n56a1Ov6UismnTJutF586dnQv26NEjk8qBPMdz7IDI\nlZWVXXDBBU899ZSIvPjii/3791+6dOlXX30lIi1btrTOcPv27bPv/rOShJN9RZd10kppZqeS\nkhJ7xM360XrRqFEje6LrgvQMDRs2bMaMGSLyl7/8pW7duvZE75z79u07cuSIvrWvv/5aRP74\nxz+OHj3amlKvXr0OHToUFxdv3rzZ2qVeJSUl5eXl9o/eK/SPHDny+eefuyYmk0kxO3ZplFS7\ndm07VWuEvqX79u2z46Dds+tlfizS49r8NFYXuKXJZHL//v3Wa/uDp/wRiBl67IBsGDVqlPXC\netjH/PnzrR8vueSSWrVqiUjdunXta7p3797tWtyOaNYdDynNnFsDBgywUuPixYtff/11EWnU\nqJF9UbxT3bp1rcejiMjUqVOVQwz/9V//tWfPHnuw+JJLLtmxY8cnn3yyevXqiy66KKJNCDx2\naZRUp04da1mNKLa0rKzMXu/evXv9ZjM8FmmX4dr8KFbnfHCMa0u9/2WAOCHYAdkwaNCg448/\nXkQ+/fTT//mf/7HDgT2Wl0gk7EHS1atXO5c9dOjQJ598Yr22xkxTmjm3ioqKBg8eLCJLly59\n4403RGTIkCFFRUXeOROJhD04aA/Mea1atWrfvn3W61tvvdXugPz000/TLrJRo0beJGHHgsBj\nF0VJETVbq1Yte1zSNVL/9NNP33XXXXfddderr75qeCzCEtHq2rZta734+9//7pxuXQ8AxBXB\nDsiGoqIi+7FtkydPtgb+TjrpJPseCBGxH402f/78L774wp7+2GOPWZ1wiUTCHsRMaebMvffe\ne/Pnz58/f77zERKGrDIqKyut0TRNVRUVFdaL5557zs40hw8fvvTSS6+66qrx48dv3rzZuvTN\nYj1aWUQ++eQTKzU6J4Yl8NhFVFJEzVo5W0Tmz5//7bffWq937dp1ww03TJgwYcKECdbHyeRY\n+K3CeQGl36ixSyar8/Pd737XerFgwYJdu3ZZr/fs2eO8cxyIoZRvtwCQliVLlrj+9/361792\nzrBr1y7r4b0i0qlTp+nTp8+dO/fWW2+1r4Rz3saY0sx+d1za3SR//OMf7Yn2tfkdO3a0J6b3\nuBNrSmVlpV1VaWlpZWWlq037rtj169fb/VL9+vV76aWXXnvtNTvC9ujRo7q6evPmzfYo3g9/\n+MM1a9YsWLCgVatW9lNaGjRo8O67727bts1vq//5z3/ah2DNmjVBxy2ZDDp2mZekPEYRbenn\nn39uX512xhlnPPvss0899ZT9LSBt27a1ngxiciz8dld1dbXdKdu/f//Zs2e/9tprym20Ga4u\npS11HrXevXvPmjXr8ccfP/300+3LUmvVqmVy9IHCQrADssR+9od9UnE+x8ui+TKJ4cOHu54c\nZj5zboNdMpm0e4mcj032BjurVDsFOrVq1cp+cIb9vA9by5YtN27c2LJlS3vKpEmTQgx2gccu\nw5KSPscooi1duHCh8gaC5s2br169OqVj4ee8885zLmIdd/0jXUxWl+qW/tu//ZurtXr16k2b\nNs3+8fDhw/oNAQoOQ7FAliQSiZEjR9o/nn322XaXm+3kk0/+5JNP7rrrrtNPP71hw4ZFRUUt\nWrS48MILFyxY4P3CpZRmzq0LL7zQehE4OjxixIhVq1ZdddVV7du3LykpqVu3bs+ePW+//fY1\na9bYj3G55557fv3rX3fv3r2srKxVq1ZXX331ihUr2rVr9/jjj3fp0qVOnTqtW7e2n2EWisBj\nF1FJETU7ZMiQjz/++Prrr+/cubP1VJEePXqMHz9+zZo1zmsDTI6FnwcffHDYsGGNGjUqLS1t\n3779mWeeGVhVJqvz89BDD91zzz1dunQpLi5u2rTpiBEj3nvvvX/913+1Z9DcQQIUqEQymcx1\nDQAAAAgBPXYAAAAxQbADAACICYIdAABATBDsAAAAYoJgBwAAEBMEOwAAgJgg2AEAAMQEwQ4A\nACAmCHYAAAAxQbADAACICYIdAABATBDsAAAAYoJgBwAAEBMEOwAAgJgg2AEAAMQEwQ4AACAm\nCHYAAAAxQbADAACICYIdAABATBDsAAAAYoJgBwAAEBMEOwAAgJgg2AEAAMQEwQ4AACAmCHYA\nAAAxQbADAACICYIdAABATBDsAAAAYoJgBwAAEBMEOwAAgJgg2AEAAMQEwQ4AACAmCHYAAAAx\nQbADAACICYIdAABATBDsAAAAYoJgBwAAEBMEOwAAgJiok+sCCsOHH35YXV2d6yoAAEBeqFOn\nzsknn5zrKhQIdsFWrlx5+umn57oKAACQR1asWHHaaaflugo3gl2wgwcPisiBAweKi4tzXQsA\nAMixgwcPlpSUWPEg3xResEsmkxs3btywYUNlZaWINGzYsHPnzm3atMl1XQAAADlWSMFu165d\nU6ZMmTlz5vbt211vtW3b9uqrrx47dmxZWVlOagMAAMi5ggl2W7Zs6dev38aNGzt37jx48OAT\nTzyxXr16IrJ79+7169e/+eabEydOnDt37htvvNG4ceNcFwsAAJADBRPsJkyY8MUXX8yePfvH\nP/6x993Dhw8/8MADN9xwwy9/+cupU6dmvzwAAICcK5jn2L300kuXXXaZMtWJSO3atUePHn3R\nRRc9//zzWS4MAAAgTxRMsNu5c2fHjh3183Tr1m3btm3ZqQcAACDfFEywa9my5YcffqifZ9Wq\nVS1btsxOPQAAAPmmYILdsGHDnnvuud/+9rcHDhzwvrt3795Jkya98MILI0eOzH5tAAAA+SCR\nTCZzXYORb775ZuDAgR988EF5eXmfPn3atGlTv379ZDK5Z8+ezz//fPny5fv27evfv//LL79c\nv379cFf9zjvv9OvXjwcUAwAAOfqA4rfffvvMM8/MdS1uBXNXbKNGjd59993p06c/8cQTS5Ys\nOXz4sP1WUVFR7969r7rqqquuuqp27do5LBIAACCHCibYiUhxcfFNN9100003VVVV/fOf/7S+\neaJBgwZt27alLw0AAKCQgp2ttLS0c+fOua4CAAAgvxTMzRMAAADQK8geO6X169dfe+21IrJo\n0SLzpfbu3fub3/ymqqpKM8/nn3+eaXEAAADRi0+wq6ysfP3111Ndas+ePStWrDh48KBmns2b\nN4sId8UCAIA8F59g17Vr1zVr1qS6VLNmzV566SX9PA888MB1112XSCTSLQ0AACAb4hPsSktL\ne/bsmesqAAAAciZWN0/s2rVr06ZNua4CAAAgNwop2H300UdDhgxp165d//79Z8yY4XxGseXX\nv/51+/btc1IbAABAzhXMUOzbb789cODAAwcO1K1b98svv1y2bNns2bPnzZvXuHHjXJcGAACQ\nFwqmx+5Xv/rVkSNH5s2bt2fPnsrKyt/97nfvvPNORUXF3r17c10aUHi4GQgAYqlggt1HH300\ncuTIYcOGJRKJkpKSm2666dVXX/3www8vuugi75gsAA0r1ZHtACB+CibYbd26tUOHDs4p55xz\nzkMPPfTyyy/ffPPNuaoKKETJo3JdCAAgZAVzjV2zZs1Wr17tmnjZZZetXbv2V7/6VevWrW+9\n9dacFAYAAJAnCibYDR8+/L777ps2bdq1115bVFRkT58yZcqXX375H//xH19++SVjsgAAoCYr\nmGA3ceLE+fPn//znP3/hhRf+8pe/2NMTicSjjz7asGHDqVOn5rA8AACAnCuYa+yOP/74999/\nf/To0d6vl0gkEvfee+/cuXM7duyYk9oAAADyQcH02InICSecMH36dL93hw8fPnz48GzWAwAA\nkFcKpscOAAAAegQ7AACAmCDYAQAAxATBDgAAICYIdgAAADFBsAPiL5FI8M2wAFATEOyAAkZi\nAwA4FdJz7ACkJ5lM5roEAEA2EOyAAkZiAwA4MRQLAAAQEwQ7AACAmCDYAQAAxATBDgAAICYI\ndgAAADFBsAMKCQ+uAwBoEOyA+CD2AUANx3PsgELCg+sAABoEOyA+iH0AUMMxFAsAABATBDsA\nAICYINgBAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsAAICY\nINgBAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsAAICYINgB\nAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsAAICYINgBAADE\nBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsAAICYINgBAADEBMEO\nyLZEIpFIJHJdBQAghgh2AAAAMVEn1wUANU4ymcx1CQCAeKLHDgAAICYIdgAAADFBsAMAAIgJ\ngh0AAEBMEOwAAABigmAHAAAQEwQ7AACAmCDYAfhffCUGABQ6gh0AAEBM8M0TAP4XX4kBAIWO\nHjsAAICYINgBAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsA\nAICYINgBAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsAAICY\nINgBAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABATBDsAAICYINgB\nAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYIcaJ5FIJBKJXFcBAED4CHYA\nAAAxUSfXBQDZlkwmc10CAACRoMcOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcA\nABATBDsAAICYINgBAADEBMEOAAAgJgh2AAAAMUGwAwAAiAmCHQAAQEwQ7AAAAGKCYAcAABAT\nBDsAAICYINgBAADEBMEOAAAgJgh2QPwlEolEIpHrKgAAkSPYAQAAxESdXBcAIHLJZDLXJQAA\nsoEeOwAAgJgg2AEAAMQEwQ4AACAmCHYAAAAxQbADAACICYIdAABATBDsAAAAYoJgBwAAEBM8\noBg1jvXlWjX2mb32d4vV2D0AADFGjx1Qs1h5jlQHALFEjx1qHDINewAA4ooeO8SBPbwIAEBN\nRrBDwSPVAQBgIdih4DGwCACAhWCHwpNIJFy9dGS7Gsj7MQAAEOwAAABigrtiUXjon4PwMQAA\nFXrsABQwRmMBwIlgB6Cwke0AwEawQzZwnTuiw5gsANgIdsgGTr2ICB8tAHAi2CFL4nQCzvPe\nx8RR4TYYVmsAgOgQ7IDUWBEnn4OOlaHjlKQBAIZ43AkKmxWwshliCiIwhVtkQWwyAEDosQMA\nAIgNeuxQ2OhMAgDARo8d8kU+X7UGAEBBINghL5DqAADIHMEOeSHPR1R53gcAoCAQ7JAvsp/t\nzONanudOAAAsBDvACNkOAJD/uCsWBSmUx9eR1QAAMUOPHQAAQEwUdo/dwYMHP/zwwz179rRr\n1659+/a5LgfZk7edbdn/JgwAAGwF02N31113vfHGG84pDzzwQPPmzfv06XPOOed06NDhtNNO\nW716da7KQxS4ERUAgJQUTLCbMGHCa6+9Zv/40ksvXXfddfv27bvwwguvvfbafv36vf/++wMG\nDFi/fn0Oi4SfxFFpLBjW2qNr3ymZTNJdBwDIlUIdir3pppsaNmz47rvvduvWzZry/PPPjxgx\nYsqUKY888khua0NYspOQEokEUQyhYCAeQM4VTI+d044dOz777LMxY8bYqU5Ehg8ffsEFF/z5\nz3/OYWHQy9UJT9OLRgcbACBOCjLYVVVViYgz1Vl69uy5ffv2XFQEoCCF+50i/J0AIOcKcii2\nZcuWDRs2/OKLL1zTv/zyy/Ly8pyUBD3OdgAAZEEh9dj94x//WLly5bp163bt2jV69OiHH354\n37599rt///vfn3322X79+uWwwtjjK1MRM/zJASBmCqnH7plnnnnmmWecU1555ZUf/ehHIvL0\n00//7Gc/279//4QJE3JUHVDYTG4iievNAdxAAyA2CibYPfroo984fPvtt998803jxo2td7/5\n5ptGjRrNmjXr9NNPz22d8RbWyS+u+aDQ1dh8UzO3GkAsFUywu/LKKzXvXn755dddd12tWoU0\nsoyYiUFaDSw+mUxaw/EFvZkucdoWACiYYKdXv379XJeAFHAqzUMcFACIgZgEOyDnakgwqiGb\nCQAFKj7Bbv369ddee62ILFq0yHypDRs2dO3a9dChQ4Fzcj5DnMRg4BgA4BWfYFdZWfn666+n\nulT79u3feOON/fv3a+ZZuHDhvffey2M+4IeQBADIE6bBbtOmTevWrfvqq6++/fbbhg0bnnDC\nCZ06dWrXrl2UtaWma9eua9asSXWpRCIR+Oi79evXp1sUchN6SFqBMtk5Mbt5AgDiJCDYbdiw\nYerUqa+88sq6deu873bq1Gnw4MG/+MUv2rdvH015KSgtLe3Zs2euq0CYCiWfuSqMd+7JVdd1\n9j8M9pbG+GgCiB/fYLdjx45x48Y98cQT1dXVTZs2HTlyZOfOnZs2bdqoUaNvvvlm+/btn332\n2eLFi//whz/MmDHj8ssv/3//7/81adIkm6V77dy5c9euXZ06dcptGXDKyUkxh2diKw3EONtZ\nTzzJdRXZYG1pXI8jgLhSB7slS5aMHDly586dF1988c0333zKKacof5Unk8lVq1b97ne/e/zx\nxxcuXDh79uyzzz474oJ1fvOb3/z617/mF3FsFOKhrAm5p+aE9UL8BAKo4dRP9B00aFDXrl0/\n/fTTJ5988tRTT/U7USUSiVNPPfXJJ5/8+9//3rVr13PPPTfKUoHCkEwmCQQAgJxQB7tx48Yt\nXry4Y8eOhq106tRp8eLF48aNC68wFBLr2whyXUXOJBxyXQsAoEZTD8VOnjw51YZq166dxlLm\nTjvttMB5Nm/eHF0BQNYUyl0jAIB8UzDPsVu1apWIFBUVaeaprq7OVjk4RsFFkHAvii+4zQcA\nxJV6KDYP3XrrrfXq1fv444+r/I0dOzbXZaJgBA6b5mRc1RrP5So9AEB6CibY3XnnnZ06dRo1\napTJ13+hpkn1+rbA5GQ/tSTTygAAyKKCCXZFRUVPPfXU3/72t9tuuy3XtaCmyH63GX11AIBM\nFMw1diLSrVu3rVu3ai6kO++88xo1apTNkpAnQg9DpCsAQCEqpGAnIg0aNNC8e/bZZ+f2CcnI\nIe4kBQAgONglk8k5c+Y88cQTX3zxhfL6to8//jiCwgAAAJCa4GB3zz333HrrrSJSt25d/dNG\ngBx2m9FXl0N8pyoA5IngYHfvvfdWVFTMmDGjQ4cOWSgIyALGbUNHtgOAfBAc7LZt2zZnzhxS\nHQzlydk9+9GtJofFmrnVLvbDcdgbAHIo+HEnzZo14/cUTBTKU99qcgKDV1hf8mt9olL6XBXK\nfxkABSQ42I0aNWrmzJlZKAWFLt5RKfD0n8kj6MLKFnZrYTWFlKTxAeBgAQhX8FDsxIkTR4wY\ncemll15++eVt27b13j/RqVOnaGpD4Um1u4LbLCKS/X1buP2gOay5EHcXgDwXHOzKy8utF08/\n/bRyBn43IW3R5Y/CemRxuI3zXxIAaqzgYDdq1Kji4uI6dQrsUcbIf+SPOOFoAkA+CI5rfh11\nQP4r3PFBAADSkEI/3FdfffXZZ5/t3bu3vLy8S5cufCsr8kQNf4JaDMJrDT+CABCi4LtiRWTZ\nsmV9+/Zt0qTJmWeeee655/bt2/e44477/ve/z5eJQUK6rc+wEe/do9aPfotncqcqskN/BAEA\nKQnusVu+fPn3v//96urqs846q0uXLmVlZXv37v3kk08WL17cr1+/5cuXd+nSJQuFIp9l2ONi\nn9q9jQS2TG4r9D1Q6PUDQF4JDnZ33XVXkyZN/vKXv3Tt2tU5fdWqVT/4wQ9++cqG2DwAACAA\nSURBVMtfchFeDZf5iTmZTGo6bJzZLh4hIAaDpwCA/BQ8FPvOO++MHj3alepE5JRTThk9evTi\nxYujKQw1iyblEIAAADAU3GP37bfftm7dWvlWu3btvv7667BLQg2lDHCxTHWx3CgAQD4I7rFr\n2rTp2rVrlW998sknTZs2Dbsk5KkCvbw93G/rAgAgnwUHu0GDBt13330vvPCCs5shmUzOmzdv\n+vTp5513XpTlIb+QkJTYLXDhzwkAuRI8FDtp0qSXX3552LBhzZs37969e7169ay7Yrdu3dqi\nRYtJkyZloUrkCSvcF9a1/8o7bZXTzWfwzsyT2AAA+SC4x65du3YrV6684oor9u/fv3jx4hdf\nfHHx4sUHDx68+uqr33//fb/L7xA/BBcla7ewc+DEAxQB5IrRN0+0adPmscceSyaTW7du3bt3\nb/369Zs3bx51ZUhDdvrS8u2MpdlqZUdaYP36GZxDbKQ6xB690UBhUQe7rVu3lpSUNG7c2Hpt\nT08kEvXr13dNJOTVNHayyfNf9yEOkroa0T94D4gNPudAwVEHuxYtWlRUVLz66qvWa30TeX52\nr1GycyysWKO/Ri2bnwq/dYVVg+vcZthLl2rvaQFduVgoyR6Z428YoOCog93IkSN79eplv85i\nPSgMgXcexEksz22Z50hSXQ3BgQYKizrYzZo1S/kaCBTLGJTquS2N2FRAp8+oSy2gzksAyDfB\nd8UuW7bM7+slli9fPnfu3LBLQsGzT8mZJLw8fBJYvpWUdj0kJwCIq+Bg179//7feekv51tKl\nS6+55pqwS0KshJiE8idXBVbC0y4ywd4DgLT5Pu5k3bp169ats16vWrWqtLTUNcP+/ftnz559\n4MCBCKtDgdDcLZH2GTrcWx9CaS3StJHN0VtiEwDElW+wmzNnzvjx463XkydP9pttxIgR4ReF\ngqLpuwo3QORPHEmvEgZAAQBR8w1248aNu+KKK1asWHHBBRdcdtll3bt3d81Qu3btDh06DB06\nNOIKke+yc7dE2o9QKZQgFW6dUYfInIRUkjEABNJ980SLFi2GDh06ZMiQ0aNH9+3b1zvD3r17\nd+7cyQOKkZ3bJNPLdtE9VC/VltMuI9xAk2/xKN/qAYCCFnzzxMKFC5WpTkReeOGFU089NeyS\nAIVMLqi373UI/faLPLmZwyvq+w9ycn8DN1UAQCCj74r96quvZs2atWnTpurqantiVVXVwoUL\n9+zZE1ltiIkYPws3a1/1kc3W8uSLQwAAaQgOdps2berTp8+OHTsUC9epM2HChAiqAo6RSTS0\nlrL66uJ6M0d6lHuVB+MBQEELDnZ33HFHVVXVtGnTunXrNnDgwIceeqh169ZLliyZOXPmww8/\nXFFRkYUqUVhcp/lMzvd5O9YZFte+Mk+fEWWpWH5xCADUHMHBbunSpWPGjBkzZkxVVZWI9OjR\no2/fvhUVFSNHjhw4cOCCBQv69esXfZ2IVnopIZtjdhmuKM87k6w9meVEldKjB/WfkDzfvVCi\nnxWIpeCbJ7Zs2dKhQwcRqVWrlogcPHjQmt6rV68xY8ZMmjQp0voQnQzvJLBvR/C+FeJF7smj\nJMWCo/iaiijadG5gSvuNOwnoWQQAr+BgV15evm3bNhEpLi6uX7/+hg0b7Le6d+++cuXKCKtD\ntqSREuwsksbq8ufLwaJWQFuqLzU/c2TU+7ZQjl168vOYAsiQ0XfF3n///UuWLBGRk046afr0\n6fadsIsXLy4pKYm0PkQn8weIZPmsYPfbGc4cYnn29pq06U1IIcY7ZTsFFB/DlYUveauZOxZA\n4QoOdrfddtvOnTvHjh0rItdcc83KlSu7d+8+fPjwU0455cEHHzz33HOjLxIFI3GUfrbougr8\n1q6cnoU8FMWWRlRz6KVGvXuj/rsikz5pAMiV4Jsn+vTps2zZsuXLl4vIlVde+dlnn02dOnXe\nvHmJRGLo0KFTp06NvkjkHc1F9FnoyctVZ2HaM5s8Os5wLcp5CB/hsg9HTdix9v8m7qUA4sHo\nAcW9e/fu3bu3iCQSibvvvnvixIlbt25t1qxZWVlZxOWh8GR4YrD7ePwSjD7VpXTbZv6cw2L5\nkJH82b3wE79PHQB1sNu6dWtJSUnjxo2t194ZSktLv/3222+//VZE+K7YmiBrf80bRrfsd9pF\nrbA2J969O3HdLi/nXxQ1Z6uBeFMHuxYtWlRUVLz66qvWa30T/DqAOZNAZv6E3nz47MU74iD2\n+OgCMaMOdiNHjuzVq5f9Oov1IE+F8tvf77YGw1tNXWU4X8c+XTn3kr2xOdzqWI4dA0AMqIPd\nrFmzlK+BTPilAWdSMYwpft+O4Ep72Rk7TntZw2QWxZOWA9cbOE+MYzQAFK7gx52cffbZDz74\n4DfffJOFalC4zJ9tkXCwppg8oE7//RN+IcN+FJlzkYLranJtnfk3VRg+faZAxXjTACBtwcFu\n6dKlP/vZz5o3b/6jH/1o3rx59leKAV6u/OR33lWOqKb0/GFltrP6/Jx5UR/4cs78gRrpdY/5\nPYnNZL159bAPMhwAGAoOdv/4xz9+//vfn3rqqfPmzRs+fHjz5s2vvfbapUuX5s8vfRiK9Oxo\nmAP8ZgvsgrKLTx6lWYtrM13z2xEwsFrXqvUTDUNtlsX4/2leRU8AyBPBwa5169a/+MUv3nnn\nHSvhde3a9cEHH/ze977Xvn3722+/fe3atVmoEnlIOcznzU9+iyvfDcxDgWkppTN9iNkrT2Kc\nS/7kS8msGDIcABgKDnY2V8I78cQT//u//7t79+7RFYdwZXh2zOTEnHnCMK88w77D9OZUdgqa\nNB6uvEpyAIDsSyHY2erVq3f88ce3bt26QYMGoReE6BgOKSrf8rumLY0eMv21d67r5Lzvmq8u\nVemlIs0iuYpZrmHr7BeglFfFAEBcGX2lmGX79u3z58+fO3fu4sWLq6urGzZsOHz48FGjRkVX\nHPJHMrMvgU3pmraU2Akm7Rk0C+qHksURVfMksqR0A4qJtPcewpU/nzEAeS442G3evHnevHlz\n585dunTp4cOHy8rKhg0bdskllwwePLikpCQLJSIsmrsWTObP8LxiLx4YwjJflznlLbSpBiO/\nanN1JiYBxBLZDoCJ4GDXpk2bZDJZp06dc889d9SoURdeeGF5eXkWKkNcaTqB0ujYM7n6Tf9u\nGudL5UBnnnRuhVtGzjcHNo4FABPBwa5fv36jRo368Y9/3KRJkywUhJyoyZ0BrnAWm0vB8iRo\nIhQcRwCGgoPd0qVLrReVlZX/+Mc/WrVq1ahRo4irQg5kLdvZt0c4LwjzDtRmnktSHdhVPpfO\nfPw6T0692SwjV9mxJv8dAgB6RnfFvvnmm6eddlqDBg169uz517/+1Zo4dOjQ119/PcrakD3e\nbqqEg99Sfu9m+GCU9Bb0Sjq+dyFwWyTUgGK+B6K7c9ak6zHDtefqkS7ZXykAFIrgYLd8+fJB\ngwb9z//8T0VFhT1xx44dK1asGDx48Pvvvx9leShgmhOwM3N471pwLph5cNS04G0nH55FV1jS\nuzwxm2sEgBolONhNnjy5efPmn3zyyWOPPWZPbNKkyYcffti8efM777wzwuqQa96TqDMM+UUf\nw9tL9c/GCzdXOVvOAvN15TY+ZnntYXW2ke0AwE9wsPvrX/96/fXXt27d2jW9adOm11133Vtv\nvRVNYcgxZ8DyvhvYkWYv6zcG6u2cUzboFxxTjU2uGpIOgdulrLygJY6VxrLprZdABgBRCw52\n3377bZs2bZRvtWjRYs+ePWGXhHxkn871fXhOzlzoXSqwCy0w8ynDon78V9Og4Xo186SdeKJL\niq6SvOlWjt0t0V3w51ppoYhBiAdQ0wTfFdu8efO1a9cq33rrrbdatmwZdknII8owp7yb1Ttb\n4HQ971qsDjy/c61f2NL0O6ZRrTJNBsbTwBn0jaTEpCnnBY6ptm9yQ4Z3zsD9kAWp7mRSHYBC\nFNxjN3jw4BkzZnzwwQfOibt27br99tsfffTRIUOGRFYbskFz9jI8sYVynVZgX5H3Lgf95X3i\nyEyGqzOZXzl0a79l3vvluk4xxFTnbdx+N3BFIVaSnz1zKWW1/NwEANALDna//OUv69evf8YZ\nZ1gZbvz48aecckqLFi3uvvvutm3bTpw4MfoiERU7B+jHUv3ezfDM57de78Vwcmwnk+GFd6lm\nJsO+PW9U0iyifNfk+sL0pNo9GQXlJocYGTORag35UDMApCQ42DVv3nzlypXXXHPN559/LiKr\nV69evXp1eXn59ddfv2LFimbNmkVfJKJiOOyY9sVY5tnFPvH7hZ5wY5C3d82ebjK/yfneuYiy\nuzGi0BBi51/2xyIjXSkpDUBNEHyNnYg0bdp0xowZ06dP3759e2VlZXl5OXkuNpRnO+/JVRmD\n9JRXj/mNEupXpxwA1c9jUpuk9e206QmsMO2r0PLh8jUAQJ4wCnaWRCLRrFkzIl3NEXg9VuDi\nyoBoEsJcYcVvEeftFIbNKkcJ/ToINSvVr8jVSNqpK4ehreasFADiRB3s+vbta7j8wYMHXfdV\noEA5M4RzVFSZhAzbdLbjatyvfX0cdN0kK45sZxgWrTm9nYjmLRjO7NeCZrQ3jQahUbh9mYVb\nOYCcUwe7lStXOn+sVavWoUOHrNfOM1PDhg0bNGgQaX1IW9TnhkyGDv2W0t+p6spVzjynTJAu\nfvnP7+aGwPHfQMoWUgqFnNpzK+0EDwC5or55otphx44dffv2HTNmzOrVq/fv33/kyJHdu3cv\nW7bs4osv7t2795o1a7JcMVzCukTMO2KY4RiiMmYpnw+iuWReX5L3Xe9E53qdnYWuyJiFK+3S\nu2cisLDo7sModBnumehuXg7EMQWQtuC7YseOHduiRYtp06adfPLJpaWlIlJeXt6vX79nnnmm\nrKzslltuib5I+NKce9LLEGnfx2o4s3Ndfi9c8ysTp2ZFon1Mnd8UV170jggH7k/N3ks7OOYk\nVcCi+TsBAPJWcLB78cUXKyoqlG8NGDBgwYIFYZeEFKTXCeSSSSLx9nhZr/2qSvsKM81metce\nuArXfvMbjU2J+VFIKT3L0Y0KMeRlp4cymyuKCKkOQMEJvit29+7dO3bsUL61c+fO3bt3h10S\nQpPqZWH6S980F6655tS3oK8nsFpNMZoMYRgonWPE5onENbKsX0uqV9rZ2Y6QAQAIFNxj1717\n9/vuu2/FihWu6cuXL3/kkUe6du0aTWHIHn0icYUSfX+YPrrph4zN16UvUhOADPv2AtvJRHo9\nrOEWk7VLuFKNyACADAX32E2ePHnYsGF9+vTp1KlT+/btS0tLq6qqNm7cuG7dukQiMW3atCxU\nifSEcvI27PZLaB8ppzm7p9GtaI/2et8ybCSNxpVSLV4c6TbPO+HS2DRNOwCA7AgOdueff/6S\nJUumTJmyZMmSdevWWROLi4sHDBgwbtw4v8vvULg0gSNwNDZ57CNFNCOwqY6oOleRUmt+mxB4\nB4ZSWHHH1UhKzYZYQxbQYwcA2WT0zRNnnXXWK6+8cuTIkS1btuzbt6+srKx58+Z16qTwrRUo\nLM5w5ndjgXL009lj573d1eR2ioTjgcOanjm7Bm+OVLZpT/SLrYljH4lnKOeDqtFJtU79hZjh\n1JQi/QcPAGIphXBWq1atVq1aRVcKssPZfRLuCS+ljhllr55rJNeVHV3tO1twLq6c4qpTn+28\nGxVpMvDugQzvL8mhwupNNBfX7QIQP/S6IWWagVplV1+qT/fwct1463ftvzfeuZrNcFjQcNlU\ng1qGQlxFJk1pknSu2CXl+eWMNQThGMgOgl3hyfD3Y1L1zVr6sVfzSvx6vDTLakZITbrWNAO1\nrhnSuwc281s0AufX78DMpXQBYobJLKL6c54Gcl4AABgi2BWe0M++9gCo8l0v86DgrdO7rKuD\nTTmo6gyjhqd5Vyef861MTtKuO0X85gllXTmR84JdnxC/+6zN5XyLYOFAANkR/Bw75KFwf0Vm\ncsq0b3EQ1ZPhnJfKudZiOHKX9q2jyaOcXYN+Y7h28d5NcG6dqK7/8y7irDYsmbSm2eosrD1D\nyg8PAMAPPXb4P8qePG8/mfeuBdGe+52RyDu/sgVll54Y99xknmOUu0JUOSPpeHqfHBtYUx00\n18/vt+2hhDZNI5n3maVURuZj35rGM2wHAPIfPXYQ0XbquLqsnNPl2MTjvGDOrx9L2X4mlft1\nFjrfzSQL+iU882XD6u7KVS7J2z4z888YANQc6h67vn37Gi5/8ODBDz74ILx6kG2a6OPqS9PE\nF/0dEoZNKTvz/K7SE09XWaq8HWxJ1W0lrsI0+cbv0jpNYvbOaYWVKHqt/GShBzSsMryLGB56\nfePmV5cCQP5TB7uVK1c6f6xVq9ahQ4es186zTsOGDRs0aBBpfcgH3hOePnt5pzsH2pQ3SRjW\n4L2Gz1uJSajyq1x/ak8vfGRhEZeaM/IYyjb6BXoAKETqodhqhx07dvTt23fMmDGrV6/ev3//\nkSNHdu/evWzZsosvvrh3795r1qzJcsUIl8mV9SZDq84hUX3GUsZEu3FvHNTXbA8B+w3FOltW\nNq48qbtGeEMc8nPuTH3PnH6fB+4oZWen4Si5yeHO85FQ75UDmmoDb4UBgEIRfI3d2LFjW7Ro\nMW3atJNPPrm0tFREysvL+/Xr98wzz5SVld1yyy3RF4lwaC5Ec6WfwNO283I6ZSoy7P/wxkpl\n1jHvTUnvDK1Mb66sadJa4H5z5jnv1XupFux9bfEOaps0qAyCfvPn7YV3FsIZgBorONi9+OKL\nFRUVyrcGDBiwYMGCsEtC9vh1lbk4b4xwLuucwduyZlnvnEnHY1OcC8qxack5xbUJyrU4GxFP\nD6LmR2/x5n1p3g307ivDK9v8ZvNW7sqUyrUkHTQNmpcXOE/m0otoyj8YMtznmcj/Dk4AsREc\n7Hbv3r1jxw7lWzt37ty9e3fYJSHbXOlHSZ+lzPu0AnkTmLda18yiSlrKLVKeXDVBx5WTXCt1\ntqnPl84F/faSYaRwZVxXwa7AF9iatwXNFMNokkmI8es0LXT6Y0Hm02DnAKkKDnbdu3e/7777\nVqxY4Zq+fPnyRx55pGvXrtEUhkwps45fH5g9g3dm8QxKOmf2i0rKSpSRyO5wcr3lLEPZu6bv\nnXI1pc9bhnnFuXZR7Q0lZbWutdv7R1+Mvk5vDRkGrMBllTOYd0kGrjSi/jO/tRvuq0ySa/YV\nSp0AwhL8gOLJkycPGzasT58+nTp1at++fWlpaVVV1caNG9etW5dIJKZNm5aFKhEiTY+R38x+\n3Vd+yUYZ9ezTtjJ/aEpNqm7FcC6ibNO5oGuKssdLuV7v9MBuLT/Jo7deaho3acd8ja6WU9pe\nZ2q3p3sjaVgVZrJUuPw+S07JtO6iNeyEjkIa1eaPwq0cyJXgYHf++ecvWbJkypQpS5YsWbdu\nnTWxuLh4wIAB48aN87v8DjnnjF+uc7YzMBmGJ8N1ybH5TxMplL2Aygb16/Uu7upaC9wKv1Ao\nx+465dBnSjVrViQG8ctkFcr5Q+yzKeiIoGfyUXHOHGkx4SqsagFkyOgrxc4666xXXnnlyJEj\nW7Zs2bdvX1lZWfPmzevU4evICkZC9d1fJonH5FTnGmd0dU2Jp+NNOUUZN10ze9ernO5aUDPF\nrlMZ2jRdjJqq9EIfFzOMg66+N/EcAn3odK0x7cHWPBe/LZKYbhQAjeBwtmDBgo4dO/bo0aNW\nrVqtWrXKQk0IXWBfkTf5eU/eJt0/mtRopyhlC3YBysFTZY+g+aiZHNuN58ydmvSmzHyuGKQc\ndDasJ7AL0Cu9NBkuw7WHVar3AoBw289QnpQBALbgmydGjhy5cOHCLJSCcHk7vSzJo5SLOGOT\nk35F3gYNY4pzQW+R4gheynmUzbra9BuuNalQk0T9Fg/cXXomO9xVhuGG+B0gwxbsmTPcwFQV\nRO9gNncIAOgFB7uzzjrrzTffPHLkSBaqgQnzM6vhbN5r0TTn0aSDa0V+KU3ZiPJH/TCxa8BX\nDGKQd0O8HX7ebTEs3rmg3z5RLm6SpfzSZFgRR7ONoUQ381ID16VsJ8RdkQnDo5k/spzLAWRf\n8FDsk08+edNNNw0ZMuTyyy//l3/5l4YNG7pm6NSpUzS1IVN+nVWi7e4KbNY1OubqVDPhCnDe\nZOlt3/mjybqSjgvLlI07KzfMo2Kwf1z7JNVTvr14iAHOLxjp07O+wrAQMvJHiJ86ADkUHOya\nN29uvXj11VeVM/C7IMsMO3uUnVWuMUply66uu8CMok9FSe3l+fqtcPaHObfL9cK7oLed5LE3\navjN6ZpBWYyLqzZ7YhqnSWXHp18jyrcSPtcjaio3mZgGw4PObw+n9P6npETzcRKyHRALwcFu\n5MiRxcXFRUVF/G1diFzpxy/KKKcof8XrF1Gu0RukzHvInI14M2Ly2Dttk54BZb9eQM0U74/K\npBj4ozJ1Kaf70ZxrXb2kft2EeXWSzjA0ZCH01GTp/SkCIA8FB7tZs2b5vbV3797KyspQ64Fa\nSr9zXbFJ2aWkHND0prrAy9cCc55z1clj70V1XZzkjWXO1vRjyt5w6Vew39oN++SU7GUz7PbQ\nR0w5NjUqg2waK42Iqxg6hEzkdudwaIB4CL55QuOFF1449dRTwyoFftK4iE3TlDPSuXKVd077\nraTndshUI4V9xZuyHvNM6Tfga2cdZXegMsi62vQb2A2c7uohUwYyV21pUPbSpdpmJp8i/T7U\nc2XoNNrJcO95ZbI5+baWgsNuAaJj9JDhr776atasWZs2baqurrYnVlVVLVy4cM+ePZHVhv+V\nxvnMGyOU3Xhy7FV0mrV7u8TS+9XsFw2dPzpfuLobNR2Erjr9hiO9jYujA8yvNuU8rvYT/hfz\nOdsMPJp+s2Uea5TRMGvSCKBZrjAnKwWAcAUHu02bNvXp02fHjh2KhevUmTBhQgRVISPm52+/\n3i9nkDJJb95hwbCYD94pY6JfqvNy5ja/Ljpns4ZDjc4y/CrUdyWGlTOcn4f0jlR6f2B4F/T7\nGyPLsrPetNcS4qHPQzHeNCDngoPdHXfcUVVVNW3atG7dug0cOPChhx5q3br1kiVLZs6c+fDD\nD/NdsXkojTO3t7vLG+n8uqwsoWQ7TXdXYM3eFsQsOrjm11xp5+pTdHWIJo69bUK5A/Wbo+w1\nDOsE760wr3g7kr0To5Y/uyXcQx8KujOBQhEc7JYuXTpmzJgxY8ZUVVWJSI8ePfr27VtRUTFy\n5MiBAwcuWLCgX79+0dcJU36/f5VdTX6Dm95xTPsqMWdrSpnEO7+BWvs079eJ6EqErpTp1zfm\n1zPnN6rrXZdrusk2utpxHi9NV5b5aVUzZ1hdbuYCs2x6zerFIIIUdPEAciv45oktW7Z06NBB\nRGrVqiUiBw8etKb36tVrzJgxkyZNirQ+RCRxlPNH67UrErnClmtO+y2TzJdhtZrGXX0bdk7y\nyzcmV7C5tkgZNP2SqOH4deA85pR5VzkxC9JYr99Byf+Ik6udnGX29aM1YWOBghYc7MrLy7dt\n2yYixcXF9evX37Bhg/1W9+7dV65cGWF1SJ0mzUgGYUKZ4eTY9KPsfIqCs0PRG79c6dOe05tW\nvf2XzkW882hyhnPt4r+7lAWYBxrlphku61eDXkFEK5dCrBkAwhIc7Pr373///fcvWbJERE46\n6aTp06fbd8IuXry4pKQk0vqQHr8E4B1k1AzaulpztuDHTnWBoSG9U6/mnO0NSSmlWGXUk2O3\n3dtp5/ejt/Es9HPoA2KqeyNwfs08KUWr/OwBMq8qtzkyy11o+Raa6UEEvIKD3W233bZz586x\nY8eKyDXXXLNy5cru3bsPHz78lFNOefDBB88999zoi4QR/e8453iislPNZHTS2ZRyjSZdSq5G\nAlfnfCuwcU32ck3UdKTZM+h3kT045ZrflRH9egrtd81PTvaCyripbETZvje7e9/Njnw+K+dJ\nbVnILnmypQBCEXzzRJ8+fZYtW7Z8+XIRufLKKz/77LOpU6fOmzcvkUgMHTp06tSp0ReJlLn6\naew04OrQ0vRv+f2uD/cckPTc4mC4UldmUs6TUD0TWJ9p7KVc+y3hc3+iMy5rmvVW67dezQwp\nLes97mkcOP2+8n7GUm0/7aWyI59rs2VeZPZzfIg7tiCOEZBlRg8o7t27d+/evUUkkUjcfffd\nEydO3Lp1a7NmzcrKyiIuD6lJtY8qsDX9L/304oKTZiwvpZxnuCJNs4E7J7Av0H7L2/OnWcok\n83mXdf5ot6BJYK729Rvr6obUF5aGTCJsWAVIju7YTUPUleTPlgIIhVGwcyktLW3Xrl3YlSBT\n+lTnJ/ujMCZxMDtVKU/wzgio76iz2VuU8DwpxnyMNZ0N8FSlD5HmiUozSqscsvcO9JuUqulo\nNCkyxuK6H+K3RUAeUge7vn37Gi5/8ODBDz74ILx6EJWE6il0rn4sV+Qyj1+ZdN0FBqBMOg41\n0517I6UeLO/rhOPRd8596x0IdlbluiDPb+3eA+f3Y2A/omYbTXjDq3N6So2Y7/woEC/yX867\ndYHCpQ52roeY1KpV69ChQ9Zr5/+3hg0bNmjQINL6IEGXcJn8ce+6FMyvfUk9AXhHHgN5L+bz\njjOmlC/tpQJn9mvWexWa97VrS/36R9PbFfrTmMlJzpnOJSgCGlblWiRwqDeN7mHNEHNNVsP3\ng9+fEABMqO+KrXbYsWNH3759x4wZs3r16v379x85cmT37t3Lli27+OKLe/fuvWbNmixXXAMl\nVTexptFCJvMkHZzTQxkz9UY9TRl+i9upyyQDeedRJrbAq+KU45LOOOVKvc7hXfuFX8eeco36\nDcnwRKjJ6Pr4bj7C66w5vWpT/UMiInlSRijybVsy/40H1GTBjzsZO3ZsixYtpk2bdvLJJ5eW\nlopIeXl5v379nnnmmbKysltuuSX6Imsu87CiXCrh4JrHe3L1m9PVbHQnN+HYwQAAIABJREFU\nAMNrs1yLpHHRmL1L9QnDGb/MO6WchQWGYOUU1052pTr9eKvfj+Y7KpNF0pZvwSJ+Cm73kuqA\ntAUHuxdffLGiokL51oABAxYsWBB2SUiNfQL2dg5plkr1F31EJwa/aKXsCVNWZZKWxHNjgSak\nOodHXX2BzmJcldtzik8CdiUzZceesw/PW5g57yZ431Vuu6YHMZs5Ty+NSqIIjvmzQwJpjril\ngLYFQKDgYLd79+4dO3Yo39q5c+fu3bvDLgn/x/AXrvO8lfC5b9Gvndz+Ke+3dsO4Zs7bY2dP\nT6ZyYZwmsWlqdgVuv+43k1RnUqqmBX30j+jDENgTHMVKo2i/QHsWGdkEapTgYNe9e/f77rtv\nxYoVrunLly9/5JFHunbtGk1hNV0apxBnD1B6KzX/1R/6ScLZX2XOfH5vmNM36+y70gyqOiOa\na8zUr01vPYHb4hclA1vzVu4an9UXmR3ZiR01MNZ4B/SV8xRiVAWgEfwcu8mTJw8bNqxPnz6d\nOnVq3759aWlpVVXVxo0b161bl0gkpk2bloUqoecXKZRDtPr8YSi6rh1X/5m+Oy2lMvRDUa6u\nLFdQC4wFfiObrsX92vHO6WpZsx/sgbak6slwynVpKlG2b7j53uRquK5IU1eIjRdQOnR+KnJd\nC4CsCg52559//pIlS6ZMmbJkyZJ169ZZE4uLiwcMGDBu3Di/y++QoVT7olw9MfowlJ/Dssoa\nAktKaSA16fNwFpO+DcO1p5SBDFOjs2XNpXuB+yGl1TkXMUmNyv4/8xWZ16NsNo1NizfD/xfs\nMSB+jL554qyzznrllVeOHDmyZcuWffv2lZWVNW/evE6ddL61AhFRXjuV53+vp5TJ/FrQNxvY\nPRkYywx7m5TBwq8fS9OOcrprnNo75m53jKWxP/WRKOl4CLP4f6Kyk+osyhpMenbz+f8CAIQl\nhXBWq1atVq1aRVcKUpLw3EHpHce0uIJOLK+qcQ5levu00ht29Lssz2/U1aRNzRVvok2HzkU0\nMUW/pakOKCs/YJLrPxg0GTTLlYSbF0NPn/EOsoR1wE9wsEsmk3PmzHniiSe++OIL+/snnD7+\n+OMICkMw1/nV1W0TyrV0kTIcKtLP5oqzypFKwzFEk0Fq1+0FgS07u7vk2POQyZlJuaCrfcNM\n6bdG85zk/ESZZ7vMN9Ok2iiWyn8hhhtyEhAbwcHunnvuufXWW0Wkbt26RUVF0ZcEU35Dcq6J\nokoYytbyMAUGpjr9PIEbrumH8+sFNNlL3iNicu2g3wijJoqZX9KnLMBVnskfA5Ge+zVV5aFw\nCwt9M9Pbe3l+/YatIIoEciI42N17770VFRUzZszo0KFDFgqCH9dAmOvsHthzY9J+KHWay0KU\nTON+An0m87vxwi9SB54mlfclONvUj376nbxdsV752k90Y4t+1Xqjs0kNeR77siCicdtCyXYA\nlIKD3bZt2+bMmUOqyy3XUGNgL04e9r15pVpkdNuluSfAvI/TzhlJz90bztd+0dDvR1EddL/c\n5q3QZDbvu84B7ujO8d78KsabFtZ6Yyz2GwhAKTjYNWvWjF8QOee9cs6bRWJ/ujJMdWnkP2Xk\ncoYz/Zis3+VoLpqOEH0L3i491xTvoTe8tE7f3WgipTs5vDU4/2JxJTznKgzbD/2/QDz+T8Vj\nKwAYCv7miVGjRs2cOTMLpUDP7/eyFfKyXEyeUEaZNPaGvpfIZDhVfNKh34rsEGP38NnH0TXO\nLo4eO1fHrd8qDAegAykrUS7ryrvmh8C50/QtGzZrONZsnnJcqTes/2tZ/m+b0vbW8AhYoIcY\nsAX32E2cOHHEiBGXXnrp5Zdf3rZtW+/9E506dYqmNqiZnN3jJ8NxWL9eN5tft5ZfF51fMZo+\nOb+xVGeDJp0rfqsOzJTe66jMexA1a7SDaYaBwLBbLtUWMhGPiBOPrcgOv2tdzJcVdjhyLTjY\nlZeXWy+efvpp5Qx8iPNBLEOeeXeO3wzOd135wy+leV97Y5zfdP0AqKZP0dlg4Ciq911XJfqE\npG9T89/ZcMFUfyGYdx96FzTZcO+60vuVZb6syZzp7SV+2UbN+38wk6YybwRIQ3CwGzVqVHFx\nMd8zkW8CL/yKgZS2JaUuPX2S87bm/HXv7DF19rQ5k1N6CS/ViKOs0NV5FunnITDMmfdhpJfJ\nvDuQs6lmR5lfUZBNrvWmcRDDrTztdvjsIU8ExzW/jjoR2bt3b2VlZaj1wJTyhJ3heGXhMtnw\nVC/8SmPBwEX0w8EZ8uuoyzzupFdtGjk11YFX1/5U9uqJTwBNKQ0E1paI+D7iGhIaauavLyBc\nwTdPaLzwwgunnnpqWKUgVZrf9TXkNGBL9Yo3zQz6sUhNH1viKG+DmqSYPEq5dm8foWbw17U6\ne2Z9m96S/GiuyfOb33wpu+NTuQ/17eiH6V1hLqyY7mzKLt68wbQTTOCyyh1lchRSPb5hcX3+\n06ghV5UD+clogPWrr76aNWvWpk2bqqur7YlVVVULFy7cs2dPZLVBzTv2mjz20nvhD9+j9Kd8\nOfb0rz89e/tjvIOz3pldK3LOr8xb4p9IlOnQWYM+kjob1DflLSOwJ8xwRRpJ1WC34VLeFTkP\npd+eSWl3BWYpv6OmX1EmTDo4XX9CeN8iD2WCoX/kp+Bgt2nTpj59+uzYsUOxcJ06EyZMiKAq\n+FKGkrjGuKyNLGtWpAx8yhFVZSpy0cQ78wL8Eph3QeV0Vz3e+GL3tOmDUUqNK6t1hbk04qDf\nzEnHFYeBM6e3Fm8uTymPpleGHPvh0RwaV9CM66+IHMrwfwcQneBgd8cdd1RVVU2bNq1bt24D\nBw586KGHWrduvWTJkpkzZz788MMVFRVZqBJKrjNx/H53p7pFyqRl0khgqHK16VzKr3POpH3x\nOTGYDJ4qO2D8aHoKNQsm/a/PMzyZ2elK2fvl10UaSmeSHU+dlaS0bEr0vbCh0xwaZQGBnzGk\nyvy/gPnMQCiCg93SpUvHjBkzZsyYqqoqEenRo0ffvn0rKipGjhw5cODABQsW9OvXL/o6ayjv\nLwVnV4TzhagSRk0TGLACBe5DfURzthPYeNoxVMn8zOEXs6Iow9XdqMk9aa/R5MTp19npmieK\ns6+yXzalFaWXrUkSQI0VfPPEli1brC+KrVWrlogcPHjQmt6rV68xY8ZMmjQp0vqg5B2EJdU5\nhXVWS3ie4ptey8rL1OzpiaNSKkw81+97c63rXU0N9mvzGgwr9GvTL+Q5/0pxbZRri8yLUSZI\n5z706ztMaRXR5cJcSfVjCa+IPhiARnCwKy8v37Ztm4gUFxfXr19/w4YN9lvdu3dfuXJlhNXV\neJkPJtZMrguMnPQhRjks6Ix3ru6upOeqee/vcb+zo7OrVbO4t7AMr9Ay7/5J+5zkSmPJo/TN\nBsYI+xC4egFNinQdd2+SU7YTWJJJza4/DFLapWSCmCErIwuCh2L79+9///33n3XWWQMGDDjp\npJOmT58+YsSI+vXri8jixYtLSkqiLxL/J/CXQva77jqKfE/E/U1zIjtFNop8LrIzm9UESXhu\nvUw6LlXMZPTWZAxXGR28M3hnVvZv+eV+7xSTfOBdqX7gUtOsq/vN1aByWb+avfsn4bluz1wU\nfZNR05SqPwqRrjpcUW8IUKMEB7vbbrvt7LPPHjt27MqVK6+55pqrrrqqe/fup5122saNG1ev\nXn3ppZdmocqaTNO/4tcPFHlNIiLSQ2S4yI9EThb5UqTq2HcPiDQXaSwiIrtFNopsFPlE5A2R\nd0T2ZadEH960oZzut0ji2GscxXNDgP7oaOKj+YlN0yWpWWMaJ8708pPNOwaq2XZlkd785wy1\nUdSsWbtrLfoZIpLJHyHIOcIrsiA42PXp02fZsmXLly8XkSuvvPKzzz6bOnXqvHnzEonE0KFD\np06dGn2R+F+uc4mrdyc7v+g7ilwl8iORLiKrReaKXCLyic/MDUXai7Q7+u9ZImNFjoj8VWSx\nyGKR90SqfZbNUOAv0MBhbpNdmmrfmKaLy5UU9etVdll5u6MCuxJdI8uu9vUFOxsxrNk5v77N\ntLlGaTNvMG/FaevitC1AzqXzp21VVdXWrVubNWtWVlYWRU355oEHHrjuuusqKyutAeiccF0M\n5B1GzI72Iu+KfCEyS2SeyPrUW6gn0k/kX0UGipwqsldkgcjTIn/JIOGlvROSnivknNJu029Z\n/VuBV9EFDuZ639V37Jl3++kpx3A1U8Jar76e6Nqv4WpCaAYCHTx4sKSk5O233z7zzDNzXYtb\n8M0Ty5Yt+/rrr51TSktL27VrV1ZWtnz58rlz50ZWW83lN1ao/DFrjhN5WeQDkb4iv00r1YnI\nXpE/i4wX6SPSRORqkXoi80S+FJkmcqZIGtuW9g5xDbC63jXvgpKjF8UH9gJqriRzxSCT/Odc\ntbJ9Z1XeZjM5NwfedmCyYKTsHRvFGrkEHkDeCg52/fv3f+utt5RvLV269Jprrgm7pJoucPjM\ndbbOjhKReSL7RS4Kb/B0l8hzIsNFmouMF+km8pbIBpFJIi1CWkVKnGdrkzO3q2fI/GSvP3yB\n13UljjJfyq+XRVmJsn1NGa6Z/S6Vc/WP6j/AmSQnZ+Mx7lgy/IhmGECVn7TC2qukcNQ0vtfY\nrVu3bt26ddbrVatWlZaWumbYv3//7NmzDxw4EGF1NZLr8inlqKv3RaQSIo+KtBf5rkgU3w38\njcjDIg+LtBAZJXKtyG0ic0XuE3k3gtXppbpLlWnJey50pUZl+HBdaSeqkKTstDOvWT/67JxH\nHz2Va0x4HvuXE66dlvlAs3eAO1fbmOVh0LzNQzk8BED+8w12c+bMGT9+vPV68uTJfrONGDEi\n/KJqPGVWsF9n/7ft3SKDRfqLbI54RVtEficyVWSQyM9FlomsErlPZJZInvwB4b1KzD7XutKb\nCb/zU8L/e041sUwT1Fw/Bp4XDc+a5h9L80Riz+nKuJlLozW7b9JvsDuHTLYl7b2X8LmrJh94\nD4pe5vWTI1FYfIPduHHjrrjiihUrVlxwwQWXXXZZ9+7dXTPUrl27Q4cOQ4cOjbhC/B+/npJI\n/UzkFpHBImuytcYjIq+KvCrSSWS0yL0i/y3ye5EZIruzVYOfwAFKF00as89PzrfsQ+wd3BT/\nXihl7Agc61QOwvoFTQn6+DkrN+nFDJRG/a4FnWnStXbDoBnKhoQoO/HC+3nLck+hXpbLSDVH\nAjmne9xJixYthg4dOmTIkNGjR/ft2zdrNcH5a1R5Wsqa80Smi1wjsigXa18ncrPIRJGfiowV\n+U+R+0T+IPJVLopRMhkTV15/pk9Imjk1PXzKiZoePmUKzOQU7mzEpEPRtS7zvro0/juEGMhi\nf4LPbXjNNyY93EBeCX6O3cKFC0Xk8OHDtWvXtqYcOHBg9erVxcXFvXr1yv7//2QyuXHjxg0b\nNlRWVopIw4YNO3fu3KZNmyyXEZ3AXZq1qNdT5FmRu0Uei3pNWntE7hX5o8gVIv8pcpPIn0R+\nK7LFZ/5wT0sZtuYKT8pr75xz2gfX1Tnn7MbzXvJlL+IKSX6Vez9CfjnSezmgaylvmDOPnhr6\ny9r8mlJeXeccKw9sIfvC6gwLN3z4dQnXTDV881FwgoPd4cOH//3f/3379u3PPfeciGzatGng\nwIHWN8aeddZZr7zyStae7rZr164pU6bMnDlz+/btrrfatm179dVXjx07NjaP1lP+Ys1mjG4k\n8rzIqyL/lbVVah0UeVDkEZGLRcaLjBaZITJF5GvPnPq9lGpQS+++BM27+o5YZ85z9bf53a9g\nz6BZtWt+ZQLzdphpBpdFlQv9+A1m+f2oLCxVfr2Y3vazI71VGy6Vtd8MeTUmC0ApONj95je/\nmTFjxs0332z9OGbMmI0bN15//fWJROL++++fNm3auHHjIi5SRGTLli39+vXbuHFj586dBw8e\nfOKJJ9arV09Edu/evX79+jfffHPixIlz58594403GjdunIV6omMyKBb17/FaIk+JHBD5N5G8\n+hV+WOQpkWdEhovcLfJvIneLTPN8oVnaTLKUH/01dsruMeV6/RpXXsSmXND8SjtXqa5+O+XQ\nqrMDzLld+mRp3p+k3Dn6mUUVXk0WNylGuTcybDas8iwMmwI4RjJIz549hw8fbr3+4osvEonE\nT3/6U+vHq666qlevXoEthOKnP/1pUVHR7Nmzle9WV1dPnz49kUjceOONoa/6/vvvF5HKysrQ\nW9azD5DrRRbcKbJLpFPW1peWIpExIttFNolcavJIxsykuv9dx9E50TWD32x+nwRlVfpPjma9\n3gI0H0W/194C/BY3EVikpmZnbemtwruuwJVmLsMGwy0mH2oIfQ/nZBWIMetZb2+//XauC1EI\nPhtu2rRp0KBB1uvXXnstmUyOGjXK+rF3796bNm0KbCEUL7300mWXXfbjH/9Y+W7t2rVHjx59\n0UUXPf/889mpJwusI+SakoX1DhMZJ3KpyLrwVprUppP0HBKZLtJJ5EmRP4msEPnXsJo+yllt\n4Ahv8thTvt8iJv093j6npCdP+P2onKJp3EW5oLNB5WuTHWUyXKupQV+z3Xdo8knTX4Bosjpn\nU/nQYZad3wyB8mFXAAgOds7/q4sWLapXr17//v2tH5PJ5KFDh6Iq7Vg7d+7s2LGjfp5u3bpt\n27YtO/VkR/ZPG91EHhepc9ddLztqCLH90Ddnt8gdIv8islpkkchskbY+c6Zx8tNfM6efWZ8e\nlEdWGchcw52uyJg4yjmn819lBDRJNs6W9WXr6VvwqyTh4JyuDLXOvaFsTTy7UQw+D87Yqo/O\nqe6WwO3ybkUmvwoCl1Xu6lTp/5zIbWu5WgWQE8HB7sQTT7S+Umzbtm0vvvjioEGDiouLrbc+\n/PDD1q1bR1vgUS1btvzwww/186xataply5bZqSdXIv1l1EBknsjrIrXuuCOwDPNms/Dbc7PI\nT0XOEGkjslZkgoj7m1IiTqjeH5VdR8qztTOBJVQ3t3oXCSzMlfwMtyIlftnLxTE+oI7C9r7y\nC7uuNSpXYb8wyazKbUlpfueqTT7eaSS/9OpJj2HSjbfMoy2QJ4KD3SWXXPL000+feeaZp556\n6p49e2688UZr+hNPPPH4449n7QHFw4YNe+655377298qv8Rs7969kyZNeuGFF0aOHJmderJD\n2TEQ0W+f5JEjT4gcEbkirRsmkh72dMNzf1pVH2OlyJkio0VGi/xN5ILUW/Arwzs9cE5vl4zr\nR78W7N3l7dnyW6OzY0mZopyLOzvtlP153npM5lEmKud0i3eNylThlwVdzSp3grds18b6lapc\nV0pr189pv9ZndzHIWH7V+oXjwAZreKoD4iT4VrWqqqrrr7/+2WefLS4uvvPOO3/+859b01u2\nbHnCCSe8+eab2bkL9Ztvvhk4cOAHH3xQXl7ep0+fNm3a1K9fP5lM7tmz5/PPP1++fPm+ffv6\n9+//8ssvh/74lQceeOC6666rrKzM2oNdbM4zQdR/Td4u8h8ifUQ+Vb2bPPZ2UVGdWjRhQtlI\ndFvUQGSiyL+LLBa54ejFguEyrF95/rZ3oN+eVLafPPb2TNee9MY1TbJ0HUpR5Ty7QuXM3mL0\nScVvA/VV+dWmrDbVrfNuqUnZJq2J53DrN82kNb/NV2ZWw1xImMsHHIsCdfDgwZKSkrfffvvM\nM8/MdS0eyXS9++67hw4dSnvxNBw4cOB3v/tdr1697EclW4qKivr27funP/2puro6ivXm5K7Y\nLH8MvidySCT53HOu6Zqq/KZotsX5o/dF6LqKvC6yT2S8SJHB/GlU4reId7pze5X7QTlbqvUo\nl9UcF78jKD4HyKRZZdn6T4JrvX6fE+8qNDUEzmPSQtrMG09vK/Q7J8Ty0paFVWQoHyo0+d+E\n/JTPd8UW5DelVFVV/fOf/7S+eaJBgwZt27a1L/uLQk567DRdL+FKJpOyY8fmpk3ni9ygfNdR\nj/ctbw+cc7rf4knPI9P8astw2xMil4v8VmSryM9E3k1xcVcBoVTr3SfKo+z8j5kIelyw31Hw\n490o1+o0q/CW51wqcL2uQ6/8tGjK8L6rWaNJx57JL0Bvb2vO5X83DxUaFpDbGpC2fO6xC35A\ncR4qLS3t3LlzrquIlvL0H0XCq5VIvCRy3imnjHn33TElJcrfNeYTTfg1Yh6hUliXyOMiL4nc\nI7JM5H6R20S+NV7c8DKspPY6Qte2KGdTbrt+iNZeqfUnmmseMT5veUchlekwlCOirDOlpezC\nXHtJ+f8l80gXFvPUa9hg/kcB5VHLXTkKOa/H+zcVEIqon+qK0ISY6py/Sm4VOUvkX1atSpSW\n2u8qT8DKX0Cuc7/9o+a0Kv4deK4Zwtrkr0SuEBkkMkjkE5Fhx6468/YDLzfUTPfr+vJrRNnT\nGdillHDwzmlP1MRTw5pN+H02vJ8K/adI+a6mj9m7ulTLjqJZ17LpLRh6JXYLYRUDJVIdolCQ\nPXZK69evv/baa0Vk0aJF5kvt3bv3t7/97f79+zXzrF69OtPiUhfpr1S78TNF7hK5XOQzx3Tl\nH5EmUwL59Wxl5/zxush3RCaIPCcyW+TfRXZmZdWG3XXerObtl/LyzuDM1n71eDOKpkiTrlm/\naO43NJzw3B7r13Ig5SdTf1gNI5pF05Fs3t1i+EeOK5Xm9qyv+W2QKuILkE3xCXaVlZWvv/56\nGku99957+scsb968WeL4u+k4kWdEin72s1l/+pMYX56VTa4hRWUAUk7xs1/kNpHnRR4V+Vjk\nepH5qRTjijiasGXc6jG9ZZp3XTV4w5l+J2iSnOFOlmPP8c6PSnpJ3TwxONeVSfRxLZLSsppO\nZfPxZWXN3vG4sP7YyPA/skkYzZ/fFQBs8Rngr6qqWrdunYj07Nkz3Jazf/NE1N1IyWRSkkm5\n4AL5/HP5618Tdeu63k3ppJvJyTVXYz0lIhNF/sPRdaehrNaV8wxDlbcpTU71a1Y5gzd3ennT\nmPJYe/uQvC1rgp2rbP2oqzLQKOvxKzKldOi3CYZLeX/M5MPvV1gaLSsb9K4rRKH05GVfvv3t\nisKVzzdPqK+xe//999NoK72lwlJaWtqzZ8/QU1022cOUyWNFsrKpU/e8+GK3jz6SsjLnitKI\nXImj0qgiwg1Urct+fUDkdpHvinzH4FHG+r60lLrKTPZVYEeg/sytT2muAVD9OK94PpPKppQ1\n2B8n1/ZaP3q3Rb8PAzfTVa13umYpDf0+T6M1bxR2/b9Lr2VbJn8ppfS/OB+ykeZwB36wgf/P\n3pnHS1Gc6/8ZdmQNbqiBoEIUL1ERUC77JosazBUMcYv+1CBXjeK+XDXGJe6JRIniFmPivuTm\nRlQQWdWoYdEQNQmCxg1EVkHQI5z5/XGgqVPL29U9Pd2zPN8PHz5zuqveequ6p+uZp7pnKhi7\nsOvTp88dd9wRKdAdd9yRuW5dvXp1nWlHXGybWhYtqjn//LOAfxh7ra9DY0ZOQCGY5jWtEHv6\ndGFe5ecDPYH7gKeAB4E2Yl3XJCErEp/J0tVr69qftZa515zYcvVR9YSpKswxDwJGnfut6UVq\n2oqn36YWCzKxSlW1g2o3451+kU6YxM/wND8vlSzW8efIkGrALuzGjx9/zjnn9OnT5/nnnw8N\n8fzzz/fp0+ecc875yU9+knR60bjlllvK+mtQ1A/0iX/W3HFF27wZJ574R+Ah24yiGSo+YYPg\nkXJW+5iz3c7vast/l08vvgauAPoB/wm8BQwOreCHrGZcaGMiV7FKELVdz+PoMtVQXzu6NJBP\nv0JHQz7urhnautGViUuqurZoH3Lyyd2BmqywMI9C4YZiueBKOK+QflaEZI794Yk6++2nP/3p\nEUcc0bVr1+HDhw8ePLhz58677LJLmzZt1q9fv2rVqvfee2/WrFnTp09/9913d95550ceeeS4\n445LOfvKRrNVCpR6O6b5iy/Ghg3j1q4d17ZtEmnuiB+7rmnbCCVdiy+FpPQ60B24GXgR+DVw\nOfCVZ02/9ARCtazVqIukgJNSJNZ8XLtMaeiKKatDQaSa3fQ5kUyFavUOo+Lf38LbyiRyOgjn\naoKnMSGVjfOp2OOOO+773//+nXfeefvtt0+aNGnSpEnWYu3bt7/xxhvPOuus9H9HtapIzMB7\n/nncdRdmzIBD1cW+bkbVNNYpMHRijiqD/NkEnA38H/AAMBz4MbAwehDTb3MVsBIq2rRB00q6\nDCe1gKe1Gdh4LhnkasKFdfXTJ05+++Md8rweyaSMR2lKitLMihCSIdLXnbRs2fLSSy+9+OKL\nFyxY8NJLLy1ZsmTVqlVffPFF69atd9llly5dugwbNuyQQw5p0CCNbznu2bNnaJm67yUpa4LZ\nK/GJKp/P75bL/Q1of/HFGDSoGFNgsedUbXDk8vEU3nTge8BvgNeAq4CbgdqwgP5tWe23UBnn\nklY+jZoWmmCPaQ5xaFhZoplpy3XN7YIWNIlx7iV4umrHxXyRbHNqo3CfIeWI7NanmQkh5Uv4\n99g1aNCgV69evXr1SiEbgUWLFgFo3Fj6MfctW7aklU7RyXs8sRgpDoD7gOVA+2uvTfwGvuIR\nT4AW0sG1wHHAj4C7geHA4I8+ynXooJpGkdqy1pJts0AN+PQiX/9bTiJ13BxbdYu/tWbKzaCk\nZiJqjqZLofrfVBcJzzsCC29C7aY2FAk2HXiZSQUkKVPBhy/Zs72CB6pIhJttL7/88po1a6y7\n3njjjaeffjrplOxcdNFFLVq0+Pvf//6VmwsvvDCdZIpHIrfTqex4g91zz+iddur+zjto0qRc\nbis2/a3AUnLpj6T69RhwMNAEWNOhw3/ZkvHENKtge+5B3h5Q1/FgkVQz2KwKUh0oFU1juRaR\nrclEPX+0VK1mns8qrWtkfESwvx/pg3aYzOS1/god91TwVqwjJnewqB/qCulLtcGB8oQDFYNw\nYde/f/+5c+dad82bNy+1J2Gvvfbazp07H3fccfKvRFQYScmU/XOh18D3AAAgAElEQVQ5nH8+\nbr4ZXbsmEjAdrN3XJg/tdYKfFD8ABgF3AE8AU4CdwsoXiOf1K9TfCqZ2VXzkjUdf1Y8QPmrJ\nVJxafLV1zRtz+XZwSLd0Vi3NvbF1iTYOVgVpqjpXW55pFJJtjFpZhUWF6sWy+Ggdm3LxDioV\n51Lse++9F3wn3KJFi5pt/4X4gM2bNz/xxBNff/11EbNTaNy48cMPP9yjR4/LL7/8lltuSafR\nlFHnBvnS708+n8c336BPH+y2G848s/AkU0a4OgjuZlLTwBbgauBF4A/AAuB4YFEicd3LptZO\nWQsL1WUE5eSz/BoqnUMH36Wu1O3WMqY36WnCuUbb1bpcUm7RZ8TyyipqaLsJflZxZZhIi5zI\no8IR84GjFAOnsHvqqacuu+yyutfXXHONq9jYsWOTT8pB165dV6xYIdxIN2rUqLaJfoVHhoRO\nRZ5BAOC66/Dvf+PZZ1H+n3pV/wmGP+RCKONT/RXgYOBu4C/ARcCdQIFXGnOOV7fnbPfz5ZUV\nWLW8VkxbHDSDaMW0OOrw+qgla9M+5Oo/WCCEdTUdJCybcKqVKERTw1pjylrHtbgsRPDJRx7S\neBOep6orKTi1ExIJp7C79NJLTz755L/+9a9HH330SSeddMABB2gFGjZsuM8++4wePbrIGdaj\ndevWwt6BAwcOHDgwtWQSR7Cg4rNgAW64AY8+it13TzJsRiSid1U8o60HjgNOA34NDAFOA6y3\nnaqKU4isLWiar80goXN8qCPlqugq7ClErIpHU96uimZw/yncU30mSyCsZfGnDaCnUDNJ5Gwv\ntv2W/lEghMhIT8Xuscceo0ePPvLII88888zevXunlhOJjSoN8/k8vv4aJ5+MY4/FmDFZp5YY\nghEi3LSUSNP3A38BHgcWAccBr7obSqRFM5rQd2ENV9tllQvamRMpf88lRWstsyPxIpi74h2C\nRDSKvM5eh6ceSl8zRWqxBO09Qkj4wxPjx49v1apVCqkQFHw/ja4ArroKa9Yg4s/+liMJTjDy\n4L8DHAq8AMwBLvN5/ySdgAvVRgpUmtV7w3YbKTCfrGvBQVitCe212Zb1WKgNySUj9VfYqxUr\nsLlgxII/zTS0ArJbGTuTqAhpJBK8SJE1ijdiBZ4YhJQg4RPTuHHjnn322RRSIQEFXmi2XW1f\nfRW33YZ770W7dsmkVRqoK1wx1rYKT2AzcAZwAnAJ8AKwW+ER3WizsnUS0vSW9gJhq6ioL1PM\n6lqL8p/WOFbHMdCgQR+LOnknGM21CCu0bhXQlUEFaDtCKoxwYdevX785c+bU1tamkE01k/B1\nf9MmnHIKTj4ZRx5ZaKjSw/8Sr1ks5kqlK3joIXgCOARoBywE+ntm45enyySLFxCKsHCV9GzF\n9PwC3RaIM8Gp0twsOBSSnK3VHtOqm2kIWVmFshlQTkbGKs2LpIcq2H8q0ogV1dEkJBPCf3ni\nD3/4w3nnnXfkkUf++Mc//u53v9umTRutQOfOnYuTWzVSoHWx4wp16aX4+mv86ldJJVZSCCti\n6pZAPYTKGnO7zyFYBvQFfgnMBC4HbnU/LRt6WAv3aIPu+0/toSNjJqZWMbWdWR02oRz7A4xn\nv9QW5SZkVSeE9TFEVazdT4qiKsUSobJ7R0iyhAu79u3b17144YUXrAX4lksEczksPrNm4Te/\nwQsvQHyIuEyRF8Jyxn1mRU3ma+AsYB5wD9APOAVYaysWKY28+Gitda+n32ZVAGocLaY5koGY\nC33X+6sNfxtVS8xsKOpBN9tyCX3P6gXWiqoXzbrp1CKElDLhwm7cuHFNmjRp3LhxpTr8JUJi\nw7thA049FRMmYNiwZAKWCabgCB1SbR6NfQgeAxYBTwELgB8C8+NFEReOhd5ptqVL3JiKDX7q\nMGd7lsIM4pIIri2B1Wct5qMdQ4mnWlzlS1b9pPMZxh/XsFNEEpIO4cLuscceSyEPUjjbrpgX\nXoiGDXHTTVmnkx6mTDFdqFDdFnWtVuOfwKHAb4CXgfOAu6L1YIc2ck17mgwydwk+nLWKoAas\nU7IW0CoNXT1yZehqUe2sWV0bKHnEtD8LEXna0SmGQPGMGTqeibcY2johpHQIF3YkHZK5u27m\nTNx/P2bORIsWCeZWsrhuexLu67IiyCD/I7IZ+H/APGAy8J/ABGDT9uChcQTbzHVTl0u4RFKo\nobcYBm2ZW7TqVnEWinqXnhlECxh1YTdx00hbnq5gIg2UVlheKCeEFJtwYXfwwQc3bdrU8zL9\n2muvFZwSiUC9ifnLL3H66ZgwAQMGZJpUxsRYVE1qTRbAA8CbwFPAX4CxwBIjflSsi6qmdWS1\n9HxWWiMtXrvUWyHLly5XTPNfzZVfIbhZptTWKyPhMgut8svldBaeA7bfZJlIQEJIkQgXditW\nrPjiiy82b95c96d6iWnevHlNTU0Rs6s+YqiKHdfZyy9HbS1uvDH5tEoVl74pZO4pfO5fCPQA\n/gD8FTg5bkBZiLgWl81b8VzBTQeubs42NZMqFGSFoW1Uw8J2aDQ7x2oKqhHMQYBxrOMZh6Fh\nrVtihAo9Oq5R8jQskyJSW7LCTiKdeqQ8FFrToPVISp7w77F79913e/TocdZZZy1cuHDz5s21\ntbXr16+fM2fOMccc079//zVr1mxRSCHjyibSPFTv+vLqq5g8Gffcg5Ytk0+rfLBec825P6BI\naawFvg/cBjwN3AA0jB7BXP10vdasFHPxFKLcCUpaN8ImMUOXlf2nXmt3rCalNc/QyInoMxM1\nSWvCrlqIrjWFc9W6XT0TEjzDS8qrK1PnlZB0CBd2F1xwQefOne+8887u3bs3a9YMQOvWrQcM\nGPD00083aNDgggsuKH6SVYH/9GCp8tVXOPVUnHwyhg8vSnIljOe4WRccizo91ALXAkcCPwGm\nA7sWEMoqqqzaQhuNnOMhCcEDU3fJY2suCFoFtPanLBmtXXa1GwR0ZavqG00eCV0rUMFoHTT7\nKAc3W49xZUiHrBLLUGKWlLolxEW4sHv22Wf797d/tf6wYcP+7//+L+mUqhGrNxOBq6/Ghg24\n7bYkc6poYiyp+BQ2y0wDDgHaAAuAXtGDR8pQUEimi2PVQNY/hclbNefUhjzTNhW2qQW17Voy\nqgfmajRXH1eBqNmauQmthwZPmdCsipp2aY4JIRVDuLD74osvVqxYYd21cuXK9evXJ51SNRJM\nuv7Xux0z6Pz5uO023HUX2rYtZo4lR6AqfGSEObCRqlsjeJb5EOgHzADmAacbe4UEzPNBztbc\nZRpUwQtrYc2E0+K4WneNreYpqq1YfTtrbtYWNa9OLmzNzacX5l4fRWJ2UNsVNcPQWlr51GQT\n7StCSpNwYXfAAQdMnjz59ddf17a/8sorDzzwwP7771+cxKqOSKpuBzU1OO00HHssRo8uQlIV\niCAUitfoV8CpQNO7754M3AM0VXYJTpKaHhQBhPp2kbpLs7VyjnvdhBateleNo+710XmmppQF\nmaf0dG3R0hAUkjaGApoG9amiNepfPpJEs0pq/6wKKVAIpaYI6SCSCiP8qdirr776mGOO6d27\n9957773vvvs2b9588+bNy5YtW7ZsWS6Xu/vuu1PIkphsm2tvuAHLl+/6t7+tevTRkrpWpkC8\n/gaiwSUy1PiaLAiVXyHT8IQJhwGv7bXXQZ98Mhb4yC9V1xZVZAjtFpJz3vFtKTlj+dVaF/Ul\nkdUFjEouyjeYaK3nlCd/raEitS70XUhGkKGBVnbJcWtAa+FELgVqwq7kow5CpEZTxnPYCSl9\nwh270aNHv/TSSyNGjFi+fPmMGTP+/Oc/z5gx4+OPPx4yZMj06dPHjBmTQpbVQNSPjPl8Hm+/\njV/8AnfcsapIOVUQpnES6liYzpNPK3KB14H2n3yyCZgP5GfN8qzu0qCm2+c/OakSR106tKol\nUwAJxpLVV1NbVAuEOlKhppQQKjjoVptTbsXahDVDOf9IxVD/WMvOltU0TcoMsybsslrL3fGi\nniMVhtcvTwwcOHDgwIG1tbXLly/ftGlT8+bN99hjj4YNY3yHA7EQ85pYW4vx4zFiBMaNy48b\nl3RSVYGqCTRBE+mg5I17qgQn7DPgcOAm4KeDB58LTFJ2WSWLK07U7VpYaxVTYaivQy1Js3W1\nLXWQ1a55Onlyf12aQysZYy01KO8anNDyVhdK3Ziv/x1+kdR5aJl4HpiWhirgTCmZoKrLSmNR\n25FKIlzY1dbWNmjQAECDBg322muvzz//fO7cuRs3bhw8eHDHjh2LnyHR2XYN+s1vsHgx3n47\n63TKFW3Cc01OnvOW58RfV2wLcAGwELgX6AmMBza7MxSSt8ZHfR3m6Y1Z81QjxPO0giDWwmor\nsB0L/+lWXWC1hooUzdWEq1G5vGz4mcabLG1De5HUaqY1gv/GsiPDVWBCkkVaip07d2737t1n\nKQtGL774YufOnceOHXvKKafsu+++d9xxR/EzrHzirJ588gkuuwzXX48OHYqTVMVijrZrCc8/\npv9SlFrsYaAv0B94GfiOI1XtRSRcVpa1CWsrpsfm0yKUNVCrGILxhESwYOpaAZTlmqmNctvR\nyrjCRl1SFBoV8k8Hz8XZ0KzMAhQ9hJQFTmG3cOHCUaNGvfnmm8uXL6/bsnHjxuOPP76mpubC\nCy+89tprd9ttt4kTJy5YsCCtVInC2Weja1ecdVbWeZQxgZhQ/6x7rU5pRZ2SFwE9gXXAfGCI\nkV7wWvbn1Nf++tK/X4FANMfKVG9WkeRSTj7aQvvTlFMaVlnj09/ED3SopI6qk1ynqLVkpMgC\nhQ9Lyro2NnE+YEenLIaClDtOYXfrrbfW1NRMnTr1xBNPrNvyyCOPrFq16pZbbrnllluuuOKK\nWbNmNWzY8K677korVbKdZ57B1Km491400A9fuVxDs0XTHHDcJOeqXsjV36y7ChgB/A6YBpxn\nS1JeyDPD5mxfVmJabmaXXa81LZV3rz+6UrV6P9YMVb3oKhw6/poACi3medytWL1DYbtasZC3\nqpynT/BQyevS0OV+kck8/7IePVIWOO+xe/XVV4844ogjjjgi2PLcc881adLkpJNOqvvzu9/9\n7vDhw+fNm1f0HIlCft06HHAALroI3/te1rlUCFalkrPdHBbsit2WNewW4EJgAXAf0B04w3HL\nnRw2NMOc+245MyWzpGZhmkpRk6GaFDPLwxANWpkgh1CXTkjDmqpZHorS1cSZT12zvDmkZu+s\nZ53cokoK9lIiTajnUuyABVYvHSJ1QT51CXHhdOxWrFjRvXv34M98Pj9v3rzevXu3adMm2Ljf\nfvt9/PHHxU2wCogmFC65BC1a4MorrTvTWU0od8xRcnkqWplEWncd7keBnRYuHADMAzp4lA8l\n1NRxNaGJM2sC2pqgnKQ2PwlOoZCkT3l/AnkKm74MreujM4LRC8rnFCAOL4xR0rbLSfpfB4SS\nrlZSvsgkLunK6yJJe4/EQHp4olWrVsHrd955Z82aNX379tUK1NTUFCu1asLz3dsPwL33YsoU\nNGtW5IyqBXXSzToX5A45pBfwBTAfGKBsjzcPxe5Rrj6RmtBEjFwyXnox6mopyUdcte58ln21\n10FYdQRM01GWy3Ixa8lC8DnQoWWKgdpiGamxpCgvDUpKB6ewa9++/WeffRb8+dJLLwHo37+/\nWuazzz5TDTxSVJoC9wAP1NZi8OCsc6lwrJYekptahDifA8OBx4AZwJliSjEInSZDLShzmdJa\nxbQArTefaWaManFpW0LzVG0wU2yhvt7KK1jjqzFNvy3SsQgKW9de6/aqucEYlqSIqsy03Aok\n3glMZUNIDJzCrlu3bk8//fTWrVsBfPPNN/fdd1+LFi0GDRoUFNi6devUqVO7du2aQpYVTM77\nLopLgHbAqatXFzulqsKl4RDFPomKHGcLcC4wHrgNuA9omtzt3qaGMBMTzsZA1qgRhMSEAbTq\nPKGiK46wSxVtwmqj+doq3XLuW/00JaqeP3nb0wlmZG2XtZZaJimt41LDZrFs1VUmZiEh5YtT\n2J188snLli0bNmzY5MmTx4wZs3jx4tNOO6158+Z1e2tray+99NKPPvro6KOPTivVisXnmtUF\nuAzY/Q9/QLt2KaRUnWjTmzbrWydvrXoiOdS9eBAYCIwEZgF7FB7Xm5ztCQazgCAFVNEmmGFB\nYVNdqT5ZJB1jVjezMjWl1ZeybinwEKvuHRyiUxv8BBWVNZpP/CDnEjHPSlPnFS+r0uwvKWWc\nT8WOGTNm7NixTz311OzZswH06NHjuuuuC/aOHj166tSpnTt3njBhQgpZkslAs6FDccIJWSdS\nsbgknTnrh4qVQlCDvAH0AJ4B5gPHFPYUheyumSuScpWc7QFbs4rmPGlLe+ZeoSGzO5oqCrXB\nzJhWPacGzCm/vRHJmNTSCNVDWQkmf1WXLbLhWgoZpknmBiopfZzCrkGDBk888cTs2bPfeeed\nDh06HHHEEY0a7Sj87W9/e+TIkffee2/Lli1TybNikeeMOo4DDm/aFL/5TTopkQDTu5IPVl75\nXg9BRclekcpnwGDgTmAO8N/Ab/1OmKg5mxt9MhTEk7WAtsQpL8KqC6OuJjQtLkc2E9MOrixh\nfabSdHSGtZWSmuyDDIuUVen0VKV4Wbk+cxLiQvqt2FwuN3jw4MG2W/UnT57csGHDomVFdtAG\nuA245uuvr/rud7POpboIJvtQxRDoJ9kY04J4zgQ1wHhgETAF6A409riye0oceRFQ6EvsOSxU\n9cKmF4NMNHXoEw2GEnJpuKQmZs2hFAoIREom3mRfbBlaiAQpcaczQyJ9riNVi/R1JwJUdalx\nPbARuCHrNKoZ6z1bZoF4kT1L3gUMBX4IvAjs4hHWTEmQcTmF4E9Tl1idA9PUDF64VlGFAmZH\nzD/VjdprQRBoelFNwHV8zREIPdBaZGsvgpKuIHIr1uNiRnMZkK6wCZLfTgptVSEcWBJKTGFH\nEkR4o/YCJgBdpk/fzDdzilinZ3VvIq1EnWXnAb2AVsB84GB3MavSciVgagiXbFL/9JQdprtm\nunGauRiqF9X4mm8nlNRy8xG+/piC0lR1pqz0MXdjpCEcPm1jbO0V+jknIPaQUhcSUggUdtnj\nukQ2BO4GngBw+OHpZkScs3Xwf4yJxxQlUfkI6AfMA14BjnM0AUMtRU3SM71AJMGY7M0XqqrT\nCrvklyrCrBULRDug8Bgu7bhb/UJrFa1WLuwBjkinhxZQCELBREg1IN1jR7LlbGBf4JBPP806\nkWokX/8RB+1FbGTF48NXwEnAIuAh4GDgcmBL2LfTWXe5SprWmhxQK2y1wVzjpsmRQCOqA67m\nkzfux88bP/Bqxap01chmF0KPdYGWW6Rx9jzxUhBtLjmePp6HiZAqhI5dibIXcA1wOYA90vwW\nM7ID2ZuJMaPLS2ORsvolMBI4DdgyYkQ7vwh5hUgtRs1NbSsYKJc60dYog42aY2q18VxHwbXL\n5ahZy+TqY5bJ27AOixZBE6bWKlpb8vlWyDGNeiYX+xQihBQOhV2WCFfVXwH/BO7m8+0lQIGH\nINlZMEjmJeBQ4G/Tpr0B/Id3Rc+J3N+kND0ws5UCza1C0lNViKaoTBmqiUgBdTBN0W+W1Pa6\nsrLmL2dSJORPL7E/2yQI9SUhLijsssR1YTocOAb4b6A25YTKmeJNM9bDVArzyjKgD/Am8Bpw\njLE3RnrqkmgdLqtJ3Ri7ISFJOWZUo85fWaouo2DFRe2yT3mXVLLmkJSuCiJro+cjN60iXssw\ncU1PCAmFwq7kaALcAdwDLADAm0j8KPb8oa4Sai9C0bRRsorwS+CHwC+Ax4Hr67+ftQxdssya\nqtk7ubqqD0JzFrqvxbG6X6bu1AKqskMgZ2BN0jUUnieAaQ36ZOUT2R+fmOagWbNVVa+2XWg9\nasIFknKLPsObucFJqgoKuyyxvtXPA9oBVwCgqvPGcy5PFv8rtaoSEr/E54EbgO8D/w38GWjr\nKCZoF1e2pmXlmuk1s6cOl/8E7yc5tHFTS1pdNDVVLR8zbWuLqC8uXR8YNLEuBzdH3noO+Gg+\nLclkz6V4bx/hLArelelfx6iiSDVDYVdadACuBC4H1lDVRSSd4RKW5wQSmWbkRl8AegEdAc9b\n7tTETPdF1Q2BuNHUW+yEtSCqeNK0i6CDXZn4KDbrFnWX0KgpYa1txTvi5qkliMuoYWNXDw0l\nxHedAIV0x6dumhdPn+HNRN2SqoXCLkvMt/ptwN+BBwDwQ2fJY7pZMa7dCV7ulwK9gbeA14Ax\nBcQJtYJcrpjmdVmDCF5doGBkBSZs9JFcpjR3FVa3Cwaba6CEnsY4VQKpbc0wcRs4Q6xnTiSB\nSwlFqhwKuxJiGHAMcNb2ZyZ4eSpxgsmmkI/j/vOxz+Qd3HL3GPALIPjhP9k5c0kQVTeE5iYk\n6ZJlrsS0iqbQtG70HElNdMoJa8dX64hZ1+yOT4bCkTXFZTwBF0n5xdaILjWvIbxfQocrhati\nJalkUp1Q2JUKdc9M3MtnJsoZYZpP7YDW3XJ3FHAGMBVoZ0ssZuT6RpeGIE3U6mp5q1yz7orR\nnFlMiBlaTN2ommcuH1HrrKliVWXpeW4I4x/vo4UwwoLqDQpYBTEKO9VdETSHNXZ8f6jtSPlC\nYVcqnAfsAvxP1mkQf6y+kVkG2xc3rbuKxDSgF9A+7IdlYxDVN5K9MWtJc0u+Pq5iZotCqpqG\ncPlDmhozC/vLMnNLVG2nIYyYtXV1r3ZQhM8kcgLWhgrBao6mj2DNUvOREofCriT4NnAFcBmw\nJutMiA/WCR7uWc06E7gsn6So+5a7V4FXgBPDClvT0GyqUBHg6aLB73lbTzxn2VDxbRbTuu+q\nIiN4bNpr1/CGYvUXfRLTypudFdRVtqrLSugnB0/MXrvOEEJKEwq7zFAvQ7cB72x/ZoKUBVbP\nxtRq2U4Dm4ATgcuA3wKTgMb190ZaARQKCBaRUMZTfJiLs6EVg1qqw2edqoOwQh/VptUt1mI+\nG62pmhWFWlovzFrCcq0pKOVMQlt3fcjxJKcQo3pq+IhdQkqERlknQDAEGAP05u9MlBWCegvk\nnTbRCr5dUfk18BbwONAd+CGwwta0dU0t0vKTq7ypqFSJpmkLTS7HGBwtZlTZ4Tpq1mTy7jvt\n/FvUmjA1aGjCoVtCqxdyEmqjHUMdxjvKZuvUW4TUQccuM+ouQ42AScBvgflZ50P88ZlCBFVk\neifFZg7QA2gMLAT6+VVRJZHP1Cv7dqHLtUExuaTVKFX/1JY1XfmozVk3CkmaofydWjWxxG2q\nUN/RtQtxtZ3ZXIw4qrEayaqkkiPECoVdNgSXsAnAt+s/M0Grv4xwSTRzo6nzUl57+gQYCDwN\nzGvceGIBcTxPzmAErEZX8Fozq4QWtbUwl4Wm/mmqwNAVVaEjoW9MlxMpREb9EfAJXgjWRU/z\nHI63Npqy2OJFkhAXFHZZsjPwc+AaYGXWmZDYuJbPQufpYiZlpwb4KXDSN9/8aqedHgNahpUv\nfIVRdac0QRZqzrlW2VyFVV3lSh5ufeZSMy4l5Eo1qGWVtlpdYQQ8xznU11TbLbYYKiR+zn2n\nYwyVKVeJp1zl5pIKRUjhUNhlyTXAZ8CdWadBEsFl3YWWidSE8KcnfwDwl7/0AF4D9hNLhk5X\nQgKqegv0llUkBS9Mv00TRrIQFBLI226GU4WX6eSFNmHKVmu/QskbBMkLilBN0lPbWePIKXl2\noXBkEQb3eMZTVMmai8JRyIpkZSspOyjssiGfzx8InAGcB3yTdTKkEEwPJpg2tBcxImtbtIt1\n7Gt37qCDegHLgDeA/BNPxAvi1ZBhzqlKy9wrxFHnKnVgVd0GcUxcbpAW07rXJaSsMa3nQwys\nJ4Cp9kJzSHOa1ySyZy21F0K2ufoIrYQOe4KyNX0dTIgMhV1mTAKeA6ZlnQYpEtY5GPUdCHlK\nKOpkvA44GrgB2PLDH/4aaOIoVvh0ZRpgwqysqTd5gg9N2PXa2nog2lSx6NLWOY8ftA2VU5pY\njHS4ZSXhedQSF3xqX6w+lqtFYbv2qQBG71x6PRMtWyJQaFY5FHbZ8MNc7j+B87JOg6RMqNlj\nUrwLdB64ERgGjAXmAh1tZWRTzapc5UnFnAVdjosg3VSJXDeFq36PGd+UGqGTsSDgYmD6TGZb\nKkEHXR8P1LDmazN4smeRqy1TeMVo13p8VblvtihnlQIlqO0KJMPBJIVDYZcFmzc/8Z3v/ApY\nmnUipNho866mP8yrp8siKh5zgEOATcBC4AhHGc8845lP1iouBayaN3nje++0rISweeP76qzk\n3Y/rqtVdi6GhaljYZbUYXbXinSSJ+zramJjBI42V1Z+T4/gXSBbaY6SkoLDLgltuQU3NL7LO\nghQPH9FgFobtNvwUWAEcDtwN/B9wvfit5YIxFlolKpqeC5Scp3x0CYU6tDHXylgtNE0raKaa\nVUQGaNUFu8slVa399UkpNIhJaJlCZFMkGRpoO5/mBDuzqFSGqtOGK2VlTJKFwi51PvoIN92E\nG2/cYNvJ91Kl4mlg5GN9Eb9/Dq5dW4ErgNHAeGAm8G1HMdO3CzVOYFNFofmY7crKJlB7Zqox\nxKgaRDDeXAHNOdJ87aornyQuRejSozGwjmSoZWimWr6kpghLBLW/VdXxCobCLnXuuAMHHoiT\nTso6D1JEVIfJ3CVsieHYRZJHcoHngEOAHLAIOCqJgC5BZq3u+tOsbg3oEj1R/SHUf942CC5I\nMdfhViWXj1wwD2VQK/Qom8lbMw8dkECLy82hfqciDbIreLKKKoZVmQKlk0mAduZkmwxJBAq7\n1LnkEjz7LErsvU2KgWvOk6+eUV2xZOeJj4DBwBTgf4HbHE/LuvJRU7VqFM8cBHtPMD6t85O/\nMeZCdgq1kmaLoYrQPx9NJqrx68jXvyPQqmx8dJiWjHwGaoNfIILmkzd6SmfPHKpN31Rhlysb\nCrvU2Xln7LxzqX1oI4mjzq8Ql+dC4xTvbLFe0LcAVwAjgW0l0c8AACAASURBVOOBl4H8Uv0h\nH61raoTAXpLduFAEk0+bzl0RXNJQy9NsVNNMWnVVXWkRfKRnpGEJ9erMaJ7GnufpJ597iasB\nWdWFajs5cuGpUjWScoHCLgMKdxFI6aPOoK45KUGbIR5C0zOAg4F1wPp99/2h4+TU8jdfxM7K\nnP+EmKFTvlVUBbt8AuYdj8eaRzBQY7LOU1+E+otqW2ZMzZlzXUbUlIRWIpHO2SukrY5kCpqJ\nH8hJWUBhlwEUcFWCuaQVvC6LGeIzYCRwI/AwgNNOy2/cqBUo6pns+flHtXNMGR3JP9Oa0ORX\nIaiazMzQ7IinYPKRMoVYdCXiK8OhfYuUnlCYl25SFlDYEVJ0tPU7666oAZPJLExi1gI3Av0B\nzJ79r5YtD4lS1x+zO1pkzZRSa6nbZUfNZc6Zgtvl7fmoKCEf80/r+SAHD2r5G5Cxz5bQtd1i\nm2RWN7SQs64QAUpVR8oFCrsM8L8ikwpAW3I1RUYMeyB0ckr2dHoNwKJF84G/ABcAMSZG2f0K\n7Y5LtLlcN2th2ISI4NuZksKKav5pxxrG4XZJT2tMV4vWrDTZJwSPFNazeoEKL1Ir8T4I+ddK\nYUmXkKJCYUdIcQmMJXOBT1irDbRCvDkmxnQuN5Rr0+b4fP504GdA7YgR7aNWV4ROghOn3E2t\nFVVba8ILymFyteIyEU0LTTuOrjy1cTADWuvGHj1NcZq7rBm6kk8Q86CoCcgfb6LmRtFGqgEK\nuwzglaXaEPwqbQo3rZ1IU1dRXZNcLvd74BDgr9OmvQWMrt+oT56acJGLCXtl/0/d5TMg2vpp\n8ML0h6wrg2Ymal1Npcmtu/AXPS6x6ElsS0xONbTR2M35U2x5SkjpQGFXElDqVS2mqxSpvEYK\ns9d7QF/gXuAZ4H6gVZSlZB/zKe/xlcKhqldVkJrC81l41bYINp61gLUtNZqsFzXl6jri5pJr\nbLkstJKOxRVPUMqDYw5+yWq7eMvlJdsdkjkUdoRkTKA81Bea7xK8LsbVPOqE+g1wBdAfGAC8\nBQxwFPNJVZBTamL+xpvmk8GmzwLZp70W0rO6eq4E5AzlXniac/GUSjA4kWqZ7car6Hk+FO8M\nDxXomVDiopOUIxR2JQHf1dWDPHXFnnsKXD6LUesvwMHAdGAmcDPQtJjtmlafq4ynMpCHy+o1\nCjl4epayihUwDTlNZfo7XqEyQhhAHwkS4zwskpJz2ajFbjoqguiUa6XgpJIyhcKuJOBblMBx\ndz9SOT3iNfElMAH4PnAi8Ffg4IIjhyoh10Zzl1WcuV5r/qhq1EWymlR8SlpbD0xEOZSq7eSS\nkSSvgCZBXAGtR1wWIimf7a62iuQXyuSi/zw0ITIUdhlQCh8TSVbI9zNpL+D9gd7zpBLuSfKp\nbg34PPA94F3gDeBnQOMokV1KTu6vWswsad2oaTW5xWDMtb1WTZOIdxIvgr9FB9uJZNUxmrL0\nbNFHWfq0iFQujy5VnT6cC0gxoLAjJG1CHR1Ef6jCv2mfYv6zXV3A1cA44ATgLOAN4KAoTbjW\nyKLekRZVYKlCJ1TB5JSHMKwKSc0hUiauZV9zJVGIqa3MCsOSyIkUnKLaQfT05ASrr0CVHMNy\n08zClHUevTpSDCjssofv7WrD036rm2aCuSrN8yTe9P8k0A1YCrwBXFnfunNZaOZMrLlippFm\nog2UbPwEMXP1v99YeGFtUW03VDlZM9e8MdP0cukMqz3mM1ChWfkII+3ABcnIulOTxWoooaEi\nGVqmJM2QzBMglQeFHSEZIGgU9U/N5xBKlsj0sBIYC/wYOAd4DThw+3b/5F1zed79jc2CArA2\n5CPdzF1qKy4RqWkdTaWhvhMZW7Wo2tcVTZWMwV6rUhSGzqVHreX9dVggiIXWZaLmLGQSo3VC\nShwKu+wp0qdSUqZYJyd5BgpdCEuZx4FuwAfAfOAa2wOz/ue8VWD51woVfFbnxqxl3SsXc220\nOpEqarFc/SXpoEVrMc88tRPMzETL39pNlwlnRbAYBUtSMCytWRFC6qCwy57SmY9JtriEQr7+\nXfyq9xPsjb1uFdXh8Cmfz+c/A8YAJwA/ARYBfW1tmXpFyM3fDfIpZoYNja/6TNZGXVrNlVu+\n/vKxoHtMd9C69OmSlcKY5B2P/WobfQbHut0lOuVaPm3xskmICwo7QkoIq7bTplh/O0pGFQ3+\n6UUyzJ4EDgBeA+YCdwKt69cVQkXqgvVPl/kk6DNrXas40zb6j2FUGa1lrjaXU364zGWzaZ8B\nXNkKssw/bUFBWmVovFA++bjeL4RUCRR2hGSGNuu45jPB9XGF9Wzdp1iBrAVOBUYAo4C/A0dG\n6UKosDCNTNQXrNrGAPNPbYu83qoeONVDlYfUpajkpVIX1jKmd6uNpGoQqn6hGdxHAXtSuLsW\n1ZOmniPVDIUdIZnhP+G5vKikpt5iMwP4HvAk8CfgyVxur4jVQ/toXdPUXpvLmnUEQk1eTwzV\nCtpB0QprNlu8o+ay6MzmgjLawqXnKZeUMPIRqT4H17M5YRwSoSzea/5Q/lYqFHaEZEkkbac6\nQwlOV4VjXaDU2ARcCBwGdALeBfK3397QiFBI60jOqsy5Vzmtc6HVFXMlKRcocK7VctZsS62M\nj9qLJ4xM19CljP07G5pJEMp64JKCSoiUPhR2hJQ0wkyv2XhpZWRB1jQqC4DewKXAuokT/woc\nBsDhO2pzs9DBQL6YZTzlVFDAjOCqpaoHH4XkI/usZeQhDVR+vv538oUKRLWiNR/r9qi6Uygs\nHDVrPp4tFvWNIGSbgvuVePziyV+SLRR2GcP3FfEkmDn8FY9ngaTwnHhqgd8AbVeseAd4rUGD\nuwGsXWuNpukbWbdpCQRSSZ69XDkL8kseT1NKqh0JlQWhqboEhOYaapGjagK1SozqmmtoeoSR\nAqq1fNzWSKkWg2KIvFLoFykXKOwIKWk8TQKr46Xu8nGVCswzUvlc+/YnAkNrawcBq9q1OwMI\nXZk1tZ3Wa8FkClVjmmeWt914p0orYfBNBWMVNK7p36d6kIDmGgb/h1qPmsnnSQyPJ5IjWAiy\n7C62MCq2+0ULgPhDYUdIeePpOZXmOtFM4EDgZuBmYAEwUNnlo+2sJpBVssSziFwVXR6Sj84W\nZmitXdOAFDAdO63FSIpHHViXTDRHWOi1q2nh7PU5gqZatcZM0+7KhZnEsaG2I55Q2KUNHXXi\nSeiCjrq8JS9TqsVK7QysAW7O51svX74QmAk8DnQMe5tYdZXPOqkZIcDUZ1oZl4j0ac7VkJyS\ntnxpHlDNZRTyiZRnMVYSC2/UdXx9Uk1cabmyLbU3F6lOGmWdQFXDT2DEH+viYLAL9eUObEoF\n0See/PYfudfWKGP1wEkQ8C7g18A/gJuBllHufsN2p0TITZOD1uVRLVqgnFzrierr2MMSGty1\nSz2smtsXKnMD8eo6qTxzDuKYKtPaEVU0e5qXkTamcFH1OccIyRA6doSUKC6bIdBtkTyDeFOO\noBGLwV+BPsB44HTgX8D/s12hQsWKFZeMC/XeVPFk6ifNLXNpbqFkgVLAVEjmIRMQ1mHj+XaR\nfFOz0XjNZQI1HClZKOwIKRtkk6nwsKVAHvgD8F1gCvBrYAEwtH6BAvuuKo9QwWo6nXLrgRgS\n9E2Co209H+ItRid4RgV9dynd4p1v6SwiF7ULycKl4eqEwo6QEsU6S6lOT77+r8hrBay1XA15\nliwQn+B1ZTYB1wDfBeYD04D8kUd2jdiQJixyCvBTAP7DIrh96nGxunehqGl7emlCW+o4mAIl\nXlv+fREQGkpHriVF6aRaOpmQlKGwI6TMsM7H1tfqxkLW5pLCJ7haZjnwE+AQYPrUqX8DpgD5\nTz5R+x7qnQjKWNhixWfxVM5HO0zycfRHVWCyNjJFHhQRHHoWCQnESFvI00zSP5N8po8HaZ+y\nsqVcbEWSOBR2hJQoPis+riU/6/YyvdD/DRgBfB84FNi011647LL8mjV1u9R1T5VIHo9mqpkC\nzjSu/A+N2Yq1utVK1BpVvTSI6kFz3czEXLauuctHOheiY9TBjx3EDFjgx5vYtcwxzNZuLNO3\nPCkQCrvM4FuOJIJVwwXTmzCFp5NeUrwA9AB+AuCJJ9C580VAcwDKMw2enZL1UNQqQQFTlsHx\nkETwIqeA+kcw6rqk6QXKCcsU7/SQk3d9SnFttKIFUQ9NrJSrl2wlKSkECjtCKhlzfU1VGMEs\nWBY6rxZ4BGi6bNlP16y5APgX8BOg8fa9Pqt4od2Mas+gvobQgqiiU8tKFmeeh8OVrantzAQC\n289UperGUAo/edTq/v6iZ0AtToxUfWoJB8KVDyHFg8KOkApB865c04nVKCqjj+Y1wJ1AZ+Be\n4BbgH8DJQGO/e+kEN8u1ammWlBWYMIu7VIsmEK2LoVbzz8xH3Rh6TE2154rmim+Vp7LEERSn\nnEBoj9QC6ultuneJUEZvmXgEQxdJ6JNSgMKOkIol1KUo30v2RuAaYB/gUeBO4G3ghPqXM5cl\npm3XxI0si62rpabOM7WypjOs+sNMVZ1Wg/hW2efaq7YSqvjVINqkLpS34tJ85i6hZOkjHBd5\nuxmh2BTyNtc+eJDSh8KOkApBmEUENVDWrAGuAPYG/heYAiwGfghEmr40p0qzM3PKTzhogkx7\nAfFLT4IXsh1lKkJPVB0Wmo9ZVwviwiUxrdutqtS/O66NwgCqL6x9Sfy0TyRaPL2V8uexyrhc\nVA8UdtnA9wkpBi4vRJsDysURgd87ZRVwCbAvMA14EKg94IDjgYYFN60acpHmUVnK5OovB/tf\nCkyRZNUuntHMbM1WEPZtdq4OmmFd8tHqQUbtQiYklWc6/a2Yz3LEBwo7QioKT/2RK/jZyXTw\nl1OfAecD+wAYNeoe4F0g/+CDjUOrhcmj0FFSC8QwqLRVV22X6oqpL6wuYDwXx2d9NigJ48MD\nbFpN/T/2aWb2RfakA2SpWlLEHhwKNSJAYUdI5WCdYhFl8i53VgC5227bG3gKwE9/+k9gPNAE\ngPf3aGhmmMt5CjBtUcHlUlcqXU6VphTdfbXkoLWi/amuMru647+IaXXmgvIu89i1xT+4DyWr\n5AhJAQo7QioQ67xovUHKh9JXhFqPPgcuB3besOEh4EZgGXABgA0bzFpRNYeP0ymIJ2Fh10eL\nmLrQFKBCetquqOvLue1PYIQakwXKMhjyOlKSWsUCP9L4j1IpW4OkqqCwS5sqMU5ItljnxXiT\nJfxEQzoICZhz6hrgaqATMAm4AFjbuvX1wG7uKv5Wmec4CLacWiDQTKjvdQnuoNyKViVUbVgL\nuNSnbPhZt2sZ5mw/pxYaxJ8CR4OQcofCjpDKRFh5NJfk5Fqhu1LDNSsLs/UXwC3A3sDFwFjg\nA+A3dbfiRQkilIyqUQJLybUwmt+OVXeatYKA2pKrtUdmtoKJaNqBrk5pAUOlW5HUVYIfm4Nx\n9o8ZWpKykqQDhR0hVUHhk0pZT0tfA/cBXYGTgJ7Av4CngD5+dbV1PRhCLedYnzX1X7DF1HOq\nwPJxxdQ4ps6DIZ4EOWiNbK5TByXNBVkZM45ruIp9glmbFkRtKXyYKSPK9/pQYVDYEVKxCLZc\nMMNZ/RsfynHOqwWeBg4FhgGNgXnAX4BjgYbu7+OAbUyCLbKnBUOoqa1YC6uvrcfIJThcFl0o\noQovdK8ayrS4YqdkRvYvkAhRz3DBKA0CluO7xpPYZyBJHAo7QqoC66Sirf0l0krhQZLFmtJs\n4GigK7AIeBBYAuD221sD8H5g03xtbVdYrzRfC1aW9X+1ivk/bLrT32MLgrvMNuGcMcfQ58Qw\nmxNyUw1ObZeagzWamZsp5WMLlKpVNvQ4SwcKO0IqFnnq9Q/iEzBq2NQQUvoXcCbQEbgfWH7e\neR8DdwDfjRUt1EKTtUKwGisPoCbj1PKBjgm0Ud7x27hWJWTmaf3fxJqeGjPqZwYtgiYxc8pv\no6nl/Y3n4p2ieYUiNVHiVG3HSw0KO0KqDp+lIpd1FNX7KX1WA9cDnYD/BnoB/wCeB0bZLo7a\nmMRYcLQqLVd11ZYLtvgYWoHsU/9UD65mwsF2uP3dXFPqBXUFR01AaE44BEGe6ri5+hWaQDyN\nUjFvClLWUNgRUnVo85+KubFK5qoa4GGgN/CfwGrgf4F3gXOA/Lp1QRltKDThJWgO2GSHhqAk\nQg+B4FcFBp4ZRPX/NG8Moog0lSKMQcjbfgHCpy9qYWsToYSK0SIZS4nYhFXydiNFhcKOkCol\n6vRWJctMrwMnAh2BR4FLgC/btp0CHGwU0+w6q6pzWXousaKqIpfbpKrDuj81BSZ4b64uuwQr\njJXZoEXYjDFrE5oHbCasNWTmkwk+5qiGa4T9FW0k7UuICwo7QqoRcxKSl+dUF8c1gWWr+UJb\nj5TeZ9u/3PhUYD9gEfAKcDzQdHsBwdp0mWS5+g+6CsNoLm4GETzVjyoNtSMbtKtFMNcxhfhW\nzICyi+kTJE1iiDmtrkvaen4ikrWvfybxKpKKgcKOELKNXP3vqtDmeE1SyNIwfVxZaQVCUSPU\n5POP5/ODgG7Am8BdwIfATcC+7tZdsinYGGg+TfzFm8tN0eYK6PLJtC1q/qoiCZUmru6Y7Wqj\nZHqBiCsrEzwDszKnE2mU2q7KobAjhGwj6qTib0KkRuFTmimG8vn828BZwLeBq4DhwBJgOjAG\naGyr65KYnkPhMti0UFZhbfp5oWLONWLWdoMBsXqBrlquJOXEPGu5gsQ4EwoRc8kKwdLJhJQj\nFHaEEMD2NRPCWqH/amwFmAdBFzYAU4DuwH8CHwG/Az4Ergf2FmupWzRfqo7QaVjWWJ61TCnm\nSthfGWjGm9qEv9UX7FXLC1ZiaBzUV9hRSdb5IyR9KOwIIU60Wdm6iKburewZMej+68BpwF7A\ndcD3gfeA6cA45Q48rVbw2rUe6lItue2Y29UX+frALYPUMjBstpwCDAvQbNRML1RRaaeQuVeu\nbqImoHW/9I0rikhSDCjsCCG+mPO9RqlNqMlmovV6PTAZOBDoA3wI3At8DPwqlzvAUUtIJtQt\nM9WYVt1cFbUGCcrLxpiPGLWm51ndjGCNFlpSoMClTEouUr5Q2BFCwvGZ5EIXB4uNVfEIeigG\n1nF4HTgd2BO4DOgDvA28ApwGtHLX9VydDM3BKoAEmWh141BfkFlljVVxav6fnLy1aVd6PgHL\nDmvXKqZ3pHSgsCOEhGCutamY/ors5xUP2RkqtgGzEbgP6A0cCLwB3AgsBx4EBgBBw9ZVSHP0\nrA6Zj+JRxZPakGnOmXpXXr1VQ6melithrbqP6FR7oXYnp9y9Z21I3asVCPWY5bQjVfGBLiBJ\nAQo7QoiONqvJk5z/rFnZs1owRIuB84C9gFOAXYGZwD+B/wHyH35orRh7qVGzvjT/TFg/FXw+\nVb1Z47gOt+dpYK7VWhNT9ZwqK8185DMzNJ80oT9H0oHCjpBqJJKHEUz2nuXLdAIrMGdtPGuA\np4AjgY7Ab4EfA7UdO76Yy50A7BQ3GR8RY9qrof3SfD4fw1Vz70KXX1WfT87fakP65G9mEvxJ\n+41UFRR2hJBwtKXDMpVuMrHnaXkoPgVuAPYH+gMfAJOB5cC9QD9lidYMKC8rW7PVDDxXRdTX\nW/7dkcfHPCVcHpururzFGj9DaSWLYEIyhMKOkGokqjKzLoqFGjZlR9SOWF0ou1oCXgXGA3sA\nE4AOwBxgCfAz9+9YeLYeKWFXnqqzJd/K5vpTSM/qosGtL9Uy1j5aBa4waMWQXxV25pNKgsKO\nEBKC/+qYOYWX18wXVQFYNZAcZDPwKDAS6AjcA4wFlgAvA2cA3/JuNxjY0JVNrbwrec1/Mlcw\nTREm24dq0z4ZusgpwK0RrTHVirTW/MnWCiWFQ2FHCImAa57WZlltPk6NUtCRpk1l5RPgZuB7\nQA/gr8DVwHLgKeAH9b/o2CWOhbHVEtBUkRnQ/FO+684qp+IdaDNJM7KrSvCn1RqEoQLlsLGh\nBiKlBoUdISQagv2TuTuS+SxrNbRkFubz5wEdgGOAb4CHgeXAlO3fkxLDFITbZA1Em4+RFlQx\nJbvWlmwfujIMagkK1XqyWdeLrXLTlZu1xRgfRTI/34pB2RntRIPCjhCSMC6nhLMFHPYYgC3A\nc8CP8vn2wESgEzAT+AD4BfA9o67LmnLJoLoX5tqrGcpay5q8y2zTXsvSxxrcFFhRpYZpN/pU\n97RazSqElBQUdoSQxKibgFWzxOUAZT4jFiOBvMev5YZaRxuAh4ARQAfgl8DhwN+AxcDlAN5/\nX23FFCKy9RW8lhNwbfHRPapwFFwxa3qmnhMG02eFWhsoVz5aWFeLBRLVC5RDJRKHVCoUdoSQ\nVImx4BUVH+FYjARCbTABM+flwCSgF7Af8ATwYyC/zz6vAufkcu2NRuXVUriNt9C+aButsswV\n00d+ac1Zdb+P82eVaK71WfXPQs6E9DUWVR0JhcKOEJIAmicUumhYVMyFvBQadbUeinC3Wd32\nfwHXAvsDPYBXgAuBj4H80KHjc7l2SouePqggqsysgpjW4xiEMpdoXcmo21V7T20lUndcaWv5\nqMG1JEM1cWhzyZYMjRO1SrE/R5FSg8KOEFIUTPtKnbytU295kVTaggzVxnARcBGwNzAImPzS\nS9cCK4CpwI+BNobppcomGMLCTN4lx7U4cDy7oEYwV1RDpXZO+UFYc4hCVZGmF0tWxySSW5m+\nX0hqUNgRQpLHdJ40VZHavU1mMqUWVhM0VsGkbqwFXgbOyuf3AkYBnwC/BFYA/wvg4Ydbh63D\nwhht08oKjeCzS7YGNUdQQzgfBG0UKpsEc1QrUBZ4ysTy6hQpHAo7QkjyuOYS6zxUyhZLCoT2\n3TVoW4GXtv+axRhgHYCzz14J/BE4IZdrVd8rVQdZWz9Vt/us0uYc35bicxxdcsq1Kyk8z7Fq\nPg/TQTsPOeDFgMKOEFIUzAW43PZ71Xk1T5ZvgOeAUwB89tmxwHpgMrASwDHHnAC0BmCsgHsq\nMFUayvfMCSuz6p+hN7Rp68hmMTOIa5eAULKMTs70rTi+ecuCRlknQAipFtQJ21yNFWrVvSj9\nGcW/U8VoGtuH6FmgCTAcGPPHP94B3A9g9GiMGZNfs0ZI0rrdlHeBOrdW1+SjmptP/mqjPgj+\nn2ujueIcI9vYxA6uvnGSSycOBSagaX1XMeuhIZ7QsSOEFAXrHVTBxtB7oYLXgfOXvj9hlSkC\nkZL0DyuUdKneGuBZ4P8B7YEfANhtN1xwQU27dhg1Cvfeu6tRxaXq5EXSRCSRp9Un2HsxGnVR\n+koi0ieHxA220h8fAgo7QkhqaIuzwjKfazZK0w8T3CkB11TqmhF9ZkohAXOlW6MGeAHAffc1\nXr36KGDKCy/gqquWA7OBc4AOYU1ot+VZM88pWCOYuwKl7iovdDMe6lmXwi191qZde30yCdKu\nEmmV/qe4SoLCjhCSBtrsFcxSPq6Pj8mXODGaM30mOZrWqUQ8PLNYXStbgBeBCUDDFSsazp27\nCDgP+DfwOpC/8cb9jIqq7DZ3uRIwBVxohoHMsuo5dUkuxkwvKE45mqxWIyVQSHWNqCNAeVSd\nUNgRQtLAJRTMMpUxG4VqAquP5R+87oXgeqqOmrq3FkD//hPz+U75fE9gBoDf/e4fQP6AA64H\n8n/9a762VkhGTVtVXWqjmj1mOmRqPqbiN+1bzd7Thk4eN0GJutoyuyzED0U+mSvjbCclBR+e\nIISkROgMV/dCUwApG3UufDKJlGrhXZMjqOpKECsL8nkA++dyP3jnnf8C8r16fQz8CRgKzAG+\nCbu93XQcg+3C4c7Vv3vPuuDruWaqNae1a92o1XVpPjOlqPgMReyw2crBxDtFEoSOHSGkRFEX\n4KwmkE/1kqXApV6f7a6G1Fm5zqP6J3AT0BvoANwInD1s2PN1X5hy4ol46qmWRls522MippFm\n6jZzo08XzP6aZwUUIRXbdbMan7FPpNgryCWO9QiS0oHCjhBSQpirk6YnBD/TItmJxz9agYtr\n/nUDdRXpbrCginXvJ8BvgNyMGY3Xrv3WI4+gpgannrqhWbP8EUeckcvtZei2vIK1I+aqMRxD\npIVydUrQbUFMTbb6L8XGxpVtMVRdUmn7nzNmAqhEwVoxUNgRQkoOH+Mknn2VArHny6C6/y6f\ntqzWWnjJtm1x3HG5J5/E55/jmWemPPfc1cDHAA47LH/ddfm//c2am2myqloteB2kbSo5uS+h\nXUtQq0XNBKmfe9ZPQekYackK4kRCkQDeY0cIKQlME05bMUR9SybYYk5vgiMViaTiFLsh0xtT\nh0sbKEHVof44byvfrFldgf8GegDff+ONH7zxxoFXXpnfZx+MHo2ZM9G/Pxo3riujNWEKSu34\namVctl/ULT67EsGUs6HFkm06V8C9bgXm4++am2UKSZuEQseOEFK6WFWI+infVaCQRjU1mQIF\nNqQOiMucC21CG1XLqiIwH/gZcBCAZcsmLls28/bbMXIkdtsNxx2Xf+SRul+2EOJbM9Ey11Z4\ng0xka8fH+JHPpRSso8Kb0A60Zk+aW0oZ1bLNOpdKg8KOEFISWJfkVIsOcQVQ1JlDbqXY85DW\n5UJwDaNP60LFfD6PTp0mAUPyeXz+Oe66C7ncuuOP39Ku3UtA/pe/7GwkAPd9daH5W+1MednR\n3CW7lTJCLUGXmCIsRtOhiUXtkWvcIsUJIoQu8bs6XsjbmYRCYUcIKVGsU4LnZGAu+bkCRkVL\nINKcLdwQZtqE1p6GLjtq03bQRGCAudoNXqvGjzWZwFQDgDZt8KMf4ZFH2tbUHA78DXjv/POX\nAO8CNwP5OXOwdas5Yq4/tVb8+x6Ka7XU8/AlIkHUtuK5d4ULROs4RFW9wtIztVopQGFHCCkt\nhE/58sSmipWcchOPKVP8Y4bimqE9pz3/pdLQYqErlWpbLtmUU7CG0qTJjpKNG88GzgO6AF2B\n/W+++aL+/TFkyJpGjR4Gjs/lsHatNYgrf1U9hBpjkQPYegAAIABJREFUWoEY7mDipppVZ3vi\n/+klRs6ukYkUSniHRopAIVgM+PAEIaQSMI2fXNhv2AuhrLpHdf7kAp6NCuJJc908A1ojuHSn\n5tMIvpq27mZ2X5PR27ZedBFWr273wgvHP/vsqMcew267oW9fHHUUjjoK+++vdkrrrFWrhXbT\nmoy5y+ydq5taQ3ImUVG7aS0g7Coccxw82xKGiJQOFHaEkDJD83uEaSZUflnLWJVQqGhL0Hvw\nScmqadTyphyMNB+rc79rtdRMwyJJTzghd+KJDYH+W7YcNWfOBStW4KKLsO++OPJIHHVUU0eG\n1rVCzZOzbtc0vU8f5ZL+hzW2WrImk4J4ylyfZZ5ApcKlWEJIuRJjYkh/6afwNT7/u83UtuRF\nQP/lYyGUVt6sHlTcCswGLgRy//wnlizB2Wfj7bdx1FGrgKeB/H335T/9VIsTbxFTXlWUV95j\nLODGK6mWj3QgMifxBWtSDCjsCCGljs8tXzLx1uY8CbV8CllWC5UakcScufSpxpHtPauPaK7k\nquIpX59t1Tp3xsSJmDEDq1b9GFgP4H/+B3vtNR+4NpfD66+jtlbNQViWLVyK+RQ27UNZcboM\nXblWmpopdlvx1DZJGQo7QkhZktuOXKxuDvM3vXz2mpmEBok9HYaqhKhzradF5Fr9tNZ1CYWQ\n9etWrf4InArg00/xxht/BkYC6NMH7dvj5JPzjz+OdeuEnLVdMTSHPIzx/MJIZ44syq26uahQ\nt1UG5X2PXU1NzVtvvbVx48ZOnTrtvffeWadDCCkKMRa5rLdPqYtfwp1hSeEZU1iSi0poR/KO\nX+ywumKa16iWydl+Q8LsjstOC/buKNCz59V1r1euxPPP49lnMX48vvwSffpcCtzw1ltaW9qh\nFM4Qs7yaiauwMIyhBq25pUA3zqebodUTsQNdgxYaPMEcSChl49hdd911s2bNUrdMmTKlffv2\nhx566JAhQ/bZZ5+ePXu++eabWaVHCEkZVWRo64Auf85lSLh0idB0Ybk7Ww/ix57C5dXJUOvL\nOkSmqySYkYLt5IoQbM/lcthtN5x8Mp58ssn69YO3bMFhh93QrRsOOggdO2LCBPzpTy0dYeVB\nc/mU+fo/XGt2OSrWNHxUu1kx2CJ4okjUd3QlHxqEPl9JUTbC7sorr5w2bVrw59SpUydMmLBp\n06b/+q//OuOMM/r27btgwYJBgwYtXbo0wyQJIVlhyjtZzAUlzTihxbQyggjwlGj+k7GPNWI1\njcwgVotOCxLqTrncGi14JLepLuw3wGwgd8stWLwYH3yAyy/HJ5/g+OM3NG2aHzYs/8tf7udo\ny9ojtaSm3YPRsKbtmbOA56nogyq+fUJpfSweVpmuFaBdlxrluhR73nnntWnT5i9/+UvXrl3r\ntjzzzDNjx469/vrrH3jggWxzI4Skg8/CmUpe+eLiBD2GUBsswZihVmKk+V6rYq1u3WJ6S64g\neeML84TkrTHrRdu8GbNn4/nncddd/wCwzz4YNQrPPYdBg7DTTnJ8rUc+/lyyWsT/RLXiM4zC\ncYlN6CCEGqWhHw98WiH+lI1jp/L5558vWbLkrLPOClQdgGOOOeboo4+ePn16hokRQjJBW6sy\n3TvY1hOtda2RkdbEY07MkapHmvWt1WVv0rrLU7HJrVtllsVTbN4cI0di0iT8619YsgTnnov3\n3sPYsdh552m53MRcDv/6lxk5nl1UjCNeSEyXsyj3LmrffVZd5TIUaplTlsLuq6++AqCqujq6\ndeu2cuXKLDIihGSAddKyzjqutUVNAprBNQkYmk/oFhmrHpWn7dBWVEVrkUruxyDUCNYmzMjW\nYuYIB6apuraolQ8WHPP12RG3c2eccw5eeAGrVuGZZ5YAZwHYbz907oyzz8bUqdi0yZV2icsO\nl3gq9meMZJeeYQy159mldr/w9esqpCyXYvfcc882bdp8/PHH2vZPP/20VatWmaRECCkRfOSX\nS/ype/0dKSF44XOSvNwsLM+ZHXGNTLzl3UBkuKrnlecSzBZltWqOoZx8Pp8/u67AkiV47jlM\nm4Zjj0U+j/79MXIkRo2CYQT4UC7mUyF5Rq0bezSoz1KjnBy7Dz/8cP78+e+9997atWvPPPPM\n+++/f9OmTcHef/zjH48//njfvn0zzJAQkhXyJ3vNFdBMBfWFLIOsFT1zsEZz7bKaWJGw3hvn\nci4FE0uLExiZprkYNOEaba1HVlcyiBDU9enjjshduuDcc/Hcc1i9Gs88g65dcffdOOAAdOqE\nCROOyeVaJ3SAiofVWE2nXRSzy6GHUj05XZ4f8aGcHLtHH3300UcfVbc8//zzY8aMAfDII4+M\nHz9+8+bNV155ZUbZEUJKFNkfUsnbvgAvrzxyoYW11o2UWKQCZhOyYSa0oi0xy2E9R8/qcZqS\nTi0f7DUX40JbtyrXHeWbN8eoURg1CpMmYelSvPACpk37PdAEwODBGDECI0fioIMgPqRiVRWq\nh5q45ghM1kiRC0nDHPlMtF2RxlOjXCzYAikbYffb3/52ncL69evXrVv3rW99q27vunXr2rZt\n+9hjj/Xq1SvbPAkhmRBj+nd5cuZrVZ0IDZm6MFf/C36F5H1Eoasv1gg+AYXV0kgTrWZzulxP\nq8jTpExUfWwVXvrGfffFWWfhrLNa1NRg7lxMm4aHH8bll2P33TFiBEaMwOGHY5ddQiMLaSdF\nhpojHWklJ5BV0xVGlkcxQTZu3LjTTjs1aFCUleUpU6ZMmDBhw4YNLVu2LEZ8QkixUfWWqc9C\nP8cLFpQqRHxex0PwvRKPaVWi8tBpFSEOuNUdVFWFYB+GyjjVTrOW38HHH2PaNEybhhkzsH49\nevbcJvIOOwyNvCwPq9FYGVMqCaWmpqZp06avvPJKnz59ss5Fp5zusRNo2bJlkVQdIaRMEdYo\nXdIklLxxe5m6y9W6fxOmj+jjLPpsN0O57reLLR9zCq4CZp6B9aWmFBTwtM0C5Fuy6uX27W/j\ntNPwxBP4/HPMm4cRIzBtGgYOxK67YuxY3HMP/v1vrV9atECGal0QEi5ThGNKShCKIUJIxaKq\nBKvZo8kvWQtqMk5dqDXXNM0qWkBrKLOYphvUbK0SMMYEbF3h9YwjdNMcnyDtQLSZowplQNQI\nPml7jqpOw4bo0wfXXIPXX99561ZMmYLWrfHzn6NTJ+y/P849Nz91aguPhUL5iFcMFHmlT9nc\nYxfK0qVLzzjjDAAzZszwr7Vs2bL999//m2++CS3JU5mQssNnllWto6jlzSYEaaWuMwrXE21d\nUogs/2mN7NplXQOV97q672rR7FQQQTA+Q5Ezzyk3PvoEXwPkxo3bVmzxYkyfjmnTcO+9G5s2\nxdChGD4cI0Zoj1x4Jhy+NBxW11q9kLD+yEeZlBqVc0PAm2++2b17d0Q85/L5/CuvvFL3jccu\n3n777YkTJ3799ddNmjQpNEtCSKWgzanCUp36J2xqQzCftJvG/DEllLbXXBi1LpVqQVy6VmtC\n7pRWxeeiLctuU99owV2Lzq4t9cpv3oy5czF9OqZPx9//jt13x7Bh2x65aN8+NHMzmai4qhcY\nVgtVsmIgHfEalVK+x65yHLv9999/8eLFUWvlcrl+/frJZXby/glCQkj1YGojq/YynSp54bXu\ndc54xtaMbwrHSB5eUN1qEJoSR1BIrvSsy8eeGsvsSKj4kAsIbqK1jzto3jw3cuS2CJ98ghdf\nxPTpOP98rF6NAw/E4Yfj8MPRvz+aNzeDa2vK8ZCrJ6LqEEvblabkIpUj7Jo1a9atW7essyCE\nVAvCdGhaR1YDSS0AQ0/4LK1a59S8+0tM1AKw3f3m7wv6aDu1pDYOPkamFlkWbdbDEbz2z1bo\nIADstRdOOQWnnILaWixatM3G+/Wv0bAh+vbF8OEYPhwHHhiacCIkFT+dbGNTsomVLOUn7PL5\n/Pvvv79s2bINGzYAaNOmTZcuXTp06JB1XoSQ6kKYDtXtQbHQ9VBzr3WLVRS67D2Xz+dKw6e8\nC2tiZp5wOD3C4q+PpRREznn/coNL9slKfRsNGqBHD/Togcsuw5dfYvZsvPgifvc7XHwxdt8d\nQ4fmH3wQn3yCvfbS+hKaVSbESyxSrVLWjhVGOQm7tWvXXn/99b///e9Xrlyp7erYsePpp59+\n4YUXNlfMcEIIKSrCoqrrT012qOpQnfs9nTOXNHQpJ1P3mCXNXVqGkbJS1yIF1SVYa1oOsfGR\nVq5l4pCKLVrgyCNx5JEA8MknmD4dM2bg4ouxciUOOACHH45hw1oAXxaSfZkTe7WXxKBsRnn5\n8uV9+/Z9//33u3Tp0rdv3+985zstWrQA8MUXXyxdunTOnDmffvrpQQcdNGvWrODnKJLi1Vdf\n7du3Lx+eIISEok1dVq3muuq6bDPzvjRZArruYzNzkFPScrB2MLQLrtGQI4Q2EWPm8lEVMYWd\nlXweb72FGTPw4ouYNw9btqB37zqRh169tO9ArgbFU2F95MMTCXDllVd+/PHHTzzxxLHHHmvu\n3bp165QpU84+++yf//znt99+e/rpEUIqjHjTuea9qcuIoZE9fSn1NjV5XdUqs6xC0FXe2kFB\nsaldEFZ1rRGyuiNNGBm5oosdvTv4YBx8MC68EJs349VX8eKL+N//xdVXo1UrDBqEww/H0KHY\nf3/tHsdijIPweSM1KknVlThlo6D32GOPI4444v777xfK/OhHP3r11Vc//PDDZJumY0dIFRJq\nsIXWlZcvg72meeZq2v9mf1NRuTw/l4dnDWv2Tt4it242Gm8+sooh/5jawSp8TgyRTatW4aWX\nMGMGZszABx/g29/GsGEYNgxDh+b22MMz5wJTKvEb/soCOnYJsHr16n333Vcu07Vr1z/+8Y/p\n5EMIqWwEC8eKsIqnxjT3usRE6A18rvSsS6ies3joKrA1lBBckFxaK7FVHWxjmPd75AJhXYtB\nyJmzyy4YNw7jxgHAe+9tU3jnnIO1a/PdutUpPGzYgFatCs9ETUn4k1QYZSPs9txzz7feeksu\ns2jRoj333DOdfAghFU/s+U+Y2lWdpD5SoBYIlXSF55lXHqRQF4vrllNdtQSnR9tlCkRXzq7H\nKQS0MQxdNQ5NPlKLnoW96NwZnTtjwgRs3Yo338SLL2LmTNx9N7ZswWGHYehQDBuGww5D/cWi\nAg1OUvGUzW/F/uAHP3jyySdvvfXWr7/+2tz75Zdf/uxnP/vTn/40ru5jECGEpIumTlSvKwCx\nxJmm/IKNropaJtbqrqatdlpU1eWjEbUq/vFh85+EUU3ckKtDyDlOKw0bokcPXHoppk/H2rWY\nNg0DBmDaNAwejHbtMGoUbr0Vixahthb1PxVkQugZVWHtlh1lc4/dunXrhg4dunDhwlatWh16\n6KEdOnRo2bJlPp/fuHHjv//97zfeeGPTpk39+/d/7rnnWrZsmWzTvMeOEBKDGBaRcLuYdRnX\n8wYyONRbMa7/rsTUrhXvEQFr00VqKI221q7F7NmYNQszZuDdd7HLLhg0CEOHYuhQdOlS3Kbd\nZHWLXkndGljK99iVjbADUFNTM3ny5Iceemjx4sVbt24Ntjdu3LhHjx6nnnrqqaee2rBhw8Tb\npbAjhMQjnanIuugpiA/z2Y54AjFGhq7qiQxUmhN/UR/jtQdfsWLbWu3MmfjwQ3TogCFDMHQo\nhgwJvgaZpAaFXcJ89dVXH330Ud0vT7Ru3bpjx45FlVwUdoSQePg8QFp4dUHYhd4P5/m8ZOHK\nLDSyXD0RfNzEzPE6Q5YswaxZeOklzJyJVauw334YMgRDhmDQIOyyS0qJVjelLOzK5uEJlWbN\nmnXJzoUmhBBPkpILVoWnPqlgFhZad912Jt835vmQqakUZWGadzwS6/+ghj/WJy1iP2BRJINQ\neBxkB126oEsXjB+PfB6LF2PmTLz0Ek49FV9+iQMPxODBGDIEAwagdetkcyNlQVkKO0IIKVMi\n6QDXw6Q+Giu2TSjIQf/vZ3GpJWvdNN2ypFov6rJvhLC5HA48EAceiIkTsWUL5s/f5uTdfTdq\natCzJ4YMweDB6NsXO+1UjFRJCUJhRwghGRAoAx+J4LPAahaOSuhddy4zSRV8WhDza1AiJQOb\nFozaQf+V1tJZkI1Do0bo3Ru9e+Oyy/D113jttW035N12GwAcdtg2kde7N5o2zTpXUkTK5utO\nCCGEuJC/CSLqV5ZExfwCjqBFn6ZdyQfbC/mSiyJ9QUaxh7RQmjbFwIH4+c8xbx7WrMGf/4w+\nffD88xg6FN/6FoYNw3XX4ZVX8M03WSdKkoeOHSGEZEAME0suHOn7UAQTK55eSfBmuNgmnyux\nav/ysxYtMHw4hg8HgC++wNy5mDULf/wjfvYzNG+Ovn0xaBAGD0bPnmhESVAJ8CgSQkjZI9zB\n5nq01sR/nddsqJB7zpKVmMWLUwm0bo2jjsJRRwHAunWYOxczZ+Lxx3HFFWjRAv36YdAgDBqE\nHj1QhO8OI+lAYUcIIWVPoNUCyaU6XpqSK8TEMhsSyNz8IxJt22L0aIweDQBr1mD2bMyejd//\nHpdeilat0L//NpHXvTtFXnlBYUcIIZVD6POqPsXk75zzbMIfirbsadcOxxyDY44BgM8/x5w5\nmD0bDz6Iiy9Gq1YYMAADB2LgQBxyCEVe6UNhRwghZU86d5JFbUV29RL5EhaSPLvuirFjMXYs\nAKxciTlzMGfODpHXv/8Okcd78koSHhVCCKkEEpQ+/o9ohFK8FVuSBrvthmOPxbHHAsDKlZg7\nF3Pm4KGHcMklaNkS/fptE3l88KKU4JEghJCyp6R+FEslQY1YIFzwLZTddtvh5H3+OebNw+zZ\nePhhXHYZWrRA377bVmx79QJ/fjNTKOwIIaS88fyxrwTbgp9CSl+6VbBuKy1huuuuO+7JW70a\n8+Zhzhw89RSuugpNm6J3721O3mGHoVmzrHOtOijsCCGkvHH93Gq5txWDxL+fr6Qo0S7svDN+\n8AP84AcAsG4dXn4Zc+Zg6lRcey0aNcKhh2LgQPTvjz590KJF1rlWBRR2hBBS9hRpyre6RCUq\nLwCUdm4FUh5da9t2x/fkbdiAV17Z9lV5N92E2lr07IkBAzBgAPr2Rdu2WedasVDYEUIIKQop\ne3vlIX2qh1atMHIkRo4EgE2b8Npr2569uOMO1NTgwAMxYAD698eAAdh116xzrSgo7AghhNgp\n/NvpCAGAnXbCkCEYMgQAvv4a8+djzhzMm4cHHsDGjdh//20Kb8AAdOiQda5lD4UdIYSQ5OGP\ntBI7TZuib1/07QsAW7di0aJtTt4552DNGnTqtE3k9e+P/fbLOteyhMKOEEJIUQh+WBZcJyVW\nGjZEz57o2RPnn498Hm+/jblzMW8efvYzfPopdt8d/ftv+3fggfzRC08o7AghhBCSNbkcunVD\nt24480wAWLp02wO2d96Jc89F69bo2xf9+qF/f/TqxW9REaCwI4QQUkTq1mRTeJCC1mBFse++\n2HdfnHwyAKxYgXnzMG8ennwSV12FRo3Qqxf69UO/fnzA1oTCjhBCSMYkosl4V1/F0r79jl82\n27gRr72Gl1/GK69g0iRs3ox99sGwYdt++qJTp4xTLQEo7AghhBSX1Fw02nWVT8uWGDYMw4YB\nwJYteOutbSLv/POxejX22GObjdevHw45BFUp9CnsCCGEZAwFGYlDo0bo0QM9euDccwFg2TLM\nmIGXX8akSZg4Ea1b49BDt5l5vXqhadOs002JBlknQAghhBBSMPvsg/Hj8dBDWLYM77+PO+/E\n3nvjd7/DgAHYeWcMG4arr8Z772WdZdGhY0cIIYQUBT7PkRmdOqFTJ5x0EgCsWrXtx81eeAGt\nW+P887NOrrhQ2BFCCCGkctllFxx9NI4+Ous8UoLCjhBCCCkK9OpI+vAeO0IIIYSQCoHCjhBC\nCCGkQqCwI4QQQgipECjsCCGEEEIqBAo7QgghhJAKgcKOEEIIIaRCoLAjhBBCCKkQKOwIIYQQ\nQioECjtCCCGEkAqBwo4QQgghpEKgsCOEEEIIqRAo7AghhBBCKgQKO0IIIYSQCoHCjhBCCCGk\nQqCwI4QQQgipECjsCCGEEEIqBAo7QgghhJAKgcKOEEIIIaRCoLAjhBBCCKkQKOwIIYQQQiqE\nRlknUAY0adIEQNOmTbNOhBBCCCGlQp08KDVy+Xw+6xzKgLfeemvLli1ZZ5EM8+fPnzBhwu9/\n//usEyEE77///lVXXXXvvfc2a9Ys61xItfPpp59ecskl06dPb9euXda5kDKgUaNGBx10UNZZ\nWKCwqzpmz549ePBgHndSCsyfP79Xr14bNmxo2bJl1rmQaufdd9894IADVqxYsfvuu2edCyHx\n4T12hBBCCCEVAoUdIYQQQkiFQGFHCCGEEFIhUNgRQgghhFQIFHaEEEIIIRUChR0hhBBCSIVA\nYUcIIYQQUiFQ2BFCCCGEVAgUdoQQQgghFQKFXdXRpEmT0vx5O1KFNGnSpEGDBo0a8UerSfY0\nadIkl8s1btw460QIKQj+pFjVkc/nP/jgg7333jvrRAgBgGXLlu2zzz5ZZ0EIwLORVAQUdoQQ\nQgghFQKXYgkhhBBCKgQKO0IIIYSQCoHCjhBCCCGkQqCwI4QQQgipECjsCCGEEEIqBAo7Qggh\nhJAKgcKOEEIIIaRCoLAjhBBCCKkQKOwIIYQQQioECjtCCCGEkAqBwo4QQgghpEKgsCOEEEII\nqRAo7AghhBBCKgQKO0IIIYSQCoHCjhBCCCGkQqCwqyLWrVs3ceLETp06NWnSZM899zz99NOX\nL1+edVKkGnnwwQdzNq677rqsUyNVwTfffHPZZZc1bNiwZ8+e5l5eKklZ0yjrBEhK1NTUDB06\ndOHChWPGjDnkkEOWLl360EMPzZw5c8GCBd/61reyzo5UF+vWrQNw3HHHdezYUd3et2/fjDIi\nVcS777574oknLlmyxLqXl0pS7lDYVQuTJ09euHDhTTfddPHFF9dtGTFixLhx466//vpbb701\n29xItVEn7M4//3yrX0JI8fjiiy969OjxH//xHwsXLuzWrZtZgJdKUu7k8vl81jmQNOjevfvS\npUs///zzpk2bBhu7dOnyxRdfrFixIpfLZZgbqTYmTpw4adKkJUuWdO7cOetcSHWxZs2aX/zi\nFzfccEPjxo2bNWvWrVu3+fPnqwV4qSTlDu+xqwq++uqrxYsXH3rooeqlCkC/fv1Wrlz5/vvv\nZ5UYqU7qHLu2bdtu3br1448/XrVqVdYZkWqhXbt2t956a+PGja17eakkFQCFXVXw0Ucfbd26\ntUOHDtr273znOwCWLVuWRVKkelm/fj2A22+/fdddd+3QocOuu+663377PfLII1nnRaodXipJ\nBcB77KqCDRs2AGjRooW2vWXLlsFeQlKjzrF79NFHL7744r322uvdd9+dPHnyCSecsGHDhjPO\nOCPr7Ej1wkslqQAo7KoI8+6QujssedcISZkrr7zy7LPPHvn/27vXoCauPgzgJxKCRMNNbiZg\nMBW5iAqBqjhYFWgFtEBFLVKKpVAVFfGC4qUqYLVxsPWOHVqVih1EphQ6ikPraKlUFItKBUEF\nAWkJVRGUFJEG8n5Y3TcNIaCiaZfn98mcPbv7383M8rhn98THh/4LGhoaKhaL161bFx4ezuFw\ntFse9HO4VMJ/GoZi+wUDAwOi7r+bDx8+JITweDwt1AT9mKenZ1BQkPJ9EUdHRz8/v/v375eU\nlGixMOjncKkEBkCw6xeGDRvGZrNra2tV2quqqgghtra22igK4B/Mzc0JITKZTNuFQP+FSyUw\nAIJdv8DhcFxdXYuKilpbW+nGzs7O/Px8a2trlUliAV4qmUy2f//+9PR0lfaysjLy9Cl1AK3A\npRIYAMGuv4iIiGhtbU1KSqJbUlJS6uvrIyMjtVgV9ENcLnfLli3z58+vqKigG3NycgoKClxc\nXEQikRZrA8ClEv7rMEFxf9HR0TF16tSzZ88GBASIxeLy8vKMjAwnJ6fz589zuVxtVwf9y/ff\nfx8YGMjlcoODg/l8fmlpaXZ2No/HO3PmjFgs1nZ1wGT5+fknT56k/r19+3YzM7N58+ZRH1et\nWjVkyBBcKuG/DsGuH5HJZAkJCZmZmfX19ebm5oGBgYmJiSYmJtquC/qjwsLCzZs3FxYWymQy\nc3Nzb2/vDRs24Ico4GWTSCRr165Vu4j+KRRcKuE/DcEOAAAAgCHwjB0AAAAAQyDYAQAAADAE\ngh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDY\nAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0A\nAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAA\nADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAAQyDYAQAAADAEgh0AAAAA\nQyDYAQAAADAEgh1AfxccHMxisX7//fdXvN8lS5bo6ekVFxe/4v2+MtSJbWho0HYhr9TGjRs5\nHE5+fr62CwHopxDsAKCPNTU1xcbGCoVCPT294cOHBwYGnj9/XqVPenr6vn37tm/f7urqqpUi\ne0MikVRWVj736s7OztOmTdPT0+vDkl5cc3PzsmXLbGxsOBwOn8+PjIyUSqUa+qemprLU+eST\nT9T237Rpk7u7+5w5c+7evftyjgAANGEpFApt1wAA2hQcHJyRkVFXV2dlZfXiW7t//76rq2tN\nTc306dPFYvGtW7cyMjLYbHZRUdHo0aOpPjKZzMbGxtbWtrCw8MX3+JJIpVI+n3/y5EkfHx9t\n19Jn2tvb3d3dL126FBQUJBaLq6qq0tLSrKysiouLjY2N1a6yc+fO5cuXz507d9iwYcrt06ZN\nmzp1qtpVKisr7e3t582bd+DAgb4/BgDQTAEKu6TzAAANSUlEQVQA/du7775LCKmrq+uTrS1e\nvJgQsmfPHrrl22+/JYT4+fnRLRKJhBBy4sSJPtnjS5KTk0MIOXnypLYL6ZlQKFy5cmVven7+\n+eeEkG3bttEtGRkZhBANq2/atIkQcvHixWcqKSQkhM1m37p165nWAoAXh6FYAPiH2tra8PBw\ngUDA4XBMTU39/f2LioqUO5w4cWLcuHFcLtfS0jImJubRo0fW1tZisZhaqqur6+XltWDBArr/\nO++8o6+vX1ZWRn3s7OzcuXOnvb29n5+f5koaGhoiIyMFAsGgQYPGjh27a9cuuVzeyzpnzJjB\nYrGam5vpFrlczmKxvL29qY8hISEsFksmk8XFxdnY2Ojp6VlbW+/YsUOhUFCrBwQEEEJ8fX1Z\nLFZBQYHaCh8/fpyUlDR27FhDQ0MejzdmzJikpKTOzk5qKf2MXU1NjdrRTFNTU3pTf/755+LF\ni4VCIYfDMTMzCwwMvHjxoubz8xwOHz7M4/FiYmLoljlz5owYMSItLU3RzegNdQ6NjIyeaUcr\nVqyQy+U7d+58kWoB4DmwtV0AAPyL1NXVjRs3rrW1NSoqatSoUX/88UdycvIbb7xx6tQpDw8P\nQsjPP/8cEBBgZma2Zs0aU1PTzMzM4ODglpYWgUBAbWHHjh0q22xvb5fL5fQ476VLlxoaGubM\nmaO5krt377q5uclksrCwMKFQ+NNPPy1btuzq1atfffVVb+rsEYfDIYTMmjVr+PDhR48e7ezs\nTEhIWLFihZGRUXh4+Mcff2xiYpKWlrZx40YXFxdHR0e1G4mKijp06FBISEhUVBSLxcrLy1u9\nenVtbe3evXuVu5mamn755ZfKLSUlJXv37rW3t6cPdvz48c3NzQsXLnRycqqrq0tOTp40aVJe\nXt7kyZN7czi90dbWdvXq1SlTpqg89ufh4ZGamlpdXS0SibquRQe7jo4OqVQ6cOBA5TzaHbFY\nbGZmlpubu2vXrr6qHwB6Rdu3DAFAy5SHYufNm0cIycrKopdeu3ZNR0dnwoQJ1Mc333yTKA3M\nyeVy6kGr8ePHd7d96k87PTj76aefEkKys7M1VxUVFUUIycvLo1umT59OCCktLe1NnVTnpqYm\nusPff/9NCPHy8qI+RkREEELmzp1Ld6iqqiKEzJgxQ7lOzUOxXC7X3d1duWX58uVBQUFyuVzx\n9MRKpVKVte7fvy8SiUxNTWtra+mDZbPZysOdt2/f5vF4bm5u3e36phKBQBAREUF/7LpHyo0b\nNwghH3zwgUo7Ndj6448/ql0rMDCQELJ+/Xr6IbyRI0d+88033RVGow6/urq6x54A0Idwxw4A\nnlAoFNnZ2RYWFtTfcoqDg4O7u3tBQUFjY+OQIUPOnj1rb2/v5uZGLdXR0YmLiztz5kx328zP\nz1+1apWHh8fChQuplps3bxJCRowYobmSY8eOWVtbUzmSsnv37pUrV1pYWPSmzl4eMhUQKSKR\niMvlPtO0L7q6urW1tXfu3DE3N6daqIfYNFAoFKGhobW1tXl5edTrCAqFIjMzc8yYMVZWVvTc\nKLq6uhMnTszLy5PJZIMHD1bZiFwut7W1VW45cOAA/aZCQEBAdnZ21123tLQQQgYNGqTSTm2f\nWtoVdccuPT199erVAoGgvLx837597733XktLi/KAe1dUhZWVlTY2Nhq6AUDfQrADgCcaGhoe\nPHjg6urKYrGU2+3s7AoKCm7cuOHg4NDW1qaSySZOnNjdBtPT08PDw52cnHJyctjsJ1ebe/fu\nEUI0D+dJpdLGxkaxWKxciUgkosYKpVKp5jrd3d17ecgqb3rq6upSN/Z6KTExMSYmxtbWNiAg\nYOrUqW+99RY9JN2dhISE3NxciUTi5eVFtdy5c+fevXv37t0bOnRo1/63b9/uOhCso6OTmZlJ\nf1y0aNGECRPCwsKoj5prUDlphBCFQqG2nbJhw4YlS5b4+PjQiTA0NFQsFq9bty48PJwa0VaL\nCrvU1w0ArwyCHQA88ddffxF1d3T09fWppY2NjYQQLpervJTH4+no6KisolAo4uPjExMTfXx8\njh07xuPx6EUPHz4khBgaGmqo5NGjR4SQ7maA67FODVtWoaur2/vOXS1dutTJyWnPnj1ZWVlp\naWksFsvX1zc5OVkoFKrtn5ubm5iYOHPmzLi4OLqRulXm7OxMDf6q4PP5XRtZLNasWbPoj7Gx\nsSNHjlRuUcvAwICouzNHfSPK35EyT09PlRZHR0c/P7/vvvuupKTk9ddf72531PsWDx480FwV\nAPQtBDsAeIIakusajKgWHo9HxaC2tjblpa2trR0dHcotCoUiMjLy4MGD0dHRO3bsUIl9VLx4\n8ODBwIEDu6vE0tKSPB0EfI461a7V3t7e3e5ehKenp6en5+PHj8+ePXvkyJHDhw97e3uXlZV1\nvZV169at0NBQOzu71NRU5Xa64Jc9Yd6wYcPYbHZtba1KO/VwocrYrmbU3TiZTKahD/X1aU7w\nANDnMN0JADxhaWlpYmJSXl6u+OfMF9euXWOxWHZ2dpaWlgMGDFBJBhcuXFDZzvLlyw8ePLh1\n69bdu3d3vZlHDcJSN/+6M2jQIDMzs/LycuWB0evXr+/du7esrKzHOsnTW3HKq1dXV/d0Ap6f\nnp6et7d3amrqwoULKysrr1y5otLh0aNHM2fOlMvlWVlZKtHTwsLC1NS0oqJCJcj2+S83cDgc\nV1fXoqKi1tZWurGzszM/P9/a2lplVJoik8n279+fnp6u0k5NXtPdjUkKVX9vXqEFgD6EYAcA\n/zdz5kypVErNzUu5cuVKUVGRp6enkZERh8Nxc3P77bffKioqqKUdHR3btm1T3kJWVtauXbti\nYmLWrl2rdhf0M/WaKwkICGhsbPz666/plvj4+Ojo6MePH/dYJyGEel6tvLyc7nD48OFenYKn\nqEhKDQqrdf78eYFAoLLZAQMGEHUjvAsWLCgpKTl06JCDg0PXTc2ePbutrS0pKYluuXv37pgx\nY95+++1nqrlHERERra2tyjtKSUmpr6+PjIykPra1tV25coW6h0cI4XK5W7ZsmT9/Pv2NE0Jy\ncnIKCgpcXFzUTo9C681bMgDQ5zAUCwD/l5CQcPz48ffff3/p0qV2dnY1NTX79u0bPHgw/bLn\nqlWrZs+e7efnt2jRIgMDgyNHjohEIuWH4VavXk0I6ezsXLNmjcrG4+LijI2NqZcGTp8+7e/v\nr6GSTZs2HT9+PCoqqqSkRCgU5ufnHz9+PCwsjJoJucc6w8LC9u/fv2LFiqSkJC6Xm5OTU1hY\n2N0orVpUapFIJNXV1ZMmTer6MJmbm5uJiclHH31UUFDg7OzMYrF+/fXX1NRUDw8PZ2dn5Z5H\njhxJS0tzdnZuamqi5uGj+fj4WFlZxcfHnzhxYuvWrVKpdPLkyfX19V988UVjY+PSpUt7U2pN\nTU0vD+rDDz9MS0uLj4+/fPmyWCwuLy/PyMgYPXp0bGws1aGystLFxcXLy+vUqVOEkAEDBiQn\nJwcGBrq5uQUHB/P5/NLS0uzsbAMDA5UDUaFQKE6fPj1ixAi8EgvwqmlpmhUA+LdQ+Umx27dv\nh4eHDx06lM1mm5ubBwcHX7t2Tbn/gQMH7OzsOByOUChcv359e3s7h8OZOHEitVTD1Yaa0qyj\no8PCwsLBwaHHwmpqakJDQ83NzXV1dUUi0WeffUbND9fLOlNTUx0dHfX19S0sLObPn9/c3Mzn\n8z08PKil1Dx2N2/eVF7F0NBw1KhR1L/b29uDgoL09fWNjY0zMzPVVtjY2Lhs2bLXXnuNy+Ua\nGhqOHTt269atLS0tyidWKpWuX7++u3NCz5MnlUqjoqKsra3ZbLaRkZG/v/+FCxd6PEXPoaWl\nJTY2VigU6urqCgSCxYsXNzY20kuvXr1KlGb7o5w7d87X19fIyIjNZvP5/LCwMJXz1lVxcTEh\nJDo6+mUcAgBowFJovBADAGj28OFDQ0NDf39/5YFRzSQSydq1a3Nzc319fV9qbaAtoaGhGRkZ\n169f1zxcCwB9Ds/YAcAzOHTo0JQpU6j7MRTqHc9e/pAXZcmSJUOGDNm8eXOflwf/BlVVVUeP\nHg0LC0OqA3j1cMcOAJ7BhQsXJk+ebGxsHBUVxefzL1++nJKSwufzS0pKnul34tPT00NCQnbv\n3h0dHf3yqoVXr6Ojw9PTs6KiorS01MzMTNvlAPQ7CHYA8Gx++eWXLVu2FBcXNzU1mZubT5s2\nbfPmzWrn0dUsOjo6JSXl3Llzrq6uL6NO0IqNGzdKJJIffvhhypQp2q4FoD9CsAMAAABgCDxj\nBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYA\nAAAADIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADIFgBwAA\nAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAA\nDIFgBwAAAMAQCHYAAAAADIFgBwAAAMAQCHYAAAAADPE/C6YHVkF4A+EAAAAASUVORK5CYII=",
      "text/plain": [
       "Plot with title “voom: Mean-variance trend”"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# splice_list <- c(\"a3ss_\",\"a5ss_\",\"mxe_\",\"ri_\",\"se_\")\n",
    "\n",
    "splice_list  <- c( \"se_\")\n",
    "\n",
    "# parameters that do not change with splice_type\n",
    "tissue_list  <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "tissue_list\n",
    "length(tissue_list)\n",
    "\n",
    "metadata     <- ocounts_metadata\n",
    "plot         <- TRUE\n",
    "dup          <- FALSE\n",
    "obj          <- reduced_obj2\n",
    "\n",
    "# parameters that change with each splice type (3)\n",
    "# fromGTF\n",
    "# ijc\n",
    "# sjc\n",
    "\n",
    "for (tissue_index in 1:length(tissue_list)) {\n",
    "\n",
    "    # a3ss\n",
    "    splice_type = \"a3ss_\"\n",
    "    res = splice_list %in% splice_type\n",
    "    tissue_of_interest  = as.vector(as.character(tissue_list[tissue_index]))\n",
    "    message (\"processing \",\n",
    "             paste(tissue_of_interest), \" \\n\")\n",
    "    \n",
    "    if (sum(res) == 1) {\n",
    "        message (\"splice_list does contain\\n\",\n",
    "             paste(splice_type), \" continuing with processing\\n\")\n",
    "        fromGTF             = a3ss_fromGTF_no_chrY\n",
    "        ijc                 = a3ss_jc_ijc_no_chrY\n",
    "        sjc                 = a3ss_jc_sjc_no_chrY\n",
    "        print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    }\n",
    "    # a5ss\n",
    "    splice_type = \"a5ss_\"\n",
    "    res = splice_list %in% splice_type\n",
    "    if (sum(res) == 1) {\n",
    "        message (\"splice_list does contain\\n\",\n",
    "             paste(splice_type), \" continuing with processing\\n\")\n",
    "        fromGTF             = a5ss_fromGTF_no_chrY\n",
    "        ijc                 = a5ss_jc_ijc_no_chrY\n",
    "        sjc                 = a5ss_jc_sjc_no_chrY\n",
    "        print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    }\n",
    "    # mxe\n",
    "    splice_type = \"mxe_\"\n",
    "    res = splice_list %in% splice_type\n",
    "    if (sum(res) == 1) {\n",
    "        message (\"splice_list does contain\\n\",\n",
    "             paste(splice_type), \" continuing with processing\\n\")\n",
    "        fromGTF             = mxe_fromGTF_no_chrY\n",
    "        ijc                 = mxe_jc_ijc_no_chrY\n",
    "        sjc                 = mxe_jc_sjc_no_chrY\n",
    "        print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    }\n",
    "    # ri\n",
    "    splice_type = \"ri_\"\n",
    "    res = splice_list %in% splice_type\n",
    "    if (sum(res) == 1) {\n",
    "        message (\"splice_list does contain\\n\",\n",
    "             paste(splice_type), \" continuing with processing\\n\")\n",
    "        fromGTF             = ri_fromGTF_no_chrY\n",
    "        ijc                 = ri_jc_ijc_no_chrY\n",
    "        sjc                 = ri_jc_sjc_no_chrY\n",
    "        print_exploratory_plots (plot, \n",
    "                             dup,\n",
    "                             tissue_of_interest, \n",
    "                             splice_type, \n",
    "                             fromGTF, \n",
    "                             tissue_list, \n",
    "                             ijc, \n",
    "                             sjc, \n",
    "                             obj, \n",
    "                             metadata )\n",
    "    }\n",
    "    # se\n",
    "    splice_type = \"se_\"\n",
    "    res = splice_list %in% splice_type\n",
    "    if (sum(res) == 1) {\n",
    "        message (\"splice_list does contain\\n\",\n",
    "             paste(splice_type), \" continuing with processing\\n\")\n",
    "        fromGTF             = se_fromGTF_no_chrY\n",
    "        ijc                 = se_jc_ijc_no_chrY\n",
    "        sjc                 = se_jc_sjc_no_chrY\n",
    "        print_exploratory_plots (plot, \n",
    "                              dup,\n",
    "                              tissue_of_interest, \n",
    "                              splice_type, \n",
    "                              fromGTF, \n",
    "                              tissue_list, \n",
    "                              ijc, \n",
    "                              sjc, \n",
    "                              obj, \n",
    "                              metadata )\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"AllTissueAllSplicingJunctionAnalysis\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the artefacts in the `..data/` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -type f -exec sha256sum {} \\\\;  >  ../metadata/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\")\n",
    "\n",
    "data.table::fread(paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\"), header = FALSE, col.names = c(\"sha256sum\", \"file\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../metadata/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../metadata/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
