{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysis Notebook - Count Genes and Events\n",
    "\n",
    "This notebook processes the raw counts as provided by rMATS 3.2.5 with modifications because this rMATS was run at scale with the entirety of GTEx.   To so, please refer to the methods in the paper, but in brief, each sample was run against another.  This method of running rMATS 3.2.5, permitted us leveraging the counting of the reads done by rMATS together with the scrutinization of rMATS against the provided GTF.   In this case, gencode's complete annotated gtf was scanned by rMATS 3.2.5, the specific possible alternative splicing events catalogued and used in the `fromGTF` descriptions which capture the specific genomic locations for each of five alternative splicing types.  Then the short read RNA sequencing files collected by GTEx and annotated by sex and tissue were aligned and counted against the specific junctions as described by these fromGTF files.   It is these counts that were used in the analyses for this paper.  \n",
    "\n",
    "This notebook, counts the genes and events and performs several descriptive statistical analysis. It  produces outputs needed for downstream visualizations, which are in separate notebooks for added clarity.\n",
    "\n",
    "## 1 Introduction\n",
    "\n",
    "## 1.1 Data files needed by this notebook\n",
    "\n",
    "Input files have been pulled from results, the data were primarily obtained through the execution of the Nextflow workflow, rmats-nf, found in the GitHub repository, https://github.com/lifebit-ai/rmats-nf or the notebook.   How the data were generated are noted for each of the files.\n",
    "\n",
    "1. **`*`.model_B_sex_as_event.csv** One for each tissue, generated by running the notebook `differentialSplicingJunctionAnalysis.ipynb` as a Nextflow workflow\n",
    "2. **`*`.model_B_sex_as_event_refined.csv** One for each tissue, these are the statistically significant results FC > 1.5 and p-value < 0.05 generated by running the notebook `differentialSplicingJunctionAnalysis.ipynb` as a Nextflow workflow.\n",
    "3. **fromGTF.`*`.txt** One for each splicing type, generated within rmats-nf, a Nextflow workflow found here `https://github.com/lifebit-ai/rmats-nf` (fromGTF.A3SS.txt, fromGTF.A5SS.txt, fromGTF.MXE.txt, fromGTF.RI.txt,fromGTF.SE.txt)\n",
    "4. **`*`.DGE.csv** One for each tissue, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "5. **`*`.DGE_refined.csv** One for each tissue, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "6. **srr_pdata.csv** Sequence Run (SRR) data merged with phenotype data, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "7. **rmats_final.`*`.jc.ijc.txt.gz** One for each splicing type, this is a matrix of all included junction (ijc) counts for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`. \n",
    "8. **rmats_final.`*`.jc.inclen.txt.gz** One for each splicing type, this is a matrix of inclusion length (inclen) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "9. **rmats_final.`*`.jc.inclen.txt.gz** One for each splicing type, this is a matrix of inclen for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "10. **rmats_final.`*`.jc.inc.txt.gz** One for each splicing type, this is a matrix of percent spliced in (inc) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "11. **rmats_final.`*`.jc.sjc.txt.gz** One for each splicing type, this is a matrix of skipped junction counts (sjc) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "12. **rmats_final.`*`.jc.skiplen.txt.gz** One for each splicing type, this is a matrix of skipped length (skiplen) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`\n",
    "\n",
    "Two files needed from the `sbas/assets` directory\n",
    "\n",
    "13. **tissues.csv** This file curated by each tissue, the number of female and male samples and provides a display name for the graphics.\n",
    "14. **all_gene_dge.tsv** This file is an intersection of all the differentially expressed genes (not just the significant but all) for all the tissues considered in this study generated by a custom script.\n",
    "\n",
    "\n",
    "## 1.2 Data files created by this notebook\n",
    "\n",
    "Output text files are written to the ``sbas/data`` directory (this notebook is in the ``sbas/jupyter`` directory). \n",
    "\n",
    "1.  **gene_as.tsv**: Significant alternative splicing events per gene, adjusted p-value <= 0.05, fold change >= 1.5.\n",
    "2.  **all_gene_as.tsv**: all alternative splicing events\n",
    "3.  **gene_dge.tsv**: Significant differential gene expression, adjusted p-value <= 0.05, fold change >= 1.5\n",
    "4.  **all_gene_dge.tsv**: all differential gene event files is NOT produced here but is found in the assets directory ../assets/all_gene_dge.tsv\n",
    "5.  **genesWithCommonAS.tsv**: genes (as geneSymbol, the number of splicing events, and the number of tissues the event occurs in)\n",
    "6.  **Total_AS_by_chr.tsv**: Total alternative splicing events per chromosome\n",
    "7.  **Total_AS_by_geneSymbol.tsv**: Count the number of tissues in which specific genes show significant alternative splicing\n",
    "8.  **DGE_by_geneSymbol.tsv***: Most highly expressed genes by tissue\n",
    "9.  **Total_AS_by_tissue.tsv**: Count the number of significant splicing events per tissue\n",
    "10. **Total_AS_by_splicingtype.tsv**: Count number of significant splicing events for each of the 5' alternative splicing categories\n",
    "11. **SplicingIndex_chr.tsv**: Splicing index by chr (number of sigificant AS events per 1000 exons)\n",
    "\n",
    "## 2 Run Analyses\n",
    "\n",
    "## 2.1 Load Libraries\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#conda install r-dplyr bioconductor-biobase r-tibble r-r.utils bioconductor-rtracklayer -y\n",
    "start_time <- Sys.time()\n",
    "suppressMessages({\n",
    "    options(warn = -1) \n",
    "    library(dplyr)\n",
    "    library(Biobase)\n",
    "    library(tibble)\n",
    "    library(R.utils)\n",
    "    library(rtracklayer)\n",
    "})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.2 Read in Curated Tissue List\n",
    "\n",
    "Read in a curated list of tissues, noting sample counts per each sex, male and female and also a curated tissue name for display.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Adipose (sc)'</li><li>'Adipose (v)'</li><li>'Adrenal gland'</li><li>'Aorta'</li><li>'Atrial appendage'</li><li>'Breast'</li><li>'Caudate'</li><li>'Cerebellar hemisphere'</li><li>'Cerebellum'</li><li>'Coronary artery'</li><li>'Cortex'</li><li>'EBV-lymphocytes'</li><li>'Esophagus (gej)'</li><li>'Esophagus (m)'</li><li>'Esophagus (mu)'</li><li>'Fibroblasts'</li><li>'Frontal cortex'</li><li>'Hippocampus'</li><li>'Hypothalamus'</li><li>'Left ventricle'</li><li>'Liver'</li><li>'Lung'</li><li>'Nucleus accumbens'</li><li>'Pancreas'</li><li>'Pituitary'</li><li>'Putamen'</li><li>'Sigmoid colon'</li><li>'Skeletal muscle'</li><li>'Skin (exposed)'</li><li>'Skin (not exposed)'</li><li>'Small intestine'</li><li>'Spinal cord'</li><li>'Spleen'</li><li>'Stomach'</li><li>'Thyroid'</li><li>'Tibial artery'</li><li>'Tibial nerve'</li><li>'Transverse colon'</li><li>'Whole blood'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Adipose (sc)'\n",
       "\\item 'Adipose (v)'\n",
       "\\item 'Adrenal gland'\n",
       "\\item 'Aorta'\n",
       "\\item 'Atrial appendage'\n",
       "\\item 'Breast'\n",
       "\\item 'Caudate'\n",
       "\\item 'Cerebellar hemisphere'\n",
       "\\item 'Cerebellum'\n",
       "\\item 'Coronary artery'\n",
       "\\item 'Cortex'\n",
       "\\item 'EBV-lymphocytes'\n",
       "\\item 'Esophagus (gej)'\n",
       "\\item 'Esophagus (m)'\n",
       "\\item 'Esophagus (mu)'\n",
       "\\item 'Fibroblasts'\n",
       "\\item 'Frontal cortex'\n",
       "\\item 'Hippocampus'\n",
       "\\item 'Hypothalamus'\n",
       "\\item 'Left ventricle'\n",
       "\\item 'Liver'\n",
       "\\item 'Lung'\n",
       "\\item 'Nucleus accumbens'\n",
       "\\item 'Pancreas'\n",
       "\\item 'Pituitary'\n",
       "\\item 'Putamen'\n",
       "\\item 'Sigmoid colon'\n",
       "\\item 'Skeletal muscle'\n",
       "\\item 'Skin (exposed)'\n",
       "\\item 'Skin (not exposed)'\n",
       "\\item 'Small intestine'\n",
       "\\item 'Spinal cord'\n",
       "\\item 'Spleen'\n",
       "\\item 'Stomach'\n",
       "\\item 'Thyroid'\n",
       "\\item 'Tibial artery'\n",
       "\\item 'Tibial nerve'\n",
       "\\item 'Transverse colon'\n",
       "\\item 'Whole blood'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Adipose (sc)'\n",
       "2. 'Adipose (v)'\n",
       "3. 'Adrenal gland'\n",
       "4. 'Aorta'\n",
       "5. 'Atrial appendage'\n",
       "6. 'Breast'\n",
       "7. 'Caudate'\n",
       "8. 'Cerebellar hemisphere'\n",
       "9. 'Cerebellum'\n",
       "10. 'Coronary artery'\n",
       "11. 'Cortex'\n",
       "12. 'EBV-lymphocytes'\n",
       "13. 'Esophagus (gej)'\n",
       "14. 'Esophagus (m)'\n",
       "15. 'Esophagus (mu)'\n",
       "16. 'Fibroblasts'\n",
       "17. 'Frontal cortex'\n",
       "18. 'Hippocampus'\n",
       "19. 'Hypothalamus'\n",
       "20. 'Left ventricle'\n",
       "21. 'Liver'\n",
       "22. 'Lung'\n",
       "23. 'Nucleus accumbens'\n",
       "24. 'Pancreas'\n",
       "25. 'Pituitary'\n",
       "26. 'Putamen'\n",
       "27. 'Sigmoid colon'\n",
       "28. 'Skeletal muscle'\n",
       "29. 'Skin (exposed)'\n",
       "30. 'Skin (not exposed)'\n",
       "31. 'Small intestine'\n",
       "32. 'Spinal cord'\n",
       "33. 'Spleen'\n",
       "34. 'Stomach'\n",
       "35. 'Thyroid'\n",
       "36. 'Tibial artery'\n",
       "37. 'Tibial nerve'\n",
       "38. 'Transverse colon'\n",
       "39. 'Whole blood'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Adipose (sc)\"          \"Adipose (v)\"           \"Adrenal gland\"        \n",
       " [4] \"Aorta\"                 \"Atrial appendage\"      \"Breast\"               \n",
       " [7] \"Caudate\"               \"Cerebellar hemisphere\" \"Cerebellum\"           \n",
       "[10] \"Coronary artery\"       \"Cortex\"                \"EBV-lymphocytes\"      \n",
       "[13] \"Esophagus (gej)\"       \"Esophagus (m)\"         \"Esophagus (mu)\"       \n",
       "[16] \"Fibroblasts\"           \"Frontal cortex\"        \"Hippocampus\"          \n",
       "[19] \"Hypothalamus\"          \"Left ventricle\"        \"Liver\"                \n",
       "[22] \"Lung\"                  \"Nucleus accumbens\"     \"Pancreas\"             \n",
       "[25] \"Pituitary\"             \"Putamen\"               \"Sigmoid colon\"        \n",
       "[28] \"Skeletal muscle\"       \"Skin (exposed)\"        \"Skin (not exposed)\"   \n",
       "[31] \"Small intestine\"       \"Spinal cord\"           \"Spleen\"               \n",
       "[34] \"Stomach\"               \"Thyroid\"               \"Tibial artery\"        \n",
       "[37] \"Tibial nerve\"          \"Transverse colon\"      \"Whole blood\"          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted 39 different tissues with at least 50 samples in both M & f\n",
      "\n"
     ]
    }
   ],
   "source": [
    "tissue_reduction_filename <- \"../assets/tissues.tsv\"\n",
    "tissue_reduction <- read.table(tissue_reduction_filename, header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$display_name != \"n/a\",]\n",
    "tissue_reduction$display_name <- factor(tissue_reduction$display_name)\n",
    "levels(tissue_reduction$display_name)\n",
    "message(\"We extracted \", length(levels(tissue_reduction$display_name)),\" different tissues with at least 50 samples in both M & f\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.3 Read in refined differential AS events\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted 195 model_B_sex_as_events_refined.csv files\n",
      "\n"
     ]
    }
   ],
   "source": [
    "significant_results_dir = \"../data/\"\n",
    "pattern = \"model_B_sex_as_events_refined.csv\"\n",
    "files <- list.files(path = significant_results_dir, pattern = pattern)\n",
    "as_types <- c(\"a3ss\", \"a5ss\", \"mxe\", \"ri\", \"se\")\n",
    "message(\"We extracted \", length(files), \" model_B_sex_as_events_refined.csv files\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.4 Read in the AS Events annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "a3ss_annot <- read.table(file = \"../data/fromGTF.A3SS.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "a5ss_annot <- read.table(file = \"../data/fromGTF.A5SS.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "mxe_annot <- read.table(file = \"../data/fromGTF.MXE.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "ri_annot <- read.table(file = \"../data/fromGTF.RI.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "se_annot <- read.table(file = \"../data/fromGTF.SE.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>GeneID</th><th scope=col>geneSymbol</th><th scope=col>chr</th><th scope=col>strand</th><th scope=col>exonStart_0base</th><th scope=col>exonEnd</th><th scope=col>upstreamES</th><th scope=col>upstreamEE</th><th scope=col>downstreamES</th><th scope=col>downstreamEE</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21287990</td><td>21288091</td><td>21284709</td><td>21284969</td><td>21295674</td><td>21295769</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21303182</td><td>21303234</td><td>21302142</td><td>21302259</td><td>21304425</td><td>21304553</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21287990</td><td>21288091</td><td>21296085</td><td>21296143</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21287990</td><td>21288091</td><td>21298412</td><td>21298479</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21284710</td><td>21284969</td><td>21296085</td><td>21296143</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21284710</td><td>21284969</td><td>21298412</td><td>21298479</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID & GeneID & geneSymbol & chr & strand & exonStart\\_0base & exonEnd & upstreamES & upstreamEE & downstreamES & downstreamEE\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21287990 & 21288091 & 21284709 & 21284969 & 21295674 & 21295769\\\\\n",
       "\t2 & 2 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21303182 & 21303234 & 21302142 & 21302259 & 21304425 & 21304553\\\\\n",
       "\t3 & 3 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21287990 & 21288091 & 21296085 & 21296143\\\\\n",
       "\t4 & 4 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21287990 & 21288091 & 21298412 & 21298479\\\\\n",
       "\t5 & 5 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21284710 & 21284969 & 21296085 & 21296143\\\\\n",
       "\t6 & 6 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21284710 & 21284969 & 21298412 & 21298479\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | ID &lt;int&gt; | GeneID &lt;chr&gt; | geneSymbol &lt;chr&gt; | chr &lt;chr&gt; | strand &lt;chr&gt; | exonStart_0base &lt;int&gt; | exonEnd &lt;int&gt; | upstreamES &lt;int&gt; | upstreamEE &lt;int&gt; | downstreamES &lt;int&gt; | downstreamEE &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21287990 | 21288091 | 21284709 | 21284969 | 21295674 | 21295769 |\n",
       "| 2 | 2 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21303182 | 21303234 | 21302142 | 21302259 | 21304425 | 21304553 |\n",
       "| 3 | 3 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21287990 | 21288091 | 21296085 | 21296143 |\n",
       "| 4 | 4 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21287990 | 21288091 | 21298412 | 21298479 |\n",
       "| 5 | 5 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21284710 | 21284969 | 21296085 | 21296143 |\n",
       "| 6 | 6 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21284710 | 21284969 | 21298412 | 21298479 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID GeneID             geneSymbol chr   strand exonStart_0base exonEnd \n",
       "1 1  ENSG00000034152.18 MAP2K3     chr17 +      21287990        21288091\n",
       "2 2  ENSG00000034152.18 MAP2K3     chr17 +      21303182        21303234\n",
       "3 3  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "4 4  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "5 5  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "6 6  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "  upstreamES upstreamEE downstreamES downstreamEE\n",
       "1 21284709   21284969   21295674     21295769    \n",
       "2 21302142   21302259   21304425     21304553    \n",
       "3 21287990   21288091   21296085     21296143    \n",
       "4 21287990   21288091   21298412     21298479    \n",
       "5 21284710   21284969   21296085     21296143    \n",
       "6 21284710   21284969   21298412     21298479    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(se_annot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.5 create_as_structure \n",
    "\n",
    "This function doees an aggregation of the alternative splicing events - good for all events and the significantly expressed events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "create_as_structure <- function ( results_dir, files, all_or_das, pattern, tissue_reduction) {\n",
    "    gene_as = data.frame()\n",
    "    counts <- rep(NA, length(files))\n",
    "    message(\"\\nnumber of files:\", paste(length(files)), collapse = \"\")\n",
    "    for (i in 1:length(files)) {\n",
    "       lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "       if (dim(lines)[1] > 0) {\n",
    "           event     <- as.vector(as.character(rownames(lines)))\n",
    "           tissue1   <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "           counts[i] <- dim(lines)[1]\n",
    "           event_idx <- substring(event, regexpr(\"[0-9]+$\", event))\n",
    "           res       <- data.frame()\n",
    "           if (grepl(\"^a3ss_\", files[i])) {\n",
    "               # remove the first 5 letters of the string \n",
    "               tissue2 <- substring(tissue1,6)\n",
    "               idx <- match(event_idx, a3ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"A3SS\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- a3ss_annot$geneSymbol[idx],\n",
    "                              GeneID       <- a3ss_annot$GeneID[idx],\n",
    "                              chr          <- a3ss_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "            \n",
    "           } else if (grepl(\"^a5ss_\", files[i])) {\n",
    "               # remove the first 5 letters of the string \n",
    "               tissue2 <- substring(tissue1,6)\n",
    "               idx <- match(event_idx, a5ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"A5SS\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- a5ss_annot$geneSymbol[idx],\n",
    "                              GeneID       <- a5ss_annot$GeneID[idx],\n",
    "                              chr          <- a5ss_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                               \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^mxe_\", files[i])) {\n",
    "               # remove the first 4 letters of the string \n",
    "               tissue2 <- substring(tissue1,5)\n",
    "               idx <- match(event_idx, a3ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"MXE\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- mxe_annot$geneSymbol[idx],\n",
    "                              GeneID       <- mxe_annot$GeneID[idx],\n",
    "                              chr          <- mxe_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^se_\", files[i])) {\n",
    "               # remove the first 3 letters of the string \n",
    "               tissue2 <- substring(tissue1,4)\n",
    "               idx <- match(event_idx, se_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"SE\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- se_annot$geneSymbol[idx],\n",
    "                              GeneID       <- se_annot$GeneID[idx],\n",
    "                              chr          <- se_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^ri_\", files[i])){\n",
    "               # remove the first 3 letters of the string \n",
    "               tissue2 <- substring(tissue1,4)\n",
    "               idx <- match(event_idx, ri_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"RI\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- ri_annot$geneSymbol[idx],\n",
    "                              GeneID       <- ri_annot$GeneID[idx],\n",
    "                              chr          <- ri_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           }\n",
    "        \n",
    "       } #if has sig. events\n",
    "    \n",
    "   } #for all files\n",
    "   colnames(gene_as) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\"GeneSymbol\",\n",
    "                           \"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "   n_unique_genes <- length(summary(as.factor(gene_as$GeneSymbol),maxsum=50000))\n",
    "   message(\"For the run for \", all_or_das, \" run\")\n",
    "   message(\"We extracted a total of \",nrow(gene_as),\" alternative splicing events (gene_as)\")\n",
    "   message(\"This includes \", n_unique_genes, \" total genes\")\n",
    "   return (gene_as)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.6 create_dge_structure \n",
    "\n",
    "This function does an aggregation of the differential gene expression events - good for all events and the significantly expressed events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "create_dge_structure <- function ( results_dir, files, all_or_dge, pattern, map_pattern, tissue_reduction) {\n",
    "   gene_dge = data.frame()\n",
    "   counts <- rep(NA, length(files))\n",
    "   for (i in 1:length(files)) {\n",
    "      lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "      if (dim(lines)[1] > 0) {\n",
    "         tissue1    <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "         map_lines  <- read.table(file=paste0(paste0(results_dir, tissue1),map_pattern),\n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "         counts[i]  <- dim(lines)[1]    \n",
    "         tissue1    <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "         map_lines  <- read.table(file=paste0(paste0(results_dir, tissue1),map_pattern),\n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "         ensg_ver   <- as.vector(as.character(rownames(lines)))\n",
    "         ensg_no_ver<- as.vector(as.character(map_lines$ensg_names))\n",
    "         ensg_genes <- as.vector(as.character(map_lines$ensg_genes))\n",
    "         counts[i]  <- dim(lines)[1]  \n",
    "         res <- data.frame(Tissue       <- tissue1,\n",
    "                           ENSG_ver     <- ensg_ver,\n",
    "                           ENSG_no_ver  <- ensg_no_ver,\n",
    "                           GeneSymbol   <- ensg_genes,\n",
    "                           counts       <- counts[i],\n",
    "                           Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue1, \"display_name\"],\n",
    "                           logFC        <- lines$logFC,\n",
    "                           AveExpr      <- lines$AveExpr,\n",
    "                           t            <- lines$t,\n",
    "                           PValue       <- lines$P.Value,\n",
    "                           AdjPVal      <- lines$adj.P.Val,\n",
    "                           B            <- lines$B)\n",
    "         colnames(res) <- c(\"Tissue\",\"ENSG_ver\",\"ENSG_no_ver\",\"GeneSymbol\",\"counts\",\"Display\",\n",
    "                            \"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "         gene_dge <- rbind(gene_dge, res)\n",
    "       } #if has sig. events\n",
    "    } #for all files\n",
    "    colnames(gene_dge) <- c(\"Tissue\",\"ENSG_ver\",\"ENSG_no_ver\",\"GeneSymbol\",\"counts\",\"Display\",\n",
    "                        \"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "    n_unique_genes <- length(summary(as.factor(gene_dge$GeneSymbol),maxsum=50000))\n",
    "    message(\"For the run for \", all_or_dge, \"run\")\n",
    "    message(\"We extracted a total of \",nrow(gene_dge),\" gene events (gene_dge)\")\n",
    "    message(\"This includes \", n_unique_genes, \" total genes\")\n",
    "    return(gene_dge)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.7 Read in the alternative splicing results\n",
    "\n",
    "We will create an aggregation of  all the results and all the significant results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Length of all_files: 195\n",
      "\n",
      "Length of significant_files: 195\n",
      "\n",
      "\n",
      "number of files:195\n",
      "\n",
      "For the run for differentially significant alternative splicing run\n",
      "\n",
      "We extracted a total of 1278 alternative splicing events (gene_as)\n",
      "\n",
      "This includes 888 total genes\n",
      "\n",
      "\n",
      "number of files:195\n",
      "\n",
      "For the run for all alternatively spliced run\n",
      "\n",
      "We extracted a total of 1140938 alternative splicing events (gene_as)\n",
      "\n",
      "This includes 12681 total genes\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneJunction</th><th scope=col>ASE</th><th scope=col>ASE_IDX</th><th scope=col>Tissue</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>GeneSymbol</th><th scope=col>GeneID</th><th scope=col>chr</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SCO1-8452</td><td>A3SS</td><td>8452</td><td>adrenal_gland  </td><td>1</td><td>Adrenal gland  </td><td>SCO1</td><td>ENSG00000133028.12</td><td>chr17</td><td> 0.7715834</td><td>5.028635</td><td>  4.906848</td><td>1.533202e-06</td><td>8.170436e-03</td><td> 0.8561371</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>XIST-2252</td><td>A3SS</td><td>2252</td><td>artery_coronary</td><td>1</td><td>Coronary artery</td><td>XIST</td><td>ENSG00000229807.11</td><td>chrX </td><td>-2.1880965</td><td>4.327079</td><td>-10.329346</td><td>4.631685e-21</td><td>2.464057e-17</td><td>16.7535120</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & GeneJunction & ASE & ASE\\_IDX & Tissue & counts & Display & GeneSymbol & GeneID & chr & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <fct> & <fct> & <int> & <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & SCO1-8452 & A3SS & 8452 & adrenal\\_gland   & 1 & Adrenal gland   & SCO1 & ENSG00000133028.12 & chr17 &  0.7715834 & 5.028635 &   4.906848 & 1.533202e-06 & 8.170436e-03 &  0.8561371\\\\\n",
       "\t2 & XIST-2252 & A3SS & 2252 & artery\\_coronary & 1 & Coronary artery & XIST & ENSG00000229807.11 & chrX  & -2.1880965 & 4.327079 & -10.329346 & 4.631685e-21 & 2.464057e-17 & 16.7535120\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 15\n",
       "\n",
       "| <!--/--> | GeneJunction &lt;fct&gt; | ASE &lt;fct&gt; | ASE_IDX &lt;int&gt; | Tissue &lt;fct&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | GeneSymbol &lt;fct&gt; | GeneID &lt;fct&gt; | chr &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | SCO1-8452 | A3SS | 8452 | adrenal_gland   | 1 | Adrenal gland   | SCO1 | ENSG00000133028.12 | chr17 |  0.7715834 | 5.028635 |   4.906848 | 1.533202e-06 | 8.170436e-03 |  0.8561371 |\n",
       "| 2 | XIST-2252 | A3SS | 2252 | artery_coronary | 1 | Coronary artery | XIST | ENSG00000229807.11 | chrX  | -2.1880965 | 4.327079 | -10.329346 | 4.631685e-21 | 2.464057e-17 | 16.7535120 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneJunction ASE  ASE_IDX Tissue          counts Display         GeneSymbol\n",
       "1 SCO1-8452    A3SS 8452    adrenal_gland   1      Adrenal gland   SCO1      \n",
       "2 XIST-2252    A3SS 2252    artery_coronary 1      Coronary artery XIST      \n",
       "  GeneID             chr   logFC      AveExpr  t          PValue      \n",
       "1 ENSG00000133028.12 chr17  0.7715834 5.028635   4.906848 1.533202e-06\n",
       "2 ENSG00000229807.11 chrX  -2.1880965 4.327079 -10.329346 4.631685e-21\n",
       "  AdjPVal      B         \n",
       "1 8.170436e-03  0.8561371\n",
       "2 2.464057e-17 16.7535120"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "results_dir         <- \"../data/\"\n",
    "significant_pattern <- \"_AS_model_B_sex_as_events_refined.csv\"\n",
    "significant_files   <- list.files(path = results_dir, pattern = significant_pattern)\n",
    "all_pattern         <- \"_AS_model_B_sex_as_events.csv\"\n",
    "all_files           <- list.files(path = results_dir, pattern = all_pattern)\n",
    "as_types            <- c(\"a3ss\", \"a5ss\", \"mxe\", \"ri\", \"se\")\n",
    "message(\"Length of all_files: \", length(all_files))\n",
    "message(\"Length of significant_files: \", length(significant_files))\n",
    "\n",
    "gene_as     <- create_as_structure (results_dir      <- results_dir, \n",
    "                                    files            <- significant_files,\n",
    "                                    all_or_das       <- \"differentially significant alternative splicing\",\n",
    "                                    pattern          <- significant_pattern, \n",
    "                                    tissue_reduction <- tissue_reduction)\n",
    "all_gene_as <- create_as_structure (results_dir      <- results_dir, \n",
    "                                    files            <- all_files, \n",
    "                                    all_or_das       <- \"all alternatively spliced\",\n",
    "                                    pattern          <- all_pattern, \n",
    "                                    tissue_reduction <- tissue_reduction)\n",
    "head(gene_as,2)\n",
    "gene_as$Tissue <- factor(gene_as$Tissue)\n",
    "write.table(gene_as, \"../data/gene_as.tsv\", quote=FALSE, sep=\"\\t\")\n",
    "write.table(all_gene_as, \"../data/all_gene_as.tsv\", quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.8 Create a genes-id file capturing the unique gene-junction locations in a single file\n",
    "\n",
    "rMATS 3.2.5 unique junction ids by splicing event tied together with gene names and these identifiers useful for downstream analyses and investigations.\n",
    "only the SE significant AS events here\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The first file from ^se_*AS_model_B_sex_as_events_refined.csv: se_adipose_subcutaneous_AS_model_B_sex_as_events_refined.csv\n",
      "\n",
      "Done writing 39 files.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results_dir         <- \"../data/\"\n",
    "significant_pattern <- \"^se_*AS_model_B_sex_as_events_refined.csv\"\n",
    "files   <- list.files(path =results_dir, pattern = glob2rx(significant_pattern))\n",
    "message(\"The first file from ^se_*AS_model_B_sex_as_events_refined.csv: \", files[1])\n",
    "pattern=\"_AS_model_B_sex_as_events_refined.csv\"\n",
    "geneids <- data.frame()\n",
    "for (i in 1:length(files)) {\n",
    "    lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "    \n",
    "    if (dim(lines)[1] > 0) {\n",
    "           event     <- as.vector(as.character(rownames(lines)))\n",
    "           tissue1   <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "           event_idx <- substring(event, regexpr(\"[0-9]+$\", event))\n",
    "           res       <- data.frame()\n",
    "           tissue2 <- substring(tissue1,4)\n",
    "           idx <- match(event_idx, se_annot$ID)\n",
    "           res <- data.frame(geneIDs      <- event,\n",
    "                             ID           <- event_idx,\n",
    "                             GeneSymbol   <- se_annot$geneSymbol[idx],\n",
    "                             GeneID       <- se_annot$GeneID[idx],\n",
    "                             chr          <- se_annot$chr[idx])\n",
    "           outfilename <- paste0(paste0(\"../data/se_\",tissue2),\"_geneids.tsv\")\n",
    "           write.table(res, outfilename, quote=FALSE, sep=\"\\t\")\n",
    "           \n",
    "     }\n",
    "}\n",
    "message(\"Done writing \", length(files), \" files.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.9 Read in the differential gene expression results\n",
    "\n",
    "Here we create an aggregation of al the significant results differential gene expression events.\n",
    "Note the all_gene_dge.tsv may be found in the assets directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "195"
      ],
      "text/latex": [
       "195"
      ],
      "text/markdown": [
       "195"
      ],
      "text/plain": [
       "[1] 195"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for differential gene expressionrun\n",
      "\n",
      "We extracted a total of 4417 gene events (gene_dge)\n",
      "\n",
      "This includes 3221 total genes\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Tissue</th><th scope=col>ENSG_ver</th><th scope=col>ENSG_no_ver</th><th scope=col>GeneSymbol</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>adipose_subcutaneous</td><td>ENSG00000147050.14</td><td>ENSG00000147050</td><td>KDM6A     </td><td>272</td><td>Adipose (sc)</td><td>0.6119682</td><td> 5.030862</td><td>35.53259</td><td>9.295961e-156</td><td>1.498695e-151</td><td>344.27452</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>adipose_subcutaneous</td><td>ENSG00000224525.2 </td><td>ENSG00000224525</td><td>AL591686.1</td><td>272</td><td>Adipose (sc)</td><td>2.2810767</td><td>-1.464120</td><td>14.44699</td><td> 2.450714e-41</td><td> 3.591855e-38</td><td> 81.05687</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & Tissue & ENSG\\_ver & ENSG\\_no\\_ver & GeneSymbol & counts & Display & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & adipose\\_subcutaneous & ENSG00000147050.14 & ENSG00000147050 & KDM6A      & 272 & Adipose (sc) & 0.6119682 &  5.030862 & 35.53259 & 9.295961e-156 & 1.498695e-151 & 344.27452\\\\\n",
       "\t2 & adipose\\_subcutaneous & ENSG00000224525.2  & ENSG00000224525 & AL591686.1 & 272 & Adipose (sc) & 2.2810767 & -1.464120 & 14.44699 &  2.450714e-41 &  3.591855e-38 &  81.05687\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 12\n",
       "\n",
       "| <!--/--> | Tissue &lt;fct&gt; | ENSG_ver &lt;fct&gt; | ENSG_no_ver &lt;fct&gt; | GeneSymbol &lt;fct&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | adipose_subcutaneous | ENSG00000147050.14 | ENSG00000147050 | KDM6A      | 272 | Adipose (sc) | 0.6119682 |  5.030862 | 35.53259 | 9.295961e-156 | 1.498695e-151 | 344.27452 |\n",
       "| 2 | adipose_subcutaneous | ENSG00000224525.2  | ENSG00000224525 | AL591686.1 | 272 | Adipose (sc) | 2.2810767 | -1.464120 | 14.44699 |  2.450714e-41 |  3.591855e-38 |  81.05687 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue               ENSG_ver           ENSG_no_ver     GeneSymbol counts\n",
       "1 adipose_subcutaneous ENSG00000147050.14 ENSG00000147050 KDM6A      272   \n",
       "2 adipose_subcutaneous ENSG00000224525.2  ENSG00000224525 AL591686.1 272   \n",
       "  Display      logFC     AveExpr   t        PValue        AdjPVal      \n",
       "1 Adipose (sc) 0.6119682  5.030862 35.53259 9.295961e-156 1.498695e-151\n",
       "2 Adipose (sc) 2.2810767 -1.464120 14.44699  2.450714e-41  3.591855e-38\n",
       "  B        \n",
       "1 344.27452\n",
       "2  81.05687"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "results_dir             <- \"../data/\"\n",
    "significant_dge_pattern <- \"_DGE_refined.csv\"\n",
    "significant_dge_files   <- list.files(path = results_dir, pattern = significant_dge_pattern)\n",
    "map_pattern             <- \"_DGE_ensg_map.csv\"\n",
    "length(significant_files)\n",
    "\n",
    "gene_dge     <- create_dge_structure (results_dir      <- results_dir, \n",
    "                                      files            <- significant_dge_files, \n",
    "                                      all_or_dge       <- \"differential gene expression\",\n",
    "                                      pattern          <- significant_dge_pattern, \n",
    "                                      map_pattern      <- map_pattern,\n",
    "                                      tissue_reduction <- tissue_reduction)\n",
    "\n",
    "head(gene_dge,2)\n",
    "gene_dge$Tissue <- factor(gene_dge$Tissue)\n",
    "write.table(gene_dge,     \"../data/gene_dge.tsv\",     quote=FALSE, sep=\"\\t\") "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.10 Read in Gencode (v30) Complete Annotation file.\n",
    "\n",
    "Load in the gencode.v30.annotation.gtf file for additional annotation.\n",
    "chr information for summary data later, use the annotation we used for rMATS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "downloading gencode v30 annotation\n",
      "\n",
      "\n",
      "Done!\n",
      "\n",
      "\n",
      "Unzipping compressed file gencode.v30.annotation.gtf.gz..\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)\n",
       "attr(,\"status\")\n",
       "[1] 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done! gencode.v30.annotation.gtf can be found in ../data/\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>GeneSymbol</th><th scope=col>ENSG</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>DDX11L1    </td><td>ENSG00000223972.5</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>WASH7P     </td><td>ENSG00000227232.5</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>MIR6859-1  </td><td>ENSG00000278267.1</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>chr1</td><td>MIR1302-2HG</td><td>ENSG00000243485.5</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>chr1</td><td>MIR1302-2  </td><td>ENSG00000284332.1</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>chr1</td><td>FAM138A    </td><td>ENSG00000237613.2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & GeneSymbol & ENSG\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & DDX11L1     & ENSG00000223972.5\\\\\n",
       "\t13 & chr1 & WASH7P      & ENSG00000227232.5\\\\\n",
       "\t26 & chr1 & MIR6859-1   & ENSG00000278267.1\\\\\n",
       "\t29 & chr1 & MIR1302-2HG & ENSG00000243485.5\\\\\n",
       "\t37 & chr1 & MIR1302-2   & ENSG00000284332.1\\\\\n",
       "\t40 & chr1 & FAM138A     & ENSG00000237613.2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | chr &lt;fct&gt; | GeneSymbol &lt;chr&gt; | ENSG &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chr1 | DDX11L1     | ENSG00000223972.5 |\n",
       "| 13 | chr1 | WASH7P      | ENSG00000227232.5 |\n",
       "| 26 | chr1 | MIR6859-1   | ENSG00000278267.1 |\n",
       "| 29 | chr1 | MIR1302-2HG | ENSG00000243485.5 |\n",
       "| 37 | chr1 | MIR1302-2   | ENSG00000284332.1 |\n",
       "| 40 | chr1 | FAM138A     | ENSG00000237613.2 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  GeneSymbol  ENSG             \n",
       "1  chr1 DDX11L1     ENSG00000223972.5\n",
       "13 chr1 WASH7P      ENSG00000227232.5\n",
       "26 chr1 MIR6859-1   ENSG00000278267.1\n",
       "29 chr1 MIR1302-2HG ENSG00000243485.5\n",
       "37 chr1 MIR1302-2   ENSG00000284332.1\n",
       "40 chr1 FAM138A     ENSG00000237613.2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "message(\"downloading gencode v30 annotation\\n\")\n",
    "system(\"wget -O ../data/gencode.v30.annotation.gtf.gz ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gtf.gz\")\n",
    "message(\"Done!\\n\")\n",
    "message(\"Unzipping compressed file gencode.v30.annotation.gtf.gz..\")\n",
    "system(\"gunzip ../data/gencode.v30.annotation.gtf.gz\", intern = TRUE)\n",
    "message(\"Done! gencode.v30.annotation.gtf can be found in ../data/\")\n",
    "\n",
    "gencode <- import(\"../data/gencode.v30.annotation.gtf\")\n",
    "gtf.df <- as.data.frame (gencode)\n",
    "chr_genes <- unique(gtf.df[,c(\"seqnames\",\"gene_name\",\"gene_id\")])\n",
    "colnames(chr_genes) <- c(\"chr\",\"GeneSymbol\", \"ENSG\")\n",
    "head(chr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>GeneSymbol</th><th scope=col>ENSG</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>DDX11L1    </td><td>ENSG00000223972</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>WASH7P     </td><td>ENSG00000227232</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>MIR6859-1  </td><td>ENSG00000278267</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>chr1</td><td>MIR1302-2HG</td><td>ENSG00000243485</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>chr1</td><td>MIR1302-2  </td><td>ENSG00000284332</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>chr1</td><td>FAM138A    </td><td>ENSG00000237613</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & GeneSymbol & ENSG\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & DDX11L1     & ENSG00000223972\\\\\n",
       "\t13 & chr1 & WASH7P      & ENSG00000227232\\\\\n",
       "\t26 & chr1 & MIR6859-1   & ENSG00000278267\\\\\n",
       "\t29 & chr1 & MIR1302-2HG & ENSG00000243485\\\\\n",
       "\t37 & chr1 & MIR1302-2   & ENSG00000284332\\\\\n",
       "\t40 & chr1 & FAM138A     & ENSG00000237613\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | chr &lt;fct&gt; | GeneSymbol &lt;chr&gt; | ENSG &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chr1 | DDX11L1     | ENSG00000223972 |\n",
       "| 13 | chr1 | WASH7P      | ENSG00000227232 |\n",
       "| 26 | chr1 | MIR6859-1   | ENSG00000278267 |\n",
       "| 29 | chr1 | MIR1302-2HG | ENSG00000243485 |\n",
       "| 37 | chr1 | MIR1302-2   | ENSG00000284332 |\n",
       "| 40 | chr1 | FAM138A     | ENSG00000237613 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  GeneSymbol  ENSG           \n",
       "1  chr1 DDX11L1     ENSG00000223972\n",
       "13 chr1 WASH7P      ENSG00000227232\n",
       "26 chr1 MIR6859-1   ENSG00000278267\n",
       "29 chr1 MIR1302-2HG ENSG00000243485\n",
       "37 chr1 MIR1302-2   ENSG00000284332\n",
       "40 chr1 FAM138A     ENSG00000237613"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for (i in 1:dim(chr_genes)[1]) {\n",
    "    chr_genes$ENSG[i] <- as.character(strsplit(chr_genes$ENSG[i],'\\\\.\\\\w+$'))\n",
    "}\n",
    "head(chr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Tissue</th><th scope=col>ENSG_ver</th><th scope=col>ENSG_no_ver</th><th scope=col>GeneSymbol</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th><th scope=col>chr</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>adipose_subcutaneous</td><td>ENSG00000147050.14</td><td>ENSG00000147050</td><td>KDM6A     </td><td>272</td><td>Adipose (sc)</td><td> 0.6119682</td><td> 5.0308617</td><td> 35.53259</td><td>9.295961e-156</td><td>1.498695e-151</td><td>344.27452</td><td>chrX </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>adipose_subcutaneous</td><td>ENSG00000224525.2 </td><td>ENSG00000224525</td><td>AL591686.1</td><td>272</td><td>Adipose (sc)</td><td> 2.2810767</td><td>-1.4641204</td><td> 14.44699</td><td> 2.450714e-41</td><td> 3.591855e-38</td><td> 81.05687</td><td>chr1 </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>adipose_subcutaneous</td><td>ENSG00000115041.12</td><td>ENSG00000115041</td><td>KCNIP3    </td><td>272</td><td>Adipose (sc)</td><td>-0.9052185</td><td> 3.4532379</td><td>-13.70790</td><td> 7.695915e-38</td><td> 8.862395e-35</td><td> 75.10723</td><td>chr2 </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>adipose_subcutaneous</td><td>ENSG00000258484.3 </td><td>ENSG00000258484</td><td>SPESP1    </td><td>272</td><td>Adipose (sc)</td><td>-0.8918130</td><td> 0.9410518</td><td>-13.10456</td><td> 4.615796e-35</td><td> 4.650991e-32</td><td> 68.23064</td><td>chr15</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>adipose_subcutaneous</td><td>ENSG00000134339.8 </td><td>ENSG00000134339</td><td>SAA2      </td><td>272</td><td>Adipose (sc)</td><td> 3.1082279</td><td> 2.7856727</td><td> 12.87586</td><td> 5.017765e-34</td><td> 4.257705e-31</td><td> 66.45132</td><td>chr11</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>adipose_subcutaneous</td><td>ENSG00000103449.11</td><td>ENSG00000103449</td><td>SALL1     </td><td>272</td><td>Adipose (sc)</td><td> 1.2692252</td><td>-0.2424580</td><td> 12.72143</td><td> 2.470418e-33</td><td> 1.991404e-30</td><td> 64.14054</td><td>chr16</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & Tissue & ENSG\\_ver & ENSG\\_no\\_ver & GeneSymbol & counts & Display & logFC & AveExpr & t & PValue & AdjPVal & B & chr\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & adipose\\_subcutaneous & ENSG00000147050.14 & ENSG00000147050 & KDM6A      & 272 & Adipose (sc) &  0.6119682 &  5.0308617 &  35.53259 & 9.295961e-156 & 1.498695e-151 & 344.27452 & chrX \\\\\n",
       "\t2 & adipose\\_subcutaneous & ENSG00000224525.2  & ENSG00000224525 & AL591686.1 & 272 & Adipose (sc) &  2.2810767 & -1.4641204 &  14.44699 &  2.450714e-41 &  3.591855e-38 &  81.05687 & chr1 \\\\\n",
       "\t3 & adipose\\_subcutaneous & ENSG00000115041.12 & ENSG00000115041 & KCNIP3     & 272 & Adipose (sc) & -0.9052185 &  3.4532379 & -13.70790 &  7.695915e-38 &  8.862395e-35 &  75.10723 & chr2 \\\\\n",
       "\t4 & adipose\\_subcutaneous & ENSG00000258484.3  & ENSG00000258484 & SPESP1     & 272 & Adipose (sc) & -0.8918130 &  0.9410518 & -13.10456 &  4.615796e-35 &  4.650991e-32 &  68.23064 & chr15\\\\\n",
       "\t5 & adipose\\_subcutaneous & ENSG00000134339.8  & ENSG00000134339 & SAA2       & 272 & Adipose (sc) &  3.1082279 &  2.7856727 &  12.87586 &  5.017765e-34 &  4.257705e-31 &  66.45132 & chr11\\\\\n",
       "\t6 & adipose\\_subcutaneous & ENSG00000103449.11 & ENSG00000103449 & SALL1      & 272 & Adipose (sc) &  1.2692252 & -0.2424580 &  12.72143 &  2.470418e-33 &  1.991404e-30 &  64.14054 & chr16\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 13\n",
       "\n",
       "| <!--/--> | Tissue &lt;fct&gt; | ENSG_ver &lt;fct&gt; | ENSG_no_ver &lt;fct&gt; | GeneSymbol &lt;fct&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; | chr &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | adipose_subcutaneous | ENSG00000147050.14 | ENSG00000147050 | KDM6A      | 272 | Adipose (sc) |  0.6119682 |  5.0308617 |  35.53259 | 9.295961e-156 | 1.498695e-151 | 344.27452 | chrX  |\n",
       "| 2 | adipose_subcutaneous | ENSG00000224525.2  | ENSG00000224525 | AL591686.1 | 272 | Adipose (sc) |  2.2810767 | -1.4641204 |  14.44699 |  2.450714e-41 |  3.591855e-38 |  81.05687 | chr1  |\n",
       "| 3 | adipose_subcutaneous | ENSG00000115041.12 | ENSG00000115041 | KCNIP3     | 272 | Adipose (sc) | -0.9052185 |  3.4532379 | -13.70790 |  7.695915e-38 |  8.862395e-35 |  75.10723 | chr2  |\n",
       "| 4 | adipose_subcutaneous | ENSG00000258484.3  | ENSG00000258484 | SPESP1     | 272 | Adipose (sc) | -0.8918130 |  0.9410518 | -13.10456 |  4.615796e-35 |  4.650991e-32 |  68.23064 | chr15 |\n",
       "| 5 | adipose_subcutaneous | ENSG00000134339.8  | ENSG00000134339 | SAA2       | 272 | Adipose (sc) |  3.1082279 |  2.7856727 |  12.87586 |  5.017765e-34 |  4.257705e-31 |  66.45132 | chr11 |\n",
       "| 6 | adipose_subcutaneous | ENSG00000103449.11 | ENSG00000103449 | SALL1      | 272 | Adipose (sc) |  1.2692252 | -0.2424580 |  12.72143 |  2.470418e-33 |  1.991404e-30 |  64.14054 | chr16 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue               ENSG_ver           ENSG_no_ver     GeneSymbol counts\n",
       "1 adipose_subcutaneous ENSG00000147050.14 ENSG00000147050 KDM6A      272   \n",
       "2 adipose_subcutaneous ENSG00000224525.2  ENSG00000224525 AL591686.1 272   \n",
       "3 adipose_subcutaneous ENSG00000115041.12 ENSG00000115041 KCNIP3     272   \n",
       "4 adipose_subcutaneous ENSG00000258484.3  ENSG00000258484 SPESP1     272   \n",
       "5 adipose_subcutaneous ENSG00000134339.8  ENSG00000134339 SAA2       272   \n",
       "6 adipose_subcutaneous ENSG00000103449.11 ENSG00000103449 SALL1      272   \n",
       "  Display      logFC      AveExpr    t         PValue        AdjPVal      \n",
       "1 Adipose (sc)  0.6119682  5.0308617  35.53259 9.295961e-156 1.498695e-151\n",
       "2 Adipose (sc)  2.2810767 -1.4641204  14.44699  2.450714e-41  3.591855e-38\n",
       "3 Adipose (sc) -0.9052185  3.4532379 -13.70790  7.695915e-38  8.862395e-35\n",
       "4 Adipose (sc) -0.8918130  0.9410518 -13.10456  4.615796e-35  4.650991e-32\n",
       "5 Adipose (sc)  3.1082279  2.7856727  12.87586  5.017765e-34  4.257705e-31\n",
       "6 Adipose (sc)  1.2692252 -0.2424580  12.72143  2.470418e-33  1.991404e-30\n",
       "  B         chr  \n",
       "1 344.27452 chrX \n",
       "2  81.05687 chr1 \n",
       "3  75.10723 chr2 \n",
       "4  68.23064 chr15\n",
       "5  66.45132 chr11\n",
       "6  64.14054 chr16"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "i = 1\n",
    "chr <- rep(\"NA\",dim(gene_dge)[1])\n",
    "gene_dge$chr <- chr\n",
    "for (i in 1:dim(gene_dge)[1]) {\n",
    "    match  <- as.character(chr_genes$ENSG) %in% as.character((gene_dge$ENSG_no_ver[i]))\n",
    "    if (sum(match==TRUE)== 1) {\n",
    "        chr[i] <- as.character(chr_genes[match,]$chr)\n",
    "        gene_dge$chr[i] <- chr[i]\n",
    "    } else if (sum(match==TRUE)>1) {\n",
    "        all <- as.vector(as.character(chr_genes[match,]$chr))\n",
    "        gene_dge$chr[i] <- as.character(all[1])\n",
    "    }\n",
    "}\n",
    "head(gene_dge)\n",
    "\n",
    "write.table(gene_dge, \"../data/gene_dge.tsv\", quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.11 Summary AS \n",
    "\n",
    "Capture descriptive statistics of the signficiantly alternatively splicedgene_as and significantly differentially gene expressed, gene_dge regarding events by tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Tissue</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>pancreas                      </td><td>1</td></tr>\n",
       "\t<tr><td>small_intestine_terminal_ileum</td><td>1</td></tr>\n",
       "\t<tr><td>adrenal_gland                 </td><td>2</td></tr>\n",
       "\t<tr><td>artery_coronary               </td><td>2</td></tr>\n",
       "\t<tr><td>brain_cortex                  </td><td>2</td></tr>\n",
       "\t<tr><td>stomach                       </td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Tissue & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t pancreas                       & 1\\\\\n",
       "\t small\\_intestine\\_terminal\\_ileum & 1\\\\\n",
       "\t adrenal\\_gland                  & 2\\\\\n",
       "\t artery\\_coronary                & 2\\\\\n",
       "\t brain\\_cortex                   & 2\\\\\n",
       "\t stomach                        & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Tissue &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| pancreas                       | 1 |\n",
       "| small_intestine_terminal_ileum | 1 |\n",
       "| adrenal_gland                  | 2 |\n",
       "| artery_coronary                | 2 |\n",
       "| brain_cortex                   | 2 |\n",
       "| stomach                        | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue                         n\n",
       "1 pancreas                       1\n",
       "2 small_intestine_terminal_ileum 1\n",
       "3 adrenal_gland                  2\n",
       "4 artery_coronary                2\n",
       "5 brain_cortex                   2\n",
       "6 stomach                        2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum splicing events per tissue 1 maximum splicing events per tissue 996\n",
      "\n",
      "Sum of significant splicing events per tissue less than 10037\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Tissue</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>brain_spinal_cord_cervical_c_1       </td><td> 5</td></tr>\n",
       "\t<tr><td>brain_cortex                         </td><td> 7</td></tr>\n",
       "\t<tr><td>brain_hippocampus                    </td><td> 7</td></tr>\n",
       "\t<tr><td>brain_nucleus_accumbens_basal_ganglia</td><td> 8</td></tr>\n",
       "\t<tr><td>small_intestine_terminal_ileum       </td><td>10</td></tr>\n",
       "\t<tr><td>brain_caudate_basal_ganglia          </td><td>12</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Tissue & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t brain\\_spinal\\_cord\\_cervical\\_c\\_1        &  5\\\\\n",
       "\t brain\\_cortex                          &  7\\\\\n",
       "\t brain\\_hippocampus                     &  7\\\\\n",
       "\t brain\\_nucleus\\_accumbens\\_basal\\_ganglia &  8\\\\\n",
       "\t small\\_intestine\\_terminal\\_ileum        & 10\\\\\n",
       "\t brain\\_caudate\\_basal\\_ganglia           & 12\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Tissue &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| brain_spinal_cord_cervical_c_1        |  5 |\n",
       "| brain_cortex                          |  7 |\n",
       "| brain_hippocampus                     |  7 |\n",
       "| brain_nucleus_accumbens_basal_ganglia |  8 |\n",
       "| small_intestine_terminal_ileum        | 10 |\n",
       "| brain_caudate_basal_ganglia           | 12 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue                                n \n",
       "1 brain_spinal_cord_cervical_c_1         5\n",
       "2 brain_cortex                           7\n",
       "3 brain_hippocampus                      7\n",
       "4 brain_nucleus_accumbens_basal_ganglia  8\n",
       "5 small_intestine_terminal_ileum        10\n",
       "6 brain_caudate_basal_ganglia           12"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum gene expression events per tissue 5 maximum gene expression events per tissue 2296\n",
      "\n"
     ]
    }
   ],
   "source": [
    "XY <- gene_as %>% group_by(Tissue) %>% tally()\n",
    "XY <- XY[order(XY$n),decreasing=TRUE]\n",
    "head(XY)\n",
    "message(\"Minimum splicing events per tissue \", min(XY$n), \" maximum splicing events per tissue \", max(XY$n))\n",
    "message(\"Sum of significant splicing events per tissue less than 100\", sum(XY$n<100))\n",
    "\n",
    "XY <- gene_dge %>% group_by(Tissue) %>% tally()\n",
    "XY <- XY[order(XY$n),decreasing=TRUE]\n",
    "head(XY)\n",
    "message(\"Minimum gene expression events per tissue \", min(XY$n), \" maximum gene expression events per tissue \", max(XY$n))\n",
    "# table(gtf.df[,c(\"gene_type\")])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3 Data Structures for Figures"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.1 gene_as.tsv\n",
    "\n",
    "This file contains (description)\n",
    "Here is a typical line\n",
    "<pre>\n",
    "A data.frame: 6 × 15\n",
    "GeneJunction\tASE\tASE_IDX\tTissue\tcounts\tDisplay\tGeneSymbol\tGeneID\tchr\tlogFC\tAveExpr\tt\tPValue\tAdjPVal\tB\n",
    "<fct>\t<fct>\t<int>\t<fct>\t<int>\t<fct>\t<fct>\t<fct>\t<fct>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\n",
    "1\tXIST-2253\tA3SS\t2253\tadipose_subcutaneous\t4\tAdipose (sc)\tXIST\tENSG00000229807.11\tchrX\t-4.4086049\t3.196317\t-36.488970\t4.635568e-154\t3.893877e-150\t310.016049\n",
    "2\tXIST-2252\tA3SS\t2252\tadipose_subcutaneous\t4\tAdipose (sc)\tXIST\tENSG00000229807.11\tchrX\t-2.4147126\t3.647690\t-21.921057\t1.444102e-78\t6.065229e-75\t160.028167\n",
    "3\tGREB1L-4933\tA3SS\t4933\tadipose_subcutaneous\t4\tAdipose (sc)\tGREB1L\tENSG00000141449.14\tchr18\t1.2793173\t2.115005\t7.123138\t3.052112e-12\t8.545914e-09\t16.692429\n",
    "4\tRHCG-1776\tA3SS\t1776\tadipose_subcutaneous\t4\tAdipose (sc)\tRHCG\tENSG00000140519.14\tchr15\t-0.6930009\t1.636472\t-3.922124\t9.797866e-05\t3.919146e-02\t1.142232\n",
    "5\tXIST-2253\tA3SS\t2253\tadipose_visceral_omentum\t12\tAdipose (v)\tXIST\tENSG00000229807.11\tchrX\t-4.4403352\t3.113532\t-33.950800\t2.654474e-123\t2.209585e-119\t241.826117\n",
    "6\tXIST-2252\tA3SS\t2252\tadipose_visceral_omentum\t12\tAdipose (v)\tXIST\tENSG00000229807.11\tchrX\t-2.4506832\t3.650617\t-18.890779\t2.817671e-58\t1.172715e-54\t114.731682\n",
    "</pre>\n",
    "There are 2887 significant events in the file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 1,278\n",
      "Columns: 15\n",
      "$ GeneJunction \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m SCO1-8452, XIST-2252, CPT1C-6427, DEPDC5-1839, CLN5-8117…\n",
      "$ ASE          \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m A3SS, A3SS, A3SS, A3SS, A3SS, A3SS, A3SS, A3SS, A3SS, A3…\n",
      "$ ASE_IDX      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 8452, 2252, 6427, 1839, 8117, 453, 3523, 7852, 7853, 165…\n",
      "$ Tissue       \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m adrenal_gland, artery_coronary, brain_cortex, brain_hypo…\n",
      "$ counts       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 1, 1, 1, 2, 2, 2, 2, 104, 104, 104, 104, 104, 104, 104, …\n",
      "$ Display      \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m Adrenal gland, Coronary artery, Cortex, Hypothalamus, Hy…\n",
      "$ GeneSymbol   \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m SCO1, XIST, CPT1C, DEPDC5, CLN5, ANKHD1-EIF4EBP3, DGUOK,…\n",
      "$ GeneID       \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m ENSG00000133028.12, ENSG00000229807.11, ENSG00000169169.…\n",
      "$ chr          \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr17, chrX, chr19, chr22, chr13, chr5, chr2, chr10, chr…\n",
      "$ logFC        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 0.7715834, -2.1880965, -0.8246151, -1.2651387, -1.240754…\n",
      "$ AveExpr      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 5.028635, 4.327079, 6.908922, 3.521844, 4.678101, 6.4593…\n",
      "$ t            \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 4.906848, -10.329346, -5.042202, -5.035619, -4.930630, 4…\n",
      "$ PValue       \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 1.533202e-06, 4.631685e-21, 9.757441e-07, 1.099810e-06, …\n",
      "$ AdjPVal      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 8.170436e-03, 2.464057e-17, 5.117778e-03, 4.663317e-03, …\n",
      "$ B            \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 0.8561371, 16.7535120, 5.1676834, 4.2503727, 4.1488842, …\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "38"
      ],
      "text/latex": [
       "38"
      ],
      "text/markdown": [
       "38"
      ],
      "text/plain": [
       "[1] 38"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE \n",
       " 1278 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "         Adipose (sc)           Adipose (v)         Adrenal gland \n",
       "                    3                     3                     2 \n",
       "                Aorta      Atrial appendage                Breast \n",
       "                   10                     5                   996 \n",
       "              Caudate Cerebellar hemisphere            Cerebellum \n",
       "                    2                     2                    34 \n",
       "      Coronary artery                Cortex       EBV-lymphocytes \n",
       "                    2                     2                     2 \n",
       "      Esophagus (gej)         Esophagus (m)        Esophagus (mu) \n",
       "                    6                     3                    53 \n",
       "          Fibroblasts        Frontal cortex           Hippocampus \n",
       "                    3                     2                    14 \n",
       "         Hypothalamus        Left ventricle                 Liver \n",
       "                    3                     0                     5 \n",
       "                 Lung     Nucleus accumbens              Pancreas \n",
       "                    3                     3                     1 \n",
       "            Pituitary               Putamen         Sigmoid colon \n",
       "                   11                     2                     4 \n",
       "      Skeletal muscle        Skin (exposed)    Skin (not exposed) \n",
       "                   25                     6                     2 \n",
       "      Small intestine           Spinal cord                Spleen \n",
       "                    1                     3                    35 \n",
       "              Stomach               Thyroid         Tibial artery \n",
       "                    2                    13                     7 \n",
       "         Tibial nerve      Transverse colon           Whole blood \n",
       "                    3                     3                     2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneJunction'</li><li>'ASE'</li><li>'ASE_IDX'</li><li>'Tissue'</li><li>'counts'</li><li>'Display'</li><li>'GeneSymbol'</li><li>'GeneID'</li><li>'chr'</li><li>'logFC'</li><li>'AveExpr'</li><li>'t'</li><li>'PValue'</li><li>'AdjPVal'</li><li>'B'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneJunction'\n",
       "\\item 'ASE'\n",
       "\\item 'ASE\\_IDX'\n",
       "\\item 'Tissue'\n",
       "\\item 'counts'\n",
       "\\item 'Display'\n",
       "\\item 'GeneSymbol'\n",
       "\\item 'GeneID'\n",
       "\\item 'chr'\n",
       "\\item 'logFC'\n",
       "\\item 'AveExpr'\n",
       "\\item 't'\n",
       "\\item 'PValue'\n",
       "\\item 'AdjPVal'\n",
       "\\item 'B'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneJunction'\n",
       "2. 'ASE'\n",
       "3. 'ASE_IDX'\n",
       "4. 'Tissue'\n",
       "5. 'counts'\n",
       "6. 'Display'\n",
       "7. 'GeneSymbol'\n",
       "8. 'GeneID'\n",
       "9. 'chr'\n",
       "10. 'logFC'\n",
       "11. 'AveExpr'\n",
       "12. 't'\n",
       "13. 'PValue'\n",
       "14. 'AdjPVal'\n",
       "15. 'B'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GeneJunction\" \"ASE\"          \"ASE_IDX\"      \"Tissue\"       \"counts\"      \n",
       " [6] \"Display\"      \"GeneSymbol\"   \"GeneID\"       \"chr\"          \"logFC\"       \n",
       "[11] \"AveExpr\"      \"t\"            \"PValue\"       \"AdjPVal\"      \"B\"           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneJunction</th><th scope=col>ASE</th><th scope=col>ASE_IDX</th><th scope=col>Tissue</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>GeneSymbol</th><th scope=col>GeneID</th><th scope=col>chr</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SCO1-8452          </td><td>A3SS</td><td>8452</td><td>adrenal_gland                 </td><td>1</td><td>Adrenal gland  </td><td>SCO1           </td><td>ENSG00000133028.12</td><td>chr17</td><td> 0.7715834</td><td>5.028635</td><td>  4.906848</td><td>1.533202e-06</td><td>8.170436e-03</td><td> 0.8561371</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>XIST-2252          </td><td>A3SS</td><td>2252</td><td>artery_coronary               </td><td>1</td><td>Coronary artery</td><td>XIST           </td><td>ENSG00000229807.11</td><td>chrX </td><td>-2.1880965</td><td>4.327079</td><td>-10.329346</td><td>4.631685e-21</td><td>2.464057e-17</td><td>16.7535120</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>CPT1C-6427         </td><td>A3SS</td><td>6427</td><td>brain_cortex                  </td><td>1</td><td>Cortex         </td><td>CPT1C          </td><td>ENSG00000169169.14</td><td>chr19</td><td>-0.8246151</td><td>6.908922</td><td> -5.042202</td><td>9.757441e-07</td><td>5.117778e-03</td><td> 5.1676834</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>DEPDC5-1839        </td><td>A3SS</td><td>1839</td><td>brain_hypothalamus            </td><td>2</td><td>Hypothalamus   </td><td>DEPDC5         </td><td>ENSG00000100150.18</td><td>chr22</td><td>-1.2651387</td><td>3.521844</td><td> -5.035619</td><td>1.099810e-06</td><td>4.663317e-03</td><td> 4.2503727</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>CLN5-8117          </td><td>A3SS</td><td>8117</td><td>brain_hypothalamus            </td><td>2</td><td>Hypothalamus   </td><td>CLN5           </td><td>ENSG00000102805.15</td><td>chr13</td><td>-1.2407543</td><td>4.678101</td><td> -4.930630</td><td>1.777517e-06</td><td>4.663317e-03</td><td> 4.1488842</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ANKHD1-EIF4EBP3-453</td><td>A3SS</td><td> 453</td><td>brain_spinal_cord_cervical_c_1</td><td>2</td><td>Spinal cord    </td><td>ANKHD1-EIF4EBP3</td><td>ENSG00000254996.5 </td><td>chr5 </td><td> 0.6105820</td><td>6.459352</td><td>  4.440250</td><td>1.825438e-05</td><td>4.818212e-02</td><td> 2.5615289</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & GeneJunction & ASE & ASE\\_IDX & Tissue & counts & Display & GeneSymbol & GeneID & chr & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <fct> & <fct> & <int> & <fct> & <int> & <fct> & <fct> & <fct> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & SCO1-8452           & A3SS & 8452 & adrenal\\_gland                  & 1 & Adrenal gland   & SCO1            & ENSG00000133028.12 & chr17 &  0.7715834 & 5.028635 &   4.906848 & 1.533202e-06 & 8.170436e-03 &  0.8561371\\\\\n",
       "\t2 & XIST-2252           & A3SS & 2252 & artery\\_coronary                & 1 & Coronary artery & XIST            & ENSG00000229807.11 & chrX  & -2.1880965 & 4.327079 & -10.329346 & 4.631685e-21 & 2.464057e-17 & 16.7535120\\\\\n",
       "\t3 & CPT1C-6427          & A3SS & 6427 & brain\\_cortex                   & 1 & Cortex          & CPT1C           & ENSG00000169169.14 & chr19 & -0.8246151 & 6.908922 &  -5.042202 & 9.757441e-07 & 5.117778e-03 &  5.1676834\\\\\n",
       "\t4 & DEPDC5-1839         & A3SS & 1839 & brain\\_hypothalamus             & 2 & Hypothalamus    & DEPDC5          & ENSG00000100150.18 & chr22 & -1.2651387 & 3.521844 &  -5.035619 & 1.099810e-06 & 4.663317e-03 &  4.2503727\\\\\n",
       "\t5 & CLN5-8117           & A3SS & 8117 & brain\\_hypothalamus             & 2 & Hypothalamus    & CLN5            & ENSG00000102805.15 & chr13 & -1.2407543 & 4.678101 &  -4.930630 & 1.777517e-06 & 4.663317e-03 &  4.1488842\\\\\n",
       "\t6 & ANKHD1-EIF4EBP3-453 & A3SS &  453 & brain\\_spinal\\_cord\\_cervical\\_c\\_1 & 2 & Spinal cord     & ANKHD1-EIF4EBP3 & ENSG00000254996.5  & chr5  &  0.6105820 & 6.459352 &   4.440250 & 1.825438e-05 & 4.818212e-02 &  2.5615289\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | GeneJunction &lt;fct&gt; | ASE &lt;fct&gt; | ASE_IDX &lt;int&gt; | Tissue &lt;fct&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | GeneSymbol &lt;fct&gt; | GeneID &lt;fct&gt; | chr &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | SCO1-8452           | A3SS | 8452 | adrenal_gland                  | 1 | Adrenal gland   | SCO1            | ENSG00000133028.12 | chr17 |  0.7715834 | 5.028635 |   4.906848 | 1.533202e-06 | 8.170436e-03 |  0.8561371 |\n",
       "| 2 | XIST-2252           | A3SS | 2252 | artery_coronary                | 1 | Coronary artery | XIST            | ENSG00000229807.11 | chrX  | -2.1880965 | 4.327079 | -10.329346 | 4.631685e-21 | 2.464057e-17 | 16.7535120 |\n",
       "| 3 | CPT1C-6427          | A3SS | 6427 | brain_cortex                   | 1 | Cortex          | CPT1C           | ENSG00000169169.14 | chr19 | -0.8246151 | 6.908922 |  -5.042202 | 9.757441e-07 | 5.117778e-03 |  5.1676834 |\n",
       "| 4 | DEPDC5-1839         | A3SS | 1839 | brain_hypothalamus             | 2 | Hypothalamus    | DEPDC5          | ENSG00000100150.18 | chr22 | -1.2651387 | 3.521844 |  -5.035619 | 1.099810e-06 | 4.663317e-03 |  4.2503727 |\n",
       "| 5 | CLN5-8117           | A3SS | 8117 | brain_hypothalamus             | 2 | Hypothalamus    | CLN5            | ENSG00000102805.15 | chr13 | -1.2407543 | 4.678101 |  -4.930630 | 1.777517e-06 | 4.663317e-03 |  4.1488842 |\n",
       "| 6 | ANKHD1-EIF4EBP3-453 | A3SS |  453 | brain_spinal_cord_cervical_c_1 | 2 | Spinal cord     | ANKHD1-EIF4EBP3 | ENSG00000254996.5  | chr5  |  0.6105820 | 6.459352 |   4.440250 | 1.825438e-05 | 4.818212e-02 |  2.5615289 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneJunction        ASE  ASE_IDX Tissue                         counts\n",
       "1 SCO1-8452           A3SS 8452    adrenal_gland                  1     \n",
       "2 XIST-2252           A3SS 2252    artery_coronary                1     \n",
       "3 CPT1C-6427          A3SS 6427    brain_cortex                   1     \n",
       "4 DEPDC5-1839         A3SS 1839    brain_hypothalamus             2     \n",
       "5 CLN5-8117           A3SS 8117    brain_hypothalamus             2     \n",
       "6 ANKHD1-EIF4EBP3-453 A3SS  453    brain_spinal_cord_cervical_c_1 2     \n",
       "  Display         GeneSymbol      GeneID             chr   logFC      AveExpr \n",
       "1 Adrenal gland   SCO1            ENSG00000133028.12 chr17  0.7715834 5.028635\n",
       "2 Coronary artery XIST            ENSG00000229807.11 chrX  -2.1880965 4.327079\n",
       "3 Cortex          CPT1C           ENSG00000169169.14 chr19 -0.8246151 6.908922\n",
       "4 Hypothalamus    DEPDC5          ENSG00000100150.18 chr22 -1.2651387 3.521844\n",
       "5 Hypothalamus    CLN5            ENSG00000102805.15 chr13 -1.2407543 4.678101\n",
       "6 Spinal cord     ANKHD1-EIF4EBP3 ENSG00000254996.5  chr5   0.6105820 6.459352\n",
       "  t          PValue       AdjPVal      B         \n",
       "1   4.906848 1.533202e-06 8.170436e-03  0.8561371\n",
       "2 -10.329346 4.631685e-21 2.464057e-17 16.7535120\n",
       "3  -5.042202 9.757441e-07 5.117778e-03  5.1676834\n",
       "4  -5.035619 1.099810e-06 4.663317e-03  4.2503727\n",
       "5  -4.930630 1.777517e-06 4.663317e-03  4.1488842\n",
       "6   4.440250 1.825438e-05 4.818212e-02  2.5615289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "glimpse(gene_as)\n",
    "gene_as$Tissue <- factor(gene_as$Tissue)\n",
    "length(levels(gene_as$Tissue))\n",
    "table(is.na(gene_as$Display))\n",
    "table(gene_as$Display)\n",
    "colnames(gene_as)\n",
    "head(gene_as)\n",
    "tissue_reduction$display_name <- factor(tissue_reduction$display_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 1278 total significant alternative splicing events (gene_as)\n",
      "\n",
      "There were 100 total significant alternative splicing events on the X chromosome (gene_as)\n",
      "\n",
      "i.e., 7.82472613458529% of all significant AS events were on the X chromosome\n",
      "\n",
      "gene_as now has 39 tissues and 5 ASE categories\n",
      "\n",
      "ASE:\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>A3SS</dt><dd>130</dd><dt>A5SS</dt><dd>107</dd><dt>MXE</dt><dd>77</dd><dt>RI</dt><dd>178</dd><dt>SE</dt><dd>786</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[A3SS] 130\n",
       "\\item[A5SS] 107\n",
       "\\item[MXE] 77\n",
       "\\item[RI] 178\n",
       "\\item[SE] 786\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "A3SS\n",
       ":   130A5SS\n",
       ":   107MXE\n",
       ":   77RI\n",
       ":   178SE\n",
       ":   786\n",
       "\n"
      ],
      "text/plain": [
       "A3SS A5SS  MXE   RI   SE \n",
       " 130  107   77  178  786 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x_as_events <- gene_as[gene_as$chr==\"chrX\",]\n",
    "message(\"There were \",nrow(gene_as),\" total significant alternative splicing events (gene_as)\")\n",
    "message(\"There were \",nrow(x_as_events),\" total significant alternative splicing events on the X chromosome (gene_as)\")\n",
    "message(\"i.e., \", (100*nrow(x_as_events)/nrow(gene_as)), \"% of all significant AS events were on the X chromosome\")\n",
    "\n",
    "numberOfUniqueTissues <- length(summary(as.factor(gene_as$Display),maxsum=500))\n",
    "numberOfASEmechanisms <- length(summary(as.factor(gene_as$ASE),maxsum=500))\n",
    "\n",
    "message(\"gene_as now has \",numberOfUniqueTissues, \" tissues and \", numberOfASEmechanisms, \" ASE categories\")\n",
    "message(\"ASE:\")\n",
    "summary(as.factor(gene_as$ASE),maxsum=500)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.2 gene_dge.tsv\n",
    "\n",
    "This file contains (description)\n",
    "Here is a typical line\n",
    "<pre>\n",
    "Tissue  ENSG_ver        ENSG_no_ver     GeneSymbol      counts  Display logFC   AveExpr t       PValue  AdjPVal B\n",
    "1       adipose_subcutaneous    ENSG00000176728.7       ENSG00000176728 TTTY14  765     Adipose (sc)    -7.98216577151896     -0.928812923511535       -139.823010017733       0       0       1107.42326360464\n",
    "2       adipose_subcutaneous    ENSG00000231535.5       ENSG00000231535 LINC00278       765     Adipose (sc)    -6.09542040758638      -2.77656379347601       -126.913818678612       0       0       1050.36559888639\n",
    "3       adipose_subcutaneous    ENSG00000129824.15      ENSG00000129824 RPS4Y1  765     Adipose (sc)    -9.6641901864726      4.63528767282141 -125.827094717734       0       0       1041.87660796556\n",
    "4       adipose_subcutaneous    ENSG00000067646.11      ENSG00000067646 ZFY     765     Adipose (sc)    -9.50458982938477     0.672755457406984        -125.037143030325       0       0       1033.61131617113\n",
    "5       adipose_subcutaneous    ENSG00000229807.10      ENSG00000229807 XIST    765     Adipose (sc)    9.89280986473167      1.23756039627052 121.69689757218 0       0       1030.17757492281\n",
    "6       adipose_subcutaneous    ENSG00000229236.1       ENSG00000229236 TTTY10  765     Adipose (sc)    -6.20901295440725     -2.74524363170072        -122.540297482165       0       0       1029.41424065532\n",
    "7       adipose_subcutaneous    ENSG00000233864.7       ENSG00000233864 TTTY15  765     Adipose (sc)    -8.19361688496523     -0.741097276206495       -122.47199454746        0       0       1027.58175958647\n",
    "8       adipose_subcutaneous    ENSG00000260197.1       ENSG00000260197 AC010889.1      765     Adipose (sc)    -8.52835806068555      -0.686009457030557      -119.790486729538       0       0       1015.85291786821\n",
    "9       adipose_subcutaneous    ENSG00000183878.15      ENSG00000183878 UTY     765     Adipose (sc)    -9.52139275438866     1.60375445153084 -110.599936868261       0       0       953.069478992754\n",
    "</pre>\n",
    "There are 7417 significant events in the file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 4417 total significant differential gene expression events (gene_dge)\n",
      "\n",
      "There were 328 total significant differential gene expression events on the X chromosome (gene_as)\n",
      "\n",
      "i.e., 7.42585465247906% of all significant DGE events were on the X chromosome\n",
      "\n",
      "gene_dge now has 39 tissues\n",
      "\n"
     ]
    }
   ],
   "source": [
    "x_dge_events <- gene_dge[gene_dge$chr==\"chrX\",]\n",
    "message(\"There were \",nrow(gene_dge),\" total significant differential gene expression events (gene_dge)\")\n",
    "message(\"There were \",nrow(x_dge_events),\" total significant differential gene expression events on the X chromosome (gene_as)\")\n",
    "message(\"i.e., \", (100*nrow(x_dge_events)/nrow(gene_dge)), \"% of all significant DGE events were on the X chromosome\")\n",
    "\n",
    "numberOfUniqueTissues <- length(summary(as.factor(gene_dge$Display),maxsum=500))\n",
    "\n",
    "message(\"gene_dge now has \",numberOfUniqueTissues, \" tissues\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.3 Count events by chromosome\n",
    "\n",
    "Count the number of significant alternative splicing events per chromosome and save to the file **Total_AS_by_chr.tsv**."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.3.1 by alternative splicing events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "23"
      ],
      "text/latex": [
       "23"
      ],
      "text/markdown": [
       "23"
      ],
      "text/plain": [
       "[1] 23"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 23 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>129</td></tr>\n",
       "\t<tr><td>chrX </td><td>100</td></tr>\n",
       "\t<tr><td>chr17</td><td> 97</td></tr>\n",
       "\t<tr><td>chr16</td><td> 95</td></tr>\n",
       "\t<tr><td>chr19</td><td> 94</td></tr>\n",
       "\t<tr><td>chr11</td><td> 79</td></tr>\n",
       "\t<tr><td>chr2 </td><td> 75</td></tr>\n",
       "\t<tr><td>chr12</td><td> 68</td></tr>\n",
       "\t<tr><td>chr3 </td><td> 63</td></tr>\n",
       "\t<tr><td>chr10</td><td> 56</td></tr>\n",
       "\t<tr><td>chr6 </td><td> 53</td></tr>\n",
       "\t<tr><td>chr7 </td><td> 49</td></tr>\n",
       "\t<tr><td>chr4 </td><td> 42</td></tr>\n",
       "\t<tr><td>chr9 </td><td> 38</td></tr>\n",
       "\t<tr><td>chr22</td><td> 36</td></tr>\n",
       "\t<tr><td>chr8 </td><td> 36</td></tr>\n",
       "\t<tr><td>chr14</td><td> 34</td></tr>\n",
       "\t<tr><td>chr15</td><td> 33</td></tr>\n",
       "\t<tr><td>chr5 </td><td> 32</td></tr>\n",
       "\t<tr><td>chr20</td><td> 32</td></tr>\n",
       "\t<tr><td>chr13</td><td> 14</td></tr>\n",
       "\t<tr><td>chr18</td><td> 13</td></tr>\n",
       "\t<tr><td>chr21</td><td> 10</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 23 × 2\n",
       "\\begin{tabular}{ll}\n",
       " chr & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 129\\\\\n",
       "\t chrX  & 100\\\\\n",
       "\t chr17 &  97\\\\\n",
       "\t chr16 &  95\\\\\n",
       "\t chr19 &  94\\\\\n",
       "\t chr11 &  79\\\\\n",
       "\t chr2  &  75\\\\\n",
       "\t chr12 &  68\\\\\n",
       "\t chr3  &  63\\\\\n",
       "\t chr10 &  56\\\\\n",
       "\t chr6  &  53\\\\\n",
       "\t chr7  &  49\\\\\n",
       "\t chr4  &  42\\\\\n",
       "\t chr9  &  38\\\\\n",
       "\t chr22 &  36\\\\\n",
       "\t chr8  &  36\\\\\n",
       "\t chr14 &  34\\\\\n",
       "\t chr15 &  33\\\\\n",
       "\t chr5  &  32\\\\\n",
       "\t chr20 &  32\\\\\n",
       "\t chr13 &  14\\\\\n",
       "\t chr18 &  13\\\\\n",
       "\t chr21 &  10\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 23 × 2\n",
       "\n",
       "| chr &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 129 |\n",
       "| chrX  | 100 |\n",
       "| chr17 |  97 |\n",
       "| chr16 |  95 |\n",
       "| chr19 |  94 |\n",
       "| chr11 |  79 |\n",
       "| chr2  |  75 |\n",
       "| chr12 |  68 |\n",
       "| chr3  |  63 |\n",
       "| chr10 |  56 |\n",
       "| chr6  |  53 |\n",
       "| chr7  |  49 |\n",
       "| chr4  |  42 |\n",
       "| chr9  |  38 |\n",
       "| chr22 |  36 |\n",
       "| chr8  |  36 |\n",
       "| chr14 |  34 |\n",
       "| chr15 |  33 |\n",
       "| chr5  |  32 |\n",
       "| chr20 |  32 |\n",
       "| chr13 |  14 |\n",
       "| chr18 |  13 |\n",
       "| chr21 |  10 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr   n  \n",
       "1  chr1  129\n",
       "2  chrX  100\n",
       "3  chr17  97\n",
       "4  chr16  95\n",
       "5  chr19  94\n",
       "6  chr11  79\n",
       "7  chr2   75\n",
       "8  chr12  68\n",
       "9  chr3   63\n",
       "10 chr10  56\n",
       "11 chr6   53\n",
       "12 chr7   49\n",
       "13 chr4   42\n",
       "14 chr9   38\n",
       "15 chr22  36\n",
       "16 chr8   36\n",
       "17 chr14  34\n",
       "18 chr15  33\n",
       "19 chr5   32\n",
       "20 chr20  32\n",
       "21 chr13  14\n",
       "22 chr18  13\n",
       "23 chr21  10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 23\n",
      "Columns: 2\n",
      "$ chr \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr1, chrX, chr17, chr16, chr19, chr11, chr2, chr12, chr3, chr10,…\n",
      "$ n   \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 129, 100, 97, 95, 94, 79, 75, 68, 63, 56, 53, 49, 42, 38, 36, 36,…\n"
     ]
    }
   ],
   "source": [
    "total_as_by_chr <- gene_as          %>% \n",
    "                   group_by(chr)    %>% \n",
    "                   count(chr)       %>% \n",
    "                   arrange(desc(n)) %>% \n",
    "                   as.data.frame()\n",
    "total_as_by_chr$chr <- factor(total_as_by_chr$chr, levels = total_as_by_chr$chr)\n",
    "length(total_as_by_chr$chr)\n",
    "total_as_by_chr\n",
    "glimpse(total_as_by_chr)\n",
    "write.table(total_as_by_chr, file= \"../data/Total_AS_by_chr.tsv\", sep=\"\\t\", quote = FALSE, row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.3.2 by gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "26"
      ],
      "text/latex": [
       "26"
      ],
      "text/markdown": [
       "26"
      ],
      "text/plain": [
       "[1] 26"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 26 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>431</td></tr>\n",
       "\t<tr><td>chr2 </td><td>346</td></tr>\n",
       "\t<tr><td>chrX </td><td>328</td></tr>\n",
       "\t<tr><td>chr11</td><td>250</td></tr>\n",
       "\t<tr><td>chr12</td><td>234</td></tr>\n",
       "\t<tr><td>chr19</td><td>224</td></tr>\n",
       "\t<tr><td>chr17</td><td>216</td></tr>\n",
       "\t<tr><td>chr3 </td><td>212</td></tr>\n",
       "\t<tr><td>chr6 </td><td>199</td></tr>\n",
       "\t<tr><td>chr7 </td><td>196</td></tr>\n",
       "\t<tr><td>chr14</td><td>195</td></tr>\n",
       "\t<tr><td>chr4 </td><td>187</td></tr>\n",
       "\t<tr><td>chr5 </td><td>178</td></tr>\n",
       "\t<tr><td>chr10</td><td>153</td></tr>\n",
       "\t<tr><td>chr9 </td><td>150</td></tr>\n",
       "\t<tr><td>chr20</td><td>148</td></tr>\n",
       "\t<tr><td>chr15</td><td>147</td></tr>\n",
       "\t<tr><td>chr16</td><td>145</td></tr>\n",
       "\t<tr><td>chr22</td><td>130</td></tr>\n",
       "\t<tr><td>chr8 </td><td>128</td></tr>\n",
       "\t<tr><td>chr18</td><td> 67</td></tr>\n",
       "\t<tr><td>chr21</td><td> 58</td></tr>\n",
       "\t<tr><td>chr13</td><td> 57</td></tr>\n",
       "\t<tr><td>NA   </td><td> 28</td></tr>\n",
       "\t<tr><td>chrM </td><td>  7</td></tr>\n",
       "\t<tr><td>chrY </td><td>  3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 2\n",
       "\\begin{tabular}{ll}\n",
       " chr & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 431\\\\\n",
       "\t chr2  & 346\\\\\n",
       "\t chrX  & 328\\\\\n",
       "\t chr11 & 250\\\\\n",
       "\t chr12 & 234\\\\\n",
       "\t chr19 & 224\\\\\n",
       "\t chr17 & 216\\\\\n",
       "\t chr3  & 212\\\\\n",
       "\t chr6  & 199\\\\\n",
       "\t chr7  & 196\\\\\n",
       "\t chr14 & 195\\\\\n",
       "\t chr4  & 187\\\\\n",
       "\t chr5  & 178\\\\\n",
       "\t chr10 & 153\\\\\n",
       "\t chr9  & 150\\\\\n",
       "\t chr20 & 148\\\\\n",
       "\t chr15 & 147\\\\\n",
       "\t chr16 & 145\\\\\n",
       "\t chr22 & 130\\\\\n",
       "\t chr8  & 128\\\\\n",
       "\t chr18 &  67\\\\\n",
       "\t chr21 &  58\\\\\n",
       "\t chr13 &  57\\\\\n",
       "\t NA    &  28\\\\\n",
       "\t chrM  &   7\\\\\n",
       "\t chrY  &   3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 2\n",
       "\n",
       "| chr &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 431 |\n",
       "| chr2  | 346 |\n",
       "| chrX  | 328 |\n",
       "| chr11 | 250 |\n",
       "| chr12 | 234 |\n",
       "| chr19 | 224 |\n",
       "| chr17 | 216 |\n",
       "| chr3  | 212 |\n",
       "| chr6  | 199 |\n",
       "| chr7  | 196 |\n",
       "| chr14 | 195 |\n",
       "| chr4  | 187 |\n",
       "| chr5  | 178 |\n",
       "| chr10 | 153 |\n",
       "| chr9  | 150 |\n",
       "| chr20 | 148 |\n",
       "| chr15 | 147 |\n",
       "| chr16 | 145 |\n",
       "| chr22 | 130 |\n",
       "| chr8  | 128 |\n",
       "| chr18 |  67 |\n",
       "| chr21 |  58 |\n",
       "| chr13 |  57 |\n",
       "| NA    |  28 |\n",
       "| chrM  |   7 |\n",
       "| chrY  |   3 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr   n  \n",
       "1  chr1  431\n",
       "2  chr2  346\n",
       "3  chrX  328\n",
       "4  chr11 250\n",
       "5  chr12 234\n",
       "6  chr19 224\n",
       "7  chr17 216\n",
       "8  chr3  212\n",
       "9  chr6  199\n",
       "10 chr7  196\n",
       "11 chr14 195\n",
       "12 chr4  187\n",
       "13 chr5  178\n",
       "14 chr10 153\n",
       "15 chr9  150\n",
       "16 chr20 148\n",
       "17 chr15 147\n",
       "18 chr16 145\n",
       "19 chr22 130\n",
       "20 chr8  128\n",
       "21 chr18  67\n",
       "22 chr21  58\n",
       "23 chr13  57\n",
       "24 NA     28\n",
       "25 chrM    7\n",
       "26 chrY    3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 26\n",
      "Columns: 2\n",
      "$ chr \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr1, chr2, chrX, chr11, chr12, chr19, chr17, chr3, chr6, chr7, c…\n",
      "$ n   \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 431, 346, 328, 250, 234, 224, 216, 212, 199, 196, 195, 187, 178, …\n"
     ]
    }
   ],
   "source": [
    "total_dge_by_chr <- gene_dge          %>% \n",
    "                   group_by(chr)    %>% \n",
    "                   count(chr)       %>% \n",
    "                   arrange(desc(n)) %>% \n",
    "                   as.data.frame()\n",
    "total_dge_by_chr$chr <- factor(total_dge_by_chr$chr, levels = total_dge_by_chr$chr)\n",
    "length(total_dge_by_chr$chr)\n",
    "total_dge_by_chr\n",
    "glimpse(total_dge_by_chr)\n",
    "write.table(total_dge_by_chr, file= \"../data/Total_DGE_by_chr.tsv\", sep=\"\\t\", quote = FALSE, row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.4 Count events by genes "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.4.1 by alternative splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "888"
      ],
      "text/latex": [
       "888"
      ],
      "text/markdown": [
       "888"
      ],
      "text/plain": [
       "[1] 888"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneSymbol</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>XIST  </td><td>27</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>KDM5C </td><td>23</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>DMKN  </td><td> 9</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>CD44  </td><td> 9</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>SORBS2</td><td> 9</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>DDX3X </td><td> 8</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>ADGRG1</td><td> 7</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>MUC1  </td><td> 7</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>SPG7  </td><td> 7</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>DTNB  </td><td> 6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & GeneSymbol & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & XIST   & 27\\\\\n",
       "\t2 & KDM5C  & 23\\\\\n",
       "\t3 & DMKN   &  9\\\\\n",
       "\t4 & CD44   &  9\\\\\n",
       "\t5 & SORBS2 &  9\\\\\n",
       "\t6 & DDX3X  &  8\\\\\n",
       "\t7 & ADGRG1 &  7\\\\\n",
       "\t8 & MUC1   &  7\\\\\n",
       "\t9 & SPG7   &  7\\\\\n",
       "\t10 & DTNB   &  6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | GeneSymbol &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | XIST   | 27 |\n",
       "| 2 | KDM5C  | 23 |\n",
       "| 3 | DMKN   |  9 |\n",
       "| 4 | CD44   |  9 |\n",
       "| 5 | SORBS2 |  9 |\n",
       "| 6 | DDX3X  |  8 |\n",
       "| 7 | ADGRG1 |  7 |\n",
       "| 8 | MUC1   |  7 |\n",
       "| 9 | SPG7   |  7 |\n",
       "| 10 | DTNB   |  6 |\n",
       "\n"
      ],
      "text/plain": [
       "   GeneSymbol n \n",
       "1  XIST       27\n",
       "2  KDM5C      23\n",
       "3  DMKN        9\n",
       "4  CD44        9\n",
       "5  SORBS2      9\n",
       "6  DDX3X       8\n",
       "7  ADGRG1      7\n",
       "8  MUC1        7\n",
       "9  SPG7        7\n",
       "10 DTNB        6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_geneSymbol <- gene_as %>% \n",
    "                          group_by(GeneSymbol) %>% \n",
    "                          count(GeneSymbol)    %>% \n",
    "                          arrange(desc(n))     %>% \n",
    "                          as.data.frame()\n",
    "total_as_by_geneSymbol$GeneSymbol <- factor(total_as_by_geneSymbol$GeneSymbol, \n",
    "                                            levels = total_as_by_geneSymbol$GeneSymbol)\n",
    "length(total_as_by_geneSymbol$GeneSymbol)\n",
    "head(total_as_by_geneSymbol,10)\n",
    "write.table(total_as_by_geneSymbol, file = \"../data/Total_AS_by_geneSymbol.tsv\", sep = \"\\t\", quote=FALSE, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.5 Count most frequent splicing by tissue\n",
    "\n",
    "### 3.5.1 by alternative splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Display</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Breast         </td><td>996</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Esophagus (mu) </td><td> 53</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Spleen         </td><td> 35</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Cerebellum     </td><td> 34</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Skeletal muscle</td><td> 25</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Hippocampus    </td><td> 14</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Thyroid        </td><td> 13</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>Pituitary      </td><td> 11</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Aorta          </td><td> 10</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Tibial artery  </td><td>  7</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Display & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Breast          & 996\\\\\n",
       "\t2 & Esophagus (mu)  &  53\\\\\n",
       "\t3 & Spleen          &  35\\\\\n",
       "\t4 & Cerebellum      &  34\\\\\n",
       "\t5 & Skeletal muscle &  25\\\\\n",
       "\t6 & Hippocampus     &  14\\\\\n",
       "\t7 & Thyroid         &  13\\\\\n",
       "\t8 & Pituitary       &  11\\\\\n",
       "\t9 & Aorta           &  10\\\\\n",
       "\t10 & Tibial artery   &   7\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | Display &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Breast          | 996 |\n",
       "| 2 | Esophagus (mu)  |  53 |\n",
       "| 3 | Spleen          |  35 |\n",
       "| 4 | Cerebellum      |  34 |\n",
       "| 5 | Skeletal muscle |  25 |\n",
       "| 6 | Hippocampus     |  14 |\n",
       "| 7 | Thyroid         |  13 |\n",
       "| 8 | Pituitary       |  11 |\n",
       "| 9 | Aorta           |  10 |\n",
       "| 10 | Tibial artery   |   7 |\n",
       "\n"
      ],
      "text/plain": [
       "   Display         n  \n",
       "1  Breast          996\n",
       "2  Esophagus (mu)   53\n",
       "3  Spleen           35\n",
       "4  Cerebellum       34\n",
       "5  Skeletal muscle  25\n",
       "6  Hippocampus      14\n",
       "7  Thyroid          13\n",
       "8  Pituitary        11\n",
       "9  Aorta            10\n",
       "10 Tibial artery     7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "38"
      ],
      "text/latex": [
       "38"
      ],
      "text/markdown": [
       "38"
      ],
      "text/plain": [
       "[1] 38"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_tissue <- gene_as %>% \n",
    "                      group_by(Display) %>% \n",
    "                      count(Display)    %>% \n",
    "                      arrange(desc(n))  %>% \n",
    "                      as.data.frame()\n",
    "total_as_by_tissue$Display <- factor(total_as_by_tissue$Display, \n",
    "                                     levels = total_as_by_tissue$Display)\n",
    "head(total_as_by_tissue,10)\n",
    "length(total_as_by_tissue$Display)\n",
    "write.table(total_as_by_tissue, file = \"../data/Total_AS_by_tissue.tsv\", sep = \"\\t\", row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.5.2 by gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Display</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Breast            </td><td>2296</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Adipose (sc)      </td><td> 272</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Pituitary         </td><td> 256</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Thyroid           </td><td> 164</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Adipose (v)       </td><td> 156</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Skin (not exposed)</td><td> 129</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Skin (exposed)    </td><td> 116</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>Skeletal muscle   </td><td> 102</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Left ventricle    </td><td>  85</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Fibroblasts       </td><td>  84</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Display & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Breast             & 2296\\\\\n",
       "\t2 & Adipose (sc)       &  272\\\\\n",
       "\t3 & Pituitary          &  256\\\\\n",
       "\t4 & Thyroid            &  164\\\\\n",
       "\t5 & Adipose (v)        &  156\\\\\n",
       "\t6 & Skin (not exposed) &  129\\\\\n",
       "\t7 & Skin (exposed)     &  116\\\\\n",
       "\t8 & Skeletal muscle    &  102\\\\\n",
       "\t9 & Left ventricle     &   85\\\\\n",
       "\t10 & Fibroblasts        &   84\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | Display &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Breast             | 2296 |\n",
       "| 2 | Adipose (sc)       |  272 |\n",
       "| 3 | Pituitary          |  256 |\n",
       "| 4 | Thyroid            |  164 |\n",
       "| 5 | Adipose (v)        |  156 |\n",
       "| 6 | Skin (not exposed) |  129 |\n",
       "| 7 | Skin (exposed)     |  116 |\n",
       "| 8 | Skeletal muscle    |  102 |\n",
       "| 9 | Left ventricle     |   85 |\n",
       "| 10 | Fibroblasts        |   84 |\n",
       "\n"
      ],
      "text/plain": [
       "   Display            n   \n",
       "1  Breast             2296\n",
       "2  Adipose (sc)        272\n",
       "3  Pituitary           256\n",
       "4  Thyroid             164\n",
       "5  Adipose (v)         156\n",
       "6  Skin (not exposed)  129\n",
       "7  Skin (exposed)      116\n",
       "8  Skeletal muscle     102\n",
       "9  Left ventricle       85\n",
       "10 Fibroblasts          84"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#glimpse(gene_dge)\n",
    "gene_dge$GeneSymbol <- factor(gene_dge$GeneSymbol)\n",
    "total_dge_by_tissue <- gene_dge %>% \n",
    "                          select(c(GeneSymbol, Display, logFC)) %>%\n",
    "                          group_by(Display) %>%\n",
    "                          arrange(desc(logFC)) %>%\n",
    "                          tally() %>%\n",
    "                          arrange(desc(n)) %>%\n",
    "                          as.data.frame()\n",
    "head(total_dge_by_tissue,10)\n",
    "length(total_dge_by_tissue$Display)\n",
    "write.table(total_dge_by_tissue, file = \"../data/Total_DGE_by_tissue.tsv\", sep = \"\\t\", quote=FALSE, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  3.6 Significant Count by splicing type \n",
    "We define **significant** to be FC > 1.5 and pVal < 0.05\n",
    "\n",
    "Our starting values were the significant events, all meeting the criteria FC > 1.5 and pVal < 0.05\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 5 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ASE</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SE  </td><td>786</td></tr>\n",
       "\t<tr><td>RI  </td><td>178</td></tr>\n",
       "\t<tr><td>A3SS</td><td>130</td></tr>\n",
       "\t<tr><td>A5SS</td><td>107</td></tr>\n",
       "\t<tr><td>MXE </td><td> 77</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ASE & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t SE   & 786\\\\\n",
       "\t RI   & 178\\\\\n",
       "\t A3SS & 130\\\\\n",
       "\t A5SS & 107\\\\\n",
       "\t MXE  &  77\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 2\n",
       "\n",
       "| ASE &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| SE   | 786 |\n",
       "| RI   | 178 |\n",
       "| A3SS | 130 |\n",
       "| A5SS | 107 |\n",
       "| MXE  |  77 |\n",
       "\n"
      ],
      "text/plain": [
       "  ASE  n  \n",
       "1 SE   786\n",
       "2 RI   178\n",
       "3 A3SS 130\n",
       "4 A5SS 107\n",
       "5 MXE   77"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_splicingtype <- gene_as %>% \n",
    "                            group_by(ASE)    %>% \n",
    "                            count(ASE)       %>% \n",
    "                            arrange(desc(n)) %>%\n",
    "                            as.data.frame()\n",
    "total_as_by_splicingtype$ASE <- factor(total_as_by_splicingtype$ASE, levels = total_as_by_splicingtype$ASE)\n",
    "total_as_by_splicingtype\n",
    "write.table(total_as_by_splicingtype, file= \"../data/Total_AS_by_splicingtype.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  3.7 Significant Count by splicing type (significant == FC > 1.5 and pVal < 0.05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "A3SS_keep\n",
       "FALSE  TRUE \n",
       " 1148   130 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A5SS_keep\n",
       "FALSE  TRUE \n",
       " 1171   107 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "MXE_keep\n",
       "FALSE  TRUE \n",
       " 1201    77 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "SE_keep\n",
       "FALSE  TRUE \n",
       "  492   786 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RI_keep\n",
       "FALSE  TRUE \n",
       " 1100   178 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>130</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 130\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 130\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 130  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>107</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 107\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 107\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 107  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>77</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 77\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 77\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 77 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>786</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 786\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 786\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 786  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>178</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 178\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 178\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 178  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "A3SS_keep <- as.character(gene_as$ASE) %in% \"A3SS\"\n",
    "table(A3SS_keep)\n",
    "A3SS.gene_as <- data.frame(gene_as[A3SS_keep == TRUE,])\n",
    "\n",
    "A5SS_keep <- as.character(gene_as$ASE) %in% \"A5SS\"\n",
    "table(A5SS_keep)\n",
    "A5SS.gene_as <- data.frame(gene_as[A5SS_keep == TRUE,])\n",
    "\n",
    "MXE_keep  <- as.character(gene_as$ASE) %in% \"MXE\"\n",
    "table(MXE_keep)\n",
    "MXE.gene_as <- data.frame(gene_as[MXE_keep == TRUE,])\n",
    "\n",
    "SE_keep   <- as.character(gene_as$ASE) %in% \"SE\"\n",
    "table(SE_keep)\n",
    "SE.gene_as <- data.frame(gene_as[SE_keep == TRUE,])\n",
    "\n",
    "RI_keep   <- as.character(gene_as$ASE) %in% \"RI\"\n",
    "table(RI_keep)\n",
    "RI.gene_as <- data.frame(gene_as[RI_keep == TRUE,])\n",
    "\n",
    "dim(A3SS.gene_as)\n",
    "dim(A5SS.gene_as)\n",
    "dim(MXE.gene_as)\n",
    "dim(SE.gene_as)\n",
    "dim(RI.gene_as)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.8 Siginficant spliced by Gene for each splicing factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for A3SS\n",
      "116 \n",
      "\n",
      "Significant spliced genes for A5SS\n",
      "100 \n",
      "\n",
      "Significant spliced genes for MXE\n",
      "50 \n",
      "\n",
      "Significant spliced genes for RI\n",
      "159 \n",
      "\n",
      "Significant spliced genes for SE\n",
      "589 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "A3SS.res <- A3SS.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "A3SS.res$GeneSymbol <- factor(A3SS.res$GeneSymbol, levels = A3SS.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for A3SS\\n\",\n",
    "        paste(length(A3SS.res$GeneSymbol)), collapse=\" \")\n",
    "#head(A3SS.res)\n",
    "\n",
    "A5SS.res <- A5SS.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "A5SS.res$GeneSymbol <- factor(A5SS.res$GeneSymbol, levels = A5SS.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for A5SS\\n\",\n",
    "        paste(length(A5SS.res$GeneSymbol)), collapse=\" \")\n",
    "#head(A5SS.res)\n",
    "\n",
    "MXE.res <- MXE.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "MXE.res$GeneSymbol <- factor(MXE.res$GeneSymbol, levels = MXE.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for MXE\\n\",\n",
    "        paste(length(MXE.res$GeneSymbol)), collapse=\" \")\n",
    "#head(MXE.res)\n",
    "\n",
    "RI.res <- RI.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "RI.res$GeneSymbol <- factor(RI.res$GeneSymbol, levels = RI.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for RI\\n\",\n",
    "        paste(length(RI.res$GeneSymbol)), collapse=\" \")\n",
    "#head(RI.res)\n",
    "\n",
    "SE.res <- SE.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "SE.res$GeneSymbol <- factor(SE.res$GeneSymbol, levels = SE.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for SE\\n\",\n",
    "        paste(length(SE.res$GeneSymbol)), collapse=\" \")\n",
    "#head(SE.res)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.9 Count most frequent spliced genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "888"
      ],
      "text/latex": [
       "888"
      ],
      "text/markdown": [
       "888"
      ],
      "text/plain": [
       "[1] 888"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneSymbol</th><th scope=col>n</th><th scope=col>Tissues</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>XIST  </td><td>27</td><td>23</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>KDM5C </td><td>23</td><td>22</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>DMKN  </td><td> 9</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>CD44  </td><td> 9</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>SORBS2</td><td> 9</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>DDX3X </td><td> 8</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & GeneSymbol & n & Tissues\\\\\n",
       "  & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & XIST   & 27 & 23\\\\\n",
       "\t2 & KDM5C  & 23 & 22\\\\\n",
       "\t3 & DMKN   &  9 &  1\\\\\n",
       "\t4 & CD44   &  9 &  1\\\\\n",
       "\t5 & SORBS2 &  9 &  1\\\\\n",
       "\t6 & DDX3X  &  8 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | GeneSymbol &lt;fct&gt; | n &lt;int&gt; | Tissues &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | XIST   | 27 | 23 |\n",
       "| 2 | KDM5C  | 23 | 22 |\n",
       "| 3 | DMKN   |  9 |  1 |\n",
       "| 4 | CD44   |  9 |  1 |\n",
       "| 5 | SORBS2 |  9 |  1 |\n",
       "| 6 | DDX3X  |  8 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneSymbol n  Tissues\n",
       "1 XIST       27 23     \n",
       "2 KDM5C      23 22     \n",
       "3 DMKN        9  1     \n",
       "4 CD44        9  1     \n",
       "5 SORBS2      9  1     \n",
       "6 DDX3X       8  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genesMostFrequentlySpliced <- gene_as %>% \n",
    "                              group_by(GeneSymbol) %>% \n",
    "                              count(GeneSymbol)    %>% \n",
    "                              arrange(desc(n))     %>% \n",
    "                              as.data.frame()\n",
    "genesMostFrequentlySpliced$GeneSymbol <- factor(genesMostFrequentlySpliced$GeneSymbol, \n",
    "                                                levels = genesMostFrequentlySpliced$GeneSymbol)\n",
    "length(genesMostFrequentlySpliced$GeneSymbol)\n",
    "\n",
    "#Add number of tissues\n",
    "nTissues <- rep(NA, length(genesMostFrequentlySpliced))\n",
    "for (i in 1:nrow(genesMostFrequentlySpliced)) {\n",
    "    df_gene <- gene_as %>% \n",
    "               filter(GeneSymbol == genesMostFrequentlySpliced$GeneSymbol[i])\n",
    "    nTissues[i] <- length(unique(df_gene$Tissue))\n",
    "}\n",
    "genesMostFrequentlySpliced$Tissues <- nTissues\n",
    "head(genesMostFrequentlySpliced)\n",
    "write.table(genesMostFrequentlySpliced, file = \"../data/genesWithCommonAS.tsv\", sep = \"\\t\", quote = F, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.10 Count most frequent spliced chromosomes\n",
    "To get an indication of which chromosome has the most frequent slicing event (regardless of type)\n",
    "We create an index based upon the number of exons per chromosome.\n",
    "\n",
    "get the annotation file, at this writing, gencode.v30.annotation.gtf\n",
    "The information as to the number of exons within the chromosome may be found there"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 23\n",
      "Columns: 4\n",
      "$ chr        \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chrX, chr16, chr19, chr22, chr17, chr10, chr20, chr1, chr1…\n",
      "$ n          \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 100, 95, 94, 36, 97, 56, 32, 129, 79, 53, 68, 42, 38, 49, …\n",
      "$ ExonCounts \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 40029, 61199, 74466, 28655, 78291, 47124, 28506, 118996, 7…\n",
      "$ Index      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 2.4981888, 1.5523129, 1.2623211, 1.2563252, 1.2389674, 1.1…\n"
     ]
    }
   ],
   "source": [
    "exons <- gencode[ gencode$type == \"exon\", ]\n",
    "exons <- as.data.frame(exons)\n",
    "\n",
    "#Obtain chromosomes we have splicing information for (recall we did not use chr Y in our analysis)\n",
    "all_chr <- as.character(unique(gene_as$chr))\n",
    "chr_counts <- rep(0, length(all_chr))\n",
    "\n",
    "\n",
    "for (i in 1:length(all_chr)) {\n",
    "  chr_counts[i] <- nrow(exons[exons$seqnames == all_chr[i], ])\n",
    "}\n",
    "\n",
    "exon_counts <- data.frame(chr = all_chr, counts = chr_counts)\n",
    "\n",
    "# Count most frequent spliced chromosomes\n",
    "res <- gene_as %>% group_by(chr) %>% count(chr) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "res$chr <- factor(res$chr, levels = res$chr)\n",
    "\n",
    "idx <- match(res$chr, exon_counts$chr)\n",
    "\n",
    "res$ExonCounts <- exon_counts$counts[idx]\n",
    "\n",
    "res$Index <- (res$n / res$ExonCounts) * 1000\n",
    "\n",
    "res_sorted <- res %>% arrange(desc(Index))\n",
    "res_sorted$chr <- factor(res_sorted$chr, levels = res_sorted$chr)\n",
    "glimpse(res_sorted)\n",
    "write.table(res_sorted, file = \"../data/SplicingIndex_chr.tsv\", sep = \"\\t\", quote = F, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.11 Overlap between Differential Gene Expression and Differential Alternative Splicing\n",
    "\n",
    "First gather the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>A2M</li><li>AADAC</li><li>AADAC</li><li>AADAC</li><li>AADACL3</li><li>AARD</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'A2M'</li><li>'AADAC'</li><li>'AADACL3'</li><li>'AARD'</li><li>'AATBC'</li><li>'AATK'</li><li>'ABCA12'</li><li>'ABCA13'</li><li>'ABCA4'</li><li>'ABCA8'</li><li>'ABCB1'</li><li>'ABCB4'</li><li>'ABCB5'</li><li>'ABCC11'</li><li>'ABCC3'</li><li>'ABCC6'</li><li>'ABCC8'</li><li>'ABCC9'</li><li>'ABI3BP'</li><li>'ABLIM2'</li><li>'ABLIM3'</li><li>'AC002066.1'</li><li>'AC002546.1'</li><li>'AC003986.2'</li><li>'AC004130.2'</li><li>'AC004160.1'</li><li>'AC004233.2'</li><li>'AC004540.1'</li><li>'AC004947.1'</li><li>'AC005064.1'</li><li>'AC005077.4'</li><li>'AC005082.1'</li><li>'AC005083.1'</li><li>'AC005162.2'</li><li>'AC005165.1'</li><li>'AC005323.1'</li><li>'AC005336.1'</li><li>'AC005550.2'</li><li>'AC005674.2'</li><li>'AC005831.1'</li><li>'AC005906.2'</li><li>'AC006946.2'</li><li>'AC007423.1'</li><li>'AC007666.1'</li><li>'AC007743.1'</li><li>'AC007906.2'</li><li>'AC008014.1'</li><li>'AC008035.1'</li><li>'AC008105.1'</li><li>'AC008268.1'</li><li>'AC008592.4'</li><li>'AC008763.1'</li><li>'AC008894.3'</li><li>'AC008957.3'</li><li>'AC010203.1'</li><li>'AC010329.1'</li><li>'AC010457.1'</li><li>'AC010503.5'</li><li>'AC010655.3'</li><li>'AC010735.2'</li><li>'AC010761.4'</li><li>'AC010976.2'</li><li>'AC011008.2'</li><li>'AC011337.1'</li><li>'AC011352.1'</li><li>'AC011476.3'</li><li>'AC011503.1'</li><li>'AC012313.2'</li><li>'AC012368.1'</li><li>'AC015712.1'</li><li>'AC015712.2'</li><li>'AC015712.4'</li><li>'AC015802.6'</li><li>'AC016705.2'</li><li>'AC016738.1'</li><li>'AC018553.1'</li><li>'AC018629.1'</li><li>'AC019205.1'</li><li>'AC020604.1'</li><li>'AC020636.1'</li><li>'AC020659.1'</li><li>'AC020912.1'</li><li>'AC020951.1'</li><li>'AC022034.1'</li><li>'AC022144.1'</li><li>'AC022148.1'</li><li>'AC022497.1'</li><li>'AC022706.1'</li><li>'AC023301.1'</li><li>'AC024560.2'</li><li>'AC024940.2'</li><li>'AC025569.1'</li><li>'AC027117.1'</li><li>'AC036108.2'</li><li>'AC036108.3'</li><li>'AC048341.1'</li><li>'AC055854.1'</li><li>'AC062004.1'</li><li>'AC068473.3'</li><li>'AC068587.4'</li><li>'AC069148.1'</li><li>'AC073342.2'</li><li>'AC073850.1'</li><li>'AC078850.1'</li><li>'AC079313.2'</li><li>'AC080013.1'</li><li>'AC080013.4'</li><li>'AC083843.3'</li><li>'AC083843.4'</li><li>'AC084871.1'</li><li>'AC087482.1'</li><li>'AC087521.1'</li><li>'AC087623.1'</li><li>'AC087741.1'</li><li>'AC091182.1'</li><li>'AC091544.4'</li><li>'AC092164.1'</li><li>'AC092171.5'</li><li>'AC092490.1'</li><li>'AC092535.5'</li><li>'AC092746.1'</li><li>'AC093001.1'</li><li>'AC093159.1'</li><li>'AC093162.3'</li><li>'AC093278.2'</li><li>'AC093425.1'</li><li>'AC093627.4'</li><li>'AC093627.6'</li><li>'AC093642.2'</li><li>'AC093866.1'</li><li>'AC096677.1'</li><li>'AC100793.4'</li><li>'AC103740.1'</li><li>'AC104024.2'</li><li>'AC104118.1'</li><li>'AC104260.2'</li><li>'AC104695.4'</li><li>'AC104809.2'</li><li>'AC105391.1'</li><li>'AC105942.1'</li><li>'AC106881.1'</li><li>'AC107204.1'</li><li>'AC107464.2'</li><li>'AC107884.1'</li><li>'AC108058.1'</li><li>'AC108134.1'</li><li>'AC108472.1'</li><li>'AC108704.2'</li><li>'AC108860.2'</li><li>'AC112206.2'</li><li>'AC114811.2'</li><li>'AC123912.2'</li><li>'AC126323.1'</li><li>'AC130466.1'</li><li>'AC136475.3'</li><li>'AC139769.1'</li><li>'AC141557.1'</li><li>'AC233266.2'</li><li>'AC237221.1'</li><li>'AC243772.2'</li><li>'AC244453.2'</li><li>'AC245041.2'</li><li>'AC245060.5'</li><li>'AC245100.2'</li><li>'AC245100.4'</li><li>'ACADL'</li><li>'ACAN'</li><li>'ACE2'</li><li>'ACER2'</li><li>'ACKR1'</li><li>'ACKR4P1'</li><li>'ACOT11'</li><li>'ACOT4'</li><li>'ACP5'</li><li>'ACR'</li><li>'ACSM1'</li><li>'ACSM3'</li><li>'ACSM5'</li><li>'ACSM6'</li><li>'ACTA2-AS1'</li><li>'ACTC1'</li><li>'ACTG2'</li><li>'ACTN3'</li><li>'ACTR3C'</li><li>'ADA'</li><li>'ADAD2'</li><li>'ADAM23'</li><li>'ADAM28'</li><li>'ADAMDEC1'</li><li>'ADAMTS14'</li><li>'ADAMTS18'</li><li>'ADAMTS2'</li><li>'ADAMTS3'</li><li>'ADAMTS4'</li><li>'ADAMTS7'</li><li>'ADAMTS9'</li><li>'ADAMTS9-AS1'</li><li>'ADAMTSL1'</li><li>'ADAMTSL5'</li><li>'ADARB2'</li><li>⋯</li><li>'TNFSF14'</li><li>'TNFSF15'</li><li>'TNIK'</li><li>'TNK1'</li><li>'TNMD'</li><li>'TNN'</li><li>'TNNI1'</li><li>'TNNI2'</li><li>'TNNT2'</li><li>'TNS4'</li><li>'TOX2'</li><li>'TOX3'</li><li>'TP53INP2'</li><li>'TP63'</li><li>'TP73'</li><li>'TPH1'</li><li>'TPO'</li><li>'TPPP3'</li><li>'TPRG1'</li><li>'TPSAB1'</li><li>'TPSB2'</li><li>'TPT1P12'</li><li>'TPTE2P5'</li><li>'TPTEP1'</li><li>'TRABD2B'</li><li>'TRAC'</li><li>'TRAF3IP3'</li><li>'TRAF4'</li><li>'TRARG1'</li><li>'TRAT1'</li><li>'TRBC1'</li><li>'TRBC2'</li><li>'TRDN'</li><li>'TREM2'</li><li>'TREML1'</li><li>'TRG-AS1'</li><li>'TRGC2'</li><li>'TRHDE'</li><li>'TRHDE-AS1'</li><li>'TRIL'</li><li>'TRIM17'</li><li>'TRIM2'</li><li>'TRIM29'</li><li>'TRIM46'</li><li>'TRIM55'</li><li>'TRIM9'</li><li>'TRMT9B'</li><li>'TRPA1'</li><li>'TRPC4'</li><li>'TRPC6'</li><li>'TRPC7'</li><li>'TRPM2-AS'</li><li>'TRPS1'</li><li>'TRPV4'</li><li>'TRPV6'</li><li>'TSHB'</li><li>'TSHZ2'</li><li>'TSKS'</li><li>'TSPAN1'</li><li>'TSPAN11'</li><li>'TSPAN13'</li><li>'TSPAN33'</li><li>'TSPAN5'</li><li>'TSPAN7'</li><li>'TSPAN8'</li><li>'TSPEAR'</li><li>'TSPEAR-AS1'</li><li>'TSPEAR-AS2'</li><li>'TSPOAP1'</li><li>'TTC22'</li><li>'TTC23L'</li><li>'TTC6'</li><li>'TTC9'</li><li>'TTYH1'</li><li>'TUBA3D'</li><li>'TUBBP5'</li><li>'TUFT1'</li><li>'TUSC8'</li><li>'TXLNG'</li><li>'UBASH3B'</li><li>'UBE2Q2P2'</li><li>'UBXN10'</li><li>'UCHL1'</li><li>'UCN3'</li><li>'UGT1A2P'</li><li>'UGT2A1'</li><li>'UGT2B17'</li><li>'UGT3A2'</li><li>'UGT8'</li><li>'UNC13C'</li><li>'UNC5B'</li><li>'UNC5C'</li><li>'UPK1B'</li><li>'UPK3B'</li><li>'UPP1'</li><li>'USHBP1'</li><li>'USP32P2'</li><li>'USP43'</li><li>'USP6'</li><li>'UTS2R'</li><li>'VAMP5'</li><li>'VANGL2'</li><li>'VASH2'</li><li>'VAT1L'</li><li>'VAV3'</li><li>'VAX2'</li><li>'VDR'</li><li>'VEGFD'</li><li>'VGF'</li><li>'VGLL1'</li><li>'VIL1'</li><li>'VIM'</li><li>'VIM-AS1'</li><li>'VIPR2'</li><li>'VIT'</li><li>'VLDLR-AS1'</li><li>'VMO1'</li><li>'VN1R53P'</li><li>'VNN2'</li><li>'VPS9D1-AS1'</li><li>'VSIG1'</li><li>'VSIG4'</li><li>'VSNL1'</li><li>'VSTM2A'</li><li>'VSTM5'</li><li>'VTCN1'</li><li>'VWC2'</li><li>'VWCE'</li><li>'VWDE'</li><li>'VWF'</li><li>'WDR62'</li><li>'WDR86'</li><li>'WDR86-AS1'</li><li>'WEE1'</li><li>'WFDC1'</li><li>'WFDC2'</li><li>'WFDC21P'</li><li>'WHRN'</li><li>'WIF1'</li><li>'WIPF3'</li><li>'WNK2'</li><li>'WNK4'</li><li>'WNT10A'</li><li>'WNT10B'</li><li>'WNT11'</li><li>'WNT2'</li><li>'WNT4'</li><li>'WNT5A'</li><li>'WNT7B'</li><li>'WSCD1'</li><li>'WSCD2'</li><li>'WWC1'</li><li>'XDH'</li><li>'XG'</li><li>'XIST'</li><li>'XKR4'</li><li>'XPOTP1'</li><li>'Z84485.1'</li><li>'Z84492.1'</li><li>'ZAP70'</li><li>'ZBED1'</li><li>'ZBED2'</li><li>'ZBED6CL'</li><li>'ZBED9'</li><li>'ZBP1'</li><li>'ZCCHC12'</li><li>'ZDHHC19'</li><li>'ZDHHC8P1'</li><li>'ZFP57'</li><li>'ZFX'</li><li>'ZG16B'</li><li>'ZMYND15'</li><li>'ZNF165'</li><li>'ZNF185'</li><li>'ZNF215'</li><li>'ZNF296'</li><li>'ZNF334'</li><li>'ZNF33B'</li><li>'ZNF365'</li><li>'ZNF366'</li><li>'ZNF385B'</li><li>'ZNF385C'</li><li>'ZNF385D'</li><li>'ZNF469'</li><li>'ZNF497'</li><li>'ZNF536'</li><li>'ZNF552'</li><li>'ZNF608'</li><li>'ZNF667'</li><li>'ZNF689'</li><li>'ZNF711'</li><li>'ZNF726'</li><li>'ZNF750'</li><li>'ZNF774'</li><li>'ZNF812P'</li><li>'ZNF831'</li><li>'ZNF860'</li><li>'ZNF98'</li><li>'ZRSR2'</li><li>'ZYG11A'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item A2M\n",
       "\\item AADAC\n",
       "\\item AADAC\n",
       "\\item AADAC\n",
       "\\item AADACL3\n",
       "\\item AARD\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'A2M'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADACL3'\n",
       "\\item 'AARD'\n",
       "\\item 'AATBC'\n",
       "\\item 'AATK'\n",
       "\\item 'ABCA12'\n",
       "\\item 'ABCA13'\n",
       "\\item 'ABCA4'\n",
       "\\item 'ABCA8'\n",
       "\\item 'ABCB1'\n",
       "\\item 'ABCB4'\n",
       "\\item 'ABCB5'\n",
       "\\item 'ABCC11'\n",
       "\\item 'ABCC3'\n",
       "\\item 'ABCC6'\n",
       "\\item 'ABCC8'\n",
       "\\item 'ABCC9'\n",
       "\\item 'ABI3BP'\n",
       "\\item 'ABLIM2'\n",
       "\\item 'ABLIM3'\n",
       "\\item 'AC002066.1'\n",
       "\\item 'AC002546.1'\n",
       "\\item 'AC003986.2'\n",
       "\\item 'AC004130.2'\n",
       "\\item 'AC004160.1'\n",
       "\\item 'AC004233.2'\n",
       "\\item 'AC004540.1'\n",
       "\\item 'AC004947.1'\n",
       "\\item 'AC005064.1'\n",
       "\\item 'AC005077.4'\n",
       "\\item 'AC005082.1'\n",
       "\\item 'AC005083.1'\n",
       "\\item 'AC005162.2'\n",
       "\\item 'AC005165.1'\n",
       "\\item 'AC005323.1'\n",
       "\\item 'AC005336.1'\n",
       "\\item 'AC005550.2'\n",
       "\\item 'AC005674.2'\n",
       "\\item 'AC005831.1'\n",
       "\\item 'AC005906.2'\n",
       "\\item 'AC006946.2'\n",
       "\\item 'AC007423.1'\n",
       "\\item 'AC007666.1'\n",
       "\\item 'AC007743.1'\n",
       "\\item 'AC007906.2'\n",
       "\\item 'AC008014.1'\n",
       "\\item 'AC008035.1'\n",
       "\\item 'AC008105.1'\n",
       "\\item 'AC008268.1'\n",
       "\\item 'AC008592.4'\n",
       "\\item 'AC008763.1'\n",
       "\\item 'AC008894.3'\n",
       "\\item 'AC008957.3'\n",
       "\\item 'AC010203.1'\n",
       "\\item 'AC010329.1'\n",
       "\\item 'AC010457.1'\n",
       "\\item 'AC010503.5'\n",
       "\\item 'AC010655.3'\n",
       "\\item 'AC010735.2'\n",
       "\\item 'AC010761.4'\n",
       "\\item 'AC010976.2'\n",
       "\\item 'AC011008.2'\n",
       "\\item 'AC011337.1'\n",
       "\\item 'AC011352.1'\n",
       "\\item 'AC011476.3'\n",
       "\\item 'AC011503.1'\n",
       "\\item 'AC012313.2'\n",
       "\\item 'AC012368.1'\n",
       "\\item 'AC015712.1'\n",
       "\\item 'AC015712.2'\n",
       "\\item 'AC015712.4'\n",
       "\\item 'AC015802.6'\n",
       "\\item 'AC016705.2'\n",
       "\\item 'AC016738.1'\n",
       "\\item 'AC018553.1'\n",
       "\\item 'AC018629.1'\n",
       "\\item 'AC019205.1'\n",
       "\\item 'AC020604.1'\n",
       "\\item 'AC020636.1'\n",
       "\\item 'AC020659.1'\n",
       "\\item 'AC020912.1'\n",
       "\\item 'AC020951.1'\n",
       "\\item 'AC022034.1'\n",
       "\\item 'AC022144.1'\n",
       "\\item 'AC022148.1'\n",
       "\\item 'AC022497.1'\n",
       "\\item 'AC022706.1'\n",
       "\\item 'AC023301.1'\n",
       "\\item 'AC024560.2'\n",
       "\\item 'AC024940.2'\n",
       "\\item 'AC025569.1'\n",
       "\\item 'AC027117.1'\n",
       "\\item 'AC036108.2'\n",
       "\\item 'AC036108.3'\n",
       "\\item 'AC048341.1'\n",
       "\\item 'AC055854.1'\n",
       "\\item 'AC062004.1'\n",
       "\\item 'AC068473.3'\n",
       "\\item 'AC068587.4'\n",
       "\\item 'AC069148.1'\n",
       "\\item 'AC073342.2'\n",
       "\\item 'AC073850.1'\n",
       "\\item 'AC078850.1'\n",
       "\\item 'AC079313.2'\n",
       "\\item 'AC080013.1'\n",
       "\\item 'AC080013.4'\n",
       "\\item 'AC083843.3'\n",
       "\\item 'AC083843.4'\n",
       "\\item 'AC084871.1'\n",
       "\\item 'AC087482.1'\n",
       "\\item 'AC087521.1'\n",
       "\\item 'AC087623.1'\n",
       "\\item 'AC087741.1'\n",
       "\\item 'AC091182.1'\n",
       "\\item 'AC091544.4'\n",
       "\\item 'AC092164.1'\n",
       "\\item 'AC092171.5'\n",
       "\\item 'AC092490.1'\n",
       "\\item 'AC092535.5'\n",
       "\\item 'AC092746.1'\n",
       "\\item 'AC093001.1'\n",
       "\\item 'AC093159.1'\n",
       "\\item 'AC093162.3'\n",
       "\\item 'AC093278.2'\n",
       "\\item 'AC093425.1'\n",
       "\\item 'AC093627.4'\n",
       "\\item 'AC093627.6'\n",
       "\\item 'AC093642.2'\n",
       "\\item 'AC093866.1'\n",
       "\\item 'AC096677.1'\n",
       "\\item 'AC100793.4'\n",
       "\\item 'AC103740.1'\n",
       "\\item 'AC104024.2'\n",
       "\\item 'AC104118.1'\n",
       "\\item 'AC104260.2'\n",
       "\\item 'AC104695.4'\n",
       "\\item 'AC104809.2'\n",
       "\\item 'AC105391.1'\n",
       "\\item 'AC105942.1'\n",
       "\\item 'AC106881.1'\n",
       "\\item 'AC107204.1'\n",
       "\\item 'AC107464.2'\n",
       "\\item 'AC107884.1'\n",
       "\\item 'AC108058.1'\n",
       "\\item 'AC108134.1'\n",
       "\\item 'AC108472.1'\n",
       "\\item 'AC108704.2'\n",
       "\\item 'AC108860.2'\n",
       "\\item 'AC112206.2'\n",
       "\\item 'AC114811.2'\n",
       "\\item 'AC123912.2'\n",
       "\\item 'AC126323.1'\n",
       "\\item 'AC130466.1'\n",
       "\\item 'AC136475.3'\n",
       "\\item 'AC139769.1'\n",
       "\\item 'AC141557.1'\n",
       "\\item 'AC233266.2'\n",
       "\\item 'AC237221.1'\n",
       "\\item 'AC243772.2'\n",
       "\\item 'AC244453.2'\n",
       "\\item 'AC245041.2'\n",
       "\\item 'AC245060.5'\n",
       "\\item 'AC245100.2'\n",
       "\\item 'AC245100.4'\n",
       "\\item 'ACADL'\n",
       "\\item 'ACAN'\n",
       "\\item 'ACE2'\n",
       "\\item 'ACER2'\n",
       "\\item 'ACKR1'\n",
       "\\item 'ACKR4P1'\n",
       "\\item 'ACOT11'\n",
       "\\item 'ACOT4'\n",
       "\\item 'ACP5'\n",
       "\\item 'ACR'\n",
       "\\item 'ACSM1'\n",
       "\\item 'ACSM3'\n",
       "\\item 'ACSM5'\n",
       "\\item 'ACSM6'\n",
       "\\item 'ACTA2-AS1'\n",
       "\\item 'ACTC1'\n",
       "\\item 'ACTG2'\n",
       "\\item 'ACTN3'\n",
       "\\item 'ACTR3C'\n",
       "\\item 'ADA'\n",
       "\\item 'ADAD2'\n",
       "\\item 'ADAM23'\n",
       "\\item 'ADAM28'\n",
       "\\item 'ADAMDEC1'\n",
       "\\item 'ADAMTS14'\n",
       "\\item 'ADAMTS18'\n",
       "\\item 'ADAMTS2'\n",
       "\\item 'ADAMTS3'\n",
       "\\item 'ADAMTS4'\n",
       "\\item 'ADAMTS7'\n",
       "\\item 'ADAMTS9'\n",
       "\\item 'ADAMTS9-AS1'\n",
       "\\item 'ADAMTSL1'\n",
       "\\item 'ADAMTSL5'\n",
       "\\item 'ADARB2'\n",
       "\\item ⋯\n",
       "\\item 'TNFSF14'\n",
       "\\item 'TNFSF15'\n",
       "\\item 'TNIK'\n",
       "\\item 'TNK1'\n",
       "\\item 'TNMD'\n",
       "\\item 'TNN'\n",
       "\\item 'TNNI1'\n",
       "\\item 'TNNI2'\n",
       "\\item 'TNNT2'\n",
       "\\item 'TNS4'\n",
       "\\item 'TOX2'\n",
       "\\item 'TOX3'\n",
       "\\item 'TP53INP2'\n",
       "\\item 'TP63'\n",
       "\\item 'TP73'\n",
       "\\item 'TPH1'\n",
       "\\item 'TPO'\n",
       "\\item 'TPPP3'\n",
       "\\item 'TPRG1'\n",
       "\\item 'TPSAB1'\n",
       "\\item 'TPSB2'\n",
       "\\item 'TPT1P12'\n",
       "\\item 'TPTE2P5'\n",
       "\\item 'TPTEP1'\n",
       "\\item 'TRABD2B'\n",
       "\\item 'TRAC'\n",
       "\\item 'TRAF3IP3'\n",
       "\\item 'TRAF4'\n",
       "\\item 'TRARG1'\n",
       "\\item 'TRAT1'\n",
       "\\item 'TRBC1'\n",
       "\\item 'TRBC2'\n",
       "\\item 'TRDN'\n",
       "\\item 'TREM2'\n",
       "\\item 'TREML1'\n",
       "\\item 'TRG-AS1'\n",
       "\\item 'TRGC2'\n",
       "\\item 'TRHDE'\n",
       "\\item 'TRHDE-AS1'\n",
       "\\item 'TRIL'\n",
       "\\item 'TRIM17'\n",
       "\\item 'TRIM2'\n",
       "\\item 'TRIM29'\n",
       "\\item 'TRIM46'\n",
       "\\item 'TRIM55'\n",
       "\\item 'TRIM9'\n",
       "\\item 'TRMT9B'\n",
       "\\item 'TRPA1'\n",
       "\\item 'TRPC4'\n",
       "\\item 'TRPC6'\n",
       "\\item 'TRPC7'\n",
       "\\item 'TRPM2-AS'\n",
       "\\item 'TRPS1'\n",
       "\\item 'TRPV4'\n",
       "\\item 'TRPV6'\n",
       "\\item 'TSHB'\n",
       "\\item 'TSHZ2'\n",
       "\\item 'TSKS'\n",
       "\\item 'TSPAN1'\n",
       "\\item 'TSPAN11'\n",
       "\\item 'TSPAN13'\n",
       "\\item 'TSPAN33'\n",
       "\\item 'TSPAN5'\n",
       "\\item 'TSPAN7'\n",
       "\\item 'TSPAN8'\n",
       "\\item 'TSPEAR'\n",
       "\\item 'TSPEAR-AS1'\n",
       "\\item 'TSPEAR-AS2'\n",
       "\\item 'TSPOAP1'\n",
       "\\item 'TTC22'\n",
       "\\item 'TTC23L'\n",
       "\\item 'TTC6'\n",
       "\\item 'TTC9'\n",
       "\\item 'TTYH1'\n",
       "\\item 'TUBA3D'\n",
       "\\item 'TUBBP5'\n",
       "\\item 'TUFT1'\n",
       "\\item 'TUSC8'\n",
       "\\item 'TXLNG'\n",
       "\\item 'UBASH3B'\n",
       "\\item 'UBE2Q2P2'\n",
       "\\item 'UBXN10'\n",
       "\\item 'UCHL1'\n",
       "\\item 'UCN3'\n",
       "\\item 'UGT1A2P'\n",
       "\\item 'UGT2A1'\n",
       "\\item 'UGT2B17'\n",
       "\\item 'UGT3A2'\n",
       "\\item 'UGT8'\n",
       "\\item 'UNC13C'\n",
       "\\item 'UNC5B'\n",
       "\\item 'UNC5C'\n",
       "\\item 'UPK1B'\n",
       "\\item 'UPK3B'\n",
       "\\item 'UPP1'\n",
       "\\item 'USHBP1'\n",
       "\\item 'USP32P2'\n",
       "\\item 'USP43'\n",
       "\\item 'USP6'\n",
       "\\item 'UTS2R'\n",
       "\\item 'VAMP5'\n",
       "\\item 'VANGL2'\n",
       "\\item 'VASH2'\n",
       "\\item 'VAT1L'\n",
       "\\item 'VAV3'\n",
       "\\item 'VAX2'\n",
       "\\item 'VDR'\n",
       "\\item 'VEGFD'\n",
       "\\item 'VGF'\n",
       "\\item 'VGLL1'\n",
       "\\item 'VIL1'\n",
       "\\item 'VIM'\n",
       "\\item 'VIM-AS1'\n",
       "\\item 'VIPR2'\n",
       "\\item 'VIT'\n",
       "\\item 'VLDLR-AS1'\n",
       "\\item 'VMO1'\n",
       "\\item 'VN1R53P'\n",
       "\\item 'VNN2'\n",
       "\\item 'VPS9D1-AS1'\n",
       "\\item 'VSIG1'\n",
       "\\item 'VSIG4'\n",
       "\\item 'VSNL1'\n",
       "\\item 'VSTM2A'\n",
       "\\item 'VSTM5'\n",
       "\\item 'VTCN1'\n",
       "\\item 'VWC2'\n",
       "\\item 'VWCE'\n",
       "\\item 'VWDE'\n",
       "\\item 'VWF'\n",
       "\\item 'WDR62'\n",
       "\\item 'WDR86'\n",
       "\\item 'WDR86-AS1'\n",
       "\\item 'WEE1'\n",
       "\\item 'WFDC1'\n",
       "\\item 'WFDC2'\n",
       "\\item 'WFDC21P'\n",
       "\\item 'WHRN'\n",
       "\\item 'WIF1'\n",
       "\\item 'WIPF3'\n",
       "\\item 'WNK2'\n",
       "\\item 'WNK4'\n",
       "\\item 'WNT10A'\n",
       "\\item 'WNT10B'\n",
       "\\item 'WNT11'\n",
       "\\item 'WNT2'\n",
       "\\item 'WNT4'\n",
       "\\item 'WNT5A'\n",
       "\\item 'WNT7B'\n",
       "\\item 'WSCD1'\n",
       "\\item 'WSCD2'\n",
       "\\item 'WWC1'\n",
       "\\item 'XDH'\n",
       "\\item 'XG'\n",
       "\\item 'XIST'\n",
       "\\item 'XKR4'\n",
       "\\item 'XPOTP1'\n",
       "\\item 'Z84485.1'\n",
       "\\item 'Z84492.1'\n",
       "\\item 'ZAP70'\n",
       "\\item 'ZBED1'\n",
       "\\item 'ZBED2'\n",
       "\\item 'ZBED6CL'\n",
       "\\item 'ZBED9'\n",
       "\\item 'ZBP1'\n",
       "\\item 'ZCCHC12'\n",
       "\\item 'ZDHHC19'\n",
       "\\item 'ZDHHC8P1'\n",
       "\\item 'ZFP57'\n",
       "\\item 'ZFX'\n",
       "\\item 'ZG16B'\n",
       "\\item 'ZMYND15'\n",
       "\\item 'ZNF165'\n",
       "\\item 'ZNF185'\n",
       "\\item 'ZNF215'\n",
       "\\item 'ZNF296'\n",
       "\\item 'ZNF334'\n",
       "\\item 'ZNF33B'\n",
       "\\item 'ZNF365'\n",
       "\\item 'ZNF366'\n",
       "\\item 'ZNF385B'\n",
       "\\item 'ZNF385C'\n",
       "\\item 'ZNF385D'\n",
       "\\item 'ZNF469'\n",
       "\\item 'ZNF497'\n",
       "\\item 'ZNF536'\n",
       "\\item 'ZNF552'\n",
       "\\item 'ZNF608'\n",
       "\\item 'ZNF667'\n",
       "\\item 'ZNF689'\n",
       "\\item 'ZNF711'\n",
       "\\item 'ZNF726'\n",
       "\\item 'ZNF750'\n",
       "\\item 'ZNF774'\n",
       "\\item 'ZNF812P'\n",
       "\\item 'ZNF831'\n",
       "\\item 'ZNF860'\n",
       "\\item 'ZNF98'\n",
       "\\item 'ZRSR2'\n",
       "\\item 'ZYG11A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. A2M\n",
       "2. AADAC\n",
       "3. AADAC\n",
       "4. AADAC\n",
       "5. AADACL3\n",
       "6. AARD\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'A2M'\n",
       "2. 'AADAC'\n",
       "3. 'AADACL3'\n",
       "4. 'AARD'\n",
       "5. 'AATBC'\n",
       "6. 'AATK'\n",
       "7. 'ABCA12'\n",
       "8. 'ABCA13'\n",
       "9. 'ABCA4'\n",
       "10. 'ABCA8'\n",
       "11. 'ABCB1'\n",
       "12. 'ABCB4'\n",
       "13. 'ABCB5'\n",
       "14. 'ABCC11'\n",
       "15. 'ABCC3'\n",
       "16. 'ABCC6'\n",
       "17. 'ABCC8'\n",
       "18. 'ABCC9'\n",
       "19. 'ABI3BP'\n",
       "20. 'ABLIM2'\n",
       "21. 'ABLIM3'\n",
       "22. 'AC002066.1'\n",
       "23. 'AC002546.1'\n",
       "24. 'AC003986.2'\n",
       "25. 'AC004130.2'\n",
       "26. 'AC004160.1'\n",
       "27. 'AC004233.2'\n",
       "28. 'AC004540.1'\n",
       "29. 'AC004947.1'\n",
       "30. 'AC005064.1'\n",
       "31. 'AC005077.4'\n",
       "32. 'AC005082.1'\n",
       "33. 'AC005083.1'\n",
       "34. 'AC005162.2'\n",
       "35. 'AC005165.1'\n",
       "36. 'AC005323.1'\n",
       "37. 'AC005336.1'\n",
       "38. 'AC005550.2'\n",
       "39. 'AC005674.2'\n",
       "40. 'AC005831.1'\n",
       "41. 'AC005906.2'\n",
       "42. 'AC006946.2'\n",
       "43. 'AC007423.1'\n",
       "44. 'AC007666.1'\n",
       "45. 'AC007743.1'\n",
       "46. 'AC007906.2'\n",
       "47. 'AC008014.1'\n",
       "48. 'AC008035.1'\n",
       "49. 'AC008105.1'\n",
       "50. 'AC008268.1'\n",
       "51. 'AC008592.4'\n",
       "52. 'AC008763.1'\n",
       "53. 'AC008894.3'\n",
       "54. 'AC008957.3'\n",
       "55. 'AC010203.1'\n",
       "56. 'AC010329.1'\n",
       "57. 'AC010457.1'\n",
       "58. 'AC010503.5'\n",
       "59. 'AC010655.3'\n",
       "60. 'AC010735.2'\n",
       "61. 'AC010761.4'\n",
       "62. 'AC010976.2'\n",
       "63. 'AC011008.2'\n",
       "64. 'AC011337.1'\n",
       "65. 'AC011352.1'\n",
       "66. 'AC011476.3'\n",
       "67. 'AC011503.1'\n",
       "68. 'AC012313.2'\n",
       "69. 'AC012368.1'\n",
       "70. 'AC015712.1'\n",
       "71. 'AC015712.2'\n",
       "72. 'AC015712.4'\n",
       "73. 'AC015802.6'\n",
       "74. 'AC016705.2'\n",
       "75. 'AC016738.1'\n",
       "76. 'AC018553.1'\n",
       "77. 'AC018629.1'\n",
       "78. 'AC019205.1'\n",
       "79. 'AC020604.1'\n",
       "80. 'AC020636.1'\n",
       "81. 'AC020659.1'\n",
       "82. 'AC020912.1'\n",
       "83. 'AC020951.1'\n",
       "84. 'AC022034.1'\n",
       "85. 'AC022144.1'\n",
       "86. 'AC022148.1'\n",
       "87. 'AC022497.1'\n",
       "88. 'AC022706.1'\n",
       "89. 'AC023301.1'\n",
       "90. 'AC024560.2'\n",
       "91. 'AC024940.2'\n",
       "92. 'AC025569.1'\n",
       "93. 'AC027117.1'\n",
       "94. 'AC036108.2'\n",
       "95. 'AC036108.3'\n",
       "96. 'AC048341.1'\n",
       "97. 'AC055854.1'\n",
       "98. 'AC062004.1'\n",
       "99. 'AC068473.3'\n",
       "100. 'AC068587.4'\n",
       "101. 'AC069148.1'\n",
       "102. 'AC073342.2'\n",
       "103. 'AC073850.1'\n",
       "104. 'AC078850.1'\n",
       "105. 'AC079313.2'\n",
       "106. 'AC080013.1'\n",
       "107. 'AC080013.4'\n",
       "108. 'AC083843.3'\n",
       "109. 'AC083843.4'\n",
       "110. 'AC084871.1'\n",
       "111. 'AC087482.1'\n",
       "112. 'AC087521.1'\n",
       "113. 'AC087623.1'\n",
       "114. 'AC087741.1'\n",
       "115. 'AC091182.1'\n",
       "116. 'AC091544.4'\n",
       "117. 'AC092164.1'\n",
       "118. 'AC092171.5'\n",
       "119. 'AC092490.1'\n",
       "120. 'AC092535.5'\n",
       "121. 'AC092746.1'\n",
       "122. 'AC093001.1'\n",
       "123. 'AC093159.1'\n",
       "124. 'AC093162.3'\n",
       "125. 'AC093278.2'\n",
       "126. 'AC093425.1'\n",
       "127. 'AC093627.4'\n",
       "128. 'AC093627.6'\n",
       "129. 'AC093642.2'\n",
       "130. 'AC093866.1'\n",
       "131. 'AC096677.1'\n",
       "132. 'AC100793.4'\n",
       "133. 'AC103740.1'\n",
       "134. 'AC104024.2'\n",
       "135. 'AC104118.1'\n",
       "136. 'AC104260.2'\n",
       "137. 'AC104695.4'\n",
       "138. 'AC104809.2'\n",
       "139. 'AC105391.1'\n",
       "140. 'AC105942.1'\n",
       "141. 'AC106881.1'\n",
       "142. 'AC107204.1'\n",
       "143. 'AC107464.2'\n",
       "144. 'AC107884.1'\n",
       "145. 'AC108058.1'\n",
       "146. 'AC108134.1'\n",
       "147. 'AC108472.1'\n",
       "148. 'AC108704.2'\n",
       "149. 'AC108860.2'\n",
       "150. 'AC112206.2'\n",
       "151. 'AC114811.2'\n",
       "152. 'AC123912.2'\n",
       "153. 'AC126323.1'\n",
       "154. 'AC130466.1'\n",
       "155. 'AC136475.3'\n",
       "156. 'AC139769.1'\n",
       "157. 'AC141557.1'\n",
       "158. 'AC233266.2'\n",
       "159. 'AC237221.1'\n",
       "160. 'AC243772.2'\n",
       "161. 'AC244453.2'\n",
       "162. 'AC245041.2'\n",
       "163. 'AC245060.5'\n",
       "164. 'AC245100.2'\n",
       "165. 'AC245100.4'\n",
       "166. 'ACADL'\n",
       "167. 'ACAN'\n",
       "168. 'ACE2'\n",
       "169. 'ACER2'\n",
       "170. 'ACKR1'\n",
       "171. 'ACKR4P1'\n",
       "172. 'ACOT11'\n",
       "173. 'ACOT4'\n",
       "174. 'ACP5'\n",
       "175. 'ACR'\n",
       "176. 'ACSM1'\n",
       "177. 'ACSM3'\n",
       "178. 'ACSM5'\n",
       "179. 'ACSM6'\n",
       "180. 'ACTA2-AS1'\n",
       "181. 'ACTC1'\n",
       "182. 'ACTG2'\n",
       "183. 'ACTN3'\n",
       "184. 'ACTR3C'\n",
       "185. 'ADA'\n",
       "186. 'ADAD2'\n",
       "187. 'ADAM23'\n",
       "188. 'ADAM28'\n",
       "189. 'ADAMDEC1'\n",
       "190. 'ADAMTS14'\n",
       "191. 'ADAMTS18'\n",
       "192. 'ADAMTS2'\n",
       "193. 'ADAMTS3'\n",
       "194. 'ADAMTS4'\n",
       "195. 'ADAMTS7'\n",
       "196. 'ADAMTS9'\n",
       "197. 'ADAMTS9-AS1'\n",
       "198. 'ADAMTSL1'\n",
       "199. 'ADAMTSL5'\n",
       "200. 'ADARB2'\n",
       "201. ⋯\n",
       "202. 'TNFSF14'\n",
       "203. 'TNFSF15'\n",
       "204. 'TNIK'\n",
       "205. 'TNK1'\n",
       "206. 'TNMD'\n",
       "207. 'TNN'\n",
       "208. 'TNNI1'\n",
       "209. 'TNNI2'\n",
       "210. 'TNNT2'\n",
       "211. 'TNS4'\n",
       "212. 'TOX2'\n",
       "213. 'TOX3'\n",
       "214. 'TP53INP2'\n",
       "215. 'TP63'\n",
       "216. 'TP73'\n",
       "217. 'TPH1'\n",
       "218. 'TPO'\n",
       "219. 'TPPP3'\n",
       "220. 'TPRG1'\n",
       "221. 'TPSAB1'\n",
       "222. 'TPSB2'\n",
       "223. 'TPT1P12'\n",
       "224. 'TPTE2P5'\n",
       "225. 'TPTEP1'\n",
       "226. 'TRABD2B'\n",
       "227. 'TRAC'\n",
       "228. 'TRAF3IP3'\n",
       "229. 'TRAF4'\n",
       "230. 'TRARG1'\n",
       "231. 'TRAT1'\n",
       "232. 'TRBC1'\n",
       "233. 'TRBC2'\n",
       "234. 'TRDN'\n",
       "235. 'TREM2'\n",
       "236. 'TREML1'\n",
       "237. 'TRG-AS1'\n",
       "238. 'TRGC2'\n",
       "239. 'TRHDE'\n",
       "240. 'TRHDE-AS1'\n",
       "241. 'TRIL'\n",
       "242. 'TRIM17'\n",
       "243. 'TRIM2'\n",
       "244. 'TRIM29'\n",
       "245. 'TRIM46'\n",
       "246. 'TRIM55'\n",
       "247. 'TRIM9'\n",
       "248. 'TRMT9B'\n",
       "249. 'TRPA1'\n",
       "250. 'TRPC4'\n",
       "251. 'TRPC6'\n",
       "252. 'TRPC7'\n",
       "253. 'TRPM2-AS'\n",
       "254. 'TRPS1'\n",
       "255. 'TRPV4'\n",
       "256. 'TRPV6'\n",
       "257. 'TSHB'\n",
       "258. 'TSHZ2'\n",
       "259. 'TSKS'\n",
       "260. 'TSPAN1'\n",
       "261. 'TSPAN11'\n",
       "262. 'TSPAN13'\n",
       "263. 'TSPAN33'\n",
       "264. 'TSPAN5'\n",
       "265. 'TSPAN7'\n",
       "266. 'TSPAN8'\n",
       "267. 'TSPEAR'\n",
       "268. 'TSPEAR-AS1'\n",
       "269. 'TSPEAR-AS2'\n",
       "270. 'TSPOAP1'\n",
       "271. 'TTC22'\n",
       "272. 'TTC23L'\n",
       "273. 'TTC6'\n",
       "274. 'TTC9'\n",
       "275. 'TTYH1'\n",
       "276. 'TUBA3D'\n",
       "277. 'TUBBP5'\n",
       "278. 'TUFT1'\n",
       "279. 'TUSC8'\n",
       "280. 'TXLNG'\n",
       "281. 'UBASH3B'\n",
       "282. 'UBE2Q2P2'\n",
       "283. 'UBXN10'\n",
       "284. 'UCHL1'\n",
       "285. 'UCN3'\n",
       "286. 'UGT1A2P'\n",
       "287. 'UGT2A1'\n",
       "288. 'UGT2B17'\n",
       "289. 'UGT3A2'\n",
       "290. 'UGT8'\n",
       "291. 'UNC13C'\n",
       "292. 'UNC5B'\n",
       "293. 'UNC5C'\n",
       "294. 'UPK1B'\n",
       "295. 'UPK3B'\n",
       "296. 'UPP1'\n",
       "297. 'USHBP1'\n",
       "298. 'USP32P2'\n",
       "299. 'USP43'\n",
       "300. 'USP6'\n",
       "301. 'UTS2R'\n",
       "302. 'VAMP5'\n",
       "303. 'VANGL2'\n",
       "304. 'VASH2'\n",
       "305. 'VAT1L'\n",
       "306. 'VAV3'\n",
       "307. 'VAX2'\n",
       "308. 'VDR'\n",
       "309. 'VEGFD'\n",
       "310. 'VGF'\n",
       "311. 'VGLL1'\n",
       "312. 'VIL1'\n",
       "313. 'VIM'\n",
       "314. 'VIM-AS1'\n",
       "315. 'VIPR2'\n",
       "316. 'VIT'\n",
       "317. 'VLDLR-AS1'\n",
       "318. 'VMO1'\n",
       "319. 'VN1R53P'\n",
       "320. 'VNN2'\n",
       "321. 'VPS9D1-AS1'\n",
       "322. 'VSIG1'\n",
       "323. 'VSIG4'\n",
       "324. 'VSNL1'\n",
       "325. 'VSTM2A'\n",
       "326. 'VSTM5'\n",
       "327. 'VTCN1'\n",
       "328. 'VWC2'\n",
       "329. 'VWCE'\n",
       "330. 'VWDE'\n",
       "331. 'VWF'\n",
       "332. 'WDR62'\n",
       "333. 'WDR86'\n",
       "334. 'WDR86-AS1'\n",
       "335. 'WEE1'\n",
       "336. 'WFDC1'\n",
       "337. 'WFDC2'\n",
       "338. 'WFDC21P'\n",
       "339. 'WHRN'\n",
       "340. 'WIF1'\n",
       "341. 'WIPF3'\n",
       "342. 'WNK2'\n",
       "343. 'WNK4'\n",
       "344. 'WNT10A'\n",
       "345. 'WNT10B'\n",
       "346. 'WNT11'\n",
       "347. 'WNT2'\n",
       "348. 'WNT4'\n",
       "349. 'WNT5A'\n",
       "350. 'WNT7B'\n",
       "351. 'WSCD1'\n",
       "352. 'WSCD2'\n",
       "353. 'WWC1'\n",
       "354. 'XDH'\n",
       "355. 'XG'\n",
       "356. 'XIST'\n",
       "357. 'XKR4'\n",
       "358. 'XPOTP1'\n",
       "359. 'Z84485.1'\n",
       "360. 'Z84492.1'\n",
       "361. 'ZAP70'\n",
       "362. 'ZBED1'\n",
       "363. 'ZBED2'\n",
       "364. 'ZBED6CL'\n",
       "365. 'ZBED9'\n",
       "366. 'ZBP1'\n",
       "367. 'ZCCHC12'\n",
       "368. 'ZDHHC19'\n",
       "369. 'ZDHHC8P1'\n",
       "370. 'ZFP57'\n",
       "371. 'ZFX'\n",
       "372. 'ZG16B'\n",
       "373. 'ZMYND15'\n",
       "374. 'ZNF165'\n",
       "375. 'ZNF185'\n",
       "376. 'ZNF215'\n",
       "377. 'ZNF296'\n",
       "378. 'ZNF334'\n",
       "379. 'ZNF33B'\n",
       "380. 'ZNF365'\n",
       "381. 'ZNF366'\n",
       "382. 'ZNF385B'\n",
       "383. 'ZNF385C'\n",
       "384. 'ZNF385D'\n",
       "385. 'ZNF469'\n",
       "386. 'ZNF497'\n",
       "387. 'ZNF536'\n",
       "388. 'ZNF552'\n",
       "389. 'ZNF608'\n",
       "390. 'ZNF667'\n",
       "391. 'ZNF689'\n",
       "392. 'ZNF711'\n",
       "393. 'ZNF726'\n",
       "394. 'ZNF750'\n",
       "395. 'ZNF774'\n",
       "396. 'ZNF812P'\n",
       "397. 'ZNF831'\n",
       "398. 'ZNF860'\n",
       "399. 'ZNF98'\n",
       "400. 'ZRSR2'\n",
       "401. 'ZYG11A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] A2M     AADAC   AADAC   AADAC   AADACL3 AARD   \n",
       "3221 Levels: A2M AADAC AADACL3 AARD AATBC AATK ABCA12 ABCA13 ABCA4 ... ZYG11A"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_AS_Genes <- read.table(file=\"../data/Total_AS_by_geneSymbol.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "sigAsGenes <- sort(total_AS_Genes$GeneSymbol)\n",
    "dge <- read.table(\"../data/gene_dge.tsv\", sep = \"\\t\", header = FALSE, row.names=1, skip = 1)\n",
    "dge_genes <- sort(dge$V5)\n",
    "head(dge_genes)\n",
    "all_genes_data <- read.table(\"../assets/all_gene_dge.tsv\")\n",
    "names(all_genes_data) <- c(\"GeneSymbol\", \"ensg\")\n",
    "all_genes <- sort(all_genes_data$GeneSymbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>ENSG00000000003.14</li><li>ENSG00000000005.5</li><li>ENSG00000000419.12</li><li>ENSG00000000457.13</li><li>ENSG00000000460.16</li><li>ENSG00000000938.12</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'ENSG00000000003.14'</li><li>'ENSG00000000005.5'</li><li>'ENSG00000000419.12'</li><li>'ENSG00000000457.13'</li><li>'ENSG00000000460.16'</li><li>'ENSG00000000938.12'</li><li>'ENSG00000000971.15'</li><li>'ENSG00000001036.13'</li><li>'ENSG00000001084.10'</li><li>'ENSG00000001167.14'</li><li>'ENSG00000001460.17'</li><li>'ENSG00000001461.16'</li><li>'ENSG00000001497.16'</li><li>'ENSG00000001561.6'</li><li>'ENSG00000001617.11'</li><li>'ENSG00000001626.14'</li><li>'ENSG00000001629.9'</li><li>'ENSG00000001630.15'</li><li>'ENSG00000001631.15'</li><li>'ENSG00000002016.17'</li><li>'ENSG00000002330.13'</li><li>'ENSG00000002549.12'</li><li>'ENSG00000002586.18'</li><li>'ENSG00000002587.9'</li><li>'ENSG00000002726.20'</li><li>'ENSG00000002745.12'</li><li>'ENSG00000002746.14'</li><li>'ENSG00000002822.15'</li><li>'ENSG00000002834.17'</li><li>'ENSG00000002919.14'</li><li>'ENSG00000002933.7'</li><li>'ENSG00000003056.7'</li><li>'ENSG00000003096.14'</li><li>'ENSG00000003137.8'</li><li>'ENSG00000003147.17'</li><li>'ENSG00000003249.13'</li><li>'ENSG00000003393.14'</li><li>'ENSG00000003400.14'</li><li>'ENSG00000003402.19'</li><li>'ENSG00000003436.15'</li><li>'ENSG00000003509.15'</li><li>'ENSG00000003756.16'</li><li>'ENSG00000003987.13'</li><li>'ENSG00000003989.17'</li><li>'ENSG00000004059.10'</li><li>'ENSG00000004139.13'</li><li>'ENSG00000004142.11'</li><li>'ENSG00000004399.12'</li><li>'ENSG00000004455.16'</li><li>'ENSG00000004468.12'</li><li>'ENSG00000004478.7'</li><li>'ENSG00000004487.15'</li><li>'ENSG00000004534.14'</li><li>'ENSG00000004660.14'</li><li>'ENSG00000004700.15'</li><li>'ENSG00000004766.15'</li><li>'ENSG00000004776.12'</li><li>'ENSG00000004777.18'</li><li>'ENSG00000004779.9'</li><li>'ENSG00000004799.7'</li><li>'ENSG00000004809.13'</li><li>'ENSG00000004838.13'</li><li>'ENSG00000004846.16'</li><li>'ENSG00000004848.7'</li><li>'ENSG00000004864.13'</li><li>'ENSG00000004866.20'</li><li>'ENSG00000004897.11'</li><li>'ENSG00000004939.13'</li><li>'ENSG00000004948.13'</li><li>'ENSG00000004961.14'</li><li>'ENSG00000004975.11'</li><li>'ENSG00000005001.9'</li><li>'ENSG00000005007.12'</li><li>'ENSG00000005020.12'</li><li>'ENSG00000005022.5'</li><li>'ENSG00000005059.15'</li><li>'ENSG00000005073.5'</li><li>'ENSG00000005075.15'</li><li>'ENSG00000005100.12'</li><li>'ENSG00000005102.12'</li><li>'ENSG00000005108.15'</li><li>'ENSG00000005156.11'</li><li>'ENSG00000005175.9'</li><li>'ENSG00000005187.11'</li><li>'ENSG00000005189.19'</li><li>'ENSG00000005194.14'</li><li>'ENSG00000005206.16'</li><li>'ENSG00000005238.19'</li><li>'ENSG00000005243.9'</li><li>'ENSG00000005249.12'</li><li>'ENSG00000005302.17'</li><li>'ENSG00000005339.14'</li><li>'ENSG00000005379.15'</li><li>'ENSG00000005381.7'</li><li>'ENSG00000005421.8'</li><li>'ENSG00000005436.13'</li><li>'ENSG00000005448.16'</li><li>'ENSG00000005469.11'</li><li>'ENSG00000005471.15'</li><li>'ENSG00000005483.20'</li><li>'ENSG00000005486.16'</li><li>'ENSG00000005513.9'</li><li>'ENSG00000005700.14'</li><li>'ENSG00000005801.16'</li><li>'ENSG00000005810.17'</li><li>'ENSG00000005812.10'</li><li>'ENSG00000005844.17'</li><li>'ENSG00000005882.11'</li><li>'ENSG00000005884.17'</li><li>'ENSG00000005889.15'</li><li>'ENSG00000005893.15'</li><li>'ENSG00000005961.17'</li><li>'ENSG00000005981.12'</li><li>'ENSG00000006007.11'</li><li>'ENSG00000006015.17'</li><li>'ENSG00000006016.10'</li><li>'ENSG00000006025.11'</li><li>'ENSG00000006042.11'</li><li>'ENSG00000006047.12'</li><li>'ENSG00000006059.3'</li><li>'ENSG00000006062.14'</li><li>'ENSG00000006071.12'</li><li>'ENSG00000006116.3'</li><li>'ENSG00000006118.14'</li><li>'ENSG00000006125.17'</li><li>'ENSG00000006128.11'</li><li>'ENSG00000006194.9'</li><li>'ENSG00000006210.6'</li><li>'ENSG00000006282.20'</li><li>'ENSG00000006283.17'</li><li>'ENSG00000006327.13'</li><li>'ENSG00000006377.10'</li><li>'ENSG00000006432.15'</li><li>'ENSG00000006451.7'</li><li>'ENSG00000006453.13'</li><li>'ENSG00000006459.10'</li><li>'ENSG00000006468.13'</li><li>'ENSG00000006530.15'</li><li>'ENSG00000006534.15'</li><li>'ENSG00000006555.10'</li><li>'ENSG00000006576.16'</li><li>'ENSG00000006606.8'</li><li>'ENSG00000006607.13'</li><li>'ENSG00000006611.15'</li><li>'ENSG00000006625.17'</li><li>'ENSG00000006634.7'</li><li>'ENSG00000006638.11'</li><li>'ENSG00000006652.13'</li><li>'ENSG00000006659.12'</li><li>'ENSG00000006695.10'</li><li>'ENSG00000006704.10'</li><li>'ENSG00000006712.14'</li><li>'ENSG00000006715.15'</li><li>'ENSG00000006740.16'</li><li>'ENSG00000006744.18'</li><li>'ENSG00000006747.14'</li><li>'ENSG00000006756.15'</li><li>'ENSG00000006757.11'</li><li>'ENSG00000006788.13'</li><li>'ENSG00000006831.9'</li><li>'ENSG00000006837.11'</li><li>'ENSG00000007001.12'</li><li>'ENSG00000007038.10'</li><li>'ENSG00000007047.14'</li><li>'ENSG00000007062.11'</li><li>'ENSG00000007080.10'</li><li>'ENSG00000007129.17'</li><li>'ENSG00000007168.12'</li><li>'ENSG00000007171.16'</li><li>'ENSG00000007174.17'</li><li>'ENSG00000007202.14'</li><li>'ENSG00000007216.14'</li><li>'ENSG00000007237.18'</li><li>'ENSG00000007255.10'</li><li>'ENSG00000007264.14'</li><li>'ENSG00000007306.14'</li><li>'ENSG00000007312.12'</li><li>'ENSG00000007314.12'</li><li>'ENSG00000007341.18'</li><li>'ENSG00000007350.16'</li><li>'ENSG00000007372.21'</li><li>'ENSG00000007376.7'</li><li>'ENSG00000007384.15'</li><li>'ENSG00000007392.16'</li><li>'ENSG00000007402.11'</li><li>'ENSG00000007516.13'</li><li>'ENSG00000007520.3'</li><li>'ENSG00000007541.16'</li><li>'ENSG00000007545.15'</li><li>'ENSG00000007866.19'</li><li>'ENSG00000007908.15'</li><li>'ENSG00000007923.15'</li><li>'ENSG00000007933.12'</li><li>'ENSG00000007944.14'</li><li>'ENSG00000007952.17'</li><li>'ENSG00000007968.6'</li><li>'ENSG00000008018.8'</li><li>'ENSG00000008056.13'</li><li>'ENSG00000008083.13'</li><li>'ENSG00000008086.11'</li><li>⋯</li><li>'ENSG00000280623.1'</li><li>'ENSG00000280639.1'</li><li>'ENSG00000280649.2'</li><li>'ENSG00000280650.1'</li><li>'ENSG00000280660.1'</li><li>'ENSG00000280670.2'</li><li>'ENSG00000280693.2'</li><li>'ENSG00000280721.1'</li><li>'ENSG00000280734.2'</li><li>'ENSG00000280739.2'</li><li>'ENSG00000280758.1'</li><li>'ENSG00000280776.1'</li><li>'ENSG00000280789.1'</li><li>'ENSG00000280798.1'</li><li>'ENSG00000280809.2'</li><li>'ENSG00000280828.1'</li><li>'ENSG00000280832.1'</li><li>'ENSG00000280837.1'</li><li>'ENSG00000280870.2'</li><li>'ENSG00000280913.2'</li><li>'ENSG00000280916.1'</li><li>'ENSG00000280927.1'</li><li>'ENSG00000280936.1'</li><li>'ENSG00000281005.1'</li><li>'ENSG00000281026.1'</li><li>'ENSG00000281097.1'</li><li>'ENSG00000281100.1'</li><li>'ENSG00000281103.1'</li><li>'ENSG00000281106.2'</li><li>'ENSG00000281128.1'</li><li>'ENSG00000281160.1'</li><li>'ENSG00000281162.2'</li><li>'ENSG00000281181.1'</li><li>'ENSG00000281183.1'</li><li>'ENSG00000281186.1'</li><li>'ENSG00000281195.1'</li><li>'ENSG00000281205.1'</li><li>'ENSG00000281207.1'</li><li>'ENSG00000281327.1'</li><li>'ENSG00000281332.1'</li><li>'ENSG00000281344.1'</li><li>'ENSG00000281357.2'</li><li>'ENSG00000281358.1'</li><li>'ENSG00000281376.1'</li><li>'ENSG00000281383.1'</li><li>'ENSG00000281386.1'</li><li>'ENSG00000281392.1'</li><li>'ENSG00000281398.2'</li><li>'ENSG00000281404.1'</li><li>'ENSG00000281406.1'</li><li>'ENSG00000281468.1'</li><li>'ENSG00000281490.1'</li><li>'ENSG00000281501.1'</li><li>'ENSG00000281530.1'</li><li>'ENSG00000281538.1'</li><li>'ENSG00000281571.2'</li><li>'ENSG00000281649.1'</li><li>'ENSG00000281655.1'</li><li>'ENSG00000281692.1'</li><li>'ENSG00000281706.2'</li><li>'ENSG00000281721.1'</li><li>'ENSG00000281732.1'</li><li>'ENSG00000281756.1'</li><li>'ENSG00000281769.1'</li><li>'ENSG00000281831.1'</li><li>'ENSG00000281852.1'</li><li>'ENSG00000281880.1'</li><li>'ENSG00000281896.1'</li><li>'ENSG00000281903.2'</li><li>'ENSG00000281912.1'</li><li>'ENSG00000281969.1'</li><li>'ENSG00000281974.1'</li><li>'ENSG00000281990.1'</li><li>'ENSG00000281991.1'</li><li>'ENSG00000282012.1'</li><li>'ENSG00000282022.1'</li><li>'ENSG00000282024.1'</li><li>'ENSG00000282033.1'</li><li>'ENSG00000282057.1'</li><li>'ENSG00000282122.1'</li><li>'ENSG00000282164.2'</li><li>'ENSG00000282221.1'</li><li>'ENSG00000282375.1'</li><li>'ENSG00000282386.1'</li><li>'ENSG00000282390.1'</li><li>'ENSG00000282416.1'</li><li>'ENSG00000282419.2'</li><li>'ENSG00000282458.1'</li><li>'ENSG00000282508.1'</li><li>'ENSG00000282556.1'</li><li>'ENSG00000282572.2'</li><li>'ENSG00000282600.2'</li><li>'ENSG00000282608.1'</li><li>'ENSG00000282639.1'</li><li>'ENSG00000282651.2'</li><li>'ENSG00000282728.1'</li><li>'ENSG00000282826.1'</li><li>'ENSG00000282840.1'</li><li>'ENSG00000282851.1'</li><li>'ENSG00000282881.1'</li><li>'ENSG00000282885.1'</li><li>'ENSG00000282890.1'</li><li>'ENSG00000282917.1'</li><li>'ENSG00000282933.2'</li><li>'ENSG00000282936.1'</li><li>'ENSG00000282939.1'</li><li>'ENSG00000282943.1'</li><li>'ENSG00000282951.1'</li><li>'ENSG00000283000.1'</li><li>'ENSG00000283003.1'</li><li>'ENSG00000283025.1'</li><li>'ENSG00000283031.1'</li><li>'ENSG00000283041.1'</li><li>'ENSG00000283050.1'</li><li>'ENSG00000283051.1'</li><li>'ENSG00000283063.1'</li><li>'ENSG00000283071.1'</li><li>'ENSG00000283078.1'</li><li>'ENSG00000283095.1'</li><li>'ENSG00000283098.1'</li><li>'ENSG00000283103.1'</li><li>'ENSG00000283122.1'</li><li>'ENSG00000283132.1'</li><li>'ENSG00000283154.2'</li><li>'ENSG00000283156.1'</li><li>'ENSG00000283162.1'</li><li>'ENSG00000283175.1'</li><li>'ENSG00000283178.1'</li><li>'ENSG00000283183.1'</li><li>'ENSG00000283196.1'</li><li>'ENSG00000283199.1'</li><li>'ENSG00000283213.1'</li><li>'ENSG00000283227.1'</li><li>'ENSG00000283235.1'</li><li>'ENSG00000283236.1'</li><li>'ENSG00000283267.1'</li><li>'ENSG00000283270.1'</li><li>'ENSG00000283288.1'</li><li>'ENSG00000283297.1'</li><li>'ENSG00000283307.1'</li><li>'ENSG00000283312.1'</li><li>'ENSG00000283342.1'</li><li>'ENSG00000283355.1'</li><li>'ENSG00000283361.1'</li><li>'ENSG00000283366.1'</li><li>'ENSG00000283375.1'</li><li>'ENSG00000283426.1'</li><li>'ENSG00000283440.1'</li><li>'ENSG00000283445.1'</li><li>'ENSG00000283458.1'</li><li>'ENSG00000283462.1'</li><li>'ENSG00000283503.1'</li><li>'ENSG00000283518.1'</li><li>'ENSG00000283525.1'</li><li>'ENSG00000283526.1'</li><li>'ENSG00000283538.1'</li><li>'ENSG00000283554.1'</li><li>'ENSG00000283559.1'</li><li>'ENSG00000283597.2'</li><li>'ENSG00000283602.1'</li><li>'ENSG00000283632.2'</li><li>'ENSG00000283633.1'</li><li>'ENSG00000283646.1'</li><li>'ENSG00000283662.1'</li><li>'ENSG00000283667.1'</li><li>'ENSG00000283674.1'</li><li>'ENSG00000283689.1'</li><li>'ENSG00000283692.1'</li><li>'ENSG00000283696.1'</li><li>'ENSG00000283703.1'</li><li>'ENSG00000283709.1'</li><li>'ENSG00000283766.1'</li><li>'ENSG00000283787.1'</li><li>'ENSG00000283828.1'</li><li>'ENSG00000283839.1'</li><li>'ENSG00000283886.1'</li><li>'ENSG00000283897.1'</li><li>'ENSG00000283906.1'</li><li>'ENSG00000283959.1'</li><li>'ENSG00000283992.1'</li><li>'ENSG00000283994.1'</li><li>'ENSG00000284024.1'</li><li>'ENSG00000284048.1'</li><li>'ENSG00000284052.1'</li><li>'ENSG00000284070.1'</li><li>'ENSG00000284128.1'</li><li>'ENSG00000284130.1'</li><li>'ENSG00000284196.1'</li><li>'ENSG00000284237.1'</li><li>'ENSG00000284308.1'</li><li>'ENSG00000284309.1'</li><li>'ENSG00000284377.1'</li><li>'ENSG00000284391.1'</li><li>'ENSG00000284413.1'</li><li>'ENSG00000284428.1'</li><li>'ENSG00000284430.1'</li><li>'ENSG00000284471.1'</li><li>'ENSG00000284526.1'</li><li>'ENSG00000284543.1'</li><li>'ENSG00000284600.1'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item ENSG00000000003.14\n",
       "\\item ENSG00000000005.5\n",
       "\\item ENSG00000000419.12\n",
       "\\item ENSG00000000457.13\n",
       "\\item ENSG00000000460.16\n",
       "\\item ENSG00000000938.12\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'ENSG00000000003.14'\n",
       "\\item 'ENSG00000000005.5'\n",
       "\\item 'ENSG00000000419.12'\n",
       "\\item 'ENSG00000000457.13'\n",
       "\\item 'ENSG00000000460.16'\n",
       "\\item 'ENSG00000000938.12'\n",
       "\\item 'ENSG00000000971.15'\n",
       "\\item 'ENSG00000001036.13'\n",
       "\\item 'ENSG00000001084.10'\n",
       "\\item 'ENSG00000001167.14'\n",
       "\\item 'ENSG00000001460.17'\n",
       "\\item 'ENSG00000001461.16'\n",
       "\\item 'ENSG00000001497.16'\n",
       "\\item 'ENSG00000001561.6'\n",
       "\\item 'ENSG00000001617.11'\n",
       "\\item 'ENSG00000001626.14'\n",
       "\\item 'ENSG00000001629.9'\n",
       "\\item 'ENSG00000001630.15'\n",
       "\\item 'ENSG00000001631.15'\n",
       "\\item 'ENSG00000002016.17'\n",
       "\\item 'ENSG00000002330.13'\n",
       "\\item 'ENSG00000002549.12'\n",
       "\\item 'ENSG00000002586.18'\n",
       "\\item 'ENSG00000002587.9'\n",
       "\\item 'ENSG00000002726.20'\n",
       "\\item 'ENSG00000002745.12'\n",
       "\\item 'ENSG00000002746.14'\n",
       "\\item 'ENSG00000002822.15'\n",
       "\\item 'ENSG00000002834.17'\n",
       "\\item 'ENSG00000002919.14'\n",
       "\\item 'ENSG00000002933.7'\n",
       "\\item 'ENSG00000003056.7'\n",
       "\\item 'ENSG00000003096.14'\n",
       "\\item 'ENSG00000003137.8'\n",
       "\\item 'ENSG00000003147.17'\n",
       "\\item 'ENSG00000003249.13'\n",
       "\\item 'ENSG00000003393.14'\n",
       "\\item 'ENSG00000003400.14'\n",
       "\\item 'ENSG00000003402.19'\n",
       "\\item 'ENSG00000003436.15'\n",
       "\\item 'ENSG00000003509.15'\n",
       "\\item 'ENSG00000003756.16'\n",
       "\\item 'ENSG00000003987.13'\n",
       "\\item 'ENSG00000003989.17'\n",
       "\\item 'ENSG00000004059.10'\n",
       "\\item 'ENSG00000004139.13'\n",
       "\\item 'ENSG00000004142.11'\n",
       "\\item 'ENSG00000004399.12'\n",
       "\\item 'ENSG00000004455.16'\n",
       "\\item 'ENSG00000004468.12'\n",
       "\\item 'ENSG00000004478.7'\n",
       "\\item 'ENSG00000004487.15'\n",
       "\\item 'ENSG00000004534.14'\n",
       "\\item 'ENSG00000004660.14'\n",
       "\\item 'ENSG00000004700.15'\n",
       "\\item 'ENSG00000004766.15'\n",
       "\\item 'ENSG00000004776.12'\n",
       "\\item 'ENSG00000004777.18'\n",
       "\\item 'ENSG00000004779.9'\n",
       "\\item 'ENSG00000004799.7'\n",
       "\\item 'ENSG00000004809.13'\n",
       "\\item 'ENSG00000004838.13'\n",
       "\\item 'ENSG00000004846.16'\n",
       "\\item 'ENSG00000004848.7'\n",
       "\\item 'ENSG00000004864.13'\n",
       "\\item 'ENSG00000004866.20'\n",
       "\\item 'ENSG00000004897.11'\n",
       "\\item 'ENSG00000004939.13'\n",
       "\\item 'ENSG00000004948.13'\n",
       "\\item 'ENSG00000004961.14'\n",
       "\\item 'ENSG00000004975.11'\n",
       "\\item 'ENSG00000005001.9'\n",
       "\\item 'ENSG00000005007.12'\n",
       "\\item 'ENSG00000005020.12'\n",
       "\\item 'ENSG00000005022.5'\n",
       "\\item 'ENSG00000005059.15'\n",
       "\\item 'ENSG00000005073.5'\n",
       "\\item 'ENSG00000005075.15'\n",
       "\\item 'ENSG00000005100.12'\n",
       "\\item 'ENSG00000005102.12'\n",
       "\\item 'ENSG00000005108.15'\n",
       "\\item 'ENSG00000005156.11'\n",
       "\\item 'ENSG00000005175.9'\n",
       "\\item 'ENSG00000005187.11'\n",
       "\\item 'ENSG00000005189.19'\n",
       "\\item 'ENSG00000005194.14'\n",
       "\\item 'ENSG00000005206.16'\n",
       "\\item 'ENSG00000005238.19'\n",
       "\\item 'ENSG00000005243.9'\n",
       "\\item 'ENSG00000005249.12'\n",
       "\\item 'ENSG00000005302.17'\n",
       "\\item 'ENSG00000005339.14'\n",
       "\\item 'ENSG00000005379.15'\n",
       "\\item 'ENSG00000005381.7'\n",
       "\\item 'ENSG00000005421.8'\n",
       "\\item 'ENSG00000005436.13'\n",
       "\\item 'ENSG00000005448.16'\n",
       "\\item 'ENSG00000005469.11'\n",
       "\\item 'ENSG00000005471.15'\n",
       "\\item 'ENSG00000005483.20'\n",
       "\\item 'ENSG00000005486.16'\n",
       "\\item 'ENSG00000005513.9'\n",
       "\\item 'ENSG00000005700.14'\n",
       "\\item 'ENSG00000005801.16'\n",
       "\\item 'ENSG00000005810.17'\n",
       "\\item 'ENSG00000005812.10'\n",
       "\\item 'ENSG00000005844.17'\n",
       "\\item 'ENSG00000005882.11'\n",
       "\\item 'ENSG00000005884.17'\n",
       "\\item 'ENSG00000005889.15'\n",
       "\\item 'ENSG00000005893.15'\n",
       "\\item 'ENSG00000005961.17'\n",
       "\\item 'ENSG00000005981.12'\n",
       "\\item 'ENSG00000006007.11'\n",
       "\\item 'ENSG00000006015.17'\n",
       "\\item 'ENSG00000006016.10'\n",
       "\\item 'ENSG00000006025.11'\n",
       "\\item 'ENSG00000006042.11'\n",
       "\\item 'ENSG00000006047.12'\n",
       "\\item 'ENSG00000006059.3'\n",
       "\\item 'ENSG00000006062.14'\n",
       "\\item 'ENSG00000006071.12'\n",
       "\\item 'ENSG00000006116.3'\n",
       "\\item 'ENSG00000006118.14'\n",
       "\\item 'ENSG00000006125.17'\n",
       "\\item 'ENSG00000006128.11'\n",
       "\\item 'ENSG00000006194.9'\n",
       "\\item 'ENSG00000006210.6'\n",
       "\\item 'ENSG00000006282.20'\n",
       "\\item 'ENSG00000006283.17'\n",
       "\\item 'ENSG00000006327.13'\n",
       "\\item 'ENSG00000006377.10'\n",
       "\\item 'ENSG00000006432.15'\n",
       "\\item 'ENSG00000006451.7'\n",
       "\\item 'ENSG00000006453.13'\n",
       "\\item 'ENSG00000006459.10'\n",
       "\\item 'ENSG00000006468.13'\n",
       "\\item 'ENSG00000006530.15'\n",
       "\\item 'ENSG00000006534.15'\n",
       "\\item 'ENSG00000006555.10'\n",
       "\\item 'ENSG00000006576.16'\n",
       "\\item 'ENSG00000006606.8'\n",
       "\\item 'ENSG00000006607.13'\n",
       "\\item 'ENSG00000006611.15'\n",
       "\\item 'ENSG00000006625.17'\n",
       "\\item 'ENSG00000006634.7'\n",
       "\\item 'ENSG00000006638.11'\n",
       "\\item 'ENSG00000006652.13'\n",
       "\\item 'ENSG00000006659.12'\n",
       "\\item 'ENSG00000006695.10'\n",
       "\\item 'ENSG00000006704.10'\n",
       "\\item 'ENSG00000006712.14'\n",
       "\\item 'ENSG00000006715.15'\n",
       "\\item 'ENSG00000006740.16'\n",
       "\\item 'ENSG00000006744.18'\n",
       "\\item 'ENSG00000006747.14'\n",
       "\\item 'ENSG00000006756.15'\n",
       "\\item 'ENSG00000006757.11'\n",
       "\\item 'ENSG00000006788.13'\n",
       "\\item 'ENSG00000006831.9'\n",
       "\\item 'ENSG00000006837.11'\n",
       "\\item 'ENSG00000007001.12'\n",
       "\\item 'ENSG00000007038.10'\n",
       "\\item 'ENSG00000007047.14'\n",
       "\\item 'ENSG00000007062.11'\n",
       "\\item 'ENSG00000007080.10'\n",
       "\\item 'ENSG00000007129.17'\n",
       "\\item 'ENSG00000007168.12'\n",
       "\\item 'ENSG00000007171.16'\n",
       "\\item 'ENSG00000007174.17'\n",
       "\\item 'ENSG00000007202.14'\n",
       "\\item 'ENSG00000007216.14'\n",
       "\\item 'ENSG00000007237.18'\n",
       "\\item 'ENSG00000007255.10'\n",
       "\\item 'ENSG00000007264.14'\n",
       "\\item 'ENSG00000007306.14'\n",
       "\\item 'ENSG00000007312.12'\n",
       "\\item 'ENSG00000007314.12'\n",
       "\\item 'ENSG00000007341.18'\n",
       "\\item 'ENSG00000007350.16'\n",
       "\\item 'ENSG00000007372.21'\n",
       "\\item 'ENSG00000007376.7'\n",
       "\\item 'ENSG00000007384.15'\n",
       "\\item 'ENSG00000007392.16'\n",
       "\\item 'ENSG00000007402.11'\n",
       "\\item 'ENSG00000007516.13'\n",
       "\\item 'ENSG00000007520.3'\n",
       "\\item 'ENSG00000007541.16'\n",
       "\\item 'ENSG00000007545.15'\n",
       "\\item 'ENSG00000007866.19'\n",
       "\\item 'ENSG00000007908.15'\n",
       "\\item 'ENSG00000007923.15'\n",
       "\\item 'ENSG00000007933.12'\n",
       "\\item 'ENSG00000007944.14'\n",
       "\\item 'ENSG00000007952.17'\n",
       "\\item 'ENSG00000007968.6'\n",
       "\\item 'ENSG00000008018.8'\n",
       "\\item 'ENSG00000008056.13'\n",
       "\\item 'ENSG00000008083.13'\n",
       "\\item 'ENSG00000008086.11'\n",
       "\\item ⋯\n",
       "\\item 'ENSG00000280623.1'\n",
       "\\item 'ENSG00000280639.1'\n",
       "\\item 'ENSG00000280649.2'\n",
       "\\item 'ENSG00000280650.1'\n",
       "\\item 'ENSG00000280660.1'\n",
       "\\item 'ENSG00000280670.2'\n",
       "\\item 'ENSG00000280693.2'\n",
       "\\item 'ENSG00000280721.1'\n",
       "\\item 'ENSG00000280734.2'\n",
       "\\item 'ENSG00000280739.2'\n",
       "\\item 'ENSG00000280758.1'\n",
       "\\item 'ENSG00000280776.1'\n",
       "\\item 'ENSG00000280789.1'\n",
       "\\item 'ENSG00000280798.1'\n",
       "\\item 'ENSG00000280809.2'\n",
       "\\item 'ENSG00000280828.1'\n",
       "\\item 'ENSG00000280832.1'\n",
       "\\item 'ENSG00000280837.1'\n",
       "\\item 'ENSG00000280870.2'\n",
       "\\item 'ENSG00000280913.2'\n",
       "\\item 'ENSG00000280916.1'\n",
       "\\item 'ENSG00000280927.1'\n",
       "\\item 'ENSG00000280936.1'\n",
       "\\item 'ENSG00000281005.1'\n",
       "\\item 'ENSG00000281026.1'\n",
       "\\item 'ENSG00000281097.1'\n",
       "\\item 'ENSG00000281100.1'\n",
       "\\item 'ENSG00000281103.1'\n",
       "\\item 'ENSG00000281106.2'\n",
       "\\item 'ENSG00000281128.1'\n",
       "\\item 'ENSG00000281160.1'\n",
       "\\item 'ENSG00000281162.2'\n",
       "\\item 'ENSG00000281181.1'\n",
       "\\item 'ENSG00000281183.1'\n",
       "\\item 'ENSG00000281186.1'\n",
       "\\item 'ENSG00000281195.1'\n",
       "\\item 'ENSG00000281205.1'\n",
       "\\item 'ENSG00000281207.1'\n",
       "\\item 'ENSG00000281327.1'\n",
       "\\item 'ENSG00000281332.1'\n",
       "\\item 'ENSG00000281344.1'\n",
       "\\item 'ENSG00000281357.2'\n",
       "\\item 'ENSG00000281358.1'\n",
       "\\item 'ENSG00000281376.1'\n",
       "\\item 'ENSG00000281383.1'\n",
       "\\item 'ENSG00000281386.1'\n",
       "\\item 'ENSG00000281392.1'\n",
       "\\item 'ENSG00000281398.2'\n",
       "\\item 'ENSG00000281404.1'\n",
       "\\item 'ENSG00000281406.1'\n",
       "\\item 'ENSG00000281468.1'\n",
       "\\item 'ENSG00000281490.1'\n",
       "\\item 'ENSG00000281501.1'\n",
       "\\item 'ENSG00000281530.1'\n",
       "\\item 'ENSG00000281538.1'\n",
       "\\item 'ENSG00000281571.2'\n",
       "\\item 'ENSG00000281649.1'\n",
       "\\item 'ENSG00000281655.1'\n",
       "\\item 'ENSG00000281692.1'\n",
       "\\item 'ENSG00000281706.2'\n",
       "\\item 'ENSG00000281721.1'\n",
       "\\item 'ENSG00000281732.1'\n",
       "\\item 'ENSG00000281756.1'\n",
       "\\item 'ENSG00000281769.1'\n",
       "\\item 'ENSG00000281831.1'\n",
       "\\item 'ENSG00000281852.1'\n",
       "\\item 'ENSG00000281880.1'\n",
       "\\item 'ENSG00000281896.1'\n",
       "\\item 'ENSG00000281903.2'\n",
       "\\item 'ENSG00000281912.1'\n",
       "\\item 'ENSG00000281969.1'\n",
       "\\item 'ENSG00000281974.1'\n",
       "\\item 'ENSG00000281990.1'\n",
       "\\item 'ENSG00000281991.1'\n",
       "\\item 'ENSG00000282012.1'\n",
       "\\item 'ENSG00000282022.1'\n",
       "\\item 'ENSG00000282024.1'\n",
       "\\item 'ENSG00000282033.1'\n",
       "\\item 'ENSG00000282057.1'\n",
       "\\item 'ENSG00000282122.1'\n",
       "\\item 'ENSG00000282164.2'\n",
       "\\item 'ENSG00000282221.1'\n",
       "\\item 'ENSG00000282375.1'\n",
       "\\item 'ENSG00000282386.1'\n",
       "\\item 'ENSG00000282390.1'\n",
       "\\item 'ENSG00000282416.1'\n",
       "\\item 'ENSG00000282419.2'\n",
       "\\item 'ENSG00000282458.1'\n",
       "\\item 'ENSG00000282508.1'\n",
       "\\item 'ENSG00000282556.1'\n",
       "\\item 'ENSG00000282572.2'\n",
       "\\item 'ENSG00000282600.2'\n",
       "\\item 'ENSG00000282608.1'\n",
       "\\item 'ENSG00000282639.1'\n",
       "\\item 'ENSG00000282651.2'\n",
       "\\item 'ENSG00000282728.1'\n",
       "\\item 'ENSG00000282826.1'\n",
       "\\item 'ENSG00000282840.1'\n",
       "\\item 'ENSG00000282851.1'\n",
       "\\item 'ENSG00000282881.1'\n",
       "\\item 'ENSG00000282885.1'\n",
       "\\item 'ENSG00000282890.1'\n",
       "\\item 'ENSG00000282917.1'\n",
       "\\item 'ENSG00000282933.2'\n",
       "\\item 'ENSG00000282936.1'\n",
       "\\item 'ENSG00000282939.1'\n",
       "\\item 'ENSG00000282943.1'\n",
       "\\item 'ENSG00000282951.1'\n",
       "\\item 'ENSG00000283000.1'\n",
       "\\item 'ENSG00000283003.1'\n",
       "\\item 'ENSG00000283025.1'\n",
       "\\item 'ENSG00000283031.1'\n",
       "\\item 'ENSG00000283041.1'\n",
       "\\item 'ENSG00000283050.1'\n",
       "\\item 'ENSG00000283051.1'\n",
       "\\item 'ENSG00000283063.1'\n",
       "\\item 'ENSG00000283071.1'\n",
       "\\item 'ENSG00000283078.1'\n",
       "\\item 'ENSG00000283095.1'\n",
       "\\item 'ENSG00000283098.1'\n",
       "\\item 'ENSG00000283103.1'\n",
       "\\item 'ENSG00000283122.1'\n",
       "\\item 'ENSG00000283132.1'\n",
       "\\item 'ENSG00000283154.2'\n",
       "\\item 'ENSG00000283156.1'\n",
       "\\item 'ENSG00000283162.1'\n",
       "\\item 'ENSG00000283175.1'\n",
       "\\item 'ENSG00000283178.1'\n",
       "\\item 'ENSG00000283183.1'\n",
       "\\item 'ENSG00000283196.1'\n",
       "\\item 'ENSG00000283199.1'\n",
       "\\item 'ENSG00000283213.1'\n",
       "\\item 'ENSG00000283227.1'\n",
       "\\item 'ENSG00000283235.1'\n",
       "\\item 'ENSG00000283236.1'\n",
       "\\item 'ENSG00000283267.1'\n",
       "\\item 'ENSG00000283270.1'\n",
       "\\item 'ENSG00000283288.1'\n",
       "\\item 'ENSG00000283297.1'\n",
       "\\item 'ENSG00000283307.1'\n",
       "\\item 'ENSG00000283312.1'\n",
       "\\item 'ENSG00000283342.1'\n",
       "\\item 'ENSG00000283355.1'\n",
       "\\item 'ENSG00000283361.1'\n",
       "\\item 'ENSG00000283366.1'\n",
       "\\item 'ENSG00000283375.1'\n",
       "\\item 'ENSG00000283426.1'\n",
       "\\item 'ENSG00000283440.1'\n",
       "\\item 'ENSG00000283445.1'\n",
       "\\item 'ENSG00000283458.1'\n",
       "\\item 'ENSG00000283462.1'\n",
       "\\item 'ENSG00000283503.1'\n",
       "\\item 'ENSG00000283518.1'\n",
       "\\item 'ENSG00000283525.1'\n",
       "\\item 'ENSG00000283526.1'\n",
       "\\item 'ENSG00000283538.1'\n",
       "\\item 'ENSG00000283554.1'\n",
       "\\item 'ENSG00000283559.1'\n",
       "\\item 'ENSG00000283597.2'\n",
       "\\item 'ENSG00000283602.1'\n",
       "\\item 'ENSG00000283632.2'\n",
       "\\item 'ENSG00000283633.1'\n",
       "\\item 'ENSG00000283646.1'\n",
       "\\item 'ENSG00000283662.1'\n",
       "\\item 'ENSG00000283667.1'\n",
       "\\item 'ENSG00000283674.1'\n",
       "\\item 'ENSG00000283689.1'\n",
       "\\item 'ENSG00000283692.1'\n",
       "\\item 'ENSG00000283696.1'\n",
       "\\item 'ENSG00000283703.1'\n",
       "\\item 'ENSG00000283709.1'\n",
       "\\item 'ENSG00000283766.1'\n",
       "\\item 'ENSG00000283787.1'\n",
       "\\item 'ENSG00000283828.1'\n",
       "\\item 'ENSG00000283839.1'\n",
       "\\item 'ENSG00000283886.1'\n",
       "\\item 'ENSG00000283897.1'\n",
       "\\item 'ENSG00000283906.1'\n",
       "\\item 'ENSG00000283959.1'\n",
       "\\item 'ENSG00000283992.1'\n",
       "\\item 'ENSG00000283994.1'\n",
       "\\item 'ENSG00000284024.1'\n",
       "\\item 'ENSG00000284048.1'\n",
       "\\item 'ENSG00000284052.1'\n",
       "\\item 'ENSG00000284070.1'\n",
       "\\item 'ENSG00000284128.1'\n",
       "\\item 'ENSG00000284130.1'\n",
       "\\item 'ENSG00000284196.1'\n",
       "\\item 'ENSG00000284237.1'\n",
       "\\item 'ENSG00000284308.1'\n",
       "\\item 'ENSG00000284309.1'\n",
       "\\item 'ENSG00000284377.1'\n",
       "\\item 'ENSG00000284391.1'\n",
       "\\item 'ENSG00000284413.1'\n",
       "\\item 'ENSG00000284428.1'\n",
       "\\item 'ENSG00000284430.1'\n",
       "\\item 'ENSG00000284471.1'\n",
       "\\item 'ENSG00000284526.1'\n",
       "\\item 'ENSG00000284543.1'\n",
       "\\item 'ENSG00000284600.1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. ENSG00000000003.14\n",
       "2. ENSG00000000005.5\n",
       "3. ENSG00000000419.12\n",
       "4. ENSG00000000457.13\n",
       "5. ENSG00000000460.16\n",
       "6. ENSG00000000938.12\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'ENSG00000000003.14'\n",
       "2. 'ENSG00000000005.5'\n",
       "3. 'ENSG00000000419.12'\n",
       "4. 'ENSG00000000457.13'\n",
       "5. 'ENSG00000000460.16'\n",
       "6. 'ENSG00000000938.12'\n",
       "7. 'ENSG00000000971.15'\n",
       "8. 'ENSG00000001036.13'\n",
       "9. 'ENSG00000001084.10'\n",
       "10. 'ENSG00000001167.14'\n",
       "11. 'ENSG00000001460.17'\n",
       "12. 'ENSG00000001461.16'\n",
       "13. 'ENSG00000001497.16'\n",
       "14. 'ENSG00000001561.6'\n",
       "15. 'ENSG00000001617.11'\n",
       "16. 'ENSG00000001626.14'\n",
       "17. 'ENSG00000001629.9'\n",
       "18. 'ENSG00000001630.15'\n",
       "19. 'ENSG00000001631.15'\n",
       "20. 'ENSG00000002016.17'\n",
       "21. 'ENSG00000002330.13'\n",
       "22. 'ENSG00000002549.12'\n",
       "23. 'ENSG00000002586.18'\n",
       "24. 'ENSG00000002587.9'\n",
       "25. 'ENSG00000002726.20'\n",
       "26. 'ENSG00000002745.12'\n",
       "27. 'ENSG00000002746.14'\n",
       "28. 'ENSG00000002822.15'\n",
       "29. 'ENSG00000002834.17'\n",
       "30. 'ENSG00000002919.14'\n",
       "31. 'ENSG00000002933.7'\n",
       "32. 'ENSG00000003056.7'\n",
       "33. 'ENSG00000003096.14'\n",
       "34. 'ENSG00000003137.8'\n",
       "35. 'ENSG00000003147.17'\n",
       "36. 'ENSG00000003249.13'\n",
       "37. 'ENSG00000003393.14'\n",
       "38. 'ENSG00000003400.14'\n",
       "39. 'ENSG00000003402.19'\n",
       "40. 'ENSG00000003436.15'\n",
       "41. 'ENSG00000003509.15'\n",
       "42. 'ENSG00000003756.16'\n",
       "43. 'ENSG00000003987.13'\n",
       "44. 'ENSG00000003989.17'\n",
       "45. 'ENSG00000004059.10'\n",
       "46. 'ENSG00000004139.13'\n",
       "47. 'ENSG00000004142.11'\n",
       "48. 'ENSG00000004399.12'\n",
       "49. 'ENSG00000004455.16'\n",
       "50. 'ENSG00000004468.12'\n",
       "51. 'ENSG00000004478.7'\n",
       "52. 'ENSG00000004487.15'\n",
       "53. 'ENSG00000004534.14'\n",
       "54. 'ENSG00000004660.14'\n",
       "55. 'ENSG00000004700.15'\n",
       "56. 'ENSG00000004766.15'\n",
       "57. 'ENSG00000004776.12'\n",
       "58. 'ENSG00000004777.18'\n",
       "59. 'ENSG00000004779.9'\n",
       "60. 'ENSG00000004799.7'\n",
       "61. 'ENSG00000004809.13'\n",
       "62. 'ENSG00000004838.13'\n",
       "63. 'ENSG00000004846.16'\n",
       "64. 'ENSG00000004848.7'\n",
       "65. 'ENSG00000004864.13'\n",
       "66. 'ENSG00000004866.20'\n",
       "67. 'ENSG00000004897.11'\n",
       "68. 'ENSG00000004939.13'\n",
       "69. 'ENSG00000004948.13'\n",
       "70. 'ENSG00000004961.14'\n",
       "71. 'ENSG00000004975.11'\n",
       "72. 'ENSG00000005001.9'\n",
       "73. 'ENSG00000005007.12'\n",
       "74. 'ENSG00000005020.12'\n",
       "75. 'ENSG00000005022.5'\n",
       "76. 'ENSG00000005059.15'\n",
       "77. 'ENSG00000005073.5'\n",
       "78. 'ENSG00000005075.15'\n",
       "79. 'ENSG00000005100.12'\n",
       "80. 'ENSG00000005102.12'\n",
       "81. 'ENSG00000005108.15'\n",
       "82. 'ENSG00000005156.11'\n",
       "83. 'ENSG00000005175.9'\n",
       "84. 'ENSG00000005187.11'\n",
       "85. 'ENSG00000005189.19'\n",
       "86. 'ENSG00000005194.14'\n",
       "87. 'ENSG00000005206.16'\n",
       "88. 'ENSG00000005238.19'\n",
       "89. 'ENSG00000005243.9'\n",
       "90. 'ENSG00000005249.12'\n",
       "91. 'ENSG00000005302.17'\n",
       "92. 'ENSG00000005339.14'\n",
       "93. 'ENSG00000005379.15'\n",
       "94. 'ENSG00000005381.7'\n",
       "95. 'ENSG00000005421.8'\n",
       "96. 'ENSG00000005436.13'\n",
       "97. 'ENSG00000005448.16'\n",
       "98. 'ENSG00000005469.11'\n",
       "99. 'ENSG00000005471.15'\n",
       "100. 'ENSG00000005483.20'\n",
       "101. 'ENSG00000005486.16'\n",
       "102. 'ENSG00000005513.9'\n",
       "103. 'ENSG00000005700.14'\n",
       "104. 'ENSG00000005801.16'\n",
       "105. 'ENSG00000005810.17'\n",
       "106. 'ENSG00000005812.10'\n",
       "107. 'ENSG00000005844.17'\n",
       "108. 'ENSG00000005882.11'\n",
       "109. 'ENSG00000005884.17'\n",
       "110. 'ENSG00000005889.15'\n",
       "111. 'ENSG00000005893.15'\n",
       "112. 'ENSG00000005961.17'\n",
       "113. 'ENSG00000005981.12'\n",
       "114. 'ENSG00000006007.11'\n",
       "115. 'ENSG00000006015.17'\n",
       "116. 'ENSG00000006016.10'\n",
       "117. 'ENSG00000006025.11'\n",
       "118. 'ENSG00000006042.11'\n",
       "119. 'ENSG00000006047.12'\n",
       "120. 'ENSG00000006059.3'\n",
       "121. 'ENSG00000006062.14'\n",
       "122. 'ENSG00000006071.12'\n",
       "123. 'ENSG00000006116.3'\n",
       "124. 'ENSG00000006118.14'\n",
       "125. 'ENSG00000006125.17'\n",
       "126. 'ENSG00000006128.11'\n",
       "127. 'ENSG00000006194.9'\n",
       "128. 'ENSG00000006210.6'\n",
       "129. 'ENSG00000006282.20'\n",
       "130. 'ENSG00000006283.17'\n",
       "131. 'ENSG00000006327.13'\n",
       "132. 'ENSG00000006377.10'\n",
       "133. 'ENSG00000006432.15'\n",
       "134. 'ENSG00000006451.7'\n",
       "135. 'ENSG00000006453.13'\n",
       "136. 'ENSG00000006459.10'\n",
       "137. 'ENSG00000006468.13'\n",
       "138. 'ENSG00000006530.15'\n",
       "139. 'ENSG00000006534.15'\n",
       "140. 'ENSG00000006555.10'\n",
       "141. 'ENSG00000006576.16'\n",
       "142. 'ENSG00000006606.8'\n",
       "143. 'ENSG00000006607.13'\n",
       "144. 'ENSG00000006611.15'\n",
       "145. 'ENSG00000006625.17'\n",
       "146. 'ENSG00000006634.7'\n",
       "147. 'ENSG00000006638.11'\n",
       "148. 'ENSG00000006652.13'\n",
       "149. 'ENSG00000006659.12'\n",
       "150. 'ENSG00000006695.10'\n",
       "151. 'ENSG00000006704.10'\n",
       "152. 'ENSG00000006712.14'\n",
       "153. 'ENSG00000006715.15'\n",
       "154. 'ENSG00000006740.16'\n",
       "155. 'ENSG00000006744.18'\n",
       "156. 'ENSG00000006747.14'\n",
       "157. 'ENSG00000006756.15'\n",
       "158. 'ENSG00000006757.11'\n",
       "159. 'ENSG00000006788.13'\n",
       "160. 'ENSG00000006831.9'\n",
       "161. 'ENSG00000006837.11'\n",
       "162. 'ENSG00000007001.12'\n",
       "163. 'ENSG00000007038.10'\n",
       "164. 'ENSG00000007047.14'\n",
       "165. 'ENSG00000007062.11'\n",
       "166. 'ENSG00000007080.10'\n",
       "167. 'ENSG00000007129.17'\n",
       "168. 'ENSG00000007168.12'\n",
       "169. 'ENSG00000007171.16'\n",
       "170. 'ENSG00000007174.17'\n",
       "171. 'ENSG00000007202.14'\n",
       "172. 'ENSG00000007216.14'\n",
       "173. 'ENSG00000007237.18'\n",
       "174. 'ENSG00000007255.10'\n",
       "175. 'ENSG00000007264.14'\n",
       "176. 'ENSG00000007306.14'\n",
       "177. 'ENSG00000007312.12'\n",
       "178. 'ENSG00000007314.12'\n",
       "179. 'ENSG00000007341.18'\n",
       "180. 'ENSG00000007350.16'\n",
       "181. 'ENSG00000007372.21'\n",
       "182. 'ENSG00000007376.7'\n",
       "183. 'ENSG00000007384.15'\n",
       "184. 'ENSG00000007392.16'\n",
       "185. 'ENSG00000007402.11'\n",
       "186. 'ENSG00000007516.13'\n",
       "187. 'ENSG00000007520.3'\n",
       "188. 'ENSG00000007541.16'\n",
       "189. 'ENSG00000007545.15'\n",
       "190. 'ENSG00000007866.19'\n",
       "191. 'ENSG00000007908.15'\n",
       "192. 'ENSG00000007923.15'\n",
       "193. 'ENSG00000007933.12'\n",
       "194. 'ENSG00000007944.14'\n",
       "195. 'ENSG00000007952.17'\n",
       "196. 'ENSG00000007968.6'\n",
       "197. 'ENSG00000008018.8'\n",
       "198. 'ENSG00000008056.13'\n",
       "199. 'ENSG00000008083.13'\n",
       "200. 'ENSG00000008086.11'\n",
       "201. ⋯\n",
       "202. 'ENSG00000280623.1'\n",
       "203. 'ENSG00000280639.1'\n",
       "204. 'ENSG00000280649.2'\n",
       "205. 'ENSG00000280650.1'\n",
       "206. 'ENSG00000280660.1'\n",
       "207. 'ENSG00000280670.2'\n",
       "208. 'ENSG00000280693.2'\n",
       "209. 'ENSG00000280721.1'\n",
       "210. 'ENSG00000280734.2'\n",
       "211. 'ENSG00000280739.2'\n",
       "212. 'ENSG00000280758.1'\n",
       "213. 'ENSG00000280776.1'\n",
       "214. 'ENSG00000280789.1'\n",
       "215. 'ENSG00000280798.1'\n",
       "216. 'ENSG00000280809.2'\n",
       "217. 'ENSG00000280828.1'\n",
       "218. 'ENSG00000280832.1'\n",
       "219. 'ENSG00000280837.1'\n",
       "220. 'ENSG00000280870.2'\n",
       "221. 'ENSG00000280913.2'\n",
       "222. 'ENSG00000280916.1'\n",
       "223. 'ENSG00000280927.1'\n",
       "224. 'ENSG00000280936.1'\n",
       "225. 'ENSG00000281005.1'\n",
       "226. 'ENSG00000281026.1'\n",
       "227. 'ENSG00000281097.1'\n",
       "228. 'ENSG00000281100.1'\n",
       "229. 'ENSG00000281103.1'\n",
       "230. 'ENSG00000281106.2'\n",
       "231. 'ENSG00000281128.1'\n",
       "232. 'ENSG00000281160.1'\n",
       "233. 'ENSG00000281162.2'\n",
       "234. 'ENSG00000281181.1'\n",
       "235. 'ENSG00000281183.1'\n",
       "236. 'ENSG00000281186.1'\n",
       "237. 'ENSG00000281195.1'\n",
       "238. 'ENSG00000281205.1'\n",
       "239. 'ENSG00000281207.1'\n",
       "240. 'ENSG00000281327.1'\n",
       "241. 'ENSG00000281332.1'\n",
       "242. 'ENSG00000281344.1'\n",
       "243. 'ENSG00000281357.2'\n",
       "244. 'ENSG00000281358.1'\n",
       "245. 'ENSG00000281376.1'\n",
       "246. 'ENSG00000281383.1'\n",
       "247. 'ENSG00000281386.1'\n",
       "248. 'ENSG00000281392.1'\n",
       "249. 'ENSG00000281398.2'\n",
       "250. 'ENSG00000281404.1'\n",
       "251. 'ENSG00000281406.1'\n",
       "252. 'ENSG00000281468.1'\n",
       "253. 'ENSG00000281490.1'\n",
       "254. 'ENSG00000281501.1'\n",
       "255. 'ENSG00000281530.1'\n",
       "256. 'ENSG00000281538.1'\n",
       "257. 'ENSG00000281571.2'\n",
       "258. 'ENSG00000281649.1'\n",
       "259. 'ENSG00000281655.1'\n",
       "260. 'ENSG00000281692.1'\n",
       "261. 'ENSG00000281706.2'\n",
       "262. 'ENSG00000281721.1'\n",
       "263. 'ENSG00000281732.1'\n",
       "264. 'ENSG00000281756.1'\n",
       "265. 'ENSG00000281769.1'\n",
       "266. 'ENSG00000281831.1'\n",
       "267. 'ENSG00000281852.1'\n",
       "268. 'ENSG00000281880.1'\n",
       "269. 'ENSG00000281896.1'\n",
       "270. 'ENSG00000281903.2'\n",
       "271. 'ENSG00000281912.1'\n",
       "272. 'ENSG00000281969.1'\n",
       "273. 'ENSG00000281974.1'\n",
       "274. 'ENSG00000281990.1'\n",
       "275. 'ENSG00000281991.1'\n",
       "276. 'ENSG00000282012.1'\n",
       "277. 'ENSG00000282022.1'\n",
       "278. 'ENSG00000282024.1'\n",
       "279. 'ENSG00000282033.1'\n",
       "280. 'ENSG00000282057.1'\n",
       "281. 'ENSG00000282122.1'\n",
       "282. 'ENSG00000282164.2'\n",
       "283. 'ENSG00000282221.1'\n",
       "284. 'ENSG00000282375.1'\n",
       "285. 'ENSG00000282386.1'\n",
       "286. 'ENSG00000282390.1'\n",
       "287. 'ENSG00000282416.1'\n",
       "288. 'ENSG00000282419.2'\n",
       "289. 'ENSG00000282458.1'\n",
       "290. 'ENSG00000282508.1'\n",
       "291. 'ENSG00000282556.1'\n",
       "292. 'ENSG00000282572.2'\n",
       "293. 'ENSG00000282600.2'\n",
       "294. 'ENSG00000282608.1'\n",
       "295. 'ENSG00000282639.1'\n",
       "296. 'ENSG00000282651.2'\n",
       "297. 'ENSG00000282728.1'\n",
       "298. 'ENSG00000282826.1'\n",
       "299. 'ENSG00000282840.1'\n",
       "300. 'ENSG00000282851.1'\n",
       "301. 'ENSG00000282881.1'\n",
       "302. 'ENSG00000282885.1'\n",
       "303. 'ENSG00000282890.1'\n",
       "304. 'ENSG00000282917.1'\n",
       "305. 'ENSG00000282933.2'\n",
       "306. 'ENSG00000282936.1'\n",
       "307. 'ENSG00000282939.1'\n",
       "308. 'ENSG00000282943.1'\n",
       "309. 'ENSG00000282951.1'\n",
       "310. 'ENSG00000283000.1'\n",
       "311. 'ENSG00000283003.1'\n",
       "312. 'ENSG00000283025.1'\n",
       "313. 'ENSG00000283031.1'\n",
       "314. 'ENSG00000283041.1'\n",
       "315. 'ENSG00000283050.1'\n",
       "316. 'ENSG00000283051.1'\n",
       "317. 'ENSG00000283063.1'\n",
       "318. 'ENSG00000283071.1'\n",
       "319. 'ENSG00000283078.1'\n",
       "320. 'ENSG00000283095.1'\n",
       "321. 'ENSG00000283098.1'\n",
       "322. 'ENSG00000283103.1'\n",
       "323. 'ENSG00000283122.1'\n",
       "324. 'ENSG00000283132.1'\n",
       "325. 'ENSG00000283154.2'\n",
       "326. 'ENSG00000283156.1'\n",
       "327. 'ENSG00000283162.1'\n",
       "328. 'ENSG00000283175.1'\n",
       "329. 'ENSG00000283178.1'\n",
       "330. 'ENSG00000283183.1'\n",
       "331. 'ENSG00000283196.1'\n",
       "332. 'ENSG00000283199.1'\n",
       "333. 'ENSG00000283213.1'\n",
       "334. 'ENSG00000283227.1'\n",
       "335. 'ENSG00000283235.1'\n",
       "336. 'ENSG00000283236.1'\n",
       "337. 'ENSG00000283267.1'\n",
       "338. 'ENSG00000283270.1'\n",
       "339. 'ENSG00000283288.1'\n",
       "340. 'ENSG00000283297.1'\n",
       "341. 'ENSG00000283307.1'\n",
       "342. 'ENSG00000283312.1'\n",
       "343. 'ENSG00000283342.1'\n",
       "344. 'ENSG00000283355.1'\n",
       "345. 'ENSG00000283361.1'\n",
       "346. 'ENSG00000283366.1'\n",
       "347. 'ENSG00000283375.1'\n",
       "348. 'ENSG00000283426.1'\n",
       "349. 'ENSG00000283440.1'\n",
       "350. 'ENSG00000283445.1'\n",
       "351. 'ENSG00000283458.1'\n",
       "352. 'ENSG00000283462.1'\n",
       "353. 'ENSG00000283503.1'\n",
       "354. 'ENSG00000283518.1'\n",
       "355. 'ENSG00000283525.1'\n",
       "356. 'ENSG00000283526.1'\n",
       "357. 'ENSG00000283538.1'\n",
       "358. 'ENSG00000283554.1'\n",
       "359. 'ENSG00000283559.1'\n",
       "360. 'ENSG00000283597.2'\n",
       "361. 'ENSG00000283602.1'\n",
       "362. 'ENSG00000283632.2'\n",
       "363. 'ENSG00000283633.1'\n",
       "364. 'ENSG00000283646.1'\n",
       "365. 'ENSG00000283662.1'\n",
       "366. 'ENSG00000283667.1'\n",
       "367. 'ENSG00000283674.1'\n",
       "368. 'ENSG00000283689.1'\n",
       "369. 'ENSG00000283692.1'\n",
       "370. 'ENSG00000283696.1'\n",
       "371. 'ENSG00000283703.1'\n",
       "372. 'ENSG00000283709.1'\n",
       "373. 'ENSG00000283766.1'\n",
       "374. 'ENSG00000283787.1'\n",
       "375. 'ENSG00000283828.1'\n",
       "376. 'ENSG00000283839.1'\n",
       "377. 'ENSG00000283886.1'\n",
       "378. 'ENSG00000283897.1'\n",
       "379. 'ENSG00000283906.1'\n",
       "380. 'ENSG00000283959.1'\n",
       "381. 'ENSG00000283992.1'\n",
       "382. 'ENSG00000283994.1'\n",
       "383. 'ENSG00000284024.1'\n",
       "384. 'ENSG00000284048.1'\n",
       "385. 'ENSG00000284052.1'\n",
       "386. 'ENSG00000284070.1'\n",
       "387. 'ENSG00000284128.1'\n",
       "388. 'ENSG00000284130.1'\n",
       "389. 'ENSG00000284196.1'\n",
       "390. 'ENSG00000284237.1'\n",
       "391. 'ENSG00000284308.1'\n",
       "392. 'ENSG00000284309.1'\n",
       "393. 'ENSG00000284377.1'\n",
       "394. 'ENSG00000284391.1'\n",
       "395. 'ENSG00000284413.1'\n",
       "396. 'ENSG00000284428.1'\n",
       "397. 'ENSG00000284430.1'\n",
       "398. 'ENSG00000284471.1'\n",
       "399. 'ENSG00000284526.1'\n",
       "400. 'ENSG00000284543.1'\n",
       "401. 'ENSG00000284600.1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] ENSG00000000003.14 ENSG00000000005.5  ENSG00000000419.12 ENSG00000000457.13\n",
       "[5] ENSG00000000460.16 ENSG00000000938.12\n",
       "24185 Levels: ENSG00000000003.14 ENSG00000000005.5 ... ENSG00000284600.1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(all_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.12 We then do a hypergeometric/Fisher test to look for overrepresentation\n",
    "The universe consists of all genes with at least one read (all_genes_data).\n",
    "So we have\n",
    "\n",
    "|  \t|  DGE+| DGE-|\n",
    "|-\t|-\t|-\t|\n",
    "| DAS+|  a|  b|\n",
    "| DAS-|  c| d|"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of sigAsGenes 888\n",
      "\n",
      "Number of genes that are NOT sigAs 24185\n",
      "\n",
      "Number of DGE genes4417\n",
      "\n",
      "Number of genes that are NOT DGE 24172\n",
      "\n",
      "a: 258; b: 0; c: 13; d: 24172\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"Number of sigAsGenes \", length(sigAsGenes))\n",
    "notSigAs <- setdiff(all_genes,sigAsGenes)\n",
    "message(\"Number of genes that are NOT sigAs \", length(notSigAs))\n",
    "message(\"Number of DGE genes\", length(dge_genes))\n",
    "notDGE <- setdiff(all_genes,dge_genes)\n",
    "message(\"Number of genes that are NOT DGE \", length(notDGE))\n",
    "a <- intersect(sigAsGenes, dge_genes)\n",
    "b <- intersect(sigAsGenes, notDGE)\n",
    "c <- intersect(notSigAs, dge_genes)\n",
    "d <- intersect(notSigAs, notDGE)\n",
    "message(\"a: \", length(a), \"; b: \",  length(b), \"; c: \",  length(c), \"; d: \",  length(d))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tFisher's Exact Test for Count Data\n",
       "\n",
       "data:  m\n",
       "p-value < 2.2e-16\n",
       "alternative hypothesis: true odds ratio is not equal to 1\n",
       "95 percent confidence interval:\n",
       " 8402.051      Inf\n",
       "sample estimates:\n",
       "odds ratio \n",
       "       Inf \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "m <- matrix(c(length(a),length(b),length(c),length(d)), nrow=2,byrow = TRUE)\n",
    "fisher.test(m)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Appendix - Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Appendix 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'countGenesAndEvents'"
      ],
      "text/latex": [
       "'countGenesAndEvents'"
      ],
      "text/markdown": [
       "'countGenesAndEvents'"
      ],
      "text/plain": [
       "[1] \"countGenesAndEvents\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/gene_as.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/all_gene_as.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/gene_dge.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/Total_AS_by_chr.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/Total_AS_by_geneSymbol.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/Total_AS_by_tissue.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/Total_AS_by_splicingtype.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/genesWithCommonAS.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256 checksums of the file `../data/SplicingIndex_chr.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"countGenesAndEvents\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/gene_as.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_as.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/all_gene_as.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_dge.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/gene_dge.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_as.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_chr.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_chr.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_geneSymbol.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_geneSymbol.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_tissue.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_tissue.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_splicingtype.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_splicingtype.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/genesWithCommonAS.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name genesWithCommonAS.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/SplicingIndex_chr.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name SplicingIndex_chr.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Appendix 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "lines_to_next_cell": 0
   },
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../data/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../data/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../data/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../data/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Time difference of 4.300307 mins"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "end_time <- Sys.time()\n",
    "end_time - start_time"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "encoding": "# -*- coding: utf-8 -*-",
   "main_language": "R",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
