{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3b1e52cf",
   "metadata": {
    "papermill": {
     "duration": 0.029964,
     "end_time": "2021-05-21T02:43:54.123072",
     "exception": false,
     "start_time": "2021-05-21T02:43:54.093108",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Analysis Notebook - Count Genes and Events\n",
    "\n",
    "This notebook processes the raw counts as provided by rMATS 3.2.5 with modifications because this rMATS was run at scale with the entirety of GTEx.   To so, please refer to the methods in the paper, but in brief, each sample was run against another.  This method of running rMATS 3.2.5, permitted us leveraging the counting of the reads done by rMATS together with the scrutinization of rMATS against the provided GTF.   In this case, gencode's complete annotated gtf was scanned by rMATS 3.2.5, the specific possible alternative splicing events catalogued and used in the `fromGTF` descriptions which capture the specific genomic locations for each of five alternative splicing types.  Then the short read RNA sequencing files collected by GTEx and annotated by sex and tissue were aligned and counted against the specific junctions as described by these fromGTF files.   It is these counts that were used in the analyses for this paper.  \n",
    "\n",
    "This notebook, counts the genes and events and performs several descriptive statistical analysis. It  produces outputs needed for downstream visualizations, which are in separate notebooks for added clarity.\n",
    "\n",
    "## 1 Introduction\n",
    "\n",
    "## 1.1 Data files needed by this notebook\n",
    "\n",
    "Input files have been pulled from results, the data were primarily obtained through the execution of the Nextflow workflow, rmats-nf, found in the GitHub repository, https://github.com/lifebit-ai/rmats-nf or the notebook.   How the data were generated are noted for each of the files.\n",
    "\n",
    "1. **`*`.model_B_sex_as_event.csv** One for each tissue, generated by running the notebook `differentialSplicingJunctionAnalysis.ipynb` as a Nextflow workflow\n",
    "2. **`*`.model_B_sex_as_event_refined.csv** One for each tissue, these are the statistically significant results FC > 1.5 and p-value < 0.05 generated by running the notebook `differentialSplicingJunctionAnalysis.ipynb` as a Nextflow workflow.\n",
    "3. **fromGTF.`*`.txt** One for each splicing type, generated within rmats-nf, a Nextflow workflow found here `https://github.com/lifebit-ai/rmats-nf` (fromGTF.A3SS.txt, fromGTF.A5SS.txt, fromGTF.MXE.txt, fromGTF.RI.txt,fromGTF.SE.txt)\n",
    "4. **`*`.DGE.csv** One for each tissue, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "5. **`*`.DGE_refined.csv** One for each tissue, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "6. **srr_pdata.csv** Sequence Run (SRR) data merged with phenotype data, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "7. **rmats_final.`*`.jc.ijc.txt.gz** One for each splicing type, this is a matrix of all included junction (ijc) counts for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`. \n",
    "8. **rmats_final.`*`.jc.inclen.txt.gz** One for each splicing type, this is a matrix of inclusion length (inclen) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "9. **rmats_final.`*`.jc.inclen.txt.gz** One for each splicing type, this is a matrix of inclen for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "10. **rmats_final.`*`.jc.inc.txt.gz** One for each splicing type, this is a matrix of percent spliced in (inc) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "11. **rmats_final.`*`.jc.sjc.txt.gz** One for each splicing type, this is a matrix of skipped junction counts (sjc) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "12. **rmats_final.`*`.jc.skiplen.txt.gz** One for each splicing type, this is a matrix of skipped length (skiplen) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`\n",
    "\n",
    "Two files needed from the `sbas/assets` directory\n",
    "\n",
    "13. **tissues.csv** This file curated by each tissue, the number of female and male samples and provides a display name for the graphics.\n",
    "14. **all_gene_dge.tsv** This file is an intersection of all the differentially expressed genes (not just the significant but all) for all the tissues considered in this study generated by a custom script.\n",
    "\n",
    "\n",
    "## 1.2 Data files created by this notebook\n",
    "\n",
    "Output text files are written to the ``sbas/data`` directory (this notebook is in the ``sbas/jupyter`` directory). \n",
    "\n",
    "1.  **gene_as.tsv**: Significant alternative splicing events per gene, adjusted p-value <= 0.05, fold change >= 1.5.\n",
    "2.  **all_gene_as.tsv**: all alternative splicing events\n",
    "3.  **gene_dge.tsv**: Significant differential gene expression, adjusted p-value <= 0.05, fold change >= 1.5\n",
    "4.  **all_gene_dge.tsv**: all differential gene event files is NOT produced here but is found in the assets directory ../assets/all_gene_dge.tsv\n",
    "5.  **genesWithCommonAS.tsv**: genes (as geneSymbol, the number of splicing events, and the number of tissues the event occurs in)\n",
    "6.  **Total_AS_by_chr.tsv**: Total alternative splicing events per chromosome\n",
    "7.  **Total_AS_by_geneSymbol.tsv**: Count the number of tissues in which specific genes show significant alternative splicing\n",
    "8.  **DGE_by_geneSymbol.tsv***: Most highly expressed genes by tissue\n",
    "9.  **Total_AS_by_tissue.tsv**: Count the number of significant splicing events per tissue\n",
    "10. **Total_AS_by_splicingtype.tsv**: Count number of significant splicing events for each of the 5' alternative splicing categories\n",
    "11. **SplicingIndex_chr.tsv**: Splicing index by chr (number of sigificant AS events per 1000 exons)\n",
    "\n",
    "## 2 Run Analyses\n",
    "\n",
    "## 2.1 Load Libraries\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "71d6a779",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:43:54.257350Z",
     "iopub.status.busy": "2021-05-21T02:43:54.254717Z",
     "iopub.status.idle": "2021-05-21T02:44:01.821766Z",
     "shell.execute_reply": "2021-05-21T02:44:01.820431Z"
    },
    "papermill": {
     "duration": 7.669701,
     "end_time": "2021-05-21T02:44:01.821974",
     "exception": false,
     "start_time": "2021-05-21T02:43:54.152273",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#conda install r-dplyr bioconductor-biobase r-tibble r-r.utils bioconductor-rtracklayer -y\n",
    "start_time <- Sys.time()\n",
    "suppressMessages({\n",
    "    options(warn = -1) \n",
    "    library(dplyr)\n",
    "    library(Biobase)\n",
    "    library(tibble)\n",
    "    library(R.utils)\n",
    "    library(rtracklayer)\n",
    "})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a181eed4",
   "metadata": {
    "papermill": {
     "duration": 0.028156,
     "end_time": "2021-05-21T02:44:01.878751",
     "exception": false,
     "start_time": "2021-05-21T02:44:01.850595",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.2 Read in Curated Tissue List\n",
    "\n",
    "Read in a curated list of tissues, noting sample counts per each sex, male and female and also a curated tissue name for display.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "96fd98c3",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:01.967689Z",
     "iopub.status.busy": "2021-05-21T02:44:01.938746Z",
     "iopub.status.idle": "2021-05-21T02:44:02.031096Z",
     "shell.execute_reply": "2021-05-21T02:44:02.028240Z"
    },
    "papermill": {
     "duration": 0.124408,
     "end_time": "2021-05-21T02:44:02.031242",
     "exception": false,
     "start_time": "2021-05-21T02:44:01.906834",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Adipose (sc)'</li><li>'Adipose (v)'</li><li>'Adrenal gland'</li><li>'Aorta'</li><li>'Atrial appendage'</li><li>'Breast'</li><li>'Caudate'</li><li>'Cerebellar hemisphere'</li><li>'Cerebellum'</li><li>'Coronary artery'</li><li>'Cortex'</li><li>'EBV-lymphocytes'</li><li>'Esophagus (gej)'</li><li>'Esophagus (m)'</li><li>'Esophagus (mu)'</li><li>'Fibroblasts'</li><li>'Frontal cortex'</li><li>'Hippocampus'</li><li>'Hypothalamus'</li><li>'Left ventricle'</li><li>'Liver'</li><li>'Lung'</li><li>'Nucleus accumbens'</li><li>'Pancreas'</li><li>'Pituitary'</li><li>'Putamen'</li><li>'Sigmoid colon'</li><li>'Skeletal muscle'</li><li>'Skin (exposed)'</li><li>'Skin (not exposed)'</li><li>'Small intestine'</li><li>'Spinal cord'</li><li>'Spleen'</li><li>'Stomach'</li><li>'Thyroid'</li><li>'Tibial artery'</li><li>'Tibial nerve'</li><li>'Transverse colon'</li><li>'Whole blood'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Adipose (sc)'\n",
       "\\item 'Adipose (v)'\n",
       "\\item 'Adrenal gland'\n",
       "\\item 'Aorta'\n",
       "\\item 'Atrial appendage'\n",
       "\\item 'Breast'\n",
       "\\item 'Caudate'\n",
       "\\item 'Cerebellar hemisphere'\n",
       "\\item 'Cerebellum'\n",
       "\\item 'Coronary artery'\n",
       "\\item 'Cortex'\n",
       "\\item 'EBV-lymphocytes'\n",
       "\\item 'Esophagus (gej)'\n",
       "\\item 'Esophagus (m)'\n",
       "\\item 'Esophagus (mu)'\n",
       "\\item 'Fibroblasts'\n",
       "\\item 'Frontal cortex'\n",
       "\\item 'Hippocampus'\n",
       "\\item 'Hypothalamus'\n",
       "\\item 'Left ventricle'\n",
       "\\item 'Liver'\n",
       "\\item 'Lung'\n",
       "\\item 'Nucleus accumbens'\n",
       "\\item 'Pancreas'\n",
       "\\item 'Pituitary'\n",
       "\\item 'Putamen'\n",
       "\\item 'Sigmoid colon'\n",
       "\\item 'Skeletal muscle'\n",
       "\\item 'Skin (exposed)'\n",
       "\\item 'Skin (not exposed)'\n",
       "\\item 'Small intestine'\n",
       "\\item 'Spinal cord'\n",
       "\\item 'Spleen'\n",
       "\\item 'Stomach'\n",
       "\\item 'Thyroid'\n",
       "\\item 'Tibial artery'\n",
       "\\item 'Tibial nerve'\n",
       "\\item 'Transverse colon'\n",
       "\\item 'Whole blood'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Adipose (sc)'\n",
       "2. 'Adipose (v)'\n",
       "3. 'Adrenal gland'\n",
       "4. 'Aorta'\n",
       "5. 'Atrial appendage'\n",
       "6. 'Breast'\n",
       "7. 'Caudate'\n",
       "8. 'Cerebellar hemisphere'\n",
       "9. 'Cerebellum'\n",
       "10. 'Coronary artery'\n",
       "11. 'Cortex'\n",
       "12. 'EBV-lymphocytes'\n",
       "13. 'Esophagus (gej)'\n",
       "14. 'Esophagus (m)'\n",
       "15. 'Esophagus (mu)'\n",
       "16. 'Fibroblasts'\n",
       "17. 'Frontal cortex'\n",
       "18. 'Hippocampus'\n",
       "19. 'Hypothalamus'\n",
       "20. 'Left ventricle'\n",
       "21. 'Liver'\n",
       "22. 'Lung'\n",
       "23. 'Nucleus accumbens'\n",
       "24. 'Pancreas'\n",
       "25. 'Pituitary'\n",
       "26. 'Putamen'\n",
       "27. 'Sigmoid colon'\n",
       "28. 'Skeletal muscle'\n",
       "29. 'Skin (exposed)'\n",
       "30. 'Skin (not exposed)'\n",
       "31. 'Small intestine'\n",
       "32. 'Spinal cord'\n",
       "33. 'Spleen'\n",
       "34. 'Stomach'\n",
       "35. 'Thyroid'\n",
       "36. 'Tibial artery'\n",
       "37. 'Tibial nerve'\n",
       "38. 'Transverse colon'\n",
       "39. 'Whole blood'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Adipose (sc)\"          \"Adipose (v)\"           \"Adrenal gland\"        \n",
       " [4] \"Aorta\"                 \"Atrial appendage\"      \"Breast\"               \n",
       " [7] \"Caudate\"               \"Cerebellar hemisphere\" \"Cerebellum\"           \n",
       "[10] \"Coronary artery\"       \"Cortex\"                \"EBV-lymphocytes\"      \n",
       "[13] \"Esophagus (gej)\"       \"Esophagus (m)\"         \"Esophagus (mu)\"       \n",
       "[16] \"Fibroblasts\"           \"Frontal cortex\"        \"Hippocampus\"          \n",
       "[19] \"Hypothalamus\"          \"Left ventricle\"        \"Liver\"                \n",
       "[22] \"Lung\"                  \"Nucleus accumbens\"     \"Pancreas\"             \n",
       "[25] \"Pituitary\"             \"Putamen\"               \"Sigmoid colon\"        \n",
       "[28] \"Skeletal muscle\"       \"Skin (exposed)\"        \"Skin (not exposed)\"   \n",
       "[31] \"Small intestine\"       \"Spinal cord\"           \"Spleen\"               \n",
       "[34] \"Stomach\"               \"Thyroid\"               \"Tibial artery\"        \n",
       "[37] \"Tibial nerve\"          \"Transverse colon\"      \"Whole blood\"          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted 39 different tissues with at least 50 samples in both M & f\n",
      "\n"
     ]
    }
   ],
   "source": [
    "tissue_reduction_filename <- \"../assets/tissues.tsv\"\n",
    "tissue_reduction <- read.table(tissue_reduction_filename, header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$display_name != \"n/a\",]\n",
    "tissue_reduction$display_name <- factor(tissue_reduction$display_name)\n",
    "levels(tissue_reduction$display_name)\n",
    "message(\"We extracted \", length(levels(tissue_reduction$display_name)),\" different tissues with at least 50 samples in both M & f\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9501ece4",
   "metadata": {
    "papermill": {
     "duration": 0.029133,
     "end_time": "2021-05-21T02:44:02.090642",
     "exception": false,
     "start_time": "2021-05-21T02:44:02.061509",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.3 Read in refined differential AS events\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "229974a7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:02.157811Z",
     "iopub.status.busy": "2021-05-21T02:44:02.156217Z",
     "iopub.status.idle": "2021-05-21T02:44:02.213299Z",
     "shell.execute_reply": "2021-05-21T02:44:02.210243Z"
    },
    "papermill": {
     "duration": 0.093722,
     "end_time": "2021-05-21T02:44:02.213451",
     "exception": false,
     "start_time": "2021-05-21T02:44:02.119729",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted 195 model_B_sex_as_events_refined.csv files\n",
      "\n"
     ]
    }
   ],
   "source": [
    "significant_results_dir = \"../data/\"\n",
    "pattern = \"model_B_sex_as_events_refined.csv\"\n",
    "files <- list.files(path = significant_results_dir, pattern = pattern)\n",
    "as_types <- c(\"a3ss\", \"a5ss\", \"mxe\", \"ri\", \"se\")\n",
    "message(\"We extracted \", length(files), \" model_B_sex_as_events_refined.csv files\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e4aea53",
   "metadata": {
    "papermill": {
     "duration": 0.03028,
     "end_time": "2021-05-21T02:44:02.275308",
     "exception": false,
     "start_time": "2021-05-21T02:44:02.245028",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.4 Read in the AS Events annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "86530ca8",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:02.344862Z",
     "iopub.status.busy": "2021-05-21T02:44:02.342393Z",
     "iopub.status.idle": "2021-05-21T02:44:03.575485Z",
     "shell.execute_reply": "2021-05-21T02:44:03.574561Z"
    },
    "papermill": {
     "duration": 1.270255,
     "end_time": "2021-05-21T02:44:03.575705",
     "exception": false,
     "start_time": "2021-05-21T02:44:02.305450",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "a3ss_annot <- read.table(file = \"../data/fromGTF.A3SS.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "a5ss_annot <- read.table(file = \"../data/fromGTF.A5SS.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "mxe_annot <- read.table(file = \"../data/fromGTF.MXE.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "ri_annot <- read.table(file = \"../data/fromGTF.RI.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)\n",
    "se_annot <- read.table(file = \"../data/fromGTF.SE.txt\", sep = \"\\t\", quote = \"\\\"\", header = T, stringsAsFactors = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ea63060a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:03.659132Z",
     "iopub.status.busy": "2021-05-21T02:44:03.657852Z",
     "iopub.status.idle": "2021-05-21T02:44:03.684858Z",
     "shell.execute_reply": "2021-05-21T02:44:03.683643Z"
    },
    "papermill": {
     "duration": 0.078407,
     "end_time": "2021-05-21T02:44:03.684991",
     "exception": false,
     "start_time": "2021-05-21T02:44:03.606584",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>GeneID</th><th scope=col>geneSymbol</th><th scope=col>chr</th><th scope=col>strand</th><th scope=col>exonStart_0base</th><th scope=col>exonEnd</th><th scope=col>upstreamES</th><th scope=col>upstreamEE</th><th scope=col>downstreamES</th><th scope=col>downstreamEE</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21287990</td><td>21288091</td><td>21284709</td><td>21284969</td><td>21295674</td><td>21295769</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21303182</td><td>21303234</td><td>21302142</td><td>21302259</td><td>21304425</td><td>21304553</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>3</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21287990</td><td>21288091</td><td>21296085</td><td>21296143</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>4</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21287990</td><td>21288091</td><td>21298412</td><td>21298479</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>5</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21284710</td><td>21284969</td><td>21296085</td><td>21296143</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>6</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21284710</td><td>21284969</td><td>21298412</td><td>21298479</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID & GeneID & geneSymbol & chr & strand & exonStart\\_0base & exonEnd & upstreamES & upstreamEE & downstreamES & downstreamEE\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21287990 & 21288091 & 21284709 & 21284969 & 21295674 & 21295769\\\\\n",
       "\t2 & 2 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21303182 & 21303234 & 21302142 & 21302259 & 21304425 & 21304553\\\\\n",
       "\t3 & 3 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21287990 & 21288091 & 21296085 & 21296143\\\\\n",
       "\t4 & 4 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21287990 & 21288091 & 21298412 & 21298479\\\\\n",
       "\t5 & 5 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21284710 & 21284969 & 21296085 & 21296143\\\\\n",
       "\t6 & 6 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21284710 & 21284969 & 21298412 & 21298479\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | ID &lt;int&gt; | GeneID &lt;chr&gt; | geneSymbol &lt;chr&gt; | chr &lt;chr&gt; | strand &lt;chr&gt; | exonStart_0base &lt;int&gt; | exonEnd &lt;int&gt; | upstreamES &lt;int&gt; | upstreamEE &lt;int&gt; | downstreamES &lt;int&gt; | downstreamEE &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21287990 | 21288091 | 21284709 | 21284969 | 21295674 | 21295769 |\n",
       "| 2 | 2 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21303182 | 21303234 | 21302142 | 21302259 | 21304425 | 21304553 |\n",
       "| 3 | 3 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21287990 | 21288091 | 21296085 | 21296143 |\n",
       "| 4 | 4 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21287990 | 21288091 | 21298412 | 21298479 |\n",
       "| 5 | 5 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21284710 | 21284969 | 21296085 | 21296143 |\n",
       "| 6 | 6 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21284710 | 21284969 | 21298412 | 21298479 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID GeneID             geneSymbol chr   strand exonStart_0base exonEnd \n",
       "1 1  ENSG00000034152.18 MAP2K3     chr17 +      21287990        21288091\n",
       "2 2  ENSG00000034152.18 MAP2K3     chr17 +      21303182        21303234\n",
       "3 3  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "4 4  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "5 5  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "6 6  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "  upstreamES upstreamEE downstreamES downstreamEE\n",
       "1 21284709   21284969   21295674     21295769    \n",
       "2 21302142   21302259   21304425     21304553    \n",
       "3 21287990   21288091   21296085     21296143    \n",
       "4 21287990   21288091   21298412     21298479    \n",
       "5 21284710   21284969   21296085     21296143    \n",
       "6 21284710   21284969   21298412     21298479    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(se_annot)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f810ec83",
   "metadata": {
    "papermill": {
     "duration": 0.03069,
     "end_time": "2021-05-21T02:44:03.745826",
     "exception": false,
     "start_time": "2021-05-21T02:44:03.715136",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.5 create_as_structure \n",
    "\n",
    "This function doees an aggregation of the alternative splicing events - good for all events and the significantly expressed events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9dd7c95b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:03.811794Z",
     "iopub.status.busy": "2021-05-21T02:44:03.810317Z",
     "iopub.status.idle": "2021-05-21T02:44:03.824018Z",
     "shell.execute_reply": "2021-05-21T02:44:03.823005Z"
    },
    "papermill": {
     "duration": 0.04836,
     "end_time": "2021-05-21T02:44:03.824147",
     "exception": false,
     "start_time": "2021-05-21T02:44:03.775787",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "create_as_structure <- function ( results_dir, files, all_or_das, pattern, tissue_reduction) {\n",
    "    gene_as = data.frame()\n",
    "    counts <- rep(NA, length(files))\n",
    "    message(\"\\nnumber of files:\", paste(length(files)), collapse = \"\")\n",
    "    for (i in 1:length(files)) {\n",
    "       lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "       if (dim(lines)[1] > 0) {\n",
    "           event     <- as.vector(as.character(rownames(lines)))\n",
    "           tissue1   <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "           counts[i] <- dim(lines)[1]\n",
    "           event_idx <- substring(event, regexpr(\"[0-9]+$\", event))\n",
    "           res       <- data.frame()\n",
    "           if (grepl(\"^a3ss_\", files[i])) {\n",
    "               # remove the first 5 letters of the string \n",
    "               tissue2 <- substring(tissue1,6)\n",
    "               idx <- match(event_idx, a3ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"A3SS\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- a3ss_annot$geneSymbol[idx],\n",
    "                              GeneID       <- a3ss_annot$GeneID[idx],\n",
    "                              chr          <- a3ss_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "            \n",
    "           } else if (grepl(\"^a5ss_\", files[i])) {\n",
    "               # remove the first 5 letters of the string \n",
    "               tissue2 <- substring(tissue1,6)\n",
    "               idx <- match(event_idx, a5ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"A5SS\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- a5ss_annot$geneSymbol[idx],\n",
    "                              GeneID       <- a5ss_annot$GeneID[idx],\n",
    "                              chr          <- a5ss_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                               \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^mxe_\", files[i])) {\n",
    "               # remove the first 4 letters of the string \n",
    "               tissue2 <- substring(tissue1,5)\n",
    "               idx <- match(event_idx, a3ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"MXE\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- mxe_annot$geneSymbol[idx],\n",
    "                              GeneID       <- mxe_annot$GeneID[idx],\n",
    "                              chr          <- mxe_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^se_\", files[i])) {\n",
    "               # remove the first 3 letters of the string \n",
    "               tissue2 <- substring(tissue1,4)\n",
    "               idx <- match(event_idx, se_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"SE\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- se_annot$geneSymbol[idx],\n",
    "                              GeneID       <- se_annot$GeneID[idx],\n",
    "                              chr          <- se_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^ri_\", files[i])){\n",
    "               # remove the first 3 letters of the string \n",
    "               tissue2 <- substring(tissue1,4)\n",
    "               idx <- match(event_idx, ri_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"RI\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- ri_annot$geneSymbol[idx],\n",
    "                              GeneID       <- ri_annot$GeneID[idx],\n",
    "                              chr          <- ri_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           }\n",
    "        \n",
    "       } #if has sig. events\n",
    "    \n",
    "   } #for all files\n",
    "   colnames(gene_as) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\"GeneSymbol\",\n",
    "                           \"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "   n_unique_genes <- length(summary(as.factor(gene_as$GeneSymbol),maxsum=50000))\n",
    "   message(\"For the run for \", all_or_das, \" run\")\n",
    "   message(\"We extracted a total of \",nrow(gene_as),\" alternative splicing events (gene_as)\")\n",
    "   message(\"This includes \", n_unique_genes, \" total genes\")\n",
    "   return (gene_as)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e33c87b3",
   "metadata": {
    "papermill": {
     "duration": 0.030809,
     "end_time": "2021-05-21T02:44:03.885692",
     "exception": false,
     "start_time": "2021-05-21T02:44:03.854883",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.6 create_dge_structure \n",
    "\n",
    "This function does an aggregation of the differential gene expression events - good for all events and the significantly expressed events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fdbcab91",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:03.950459Z",
     "iopub.status.busy": "2021-05-21T02:44:03.949238Z",
     "iopub.status.idle": "2021-05-21T02:44:03.962038Z",
     "shell.execute_reply": "2021-05-21T02:44:03.960973Z"
    },
    "papermill": {
     "duration": 0.04666,
     "end_time": "2021-05-21T02:44:03.962183",
     "exception": false,
     "start_time": "2021-05-21T02:44:03.915523",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "create_dge_structure <- function ( results_dir, files, all_or_dge, pattern, map_pattern, tissue_reduction) {\n",
    "   gene_dge = data.frame()\n",
    "   counts <- rep(NA, length(files))\n",
    "   for (i in 1:length(files)) {\n",
    "      lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "      if (dim(lines)[1] > 0) {\n",
    "         tissue1    <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "         map_lines  <- read.table(file=paste0(paste0(results_dir, tissue1),map_pattern),\n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "         counts[i]  <- dim(lines)[1]    \n",
    "         tissue1    <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "         map_lines  <- read.table(file=paste0(paste0(results_dir, tissue1),map_pattern),\n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "         ensg_ver   <- as.vector(as.character(rownames(lines)))\n",
    "         ensg_no_ver<- as.vector(as.character(map_lines$ensg_names))\n",
    "         ensg_genes <- as.vector(as.character(map_lines$ensg_genes))\n",
    "         counts[i]  <- dim(lines)[1]  \n",
    "         res <- data.frame(Tissue       <- tissue1,\n",
    "                           ENSG_ver     <- ensg_ver,\n",
    "                           ENSG_no_ver  <- ensg_no_ver,\n",
    "                           GeneSymbol   <- ensg_genes,\n",
    "                           counts       <- counts[i],\n",
    "                           Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue1, \"display_name\"],\n",
    "                           logFC        <- lines$logFC,\n",
    "                           AveExpr      <- lines$AveExpr,\n",
    "                           t            <- lines$t,\n",
    "                           PValue       <- lines$P.Value,\n",
    "                           AdjPVal      <- lines$adj.P.Val,\n",
    "                           B            <- lines$B)\n",
    "         colnames(res) <- c(\"Tissue\",\"ENSG_ver\",\"ENSG_no_ver\",\"GeneSymbol\",\"counts\",\"Display\",\n",
    "                            \"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "         gene_dge <- rbind(gene_dge, res)\n",
    "       } #if has sig. events\n",
    "    } #for all files\n",
    "    colnames(gene_dge) <- c(\"Tissue\",\"ENSG_ver\",\"ENSG_no_ver\",\"GeneSymbol\",\"counts\",\"Display\",\n",
    "                        \"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "    n_unique_genes <- length(summary(as.factor(gene_dge$GeneSymbol),maxsum=50000))\n",
    "    message(\"For the run for \", all_or_dge, \"run\")\n",
    "    message(\"We extracted a total of \",nrow(gene_dge),\" gene events (gene_dge)\")\n",
    "    message(\"This includes \", n_unique_genes, \" total genes\")\n",
    "    return(gene_dge)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3e6cb6d",
   "metadata": {
    "papermill": {
     "duration": 0.030941,
     "end_time": "2021-05-21T02:44:04.023726",
     "exception": false,
     "start_time": "2021-05-21T02:44:03.992785",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.7 Read in the alternative splicing results\n",
    "\n",
    "We will create an aggregation of  all the results and all the significant results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a2da57a1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:44:04.091685Z",
     "iopub.status.busy": "2021-05-21T02:44:04.090482Z",
     "iopub.status.idle": "2021-05-21T02:45:41.653043Z",
     "shell.execute_reply": "2021-05-21T02:45:41.651953Z"
    },
    "papermill": {
     "duration": 97.598076,
     "end_time": "2021-05-21T02:45:41.653198",
     "exception": false,
     "start_time": "2021-05-21T02:44:04.055122",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Length of all_files: 195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Length of significant_files: 195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "number of files:195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for differentially significant alternative splicing run\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted a total of 312 alternative splicing events (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "This includes 254 total genes\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "number of files:195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for all alternatively spliced run\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted a total of 1140938 alternative splicing events (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "This includes 12681 total genes\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneJunction</th><th scope=col>ASE</th><th scope=col>ASE_IDX</th><th scope=col>Tissue</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>GeneSymbol</th><th scope=col>GeneID</th><th scope=col>chr</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CYB5R1-2513</td><td>A3SS</td><td>2513</td><td>brain_hippocampus </td><td>1</td><td>Hippocampus </td><td>CYB5R1</td><td>ENSG00000159348.13</td><td>chr1 </td><td> 4.248889</td><td>5.779254</td><td> 5.126318</td><td>7.07777e-07</td><td>0.003560118</td><td>5.185446</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ANK3-4589  </td><td>A3SS</td><td>4589</td><td>brain_hypothalamus</td><td>1</td><td>Hypothalamus</td><td>ANK3  </td><td>ENSG00000151150.22</td><td>chr10</td><td>-6.419966</td><td>4.170207</td><td>-4.656180</td><td>6.14524e-06</td><td>0.032244072</td><td>3.210014</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & GeneJunction & ASE & ASE\\_IDX & Tissue & counts & Display & GeneSymbol & GeneID & chr & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <int> & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & CYB5R1-2513 & A3SS & 2513 & brain\\_hippocampus  & 1 & Hippocampus  & CYB5R1 & ENSG00000159348.13 & chr1  &  4.248889 & 5.779254 &  5.126318 & 7.07777e-07 & 0.003560118 & 5.185446\\\\\n",
       "\t2 & ANK3-4589   & A3SS & 4589 & brain\\_hypothalamus & 1 & Hypothalamus & ANK3   & ENSG00000151150.22 & chr10 & -6.419966 & 4.170207 & -4.656180 & 6.14524e-06 & 0.032244072 & 3.210014\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 15\n",
       "\n",
       "| <!--/--> | GeneJunction &lt;chr&gt; | ASE &lt;chr&gt; | ASE_IDX &lt;int&gt; | Tissue &lt;chr&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | GeneSymbol &lt;chr&gt; | GeneID &lt;chr&gt; | chr &lt;chr&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | CYB5R1-2513 | A3SS | 2513 | brain_hippocampus  | 1 | Hippocampus  | CYB5R1 | ENSG00000159348.13 | chr1  |  4.248889 | 5.779254 |  5.126318 | 7.07777e-07 | 0.003560118 | 5.185446 |\n",
       "| 2 | ANK3-4589   | A3SS | 4589 | brain_hypothalamus | 1 | Hypothalamus | ANK3   | ENSG00000151150.22 | chr10 | -6.419966 | 4.170207 | -4.656180 | 6.14524e-06 | 0.032244072 | 3.210014 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneJunction ASE  ASE_IDX Tissue             counts Display      GeneSymbol\n",
       "1 CYB5R1-2513  A3SS 2513    brain_hippocampus  1      Hippocampus  CYB5R1    \n",
       "2 ANK3-4589    A3SS 4589    brain_hypothalamus 1      Hypothalamus ANK3      \n",
       "  GeneID             chr   logFC     AveExpr  t         PValue      AdjPVal    \n",
       "1 ENSG00000159348.13 chr1   4.248889 5.779254  5.126318 7.07777e-07 0.003560118\n",
       "2 ENSG00000151150.22 chr10 -6.419966 4.170207 -4.656180 6.14524e-06 0.032244072\n",
       "  B       \n",
       "1 5.185446\n",
       "2 3.210014"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "results_dir         <- \"../data/\"\n",
    "significant_pattern <- \"_AS_model_B_sex_as_events_refined.csv\"\n",
    "significant_files   <- list.files(path = results_dir, pattern = significant_pattern)\n",
    "all_pattern         <- \"_AS_model_B_sex_as_events.csv\"\n",
    "all_files           <- list.files(path = results_dir, pattern = all_pattern)\n",
    "as_types            <- c(\"a3ss\", \"a5ss\", \"mxe\", \"ri\", \"se\")\n",
    "message(\"Length of all_files: \", length(all_files))\n",
    "message(\"Length of significant_files: \", length(significant_files))\n",
    "\n",
    "gene_as     <- create_as_structure (results_dir      <- results_dir, \n",
    "                                    files            <- significant_files,\n",
    "                                    all_or_das       <- \"differentially significant alternative splicing\",\n",
    "                                    pattern          <- significant_pattern, \n",
    "                                    tissue_reduction <- tissue_reduction)\n",
    "all_gene_as <- create_as_structure (results_dir      <- results_dir, \n",
    "                                    files            <- all_files, \n",
    "                                    all_or_das       <- \"all alternatively spliced\",\n",
    "                                    pattern          <- all_pattern, \n",
    "                                    tissue_reduction <- tissue_reduction)\n",
    "head(gene_as,2)\n",
    "gene_as$Tissue <- factor(gene_as$Tissue)\n",
    "write.table(gene_as, \"../data/gene_as.tsv\", quote=FALSE, sep=\"\\t\")\n",
    "write.table(all_gene_as, \"../data/all_gene_as.tsv\", quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "652731e2",
   "metadata": {
    "papermill": {
     "duration": 0.033515,
     "end_time": "2021-05-21T02:45:41.721291",
     "exception": false,
     "start_time": "2021-05-21T02:45:41.687776",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.8 Create a genes-id file capturing the unique gene-junction locations in a single file\n",
    "\n",
    "rMATS 3.2.5 unique junction ids by splicing event tied together with gene names and these identifiers useful for downstream analyses and investigations.\n",
    "only the SE significant AS events here\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b82c6e81",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:45:41.792967Z",
     "iopub.status.busy": "2021-05-21T02:45:41.791696Z",
     "iopub.status.idle": "2021-05-21T02:45:42.094184Z",
     "shell.execute_reply": "2021-05-21T02:45:42.093053Z"
    },
    "papermill": {
     "duration": 0.34001,
     "end_time": "2021-05-21T02:45:42.094307",
     "exception": false,
     "start_time": "2021-05-21T02:45:41.754297",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The first file from ^se_*AS_model_B_sex_as_events_refined.csv: se_adipose_subcutaneous_AS_model_B_sex_as_events_refined.csv\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done writing 39 files.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results_dir         <- \"../data/\"\n",
    "significant_pattern <- \"^se_*AS_model_B_sex_as_events_refined.csv\"\n",
    "files   <- list.files(path =results_dir, pattern = glob2rx(significant_pattern))\n",
    "message(\"The first file from ^se_*AS_model_B_sex_as_events_refined.csv: \", files[1])\n",
    "pattern=\"_AS_model_B_sex_as_events_refined.csv\"\n",
    "geneids <- data.frame()\n",
    "for (i in 1:length(files)) {\n",
    "    lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "    \n",
    "    if (dim(lines)[1] > 0) {\n",
    "           event     <- as.vector(as.character(rownames(lines)))\n",
    "           tissue1   <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "           event_idx <- substring(event, regexpr(\"[0-9]+$\", event))\n",
    "           res       <- data.frame()\n",
    "           tissue2 <- substring(tissue1,4)\n",
    "           idx <- match(event_idx, se_annot$ID)\n",
    "           res <- data.frame(geneIDs      <- event,\n",
    "                             ID           <- event_idx,\n",
    "                             GeneSymbol   <- se_annot$geneSymbol[idx],\n",
    "                             GeneID       <- se_annot$GeneID[idx],\n",
    "                             chr          <- se_annot$chr[idx])\n",
    "           outfilename <- paste0(paste0(\"../data/se_\",tissue2),\"_geneids.tsv\")\n",
    "           write.table(res, outfilename, quote=FALSE, sep=\"\\t\")\n",
    "           \n",
    "     }\n",
    "}\n",
    "message(\"Done writing \", length(files), \" files.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4d39658",
   "metadata": {
    "papermill": {
     "duration": 0.033896,
     "end_time": "2021-05-21T02:45:42.162221",
     "exception": false,
     "start_time": "2021-05-21T02:45:42.128325",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.9 Read in the differential gene expression results\n",
    "\n",
    "Here we create an aggregation of al the significant results differential gene expression events.\n",
    "Note the all_gene_dge.tsv may be found in the assets directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c14b2c63",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:45:42.234648Z",
     "iopub.status.busy": "2021-05-21T02:45:42.233404Z",
     "iopub.status.idle": "2021-05-21T02:45:43.240677Z",
     "shell.execute_reply": "2021-05-21T02:45:43.239880Z"
    },
    "papermill": {
     "duration": 1.044995,
     "end_time": "2021-05-21T02:45:43.240842",
     "exception": false,
     "start_time": "2021-05-21T02:45:42.195847",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "195"
      ],
      "text/latex": [
       "195"
      ],
      "text/markdown": [
       "195"
      ],
      "text/plain": [
       "[1] 195"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for differential gene expressionrun\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted a total of 4219 gene events (gene_dge)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "This includes 3148 total genes\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Tissue</th><th scope=col>ENSG_ver</th><th scope=col>ENSG_no_ver</th><th scope=col>GeneSymbol</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>adipose_subcutaneous</td><td>ENSG00000147050.14</td><td>ENSG00000147050</td><td>KDM6A </td><td>247</td><td>Adipose (sc)</td><td> 0.6133494</td><td>5.030862</td><td> 35.85874</td><td>3.150846e-157</td><td>5.079794e-153</td><td>347.77639</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>adipose_subcutaneous</td><td>ENSG00000115041.12</td><td>ENSG00000115041</td><td>KCNIP3</td><td>247</td><td>Adipose (sc)</td><td>-0.9458500</td><td>3.453238</td><td>-15.42637</td><td> 4.374046e-46</td><td> 6.410760e-43</td><td> 93.87157</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & Tissue & ENSG\\_ver & ENSG\\_no\\_ver & GeneSymbol & counts & Display & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & adipose\\_subcutaneous & ENSG00000147050.14 & ENSG00000147050 & KDM6A  & 247 & Adipose (sc) &  0.6133494 & 5.030862 &  35.85874 & 3.150846e-157 & 5.079794e-153 & 347.77639\\\\\n",
       "\t2 & adipose\\_subcutaneous & ENSG00000115041.12 & ENSG00000115041 & KCNIP3 & 247 & Adipose (sc) & -0.9458500 & 3.453238 & -15.42637 &  4.374046e-46 &  6.410760e-43 &  93.87157\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 12\n",
       "\n",
       "| <!--/--> | Tissue &lt;chr&gt; | ENSG_ver &lt;chr&gt; | ENSG_no_ver &lt;chr&gt; | GeneSymbol &lt;chr&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | adipose_subcutaneous | ENSG00000147050.14 | ENSG00000147050 | KDM6A  | 247 | Adipose (sc) |  0.6133494 | 5.030862 |  35.85874 | 3.150846e-157 | 5.079794e-153 | 347.77639 |\n",
       "| 2 | adipose_subcutaneous | ENSG00000115041.12 | ENSG00000115041 | KCNIP3 | 247 | Adipose (sc) | -0.9458500 | 3.453238 | -15.42637 |  4.374046e-46 |  6.410760e-43 |  93.87157 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue               ENSG_ver           ENSG_no_ver     GeneSymbol counts\n",
       "1 adipose_subcutaneous ENSG00000147050.14 ENSG00000147050 KDM6A      247   \n",
       "2 adipose_subcutaneous ENSG00000115041.12 ENSG00000115041 KCNIP3     247   \n",
       "  Display      logFC      AveExpr  t         PValue        AdjPVal      \n",
       "1 Adipose (sc)  0.6133494 5.030862  35.85874 3.150846e-157 5.079794e-153\n",
       "2 Adipose (sc) -0.9458500 3.453238 -15.42637  4.374046e-46  6.410760e-43\n",
       "  B        \n",
       "1 347.77639\n",
       "2  93.87157"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "results_dir             <- \"../data/\"\n",
    "significant_dge_pattern <- \"_DGE_refined.csv\"\n",
    "significant_dge_files   <- list.files(path = results_dir, pattern = significant_dge_pattern)\n",
    "map_pattern             <- \"_DGE_ensg_map.csv\"\n",
    "length(significant_files)\n",
    "\n",
    "gene_dge     <- create_dge_structure (results_dir      <- results_dir, \n",
    "                                      files            <- significant_dge_files, \n",
    "                                      all_or_dge       <- \"differential gene expression\",\n",
    "                                      pattern          <- significant_dge_pattern, \n",
    "                                      map_pattern      <- map_pattern,\n",
    "                                      tissue_reduction <- tissue_reduction)\n",
    "\n",
    "head(gene_dge,2)\n",
    "gene_dge$Tissue <- factor(gene_dge$Tissue)\n",
    "write.table(gene_dge,     \"../data/gene_dge.tsv\",     quote=FALSE, sep=\"\\t\") "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2d0da13",
   "metadata": {
    "papermill": {
     "duration": 0.034685,
     "end_time": "2021-05-21T02:45:43.311227",
     "exception": false,
     "start_time": "2021-05-21T02:45:43.276542",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.10 Read in Gencode (v30) Complete Annotation file.\n",
    "\n",
    "Load in the gencode.v30.annotation.gtf file for additional annotation.\n",
    "chr information for summary data later, use the annotation we used for rMATS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f19dd94d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:45:43.385576Z",
     "iopub.status.busy": "2021-05-21T02:45:43.384447Z",
     "iopub.status.idle": "2021-05-21T02:46:30.208815Z",
     "shell.execute_reply": "2021-05-21T02:46:30.209319Z"
    },
    "papermill": {
     "duration": 46.86371,
     "end_time": "2021-05-21T02:46:30.209500",
     "exception": false,
     "start_time": "2021-05-21T02:45:43.345790",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "downloading gencode v30 annotation\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unzipping compressed file gencode.v30.annotation.gtf.gz..\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)\n",
       "attr(,\"status\")\n",
       "[1] 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done! gencode.v30.annotation.gtf can be found in ../data/\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>GeneSymbol</th><th scope=col>ENSG</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>DDX11L1    </td><td>ENSG00000223972.5</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>WASH7P     </td><td>ENSG00000227232.5</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>MIR6859-1  </td><td>ENSG00000278267.1</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>chr1</td><td>MIR1302-2HG</td><td>ENSG00000243485.5</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>chr1</td><td>MIR1302-2  </td><td>ENSG00000284332.1</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>chr1</td><td>FAM138A    </td><td>ENSG00000237613.2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & GeneSymbol & ENSG\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & DDX11L1     & ENSG00000223972.5\\\\\n",
       "\t13 & chr1 & WASH7P      & ENSG00000227232.5\\\\\n",
       "\t26 & chr1 & MIR6859-1   & ENSG00000278267.1\\\\\n",
       "\t29 & chr1 & MIR1302-2HG & ENSG00000243485.5\\\\\n",
       "\t37 & chr1 & MIR1302-2   & ENSG00000284332.1\\\\\n",
       "\t40 & chr1 & FAM138A     & ENSG00000237613.2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | chr &lt;fct&gt; | GeneSymbol &lt;chr&gt; | ENSG &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chr1 | DDX11L1     | ENSG00000223972.5 |\n",
       "| 13 | chr1 | WASH7P      | ENSG00000227232.5 |\n",
       "| 26 | chr1 | MIR6859-1   | ENSG00000278267.1 |\n",
       "| 29 | chr1 | MIR1302-2HG | ENSG00000243485.5 |\n",
       "| 37 | chr1 | MIR1302-2   | ENSG00000284332.1 |\n",
       "| 40 | chr1 | FAM138A     | ENSG00000237613.2 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  GeneSymbol  ENSG             \n",
       "1  chr1 DDX11L1     ENSG00000223972.5\n",
       "13 chr1 WASH7P      ENSG00000227232.5\n",
       "26 chr1 MIR6859-1   ENSG00000278267.1\n",
       "29 chr1 MIR1302-2HG ENSG00000243485.5\n",
       "37 chr1 MIR1302-2   ENSG00000284332.1\n",
       "40 chr1 FAM138A     ENSG00000237613.2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "message(\"downloading gencode v30 annotation\\n\")\n",
    "system(\"wget -O ../data/gencode.v30.annotation.gtf.gz ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gtf.gz\")\n",
    "message(\"Done!\\n\")\n",
    "message(\"Unzipping compressed file gencode.v30.annotation.gtf.gz..\")\n",
    "system(\"gunzip ../data/gencode.v30.annotation.gtf.gz\", intern = TRUE)\n",
    "message(\"Done! gencode.v30.annotation.gtf can be found in ../data/\")\n",
    "\n",
    "gencode <- import(\"../data/gencode.v30.annotation.gtf\")\n",
    "gtf.df <- as.data.frame (gencode)\n",
    "chr_genes <- unique(gtf.df[,c(\"seqnames\",\"gene_name\",\"gene_id\")])\n",
    "colnames(chr_genes) <- c(\"chr\",\"GeneSymbol\", \"ENSG\")\n",
    "head(chr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "6b530021",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:30.289669Z",
     "iopub.status.busy": "2021-05-21T02:46:30.288356Z",
     "iopub.status.idle": "2021-05-21T02:46:41.043293Z",
     "shell.execute_reply": "2021-05-21T02:46:41.042004Z"
    },
    "papermill": {
     "duration": 10.796766,
     "end_time": "2021-05-21T02:46:41.043469",
     "exception": false,
     "start_time": "2021-05-21T02:46:30.246703",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>GeneSymbol</th><th scope=col>ENSG</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>DDX11L1    </td><td>ENSG00000223972</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>WASH7P     </td><td>ENSG00000227232</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>MIR6859-1  </td><td>ENSG00000278267</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>chr1</td><td>MIR1302-2HG</td><td>ENSG00000243485</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>chr1</td><td>MIR1302-2  </td><td>ENSG00000284332</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>chr1</td><td>FAM138A    </td><td>ENSG00000237613</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & GeneSymbol & ENSG\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & DDX11L1     & ENSG00000223972\\\\\n",
       "\t13 & chr1 & WASH7P      & ENSG00000227232\\\\\n",
       "\t26 & chr1 & MIR6859-1   & ENSG00000278267\\\\\n",
       "\t29 & chr1 & MIR1302-2HG & ENSG00000243485\\\\\n",
       "\t37 & chr1 & MIR1302-2   & ENSG00000284332\\\\\n",
       "\t40 & chr1 & FAM138A     & ENSG00000237613\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | chr &lt;fct&gt; | GeneSymbol &lt;chr&gt; | ENSG &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chr1 | DDX11L1     | ENSG00000223972 |\n",
       "| 13 | chr1 | WASH7P      | ENSG00000227232 |\n",
       "| 26 | chr1 | MIR6859-1   | ENSG00000278267 |\n",
       "| 29 | chr1 | MIR1302-2HG | ENSG00000243485 |\n",
       "| 37 | chr1 | MIR1302-2   | ENSG00000284332 |\n",
       "| 40 | chr1 | FAM138A     | ENSG00000237613 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  GeneSymbol  ENSG           \n",
       "1  chr1 DDX11L1     ENSG00000223972\n",
       "13 chr1 WASH7P      ENSG00000227232\n",
       "26 chr1 MIR6859-1   ENSG00000278267\n",
       "29 chr1 MIR1302-2HG ENSG00000243485\n",
       "37 chr1 MIR1302-2   ENSG00000284332\n",
       "40 chr1 FAM138A     ENSG00000237613"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for (i in 1:dim(chr_genes)[1]) {\n",
    "    chr_genes$ENSG[i] <- as.character(strsplit(chr_genes$ENSG[i],'\\\\.\\\\w+$'))\n",
    "}\n",
    "head(chr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "97863e79",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:41.125304Z",
     "iopub.status.busy": "2021-05-21T02:46:41.124265Z",
     "iopub.status.idle": "2021-05-21T02:46:52.423509Z",
     "shell.execute_reply": "2021-05-21T02:46:52.422328Z"
    },
    "papermill": {
     "duration": 11.341292,
     "end_time": "2021-05-21T02:46:52.423687",
     "exception": false,
     "start_time": "2021-05-21T02:46:41.082395",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Tissue</th><th scope=col>ENSG_ver</th><th scope=col>ENSG_no_ver</th><th scope=col>GeneSymbol</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th><th scope=col>chr</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>adipose_subcutaneous</td><td>ENSG00000147050.14</td><td>ENSG00000147050</td><td>KDM6A     </td><td>247</td><td>Adipose (sc)</td><td> 0.6133494</td><td> 5.0308617</td><td> 35.85874</td><td>3.150846e-157</td><td>5.079794e-153</td><td>347.77639</td><td>chrX </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>adipose_subcutaneous</td><td>ENSG00000115041.12</td><td>ENSG00000115041</td><td>KCNIP3    </td><td>247</td><td>Adipose (sc)</td><td>-0.9458500</td><td> 3.4532379</td><td>-15.42637</td><td> 4.374046e-46</td><td> 6.410760e-43</td><td> 93.87157</td><td>chr2 </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>adipose_subcutaneous</td><td>ENSG00000224525.2 </td><td>ENSG00000224525</td><td>AL591686.1</td><td>247</td><td>Adipose (sc)</td><td> 2.2244583</td><td>-1.4641204</td><td> 14.45808</td><td> 2.280549e-41</td><td> 2.828232e-38</td><td> 81.34529</td><td>chr1 </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>adipose_subcutaneous</td><td>ENSG00000141449.14</td><td>ENSG00000141449</td><td>GREB1L    </td><td>247</td><td>Adipose (sc)</td><td>-1.4860060</td><td> 2.7133806</td><td>-13.23114</td><td> 1.272824e-35</td><td> 1.207086e-32</td><td> 69.99979</td><td>chr18</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>adipose_subcutaneous</td><td>ENSG00000258484.3 </td><td>ENSG00000258484</td><td>SPESP1    </td><td>247</td><td>Adipose (sc)</td><td>-0.8885441</td><td> 0.9410518</td><td>-13.02960</td><td> 1.047185e-34</td><td> 8.441355e-32</td><td> 67.50586</td><td>chr15</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>adipose_subcutaneous</td><td>ENSG00000134339.8 </td><td>ENSG00000134339</td><td>SAA2      </td><td>247</td><td>Adipose (sc)</td><td> 2.9752333</td><td> 2.7856727</td><td> 12.82938</td><td> 8.397650e-34</td><td> 6.260957e-31</td><td> 65.94905</td><td>chr11</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & Tissue & ENSG\\_ver & ENSG\\_no\\_ver & GeneSymbol & counts & Display & logFC & AveExpr & t & PValue & AdjPVal & B & chr\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr> & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & adipose\\_subcutaneous & ENSG00000147050.14 & ENSG00000147050 & KDM6A      & 247 & Adipose (sc) &  0.6133494 &  5.0308617 &  35.85874 & 3.150846e-157 & 5.079794e-153 & 347.77639 & chrX \\\\\n",
       "\t2 & adipose\\_subcutaneous & ENSG00000115041.12 & ENSG00000115041 & KCNIP3     & 247 & Adipose (sc) & -0.9458500 &  3.4532379 & -15.42637 &  4.374046e-46 &  6.410760e-43 &  93.87157 & chr2 \\\\\n",
       "\t3 & adipose\\_subcutaneous & ENSG00000224525.2  & ENSG00000224525 & AL591686.1 & 247 & Adipose (sc) &  2.2244583 & -1.4641204 &  14.45808 &  2.280549e-41 &  2.828232e-38 &  81.34529 & chr1 \\\\\n",
       "\t4 & adipose\\_subcutaneous & ENSG00000141449.14 & ENSG00000141449 & GREB1L     & 247 & Adipose (sc) & -1.4860060 &  2.7133806 & -13.23114 &  1.272824e-35 &  1.207086e-32 &  69.99979 & chr18\\\\\n",
       "\t5 & adipose\\_subcutaneous & ENSG00000258484.3  & ENSG00000258484 & SPESP1     & 247 & Adipose (sc) & -0.8885441 &  0.9410518 & -13.02960 &  1.047185e-34 &  8.441355e-32 &  67.50586 & chr15\\\\\n",
       "\t6 & adipose\\_subcutaneous & ENSG00000134339.8  & ENSG00000134339 & SAA2       & 247 & Adipose (sc) &  2.9752333 &  2.7856727 &  12.82938 &  8.397650e-34 &  6.260957e-31 &  65.94905 & chr11\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 13\n",
       "\n",
       "| <!--/--> | Tissue &lt;fct&gt; | ENSG_ver &lt;chr&gt; | ENSG_no_ver &lt;chr&gt; | GeneSymbol &lt;chr&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; | chr &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | adipose_subcutaneous | ENSG00000147050.14 | ENSG00000147050 | KDM6A      | 247 | Adipose (sc) |  0.6133494 |  5.0308617 |  35.85874 | 3.150846e-157 | 5.079794e-153 | 347.77639 | chrX  |\n",
       "| 2 | adipose_subcutaneous | ENSG00000115041.12 | ENSG00000115041 | KCNIP3     | 247 | Adipose (sc) | -0.9458500 |  3.4532379 | -15.42637 |  4.374046e-46 |  6.410760e-43 |  93.87157 | chr2  |\n",
       "| 3 | adipose_subcutaneous | ENSG00000224525.2  | ENSG00000224525 | AL591686.1 | 247 | Adipose (sc) |  2.2244583 | -1.4641204 |  14.45808 |  2.280549e-41 |  2.828232e-38 |  81.34529 | chr1  |\n",
       "| 4 | adipose_subcutaneous | ENSG00000141449.14 | ENSG00000141449 | GREB1L     | 247 | Adipose (sc) | -1.4860060 |  2.7133806 | -13.23114 |  1.272824e-35 |  1.207086e-32 |  69.99979 | chr18 |\n",
       "| 5 | adipose_subcutaneous | ENSG00000258484.3  | ENSG00000258484 | SPESP1     | 247 | Adipose (sc) | -0.8885441 |  0.9410518 | -13.02960 |  1.047185e-34 |  8.441355e-32 |  67.50586 | chr15 |\n",
       "| 6 | adipose_subcutaneous | ENSG00000134339.8  | ENSG00000134339 | SAA2       | 247 | Adipose (sc) |  2.9752333 |  2.7856727 |  12.82938 |  8.397650e-34 |  6.260957e-31 |  65.94905 | chr11 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue               ENSG_ver           ENSG_no_ver     GeneSymbol counts\n",
       "1 adipose_subcutaneous ENSG00000147050.14 ENSG00000147050 KDM6A      247   \n",
       "2 adipose_subcutaneous ENSG00000115041.12 ENSG00000115041 KCNIP3     247   \n",
       "3 adipose_subcutaneous ENSG00000224525.2  ENSG00000224525 AL591686.1 247   \n",
       "4 adipose_subcutaneous ENSG00000141449.14 ENSG00000141449 GREB1L     247   \n",
       "5 adipose_subcutaneous ENSG00000258484.3  ENSG00000258484 SPESP1     247   \n",
       "6 adipose_subcutaneous ENSG00000134339.8  ENSG00000134339 SAA2       247   \n",
       "  Display      logFC      AveExpr    t         PValue        AdjPVal      \n",
       "1 Adipose (sc)  0.6133494  5.0308617  35.85874 3.150846e-157 5.079794e-153\n",
       "2 Adipose (sc) -0.9458500  3.4532379 -15.42637  4.374046e-46  6.410760e-43\n",
       "3 Adipose (sc)  2.2244583 -1.4641204  14.45808  2.280549e-41  2.828232e-38\n",
       "4 Adipose (sc) -1.4860060  2.7133806 -13.23114  1.272824e-35  1.207086e-32\n",
       "5 Adipose (sc) -0.8885441  0.9410518 -13.02960  1.047185e-34  8.441355e-32\n",
       "6 Adipose (sc)  2.9752333  2.7856727  12.82938  8.397650e-34  6.260957e-31\n",
       "  B         chr  \n",
       "1 347.77639 chrX \n",
       "2  93.87157 chr2 \n",
       "3  81.34529 chr1 \n",
       "4  69.99979 chr18\n",
       "5  67.50586 chr15\n",
       "6  65.94905 chr11"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "i = 1\n",
    "chr <- rep(\"NA\",dim(gene_dge)[1])\n",
    "gene_dge$chr <- chr\n",
    "for (i in 1:dim(gene_dge)[1]) {\n",
    "    match  <- as.character(chr_genes$ENSG) %in% as.character((gene_dge$ENSG_no_ver[i]))\n",
    "    if (sum(match==TRUE)== 1) {\n",
    "        chr[i] <- as.character(chr_genes[match,]$chr)\n",
    "        gene_dge$chr[i] <- chr[i]\n",
    "    } else if (sum(match==TRUE)>1) {\n",
    "        all <- as.vector(as.character(chr_genes[match,]$chr))\n",
    "        gene_dge$chr[i] <- as.character(all[1])\n",
    "    }\n",
    "}\n",
    "head(gene_dge)\n",
    "\n",
    "write.table(gene_dge, \"../data/gene_dge.tsv\", quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28f70102",
   "metadata": {
    "papermill": {
     "duration": 0.037731,
     "end_time": "2021-05-21T02:46:52.499881",
     "exception": false,
     "start_time": "2021-05-21T02:46:52.462150",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.11 Summary AS \n",
    "\n",
    "Capture descriptive statistics of the signficiantly alternatively splicedgene_as and significantly differentially gene expressed, gene_dge regarding events by tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "268a82d7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:52.580749Z",
     "iopub.status.busy": "2021-05-21T02:46:52.579530Z",
     "iopub.status.idle": "2021-05-21T02:46:52.651513Z",
     "shell.execute_reply": "2021-05-21T02:46:52.650192Z"
    },
    "papermill": {
     "duration": 0.11412,
     "end_time": "2021-05-21T02:46:52.651661",
     "exception": false,
     "start_time": "2021-05-21T02:46:52.537541",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Tissue</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>adipose_subcutaneous               </td><td>1</td></tr>\n",
       "\t<tr><td>esophagus_gastroesophageal_junction</td><td>1</td></tr>\n",
       "\t<tr><td>esophagus_mucosa                   </td><td>1</td></tr>\n",
       "\t<tr><td>nerve_tibial                       </td><td>1</td></tr>\n",
       "\t<tr><td>small_intestine_terminal_ileum     </td><td>1</td></tr>\n",
       "\t<tr><td>brain_hippocampus                  </td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Tissue & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t adipose\\_subcutaneous                & 1\\\\\n",
       "\t esophagus\\_gastroesophageal\\_junction & 1\\\\\n",
       "\t esophagus\\_mucosa                    & 1\\\\\n",
       "\t nerve\\_tibial                        & 1\\\\\n",
       "\t small\\_intestine\\_terminal\\_ileum      & 1\\\\\n",
       "\t brain\\_hippocampus                   & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Tissue &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| adipose_subcutaneous                | 1 |\n",
       "| esophagus_gastroesophageal_junction | 1 |\n",
       "| esophagus_mucosa                    | 1 |\n",
       "| nerve_tibial                        | 1 |\n",
       "| small_intestine_terminal_ileum      | 1 |\n",
       "| brain_hippocampus                   | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue                              n\n",
       "1 adipose_subcutaneous                1\n",
       "2 esophagus_gastroesophageal_junction 1\n",
       "3 esophagus_mucosa                    1\n",
       "4 nerve_tibial                        1\n",
       "5 small_intestine_terminal_ileum      1\n",
       "6 brain_hippocampus                   2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum splicing events per tissue 1 maximum splicing events per tissue 199\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sum of significant splicing events per tissue less than 10027\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Tissue</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>brain_spinal_cord_cervical_c_1</td><td> 5</td></tr>\n",
       "\t<tr><td>brain_cortex                  </td><td> 7</td></tr>\n",
       "\t<tr><td>brain_hippocampus             </td><td> 8</td></tr>\n",
       "\t<tr><td>colon_transverse              </td><td> 8</td></tr>\n",
       "\t<tr><td>small_intestine_terminal_ileum</td><td> 9</td></tr>\n",
       "\t<tr><td>colon_sigmoid                 </td><td>10</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Tissue & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t brain\\_spinal\\_cord\\_cervical\\_c\\_1 &  5\\\\\n",
       "\t brain\\_cortex                   &  7\\\\\n",
       "\t brain\\_hippocampus              &  8\\\\\n",
       "\t colon\\_transverse               &  8\\\\\n",
       "\t small\\_intestine\\_terminal\\_ileum &  9\\\\\n",
       "\t colon\\_sigmoid                  & 10\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Tissue &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| brain_spinal_cord_cervical_c_1 |  5 |\n",
       "| brain_cortex                   |  7 |\n",
       "| brain_hippocampus              |  8 |\n",
       "| colon_transverse               |  8 |\n",
       "| small_intestine_terminal_ileum |  9 |\n",
       "| colon_sigmoid                  | 10 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue                         n \n",
       "1 brain_spinal_cord_cervical_c_1  5\n",
       "2 brain_cortex                    7\n",
       "3 brain_hippocampus               8\n",
       "4 colon_transverse                8\n",
       "5 small_intestine_terminal_ileum  9\n",
       "6 colon_sigmoid                  10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum gene expression events per tissue 5 maximum gene expression events per tissue 2301\n",
      "\n"
     ]
    }
   ],
   "source": [
    "XY <- gene_as %>% group_by(Tissue) %>% tally()\n",
    "XY <- XY[order(XY$n),decreasing=TRUE]\n",
    "head(XY)\n",
    "message(\"Minimum splicing events per tissue \", min(XY$n), \" maximum splicing events per tissue \", max(XY$n))\n",
    "message(\"Sum of significant splicing events per tissue less than 100\", sum(XY$n<100))\n",
    "\n",
    "XY <- gene_dge %>% group_by(Tissue) %>% tally()\n",
    "XY <- XY[order(XY$n),decreasing=TRUE]\n",
    "head(XY)\n",
    "message(\"Minimum gene expression events per tissue \", min(XY$n), \" maximum gene expression events per tissue \", max(XY$n))\n",
    "# table(gtf.df[,c(\"gene_type\")])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5a38f01",
   "metadata": {
    "papermill": {
     "duration": 0.04034,
     "end_time": "2021-05-21T02:46:52.732379",
     "exception": false,
     "start_time": "2021-05-21T02:46:52.692039",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3 Data Structures for Figures"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d99e014",
   "metadata": {
    "papermill": {
     "duration": 0.03958,
     "end_time": "2021-05-21T02:46:52.812125",
     "exception": false,
     "start_time": "2021-05-21T02:46:52.772545",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.1 gene_as.tsv\n",
    "\n",
    "This file contains (description)\n",
    "Here is a typical line\n",
    "<pre>\n",
    "A data.frame: 6 × 15\n",
    "GeneJunction\tASE\tASE_IDX\tTissue\tcounts\tDisplay\tGeneSymbol\tGeneID\tchr\tlogFC\tAveExpr\tt\tPValue\tAdjPVal\tB\n",
    "<fct>\t<fct>\t<int>\t<fct>\t<int>\t<fct>\t<fct>\t<fct>\t<fct>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\n",
    "1\tXIST-2253\tA3SS\t2253\tadipose_subcutaneous\t4\tAdipose (sc)\tXIST\tENSG00000229807.11\tchrX\t-4.4086049\t3.196317\t-36.488970\t4.635568e-154\t3.893877e-150\t310.016049\n",
    "2\tXIST-2252\tA3SS\t2252\tadipose_subcutaneous\t4\tAdipose (sc)\tXIST\tENSG00000229807.11\tchrX\t-2.4147126\t3.647690\t-21.921057\t1.444102e-78\t6.065229e-75\t160.028167\n",
    "3\tGREB1L-4933\tA3SS\t4933\tadipose_subcutaneous\t4\tAdipose (sc)\tGREB1L\tENSG00000141449.14\tchr18\t1.2793173\t2.115005\t7.123138\t3.052112e-12\t8.545914e-09\t16.692429\n",
    "4\tRHCG-1776\tA3SS\t1776\tadipose_subcutaneous\t4\tAdipose (sc)\tRHCG\tENSG00000140519.14\tchr15\t-0.6930009\t1.636472\t-3.922124\t9.797866e-05\t3.919146e-02\t1.142232\n",
    "5\tXIST-2253\tA3SS\t2253\tadipose_visceral_omentum\t12\tAdipose (v)\tXIST\tENSG00000229807.11\tchrX\t-4.4403352\t3.113532\t-33.950800\t2.654474e-123\t2.209585e-119\t241.826117\n",
    "6\tXIST-2252\tA3SS\t2252\tadipose_visceral_omentum\t12\tAdipose (v)\tXIST\tENSG00000229807.11\tchrX\t-2.4506832\t3.650617\t-18.890779\t2.817671e-58\t1.172715e-54\t114.731682\n",
    "</pre>\n",
    "There are 2887 significant events in the file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3aa06d7a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:52.897344Z",
     "iopub.status.busy": "2021-05-21T02:46:52.896163Z",
     "iopub.status.idle": "2021-05-21T02:46:52.996157Z",
     "shell.execute_reply": "2021-05-21T02:46:52.994975Z"
    },
    "papermill": {
     "duration": 0.144112,
     "end_time": "2021-05-21T02:46:52.996291",
     "exception": false,
     "start_time": "2021-05-21T02:46:52.852179",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 312\n",
      "Columns: 15\n",
      "$ GeneJunction \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"CYB5R1-2513\", \"ANK3-4589\", \"RPS24-7852\", \"ODF2-2715\", \"I…\n",
      "$ ASE          \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"…\n",
      "$ ASE_IDX      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 2513, 4589, 7852, 2715, 8102, 3445, 8615, 2128, 3350, 225…\n",
      "$ Tissue       \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m brain_hippocampus, brain_hypothalamus, breast_mammary_tis…\n",
      "$ counts       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 1, 1, 7, 7, 7, 7, 7, 7, 7, 1, 1, 1, 1, 3, 3, 3, 1, 1, 1, …\n",
      "$ Display      \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m Hippocampus, Hypothalamus, Breast, Breast, Breast, Breast…\n",
      "$ GeneSymbol   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"CYB5R1\", \"ANK3\", \"RPS24\", \"ODF2\", \"IP6K2\", \"PITRM1\", \"PP…\n",
      "$ GeneID       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"ENSG00000159348.13\", \"ENSG00000151150.22\", \"ENSG00000138…\n",
      "$ chr          \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"chr1\", \"chr10\", \"chr10\", \"chr9\", \"chr3\", \"chr10\", \"chr10…\n",
      "$ logFC        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 4.2488889, -6.4199664, -3.5211998, 2.0702900, 2.7260476, …\n",
      "$ AveExpr      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 5.779254, 4.170207, 10.731240, 4.540094, 1.621525, 7.3524…\n",
      "$ t            \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 5.126318, -4.656180, -4.756402, 4.396519, 4.452328, -4.25…\n",
      "$ PValue       \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 7.077770e-07, 6.145240e-06, 2.899873e-06, 1.450855e-05, 1…\n",
      "$ AdjPVal      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 0.0035601181, 0.0322440717, 0.0160536974, 0.0267731030, 0…\n",
      "$ B            \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 5.185446, 3.210014, 4.243538, 2.720328, 2.634277, 2.30367…\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "28"
      ],
      "text/latex": [
       "28"
      ],
      "text/markdown": [
       "28"
      ],
      "text/plain": [
       "[1] 28"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE \n",
       "  312 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "         Adipose (sc)           Adipose (v)         Adrenal gland \n",
       "                    1                     6                     0 \n",
       "                Aorta      Atrial appendage                Breast \n",
       "                    3                     0                   199 \n",
       "              Caudate Cerebellar hemisphere            Cerebellum \n",
       "                    0                     6                     9 \n",
       "      Coronary artery                Cortex       EBV-lymphocytes \n",
       "                    0                    13                     3 \n",
       "      Esophagus (gej)         Esophagus (m)        Esophagus (mu) \n",
       "                    1                     1                     6 \n",
       "          Fibroblasts        Frontal cortex           Hippocampus \n",
       "                    2                     0                     2 \n",
       "         Hypothalamus        Left ventricle                 Liver \n",
       "                    3                     0                     4 \n",
       "                 Lung     Nucleus accumbens              Pancreas \n",
       "                    0                     2                     3 \n",
       "            Pituitary               Putamen         Sigmoid colon \n",
       "                    5                     4                    13 \n",
       "      Skeletal muscle        Skin (exposed)    Skin (not exposed) \n",
       "                    2                     7                     0 \n",
       "      Small intestine           Spinal cord                Spleen \n",
       "                    1                     0                     4 \n",
       "              Stomach               Thyroid         Tibial artery \n",
       "                    5                     3                     0 \n",
       "         Tibial nerve      Transverse colon           Whole blood \n",
       "                    1                     3                     0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneJunction'</li><li>'ASE'</li><li>'ASE_IDX'</li><li>'Tissue'</li><li>'counts'</li><li>'Display'</li><li>'GeneSymbol'</li><li>'GeneID'</li><li>'chr'</li><li>'logFC'</li><li>'AveExpr'</li><li>'t'</li><li>'PValue'</li><li>'AdjPVal'</li><li>'B'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneJunction'\n",
       "\\item 'ASE'\n",
       "\\item 'ASE\\_IDX'\n",
       "\\item 'Tissue'\n",
       "\\item 'counts'\n",
       "\\item 'Display'\n",
       "\\item 'GeneSymbol'\n",
       "\\item 'GeneID'\n",
       "\\item 'chr'\n",
       "\\item 'logFC'\n",
       "\\item 'AveExpr'\n",
       "\\item 't'\n",
       "\\item 'PValue'\n",
       "\\item 'AdjPVal'\n",
       "\\item 'B'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneJunction'\n",
       "2. 'ASE'\n",
       "3. 'ASE_IDX'\n",
       "4. 'Tissue'\n",
       "5. 'counts'\n",
       "6. 'Display'\n",
       "7. 'GeneSymbol'\n",
       "8. 'GeneID'\n",
       "9. 'chr'\n",
       "10. 'logFC'\n",
       "11. 'AveExpr'\n",
       "12. 't'\n",
       "13. 'PValue'\n",
       "14. 'AdjPVal'\n",
       "15. 'B'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GeneJunction\" \"ASE\"          \"ASE_IDX\"      \"Tissue\"       \"counts\"      \n",
       " [6] \"Display\"      \"GeneSymbol\"   \"GeneID\"       \"chr\"          \"logFC\"       \n",
       "[11] \"AveExpr\"      \"t\"            \"PValue\"       \"AdjPVal\"      \"B\"           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneJunction</th><th scope=col>ASE</th><th scope=col>ASE_IDX</th><th scope=col>Tissue</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>GeneSymbol</th><th scope=col>GeneID</th><th scope=col>chr</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CYB5R1-2513</td><td>A3SS</td><td>2513</td><td>brain_hippocampus    </td><td>1</td><td>Hippocampus </td><td>CYB5R1</td><td>ENSG00000159348.13</td><td>chr1 </td><td> 4.248889</td><td> 5.779254</td><td> 5.126318</td><td>7.077770e-07</td><td>0.003560118</td><td>5.185446</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ANK3-4589  </td><td>A3SS</td><td>4589</td><td>brain_hypothalamus   </td><td>1</td><td>Hypothalamus</td><td>ANK3  </td><td>ENSG00000151150.22</td><td>chr10</td><td>-6.419966</td><td> 4.170207</td><td>-4.656180</td><td>6.145240e-06</td><td>0.032244072</td><td>3.210014</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>RPS24-7852 </td><td>A3SS</td><td>7852</td><td>breast_mammary_tissue</td><td>7</td><td>Breast      </td><td>RPS24 </td><td>ENSG00000138326.19</td><td>chr10</td><td>-3.521200</td><td>10.731240</td><td>-4.756402</td><td>2.899873e-06</td><td>0.016053697</td><td>4.243538</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ODF2-2715  </td><td>A3SS</td><td>2715</td><td>breast_mammary_tissue</td><td>7</td><td>Breast      </td><td>ODF2  </td><td>ENSG00000136811.16</td><td>chr9 </td><td> 2.070290</td><td> 4.540094</td><td> 4.396519</td><td>1.450855e-05</td><td>0.026773103</td><td>2.720328</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>IP6K2-8102 </td><td>A3SS</td><td>8102</td><td>breast_mammary_tissue</td><td>7</td><td>Breast      </td><td>IP6K2 </td><td>ENSG00000068745.15</td><td>chr3 </td><td> 2.726048</td><td> 1.621525</td><td> 4.452328</td><td>1.135499e-05</td><td>0.026773103</td><td>2.634277</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>PITRM1-3445</td><td>A3SS</td><td>3445</td><td>breast_mammary_tissue</td><td>7</td><td>Breast      </td><td>PITRM1</td><td>ENSG00000107959.16</td><td>chr10</td><td>-1.178564</td><td> 7.352415</td><td>-4.255573</td><td>2.664085e-05</td><td>0.036870943</td><td>2.303679</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & GeneJunction & ASE & ASE\\_IDX & Tissue & counts & Display & GeneSymbol & GeneID & chr & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <chr> & <chr> & <int> & <fct> & <int> & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & CYB5R1-2513 & A3SS & 2513 & brain\\_hippocampus     & 1 & Hippocampus  & CYB5R1 & ENSG00000159348.13 & chr1  &  4.248889 &  5.779254 &  5.126318 & 7.077770e-07 & 0.003560118 & 5.185446\\\\\n",
       "\t2 & ANK3-4589   & A3SS & 4589 & brain\\_hypothalamus    & 1 & Hypothalamus & ANK3   & ENSG00000151150.22 & chr10 & -6.419966 &  4.170207 & -4.656180 & 6.145240e-06 & 0.032244072 & 3.210014\\\\\n",
       "\t3 & RPS24-7852  & A3SS & 7852 & breast\\_mammary\\_tissue & 7 & Breast       & RPS24  & ENSG00000138326.19 & chr10 & -3.521200 & 10.731240 & -4.756402 & 2.899873e-06 & 0.016053697 & 4.243538\\\\\n",
       "\t4 & ODF2-2715   & A3SS & 2715 & breast\\_mammary\\_tissue & 7 & Breast       & ODF2   & ENSG00000136811.16 & chr9  &  2.070290 &  4.540094 &  4.396519 & 1.450855e-05 & 0.026773103 & 2.720328\\\\\n",
       "\t5 & IP6K2-8102  & A3SS & 8102 & breast\\_mammary\\_tissue & 7 & Breast       & IP6K2  & ENSG00000068745.15 & chr3  &  2.726048 &  1.621525 &  4.452328 & 1.135499e-05 & 0.026773103 & 2.634277\\\\\n",
       "\t6 & PITRM1-3445 & A3SS & 3445 & breast\\_mammary\\_tissue & 7 & Breast       & PITRM1 & ENSG00000107959.16 & chr10 & -1.178564 &  7.352415 & -4.255573 & 2.664085e-05 & 0.036870943 & 2.303679\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | GeneJunction &lt;chr&gt; | ASE &lt;chr&gt; | ASE_IDX &lt;int&gt; | Tissue &lt;fct&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | GeneSymbol &lt;chr&gt; | GeneID &lt;chr&gt; | chr &lt;chr&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | CYB5R1-2513 | A3SS | 2513 | brain_hippocampus     | 1 | Hippocampus  | CYB5R1 | ENSG00000159348.13 | chr1  |  4.248889 |  5.779254 |  5.126318 | 7.077770e-07 | 0.003560118 | 5.185446 |\n",
       "| 2 | ANK3-4589   | A3SS | 4589 | brain_hypothalamus    | 1 | Hypothalamus | ANK3   | ENSG00000151150.22 | chr10 | -6.419966 |  4.170207 | -4.656180 | 6.145240e-06 | 0.032244072 | 3.210014 |\n",
       "| 3 | RPS24-7852  | A3SS | 7852 | breast_mammary_tissue | 7 | Breast       | RPS24  | ENSG00000138326.19 | chr10 | -3.521200 | 10.731240 | -4.756402 | 2.899873e-06 | 0.016053697 | 4.243538 |\n",
       "| 4 | ODF2-2715   | A3SS | 2715 | breast_mammary_tissue | 7 | Breast       | ODF2   | ENSG00000136811.16 | chr9  |  2.070290 |  4.540094 |  4.396519 | 1.450855e-05 | 0.026773103 | 2.720328 |\n",
       "| 5 | IP6K2-8102  | A3SS | 8102 | breast_mammary_tissue | 7 | Breast       | IP6K2  | ENSG00000068745.15 | chr3  |  2.726048 |  1.621525 |  4.452328 | 1.135499e-05 | 0.026773103 | 2.634277 |\n",
       "| 6 | PITRM1-3445 | A3SS | 3445 | breast_mammary_tissue | 7 | Breast       | PITRM1 | ENSG00000107959.16 | chr10 | -1.178564 |  7.352415 | -4.255573 | 2.664085e-05 | 0.036870943 | 2.303679 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneJunction ASE  ASE_IDX Tissue                counts Display     \n",
       "1 CYB5R1-2513  A3SS 2513    brain_hippocampus     1      Hippocampus \n",
       "2 ANK3-4589    A3SS 4589    brain_hypothalamus    1      Hypothalamus\n",
       "3 RPS24-7852   A3SS 7852    breast_mammary_tissue 7      Breast      \n",
       "4 ODF2-2715    A3SS 2715    breast_mammary_tissue 7      Breast      \n",
       "5 IP6K2-8102   A3SS 8102    breast_mammary_tissue 7      Breast      \n",
       "6 PITRM1-3445  A3SS 3445    breast_mammary_tissue 7      Breast      \n",
       "  GeneSymbol GeneID             chr   logFC     AveExpr   t        \n",
       "1 CYB5R1     ENSG00000159348.13 chr1   4.248889  5.779254  5.126318\n",
       "2 ANK3       ENSG00000151150.22 chr10 -6.419966  4.170207 -4.656180\n",
       "3 RPS24      ENSG00000138326.19 chr10 -3.521200 10.731240 -4.756402\n",
       "4 ODF2       ENSG00000136811.16 chr9   2.070290  4.540094  4.396519\n",
       "5 IP6K2      ENSG00000068745.15 chr3   2.726048  1.621525  4.452328\n",
       "6 PITRM1     ENSG00000107959.16 chr10 -1.178564  7.352415 -4.255573\n",
       "  PValue       AdjPVal     B       \n",
       "1 7.077770e-07 0.003560118 5.185446\n",
       "2 6.145240e-06 0.032244072 3.210014\n",
       "3 2.899873e-06 0.016053697 4.243538\n",
       "4 1.450855e-05 0.026773103 2.720328\n",
       "5 1.135499e-05 0.026773103 2.634277\n",
       "6 2.664085e-05 0.036870943 2.303679"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "glimpse(gene_as)\n",
    "gene_as$Tissue <- factor(gene_as$Tissue)\n",
    "length(levels(gene_as$Tissue))\n",
    "table(is.na(gene_as$Display))\n",
    "table(gene_as$Display)\n",
    "colnames(gene_as)\n",
    "head(gene_as)\n",
    "tissue_reduction$display_name <- factor(tissue_reduction$display_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "1850a5f9",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:53.087271Z",
     "iopub.status.busy": "2021-05-21T02:46:53.085954Z",
     "iopub.status.idle": "2021-05-21T02:46:53.121634Z",
     "shell.execute_reply": "2021-05-21T02:46:53.120429Z"
    },
    "papermill": {
     "duration": 0.082595,
     "end_time": "2021-05-21T02:46:53.121765",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.039170",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 312 total significant alternative splicing events (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 18 total significant alternative splicing events on the X chromosome (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "i.e., 5.76923076923077% of all significant AS events were on the X chromosome\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "gene_as now has 39 tissues and 5 ASE categories\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ASE:\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>A3SS</dt><dd>20</dd><dt>A5SS</dt><dd>20</dd><dt>MXE</dt><dd>18</dd><dt>RI</dt><dd>32</dd><dt>SE</dt><dd>222</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[A3SS] 20\n",
       "\\item[A5SS] 20\n",
       "\\item[MXE] 18\n",
       "\\item[RI] 32\n",
       "\\item[SE] 222\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "A3SS\n",
       ":   20A5SS\n",
       ":   20MXE\n",
       ":   18RI\n",
       ":   32SE\n",
       ":   222\n",
       "\n"
      ],
      "text/plain": [
       "A3SS A5SS  MXE   RI   SE \n",
       "  20   20   18   32  222 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x_as_events <- gene_as[gene_as$chr==\"chrX\",]\n",
    "message(\"There were \",nrow(gene_as),\" total significant alternative splicing events (gene_as)\")\n",
    "message(\"There were \",nrow(x_as_events),\" total significant alternative splicing events on the X chromosome (gene_as)\")\n",
    "message(\"i.e., \", (100*nrow(x_as_events)/nrow(gene_as)), \"% of all significant AS events were on the X chromosome\")\n",
    "\n",
    "numberOfUniqueTissues <- length(summary(as.factor(gene_as$Display),maxsum=500))\n",
    "numberOfASEmechanisms <- length(summary(as.factor(gene_as$ASE),maxsum=500))\n",
    "\n",
    "message(\"gene_as now has \",numberOfUniqueTissues, \" tissues and \", numberOfASEmechanisms, \" ASE categories\")\n",
    "message(\"ASE:\")\n",
    "summary(as.factor(gene_as$ASE),maxsum=500)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97a0f3b3",
   "metadata": {
    "papermill": {
     "duration": 0.044195,
     "end_time": "2021-05-21T02:46:53.210420",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.166225",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.2 gene_dge.tsv\n",
    "\n",
    "This file contains (description)\n",
    "Here is a typical line\n",
    "<pre>\n",
    "Tissue  ENSG_ver        ENSG_no_ver     GeneSymbol      counts  Display logFC   AveExpr t       PValue  AdjPVal B\n",
    "1       adipose_subcutaneous    ENSG00000176728.7       ENSG00000176728 TTTY14  765     Adipose (sc)    -7.98216577151896     -0.928812923511535       -139.823010017733       0       0       1107.42326360464\n",
    "2       adipose_subcutaneous    ENSG00000231535.5       ENSG00000231535 LINC00278       765     Adipose (sc)    -6.09542040758638      -2.77656379347601       -126.913818678612       0       0       1050.36559888639\n",
    "3       adipose_subcutaneous    ENSG00000129824.15      ENSG00000129824 RPS4Y1  765     Adipose (sc)    -9.6641901864726      4.63528767282141 -125.827094717734       0       0       1041.87660796556\n",
    "4       adipose_subcutaneous    ENSG00000067646.11      ENSG00000067646 ZFY     765     Adipose (sc)    -9.50458982938477     0.672755457406984        -125.037143030325       0       0       1033.61131617113\n",
    "5       adipose_subcutaneous    ENSG00000229807.10      ENSG00000229807 XIST    765     Adipose (sc)    9.89280986473167      1.23756039627052 121.69689757218 0       0       1030.17757492281\n",
    "6       adipose_subcutaneous    ENSG00000229236.1       ENSG00000229236 TTTY10  765     Adipose (sc)    -6.20901295440725     -2.74524363170072        -122.540297482165       0       0       1029.41424065532\n",
    "7       adipose_subcutaneous    ENSG00000233864.7       ENSG00000233864 TTTY15  765     Adipose (sc)    -8.19361688496523     -0.741097276206495       -122.47199454746        0       0       1027.58175958647\n",
    "8       adipose_subcutaneous    ENSG00000260197.1       ENSG00000260197 AC010889.1      765     Adipose (sc)    -8.52835806068555      -0.686009457030557      -119.790486729538       0       0       1015.85291786821\n",
    "9       adipose_subcutaneous    ENSG00000183878.15      ENSG00000183878 UTY     765     Adipose (sc)    -9.52139275438866     1.60375445153084 -110.599936868261       0       0       953.069478992754\n",
    "</pre>\n",
    "There are 7417 significant events in the file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0269f3f0",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:53.304651Z",
     "iopub.status.busy": "2021-05-21T02:46:53.303534Z",
     "iopub.status.idle": "2021-05-21T02:46:53.330676Z",
     "shell.execute_reply": "2021-05-21T02:46:53.329471Z"
    },
    "papermill": {
     "duration": 0.075774,
     "end_time": "2021-05-21T02:46:53.330804",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.255030",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 4219 total significant differential gene expression events (gene_dge)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 307 total significant differential gene expression events on the X chromosome (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "i.e., 7.27660583076558% of all significant DGE events were on the X chromosome\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "gene_dge now has 39 tissues\n",
      "\n"
     ]
    }
   ],
   "source": [
    "x_dge_events <- gene_dge[gene_dge$chr==\"chrX\",]\n",
    "message(\"There were \",nrow(gene_dge),\" total significant differential gene expression events (gene_dge)\")\n",
    "message(\"There were \",nrow(x_dge_events),\" total significant differential gene expression events on the X chromosome (gene_as)\")\n",
    "message(\"i.e., \", (100*nrow(x_dge_events)/nrow(gene_dge)), \"% of all significant DGE events were on the X chromosome\")\n",
    "\n",
    "numberOfUniqueTissues <- length(summary(as.factor(gene_dge$Display),maxsum=500))\n",
    "\n",
    "message(\"gene_dge now has \",numberOfUniqueTissues, \" tissues\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f98c3d69",
   "metadata": {
    "papermill": {
     "duration": 0.045625,
     "end_time": "2021-05-21T02:46:53.421400",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.375775",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.3 Count events by chromosome\n",
    "\n",
    "Count the number of significant alternative splicing events per chromosome and save to the file **Total_AS_by_chr.tsv**."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1419491e",
   "metadata": {
    "papermill": {
     "duration": 0.044958,
     "end_time": "2021-05-21T02:46:53.510736",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.465778",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.3.1 by alternative splicing events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "65c50ed7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:53.605026Z",
     "iopub.status.busy": "2021-05-21T02:46:53.603815Z",
     "iopub.status.idle": "2021-05-21T02:46:53.658527Z",
     "shell.execute_reply": "2021-05-21T02:46:53.657492Z"
    },
    "papermill": {
     "duration": 0.103073,
     "end_time": "2021-05-21T02:46:53.658695",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.555622",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "23"
      ],
      "text/latex": [
       "23"
      ],
      "text/markdown": [
       "23"
      ],
      "text/plain": [
       "[1] 23"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 23 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>26</td></tr>\n",
       "\t<tr><td>chr17</td><td>26</td></tr>\n",
       "\t<tr><td>chr12</td><td>22</td></tr>\n",
       "\t<tr><td>chr2 </td><td>22</td></tr>\n",
       "\t<tr><td>chr16</td><td>21</td></tr>\n",
       "\t<tr><td>chr10</td><td>20</td></tr>\n",
       "\t<tr><td>chrX </td><td>18</td></tr>\n",
       "\t<tr><td>chr19</td><td>17</td></tr>\n",
       "\t<tr><td>chr3 </td><td>16</td></tr>\n",
       "\t<tr><td>chr7 </td><td>16</td></tr>\n",
       "\t<tr><td>chr11</td><td>14</td></tr>\n",
       "\t<tr><td>chr4 </td><td>13</td></tr>\n",
       "\t<tr><td>chr9 </td><td>13</td></tr>\n",
       "\t<tr><td>chr14</td><td>10</td></tr>\n",
       "\t<tr><td>chr15</td><td>10</td></tr>\n",
       "\t<tr><td>chr6 </td><td> 8</td></tr>\n",
       "\t<tr><td>chr20</td><td> 7</td></tr>\n",
       "\t<tr><td>chr22</td><td> 7</td></tr>\n",
       "\t<tr><td>chr18</td><td> 6</td></tr>\n",
       "\t<tr><td>chr8 </td><td> 6</td></tr>\n",
       "\t<tr><td>chr13</td><td> 5</td></tr>\n",
       "\t<tr><td>chr5 </td><td> 5</td></tr>\n",
       "\t<tr><td>chr21</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 23 × 2\n",
       "\\begin{tabular}{ll}\n",
       " chr & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 26\\\\\n",
       "\t chr17 & 26\\\\\n",
       "\t chr12 & 22\\\\\n",
       "\t chr2  & 22\\\\\n",
       "\t chr16 & 21\\\\\n",
       "\t chr10 & 20\\\\\n",
       "\t chrX  & 18\\\\\n",
       "\t chr19 & 17\\\\\n",
       "\t chr3  & 16\\\\\n",
       "\t chr7  & 16\\\\\n",
       "\t chr11 & 14\\\\\n",
       "\t chr4  & 13\\\\\n",
       "\t chr9  & 13\\\\\n",
       "\t chr14 & 10\\\\\n",
       "\t chr15 & 10\\\\\n",
       "\t chr6  &  8\\\\\n",
       "\t chr20 &  7\\\\\n",
       "\t chr22 &  7\\\\\n",
       "\t chr18 &  6\\\\\n",
       "\t chr8  &  6\\\\\n",
       "\t chr13 &  5\\\\\n",
       "\t chr5  &  5\\\\\n",
       "\t chr21 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 23 × 2\n",
       "\n",
       "| chr &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 26 |\n",
       "| chr17 | 26 |\n",
       "| chr12 | 22 |\n",
       "| chr2  | 22 |\n",
       "| chr16 | 21 |\n",
       "| chr10 | 20 |\n",
       "| chrX  | 18 |\n",
       "| chr19 | 17 |\n",
       "| chr3  | 16 |\n",
       "| chr7  | 16 |\n",
       "| chr11 | 14 |\n",
       "| chr4  | 13 |\n",
       "| chr9  | 13 |\n",
       "| chr14 | 10 |\n",
       "| chr15 | 10 |\n",
       "| chr6  |  8 |\n",
       "| chr20 |  7 |\n",
       "| chr22 |  7 |\n",
       "| chr18 |  6 |\n",
       "| chr8  |  6 |\n",
       "| chr13 |  5 |\n",
       "| chr5  |  5 |\n",
       "| chr21 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr   n \n",
       "1  chr1  26\n",
       "2  chr17 26\n",
       "3  chr12 22\n",
       "4  chr2  22\n",
       "5  chr16 21\n",
       "6  chr10 20\n",
       "7  chrX  18\n",
       "8  chr19 17\n",
       "9  chr3  16\n",
       "10 chr7  16\n",
       "11 chr11 14\n",
       "12 chr4  13\n",
       "13 chr9  13\n",
       "14 chr14 10\n",
       "15 chr15 10\n",
       "16 chr6   8\n",
       "17 chr20  7\n",
       "18 chr22  7\n",
       "19 chr18  6\n",
       "20 chr8   6\n",
       "21 chr13  5\n",
       "22 chr5   5\n",
       "23 chr21  4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 23\n",
      "Columns: 2\n",
      "$ chr \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr1, chr17, chr12, chr2, chr16, chr10, chrX, chr19, chr3, chr7, c…\n",
      "$ n   \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 26, 26, 22, 22, 21, 20, 18, 17, 16, 16, 14, 13, 13, 10, 10, 8, 7, …\n"
     ]
    }
   ],
   "source": [
    "total_as_by_chr <- gene_as          %>% \n",
    "                   group_by(chr)    %>% \n",
    "                   count(chr)       %>% \n",
    "                   arrange(desc(n)) %>% \n",
    "                   as.data.frame()\n",
    "total_as_by_chr$chr <- factor(total_as_by_chr$chr, levels = total_as_by_chr$chr)\n",
    "length(total_as_by_chr$chr)\n",
    "total_as_by_chr\n",
    "glimpse(total_as_by_chr)\n",
    "write.table(total_as_by_chr, file= \"../data/Total_AS_by_chr.tsv\", sep=\"\\t\", quote = FALSE, row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bdb8e359",
   "metadata": {
    "papermill": {
     "duration": 0.045801,
     "end_time": "2021-05-21T02:46:53.750455",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.704654",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.3.2 by gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "553e72a6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:53.847089Z",
     "iopub.status.busy": "2021-05-21T02:46:53.845999Z",
     "iopub.status.idle": "2021-05-21T02:46:53.896859Z",
     "shell.execute_reply": "2021-05-21T02:46:53.895840Z"
    },
    "papermill": {
     "duration": 0.101272,
     "end_time": "2021-05-21T02:46:53.896991",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.795719",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "26"
      ],
      "text/latex": [
       "26"
      ],
      "text/markdown": [
       "26"
      ],
      "text/plain": [
       "[1] 26"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>437</td></tr>\n",
       "\t<tr><td>chr2 </td><td>321</td></tr>\n",
       "\t<tr><td>chrX </td><td>307</td></tr>\n",
       "\t<tr><td>chr11</td><td>237</td></tr>\n",
       "\t<tr><td>chr12</td><td>217</td></tr>\n",
       "\t<tr><td>chr19</td><td>206</td></tr>\n",
       "\t<tr><td>chr6 </td><td>202</td></tr>\n",
       "\t<tr><td>chr3 </td><td>195</td></tr>\n",
       "\t<tr><td>chr17</td><td>191</td></tr>\n",
       "\t<tr><td>chr7 </td><td>181</td></tr>\n",
       "\t<tr><td>chr4 </td><td>180</td></tr>\n",
       "\t<tr><td>chr14</td><td>175</td></tr>\n",
       "\t<tr><td>chr5 </td><td>173</td></tr>\n",
       "\t<tr><td>chr20</td><td>152</td></tr>\n",
       "\t<tr><td>chr16</td><td>150</td></tr>\n",
       "\t<tr><td>chr9 </td><td>149</td></tr>\n",
       "\t<tr><td>chr10</td><td>144</td></tr>\n",
       "\t<tr><td>chr15</td><td>135</td></tr>\n",
       "\t<tr><td>chr22</td><td>133</td></tr>\n",
       "\t<tr><td>chr8 </td><td>125</td></tr>\n",
       "\t<tr><td>chr13</td><td> 64</td></tr>\n",
       "\t<tr><td>chr18</td><td> 61</td></tr>\n",
       "\t<tr><td>chr21</td><td> 54</td></tr>\n",
       "\t<tr><td>NA   </td><td> 25</td></tr>\n",
       "\t<tr><td>chrY </td><td>  3</td></tr>\n",
       "\t<tr><td>chrM </td><td>  2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 2\n",
       "\\begin{tabular}{ll}\n",
       " chr & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 437\\\\\n",
       "\t chr2  & 321\\\\\n",
       "\t chrX  & 307\\\\\n",
       "\t chr11 & 237\\\\\n",
       "\t chr12 & 217\\\\\n",
       "\t chr19 & 206\\\\\n",
       "\t chr6  & 202\\\\\n",
       "\t chr3  & 195\\\\\n",
       "\t chr17 & 191\\\\\n",
       "\t chr7  & 181\\\\\n",
       "\t chr4  & 180\\\\\n",
       "\t chr14 & 175\\\\\n",
       "\t chr5  & 173\\\\\n",
       "\t chr20 & 152\\\\\n",
       "\t chr16 & 150\\\\\n",
       "\t chr9  & 149\\\\\n",
       "\t chr10 & 144\\\\\n",
       "\t chr15 & 135\\\\\n",
       "\t chr22 & 133\\\\\n",
       "\t chr8  & 125\\\\\n",
       "\t chr13 &  64\\\\\n",
       "\t chr18 &  61\\\\\n",
       "\t chr21 &  54\\\\\n",
       "\t NA    &  25\\\\\n",
       "\t chrY  &   3\\\\\n",
       "\t chrM  &   2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 2\n",
       "\n",
       "| chr &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 437 |\n",
       "| chr2  | 321 |\n",
       "| chrX  | 307 |\n",
       "| chr11 | 237 |\n",
       "| chr12 | 217 |\n",
       "| chr19 | 206 |\n",
       "| chr6  | 202 |\n",
       "| chr3  | 195 |\n",
       "| chr17 | 191 |\n",
       "| chr7  | 181 |\n",
       "| chr4  | 180 |\n",
       "| chr14 | 175 |\n",
       "| chr5  | 173 |\n",
       "| chr20 | 152 |\n",
       "| chr16 | 150 |\n",
       "| chr9  | 149 |\n",
       "| chr10 | 144 |\n",
       "| chr15 | 135 |\n",
       "| chr22 | 133 |\n",
       "| chr8  | 125 |\n",
       "| chr13 |  64 |\n",
       "| chr18 |  61 |\n",
       "| chr21 |  54 |\n",
       "| NA    |  25 |\n",
       "| chrY  |   3 |\n",
       "| chrM  |   2 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr   n  \n",
       "1  chr1  437\n",
       "2  chr2  321\n",
       "3  chrX  307\n",
       "4  chr11 237\n",
       "5  chr12 217\n",
       "6  chr19 206\n",
       "7  chr6  202\n",
       "8  chr3  195\n",
       "9  chr17 191\n",
       "10 chr7  181\n",
       "11 chr4  180\n",
       "12 chr14 175\n",
       "13 chr5  173\n",
       "14 chr20 152\n",
       "15 chr16 150\n",
       "16 chr9  149\n",
       "17 chr10 144\n",
       "18 chr15 135\n",
       "19 chr22 133\n",
       "20 chr8  125\n",
       "21 chr13  64\n",
       "22 chr18  61\n",
       "23 chr21  54\n",
       "24 NA     25\n",
       "25 chrY    3\n",
       "26 chrM    2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 26\n",
      "Columns: 2\n",
      "$ chr \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr1, chr2, chrX, chr11, chr12, chr19, chr6, chr3, chr17, chr7, ch…\n",
      "$ n   \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 437, 321, 307, 237, 217, 206, 202, 195, 191, 181, 180, 175, 173, 1…\n"
     ]
    }
   ],
   "source": [
    "total_dge_by_chr <- gene_dge          %>% \n",
    "                   group_by(chr)    %>% \n",
    "                   count(chr)       %>% \n",
    "                   arrange(desc(n)) %>% \n",
    "                   as.data.frame()\n",
    "total_dge_by_chr$chr <- factor(total_dge_by_chr$chr, levels = total_dge_by_chr$chr)\n",
    "length(total_dge_by_chr$chr)\n",
    "total_dge_by_chr\n",
    "glimpse(total_dge_by_chr)\n",
    "write.table(total_dge_by_chr, file= \"../data/Total_DGE_by_chr.tsv\", sep=\"\\t\", quote = FALSE, row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c6c2b72e",
   "metadata": {
    "papermill": {
     "duration": 0.047026,
     "end_time": "2021-05-21T02:46:53.991087",
     "exception": false,
     "start_time": "2021-05-21T02:46:53.944061",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.4 Count events by genes "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfa68b06",
   "metadata": {
    "papermill": {
     "duration": 0.046674,
     "end_time": "2021-05-21T02:46:54.084851",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.038177",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.4.1 by alternative splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8b97a8d7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:54.185011Z",
     "iopub.status.busy": "2021-05-21T02:46:54.183838Z",
     "iopub.status.idle": "2021-05-21T02:46:54.232829Z",
     "shell.execute_reply": "2021-05-21T02:46:54.231706Z"
    },
    "papermill": {
     "duration": 0.100935,
     "end_time": "2021-05-21T02:46:54.232961",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.132026",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "254"
      ],
      "text/latex": [
       "254"
      ],
      "text/markdown": [
       "254"
      ],
      "text/plain": [
       "[1] 254"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneSymbol</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>KDM5C  </td><td>8</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SORBS2 </td><td>5</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ABCD4  </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BNIP2  </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>FGFR2  </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MLH1   </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>RPLP0  </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>SNRNP70</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>ABI1   </td><td>2</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>ACADVL </td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & GeneSymbol & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & KDM5C   & 8\\\\\n",
       "\t2 & SORBS2  & 5\\\\\n",
       "\t3 & ABCD4   & 3\\\\\n",
       "\t4 & BNIP2   & 3\\\\\n",
       "\t5 & FGFR2   & 3\\\\\n",
       "\t6 & MLH1    & 3\\\\\n",
       "\t7 & RPLP0   & 3\\\\\n",
       "\t8 & SNRNP70 & 3\\\\\n",
       "\t9 & ABI1    & 2\\\\\n",
       "\t10 & ACADVL  & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | GeneSymbol &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | KDM5C   | 8 |\n",
       "| 2 | SORBS2  | 5 |\n",
       "| 3 | ABCD4   | 3 |\n",
       "| 4 | BNIP2   | 3 |\n",
       "| 5 | FGFR2   | 3 |\n",
       "| 6 | MLH1    | 3 |\n",
       "| 7 | RPLP0   | 3 |\n",
       "| 8 | SNRNP70 | 3 |\n",
       "| 9 | ABI1    | 2 |\n",
       "| 10 | ACADVL  | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "   GeneSymbol n\n",
       "1  KDM5C      8\n",
       "2  SORBS2     5\n",
       "3  ABCD4      3\n",
       "4  BNIP2      3\n",
       "5  FGFR2      3\n",
       "6  MLH1       3\n",
       "7  RPLP0      3\n",
       "8  SNRNP70    3\n",
       "9  ABI1       2\n",
       "10 ACADVL     2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_geneSymbol <- gene_as %>% \n",
    "                          group_by(GeneSymbol) %>% \n",
    "                          count(GeneSymbol)    %>% \n",
    "                          arrange(desc(n))     %>% \n",
    "                          as.data.frame()\n",
    "total_as_by_geneSymbol$GeneSymbol <- factor(total_as_by_geneSymbol$GeneSymbol, \n",
    "                                            levels = total_as_by_geneSymbol$GeneSymbol)\n",
    "length(total_as_by_geneSymbol$GeneSymbol)\n",
    "head(total_as_by_geneSymbol,10)\n",
    "write.table(total_as_by_geneSymbol, file = \"../data/Total_AS_by_geneSymbol.tsv\", sep = \"\\t\", quote=FALSE, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ce5ba2d",
   "metadata": {
    "papermill": {
     "duration": 0.048001,
     "end_time": "2021-05-21T02:46:54.328825",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.280824",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.5 Count most frequent splicing by tissue\n",
    "\n",
    "### 3.5.1 by alternative splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "fbaa6b4e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:54.430192Z",
     "iopub.status.busy": "2021-05-21T02:46:54.428775Z",
     "iopub.status.idle": "2021-05-21T02:46:54.473235Z",
     "shell.execute_reply": "2021-05-21T02:46:54.471919Z"
    },
    "papermill": {
     "duration": 0.096378,
     "end_time": "2021-05-21T02:46:54.473370",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.376992",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Display</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Breast               </td><td>199</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Cortex               </td><td> 13</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Sigmoid colon        </td><td> 13</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Cerebellum           </td><td>  9</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Skin (exposed)       </td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Adipose (v)          </td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Cerebellar hemisphere</td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>Esophagus (mu)       </td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Pituitary            </td><td>  5</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Stomach              </td><td>  5</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Display & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Breast                & 199\\\\\n",
       "\t2 & Cortex                &  13\\\\\n",
       "\t3 & Sigmoid colon         &  13\\\\\n",
       "\t4 & Cerebellum            &   9\\\\\n",
       "\t5 & Skin (exposed)        &   7\\\\\n",
       "\t6 & Adipose (v)           &   6\\\\\n",
       "\t7 & Cerebellar hemisphere &   6\\\\\n",
       "\t8 & Esophagus (mu)        &   6\\\\\n",
       "\t9 & Pituitary             &   5\\\\\n",
       "\t10 & Stomach               &   5\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | Display &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Breast                | 199 |\n",
       "| 2 | Cortex                |  13 |\n",
       "| 3 | Sigmoid colon         |  13 |\n",
       "| 4 | Cerebellum            |   9 |\n",
       "| 5 | Skin (exposed)        |   7 |\n",
       "| 6 | Adipose (v)           |   6 |\n",
       "| 7 | Cerebellar hemisphere |   6 |\n",
       "| 8 | Esophagus (mu)        |   6 |\n",
       "| 9 | Pituitary             |   5 |\n",
       "| 10 | Stomach               |   5 |\n",
       "\n"
      ],
      "text/plain": [
       "   Display               n  \n",
       "1  Breast                199\n",
       "2  Cortex                 13\n",
       "3  Sigmoid colon          13\n",
       "4  Cerebellum              9\n",
       "5  Skin (exposed)          7\n",
       "6  Adipose (v)             6\n",
       "7  Cerebellar hemisphere   6\n",
       "8  Esophagus (mu)          6\n",
       "9  Pituitary               5\n",
       "10 Stomach                 5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "28"
      ],
      "text/latex": [
       "28"
      ],
      "text/markdown": [
       "28"
      ],
      "text/plain": [
       "[1] 28"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_tissue <- gene_as %>% \n",
    "                      group_by(Display) %>% \n",
    "                      count(Display)    %>% \n",
    "                      arrange(desc(n))  %>% \n",
    "                      as.data.frame()\n",
    "total_as_by_tissue$Display <- factor(total_as_by_tissue$Display, \n",
    "                                     levels = total_as_by_tissue$Display)\n",
    "head(total_as_by_tissue,10)\n",
    "length(total_as_by_tissue$Display)\n",
    "write.table(total_as_by_tissue, file = \"../data/Total_AS_by_tissue.tsv\", sep = \"\\t\", row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4dffc6d2",
   "metadata": {
    "papermill": {
     "duration": 0.048472,
     "end_time": "2021-05-21T02:46:54.570169",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.521697",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.5.2 by gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "637be747",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:54.672729Z",
     "iopub.status.busy": "2021-05-21T02:46:54.671540Z",
     "iopub.status.idle": "2021-05-21T02:46:54.732591Z",
     "shell.execute_reply": "2021-05-21T02:46:54.731379Z"
    },
    "papermill": {
     "duration": 0.11422,
     "end_time": "2021-05-21T02:46:54.732720",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.618500",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Display</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Breast            </td><td>2301</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Pituitary         </td><td> 265</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Adipose (sc)      </td><td> 247</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Thyroid           </td><td> 132</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Frontal cortex    </td><td> 117</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Skin (not exposed)</td><td> 115</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Fibroblasts       </td><td>  94</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>Skin (exposed)    </td><td>  93</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Skeletal muscle   </td><td>  78</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Putamen           </td><td>  65</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Display & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Breast             & 2301\\\\\n",
       "\t2 & Pituitary          &  265\\\\\n",
       "\t3 & Adipose (sc)       &  247\\\\\n",
       "\t4 & Thyroid            &  132\\\\\n",
       "\t5 & Frontal cortex     &  117\\\\\n",
       "\t6 & Skin (not exposed) &  115\\\\\n",
       "\t7 & Fibroblasts        &   94\\\\\n",
       "\t8 & Skin (exposed)     &   93\\\\\n",
       "\t9 & Skeletal muscle    &   78\\\\\n",
       "\t10 & Putamen            &   65\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | Display &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Breast             | 2301 |\n",
       "| 2 | Pituitary          |  265 |\n",
       "| 3 | Adipose (sc)       |  247 |\n",
       "| 4 | Thyroid            |  132 |\n",
       "| 5 | Frontal cortex     |  117 |\n",
       "| 6 | Skin (not exposed) |  115 |\n",
       "| 7 | Fibroblasts        |   94 |\n",
       "| 8 | Skin (exposed)     |   93 |\n",
       "| 9 | Skeletal muscle    |   78 |\n",
       "| 10 | Putamen            |   65 |\n",
       "\n"
      ],
      "text/plain": [
       "   Display            n   \n",
       "1  Breast             2301\n",
       "2  Pituitary           265\n",
       "3  Adipose (sc)        247\n",
       "4  Thyroid             132\n",
       "5  Frontal cortex      117\n",
       "6  Skin (not exposed)  115\n",
       "7  Fibroblasts          94\n",
       "8  Skin (exposed)       93\n",
       "9  Skeletal muscle      78\n",
       "10 Putamen              65"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#glimpse(gene_dge)\n",
    "gene_dge$GeneSymbol <- factor(gene_dge$GeneSymbol)\n",
    "total_dge_by_tissue <- gene_dge %>% \n",
    "                          select(c(GeneSymbol, Display, logFC)) %>%\n",
    "                          group_by(Display) %>%\n",
    "                          arrange(desc(logFC)) %>%\n",
    "                          tally() %>%\n",
    "                          arrange(desc(n)) %>%\n",
    "                          as.data.frame()\n",
    "head(total_dge_by_tissue,10)\n",
    "length(total_dge_by_tissue$Display)\n",
    "write.table(total_dge_by_tissue, file = \"../data/Total_DGE_by_tissue.tsv\", sep = \"\\t\", quote=FALSE, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21ae3094",
   "metadata": {
    "papermill": {
     "duration": 0.049465,
     "end_time": "2021-05-21T02:46:54.831692",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.782227",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "###  3.6 Significant Count by splicing type \n",
    "We define **significant** to be FC > 1.5 and pVal < 0.05\n",
    "\n",
    "Our starting values were the significant events, all meeting the criteria FC > 1.5 and pVal < 0.05\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "15bde520",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:54.940256Z",
     "iopub.status.busy": "2021-05-21T02:46:54.938946Z",
     "iopub.status.idle": "2021-05-21T02:46:54.977317Z",
     "shell.execute_reply": "2021-05-21T02:46:54.976058Z"
    },
    "papermill": {
     "duration": 0.094698,
     "end_time": "2021-05-21T02:46:54.977450",
     "exception": false,
     "start_time": "2021-05-21T02:46:54.882752",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 5 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ASE</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SE  </td><td>222</td></tr>\n",
       "\t<tr><td>RI  </td><td> 32</td></tr>\n",
       "\t<tr><td>A3SS</td><td> 20</td></tr>\n",
       "\t<tr><td>A5SS</td><td> 20</td></tr>\n",
       "\t<tr><td>MXE </td><td> 18</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ASE & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t SE   & 222\\\\\n",
       "\t RI   &  32\\\\\n",
       "\t A3SS &  20\\\\\n",
       "\t A5SS &  20\\\\\n",
       "\t MXE  &  18\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 2\n",
       "\n",
       "| ASE &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| SE   | 222 |\n",
       "| RI   |  32 |\n",
       "| A3SS |  20 |\n",
       "| A5SS |  20 |\n",
       "| MXE  |  18 |\n",
       "\n"
      ],
      "text/plain": [
       "  ASE  n  \n",
       "1 SE   222\n",
       "2 RI    32\n",
       "3 A3SS  20\n",
       "4 A5SS  20\n",
       "5 MXE   18"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_splicingtype <- gene_as %>% \n",
    "                            group_by(ASE)    %>% \n",
    "                            count(ASE)       %>% \n",
    "                            arrange(desc(n)) %>%\n",
    "                            as.data.frame()\n",
    "total_as_by_splicingtype$ASE <- factor(total_as_by_splicingtype$ASE, levels = total_as_by_splicingtype$ASE)\n",
    "total_as_by_splicingtype\n",
    "write.table(total_as_by_splicingtype, file= \"../data/Total_AS_by_splicingtype.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f557974",
   "metadata": {
    "papermill": {
     "duration": 0.050864,
     "end_time": "2021-05-21T02:46:55.079053",
     "exception": false,
     "start_time": "2021-05-21T02:46:55.028189",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "###  3.7 Significant Count by splicing type (significant == FC > 1.5 and pVal < 0.05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f0d0e954",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:55.187051Z",
     "iopub.status.busy": "2021-05-21T02:46:55.185691Z",
     "iopub.status.idle": "2021-05-21T02:46:55.260551Z",
     "shell.execute_reply": "2021-05-21T02:46:55.259343Z"
    },
    "lines_to_next_cell": 0,
    "papermill": {
     "duration": 0.130141,
     "end_time": "2021-05-21T02:46:55.260685",
     "exception": false,
     "start_time": "2021-05-21T02:46:55.130544",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "A3SS_keep\n",
       "FALSE  TRUE \n",
       "  292    20 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A5SS_keep\n",
       "FALSE  TRUE \n",
       "  292    20 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "MXE_keep\n",
       "FALSE  TRUE \n",
       "  294    18 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "SE_keep\n",
       "FALSE  TRUE \n",
       "   90   222 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RI_keep\n",
       "FALSE  TRUE \n",
       "  280    32 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>20</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 20\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 20\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 20 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>20</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 20\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 20\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 20 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>18</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 18\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 18\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 18 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>222</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 222\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 222\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 222  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>32</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 32\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 32\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 32 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "A3SS_keep <- as.character(gene_as$ASE) %in% \"A3SS\"\n",
    "table(A3SS_keep)\n",
    "A3SS.gene_as <- data.frame(gene_as[A3SS_keep == TRUE,])\n",
    "\n",
    "A5SS_keep <- as.character(gene_as$ASE) %in% \"A5SS\"\n",
    "table(A5SS_keep)\n",
    "A5SS.gene_as <- data.frame(gene_as[A5SS_keep == TRUE,])\n",
    "\n",
    "MXE_keep  <- as.character(gene_as$ASE) %in% \"MXE\"\n",
    "table(MXE_keep)\n",
    "MXE.gene_as <- data.frame(gene_as[MXE_keep == TRUE,])\n",
    "\n",
    "SE_keep   <- as.character(gene_as$ASE) %in% \"SE\"\n",
    "table(SE_keep)\n",
    "SE.gene_as <- data.frame(gene_as[SE_keep == TRUE,])\n",
    "\n",
    "RI_keep   <- as.character(gene_as$ASE) %in% \"RI\"\n",
    "table(RI_keep)\n",
    "RI.gene_as <- data.frame(gene_as[RI_keep == TRUE,])\n",
    "\n",
    "dim(A3SS.gene_as)\n",
    "dim(A5SS.gene_as)\n",
    "dim(MXE.gene_as)\n",
    "dim(SE.gene_as)\n",
    "dim(RI.gene_as)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ae0fc2d",
   "metadata": {
    "papermill": {
     "duration": 0.055139,
     "end_time": "2021-05-21T02:46:55.370251",
     "exception": false,
     "start_time": "2021-05-21T02:46:55.315112",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.8 Siginficant spliced by Gene for each splicing factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4679e54e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:55.484994Z",
     "iopub.status.busy": "2021-05-21T02:46:55.483765Z",
     "iopub.status.idle": "2021-05-21T02:46:55.589486Z",
     "shell.execute_reply": "2021-05-21T02:46:55.588274Z"
    },
    "papermill": {
     "duration": 0.164548,
     "end_time": "2021-05-21T02:46:55.589615",
     "exception": false,
     "start_time": "2021-05-21T02:46:55.425067",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for A3SS\n",
      "20 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for A5SS\n",
      "19 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for MXE\n",
      "14 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for RI\n",
      "30 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for SE\n",
      "181 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "A3SS.res <- A3SS.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "A3SS.res$GeneSymbol <- factor(A3SS.res$GeneSymbol, levels = A3SS.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for A3SS\\n\",\n",
    "        paste(length(A3SS.res$GeneSymbol)), collapse=\" \")\n",
    "#head(A3SS.res)\n",
    "\n",
    "A5SS.res <- A5SS.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "A5SS.res$GeneSymbol <- factor(A5SS.res$GeneSymbol, levels = A5SS.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for A5SS\\n\",\n",
    "        paste(length(A5SS.res$GeneSymbol)), collapse=\" \")\n",
    "#head(A5SS.res)\n",
    "\n",
    "MXE.res <- MXE.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "MXE.res$GeneSymbol <- factor(MXE.res$GeneSymbol, levels = MXE.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for MXE\\n\",\n",
    "        paste(length(MXE.res$GeneSymbol)), collapse=\" \")\n",
    "#head(MXE.res)\n",
    "\n",
    "RI.res <- RI.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "RI.res$GeneSymbol <- factor(RI.res$GeneSymbol, levels = RI.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for RI\\n\",\n",
    "        paste(length(RI.res$GeneSymbol)), collapse=\" \")\n",
    "#head(RI.res)\n",
    "\n",
    "SE.res <- SE.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "SE.res$GeneSymbol <- factor(SE.res$GeneSymbol, levels = SE.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for SE\\n\",\n",
    "        paste(length(SE.res$GeneSymbol)), collapse=\" \")\n",
    "#head(SE.res)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7b783f4",
   "metadata": {
    "papermill": {
     "duration": 0.055152,
     "end_time": "2021-05-21T02:46:55.700610",
     "exception": false,
     "start_time": "2021-05-21T02:46:55.645458",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.9 Count most frequent spliced genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "838a0c41",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:55.818088Z",
     "iopub.status.busy": "2021-05-21T02:46:55.816709Z",
     "iopub.status.idle": "2021-05-21T02:46:56.421045Z",
     "shell.execute_reply": "2021-05-21T02:46:56.419996Z"
    },
    "papermill": {
     "duration": 0.664776,
     "end_time": "2021-05-21T02:46:56.421177",
     "exception": false,
     "start_time": "2021-05-21T02:46:55.756401",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "254"
      ],
      "text/latex": [
       "254"
      ],
      "text/markdown": [
       "254"
      ],
      "text/plain": [
       "[1] 254"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneSymbol</th><th scope=col>n</th><th scope=col>Tissues</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>KDM5C </td><td>8</td><td>8</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SORBS2</td><td>5</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ABCD4 </td><td>3</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>BNIP2 </td><td>3</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>FGFR2 </td><td>3</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>MLH1  </td><td>3</td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & GeneSymbol & n & Tissues\\\\\n",
       "  & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & KDM5C  & 8 & 8\\\\\n",
       "\t2 & SORBS2 & 5 & 1\\\\\n",
       "\t3 & ABCD4  & 3 & 1\\\\\n",
       "\t4 & BNIP2  & 3 & 1\\\\\n",
       "\t5 & FGFR2  & 3 & 1\\\\\n",
       "\t6 & MLH1   & 3 & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | GeneSymbol &lt;fct&gt; | n &lt;int&gt; | Tissues &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | KDM5C  | 8 | 8 |\n",
       "| 2 | SORBS2 | 5 | 1 |\n",
       "| 3 | ABCD4  | 3 | 1 |\n",
       "| 4 | BNIP2  | 3 | 1 |\n",
       "| 5 | FGFR2  | 3 | 1 |\n",
       "| 6 | MLH1   | 3 | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneSymbol n Tissues\n",
       "1 KDM5C      8 8      \n",
       "2 SORBS2     5 1      \n",
       "3 ABCD4      3 1      \n",
       "4 BNIP2      3 1      \n",
       "5 FGFR2      3 1      \n",
       "6 MLH1       3 3      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genesMostFrequentlySpliced <- gene_as %>% \n",
    "                              group_by(GeneSymbol) %>% \n",
    "                              count(GeneSymbol)    %>% \n",
    "                              arrange(desc(n))     %>% \n",
    "                              as.data.frame()\n",
    "genesMostFrequentlySpliced$GeneSymbol <- factor(genesMostFrequentlySpliced$GeneSymbol, \n",
    "                                                levels = genesMostFrequentlySpliced$GeneSymbol)\n",
    "length(genesMostFrequentlySpliced$GeneSymbol)\n",
    "\n",
    "#Add number of tissues\n",
    "nTissues <- rep(NA, length(genesMostFrequentlySpliced))\n",
    "for (i in 1:nrow(genesMostFrequentlySpliced)) {\n",
    "    df_gene <- gene_as %>% \n",
    "               filter(GeneSymbol == genesMostFrequentlySpliced$GeneSymbol[i])\n",
    "    nTissues[i] <- length(unique(df_gene$Tissue))\n",
    "}\n",
    "genesMostFrequentlySpliced$Tissues <- nTissues\n",
    "head(genesMostFrequentlySpliced)\n",
    "write.table(genesMostFrequentlySpliced, file = \"../data/genesWithCommonAS.tsv\", sep = \"\\t\", quote = F, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b370b105",
   "metadata": {
    "papermill": {
     "duration": 0.056378,
     "end_time": "2021-05-21T02:46:56.534144",
     "exception": false,
     "start_time": "2021-05-21T02:46:56.477766",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.10 Count most frequent spliced chromosomes\n",
    "To get an indication of which chromosome has the most frequent slicing event (regardless of type)\n",
    "We create an index based upon the number of exons per chromosome.\n",
    "\n",
    "get the annotation file, at this writing, gencode.v30.annotation.gtf\n",
    "The information as to the number of exons within the chromosome may be found there"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "011b4d79",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:46:56.652887Z",
     "iopub.status.busy": "2021-05-21T02:46:56.651730Z",
     "iopub.status.idle": "2021-05-21T02:47:01.733827Z",
     "shell.execute_reply": "2021-05-21T02:47:01.731919Z"
    },
    "papermill": {
     "duration": 5.142958,
     "end_time": "2021-05-21T02:47:01.734184",
     "exception": false,
     "start_time": "2021-05-21T02:46:56.591226",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 23\n",
      "Columns: 4\n",
      "$ chr        \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chrX, chr10, chr16, chr17, chr12, chr9, chr21, chr18, chr7,…\n",
      "$ n          \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 18, 20, 21, 26, 22, 13, 4, 6, 16, 13, 7, 7, 5, 17, 10, 22, …\n",
      "$ ExonCounts \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 40029, 47124, 61199, 78291, 73881, 46079, 14629, 22410, 619…\n",
      "$ Index      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 0.44967399, 0.42441219, 0.34314286, 0.33209437, 0.29777615,…\n"
     ]
    }
   ],
   "source": [
    "exons <- gencode[ gencode$type == \"exon\", ]\n",
    "exons <- as.data.frame(exons)\n",
    "\n",
    "#Obtain chromosomes we have splicing information for (recall we did not use chr Y in our analysis)\n",
    "all_chr <- as.character(unique(gene_as$chr))\n",
    "chr_counts <- rep(0, length(all_chr))\n",
    "\n",
    "\n",
    "for (i in 1:length(all_chr)) {\n",
    "  chr_counts[i] <- nrow(exons[exons$seqnames == all_chr[i], ])\n",
    "}\n",
    "\n",
    "exon_counts <- data.frame(chr = all_chr, counts = chr_counts)\n",
    "\n",
    "# Count most frequent spliced chromosomes\n",
    "res <- gene_as %>% group_by(chr) %>% count(chr) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "res$chr <- factor(res$chr, levels = res$chr)\n",
    "\n",
    "idx <- match(res$chr, exon_counts$chr)\n",
    "\n",
    "res$ExonCounts <- exon_counts$counts[idx]\n",
    "\n",
    "res$Index <- (res$n / res$ExonCounts) * 1000\n",
    "\n",
    "res_sorted <- res %>% arrange(desc(Index))\n",
    "res_sorted$chr <- factor(res_sorted$chr, levels = res_sorted$chr)\n",
    "glimpse(res_sorted)\n",
    "write.table(res_sorted, file = \"../data/SplicingIndex_chr.tsv\", sep = \"\\t\", quote = F, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "128439dc",
   "metadata": {
    "papermill": {
     "duration": 0.101242,
     "end_time": "2021-05-21T02:47:01.942485",
     "exception": false,
     "start_time": "2021-05-21T02:47:01.841243",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.11 Overlap between Differential Gene Expression and Differential Alternative Splicing\n",
    "\n",
    "First gather the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "3fde4811",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:47:02.176077Z",
     "iopub.status.busy": "2021-05-21T02:47:02.173776Z",
     "iopub.status.idle": "2021-05-21T02:47:02.424774Z",
     "shell.execute_reply": "2021-05-21T02:47:02.423043Z"
    },
    "papermill": {
     "duration": 0.378471,
     "end_time": "2021-05-21T02:47:02.425070",
     "exception": false,
     "start_time": "2021-05-21T02:47:02.046599",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'A2M'</li><li>'AADAC'</li><li>'AADAC'</li><li>'AADACL3'</li><li>'AARD'</li><li>'AATBC'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A2M'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADACL3'\n",
       "\\item 'AARD'\n",
       "\\item 'AATBC'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A2M'\n",
       "2. 'AADAC'\n",
       "3. 'AADAC'\n",
       "4. 'AADACL3'\n",
       "5. 'AARD'\n",
       "6. 'AATBC'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A2M\"     \"AADAC\"   \"AADAC\"   \"AADACL3\" \"AARD\"    \"AATBC\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_AS_Genes <- read.table(file=\"../data/Total_AS_by_geneSymbol.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "sigAsGenes <- sort(total_AS_Genes$GeneSymbol)\n",
    "dge <- read.table(\"../data/gene_dge.tsv\", sep = \"\\t\", header = FALSE, row.names=1, skip = 1)\n",
    "dge_genes <- sort(dge$V5)\n",
    "head(dge_genes)\n",
    "all_genes_data <- read.table(\"../assets/all_gene_dge.tsv\")\n",
    "names(all_genes_data) <- c(\"GeneSymbol\", \"ensg\")\n",
    "all_genes <- sort(all_genes_data$GeneSymbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "3d1c6abb",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:47:02.629189Z",
     "iopub.status.busy": "2021-05-21T02:47:02.627864Z",
     "iopub.status.idle": "2021-05-21T02:47:02.642586Z",
     "shell.execute_reply": "2021-05-21T02:47:02.641482Z"
    },
    "papermill": {
     "duration": 0.102312,
     "end_time": "2021-05-21T02:47:02.642721",
     "exception": false,
     "start_time": "2021-05-21T02:47:02.540409",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ENSG00000000003.14'</li><li>'ENSG00000000005.5'</li><li>'ENSG00000000419.12'</li><li>'ENSG00000000457.13'</li><li>'ENSG00000000460.16'</li><li>'ENSG00000000938.12'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ENSG00000000003.14'\n",
       "\\item 'ENSG00000000005.5'\n",
       "\\item 'ENSG00000000419.12'\n",
       "\\item 'ENSG00000000457.13'\n",
       "\\item 'ENSG00000000460.16'\n",
       "\\item 'ENSG00000000938.12'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ENSG00000000003.14'\n",
       "2. 'ENSG00000000005.5'\n",
       "3. 'ENSG00000000419.12'\n",
       "4. 'ENSG00000000457.13'\n",
       "5. 'ENSG00000000460.16'\n",
       "6. 'ENSG00000000938.12'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"ENSG00000000003.14\" \"ENSG00000000005.5\"  \"ENSG00000000419.12\"\n",
       "[4] \"ENSG00000000457.13\" \"ENSG00000000460.16\" \"ENSG00000000938.12\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(all_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86dca4f6",
   "metadata": {
    "papermill": {
     "duration": 0.058258,
     "end_time": "2021-05-21T02:47:02.759589",
     "exception": false,
     "start_time": "2021-05-21T02:47:02.701331",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.12 We then do a hypergeometric/Fisher test to look for overrepresentation\n",
    "The universe consists of all genes with at least one read (all_genes_data).\n",
    "So we have\n",
    "\n",
    "|  \t|  DGE+| DGE-|\n",
    "|-\t|-\t|-\t|\n",
    "| DAS+|  a|  b|\n",
    "| DAS-|  c| d|"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "2f8e283c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:47:02.880881Z",
     "iopub.status.busy": "2021-05-21T02:47:02.879590Z",
     "iopub.status.idle": "2021-05-21T02:47:02.930894Z",
     "shell.execute_reply": "2021-05-21T02:47:02.929752Z"
    },
    "papermill": {
     "duration": 0.113746,
     "end_time": "2021-05-21T02:47:02.931059",
     "exception": false,
     "start_time": "2021-05-21T02:47:02.817313",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of sigAsGenes 254\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of genes that are NOT sigAs 24185\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of DGE genes4219\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of genes that are NOT DGE 24185\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "a: 12; b: 0; c: 0; d: 24185\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"Number of sigAsGenes \", length(sigAsGenes))\n",
    "notSigAs <- setdiff(all_genes,sigAsGenes)\n",
    "message(\"Number of genes that are NOT sigAs \", length(notSigAs))\n",
    "message(\"Number of DGE genes\", length(dge_genes))\n",
    "notDGE <- setdiff(all_genes,dge_genes)\n",
    "message(\"Number of genes that are NOT DGE \", length(notDGE))\n",
    "a <- intersect(sigAsGenes, dge_genes)\n",
    "b <- intersect(sigAsGenes, notDGE)\n",
    "c <- intersect(notSigAs, dge_genes)\n",
    "d <- intersect(notSigAs, notDGE)\n",
    "message(\"a: \", length(a), \"; b: \",  length(b), \"; c: \",  length(c), \"; d: \",  length(d))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "708898f1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:47:03.054403Z",
     "iopub.status.busy": "2021-05-21T02:47:03.052935Z",
     "iopub.status.idle": "2021-05-21T02:47:03.073760Z",
     "shell.execute_reply": "2021-05-21T02:47:03.072627Z"
    },
    "papermill": {
     "duration": 0.083457,
     "end_time": "2021-05-21T02:47:03.073898",
     "exception": false,
     "start_time": "2021-05-21T02:47:02.990441",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tFisher's Exact Test for Count Data\n",
       "\n",
       "data:  m\n",
       "p-value < 2.2e-16\n",
       "alternative hypothesis: true odds ratio is not equal to 1\n",
       "95 percent confidence interval:\n",
       " 8402.051      Inf\n",
       "sample estimates:\n",
       "odds ratio \n",
       "       Inf \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "m <- matrix(c(length(a),length(b),length(c),length(d)), nrow=2,byrow = TRUE)\n",
    "fisher.test(m)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b01a8d81",
   "metadata": {
    "papermill": {
     "duration": 0.060172,
     "end_time": "2021-05-21T02:47:03.193659",
     "exception": false,
     "start_time": "2021-05-21T02:47:03.133487",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Appendix - Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2194f233",
   "metadata": {
    "papermill": {
     "duration": 0.059592,
     "end_time": "2021-05-21T02:47:03.313540",
     "exception": false,
     "start_time": "2021-05-21T02:47:03.253948",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Appendix 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "0a1f139a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:47:03.437312Z",
     "iopub.status.busy": "2021-05-21T02:47:03.436119Z",
     "iopub.status.idle": "2021-05-21T02:47:04.245141Z",
     "shell.execute_reply": "2021-05-21T02:47:04.243568Z"
    },
    "papermill": {
     "duration": 0.872386,
     "end_time": "2021-05-21T02:47:04.245347",
     "exception": false,
     "start_time": "2021-05-21T02:47:03.372961",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'countGenesAndEvents'"
      ],
      "text/latex": [
       "'countGenesAndEvents'"
      ],
      "text/markdown": [
       "'countGenesAndEvents'"
      ],
      "text/plain": [
       "[1] \"countGenesAndEvents\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/gene_as.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/all_gene_as.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/gene_dge.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_chr.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_geneSymbol.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_tissue.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_splicingtype.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/genesWithCommonAS.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/SplicingIndex_chr.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"countGenesAndEvents\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/gene_as.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_as.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/all_gene_as.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_dge.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/gene_dge.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_as.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_chr.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_chr.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_geneSymbol.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_geneSymbol.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_tissue.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_tissue.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_splicingtype.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_splicingtype.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/genesWithCommonAS.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name genesWithCommonAS.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/SplicingIndex_chr.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name SplicingIndex_chr.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5a68839",
   "metadata": {
    "papermill": {
     "duration": 0.066503,
     "end_time": "2021-05-21T02:47:04.379347",
     "exception": false,
     "start_time": "2021-05-21T02:47:04.312844",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Appendix 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "raw",
   "id": "40048015",
   "metadata": {
    "lines_to_next_cell": 0,
    "papermill": {
     "duration": 0.067096,
     "end_time": "2021-05-21T02:47:04.513802",
     "exception": false,
     "start_time": "2021-05-21T02:47:04.446706",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../data/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../data/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../data/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../data/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "7cc70f16",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T02:47:04.655050Z",
     "iopub.status.busy": "2021-05-21T02:47:04.653374Z",
     "iopub.status.idle": "2021-05-21T02:47:04.675645Z",
     "shell.execute_reply": "2021-05-21T02:47:04.674200Z"
    },
    "papermill": {
     "duration": 0.094379,
     "end_time": "2021-05-21T02:47:04.675773",
     "exception": false,
     "start_time": "2021-05-21T02:47:04.581394",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Time difference of 3.172908 mins"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "end_time <- Sys.time()\n",
    "end_time - start_time"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "encoding": "# -*- coding: utf-8 -*-",
   "main_language": "R",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  },
  "papermill": {
   "duration": 191.741801,
   "end_time": "2021-05-21T02:47:04.956662",
   "environment_variables": {},
   "exception": null,
   "input_path": "countGenesAndEvents.ipynb",
   "output_path": "countGenesAndEvents.ipynb",
   "parameters": {},
   "start_time": "2021-05-21T02:43:53.214861",
   "version": "2.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}