{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7879e876",
   "metadata": {
    "papermill": {
     "duration": 0.027739,
     "end_time": "2021-06-27T18:40:09.349631",
     "exception": false,
     "start_time": "2021-06-27T18:40:09.321892",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Analysis Notebook - Count Genes and Events\n",
    "\n",
    "This notebook processes the raw counts as provided by rMATS 3.2.5 with modifications because this rMATS was run at scale with the entirety of GTEx.   To so, please refer to the methods in the paper, but in brief, each sample was run against another.  This method of running rMATS 3.2.5, permitted us leveraging the counting of the reads done by rMATS together with the scrutinization of rMATS against the provided GTF.   In this case, gencode's complete annotated gtf was scanned by rMATS 3.2.5, the specific possible alternative splicing events catalogued and used in the `fromGTF` descriptions which capture the specific genomic locations for each of five alternative splicing types.  Then the short read RNA sequencing files collected by GTEx and annotated by sex and tissue were aligned and counted against the specific junctions as described by these fromGTF files.   It is these counts that were used in the analyses for this paper.  \n",
    "\n",
    "This notebook, counts the genes and events and performs several descriptive statistical analysis. It  produces outputs needed for downstream visualizations, which are in separate notebooks for added clarity.\n",
    "\n",
    "## 1 Introduction\n",
    "\n",
    "## 1.1 Data files needed by this notebook\n",
    "\n",
    "Input files have been pulled from results, the data were primarily obtained through the execution of the Nextflow workflow, rmats-nf, found in the GitHub repository, https://github.com/lifebit-ai/rmats-nf or the notebook.   How the data were generated are noted for each of the files.\n",
    "\n",
    "1. **`*`.model_B_sex_as_event.csv** One for each tissue, generated by running the notebook `differentialSplicingJunctionAnalysis.ipynb` as a Nextflow workflow\n",
    "2. **`*`.model_B_sex_as_event_refined.csv** One for each tissue, these are the statistically significant results FC > 1.5 and p-value < 0.05 generated by running the notebook `differentialSplicingJunctionAnalysis.ipynb` as a Nextflow workflow.\n",
    "3. **fromGTF.`*`.txt** One for each splicing type, generated within rmats-nf, a Nextflow workflow found here `https://github.com/lifebit-ai/rmats-nf` (fromGTF.A3SS.txt, fromGTF.A5SS.txt, fromGTF.MXE.txt, fromGTF.RI.txt,fromGTF.SE.txt)\n",
    "4. **`*`.DGE.csv** One for each tissue, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "5. **`*`.DGE_refined.csv** One for each tissue, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "6. **srr_pdata.csv** Sequence Run (SRR) data merged with phenotype data, generated by `differentialGeneExpressionAnalysis.ipynb`\n",
    "7. **rmats_final.`*`.jc.ijc.txt.gz** One for each splicing type, this is a matrix of all included junction (ijc) counts for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`. \n",
    "8. **rmats_final.`*`.jc.inclen.txt.gz** One for each splicing type, this is a matrix of inclusion length (inclen) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "9. **rmats_final.`*`.jc.inclen.txt.gz** One for each splicing type, this is a matrix of inclen for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "10. **rmats_final.`*`.jc.inc.txt.gz** One for each splicing type, this is a matrix of percent spliced in (inc) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "11. **rmats_final.`*`.jc.sjc.txt.gz** One for each splicing type, this is a matrix of skipped junction counts (sjc) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`.\n",
    "12. **rmats_final.`*`.jc.skiplen.txt.gz** One for each splicing type, this is a matrix of skipped length (skiplen) for each junction and for each sample (SRR) generated by rmats-nf found here `https://github.com/lifebit-ai/rmats-nf`\n",
    "\n",
    "Two files needed from the `sbas/assets` directory\n",
    "\n",
    "13. **tissues.csv** This file curated by each tissue, the number of female and male samples and provides a display name for the graphics.\n",
    "14. **all_gene_dge.tsv** This file is an intersection of all the differentially expressed genes (not just the significant but all) for all the tissues considered in this study generated by a custom script.\n",
    "\n",
    "\n",
    "## 1.2 Data files created by this notebook\n",
    "\n",
    "Output text files are written to the ``sbas/data`` directory (this notebook is in the ``sbas/jupyter`` directory). \n",
    "\n",
    "1.  **gene_as.tsv**: Significant alternative splicing events per gene, adjusted p-value <= 0.05, fold change >= 1.5.\n",
    "2.  **all_gene_as.tsv**: all alternative splicing events\n",
    "3.  **gene_dge.tsv**: Significant differential gene expression, adjusted p-value <= 0.05, fold change >= 1.5\n",
    "4.  **all_gene_dge.tsv**: all differential gene event files is NOT produced here but is found in the assets directory ../assets/all_gene_dge.tsv\n",
    "5.  **genesWithCommonAS.tsv**: genes (as geneSymbol, the number of splicing events, and the number of tissues the event occurs in)\n",
    "6.  **Total_AS_by_chr.tsv**: Total alternative splicing events per chromosome\n",
    "7.  **Total_AS_by_geneSymbol.tsv**: Count the number of tissues in which specific genes show significant alternative splicing\n",
    "8.  **DGE_by_geneSymbol.tsv***: Most highly expressed genes by tissue\n",
    "9.  **Total_AS_by_tissue.tsv**: Count the number of significant splicing events per tissue\n",
    "10. **Total_AS_by_splicingtype.tsv**: Count number of significant splicing events for each of the 5' alternative splicing categories\n",
    "11. **SplicingIndex_chr.tsv**: Splicing index by chr (number of sigificant AS events per 1000 exons)\n",
    "\n",
    "## 2 Run Analyses\n",
    "\n",
    "## 2.1 Load Libraries\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "18bf101c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:09.514209Z",
     "iopub.status.busy": "2021-06-27T18:40:09.511864Z",
     "iopub.status.idle": "2021-06-27T18:40:17.270938Z",
     "shell.execute_reply": "2021-06-27T18:40:17.269898Z"
    },
    "papermill": {
     "duration": 7.895192,
     "end_time": "2021-06-27T18:40:17.271155",
     "exception": false,
     "start_time": "2021-06-27T18:40:09.375963",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#conda install r-dplyr bioconductor-biobase r-tibble r-r.utils bioconductor-rtracklayer -y\n",
    "start_time <- Sys.time()\n",
    "suppressMessages({\n",
    "    options(warn = -1) \n",
    "    library(readr)\n",
    "    library(dplyr)\n",
    "    library(Biobase)\n",
    "    library(tibble)\n",
    "    library(R.utils)\n",
    "    library(rtracklayer)\n",
    "})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "780bc774",
   "metadata": {
    "papermill": {
     "duration": 0.026057,
     "end_time": "2021-06-27T18:40:17.324252",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.298195",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.2 Read in Curated Tissue List\n",
    "\n",
    "Read in a curated list of tissues, noting sample counts per each sex, male and female and also a curated tissue name for display.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1e8b8da8",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:17.413567Z",
     "iopub.status.busy": "2021-06-27T18:40:17.380138Z",
     "iopub.status.idle": "2021-06-27T18:40:17.425060Z",
     "shell.execute_reply": "2021-06-27T18:40:17.423691Z"
    },
    "papermill": {
     "duration": 0.074921,
     "end_time": "2021-06-27T18:40:17.425185",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.350264",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(readr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "96e68c2e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:17.483378Z",
     "iopub.status.busy": "2021-06-27T18:40:17.481696Z",
     "iopub.status.idle": "2021-06-27T18:40:17.617674Z",
     "shell.execute_reply": "2021-06-27T18:40:17.616209Z"
    },
    "papermill": {
     "duration": 0.166769,
     "end_time": "2021-06-27T18:40:17.617957",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.451188",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Adipose (sc)'</li><li>'Adipose (v)'</li><li>'Adrenal gland'</li><li>'Aorta'</li><li>'Atrial appendage'</li><li>'Breast'</li><li>'Caudate'</li><li>'Cerebellar hemisphere'</li><li>'Cerebellum'</li><li>'Coronary artery'</li><li>'Cortex'</li><li>'EBV-lymphocytes'</li><li>'Esophagus (gej)'</li><li>'Esophagus (m)'</li><li>'Esophagus (mu)'</li><li>'Fibroblasts'</li><li>'Frontal cortex'</li><li>'Hippocampus'</li><li>'Hypothalamus'</li><li>'Left ventricle'</li><li>'Liver'</li><li>'Lung'</li><li>'Nucleus accumbens'</li><li>'Pancreas'</li><li>'Pituitary'</li><li>'Putamen'</li><li>'Sigmoid colon'</li><li>'Skeletal muscle'</li><li>'Skin (exposed)'</li><li>'Skin (not exposed)'</li><li>'Small intestine'</li><li>'Spinal cord'</li><li>'Spleen'</li><li>'Stomach'</li><li>'Thyroid'</li><li>'Tibial artery'</li><li>'Tibial nerve'</li><li>'Transverse colon'</li><li>'Whole blood'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Adipose (sc)'\n",
       "\\item 'Adipose (v)'\n",
       "\\item 'Adrenal gland'\n",
       "\\item 'Aorta'\n",
       "\\item 'Atrial appendage'\n",
       "\\item 'Breast'\n",
       "\\item 'Caudate'\n",
       "\\item 'Cerebellar hemisphere'\n",
       "\\item 'Cerebellum'\n",
       "\\item 'Coronary artery'\n",
       "\\item 'Cortex'\n",
       "\\item 'EBV-lymphocytes'\n",
       "\\item 'Esophagus (gej)'\n",
       "\\item 'Esophagus (m)'\n",
       "\\item 'Esophagus (mu)'\n",
       "\\item 'Fibroblasts'\n",
       "\\item 'Frontal cortex'\n",
       "\\item 'Hippocampus'\n",
       "\\item 'Hypothalamus'\n",
       "\\item 'Left ventricle'\n",
       "\\item 'Liver'\n",
       "\\item 'Lung'\n",
       "\\item 'Nucleus accumbens'\n",
       "\\item 'Pancreas'\n",
       "\\item 'Pituitary'\n",
       "\\item 'Putamen'\n",
       "\\item 'Sigmoid colon'\n",
       "\\item 'Skeletal muscle'\n",
       "\\item 'Skin (exposed)'\n",
       "\\item 'Skin (not exposed)'\n",
       "\\item 'Small intestine'\n",
       "\\item 'Spinal cord'\n",
       "\\item 'Spleen'\n",
       "\\item 'Stomach'\n",
       "\\item 'Thyroid'\n",
       "\\item 'Tibial artery'\n",
       "\\item 'Tibial nerve'\n",
       "\\item 'Transverse colon'\n",
       "\\item 'Whole blood'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Adipose (sc)'\n",
       "2. 'Adipose (v)'\n",
       "3. 'Adrenal gland'\n",
       "4. 'Aorta'\n",
       "5. 'Atrial appendage'\n",
       "6. 'Breast'\n",
       "7. 'Caudate'\n",
       "8. 'Cerebellar hemisphere'\n",
       "9. 'Cerebellum'\n",
       "10. 'Coronary artery'\n",
       "11. 'Cortex'\n",
       "12. 'EBV-lymphocytes'\n",
       "13. 'Esophagus (gej)'\n",
       "14. 'Esophagus (m)'\n",
       "15. 'Esophagus (mu)'\n",
       "16. 'Fibroblasts'\n",
       "17. 'Frontal cortex'\n",
       "18. 'Hippocampus'\n",
       "19. 'Hypothalamus'\n",
       "20. 'Left ventricle'\n",
       "21. 'Liver'\n",
       "22. 'Lung'\n",
       "23. 'Nucleus accumbens'\n",
       "24. 'Pancreas'\n",
       "25. 'Pituitary'\n",
       "26. 'Putamen'\n",
       "27. 'Sigmoid colon'\n",
       "28. 'Skeletal muscle'\n",
       "29. 'Skin (exposed)'\n",
       "30. 'Skin (not exposed)'\n",
       "31. 'Small intestine'\n",
       "32. 'Spinal cord'\n",
       "33. 'Spleen'\n",
       "34. 'Stomach'\n",
       "35. 'Thyroid'\n",
       "36. 'Tibial artery'\n",
       "37. 'Tibial nerve'\n",
       "38. 'Transverse colon'\n",
       "39. 'Whole blood'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Adipose (sc)\"          \"Adipose (v)\"           \"Adrenal gland\"        \n",
       " [4] \"Aorta\"                 \"Atrial appendage\"      \"Breast\"               \n",
       " [7] \"Caudate\"               \"Cerebellar hemisphere\" \"Cerebellum\"           \n",
       "[10] \"Coronary artery\"       \"Cortex\"                \"EBV-lymphocytes\"      \n",
       "[13] \"Esophagus (gej)\"       \"Esophagus (m)\"         \"Esophagus (mu)\"       \n",
       "[16] \"Fibroblasts\"           \"Frontal cortex\"        \"Hippocampus\"          \n",
       "[19] \"Hypothalamus\"          \"Left ventricle\"        \"Liver\"                \n",
       "[22] \"Lung\"                  \"Nucleus accumbens\"     \"Pancreas\"             \n",
       "[25] \"Pituitary\"             \"Putamen\"               \"Sigmoid colon\"        \n",
       "[28] \"Skeletal muscle\"       \"Skin (exposed)\"        \"Skin (not exposed)\"   \n",
       "[31] \"Small intestine\"       \"Spinal cord\"           \"Spleen\"               \n",
       "[34] \"Stomach\"               \"Thyroid\"               \"Tibial artery\"        \n",
       "[37] \"Tibial nerve\"          \"Transverse colon\"      \"Whole blood\"          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted 39 different tissues with at least 50 samples in both M & f\n",
      "\n"
     ]
    }
   ],
   "source": [
    "tissue_reduction_filename <- \"../assets/tissues.tsv\"\n",
    "#tissue_reduction <- read.table(tissue_reduction_filename, header=TRUE, sep=\"\\t\",\n",
    "#                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "tissue_reduction <- data.table::fread(tissue_reduction_filename)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$display_name != \"n/a\",]\n",
    "tissue_reduction$display_name <- factor(tissue_reduction$display_name)\n",
    "levels(tissue_reduction$display_name)\n",
    "message(\"We extracted \", length(levels(tissue_reduction$display_name)),\" different tissues with at least 50 samples in both M & f\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14505579",
   "metadata": {
    "papermill": {
     "duration": 0.02691,
     "end_time": "2021-06-27T18:40:17.672085",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.645175",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.3 Read in refined differential AS events\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "59aeae15",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:17.731977Z",
     "iopub.status.busy": "2021-06-27T18:40:17.730249Z",
     "iopub.status.idle": "2021-06-27T18:40:17.770364Z",
     "shell.execute_reply": "2021-06-27T18:40:17.768813Z"
    },
    "papermill": {
     "duration": 0.071538,
     "end_time": "2021-06-27T18:40:17.770495",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.698957",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted 195 model_B_sex_as_events_refined.csv files\n",
      "\n"
     ]
    }
   ],
   "source": [
    "significant_results_dir = \"../data/\"\n",
    "pattern = \"model_B_sex_as_events_refined.csv\"\n",
    "files <- list.files(path = significant_results_dir, pattern = pattern)\n",
    "as_types <- c(\"a3ss\", \"a5ss\", \"mxe\", \"ri\", \"se\")\n",
    "message(\"We extracted \", length(files), \" model_B_sex_as_events_refined.csv files\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0be3e8ad",
   "metadata": {
    "papermill": {
     "duration": 0.027001,
     "end_time": "2021-06-27T18:40:17.824896",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.797895",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.4 Read in the AS Events annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4c175624",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:17.884549Z",
     "iopub.status.busy": "2021-06-27T18:40:17.883037Z",
     "iopub.status.idle": "2021-06-27T18:40:17.963183Z",
     "shell.execute_reply": "2021-06-27T18:40:17.961694Z"
    },
    "papermill": {
     "duration": 0.111601,
     "end_time": "2021-06-27T18:40:17.963330",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.851729",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "a3ss_annot <- data.table::fread(\"../data/fromGTF.A3SS.txt\")\n",
    "a5ss_annot <- data.table::fread(\"../data/fromGTF.A5SS.txt\")\n",
    "mxe_annot  <- data.table::fread(\"../data/fromGTF.MXE.txt\")\n",
    "ri_annot   <- data.table::fread(\"../data/fromGTF.RI.txt\")\n",
    "se_annot   <- data.table::fread(\"../data/fromGTF.SE.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7f58ac0f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:18.024630Z",
     "iopub.status.busy": "2021-06-27T18:40:18.022918Z",
     "iopub.status.idle": "2021-06-27T18:40:18.134785Z",
     "shell.execute_reply": "2021-06-27T18:40:18.133530Z"
    },
    "papermill": {
     "duration": 0.144143,
     "end_time": "2021-06-27T18:40:18.134922",
     "exception": false,
     "start_time": "2021-06-27T18:40:17.990779",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>GeneID</th><th scope=col>geneSymbol</th><th scope=col>chr</th><th scope=col>strand</th><th scope=col>exonStart_0base</th><th scope=col>exonEnd</th><th scope=col>upstreamES</th><th scope=col>upstreamEE</th><th scope=col>downstreamES</th><th scope=col>downstreamEE</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21287990</td><td>21288091</td><td>21284709</td><td>21284969</td><td>21295674</td><td>21295769</td></tr>\n",
       "\t<tr><td>2</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21303182</td><td>21303234</td><td>21302142</td><td>21302259</td><td>21304425</td><td>21304553</td></tr>\n",
       "\t<tr><td>3</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21287990</td><td>21288091</td><td>21296085</td><td>21296143</td></tr>\n",
       "\t<tr><td>4</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21287990</td><td>21288091</td><td>21298412</td><td>21298479</td></tr>\n",
       "\t<tr><td>5</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21284710</td><td>21284969</td><td>21296085</td><td>21296143</td></tr>\n",
       "\t<tr><td>6</td><td>ENSG00000034152.18</td><td>MAP2K3</td><td>chr17</td><td>+</td><td>21295674</td><td>21295769</td><td>21284710</td><td>21284969</td><td>21298412</td><td>21298479</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " ID & GeneID & geneSymbol & chr & strand & exonStart\\_0base & exonEnd & upstreamES & upstreamEE & downstreamES & downstreamEE\\\\\n",
       " <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21287990 & 21288091 & 21284709 & 21284969 & 21295674 & 21295769\\\\\n",
       "\t 2 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21303182 & 21303234 & 21302142 & 21302259 & 21304425 & 21304553\\\\\n",
       "\t 3 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21287990 & 21288091 & 21296085 & 21296143\\\\\n",
       "\t 4 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21287990 & 21288091 & 21298412 & 21298479\\\\\n",
       "\t 5 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21284710 & 21284969 & 21296085 & 21296143\\\\\n",
       "\t 6 & ENSG00000034152.18 & MAP2K3 & chr17 & + & 21295674 & 21295769 & 21284710 & 21284969 & 21298412 & 21298479\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 11\n",
       "\n",
       "| ID &lt;int&gt; | GeneID &lt;chr&gt; | geneSymbol &lt;chr&gt; | chr &lt;chr&gt; | strand &lt;chr&gt; | exonStart_0base &lt;int&gt; | exonEnd &lt;int&gt; | upstreamES &lt;int&gt; | upstreamEE &lt;int&gt; | downstreamES &lt;int&gt; | downstreamEE &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21287990 | 21288091 | 21284709 | 21284969 | 21295674 | 21295769 |\n",
       "| 2 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21303182 | 21303234 | 21302142 | 21302259 | 21304425 | 21304553 |\n",
       "| 3 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21287990 | 21288091 | 21296085 | 21296143 |\n",
       "| 4 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21287990 | 21288091 | 21298412 | 21298479 |\n",
       "| 5 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21284710 | 21284969 | 21296085 | 21296143 |\n",
       "| 6 | ENSG00000034152.18 | MAP2K3 | chr17 | + | 21295674 | 21295769 | 21284710 | 21284969 | 21298412 | 21298479 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID GeneID             geneSymbol chr   strand exonStart_0base exonEnd \n",
       "1 1  ENSG00000034152.18 MAP2K3     chr17 +      21287990        21288091\n",
       "2 2  ENSG00000034152.18 MAP2K3     chr17 +      21303182        21303234\n",
       "3 3  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "4 4  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "5 5  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "6 6  ENSG00000034152.18 MAP2K3     chr17 +      21295674        21295769\n",
       "  upstreamES upstreamEE downstreamES downstreamEE\n",
       "1 21284709   21284969   21295674     21295769    \n",
       "2 21302142   21302259   21304425     21304553    \n",
       "3 21287990   21288091   21296085     21296143    \n",
       "4 21287990   21288091   21298412     21298479    \n",
       "5 21284710   21284969   21296085     21296143    \n",
       "6 21284710   21284969   21298412     21298479    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(se_annot)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f435c2e6",
   "metadata": {
    "papermill": {
     "duration": 0.028923,
     "end_time": "2021-06-27T18:40:18.191643",
     "exception": false,
     "start_time": "2021-06-27T18:40:18.162720",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.5 create_as_structure \n",
    "\n",
    "This function doees an aggregation of the alternative splicing events - good for all events and the significantly expressed events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fd7cf418",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:18.253756Z",
     "iopub.status.busy": "2021-06-27T18:40:18.252037Z",
     "iopub.status.idle": "2021-06-27T18:40:18.266097Z",
     "shell.execute_reply": "2021-06-27T18:40:18.264925Z"
    },
    "papermill": {
     "duration": 0.046896,
     "end_time": "2021-06-27T18:40:18.266218",
     "exception": false,
     "start_time": "2021-06-27T18:40:18.219322",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "create_as_structure <- function ( results_dir, files, all_or_das, pattern, tissue_reduction) {\n",
    "    gene_as = data.frame()\n",
    "    counts <- rep(NA, length(files))\n",
    "    message(\"\\nnumber of files:\", paste(length(files)), collapse = \"\")\n",
    "    for (i in 1:length(files)) {\n",
    "       lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "       if (dim(lines)[1] > 0) {\n",
    "           event     <- as.vector(as.character(rownames(lines)))\n",
    "           tissue1   <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "           counts[i] <- dim(lines)[1]\n",
    "           event_idx <- substring(event, regexpr(\"[0-9]+$\", event))\n",
    "           res       <- data.frame()\n",
    "           if (grepl(\"^a3ss_\", files[i])) {\n",
    "               # remove the first 5 letters of the string \n",
    "               tissue2 <- substring(tissue1,6)\n",
    "               idx <- match(event_idx, a3ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"A3SS\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- a3ss_annot$geneSymbol[idx],\n",
    "                              GeneID       <- a3ss_annot$GeneID[idx],\n",
    "                              chr          <- a3ss_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "            \n",
    "           } else if (grepl(\"^a5ss_\", files[i])) {\n",
    "               # remove the first 5 letters of the string \n",
    "               tissue2 <- substring(tissue1,6)\n",
    "               idx <- match(event_idx, a5ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"A5SS\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- a5ss_annot$geneSymbol[idx],\n",
    "                              GeneID       <- a5ss_annot$GeneID[idx],\n",
    "                              chr          <- a5ss_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                               \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^mxe_\", files[i])) {\n",
    "               # remove the first 4 letters of the string \n",
    "               tissue2 <- substring(tissue1,5)\n",
    "               idx <- match(event_idx, a3ss_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"MXE\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- mxe_annot$geneSymbol[idx],\n",
    "                              GeneID       <- mxe_annot$GeneID[idx],\n",
    "                              chr          <- mxe_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^se_\", files[i])) {\n",
    "               # remove the first 3 letters of the string \n",
    "               tissue2 <- substring(tissue1,4)\n",
    "               idx <- match(event_idx, se_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"SE\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- se_annot$geneSymbol[idx],\n",
    "                              GeneID       <- se_annot$GeneID[idx],\n",
    "                              chr          <- se_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           } else if (grepl(\"^ri_\", files[i])){\n",
    "               # remove the first 3 letters of the string \n",
    "               tissue2 <- substring(tissue1,4)\n",
    "               idx <- match(event_idx, ri_annot$ID)\n",
    "               res <- data.frame(GeneJunction <- event,\n",
    "                              ASE          <- \"RI\", \n",
    "                              ASE_IDX      <- idx,\n",
    "                              Tissue       <- tissue2,\n",
    "                              counts       <- counts[i],\n",
    "                              Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue2, \"display_name\"],\n",
    "                              GeneSymbol   <- ri_annot$geneSymbol[idx],\n",
    "                              GeneID       <- ri_annot$GeneID[idx],\n",
    "                              chr          <- ri_annot$chr[idx],\n",
    "                              logFC        <- lines$logFC,\n",
    "                              AveExpr      <- lines$AveExpr,\n",
    "                              t            <- lines$t,\n",
    "                              PValue       <- lines$P.Value,\n",
    "                              AdjPVal      <- lines$adj.P.Val,\n",
    "                              B            <- lines$B)\n",
    "               colnames(res) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\n",
    "                                  \"GeneSymbol\",\"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "               gene_as <- rbind(gene_as,res)\n",
    "           }\n",
    "        \n",
    "       } #if has sig. events\n",
    "    \n",
    "   } #for all files\n",
    "   colnames(gene_as) <- c(\"GeneJunction\",\"ASE\",\"ASE_IDX\",\"Tissue\",\"counts\",\"Display\",\"GeneSymbol\",\n",
    "                           \"GeneID\",\"chr\",\"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "   n_unique_genes <- length(summary(as.factor(gene_as$GeneSymbol),maxsum=50000))\n",
    "   message(\"For the run for \", all_or_das, \" run\")\n",
    "   message(\"We extracted a total of \",nrow(gene_as),\" alternative splicing events (gene_as)\")\n",
    "   message(\"This includes \", n_unique_genes, \" total genes\")\n",
    "   return (gene_as)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "287c4c6d",
   "metadata": {
    "papermill": {
     "duration": 0.028214,
     "end_time": "2021-06-27T18:40:18.322896",
     "exception": false,
     "start_time": "2021-06-27T18:40:18.294682",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.6 create_dge_structure \n",
    "\n",
    "This function does an aggregation of the differential gene expression events - good for all events and the significantly expressed events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b9aa4112",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:18.384129Z",
     "iopub.status.busy": "2021-06-27T18:40:18.382657Z",
     "iopub.status.idle": "2021-06-27T18:40:18.394772Z",
     "shell.execute_reply": "2021-06-27T18:40:18.393566Z"
    },
    "papermill": {
     "duration": 0.044035,
     "end_time": "2021-06-27T18:40:18.394908",
     "exception": false,
     "start_time": "2021-06-27T18:40:18.350873",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "create_dge_structure <- function ( results_dir, files, all_or_dge, pattern, map_pattern, tissue_reduction) {\n",
    "   gene_dge = data.frame()\n",
    "   counts <- rep(NA, length(files))\n",
    "   for (i in 1:length(files)) {\n",
    "      lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "      if (dim(lines)[1] > 0) {\n",
    "         tissue1    <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "         map_lines  <- read.table(file=paste0(paste0(results_dir, tissue1),map_pattern),\n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "         counts[i]  <- dim(lines)[1]    \n",
    "         ensg_ver   <- as.vector(as.character(rownames(lines)))\n",
    "         ensg_no_ver<- as.vector(as.character(map_lines$ensg_names))\n",
    "         ensg_genes <- as.vector(as.character(map_lines$ensg_genes))\n",
    "         counts[i]  <- dim(lines)[1]  \n",
    "         res <- data.frame(Tissue       <- tissue1,\n",
    "                           ENSG_ver     <- ensg_ver,\n",
    "                           ENSG_no_ver  <- ensg_no_ver,\n",
    "                           GeneSymbol   <- ensg_genes,\n",
    "                           counts       <- counts[i],\n",
    "                           Display      <- tissue_reduction[tissue_reduction$SMTSD == tissue1, \"display_name\"],\n",
    "                           logFC        <- lines$logFC,\n",
    "                           AveExpr      <- lines$AveExpr,\n",
    "                           t            <- lines$t,\n",
    "                           PValue       <- lines$P.Value,\n",
    "                           AdjPVal      <- lines$adj.P.Val,\n",
    "                           B            <- lines$B)\n",
    "         colnames(res) <- c(\"Tissue\",\"ENSG_ver\",\"ENSG_no_ver\",\"GeneSymbol\",\"counts\",\"Display\",\n",
    "                            \"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "         gene_dge <- rbind(gene_dge, res)\n",
    "       } #if has sig. events\n",
    "    } #for all files\n",
    "    colnames(gene_dge) <- c(\"Tissue\",\"ENSG_ver\",\"ENSG_no_ver\",\"GeneSymbol\",\"counts\",\"Display\",\n",
    "                        \"logFC\",\"AveExpr\",\"t\",\"PValue\",\"AdjPVal\",\"B\")\n",
    "    n_unique_genes <- length(summary(as.factor(gene_dge$GeneSymbol),maxsum=50000))\n",
    "    message(\"For the run for \", all_or_dge, \"run\")\n",
    "    message(\"We extracted a total of \",nrow(gene_dge),\" gene events (gene_dge)\")\n",
    "    message(\"This includes \", n_unique_genes, \" total genes\")\n",
    "    return(gene_dge)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fccf9d9",
   "metadata": {
    "papermill": {
     "duration": 0.02801,
     "end_time": "2021-06-27T18:40:18.450759",
     "exception": false,
     "start_time": "2021-06-27T18:40:18.422749",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.7 Read in the alternative splicing results\n",
    "\n",
    "We will create an aggregation of  all the results and all the significant results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "dfd86e0d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:40:18.511711Z",
     "iopub.status.busy": "2021-06-27T18:40:18.510139Z",
     "iopub.status.idle": "2021-06-27T18:41:55.255533Z",
     "shell.execute_reply": "2021-06-27T18:41:55.254386Z"
    },
    "papermill": {
     "duration": 96.777137,
     "end_time": "2021-06-27T18:41:55.255668",
     "exception": false,
     "start_time": "2021-06-27T18:40:18.478531",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Length of all_files: 195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Length of significant_files: 195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "number of files:195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for differentially significant alternative splicing run\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted a total of 743 alternative splicing events (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "This includes 581 total genes\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "number of files:195\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for all alternatively spliced run\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted a total of 1132598 alternative splicing events (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "This includes 12681 total genes\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneJunction</th><th scope=col>ASE</th><th scope=col>ASE_IDX</th><th scope=col>Tissue</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>GeneSymbol</th><th scope=col>GeneID</th><th scope=col>chr</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SCO1-8452</td><td>A3SS</td><td>8452</td><td>adrenal_gland  </td><td>1</td><td>Adrenal gland  </td><td>SCO1</td><td>ENSG00000133028.12</td><td>chr17</td><td> 0.7715834</td><td>5.028635</td><td>  4.906848</td><td>1.533202e-06</td><td>8.170436e-03</td><td> 0.8561371</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>XIST-2252</td><td>A3SS</td><td>2252</td><td>artery_coronary</td><td>1</td><td>Coronary artery</td><td>XIST</td><td>ENSG00000229807.11</td><td>chrX </td><td>-2.1880965</td><td>4.327079</td><td>-10.329346</td><td>4.631685e-21</td><td>2.464057e-17</td><td>16.7535120</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & GeneJunction & ASE & ASE\\_IDX & Tissue & counts & Display & GeneSymbol & GeneID & chr & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <int> & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & SCO1-8452 & A3SS & 8452 & adrenal\\_gland   & 1 & Adrenal gland   & SCO1 & ENSG00000133028.12 & chr17 &  0.7715834 & 5.028635 &   4.906848 & 1.533202e-06 & 8.170436e-03 &  0.8561371\\\\\n",
       "\t2 & XIST-2252 & A3SS & 2252 & artery\\_coronary & 1 & Coronary artery & XIST & ENSG00000229807.11 & chrX  & -2.1880965 & 4.327079 & -10.329346 & 4.631685e-21 & 2.464057e-17 & 16.7535120\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 15\n",
       "\n",
       "| <!--/--> | GeneJunction &lt;chr&gt; | ASE &lt;chr&gt; | ASE_IDX &lt;int&gt; | Tissue &lt;chr&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | GeneSymbol &lt;chr&gt; | GeneID &lt;chr&gt; | chr &lt;chr&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | SCO1-8452 | A3SS | 8452 | adrenal_gland   | 1 | Adrenal gland   | SCO1 | ENSG00000133028.12 | chr17 |  0.7715834 | 5.028635 |   4.906848 | 1.533202e-06 | 8.170436e-03 |  0.8561371 |\n",
       "| 2 | XIST-2252 | A3SS | 2252 | artery_coronary | 1 | Coronary artery | XIST | ENSG00000229807.11 | chrX  | -2.1880965 | 4.327079 | -10.329346 | 4.631685e-21 | 2.464057e-17 | 16.7535120 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneJunction ASE  ASE_IDX Tissue          counts Display         GeneSymbol\n",
       "1 SCO1-8452    A3SS 8452    adrenal_gland   1      Adrenal gland   SCO1      \n",
       "2 XIST-2252    A3SS 2252    artery_coronary 1      Coronary artery XIST      \n",
       "  GeneID             chr   logFC      AveExpr  t          PValue      \n",
       "1 ENSG00000133028.12 chr17  0.7715834 5.028635   4.906848 1.533202e-06\n",
       "2 ENSG00000229807.11 chrX  -2.1880965 4.327079 -10.329346 4.631685e-21\n",
       "  AdjPVal      B         \n",
       "1 8.170436e-03  0.8561371\n",
       "2 2.464057e-17 16.7535120"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "results_dir         <- \"../data/\"\n",
    "significant_pattern <- \"_AS_model_B_sex_as_events_refined.csv\"\n",
    "significant_files   <- list.files(path = results_dir, pattern = significant_pattern)\n",
    "all_pattern         <- \"_AS_model_B_sex_as_events.csv\"\n",
    "all_files           <- list.files(path = results_dir, pattern = all_pattern)\n",
    "as_types            <- c(\"a3ss\", \"a5ss\", \"mxe\", \"ri\", \"se\")\n",
    "message(\"Length of all_files: \", length(all_files))\n",
    "message(\"Length of significant_files: \", length(significant_files))\n",
    "\n",
    "gene_as     <- create_as_structure (results_dir      <- results_dir, \n",
    "                                    files            <- significant_files,\n",
    "                                    all_or_das       <- \"differentially significant alternative splicing\",\n",
    "                                    pattern          <- significant_pattern, \n",
    "                                    tissue_reduction <- tissue_reduction)\n",
    "all_gene_as <- create_as_structure (results_dir      <- results_dir, \n",
    "                                    files            <- all_files, \n",
    "                                    all_or_das       <- \"all alternatively spliced\",\n",
    "                                    pattern          <- all_pattern, \n",
    "                                    tissue_reduction <- tissue_reduction)\n",
    "head(gene_as,2)\n",
    "gene_as$Tissue <- factor(gene_as$Tissue)\n",
    "write.table(gene_as, \"../data/gene_as.tsv\", quote=FALSE, sep=\"\\t\")\n",
    "write.table(all_gene_as, \"../data/all_gene_as.tsv\", quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c99fff42",
   "metadata": {
    "papermill": {
     "duration": 0.030677,
     "end_time": "2021-06-27T18:41:55.317715",
     "exception": false,
     "start_time": "2021-06-27T18:41:55.287038",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.8 Create a genes-id file capturing the unique gene-junction locations in a single file\n",
    "\n",
    "rMATS 3.2.5 unique junction ids by splicing event tied together with gene names and these identifiers useful for downstream analyses and investigations.\n",
    "only the SE significant AS events here\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "711aa975",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:41:55.384604Z",
     "iopub.status.busy": "2021-06-27T18:41:55.383262Z",
     "iopub.status.idle": "2021-06-27T18:41:55.752999Z",
     "shell.execute_reply": "2021-06-27T18:41:55.751621Z"
    },
    "papermill": {
     "duration": 0.404701,
     "end_time": "2021-06-27T18:41:55.753203",
     "exception": false,
     "start_time": "2021-06-27T18:41:55.348502",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The first file from ^se_*AS_model_B_sex_as_events_refined.csv: se_adipose_subcutaneous_AS_model_B_sex_as_events_refined.csv\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done writing 39 files.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "results_dir         <- \"../data/\"\n",
    "significant_pattern <- \"^se_*AS_model_B_sex_as_events_refined.csv\"\n",
    "files   <- list.files(path =results_dir, pattern = glob2rx(significant_pattern))\n",
    "message(\"The first file from ^se_*AS_model_B_sex_as_events_refined.csv: \", files[1])\n",
    "pattern=\"_AS_model_B_sex_as_events_refined.csv\"\n",
    "geneids <- data.frame()\n",
    "for (i in 1:length(files)) {\n",
    "    lines  <- read.table(file=paste0(results_dir, files[i]), \n",
    "                                     header = TRUE, sep = \",\", quote = \"\\\"'\", skipNul = FALSE)\n",
    "    \n",
    "    if (dim(lines)[1] > 0) {\n",
    "           event     <- as.vector(as.character(rownames(lines)))\n",
    "           tissue1   <- gsub(pattern,\"\", files[i], fixed = TRUE)\n",
    "           event_idx <- substring(event, regexpr(\"[0-9]+$\", event))\n",
    "           res       <- data.frame()\n",
    "           tissue2 <- substring(tissue1,4)\n",
    "           idx <- match(event_idx, se_annot$ID)\n",
    "           res <- data.frame(geneIDs      <- event,\n",
    "                             ID           <- event_idx,\n",
    "                             GeneSymbol   <- se_annot$geneSymbol[idx],\n",
    "                             GeneID       <- se_annot$GeneID[idx],\n",
    "                             chr          <- se_annot$chr[idx])\n",
    "           outfilename <- paste0(paste0(\"../data/se_\",tissue2),\"_geneids.tsv\")\n",
    "           write.table(res, outfilename, quote=FALSE, sep=\"\\t\")\n",
    "           \n",
    "     }\n",
    "}\n",
    "message(\"Done writing \", length(files), \" files.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfe3a9f1",
   "metadata": {
    "papermill": {
     "duration": 0.031108,
     "end_time": "2021-06-27T18:41:55.816515",
     "exception": false,
     "start_time": "2021-06-27T18:41:55.785407",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.9 Read in the differential gene expression results\n",
    "\n",
    "Here we create an aggregation of al the significant results differential gene expression events.\n",
    "Note the all_gene_dge.tsv may be found in the assets directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "79f6d12e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:41:55.884246Z",
     "iopub.status.busy": "2021-06-27T18:41:55.882928Z",
     "iopub.status.idle": "2021-06-27T18:41:56.279753Z",
     "shell.execute_reply": "2021-06-27T18:41:56.278688Z"
    },
    "papermill": {
     "duration": 0.432504,
     "end_time": "2021-06-27T18:41:56.279878",
     "exception": false,
     "start_time": "2021-06-27T18:41:55.847374",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "195"
      ],
      "text/latex": [
       "195"
      ],
      "text/markdown": [
       "195"
      ],
      "text/plain": [
       "[1] 195"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "For the run for differential gene expressionrun\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "We extracted a total of 4417 gene events (gene_dge)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "This includes 3221 total genes\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Tissue</th><th scope=col>ENSG_ver</th><th scope=col>ENSG_no_ver</th><th scope=col>GeneSymbol</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>adipose_subcutaneous</td><td>ENSG00000147050.14</td><td>ENSG00000147050</td><td>KDM6A     </td><td>272</td><td>Adipose (sc)</td><td>0.6119682</td><td> 5.030862</td><td>35.53259</td><td>9.295961e-156</td><td>1.498695e-151</td><td>344.27452</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>adipose_subcutaneous</td><td>ENSG00000224525.2 </td><td>ENSG00000224525</td><td>AL591686.1</td><td>272</td><td>Adipose (sc)</td><td>2.2810767</td><td>-1.464120</td><td>14.44699</td><td> 2.450714e-41</td><td> 3.591855e-38</td><td> 81.05687</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & Tissue & ENSG\\_ver & ENSG\\_no\\_ver & GeneSymbol & counts & Display & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & adipose\\_subcutaneous & ENSG00000147050.14 & ENSG00000147050 & KDM6A      & 272 & Adipose (sc) & 0.6119682 &  5.030862 & 35.53259 & 9.295961e-156 & 1.498695e-151 & 344.27452\\\\\n",
       "\t2 & adipose\\_subcutaneous & ENSG00000224525.2  & ENSG00000224525 & AL591686.1 & 272 & Adipose (sc) & 2.2810767 & -1.464120 & 14.44699 &  2.450714e-41 &  3.591855e-38 &  81.05687\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 12\n",
       "\n",
       "| <!--/--> | Tissue &lt;chr&gt; | ENSG_ver &lt;chr&gt; | ENSG_no_ver &lt;chr&gt; | GeneSymbol &lt;chr&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | adipose_subcutaneous | ENSG00000147050.14 | ENSG00000147050 | KDM6A      | 272 | Adipose (sc) | 0.6119682 |  5.030862 | 35.53259 | 9.295961e-156 | 1.498695e-151 | 344.27452 |\n",
       "| 2 | adipose_subcutaneous | ENSG00000224525.2  | ENSG00000224525 | AL591686.1 | 272 | Adipose (sc) | 2.2810767 | -1.464120 | 14.44699 |  2.450714e-41 |  3.591855e-38 |  81.05687 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue               ENSG_ver           ENSG_no_ver     GeneSymbol counts\n",
       "1 adipose_subcutaneous ENSG00000147050.14 ENSG00000147050 KDM6A      272   \n",
       "2 adipose_subcutaneous ENSG00000224525.2  ENSG00000224525 AL591686.1 272   \n",
       "  Display      logFC     AveExpr   t        PValue        AdjPVal      \n",
       "1 Adipose (sc) 0.6119682  5.030862 35.53259 9.295961e-156 1.498695e-151\n",
       "2 Adipose (sc) 2.2810767 -1.464120 14.44699  2.450714e-41  3.591855e-38\n",
       "  B        \n",
       "1 344.27452\n",
       "2  81.05687"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "results_dir             <- \"../data/\"\n",
    "significant_dge_pattern <- \"_DGE_refined.csv\"\n",
    "significant_dge_files   <- list.files(path = results_dir, pattern = significant_dge_pattern)\n",
    "map_pattern             <- \"_DGE_ensg_map.csv\"\n",
    "length(significant_files)\n",
    "\n",
    "gene_dge     <- create_dge_structure (results_dir      <- results_dir, \n",
    "                                      files            <- significant_dge_files, \n",
    "                                      all_or_dge       <- \"differential gene expression\",\n",
    "                                      pattern          <- significant_dge_pattern, \n",
    "                                      map_pattern      <- map_pattern,\n",
    "                                      tissue_reduction <- tissue_reduction)\n",
    "\n",
    "head(gene_dge,2)\n",
    "gene_dge$Tissue <- factor(gene_dge$Tissue)\n",
    "write.table(gene_dge,     \"../data/gene_dge.tsv\",     quote=FALSE, sep=\"\\t\") "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c9a4b55",
   "metadata": {
    "papermill": {
     "duration": 0.032928,
     "end_time": "2021-06-27T18:41:56.345117",
     "exception": false,
     "start_time": "2021-06-27T18:41:56.312189",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.10 Read in Gencode (v30) Complete Annotation file.\n",
    "\n",
    "Load in the gencode.v30.annotation.gtf file for additional annotation.\n",
    "chr information for summary data later, use the annotation we used for rMATS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "57c17038",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:41:56.414329Z",
     "iopub.status.busy": "2021-06-27T18:41:56.412997Z",
     "iopub.status.idle": "2021-06-27T18:42:48.760615Z",
     "shell.execute_reply": "2021-06-27T18:42:48.761015Z"
    },
    "papermill": {
     "duration": 52.383783,
     "end_time": "2021-06-27T18:42:48.761168",
     "exception": false,
     "start_time": "2021-06-27T18:41:56.377385",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "downloading gencode v30 annotation\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unzipping compressed file gencode.v30.annotation.gtf.gz..\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)\n",
       "attr(,\"status\")\n",
       "[1] 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done! gencode.v30.annotation.gtf can be found in ../data/\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>GeneSymbol</th><th scope=col>ENSG</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>DDX11L1    </td><td>ENSG00000223972.5</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>WASH7P     </td><td>ENSG00000227232.5</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>MIR6859-1  </td><td>ENSG00000278267.1</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>chr1</td><td>MIR1302-2HG</td><td>ENSG00000243485.5</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>chr1</td><td>MIR1302-2  </td><td>ENSG00000284332.1</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>chr1</td><td>FAM138A    </td><td>ENSG00000237613.2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & GeneSymbol & ENSG\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & DDX11L1     & ENSG00000223972.5\\\\\n",
       "\t13 & chr1 & WASH7P      & ENSG00000227232.5\\\\\n",
       "\t26 & chr1 & MIR6859-1   & ENSG00000278267.1\\\\\n",
       "\t29 & chr1 & MIR1302-2HG & ENSG00000243485.5\\\\\n",
       "\t37 & chr1 & MIR1302-2   & ENSG00000284332.1\\\\\n",
       "\t40 & chr1 & FAM138A     & ENSG00000237613.2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | chr &lt;fct&gt; | GeneSymbol &lt;chr&gt; | ENSG &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chr1 | DDX11L1     | ENSG00000223972.5 |\n",
       "| 13 | chr1 | WASH7P      | ENSG00000227232.5 |\n",
       "| 26 | chr1 | MIR6859-1   | ENSG00000278267.1 |\n",
       "| 29 | chr1 | MIR1302-2HG | ENSG00000243485.5 |\n",
       "| 37 | chr1 | MIR1302-2   | ENSG00000284332.1 |\n",
       "| 40 | chr1 | FAM138A     | ENSG00000237613.2 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  GeneSymbol  ENSG             \n",
       "1  chr1 DDX11L1     ENSG00000223972.5\n",
       "13 chr1 WASH7P      ENSG00000227232.5\n",
       "26 chr1 MIR6859-1   ENSG00000278267.1\n",
       "29 chr1 MIR1302-2HG ENSG00000243485.5\n",
       "37 chr1 MIR1302-2   ENSG00000284332.1\n",
       "40 chr1 FAM138A     ENSG00000237613.2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "message(\"downloading gencode v30 annotation\\n\")\n",
    "system(\"wget -O ../data/gencode.v30.annotation.gtf.gz ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gtf.gz\")\n",
    "message(\"Done!\\n\")\n",
    "message(\"Unzipping compressed file gencode.v30.annotation.gtf.gz..\")\n",
    "system(\"gunzip ../data/gencode.v30.annotation.gtf.gz\", intern = TRUE)\n",
    "message(\"Done! gencode.v30.annotation.gtf can be found in ../data/\")\n",
    "\n",
    "gencode <- import(\"../data/gencode.v30.annotation.gtf\")\n",
    "gtf.df <- as.data.frame (gencode)\n",
    "chr_genes <- unique(gtf.df[,c(\"seqnames\",\"gene_name\",\"gene_id\")])\n",
    "colnames(chr_genes) <- c(\"chr\",\"GeneSymbol\", \"ENSG\")\n",
    "head(chr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "395a7639",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:42:48.834694Z",
     "iopub.status.busy": "2021-06-27T18:42:48.833365Z",
     "iopub.status.idle": "2021-06-27T18:42:58.402312Z",
     "shell.execute_reply": "2021-06-27T18:42:58.401017Z"
    },
    "papermill": {
     "duration": 9.606843,
     "end_time": "2021-06-27T18:42:58.402472",
     "exception": false,
     "start_time": "2021-06-27T18:42:48.795629",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>GeneSymbol</th><th scope=col>ENSG</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>DDX11L1    </td><td>ENSG00000223972</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1</td><td>WASH7P     </td><td>ENSG00000227232</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>chr1</td><td>MIR6859-1  </td><td>ENSG00000278267</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>chr1</td><td>MIR1302-2HG</td><td>ENSG00000243485</td></tr>\n",
       "\t<tr><th scope=row>37</th><td>chr1</td><td>MIR1302-2  </td><td>ENSG00000284332</td></tr>\n",
       "\t<tr><th scope=row>40</th><td>chr1</td><td>FAM138A    </td><td>ENSG00000237613</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & chr & GeneSymbol & ENSG\\\\\n",
       "  & <fct> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & DDX11L1     & ENSG00000223972\\\\\n",
       "\t13 & chr1 & WASH7P      & ENSG00000227232\\\\\n",
       "\t26 & chr1 & MIR6859-1   & ENSG00000278267\\\\\n",
       "\t29 & chr1 & MIR1302-2HG & ENSG00000243485\\\\\n",
       "\t37 & chr1 & MIR1302-2   & ENSG00000284332\\\\\n",
       "\t40 & chr1 & FAM138A     & ENSG00000237613\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | chr &lt;fct&gt; | GeneSymbol &lt;chr&gt; | ENSG &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | chr1 | DDX11L1     | ENSG00000223972 |\n",
       "| 13 | chr1 | WASH7P      | ENSG00000227232 |\n",
       "| 26 | chr1 | MIR6859-1   | ENSG00000278267 |\n",
       "| 29 | chr1 | MIR1302-2HG | ENSG00000243485 |\n",
       "| 37 | chr1 | MIR1302-2   | ENSG00000284332 |\n",
       "| 40 | chr1 | FAM138A     | ENSG00000237613 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr  GeneSymbol  ENSG           \n",
       "1  chr1 DDX11L1     ENSG00000223972\n",
       "13 chr1 WASH7P      ENSG00000227232\n",
       "26 chr1 MIR6859-1   ENSG00000278267\n",
       "29 chr1 MIR1302-2HG ENSG00000243485\n",
       "37 chr1 MIR1302-2   ENSG00000284332\n",
       "40 chr1 FAM138A     ENSG00000237613"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for (i in 1:dim(chr_genes)[1]) {\n",
    "    chr_genes$ENSG[i] <- as.character(strsplit(chr_genes$ENSG[i],'\\\\.\\\\w+$'))\n",
    "}\n",
    "head(chr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a3a0cff4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:42:58.477463Z",
     "iopub.status.busy": "2021-06-27T18:42:58.476170Z",
     "iopub.status.idle": "2021-06-27T18:43:07.853577Z",
     "shell.execute_reply": "2021-06-27T18:43:07.852570Z"
    },
    "papermill": {
     "duration": 9.416549,
     "end_time": "2021-06-27T18:43:07.853702",
     "exception": false,
     "start_time": "2021-06-27T18:42:58.437153",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Tissue</th><th scope=col>ENSG_ver</th><th scope=col>ENSG_no_ver</th><th scope=col>GeneSymbol</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th><th scope=col>chr</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>adipose_subcutaneous</td><td>ENSG00000147050.14</td><td>ENSG00000147050</td><td>KDM6A     </td><td>272</td><td>Adipose (sc)</td><td> 0.6119682</td><td> 5.0308617</td><td> 35.53259</td><td>9.295961e-156</td><td>1.498695e-151</td><td>344.27452</td><td>chrX </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>adipose_subcutaneous</td><td>ENSG00000224525.2 </td><td>ENSG00000224525</td><td>AL591686.1</td><td>272</td><td>Adipose (sc)</td><td> 2.2810767</td><td>-1.4641204</td><td> 14.44699</td><td> 2.450714e-41</td><td> 3.591855e-38</td><td> 81.05687</td><td>chr1 </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>adipose_subcutaneous</td><td>ENSG00000115041.12</td><td>ENSG00000115041</td><td>KCNIP3    </td><td>272</td><td>Adipose (sc)</td><td>-0.9052185</td><td> 3.4532379</td><td>-13.70790</td><td> 7.695915e-38</td><td> 8.862395e-35</td><td> 75.10723</td><td>chr2 </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>adipose_subcutaneous</td><td>ENSG00000258484.3 </td><td>ENSG00000258484</td><td>SPESP1    </td><td>272</td><td>Adipose (sc)</td><td>-0.8918130</td><td> 0.9410518</td><td>-13.10456</td><td> 4.615796e-35</td><td> 4.650991e-32</td><td> 68.23064</td><td>chr15</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>adipose_subcutaneous</td><td>ENSG00000134339.8 </td><td>ENSG00000134339</td><td>SAA2      </td><td>272</td><td>Adipose (sc)</td><td> 3.1082279</td><td> 2.7856727</td><td> 12.87586</td><td> 5.017765e-34</td><td> 4.257705e-31</td><td> 66.45132</td><td>chr11</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>adipose_subcutaneous</td><td>ENSG00000103449.11</td><td>ENSG00000103449</td><td>SALL1     </td><td>272</td><td>Adipose (sc)</td><td> 1.2692252</td><td>-0.2424580</td><td> 12.72143</td><td> 2.470418e-33</td><td> 1.991404e-30</td><td> 64.14054</td><td>chr16</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & Tissue & ENSG\\_ver & ENSG\\_no\\_ver & GeneSymbol & counts & Display & logFC & AveExpr & t & PValue & AdjPVal & B & chr\\\\\n",
       "  & <fct> & <chr> & <chr> & <chr> & <int> & <fct> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & adipose\\_subcutaneous & ENSG00000147050.14 & ENSG00000147050 & KDM6A      & 272 & Adipose (sc) &  0.6119682 &  5.0308617 &  35.53259 & 9.295961e-156 & 1.498695e-151 & 344.27452 & chrX \\\\\n",
       "\t2 & adipose\\_subcutaneous & ENSG00000224525.2  & ENSG00000224525 & AL591686.1 & 272 & Adipose (sc) &  2.2810767 & -1.4641204 &  14.44699 &  2.450714e-41 &  3.591855e-38 &  81.05687 & chr1 \\\\\n",
       "\t3 & adipose\\_subcutaneous & ENSG00000115041.12 & ENSG00000115041 & KCNIP3     & 272 & Adipose (sc) & -0.9052185 &  3.4532379 & -13.70790 &  7.695915e-38 &  8.862395e-35 &  75.10723 & chr2 \\\\\n",
       "\t4 & adipose\\_subcutaneous & ENSG00000258484.3  & ENSG00000258484 & SPESP1     & 272 & Adipose (sc) & -0.8918130 &  0.9410518 & -13.10456 &  4.615796e-35 &  4.650991e-32 &  68.23064 & chr15\\\\\n",
       "\t5 & adipose\\_subcutaneous & ENSG00000134339.8  & ENSG00000134339 & SAA2       & 272 & Adipose (sc) &  3.1082279 &  2.7856727 &  12.87586 &  5.017765e-34 &  4.257705e-31 &  66.45132 & chr11\\\\\n",
       "\t6 & adipose\\_subcutaneous & ENSG00000103449.11 & ENSG00000103449 & SALL1      & 272 & Adipose (sc) &  1.2692252 & -0.2424580 &  12.72143 &  2.470418e-33 &  1.991404e-30 &  64.14054 & chr16\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 13\n",
       "\n",
       "| <!--/--> | Tissue &lt;fct&gt; | ENSG_ver &lt;chr&gt; | ENSG_no_ver &lt;chr&gt; | GeneSymbol &lt;chr&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; | chr &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | adipose_subcutaneous | ENSG00000147050.14 | ENSG00000147050 | KDM6A      | 272 | Adipose (sc) |  0.6119682 |  5.0308617 |  35.53259 | 9.295961e-156 | 1.498695e-151 | 344.27452 | chrX  |\n",
       "| 2 | adipose_subcutaneous | ENSG00000224525.2  | ENSG00000224525 | AL591686.1 | 272 | Adipose (sc) |  2.2810767 | -1.4641204 |  14.44699 |  2.450714e-41 |  3.591855e-38 |  81.05687 | chr1  |\n",
       "| 3 | adipose_subcutaneous | ENSG00000115041.12 | ENSG00000115041 | KCNIP3     | 272 | Adipose (sc) | -0.9052185 |  3.4532379 | -13.70790 |  7.695915e-38 |  8.862395e-35 |  75.10723 | chr2  |\n",
       "| 4 | adipose_subcutaneous | ENSG00000258484.3  | ENSG00000258484 | SPESP1     | 272 | Adipose (sc) | -0.8918130 |  0.9410518 | -13.10456 |  4.615796e-35 |  4.650991e-32 |  68.23064 | chr15 |\n",
       "| 5 | adipose_subcutaneous | ENSG00000134339.8  | ENSG00000134339 | SAA2       | 272 | Adipose (sc) |  3.1082279 |  2.7856727 |  12.87586 |  5.017765e-34 |  4.257705e-31 |  66.45132 | chr11 |\n",
       "| 6 | adipose_subcutaneous | ENSG00000103449.11 | ENSG00000103449 | SALL1      | 272 | Adipose (sc) |  1.2692252 | -0.2424580 |  12.72143 |  2.470418e-33 |  1.991404e-30 |  64.14054 | chr16 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue               ENSG_ver           ENSG_no_ver     GeneSymbol counts\n",
       "1 adipose_subcutaneous ENSG00000147050.14 ENSG00000147050 KDM6A      272   \n",
       "2 adipose_subcutaneous ENSG00000224525.2  ENSG00000224525 AL591686.1 272   \n",
       "3 adipose_subcutaneous ENSG00000115041.12 ENSG00000115041 KCNIP3     272   \n",
       "4 adipose_subcutaneous ENSG00000258484.3  ENSG00000258484 SPESP1     272   \n",
       "5 adipose_subcutaneous ENSG00000134339.8  ENSG00000134339 SAA2       272   \n",
       "6 adipose_subcutaneous ENSG00000103449.11 ENSG00000103449 SALL1      272   \n",
       "  Display      logFC      AveExpr    t         PValue        AdjPVal      \n",
       "1 Adipose (sc)  0.6119682  5.0308617  35.53259 9.295961e-156 1.498695e-151\n",
       "2 Adipose (sc)  2.2810767 -1.4641204  14.44699  2.450714e-41  3.591855e-38\n",
       "3 Adipose (sc) -0.9052185  3.4532379 -13.70790  7.695915e-38  8.862395e-35\n",
       "4 Adipose (sc) -0.8918130  0.9410518 -13.10456  4.615796e-35  4.650991e-32\n",
       "5 Adipose (sc)  3.1082279  2.7856727  12.87586  5.017765e-34  4.257705e-31\n",
       "6 Adipose (sc)  1.2692252 -0.2424580  12.72143  2.470418e-33  1.991404e-30\n",
       "  B         chr  \n",
       "1 344.27452 chrX \n",
       "2  81.05687 chr1 \n",
       "3  75.10723 chr2 \n",
       "4  68.23064 chr15\n",
       "5  66.45132 chr11\n",
       "6  64.14054 chr16"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "i = 1\n",
    "chr <- rep(\"NA\",dim(gene_dge)[1])\n",
    "gene_dge$chr <- chr\n",
    "for (i in 1:dim(gene_dge)[1]) {\n",
    "    match  <- as.character(chr_genes$ENSG) %in% as.character((gene_dge$ENSG_no_ver[i]))\n",
    "    if (sum(match==TRUE)== 1) {\n",
    "        chr[i] <- as.character(chr_genes[match,]$chr)\n",
    "        gene_dge$chr[i] <- chr[i]\n",
    "    } else if (sum(match==TRUE)>1) {\n",
    "        all <- as.vector(as.character(chr_genes[match,]$chr))\n",
    "        gene_dge$chr[i] <- as.character(all[1])\n",
    "    }\n",
    "}\n",
    "head(gene_dge)\n",
    "\n",
    "write.table(gene_dge, \"../data/gene_dge.tsv\", quote=FALSE, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86897f9d",
   "metadata": {
    "papermill": {
     "duration": 0.03521,
     "end_time": "2021-06-27T18:43:07.925261",
     "exception": false,
     "start_time": "2021-06-27T18:43:07.890051",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 2.11 Summary AS \n",
    "\n",
    "Capture descriptive statistics of the signficiantly alternatively splicedgene_as and significantly differentially gene expressed, gene_dge regarding events by tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "06365572",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:08.000594Z",
     "iopub.status.busy": "2021-06-27T18:43:07.998953Z",
     "iopub.status.idle": "2021-06-27T18:43:08.072165Z",
     "shell.execute_reply": "2021-06-27T18:43:08.071042Z"
    },
    "papermill": {
     "duration": 0.112107,
     "end_time": "2021-06-27T18:43:08.072376",
     "exception": false,
     "start_time": "2021-06-27T18:43:07.960269",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Tissue</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>esophagus_mucosa</td><td>1</td></tr>\n",
       "\t<tr><td>liver           </td><td>1</td></tr>\n",
       "\t<tr><td>lung            </td><td>1</td></tr>\n",
       "\t<tr><td>whole_blood     </td><td>1</td></tr>\n",
       "\t<tr><td>adrenal_gland   </td><td>2</td></tr>\n",
       "\t<tr><td>artery_coronary </td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Tissue & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t esophagus\\_mucosa & 1\\\\\n",
       "\t liver            & 1\\\\\n",
       "\t lung             & 1\\\\\n",
       "\t whole\\_blood      & 1\\\\\n",
       "\t adrenal\\_gland    & 2\\\\\n",
       "\t artery\\_coronary  & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Tissue &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| esophagus_mucosa | 1 |\n",
       "| liver            | 1 |\n",
       "| lung             | 1 |\n",
       "| whole_blood      | 1 |\n",
       "| adrenal_gland    | 2 |\n",
       "| artery_coronary  | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue           n\n",
       "1 esophagus_mucosa 1\n",
       "2 liver            1\n",
       "3 lung             1\n",
       "4 whole_blood      1\n",
       "5 adrenal_gland    2\n",
       "6 artery_coronary  2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum splicing events per tissue 1 maximum splicing events per tissue 521\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Sum of significant splicing events per tissue less than 100 35\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Tissue</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>brain_spinal_cord_cervical_c_1       </td><td> 5</td></tr>\n",
       "\t<tr><td>brain_cortex                         </td><td> 7</td></tr>\n",
       "\t<tr><td>brain_hippocampus                    </td><td> 7</td></tr>\n",
       "\t<tr><td>brain_nucleus_accumbens_basal_ganglia</td><td> 8</td></tr>\n",
       "\t<tr><td>small_intestine_terminal_ileum       </td><td>10</td></tr>\n",
       "\t<tr><td>brain_caudate_basal_ganglia          </td><td>12</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Tissue & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t brain\\_spinal\\_cord\\_cervical\\_c\\_1        &  5\\\\\n",
       "\t brain\\_cortex                          &  7\\\\\n",
       "\t brain\\_hippocampus                     &  7\\\\\n",
       "\t brain\\_nucleus\\_accumbens\\_basal\\_ganglia &  8\\\\\n",
       "\t small\\_intestine\\_terminal\\_ileum        & 10\\\\\n",
       "\t brain\\_caudate\\_basal\\_ganglia           & 12\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Tissue &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| brain_spinal_cord_cervical_c_1        |  5 |\n",
       "| brain_cortex                          |  7 |\n",
       "| brain_hippocampus                     |  7 |\n",
       "| brain_nucleus_accumbens_basal_ganglia |  8 |\n",
       "| small_intestine_terminal_ileum        | 10 |\n",
       "| brain_caudate_basal_ganglia           | 12 |\n",
       "\n"
      ],
      "text/plain": [
       "  Tissue                                n \n",
       "1 brain_spinal_cord_cervical_c_1         5\n",
       "2 brain_cortex                           7\n",
       "3 brain_hippocampus                      7\n",
       "4 brain_nucleus_accumbens_basal_ganglia  8\n",
       "5 small_intestine_terminal_ileum        10\n",
       "6 brain_caudate_basal_ganglia           12"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Minimum gene expression events per tissue 5 maximum gene expression events per tissue 2296\n",
      "\n"
     ]
    }
   ],
   "source": [
    "XY <- gene_as %>% group_by(Tissue) %>% tally()\n",
    "XY <- XY[order(XY$n),decreasing=TRUE]\n",
    "head(XY)\n",
    "message(\"Minimum splicing events per tissue \", min(XY$n), \" maximum splicing events per tissue \", max(XY$n))\n",
    "message(\"Sum of significant splicing events per tissue less than 100 \", sum(XY$n<100))\n",
    "\n",
    "XY <- gene_dge %>% group_by(Tissue) %>% tally()\n",
    "XY <- XY[order(XY$n),decreasing=TRUE]\n",
    "head(XY)\n",
    "message(\"Minimum gene expression events per tissue \", min(XY$n), \" maximum gene expression events per tissue \", max(XY$n))\n",
    "# table(gtf.df[,c(\"gene_type\")])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4fd6ae8",
   "metadata": {
    "papermill": {
     "duration": 0.036677,
     "end_time": "2021-06-27T18:43:08.146321",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.109644",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3 Data Structures for Figures"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7fc21bfc",
   "metadata": {
    "papermill": {
     "duration": 0.036556,
     "end_time": "2021-06-27T18:43:08.219431",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.182875",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.1 gene_as.tsv\n",
    "\n",
    "This file contains (description)\n",
    "Here is a typical line\n",
    "<pre>\n",
    "A data.frame: 6 × 15\n",
    "GeneJunction\tASE\tASE_IDX\tTissue\tcounts\tDisplay\tGeneSymbol\tGeneID\tchr\tlogFC\tAveExpr\tt\tPValue\tAdjPVal\tB\n",
    "<fct>\t<fct>\t<int>\t<fct>\t<int>\t<fct>\t<fct>\t<fct>\t<fct>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\t<dbl>\n",
    "1\tXIST-2253\tA3SS\t2253\tadipose_subcutaneous\t4\tAdipose (sc)\tXIST\tENSG00000229807.11\tchrX\t-4.4086049\t3.196317\t-36.488970\t4.635568e-154\t3.893877e-150\t310.016049\n",
    "2\tXIST-2252\tA3SS\t2252\tadipose_subcutaneous\t4\tAdipose (sc)\tXIST\tENSG00000229807.11\tchrX\t-2.4147126\t3.647690\t-21.921057\t1.444102e-78\t6.065229e-75\t160.028167\n",
    "3\tGREB1L-4933\tA3SS\t4933\tadipose_subcutaneous\t4\tAdipose (sc)\tGREB1L\tENSG00000141449.14\tchr18\t1.2793173\t2.115005\t7.123138\t3.052112e-12\t8.545914e-09\t16.692429\n",
    "4\tRHCG-1776\tA3SS\t1776\tadipose_subcutaneous\t4\tAdipose (sc)\tRHCG\tENSG00000140519.14\tchr15\t-0.6930009\t1.636472\t-3.922124\t9.797866e-05\t3.919146e-02\t1.142232\n",
    "5\tXIST-2253\tA3SS\t2253\tadipose_visceral_omentum\t12\tAdipose (v)\tXIST\tENSG00000229807.11\tchrX\t-4.4403352\t3.113532\t-33.950800\t2.654474e-123\t2.209585e-119\t241.826117\n",
    "6\tXIST-2252\tA3SS\t2252\tadipose_visceral_omentum\t12\tAdipose (v)\tXIST\tENSG00000229807.11\tchrX\t-2.4506832\t3.650617\t-18.890779\t2.817671e-58\t1.172715e-54\t114.731682\n",
    "</pre>\n",
    "There are 2887 significant events in the file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fda0078f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:08.298765Z",
     "iopub.status.busy": "2021-06-27T18:43:08.297483Z",
     "iopub.status.idle": "2021-06-27T18:43:08.397066Z",
     "shell.execute_reply": "2021-06-27T18:43:08.395921Z"
    },
    "papermill": {
     "duration": 0.140844,
     "end_time": "2021-06-27T18:43:08.397186",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.256342",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 743\n",
      "Columns: 15\n",
      "$ GeneJunction \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"SCO1-8452\", \"XIST-2252\", \"CPT1C-6427\", \"DEPDC5-1839\", \"C…\n",
      "$ ASE          \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"A3SS\", \"…\n",
      "$ ASE_IDX      \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 8452, 2252, 6427, 1839, 8117, 453, 3523, 7852, 7853, 1653…\n",
      "$ Tissue       \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m adrenal_gland, artery_coronary, brain_cortex, brain_hypot…\n",
      "$ counts       \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 1, 1, 1, 2, 2, 2, 2, 104, 104, 104, 104, 104, 104, 104, 1…\n",
      "$ Display      \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m Adrenal gland, Coronary artery, Cortex, Hypothalamus, Hyp…\n",
      "$ GeneSymbol   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"SCO1\", \"XIST\", \"CPT1C\", \"DEPDC5\", \"CLN5\", \"ANKHD1-EIF4EB…\n",
      "$ GeneID       \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"ENSG00000133028.12\", \"ENSG00000229807.11\", \"ENSG00000169…\n",
      "$ chr          \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \"chr17\", \"chrX\", \"chr19\", \"chr22\", \"chr13\", \"chr5\", \"chr2…\n",
      "$ logFC        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 0.7715834, -2.1880965, -0.8246151, -1.2651387, -1.2407543…\n",
      "$ AveExpr      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 5.028635, 4.327079, 6.908922, 3.521844, 4.678101, 6.45935…\n",
      "$ t            \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 4.906848, -10.329346, -5.042202, -5.035619, -4.930630, 4.…\n",
      "$ PValue       \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 1.533202e-06, 4.631685e-21, 9.757441e-07, 1.099810e-06, 1…\n",
      "$ AdjPVal      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 8.170436e-03, 2.464057e-17, 5.117778e-03, 4.663317e-03, 4…\n",
      "$ B            \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 0.8561371, 16.7535120, 5.1676834, 4.2503727, 4.1488842, 2…\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "36"
      ],
      "text/latex": [
       "36"
      ],
      "text/markdown": [
       "36"
      ],
      "text/plain": [
       "[1] 36"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE \n",
       "  743 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "         Adipose (sc)           Adipose (v)         Adrenal gland \n",
       "                    3                     3                     2 \n",
       "                Aorta      Atrial appendage                Breast \n",
       "                   10                     4                   521 \n",
       "              Caudate Cerebellar hemisphere            Cerebellum \n",
       "                    2                     2                    34 \n",
       "      Coronary artery                Cortex       EBV-lymphocytes \n",
       "                    2                     2                     3 \n",
       "      Esophagus (gej)         Esophagus (m)        Esophagus (mu) \n",
       "                    6                     1                    32 \n",
       "          Fibroblasts        Frontal cortex           Hippocampus \n",
       "                    3                     2                    14 \n",
       "         Hypothalamus        Left ventricle                 Liver \n",
       "                    3                     0                     1 \n",
       "                 Lung     Nucleus accumbens              Pancreas \n",
       "                    1                     3                     2 \n",
       "            Pituitary               Putamen         Sigmoid colon \n",
       "                    5                     2                    14 \n",
       "      Skeletal muscle        Skin (exposed)    Skin (not exposed) \n",
       "                    7                     6                     0 \n",
       "      Small intestine           Spinal cord                Spleen \n",
       "                    0                     3                    28 \n",
       "              Stomach               Thyroid         Tibial artery \n",
       "                    2                     5                     7 \n",
       "         Tibial nerve      Transverse colon           Whole blood \n",
       "                    3                     4                     1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneJunction'</li><li>'ASE'</li><li>'ASE_IDX'</li><li>'Tissue'</li><li>'counts'</li><li>'Display'</li><li>'GeneSymbol'</li><li>'GeneID'</li><li>'chr'</li><li>'logFC'</li><li>'AveExpr'</li><li>'t'</li><li>'PValue'</li><li>'AdjPVal'</li><li>'B'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneJunction'\n",
       "\\item 'ASE'\n",
       "\\item 'ASE\\_IDX'\n",
       "\\item 'Tissue'\n",
       "\\item 'counts'\n",
       "\\item 'Display'\n",
       "\\item 'GeneSymbol'\n",
       "\\item 'GeneID'\n",
       "\\item 'chr'\n",
       "\\item 'logFC'\n",
       "\\item 'AveExpr'\n",
       "\\item 't'\n",
       "\\item 'PValue'\n",
       "\\item 'AdjPVal'\n",
       "\\item 'B'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneJunction'\n",
       "2. 'ASE'\n",
       "3. 'ASE_IDX'\n",
       "4. 'Tissue'\n",
       "5. 'counts'\n",
       "6. 'Display'\n",
       "7. 'GeneSymbol'\n",
       "8. 'GeneID'\n",
       "9. 'chr'\n",
       "10. 'logFC'\n",
       "11. 'AveExpr'\n",
       "12. 't'\n",
       "13. 'PValue'\n",
       "14. 'AdjPVal'\n",
       "15. 'B'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GeneJunction\" \"ASE\"          \"ASE_IDX\"      \"Tissue\"       \"counts\"      \n",
       " [6] \"Display\"      \"GeneSymbol\"   \"GeneID\"       \"chr\"          \"logFC\"       \n",
       "[11] \"AveExpr\"      \"t\"            \"PValue\"       \"AdjPVal\"      \"B\"           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneJunction</th><th scope=col>ASE</th><th scope=col>ASE_IDX</th><th scope=col>Tissue</th><th scope=col>counts</th><th scope=col>Display</th><th scope=col>GeneSymbol</th><th scope=col>GeneID</th><th scope=col>chr</th><th scope=col>logFC</th><th scope=col>AveExpr</th><th scope=col>t</th><th scope=col>PValue</th><th scope=col>AdjPVal</th><th scope=col>B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SCO1-8452          </td><td>A3SS</td><td>8452</td><td>adrenal_gland                 </td><td>1</td><td>Adrenal gland  </td><td>SCO1           </td><td>ENSG00000133028.12</td><td>chr17</td><td> 0.7715834</td><td>5.028635</td><td>  4.906848</td><td>1.533202e-06</td><td>8.170436e-03</td><td> 0.8561371</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>XIST-2252          </td><td>A3SS</td><td>2252</td><td>artery_coronary               </td><td>1</td><td>Coronary artery</td><td>XIST           </td><td>ENSG00000229807.11</td><td>chrX </td><td>-2.1880965</td><td>4.327079</td><td>-10.329346</td><td>4.631685e-21</td><td>2.464057e-17</td><td>16.7535120</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>CPT1C-6427         </td><td>A3SS</td><td>6427</td><td>brain_cortex                  </td><td>1</td><td>Cortex         </td><td>CPT1C          </td><td>ENSG00000169169.14</td><td>chr19</td><td>-0.8246151</td><td>6.908922</td><td> -5.042202</td><td>9.757441e-07</td><td>5.117778e-03</td><td> 5.1676834</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>DEPDC5-1839        </td><td>A3SS</td><td>1839</td><td>brain_hypothalamus            </td><td>2</td><td>Hypothalamus   </td><td>DEPDC5         </td><td>ENSG00000100150.18</td><td>chr22</td><td>-1.2651387</td><td>3.521844</td><td> -5.035619</td><td>1.099810e-06</td><td>4.663317e-03</td><td> 4.2503727</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>CLN5-8117          </td><td>A3SS</td><td>8117</td><td>brain_hypothalamus            </td><td>2</td><td>Hypothalamus   </td><td>CLN5           </td><td>ENSG00000102805.15</td><td>chr13</td><td>-1.2407543</td><td>4.678101</td><td> -4.930630</td><td>1.777517e-06</td><td>4.663317e-03</td><td> 4.1488842</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ANKHD1-EIF4EBP3-453</td><td>A3SS</td><td> 453</td><td>brain_spinal_cord_cervical_c_1</td><td>2</td><td>Spinal cord    </td><td>ANKHD1-EIF4EBP3</td><td>ENSG00000254996.5 </td><td>chr5 </td><td> 0.6105820</td><td>6.459352</td><td>  4.440250</td><td>1.825438e-05</td><td>4.818212e-02</td><td> 2.5615289</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & GeneJunction & ASE & ASE\\_IDX & Tissue & counts & Display & GeneSymbol & GeneID & chr & logFC & AveExpr & t & PValue & AdjPVal & B\\\\\n",
       "  & <chr> & <chr> & <int> & <fct> & <int> & <fct> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & SCO1-8452           & A3SS & 8452 & adrenal\\_gland                  & 1 & Adrenal gland   & SCO1            & ENSG00000133028.12 & chr17 &  0.7715834 & 5.028635 &   4.906848 & 1.533202e-06 & 8.170436e-03 &  0.8561371\\\\\n",
       "\t2 & XIST-2252           & A3SS & 2252 & artery\\_coronary                & 1 & Coronary artery & XIST            & ENSG00000229807.11 & chrX  & -2.1880965 & 4.327079 & -10.329346 & 4.631685e-21 & 2.464057e-17 & 16.7535120\\\\\n",
       "\t3 & CPT1C-6427          & A3SS & 6427 & brain\\_cortex                   & 1 & Cortex          & CPT1C           & ENSG00000169169.14 & chr19 & -0.8246151 & 6.908922 &  -5.042202 & 9.757441e-07 & 5.117778e-03 &  5.1676834\\\\\n",
       "\t4 & DEPDC5-1839         & A3SS & 1839 & brain\\_hypothalamus             & 2 & Hypothalamus    & DEPDC5          & ENSG00000100150.18 & chr22 & -1.2651387 & 3.521844 &  -5.035619 & 1.099810e-06 & 4.663317e-03 &  4.2503727\\\\\n",
       "\t5 & CLN5-8117           & A3SS & 8117 & brain\\_hypothalamus             & 2 & Hypothalamus    & CLN5            & ENSG00000102805.15 & chr13 & -1.2407543 & 4.678101 &  -4.930630 & 1.777517e-06 & 4.663317e-03 &  4.1488842\\\\\n",
       "\t6 & ANKHD1-EIF4EBP3-453 & A3SS &  453 & brain\\_spinal\\_cord\\_cervical\\_c\\_1 & 2 & Spinal cord     & ANKHD1-EIF4EBP3 & ENSG00000254996.5  & chr5  &  0.6105820 & 6.459352 &   4.440250 & 1.825438e-05 & 4.818212e-02 &  2.5615289\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 15\n",
       "\n",
       "| <!--/--> | GeneJunction &lt;chr&gt; | ASE &lt;chr&gt; | ASE_IDX &lt;int&gt; | Tissue &lt;fct&gt; | counts &lt;int&gt; | Display &lt;fct&gt; | GeneSymbol &lt;chr&gt; | GeneID &lt;chr&gt; | chr &lt;chr&gt; | logFC &lt;dbl&gt; | AveExpr &lt;dbl&gt; | t &lt;dbl&gt; | PValue &lt;dbl&gt; | AdjPVal &lt;dbl&gt; | B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | SCO1-8452           | A3SS | 8452 | adrenal_gland                  | 1 | Adrenal gland   | SCO1            | ENSG00000133028.12 | chr17 |  0.7715834 | 5.028635 |   4.906848 | 1.533202e-06 | 8.170436e-03 |  0.8561371 |\n",
       "| 2 | XIST-2252           | A3SS | 2252 | artery_coronary                | 1 | Coronary artery | XIST            | ENSG00000229807.11 | chrX  | -2.1880965 | 4.327079 | -10.329346 | 4.631685e-21 | 2.464057e-17 | 16.7535120 |\n",
       "| 3 | CPT1C-6427          | A3SS | 6427 | brain_cortex                   | 1 | Cortex          | CPT1C           | ENSG00000169169.14 | chr19 | -0.8246151 | 6.908922 |  -5.042202 | 9.757441e-07 | 5.117778e-03 |  5.1676834 |\n",
       "| 4 | DEPDC5-1839         | A3SS | 1839 | brain_hypothalamus             | 2 | Hypothalamus    | DEPDC5          | ENSG00000100150.18 | chr22 | -1.2651387 | 3.521844 |  -5.035619 | 1.099810e-06 | 4.663317e-03 |  4.2503727 |\n",
       "| 5 | CLN5-8117           | A3SS | 8117 | brain_hypothalamus             | 2 | Hypothalamus    | CLN5            | ENSG00000102805.15 | chr13 | -1.2407543 | 4.678101 |  -4.930630 | 1.777517e-06 | 4.663317e-03 |  4.1488842 |\n",
       "| 6 | ANKHD1-EIF4EBP3-453 | A3SS |  453 | brain_spinal_cord_cervical_c_1 | 2 | Spinal cord     | ANKHD1-EIF4EBP3 | ENSG00000254996.5  | chr5  |  0.6105820 | 6.459352 |   4.440250 | 1.825438e-05 | 4.818212e-02 |  2.5615289 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneJunction        ASE  ASE_IDX Tissue                         counts\n",
       "1 SCO1-8452           A3SS 8452    adrenal_gland                  1     \n",
       "2 XIST-2252           A3SS 2252    artery_coronary                1     \n",
       "3 CPT1C-6427          A3SS 6427    brain_cortex                   1     \n",
       "4 DEPDC5-1839         A3SS 1839    brain_hypothalamus             2     \n",
       "5 CLN5-8117           A3SS 8117    brain_hypothalamus             2     \n",
       "6 ANKHD1-EIF4EBP3-453 A3SS  453    brain_spinal_cord_cervical_c_1 2     \n",
       "  Display         GeneSymbol      GeneID             chr   logFC      AveExpr \n",
       "1 Adrenal gland   SCO1            ENSG00000133028.12 chr17  0.7715834 5.028635\n",
       "2 Coronary artery XIST            ENSG00000229807.11 chrX  -2.1880965 4.327079\n",
       "3 Cortex          CPT1C           ENSG00000169169.14 chr19 -0.8246151 6.908922\n",
       "4 Hypothalamus    DEPDC5          ENSG00000100150.18 chr22 -1.2651387 3.521844\n",
       "5 Hypothalamus    CLN5            ENSG00000102805.15 chr13 -1.2407543 4.678101\n",
       "6 Spinal cord     ANKHD1-EIF4EBP3 ENSG00000254996.5  chr5   0.6105820 6.459352\n",
       "  t          PValue       AdjPVal      B         \n",
       "1   4.906848 1.533202e-06 8.170436e-03  0.8561371\n",
       "2 -10.329346 4.631685e-21 2.464057e-17 16.7535120\n",
       "3  -5.042202 9.757441e-07 5.117778e-03  5.1676834\n",
       "4  -5.035619 1.099810e-06 4.663317e-03  4.2503727\n",
       "5  -4.930630 1.777517e-06 4.663317e-03  4.1488842\n",
       "6   4.440250 1.825438e-05 4.818212e-02  2.5615289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "glimpse(gene_as)\n",
    "gene_as$Tissue <- factor(gene_as$Tissue)\n",
    "length(levels(gene_as$Tissue))\n",
    "table(is.na(gene_as$Display))\n",
    "table(gene_as$Display)\n",
    "colnames(gene_as)\n",
    "head(gene_as)\n",
    "tissue_reduction$display_name <- factor(tissue_reduction$display_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0ef4d76c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:08.481157Z",
     "iopub.status.busy": "2021-06-27T18:43:08.479718Z",
     "iopub.status.idle": "2021-06-27T18:43:08.515290Z",
     "shell.execute_reply": "2021-06-27T18:43:08.514105Z"
    },
    "papermill": {
     "duration": 0.078671,
     "end_time": "2021-06-27T18:43:08.515425",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.436754",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 743 total significant alternative splicing events (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 68 total significant alternative splicing events on the X chromosome (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "i.e., 9.15208613728129% of all significant AS events were on the X chromosome\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "gene_as now has 39 tissues and 5 ASE categories\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ASE:\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>A3SS</dt><dd>130</dd><dt>A5SS</dt><dd>107</dd><dt>MXE</dt><dd>77</dd><dt>RI</dt><dd>178</dd><dt>SE</dt><dd>251</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[A3SS] 130\n",
       "\\item[A5SS] 107\n",
       "\\item[MXE] 77\n",
       "\\item[RI] 178\n",
       "\\item[SE] 251\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "A3SS\n",
       ":   130A5SS\n",
       ":   107MXE\n",
       ":   77RI\n",
       ":   178SE\n",
       ":   251\n",
       "\n"
      ],
      "text/plain": [
       "A3SS A5SS  MXE   RI   SE \n",
       " 130  107   77  178  251 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x_as_events <- gene_as[gene_as$chr==\"chrX\",]\n",
    "message(\"There were \",nrow(gene_as),\" total significant alternative splicing events (gene_as)\")\n",
    "message(\"There were \",nrow(x_as_events),\" total significant alternative splicing events on the X chromosome (gene_as)\")\n",
    "message(\"i.e., \", (100*nrow(x_as_events)/nrow(gene_as)), \"% of all significant AS events were on the X chromosome\")\n",
    "\n",
    "numberOfUniqueTissues <- length(summary(as.factor(gene_as$Display),maxsum=500))\n",
    "numberOfASEmechanisms <- length(summary(as.factor(gene_as$ASE),maxsum=500))\n",
    "\n",
    "message(\"gene_as now has \",numberOfUniqueTissues, \" tissues and \", numberOfASEmechanisms, \" ASE categories\")\n",
    "message(\"ASE:\")\n",
    "summary(as.factor(gene_as$ASE),maxsum=500)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a16d00f1",
   "metadata": {
    "papermill": {
     "duration": 0.040812,
     "end_time": "2021-06-27T18:43:08.597603",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.556791",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.2 gene_dge.tsv\n",
    "\n",
    "This file contains (description)\n",
    "Here is a typical line\n",
    "<pre>\n",
    "Tissue  ENSG_ver        ENSG_no_ver     GeneSymbol      counts  Display logFC   AveExpr t       PValue  AdjPVal B\n",
    "1       adipose_subcutaneous    ENSG00000176728.7       ENSG00000176728 TTTY14  765     Adipose (sc)    -7.98216577151896     -0.928812923511535       -139.823010017733       0       0       1107.42326360464\n",
    "2       adipose_subcutaneous    ENSG00000231535.5       ENSG00000231535 LINC00278       765     Adipose (sc)    -6.09542040758638      -2.77656379347601       -126.913818678612       0       0       1050.36559888639\n",
    "3       adipose_subcutaneous    ENSG00000129824.15      ENSG00000129824 RPS4Y1  765     Adipose (sc)    -9.6641901864726      4.63528767282141 -125.827094717734       0       0       1041.87660796556\n",
    "4       adipose_subcutaneous    ENSG00000067646.11      ENSG00000067646 ZFY     765     Adipose (sc)    -9.50458982938477     0.672755457406984        -125.037143030325       0       0       1033.61131617113\n",
    "5       adipose_subcutaneous    ENSG00000229807.10      ENSG00000229807 XIST    765     Adipose (sc)    9.89280986473167      1.23756039627052 121.69689757218 0       0       1030.17757492281\n",
    "6       adipose_subcutaneous    ENSG00000229236.1       ENSG00000229236 TTTY10  765     Adipose (sc)    -6.20901295440725     -2.74524363170072        -122.540297482165       0       0       1029.41424065532\n",
    "7       adipose_subcutaneous    ENSG00000233864.7       ENSG00000233864 TTTY15  765     Adipose (sc)    -8.19361688496523     -0.741097276206495       -122.47199454746        0       0       1027.58175958647\n",
    "8       adipose_subcutaneous    ENSG00000260197.1       ENSG00000260197 AC010889.1      765     Adipose (sc)    -8.52835806068555      -0.686009457030557      -119.790486729538       0       0       1015.85291786821\n",
    "9       adipose_subcutaneous    ENSG00000183878.15      ENSG00000183878 UTY     765     Adipose (sc)    -9.52139275438866     1.60375445153084 -110.599936868261       0       0       953.069478992754\n",
    "</pre>\n",
    "There are 7417 significant events in the file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "7840c5b3",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:08.684695Z",
     "iopub.status.busy": "2021-06-27T18:43:08.683306Z",
     "iopub.status.idle": "2021-06-27T18:43:08.711040Z",
     "shell.execute_reply": "2021-06-27T18:43:08.709790Z"
    },
    "papermill": {
     "duration": 0.072739,
     "end_time": "2021-06-27T18:43:08.711158",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.638419",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 4417 total significant differential gene expression events (gene_dge)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "There were 328 total significant differential gene expression events on the X chromosome (gene_as)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "i.e., 7.42585465247906% of all significant DGE events were on the X chromosome\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "gene_dge now has 39 tissues\n",
      "\n"
     ]
    }
   ],
   "source": [
    "x_dge_events <- gene_dge[gene_dge$chr==\"chrX\",]\n",
    "message(\"There were \",nrow(gene_dge),\" total significant differential gene expression events (gene_dge)\")\n",
    "message(\"There were \",nrow(x_dge_events),\" total significant differential gene expression events on the X chromosome (gene_as)\")\n",
    "message(\"i.e., \", (100*nrow(x_dge_events)/nrow(gene_dge)), \"% of all significant DGE events were on the X chromosome\")\n",
    "\n",
    "numberOfUniqueTissues <- length(summary(as.factor(gene_dge$Display),maxsum=500))\n",
    "\n",
    "message(\"gene_dge now has \",numberOfUniqueTissues, \" tissues\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "354fdf0d",
   "metadata": {
    "papermill": {
     "duration": 0.041319,
     "end_time": "2021-06-27T18:43:08.794063",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.752744",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.3 Count events by chromosome\n",
    "\n",
    "Count the number of significant alternative splicing events per chromosome and save to the file **Total_AS_by_chr.tsv**."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2829fda",
   "metadata": {
    "papermill": {
     "duration": 0.041022,
     "end_time": "2021-06-27T18:43:08.876483",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.835461",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.3.1 by alternative splicing events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "657673c5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:08.965061Z",
     "iopub.status.busy": "2021-06-27T18:43:08.963785Z",
     "iopub.status.idle": "2021-06-27T18:43:09.018435Z",
     "shell.execute_reply": "2021-06-27T18:43:09.017208Z"
    },
    "papermill": {
     "duration": 0.100224,
     "end_time": "2021-06-27T18:43:09.018559",
     "exception": false,
     "start_time": "2021-06-27T18:43:08.918335",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "23"
      ],
      "text/latex": [
       "23"
      ],
      "text/markdown": [
       "23"
      ],
      "text/plain": [
       "[1] 23"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 23 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>83</td></tr>\n",
       "\t<tr><td>chrX </td><td>68</td></tr>\n",
       "\t<tr><td>chr17</td><td>56</td></tr>\n",
       "\t<tr><td>chr19</td><td>55</td></tr>\n",
       "\t<tr><td>chr16</td><td>47</td></tr>\n",
       "\t<tr><td>chr2 </td><td>44</td></tr>\n",
       "\t<tr><td>chr11</td><td>41</td></tr>\n",
       "\t<tr><td>chr12</td><td>35</td></tr>\n",
       "\t<tr><td>chr3 </td><td>35</td></tr>\n",
       "\t<tr><td>chr10</td><td>31</td></tr>\n",
       "\t<tr><td>chr7 </td><td>29</td></tr>\n",
       "\t<tr><td>chr6 </td><td>26</td></tr>\n",
       "\t<tr><td>chr8 </td><td>25</td></tr>\n",
       "\t<tr><td>chr4 </td><td>24</td></tr>\n",
       "\t<tr><td>chr15</td><td>22</td></tr>\n",
       "\t<tr><td>chr22</td><td>22</td></tr>\n",
       "\t<tr><td>chr14</td><td>19</td></tr>\n",
       "\t<tr><td>chr20</td><td>19</td></tr>\n",
       "\t<tr><td>chr9 </td><td>18</td></tr>\n",
       "\t<tr><td>chr5 </td><td>17</td></tr>\n",
       "\t<tr><td>chr18</td><td>10</td></tr>\n",
       "\t<tr><td>chr13</td><td> 9</td></tr>\n",
       "\t<tr><td>chr21</td><td> 8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 23 × 2\n",
       "\\begin{tabular}{ll}\n",
       " chr & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 83\\\\\n",
       "\t chrX  & 68\\\\\n",
       "\t chr17 & 56\\\\\n",
       "\t chr19 & 55\\\\\n",
       "\t chr16 & 47\\\\\n",
       "\t chr2  & 44\\\\\n",
       "\t chr11 & 41\\\\\n",
       "\t chr12 & 35\\\\\n",
       "\t chr3  & 35\\\\\n",
       "\t chr10 & 31\\\\\n",
       "\t chr7  & 29\\\\\n",
       "\t chr6  & 26\\\\\n",
       "\t chr8  & 25\\\\\n",
       "\t chr4  & 24\\\\\n",
       "\t chr15 & 22\\\\\n",
       "\t chr22 & 22\\\\\n",
       "\t chr14 & 19\\\\\n",
       "\t chr20 & 19\\\\\n",
       "\t chr9  & 18\\\\\n",
       "\t chr5  & 17\\\\\n",
       "\t chr18 & 10\\\\\n",
       "\t chr13 &  9\\\\\n",
       "\t chr21 &  8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 23 × 2\n",
       "\n",
       "| chr &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 83 |\n",
       "| chrX  | 68 |\n",
       "| chr17 | 56 |\n",
       "| chr19 | 55 |\n",
       "| chr16 | 47 |\n",
       "| chr2  | 44 |\n",
       "| chr11 | 41 |\n",
       "| chr12 | 35 |\n",
       "| chr3  | 35 |\n",
       "| chr10 | 31 |\n",
       "| chr7  | 29 |\n",
       "| chr6  | 26 |\n",
       "| chr8  | 25 |\n",
       "| chr4  | 24 |\n",
       "| chr15 | 22 |\n",
       "| chr22 | 22 |\n",
       "| chr14 | 19 |\n",
       "| chr20 | 19 |\n",
       "| chr9  | 18 |\n",
       "| chr5  | 17 |\n",
       "| chr18 | 10 |\n",
       "| chr13 |  9 |\n",
       "| chr21 |  8 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr   n \n",
       "1  chr1  83\n",
       "2  chrX  68\n",
       "3  chr17 56\n",
       "4  chr19 55\n",
       "5  chr16 47\n",
       "6  chr2  44\n",
       "7  chr11 41\n",
       "8  chr12 35\n",
       "9  chr3  35\n",
       "10 chr10 31\n",
       "11 chr7  29\n",
       "12 chr6  26\n",
       "13 chr8  25\n",
       "14 chr4  24\n",
       "15 chr15 22\n",
       "16 chr22 22\n",
       "17 chr14 19\n",
       "18 chr20 19\n",
       "19 chr9  18\n",
       "20 chr5  17\n",
       "21 chr18 10\n",
       "22 chr13  9\n",
       "23 chr21  8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 23\n",
      "Columns: 2\n",
      "$ chr \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr1, chrX, chr17, chr19, chr16, chr2, chr11, chr12, chr3, chr10, …\n",
      "$ n   \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 83, 68, 56, 55, 47, 44, 41, 35, 35, 31, 29, 26, 25, 24, 22, 22, 19…\n"
     ]
    }
   ],
   "source": [
    "total_as_by_chr <- gene_as          %>% \n",
    "                   group_by(chr)    %>% \n",
    "                   count(chr)       %>% \n",
    "                   arrange(desc(n)) %>% \n",
    "                   as.data.frame()\n",
    "total_as_by_chr$chr <- factor(total_as_by_chr$chr, levels = total_as_by_chr$chr)\n",
    "length(total_as_by_chr$chr)\n",
    "total_as_by_chr\n",
    "glimpse(total_as_by_chr)\n",
    "write.table(total_as_by_chr, file= \"../data/Total_AS_by_chr.tsv\", sep=\"\\t\", quote = FALSE, row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9cde083",
   "metadata": {
    "papermill": {
     "duration": 0.043406,
     "end_time": "2021-06-27T18:43:09.105536",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.062130",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.3.2 by gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "573e53bc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:09.223481Z",
     "iopub.status.busy": "2021-06-27T18:43:09.222242Z",
     "iopub.status.idle": "2021-06-27T18:43:09.273837Z",
     "shell.execute_reply": "2021-06-27T18:43:09.272607Z"
    },
    "papermill": {
     "duration": 0.097964,
     "end_time": "2021-06-27T18:43:09.273975",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.176011",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "26"
      ],
      "text/latex": [
       "26"
      ],
      "text/markdown": [
       "26"
      ],
      "text/plain": [
       "[1] 26"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 26 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>431</td></tr>\n",
       "\t<tr><td>chr2 </td><td>346</td></tr>\n",
       "\t<tr><td>chrX </td><td>328</td></tr>\n",
       "\t<tr><td>chr11</td><td>250</td></tr>\n",
       "\t<tr><td>chr12</td><td>234</td></tr>\n",
       "\t<tr><td>chr19</td><td>224</td></tr>\n",
       "\t<tr><td>chr17</td><td>216</td></tr>\n",
       "\t<tr><td>chr3 </td><td>212</td></tr>\n",
       "\t<tr><td>chr6 </td><td>199</td></tr>\n",
       "\t<tr><td>chr7 </td><td>196</td></tr>\n",
       "\t<tr><td>chr14</td><td>195</td></tr>\n",
       "\t<tr><td>chr4 </td><td>187</td></tr>\n",
       "\t<tr><td>chr5 </td><td>178</td></tr>\n",
       "\t<tr><td>chr10</td><td>153</td></tr>\n",
       "\t<tr><td>chr9 </td><td>150</td></tr>\n",
       "\t<tr><td>chr20</td><td>148</td></tr>\n",
       "\t<tr><td>chr15</td><td>147</td></tr>\n",
       "\t<tr><td>chr16</td><td>145</td></tr>\n",
       "\t<tr><td>chr22</td><td>130</td></tr>\n",
       "\t<tr><td>chr8 </td><td>128</td></tr>\n",
       "\t<tr><td>chr18</td><td> 67</td></tr>\n",
       "\t<tr><td>chr21</td><td> 58</td></tr>\n",
       "\t<tr><td>chr13</td><td> 57</td></tr>\n",
       "\t<tr><td>NA   </td><td> 28</td></tr>\n",
       "\t<tr><td>chrM </td><td>  7</td></tr>\n",
       "\t<tr><td>chrY </td><td>  3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 26 × 2\n",
       "\\begin{tabular}{ll}\n",
       " chr & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 431\\\\\n",
       "\t chr2  & 346\\\\\n",
       "\t chrX  & 328\\\\\n",
       "\t chr11 & 250\\\\\n",
       "\t chr12 & 234\\\\\n",
       "\t chr19 & 224\\\\\n",
       "\t chr17 & 216\\\\\n",
       "\t chr3  & 212\\\\\n",
       "\t chr6  & 199\\\\\n",
       "\t chr7  & 196\\\\\n",
       "\t chr14 & 195\\\\\n",
       "\t chr4  & 187\\\\\n",
       "\t chr5  & 178\\\\\n",
       "\t chr10 & 153\\\\\n",
       "\t chr9  & 150\\\\\n",
       "\t chr20 & 148\\\\\n",
       "\t chr15 & 147\\\\\n",
       "\t chr16 & 145\\\\\n",
       "\t chr22 & 130\\\\\n",
       "\t chr8  & 128\\\\\n",
       "\t chr18 &  67\\\\\n",
       "\t chr21 &  58\\\\\n",
       "\t chr13 &  57\\\\\n",
       "\t NA    &  28\\\\\n",
       "\t chrM  &   7\\\\\n",
       "\t chrY  &   3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 26 × 2\n",
       "\n",
       "| chr &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 431 |\n",
       "| chr2  | 346 |\n",
       "| chrX  | 328 |\n",
       "| chr11 | 250 |\n",
       "| chr12 | 234 |\n",
       "| chr19 | 224 |\n",
       "| chr17 | 216 |\n",
       "| chr3  | 212 |\n",
       "| chr6  | 199 |\n",
       "| chr7  | 196 |\n",
       "| chr14 | 195 |\n",
       "| chr4  | 187 |\n",
       "| chr5  | 178 |\n",
       "| chr10 | 153 |\n",
       "| chr9  | 150 |\n",
       "| chr20 | 148 |\n",
       "| chr15 | 147 |\n",
       "| chr16 | 145 |\n",
       "| chr22 | 130 |\n",
       "| chr8  | 128 |\n",
       "| chr18 |  67 |\n",
       "| chr21 |  58 |\n",
       "| chr13 |  57 |\n",
       "| NA    |  28 |\n",
       "| chrM  |   7 |\n",
       "| chrY  |   3 |\n",
       "\n"
      ],
      "text/plain": [
       "   chr   n  \n",
       "1  chr1  431\n",
       "2  chr2  346\n",
       "3  chrX  328\n",
       "4  chr11 250\n",
       "5  chr12 234\n",
       "6  chr19 224\n",
       "7  chr17 216\n",
       "8  chr3  212\n",
       "9  chr6  199\n",
       "10 chr7  196\n",
       "11 chr14 195\n",
       "12 chr4  187\n",
       "13 chr5  178\n",
       "14 chr10 153\n",
       "15 chr9  150\n",
       "16 chr20 148\n",
       "17 chr15 147\n",
       "18 chr16 145\n",
       "19 chr22 130\n",
       "20 chr8  128\n",
       "21 chr18  67\n",
       "22 chr21  58\n",
       "23 chr13  57\n",
       "24 NA     28\n",
       "25 chrM    7\n",
       "26 chrY    3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 26\n",
      "Columns: 2\n",
      "$ chr \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chr1, chr2, chrX, chr11, chr12, chr19, chr17, chr3, chr6, chr7, ch…\n",
      "$ n   \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 431, 346, 328, 250, 234, 224, 216, 212, 199, 196, 195, 187, 178, 1…\n"
     ]
    }
   ],
   "source": [
    "total_dge_by_chr <- gene_dge          %>% \n",
    "                   group_by(chr)    %>% \n",
    "                   count(chr)       %>% \n",
    "                   arrange(desc(n)) %>% \n",
    "                   as.data.frame()\n",
    "total_dge_by_chr$chr <- factor(total_dge_by_chr$chr, levels = total_dge_by_chr$chr)\n",
    "length(total_dge_by_chr$chr)\n",
    "total_dge_by_chr\n",
    "glimpse(total_dge_by_chr)\n",
    "write.table(total_dge_by_chr, file= \"../data/Total_DGE_by_chr.tsv\", sep=\"\\t\", quote = FALSE, row.names=F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dde77b98",
   "metadata": {
    "papermill": {
     "duration": 0.043517,
     "end_time": "2021-06-27T18:43:09.361264",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.317747",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.4 Count events by genes "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e3c7208",
   "metadata": {
    "papermill": {
     "duration": 0.043828,
     "end_time": "2021-06-27T18:43:09.449125",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.405297",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.4.1 by alternative splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "75c9ea6f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:09.542256Z",
     "iopub.status.busy": "2021-06-27T18:43:09.540905Z",
     "iopub.status.idle": "2021-06-27T18:43:09.597076Z",
     "shell.execute_reply": "2021-06-27T18:43:09.595880Z"
    },
    "papermill": {
     "duration": 0.103984,
     "end_time": "2021-06-27T18:43:09.597201",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.493217",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "581"
      ],
      "text/latex": [
       "581"
      ],
      "text/markdown": [
       "581"
      ],
      "text/plain": [
       "[1] 581"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneSymbol</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>XIST  </td><td>25</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>KDM5C </td><td>14</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>MUC1  </td><td> 6</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SORBS2</td><td> 6</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>DDX3X </td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>CELSR2</td><td> 4</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>ABCD4 </td><td> 3</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>BNIP2 </td><td> 3</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>DTNA  </td><td> 3</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>EPN3  </td><td> 3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & GeneSymbol & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & XIST   & 25\\\\\n",
       "\t2 & KDM5C  & 14\\\\\n",
       "\t3 & MUC1   &  6\\\\\n",
       "\t4 & SORBS2 &  6\\\\\n",
       "\t5 & DDX3X  &  5\\\\\n",
       "\t6 & CELSR2 &  4\\\\\n",
       "\t7 & ABCD4  &  3\\\\\n",
       "\t8 & BNIP2  &  3\\\\\n",
       "\t9 & DTNA   &  3\\\\\n",
       "\t10 & EPN3   &  3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | GeneSymbol &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | XIST   | 25 |\n",
       "| 2 | KDM5C  | 14 |\n",
       "| 3 | MUC1   |  6 |\n",
       "| 4 | SORBS2 |  6 |\n",
       "| 5 | DDX3X  |  5 |\n",
       "| 6 | CELSR2 |  4 |\n",
       "| 7 | ABCD4  |  3 |\n",
       "| 8 | BNIP2  |  3 |\n",
       "| 9 | DTNA   |  3 |\n",
       "| 10 | EPN3   |  3 |\n",
       "\n"
      ],
      "text/plain": [
       "   GeneSymbol n \n",
       "1  XIST       25\n",
       "2  KDM5C      14\n",
       "3  MUC1        6\n",
       "4  SORBS2      6\n",
       "5  DDX3X       5\n",
       "6  CELSR2      4\n",
       "7  ABCD4       3\n",
       "8  BNIP2       3\n",
       "9  DTNA        3\n",
       "10 EPN3        3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_geneSymbol <- gene_as %>% \n",
    "                          group_by(GeneSymbol) %>% \n",
    "                          count(GeneSymbol)    %>% \n",
    "                          arrange(desc(n))     %>% \n",
    "                          as.data.frame()\n",
    "total_as_by_geneSymbol$GeneSymbol <- factor(total_as_by_geneSymbol$GeneSymbol, \n",
    "                                            levels = total_as_by_geneSymbol$GeneSymbol)\n",
    "length(total_as_by_geneSymbol$GeneSymbol)\n",
    "head(total_as_by_geneSymbol,10)\n",
    "write.table(total_as_by_geneSymbol, file = \"../data/Total_AS_by_geneSymbol.tsv\", sep = \"\\t\", quote=FALSE, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9ba3537",
   "metadata": {
    "papermill": {
     "duration": 0.045288,
     "end_time": "2021-06-27T18:43:09.687312",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.642024",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.5 Count most frequent splicing by tissue\n",
    "\n",
    "### 3.5.1 by alternative splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "5de7834d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:09.781329Z",
     "iopub.status.busy": "2021-06-27T18:43:09.779928Z",
     "iopub.status.idle": "2021-06-27T18:43:09.822787Z",
     "shell.execute_reply": "2021-06-27T18:43:09.821674Z"
    },
    "papermill": {
     "duration": 0.091284,
     "end_time": "2021-06-27T18:43:09.822904",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.731620",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Display</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Breast         </td><td>521</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Cerebellum     </td><td> 34</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Esophagus (mu) </td><td> 32</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Spleen         </td><td> 28</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Hippocampus    </td><td> 14</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Sigmoid colon  </td><td> 14</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Aorta          </td><td> 10</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>Skeletal muscle</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Tibial artery  </td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Esophagus (gej)</td><td>  6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Display & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Breast          & 521\\\\\n",
       "\t2 & Cerebellum      &  34\\\\\n",
       "\t3 & Esophagus (mu)  &  32\\\\\n",
       "\t4 & Spleen          &  28\\\\\n",
       "\t5 & Hippocampus     &  14\\\\\n",
       "\t6 & Sigmoid colon   &  14\\\\\n",
       "\t7 & Aorta           &  10\\\\\n",
       "\t8 & Skeletal muscle &   7\\\\\n",
       "\t9 & Tibial artery   &   7\\\\\n",
       "\t10 & Esophagus (gej) &   6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | Display &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Breast          | 521 |\n",
       "| 2 | Cerebellum      |  34 |\n",
       "| 3 | Esophagus (mu)  |  32 |\n",
       "| 4 | Spleen          |  28 |\n",
       "| 5 | Hippocampus     |  14 |\n",
       "| 6 | Sigmoid colon   |  14 |\n",
       "| 7 | Aorta           |  10 |\n",
       "| 8 | Skeletal muscle |   7 |\n",
       "| 9 | Tibial artery   |   7 |\n",
       "| 10 | Esophagus (gej) |   6 |\n",
       "\n"
      ],
      "text/plain": [
       "   Display         n  \n",
       "1  Breast          521\n",
       "2  Cerebellum       34\n",
       "3  Esophagus (mu)   32\n",
       "4  Spleen           28\n",
       "5  Hippocampus      14\n",
       "6  Sigmoid colon    14\n",
       "7  Aorta            10\n",
       "8  Skeletal muscle   7\n",
       "9  Tibial artery     7\n",
       "10 Esophagus (gej)   6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "36"
      ],
      "text/latex": [
       "36"
      ],
      "text/markdown": [
       "36"
      ],
      "text/plain": [
       "[1] 36"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_tissue <- gene_as %>% \n",
    "                      group_by(Display) %>% \n",
    "                      count(Display)    %>% \n",
    "                      arrange(desc(n))  %>% \n",
    "                      as.data.frame()\n",
    "total_as_by_tissue$Display <- factor(total_as_by_tissue$Display, \n",
    "                                     levels = total_as_by_tissue$Display)\n",
    "head(total_as_by_tissue,10)\n",
    "length(total_as_by_tissue$Display)\n",
    "write.table(total_as_by_tissue, file = \"../data/Total_AS_by_tissue.tsv\", sep = \"\\t\", row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b02efa5e",
   "metadata": {
    "papermill": {
     "duration": 0.045979,
     "end_time": "2021-06-27T18:43:09.914652",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.868673",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.5.2 by gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "762be51f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:10.011694Z",
     "iopub.status.busy": "2021-06-27T18:43:10.010355Z",
     "iopub.status.idle": "2021-06-27T18:43:10.071072Z",
     "shell.execute_reply": "2021-06-27T18:43:10.069814Z"
    },
    "papermill": {
     "duration": 0.110889,
     "end_time": "2021-06-27T18:43:10.071196",
     "exception": false,
     "start_time": "2021-06-27T18:43:09.960307",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Display</th><th scope=col>n</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Breast            </td><td>2296</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Adipose (sc)      </td><td> 272</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Pituitary         </td><td> 256</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>Thyroid           </td><td> 164</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Adipose (v)       </td><td> 156</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>Skin (not exposed)</td><td> 129</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Skin (exposed)    </td><td> 116</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>Skeletal muscle   </td><td> 102</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Left ventricle    </td><td>  85</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>Fibroblasts       </td><td>  84</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Display & n\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Breast             & 2296\\\\\n",
       "\t2 & Adipose (sc)       &  272\\\\\n",
       "\t3 & Pituitary          &  256\\\\\n",
       "\t4 & Thyroid            &  164\\\\\n",
       "\t5 & Adipose (v)        &  156\\\\\n",
       "\t6 & Skin (not exposed) &  129\\\\\n",
       "\t7 & Skin (exposed)     &  116\\\\\n",
       "\t8 & Skeletal muscle    &  102\\\\\n",
       "\t9 & Left ventricle     &   85\\\\\n",
       "\t10 & Fibroblasts        &   84\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10 × 2\n",
       "\n",
       "| <!--/--> | Display &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | Breast             | 2296 |\n",
       "| 2 | Adipose (sc)       |  272 |\n",
       "| 3 | Pituitary          |  256 |\n",
       "| 4 | Thyroid            |  164 |\n",
       "| 5 | Adipose (v)        |  156 |\n",
       "| 6 | Skin (not exposed) |  129 |\n",
       "| 7 | Skin (exposed)     |  116 |\n",
       "| 8 | Skeletal muscle    |  102 |\n",
       "| 9 | Left ventricle     |   85 |\n",
       "| 10 | Fibroblasts        |   84 |\n",
       "\n"
      ],
      "text/plain": [
       "   Display            n   \n",
       "1  Breast             2296\n",
       "2  Adipose (sc)        272\n",
       "3  Pituitary           256\n",
       "4  Thyroid             164\n",
       "5  Adipose (v)         156\n",
       "6  Skin (not exposed)  129\n",
       "7  Skin (exposed)      116\n",
       "8  Skeletal muscle     102\n",
       "9  Left ventricle       85\n",
       "10 Fibroblasts          84"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#glimpse(gene_dge)\n",
    "gene_dge$GeneSymbol <- factor(gene_dge$GeneSymbol)\n",
    "total_dge_by_tissue <- gene_dge %>% \n",
    "                          select(c(GeneSymbol, Display, logFC)) %>%\n",
    "                          group_by(Display) %>%\n",
    "                          arrange(desc(logFC)) %>%\n",
    "                          tally() %>%\n",
    "                          arrange(desc(n)) %>%\n",
    "                          as.data.frame()\n",
    "head(total_dge_by_tissue,10)\n",
    "length(total_dge_by_tissue$Display)\n",
    "write.table(total_dge_by_tissue, file = \"../data/Total_DGE_by_tissue.tsv\", sep = \"\\t\", quote=FALSE, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ffcd3dad",
   "metadata": {
    "papermill": {
     "duration": 0.045898,
     "end_time": "2021-06-27T18:43:10.164128",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.118230",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "###  3.6 Significant Count by splicing type \n",
    "We define **significant** to be FC > 1.5 and pVal < 0.05\n",
    "\n",
    "Our starting values were the significant events, all meeting the criteria FC > 1.5 and pVal < 0.05\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "734207fa",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:10.262528Z",
     "iopub.status.busy": "2021-06-27T18:43:10.261166Z",
     "iopub.status.idle": "2021-06-27T18:43:10.296622Z",
     "shell.execute_reply": "2021-06-27T18:43:10.295417Z"
    },
    "papermill": {
     "duration": 0.08625,
     "end_time": "2021-06-27T18:43:10.296746",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.210496",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 5 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ASE</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SE  </td><td>251</td></tr>\n",
       "\t<tr><td>RI  </td><td>178</td></tr>\n",
       "\t<tr><td>A3SS</td><td>130</td></tr>\n",
       "\t<tr><td>A5SS</td><td>107</td></tr>\n",
       "\t<tr><td>MXE </td><td> 77</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 2\n",
       "\\begin{tabular}{ll}\n",
       " ASE & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t SE   & 251\\\\\n",
       "\t RI   & 178\\\\\n",
       "\t A3SS & 130\\\\\n",
       "\t A5SS & 107\\\\\n",
       "\t MXE  &  77\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 2\n",
       "\n",
       "| ASE &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| SE   | 251 |\n",
       "| RI   | 178 |\n",
       "| A3SS | 130 |\n",
       "| A5SS | 107 |\n",
       "| MXE  |  77 |\n",
       "\n"
      ],
      "text/plain": [
       "  ASE  n  \n",
       "1 SE   251\n",
       "2 RI   178\n",
       "3 A3SS 130\n",
       "4 A5SS 107\n",
       "5 MXE   77"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_as_by_splicingtype <- gene_as %>% \n",
    "                            group_by(ASE)    %>% \n",
    "                            count(ASE)       %>% \n",
    "                            arrange(desc(n)) %>%\n",
    "                            as.data.frame()\n",
    "total_as_by_splicingtype$ASE <- factor(total_as_by_splicingtype$ASE, levels = total_as_by_splicingtype$ASE)\n",
    "total_as_by_splicingtype\n",
    "write.table(total_as_by_splicingtype, file= \"../data/Total_AS_by_splicingtype.tsv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "061fda14",
   "metadata": {
    "papermill": {
     "duration": 0.046717,
     "end_time": "2021-06-27T18:43:10.390641",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.343924",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "###  3.7 Significant Count by splicing type (significant == FC > 1.5 and pVal < 0.05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a74e91ee",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:10.491110Z",
     "iopub.status.busy": "2021-06-27T18:43:10.489729Z",
     "iopub.status.idle": "2021-06-27T18:43:10.561455Z",
     "shell.execute_reply": "2021-06-27T18:43:10.560132Z"
    },
    "lines_to_next_cell": 0,
    "papermill": {
     "duration": 0.124332,
     "end_time": "2021-06-27T18:43:10.561578",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.437246",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "A3SS_keep\n",
       "FALSE  TRUE \n",
       "  613   130 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "A5SS_keep\n",
       "FALSE  TRUE \n",
       "  636   107 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "MXE_keep\n",
       "FALSE  TRUE \n",
       "  666    77 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "SE_keep\n",
       "FALSE  TRUE \n",
       "  492   251 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RI_keep\n",
       "FALSE  TRUE \n",
       "  565   178 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>130</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 130\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 130\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 130  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>107</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 107\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 107\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 107  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>77</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 77\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 77\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 77 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>251</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 251\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 251\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 251  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>178</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 178\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 178\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 178  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "A3SS_keep <- as.character(gene_as$ASE) %in% \"A3SS\"\n",
    "table(A3SS_keep)\n",
    "A3SS.gene_as <- data.frame(gene_as[A3SS_keep == TRUE,])\n",
    "\n",
    "A5SS_keep <- as.character(gene_as$ASE) %in% \"A5SS\"\n",
    "table(A5SS_keep)\n",
    "A5SS.gene_as <- data.frame(gene_as[A5SS_keep == TRUE,])\n",
    "\n",
    "MXE_keep  <- as.character(gene_as$ASE) %in% \"MXE\"\n",
    "table(MXE_keep)\n",
    "MXE.gene_as <- data.frame(gene_as[MXE_keep == TRUE,])\n",
    "\n",
    "SE_keep   <- as.character(gene_as$ASE) %in% \"SE\"\n",
    "table(SE_keep)\n",
    "SE.gene_as <- data.frame(gene_as[SE_keep == TRUE,])\n",
    "\n",
    "RI_keep   <- as.character(gene_as$ASE) %in% \"RI\"\n",
    "table(RI_keep)\n",
    "RI.gene_as <- data.frame(gene_as[RI_keep == TRUE,])\n",
    "\n",
    "dim(A3SS.gene_as)\n",
    "dim(A5SS.gene_as)\n",
    "dim(MXE.gene_as)\n",
    "dim(SE.gene_as)\n",
    "dim(RI.gene_as)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd95195e",
   "metadata": {
    "papermill": {
     "duration": 0.050228,
     "end_time": "2021-06-27T18:43:10.662103",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.611875",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.8 Siginficant spliced by Gene for each splicing factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0eb8ed47",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:10.768876Z",
     "iopub.status.busy": "2021-06-27T18:43:10.767506Z",
     "iopub.status.idle": "2021-06-27T18:43:10.880267Z",
     "shell.execute_reply": "2021-06-27T18:43:10.878965Z"
    },
    "papermill": {
     "duration": 0.166962,
     "end_time": "2021-06-27T18:43:10.880396",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.713434",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for A3SS\n",
      "116 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for A5SS\n",
      "100 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for MXE\n",
      "50 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for RI\n",
      "159 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Significant spliced genes for SE\n",
      "203 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "A3SS.res <- A3SS.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "A3SS.res$GeneSymbol <- factor(A3SS.res$GeneSymbol, levels = A3SS.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for A3SS\\n\",\n",
    "        paste(length(A3SS.res$GeneSymbol)), collapse=\" \")\n",
    "#head(A3SS.res)\n",
    "\n",
    "A5SS.res <- A5SS.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "A5SS.res$GeneSymbol <- factor(A5SS.res$GeneSymbol, levels = A5SS.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for A5SS\\n\",\n",
    "        paste(length(A5SS.res$GeneSymbol)), collapse=\" \")\n",
    "#head(A5SS.res)\n",
    "\n",
    "MXE.res <- MXE.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "MXE.res$GeneSymbol <- factor(MXE.res$GeneSymbol, levels = MXE.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for MXE\\n\",\n",
    "        paste(length(MXE.res$GeneSymbol)), collapse=\" \")\n",
    "#head(MXE.res)\n",
    "\n",
    "RI.res <- RI.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "RI.res$GeneSymbol <- factor(RI.res$GeneSymbol, levels = RI.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for RI\\n\",\n",
    "        paste(length(RI.res$GeneSymbol)), collapse=\" \")\n",
    "#head(RI.res)\n",
    "\n",
    "SE.res <- SE.gene_as %>% group_by(GeneSymbol) %>% count(GeneSymbol) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "SE.res$GeneSymbol <- factor(SE.res$GeneSymbol, levels = SE.res$GeneSymbol)\n",
    "message(\"Significant spliced genes for SE\\n\",\n",
    "        paste(length(SE.res$GeneSymbol)), collapse=\" \")\n",
    "#head(SE.res)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f68f65ce",
   "metadata": {
    "papermill": {
     "duration": 0.050919,
     "end_time": "2021-06-27T18:43:10.982548",
     "exception": false,
     "start_time": "2021-06-27T18:43:10.931629",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.9 Count most frequent spliced genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "857eb5d1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:11.092366Z",
     "iopub.status.busy": "2021-06-27T18:43:11.091128Z",
     "iopub.status.idle": "2021-06-27T18:43:12.016845Z",
     "shell.execute_reply": "2021-06-27T18:43:12.015599Z"
    },
    "papermill": {
     "duration": 0.981935,
     "end_time": "2021-06-27T18:43:12.016978",
     "exception": false,
     "start_time": "2021-06-27T18:43:11.035043",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "581"
      ],
      "text/latex": [
       "581"
      ],
      "text/markdown": [
       "581"
      ],
      "text/plain": [
       "[1] 581"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GeneSymbol</th><th scope=col>n</th><th scope=col>Tissues</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>XIST  </td><td>25</td><td>23</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>KDM5C </td><td>14</td><td>14</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>MUC1  </td><td> 6</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SORBS2</td><td> 6</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>DDX3X </td><td> 5</td><td> 4</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>CELSR2</td><td> 4</td><td> 1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & GeneSymbol & n & Tissues\\\\\n",
       "  & <fct> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & XIST   & 25 & 23\\\\\n",
       "\t2 & KDM5C  & 14 & 14\\\\\n",
       "\t3 & MUC1   &  6 &  1\\\\\n",
       "\t4 & SORBS2 &  6 &  1\\\\\n",
       "\t5 & DDX3X  &  5 &  4\\\\\n",
       "\t6 & CELSR2 &  4 &  1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | GeneSymbol &lt;fct&gt; | n &lt;int&gt; | Tissues &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | XIST   | 25 | 23 |\n",
       "| 2 | KDM5C  | 14 | 14 |\n",
       "| 3 | MUC1   |  6 |  1 |\n",
       "| 4 | SORBS2 |  6 |  1 |\n",
       "| 5 | DDX3X  |  5 |  4 |\n",
       "| 6 | CELSR2 |  4 |  1 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneSymbol n  Tissues\n",
       "1 XIST       25 23     \n",
       "2 KDM5C      14 14     \n",
       "3 MUC1        6  1     \n",
       "4 SORBS2      6  1     \n",
       "5 DDX3X       5  4     \n",
       "6 CELSR2      4  1     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genesMostFrequentlySpliced <- gene_as %>% \n",
    "                              group_by(GeneSymbol) %>% \n",
    "                              count(GeneSymbol)    %>% \n",
    "                              arrange(desc(n))     %>% \n",
    "                              as.data.frame()\n",
    "genesMostFrequentlySpliced$GeneSymbol <- factor(genesMostFrequentlySpliced$GeneSymbol, \n",
    "                                                levels = genesMostFrequentlySpliced$GeneSymbol)\n",
    "length(genesMostFrequentlySpliced$GeneSymbol)\n",
    "\n",
    "#Add number of tissues\n",
    "nTissues <- rep(NA, length(genesMostFrequentlySpliced))\n",
    "for (i in 1:nrow(genesMostFrequentlySpliced)) {\n",
    "    df_gene <- gene_as %>% \n",
    "               filter(GeneSymbol == genesMostFrequentlySpliced$GeneSymbol[i])\n",
    "    nTissues[i] <- length(unique(df_gene$Tissue))\n",
    "}\n",
    "genesMostFrequentlySpliced$Tissues <- nTissues\n",
    "head(genesMostFrequentlySpliced)\n",
    "write.table(genesMostFrequentlySpliced, file = \"../data/genesWithCommonAS.tsv\", sep = \"\\t\", quote = F, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18db6ef4",
   "metadata": {
    "papermill": {
     "duration": 0.052193,
     "end_time": "2021-06-27T18:43:12.121411",
     "exception": false,
     "start_time": "2021-06-27T18:43:12.069218",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.10 Count most frequent spliced chromosomes\n",
    "To get an indication of which chromosome has the most frequent slicing event (regardless of type)\n",
    "We create an index based upon the number of exons per chromosome.\n",
    "\n",
    "get the annotation file, at this writing, gencode.v30.annotation.gtf\n",
    "The information as to the number of exons within the chromosome may be found there"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "8c6efe3d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:12.231635Z",
     "iopub.status.busy": "2021-06-27T18:43:12.230291Z",
     "iopub.status.idle": "2021-06-27T18:43:15.286413Z",
     "shell.execute_reply": "2021-06-27T18:43:15.285306Z"
    },
    "papermill": {
     "duration": 3.112799,
     "end_time": "2021-06-27T18:43:15.286747",
     "exception": false,
     "start_time": "2021-06-27T18:43:12.173948",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows: 23\n",
      "Columns: 4\n",
      "$ chr        \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m chrX, chr16, chr22, chr19, chr17, chr1, chr20, chr10, chr8,…\n",
      "$ n          \u001b[3m\u001b[90m<int>\u001b[39m\u001b[23m 68, 47, 22, 55, 56, 83, 19, 31, 25, 8, 41, 24, 35, 29, 22, …\n",
      "$ ExonCounts \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 40029, 61199, 28655, 74466, 78291, 118996, 28506, 47124, 45…\n",
      "$ Index      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m 1.6987684, 0.7679864, 0.7677543, 0.7385921, 0.7152802, 0.69…\n"
     ]
    }
   ],
   "source": [
    "exons <- gencode[ gencode$type == \"exon\", ]\n",
    "exons <- as.data.frame(exons)\n",
    "\n",
    "#Obtain chromosomes we have splicing information for (recall we did not use chr Y in our analysis)\n",
    "all_chr <- as.character(unique(gene_as$chr))\n",
    "chr_counts <- rep(0, length(all_chr))\n",
    "\n",
    "\n",
    "for (i in 1:length(all_chr)) {\n",
    "  chr_counts[i] <- nrow(exons[exons$seqnames == all_chr[i], ])\n",
    "}\n",
    "\n",
    "exon_counts <- data.frame(chr = all_chr, counts = chr_counts)\n",
    "\n",
    "# Count most frequent spliced chromosomes\n",
    "res <- gene_as %>% group_by(chr) %>% count(chr) %>% arrange(desc(n)) %>% as.data.frame()\n",
    "res$chr <- factor(res$chr, levels = res$chr)\n",
    "\n",
    "idx <- match(res$chr, exon_counts$chr)\n",
    "\n",
    "res$ExonCounts <- exon_counts$counts[idx]\n",
    "\n",
    "res$Index <- (res$n / res$ExonCounts) * 1000\n",
    "\n",
    "res_sorted <- res %>% arrange(desc(Index))\n",
    "res_sorted$chr <- factor(res_sorted$chr, levels = res_sorted$chr)\n",
    "glimpse(res_sorted)\n",
    "write.table(res_sorted, file = \"../data/SplicingIndex_chr.tsv\", sep = \"\\t\", quote = F, row.names = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2d82eb3",
   "metadata": {
    "papermill": {
     "duration": 0.052082,
     "end_time": "2021-06-27T18:43:15.391336",
     "exception": false,
     "start_time": "2021-06-27T18:43:15.339254",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.11 Overlap between Differential Gene Expression and Differential Alternative Splicing\n",
    "\n",
    "First gather the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "479ffef1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:15.501929Z",
     "iopub.status.busy": "2021-06-27T18:43:15.500662Z",
     "iopub.status.idle": "2021-06-27T18:43:15.772157Z",
     "shell.execute_reply": "2021-06-27T18:43:15.770935Z"
    },
    "papermill": {
     "duration": 0.328393,
     "end_time": "2021-06-27T18:43:15.772358",
     "exception": false,
     "start_time": "2021-06-27T18:43:15.443965",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'A2M'</li><li>'AADAC'</li><li>'AADAC'</li><li>'AADAC'</li><li>'AADACL3'</li><li>'AARD'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A2M'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADACL3'\n",
       "\\item 'AARD'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A2M'\n",
       "2. 'AADAC'\n",
       "3. 'AADAC'\n",
       "4. 'AADAC'\n",
       "5. 'AADACL3'\n",
       "6. 'AARD'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A2M\"     \"AADAC\"   \"AADAC\"   \"AADAC\"   \"AADACL3\" \"AARD\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'A1BG'</li><li>'A1BG-AS1'</li><li>'A1CF'</li><li>'A2M'</li><li>'A2M-AS1'</li><li>'A2ML1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A1BG'\n",
       "\\item 'A1BG-AS1'\n",
       "\\item 'A1CF'\n",
       "\\item 'A2M'\n",
       "\\item 'A2M-AS1'\n",
       "\\item 'A2ML1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A1BG'\n",
       "2. 'A1BG-AS1'\n",
       "3. 'A1CF'\n",
       "4. 'A2M'\n",
       "5. 'A2M-AS1'\n",
       "6. 'A2ML1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A1BG\"     \"A1BG-AS1\" \"A1CF\"     \"A2M\"      \"A2M-AS1\"  \"A2ML1\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "total_AS_Genes <- read.table(file=\"../data/Total_AS_by_geneSymbol.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "sigAsGenes <- sort(total_AS_Genes$GeneSymbol)\n",
    "dge <- read.table(\"../data/gene_dge.tsv\", sep = \"\\t\", header = FALSE, row.names=1, skip = 1)\n",
    "#dge <- data.table::fread(\"../data/gene_dge.tsv\")\n",
    "dge_genes <- sort(dge$V5)\n",
    "head(dge_genes)\n",
    "all_genes_data <- read.table(\"../assets/all_gene_dge.tsv\")\n",
    "all_genes <- sort(all_genes_data$GeneSymbol)\n",
    "head(all_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "971375f2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:15.885842Z",
     "iopub.status.busy": "2021-06-27T18:43:15.884365Z",
     "iopub.status.idle": "2021-06-27T18:43:15.898700Z",
     "shell.execute_reply": "2021-06-27T18:43:15.897549Z"
    },
    "papermill": {
     "duration": 0.072382,
     "end_time": "2021-06-27T18:43:15.898816",
     "exception": false,
     "start_time": "2021-06-27T18:43:15.826434",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'A2M'</li><li>'AADAC'</li><li>'AADAC'</li><li>'AADAC'</li><li>'AADACL3'</li><li>'AARD'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A2M'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADAC'\n",
       "\\item 'AADACL3'\n",
       "\\item 'AARD'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A2M'\n",
       "2. 'AADAC'\n",
       "3. 'AADAC'\n",
       "4. 'AADAC'\n",
       "5. 'AADACL3'\n",
       "6. 'AARD'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A2M\"     \"AADAC\"   \"AADAC\"   \"AADAC\"   \"AADACL3\" \"AARD\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(dge_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "fc343654",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:16.011482Z",
     "iopub.status.busy": "2021-06-27T18:43:16.010139Z",
     "iopub.status.idle": "2021-06-27T18:43:16.024636Z",
     "shell.execute_reply": "2021-06-27T18:43:16.023558Z"
    },
    "papermill": {
     "duration": 0.072383,
     "end_time": "2021-06-27T18:43:16.024754",
     "exception": false,
     "start_time": "2021-06-27T18:43:15.952371",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'A1BG'</li><li>'A1BG-AS1'</li><li>'A1CF'</li><li>'A2M'</li><li>'A2M-AS1'</li><li>'A2ML1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A1BG'\n",
       "\\item 'A1BG-AS1'\n",
       "\\item 'A1CF'\n",
       "\\item 'A2M'\n",
       "\\item 'A2M-AS1'\n",
       "\\item 'A2ML1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A1BG'\n",
       "2. 'A1BG-AS1'\n",
       "3. 'A1CF'\n",
       "4. 'A2M'\n",
       "5. 'A2M-AS1'\n",
       "6. 'A2ML1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A1BG\"     \"A1BG-AS1\" \"A1CF\"     \"A2M\"      \"A2M-AS1\"  \"A2ML1\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(all_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8191d66",
   "metadata": {
    "papermill": {
     "duration": 0.054175,
     "end_time": "2021-06-27T18:43:16.134118",
     "exception": false,
     "start_time": "2021-06-27T18:43:16.079943",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.12 We then do a hypergeometric/Fisher test to look for overrepresentation\n",
    "The universe consists of all genes with at least one read (all_genes_data).\n",
    "So we have\n",
    "\n",
    "|  \t|  DGE+| DGE-|\n",
    "|-\t|-\t|-\t|\n",
    "| DAS+|  a|  b|\n",
    "| DAS-|  c| d|"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "0681d4ea",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:16.251167Z",
     "iopub.status.busy": "2021-06-27T18:43:16.249969Z",
     "iopub.status.idle": "2021-06-27T18:43:16.298058Z",
     "shell.execute_reply": "2021-06-27T18:43:16.296840Z"
    },
    "papermill": {
     "duration": 0.107548,
     "end_time": "2021-06-27T18:43:16.298179",
     "exception": false,
     "start_time": "2021-06-27T18:43:16.190631",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of sigAsGenes 581\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of genes that are NOT sigAs 23480\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of DGE genes 4417\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of genes that are NOT DGE 20896\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "a: 144; b: 429; c: 3013; d: 20467\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(\"Number of sigAsGenes \", length(sigAsGenes))\n",
    "notSigAs <- setdiff(all_genes,sigAsGenes)\n",
    "message(\"Number of genes that are NOT sigAs \", length(notSigAs))\n",
    "message(\"Number of DGE genes \", length(dge_genes))\n",
    "notDGE <- setdiff(all_genes,dge_genes)\n",
    "message(\"Number of genes that are NOT DGE \", length(notDGE))\n",
    "a <- intersect(sigAsGenes, dge_genes)\n",
    "b <- intersect(sigAsGenes, notDGE)\n",
    "c <- intersect(notSigAs, dge_genes)\n",
    "d <- intersect(notSigAs, notDGE)\n",
    "message(\"a: \", length(a), \"; b: \",  length(b), \"; c: \",  length(c), \"; d: \",  length(d))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "6524b592",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:16.414572Z",
     "iopub.status.busy": "2021-06-27T18:43:16.413388Z",
     "iopub.status.idle": "2021-06-27T18:43:16.433452Z",
     "shell.execute_reply": "2021-06-27T18:43:16.432294Z"
    },
    "papermill": {
     "duration": 0.079901,
     "end_time": "2021-06-27T18:43:16.433573",
     "exception": false,
     "start_time": "2021-06-27T18:43:16.353672",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "\tFisher's Exact Test for Count Data\n",
       "\n",
       "data:  m\n",
       "p-value = 4.649e-15\n",
       "alternative hypothesis: true odds ratio is not equal to 1\n",
       "95 percent confidence interval:\n",
       " 1.867195 2.771118\n",
       "sample estimates:\n",
       "odds ratio \n",
       "  2.280031 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "m <- matrix(c(length(a),length(b),length(c),length(d)), nrow=2,byrow = TRUE)\n",
    "fisher.test(m)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d2f662f",
   "metadata": {
    "papermill": {
     "duration": 0.056006,
     "end_time": "2021-06-27T18:43:16.545134",
     "exception": false,
     "start_time": "2021-06-27T18:43:16.489128",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Appendix - Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84383710",
   "metadata": {
    "papermill": {
     "duration": 0.055933,
     "end_time": "2021-06-27T18:43:16.656892",
     "exception": false,
     "start_time": "2021-06-27T18:43:16.600959",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Appendix 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7f224f6e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:16.774107Z",
     "iopub.status.busy": "2021-06-27T18:43:16.772721Z",
     "iopub.status.idle": "2021-06-27T18:43:17.585498Z",
     "shell.execute_reply": "2021-06-27T18:43:17.584178Z"
    },
    "papermill": {
     "duration": 0.872481,
     "end_time": "2021-06-27T18:43:17.585623",
     "exception": false,
     "start_time": "2021-06-27T18:43:16.713142",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'countGenesAndEvents'"
      ],
      "text/latex": [
       "'countGenesAndEvents'"
      ],
      "text/markdown": [
       "'countGenesAndEvents'"
      ],
      "text/plain": [
       "[1] \"countGenesAndEvents\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/gene_as.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/all_gene_as.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/gene_dge.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_chr.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_geneSymbol.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_tissue.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/Total_AS_by_splicingtype.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/genesWithCommonAS.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the file `../data/SplicingIndex_chr.tsv` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"countGenesAndEvents\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/gene_as.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_as.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/all_gene_as.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_dge.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/gene_dge.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name gene_as.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_chr.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_chr.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_geneSymbol.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_geneSymbol.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_tissue.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_tissue.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/Total_AS_by_splicingtype.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name Total_AS_by_splicingtype.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/genesWithCommonAS.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name genesWithCommonAS.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Generating sha256 checksums of the file `../data/SplicingIndex_chr.tsv` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -name SplicingIndex_chr.tsv -exec sha256sum {} \\\\;  >  ../data/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d728596d",
   "metadata": {
    "papermill": {
     "duration": 0.063895,
     "end_time": "2021-06-27T18:43:17.712693",
     "exception": false,
     "start_time": "2021-06-27T18:43:17.648798",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Appendix 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "raw",
   "id": "3edd04bf",
   "metadata": {
    "lines_to_next_cell": 0,
    "papermill": {
     "duration": 0.063314,
     "end_time": "2021-06-27T18:43:17.839617",
     "exception": false,
     "start_time": "2021-06-27T18:43:17.776303",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../data/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../data/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../data/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../data/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "ab0e79b4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-06-27T18:43:17.975356Z",
     "iopub.status.busy": "2021-06-27T18:43:17.973806Z",
     "iopub.status.idle": "2021-06-27T18:43:17.994482Z",
     "shell.execute_reply": "2021-06-27T18:43:17.992868Z"
    },
    "papermill": {
     "duration": 0.090474,
     "end_time": "2021-06-27T18:43:17.994614",
     "exception": false,
     "start_time": "2021-06-27T18:43:17.904140",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Time difference of 3.140546 mins"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "end_time <- Sys.time()\n",
    "end_time - start_time"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "encoding": "# -*- coding: utf-8 -*-",
   "main_language": "R",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  },
  "papermill": {
   "duration": 190.011301,
   "end_time": "2021-06-27T18:43:18.267601",
   "environment_variables": {},
   "exception": null,
   "input_path": "countGenesAndEvents.ipynb",
   "output_path": "countGenesAndEvents.ipynb",
   "parameters": {},
   "start_time": "2021-06-27T18:40:08.256300",
   "version": "2.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}