{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Gene Expression Analysis\n",
    "\n",
    "This notebook generates the sex-biased differential gene expression analysis.   Differential Analysis (DE) was performed using voom (Law et.al., 2014) with gene expression counts with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma. \n",
    "\n",
    "Within each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression:\n",
    "\n",
    "\n",
    "           y = B0 + B1 sex + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the gene expression to be modeled sex denotes the reported sex of the subject.   The function named `fit_tissue()` performs this analysis and accepts two arguments, the `tissue` and an `object` and create the **model matrix** based  that tissue's sex. We will perform a linear fit after calculating normal factors (based on the library size) and calculate the dispersion using `voom` (mean variance model of dispersion). We are saving the resulting matrixes as files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Introduction:  Running this notebook:\n",
    "\n",
    "A few steps are needed before you can run this document on your own. The GitHub repository (https://github.com/TheJacksonLaboratory/sbas) of the project contains detailed instructions for setting up the environment in the **`dependencies/README.md`** document. Before starting with the analysis, make sure you have first completed the dependencies set up by following the instructions described there. If you have not done this already, you will need to close and restart this notebook before running it.\n",
    "\n",
    "All paths defined in this Notebook are relative to the parent directory (repository). \n",
    "\n",
    "Next - you could execute the entire notebook and it will generate output to both `../data` and diagnostic plots to `../pdf`.\n",
    "\n",
    "### 1.1 Input\n",
    "\n",
    "#### 1.1.1 SraRunTable.txt.gz \n",
    "This is the SRR run metadata manifest from the dbGaP.   This is released as part of this code release and accessible using `piggyback` from `github` with the users own `github` token.\n",
    "\n",
    "#### 1.1.2 gtex.rds \n",
    "This is generated from a forked version of yarn, explained in detail in section `1.2`.  It is an edgeR expressionSet object.   This object is built using `yarn` and R package. \n",
    "\n",
    "### 1.2 Process\n",
    "\n",
    "Using `EdgeR`, an `expressionSet object` is built, normalization factors are calculated with `calcNormFactors`, variance is modeled using mean-variance method using `Limma`'s `voom`.   A linear model is built to model the `sex-biased` differences.  Diagnostic plots and differential gene expression analysis results reported.  The function that accomplishes this step is `fit_tissue` and carried out in section 5.\n",
    "\n",
    "### 1.3 Output\n",
    "\n",
    "For each tissue, as selected and specified in the tissues.tsv file found in the `../assets` directory.  The following files are produced:\n",
    "\n",
    "#### 1.3.1 ../data/`tissue`_DGE.csv\n",
    "\n",
    "This file contains the `topTable` results, reporting the `ENSG`- gene identification,`logFC` - log fold change, `AveExpr`, `t` - the model result for the sex-bias (see section 2, `P.Value`, `adj.P.Val` - bon, and `B (FDR)`.\n",
    "\n",
    "#### 1.3.2 ../data/`tissue`_refined.csv\n",
    "\n",
    "This are the values that are differentially expressed with results that are 1.5 fold change greater than the mean and with a p-value of less than 0.05.\n",
    "\n",
    "#### 1.3.3 ../data/`tissue`_ensg_map.csv\n",
    "\n",
    "This is the mapping using `gprofiler` of the `ENSG` identifiers to their `geneSymbols` for ease of filtering prior to creating the linear model of the junctions in the computational step for the differential alternative splicing is completed.\n",
    "\n",
    "### 1.3.4  ../pdf/`tissue`-gene-y-voom-MDSplot-100.pdf\n",
    "\n",
    "These are the counts in a multi-dimensional scaling plot (MDSplot), showing the ability for the model to segregate the sex as illustrated with `red` `m` for the male and `blue` `f` for the female self-reported sex phenotypes.  In these plots, voom has been used to model the variance.\n",
    "\n",
    "### 1.3.5  ../pdf/`tissue`-gene-y-MDSplot-100.pdf\n",
    "\n",
    "These are the counts in a multi-dimensional scaling plot (MDSplot), showing the ability for the model to segregate the sex as illustrated with `red` `m` for the male and `blue` `f` for the female self-reported sex phenotypes.  In these plots, this is without the application of the results from modeling the variance with voom.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Setup\n",
    "### 2.1 Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: limma\n",
      "\n",
      "Loading required package: usethis\n",
      "\n",
      "\n",
      "Attaching package: ‘devtools’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:downloader’:\n",
      "\n",
      "    source_url\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:limma’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Setting options('download.file.method.GEOquery'='auto')\n",
      "\n",
      "Setting options('GEOquery.inmemory.gpl'=FALSE)\n",
      "\n",
      "Warning message:\n",
      "“package ‘snakecase’ was built under R version 3.6.3”\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:biomaRt’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘scales’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:readr’:\n",
      "\n",
      "    col_factor\n",
      "\n",
      "\n",
      "Loading required package: viridisLite\n",
      "\n",
      "\n",
      "Attaching package: ‘viridis’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:scales’:\n",
      "\n",
      "    viridis_pal\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(gprofiler2)\n",
    "library(downloader)\n",
    "library(readr)\n",
    "library(edgeR)\n",
    "library(biomaRt)\n",
    "library(DBI) # v >= 1.1.0 required for biomaRt\n",
    "library(devtools)\n",
    "library(yarn)\n",
    "library(statmod)\n",
    "library(piggyback)\n",
    "library(snakecase)\n",
    "library(stringr)\n",
    "library(pheatmap)\n",
    "library(magrittr)\n",
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(scales)\n",
    "library(viridis)\n",
    "library(scales)\n",
    "\n",
    "Sys.setenv(TAR = \"/bin/tar\") # for gzfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.2 Retrieving the GTEx archive\n",
    "\n",
    "We used the R package [yarn](https://bioconductor.org/packages/release/bioc/html/yarn.html) to retrieve the GTEx Biobank data. In order to download the latest GTEx version (8.0) for RNA-seq and genotype data (phs000424.v8.v2), released 2019-08-26, we created a fork of the package's GitHub repository and created a new version of the function **`yarn::downloadGTEx()`**, namely **`yarn::downloadGTExV8()`** to download this release. \n",
    "\n",
    "We used the function to perform quality control, gene filtering and normalization pre-processing on the GTEx RNA-seq data. This pipeline tested for sample sex-misidentification, merged related sub-tissues and performed tissue-aware normalization using the **`{yarn::qsmooth}`**  function ([Paulson et al, 2017](https://pubmed.ncbi.nlm.nih.gov/28974199/)).\n",
    "\n",
    "We have archived the output of the **`yarn::downloadGTExV8()`** function, which is an `ExpressionSet` object in the repo `lifebitai/lifebitCloudOSDREgtex` for replicability and decreasing the runtime of this analysis. Below we retrieve this `gtex.rds` object from the GitHub releases using the **`{ropensci/piggyback}`** package, but we have also added the relevant command to retrieve the data from GTEx and generate the `ExpressionSet` object using  **`yarn::downloadGTExV8()`**. For the current analysis we are utilising a compute resource with 8 vCPUs and 60 GB of memory available.\n",
    "\n",
    "You need to set your github token\n",
    "Sys.setenv(GITHUB_TOKEN = \"your-very-own-github-token\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    " Sys.setenv(GITHUB_TOKEN = \"your-very-own-github-token\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Did you remember to remove the token\n",
    "\n",
    "#### Download with yarn, if the data are not already in your repository\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading GTEx v8 rds object with readRDS from ../data/gtex.rds ..\n",
      "\n",
      "\n",
      "Done!\n",
      "\n",
      "\n",
      "Generating sha256sum for gtex.rds ..\n",
      "\n",
      "\n",
      "c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0  ../data/gtex.rds\n",
      "\n",
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "'ExpressionSet'"
      ],
      "text/latex": [
       "'ExpressionSet'"
      ],
      "text/markdown": [
       "'ExpressionSet'"
      ],
      "text/plain": [
       "[1] \"ExpressionSet\"\n",
       "attr(,\"package\")\n",
       "[1] \"Biobase\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>rowNames</dt><dd>17384</dd><dt>columnNames</dt><dd>67</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[rowNames] 17384\n",
       "\\item[columnNames] 67\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "rowNames\n",
       ":   17384columnNames\n",
       ":   67\n",
       "\n"
      ],
      "text/plain": [
       "   rowNames columnNames \n",
       "      17384          67 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Download with yarn if you wish, this requires several minutes to complete\n",
    "if (!(\"gtex.rds\" %in% list.files(\"../data/\"))) {\n",
    "    message(\"Downloading GTEx v8 with 'yarn::downloadGTExV8()'\")\n",
    "    obj <- yarn::downloadGTExV8(type='genes',file='../data/gtex.rds')\n",
    "    message(\"Done!\")\n",
    "\n",
    "} else {\n",
    "# Load with readRDS() if gtex.rds available in data/\n",
    "    message(\"Loading GTEx v8 rds object with readRDS from ../data/gtex.rds ..\\n\")   \n",
    "    obj <- readRDS(file = \"../data/gtex.rds\")\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Generating sha256sum for gtex.rds ..\\n\")    \n",
    "    message(system(\"sha256sum ../data/gtex.rds\", intern = TRUE))\n",
    "    message(\"Done!\\n\")\n",
    "} \n",
    "# Confirm that it is an expression set.\n",
    "# and check the dimensions of the objects, and the phenotype information of the objects\n",
    "class(obj) \n",
    "dim(phenoData(obj))\n",
    "dim(obj)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.3 download the Annotation file saved from getting the GTEx Fastq's\n",
    "\n",
    "This is the annotation file as organized by dbGaP, we use yarn as the annotations for sex have been noted to be incorrect.\n",
    "This file and the object from yarn will be joined in this way the differences can be noted, and managed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# If the SraRunTable from SRA's dbGaP not already in our directories, download with SraRunTable.noCram.noExome.noWGS.totalRNA.txt.gz \n",
    "# the annotations in this file have errors -- so for the samples we will use the yarn annotations\n",
    "if (!(\"SraRunTable.txt.gz\" %in% list.files(\"../data/\"))) {\n",
    "    piggyback::pb_download(\n",
    "        repo = \"TheJacksonLaboratory/sbas\", \n",
    "        file = \"SraRunTable.txt.gz\",\n",
    "        tag  = \"GTExV8.v1.0\", \n",
    "        dest = \"../data/\")\n",
    "    metadata          <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "} else {\n",
    "    metadata          <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.4: Quality control, preprocessing of data \n",
    "\n",
    "We observed above that our phenotype data have 2 more observations than our expression data. Let's inspect what these samples are:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "17382"
      ],
      "text/latex": [
       "17382"
      ],
      "text/markdown": [
       "17382"
      ],
      "text/plain": [
       "[1] 17382"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The non-overlapping IDs between pheno and count data are:\n",
      "\n",
      "GTEX-YF7O-2326-101833-SM-5CVN9\n",
      "GTEX-YEC3-1426-101806-SM-5PNXX\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "17384"
      ],
      "text/latex": [
       "17384"
      ],
      "text/markdown": [
       "17384"
      ],
      "text/plain": [
       "[1] 17384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "logical_match_names\n",
       "FALSE  TRUE \n",
       "    2 17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>17382</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 17382\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 17382\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 17382    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55878</dd><dt>Samples</dt><dd>17382</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55878\n",
       "\\item[Samples] 17382\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55878Samples\n",
       ":   17382\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55878    17382 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sample_names=as.vector(as.character(colnames(exprs(obj))))\n",
    "pheno_sample_names=as.vector(as.character(rownames(pData(obj))))\n",
    "length(pheno_sample_names)\n",
    "length(sample_names)\n",
    "\n",
    "if (length(pheno_sample_names) > length(sample_names)) {\n",
    "    superset <- pheno_sample_names\n",
    "    subset   <- sample_names    \n",
    "} \n",
    "\n",
    "if (length(pheno_sample_names) < length(sample_names)) {\n",
    "    superset <- sample_names\n",
    "    subset   <- pheno_sample_names   \n",
    "} \n",
    "\n",
    "non_overlaps <- setdiff( superset, subset)\n",
    "\n",
    "message(\"The non-overlapping IDs between pheno and count data are:\\n\\n\", \n",
    "        paste(non_overlaps, collapse = \"\\n\") )\n",
    "logical_match_names=superset %in% subset\n",
    "length(logical_match_names)\n",
    "table(logical_match_names)\n",
    "pData(obj) <- (pData(obj)[logical_match_names==TRUE,])\n",
    "dim(pData(obj))\n",
    "dim(obj)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.4.2   Replace all *dashes* with **dots \".\"**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>GTEX-1117F-0226-SM-5GZZ7:</strong> GTEX-1117F-0226-SM-5GZZ7\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'GTEX-1117F-0003-SM-58Q7G'</li><li>'GTEX-1117F-0003-SM-5DWSB'</li><li>'GTEX-1117F-0003-SM-6WBT7'</li><li>'GTEX-1117F-0011-R10a-SM-AHZ7F'</li><li>'GTEX-1117F-0011-R10b-SM-CYKQ8'</li><li>'GTEX-1117F-0226-SM-5GZZ7'</li><li>'GTEX-1117F-0426-SM-5EGHI'</li><li>'GTEX-1117F-0526-SM-5EGHJ'</li><li>'GTEX-1117F-0626-SM-5N9CS'</li><li>'GTEX-1117F-0726-SM-5GIEN'</li><li>'GTEX-1117F-1326-SM-5EGHH'</li><li>'GTEX-1117F-2226-SM-5N9CH'</li><li>'GTEX-1117F-2426-SM-5EGGH'</li><li>'GTEX-1117F-2526-SM-5GZY6'</li><li>'GTEX-1117F-2826-SM-5GZXL'</li><li>'GTEX-1117F-2926-SM-5GZYI'</li><li>'GTEX-1117F-3026-SM-5GZYU'</li><li>'GTEX-1117F-3226-SM-5N9CT'</li><li>'GTEX-111CU-0003-SM-58Q95'</li><li>'GTEX-111CU-0003-SM-5DWTR'</li><li>'GTEX-111CU-0003-SM-6WBUD'</li><li>'GTEX-111CU-0126-SM-5GZWZ'</li><li>'GTEX-111CU-0226-SM-5GZXC'</li><li>'GTEX-111CU-0326-SM-5GZXO'</li><li>'GTEX-111CU-0426-SM-5GZY1'</li><li>'GTEX-111CU-0526-SM-5EGHK'</li><li>'GTEX-111CU-0626-SM-5EGHL'</li><li>'GTEX-111CU-0726-SM-5GZYD'</li><li>'GTEX-111CU-0826-SM-5EGIJ'</li><li>'GTEX-111CU-0926-SM-5EGIK'</li><li>'GTEX-111CU-1026-SM-5EGIL'</li><li>'GTEX-111CU-1126-SM-5EGIM'</li><li>'GTEX-111CU-1226-SM-5EGIN'</li><li>'GTEX-111CU-1326-SM-5NQ8L'</li><li>'GTEX-111CU-1426-SM-5GZYP'</li><li>'GTEX-111CU-1526-SM-5N9FS'</li><li>'GTEX-111CU-1726-SM-5EGHM'</li><li>'GTEX-111CU-1826-SM-5GZYN'</li><li>'GTEX-111CU-1926-SM-5GZYZ'</li><li>'GTEX-111CU-2026-SM-5GZZC'</li><li>'GTEX-111CU-2226-SM-5N9G5'</li><li>'GTEX-111FC-0001-SM-58Q7U'</li><li>'GTEX-111FC-0001-SM-5DWSO'</li><li>'GTEX-111FC-0001-SM-6WBTJ'</li><li>'GTEX-111FC-0011-R10a-SM-AHZ7K'</li><li>'GTEX-111FC-0011-R10a-SM-CYKQ9'</li><li>'GTEX-111FC-0126-SM-5N9DL'</li><li>'GTEX-111FC-0226-SM-5N9B8'</li><li>'GTEX-111FC-0326-SM-5GZZ1'</li><li>'GTEX-111FC-0426-SM-5N9CV'</li><li>'GTEX-111FC-0526-SM-5GZZ8'</li><li>'GTEX-111FC-0626-SM-5N9CU'</li><li>'GTEX-111FC-0826-SM-5GZWO'</li><li>'GTEX-111FC-0826-SM-CYKRT'</li><li>'GTEX-111FC-1026-SM-5GZX1'</li><li>'GTEX-111FC-1126-SM-5GZWU'</li><li>'GTEX-111FC-1326-SM-5N9D9'</li><li>'GTEX-111FC-1426-SM-5N9C7'</li><li>'GTEX-111FC-1926-SM-5GZYC'</li><li>'GTEX-111FC-2026-SM-5GZYO'</li><li>'GTEX-111FC-2526-SM-5GZXU'</li><li>'GTEX-111FC-2926-SM-5GZY7'</li><li>'GTEX-111FC-3026-SM-5GZYJ'</li><li>'GTEX-111FC-3126-SM-5GZZ2'</li><li>'GTEX-111FC-3326-SM-5GZYV'</li><li>'GTEX-111VG-0004-SM-58Q85'</li><li>'GTEX-111VG-0004-SM-5DWSX'</li><li>'GTEX-111VG-0004-SM-6WBTS'</li><li>'GTEX-111VG-0008-SM-5Q5BG'</li><li>'GTEX-111VG-0326-SM-5GZX7'</li><li>'GTEX-111VG-0526-SM-5N9BW'</li><li>'GTEX-111VG-0726-SM-5GIDC'</li><li>'GTEX-111VG-1626-SM-5EGIO'</li><li>'GTEX-111VG-1926-SM-5GIDO'</li><li>'GTEX-111VG-2226-SM-5N9DU'</li><li>'GTEX-111VG-2326-SM-5N9BK'</li><li>'GTEX-111VG-2426-SM-5GZXD'</li><li>'GTEX-111VG-2526-SM-5GZXP'</li><li>'GTEX-111VG-2626-SM-5GZY2'</li><li>'GTEX-111YS-0004-SM-58Q7Y'</li><li>'GTEX-111YS-0004-SM-5DWSS'</li><li>'GTEX-111YS-0004-SM-6WBTN'</li><li>'GTEX-111YS-0006-SM-5NQBE'</li><li>'GTEX-111YS-0008-SM-5Q5BH'</li><li>'GTEX-111YS-0126-SM-5987T'</li><li>'GTEX-111YS-0326-SM-5GZZ3'</li><li>'GTEX-111YS-0426-SM-5987O'</li><li>'GTEX-111YS-0526-SM-5GZXJ'</li><li>'GTEX-111YS-0626-SM-5GZXV'</li><li>'GTEX-111YS-0726-SM-5GZY8'</li><li>'GTEX-111YS-0826-SM-5GZYK'</li><li>'GTEX-111YS-0926-SM-5EGGI'</li><li>'GTEX-111YS-1026-SM-5GZYE'</li><li>'GTEX-111YS-1126-SM-5GZYQ'</li><li>'GTEX-111YS-1226-SM-5EGGJ'</li><li>'GTEX-111YS-1326-SM-5EGGK'</li><li>'GTEX-111YS-1426-SM-5GID8'</li><li>'GTEX-111YS-1526-SM-5GZYW'</li><li>'GTEX-111YS-1626-SM-5GZZ9'</li><li>'GTEX-111YS-1726-SM-5GIED'</li><li>'GTEX-111YS-1826-SM-5GIEP'</li><li>'GTEX-111YS-1926-SM-5GICC'</li><li>'GTEX-111YS-2026-SM-5EGGL'</li><li>'GTEX-111YS-2126-SM-5EGGM'</li><li>'GTEX-111YS-2226-SM-5987P'</li><li>'GTEX-111YS-2326-SM-5987L'</li><li>'GTEX-111YS-2426-SM-5GZZQ'</li><li>'GTEX-1122O-0003-SM-5Q5DL'</li><li>'GTEX-1122O-0004-SM-58Q7O'</li><li>'GTEX-1122O-0004-SM-5DWSJ'</li><li>'GTEX-1122O-0004-SM-6WBTE'</li><li>'GTEX-1122O-0005-SM-5O99J'</li><li>'GTEX-1122O-0008-SM-5QGR2'</li><li>'GTEX-1122O-0126-SM-5GICA'</li><li>'GTEX-1122O-0226-SM-5N9DA'</li><li>'GTEX-1122O-0326-SM-5H124'</li><li>'GTEX-1122O-0426-SM-5H12G'</li><li>'GTEX-1122O-0526-SM-5N9DM'</li><li>'GTEX-1122O-0626-SM-5N9B9'</li><li>'GTEX-1122O-0726-SM-5GIEV'</li><li>'GTEX-1122O-0826-SM-5GICV'</li><li>'GTEX-1122O-0926-SM-5N9C9'</li><li>'GTEX-1122O-0926-SM-9YFLC'</li><li>'GTEX-1122O-1026-SM-9WYSF'</li><li>'GTEX-1122O-1126-SM-5NQ8X'</li><li>'GTEX-1122O-1226-SM-5H113'</li><li>'GTEX-1122O-1326-SM-5H11F'</li><li>'GTEX-1122O-1426-SM-5H11R'</li><li>'GTEX-1122O-1526-SM-5N9CL'</li><li>'GTEX-1122O-1626-SM-5N9CX'</li><li>'GTEX-1122O-1726-SM-5GICM'</li><li>'GTEX-1122O-1826-SM-5EGIP'</li><li>'GTEX-1122O-1926-SM-5EGIQ'</li><li>'GTEX-1122O-2026-SM-5NQ91'</li><li>'GTEX-1122O-2026-SM-9YFMG'</li><li>'GTEX-1122O-2126-SM-5EGIR'</li><li>'GTEX-1122O-2326-SM-9YFL9'</li><li>'GTEX-1122O-2426-SM-5GIDN'</li><li>'GTEX-1128S-0001-SM-58Q96'</li><li>'GTEX-1128S-0001-SM-5DWTV'</li><li>'GTEX-1128S-0001-SM-6WBUH'</li><li>'GTEX-1128S-0005-SM-5P9HI'</li><li>'GTEX-1128S-0008-SM-5Q5DP'</li><li>'GTEX-1128S-0126-SM-5H12S'</li><li>'GTEX-1128S-0326-SM-5GZZF'</li><li>'GTEX-1128S-0726-SM-5N9D6'</li><li>'GTEX-1128S-0826-SM-5GZZI'</li><li>'GTEX-1128S-0926-SM-5GZZU'</li><li>'GTEX-1128S-1426-SM-5N9EF'</li><li>'GTEX-1128S-1626-SM-5H12O'</li><li>'GTEX-1128S-2126-SM-5H12U'</li><li>'GTEX-1128S-2326-SM-5GZZY'</li><li>'GTEX-1128S-2426-SM-5H11B'</li><li>'GTEX-1128S-2526-SM-5H11N'</li><li>'GTEX-1128S-2626-SM-5H11Z'</li><li>'GTEX-1128S-2726-SM-5H12C'</li><li>'GTEX-1128S-2826-SM-5N9DI'</li><li>'GTEX-113IC-0001-SM-58Q97'</li><li>'GTEX-113IC-0001-SM-5DWSI'</li><li>'GTEX-113IC-0001-SM-DLIPU'</li><li>'GTEX-113IC-0006-SM-5NQ9C'</li><li>'GTEX-113IC-0008-SM-5QGRF'</li><li>'GTEX-113IC-0126-SM-5HL6T'</li><li>'GTEX-113IC-0226-SM-5HL5C'</li><li>'GTEX-113IC-0426-SM-5HL5O'</li><li>'GTEX-113IC-0526-SM-5HL76'</li><li>'GTEX-113JC-0001-SM-58Q7M'</li><li>'GTEX-113JC-0001-SM-5DWSH'</li><li>'GTEX-113JC-0001-SM-6WBTD'</li><li>'GTEX-113JC-0006-SM-5O997'</li><li>'GTEX-113JC-0008-SM-5QGR6'</li><li>'GTEX-113JC-0126-SM-5EGJW'</li><li>'GTEX-113JC-0726-SM-5GZZR'</li><li>'GTEX-113JC-0926-SM-5H114'</li><li>'GTEX-113JC-1026-SM-5H117'</li><li>'GTEX-113JC-1926-SM-5N9GE'</li><li>'GTEX-113JC-2226-SM-5EGJG'</li><li>'GTEX-113JC-2326-SM-5EQ4E'</li><li>'GTEX-113JC-2726-SM-5EGIS'</li><li>'GTEX-113JC-2826-SM-5EGIT'</li><li>'GTEX-117XS-0001-SM-58Q7J'</li><li>'GTEX-117XS-0001-SM-5DWSE'</li><li>'GTEX-117XS-0001-SM-6WBTA'</li><li>'GTEX-117XS-0005-SM-5PNU6'</li><li>'GTEX-117XS-0008-SM-5Q5DQ'</li><li>'GTEX-117XS-0011-R10a-SM-AHZ7G'</li><li>'GTEX-117XS-0426-SM-5GZZN'</li><li>'GTEX-117XS-0526-SM-5987Q'</li><li>'GTEX-117XS-0726-SM-5H131'</li><li>'GTEX-117XS-1926-SM-5GICO'</li><li>'GTEX-117XS-2026-SM-5GID1'</li><li>'GTEX-117XS-2226-SM-5GIE1'</li><li>'GTEX-117XS-2526-SM-5H11G'</li><li>'GTEX-117XS-2526-SM-CYKR4'</li><li>'GTEX-117XS-2726-SM-5N9BL'</li><li>'GTEX-117XS-2926-SM-5N9BX'</li><li>'GTEX-117XS-3026-SM-5N9CA'</li><li>'GTEX-117XS-3026-SM-CYKOR'</li><li>'GTEX-117XS-3126-SM-5GIDP'</li><li>'GTEX-117YW-0003-SM-58Q99'</li><li>⋯</li><li>'K-562-SM-2S1RI'</li><li>'K-562-SM-2TC5E'</li><li>'K-562-SM-2TC6R'</li><li>'K-562-SM-2TF4E'</li><li>'K-562-SM-2TF5R'</li><li>'K-562-SM-2TO4S'</li><li>'K-562-SM-2TWBO'</li><li>'K-562-SM-2XCBH'</li><li>'K-562-SM-2XCE8'</li><li>'K-562-SM-2XCFL'</li><li>'K-562-SM-2XU9E'</li><li>'K-562-SM-2XV5K'</li><li>'K-562-SM-2YUNN'</li><li>'K-562-SM-32PLC'</li><li>'K-562-SM-32PMP'</li><li>'K-562-SM-32QPQ'</li><li>'K-562-SM-33HC7'</li><li>'K-562-SM-3DB9M'</li><li>'K-562-SM-3GADY'</li><li>'K-562-SM-3GAFC'</li><li>'K-562-SM-3GIKB'</li><li>'K-562-SM-3GILO'</li><li>'K-562-SM-3K2BF'</li><li>'K-562-SM-3LK7S'</li><li>'K-562-SM-3MJHH'</li><li>'K-562-SM-3NB3I'</li><li>'K-562-SM-3NMAP'</li><li>'K-562-SM-3NMDG'</li><li>'K-562-SM-3P61Y'</li><li>'K-562-SM-43ISS'</li><li>'K-562-SM-43V9A'</li><li>'K-562-SM-46MWI'</li><li>'K-562-SM-47JYY'</li><li>'K-562-SM-48FEU'</li><li>'K-562-SM-48TE3'</li><li>'K-562-SM-48U2C'</li><li>'K-562-SM-4AD4F'</li><li>'K-562-SM-4AT3W'</li><li>'K-562-SM-4B66B'</li><li>'K-562-SM-4BONS'</li><li>'K-562-SM-4BRWK'</li><li>'K-562-SM-4DM4W'</li><li>'K-562-SM-4DXSD'</li><li>'K-562-SM-4EDPU'</li><li>'K-562-SM-4GICD'</li><li>'K-562-SM-4IHK7'</li><li>'K-562-SM-4JBIQ'</li><li>'K-562-SM-4KKZ9'</li><li>'K-562-SM-4LMI2'</li><li>'K-562-SM-4LVKX'</li><li>'K-562-SM-4M1XD'</li><li>'K-562-SM-4MVNA'</li><li>'K-562-SM-4OOR8'</li><li>'K-562-SM-4PQYN'</li><li>'K-562-SM-4QAR5'</li><li>'K-562-SM-4R59E'</li><li>'K-562-SM-4RGL3'</li><li>'K-562-SM-4RH3A'</li><li>'K-562-SM-4RTUY'</li><li>'K-562-SM-4SOHT'</li><li>'K-562-SM-4SOLN'</li><li>'K-562-SM-4T8CH'</li><li>'K-562-SM-4TT1D'</li><li>'K-562-SM-4TT6S'</li><li>'K-562-SM-4U82S'</li><li>'K-562-SM-4V81D'</li><li>'K-562-SM-4VBPT'</li><li>'K-562-SM-4VDQQ'</li><li>'K-562-SM-4W1Y5'</li><li>'K-562-SM-4WAWH'</li><li>'K-562-SM-4WKF1'</li><li>'K-562-SM-4WWAK'</li><li>'K-562-SM-4YCCQ'</li><li>'K-562-SM-51MQT'</li><li>'K-562-SM-57WAW'</li><li>'K-562-SM-59HJ4'</li><li>'K-562-SM-5CVL5'</li><li>'K-562-SM-5DUV6'</li><li>'K-562-SM-5E43B'</li><li>'K-562-SM-5GZWN'</li><li>'K-562-SM-5GZZE'</li><li>'K-562-SM-5HL4J'</li><li>'K-562-SM-5HL7A'</li><li>'K-562-SM-5IFEL'</li><li>'K-562-SM-5IFHC'</li><li>'K-562-SM-5IJB7'</li><li>'K-562-SM-5IJDX'</li><li>'K-562-SM-5J1MB'</li><li>'K-562-SM-5J2LX'</li><li>'K-562-SM-5K7TT'</li><li>'K-562-SM-5K7WK'</li><li>'K-562-SM-5K7ZB'</li><li>'K-562-SM-5KM2I'</li><li>'K-562-SM-5L3FO'</li><li>'K-562-SM-5LU2Y'</li><li>'K-562-SM-5LU8G'</li><li>'K-562-SM-5LZU1'</li><li>'K-562-SM-5LZWR'</li><li>'K-562-SM-5MR32'</li><li>'K-562-SM-5MR5S'</li><li>'K-562-SM-5N9B7'</li><li>'K-562-SM-5N9DX'</li><li>'K-562-SM-5NQ6M'</li><li>'K-562-SM-5NQ9D'</li><li>'K-562-SM-5O98J'</li><li>'K-562-SM-5O9BA'</li><li>'K-562-SM-5P9EQ'</li><li>'K-562-SM-5P9HH'</li><li>'K-562-SM-5PNU4'</li><li>'K-562-SM-5PNWU'</li><li>'K-562-SM-5Q59U'</li><li>'K-562-SM-5Q5CL'</li><li>'K-562-SM-5QGOR'</li><li>'K-562-SM-5RQHM'</li><li>'K-562-SM-66EY4'</li><li>'K-562-SM-686YS'</li><li>'K-562-SM-69LO5'</li><li>'K-562-SM-6AJ95'</li><li>'K-562-SM-6ETZI'</li><li>'K-562-SM-6LLH2'</li><li>'K-562-SM-6LPHZ'</li><li>'K-562-SM-6M461'</li><li>'K-562-SM-6PAL1'</li><li>'K-562-SM-7189B'</li><li>'K-562-SM-72D4W'</li><li>'K-562-SM-731AF'</li><li>'K-562-SM-731D6'</li><li>'K-562-SM-73KTR'</li><li>'K-562-SM-73KWI'</li><li>'K-562-SM-7938F'</li><li>'K-562-SM-793B6'</li><li>'K-562-SM-79OJI'</li><li>'K-562-SM-79OM9'</li><li>'K-562-SM-7DHKI'</li><li>'K-562-SM-7DUEE'</li><li>'K-562-SM-7EPG4'</li><li>'K-562-SM-7EWCW'</li><li>'K-562-SM-7IGL8'</li><li>'K-562-SM-7IGNY'</li><li>'K-562-SM-7KFRB'</li><li>'K-562-SM-7KUDR'</li><li>'K-562-SM-7KUKI'</li><li>'K-562-SM-7LG4B'</li><li>'K-562-SM-7LT88'</li><li>'K-562-SM-7MGVH'</li><li>'K-562-SM-7MKF2'</li><li>'K-562-SM-7MXT1'</li><li>'K-562-SM-7P8RB'</li><li>'K-562-SM-7RHFN'</li><li>'K-562-SM-9HWNP'</li><li>'K-562-SM-9HWNQ'</li><li>'K-562-SM-9HWNT'</li><li>'K-562-SM-9HWNU'</li><li>'K-562-SM-9HWNV'</li><li>'K-562-SM-9HWNW'</li><li>'K-562-SM-9HWO3'</li><li>'K-562-SM-9HWO4'</li><li>'K-562-SM-9HWOB'</li><li>'K-562-SM-9HWOD'</li><li>'K-562-SM-9HWOI'</li><li>'K-562-SM-9HWOK'</li><li>'K-562-SM-9HWOP'</li><li>'K-562-SM-9HWOQ'</li><li>'K-562-SM-9HWOR'</li><li>'K-562-SM-9HWOX'</li><li>'K-562-SM-9HWOY'</li><li>'K-562-SM-9HWP4'</li><li>'K-562-SM-9HWP5'</li><li>'K-562-SM-9HWP6'</li><li>'K-562-SM-9HWP7'</li><li>'K-562-SM-9HWP8'</li><li>'K-562-SM-9HWP9'</li><li>'K-562-SM-9HWPF'</li><li>'K-562-SM-9HWPM'</li><li>'K-562-SM-9HWPN'</li><li>'K-562-SM-9HWPV'</li><li>'K-562-SM-9HWPW'</li><li>'K-562-SM-9HWPX'</li><li>'K-562-SM-9HWPY'</li><li>'K-562-SM-9HWPZ'</li><li>'K-562-SM-9HWQ7'</li><li>'K-562-SM-9HWQ9'</li><li>'K-562-SM-9HWQF'</li><li>'K-562-SM-9HWQG'</li><li>'K-562-SM-9JGFK'</li><li>'K-562-SM-9MQJO'</li><li>'K-562-SM-CJI1Q'</li><li>'K-562-SM-CKZN7'</li><li>'K-562-SM-E9EYN'</li><li>'K-562-SM-E9EYO'</li><li>'K-562-SM-E9EYP'</li><li>'K-562-SM-E9EYT'</li><li>'K-562-SM-E9EYU'</li><li>'K-562-SM-E9EYV'</li><li>'K-562-SM-E9EYZ'</li><li>'K-562-SM-E9EZC'</li><li>'K-562-SM-E9EZI'</li><li>'K-562-SM-E9EZO'</li><li>'K-562-SM-E9EZT'</li><li>'K-562-SM-E9EZZ'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\textbf{GTEX-1117F-0226-SM-5GZZ7:} GTEX-1117F-0226-SM-5GZZ7\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'GTEX-1117F-0003-SM-58Q7G'\n",
       "\\item 'GTEX-1117F-0003-SM-5DWSB'\n",
       "\\item 'GTEX-1117F-0003-SM-6WBT7'\n",
       "\\item 'GTEX-1117F-0011-R10a-SM-AHZ7F'\n",
       "\\item 'GTEX-1117F-0011-R10b-SM-CYKQ8'\n",
       "\\item 'GTEX-1117F-0226-SM-5GZZ7'\n",
       "\\item 'GTEX-1117F-0426-SM-5EGHI'\n",
       "\\item 'GTEX-1117F-0526-SM-5EGHJ'\n",
       "\\item 'GTEX-1117F-0626-SM-5N9CS'\n",
       "\\item 'GTEX-1117F-0726-SM-5GIEN'\n",
       "\\item 'GTEX-1117F-1326-SM-5EGHH'\n",
       "\\item 'GTEX-1117F-2226-SM-5N9CH'\n",
       "\\item 'GTEX-1117F-2426-SM-5EGGH'\n",
       "\\item 'GTEX-1117F-2526-SM-5GZY6'\n",
       "\\item 'GTEX-1117F-2826-SM-5GZXL'\n",
       "\\item 'GTEX-1117F-2926-SM-5GZYI'\n",
       "\\item 'GTEX-1117F-3026-SM-5GZYU'\n",
       "\\item 'GTEX-1117F-3226-SM-5N9CT'\n",
       "\\item 'GTEX-111CU-0003-SM-58Q95'\n",
       "\\item 'GTEX-111CU-0003-SM-5DWTR'\n",
       "\\item 'GTEX-111CU-0003-SM-6WBUD'\n",
       "\\item 'GTEX-111CU-0126-SM-5GZWZ'\n",
       "\\item 'GTEX-111CU-0226-SM-5GZXC'\n",
       "\\item 'GTEX-111CU-0326-SM-5GZXO'\n",
       "\\item 'GTEX-111CU-0426-SM-5GZY1'\n",
       "\\item 'GTEX-111CU-0526-SM-5EGHK'\n",
       "\\item 'GTEX-111CU-0626-SM-5EGHL'\n",
       "\\item 'GTEX-111CU-0726-SM-5GZYD'\n",
       "\\item 'GTEX-111CU-0826-SM-5EGIJ'\n",
       "\\item 'GTEX-111CU-0926-SM-5EGIK'\n",
       "\\item 'GTEX-111CU-1026-SM-5EGIL'\n",
       "\\item 'GTEX-111CU-1126-SM-5EGIM'\n",
       "\\item 'GTEX-111CU-1226-SM-5EGIN'\n",
       "\\item 'GTEX-111CU-1326-SM-5NQ8L'\n",
       "\\item 'GTEX-111CU-1426-SM-5GZYP'\n",
       "\\item 'GTEX-111CU-1526-SM-5N9FS'\n",
       "\\item 'GTEX-111CU-1726-SM-5EGHM'\n",
       "\\item 'GTEX-111CU-1826-SM-5GZYN'\n",
       "\\item 'GTEX-111CU-1926-SM-5GZYZ'\n",
       "\\item 'GTEX-111CU-2026-SM-5GZZC'\n",
       "\\item 'GTEX-111CU-2226-SM-5N9G5'\n",
       "\\item 'GTEX-111FC-0001-SM-58Q7U'\n",
       "\\item 'GTEX-111FC-0001-SM-5DWSO'\n",
       "\\item 'GTEX-111FC-0001-SM-6WBTJ'\n",
       "\\item 'GTEX-111FC-0011-R10a-SM-AHZ7K'\n",
       "\\item 'GTEX-111FC-0011-R10a-SM-CYKQ9'\n",
       "\\item 'GTEX-111FC-0126-SM-5N9DL'\n",
       "\\item 'GTEX-111FC-0226-SM-5N9B8'\n",
       "\\item 'GTEX-111FC-0326-SM-5GZZ1'\n",
       "\\item 'GTEX-111FC-0426-SM-5N9CV'\n",
       "\\item 'GTEX-111FC-0526-SM-5GZZ8'\n",
       "\\item 'GTEX-111FC-0626-SM-5N9CU'\n",
       "\\item 'GTEX-111FC-0826-SM-5GZWO'\n",
       "\\item 'GTEX-111FC-0826-SM-CYKRT'\n",
       "\\item 'GTEX-111FC-1026-SM-5GZX1'\n",
       "\\item 'GTEX-111FC-1126-SM-5GZWU'\n",
       "\\item 'GTEX-111FC-1326-SM-5N9D9'\n",
       "\\item 'GTEX-111FC-1426-SM-5N9C7'\n",
       "\\item 'GTEX-111FC-1926-SM-5GZYC'\n",
       "\\item 'GTEX-111FC-2026-SM-5GZYO'\n",
       "\\item 'GTEX-111FC-2526-SM-5GZXU'\n",
       "\\item 'GTEX-111FC-2926-SM-5GZY7'\n",
       "\\item 'GTEX-111FC-3026-SM-5GZYJ'\n",
       "\\item 'GTEX-111FC-3126-SM-5GZZ2'\n",
       "\\item 'GTEX-111FC-3326-SM-5GZYV'\n",
       "\\item 'GTEX-111VG-0004-SM-58Q85'\n",
       "\\item 'GTEX-111VG-0004-SM-5DWSX'\n",
       "\\item 'GTEX-111VG-0004-SM-6WBTS'\n",
       "\\item 'GTEX-111VG-0008-SM-5Q5BG'\n",
       "\\item 'GTEX-111VG-0326-SM-5GZX7'\n",
       "\\item 'GTEX-111VG-0526-SM-5N9BW'\n",
       "\\item 'GTEX-111VG-0726-SM-5GIDC'\n",
       "\\item 'GTEX-111VG-1626-SM-5EGIO'\n",
       "\\item 'GTEX-111VG-1926-SM-5GIDO'\n",
       "\\item 'GTEX-111VG-2226-SM-5N9DU'\n",
       "\\item 'GTEX-111VG-2326-SM-5N9BK'\n",
       "\\item 'GTEX-111VG-2426-SM-5GZXD'\n",
       "\\item 'GTEX-111VG-2526-SM-5GZXP'\n",
       "\\item 'GTEX-111VG-2626-SM-5GZY2'\n",
       "\\item 'GTEX-111YS-0004-SM-58Q7Y'\n",
       "\\item 'GTEX-111YS-0004-SM-5DWSS'\n",
       "\\item 'GTEX-111YS-0004-SM-6WBTN'\n",
       "\\item 'GTEX-111YS-0006-SM-5NQBE'\n",
       "\\item 'GTEX-111YS-0008-SM-5Q5BH'\n",
       "\\item 'GTEX-111YS-0126-SM-5987T'\n",
       "\\item 'GTEX-111YS-0326-SM-5GZZ3'\n",
       "\\item 'GTEX-111YS-0426-SM-5987O'\n",
       "\\item 'GTEX-111YS-0526-SM-5GZXJ'\n",
       "\\item 'GTEX-111YS-0626-SM-5GZXV'\n",
       "\\item 'GTEX-111YS-0726-SM-5GZY8'\n",
       "\\item 'GTEX-111YS-0826-SM-5GZYK'\n",
       "\\item 'GTEX-111YS-0926-SM-5EGGI'\n",
       "\\item 'GTEX-111YS-1026-SM-5GZYE'\n",
       "\\item 'GTEX-111YS-1126-SM-5GZYQ'\n",
       "\\item 'GTEX-111YS-1226-SM-5EGGJ'\n",
       "\\item 'GTEX-111YS-1326-SM-5EGGK'\n",
       "\\item 'GTEX-111YS-1426-SM-5GID8'\n",
       "\\item 'GTEX-111YS-1526-SM-5GZYW'\n",
       "\\item 'GTEX-111YS-1626-SM-5GZZ9'\n",
       "\\item 'GTEX-111YS-1726-SM-5GIED'\n",
       "\\item 'GTEX-111YS-1826-SM-5GIEP'\n",
       "\\item 'GTEX-111YS-1926-SM-5GICC'\n",
       "\\item 'GTEX-111YS-2026-SM-5EGGL'\n",
       "\\item 'GTEX-111YS-2126-SM-5EGGM'\n",
       "\\item 'GTEX-111YS-2226-SM-5987P'\n",
       "\\item 'GTEX-111YS-2326-SM-5987L'\n",
       "\\item 'GTEX-111YS-2426-SM-5GZZQ'\n",
       "\\item 'GTEX-1122O-0003-SM-5Q5DL'\n",
       "\\item 'GTEX-1122O-0004-SM-58Q7O'\n",
       "\\item 'GTEX-1122O-0004-SM-5DWSJ'\n",
       "\\item 'GTEX-1122O-0004-SM-6WBTE'\n",
       "\\item 'GTEX-1122O-0005-SM-5O99J'\n",
       "\\item 'GTEX-1122O-0008-SM-5QGR2'\n",
       "\\item 'GTEX-1122O-0126-SM-5GICA'\n",
       "\\item 'GTEX-1122O-0226-SM-5N9DA'\n",
       "\\item 'GTEX-1122O-0326-SM-5H124'\n",
       "\\item 'GTEX-1122O-0426-SM-5H12G'\n",
       "\\item 'GTEX-1122O-0526-SM-5N9DM'\n",
       "\\item 'GTEX-1122O-0626-SM-5N9B9'\n",
       "\\item 'GTEX-1122O-0726-SM-5GIEV'\n",
       "\\item 'GTEX-1122O-0826-SM-5GICV'\n",
       "\\item 'GTEX-1122O-0926-SM-5N9C9'\n",
       "\\item 'GTEX-1122O-0926-SM-9YFLC'\n",
       "\\item 'GTEX-1122O-1026-SM-9WYSF'\n",
       "\\item 'GTEX-1122O-1126-SM-5NQ8X'\n",
       "\\item 'GTEX-1122O-1226-SM-5H113'\n",
       "\\item 'GTEX-1122O-1326-SM-5H11F'\n",
       "\\item 'GTEX-1122O-1426-SM-5H11R'\n",
       "\\item 'GTEX-1122O-1526-SM-5N9CL'\n",
       "\\item 'GTEX-1122O-1626-SM-5N9CX'\n",
       "\\item 'GTEX-1122O-1726-SM-5GICM'\n",
       "\\item 'GTEX-1122O-1826-SM-5EGIP'\n",
       "\\item 'GTEX-1122O-1926-SM-5EGIQ'\n",
       "\\item 'GTEX-1122O-2026-SM-5NQ91'\n",
       "\\item 'GTEX-1122O-2026-SM-9YFMG'\n",
       "\\item 'GTEX-1122O-2126-SM-5EGIR'\n",
       "\\item 'GTEX-1122O-2326-SM-9YFL9'\n",
       "\\item 'GTEX-1122O-2426-SM-5GIDN'\n",
       "\\item 'GTEX-1128S-0001-SM-58Q96'\n",
       "\\item 'GTEX-1128S-0001-SM-5DWTV'\n",
       "\\item 'GTEX-1128S-0001-SM-6WBUH'\n",
       "\\item 'GTEX-1128S-0005-SM-5P9HI'\n",
       "\\item 'GTEX-1128S-0008-SM-5Q5DP'\n",
       "\\item 'GTEX-1128S-0126-SM-5H12S'\n",
       "\\item 'GTEX-1128S-0326-SM-5GZZF'\n",
       "\\item 'GTEX-1128S-0726-SM-5N9D6'\n",
       "\\item 'GTEX-1128S-0826-SM-5GZZI'\n",
       "\\item 'GTEX-1128S-0926-SM-5GZZU'\n",
       "\\item 'GTEX-1128S-1426-SM-5N9EF'\n",
       "\\item 'GTEX-1128S-1626-SM-5H12O'\n",
       "\\item 'GTEX-1128S-2126-SM-5H12U'\n",
       "\\item 'GTEX-1128S-2326-SM-5GZZY'\n",
       "\\item 'GTEX-1128S-2426-SM-5H11B'\n",
       "\\item 'GTEX-1128S-2526-SM-5H11N'\n",
       "\\item 'GTEX-1128S-2626-SM-5H11Z'\n",
       "\\item 'GTEX-1128S-2726-SM-5H12C'\n",
       "\\item 'GTEX-1128S-2826-SM-5N9DI'\n",
       "\\item 'GTEX-113IC-0001-SM-58Q97'\n",
       "\\item 'GTEX-113IC-0001-SM-5DWSI'\n",
       "\\item 'GTEX-113IC-0001-SM-DLIPU'\n",
       "\\item 'GTEX-113IC-0006-SM-5NQ9C'\n",
       "\\item 'GTEX-113IC-0008-SM-5QGRF'\n",
       "\\item 'GTEX-113IC-0126-SM-5HL6T'\n",
       "\\item 'GTEX-113IC-0226-SM-5HL5C'\n",
       "\\item 'GTEX-113IC-0426-SM-5HL5O'\n",
       "\\item 'GTEX-113IC-0526-SM-5HL76'\n",
       "\\item 'GTEX-113JC-0001-SM-58Q7M'\n",
       "\\item 'GTEX-113JC-0001-SM-5DWSH'\n",
       "\\item 'GTEX-113JC-0001-SM-6WBTD'\n",
       "\\item 'GTEX-113JC-0006-SM-5O997'\n",
       "\\item 'GTEX-113JC-0008-SM-5QGR6'\n",
       "\\item 'GTEX-113JC-0126-SM-5EGJW'\n",
       "\\item 'GTEX-113JC-0726-SM-5GZZR'\n",
       "\\item 'GTEX-113JC-0926-SM-5H114'\n",
       "\\item 'GTEX-113JC-1026-SM-5H117'\n",
       "\\item 'GTEX-113JC-1926-SM-5N9GE'\n",
       "\\item 'GTEX-113JC-2226-SM-5EGJG'\n",
       "\\item 'GTEX-113JC-2326-SM-5EQ4E'\n",
       "\\item 'GTEX-113JC-2726-SM-5EGIS'\n",
       "\\item 'GTEX-113JC-2826-SM-5EGIT'\n",
       "\\item 'GTEX-117XS-0001-SM-58Q7J'\n",
       "\\item 'GTEX-117XS-0001-SM-5DWSE'\n",
       "\\item 'GTEX-117XS-0001-SM-6WBTA'\n",
       "\\item 'GTEX-117XS-0005-SM-5PNU6'\n",
       "\\item 'GTEX-117XS-0008-SM-5Q5DQ'\n",
       "\\item 'GTEX-117XS-0011-R10a-SM-AHZ7G'\n",
       "\\item 'GTEX-117XS-0426-SM-5GZZN'\n",
       "\\item 'GTEX-117XS-0526-SM-5987Q'\n",
       "\\item 'GTEX-117XS-0726-SM-5H131'\n",
       "\\item 'GTEX-117XS-1926-SM-5GICO'\n",
       "\\item 'GTEX-117XS-2026-SM-5GID1'\n",
       "\\item 'GTEX-117XS-2226-SM-5GIE1'\n",
       "\\item 'GTEX-117XS-2526-SM-5H11G'\n",
       "\\item 'GTEX-117XS-2526-SM-CYKR4'\n",
       "\\item 'GTEX-117XS-2726-SM-5N9BL'\n",
       "\\item 'GTEX-117XS-2926-SM-5N9BX'\n",
       "\\item 'GTEX-117XS-3026-SM-5N9CA'\n",
       "\\item 'GTEX-117XS-3026-SM-CYKOR'\n",
       "\\item 'GTEX-117XS-3126-SM-5GIDP'\n",
       "\\item 'GTEX-117YW-0003-SM-58Q99'\n",
       "\\item ⋯\n",
       "\\item 'K-562-SM-2S1RI'\n",
       "\\item 'K-562-SM-2TC5E'\n",
       "\\item 'K-562-SM-2TC6R'\n",
       "\\item 'K-562-SM-2TF4E'\n",
       "\\item 'K-562-SM-2TF5R'\n",
       "\\item 'K-562-SM-2TO4S'\n",
       "\\item 'K-562-SM-2TWBO'\n",
       "\\item 'K-562-SM-2XCBH'\n",
       "\\item 'K-562-SM-2XCE8'\n",
       "\\item 'K-562-SM-2XCFL'\n",
       "\\item 'K-562-SM-2XU9E'\n",
       "\\item 'K-562-SM-2XV5K'\n",
       "\\item 'K-562-SM-2YUNN'\n",
       "\\item 'K-562-SM-32PLC'\n",
       "\\item 'K-562-SM-32PMP'\n",
       "\\item 'K-562-SM-32QPQ'\n",
       "\\item 'K-562-SM-33HC7'\n",
       "\\item 'K-562-SM-3DB9M'\n",
       "\\item 'K-562-SM-3GADY'\n",
       "\\item 'K-562-SM-3GAFC'\n",
       "\\item 'K-562-SM-3GIKB'\n",
       "\\item 'K-562-SM-3GILO'\n",
       "\\item 'K-562-SM-3K2BF'\n",
       "\\item 'K-562-SM-3LK7S'\n",
       "\\item 'K-562-SM-3MJHH'\n",
       "\\item 'K-562-SM-3NB3I'\n",
       "\\item 'K-562-SM-3NMAP'\n",
       "\\item 'K-562-SM-3NMDG'\n",
       "\\item 'K-562-SM-3P61Y'\n",
       "\\item 'K-562-SM-43ISS'\n",
       "\\item 'K-562-SM-43V9A'\n",
       "\\item 'K-562-SM-46MWI'\n",
       "\\item 'K-562-SM-47JYY'\n",
       "\\item 'K-562-SM-48FEU'\n",
       "\\item 'K-562-SM-48TE3'\n",
       "\\item 'K-562-SM-48U2C'\n",
       "\\item 'K-562-SM-4AD4F'\n",
       "\\item 'K-562-SM-4AT3W'\n",
       "\\item 'K-562-SM-4B66B'\n",
       "\\item 'K-562-SM-4BONS'\n",
       "\\item 'K-562-SM-4BRWK'\n",
       "\\item 'K-562-SM-4DM4W'\n",
       "\\item 'K-562-SM-4DXSD'\n",
       "\\item 'K-562-SM-4EDPU'\n",
       "\\item 'K-562-SM-4GICD'\n",
       "\\item 'K-562-SM-4IHK7'\n",
       "\\item 'K-562-SM-4JBIQ'\n",
       "\\item 'K-562-SM-4KKZ9'\n",
       "\\item 'K-562-SM-4LMI2'\n",
       "\\item 'K-562-SM-4LVKX'\n",
       "\\item 'K-562-SM-4M1XD'\n",
       "\\item 'K-562-SM-4MVNA'\n",
       "\\item 'K-562-SM-4OOR8'\n",
       "\\item 'K-562-SM-4PQYN'\n",
       "\\item 'K-562-SM-4QAR5'\n",
       "\\item 'K-562-SM-4R59E'\n",
       "\\item 'K-562-SM-4RGL3'\n",
       "\\item 'K-562-SM-4RH3A'\n",
       "\\item 'K-562-SM-4RTUY'\n",
       "\\item 'K-562-SM-4SOHT'\n",
       "\\item 'K-562-SM-4SOLN'\n",
       "\\item 'K-562-SM-4T8CH'\n",
       "\\item 'K-562-SM-4TT1D'\n",
       "\\item 'K-562-SM-4TT6S'\n",
       "\\item 'K-562-SM-4U82S'\n",
       "\\item 'K-562-SM-4V81D'\n",
       "\\item 'K-562-SM-4VBPT'\n",
       "\\item 'K-562-SM-4VDQQ'\n",
       "\\item 'K-562-SM-4W1Y5'\n",
       "\\item 'K-562-SM-4WAWH'\n",
       "\\item 'K-562-SM-4WKF1'\n",
       "\\item 'K-562-SM-4WWAK'\n",
       "\\item 'K-562-SM-4YCCQ'\n",
       "\\item 'K-562-SM-51MQT'\n",
       "\\item 'K-562-SM-57WAW'\n",
       "\\item 'K-562-SM-59HJ4'\n",
       "\\item 'K-562-SM-5CVL5'\n",
       "\\item 'K-562-SM-5DUV6'\n",
       "\\item 'K-562-SM-5E43B'\n",
       "\\item 'K-562-SM-5GZWN'\n",
       "\\item 'K-562-SM-5GZZE'\n",
       "\\item 'K-562-SM-5HL4J'\n",
       "\\item 'K-562-SM-5HL7A'\n",
       "\\item 'K-562-SM-5IFEL'\n",
       "\\item 'K-562-SM-5IFHC'\n",
       "\\item 'K-562-SM-5IJB7'\n",
       "\\item 'K-562-SM-5IJDX'\n",
       "\\item 'K-562-SM-5J1MB'\n",
       "\\item 'K-562-SM-5J2LX'\n",
       "\\item 'K-562-SM-5K7TT'\n",
       "\\item 'K-562-SM-5K7WK'\n",
       "\\item 'K-562-SM-5K7ZB'\n",
       "\\item 'K-562-SM-5KM2I'\n",
       "\\item 'K-562-SM-5L3FO'\n",
       "\\item 'K-562-SM-5LU2Y'\n",
       "\\item 'K-562-SM-5LU8G'\n",
       "\\item 'K-562-SM-5LZU1'\n",
       "\\item 'K-562-SM-5LZWR'\n",
       "\\item 'K-562-SM-5MR32'\n",
       "\\item 'K-562-SM-5MR5S'\n",
       "\\item 'K-562-SM-5N9B7'\n",
       "\\item 'K-562-SM-5N9DX'\n",
       "\\item 'K-562-SM-5NQ6M'\n",
       "\\item 'K-562-SM-5NQ9D'\n",
       "\\item 'K-562-SM-5O98J'\n",
       "\\item 'K-562-SM-5O9BA'\n",
       "\\item 'K-562-SM-5P9EQ'\n",
       "\\item 'K-562-SM-5P9HH'\n",
       "\\item 'K-562-SM-5PNU4'\n",
       "\\item 'K-562-SM-5PNWU'\n",
       "\\item 'K-562-SM-5Q59U'\n",
       "\\item 'K-562-SM-5Q5CL'\n",
       "\\item 'K-562-SM-5QGOR'\n",
       "\\item 'K-562-SM-5RQHM'\n",
       "\\item 'K-562-SM-66EY4'\n",
       "\\item 'K-562-SM-686YS'\n",
       "\\item 'K-562-SM-69LO5'\n",
       "\\item 'K-562-SM-6AJ95'\n",
       "\\item 'K-562-SM-6ETZI'\n",
       "\\item 'K-562-SM-6LLH2'\n",
       "\\item 'K-562-SM-6LPHZ'\n",
       "\\item 'K-562-SM-6M461'\n",
       "\\item 'K-562-SM-6PAL1'\n",
       "\\item 'K-562-SM-7189B'\n",
       "\\item 'K-562-SM-72D4W'\n",
       "\\item 'K-562-SM-731AF'\n",
       "\\item 'K-562-SM-731D6'\n",
       "\\item 'K-562-SM-73KTR'\n",
       "\\item 'K-562-SM-73KWI'\n",
       "\\item 'K-562-SM-7938F'\n",
       "\\item 'K-562-SM-793B6'\n",
       "\\item 'K-562-SM-79OJI'\n",
       "\\item 'K-562-SM-79OM9'\n",
       "\\item 'K-562-SM-7DHKI'\n",
       "\\item 'K-562-SM-7DUEE'\n",
       "\\item 'K-562-SM-7EPG4'\n",
       "\\item 'K-562-SM-7EWCW'\n",
       "\\item 'K-562-SM-7IGL8'\n",
       "\\item 'K-562-SM-7IGNY'\n",
       "\\item 'K-562-SM-7KFRB'\n",
       "\\item 'K-562-SM-7KUDR'\n",
       "\\item 'K-562-SM-7KUKI'\n",
       "\\item 'K-562-SM-7LG4B'\n",
       "\\item 'K-562-SM-7LT88'\n",
       "\\item 'K-562-SM-7MGVH'\n",
       "\\item 'K-562-SM-7MKF2'\n",
       "\\item 'K-562-SM-7MXT1'\n",
       "\\item 'K-562-SM-7P8RB'\n",
       "\\item 'K-562-SM-7RHFN'\n",
       "\\item 'K-562-SM-9HWNP'\n",
       "\\item 'K-562-SM-9HWNQ'\n",
       "\\item 'K-562-SM-9HWNT'\n",
       "\\item 'K-562-SM-9HWNU'\n",
       "\\item 'K-562-SM-9HWNV'\n",
       "\\item 'K-562-SM-9HWNW'\n",
       "\\item 'K-562-SM-9HWO3'\n",
       "\\item 'K-562-SM-9HWO4'\n",
       "\\item 'K-562-SM-9HWOB'\n",
       "\\item 'K-562-SM-9HWOD'\n",
       "\\item 'K-562-SM-9HWOI'\n",
       "\\item 'K-562-SM-9HWOK'\n",
       "\\item 'K-562-SM-9HWOP'\n",
       "\\item 'K-562-SM-9HWOQ'\n",
       "\\item 'K-562-SM-9HWOR'\n",
       "\\item 'K-562-SM-9HWOX'\n",
       "\\item 'K-562-SM-9HWOY'\n",
       "\\item 'K-562-SM-9HWP4'\n",
       "\\item 'K-562-SM-9HWP5'\n",
       "\\item 'K-562-SM-9HWP6'\n",
       "\\item 'K-562-SM-9HWP7'\n",
       "\\item 'K-562-SM-9HWP8'\n",
       "\\item 'K-562-SM-9HWP9'\n",
       "\\item 'K-562-SM-9HWPF'\n",
       "\\item 'K-562-SM-9HWPM'\n",
       "\\item 'K-562-SM-9HWPN'\n",
       "\\item 'K-562-SM-9HWPV'\n",
       "\\item 'K-562-SM-9HWPW'\n",
       "\\item 'K-562-SM-9HWPX'\n",
       "\\item 'K-562-SM-9HWPY'\n",
       "\\item 'K-562-SM-9HWPZ'\n",
       "\\item 'K-562-SM-9HWQ7'\n",
       "\\item 'K-562-SM-9HWQ9'\n",
       "\\item 'K-562-SM-9HWQF'\n",
       "\\item 'K-562-SM-9HWQG'\n",
       "\\item 'K-562-SM-9JGFK'\n",
       "\\item 'K-562-SM-9MQJO'\n",
       "\\item 'K-562-SM-CJI1Q'\n",
       "\\item 'K-562-SM-CKZN7'\n",
       "\\item 'K-562-SM-E9EYN'\n",
       "\\item 'K-562-SM-E9EYO'\n",
       "\\item 'K-562-SM-E9EYP'\n",
       "\\item 'K-562-SM-E9EYT'\n",
       "\\item 'K-562-SM-E9EYU'\n",
       "\\item 'K-562-SM-E9EYV'\n",
       "\\item 'K-562-SM-E9EYZ'\n",
       "\\item 'K-562-SM-E9EZC'\n",
       "\\item 'K-562-SM-E9EZI'\n",
       "\\item 'K-562-SM-E9EZO'\n",
       "\\item 'K-562-SM-E9EZT'\n",
       "\\item 'K-562-SM-E9EZZ'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "**GTEX-1117F-0226-SM-5GZZ7:** GTEX-1117F-0226-SM-5GZZ7\n",
       "**Levels**: 1. 'GTEX-1117F-0003-SM-58Q7G'\n",
       "2. 'GTEX-1117F-0003-SM-5DWSB'\n",
       "3. 'GTEX-1117F-0003-SM-6WBT7'\n",
       "4. 'GTEX-1117F-0011-R10a-SM-AHZ7F'\n",
       "5. 'GTEX-1117F-0011-R10b-SM-CYKQ8'\n",
       "6. 'GTEX-1117F-0226-SM-5GZZ7'\n",
       "7. 'GTEX-1117F-0426-SM-5EGHI'\n",
       "8. 'GTEX-1117F-0526-SM-5EGHJ'\n",
       "9. 'GTEX-1117F-0626-SM-5N9CS'\n",
       "10. 'GTEX-1117F-0726-SM-5GIEN'\n",
       "11. 'GTEX-1117F-1326-SM-5EGHH'\n",
       "12. 'GTEX-1117F-2226-SM-5N9CH'\n",
       "13. 'GTEX-1117F-2426-SM-5EGGH'\n",
       "14. 'GTEX-1117F-2526-SM-5GZY6'\n",
       "15. 'GTEX-1117F-2826-SM-5GZXL'\n",
       "16. 'GTEX-1117F-2926-SM-5GZYI'\n",
       "17. 'GTEX-1117F-3026-SM-5GZYU'\n",
       "18. 'GTEX-1117F-3226-SM-5N9CT'\n",
       "19. 'GTEX-111CU-0003-SM-58Q95'\n",
       "20. 'GTEX-111CU-0003-SM-5DWTR'\n",
       "21. 'GTEX-111CU-0003-SM-6WBUD'\n",
       "22. 'GTEX-111CU-0126-SM-5GZWZ'\n",
       "23. 'GTEX-111CU-0226-SM-5GZXC'\n",
       "24. 'GTEX-111CU-0326-SM-5GZXO'\n",
       "25. 'GTEX-111CU-0426-SM-5GZY1'\n",
       "26. 'GTEX-111CU-0526-SM-5EGHK'\n",
       "27. 'GTEX-111CU-0626-SM-5EGHL'\n",
       "28. 'GTEX-111CU-0726-SM-5GZYD'\n",
       "29. 'GTEX-111CU-0826-SM-5EGIJ'\n",
       "30. 'GTEX-111CU-0926-SM-5EGIK'\n",
       "31. 'GTEX-111CU-1026-SM-5EGIL'\n",
       "32. 'GTEX-111CU-1126-SM-5EGIM'\n",
       "33. 'GTEX-111CU-1226-SM-5EGIN'\n",
       "34. 'GTEX-111CU-1326-SM-5NQ8L'\n",
       "35. 'GTEX-111CU-1426-SM-5GZYP'\n",
       "36. 'GTEX-111CU-1526-SM-5N9FS'\n",
       "37. 'GTEX-111CU-1726-SM-5EGHM'\n",
       "38. 'GTEX-111CU-1826-SM-5GZYN'\n",
       "39. 'GTEX-111CU-1926-SM-5GZYZ'\n",
       "40. 'GTEX-111CU-2026-SM-5GZZC'\n",
       "41. 'GTEX-111CU-2226-SM-5N9G5'\n",
       "42. 'GTEX-111FC-0001-SM-58Q7U'\n",
       "43. 'GTEX-111FC-0001-SM-5DWSO'\n",
       "44. 'GTEX-111FC-0001-SM-6WBTJ'\n",
       "45. 'GTEX-111FC-0011-R10a-SM-AHZ7K'\n",
       "46. 'GTEX-111FC-0011-R10a-SM-CYKQ9'\n",
       "47. 'GTEX-111FC-0126-SM-5N9DL'\n",
       "48. 'GTEX-111FC-0226-SM-5N9B8'\n",
       "49. 'GTEX-111FC-0326-SM-5GZZ1'\n",
       "50. 'GTEX-111FC-0426-SM-5N9CV'\n",
       "51. 'GTEX-111FC-0526-SM-5GZZ8'\n",
       "52. 'GTEX-111FC-0626-SM-5N9CU'\n",
       "53. 'GTEX-111FC-0826-SM-5GZWO'\n",
       "54. 'GTEX-111FC-0826-SM-CYKRT'\n",
       "55. 'GTEX-111FC-1026-SM-5GZX1'\n",
       "56. 'GTEX-111FC-1126-SM-5GZWU'\n",
       "57. 'GTEX-111FC-1326-SM-5N9D9'\n",
       "58. 'GTEX-111FC-1426-SM-5N9C7'\n",
       "59. 'GTEX-111FC-1926-SM-5GZYC'\n",
       "60. 'GTEX-111FC-2026-SM-5GZYO'\n",
       "61. 'GTEX-111FC-2526-SM-5GZXU'\n",
       "62. 'GTEX-111FC-2926-SM-5GZY7'\n",
       "63. 'GTEX-111FC-3026-SM-5GZYJ'\n",
       "64. 'GTEX-111FC-3126-SM-5GZZ2'\n",
       "65. 'GTEX-111FC-3326-SM-5GZYV'\n",
       "66. 'GTEX-111VG-0004-SM-58Q85'\n",
       "67. 'GTEX-111VG-0004-SM-5DWSX'\n",
       "68. 'GTEX-111VG-0004-SM-6WBTS'\n",
       "69. 'GTEX-111VG-0008-SM-5Q5BG'\n",
       "70. 'GTEX-111VG-0326-SM-5GZX7'\n",
       "71. 'GTEX-111VG-0526-SM-5N9BW'\n",
       "72. 'GTEX-111VG-0726-SM-5GIDC'\n",
       "73. 'GTEX-111VG-1626-SM-5EGIO'\n",
       "74. 'GTEX-111VG-1926-SM-5GIDO'\n",
       "75. 'GTEX-111VG-2226-SM-5N9DU'\n",
       "76. 'GTEX-111VG-2326-SM-5N9BK'\n",
       "77. 'GTEX-111VG-2426-SM-5GZXD'\n",
       "78. 'GTEX-111VG-2526-SM-5GZXP'\n",
       "79. 'GTEX-111VG-2626-SM-5GZY2'\n",
       "80. 'GTEX-111YS-0004-SM-58Q7Y'\n",
       "81. 'GTEX-111YS-0004-SM-5DWSS'\n",
       "82. 'GTEX-111YS-0004-SM-6WBTN'\n",
       "83. 'GTEX-111YS-0006-SM-5NQBE'\n",
       "84. 'GTEX-111YS-0008-SM-5Q5BH'\n",
       "85. 'GTEX-111YS-0126-SM-5987T'\n",
       "86. 'GTEX-111YS-0326-SM-5GZZ3'\n",
       "87. 'GTEX-111YS-0426-SM-5987O'\n",
       "88. 'GTEX-111YS-0526-SM-5GZXJ'\n",
       "89. 'GTEX-111YS-0626-SM-5GZXV'\n",
       "90. 'GTEX-111YS-0726-SM-5GZY8'\n",
       "91. 'GTEX-111YS-0826-SM-5GZYK'\n",
       "92. 'GTEX-111YS-0926-SM-5EGGI'\n",
       "93. 'GTEX-111YS-1026-SM-5GZYE'\n",
       "94. 'GTEX-111YS-1126-SM-5GZYQ'\n",
       "95. 'GTEX-111YS-1226-SM-5EGGJ'\n",
       "96. 'GTEX-111YS-1326-SM-5EGGK'\n",
       "97. 'GTEX-111YS-1426-SM-5GID8'\n",
       "98. 'GTEX-111YS-1526-SM-5GZYW'\n",
       "99. 'GTEX-111YS-1626-SM-5GZZ9'\n",
       "100. 'GTEX-111YS-1726-SM-5GIED'\n",
       "101. 'GTEX-111YS-1826-SM-5GIEP'\n",
       "102. 'GTEX-111YS-1926-SM-5GICC'\n",
       "103. 'GTEX-111YS-2026-SM-5EGGL'\n",
       "104. 'GTEX-111YS-2126-SM-5EGGM'\n",
       "105. 'GTEX-111YS-2226-SM-5987P'\n",
       "106. 'GTEX-111YS-2326-SM-5987L'\n",
       "107. 'GTEX-111YS-2426-SM-5GZZQ'\n",
       "108. 'GTEX-1122O-0003-SM-5Q5DL'\n",
       "109. 'GTEX-1122O-0004-SM-58Q7O'\n",
       "110. 'GTEX-1122O-0004-SM-5DWSJ'\n",
       "111. 'GTEX-1122O-0004-SM-6WBTE'\n",
       "112. 'GTEX-1122O-0005-SM-5O99J'\n",
       "113. 'GTEX-1122O-0008-SM-5QGR2'\n",
       "114. 'GTEX-1122O-0126-SM-5GICA'\n",
       "115. 'GTEX-1122O-0226-SM-5N9DA'\n",
       "116. 'GTEX-1122O-0326-SM-5H124'\n",
       "117. 'GTEX-1122O-0426-SM-5H12G'\n",
       "118. 'GTEX-1122O-0526-SM-5N9DM'\n",
       "119. 'GTEX-1122O-0626-SM-5N9B9'\n",
       "120. 'GTEX-1122O-0726-SM-5GIEV'\n",
       "121. 'GTEX-1122O-0826-SM-5GICV'\n",
       "122. 'GTEX-1122O-0926-SM-5N9C9'\n",
       "123. 'GTEX-1122O-0926-SM-9YFLC'\n",
       "124. 'GTEX-1122O-1026-SM-9WYSF'\n",
       "125. 'GTEX-1122O-1126-SM-5NQ8X'\n",
       "126. 'GTEX-1122O-1226-SM-5H113'\n",
       "127. 'GTEX-1122O-1326-SM-5H11F'\n",
       "128. 'GTEX-1122O-1426-SM-5H11R'\n",
       "129. 'GTEX-1122O-1526-SM-5N9CL'\n",
       "130. 'GTEX-1122O-1626-SM-5N9CX'\n",
       "131. 'GTEX-1122O-1726-SM-5GICM'\n",
       "132. 'GTEX-1122O-1826-SM-5EGIP'\n",
       "133. 'GTEX-1122O-1926-SM-5EGIQ'\n",
       "134. 'GTEX-1122O-2026-SM-5NQ91'\n",
       "135. 'GTEX-1122O-2026-SM-9YFMG'\n",
       "136. 'GTEX-1122O-2126-SM-5EGIR'\n",
       "137. 'GTEX-1122O-2326-SM-9YFL9'\n",
       "138. 'GTEX-1122O-2426-SM-5GIDN'\n",
       "139. 'GTEX-1128S-0001-SM-58Q96'\n",
       "140. 'GTEX-1128S-0001-SM-5DWTV'\n",
       "141. 'GTEX-1128S-0001-SM-6WBUH'\n",
       "142. 'GTEX-1128S-0005-SM-5P9HI'\n",
       "143. 'GTEX-1128S-0008-SM-5Q5DP'\n",
       "144. 'GTEX-1128S-0126-SM-5H12S'\n",
       "145. 'GTEX-1128S-0326-SM-5GZZF'\n",
       "146. 'GTEX-1128S-0726-SM-5N9D6'\n",
       "147. 'GTEX-1128S-0826-SM-5GZZI'\n",
       "148. 'GTEX-1128S-0926-SM-5GZZU'\n",
       "149. 'GTEX-1128S-1426-SM-5N9EF'\n",
       "150. 'GTEX-1128S-1626-SM-5H12O'\n",
       "151. 'GTEX-1128S-2126-SM-5H12U'\n",
       "152. 'GTEX-1128S-2326-SM-5GZZY'\n",
       "153. 'GTEX-1128S-2426-SM-5H11B'\n",
       "154. 'GTEX-1128S-2526-SM-5H11N'\n",
       "155. 'GTEX-1128S-2626-SM-5H11Z'\n",
       "156. 'GTEX-1128S-2726-SM-5H12C'\n",
       "157. 'GTEX-1128S-2826-SM-5N9DI'\n",
       "158. 'GTEX-113IC-0001-SM-58Q97'\n",
       "159. 'GTEX-113IC-0001-SM-5DWSI'\n",
       "160. 'GTEX-113IC-0001-SM-DLIPU'\n",
       "161. 'GTEX-113IC-0006-SM-5NQ9C'\n",
       "162. 'GTEX-113IC-0008-SM-5QGRF'\n",
       "163. 'GTEX-113IC-0126-SM-5HL6T'\n",
       "164. 'GTEX-113IC-0226-SM-5HL5C'\n",
       "165. 'GTEX-113IC-0426-SM-5HL5O'\n",
       "166. 'GTEX-113IC-0526-SM-5HL76'\n",
       "167. 'GTEX-113JC-0001-SM-58Q7M'\n",
       "168. 'GTEX-113JC-0001-SM-5DWSH'\n",
       "169. 'GTEX-113JC-0001-SM-6WBTD'\n",
       "170. 'GTEX-113JC-0006-SM-5O997'\n",
       "171. 'GTEX-113JC-0008-SM-5QGR6'\n",
       "172. 'GTEX-113JC-0126-SM-5EGJW'\n",
       "173. 'GTEX-113JC-0726-SM-5GZZR'\n",
       "174. 'GTEX-113JC-0926-SM-5H114'\n",
       "175. 'GTEX-113JC-1026-SM-5H117'\n",
       "176. 'GTEX-113JC-1926-SM-5N9GE'\n",
       "177. 'GTEX-113JC-2226-SM-5EGJG'\n",
       "178. 'GTEX-113JC-2326-SM-5EQ4E'\n",
       "179. 'GTEX-113JC-2726-SM-5EGIS'\n",
       "180. 'GTEX-113JC-2826-SM-5EGIT'\n",
       "181. 'GTEX-117XS-0001-SM-58Q7J'\n",
       "182. 'GTEX-117XS-0001-SM-5DWSE'\n",
       "183. 'GTEX-117XS-0001-SM-6WBTA'\n",
       "184. 'GTEX-117XS-0005-SM-5PNU6'\n",
       "185. 'GTEX-117XS-0008-SM-5Q5DQ'\n",
       "186. 'GTEX-117XS-0011-R10a-SM-AHZ7G'\n",
       "187. 'GTEX-117XS-0426-SM-5GZZN'\n",
       "188. 'GTEX-117XS-0526-SM-5987Q'\n",
       "189. 'GTEX-117XS-0726-SM-5H131'\n",
       "190. 'GTEX-117XS-1926-SM-5GICO'\n",
       "191. 'GTEX-117XS-2026-SM-5GID1'\n",
       "192. 'GTEX-117XS-2226-SM-5GIE1'\n",
       "193. 'GTEX-117XS-2526-SM-5H11G'\n",
       "194. 'GTEX-117XS-2526-SM-CYKR4'\n",
       "195. 'GTEX-117XS-2726-SM-5N9BL'\n",
       "196. 'GTEX-117XS-2926-SM-5N9BX'\n",
       "197. 'GTEX-117XS-3026-SM-5N9CA'\n",
       "198. 'GTEX-117XS-3026-SM-CYKOR'\n",
       "199. 'GTEX-117XS-3126-SM-5GIDP'\n",
       "200. 'GTEX-117YW-0003-SM-58Q99'\n",
       "201. ⋯\n",
       "202. 'K-562-SM-2S1RI'\n",
       "203. 'K-562-SM-2TC5E'\n",
       "204. 'K-562-SM-2TC6R'\n",
       "205. 'K-562-SM-2TF4E'\n",
       "206. 'K-562-SM-2TF5R'\n",
       "207. 'K-562-SM-2TO4S'\n",
       "208. 'K-562-SM-2TWBO'\n",
       "209. 'K-562-SM-2XCBH'\n",
       "210. 'K-562-SM-2XCE8'\n",
       "211. 'K-562-SM-2XCFL'\n",
       "212. 'K-562-SM-2XU9E'\n",
       "213. 'K-562-SM-2XV5K'\n",
       "214. 'K-562-SM-2YUNN'\n",
       "215. 'K-562-SM-32PLC'\n",
       "216. 'K-562-SM-32PMP'\n",
       "217. 'K-562-SM-32QPQ'\n",
       "218. 'K-562-SM-33HC7'\n",
       "219. 'K-562-SM-3DB9M'\n",
       "220. 'K-562-SM-3GADY'\n",
       "221. 'K-562-SM-3GAFC'\n",
       "222. 'K-562-SM-3GIKB'\n",
       "223. 'K-562-SM-3GILO'\n",
       "224. 'K-562-SM-3K2BF'\n",
       "225. 'K-562-SM-3LK7S'\n",
       "226. 'K-562-SM-3MJHH'\n",
       "227. 'K-562-SM-3NB3I'\n",
       "228. 'K-562-SM-3NMAP'\n",
       "229. 'K-562-SM-3NMDG'\n",
       "230. 'K-562-SM-3P61Y'\n",
       "231. 'K-562-SM-43ISS'\n",
       "232. 'K-562-SM-43V9A'\n",
       "233. 'K-562-SM-46MWI'\n",
       "234. 'K-562-SM-47JYY'\n",
       "235. 'K-562-SM-48FEU'\n",
       "236. 'K-562-SM-48TE3'\n",
       "237. 'K-562-SM-48U2C'\n",
       "238. 'K-562-SM-4AD4F'\n",
       "239. 'K-562-SM-4AT3W'\n",
       "240. 'K-562-SM-4B66B'\n",
       "241. 'K-562-SM-4BONS'\n",
       "242. 'K-562-SM-4BRWK'\n",
       "243. 'K-562-SM-4DM4W'\n",
       "244. 'K-562-SM-4DXSD'\n",
       "245. 'K-562-SM-4EDPU'\n",
       "246. 'K-562-SM-4GICD'\n",
       "247. 'K-562-SM-4IHK7'\n",
       "248. 'K-562-SM-4JBIQ'\n",
       "249. 'K-562-SM-4KKZ9'\n",
       "250. 'K-562-SM-4LMI2'\n",
       "251. 'K-562-SM-4LVKX'\n",
       "252. 'K-562-SM-4M1XD'\n",
       "253. 'K-562-SM-4MVNA'\n",
       "254. 'K-562-SM-4OOR8'\n",
       "255. 'K-562-SM-4PQYN'\n",
       "256. 'K-562-SM-4QAR5'\n",
       "257. 'K-562-SM-4R59E'\n",
       "258. 'K-562-SM-4RGL3'\n",
       "259. 'K-562-SM-4RH3A'\n",
       "260. 'K-562-SM-4RTUY'\n",
       "261. 'K-562-SM-4SOHT'\n",
       "262. 'K-562-SM-4SOLN'\n",
       "263. 'K-562-SM-4T8CH'\n",
       "264. 'K-562-SM-4TT1D'\n",
       "265. 'K-562-SM-4TT6S'\n",
       "266. 'K-562-SM-4U82S'\n",
       "267. 'K-562-SM-4V81D'\n",
       "268. 'K-562-SM-4VBPT'\n",
       "269. 'K-562-SM-4VDQQ'\n",
       "270. 'K-562-SM-4W1Y5'\n",
       "271. 'K-562-SM-4WAWH'\n",
       "272. 'K-562-SM-4WKF1'\n",
       "273. 'K-562-SM-4WWAK'\n",
       "274. 'K-562-SM-4YCCQ'\n",
       "275. 'K-562-SM-51MQT'\n",
       "276. 'K-562-SM-57WAW'\n",
       "277. 'K-562-SM-59HJ4'\n",
       "278. 'K-562-SM-5CVL5'\n",
       "279. 'K-562-SM-5DUV6'\n",
       "280. 'K-562-SM-5E43B'\n",
       "281. 'K-562-SM-5GZWN'\n",
       "282. 'K-562-SM-5GZZE'\n",
       "283. 'K-562-SM-5HL4J'\n",
       "284. 'K-562-SM-5HL7A'\n",
       "285. 'K-562-SM-5IFEL'\n",
       "286. 'K-562-SM-5IFHC'\n",
       "287. 'K-562-SM-5IJB7'\n",
       "288. 'K-562-SM-5IJDX'\n",
       "289. 'K-562-SM-5J1MB'\n",
       "290. 'K-562-SM-5J2LX'\n",
       "291. 'K-562-SM-5K7TT'\n",
       "292. 'K-562-SM-5K7WK'\n",
       "293. 'K-562-SM-5K7ZB'\n",
       "294. 'K-562-SM-5KM2I'\n",
       "295. 'K-562-SM-5L3FO'\n",
       "296. 'K-562-SM-5LU2Y'\n",
       "297. 'K-562-SM-5LU8G'\n",
       "298. 'K-562-SM-5LZU1'\n",
       "299. 'K-562-SM-5LZWR'\n",
       "300. 'K-562-SM-5MR32'\n",
       "301. 'K-562-SM-5MR5S'\n",
       "302. 'K-562-SM-5N9B7'\n",
       "303. 'K-562-SM-5N9DX'\n",
       "304. 'K-562-SM-5NQ6M'\n",
       "305. 'K-562-SM-5NQ9D'\n",
       "306. 'K-562-SM-5O98J'\n",
       "307. 'K-562-SM-5O9BA'\n",
       "308. 'K-562-SM-5P9EQ'\n",
       "309. 'K-562-SM-5P9HH'\n",
       "310. 'K-562-SM-5PNU4'\n",
       "311. 'K-562-SM-5PNWU'\n",
       "312. 'K-562-SM-5Q59U'\n",
       "313. 'K-562-SM-5Q5CL'\n",
       "314. 'K-562-SM-5QGOR'\n",
       "315. 'K-562-SM-5RQHM'\n",
       "316. 'K-562-SM-66EY4'\n",
       "317. 'K-562-SM-686YS'\n",
       "318. 'K-562-SM-69LO5'\n",
       "319. 'K-562-SM-6AJ95'\n",
       "320. 'K-562-SM-6ETZI'\n",
       "321. 'K-562-SM-6LLH2'\n",
       "322. 'K-562-SM-6LPHZ'\n",
       "323. 'K-562-SM-6M461'\n",
       "324. 'K-562-SM-6PAL1'\n",
       "325. 'K-562-SM-7189B'\n",
       "326. 'K-562-SM-72D4W'\n",
       "327. 'K-562-SM-731AF'\n",
       "328. 'K-562-SM-731D6'\n",
       "329. 'K-562-SM-73KTR'\n",
       "330. 'K-562-SM-73KWI'\n",
       "331. 'K-562-SM-7938F'\n",
       "332. 'K-562-SM-793B6'\n",
       "333. 'K-562-SM-79OJI'\n",
       "334. 'K-562-SM-79OM9'\n",
       "335. 'K-562-SM-7DHKI'\n",
       "336. 'K-562-SM-7DUEE'\n",
       "337. 'K-562-SM-7EPG4'\n",
       "338. 'K-562-SM-7EWCW'\n",
       "339. 'K-562-SM-7IGL8'\n",
       "340. 'K-562-SM-7IGNY'\n",
       "341. 'K-562-SM-7KFRB'\n",
       "342. 'K-562-SM-7KUDR'\n",
       "343. 'K-562-SM-7KUKI'\n",
       "344. 'K-562-SM-7LG4B'\n",
       "345. 'K-562-SM-7LT88'\n",
       "346. 'K-562-SM-7MGVH'\n",
       "347. 'K-562-SM-7MKF2'\n",
       "348. 'K-562-SM-7MXT1'\n",
       "349. 'K-562-SM-7P8RB'\n",
       "350. 'K-562-SM-7RHFN'\n",
       "351. 'K-562-SM-9HWNP'\n",
       "352. 'K-562-SM-9HWNQ'\n",
       "353. 'K-562-SM-9HWNT'\n",
       "354. 'K-562-SM-9HWNU'\n",
       "355. 'K-562-SM-9HWNV'\n",
       "356. 'K-562-SM-9HWNW'\n",
       "357. 'K-562-SM-9HWO3'\n",
       "358. 'K-562-SM-9HWO4'\n",
       "359. 'K-562-SM-9HWOB'\n",
       "360. 'K-562-SM-9HWOD'\n",
       "361. 'K-562-SM-9HWOI'\n",
       "362. 'K-562-SM-9HWOK'\n",
       "363. 'K-562-SM-9HWOP'\n",
       "364. 'K-562-SM-9HWOQ'\n",
       "365. 'K-562-SM-9HWOR'\n",
       "366. 'K-562-SM-9HWOX'\n",
       "367. 'K-562-SM-9HWOY'\n",
       "368. 'K-562-SM-9HWP4'\n",
       "369. 'K-562-SM-9HWP5'\n",
       "370. 'K-562-SM-9HWP6'\n",
       "371. 'K-562-SM-9HWP7'\n",
       "372. 'K-562-SM-9HWP8'\n",
       "373. 'K-562-SM-9HWP9'\n",
       "374. 'K-562-SM-9HWPF'\n",
       "375. 'K-562-SM-9HWPM'\n",
       "376. 'K-562-SM-9HWPN'\n",
       "377. 'K-562-SM-9HWPV'\n",
       "378. 'K-562-SM-9HWPW'\n",
       "379. 'K-562-SM-9HWPX'\n",
       "380. 'K-562-SM-9HWPY'\n",
       "381. 'K-562-SM-9HWPZ'\n",
       "382. 'K-562-SM-9HWQ7'\n",
       "383. 'K-562-SM-9HWQ9'\n",
       "384. 'K-562-SM-9HWQF'\n",
       "385. 'K-562-SM-9HWQG'\n",
       "386. 'K-562-SM-9JGFK'\n",
       "387. 'K-562-SM-9MQJO'\n",
       "388. 'K-562-SM-CJI1Q'\n",
       "389. 'K-562-SM-CKZN7'\n",
       "390. 'K-562-SM-E9EYN'\n",
       "391. 'K-562-SM-E9EYO'\n",
       "392. 'K-562-SM-E9EYP'\n",
       "393. 'K-562-SM-E9EYT'\n",
       "394. 'K-562-SM-E9EYU'\n",
       "395. 'K-562-SM-E9EYV'\n",
       "396. 'K-562-SM-E9EYZ'\n",
       "397. 'K-562-SM-E9EZC'\n",
       "398. 'K-562-SM-E9EZI'\n",
       "399. 'K-562-SM-E9EZO'\n",
       "400. 'K-562-SM-E9EZT'\n",
       "401. 'K-562-SM-E9EZZ'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "GTEX-1117F-0226-SM-5GZZ7 \n",
       "GTEX-1117F-0226-SM-5GZZ7 \n",
       "22951 Levels: GTEX-1117F-0003-SM-58Q7G ... K-562-SM-E9EZZ"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'GTEX.1117F.0226.SM.5GZZ7'"
      ],
      "text/latex": [
       "'GTEX.1117F.0226.SM.5GZZ7'"
      ],
      "text/markdown": [
       "'GTEX.1117F.0226.SM.5GZZ7'"
      ],
      "text/plain": [
       "[1] \"GTEX.1117F.0226.SM.5GZZ7\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pData(obj)$SAMPID[1]\n",
    "pData(obj)$SAMPID <- gsub('-','\\\\.',pData(obj)$SAMPID)\n",
    "pData(obj)$SAMPID[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.4.3   Synchronize Clinical Annotations and Accession Run, reducing to only tissues of interest\n",
    "\n",
    "Join the yarn metadata with the metadata we have (there are redundant samples that have been sequenced multiple times). We want to be sure that we can obtain all required Clinical Annotation information from the YARN GTEx Annotation information, as the SRA metadata is not as reliable.    This will be a one-to-many mapping, as there are multiple sequence runs per 69 samples -- expanding our data set.  There are only a handful of annotations we require: SEX, AGE, DTHHRDY (which is cause of death), SMCENTER.\n",
    "\n",
    "Note that the numbers in specific age groups expand because of the one to many relationship from sample to sequencing runs. \n",
    "\n",
    "Using results from analysis of number of samples stored in `tissues.tsv` we keep only those that are members of this reduced tissue list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "size tissue_reduction\n",
      "50 5\n",
      "\n",
      "\n",
      "size obj\n",
      "5587817382\n",
      "\n",
      "\n",
      "\n",
      "size pData(obj)\n",
      "17382 67\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " 0  1 \n",
       "11 39 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "length tissues in phenotype data\n",
      "54\n",
      "\n",
      "\n",
      "length tissues in tissue_reduction data\n",
      "39\n",
      "\n",
      "\n",
      "length tissue in samples phenotype data\n",
      "17382\n",
      "\n",
      "\n",
      "length keep obj \n",
      "17382\n",
      "\n",
      "\n",
      "how many to keep in phenotype data\n",
      "1851 15531\n",
      "\n",
      "\n",
      "size reduced_obj\n",
      "5587815531 \n",
      "\n",
      "\n",
      "size pData(reduced_obj)\n",
      "15531 67\n",
      "\n",
      "\n",
      "length tissues in phenotype data\n",
      "54\n",
      "\n",
      "\n",
      "TEST: how many to keep in to have only breast_mammary_tissue\n",
      "15072 459\n",
      "\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55878 459\n",
      "\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "459 67\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX-1117F-2826-SM-5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>breast_mammary_tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>50.2455</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX-1117F-2826-SM-5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & breast\\_mammary\\_tissue & 0008367 &  1340 & 1008 & ⋯ & 50.2455 & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX-1117F-2826-SM-5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | breast_mammary_tissue | 0008367 |  1340 | 1008 | ⋯ | 50.2455 | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX-1117F-2826-SM-5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX-1117F-2826-SM-5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                 SMUBRID SMTSISCH\n",
       "GTEX-1117F-2826-SM-5GZXL  5.80000 Breast breast_mammary_tissue 0008367  1340   \n",
       "                         SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX-1117F-2826-SM-5GZXL 1008    ⋯ 50.2455  0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX-1117F-2826-SM-5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# read in all requirements so that the stage is properly set -- \n",
    "# if it is clear here -- it will remain clear for the rest of the time\n",
    "# tissues.tsv contains the subset of files desired for analysis.\n",
    "tissue_reduction <- read.table(file=\"../assets/tissues.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "\n",
    "message(\"\\nsize tissue_reduction\\n\",\n",
    "        paste(dim(tissue_reduction), collapse=\" \"))\n",
    "message(\"\\nsize obj\\n\",\n",
    "        paste(dim(obj)), collapse=\"\\n\")\n",
    "message(\"\\nsize pData(obj)\\n\",\n",
    "        paste(dim(pData(obj)), collapse=\" \"))\n",
    "\n",
    "# only include those tissues we wish to continue with\n",
    "table(tissue_reduction$include)\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$include==1,]\n",
    "\n",
    "# create a matching tissue name to go with the expressionSet phenotype object\n",
    "pData(obj)$SMTSD       <- factor(snakecase::to_snake_case(as.character(pData(obj)$SMTSD)))\n",
    "tissue_reduction$SMTSD <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "\n",
    "message(\"\\nlength tissues in phenotype data\\n\",\n",
    "        paste(length(levels(pData(obj)$SMTSD)), collapse = \" \"))\n",
    "message(\"\\nlength tissues in tissue_reduction data\\n\",\n",
    "        paste(length(tissue_reduction$SMTSD), collapse = \" \"))\n",
    "\n",
    "keep <- pData(obj)$SMTSD %in% tissue_reduction$SMTSD\n",
    "message(\"\\nlength tissue in samples phenotype data\\n\",\n",
    "        paste(length(pData(obj)$SMTSD), collapse = \" \"))\n",
    "message(\"\\nlength keep obj \\n\",\n",
    "        paste(length(keep), collapse = \" \"))\n",
    "message(\"\\nhow many to keep in phenotype data\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "\n",
    "# both obj and pData(obj) need to be adjusted\n",
    "reduced_obj        <- obj       [          ,keep==TRUE]\n",
    "pData(reduced_obj) <- pData(obj)[keep==TRUE,          ]\n",
    "rm(keep)\n",
    "message(\"\\nsize reduced_obj\\n\",\n",
    "        paste(dim(reduced_obj)), collapse=\" \")\n",
    "message(\"\\nsize pData(reduced_obj)\\n\",\n",
    "        paste(dim(pData(reduced_obj)), collapse=\" \"))\n",
    "message(\"\\nlength tissues in phenotype data\\n\",\n",
    "        paste(length(levels(pData(reduced_obj)$SMTSD)), collapse = \" \"))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj)$SMTSD== \"breast_mammary_tissue\"\n",
    "message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = reduced_obj       [          ,keep==TRUE]\n",
    "pData(tobj) = pData(reduced_obj)[keep==TRUE,          ]\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "# end test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.4.4 Synchronize accession metadata and phenotype data\n",
    "A kind of transitive closure.   Metadata links the count data to the phenotype data.\n",
    "Begin with synchronizing accession metadata and phenotype data - which has been reduced - `reduced_obj` inputs here "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "size accession SraRunTable \n",
      "24667 80\n",
      "\n",
      "\n",
      "size reduced_obj\n",
      "55878 15531\n",
      "\n",
      "\n",
      "size pData(reduced_obj)\n",
      "15531 67\n",
      "\n",
      "\n",
      " any undefined (NA) sample names?\n",
      "0\n",
      "\n",
      "\n",
      " any undefined (NA) sample names?\n",
      "0\n",
      "\n",
      "\n",
      "length keep metadata samples in phenotype samples \n",
      "15531\n",
      "\n",
      "\n",
      "how many to keep in phenotype obj\n",
      "5159 10372\n",
      "\n",
      "\n",
      "size reduced_obj2\n",
      "55878 10372\n",
      "\n",
      "\n",
      "size pData(reduced_obj2)\n",
      "10372 67\n",
      "\n",
      "\n",
      "TEST: how many to keep in to have only breast_mammary_tissue\n",
      "10086 286\n",
      "\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55878 286\n",
      "\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "286 67\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX.1117F.2826.SM.5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>breast_mammary_tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>50.2455</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX.1117F.2826.SM.5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & breast\\_mammary\\_tissue & 0008367 &  1340 & 1008 & ⋯ & 50.2455 & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX.1117F.2826.SM.5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | breast_mammary_tissue | 0008367 |  1340 | 1008 | ⋯ | 50.2455 | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX.1117F.2826.SM.5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX.1117F.2826.SM.5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                 SMUBRID SMTSISCH\n",
       "GTEX.1117F.2826.SM.5GZXL  5.80000 Breast breast_mammary_tissue 0008367  1340   \n",
       "                         SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX.1117F.2826.SM.5GZXL 1008    ⋯ 50.2455  0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX.1117F.2826.SM.5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "length of metadata_samples\n",
      "24667\n",
      "\n",
      "\n",
      "length of phenotype_samples\n",
      "10372\n",
      "\n",
      "\n",
      "dimension of metadata\n",
      "24667 80\n",
      "\n",
      "\n",
      "length keep phenotype samples in metadata samples \n",
      "24667\n",
      "\n",
      "\n",
      "how many to keep in metadata samples\n",
      "\n",
      "\n",
      "\n",
      "dimension of reduced_metadata\n",
      "18214 80\n",
      "\n",
      "\n",
      "TEST: how many to keep in to have only breast_mammary_tissue\n",
      "10086 286\n",
      "\n",
      "\n",
      "TEST: size breast_mammary_tissue obj:tobj\n",
      "55878 286\n",
      "\n",
      "\n",
      "TEST: size phenotype object pData(tobj)\n",
      "286 67\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1 × 67</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SAMPID</th><th scope=col>SMATSSCR</th><th scope=col>SMCENTER</th><th scope=col>SMPTHNTS</th><th scope=col>SMRIN</th><th scope=col>SMTS</th><th scope=col>SMTSD</th><th scope=col>SMUBRID</th><th scope=col>SMTSISCH</th><th scope=col>SMTSPAX</th><th scope=col>⋯</th><th scope=col>SME1PCTS</th><th scope=col>SMRRNART</th><th scope=col>SME1MPRT</th><th scope=col>SMNUM5CD</th><th scope=col>SMDPMPRT</th><th scope=col>SME2PCTS</th><th scope=col>SUBJID</th><th scope=col>SEX</th><th scope=col>AGE</th><th scope=col>DTHHRDY</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GTEX.1117F.2826.SM.5GZXL</th><td>GTEX.1117F.2826.SM.5GZXL</td><td> 1</td><td>B1</td><td>2 pieces, fibrocystic changes, rep ductal/lobular elements delineated</td><td> 5.80000</td><td>Breast</td><td>breast_mammary_tissue</td><td>0008367</td><td> 1340</td><td>1008</td><td>⋯</td><td>50.2455</td><td>0.015023200</td><td>0.9943150</td><td>NA</td><td> 0</td><td>50.0068</td><td>GTEX-1117F</td><td>2</td><td>60-69</td><td> 4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 67\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & SAMPID & SMATSSCR & SMCENTER & SMPTHNTS & SMRIN & SMTS & SMTSD & SMUBRID & SMTSISCH & SMTSPAX & ⋯ & SME1PCTS & SMRRNART & SME1MPRT & SMNUM5CD & SMDPMPRT & SME2PCTS & SUBJID & SEX & AGE & DTHHRDY\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tGTEX.1117F.2826.SM.5GZXL & GTEX.1117F.2826.SM.5GZXL &  1 & B1 & 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated &  5.80000 & Breast & breast\\_mammary\\_tissue & 0008367 &  1340 & 1008 & ⋯ & 50.2455 & 0.015023200 & 0.9943150 & NA &  0 & 50.0068 & GTEX-1117F & 2 & 60-69 &  4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 67\n",
       "\n",
       "| <!--/--> | SAMPID &lt;chr&gt; | SMATSSCR &lt;fct&gt; | SMCENTER &lt;fct&gt; | SMPTHNTS &lt;fct&gt; | SMRIN &lt;fct&gt; | SMTS &lt;fct&gt; | SMTSD &lt;fct&gt; | SMUBRID &lt;fct&gt; | SMTSISCH &lt;fct&gt; | SMTSPAX &lt;fct&gt; | ⋯ ⋯ | SME1PCTS &lt;fct&gt; | SMRRNART &lt;fct&gt; | SME1MPRT &lt;fct&gt; | SMNUM5CD &lt;fct&gt; | SMDPMPRT &lt;fct&gt; | SME2PCTS &lt;fct&gt; | SUBJID &lt;fct&gt; | SEX &lt;fct&gt; | AGE &lt;fct&gt; | DTHHRDY &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GTEX.1117F.2826.SM.5GZXL | GTEX.1117F.2826.SM.5GZXL |  1 | B1 | 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated |  5.80000 | Breast | breast_mammary_tissue | 0008367 |  1340 | 1008 | ⋯ | 50.2455 | 0.015023200 | 0.9943150 | NA |  0 | 50.0068 | GTEX-1117F | 2 | 60-69 |  4 |\n",
       "\n"
      ],
      "text/plain": [
       "                         SAMPID                   SMATSSCR SMCENTER\n",
       "GTEX.1117F.2826.SM.5GZXL GTEX.1117F.2826.SM.5GZXL  1       B1      \n",
       "                         SMPTHNTS                                                             \n",
       "GTEX.1117F.2826.SM.5GZXL 2 pieces, fibrocystic changes, rep ductal/lobular elements delineated\n",
       "                         SMRIN    SMTS   SMTSD                 SMUBRID SMTSISCH\n",
       "GTEX.1117F.2826.SM.5GZXL  5.80000 Breast breast_mammary_tissue 0008367  1340   \n",
       "                         SMTSPAX ⋯ SME1PCTS SMRRNART    SME1MPRT  SMNUM5CD\n",
       "GTEX.1117F.2826.SM.5GZXL 1008    ⋯ 50.2455  0.015023200 0.9943150 NA      \n",
       "                         SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY\n",
       "GTEX.1117F.2826.SM.5GZXL  0       50.0068  GTEX-1117F 2   60-69  4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "keep\n",
       "FALSE  TRUE \n",
       "17722   492 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "TEST: how many breast_mammary_tissue samples in reduced_metadata\n",
      "17722 492\n",
      "\n",
      "\n",
      "TEST: number reduced_metadata breast_samples\n",
      "492\n",
      "\n",
      "\n",
      "TEST: number phenotype obj breast_samples\n",
      "286\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 1 × 80</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Run</th><th scope=col>analyte_type</th><th scope=col>Assay Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>biospecimen_repository</th><th scope=col>biospecimen_repository_sample_id</th><th scope=col>body_site</th><th scope=col>⋯</th><th scope=col>product_part_number (exp)</th><th scope=col>product_part_number (run)</th><th scope=col>sample_barcode (exp)</th><th scope=col>sample_barcode (run)</th><th scope=col>is_technical_control</th><th scope=col>target_set (exp)</th><th scope=col>primary_disease (exp)</th><th scope=col>secondary_accessions (run)</th><th scope=col>Alignment_Provider (run)</th><th scope=col>SAMPID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int64&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>SRR821498</td><td>RNA:Total RNA</td><td>RNA-Seq</td><td>152</td><td>7879119272</td><td>PRJNA75899</td><td>SAMN01994192</td><td>GTEx</td><td>GTEX-WI4N-1426-SM-3LK7H</td><td>Breast - Mammary Tissue</td><td>⋯</td><td></td><td></td><td>NA</td><td>NA</td><td></td><td></td><td></td><td></td><td></td><td>GTEX.WI4N.1426.SM.3LK7H</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 80\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " Run & analyte\\_type & Assay Type & AvgSpotLen & Bases & BioProject & BioSample & biospecimen\\_repository & biospecimen\\_repository\\_sample\\_id & body\\_site & ⋯ & product\\_part\\_number (exp) & product\\_part\\_number (run) & sample\\_barcode (exp) & sample\\_barcode (run) & is\\_technical\\_control & target\\_set (exp) & primary\\_disease (exp) & secondary\\_accessions (run) & Alignment\\_Provider (run) & SAMPID\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int64> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t SRR821498 & RNA:Total RNA & RNA-Seq & 152 & 7879119272 & PRJNA75899 & SAMN01994192 & GTEx & GTEX-WI4N-1426-SM-3LK7H & Breast - Mammary Tissue & ⋯ &  &  & NA & NA &  &  &  &  &  & GTEX.WI4N.1426.SM.3LK7H\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 80\n",
       "\n",
       "| Run &lt;chr&gt; | analyte_type &lt;chr&gt; | Assay Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;int64&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | biospecimen_repository &lt;chr&gt; | biospecimen_repository_sample_id &lt;chr&gt; | body_site &lt;chr&gt; | ⋯ ⋯ | product_part_number (exp) &lt;chr&gt; | product_part_number (run) &lt;chr&gt; | sample_barcode (exp) &lt;dbl&gt; | sample_barcode (run) &lt;dbl&gt; | is_technical_control &lt;chr&gt; | target_set (exp) &lt;chr&gt; | primary_disease (exp) &lt;chr&gt; | secondary_accessions (run) &lt;chr&gt; | Alignment_Provider (run) &lt;chr&gt; | SAMPID &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR821498 | RNA:Total RNA | RNA-Seq | 152 | 7879119272 | PRJNA75899 | SAMN01994192 | GTEx | GTEX-WI4N-1426-SM-3LK7H | Breast - Mammary Tissue | ⋯ | <!----> | <!----> | NA | NA | <!----> | <!----> | <!----> | <!----> | <!----> | GTEX.WI4N.1426.SM.3LK7H |\n",
       "\n"
      ],
      "text/plain": [
       "  Run       analyte_type  Assay Type AvgSpotLen Bases      BioProject\n",
       "1 SRR821498 RNA:Total RNA RNA-Seq    152        7879119272 PRJNA75899\n",
       "  BioSample    biospecimen_repository biospecimen_repository_sample_id\n",
       "1 SAMN01994192 GTEx                   GTEX-WI4N-1426-SM-3LK7H         \n",
       "  body_site               ⋯ product_part_number (exp) product_part_number (run)\n",
       "1 Breast - Mammary Tissue ⋯                                                    \n",
       "  sample_barcode (exp) sample_barcode (run) is_technical_control\n",
       "1 NA                   NA                                       \n",
       "  target_set (exp) primary_disease (exp) secondary_accessions (run)\n",
       "1                                                                  \n",
       "  Alignment_Provider (run) SAMPID                 \n",
       "1                          GTEX.WI4N.1426.SM.3LK7H"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# let's limit the phenotype object and then align the metadata file\n",
    "# input assumption is expressionSet object is now `reduced_obj`\n",
    "metadata <- data.table::fread(\"../data/SraRunTable.txt.gz\")\n",
    "metadata$SAMPID           <- gsub('-','\\\\.',metadata$'Sample Name')\n",
    "pData(reduced_obj)$SAMPID <- gsub('-','\\\\.',pData(reduced_obj)$SAMPID)\n",
    "\n",
    "message(\"\\nsize accession SraRunTable \\n\",\n",
    "        paste(dim(metadata), collapse=\" \"))\n",
    "message(\"\\nsize reduced_obj\\n\",\n",
    "        paste(dim(reduced_obj), collapse=\" \"))\n",
    "message(\"\\nsize pData(reduced_obj)\\n\",\n",
    "        paste(dim(pData(reduced_obj)), collapse=\" \"))\n",
    "rownames(pData(reduced_obj))<- pData(reduced_obj)$SAMPID\n",
    "\n",
    "# keep only those runs (as epitomized by the metadata_samples) in the phenotype set\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj)$SAMPID)\n",
    "\n",
    "# any undefined (N/A) sample names? These results will be zero\n",
    "message(\"\\n any undefined (NA) sample names?\\n\",\n",
    "        paste(sum(is.na(metadata_samples)), collapse=\" \"))\n",
    "message(\"\\n any undefined (NA) sample names?\\n\",\n",
    "        paste(sum(is.na(phenotype_samples)), collapse=\" \"))\n",
    "\n",
    "keep <- phenotype_samples %in% metadata_samples\n",
    "message(\"\\nlength keep metadata samples in phenotype samples \\n\",\n",
    "        paste(length(keep), collapse = \" \"))\n",
    "message(\"\\nhow many to keep in phenotype obj\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "\n",
    "reduced_obj2        <- reduced_obj       [          ,keep==TRUE]\n",
    "pData(reduced_obj2) <- pData(reduced_obj)[keep==TRUE,          ]\n",
    "message(\"\\nsize reduced_obj2\\n\",\n",
    "        paste(dim(reduced_obj2), collapse=\" \"))\n",
    "message(\"\\nsize pData(reduced_obj2)\\n\",\n",
    "        paste(dim(pData(reduced_obj2)), collapse=\" \"))\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj2)$SMTSD == \"breast_mammary_tissue\"\n",
    "message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = reduced_obj2       [          ,keep==TRUE]\n",
    "pData(tobj) = pData(reduced_obj2)[keep==TRUE,          ]\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "# end test\n",
    "\n",
    "# now go the other way - make sure the metadata samples are in sync with the phenotype samples\n",
    "# note that we are now with `reduced_obj2`\n",
    "metadata_samples   <- as.character(metadata$SAMPID)\n",
    "phenotype_samples  <- as.character(pData(reduced_obj2)$SAMPID)\n",
    "message(\"\\nlength of metadata_samples\\n\",\n",
    "        paste(length(metadata_samples), collapse=\" \"))\n",
    "message(\"\\nlength of phenotype_samples\\n\",\n",
    "        paste(length(phenotype_samples), collapse=\" \"))\n",
    "message(\"\\ndimension of metadata\\n\",\n",
    "        paste(dim(metadata), collapse=\" \"))\n",
    "keep <- metadata_samples %in% phenotype_samples\n",
    "message(\"\\nlength keep phenotype samples in metadata samples \\n\",\n",
    "        paste(length(keep), collapse = \" \"))\n",
    "message(\"\\nhow many to keep in metadata samples\\n\")\n",
    "\n",
    "reduced_metadata <- metadata[keep==TRUE,]\n",
    "message(\"\\ndimension of reduced_metadata\\n\",\n",
    "        paste(dim(reduced_metadata), collapse=\" \"))\n",
    "rm(keep)\n",
    "\n",
    "# test to make sure we don't have nonsense\n",
    "keep = pData(reduced_obj2)$SMTSD== \"breast_mammary_tissue\"\n",
    "message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "tobj        = reduced_obj2       [          ,keep==TRUE]\n",
    "pData(tobj) = pData(reduced_obj2)[keep==TRUE,          ]\n",
    "message(\"\\nTEST: size breast_mammary_tissue obj:tobj\\n\",\n",
    "        paste(dim(tobj), collapse=\" \"))\n",
    "message(\"\\nTEST: size phenotype object pData(tobj)\\n\",\n",
    "        paste(dim(pData(tobj)), collapse=\" \"))\n",
    "pData(tobj)[1,]\n",
    "rm(keep)\n",
    "\n",
    "breast_metadata_samples   <- as.character(reduced_metadata$SAMPID)\n",
    "breast_phenotype_samples  <- as.character(pData(tobj)$SAMPID)\n",
    "keep = breast_metadata_samples %in% breast_phenotype_samples\n",
    "table(keep)\n",
    "message(\"\\nTEST: how many breast_mammary_tissue samples in reduced_metadata\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "breast_samples <- reduced_metadata[keep,]\n",
    "message(\"\\nTEST: number reduced_metadata breast_samples\\n\",\n",
    "        paste(length(breast_samples$SAMPID), collapse=\" \"))\n",
    "message(\"\\nTEST: number phenotype obj breast_samples\\n\",\n",
    "        paste(length(pData(tobj)$SAMPID), collapse=\" \"))\n",
    "breast_samples[1,]\n",
    "# end test\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3.  Differential Expression using `edgeR`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  3.1  For each tissue, model gene expression by tissue\n",
    "\n",
    "Differential Analysis (DE) was performed using voom (Law et.al., 2014) with gene expression counts with associated precision weights, followed by linear modeling and empirical Bayes procedure using limma. \n",
    "\n",
    "Within each tissue, the following linear regression model was used to detec secually dimorphic alternative splicing event expression:\n",
    "\n",
    "\n",
    "           y = B0 + B1 sex + epsilon (error)\n",
    "           \n",
    "\n",
    "where y is the gene expression to be modeled sex denotes the reported sex of the subject"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###   3.2 remove all zero rows\n",
    "\n",
    "expressionSet object is now `reduced_obj3`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "is_zero\n",
       "FALSE  TRUE \n",
       "55814    64 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55814</dd><dt>Samples</dt><dd>10372</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55814\n",
       "\\item[Samples] 10372\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55814Samples\n",
       ":   10372\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55814    10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>10372</li><li>67</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10372\n",
       "\\item 67\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10372\n",
       "2. 67\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10372    67"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>55814</li><li>10372</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 55814\n",
       "\\item 10372\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 55814\n",
       "2. 10372\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 55814 10372"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "y_rowsums <- rowSums(exprs(reduced_obj2))\n",
    "is_zero   <- y_rowsums == 0\n",
    "table(is_zero)\n",
    "\n",
    "reduced_obj3  <- reduced_obj2[is_zero==FALSE,]\n",
    "\n",
    "dim(reduced_obj3)\n",
    "dim(pData(reduced_obj3))\n",
    "dim(exprs(reduced_obj3))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4. Separate the data by male and by female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_male   <- pData(reduced_obj3)$SEX==1\n",
    "reduced_female <- pData(reduced_obj3)$SEX==2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "reduced_obj_male   <- reduced_obj3[,reduced_male==TRUE]\n",
    "reduced_obj_female <- reduced_obj3[,reduced_female==TRUE]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55814</dd><dt>Samples</dt><dd>6796</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55814\n",
       "\\item[Samples] 6796\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55814Samples\n",
       ":   6796\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55814     6796 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>Features</dt><dd>55814</dd><dt>Samples</dt><dd>3576</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[Features] 55814\n",
       "\\item[Samples] 3576\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "Features\n",
       ":   55814Samples\n",
       ":   3576\n",
       "\n"
      ],
      "text/plain": [
       "Features  Samples \n",
       "   55814     3576 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(reduced_obj_male)\n",
    "dim(reduced_obj_female)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_groups_male <- factor(pData(reduced_obj_male)$SMTSD)\n",
    "tissue_groups_female <- factor(pData(reduced_obj_female)$SMTSD)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5. Differential gene analysis on a per tissue basis.\n",
    "Loop through the tissues and for those tissues that are shared between the two sexes perform the analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "tissue_male_female\n",
       "TRUE \n",
       "6796 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "tissue_shared_male_female\n",
       "                 adipose_subcutaneous              adipose_visceral_omentum \n",
       "                                  281                                   237 \n",
       "                        adrenal_gland                          artery_aorta \n",
       "                                  110                                   187 \n",
       "                      artery_coronary                         artery_tibial \n",
       "                                  104                                   297 \n",
       "          brain_caudate_basal_ganglia           brain_cerebellar_hemisphere \n",
       "                                  112                                   100 \n",
       "                     brain_cerebellum                          brain_cortex \n",
       "                                  120                                   111 \n",
       "            brain_frontal_cortex_ba_9                     brain_hippocampus \n",
       "                                   95                                    83 \n",
       "                   brain_hypothalamus brain_nucleus_accumbens_basal_ganglia \n",
       "                                   88                                   105 \n",
       "          brain_putamen_basal_ganglia        brain_spinal_cord_cervical_c_1 \n",
       "                                   91                                    56 \n",
       "                breast_mammary_tissue            cells_cultured_fibroblasts \n",
       "                                  175                                   225 \n",
       "    cells_ebv_transformed_lymphocytes                         colon_sigmoid \n",
       "                                   86                                   149 \n",
       "                     colon_transverse   esophagus_gastroesophageal_junction \n",
       "                                  162                                   156 \n",
       "                     esophagus_mucosa                  esophagus_muscularis \n",
       "                                  250                                   236 \n",
       "               heart_atrial_appendage                  heart_left_ventricle \n",
       "                                  198                                   201 \n",
       "                                liver                                  lung \n",
       "                                  119                                   280 \n",
       "                      muscle_skeletal                          nerve_tibial \n",
       "                                  371                                   274 \n",
       "                             pancreas                             pituitary \n",
       "                                  150                                   130 \n",
       "      skin_not_sun_exposed_suprapubic            skin_sun_exposed_lower_leg \n",
       "                                  258                                   300 \n",
       "       small_intestine_terminal_ileum                                spleen \n",
       "                                   88                                    97 \n",
       "                              stomach                               thyroid \n",
       "                                  157                                   294 \n",
       "                          whole_blood \n",
       "                                  263 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tissue_groups <- factor(pData(reduced_obj3)$SMTSD)\n",
    "tissue_male_female <- tissue_groups_male %in% tissue_groups_female\n",
    "table(tissue_male_female)\n",
    "tissue_shared_male_female <- factor(tissue_groups_male[tissue_male_female])\n",
    "table(tissue_shared_male_female)\n",
    "# SEX is coded 1 == Male\n",
    "#              2 == Female\n",
    "sex = factor(pData(reduced_obj3)$SEX)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's now define a function named `fit_tissue()`that accepts two arguments, the `tissue` and an `object` and create the **model matrix** based  that tissue's sex. We will perform a linear fit after calculating normal factors (based on the library size) and calculate the dispersion using `voom` (mean variance model of dispersion). We are saving the resulting matrixes as files.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "fit_tissue <- function (tissue, obj) {\n",
    "    tissue_true             <- pData(obj)$SMTSD == tissue\n",
    "    tissue_obj              <- obj[,tissue_true ==TRUE]\n",
    "    tissue_sex              <- factor(pData(tissue_obj)$SEX)\n",
    "    tissue_design           <- model.matrix(~tissue_sex)\n",
    "    colnames(tissue_design) <- c(\"intercept\",\"sex\")\n",
    "    \n",
    "    y_tissue       <- DGEList(counts=exprs(tissue_obj), group=tissue_sex)\n",
    "    y_tissue       <- calcNormFactors(y_tissue, method = \"RLE\")\n",
    "    y_tissue_voom  <- voom(y_tissue, tissue_design)\n",
    "    \n",
    "    sex            <- ifelse(pData(tissue_obj)$SEX==1,'male','female')\n",
    "    Gender         <- substring(sex,1,1)\n",
    "    filename       <- paste0(paste0(\"../pdf/\", snakecase::to_snake_case(tissue)),\"-gene-y-MDSplot-100.pdf\")\n",
    "    pdf (filename)\n",
    "        plotMDS(y_tissue, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "    filename       <- paste0(paste0(\"../pdf/\", snakecase::to_snake_case(tissue)),\"-gene-y-voom-MDSplot-100.pdf\")\n",
    "    pdf (filename)    \n",
    "        plotMDS(y_tissue_voom, labels=Gender, top=100, col=ifelse(Gender==\"m\",\"blue\",\"red\"), \n",
    "                gene.selection=\"common\")\n",
    "    dev.off()\n",
    "\n",
    "    fit_tissue      <- lmFit(y_tissue_voom, tissue_design)\n",
    "    fit_tissue      <- eBayes(fit_tissue, robust=TRUE)\n",
    "    results_tissue  <- topTable (fit_tissue, coef='sex', number=nrow(y_tissue))\n",
    "    results_refined <- results_tissue$adj.P.Val <= 0.05 & abs(results_tissue$logFC) >= abs(log2(1.5))\n",
    "    \n",
    "    \n",
    "    filename  = paste(paste(\"../data\",gsub(\" \",\"\",tissue), sep=\"/\"),\"DGE.csv\", sep=\"_\")\n",
    "    rfilename = paste(paste(\"../data\",gsub(\" \",\"\",tissue), sep=\"/\"),\"DGE_refined.csv\", sep=\"_\")\n",
    "    ensgfile  = paste(paste(\"../data\",gsub(\" \",\"\",tissue), sep=\"/\"),\"DGE_ensg_map.csv\", sep=\"_\")\n",
    "  \n",
    "    ensg_names <- as.character(rownames(results_tissue[results_refined,]))\n",
    "    ensg_genes <- ensg_names\n",
    "    for (i in (1:length(ensg_names))) {\n",
    "        dont_convert = 0\n",
    "        ensg <- as.character(strsplit(ensg_names[i],'\\\\.\\\\w+$'))\n",
    "        ensg_names[i] = ensg[1]\n",
    "        if (ensg_names[i] == \"ENSG00000233864\") {\n",
    "            ensg_genes[i] = as.character(\"TTTY15\")\n",
    "            dont_convert = 1\n",
    "        } \n",
    "        if (ensg_names[i] == \"ENSG00000240800\") {\n",
    "            ensg_genes[i] = as.character(\"ATP8A2P1\")\n",
    "            dong_convert = 1\n",
    "        } \n",
    "        if (!dont_convert) {\n",
    "            \n",
    "            res <- gconvert(c(as.character(ensg_names[i])),\n",
    "                                      organism = \"hsapiens\",\n",
    "                                      target = \"ENSG\",\n",
    "                                      numeric_ns = \"\", \n",
    "                                      mthreshold = Inf,\n",
    "                                      filter_na = TRUE)\n",
    "            if (!is.null(res)) {\n",
    "                ensg_genes[i] <- res$name\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "    ensg_maps <- cbind(ensg_names, ensg_genes)\n",
    "    write.table(results_tissue, filename, sep=',', quote=FALSE)\n",
    "    write.table(results_tissue[results_refined,], rfilename, sep=',', quote=FALSE)\n",
    "    write.table(ensg_maps, ensgfile, sep=',', quote=FALSE, row.names=FALSE)\n",
    "    return (results_tissue)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n",
      "No results to show\n",
      "Please make sure that the organism or namespace is correct\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# TEST: use this piece for code changes\n",
    "# TEST: tissue <- factor('breast_mammary_tissue')\n",
    "# TEST: all_logFC <- lapply(X=levels(tissue), FUN=fit_tissue, obj=reduced_obj3)\n",
    "# \n",
    "#  Production run - uncomment below\n",
    "all_logFC <- lapply(X=levels(tissue_shared_male_female), FUN=fit_tissue, obj=reduced_obj3)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>Samples:</strong> 10372"
      ],
      "text/latex": [
       "\\textbf{Samples:} 10372"
      ],
      "text/markdown": [
       "**Samples:** 10372"
      ],
      "text/plain": [
       "Samples \n",
       "  10372 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "39"
      ],
      "text/latex": [
       "39"
      ],
      "text/markdown": [
       "39"
      ],
      "text/plain": [
       "[1] 39"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(reduced_obj3)[2]\n",
    "length(levels(tissue_shared_male_female))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 6 Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "#### 6.1. Checksums with the sha256 algorithm\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating sha256 checksums of the artefacts in the `..data/` directory .. \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [],
      "text/latex": [],
      "text/markdown": [],
      "text/plain": [
       "character(0)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 120 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sha256sum</th><th scope=col>file</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>a38b7b145b0db233ba6b83ea533a7e5c01ecdac868e46996d604c5707b6780c3</td><td>./liver_DGE.csv                                     </td></tr>\n",
       "\t<tr><td>3ee0b9653d35496250df844627b6f7e2e9e9bfdad82ad1a1e75ff860a2d87eed</td><td>./esophagus_muscularis_DGE.csv                      </td></tr>\n",
       "\t<tr><td>01e26bcddec160ff640b05c0f92f63ecb7c1065969f7c12d410fb379122916b0</td><td>./brain_frontal_cortex_ba_9_DGE_ensg_map.csv        </td></tr>\n",
       "\t<tr><td>533fd5618a3df4c2d2a18bede39e1de85b67ff1d13d47353addd7316e0106078</td><td>./pituitary_DGE_refined.csv                         </td></tr>\n",
       "\t<tr><td>d00d5acac08cf0ca0add846c3097674f728b41a8a33c853b7c00d27519daeb86</td><td>./brain_cortex_DGE_refined.csv                      </td></tr>\n",
       "\t<tr><td>150d79c045deffb499df4eb827dfded1bc3e1a9813df33ba1e9448220e9ae280</td><td>./liver_DGE_ensg_map.csv                            </td></tr>\n",
       "\t<tr><td>5c30c8d636f3ce43bb29047b3a58ec76bd67998f9367ea0470ef955985a02905</td><td>./esophagus_muscularis_DGE_ensg_map.csv             </td></tr>\n",
       "\t<tr><td>25750033602385f4b2ec32b906d09d67d834d927c5328cb6f9f3b3948332e7f1</td><td>./whole_blood_DGE_refined.csv                       </td></tr>\n",
       "\t<tr><td>260c0bf8bc8a27be3aba82bc61a3d66fb8dfd73970ec3b20715e7ec485bfcb04</td><td>./heart_atrial_appendage_DGE.csv                    </td></tr>\n",
       "\t<tr><td>ca56e3f4f0b04eba5fced61bb5cfcee166164e01ed58ea13b33f86ee6fe420c2</td><td>./nerve_tibial_DGE_ensg_map.csv                     </td></tr>\n",
       "\t<tr><td>5d303aadcecb8b28264e25f56bd1f20e0dc8fa4c65dd006ea2629a31e38777c4</td><td>./heart_atrial_appendage_DGE_refined.csv            </td></tr>\n",
       "\t<tr><td>a53249f9e93a0d9dbb1d3c4f2420716926f3566c67a4a7e69bc584421896e7b2</td><td>./artery_tibial_DGE.csv                             </td></tr>\n",
       "\t<tr><td>bd04ed24fc8c544196d41063471ecbfac942f14271ef8755da0e6719ce6e6855</td><td>./brain_spinal_cord_cervical_c_1_DGE_refined.csv    </td></tr>\n",
       "\t<tr><td>bc9974cbeeabbc8d73757438f6c2391a3e2f9cac234a9f531461684f6b830f0e</td><td>./brain_hypothalamus_DGE_ensg_map.csv               </td></tr>\n",
       "\t<tr><td>f2244c536a076768e38867e857d540e76cb4b9d679189fe225d1584eebf02812</td><td>./brain_spinal_cord_cervical_c_1_DGE_ensg_map.csv   </td></tr>\n",
       "\t<tr><td>9fca162813402b542e02b80231a1f8b9ccb92ad11a074d82fa84a92800b884eb</td><td>./spleen_DGE_ensg_map.csv                           </td></tr>\n",
       "\t<tr><td>744a80d81523f9548744b3a244def8173a0811a3f50132842f347103c789880c</td><td>./pancreas_DGE_refined.csv                          </td></tr>\n",
       "\t<tr><td>e0e7b0d9c660c94660fe71f7d04a95d135057247115f03903eea7a9beb44eaaa</td><td>./brain_hippocampus_DGE_ensg_map.csv                </td></tr>\n",
       "\t<tr><td>e2ab5b48541d6eed63cd6f2b69c07490ddcefe689826088908c4f1c1670a0cf2</td><td>./brain_cortex_DGE_ensg_map.csv                     </td></tr>\n",
       "\t<tr><td>c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0</td><td>./gtex.rds                                          </td></tr>\n",
       "\t<tr><td>893b5829c9a176dfee0394aeaa8d9a303db65d5b282b26346dbf5f2b3b51c104</td><td>./thyroid_DGE_ensg_map.csv                          </td></tr>\n",
       "\t<tr><td>4069cd6b590424b5c07af679bfc2e163b372373dac0d2925afdb24f3213fb69d</td><td>./spleen_DGE_refined.csv                            </td></tr>\n",
       "\t<tr><td>8680903ad637315f6f6d909c9101f98009fe67abf10bbd0b7ba50accbf7884b1</td><td>./artery_aorta_DGE_refined.csv                      </td></tr>\n",
       "\t<tr><td>2995e91a26c65654a0c8cab7812e6518f83ca9a8f06de0f60ce253ca4c394ab7</td><td>./brain_putamen_basal_ganglia_DGE_ensg_map.csv      </td></tr>\n",
       "\t<tr><td>b1f31f4b3ce4969318c3b021e6d197c4ecf42384fce275785ecebf981fdfe295</td><td>./adrenal_gland_DGE_ensg_map.csv                    </td></tr>\n",
       "\t<tr><td>e686c40a03b666e9d8517add74956f1ab2b1efb966ce8f2c948a4fdd33e56696</td><td>./cells_cultured_fibroblasts_DGE_ensg_map.csv       </td></tr>\n",
       "\t<tr><td>c7c9a4ae8201c6d1efa53c39143f480466ac9475f9f7eb72505d1ae4ac828e91</td><td>./artery_tibial_DGE_refined.csv                     </td></tr>\n",
       "\t<tr><td>24f6b6f9999ebaf16fb0a00ddd424bfcb71d56ba012784e92b3f23eb29817cad</td><td>./colon_sigmoid_DGE_ensg_map.csv                    </td></tr>\n",
       "\t<tr><td>cda912cda3189f4d7dd8570082fccb0f0b04c71ce79affbeb05e1ed0a11b6240</td><td>./cells_ebv_transformed_lymphocytes_DGE_ensg_map.csv</td></tr>\n",
       "\t<tr><td>95ec3fc59c1746b08d4d16b558e42c53667dd0f8067be950cffbf9fa3193530d</td><td>./breast_mammary_tissue_DGE.csv                     </td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>1d01ef0841a8d106923e5233a69055a44e9725f91fd0b8087ec9264897b598f3</td><td>./whole_blood_DGE_ensg_map.csv                          </td></tr>\n",
       "\t<tr><td>56208d65c2f89d737c0355956ac3ac2622492abfe47c6cd6ab779c6c93196c61</td><td>./esophagus_mucosa_DGE_ensg_map.csv                     </td></tr>\n",
       "\t<tr><td>6d5be9fdbe52d797f68fe158e0bc8be605e16866e95215cb2bc3bd3805b9941a</td><td>./brain_cerebellum_DGE.csv                              </td></tr>\n",
       "\t<tr><td>0c66f19b598b8f492efe9d6e7471cda220553651da459a7579ef5f3ce69c7f09</td><td>./brain_frontal_cortex_ba_9_DGE_refined.csv             </td></tr>\n",
       "\t<tr><td>3bb158a6845f6b0f78649614f8ba0dd8150f9492cc9b54f847f25e643e351351</td><td>./brain_cortex_DGE.csv                                  </td></tr>\n",
       "\t<tr><td>1eabb511cc8a760e95072f68bc0448d84a0341055a9175a915b26e0360eabf6d</td><td>./lung_DGE_refined.csv                                  </td></tr>\n",
       "\t<tr><td>928dfe0e89c5f7135d1fe9d640cd03462595005f5be27ce16e1ccda87c01c2e0</td><td>./brain_hippocampus_DGE_refined.csv                     </td></tr>\n",
       "\t<tr><td>c4f653e682dbc652b842938031f6273362300f9174811b06003ec090994d6b0b</td><td>./small_intestine_terminal_ileum_DGE_ensg_map.csv       </td></tr>\n",
       "\t<tr><td>965cf032f314e3f0e03d46b57990306e682f26c2fbcb37c915ec942e4c6c78d9</td><td>./esophagus_gastroesophageal_junction_DGE_ensg_map.csv  </td></tr>\n",
       "\t<tr><td>c542fb8af1d56fd9742f4df71c250ddbda4dc3cedb1956a4b5a6cd764596aa3f</td><td>./artery_coronary_DGE_refined.csv                       </td></tr>\n",
       "\t<tr><td>5ca589e15872e4880f8c3ba7aaca955ca588f04eb7a2c573e47616687defc9c7</td><td>./brain_hippocampus_DGE.csv                             </td></tr>\n",
       "\t<tr><td>38692539c34b1d88f720658aeafd9a5487bc0f0a67cbd992e0324fd935ab5168</td><td>./brain_nucleus_accumbens_basal_ganglia_DGE_ensg_map.csv</td></tr>\n",
       "\t<tr><td>fee2a9539994682c619e9b6115fc8a78296def6ccb2847729161259db6bf4fc2</td><td>./colon_transverse_DGE.csv                              </td></tr>\n",
       "\t<tr><td>2cc5791ca6ca126b7561f60695fd4e4d2f52c6fdc6c05220b98e011255262b33</td><td>./cells_ebv_transformed_lymphocytes_DGE_refined.csv     </td></tr>\n",
       "\t<tr><td>20481d62565ca37c4c1acdc0eb64c97ae8d1ddcea55820489658f42e4825ea8d</td><td>./breast_mammary_tissue_DGE_refined.csv                 </td></tr>\n",
       "\t<tr><td>025e86c808aed60c9de41045475599f60061bf223da1bfe0c72e95d0ec7b5903</td><td>./lung_DGE_ensg_map.csv                                 </td></tr>\n",
       "\t<tr><td>68d98e3312a5f34e80eb04dd7b15c4374da3dfa17debd576bb1efc93d26241e8</td><td>./stomach_DGE_ensg_map.csv                              </td></tr>\n",
       "\t<tr><td>7990b7ea8db30990f33c84fa2bd296f51fe40a11b11753e3f8820f979ddd548f</td><td>./muscle_skeletal_DGE_ensg_map.csv                      </td></tr>\n",
       "\t<tr><td>311394371de2e4f92ea10bb72902eebe40f40160cf01a32f2ac76d2994f8ff67</td><td>./esophagus_gastroesophageal_junction_DGE.csv           </td></tr>\n",
       "\t<tr><td>42a2150a8399a854ae76a5b7f266d8c62a2c879f4561ae6c5f9ffe75df44f280</td><td>./nerve_tibial_DGE.csv                                  </td></tr>\n",
       "\t<tr><td>4bdb2128e5ecb063c61ddc29ac14cf2f288bf26f4bf468fa556cc34adb0e9b24</td><td>./stomach_DGE.csv                                       </td></tr>\n",
       "\t<tr><td>51b41d3376a554bf7653272830aac23fb1cba56288d2cf904a8a3d6d6a8709e4</td><td>./nerve_tibial_DGE_refined.csv                          </td></tr>\n",
       "\t<tr><td>294f52f3e49a37927087dcf616c3dc2bb2561b82e970e32f5c48ba01b0677c16</td><td>./brain_hypothalamus_DGE_refined.csv                    </td></tr>\n",
       "\t<tr><td>90de2ce3527a18a64ae44203ced07869713889061493588c15476dfdc8c55780</td><td>./esophagus_gastroesophageal_junction_DGE_refined.csv   </td></tr>\n",
       "\t<tr><td>66f6dff2b5c02370ea8ac82ce5aba51de70d3e911ea82adbfac8275b1c932c6c</td><td>./stomach_DGE_refined.csv                               </td></tr>\n",
       "\t<tr><td>17d047f3a2c672a1d5e753e4c2af2a33665ae78278f374a33eedfe9b88cfff6a</td><td>./cells_cultured_fibroblasts_DGE_refined.csv            </td></tr>\n",
       "\t<tr><td>1baf27cd0b0ee6df74946ee69fb005085e0d04ae9568e7589b7dfbf9e704fd6e</td><td>./small_intestine_terminal_ileum_DGE_refined.csv        </td></tr>\n",
       "\t<tr><td>a376179b2fb762e65b148c26b2e496d872696b3800c8bba3b2ddc6a3f44c9669</td><td>./adipose_visceral_omentum_DGE_ensg_map.csv             </td></tr>\n",
       "\t<tr><td>3b364672faccc397adae159deb645d6a2639e97b4328add5d8ec2f6798696150</td><td>./lung_DGE.csv                                          </td></tr>\n",
       "\t<tr><td>8291be77c7ad6cd73d9f7797658e3f1ffc197532a23385a85911ca4d4db13483</td><td>./SraRunTable.txt.gz                                    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 120 × 2\n",
       "\\begin{tabular}{ll}\n",
       " sha256sum & file\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t a38b7b145b0db233ba6b83ea533a7e5c01ecdac868e46996d604c5707b6780c3 & ./liver\\_DGE.csv                                     \\\\\n",
       "\t 3ee0b9653d35496250df844627b6f7e2e9e9bfdad82ad1a1e75ff860a2d87eed & ./esophagus\\_muscularis\\_DGE.csv                      \\\\\n",
       "\t 01e26bcddec160ff640b05c0f92f63ecb7c1065969f7c12d410fb379122916b0 & ./brain\\_frontal\\_cortex\\_ba\\_9\\_DGE\\_ensg\\_map.csv        \\\\\n",
       "\t 533fd5618a3df4c2d2a18bede39e1de85b67ff1d13d47353addd7316e0106078 & ./pituitary\\_DGE\\_refined.csv                         \\\\\n",
       "\t d00d5acac08cf0ca0add846c3097674f728b41a8a33c853b7c00d27519daeb86 & ./brain\\_cortex\\_DGE\\_refined.csv                      \\\\\n",
       "\t 150d79c045deffb499df4eb827dfded1bc3e1a9813df33ba1e9448220e9ae280 & ./liver\\_DGE\\_ensg\\_map.csv                            \\\\\n",
       "\t 5c30c8d636f3ce43bb29047b3a58ec76bd67998f9367ea0470ef955985a02905 & ./esophagus\\_muscularis\\_DGE\\_ensg\\_map.csv             \\\\\n",
       "\t 25750033602385f4b2ec32b906d09d67d834d927c5328cb6f9f3b3948332e7f1 & ./whole\\_blood\\_DGE\\_refined.csv                       \\\\\n",
       "\t 260c0bf8bc8a27be3aba82bc61a3d66fb8dfd73970ec3b20715e7ec485bfcb04 & ./heart\\_atrial\\_appendage\\_DGE.csv                    \\\\\n",
       "\t ca56e3f4f0b04eba5fced61bb5cfcee166164e01ed58ea13b33f86ee6fe420c2 & ./nerve\\_tibial\\_DGE\\_ensg\\_map.csv                     \\\\\n",
       "\t 5d303aadcecb8b28264e25f56bd1f20e0dc8fa4c65dd006ea2629a31e38777c4 & ./heart\\_atrial\\_appendage\\_DGE\\_refined.csv            \\\\\n",
       "\t a53249f9e93a0d9dbb1d3c4f2420716926f3566c67a4a7e69bc584421896e7b2 & ./artery\\_tibial\\_DGE.csv                             \\\\\n",
       "\t bd04ed24fc8c544196d41063471ecbfac942f14271ef8755da0e6719ce6e6855 & ./brain\\_spinal\\_cord\\_cervical\\_c\\_1\\_DGE\\_refined.csv    \\\\\n",
       "\t bc9974cbeeabbc8d73757438f6c2391a3e2f9cac234a9f531461684f6b830f0e & ./brain\\_hypothalamus\\_DGE\\_ensg\\_map.csv               \\\\\n",
       "\t f2244c536a076768e38867e857d540e76cb4b9d679189fe225d1584eebf02812 & ./brain\\_spinal\\_cord\\_cervical\\_c\\_1\\_DGE\\_ensg\\_map.csv   \\\\\n",
       "\t 9fca162813402b542e02b80231a1f8b9ccb92ad11a074d82fa84a92800b884eb & ./spleen\\_DGE\\_ensg\\_map.csv                           \\\\\n",
       "\t 744a80d81523f9548744b3a244def8173a0811a3f50132842f347103c789880c & ./pancreas\\_DGE\\_refined.csv                          \\\\\n",
       "\t e0e7b0d9c660c94660fe71f7d04a95d135057247115f03903eea7a9beb44eaaa & ./brain\\_hippocampus\\_DGE\\_ensg\\_map.csv                \\\\\n",
       "\t e2ab5b48541d6eed63cd6f2b69c07490ddcefe689826088908c4f1c1670a0cf2 & ./brain\\_cortex\\_DGE\\_ensg\\_map.csv                     \\\\\n",
       "\t c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0 & ./gtex.rds                                          \\\\\n",
       "\t 893b5829c9a176dfee0394aeaa8d9a303db65d5b282b26346dbf5f2b3b51c104 & ./thyroid\\_DGE\\_ensg\\_map.csv                          \\\\\n",
       "\t 4069cd6b590424b5c07af679bfc2e163b372373dac0d2925afdb24f3213fb69d & ./spleen\\_DGE\\_refined.csv                            \\\\\n",
       "\t 8680903ad637315f6f6d909c9101f98009fe67abf10bbd0b7ba50accbf7884b1 & ./artery\\_aorta\\_DGE\\_refined.csv                      \\\\\n",
       "\t 2995e91a26c65654a0c8cab7812e6518f83ca9a8f06de0f60ce253ca4c394ab7 & ./brain\\_putamen\\_basal\\_ganglia\\_DGE\\_ensg\\_map.csv      \\\\\n",
       "\t b1f31f4b3ce4969318c3b021e6d197c4ecf42384fce275785ecebf981fdfe295 & ./adrenal\\_gland\\_DGE\\_ensg\\_map.csv                    \\\\\n",
       "\t e686c40a03b666e9d8517add74956f1ab2b1efb966ce8f2c948a4fdd33e56696 & ./cells\\_cultured\\_fibroblasts\\_DGE\\_ensg\\_map.csv       \\\\\n",
       "\t c7c9a4ae8201c6d1efa53c39143f480466ac9475f9f7eb72505d1ae4ac828e91 & ./artery\\_tibial\\_DGE\\_refined.csv                     \\\\\n",
       "\t 24f6b6f9999ebaf16fb0a00ddd424bfcb71d56ba012784e92b3f23eb29817cad & ./colon\\_sigmoid\\_DGE\\_ensg\\_map.csv                    \\\\\n",
       "\t cda912cda3189f4d7dd8570082fccb0f0b04c71ce79affbeb05e1ed0a11b6240 & ./cells\\_ebv\\_transformed\\_lymphocytes\\_DGE\\_ensg\\_map.csv\\\\\n",
       "\t 95ec3fc59c1746b08d4d16b558e42c53667dd0f8067be950cffbf9fa3193530d & ./breast\\_mammary\\_tissue\\_DGE.csv                     \\\\\n",
       "\t ⋮ & ⋮\\\\\n",
       "\t 1d01ef0841a8d106923e5233a69055a44e9725f91fd0b8087ec9264897b598f3 & ./whole\\_blood\\_DGE\\_ensg\\_map.csv                          \\\\\n",
       "\t 56208d65c2f89d737c0355956ac3ac2622492abfe47c6cd6ab779c6c93196c61 & ./esophagus\\_mucosa\\_DGE\\_ensg\\_map.csv                     \\\\\n",
       "\t 6d5be9fdbe52d797f68fe158e0bc8be605e16866e95215cb2bc3bd3805b9941a & ./brain\\_cerebellum\\_DGE.csv                              \\\\\n",
       "\t 0c66f19b598b8f492efe9d6e7471cda220553651da459a7579ef5f3ce69c7f09 & ./brain\\_frontal\\_cortex\\_ba\\_9\\_DGE\\_refined.csv             \\\\\n",
       "\t 3bb158a6845f6b0f78649614f8ba0dd8150f9492cc9b54f847f25e643e351351 & ./brain\\_cortex\\_DGE.csv                                  \\\\\n",
       "\t 1eabb511cc8a760e95072f68bc0448d84a0341055a9175a915b26e0360eabf6d & ./lung\\_DGE\\_refined.csv                                  \\\\\n",
       "\t 928dfe0e89c5f7135d1fe9d640cd03462595005f5be27ce16e1ccda87c01c2e0 & ./brain\\_hippocampus\\_DGE\\_refined.csv                     \\\\\n",
       "\t c4f653e682dbc652b842938031f6273362300f9174811b06003ec090994d6b0b & ./small\\_intestine\\_terminal\\_ileum\\_DGE\\_ensg\\_map.csv       \\\\\n",
       "\t 965cf032f314e3f0e03d46b57990306e682f26c2fbcb37c915ec942e4c6c78d9 & ./esophagus\\_gastroesophageal\\_junction\\_DGE\\_ensg\\_map.csv  \\\\\n",
       "\t c542fb8af1d56fd9742f4df71c250ddbda4dc3cedb1956a4b5a6cd764596aa3f & ./artery\\_coronary\\_DGE\\_refined.csv                       \\\\\n",
       "\t 5ca589e15872e4880f8c3ba7aaca955ca588f04eb7a2c573e47616687defc9c7 & ./brain\\_hippocampus\\_DGE.csv                             \\\\\n",
       "\t 38692539c34b1d88f720658aeafd9a5487bc0f0a67cbd992e0324fd935ab5168 & ./brain\\_nucleus\\_accumbens\\_basal\\_ganglia\\_DGE\\_ensg\\_map.csv\\\\\n",
       "\t fee2a9539994682c619e9b6115fc8a78296def6ccb2847729161259db6bf4fc2 & ./colon\\_transverse\\_DGE.csv                              \\\\\n",
       "\t 2cc5791ca6ca126b7561f60695fd4e4d2f52c6fdc6c05220b98e011255262b33 & ./cells\\_ebv\\_transformed\\_lymphocytes\\_DGE\\_refined.csv     \\\\\n",
       "\t 20481d62565ca37c4c1acdc0eb64c97ae8d1ddcea55820489658f42e4825ea8d & ./breast\\_mammary\\_tissue\\_DGE\\_refined.csv                 \\\\\n",
       "\t 025e86c808aed60c9de41045475599f60061bf223da1bfe0c72e95d0ec7b5903 & ./lung\\_DGE\\_ensg\\_map.csv                                 \\\\\n",
       "\t 68d98e3312a5f34e80eb04dd7b15c4374da3dfa17debd576bb1efc93d26241e8 & ./stomach\\_DGE\\_ensg\\_map.csv                              \\\\\n",
       "\t 7990b7ea8db30990f33c84fa2bd296f51fe40a11b11753e3f8820f979ddd548f & ./muscle\\_skeletal\\_DGE\\_ensg\\_map.csv                      \\\\\n",
       "\t 311394371de2e4f92ea10bb72902eebe40f40160cf01a32f2ac76d2994f8ff67 & ./esophagus\\_gastroesophageal\\_junction\\_DGE.csv           \\\\\n",
       "\t 42a2150a8399a854ae76a5b7f266d8c62a2c879f4561ae6c5f9ffe75df44f280 & ./nerve\\_tibial\\_DGE.csv                                  \\\\\n",
       "\t 4bdb2128e5ecb063c61ddc29ac14cf2f288bf26f4bf468fa556cc34adb0e9b24 & ./stomach\\_DGE.csv                                       \\\\\n",
       "\t 51b41d3376a554bf7653272830aac23fb1cba56288d2cf904a8a3d6d6a8709e4 & ./nerve\\_tibial\\_DGE\\_refined.csv                          \\\\\n",
       "\t 294f52f3e49a37927087dcf616c3dc2bb2561b82e970e32f5c48ba01b0677c16 & ./brain\\_hypothalamus\\_DGE\\_refined.csv                    \\\\\n",
       "\t 90de2ce3527a18a64ae44203ced07869713889061493588c15476dfdc8c55780 & ./esophagus\\_gastroesophageal\\_junction\\_DGE\\_refined.csv   \\\\\n",
       "\t 66f6dff2b5c02370ea8ac82ce5aba51de70d3e911ea82adbfac8275b1c932c6c & ./stomach\\_DGE\\_refined.csv                               \\\\\n",
       "\t 17d047f3a2c672a1d5e753e4c2af2a33665ae78278f374a33eedfe9b88cfff6a & ./cells\\_cultured\\_fibroblasts\\_DGE\\_refined.csv            \\\\\n",
       "\t 1baf27cd0b0ee6df74946ee69fb005085e0d04ae9568e7589b7dfbf9e704fd6e & ./small\\_intestine\\_terminal\\_ileum\\_DGE\\_refined.csv        \\\\\n",
       "\t a376179b2fb762e65b148c26b2e496d872696b3800c8bba3b2ddc6a3f44c9669 & ./adipose\\_visceral\\_omentum\\_DGE\\_ensg\\_map.csv             \\\\\n",
       "\t 3b364672faccc397adae159deb645d6a2639e97b4328add5d8ec2f6798696150 & ./lung\\_DGE.csv                                          \\\\\n",
       "\t 8291be77c7ad6cd73d9f7797658e3f1ffc197532a23385a85911ca4d4db13483 & ./SraRunTable.txt.gz                                    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 120 × 2\n",
       "\n",
       "| sha256sum &lt;chr&gt; | file &lt;chr&gt; |\n",
       "|---|---|\n",
       "| a38b7b145b0db233ba6b83ea533a7e5c01ecdac868e46996d604c5707b6780c3 | ./liver_DGE.csv                                      |\n",
       "| 3ee0b9653d35496250df844627b6f7e2e9e9bfdad82ad1a1e75ff860a2d87eed | ./esophagus_muscularis_DGE.csv                       |\n",
       "| 01e26bcddec160ff640b05c0f92f63ecb7c1065969f7c12d410fb379122916b0 | ./brain_frontal_cortex_ba_9_DGE_ensg_map.csv         |\n",
       "| 533fd5618a3df4c2d2a18bede39e1de85b67ff1d13d47353addd7316e0106078 | ./pituitary_DGE_refined.csv                          |\n",
       "| d00d5acac08cf0ca0add846c3097674f728b41a8a33c853b7c00d27519daeb86 | ./brain_cortex_DGE_refined.csv                       |\n",
       "| 150d79c045deffb499df4eb827dfded1bc3e1a9813df33ba1e9448220e9ae280 | ./liver_DGE_ensg_map.csv                             |\n",
       "| 5c30c8d636f3ce43bb29047b3a58ec76bd67998f9367ea0470ef955985a02905 | ./esophagus_muscularis_DGE_ensg_map.csv              |\n",
       "| 25750033602385f4b2ec32b906d09d67d834d927c5328cb6f9f3b3948332e7f1 | ./whole_blood_DGE_refined.csv                        |\n",
       "| 260c0bf8bc8a27be3aba82bc61a3d66fb8dfd73970ec3b20715e7ec485bfcb04 | ./heart_atrial_appendage_DGE.csv                     |\n",
       "| ca56e3f4f0b04eba5fced61bb5cfcee166164e01ed58ea13b33f86ee6fe420c2 | ./nerve_tibial_DGE_ensg_map.csv                      |\n",
       "| 5d303aadcecb8b28264e25f56bd1f20e0dc8fa4c65dd006ea2629a31e38777c4 | ./heart_atrial_appendage_DGE_refined.csv             |\n",
       "| a53249f9e93a0d9dbb1d3c4f2420716926f3566c67a4a7e69bc584421896e7b2 | ./artery_tibial_DGE.csv                              |\n",
       "| bd04ed24fc8c544196d41063471ecbfac942f14271ef8755da0e6719ce6e6855 | ./brain_spinal_cord_cervical_c_1_DGE_refined.csv     |\n",
       "| bc9974cbeeabbc8d73757438f6c2391a3e2f9cac234a9f531461684f6b830f0e | ./brain_hypothalamus_DGE_ensg_map.csv                |\n",
       "| f2244c536a076768e38867e857d540e76cb4b9d679189fe225d1584eebf02812 | ./brain_spinal_cord_cervical_c_1_DGE_ensg_map.csv    |\n",
       "| 9fca162813402b542e02b80231a1f8b9ccb92ad11a074d82fa84a92800b884eb | ./spleen_DGE_ensg_map.csv                            |\n",
       "| 744a80d81523f9548744b3a244def8173a0811a3f50132842f347103c789880c | ./pancreas_DGE_refined.csv                           |\n",
       "| e0e7b0d9c660c94660fe71f7d04a95d135057247115f03903eea7a9beb44eaaa | ./brain_hippocampus_DGE_ensg_map.csv                 |\n",
       "| e2ab5b48541d6eed63cd6f2b69c07490ddcefe689826088908c4f1c1670a0cf2 | ./brain_cortex_DGE_ensg_map.csv                      |\n",
       "| c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0 | ./gtex.rds                                           |\n",
       "| 893b5829c9a176dfee0394aeaa8d9a303db65d5b282b26346dbf5f2b3b51c104 | ./thyroid_DGE_ensg_map.csv                           |\n",
       "| 4069cd6b590424b5c07af679bfc2e163b372373dac0d2925afdb24f3213fb69d | ./spleen_DGE_refined.csv                             |\n",
       "| 8680903ad637315f6f6d909c9101f98009fe67abf10bbd0b7ba50accbf7884b1 | ./artery_aorta_DGE_refined.csv                       |\n",
       "| 2995e91a26c65654a0c8cab7812e6518f83ca9a8f06de0f60ce253ca4c394ab7 | ./brain_putamen_basal_ganglia_DGE_ensg_map.csv       |\n",
       "| b1f31f4b3ce4969318c3b021e6d197c4ecf42384fce275785ecebf981fdfe295 | ./adrenal_gland_DGE_ensg_map.csv                     |\n",
       "| e686c40a03b666e9d8517add74956f1ab2b1efb966ce8f2c948a4fdd33e56696 | ./cells_cultured_fibroblasts_DGE_ensg_map.csv        |\n",
       "| c7c9a4ae8201c6d1efa53c39143f480466ac9475f9f7eb72505d1ae4ac828e91 | ./artery_tibial_DGE_refined.csv                      |\n",
       "| 24f6b6f9999ebaf16fb0a00ddd424bfcb71d56ba012784e92b3f23eb29817cad | ./colon_sigmoid_DGE_ensg_map.csv                     |\n",
       "| cda912cda3189f4d7dd8570082fccb0f0b04c71ce79affbeb05e1ed0a11b6240 | ./cells_ebv_transformed_lymphocytes_DGE_ensg_map.csv |\n",
       "| 95ec3fc59c1746b08d4d16b558e42c53667dd0f8067be950cffbf9fa3193530d | ./breast_mammary_tissue_DGE.csv                      |\n",
       "| ⋮ | ⋮ |\n",
       "| 1d01ef0841a8d106923e5233a69055a44e9725f91fd0b8087ec9264897b598f3 | ./whole_blood_DGE_ensg_map.csv                           |\n",
       "| 56208d65c2f89d737c0355956ac3ac2622492abfe47c6cd6ab779c6c93196c61 | ./esophagus_mucosa_DGE_ensg_map.csv                      |\n",
       "| 6d5be9fdbe52d797f68fe158e0bc8be605e16866e95215cb2bc3bd3805b9941a | ./brain_cerebellum_DGE.csv                               |\n",
       "| 0c66f19b598b8f492efe9d6e7471cda220553651da459a7579ef5f3ce69c7f09 | ./brain_frontal_cortex_ba_9_DGE_refined.csv              |\n",
       "| 3bb158a6845f6b0f78649614f8ba0dd8150f9492cc9b54f847f25e643e351351 | ./brain_cortex_DGE.csv                                   |\n",
       "| 1eabb511cc8a760e95072f68bc0448d84a0341055a9175a915b26e0360eabf6d | ./lung_DGE_refined.csv                                   |\n",
       "| 928dfe0e89c5f7135d1fe9d640cd03462595005f5be27ce16e1ccda87c01c2e0 | ./brain_hippocampus_DGE_refined.csv                      |\n",
       "| c4f653e682dbc652b842938031f6273362300f9174811b06003ec090994d6b0b | ./small_intestine_terminal_ileum_DGE_ensg_map.csv        |\n",
       "| 965cf032f314e3f0e03d46b57990306e682f26c2fbcb37c915ec942e4c6c78d9 | ./esophagus_gastroesophageal_junction_DGE_ensg_map.csv   |\n",
       "| c542fb8af1d56fd9742f4df71c250ddbda4dc3cedb1956a4b5a6cd764596aa3f | ./artery_coronary_DGE_refined.csv                        |\n",
       "| 5ca589e15872e4880f8c3ba7aaca955ca588f04eb7a2c573e47616687defc9c7 | ./brain_hippocampus_DGE.csv                              |\n",
       "| 38692539c34b1d88f720658aeafd9a5487bc0f0a67cbd992e0324fd935ab5168 | ./brain_nucleus_accumbens_basal_ganglia_DGE_ensg_map.csv |\n",
       "| fee2a9539994682c619e9b6115fc8a78296def6ccb2847729161259db6bf4fc2 | ./colon_transverse_DGE.csv                               |\n",
       "| 2cc5791ca6ca126b7561f60695fd4e4d2f52c6fdc6c05220b98e011255262b33 | ./cells_ebv_transformed_lymphocytes_DGE_refined.csv      |\n",
       "| 20481d62565ca37c4c1acdc0eb64c97ae8d1ddcea55820489658f42e4825ea8d | ./breast_mammary_tissue_DGE_refined.csv                  |\n",
       "| 025e86c808aed60c9de41045475599f60061bf223da1bfe0c72e95d0ec7b5903 | ./lung_DGE_ensg_map.csv                                  |\n",
       "| 68d98e3312a5f34e80eb04dd7b15c4374da3dfa17debd576bb1efc93d26241e8 | ./stomach_DGE_ensg_map.csv                               |\n",
       "| 7990b7ea8db30990f33c84fa2bd296f51fe40a11b11753e3f8820f979ddd548f | ./muscle_skeletal_DGE_ensg_map.csv                       |\n",
       "| 311394371de2e4f92ea10bb72902eebe40f40160cf01a32f2ac76d2994f8ff67 | ./esophagus_gastroesophageal_junction_DGE.csv            |\n",
       "| 42a2150a8399a854ae76a5b7f266d8c62a2c879f4561ae6c5f9ffe75df44f280 | ./nerve_tibial_DGE.csv                                   |\n",
       "| 4bdb2128e5ecb063c61ddc29ac14cf2f288bf26f4bf468fa556cc34adb0e9b24 | ./stomach_DGE.csv                                        |\n",
       "| 51b41d3376a554bf7653272830aac23fb1cba56288d2cf904a8a3d6d6a8709e4 | ./nerve_tibial_DGE_refined.csv                           |\n",
       "| 294f52f3e49a37927087dcf616c3dc2bb2561b82e970e32f5c48ba01b0677c16 | ./brain_hypothalamus_DGE_refined.csv                     |\n",
       "| 90de2ce3527a18a64ae44203ced07869713889061493588c15476dfdc8c55780 | ./esophagus_gastroesophageal_junction_DGE_refined.csv    |\n",
       "| 66f6dff2b5c02370ea8ac82ce5aba51de70d3e911ea82adbfac8275b1c932c6c | ./stomach_DGE_refined.csv                                |\n",
       "| 17d047f3a2c672a1d5e753e4c2af2a33665ae78278f374a33eedfe9b88cfff6a | ./cells_cultured_fibroblasts_DGE_refined.csv             |\n",
       "| 1baf27cd0b0ee6df74946ee69fb005085e0d04ae9568e7589b7dfbf9e704fd6e | ./small_intestine_terminal_ileum_DGE_refined.csv         |\n",
       "| a376179b2fb762e65b148c26b2e496d872696b3800c8bba3b2ddc6a3f44c9669 | ./adipose_visceral_omentum_DGE_ensg_map.csv              |\n",
       "| 3b364672faccc397adae159deb645d6a2639e97b4328add5d8ec2f6798696150 | ./lung_DGE.csv                                           |\n",
       "| 8291be77c7ad6cd73d9f7797658e3f1ffc197532a23385a85911ca4d4db13483 | ./SraRunTable.txt.gz                                     |\n",
       "\n"
      ],
      "text/plain": [
       "    sha256sum                                                       \n",
       "1   a38b7b145b0db233ba6b83ea533a7e5c01ecdac868e46996d604c5707b6780c3\n",
       "2   3ee0b9653d35496250df844627b6f7e2e9e9bfdad82ad1a1e75ff860a2d87eed\n",
       "3   01e26bcddec160ff640b05c0f92f63ecb7c1065969f7c12d410fb379122916b0\n",
       "4   533fd5618a3df4c2d2a18bede39e1de85b67ff1d13d47353addd7316e0106078\n",
       "5   d00d5acac08cf0ca0add846c3097674f728b41a8a33c853b7c00d27519daeb86\n",
       "6   150d79c045deffb499df4eb827dfded1bc3e1a9813df33ba1e9448220e9ae280\n",
       "7   5c30c8d636f3ce43bb29047b3a58ec76bd67998f9367ea0470ef955985a02905\n",
       "8   25750033602385f4b2ec32b906d09d67d834d927c5328cb6f9f3b3948332e7f1\n",
       "9   260c0bf8bc8a27be3aba82bc61a3d66fb8dfd73970ec3b20715e7ec485bfcb04\n",
       "10  ca56e3f4f0b04eba5fced61bb5cfcee166164e01ed58ea13b33f86ee6fe420c2\n",
       "11  5d303aadcecb8b28264e25f56bd1f20e0dc8fa4c65dd006ea2629a31e38777c4\n",
       "12  a53249f9e93a0d9dbb1d3c4f2420716926f3566c67a4a7e69bc584421896e7b2\n",
       "13  bd04ed24fc8c544196d41063471ecbfac942f14271ef8755da0e6719ce6e6855\n",
       "14  bc9974cbeeabbc8d73757438f6c2391a3e2f9cac234a9f531461684f6b830f0e\n",
       "15  f2244c536a076768e38867e857d540e76cb4b9d679189fe225d1584eebf02812\n",
       "16  9fca162813402b542e02b80231a1f8b9ccb92ad11a074d82fa84a92800b884eb\n",
       "17  744a80d81523f9548744b3a244def8173a0811a3f50132842f347103c789880c\n",
       "18  e0e7b0d9c660c94660fe71f7d04a95d135057247115f03903eea7a9beb44eaaa\n",
       "19  e2ab5b48541d6eed63cd6f2b69c07490ddcefe689826088908c4f1c1670a0cf2\n",
       "20  c3c81a2b5b1f17811d2ab828edf1d4c65e8e4a6632964db73555c4b5737fadf0\n",
       "21  893b5829c9a176dfee0394aeaa8d9a303db65d5b282b26346dbf5f2b3b51c104\n",
       "22  4069cd6b590424b5c07af679bfc2e163b372373dac0d2925afdb24f3213fb69d\n",
       "23  8680903ad637315f6f6d909c9101f98009fe67abf10bbd0b7ba50accbf7884b1\n",
       "24  2995e91a26c65654a0c8cab7812e6518f83ca9a8f06de0f60ce253ca4c394ab7\n",
       "25  b1f31f4b3ce4969318c3b021e6d197c4ecf42384fce275785ecebf981fdfe295\n",
       "26  e686c40a03b666e9d8517add74956f1ab2b1efb966ce8f2c948a4fdd33e56696\n",
       "27  c7c9a4ae8201c6d1efa53c39143f480466ac9475f9f7eb72505d1ae4ac828e91\n",
       "28  24f6b6f9999ebaf16fb0a00ddd424bfcb71d56ba012784e92b3f23eb29817cad\n",
       "29  cda912cda3189f4d7dd8570082fccb0f0b04c71ce79affbeb05e1ed0a11b6240\n",
       "30  95ec3fc59c1746b08d4d16b558e42c53667dd0f8067be950cffbf9fa3193530d\n",
       "⋮   ⋮                                                               \n",
       "91  1d01ef0841a8d106923e5233a69055a44e9725f91fd0b8087ec9264897b598f3\n",
       "92  56208d65c2f89d737c0355956ac3ac2622492abfe47c6cd6ab779c6c93196c61\n",
       "93  6d5be9fdbe52d797f68fe158e0bc8be605e16866e95215cb2bc3bd3805b9941a\n",
       "94  0c66f19b598b8f492efe9d6e7471cda220553651da459a7579ef5f3ce69c7f09\n",
       "95  3bb158a6845f6b0f78649614f8ba0dd8150f9492cc9b54f847f25e643e351351\n",
       "96  1eabb511cc8a760e95072f68bc0448d84a0341055a9175a915b26e0360eabf6d\n",
       "97  928dfe0e89c5f7135d1fe9d640cd03462595005f5be27ce16e1ccda87c01c2e0\n",
       "98  c4f653e682dbc652b842938031f6273362300f9174811b06003ec090994d6b0b\n",
       "99  965cf032f314e3f0e03d46b57990306e682f26c2fbcb37c915ec942e4c6c78d9\n",
       "100 c542fb8af1d56fd9742f4df71c250ddbda4dc3cedb1956a4b5a6cd764596aa3f\n",
       "101 5ca589e15872e4880f8c3ba7aaca955ca588f04eb7a2c573e47616687defc9c7\n",
       "102 38692539c34b1d88f720658aeafd9a5487bc0f0a67cbd992e0324fd935ab5168\n",
       "103 fee2a9539994682c619e9b6115fc8a78296def6ccb2847729161259db6bf4fc2\n",
       "104 2cc5791ca6ca126b7561f60695fd4e4d2f52c6fdc6c05220b98e011255262b33\n",
       "105 20481d62565ca37c4c1acdc0eb64c97ae8d1ddcea55820489658f42e4825ea8d\n",
       "106 025e86c808aed60c9de41045475599f60061bf223da1bfe0c72e95d0ec7b5903\n",
       "107 68d98e3312a5f34e80eb04dd7b15c4374da3dfa17debd576bb1efc93d26241e8\n",
       "108 7990b7ea8db30990f33c84fa2bd296f51fe40a11b11753e3f8820f979ddd548f\n",
       "109 311394371de2e4f92ea10bb72902eebe40f40160cf01a32f2ac76d2994f8ff67\n",
       "110 42a2150a8399a854ae76a5b7f266d8c62a2c879f4561ae6c5f9ffe75df44f280\n",
       "111 4bdb2128e5ecb063c61ddc29ac14cf2f288bf26f4bf468fa556cc34adb0e9b24\n",
       "112 51b41d3376a554bf7653272830aac23fb1cba56288d2cf904a8a3d6d6a8709e4\n",
       "113 294f52f3e49a37927087dcf616c3dc2bb2561b82e970e32f5c48ba01b0677c16\n",
       "114 90de2ce3527a18a64ae44203ced07869713889061493588c15476dfdc8c55780\n",
       "115 66f6dff2b5c02370ea8ac82ce5aba51de70d3e911ea82adbfac8275b1c932c6c\n",
       "116 17d047f3a2c672a1d5e753e4c2af2a33665ae78278f374a33eedfe9b88cfff6a\n",
       "117 1baf27cd0b0ee6df74946ee69fb005085e0d04ae9568e7589b7dfbf9e704fd6e\n",
       "118 a376179b2fb762e65b148c26b2e496d872696b3800c8bba3b2ddc6a3f44c9669\n",
       "119 3b364672faccc397adae159deb645d6a2639e97b4328add5d8ec2f6798696150\n",
       "120 8291be77c7ad6cd73d9f7797658e3f1ffc197532a23385a85911ca4d4db13483\n",
       "    file                                                    \n",
       "1   ./liver_DGE.csv                                         \n",
       "2   ./esophagus_muscularis_DGE.csv                          \n",
       "3   ./brain_frontal_cortex_ba_9_DGE_ensg_map.csv            \n",
       "4   ./pituitary_DGE_refined.csv                             \n",
       "5   ./brain_cortex_DGE_refined.csv                          \n",
       "6   ./liver_DGE_ensg_map.csv                                \n",
       "7   ./esophagus_muscularis_DGE_ensg_map.csv                 \n",
       "8   ./whole_blood_DGE_refined.csv                           \n",
       "9   ./heart_atrial_appendage_DGE.csv                        \n",
       "10  ./nerve_tibial_DGE_ensg_map.csv                         \n",
       "11  ./heart_atrial_appendage_DGE_refined.csv                \n",
       "12  ./artery_tibial_DGE.csv                                 \n",
       "13  ./brain_spinal_cord_cervical_c_1_DGE_refined.csv        \n",
       "14  ./brain_hypothalamus_DGE_ensg_map.csv                   \n",
       "15  ./brain_spinal_cord_cervical_c_1_DGE_ensg_map.csv       \n",
       "16  ./spleen_DGE_ensg_map.csv                               \n",
       "17  ./pancreas_DGE_refined.csv                              \n",
       "18  ./brain_hippocampus_DGE_ensg_map.csv                    \n",
       "19  ./brain_cortex_DGE_ensg_map.csv                         \n",
       "20  ./gtex.rds                                              \n",
       "21  ./thyroid_DGE_ensg_map.csv                              \n",
       "22  ./spleen_DGE_refined.csv                                \n",
       "23  ./artery_aorta_DGE_refined.csv                          \n",
       "24  ./brain_putamen_basal_ganglia_DGE_ensg_map.csv          \n",
       "25  ./adrenal_gland_DGE_ensg_map.csv                        \n",
       "26  ./cells_cultured_fibroblasts_DGE_ensg_map.csv           \n",
       "27  ./artery_tibial_DGE_refined.csv                         \n",
       "28  ./colon_sigmoid_DGE_ensg_map.csv                        \n",
       "29  ./cells_ebv_transformed_lymphocytes_DGE_ensg_map.csv    \n",
       "30  ./breast_mammary_tissue_DGE.csv                         \n",
       "⋮   ⋮                                                       \n",
       "91  ./whole_blood_DGE_ensg_map.csv                          \n",
       "92  ./esophagus_mucosa_DGE_ensg_map.csv                     \n",
       "93  ./brain_cerebellum_DGE.csv                              \n",
       "94  ./brain_frontal_cortex_ba_9_DGE_refined.csv             \n",
       "95  ./brain_cortex_DGE.csv                                  \n",
       "96  ./lung_DGE_refined.csv                                  \n",
       "97  ./brain_hippocampus_DGE_refined.csv                     \n",
       "98  ./small_intestine_terminal_ileum_DGE_ensg_map.csv       \n",
       "99  ./esophagus_gastroesophageal_junction_DGE_ensg_map.csv  \n",
       "100 ./artery_coronary_DGE_refined.csv                       \n",
       "101 ./brain_hippocampus_DGE.csv                             \n",
       "102 ./brain_nucleus_accumbens_basal_ganglia_DGE_ensg_map.csv\n",
       "103 ./colon_transverse_DGE.csv                              \n",
       "104 ./cells_ebv_transformed_lymphocytes_DGE_refined.csv     \n",
       "105 ./breast_mammary_tissue_DGE_refined.csv                 \n",
       "106 ./lung_DGE_ensg_map.csv                                 \n",
       "107 ./stomach_DGE_ensg_map.csv                              \n",
       "108 ./muscle_skeletal_DGE_ensg_map.csv                      \n",
       "109 ./esophagus_gastroesophageal_junction_DGE.csv           \n",
       "110 ./nerve_tibial_DGE.csv                                  \n",
       "111 ./stomach_DGE.csv                                       \n",
       "112 ./nerve_tibial_DGE_refined.csv                          \n",
       "113 ./brain_hypothalamus_DGE_refined.csv                    \n",
       "114 ./esophagus_gastroesophageal_junction_DGE_refined.csv   \n",
       "115 ./stomach_DGE_refined.csv                               \n",
       "116 ./cells_cultured_fibroblasts_DGE_refined.csv            \n",
       "117 ./small_intestine_terminal_ileum_DGE_refined.csv        \n",
       "118 ./adipose_visceral_omentum_DGE_ensg_map.csv             \n",
       "119 ./lung_DGE.csv                                          \n",
       "120 ./SraRunTable.txt.gz                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "figure_id   = \"differentialGeneExpression\"\n",
    "\n",
    "message(\"Generating sha256 checksums of the artefacts in the `..data/` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -type f -exec sha256sum {} \\\\;  >  ../metadata/\", figure_id, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "data.table::fread(paste0(\"../metadata/\", figure_id, \"_sha256sums.txt\"), header = FALSE, col.names = c(\"sha256sum\", \"file\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5.2. Library metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\n",
      "\n",
      "Done!\n",
      "\n",
      "\n",
      "Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\n",
      "\n",
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       " setting  value                       \n",
       " version  R version 3.6.1 (2019-07-05)\n",
       " os       Ubuntu 18.04.4 LTS          \n",
       " system   x86_64, linux-gnu           \n",
       " ui       X11                         \n",
       " language en_US.UTF-8                 \n",
       " collate  en_US.UTF-8                 \n",
       " ctype    en_US.UTF-8                 \n",
       " tz       Etc/UTC                     \n",
       " date     2020-06-04                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A packages_info: 24 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>package</th><th scope=col>ondiskversion</th><th scope=col>loadedversion</th><th scope=col>path</th><th scope=col>loadedpath</th><th scope=col>attached</th><th scope=col>is_base</th><th scope=col>date</th><th scope=col>source</th><th scope=col>md5ok</th><th scope=col>library</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Biobase</th><td>Biobase     </td><td>2.46.0   </td><td>2.46.0   </td><td>/opt/conda/lib/R/library/Biobase     </td><td>/opt/conda/lib/R/library/Biobase     </td><td>TRUE</td><td>FALSE</td><td>2019-10-29</td><td>Bioconductor                              </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>BiocGenerics</th><td>BiocGenerics</td><td>0.32.0   </td><td>0.32.0   </td><td>/opt/conda/lib/R/library/BiocGenerics</td><td>/opt/conda/lib/R/library/BiocGenerics</td><td>TRUE</td><td>FALSE</td><td>2019-10-29</td><td>Bioconductor                              </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>biomaRt</th><td>biomaRt     </td><td>2.42.1   </td><td>2.42.1   </td><td>/opt/conda/lib/R/library/biomaRt     </td><td>/opt/conda/lib/R/library/biomaRt     </td><td>TRUE</td><td>FALSE</td><td>2020-03-26</td><td>Bioconductor                              </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>DBI</th><td>DBI         </td><td>1.1.0    </td><td>1.1.0    </td><td>/opt/conda/lib/R/library/DBI         </td><td>/opt/conda/lib/R/library/DBI         </td><td>TRUE</td><td>FALSE</td><td>2019-12-15</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>devtools</th><td>devtools    </td><td>2.3.0    </td><td>2.3.0    </td><td>/opt/conda/lib/R/library/devtools    </td><td>/opt/conda/lib/R/library/devtools    </td><td>TRUE</td><td>FALSE</td><td>2020-04-10</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>downloader</th><td>downloader  </td><td>0.4      </td><td>0.4      </td><td>/opt/conda/lib/R/library/downloader  </td><td>/opt/conda/lib/R/library/downloader  </td><td>TRUE</td><td>FALSE</td><td>2015-07-09</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>dplyr</th><td>dplyr       </td><td>0.8.5    </td><td>0.8.5    </td><td>/opt/conda/lib/R/library/dplyr       </td><td>/opt/conda/lib/R/library/dplyr       </td><td>TRUE</td><td>FALSE</td><td>2020-03-07</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>edgeR</th><td>edgeR       </td><td>3.28.1   </td><td>3.28.1   </td><td>/opt/conda/lib/R/library/edgeR       </td><td>/opt/conda/lib/R/library/edgeR       </td><td>TRUE</td><td>FALSE</td><td>2020-02-26</td><td>Bioconductor                              </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>ggplot2</th><td>ggplot2     </td><td>3.3.0    </td><td>3.3.0    </td><td>/opt/conda/lib/R/library/ggplot2     </td><td>/opt/conda/lib/R/library/ggplot2     </td><td>TRUE</td><td>FALSE</td><td>2020-03-05</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>gprofiler2</th><td>gprofiler2  </td><td>0.1.9    </td><td>0.1.9    </td><td>/opt/conda/lib/R/library/gprofiler2  </td><td>/opt/conda/lib/R/library/gprofiler2  </td><td>TRUE</td><td>FALSE</td><td>2020-04-23</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>limma</th><td>limma       </td><td>3.42.2   </td><td>3.42.2   </td><td>/opt/conda/lib/R/library/limma       </td><td>/opt/conda/lib/R/library/limma       </td><td>TRUE</td><td>FALSE</td><td>2020-02-03</td><td>Bioconductor                              </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>magrittr</th><td>magrittr    </td><td>1.5      </td><td>1.5      </td><td>/opt/conda/lib/R/library/magrittr    </td><td>/opt/conda/lib/R/library/magrittr    </td><td>TRUE</td><td>FALSE</td><td>2014-11-22</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>pheatmap</th><td>pheatmap    </td><td>1.0.12   </td><td>1.0.12   </td><td>/opt/conda/lib/R/library/pheatmap    </td><td>/opt/conda/lib/R/library/pheatmap    </td><td>TRUE</td><td>FALSE</td><td>2019-01-04</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>piggyback</th><td>piggyback   </td><td>0.0.10.99</td><td>0.0.10.99</td><td>/opt/conda/lib/R/library/piggyback   </td><td>/opt/conda/lib/R/library/piggyback   </td><td>TRUE</td><td>FALSE</td><td>2020-06-04</td><td>Github (ropensci/piggyback@87f71e8)       </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>readr</th><td>readr       </td><td>1.3.1    </td><td>1.3.1    </td><td>/opt/conda/lib/R/library/readr       </td><td>/opt/conda/lib/R/library/readr       </td><td>TRUE</td><td>FALSE</td><td>2018-12-21</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>repr</th><td>repr        </td><td>1.1.0    </td><td>1.1.0    </td><td>/opt/conda/lib/R/library/repr        </td><td>/opt/conda/lib/R/library/repr        </td><td>TRUE</td><td>FALSE</td><td>2020-01-28</td><td>CRAN (R 3.6.3)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>scales</th><td>scales      </td><td>1.1.1    </td><td>1.1.1    </td><td>/opt/conda/lib/R/library/scales      </td><td>/opt/conda/lib/R/library/scales      </td><td>TRUE</td><td>FALSE</td><td>2020-05-11</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>snakecase</th><td>snakecase   </td><td>0.11.0   </td><td>0.11.0   </td><td>/opt/conda/lib/R/library/snakecase   </td><td>/opt/conda/lib/R/library/snakecase   </td><td>TRUE</td><td>FALSE</td><td>2019-05-25</td><td>CRAN (R 3.6.3)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>statmod</th><td>statmod     </td><td>1.4.34   </td><td>1.4.34   </td><td>/opt/conda/lib/R/library/statmod     </td><td>/opt/conda/lib/R/library/statmod     </td><td>TRUE</td><td>FALSE</td><td>2020-02-17</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>stringr</th><td>stringr     </td><td>1.4.0    </td><td>1.4.0    </td><td>/opt/conda/lib/R/library/stringr     </td><td>/opt/conda/lib/R/library/stringr     </td><td>TRUE</td><td>FALSE</td><td>2019-02-10</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>usethis</th><td>usethis     </td><td>1.6.1    </td><td>1.6.1    </td><td>/opt/conda/lib/R/library/usethis     </td><td>/opt/conda/lib/R/library/usethis     </td><td>TRUE</td><td>FALSE</td><td>2020-04-29</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>viridis</th><td>viridis     </td><td>0.5.1    </td><td>0.5.1    </td><td>/opt/conda/lib/R/library/viridis     </td><td>/opt/conda/lib/R/library/viridis     </td><td>TRUE</td><td>FALSE</td><td>2018-03-29</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>viridisLite</th><td>viridisLite </td><td>0.3.0    </td><td>0.3.0    </td><td>/opt/conda/lib/R/library/viridisLite </td><td>/opt/conda/lib/R/library/viridisLite </td><td>TRUE</td><td>FALSE</td><td>2018-02-01</td><td>CRAN (R 3.6.1)                            </td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "\t<tr><th scope=row>yarn</th><td>yarn        </td><td>1.13.1   </td><td>1.13.1   </td><td>/opt/conda/lib/R/library/yarn        </td><td>/opt/conda/lib/R/library/yarn        </td><td>TRUE</td><td>FALSE</td><td>2020-06-04</td><td>Github (TheJacksonLaboratory/yarn@3ff72c0)</td><td>NA</td><td>/opt/conda/lib/R/library</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A packages\\_info: 24 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & package & ondiskversion & loadedversion & path & loadedpath & attached & is\\_base & date & source & md5ok & library\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <chr> & <chr> & <lgl> & <fct>\\\\\n",
       "\\hline\n",
       "\tBiobase & Biobase      & 2.46.0    & 2.46.0    & /opt/conda/lib/R/library/Biobase      & /opt/conda/lib/R/library/Biobase      & TRUE & FALSE & 2019-10-29 & Bioconductor                               & NA & /opt/conda/lib/R/library\\\\\n",
       "\tBiocGenerics & BiocGenerics & 0.32.0    & 0.32.0    & /opt/conda/lib/R/library/BiocGenerics & /opt/conda/lib/R/library/BiocGenerics & TRUE & FALSE & 2019-10-29 & Bioconductor                               & NA & /opt/conda/lib/R/library\\\\\n",
       "\tbiomaRt & biomaRt      & 2.42.1    & 2.42.1    & /opt/conda/lib/R/library/biomaRt      & /opt/conda/lib/R/library/biomaRt      & TRUE & FALSE & 2020-03-26 & Bioconductor                               & NA & /opt/conda/lib/R/library\\\\\n",
       "\tDBI & DBI          & 1.1.0     & 1.1.0     & /opt/conda/lib/R/library/DBI          & /opt/conda/lib/R/library/DBI          & TRUE & FALSE & 2019-12-15 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tdevtools & devtools     & 2.3.0     & 2.3.0     & /opt/conda/lib/R/library/devtools     & /opt/conda/lib/R/library/devtools     & TRUE & FALSE & 2020-04-10 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tdownloader & downloader   & 0.4       & 0.4       & /opt/conda/lib/R/library/downloader   & /opt/conda/lib/R/library/downloader   & TRUE & FALSE & 2015-07-09 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tdplyr & dplyr        & 0.8.5     & 0.8.5     & /opt/conda/lib/R/library/dplyr        & /opt/conda/lib/R/library/dplyr        & TRUE & FALSE & 2020-03-07 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tedgeR & edgeR        & 3.28.1    & 3.28.1    & /opt/conda/lib/R/library/edgeR        & /opt/conda/lib/R/library/edgeR        & TRUE & FALSE & 2020-02-26 & Bioconductor                               & NA & /opt/conda/lib/R/library\\\\\n",
       "\tggplot2 & ggplot2      & 3.3.0     & 3.3.0     & /opt/conda/lib/R/library/ggplot2      & /opt/conda/lib/R/library/ggplot2      & TRUE & FALSE & 2020-03-05 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tgprofiler2 & gprofiler2   & 0.1.9     & 0.1.9     & /opt/conda/lib/R/library/gprofiler2   & /opt/conda/lib/R/library/gprofiler2   & TRUE & FALSE & 2020-04-23 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tlimma & limma        & 3.42.2    & 3.42.2    & /opt/conda/lib/R/library/limma        & /opt/conda/lib/R/library/limma        & TRUE & FALSE & 2020-02-03 & Bioconductor                               & NA & /opt/conda/lib/R/library\\\\\n",
       "\tmagrittr & magrittr     & 1.5       & 1.5       & /opt/conda/lib/R/library/magrittr     & /opt/conda/lib/R/library/magrittr     & TRUE & FALSE & 2014-11-22 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tpheatmap & pheatmap     & 1.0.12    & 1.0.12    & /opt/conda/lib/R/library/pheatmap     & /opt/conda/lib/R/library/pheatmap     & TRUE & FALSE & 2019-01-04 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tpiggyback & piggyback    & 0.0.10.99 & 0.0.10.99 & /opt/conda/lib/R/library/piggyback    & /opt/conda/lib/R/library/piggyback    & TRUE & FALSE & 2020-06-04 & Github (ropensci/piggyback@87f71e8)        & NA & /opt/conda/lib/R/library\\\\\n",
       "\treadr & readr        & 1.3.1     & 1.3.1     & /opt/conda/lib/R/library/readr        & /opt/conda/lib/R/library/readr        & TRUE & FALSE & 2018-12-21 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\trepr & repr         & 1.1.0     & 1.1.0     & /opt/conda/lib/R/library/repr         & /opt/conda/lib/R/library/repr         & TRUE & FALSE & 2020-01-28 & CRAN (R 3.6.3)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tscales & scales       & 1.1.1     & 1.1.1     & /opt/conda/lib/R/library/scales       & /opt/conda/lib/R/library/scales       & TRUE & FALSE & 2020-05-11 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tsnakecase & snakecase    & 0.11.0    & 0.11.0    & /opt/conda/lib/R/library/snakecase    & /opt/conda/lib/R/library/snakecase    & TRUE & FALSE & 2019-05-25 & CRAN (R 3.6.3)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tstatmod & statmod      & 1.4.34    & 1.4.34    & /opt/conda/lib/R/library/statmod      & /opt/conda/lib/R/library/statmod      & TRUE & FALSE & 2020-02-17 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tstringr & stringr      & 1.4.0     & 1.4.0     & /opt/conda/lib/R/library/stringr      & /opt/conda/lib/R/library/stringr      & TRUE & FALSE & 2019-02-10 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tusethis & usethis      & 1.6.1     & 1.6.1     & /opt/conda/lib/R/library/usethis      & /opt/conda/lib/R/library/usethis      & TRUE & FALSE & 2020-04-29 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tviridis & viridis      & 0.5.1     & 0.5.1     & /opt/conda/lib/R/library/viridis      & /opt/conda/lib/R/library/viridis      & TRUE & FALSE & 2018-03-29 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tviridisLite & viridisLite  & 0.3.0     & 0.3.0     & /opt/conda/lib/R/library/viridisLite  & /opt/conda/lib/R/library/viridisLite  & TRUE & FALSE & 2018-02-01 & CRAN (R 3.6.1)                             & NA & /opt/conda/lib/R/library\\\\\n",
       "\tyarn & yarn         & 1.13.1    & 1.13.1    & /opt/conda/lib/R/library/yarn         & /opt/conda/lib/R/library/yarn         & TRUE & FALSE & 2020-06-04 & Github (TheJacksonLaboratory/yarn@3ff72c0) & NA & /opt/conda/lib/R/library\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A packages_info: 24 × 11\n",
       "\n",
       "| <!--/--> | package &lt;chr&gt; | ondiskversion &lt;chr&gt; | loadedversion &lt;chr&gt; | path &lt;chr&gt; | loadedpath &lt;chr&gt; | attached &lt;lgl&gt; | is_base &lt;lgl&gt; | date &lt;chr&gt; | source &lt;chr&gt; | md5ok &lt;lgl&gt; | library &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Biobase | Biobase      | 2.46.0    | 2.46.0    | /opt/conda/lib/R/library/Biobase      | /opt/conda/lib/R/library/Biobase      | TRUE | FALSE | 2019-10-29 | Bioconductor                               | NA | /opt/conda/lib/R/library |\n",
       "| BiocGenerics | BiocGenerics | 0.32.0    | 0.32.0    | /opt/conda/lib/R/library/BiocGenerics | /opt/conda/lib/R/library/BiocGenerics | TRUE | FALSE | 2019-10-29 | Bioconductor                               | NA | /opt/conda/lib/R/library |\n",
       "| biomaRt | biomaRt      | 2.42.1    | 2.42.1    | /opt/conda/lib/R/library/biomaRt      | /opt/conda/lib/R/library/biomaRt      | TRUE | FALSE | 2020-03-26 | Bioconductor                               | NA | /opt/conda/lib/R/library |\n",
       "| DBI | DBI          | 1.1.0     | 1.1.0     | /opt/conda/lib/R/library/DBI          | /opt/conda/lib/R/library/DBI          | TRUE | FALSE | 2019-12-15 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| devtools | devtools     | 2.3.0     | 2.3.0     | /opt/conda/lib/R/library/devtools     | /opt/conda/lib/R/library/devtools     | TRUE | FALSE | 2020-04-10 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| downloader | downloader   | 0.4       | 0.4       | /opt/conda/lib/R/library/downloader   | /opt/conda/lib/R/library/downloader   | TRUE | FALSE | 2015-07-09 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| dplyr | dplyr        | 0.8.5     | 0.8.5     | /opt/conda/lib/R/library/dplyr        | /opt/conda/lib/R/library/dplyr        | TRUE | FALSE | 2020-03-07 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| edgeR | edgeR        | 3.28.1    | 3.28.1    | /opt/conda/lib/R/library/edgeR        | /opt/conda/lib/R/library/edgeR        | TRUE | FALSE | 2020-02-26 | Bioconductor                               | NA | /opt/conda/lib/R/library |\n",
       "| ggplot2 | ggplot2      | 3.3.0     | 3.3.0     | /opt/conda/lib/R/library/ggplot2      | /opt/conda/lib/R/library/ggplot2      | TRUE | FALSE | 2020-03-05 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| gprofiler2 | gprofiler2   | 0.1.9     | 0.1.9     | /opt/conda/lib/R/library/gprofiler2   | /opt/conda/lib/R/library/gprofiler2   | TRUE | FALSE | 2020-04-23 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| limma | limma        | 3.42.2    | 3.42.2    | /opt/conda/lib/R/library/limma        | /opt/conda/lib/R/library/limma        | TRUE | FALSE | 2020-02-03 | Bioconductor                               | NA | /opt/conda/lib/R/library |\n",
       "| magrittr | magrittr     | 1.5       | 1.5       | /opt/conda/lib/R/library/magrittr     | /opt/conda/lib/R/library/magrittr     | TRUE | FALSE | 2014-11-22 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| pheatmap | pheatmap     | 1.0.12    | 1.0.12    | /opt/conda/lib/R/library/pheatmap     | /opt/conda/lib/R/library/pheatmap     | TRUE | FALSE | 2019-01-04 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| piggyback | piggyback    | 0.0.10.99 | 0.0.10.99 | /opt/conda/lib/R/library/piggyback    | /opt/conda/lib/R/library/piggyback    | TRUE | FALSE | 2020-06-04 | Github (ropensci/piggyback@87f71e8)        | NA | /opt/conda/lib/R/library |\n",
       "| readr | readr        | 1.3.1     | 1.3.1     | /opt/conda/lib/R/library/readr        | /opt/conda/lib/R/library/readr        | TRUE | FALSE | 2018-12-21 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| repr | repr         | 1.1.0     | 1.1.0     | /opt/conda/lib/R/library/repr         | /opt/conda/lib/R/library/repr         | TRUE | FALSE | 2020-01-28 | CRAN (R 3.6.3)                             | NA | /opt/conda/lib/R/library |\n",
       "| scales | scales       | 1.1.1     | 1.1.1     | /opt/conda/lib/R/library/scales       | /opt/conda/lib/R/library/scales       | TRUE | FALSE | 2020-05-11 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| snakecase | snakecase    | 0.11.0    | 0.11.0    | /opt/conda/lib/R/library/snakecase    | /opt/conda/lib/R/library/snakecase    | TRUE | FALSE | 2019-05-25 | CRAN (R 3.6.3)                             | NA | /opt/conda/lib/R/library |\n",
       "| statmod | statmod      | 1.4.34    | 1.4.34    | /opt/conda/lib/R/library/statmod      | /opt/conda/lib/R/library/statmod      | TRUE | FALSE | 2020-02-17 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| stringr | stringr      | 1.4.0     | 1.4.0     | /opt/conda/lib/R/library/stringr      | /opt/conda/lib/R/library/stringr      | TRUE | FALSE | 2019-02-10 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| usethis | usethis      | 1.6.1     | 1.6.1     | /opt/conda/lib/R/library/usethis      | /opt/conda/lib/R/library/usethis      | TRUE | FALSE | 2020-04-29 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| viridis | viridis      | 0.5.1     | 0.5.1     | /opt/conda/lib/R/library/viridis      | /opt/conda/lib/R/library/viridis      | TRUE | FALSE | 2018-03-29 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| viridisLite | viridisLite  | 0.3.0     | 0.3.0     | /opt/conda/lib/R/library/viridisLite  | /opt/conda/lib/R/library/viridisLite  | TRUE | FALSE | 2018-02-01 | CRAN (R 3.6.1)                             | NA | /opt/conda/lib/R/library |\n",
       "| yarn | yarn         | 1.13.1    | 1.13.1    | /opt/conda/lib/R/library/yarn         | /opt/conda/lib/R/library/yarn         | TRUE | FALSE | 2020-06-04 | Github (TheJacksonLaboratory/yarn@3ff72c0) | NA | /opt/conda/lib/R/library |\n",
       "\n"
      ],
      "text/plain": [
       "             package      ondiskversion loadedversion\n",
       "Biobase      Biobase      2.46.0        2.46.0       \n",
       "BiocGenerics BiocGenerics 0.32.0        0.32.0       \n",
       "biomaRt      biomaRt      2.42.1        2.42.1       \n",
       "DBI          DBI          1.1.0         1.1.0        \n",
       "devtools     devtools     2.3.0         2.3.0        \n",
       "downloader   downloader   0.4           0.4          \n",
       "dplyr        dplyr        0.8.5         0.8.5        \n",
       "edgeR        edgeR        3.28.1        3.28.1       \n",
       "ggplot2      ggplot2      3.3.0         3.3.0        \n",
       "gprofiler2   gprofiler2   0.1.9         0.1.9        \n",
       "limma        limma        3.42.2        3.42.2       \n",
       "magrittr     magrittr     1.5           1.5          \n",
       "pheatmap     pheatmap     1.0.12        1.0.12       \n",
       "piggyback    piggyback    0.0.10.99     0.0.10.99    \n",
       "readr        readr        1.3.1         1.3.1        \n",
       "repr         repr         1.1.0         1.1.0        \n",
       "scales       scales       1.1.1         1.1.1        \n",
       "snakecase    snakecase    0.11.0        0.11.0       \n",
       "statmod      statmod      1.4.34        1.4.34       \n",
       "stringr      stringr      1.4.0         1.4.0        \n",
       "usethis      usethis      1.6.1         1.6.1        \n",
       "viridis      viridis      0.5.1         0.5.1        \n",
       "viridisLite  viridisLite  0.3.0         0.3.0        \n",
       "yarn         yarn         1.13.1        1.13.1       \n",
       "             path                                 \n",
       "Biobase      /opt/conda/lib/R/library/Biobase     \n",
       "BiocGenerics /opt/conda/lib/R/library/BiocGenerics\n",
       "biomaRt      /opt/conda/lib/R/library/biomaRt     \n",
       "DBI          /opt/conda/lib/R/library/DBI         \n",
       "devtools     /opt/conda/lib/R/library/devtools    \n",
       "downloader   /opt/conda/lib/R/library/downloader  \n",
       "dplyr        /opt/conda/lib/R/library/dplyr       \n",
       "edgeR        /opt/conda/lib/R/library/edgeR       \n",
       "ggplot2      /opt/conda/lib/R/library/ggplot2     \n",
       "gprofiler2   /opt/conda/lib/R/library/gprofiler2  \n",
       "limma        /opt/conda/lib/R/library/limma       \n",
       "magrittr     /opt/conda/lib/R/library/magrittr    \n",
       "pheatmap     /opt/conda/lib/R/library/pheatmap    \n",
       "piggyback    /opt/conda/lib/R/library/piggyback   \n",
       "readr        /opt/conda/lib/R/library/readr       \n",
       "repr         /opt/conda/lib/R/library/repr        \n",
       "scales       /opt/conda/lib/R/library/scales      \n",
       "snakecase    /opt/conda/lib/R/library/snakecase   \n",
       "statmod      /opt/conda/lib/R/library/statmod     \n",
       "stringr      /opt/conda/lib/R/library/stringr     \n",
       "usethis      /opt/conda/lib/R/library/usethis     \n",
       "viridis      /opt/conda/lib/R/library/viridis     \n",
       "viridisLite  /opt/conda/lib/R/library/viridisLite \n",
       "yarn         /opt/conda/lib/R/library/yarn        \n",
       "             loadedpath                            attached is_base date      \n",
       "Biobase      /opt/conda/lib/R/library/Biobase      TRUE     FALSE   2019-10-29\n",
       "BiocGenerics /opt/conda/lib/R/library/BiocGenerics TRUE     FALSE   2019-10-29\n",
       "biomaRt      /opt/conda/lib/R/library/biomaRt      TRUE     FALSE   2020-03-26\n",
       "DBI          /opt/conda/lib/R/library/DBI          TRUE     FALSE   2019-12-15\n",
       "devtools     /opt/conda/lib/R/library/devtools     TRUE     FALSE   2020-04-10\n",
       "downloader   /opt/conda/lib/R/library/downloader   TRUE     FALSE   2015-07-09\n",
       "dplyr        /opt/conda/lib/R/library/dplyr        TRUE     FALSE   2020-03-07\n",
       "edgeR        /opt/conda/lib/R/library/edgeR        TRUE     FALSE   2020-02-26\n",
       "ggplot2      /opt/conda/lib/R/library/ggplot2      TRUE     FALSE   2020-03-05\n",
       "gprofiler2   /opt/conda/lib/R/library/gprofiler2   TRUE     FALSE   2020-04-23\n",
       "limma        /opt/conda/lib/R/library/limma        TRUE     FALSE   2020-02-03\n",
       "magrittr     /opt/conda/lib/R/library/magrittr     TRUE     FALSE   2014-11-22\n",
       "pheatmap     /opt/conda/lib/R/library/pheatmap     TRUE     FALSE   2019-01-04\n",
       "piggyback    /opt/conda/lib/R/library/piggyback    TRUE     FALSE   2020-06-04\n",
       "readr        /opt/conda/lib/R/library/readr        TRUE     FALSE   2018-12-21\n",
       "repr         /opt/conda/lib/R/library/repr         TRUE     FALSE   2020-01-28\n",
       "scales       /opt/conda/lib/R/library/scales       TRUE     FALSE   2020-05-11\n",
       "snakecase    /opt/conda/lib/R/library/snakecase    TRUE     FALSE   2019-05-25\n",
       "statmod      /opt/conda/lib/R/library/statmod      TRUE     FALSE   2020-02-17\n",
       "stringr      /opt/conda/lib/R/library/stringr      TRUE     FALSE   2019-02-10\n",
       "usethis      /opt/conda/lib/R/library/usethis      TRUE     FALSE   2020-04-29\n",
       "viridis      /opt/conda/lib/R/library/viridis      TRUE     FALSE   2018-03-29\n",
       "viridisLite  /opt/conda/lib/R/library/viridisLite  TRUE     FALSE   2018-02-01\n",
       "yarn         /opt/conda/lib/R/library/yarn         TRUE     FALSE   2020-06-04\n",
       "             source                                     md5ok\n",
       "Biobase      Bioconductor                               NA   \n",
       "BiocGenerics Bioconductor                               NA   \n",
       "biomaRt      Bioconductor                               NA   \n",
       "DBI          CRAN (R 3.6.1)                             NA   \n",
       "devtools     CRAN (R 3.6.1)                             NA   \n",
       "downloader   CRAN (R 3.6.1)                             NA   \n",
       "dplyr        CRAN (R 3.6.1)                             NA   \n",
       "edgeR        Bioconductor                               NA   \n",
       "ggplot2      CRAN (R 3.6.1)                             NA   \n",
       "gprofiler2   CRAN (R 3.6.1)                             NA   \n",
       "limma        Bioconductor                               NA   \n",
       "magrittr     CRAN (R 3.6.1)                             NA   \n",
       "pheatmap     CRAN (R 3.6.1)                             NA   \n",
       "piggyback    Github (ropensci/piggyback@87f71e8)        NA   \n",
       "readr        CRAN (R 3.6.1)                             NA   \n",
       "repr         CRAN (R 3.6.3)                             NA   \n",
       "scales       CRAN (R 3.6.1)                             NA   \n",
       "snakecase    CRAN (R 3.6.3)                             NA   \n",
       "statmod      CRAN (R 3.6.1)                             NA   \n",
       "stringr      CRAN (R 3.6.1)                             NA   \n",
       "usethis      CRAN (R 3.6.1)                             NA   \n",
       "viridis      CRAN (R 3.6.1)                             NA   \n",
       "viridisLite  CRAN (R 3.6.1)                             NA   \n",
       "yarn         Github (TheJacksonLaboratory/yarn@3ff72c0) NA   \n",
       "             library                 \n",
       "Biobase      /opt/conda/lib/R/library\n",
       "BiocGenerics /opt/conda/lib/R/library\n",
       "biomaRt      /opt/conda/lib/R/library\n",
       "DBI          /opt/conda/lib/R/library\n",
       "devtools     /opt/conda/lib/R/library\n",
       "downloader   /opt/conda/lib/R/library\n",
       "dplyr        /opt/conda/lib/R/library\n",
       "edgeR        /opt/conda/lib/R/library\n",
       "ggplot2      /opt/conda/lib/R/library\n",
       "gprofiler2   /opt/conda/lib/R/library\n",
       "limma        /opt/conda/lib/R/library\n",
       "magrittr     /opt/conda/lib/R/library\n",
       "pheatmap     /opt/conda/lib/R/library\n",
       "piggyback    /opt/conda/lib/R/library\n",
       "readr        /opt/conda/lib/R/library\n",
       "repr         /opt/conda/lib/R/library\n",
       "scales       /opt/conda/lib/R/library\n",
       "snakecase    /opt/conda/lib/R/library\n",
       "statmod      /opt/conda/lib/R/library\n",
       "stringr      /opt/conda/lib/R/library\n",
       "usethis      /opt/conda/lib/R/library\n",
       "viridis      /opt/conda/lib/R/library\n",
       "viridisLite  /opt/conda/lib/R/library\n",
       "yarn         /opt/conda/lib/R/library"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../metadata/\", figure_id, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../metadata/\", figure_id ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
