{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Figure 2 - Characteristics of sex-biased alternative splicing.\n",
    "\n",
    "This will need to become Figure 3.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2b `Sex-biased alternative splicing index for each chromosome.`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This plot is based upon a file 'SplicingIndex_chr.tsv' that was generated by xx program.   It is sorted\n",
    "in decreasing order by index, and the size of the circle reflects XXXX."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(ggplot2)\n",
    "Sys.setenv(TAR = \"/bin/tar\") # for gzfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculation of the Sex-Biased Splicing Index\n",
    "The normalized sex-biased splicing index is defined as the number of statistically significant splicing events per 1000 exons in the chromosome."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The plot will post the ranked by splicing index on the y-axis by the chromosomes on the x-axis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# a code block\n",
    "getwd()\n",
    "dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/figure2_test_files/SplicingIndex_chr.tsv\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort by index\n",
    "res_sorted <- dat[order(dat$Index, decreasing=TRUE),]\n",
    "res_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# though we read in the `tsv` file with stringsAsFactors = true, lets ensure this is the case\n",
    "res_sorted$chr <- factor(res_sorted$chr, levels = res_sorted$chr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set the colors\n",
    "npgBlue<- rgb(60/256,84/256,136/256,1)\n",
    "npgRed <- rgb(220/256,0,0,0.5)\n",
    "npgGreen <- rgb(0,160/256,135/256,1)\n",
    "npgBrown <- rgb(126/256,97/256,72/256,1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make the plot \n",
    "figure2b <- ggplot(res_sorted, aes(x = chr, y = Index, size = n)) +\n",
    "  geom_point(color=npgBlue) +\n",
    "  theme_bw() +\n",
    "  theme(axis.text.x = element_text(size=14, angle = 270, hjust = 0.0, vjust = 0.5),\n",
    "\taxis.text.y = element_text(size=16),\n",
    "\taxis.title.x = element_blank(),\n",
    "\taxis.title.y = element_text(face=\"plain\", colour=\"black\",\n",
    "                                    size=18),\n",
    "\tlegend.title=element_blank(),\n",
    "\tlegend.text = element_text(face=\"plain\", colour=\"black\",\n",
    "                                   size=14)) +\n",
    "  scale_fill_viridis_c() +\n",
    "  ylab(paste(\"Sex-biased splicing index \")) +\n",
    "  xlab(\"Chromosomes\") +\n",
    "  guides(size = guide_legend(title = \"Number of ASE\"))\n",
    "figure2b\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "##Â save as a pdf\n",
    "pdf (\"../pdf/figure2b.pdf\")\n",
    "figure2b\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2c - Proportions of AS events that were strictly tissue specific, present in 2-5, or over 5 tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(ggplot2)\n",
    "Sys.setenv(TAR = \"/bin/tar\") # for gzfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read in the file `genesWithCommonAs.tsv` containing number of splicing events per gene and number of tissues."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/figure2_test_files/genesWithCommonAS.tsv\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting Proportions\n",
    "Making a table of the number of tissues per gene will get us the \n",
    "numbers we need for the proportions of alternative splicing events (AS) \n",
    "that are strictly tissue specific"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_specificity_cnt <- table(dat$Tissues)\n",
    "tissue_specificity_cnt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make an Array of the table\n",
    "Make an array of the table to access the values based upon the index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a <- unclass(tissue_specificity_cnt)\n",
    "a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####  Names for logic\n",
    "\n",
    "Using the names of this array contain the numbers of genes\n",
    "with those number of splicing events, which we then can use for logical reasoning, such as number of genes that have tissue specific splicing events.  The length of this vector is equal to the length of the array.\n",
    "Using this logical value within the array will allow us to sum the events for our figure."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_cnts <- as.numeric(unclass(names(a)))\n",
    "tissue_cnts == 1\n",
    "a[tissue_cnts == 1]\n",
    "low_cnt <- sum(as.numeric(a[tissue_cnts == 1]))\n",
    "low_cnt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a[tissue_cnts > 1 & tissue_cnts < 6]\n",
    "medium_cnt <- sum(as.numeric(a[tissue_cnts > 1 & tissue_cnts < 6]))\n",
    "medium_cnt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a[tissue_cnts > 4]\n",
    "high_cnt <- sum(as.numeric(a[tissue_cnts > 4]))\n",
    "high_cnt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## set the colors\n",
    "npgBlue<- rgb(60/256,84/256,136/256,1)\n",
    "npgRed <- rgb(220/256,0,0,0.8)\n",
    "npgGreen <- rgb(0,160/256,135/256,1)\n",
    "npgBrown <- rgb(126/256,97/256,72/256,1)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make the pie chart figure 2c illustrating number of genes that are spliced and present in only 1 tissue, 2-5 tissues and the number of genes that are alternatively spliced in greater than 5 tissues."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colrs <- c(npgRed,npgBlue,npgGreen)\n",
    "slices <-c(low_cnt, medium_cnt, high_cnt)\n",
    "lbls <-c(\"1 tissue\",\"2-5 tissues\",\">5 tissues\")\n",
    "pct <- round(slices/sum(slices)*100, digits=1)\n",
    "lbls <- paste(lbls, pct) # add percents to labels\n",
    "lbls <- paste(lbls,\"%\",sep=\"\") # ad % to labels\n",
    "pie(slices,labels = lbls, col=colrs,main=\"\",cex=1.5,xaxs=\"r\",yaxs=\"r\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "pdf (\"../pdf/figure2c.pdf\")\n",
    "pie(slices,labels = lbls, col=colrs,main=\"\",cex=1.5,xaxs=\"r\",yaxs=\"r\")\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2d - Genes with over 10 significant sex-biased AS events."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "PREVIOUS VERSION Most of the AS events were specific to one tissue, but slightly over 12% were found in 2-5 tissues, with only 9 AS events being found in more than 5 tissues (Fig. 2c). 27 genes were found to have >10 AS events (Fig. 2d). \n",
    "\n",
    "-- We basically need to count the number of tissues per AS event (Figure 2c) and the number of AS events per gene (FIgure 2d)\n",
    "\n",
    "\n",
    "\n",
    "The colour coding represents the number of affected tissues, sorted in decreasing order by the number of sex-biased splicing events per gene."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "According to installation instructions here https://github.com/tidyverse/dplyr it is recommended to just install the whole tidyverse."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "library(magrittr)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read in the output of the XX script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat <- read.table(file =\"https://github.com/adeslatt/sbas_test/releases/download/figure2_test_files/genesWithCommonAs.tsv\",\n",
    "                sep = \"\\t\", quote = NULL, header = TRUE, stringsAsFactors = FALSE)\n",
    "head(dat)\n",
    "dim(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove all genes that have less than 10 splicing events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "res <- dat[dat$n > 10,]\n",
    "head(res)\n",
    "dim(res)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Arrange the data in descending order in terms of the number of splicing events per gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "res <- res[order(res$n, decreasing=TRUE),]\n",
    "head(res)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make sure the GeneSymbols are treated as factors, to ease the plotting.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "res$GeneSymbol <- factor(res$GeneSymbol, levels = res$GeneSymbol)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now make the plot, using viridis to fill the dots according to the amount of splicing per tissue.  The number of splicing events per gene does not necessarily correlate with the number of tissues that gene is involved with splicing.  Fewer tissues, more splicing events, mean that within a tissue there are more splicing events.  More Tissues and fewer splicing events for that gene indicates a level of tissue specificity with the splicing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p<-ggplot(res, aes(x = GeneSymbol, y = n)) +\n",
    "  geom_point(size = 4, aes(fill = Tissues, color = Tissues)) +\n",
    "  theme_bw() +\n",
    "  theme(axis.text.x = element_text(size=14, angle = 270, hjust = 0.0, vjust = 0.5),\n",
    "        axis.text.y = element_text(size=16),\n",
    "        axis.title.x = element_blank(),\n",
    "        axis.title.y = element_text(face=\"plain\", colour=\"black\", \n",
    "                                    size=16),\n",
    "        legend.title=element_blank(),\n",
    "        legend.text = element_text(face=\"plain\", colour=\"black\", \n",
    "                                   size=16)) +\n",
    "  scale_fill_viridis_c(aesthetics = c(\"colour\", \"fill\"),\n",
    "                       option = \"plasma\",\n",
    "                       limits = c(1, 30), breaks = c(10, 20, 30)) +\n",
    "  ylab(paste(\"Number of sex-biased splicing events\"))\n",
    "\n",
    "p\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save the figure to the pdf directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "ggsave(file = \"../pdf/Figure2d.pdf\", plot = p)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2e - Genes with over 10 significant sex-biased AS events."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The colour coding represents the number of affected tissues, sorted in decreasing order by the number of sex-biased splicing events per gene."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(viridis)\n",
    "library(ggplot2)\n",
    "library(magrittr)\n",
    "Sys.setenv(TAR = \"/bin/tar\") # for gzfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reading the Significant Splicing Events by Tissue Results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Starting with the output from running xx script, this file that summarizes the significant splicing events by tissue is then limited to only those tissues that have greater than 10 splicing events, and then the distribution of the 5 splicing events, A3SS, A5SS, RI, SE, and MXE as a percentage of the whole are compared between the tissues."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df <- read.table(file =\"https://github.com/adeslatt/sbas_test/releases/download/figure2_test_files/Significant_events.tsv\",\n",
    "                sep = \"\\t\", quote = NULL, header = TRUE, stringsAsFactors = FALSE)\n",
    "head(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Another file, which created simplifying labels is read in as well.   It is produced by XX script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "totals <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/figure2e_test_file/Totals_by_tissue_annotated.txt\", \n",
    "                     sep = \"\\t\", quote = NULL, header = TRUE, stringsAsFactors = FALSE)\n",
    "head(totals)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the summarizing label, where the tissues match, annotated, to summarized output, we provide the annotation label.  This will help us make the figure with cleaner labels."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "idx <- match(df$Tissue, totals$Tissue)\n",
    "df$Label <- totals$Label[idx]\n",
    "df$Label <- factor(df$Label, levels = totals$Label)\n",
    "\n",
    "dim(df)\n",
    "dim(totals)\n",
    "head(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Just as a check, lets ensure that the total splicing events has not been altered."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sum(df$Counts)\n",
    "sum(totals$Total)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What is the distribution of the one file with annotation for the various counts of splicing events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "table(totals$Total)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One might expect greater granularity in the significant events file, that is not summarized, and as expected, the totals are lower as they are not summarized by tissue, but still broken out by each of the splicing events, which we want to preserve for this figure."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "table(df$Counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are limiting the figure to only those tissues that have greater than ten splicing events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissues_keep <- totals[totals$Total > 10,]\n",
    "head(tissues_keep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_plot <- data.frame()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a bit of a sanity check, before the loop is run, to see that the summary label works as we would like.  What we expect to see here is the same label on the far right for each of the alternative splicing events.  Looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tissue <- df[(df$Label %in% totals$Label[1]),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_df_counts <- sum(df$Counts)\n",
    "total_df_counts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Another check, before we go for the loop, to make sure all is clean -- probably not necessary to break out each of the alternative splicing events separately, but it keeps ones mind clear on these events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "length(totals$Label)\n",
    "length(tissues_keep$Label)\n",
    "tissue_splicing_events = sum(df$Counts[(df$ASE=='A3SS' | \n",
    "               df$ASE=='A5SS' |\n",
    "               df$ASE=='RI'   |\n",
    "               df$ASE=='MXE'  |\n",
    "               df$ASE=='SE'   ) & df$Label == tissues_keep$Label[1]])\n",
    "\n",
    "tissue_splicing_events\n",
    "all_splicing_events = sum(tissues_keep$Total)\n",
    "all_splicing_events\n",
    "percentage <- tissue_splicing_events/all_splicing_events*100\n",
    "percentage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_splicing_events = sum (tissues_keep$Total)\n",
    "total_splicing_events\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we create a new data frame (df_plot), which contains the label, as well as the percentages.   This will be our data for our plot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_plot <- data.frame()\n",
    "for (i in 1:length(tissues_keep$Label)) {\n",
    "    tissue_splicing_events = sum(df$Counts[(df$ASE=='A3SS' | \n",
    "                                          df$ASE=='A5SS' |\n",
    "                                          df$ASE=='RI'   |\n",
    "                                          df$ASE=='MXE'  |\n",
    "                                          df$ASE=='SE'   ) \n",
    "                                        & df$Label == tissues_keep$Label[i]])\n",
    "    \n",
    "    df_tissue      <- df[(df$Label %in% tissues_keep$Label[i]),]\n",
    "    df_tissue$Perc <- (df_tissue$Counts / tissue_splicing_events) * 100\n",
    "    if (i == 1)\n",
    "        df_plot <- df_tissue\n",
    "    else\n",
    "        df_plot <- rbind(df_plot, df_tissue)\n",
    "}  \n",
    "dim(df_plot)\n",
    "head(df_plot)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we make our plot, it will be collored by the proportion of each splicing event for each of the tissues that have greater than 10 splicing events.   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p<-ggplot(df_plot, \n",
    "       aes(x=Label, y=Perc, fill = ASE, by = Tissue)) +\n",
    "  geom_bar(stat = \"identity\", width = 0.8) +\n",
    "  theme_bw() +\n",
    "  theme(axis.text.x = element_text(size=12, angle = 270, hjust = 0.0, vjust = 0.5),\n",
    "        #axis.ticks.x = element_blank(),\n",
    "        axis.text.y = element_text(size=12),\n",
    "        axis.title.x = element_text(face=\"plain\", colour=\"black\", size=12),\n",
    "        axis.title.y = element_text(face=\"plain\", colour=\"black\", size=12),\n",
    "        legend.title=element_blank(),\n",
    "        panel.background=element_blank(),\n",
    "        panel.border=element_blank(),\n",
    "        panel.grid.major=element_blank(),\n",
    "        legend.text = element_text(face=\"plain\", colour=\"black\", size=12)\n",
    "  ) +\n",
    "  #scale_fill_brewer(palette=\"Dark2\") +\n",
    "  #scale_fill_manual(values = palette_npg) +\n",
    "  scale_fill_viridis(discrete=TRUE) +\n",
    "  ylab(\"Splicing type (%)\") +\n",
    "  xlab(\"Tissue\")\n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(file = \"../pdf/Figure2e.pdf\", plot = p)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2extra - splicing by chromosome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Percentage is shown for the entire set of 631 investigated X chromosomal genes, as well as the subsets of 248 genes with differential expression (DE) and 166 with at least one AS event. P-values were derived with Fisherâs exact test comparing genes with DE or AS to the entire set of 631 genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "library(viridis)\n",
    "library(ggplot2)\n",
    "library(magrittr)\n",
    "Sys.setenv(TAR = \"/bin/tar\") # for gzfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read in the rMATS results for parsing the gencode or other GTF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rMATS 3.2.5 parses the annotation file, gencode v.30 in this example, and creates the SE, RI, MXE, A3SS, A5SS events."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# a code block\n",
    "getwd()\n",
    "fromGTFSE.dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/rmats_final.gencode.v30/fromGTF.SE.txt\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "fromGTFSE.dat$AStype <- 'SE'\n",
    "fromGTFRI.dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/rmats_final.gencode.v30/fromGTF.RI.txt\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "fromGTFRI.dat$AStype <- 'RI'\n",
    "fromGTFMXE.dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/rmats_final.gencode.v30/fromGTF.MXE.txt\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "fromGTFMXE.dat$AStype <- 'MXE'\n",
    "fromGTFA5SS.dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/rmats_final.gencode.v30/fromGTF.A5SS.txt\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "fromGTFA5SS.dat$AStype <- 'A5SS'\n",
    "fromGTFA3SS.dat <- read.table(\"https://github.com/adeslatt/sbas_test/releases/download/rmats_final.gencode.v30/fromGTF.A3SS.txt\", \n",
    "                  stringsAsFactors = TRUE,\n",
    "                  header=TRUE)\n",
    "fromGTFA3SS.dat$AStype <- 'A3SS'\n",
    "\n",
    "head(fromGTFSE.dat)\n",
    "head(fromGTFMXE.dat)\n",
    "head(fromGTFRI.dat)\n",
    "head(fromGTFA3SS.dat)\n",
    "head(fromGTFA5SS.dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summaryAS.df <- rbind(fromGTFSE.dat[,c(\"ID\",\"GeneID\",\"geneSymbol\",\"chr\",\"strand\",\"AStype\")],\n",
    "                      fromGTFRI.dat[,c(\"ID\",\"GeneID\",\"geneSymbol\",\"chr\",\"strand\",\"AStype\")],\n",
    "                      fromGTFMXE.dat[,c(\"ID\",\"GeneID\",\"geneSymbol\",\"chr\",\"strand\",\"AStype\")],\n",
    "                      fromGTFA5SS.dat[,c(\"ID\",\"GeneID\",\"geneSymbol\",\"chr\",\"strand\",\"AStype\")],\n",
    "                      fromGTFA3SS.dat[,c(\"ID\",\"GeneID\",\"geneSymbol\",\"chr\",\"strand\",\"AStype\")])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr_specific_splicing_SE <- table(summaryAS.df$chr[summaryAS.df$AStype == \"SE\"])\n",
    "a_chr_SE <- unclass(chr_specific_splicing_SE)\n",
    "# a_chr_SE\n",
    "chr_specific_splicing_RI <- table(summaryAS.df$chr[summaryAS.df$AStype == \"RI\"])\n",
    "a_chr_RI <- unclass(chr_specific_splicing_RI)\n",
    "# a_chr_RI\n",
    "chr_specific_splicing_MXE <- table(summaryAS.df$chr[summaryAS.df$AStype == \"MXE\"])\n",
    "a_chr_MXE <- unclass(chr_specific_splicing_MXE)\n",
    "# a_chr_MXE\n",
    "chr_specific_splicing_A3SS <- table(summaryAS.df$chr[summaryAS.df$AStype == \"A3SS\"])\n",
    "a_chr_A3SS <- unclass(chr_specific_splicing_A3SS)\n",
    "#a_chr_A3SS\n",
    "chr_specific_splicing_A5SS <- table(summaryAS.df$chr[summaryAS.df$AStype == \"A5SS\"])\n",
    "a_chr_A5SS <- unclass(chr_specific_splicing_A5SS)\n",
    "print(\"A5SS (other examples commented out):\")\n",
    "table(summaryAS.df$chr[summaryAS.df$AStype == \"A5SS\"])\n",
    "#a_chr_A5SS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr_splicing_dist <- rbind(cbind(data.matrix(a_chr_SE),   chr=names(a_chr_SE),  AStype = \"SE\"),\n",
    "                           cbind(data.matrix(a_chr_RI),   chr=names(a_chr_RI),  AStype = \"RI\"),\n",
    "                           cbind(data.matrix(a_chr_MXE),  chr=names(a_chr_MXE), AStype = \"MXE\"),\n",
    "                           cbind(data.matrix(a_chr_A3SS), chr=names(a_chr_A3SS),AStype = \"A3SS\"),\n",
    "                           cbind(data.matrix(a_chr_A5SS), chr=names(a_chr_A5SS),AStype = \"A5SS\"))\n",
    "colnames(chr_splicing_dist)=c(\"Counts\",\"Chr\",\"AStype\")\n",
    "counts <- chr_splicing_dist[,\"Counts\"]\n",
    "chr_splicing_df <- data.frame(chr_splicing_dist)\n",
    "head(chr_splicing_df)\n",
    "chr_splicing_df$Counts <- as.integer(counts)\n",
    "head(chr_splicing_df)\n",
    "head(chr_splicing_df$Counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(chr_splicing_dist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosomes <- names(a_chr_SE)\n",
    "chromosomes <- chromosomes[order(chromosomes)]\n",
    "chromosomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(chr_splicing_dist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr_splicing_df[chr_splicing_df$Chr==chromosomes[1],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#chr_splicing_df$Chr\n",
    "sum(chr_splicing_df[chr_splicing_df$Chr ==chromosomes[1],\"Counts\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_chromosome     <- chr_splicing_df[chr_splicing_df$Chr ==chromosomes[1],]\n",
    "df_chromosome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "percentage <- df_chromosome$Counts/sum(df_chromosome$Counts)*100\n",
    "percentage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_plot <- data.frame()\n",
    "\n",
    "for (i in 1:length(chromosomes)) {\n",
    "    df_chromosome  <- chr_splicing_df[chr_splicing_df$Chr ==chromosomes[i],]\n",
    "    percentage     <- df_chromosome$Counts/sum(df_chromosome$Counts)*100\n",
    "    final <- cbind(df_chromosome, Perc=percentage)\n",
    "    if (i == 1)\n",
    "        df_plot <- final \n",
    "    else\n",
    "        df_plot <- rbind(df_plot, final)\n",
    "}  \n",
    "\n",
    "dim(df_plot)\n",
    "head(df_plot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p<-ggplot(df_plot, \n",
    "       aes(x=Chr, y=Perc, fill = AStype, by = Chr)) +\n",
    "  geom_bar(stat = \"identity\", width = 0.8) +\n",
    "  theme_bw() +\n",
    "  theme(axis.text.x = element_text(size=12, angle = 270, hjust = 0.0, vjust = 0.5),\n",
    "        #axis.ticks.x = element_blank(),\n",
    "        axis.text.y = element_text(size=12),\n",
    "        axis.title.x = element_text(face=\"plain\", colour=\"black\", size=12),\n",
    "        axis.title.y = element_text(face=\"plain\", colour=\"black\", size=12),\n",
    "        legend.title=element_blank(),\n",
    "        panel.background=element_blank(),\n",
    "        panel.border=element_blank(),\n",
    "        panel.grid.major=element_blank(),\n",
    "        legend.text = element_text(face=\"plain\", colour=\"black\", size=12)\n",
    "  ) +\n",
    "  #scale_fill_brewer(palette=\"Dark2\") +\n",
    "  #scale_fill_manual(values = palette_npg) +\n",
    "  scale_fill_viridis(discrete=TRUE) +\n",
    "  ylab(\"Splicing type (%)\") +\n",
    "  xlab(\"Chromosome\")\n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "ggsave(file = \"../pdf/figureSplicingByChromosome.pdf\", plot = p)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'** (British spelling), files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_id   = \"Figure2\"\n",
    "\n",
    "message(\"Generating sha256 checksums of the artefacts in the `..data/` directory .. \")\n",
    "system(paste0(\"cd ../data/ && sha256sum * > ../metadata/\", figure_id, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "data.table::fread(paste0(\"../metadata/\", figure_id, \"_sha256sums.txt\"), header = FALSE, col.names = c(\"sha256sum\", \"file\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Â 2. Libraries metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../metadata/\", figure_id, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../metadata/\", figure_id ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
