{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysis Notebook - Fisher exact test\n",
    "\n",
    "This notebook counts up differentially expressed and differentially spliced genes and calculates their overlap."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1  Read in all and significant alternative splicing and differential gene expression results\n",
    "\n",
    "The summary data captured in **all_gene_as.tsv**, **all_genes_dge_data** and significant results captured in **gene_as.tsv** and **gene_dge.tsv** (these filees are generated by **countGenesAndEvents.R**, which must be run before this notebook)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "results_dir  <- \"../data/\"\n",
    "all_genes_as_data  <- read.table(\"../data/all_gene_as.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "all_genes_dge_data <- read.table(\"../assets/all_gene_dge.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "sig_gene_as  <- read.table(file=\"../data/gene_as.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "sig_gene_dge  <- read.table(file=\"../data/gene_dge.tsv\", header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "head(sig_gene_as,2)\n",
    "head(sig_gene_dge,2)\n",
    "head(all_genes_as_data,2)\n",
    "head(all_genes_dge_data,2)\n",
    "head(all_genes_as_data$GeneSymbol,2)\n",
    "head(all_genes_dge_data$GeneSymbol,2)\n",
    "head(sig_gene_as$GeneSymbol,2)\n",
    "head(sig_gene_dge$GeneSymbol,2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2  Count up genes\n",
    "About 250 genes were found in the splicing data but not in the gene expression data, presumably related to the different processing pipelines used. We can therefore not make an assessment of whether these genes were differentially expressed or not, and thus we remove the genes prior to further analysis.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# all genes identified in the gene expression data\n",
    "allExpressionGenes <- unique(sort(all_genes_dge_data$GeneSymbol))\n",
    "# all genes identified in the splicing data\n",
    "allSplicingGenes  <- unique(sort(all_genes_as_data$GeneSymbol))\n",
    "# Genes found in splicing data but not in expression data\n",
    "orphanSplicingGenes <- setdiff(allSplicingGenes,allExpressionGenes)\n",
    "message(\"All expression genes n=\", length(allExpressionGenes),\"; all splicing genes n=\", length(allSplicingGenes), \"; splicing genes not represented in expression set n=\", length(orphanSplicingGenes))\n",
    "correctedSplicing <- setdiff(allSplicingGenes, orphanSplicingGenes)\n",
    "message(\"Note that we expect to find genes in the expression set that are not in the splicing set\")\n",
    "message(\"After removing the orphan splicing genes, we are left with  \", length(correctedSplicing), \" genes in the splicing dataset\")\n",
    "universe <- allExpressionGenes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create the sets of differentially expressed/spliced genes\n",
    "Note that we also need to correct the set of differentially spliced genes as above"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sigDGEGenes <- unique(sort(sig_gene_dge$GeneSymbol))\n",
    "sigASGenes  <- unique(sort(sig_gene_as$GeneSymbol))\n",
    "correctedSigASGenes <- setdiff(sigASGenes, orphanSplicingGenes)\n",
    "message(\"total AS (uncorrected) n=\", length(sigASGenes), \"; corrected n=\", length(correctedSigASGenes))\n",
    "total <- length(universe)\n",
    "n_dge <- length(sigDGEGenes)\n",
    "n_das <- length(correctedSigASGenes)\n",
    "message(\"significant differentially expresssed genes: n=\", n_dge, \"/\", total, \": \", 100*n_dge/total,\"%\")\n",
    "message(\"significant differentially spliced genes: n=\", n_das, \"/\", total, \": \", 100*n_das/total,\"%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dge_but_not_das <- setdiff(sigDGEGenes, correctedSigASGenes)\n",
    "das_but_not_dge <- setdiff(correctedSigASGenes, sigDGEGenes)\n",
    "dge_and_das <- intersect(sigDGEGenes, correctedSigASGenes)\n",
    "neither_dge_nor_das <- setdiff(setdiff(universe,sigDGEGenes), correctedSigASGenes)\n",
    "n_dge_but_not_das <- length(dge_but_not_das)\n",
    "n_das_but_not_dge <- length(das_but_not_dge)\n",
    "n_dge_and_das <- length(dge_and_das)\n",
    "n_neither_dge_nor_das <- length(neither_dge_nor_das)\n",
    "message(\"Differentially expressed but not differentially spliced: n=\", n_dge_but_not_das,\"/\", n_dge, \": \", 100*n_dge_but_not_das/n_dge, \"% of all DGE genes\")\n",
    "message(\"Differentially spliced but not differentially spliced: n=\", n_das_but_not_dge,\"/\", n_das, \": \", 100*n_das_but_not_dge/n_das, \"% of all DAS genes\")\n",
    "message(\"DGE and DAS: \", n_dge_and_das,\"; \", 100*n_dge_and_das/total,\"% of all genes\")\n",
    "expected_proportion <- (n_dge/total)*(n_das/total)\n",
    "message(\"By chance we would expect \", expected_proportion*total,\", or \", 100*expected_proportion, \"%\")\n",
    "message(\"Number of genes with neither DGE nor DAS \", n_neither_dge_nor_das)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2.1 Check whether the increased proportion is statistically significant\n",
    "\n",
    "Comparing differentially expressed genes with differentially alternatively spliced:\n",
    "\n",
    "|  \t|  DGE+| DGE-|\n",
    "|-\t|-\t|-\t|\n",
    "| DAS+|  a|  b|\n",
    "| DAS-|  c| d|"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "a <- n_dge_and_das\n",
    "b <- n_das_but_not_dge\n",
    "c <- n_dge_but_not_das\n",
    "d <- n_neither_dge_nor_das\n",
    "m <- matrix(c(a,b,c,d), nrow=2,byrow = TRUE)\n",
    "m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fisher.test(m)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "encoding": "# -*- coding: utf-8 -*-",
   "main_language": "R",
   "notebook_metadata_filter": "-all"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
