{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5c136c47",
   "metadata": {
    "tags": [
     "papermill-error-cell-tag"
    ]
   },
   "source": [
    "<span style=\"color:red; font-family:Helvetica Neue, Helvetica, Arial, sans-serif; font-size:2em;\">An Exception was encountered at '<a href=\"#papermill-error-cell\">In [11]</a>'.</span>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "292878df",
   "metadata": {
    "papermill": {
     "duration": 0.022735,
     "end_time": "2021-05-21T04:15:37.590439",
     "exception": false,
     "start_time": "2021-05-21T04:15:37.567704",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for plotting with MASER\n",
    "\n",
    "To accomplish the scale computing, the rMATS 3.2.5 was run in parallel in a pair-wise, sample by sample manner.   Aggregation of results was done by creating a matrix, deconvolving the output produced by rMATS 3.2.5.   To prepare specific data, genes and their transcript isoforms for plotting, we reconstruct the output putting it back into the standard rMATS 3.2.5 format.\n",
    "\n",
    "## Alternative Splice Site Types are: (se, a3ss, a5ss, mxe, ri)\n",
    "\n",
    "  * Skipped Exon events (se),\n",
    "  * Alternative 3' splice site (a3ss),\n",
    "  * Alternative 5' splice site (a5ss),\n",
    "  * Mutually exclusive exon (mxe),\n",
    "  * and retention intron (ri)\n",
    "\n",
    "## There are two different kinds of junction counts\n",
    "\n",
    "For our analysis here, we used just the jc count matrices.\n",
    "  * jc = junction counts - reads that cross the junction\n",
    "  * jcec = junction counts plus reads on the target (such as included exon)\n",
    "\n",
    "## And the count type -- there are 5 types\n",
    "\n",
    "  * inclusion levels (percent spliced in)\n",
    "  * included junction counts (ijc)\n",
    "  * skipped junction counts (sjc)\n",
    "  * inclusion length (inclen)\n",
    "  * skipped length (skiplen)\n",
    "\n",
    "## 1. Loading dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "626cc125",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:37.670240Z",
     "iopub.status.busy": "2021-05-21T04:15:37.667785Z",
     "iopub.status.idle": "2021-05-21T04:15:38.589401Z",
     "shell.execute_reply": "2021-05-21T04:15:38.588395Z"
    },
    "papermill": {
     "duration": 0.977522,
     "end_time": "2021-05-21T04:15:38.589657",
     "exception": false,
     "start_time": "2021-05-21T04:15:37.612135",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "suppressMessages({\n",
    "library(readr)\n",
    "library(edgeR)\n",
    "library(limma)\n",
    "library(Biobase)\n",
    "library(tibble)\n",
    "library(R.utils)\n",
    "library(snakecase)\n",
    "library(icesTAF)\n",
    "})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b49cc6b",
   "metadata": {
    "papermill": {
     "duration": 0.021158,
     "end_time": "2021-05-21T04:15:38.632834",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.611676",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# 1.0 Data preparation\n",
    "\n",
    "We need two things:\n",
    "\n",
    "* list of genes of interest\n",
    "* rmats results files\n",
    "\n",
    "The significant results were obtained on a per-sample bases.   \n",
    "Reconstruction of the arranged SAMPLE1 will be the male samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f33d578",
   "metadata": {
    "papermill": {
     "duration": 0.021003,
     "end_time": "2021-05-21T04:15:38.674825",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.653822",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 1.2 get rMATS GTF annotations\n",
    "\n",
    "For each splicing type, the junctions are defined, so we have 5 specific annotated splicing specific junction ID annotation files:\n",
    "fromGTF.A3SS.txt <- annotations for the alternative 3' splice site junctions\n",
    "fromGTF.A5SS.txt <- annotations for the alternative 5' splice site junctions\n",
    "fromGTF.MXE.txt <- annotations for the mutually exclusive exon junctions\n",
    "fromGTF.RI.txt <- annotations for the retained introns junctions\n",
    "fromGTF.SE.txt <- annotations for the skipped exon junctions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4ed02531",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:38.757363Z",
     "iopub.status.busy": "2021-05-21T04:15:38.722592Z",
     "iopub.status.idle": "2021-05-21T04:15:38.804201Z",
     "shell.execute_reply": "2021-05-21T04:15:38.802861Z"
    },
    "papermill": {
     "duration": 0.107509,
     "end_time": "2021-05-21T04:15:38.804330",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.696821",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "getReleasedGTFAnnotations <- function ( destDir ) {\n",
    "\n",
    "    message(\"Decompressing fromGTF.tar.gz into ../data\")\n",
    "    system(\"mkdir -p ../data && tar xvfz ../data/fromGTF.tar.gz -C ../data\", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "    message(\"Gunzipping files into ../data\")\n",
    "    system(\"gunzip ../data/fromGTF.*txt.gz\", intern = TRUE)\n",
    "    message(\"Done!\\n\")\n",
    "   return (0)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf13890c",
   "metadata": {
    "papermill": {
     "duration": 0.021202,
     "end_time": "2021-05-21T04:15:38.846726",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.825524",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 1.2.6 get reduced Tissue Data\n",
    "\n",
    "Stored in the assets subdirectory, reduced by inspection and selection focusing on those tissues with sufficient samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fe2a6c90",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:38.894269Z",
     "iopub.status.busy": "2021-05-21T04:15:38.893114Z",
     "iopub.status.idle": "2021-05-21T04:15:38.904349Z",
     "shell.execute_reply": "2021-05-21T04:15:38.903387Z"
    },
    "lines_to_next_cell": 2,
    "papermill": {
     "duration": 0.036666,
     "end_time": "2021-05-21T04:15:38.904487",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.867821",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "getTissueReduction <- function ( filename ) {\n",
    "\n",
    "    tissue_reduction <- read.table(filename, header=TRUE, sep=\"\\t\",\n",
    "                               skipNul=FALSE, stringsAsFactors = FALSE)\n",
    "    colnames(tissue_reduction)  <- c(\"SMTSD\",\"female\",\"male\",\"include\",\"display_name\")\n",
    "\n",
    "    return(tissue_reduction)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5391c71",
   "metadata": {
    "papermill": {
     "duration": 0.021639,
     "end_time": "2021-05-21T04:15:38.947420",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.925781",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 1.2.8 get GTEx phenotype data for the SRR accessions\n",
    "Transitive closure permits the association of the sequence reads, SRR Accessions, through the SraRunTable.txt (obtained from selecting annotation from the dbGaP login) with the SAMPID used with the GTEx.  This SAMPID is the means by which we can get this phenotype data and associate it with the counts data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dfe2b1d6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:38.996354Z",
     "iopub.status.busy": "2021-05-21T04:15:38.995329Z",
     "iopub.status.idle": "2021-05-21T04:15:39.006641Z",
     "shell.execute_reply": "2021-05-21T04:15:39.005755Z"
    },
    "lines_to_next_cell": 2,
    "papermill": {
     "duration": 0.037176,
     "end_time": "2021-05-21T04:15:39.006775",
     "exception": false,
     "start_time": "2021-05-21T04:15:38.969599",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "getGTExPhenoDataForSRR <- function (destDir) {\n",
    "\n",
    "    message(\"Loading srr_pdata\\n\")\n",
    "    srr_pdata <- readr::read_csv(\"../data/srr_pdata.csv\")\n",
    "    message(\"done!\\n\")\n",
    "\n",
    "    return(srr_pdata)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2a96c04",
   "metadata": {
    "papermill": {
     "duration": 0.021154,
     "end_time": "2021-05-21T04:15:39.049023",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.027869",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 1.2.9 makeCountsMatrix \n",
    "\n",
    "Given the counts filename, make a data matrix."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fe780ac3",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:39.096493Z",
     "iopub.status.busy": "2021-05-21T04:15:39.095225Z",
     "iopub.status.idle": "2021-05-21T04:15:39.107187Z",
     "shell.execute_reply": "2021-05-21T04:15:39.106327Z"
    },
    "papermill": {
     "duration": 0.037176,
     "end_time": "2021-05-21T04:15:39.107333",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.070157",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "makeCountsMatrix <- function (filename_gz) {\n",
    "    message(\"\\nloading \", paste(filename_gz), collapse=\" \")\n",
    "    counts <- data.table::fread(filename_gz)\n",
    "    message(\"done!\")\n",
    "    rownames(counts) <- counts$ID\n",
    "    counts <- counts[,-1]\n",
    "    counts <- data.matrix(counts)\n",
    "    return(counts)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ef9ecdac",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:39.160728Z",
     "iopub.status.busy": "2021-05-21T04:15:39.155233Z",
     "iopub.status.idle": "2021-05-21T04:15:39.171429Z",
     "shell.execute_reply": "2021-05-21T04:15:39.170203Z"
    },
    "papermill": {
     "duration": 0.042424,
     "end_time": "2021-05-21T04:15:39.171568",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.129144",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "makeSplicingExpressionSetObject <- function (srr_pdata, counts) {\n",
    "    message(\"making splicing expressionSet object\")\n",
    "    #\n",
    "    # match srr counts with srr_pdata - there were some srr without phenodata\n",
    "    #\n",
    "    pdata_match <- as.character(colnames(counts)) %in% as.character(srr_pdata$'SRR')\n",
    "    \n",
    "    counts <- counts[,pdata_match]\n",
    "\n",
    "    #\n",
    "    # reorder the srr_pdata to match the colnames of the counts\n",
    "    #\n",
    "    reorder_idx <- match(as.character(colnames(counts)), as.character(srr_pdata$'SRR'))\n",
    "    srr_pdata   <- srr_pdata[reorder_idx,]\n",
    "\n",
    "    #\n",
    "    # make the srr_pdata an AnnotatedDataFrame\n",
    "    #\n",
    "    metadata <- data.frame(labelDescriptions=as.character(colnames(srr_pdata)))\n",
    "    phenoData <- new(\"AnnotatedDataFrame\", data = srr_pdata, varMetadata=metadata)\n",
    "\n",
    "    #\n",
    "    # make the counts an expressionSet\n",
    "    # and provide the phenoData (the annotatedDataFrame construct from above)\n",
    "    #\n",
    "    es <- ExpressionSet(as.matrix(counts))\n",
    "    phenoData(es) <- phenoData\n",
    "    \n",
    "    message(\"made new expressionSet object\\n\",\n",
    "           paste(dim(es)), collapse = \" \")\n",
    "    message(\"dim pData(es)\\n\",\n",
    "           paste(dim(pData(es))), collapse = \" \")\n",
    "    message(\"dim exprs(es)\\n\",\n",
    "           paste(dim(exprs(es))), collapse = \" \")\n",
    "    message(\"done!\\n\")\n",
    "    \n",
    "    return(es)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b569d12e",
   "metadata": {
    "papermill": {
     "duration": 0.021181,
     "end_time": "2021-05-21T04:15:39.213773",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.192592",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## 1.3 Preprocessing \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a89bb88b",
   "metadata": {
    "papermill": {
     "duration": 0.021141,
     "end_time": "2021-05-21T04:15:39.256147",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.235006",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 1.3.1  Reduce Sample Set \n",
    "Read in all requirements so that the stage is properly set -- tissues.tsv contains the subset of files desired for analysis.\n",
    "It is found in the `assets` subdirectory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1b826c9c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:39.303971Z",
     "iopub.status.busy": "2021-05-21T04:15:39.302545Z",
     "iopub.status.idle": "2021-05-21T04:15:39.315685Z",
     "shell.execute_reply": "2021-05-21T04:15:39.314593Z"
    },
    "papermill": {
     "duration": 0.038615,
     "end_time": "2021-05-21T04:15:39.315817",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.277202",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "reduceSampleSet <- function (tissue_reduction, es) {\n",
    "\n",
    "   message(\"\\nsize tissue_reduction\\n\",\n",
    "        paste(dim(tissue_reduction), collapse=\" \"))\n",
    "   message(\"\\nsize es\\n\",\n",
    "        paste(dim(es)), collapse=\" \")\n",
    "   message(\"\\nsize pData(es)\\n\",\n",
    "        paste(dim(pData(es)), collapse=\" \"))\n",
    "   # only include those tissues we wish to continue with\n",
    "   message(\"\\n number of tissue types to keep\\n\",\n",
    "        paste(table(tissue_reduction$include)), collapse = \" \")\n",
    "   tissue_reduction <- tissue_reduction[tissue_reduction$include==1,]\n",
    "\n",
    "   # create a matching tissue name to go with the expressionSet phenotype esect\n",
    "   pData(es)$SMTSD        <- factor(snakecase::to_snake_case(as.character(pData(es)$SMTSD)))\n",
    "   tissue_reduction$SMTSD <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD)))\n",
    "\n",
    "   message(\"\\nlength tissues in phenotype data\\n\",\n",
    "        paste(length(levels(pData(es)$SMTSD)), collapse = \" \"))\n",
    "   message(\"\\nlength tissues in tissue_reduction data\\n\",\n",
    "        paste(length(tissue_reduction$SMTSD), collapse = \" \"))\n",
    "\n",
    "   keep <- pData(es)$SMTSD %in% tissue_reduction$SMTSD\n",
    "   message(\"\\nhow many to keep in phenotype data\\n\",\n",
    "        paste(table(keep), collapse = \" \"))\n",
    "\n",
    "   es        <- es       [          ,keep==TRUE]\n",
    "   pData(es)$SMTSD <- factor(pData(es)$SMTSD)\n",
    "   message(\"\\nsize reduced es\\n\",\n",
    "        paste(dim(es)), collapse=\" \")\n",
    "   message(\"\\nsize pData(es)\\n\",\n",
    "        paste(dim(pData(es)), collapse=\" \"))\n",
    "   message(\"\\nsize exprs(es)\\n\",\n",
    "        paste(dim(exprs(es)), collapse = \" \"))\n",
    "\n",
    "   # test to make sure we don't have nonsense\n",
    "#   keep = pData(es)$SMTSD== \"breast_mammary_tissue\"\n",
    "#   message(\"\\nTEST: how many to keep in to have only breast_mammary_tissue\\n\",\n",
    "#        paste(table(keep), collapse = \" \"))\n",
    "#   tes        = es       [          ,keep==TRUE]\n",
    "#   pData(tes) = pData(es)[keep==TRUE,          ]\n",
    "#   message(\"\\nTEST: size breast_mammary_tissue tes\\n\",\n",
    "#        paste(dim(tes), collapse=\" \"))\n",
    "#   message(\"\\nTEST: size phenotype object pData(tes)\\n\",\n",
    "#        paste(dim(pData(tes)), collapse=\" \"))\n",
    "   # end test\n",
    "   return (es)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae2e1b64",
   "metadata": {
    "papermill": {
     "duration": 0.021156,
     "end_time": "2021-05-21T04:15:39.358416",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.337260",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### 3.2 MAIN routine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "45c63656",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:39.405884Z",
     "iopub.status.busy": "2021-05-21T04:15:39.404775Z",
     "iopub.status.idle": "2021-05-21T04:15:39.516303Z",
     "shell.execute_reply": "2021-05-21T04:15:39.514710Z"
    },
    "papermill": {
     "duration": 0.136756,
     "end_time": "2021-05-21T04:15:39.516438",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.379682",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Decompressing fromGTF.tar.gz into ../data\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Gunzipping files into ../data\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in system(\"gunzip ../data/fromGTF.*txt.gz\", intern = TRUE):\n",
      "“running command 'gunzip ../data/fromGTF.*txt.gz' had status 1”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Done!\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getReleasedGTFAnnotations (destDir <- \"../data/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "88d2146a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:39.570633Z",
     "iopub.status.busy": "2021-05-21T04:15:39.569014Z",
     "iopub.status.idle": "2021-05-21T04:15:40.054093Z",
     "shell.execute_reply": "2021-05-21T04:15:40.052917Z"
    },
    "papermill": {
     "duration": 0.513705,
     "end_time": "2021-05-21T04:15:40.054298",
     "exception": false,
     "start_time": "2021-05-21T04:15:39.540593",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading srr_pdata\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[36m──\u001b[39m \u001b[1m\u001b[1mColumn specification\u001b[1m\u001b[22m \u001b[36m────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  SAMPID = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMATSSCR = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMCENTER = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMPTHNTS = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMTS = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMTSD = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMUBRID = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMNABTCH = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMNABTCHT = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMNABTCHD = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMGEBTCH = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMGEBTCHD = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMGEBTCHT = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMAFRZE = \u001b[31mcol_character()\u001b[39m,\n",
      "  SMGTC = \u001b[33mcol_logical()\u001b[39m,\n",
      "  SMNUMGPS = \u001b[33mcol_logical()\u001b[39m,\n",
      "  SM550NRM = \u001b[33mcol_logical()\u001b[39m,\n",
      "  SM350NRM = \u001b[33mcol_logical()\u001b[39m,\n",
      "  SMMNCPB = \u001b[33mcol_logical()\u001b[39m,\n",
      "  SMMNCV = \u001b[33mcol_logical()\u001b[39m\n",
      "  # ... with 6 more columns\n",
      ")\n",
      "\u001b[36mℹ\u001b[39m Use \u001b[30m\u001b[47m\u001b[30m\u001b[47m`spec()`\u001b[47m\u001b[30m\u001b[49m\u001b[39m for the full column specifications.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "done!\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get the metdata\n",
    "tissue_reduction <- getTissueReduction ( \"../assets/tissues.tsv\" )\n",
    "srr_pdata        <- getGTExPhenoDataForSRR (\"../data/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6c8ec10a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:40.108403Z",
     "iopub.status.busy": "2021-05-21T04:15:40.107143Z",
     "iopub.status.idle": "2021-05-21T04:15:40.135364Z",
     "shell.execute_reply": "2021-05-21T04:15:40.134375Z"
    },
    "papermill": {
     "duration": 0.057375,
     "end_time": "2021-05-21T04:15:40.135500",
     "exception": false,
     "start_time": "2021-05-21T04:15:40.078125",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "splice_list  <- c(\"se\", \"a3ss\", \"a5ss\", \"mxe\", \"ri\")\n",
    "tissue_reduction <- tissue_reduction[tissue_reduction$include==1,]\n",
    "tissue_list  <- factor(snakecase::to_snake_case(as.character(tissue_reduction$SMTSD))) \n",
    "# # Read in the list of genes of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f9f2c38",
   "metadata": {
    "tags": [
     "papermill-error-cell-tag"
    ]
   },
   "source": [
    "<span id=\"papermill-error-cell\" style=\"color:red; font-family:Helvetica Neue, Helvetica, Arial, sans-serif; font-size:2em;\">Execution using papermill encountered an exception here and stopped:</span>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bfbd6eec",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2021-05-21T04:15:40.188691Z",
     "iopub.status.busy": "2021-05-21T04:15:40.187407Z",
     "iopub.status.idle": "2021-05-21T04:15:40.207384Z",
     "shell.execute_reply": "2021-05-21T04:15:40.205914Z"
    },
    "papermill": {
     "duration": 0.048281,
     "end_time": "2021-05-21T04:15:40.207647",
     "exception": true,
     "start_time": "2021-05-21T04:15:40.159366",
     "status": "failed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in file(file, \"rt\"):\n",
      "“cannot open file '../data/genes_of_interest.tsv': No such file or directory”\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in file(file, \"rt\"): cannot open the connection\n",
     "output_type": "error",
     "traceback": [
      "Error in file(file, \"rt\"): cannot open the connection\nTraceback:\n",
      "1. read.table(\"../data/genes_of_interest.tsv\", header = TRUE, sep = \"\\t\", \n .     skipNul = FALSE, stringsAsFactors = FALSE)",
      "2. file(file, \"rt\")"
     ]
    }
   ],
   "source": [
    "genes_of_interest <- read.table(\"../data/genes_of_interest.tsv\", header=TRUE,\n",
    "                                sep=\"\\t\", skipNul=FALSE, stringsAsFactors = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d99eb4e1",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "genes_of_interest <- as.character(genes_of_interest$genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9521e52",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "gene_as           <- read.table(\"../data/genes_as_of_interest.tsv\", header=TRUE,\n",
    "                                sep=\"\\t\", skipNul=FALSE, stringsAsFactors = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74813059",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "write.table(\"SpliceType\", file=\"../data/incLevelDifferences.tsv\", append=FALSE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "write.table(\"Tissue\", file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "write.table(\"ID\", file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "write.table(\"GeneID\", file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "write.table(\"geneSymbol\", file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "write.table(\"IncLevelDifference\", file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3baab1d8",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "maser_directory = \"../data/MASER2020Sept16\"\n",
    "mkdir(maser_directory)\n",
    "\n",
    " for (tissue_index in (tissue_list)) {\n",
    "\n",
    "   tissue_dir = paste(maser_directory, tissue_index, sep=\"/\")\n",
    "   mkdir(tissue_dir)\n",
    "\n",
    "   for (splice_type in splice_list) {\n",
    "   \n",
    "    if (splice_type == \"se\") {\n",
    "\n",
    "     message(\"tissue of interest is \", tissue_index)\n",
    "\n",
    "     message(\"splice_type is \", splice_type)\n",
    "\n",
    "     # for each splice type the ReadsOnTarget output file we are generating is unique\n",
    "     file <- paste(paste(\"../data/fromGTF\",\n",
    "              snakecase::to_swap_case(as.character(splice_type)), sep=\".\"),\n",
    "      \t\"txt\", sep=\".\")\n",
    "     fromGTF          <- read.table(file=file, header=TRUE)\n",
    "\n",
    "     # Reduce the rows by focusing only on the splice_type, tissue type\n",
    "     splice_type_gene_as <- gene_as[gene_as$ASE == snakecase::to_swap_case(as.character(splice_type)),]\n",
    "     message(\"splice_type_gene_as dim is \" , dim(splice_type_gene_as))\n",
    "      \n",
    "     # Reduce rows by focusing on the tissue of interest\n",
    "     tissue              <- splice_type_gene_as$Tissue\n",
    "      \n",
    "     keep_tissue         <- tissue %in% tissue_index\n",
    "\n",
    "     if (sum(keep_tissue == TRUE) > 0) {\n",
    "\t\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_tissue == TRUE,]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aca56dde",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "      # Reduce the rows by focusing only on the genes of interest\n",
    "      gs                  <- splice_type_gene_as$GeneSymbol\n",
    "      if (length(gs) > length (genes_of_interest)) {\n",
    "           keep_gs             <- gs %in% genes_of_interest\n",
    "      } else {\n",
    "           keep_gs             <- genes_of_interest %in% gs\n",
    "      }\n",
    "      if (sum(keep_gs == TRUE) > 0) {\n",
    "        splice_type_gene_as <- splice_type_gene_as[keep_gs == TRUE,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b6cdabb",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "         # order this remaining matrix\n",
    "         idx                 <- order(splice_type_gene_as$ASE_IDX)\n",
    "         splice_type_gene_as <- splice_type_gene_as[idx,]\n",
    "  \n",
    "         # extract the junction to reduce the other matrices by\n",
    "  \t junction            <- splice_type_gene_as$ASE_IDX\n",
    "\t \n",
    "\t if (sum(is.na(junction)) == 0) {\n",
    "\t  \n",
    "          message(\"junctions are = \", junction)\n",
    "\t\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          fromGTF             <- fromGTF[junction,]\n",
    "\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.ijc.txt.gz\", sep=\".\")\n",
    "          ijc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          ijc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- ijc_counts)\n",
    "          rm (ijc_counts)\n",
    "          ijc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- ijc)\n",
    "          message(\"PASS ijc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    "\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          ijc                 <- ijc[junction,]\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.sjc.txt.gz\", sep=\".\")\n",
    "          sjc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          sjc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- sjc_counts)\n",
    "          rm(sjc_counts)\n",
    "          sjc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- sjc)\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          sjc                 <- sjc[junction,]\n",
    "          message(\"PASS sjc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    " \n",
    "          # reduce the ijc sjc matrices to just the samples that are of the tissue of interest (columns now)\n",
    "          tissue_idx          <- c(pData(ijc)$SMTSD == tissue_index)\n",
    "          ijc                 <- ijc[, tissue_idx == TRUE]\n",
    "          ijc_srrnames        <- colnames(exprs(ijc))\n",
    "          sjc                 <- sjc[, tissue_idx == TRUE]\n",
    "          sjc_srrnames        <- colnames(exprs(sjc))\n",
    "          message(\"PASS done reducing matrix to tissue_index (columns) is done\\n\")\n",
    "    \n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inc.txt.gz\", sep=\".\")\n",
    "          inc                 <- data.table::fread(file)\n",
    "          rownames(inc)       <- inc$ID\n",
    "  \t  # remove the first column -- the ID column\n",
    "          inc                 <- inc[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inclen.txt.gz\", sep=\".\")\n",
    "          inclen              <- data.table::fread(file)\n",
    "          rownames(inclen)    <- inclen$ID\n",
    "     \t  # remove the first column -- the ID column\n",
    "          inclen              <- inclen[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.skiplen.txt.gz\", sep=\".\")\n",
    "          skiplen             <- data.table::fread(file)\n",
    "          rownames(skiplen)   <- skiplen$ID\n",
    "\t  # remove the first column -- the ID column\n",
    "          skiplen             <- skiplen[,-1]\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\t\n",
    "          message(\"PASS done inc, inclen and skiplen loaded is done\\n\")\n",
    "\n",
    "          # reduce the rows to match our junctions\n",
    " \t  inc_srrnames        <- colnames(inc)\n",
    "\t  rm(keep)\n",
    " \t  keep                <- inc_srrnames %in% ijc_srrnames\n",
    "          inc                 <- matrix(as.numeric(inc    [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          inclen              <- matrix(as.numeric(inclen [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          skiplen             <- matrix(as.numeric(skiplen[,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          message (\"inc dim = \", dim(inc))\n",
    " \t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\n",
    "\t  if (dim(inc)[1] > 0) {\n",
    "           message(\"PASS done inc, inclen and skiplen reduced by columns to match ijc and insuring > 0\\n\")\n",
    "\n",
    "           # we need to further divide them into SAMPLE_1 and SAMPLE_2\n",
    "           # \n",
    "           # SAMPLE_1 data will hold all the male samples (GTEx SEX == 1)\n",
    "           # SAMPLE_2 data will hold all the female samples (GTEx SEX == 2) \n",
    "           #\n",
    "  \t   ijc_srrnames     <- as.character(colnames(exprs(ijc)))\n",
    "           male_srrnames    <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 1])))\n",
    "           keep             <- ijc_srrnames %in% male_srrnames\n",
    " \t   table (keep)\n",
    "           tmp_IJC_SAMPLE_1 <- exprs(ijc[, pData(ijc)$SEX == 1])\n",
    "           tmp_SJC_SAMPLE_1 <- exprs(sjc[, pData(ijc)$SEX == 1])\n",
    "           tmp_IncLevel1    <- matrix(as.numeric(inc[,keep==TRUE]),\n",
    "\t                    nrow = length(junction), ncol = length(male_srrnames))\n",
    "  \t   message (\"PASS tmp_IncLevel1 made\\n\")\n",
    "           message(\"PASS done separated into male SAMPLE_1\\n\")\n",
    "\n",
    "           female_srrnames  <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 2])))\n",
    "           tmp_IJC_SAMPLE_2 <- exprs(ijc[, pData(ijc)$SEX == 2])\n",
    "           tmp_SJC_SAMPLE_2 <- exprs(sjc[, pData(ijc)$SEX == 2])\n",
    "           tmp_IncLevel2    <- matrix(as.numeric(inc[, pData(ijc)$SEX == 2]),\n",
    "\t                    nrow = length(junction), ncol = length(female_srrnames))\n",
    "           message(\"PASS done separated into male SAMPLE_2\\n\")\n",
    "\n",
    "           inters = intersect(male_srrnames, female_srrnames)\n",
    "           if (length(inters) != 0) {\n",
    "             message(\"ERROR in female to male separation!\\n\")\n",
    "           } else {\n",
    "             message(\"PASS female and male samples separated\\n\")\n",
    "           }\n",
    "\n",
    "           # InclevelDifference is the difference of the rowMeans of IncLevel1 and IncLevel2\n",
    "           inclevel1_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel1), nrow = length(junction), ncol = length(male_srrnames)))\n",
    "           inclevel2_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel2), nrow = length(junction), ncol = length(female_srrnames)))\n",
    "           IncLevelDifference <- inclevel1_rm - inclevel2_rm\n",
    "           message(\"PASS done IncLevelDifference calculated \", IncLevelDifference)\n",
    "    \n",
    "           IJC_SAMPLE_1     <- tmp_IJC_SAMPLE_1[,1]\n",
    "           SJC_SAMPLE_1     <- tmp_SJC_SAMPLE_1[,1]\n",
    "           IncLevel1        <- tmp_IncLevel1[,1]\n",
    "    \n",
    "           for (i in 2:length(male_srrnames)) {\n",
    "            IJC_SAMPLE_1 <- paste(IJC_SAMPLE_1, tmp_IJC_SAMPLE_1[,i], sep=\",\")\n",
    "            SJC_SAMPLE_1 <- paste(SJC_SAMPLE_1, tmp_SJC_SAMPLE_1[,i], sep=\",\")\n",
    "            IncLevel1    <- paste(IncLevel1   , tmp_IncLevel1   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_1, SJC_SAMPLE_1 and IncLevel1 done!\\n\")\n",
    "    \n",
    "           IJC_SAMPLE_2     <- tmp_IJC_SAMPLE_2[,1]\n",
    "           SJC_SAMPLE_2     <- tmp_SJC_SAMPLE_2[,1]\n",
    "           IncLevel2        <- tmp_IncLevel2[,1]\n",
    "    \n",
    "           for (i in 2:length(female_srrnames)) {\n",
    "            IJC_SAMPLE_2 <- paste(IJC_SAMPLE_2, tmp_IJC_SAMPLE_2[,i], sep=\",\")\n",
    "            SJC_SAMPLE_2 <- paste(SJC_SAMPLE_2, tmp_SJC_SAMPLE_2[,i], sep=\",\")\n",
    "            IncLevel2    <- paste(IncLevel2   , tmp_IncLevel2   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_2, SJC_SAMPLE_2 and IncLevel2 done!\\n\")\n",
    " \n",
    "           # Ready to print out all the bits.\n",
    "           # We will paste the line\n",
    "           # first part comes from the fromGTF.splicetype.txt file\n",
    "\t   # Each splice_type is unique in their files\n",
    "\t   # Here we use fromGTF.SE.txt\n",
    "           file <- paste(paste(tissue_dir, snakecase::to_swap_case(as.character(splice_type)), sep=\"/\"),\n",
    "                             \"MATS.ReadsOnTargetAndJunctionCounts.txt\", sep= \".\")\n",
    "           write.table(\"ID\", file=file, append=FALSE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "           write.table(\"GeneID\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "           write.table(\"geneSymbol\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"chr\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"strand\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"exonStart_0base\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"exonEnd\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"upstreamES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"upstreamEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"downstreamES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"downstreamEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"ID\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SkipFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"PValue\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"FDR\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevelDifference\", file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           message(\"PASS header written to the file!\\n\")\n",
    "\n",
    "           # first part is all the information from the fromGTF\n",
    "           # The structure we have is of the dimensions of the fromGTF, reduced by row to the junctions of signficance\n",
    "           for (m in 1:(dim(fromGTF)[1])) {\n",
    "             for (n in 1:(dim(fromGTF)[2])) {\n",
    "               write.table(fromGTF[m,n], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t       \n",
    "    \t     }\n",
    "             write.table(splice_type, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(tissue_index, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,1], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,2], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,3], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS fromGTF file written to the file!\\n\")\n",
    "             #now the rest\t\t\n",
    "             write.table(IncLevelDifference[m], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(junction[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(inclen    [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(skiplen   [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$AdjPVal, file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$B      , file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevelDifference[m], file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS header written to the file!\\n\")\n",
    "             message(\"PASS row \", m)\n",
    "    \t     message(\"IJC, SJC, IncLevel for SAMPLE_1 and SAMPLE_2 written to the file!\\n\")\n",
    "\t    \n",
    "           } # for all junctions of interest\n",
    "\t   \n",
    " \t  } # and if there is > 0 junctions with inc\n",
    "#\t  rm(keep_gs)\n",
    "\t  \n",
    "         } # end !is.na(junction)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a7ced52",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#       } # and if there is > 0 genes of interest\n",
    "       rm(keep_tissue)\n",
    "       \n",
    "      } # if there are any tissues for splice type \"se\"\n",
    "      # end if splice_type is \"se\"\n",
    "    \n",
    "    } else if (splice_type == \"a3ss\") {\n",
    "\n",
    "      message(\"tissue of interest is \", tissue_index)\n",
    "\n",
    "      message(\"splice_type is \", splice_type)\n",
    "\n",
    "      # for each splice type the ReadsOnTarget output file we are generating is unique\n",
    "      file <- paste(paste(\"../data/fromGTF\",\n",
    "              snakecase::to_swap_case(as.character(splice_type)), sep=\".\"),\n",
    "      \t\"txt\", sep=\".\")\n",
    "      fromGTF          <- read.table(file=file, header=TRUE)\n",
    "\n",
    "      # Reduce the rows by focusing only on the splice_type, tissue type\n",
    "      splice_type_gene_as <- gene_as[gene_as$ASE == snakecase::to_swap_case(as.character(splice_type)),]\n",
    "      message(\"splice_type_gene_as dim is \" , dim(splice_type_gene_as))\n",
    "      \n",
    "      # Reduce rows by focusing on the tissue of interest\n",
    "      tissue              <- splice_type_gene_as$Tissue\n",
    "      \n",
    "      keep_tissue         <- tissue %in% tissue_index\n",
    "\n",
    "      if (sum(keep_tissue == TRUE) > 0) {\n",
    "\t\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_tissue == TRUE,]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46e70bb9",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "      # Reduce the rows by focusing only on the genes of interest\n",
    "      gs                  <- splice_type_gene_as$GeneSymbol\n",
    "      if (length(gs) > length (genes_of_interest)) {\n",
    "           keep_gs             <- gs %in% genes_of_interest\n",
    "      } else {\n",
    "           keep_gs             <- genes_of_interest %in% gs\n",
    "      }\n",
    "      if (sum(keep_gs == TRUE) > 0) {\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_gs == TRUE,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfa208f1",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "        # order this remaining matrix\n",
    "        idx                 <- order(splice_type_gene_as$ASE_IDX)\n",
    "        splice_type_gene_as <- splice_type_gene_as[idx,]\n",
    "  \n",
    "        # extract the junction to reduce the other matrices by\n",
    "  \tjunction            <- splice_type_gene_as$ASE_IDX\n",
    "\n",
    "\t if (sum(is.na(junction)) == 0) {\n",
    "\t \n",
    "          message(\"junctions are = \", junction)\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          fromGTF             <- fromGTF[junction,]\n",
    "\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.ijc.txt.gz\", sep=\".\")\n",
    "          ijc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          ijc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- ijc_counts)\n",
    "          rm (ijc_counts)\n",
    "          ijc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- ijc)\n",
    "          message(\"PASS ijc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    "\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          ijc                 <- ijc[junction,]\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.sjc.txt.gz\", sep=\".\")\n",
    "          sjc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          sjc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- sjc_counts)\n",
    "          rm(sjc_counts)\n",
    "          sjc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- sjc)\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          sjc                 <- sjc[junction,]\n",
    "          message(\"PASS sjc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    " \n",
    "          # reduce the ijc sjc matrices to just the samples that are of the tissue of interest (columns now)\n",
    "          tissue_idx          <- c(pData(ijc)$SMTSD == tissue_index)\n",
    "          ijc                 <- ijc[, tissue_idx == TRUE]\n",
    "          ijc_srrnames        <- colnames(exprs(ijc))\n",
    "          sjc                 <- sjc[, tissue_idx == TRUE]\n",
    "          sjc_srrnames        <- colnames(exprs(sjc))\n",
    "          message(\"PASS done reducing matrix to tissue_index (columns) is done\\n\")\n",
    "    \n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inc.txt.gz\", sep=\".\")\n",
    "          inc                 <- data.table::fread(file)\n",
    "          rownames(inc)       <- inc$ID\n",
    "  \t  # remove the first column -- the ID column\n",
    "          inc                 <- inc[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inclen.txt.gz\", sep=\".\")\n",
    "          inclen              <- data.table::fread(file)\n",
    "          rownames(inclen)    <- inclen$ID\n",
    "     \t  # remove the first column -- the ID column\n",
    "          inclen              <- inclen[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.skiplen.txt.gz\", sep=\".\")\n",
    "          skiplen             <- data.table::fread(file)\n",
    "          rownames(skiplen)   <- skiplen$ID\n",
    "\t  # remove the first column -- the ID column\n",
    "          skiplen             <- skiplen[,-1]\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\t\n",
    "          message(\"PASS done inc, inclen and skiplen loaded is done\\n\")\n",
    "\n",
    "          # reduce the rows to match our junctions\n",
    " \t  inc_srrnames        <- colnames(inc)\n",
    "\t  rm (keep)\n",
    " \t  keep                <- inc_srrnames %in% ijc_srrnames\n",
    "          inc                 <- matrix(as.numeric(inc    [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          inclen              <- matrix(as.numeric(inclen [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          skiplen             <- matrix(as.numeric(skiplen[,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          message (\"inc dim = \", dim(inc))\n",
    " \t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\n",
    "\t  if (dim(inc)[1] > 0) {\n",
    "           message(\"PASS done inc, inclen and skiplen reduced by columns to match ijc and insuring > 0\\n\")\n",
    "\n",
    "           # we need to further divide them into SAMPLE_1 and SAMPLE_2\n",
    "           # \n",
    "           # SAMPLE_1 data will hold all the male samples (GTEx SEX == 1)\n",
    "           # SAMPLE_2 data will hold all the female samples (GTEx SEX == 2) \n",
    "           #\n",
    "  \t   ijc_srrnames     <- as.character(colnames(exprs(ijc)))\n",
    "           male_srrnames    <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 1])))\n",
    "\t   rm(keep)\n",
    "           keep             <- ijc_srrnames %in% male_srrnames\n",
    " \t   table (keep)\n",
    "           tmp_IJC_SAMPLE_1 <- exprs(ijc[, pData(ijc)$SEX == 1])\n",
    "           tmp_SJC_SAMPLE_1 <- exprs(sjc[, pData(ijc)$SEX == 1])\n",
    "           tmp_IncLevel1    <- matrix(as.numeric(inc[,keep==TRUE]),\n",
    "\t                    nrow = length(junction), ncol = length(male_srrnames))\n",
    "  \t   message (\"PASS tmp_IncLevel1 made\\n\")\n",
    "           message(\"PASS done separated into male SAMPLE_1\\n\")\n",
    "\n",
    "           female_srrnames  <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 2])))\n",
    "           tmp_IJC_SAMPLE_2 <- exprs(ijc[, pData(ijc)$SEX == 2])\n",
    "           tmp_SJC_SAMPLE_2 <- exprs(sjc[, pData(ijc)$SEX == 2])\n",
    "           tmp_IncLevel2    <- matrix(as.numeric(inc[, pData(ijc)$SEX == 2]),\n",
    "\t                    nrow = length(junction), ncol = length(female_srrnames))\n",
    "           message(\"PASS done separated into male SAMPLE_2\\n\")\n",
    "\n",
    "           inters = intersect(male_srrnames, female_srrnames)\n",
    "           if (length(inters) != 0) {\n",
    "             message(\"ERROR in female to male separation!\\n\")\n",
    "           } else {\n",
    "             message(\"PASS female and male samples separated\\n\")\n",
    "           }\n",
    "\n",
    "           # InclevelDifference is the difference of the rowMeans of IncLevel1 and IncLevel2\n",
    "           inclevel1_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel1), nrow = length(junction), ncol = length(male_srrnames)))\n",
    "           inclevel2_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel2), nrow = length(junction), ncol = length(female_srrnames)))\n",
    "           IncLevelDifference <- inclevel1_rm - inclevel2_rm\n",
    "           message(\"PASS done IncLevelDifference calculated \", IncLevelDifference)\n",
    "    \n",
    "           IJC_SAMPLE_1     <- tmp_IJC_SAMPLE_1[,1]\n",
    "           SJC_SAMPLE_1     <- tmp_SJC_SAMPLE_1[,1]\n",
    "           IncLevel1        <- tmp_IncLevel1[,1]\n",
    "    \n",
    "           for (i in 2:length(male_srrnames)) {\n",
    "            IJC_SAMPLE_1 <- paste(IJC_SAMPLE_1, tmp_IJC_SAMPLE_1[,i], sep=\",\")\n",
    "            SJC_SAMPLE_1 <- paste(SJC_SAMPLE_1, tmp_SJC_SAMPLE_1[,i], sep=\",\")\n",
    "            IncLevel1    <- paste(IncLevel1   , tmp_IncLevel1   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_1, SJC_SAMPLE_1 and IncLevel1 done!\\n\")\n",
    "    \n",
    "           IJC_SAMPLE_2     <- tmp_IJC_SAMPLE_2[,1]\n",
    "           SJC_SAMPLE_2     <- tmp_SJC_SAMPLE_2[,1]\n",
    "           IncLevel2        <- tmp_IncLevel2[,1]\n",
    "    \n",
    "           for (i in 2:length(female_srrnames)) {\n",
    "            IJC_SAMPLE_2 <- paste(IJC_SAMPLE_2, tmp_IJC_SAMPLE_2[,i], sep=\",\")\n",
    "            SJC_SAMPLE_2 <- paste(SJC_SAMPLE_2, tmp_SJC_SAMPLE_2[,i], sep=\",\")\n",
    "            IncLevel2    <- paste(IncLevel2   , tmp_IncLevel2   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_2, SJC_SAMPLE_2 and IncLevel2 done!\\n\")\n",
    " \n",
    "           # Ready to print out all the bits.\n",
    "           # We will paste the line\n",
    "           # first part comes from the fromGTF.splicetype.txt file\n",
    "\t   # Each splice_type is unique in their files\n",
    "\t   # Here we use fromGTF.A3SS.txt\n",
    "           file <- paste(paste(tissue_dir, snakecase::to_swap_case(as.character(splice_type)), sep=\"/\"),\n",
    "                             \"MATS.ReadsOnTargetAndJunctionCounts.txt\", sep= \".\")\n",
    "           write.table(\"ID\", file=file, append=FALSE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t   \n",
    "           write.table(\"GeneID\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "           write.table(\"geneSymbol\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"chr\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"strand\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"longExonStart_0base\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"longExonEnd\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"shortES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"shortEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"flankingES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"flankingEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"ID.1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SkipFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"PValue\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"FDR\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevelDifference\", file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           message(\"PASS header written to the file!\\n\")\n",
    "\n",
    "           # first part is all the information from the fromGTF\n",
    "           # The structure we have is of the dimensions of the fromGTF, reduced by row to the junctions of signficance\n",
    "           for (m in 1:(dim(fromGTF)[1])) {\n",
    "             for (n in 1:(dim(fromGTF)[2])) {\n",
    "               write.table(fromGTF[m,n], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t       \n",
    "    \t     }\n",
    "             write.table(splice_type, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(tissue_index, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,1], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,2], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,3], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS fromGTF file written to the file!\\n\")\n",
    "             #now the rest\t\t\n",
    "             write.table(IncLevelDifference[m], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(junction[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(inclen    [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(skiplen   [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$AdjPVal, file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$B      , file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevelDifference[m], file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS header written to the file!\\n\")\n",
    "             message(\"PASS row \", m)\n",
    "    \t     message(\"IJC, SJC, IncLevel for SAMPLE_1 and SAMPLE_2 written to the file!\\n\")\n",
    "\t    \n",
    "\t    \n",
    "           } # for all junctions of interest\n",
    "\t   \n",
    " \t  } # and if there is > 0 junctions with inc\n",
    "#\t  rm(keep_gs)\n",
    "\t  \n",
    "         } # end !is.na(junction)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f40af2b",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#       } # and if there is > 0 genes of interest\n",
    "       rm(keep_tissue)\n",
    "       \n",
    "      } # if there are any tissues for splice type \"se\"\n",
    "    \n",
    "      # end if splice_type is \"a3ss\"\n",
    "    \n",
    "    } else if (splice_type == \"a5ss\") {\n",
    "\n",
    "      message(\"tissue of interest is \", tissue_index)\n",
    "      message(\"splice_type is \", splice_type)\n",
    "\n",
    "      # for each splice type the ReadsOnTarget output file we are generating is unique\n",
    "      file <- paste(paste(\"../data/fromGTF\",\n",
    "              snakecase::to_swap_case(as.character(splice_type)), sep=\".\"),\n",
    "      \t\"txt\", sep=\".\")\n",
    "      fromGTF          <- read.table(file=file, header=TRUE)\n",
    "\n",
    "      # Reduce the rows by focusing only on the splice_type, tissue type\n",
    "      splice_type_gene_as <- gene_as[gene_as$ASE == snakecase::to_swap_case(as.character(splice_type)),]\n",
    "      message(\"splice_type_gene_as dim is \" , dim(splice_type_gene_as))\n",
    "      \n",
    "      # Reduce rows by focusing on the tissue of interest\n",
    "      tissue              <- splice_type_gene_as$Tissue\n",
    "      \n",
    "      keep_tissue         <- tissue %in% tissue_index\n",
    "\n",
    "      if (sum(keep_tissue == TRUE) > 0) {\n",
    "\t\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_tissue == TRUE,]\t\n",
    "#       # Reduce the rows by focusing only on the genes of interest\n",
    "#       gs                  <- splice_type_gene_as$GeneSymbol\n",
    "#       if (length(gs) > length (genes_of_interest)) {\n",
    "#            keep_gs             <- gs %in% genes_of_interest\n",
    "#       } else {\n",
    "#            keep_gs             <- genes_of_interest %in% gs\n",
    "#       }\n",
    "#       if (sum(keep_gs == TRUE) > 0) {\n",
    "#        splice_type_gene_as <- splice_type_gene_as[keep_gs == TRUE,]\n",
    "\t\n",
    "        # order this remaining matrix\n",
    "        idx                 <- order(splice_type_gene_as$ASE_IDX)\n",
    "        splice_type_gene_as <- splice_type_gene_as[idx,]\n",
    "  \n",
    "        # extract the junction to reduce the other matrices by\n",
    "  \tjunction            <- splice_type_gene_as$ASE_IDX\n",
    "\n",
    "\t if (sum(is.na(junction)) == 0) {\n",
    "\t \n",
    "          message(\"junctions are = \", junction)\n",
    "\t\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          fromGTF             <- fromGTF[junction,]\n",
    "\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.ijc.txt.gz\", sep=\".\")\n",
    "          ijc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          ijc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- ijc_counts)\n",
    "          rm (ijc_counts)\n",
    "          ijc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- ijc)\n",
    "          message(\"PASS ijc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    "\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          ijc                 <- ijc[junction,]\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.sjc.txt.gz\", sep=\".\")\n",
    "          sjc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          sjc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- sjc_counts)\n",
    "          rm(sjc_counts)\n",
    "          sjc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- sjc)\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          sjc                 <- sjc[junction,]\n",
    "          message(\"PASS sjc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    " \n",
    "          # reduce the ijc sjc matrices to just the samples that are of the tissue of interest (columns now)\n",
    "          tissue_idx          <- c(pData(ijc)$SMTSD == tissue_index)\n",
    "          ijc                 <- ijc[, tissue_idx == TRUE]\n",
    "          ijc_srrnames        <- colnames(exprs(ijc))\n",
    "          sjc                 <- sjc[, tissue_idx == TRUE]\n",
    "          sjc_srrnames        <- colnames(exprs(sjc))\n",
    "          message(\"PASS done reducing matrix to tissue_index (columns) is done\\n\")\n",
    "    \n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inc.txt.gz\", sep=\".\")\n",
    "          inc                 <- data.table::fread(file)\n",
    "          rownames(inc)       <- inc$ID\n",
    "  \t  # remove the first column -- the ID column\n",
    "          inc                 <- inc[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inclen.txt.gz\", sep=\".\")\n",
    "          inclen              <- data.table::fread(file)\n",
    "          rownames(inclen)    <- inclen$ID\n",
    "     \t  # remove the first column -- the ID column\n",
    "          inclen              <- inclen[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.skiplen.txt.gz\", sep=\".\")\n",
    "          skiplen             <- data.table::fread(file)\n",
    "          rownames(skiplen)   <- skiplen$ID\n",
    "\t  # remove the first column -- the ID column\n",
    "          skiplen             <- skiplen[,-1]\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\t\n",
    "          message(\"PASS done inc, inclen and skiplen loaded is done\\n\")\n",
    "\n",
    "          # reduce the rows to match our junctions\n",
    " \t  inc_srrnames        <- colnames(inc)\n",
    " \t  keep                <- inc_srrnames %in% ijc_srrnames\n",
    "          inc                 <- matrix(as.numeric(inc    [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          inclen              <- matrix(as.numeric(inclen [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          skiplen             <- matrix(as.numeric(skiplen[,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          message (\"inc dim = \", dim(inc))\n",
    " \t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\n",
    "\t  if (dim(inc)[1] > 0) {\n",
    "           message(\"PASS done inc, inclen and skiplen reduced by columns to match ijc and insuring > 0\\n\")\n",
    "\n",
    "           # we need to further divide them into SAMPLE_1 and SAMPLE_2\n",
    "           # \n",
    "           # SAMPLE_1 data will hold all the male samples (GTEx SEX == 1)\n",
    "           # SAMPLE_2 data will hold all the female samples (GTEx SEX == 2) \n",
    "           #\n",
    "  \t   ijc_srrnames     <- as.character(colnames(exprs(ijc)))\n",
    "           male_srrnames    <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 1])))\n",
    "           keep             <- ijc_srrnames %in% male_srrnames\n",
    " \t   table (keep)\n",
    "           tmp_IJC_SAMPLE_1 <- exprs(ijc[, pData(ijc)$SEX == 1])\n",
    "           tmp_SJC_SAMPLE_1 <- exprs(sjc[, pData(ijc)$SEX == 1])\n",
    "           tmp_IncLevel1    <- matrix(as.numeric(inc[,keep==TRUE]),\n",
    "\t                    nrow = length(junction), ncol = length(male_srrnames))\n",
    "  \t   message (\"PASS tmp_IncLevel1 made\\n\")\n",
    "           message(\"PASS done separated into male SAMPLE_1\\n\")\n",
    "\n",
    "           female_srrnames  <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 2])))\n",
    "           tmp_IJC_SAMPLE_2 <- exprs(ijc[, pData(ijc)$SEX == 2])\n",
    "           tmp_SJC_SAMPLE_2 <- exprs(sjc[, pData(ijc)$SEX == 2])\n",
    "           tmp_IncLevel2    <- matrix(as.numeric(inc[, pData(ijc)$SEX == 2]),\n",
    "\t                    nrow = length(junction), ncol = length(female_srrnames))\n",
    "           message(\"PASS done separated into male SAMPLE_2\\n\")\n",
    "\n",
    "           inters = intersect(male_srrnames, female_srrnames)\n",
    "           if (length(inters) != 0) {\n",
    "             message(\"ERROR in female to male separation!\\n\")\n",
    "           } else {\n",
    "             message(\"PASS female and male samples separated\\n\")\n",
    "           }\n",
    "\n",
    "           # InclevelDifference is the difference of the rowMeans of IncLevel1 and IncLevel2\n",
    "           inclevel1_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel1), nrow = length(junction), ncol = length(male_srrnames)))\n",
    "           inclevel2_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel2), nrow = length(junction), ncol = length(female_srrnames)))\n",
    "           IncLevelDifference <- inclevel1_rm - inclevel2_rm\n",
    "           message(\"PASS done IncLevelDifference calculated \", IncLevelDifference)\n",
    "    \n",
    "           IJC_SAMPLE_1     <- tmp_IJC_SAMPLE_1[,1]\n",
    "           SJC_SAMPLE_1     <- tmp_SJC_SAMPLE_1[,1]\n",
    "           IncLevel1        <- tmp_IncLevel1[,1]\n",
    "    \n",
    "           for (i in 2:length(male_srrnames)) {\n",
    "            IJC_SAMPLE_1 <- paste(IJC_SAMPLE_1, tmp_IJC_SAMPLE_1[,i], sep=\",\")\n",
    "            SJC_SAMPLE_1 <- paste(SJC_SAMPLE_1, tmp_SJC_SAMPLE_1[,i], sep=\",\")\n",
    "            IncLevel1    <- paste(IncLevel1   , tmp_IncLevel1   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_1, SJC_SAMPLE_1 and IncLevel1 done!\\n\")\n",
    "    \n",
    "           IJC_SAMPLE_2     <- tmp_IJC_SAMPLE_2[,1]\n",
    "           SJC_SAMPLE_2     <- tmp_SJC_SAMPLE_2[,1]\n",
    "           IncLevel2        <- tmp_IncLevel2[,1]\n",
    "    \n",
    "           for (i in 2:length(female_srrnames)) {\n",
    "            IJC_SAMPLE_2 <- paste(IJC_SAMPLE_2, tmp_IJC_SAMPLE_2[,i], sep=\",\")\n",
    "            SJC_SAMPLE_2 <- paste(SJC_SAMPLE_2, tmp_SJC_SAMPLE_2[,i], sep=\",\")\n",
    "            IncLevel2    <- paste(IncLevel2   , tmp_IncLevel2   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_2, SJC_SAMPLE_2 and IncLevel2 done!\\n\")\n",
    " \n",
    "           # Ready to print out all the bits.\n",
    "           # We will paste the line\n",
    "           # first part comes from the fromGTF.splicetype.txt file\n",
    "\t   # Each splice_type is unique in their files\n",
    "\t   # Here we use fromGTF.A5SS.txt\n",
    "           file <- paste(paste(tissue_dir, snakecase::to_swap_case(as.character(splice_type)), sep=\"/\"),\n",
    "                             \"MATS.ReadsOnTargetAndJunctionCounts.txt\", sep= \".\")\n",
    "           write.table(\"ID\", file=file, append=FALSE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t   \n",
    "           write.table(\"GeneID\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "           write.table(\"geneSymbol\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"chr\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"strand\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"longExonStart_0base\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"longExonEnd\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"shortES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"shortEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"flankingES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"flankingEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"ID.1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SkipFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"PValue\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"FDR\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevelDifference\", file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           message(\"PASS header written to the file!\\n\")\n",
    "\n",
    "           # first part is all the information from the fromGTF\n",
    "           # The structure we have is of the dimensions of the fromGTF, reduced by row to the junctions of signficance\n",
    "           for (m in 1:(dim(fromGTF)[1])) {\n",
    "             for (n in 1:(dim(fromGTF)[2])) {\n",
    "               write.table(fromGTF[m,n], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t       \n",
    "    \t     }\n",
    "             write.table(splice_type, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(tissue_index, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,1], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,2], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,3], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS fromGTF file written to the file!\\n\")\n",
    "             #now the rest\t\t\n",
    "             write.table(IncLevelDifference[m], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(junction[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(inclen    [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(skiplen   [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$AdjPVal, file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$B      , file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevelDifference[m], file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS header written to the file!\\n\")\n",
    "             message(\"PASS row \", m)\n",
    "    \t     message(\"IJC, SJC, IncLevel for SAMPLE_1 and SAMPLE_2 written to the file!\\n\")\n",
    "\t    \t    \n",
    "           } # for all junctions of interest\n",
    "\t   \n",
    " \t  } # and if there is > 0 junctions with inc\n",
    "#\t  rm(keep_gs)\n",
    "\t  \n",
    "         } # end !is.na(junction)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5cbef4e6",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#       } # and if there is > 0 genes of interest\n",
    "       rm(keep_tissue)\n",
    "       \n",
    "      } # if there are any tissues for splice type \"se\"\n",
    "    \n",
    "      # end if splice_type is \"a5ss\"\n",
    "      \n",
    "    } else if (splice_type == \"mxe\") {\n",
    "\n",
    "      message(\"tissue of interest is \", tissue_index)\n",
    "      message(\"splice_type is \", splice_type)\n",
    "\n",
    "      # for each splice type the ReadsOnTarget output file we are generating is unique\n",
    "      file <- paste(paste(\"../data/fromGTF\",\n",
    "              snakecase::to_swap_case(as.character(splice_type)), sep=\".\"),\n",
    "      \t\"txt\", sep=\".\")\n",
    "      fromGTF          <- read.table(file=file, header=TRUE)\n",
    "\n",
    "      # Reduce the rows by focusing only on the splice_type, tissue type\n",
    "      splice_type_gene_as <- gene_as[gene_as$ASE == snakecase::to_swap_case(as.character(splice_type)),]\n",
    "      message(\"splice_type_gene_as dim is \" , dim(splice_type_gene_as))\n",
    "      \n",
    "      # Reduce rows by focusing on the tissue of interest\n",
    "      tissue              <- splice_type_gene_as$Tissue\n",
    "      \n",
    "      keep_tissue         <- tissue %in% tissue_index\n",
    "\n",
    "      if (sum(keep_tissue == TRUE) > 0) {\n",
    "\t\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_tissue == TRUE,]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6858a7c2",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "      # Reduce the rows by focusing only on the genes of interest\n",
    "      gs                  <- splice_type_gene_as$GeneSymbol\n",
    "      if (length(gs) > length (genes_of_interest)) {\n",
    "           keep_gs             <- gs %in% genes_of_interest\n",
    "      } else {\n",
    "           keep_gs             <- genes_of_interest %in% gs\n",
    "      }\n",
    "      if (sum(keep_gs == TRUE) > 0) {\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_gs == TRUE,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6657bb2",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "        # order this remaining matrix\n",
    "        idx                 <- order(splice_type_gene_as$ASE_IDX)\n",
    "        splice_type_gene_as <- splice_type_gene_as[idx,]\n",
    "  \n",
    "        # extract the junction to reduce the other matrices by\n",
    "  \tjunction            <- splice_type_gene_as$ASE_IDX\n",
    "\t  \n",
    "\t if (sum(is.na(junction)) == 0) {\n",
    "\n",
    "          message(\"junctions are = \", junction)\n",
    "\t\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          fromGTF             <- fromGTF[junction,]\n",
    "\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.ijc.txt.gz\", sep=\".\")\n",
    "          ijc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          ijc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- ijc_counts)\n",
    "          rm (ijc_counts)\n",
    "          ijc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- ijc)\n",
    "          message(\"PASS ijc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    "\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          ijc                 <- ijc[junction,]\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.sjc.txt.gz\", sep=\".\")\n",
    "          sjc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          sjc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- sjc_counts)\n",
    "          rm(sjc_counts)\n",
    "          sjc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- sjc)\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          sjc                 <- sjc[junction,]\n",
    "          message(\"PASS sjc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    " \n",
    "          # reduce the ijc sjc matrices to just the samples that are of the tissue of interest (columns now)\n",
    "          tissue_idx          <- c(pData(ijc)$SMTSD == tissue_index)\n",
    "          ijc                 <- ijc[, tissue_idx == TRUE]\n",
    "          ijc_srrnames        <- colnames(exprs(ijc))\n",
    "          sjc                 <- sjc[, tissue_idx == TRUE]\n",
    "          sjc_srrnames        <- colnames(exprs(sjc))\n",
    "          message(\"PASS done reducing matrix to tissue_index (columns) is done\\n\")\n",
    "    \n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inc.txt.gz\", sep=\".\")\n",
    "          inc                 <- data.table::fread(file)\n",
    "          rownames(inc)       <- inc$ID\n",
    "  \t  # remove the first column -- the ID column\n",
    "          inc                 <- inc[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inclen.txt.gz\", sep=\".\")\n",
    "          inclen              <- data.table::fread(file)\n",
    "          rownames(inclen)    <- inclen$ID\n",
    "     \t  # remove the first column -- the ID column\n",
    "          inclen              <- inclen[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.skiplen.txt.gz\", sep=\".\")\n",
    "          skiplen             <- data.table::fread(file)\n",
    "          rownames(skiplen)   <- skiplen$ID\n",
    "\t  # remove the first column -- the ID column\n",
    "          skiplen             <- skiplen[,-1]\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\t\n",
    "          message(\"PASS done inc, inclen and skiplen loaded is done\\n\")\n",
    "\n",
    "          # reduce the rows to match our junctions\n",
    " \t  inc_srrnames        <- colnames(inc)\n",
    " \t  keep                <- inc_srrnames %in% ijc_srrnames\n",
    "          inc                 <- matrix(as.numeric(inc    [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          inclen              <- matrix(as.numeric(inclen [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          skiplen             <- matrix(as.numeric(skiplen[,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          message (\"inc dim = \", dim(inc))\n",
    " \t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\n",
    "\t  if (dim(inc)[1] > 0) {\n",
    "           message(\"PASS done inc, inclen and skiplen reduced by columns to match ijc and insuring > 0\\n\")\n",
    "\n",
    "           # we need to further divide them into SAMPLE_1 and SAMPLE_2\n",
    "           # \n",
    "           # SAMPLE_1 data will hold all the male samples (GTEx SEX == 1)\n",
    "           # SAMPLE_2 data will hold all the female samples (GTEx SEX == 2) \n",
    "           #\n",
    "  \t   ijc_srrnames     <- as.character(colnames(exprs(ijc)))\n",
    "           male_srrnames    <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 1])))\n",
    "           keep             <- ijc_srrnames %in% male_srrnames\n",
    " \t   table (keep)\n",
    "           tmp_IJC_SAMPLE_1 <- exprs(ijc[, pData(ijc)$SEX == 1])\n",
    "           tmp_SJC_SAMPLE_1 <- exprs(sjc[, pData(ijc)$SEX == 1])\n",
    "           tmp_IncLevel1    <- matrix(as.numeric(inc[,keep==TRUE]),\n",
    "\t                    nrow = length(junction), ncol = length(male_srrnames))\n",
    "  \t   message (\"PASS tmp_IncLevel1 made\\n\")\n",
    "           message(\"PASS done separated into male SAMPLE_1\\n\")\n",
    "\n",
    "           female_srrnames  <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 2])))\n",
    "           tmp_IJC_SAMPLE_2 <- exprs(ijc[, pData(ijc)$SEX == 2])\n",
    "           tmp_SJC_SAMPLE_2 <- exprs(sjc[, pData(ijc)$SEX == 2])\n",
    "           tmp_IncLevel2    <- matrix(as.numeric(inc[, pData(ijc)$SEX == 2]),\n",
    "\t                    nrow = length(junction), ncol = length(female_srrnames))\n",
    "           message(\"PASS done separated into male SAMPLE_2\\n\")\n",
    "\n",
    "           inters = intersect(male_srrnames, female_srrnames)\n",
    "           if (length(inters) != 0) {\n",
    "             message(\"ERROR in female to male separation!\\n\")\n",
    "           } else {\n",
    "             message(\"PASS female and male samples separated\\n\")\n",
    "           }\n",
    "\n",
    "           # InclevelDifference is the difference of the rowMeans of IncLevel1 and IncLevel2\n",
    "           inclevel1_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel1), nrow = length(junction), ncol = length(male_srrnames)))\n",
    "           inclevel2_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel2), nrow = length(junction), ncol = length(female_srrnames)))\n",
    "           IncLevelDifference <- inclevel1_rm - inclevel2_rm\n",
    "           message(\"PASS done IncLevelDifference calculated \", IncLevelDifference)\n",
    "    \n",
    "           IJC_SAMPLE_1     <- tmp_IJC_SAMPLE_1[,1]\n",
    "           SJC_SAMPLE_1     <- tmp_SJC_SAMPLE_1[,1]\n",
    "           IncLevel1        <- tmp_IncLevel1[,1]\n",
    "    \n",
    "           for (i in 2:length(male_srrnames)) {\n",
    "            IJC_SAMPLE_1 <- paste(IJC_SAMPLE_1, tmp_IJC_SAMPLE_1[,i], sep=\",\")\n",
    "            SJC_SAMPLE_1 <- paste(SJC_SAMPLE_1, tmp_SJC_SAMPLE_1[,i], sep=\",\")\n",
    "            IncLevel1    <- paste(IncLevel1   , tmp_IncLevel1   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_1, SJC_SAMPLE_1 and IncLevel1 done!\\n\")\n",
    "    \n",
    "           IJC_SAMPLE_2     <- tmp_IJC_SAMPLE_2[,1]\n",
    "           SJC_SAMPLE_2     <- tmp_SJC_SAMPLE_2[,1]\n",
    "           IncLevel2        <- tmp_IncLevel2[,1]\n",
    "    \n",
    "           for (i in 2:length(female_srrnames)) {\n",
    "            IJC_SAMPLE_2 <- paste(IJC_SAMPLE_2, tmp_IJC_SAMPLE_2[,i], sep=\",\")\n",
    "            SJC_SAMPLE_2 <- paste(SJC_SAMPLE_2, tmp_SJC_SAMPLE_2[,i], sep=\",\")\n",
    "            IncLevel2    <- paste(IncLevel2   , tmp_IncLevel2   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_2, SJC_SAMPLE_2 and IncLevel2 done!\\n\")\n",
    " \n",
    "           # Ready to print out all the bits.\n",
    "           # We will paste the line\n",
    "           # first part comes from the fromGTF.splicetype.txt file\n",
    "\t   # Each splice_type is unique in their files\n",
    "\t   # Here we use fromGTF.MXE.txt\n",
    "           file <- paste(paste(tissue_dir, snakecase::to_swap_case(as.character(splice_type)), sep=\"/\"),\n",
    "                             \"MATS.ReadsOnTargetAndJunctionCounts.txt\", sep= \".\")\n",
    "           write.table(\"ID\", file=file, append=FALSE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t   \n",
    "           write.table(\"GeneID\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "           write.table(\"geneSymbol\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"chr\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"strand\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "\t   write.table(\"X1stExonStart_0base\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"X1stExonEnd\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"X2ndExonStart_0base\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"X2ndExonEnd\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"downstreamES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"downstreamEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"ID.1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SkipFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"PValue\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"FDR\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevelDifference\", file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           message(\"PASS header written to the file!\\n\")\n",
    "\n",
    "           # first part is all the information from the fromGTF\n",
    "           # The structure we have is of the dimensions of the fromGTF, reduced by row to the junctions of signficance\n",
    "           for (m in 1:(dim(fromGTF)[1])) {\n",
    "             for (n in 1:(dim(fromGTF)[2])) {\n",
    "               write.table(fromGTF[m,n], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t       \n",
    "    \t     }\n",
    "             write.table(splice_type, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(tissue_index, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,1], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,2], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,3], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS fromGTF file written to the file!\\n\")\n",
    "             #now the rest\t\t\n",
    "             write.table(IncLevelDifference[m], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(junction[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(inclen    [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(skiplen   [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$AdjPVal, file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$B      , file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevelDifference[m], file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS header written to the file!\\n\")\n",
    "             message(\"PASS row \", m)\n",
    "    \t     message(\"IJC, SJC, IncLevel for SAMPLE_1 and SAMPLE_2 written to the file!\\n\")\n",
    "\t    \n",
    "           } # for all junctions of interest\n",
    "\t   \n",
    " \t  } # and if there is > 0 junctions with inc\n",
    "#\t  rm(keep_gs)\n",
    "\t  \n",
    "         } # end !is.na(junction)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a536c474",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#       } # and if there is > 0 genes of interest\n",
    "       rm(keep_tissue)\n",
    "       \n",
    "      } # if there are any tissues for splice type \"se\"\n",
    "    \n",
    "      # end if splice_type is \"mxe\"\n",
    "\n",
    "    } else if (splice_type == \"ri\") {\n",
    "\n",
    "      message(\"tissue of interest is \", tissue_index)\n",
    "      message(\"splice_type is \", splice_type)\n",
    "\n",
    "      # for each splice type the ReadsOnTarget output file we are generating is unique\n",
    "      file <- paste(paste(\"../data/fromGTF\",\n",
    "              snakecase::to_swap_case(as.character(splice_type)), sep=\".\"),\n",
    "      \t\"txt\", sep=\".\")\n",
    "      fromGTF          <- read.table(file=file, header=TRUE)\n",
    "\n",
    "      # Reduce the rows by focusing only on the splice_type, tissue type\n",
    "      splice_type_gene_as <- gene_as[gene_as$ASE == snakecase::to_swap_case(as.character(splice_type)),]\n",
    "      message(\"splice_type_gene_as dim is \" , dim(splice_type_gene_as))\n",
    "      \n",
    "      # Reduce rows by focusing on the tissue of interest\n",
    "      tissue              <- splice_type_gene_as$Tissue\n",
    "      \n",
    "      keep_tissue         <- tissue %in% tissue_index\n",
    "\n",
    "      if (sum(keep_tissue == TRUE) > 0) {\n",
    "\t\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_tissue == TRUE,]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd5d71a8",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "      # Reduce the rows by focusing only on the genes of interest\n",
    "      gs                  <- splice_type_gene_as$GeneSymbol\n",
    "      if (length(gs) > length (genes_of_interest)) {\n",
    "           keep_gs             <- gs %in% genes_of_interest\n",
    "      } else {\n",
    "           keep_gs             <- genes_of_interest %in% gs\n",
    "      }      \n",
    "      if (sum(keep_gs == TRUE) > 0) {\n",
    "       splice_type_gene_as <- splice_type_gene_as[keep_gs == TRUE,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c31bc982",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "        # order this remaining matrix\n",
    "        idx                 <- order(splice_type_gene_as$ASE_IDX)\n",
    "        splice_type_gene_as <- splice_type_gene_as[idx,]\n",
    "  \n",
    "        # extract the junction to reduce the other matrices by\n",
    "  \tjunction            <- splice_type_gene_as$ASE_IDX\n",
    "\t  \n",
    "\t if (sum(is.na(junction)) == 0) {\n",
    "\t\n",
    "          message(\"junctions are = \", junction)\n",
    "\t\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          fromGTF             <- fromGTF[junction,]\n",
    "\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.ijc.txt.gz\", sep=\".\")\n",
    "          ijc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          ijc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- ijc_counts)\n",
    "          rm (ijc_counts)\n",
    "          ijc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- ijc)\n",
    "          message(\"PASS ijc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    "\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          ijc                 <- ijc[junction,]\n",
    "          file                <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.sjc.txt.gz\", sep=\".\")\n",
    "          sjc_counts          <- makeCountsMatrix( filename_gz  <- file)\n",
    "          sjc                 <- makeSplicingExpressionSetObject (srr_pdata <- srr_pdata, counts <- sjc_counts)\n",
    "          rm(sjc_counts)\n",
    "          sjc                 <- reduceSampleSet(tissue_reduction <- tissue_reduction, es <- sjc)\n",
    "          # reduce rows by keeping only the junctions we care about\n",
    "          sjc                 <- sjc[junction,]\n",
    "          message(\"PASS sjc reduced matrix to junctions of interest (rows) is done\\n\")\n",
    " \n",
    "          # reduce the ijc sjc matrices to just the samples that are of the tissue of interest (columns now)\n",
    "          tissue_idx          <- c(pData(ijc)$SMTSD == tissue_index)\n",
    "          ijc                 <- ijc[, tissue_idx == TRUE]\n",
    "          ijc_srrnames        <- colnames(exprs(ijc))\n",
    "          sjc                 <- sjc[, tissue_idx == TRUE]\n",
    "          sjc_srrnames        <- colnames(exprs(sjc))\n",
    "          message(\"PASS done reducing matrix to tissue_index (columns) is done\\n\")\n",
    "    \n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inc.txt.gz\", sep=\".\")\n",
    "          inc                 <- data.table::fread(file)\n",
    "          rownames(inc)       <- inc$ID\n",
    "  \t  # remove the first column -- the ID column\n",
    "          inc                 <- inc[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.inclen.txt.gz\", sep=\".\")\n",
    "          inclen              <- data.table::fread(file)\n",
    "          rownames(inclen)    <- inclen$ID\n",
    "     \t  # remove the first column -- the ID column\n",
    "          inclen              <- inclen[,-1]\n",
    "          file <- paste(paste(\"../data/rmats_final\",splice_type, sep=\".\"), \"jc.skiplen.txt.gz\", sep=\".\")\n",
    "          skiplen             <- data.table::fread(file)\n",
    "          rownames(skiplen)   <- skiplen$ID\n",
    "\t  # remove the first column -- the ID column\n",
    "          skiplen             <- skiplen[,-1]\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\t\n",
    "          message(\"PASS done inc, inclen and skiplen loaded is done\\n\")\n",
    "\n",
    "          # reduce the rows to match our junctions\n",
    " \t  inc_srrnames        <- colnames(inc)\n",
    " \t  keep                <- inc_srrnames %in% ijc_srrnames\n",
    "          inc                 <- matrix(as.numeric(inc    [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          inclen              <- matrix(as.numeric(inclen [,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          skiplen             <- matrix(as.numeric(skiplen[,keep == TRUE]),\n",
    "\t                          nrow=length(junction), ncol = length(inc_srrnames))\n",
    "          message (\"inc dim = \", dim(inc))\n",
    " \t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inc dim [1] = \", dim(inc)[1])\n",
    "\t  message (\"inc dim [2] = \", dim(inc)[2])\t\n",
    "\t  message (\"inclen dim [1] = \", dim(inclen)[1])\n",
    "\t  message (\"inclen dim [2] = \", dim(inclen)[2])\t\n",
    "\t  message (\"skiplen dim [1] = \", dim(skiplen)[1])\n",
    "\t  message (\"skiplen dim [2] = \", dim(skiplen)[2])\n",
    "\t  if (dim(inc)[1] > 0) {\n",
    "           message(\"PASS done inc, inclen and skiplen reduced by columns to match ijc and insuring > 0\\n\")\n",
    "\n",
    "           # we need to further divide them into SAMPLE_1 and SAMPLE_2\n",
    "           # \n",
    "           # SAMPLE_1 data will hold all the male samples (GTEx SEX == 1)\n",
    "           # SAMPLE_2 data will hold all the female samples (GTEx SEX == 2) \n",
    "           #\n",
    "  \t   ijc_srrnames     <- as.character(colnames(exprs(ijc)))\n",
    "           male_srrnames    <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 1])))\n",
    "           keep             <- ijc_srrnames %in% male_srrnames\n",
    " \t   table (keep)\n",
    "           tmp_IJC_SAMPLE_1 <- exprs(ijc[, pData(ijc)$SEX == 1])\n",
    "           tmp_SJC_SAMPLE_1 <- exprs(sjc[, pData(ijc)$SEX == 1])\n",
    "           tmp_IncLevel1    <- matrix(as.numeric(inc[,keep==TRUE]),\n",
    "\t                    nrow = length(junction), ncol = length(male_srrnames))\n",
    "  \t   message (\"PASS tmp_IncLevel1 made\\n\")\n",
    "           message(\"PASS done separated into male SAMPLE_1\\n\")\n",
    "\n",
    "           female_srrnames  <- as.character(colnames(exprs(ijc[,pData(ijc)$SEX == 2])))\n",
    "           tmp_IJC_SAMPLE_2 <- exprs(ijc[, pData(ijc)$SEX == 2])\n",
    "           tmp_SJC_SAMPLE_2 <- exprs(sjc[, pData(ijc)$SEX == 2])\n",
    "           tmp_IncLevel2    <- matrix(as.numeric(inc[, pData(ijc)$SEX == 2]),\n",
    "\t                    nrow = length(junction), ncol = length(female_srrnames))\n",
    "           message(\"PASS done separated into male SAMPLE_2\\n\")\n",
    "\n",
    "           inters = intersect(male_srrnames, female_srrnames)\n",
    "           if (length(inters) != 0) {\n",
    "             message(\"ERROR in female to male separation!\\n\")\n",
    "           } else {\n",
    "             message(\"PASS female and male samples separated\\n\")\n",
    "           }\n",
    "\n",
    "           # InclevelDifference is the difference of the rowMeans of IncLevel1 and IncLevel2\n",
    "           inclevel1_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel1), nrow = length(junction), ncol = length(male_srrnames)))\n",
    "           inclevel2_rm       <- rowMeans(matrix(as.numeric(tmp_IncLevel2), nrow = length(junction), ncol = length(female_srrnames)))\n",
    "           IncLevelDifference <- inclevel1_rm - inclevel2_rm\n",
    "           message(\"PASS done IncLevelDifference calculated \", IncLevelDifference)\n",
    "    \n",
    "           IJC_SAMPLE_1     <- tmp_IJC_SAMPLE_1[,1]\n",
    "           SJC_SAMPLE_1     <- tmp_SJC_SAMPLE_1[,1]\n",
    "           IncLevel1        <- tmp_IncLevel1[,1]\n",
    "    \n",
    "           for (i in 2:length(male_srrnames)) {\n",
    "            IJC_SAMPLE_1 <- paste(IJC_SAMPLE_1, tmp_IJC_SAMPLE_1[,i], sep=\",\")\n",
    "            SJC_SAMPLE_1 <- paste(SJC_SAMPLE_1, tmp_SJC_SAMPLE_1[,i], sep=\",\")\n",
    "            IncLevel1    <- paste(IncLevel1   , tmp_IncLevel1   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_1, SJC_SAMPLE_1 and IncLevel1 done!\\n\")\n",
    "    \n",
    "           IJC_SAMPLE_2     <- tmp_IJC_SAMPLE_2[,1]\n",
    "           SJC_SAMPLE_2     <- tmp_SJC_SAMPLE_2[,1]\n",
    "           IncLevel2        <- tmp_IncLevel2[,1]\n",
    "    \n",
    "           for (i in 2:length(female_srrnames)) {\n",
    "            IJC_SAMPLE_2 <- paste(IJC_SAMPLE_2, tmp_IJC_SAMPLE_2[,i], sep=\",\")\n",
    "            SJC_SAMPLE_2 <- paste(SJC_SAMPLE_2, tmp_SJC_SAMPLE_2[,i], sep=\",\")\n",
    "            IncLevel2    <- paste(IncLevel2   , tmp_IncLevel2   [,i], sep=\",\")\n",
    "           }\n",
    "           message(\"PASS IJC_SAMPLE_2, SJC_SAMPLE_2 and IncLevel2 done!\\n\")\n",
    " \n",
    "           # Ready to print out all the bits.\n",
    "           # We will paste the line\n",
    "           # first part comes from the fromGTF.splicetype.txt file\n",
    "\t   # Each splice_type is unique in their files\n",
    "\t   # Here we use fromGTF.RI.txt\n",
    "           file <- paste(paste(tissue_dir, snakecase::to_swap_case(as.character(splice_type)), sep=\"/\"),\n",
    "                             \"MATS.ReadsOnTargetAndJunctionCounts.txt\", sep= \".\")\n",
    "           write.table(\"ID\", file=file, append=FALSE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t   \n",
    "           write.table(\"GeneID\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "           write.table(\"geneSymbol\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"chr\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"strand\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "\t   write.table(\"riExonStart_0base\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"riExonEnd\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"upstreamES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"upstreamEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"downstreamES\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"downstreamEE\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"ID.1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SC_SAMPLE_2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"SkipFormLen\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"PValue\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"FDR\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel1\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevel2\", file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           write.table(\"IncLevelDifference\", file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "           message(\"PASS header written to the file!\\n\")\n",
    "\n",
    "           # first part is all the information from the fromGTF\n",
    "           # The structure we have is of the dimensions of the fromGTF, reduced by row to the junctions of signficance\n",
    "           for (m in 1:(dim(fromGTF)[1])) {\n",
    "             for (n in 1:(dim(fromGTF)[2])) {\n",
    "               write.table(fromGTF[m,n], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\t       \n",
    "    \t     }\n",
    "             write.table(splice_type, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(tissue_index, file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,1], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,2], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(fromGTF[m,3], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS fromGTF file written to the file!\\n\")\n",
    "             #now the rest\t\t\n",
    "             write.table(IncLevelDifference[m], file=\"../data/incLevelDifferences.tsv\", append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE) \n",
    "             write.table(junction[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(SJC_SAMPLE_2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(inclen    [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(skiplen   [m,1], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$AdjPVal, file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(splice_type_gene_as[m,]$B      , file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel1[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevel2[m], file=file, append=TRUE, eol=\"\\t\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             write.table(IncLevelDifference[m], file=file, append=TRUE, eol=\"\\n\",row.names=FALSE,col.names=FALSE,quote=FALSE)\n",
    "             message(\"PASS header written to the file!\\n\")\n",
    "             message(\"PASS row \", m)\n",
    "    \t     message(\"IJC, SJC, IncLevel for SAMPLE_1 and SAMPLE_2 written to the file!\\n\")\n",
    "\t    \n",
    "           } # for all junctions of interest\n",
    "\t   \n",
    " \t  } # and if there is > 0 junctions with inc\n",
    "#\t  rm(keep_gs)\n",
    "\t  \n",
    "         } # end !is.na(junction)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc4782f8",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#       } # and if there is > 0 genes of interest\n",
    "       rm(keep_tissue)\n",
    "       \n",
    "      } # if there are any tissues for splice type \"se\"\n",
    "    \n",
    "    } # end for splice_type \"ri\"\n",
    "\t\n",
    "  } # for all splice_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9af3cefa",
   "metadata": {
    "lines_to_next_cell": 2,
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "}  # for all tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c34574a3",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "### Appendix - Metadata\n",
    "\n",
    "For replicability and reproducibility purposes, we also print the following metadata:\n",
    "\n",
    "1. Checksums of **'artefacts'**, files generated during the analysis and stored in the folder directory **`data`**\n",
    "2. List of environment metadata, dependencies, versions of libraries using `utils::sessionInfo()` and [`devtools::session_info()`](https://devtools.r-lib.org/reference/session_info.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f9a83a3",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "### Appendix - 1. Checksums with the sha256 algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26f9f8c7",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "rm (notebookid)\n",
    "notebookid   = \"prepDataForMaser\"\n",
    "notebookid\n",
    "\n",
    "message(\"Generating sha256 checksums of the artefacts in the `..data/` directory .. \")\n",
    "system(paste0(\"cd ../data && find . -type f -exec sha256sum {} \\\\;  >  ../metadata/\", notebookid, \"_sha256sums.txt\"), intern = TRUE)\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\")\n",
    "\n",
    "data.table::fread(paste0(\"../metadata/\", notebookid, \"_sha256sums.txt\"), header = FALSE, col.names = c(\"sha256sum\", \"file\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f1caf44",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "### Appendix - 2. Libraries Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab154ba6",
   "metadata": {
    "lines_to_next_cell": 0,
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "dev_session_info   <- devtools::session_info()\n",
    "utils_session_info <- utils::sessionInfo()\n",
    "\n",
    "message(\"Saving `devtools::session_info()` objects in ../metadata/devtools_session_info.rds  ..\")\n",
    "saveRDS(dev_session_info, file = paste0(\"../metadata/\", notebookid, \"_devtools_session_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "message(\"Saving `utils::sessionInfo()` objects in ../metadata/utils_session_info.rds  ..\")\n",
    "saveRDS(utils_session_info, file = paste0(\"../metadata/\", notebookid ,\"_utils_info.rds\"))\n",
    "message(\"Done!\\n\")\n",
    "\n",
    "dev_session_info$platform\n",
    "dev_session_info$packages[dev_session_info$packages$attached==TRUE, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d25f3300",
   "metadata": {
    "lines_to_next_cell": 2,
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 3.597369,
   "end_time": "2021-05-21T04:15:40.341479",
   "environment_variables": {},
   "exception": true,
   "input_path": "prepDataForMASER.ipynb",
   "output_path": "prepDataForMASER.ipynb",
   "parameters": {},
   "start_time": "2021-05-21T04:15:36.744110",
   "version": "2.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}